---
title: "10x-data-integrate-log"
author: "Genevieve Housman"
date: "2/4/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Data Integrate

Compile 10X data using seurat.

Data being examined comes from 6 humans and 6 chimpanzees.
- GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
- GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
- GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
- HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
- HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
- HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
- HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)
- HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
- HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
- HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)
- HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
- HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
- HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)
- HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
- HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
- HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)
- HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
- HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
- HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)

Each dataset being evaluated comprises gene count data that were seperately calculated using both the hg38 human genome and the panTro6 chimpanzee genome, as well as the corresponding ortho-exon annotation in cellranger.Species-specific assignments were determined using human-chimpanzee genome combination with ortho-exon annotation.

```{r look at each sample separately - individual samples}

#BASHSCRIPT: separate-indv.sh

##!/bin/bash
##SBATCH --job-name=separate-indv
##SBATCH --output=separate-indv.out
##SBATCH --error=separate-indv.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore separate-indv.R separate-indv.out

#RSCRIPT: separate-indv.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read in files
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.rds"))

#Separate human and chimp data within each collection into separate objects
objects <- list()
idx <- 1
for (i in 1:length(batch$Sample_Name_at_Core))
{
  obj <- subset(data[[i]], species=='Human')
  objects[[idx]] <- CreateSeuratObject(obj@assays$RNA@counts, meta.data=obj@meta.data)
  idx <- idx+1
  obj <- subset(data[[i]], species=='Chimp')
  objects[[idx]] <- CreateSeuratObject(obj@assays$RNA@counts, meta.data=obj@meta.data)
  idx <- idx+1
}
rm(data,obj)

#Perform log normalization
#Reduce dimensionality
#Remove excess files
#Save datasets

#Perform log normalization on each collection
for (i in 1:length(objects)) {
  print(i)
  objects[[i]] <- NormalizeData(objects[[i]], verbose=FALSE)
  objects[[i]] <- FindVariableFeatures(objects[[i]], selection.method="vst", nfeatures=3000, verbose=FALSE)
  objects[[i]] <- ScaleData(objects[[i]], verbose=FALSE)
  objects[[i]] <- RunPCA(object=objects[[i]], npcs=100, verbose=FALSE)
  pva <- objects[[i]]@reductions$pca@stdev^2/objects[[i]]@reductions$pca@misc$total.variance
  ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
  ndim #
  objects[[i]] <- RunUMAP(objects[[i]], dims=1:ndim)
}

saveRDS(objects, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.separate-indv.rds"))

```

```{r look at each sample separately - human-chimp pairs}

#BASHSCRIPT: separate.sh

##!/bin/bash
##SBATCH --job-name=separate
##SBATCH --output=separate.out
##SBATCH --error=separate.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore separate.R separate.out

#RSCRIPT: separate.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read in files
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.rds"))

#Perform log normalization
#Reduce dimensionality
#Remove excess files
#Save datasets

#Perform log normalization on each collection
for (i in 1:length(data)) {
  print(i)
  data[[i]] <- NormalizeData(data[[i]], verbose=FALSE)
  data[[i]] <- FindVariableFeatures(data[[i]], selection.method="vst", nfeatures=3000, verbose=FALSE)
  data[[i]] <- ScaleData(data[[i]], verbose=FALSE)
  data[[i]] <- RunPCA(object=data[[i]], npcs=100, verbose=FALSE)
  pva <- data[[i]]@reductions$pca@stdev^2/data[[i]]@reductions$pca@misc$total.variance
  ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
  ndim #
  data[[i]] <- RunUMAP(data[[i]], dims=1:ndim)
}

saveRDS(data, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.separate.rds"))

```

```{r merge data without integration}

#BASHSCRIPT: merge.l.sh

##!/bin/bash
##SBATCH --job-name=merge.l
##SBATCH --output=merge.l.out
##SBATCH --error=merge.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore merge.l.R merge.l.out

#RSCRIPT: merge.l.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read in files
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.rds"))

#Merge datasets
#Perform log normalization
#Reduce dimensionality
#Remove excess files
#Save merged datasets

#combine ipscs from all samples
combo.ipsc    <- merge(data[[1]], y=c(data[[4]],data[[7]],data[[10]],data[[13]],data[[16]],data[[19]]),
                      add.cell.ids=c("H1C1.I","H1C1-r2.I","H2C2.I","H3C3.I","H4C4.I","H5C5.I","H6C6.I"))
combo.ipsc <- NormalizeData(combo.ipsc, verbose=FALSE)
combo.ipsc <- FindVariableFeatures(combo.ipsc, selection.method="vst", nfeatures=3000, verbose=FALSE)
combo.ipsc <- ScaleData(combo.ipsc, verbose=FALSE)
combo.ipsc <- RunPCA(object=combo.ipsc, npcs=100, verbose=FALSE)
pva <- combo.ipsc@reductions$pca@stdev^2/combo.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #29
combo.ipsc <- RunUMAP(combo.ipsc, dims=1:29)
saveRDS(combo.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.l.ipsc.rds"))
rm(combo.ipsc)

#combine mscs from all samples
combo.msc    <- merge(data[[2]], y=c(data[[5]],data[[8]],data[[11]],data[[14]],data[[17]],data[[20]]),
                      add.cell.ids=c("H1C1.M","H1C1-r2.M","H2C2.M","H3C3.M","H4C4.M","H5C5.M","H6C6.M"))
combo.msc <- NormalizeData(combo.msc, verbose=FALSE)
combo.msc <- FindVariableFeatures(combo.msc, selection.method="vst", nfeatures=3000, verbose=FALSE)
combo.msc <- ScaleData(combo.msc, verbose=FALSE)
combo.msc <- RunPCA(object=combo.msc, npcs=100, verbose=FALSE)
pva <- combo.msc@reductions$pca@stdev^2/combo.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #44
combo.msc <- RunUMAP(combo.msc, dims=1:44)
saveRDS(combo.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.l.msc.rds"))
rm(combo.msc)

#combine osteoblasts from all samples
combo.osteo    <- merge(data[[3]], y=c(data[[6]],data[[9]],data[[12]],data[[15]],data[[18]],data[[21]]),
                      add.cell.ids=c("H1C1.O","H1C1-r2.O","H2C2.O","H3C3.O","H4C4.O","H5C5.O","H6C6.O"))
combo.osteo <- NormalizeData(combo.osteo, verbose=FALSE)
combo.osteo <- FindVariableFeatures(combo.osteo, selection.method="vst", nfeatures=3000, verbose=FALSE)
combo.osteo <- ScaleData(combo.osteo, verbose=FALSE)
combo.osteo <- RunPCA(object=combo.osteo, npcs=100, verbose=FALSE)
pva <- combo.osteo@reductions$pca@stdev^2/combo.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #44
combo.osteo <- RunUMAP(combo.osteo, dims=1:44)
saveRDS(combo.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.l.osteo.rds"))
rm(combo.osteoblasts)

#combine all collections
combo.total  <- merge(data[[1]], y=c(data[[2]],data[[3]],data[[4]],data[[5]],data[[6]],data[[7]],data[[8]],data[[9]],data[[10]],data[[11]],
                                     data[[12]],data[[13]],data[[14]],data[[15]],data[[16]],data[[17]],data[[18]],data[[19]],data[[20]],data[[21]]),
                      add.cell.ids=c("H1C1.I","H1C1.M","H1C1.O","H1C1-r2.I","H1C1-r2.M","H1C1-r2.O","H2C2.I","H2C2.M","H2C2.O","H3C3.I","H3C3.M","H3C3.O",
                                     "H4C4.I","H4C4.M","H4C4.O","H5C5.I","H5C5.M","H5C5.O","H6C6.I","H6C6.M","H6C6.O"))
combo.total <- NormalizeData(combo.total, verbose=FALSE)
combo.total <- FindVariableFeatures(combo.total, selection.method="vst", nfeatures=3000, verbose=FALSE)
combo.total <- ScaleData(combo.total, verbose=FALSE)
combo.total <- RunPCA(object=combo.total, npcs=100, verbose=FALSE)
pva <- combo.total@reductions$pca@stdev^2/combo.total@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #49
combo.total <- RunUMAP(combo.total, dims=1:49)
saveRDS(combo.total, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.l.rds"))
rm(combo.total)

#NOTE: Within each cell type

```

```{r merge data without integration - 6K genes within species}

#BASHSCRIPT: merge-6k.l.sh

##!/bin/bash
##SBATCH --job-name=merge-6k.l
##SBATCH --output=merge-6k.l.out
##SBATCH --error=merge.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore merge-6k.l.R merge-6k.l.out

#RSCRIPT: merge-6k.l.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read in files
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.rds"))

#Merge datasets
#Perform log normalization
#Reduce dimensionality
#Remove excess files
#Save merged datasets

#combine ipscs from all samples
combo.ipsc    <- merge(data[[1]], y=c(data[[4]],data[[7]],data[[10]],data[[13]],data[[16]],data[[19]]),
                      add.cell.ids=c("H1C1.I","H1C1-r2.I","H2C2.I","H3C3.I","H4C4.I","H5C5.I","H6C6.I"))
#log normalize data
combo.ipsc <- NormalizeData(combo.ipsc, verbose=FALSE)
#find 6000 most variable genes within each species
humans <- FindVariableFeatures(subset(combo.ipsc, species=="Human"), selection.method="vst", nfeatures=6000)
chimps <- FindVariableFeatures(subset(combo.ipsc, species=="Chimp"), selection.method="vst", nfeatures=6000)
human.features <- VariableFeatures(humans)
chimp.features <- VariableFeatures(chimps)
features <- intersect(human.features, chimp.features)
length(features) #3210 variable genes intersect
head(features)
combo.ipsc <- FindVariableFeatures(combo.ipsc, selection.method="vst", nfeatures=length(features))
VariableFeatures(combo.ipsc) <- features
rm(humans,chimps,human.features,chimp.features,features)
#scale data
combo.ipsc <- ScaleData(combo.ipsc, verbose=FALSE)
#pca
combo.ipsc <- RunPCA(object=combo.ipsc, npcs=100, verbose=FALSE)
pva <- combo.ipsc@reductions$pca@stdev^2/combo.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #26
#umap
combo.ipsc <- RunUMAP(combo.ipsc, dims=1:26)
#save data
saveRDS(combo.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge-6k.l.ipsc.rds"))
rm(combo.ipsc)

#combine mscs from all samples
combo.msc    <- merge(data[[2]], y=c(data[[5]],data[[8]],data[[11]],data[[14]],data[[17]],data[[20]]),
                      add.cell.ids=c("H1C1.M","H1C1-r2.M","H2C2.M","H3C3.M","H4C4.M","H5C5.M","H6C6.M"))
#log normalize data
combo.msc <- NormalizeData(combo.msc, verbose=FALSE)
#find 6000 most variable genes within each species
humans <- FindVariableFeatures(subset(combo.msc, species=="Human"), selection.method="vst", nfeatures=6000)
chimps <- FindVariableFeatures(subset(combo.msc, species=="Chimp"), selection.method="vst", nfeatures=6000)
human.features <- VariableFeatures(humans)
chimp.features <- VariableFeatures(chimps)
features <- intersect(human.features, chimp.features)
length(features) #3210 variable genes intersect
head(features)
combo.msc <- FindVariableFeatures(combo.msc, selection.method="vst", nfeatures=length(features))
VariableFeatures(combo.msc) <- features
rm(humans,chimps,human.features,chimp.features,features)
#scale data
combo.msc <- ScaleData(combo.msc, verbose=FALSE)
#pca
combo.msc <- RunPCA(object=combo.msc, npcs=100, verbose=FALSE)
pva <- combo.msc@reductions$pca@stdev^2/combo.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #41
#umap
combo.msc <- RunUMAP(combo.msc, dims=1:41)
#save data
saveRDS(combo.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge-6k.l.msc.rds"))
rm(combo.msc)

#combine osteoblasts from all samples
combo.osteo    <- merge(data[[3]], y=c(data[[6]],data[[9]],data[[12]],data[[15]],data[[18]],data[[21]]),
                      add.cell.ids=c("H1C1.O","H1C1-r2.O","H2C2.O","H3C3.O","H4C4.O","H5C5.O","H6C6.O"))
#log normalize data
combo.osteo <- NormalizeData(combo.osteo, verbose=FALSE)
#find 6000 most variable genes within each species
humans <- FindVariableFeatures(subset(combo.osteo, species=="Human"), selection.method="vst", nfeatures=6000)
chimps <- FindVariableFeatures(subset(combo.osteo, species=="Chimp"), selection.method="vst", nfeatures=6000)
human.features <- VariableFeatures(humans)
chimp.features <- VariableFeatures(chimps)
features <- intersect(human.features, chimp.features)
length(features) #3210 variable genes intersect
head(features)
combo.osteo <- FindVariableFeatures(combo.osteo, selection.method="vst", nfeatures=length(features))
VariableFeatures(combo.osteo) <- features
rm(humans,chimps,human.features,chimp.features,features)
#scale data
combo.osteo <- ScaleData(combo.osteo, verbose=FALSE)
#pca
combo.osteo <- RunPCA(object=combo.osteo, npcs=100, verbose=FALSE)
pva <- combo.osteo@reductions$pca@stdev^2/combo.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #41
#umap
combo.osteo <- RunUMAP(combo.osteo, dims=1:41)
#save data
saveRDS(combo.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge-6k.l.osteo.rds"))
rm(combo.osteoblasts)

#combine all collections
combo.total  <- merge(data[[1]], y=c(data[[2]],data[[3]],data[[4]],data[[5]],data[[6]],data[[7]],data[[8]],data[[9]],data[[10]],data[[11]],
                                     data[[12]],data[[13]],data[[14]],data[[15]],data[[16]],data[[17]],data[[18]],data[[19]],data[[20]],data[[21]]),
                      add.cell.ids=c("H1C1.I","H1C1.M","H1C1.O","H1C1-r2.I","H1C1-r2.M","H1C1-r2.O","H2C2.I","H2C2.M","H2C2.O","H3C3.I","H3C3.M","H3C3.O",
                                     "H4C4.I","H4C4.M","H4C4.O","H5C5.I","H5C5.M","H5C5.O","H6C6.I","H6C6.M","H6C6.O"))
#log normalize data
combo.total <- NormalizeData(combo.total, verbose=FALSE)
#find 6000 most variable genes within each species
humans <- FindVariableFeatures(subset(combo.total, species=="Human"), selection.method="vst", nfeatures=6000)
chimps <- FindVariableFeatures(subset(combo.total, species=="Chimp"), selection.method="vst", nfeatures=6000)
human.features <- VariableFeatures(humans)
chimp.features <- VariableFeatures(chimps)
features <- intersect(human.features, chimp.features)
length(features) #4168 variable genes intersect
head(features)
combo.total <- FindVariableFeatures(combo.total, selection.method="vst", nfeatures=length(features))
VariableFeatures(combo.total) <- features
rm(humans,chimps,human.features,chimp.features,features)
#scale data
combo.total <- ScaleData(combo.total, verbose=FALSE)
#pca
combo.total <- RunPCA(object=combo.total, npcs=100, verbose=FALSE)
pva <- combo.total@reductions$pca@stdev^2/combo.total@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
ndim #44
#umap
combo.total <- RunUMAP(combo.total, dims=1:44)
#save data
saveRDS(combo.total, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge-6k.l.rds"))
rm(combo.total)

#DimPlot(object=combo.total, reduction="pca", pt.size=.1, group.by="species")
#DimPlot(object=combo.total, reduction="pca", pt.size=.1, group.by="Individual")
#DimPlot(object=combo.total, reduction="pca", pt.size=.1, group.by="Cell.Type")
#DimPlot(object=combo.total, reduction="pca", pt.size=.1, group.by="species", split.by="Cell.Type")
#DimPlot(object=combo.total, reduction="pca", pt.size=.1, group.by="Individual", split.by="Cell.Type")

```

```{r integrate scRNA data - separate species and cells types}

#BASHSCRIPT: rpca.l_spps-cell.sh
##!/bin/bash
##SBATCH --job-name=rpca.l_spps-cell
##SBATCH --output=rpca.l_spps-cell.out
##SBATCH --error=rpca.l_spps-cell.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_spps-cell.R rpca.l_spps-cell.out

#RSCRIPT: rpca.l_spps-cell.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.rds"))

#Define list subset
#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
#Integrate datasets
#Reduce dimensionality
#Save data

#combine ipscs from all samples
obj.ipsc  <- list(objects[[1]],objects[[4]])
obj.ipsc.features <- SelectIntegrationFeatures(object.list=obj.ipsc, nfeatures=1000)
reference_dataset <- 1 #make humans the reference
obj.ipsc <- lapply(X=obj.ipsc, FUN=function(x) {
    x <- ScaleData(x, features=obj.ipsc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.ipsc.features, verbose=FALSE)
})
obj.ipsc.anchors <- FindIntegrationAnchors(object.list=obj.ipsc, normalization.method="LogNormalize", anchor.features=obj.ipsc.features,
                                           reference=reference_dataset, reduction="rpca")
int.ipsc <- IntegrateData(anchorset=obj.ipsc.anchors, normalization.method="LogNormalize")
int.ipsc <- ScaleData(int.ipsc, verbose=FALSE)
int.ipsc <- RunPCA(object=int.ipsc, npcs=100, verbose=FALSE)
pva <- int.ipsc@reductions$pca@stdev^2/int.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.ipsc <- RunUMAP(int.ipsc, dims=1:100)
saveRDS(int.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.ipsc.rds"))
rm(obj.ipsc,int.ipsc)

#combine mscs from all samples
obj.msc   <- list(objects[[2]],objects[[5]])
obj.msc.features <- SelectIntegrationFeatures(object.list=obj.msc, nfeatures=1000)
reference_dataset <- 1 #make humans the reference
obj.msc <- lapply(X=obj.msc, FUN=function(x) {
    x <- ScaleData(x, features=obj.msc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.msc.features, verbose=FALSE)
})
obj.msc.anchors <- FindIntegrationAnchors(object.list=obj.msc, normalization.method="LogNormalize", anchor.features=obj.msc.features,
                                           reference=reference_dataset, reduction="rpca")
int.msc <- IntegrateData(anchorset=obj.msc.anchors, normalization.method="LogNormalize")
int.msc <- ScaleData(int.msc, verbose=FALSE)
int.msc <- RunPCA(object=int.msc, npcs=100, verbose=FALSE)
pva <- int.msc@reductions$pca@stdev^2/int.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.msc <- RunUMAP(int.msc, dims=1:100)
saveRDS(int.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.msc.rds"))
rm(obj.msc,int.msc)

#combine osteoblasts from all samples
obj.osteo <- list(objects[[3]],objects[[6]])
obj.osteo.features <- SelectIntegrationFeatures(object.list=obj.osteo, nfeatures=1000)
reference_dataset <- 1 #make humans the reference
obj.osteo <- lapply(X=obj.osteo, FUN=function(x) {
    x <- ScaleData(x, features=obj.osteo.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.osteo.features, verbose=FALSE)
})
obj.osteo.anchors <- FindIntegrationAnchors(object.list=obj.osteo, normalization.method="LogNormalize", anchor.features=obj.osteo.features,
                                           reference=reference_dataset, reduction="rpca")
int.osteo <- IntegrateData(anchorset=obj.osteo.anchors, normalization.method="LogNormalize")
int.osteo <- ScaleData(int.osteo, verbose=FALSE)
int.osteo <- RunPCA(object=int.osteo, npcs=100, verbose=FALSE)
pva <- int.osteo@reductions$pca@stdev^2/int.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.osteo <- RunUMAP(int.osteo, dims=1:100)
saveRDS(int.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.osteo.rds"))
rm(obj.osteo,int.osteo)

```

```{r integrate scRNA data - separate human-chimp pairs and cells types}

#BASHSCRIPT: rpca.l_pair-cell.sh
##!/bin/bash
##SBATCH --job-name=rpca.l_pair-cell
##SBATCH --output=rpca.l_pair-cell.out
##SBATCH --error=rpca.l_pair-cell.err
##SBATCH --time=03:00:00
##SBATCH --partition=bigmem2
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=12800
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_pair-cell.R rpca.l_pair-cell.out

#RSCRIPT: rpca.l_pair-cell.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.rds"))

#Define list subset
#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
#Integrate datasets
#Reduce dimensionality
#Save data

#combine ipscs from all samples
obj.ipsc  <- list(objects[[1]],objects[[4]],objects[[7]],objects[[10]],objects[[13]],objects[[16]],objects[[19]])
obj.ipsc.features <- SelectIntegrationFeatures(object.list=obj.ipsc, nfeatures=1000)
reference_dataset <- 2 #H1C1-r2 differentiated the best
obj.ipsc <- lapply(X=obj.ipsc, FUN=function(x) {
    x <- ScaleData(x, features=obj.ipsc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.ipsc.features, verbose=FALSE)
})
obj.ipsc.anchors <- FindIntegrationAnchors(object.list=obj.ipsc, normalization.method="LogNormalize", anchor.features=obj.ipsc.features,
                                           reference=reference_dataset, reduction="rpca")
int.ipsc <- IntegrateData(anchorset=obj.ipsc.anchors, normalization.method="LogNormalize")
int.ipsc <- ScaleData(int.ipsc, verbose=FALSE)
int.ipsc <- RunPCA(object=int.ipsc, npcs=100, verbose=FALSE)
pva <- int.ipsc@reductions$pca@stdev^2/int.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.ipsc <- RunUMAP(int.ipsc, dims=1:100)
saveRDS(int.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.ipsc.rds"))
rm(obj.ipsc,int.ipsc)

#combine mscs from all samples
obj.msc   <- list(objects[[2]],objects[[5]],objects[[8]],objects[[11]],objects[[14]],objects[[17]],objects[[20]])
obj.msc.features <- SelectIntegrationFeatures(object.list=obj.msc, nfeatures=1000)
reference_dataset <- 2 #H1C1-r2 differentiated the best
obj.msc <- lapply(X=obj.msc, FUN=function(x) {
    x <- ScaleData(x, features=obj.msc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.msc.features, verbose=FALSE)
})
obj.msc.anchors <- FindIntegrationAnchors(object.list=obj.msc, normalization.method="LogNormalize", anchor.features=obj.msc.features,
                                           reference=reference_dataset, reduction="rpca")
int.msc <- IntegrateData(anchorset=obj.msc.anchors, normalization.method="LogNormalize")
int.msc <- ScaleData(int.msc, verbose=FALSE)
int.msc <- RunPCA(object=int.msc, npcs=100, verbose=FALSE)
pva <- int.msc@reductions$pca@stdev^2/int.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.msc <- RunUMAP(int.msc, dims=1:100)
saveRDS(int.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.msc.rds"))
rm(obj.msc,int.msc)

#combine osteoblasts from all samples
obj.osteo   <- list(objects[[3]],objects[[6]],objects[[9]],objects[[12]],objects[[15]],objects[[18]],objects[[21]])
obj.osteo.features <- SelectIntegrationFeatures(object.list=obj.osteo, nfeatures=1000)
reference_dataset <- 2 #H1C1-r2 differentiated the best
obj.osteo <- lapply(X=obj.osteo, FUN=function(x) {
    x <- ScaleData(x, features=obj.osteo.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.osteo.features, verbose=FALSE)
})
obj.osteo.anchors <- FindIntegrationAnchors(object.list=obj.osteo, normalization.method="LogNormalize", anchor.features=obj.osteo.features,
                                           reference=reference_dataset, reduction="rpca")
int.osteo <- IntegrateData(anchorset=obj.osteo.anchors, normalization.method="LogNormalize")
int.osteo <- ScaleData(int.osteo, verbose=FALSE)
int.osteo <- RunPCA(object=int.osteo, npcs=100, verbose=FALSE)
pva <- int.osteo@reductions$pca@stdev^2/int.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.osteo <- RunUMAP(int.osteo, dims=1:100)
saveRDS(int.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.osteo.rds"))
rm(obj.osteo,int.osteo)

```

```{r integrate scRNA data - separate individuals and cells types}

#BASHSCRIPT: rpca.l_indv-cell.sh (or rpca.filter.l_indv-cell.sh / rpca.l.filterMT.l_indv-cell.sh)

##!/bin/bash
##SBATCH --job-name=rpca.l_indv-cell
##SBATCH --output=rpca.l_indv-cell.out
##SBATCH --error=rpca.l_indv-cell.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_indv-cell.R rpca.l_indv-cell.out

#RSCRIPT: rpca.l_indv-cell.R (or rpca.filter.l_indv-cell.R / rpca.filterMT.l_indv-cell.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.rds"))

#Define list subset
#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
#Integrate datasets
#Reduce dimensionality
#Save data

#combine ipscs from all samples
obj.ipsc  <- list(objects[[1]],objects[[2]],objects[[7]],objects[[8]],objects[[13]],objects[[14]],objects[[19]],objects[[20]],
                  objects[[25]],objects[[26]],objects[[31]],objects[[32]],objects[[37]],objects[[38]])
obj.ipsc.features <- SelectIntegrationFeatures(object.list=obj.ipsc, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.ipsc <- lapply(X=obj.ipsc, FUN=function(x) {
    x <- ScaleData(x, features=obj.ipsc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.ipsc.features, verbose=FALSE)
})
obj.ipsc.anchors <- FindIntegrationAnchors(object.list=obj.ipsc, normalization.method="LogNormalize", anchor.features=obj.ipsc.features,
                                           reference=reference_dataset, reduction="rpca")
int.ipsc <- IntegrateData(anchorset=obj.ipsc.anchors, normalization.method="LogNormalize")
int.ipsc <- ScaleData(int.ipsc, verbose=FALSE)
int.ipsc <- RunPCA(object=int.ipsc, npcs=100, verbose=FALSE)
pva <- int.ipsc@reductions$pca@stdev^2/int.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.ipsc <- RunUMAP(int.ipsc, dims=1:100)
saveRDS(int.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.ipsc.rds"))
rm(obj.ipsc,int.ipsc)

#combine mscs from all samples
obj.msc   <- list(objects[[3]],objects[[4]],objects[[9]],objects[[10]],objects[[15]],objects[[16]],objects[[21]],objects[[22]],
                  objects[[27]],objects[[28]],objects[[33]],objects[[34]],objects[[39]],objects[[40]])
obj.msc.features <- SelectIntegrationFeatures(object.list=obj.msc, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.msc <- lapply(X=obj.msc, FUN=function(x) {
    x <- ScaleData(x, features=obj.msc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.msc.features, verbose=FALSE)
})
obj.msc.anchors <- FindIntegrationAnchors(object.list=obj.msc, normalization.method="LogNormalize", anchor.features=obj.msc.features,
                                           reference=reference_dataset, reduction="rpca")
int.msc <- IntegrateData(anchorset=obj.msc.anchors, normalization.method="LogNormalize")
int.msc <- ScaleData(int.msc, verbose=FALSE)
int.msc <- RunPCA(object=int.msc, npcs=100, verbose=FALSE)
pva <- int.msc@reductions$pca@stdev^2/int.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.msc <- RunUMAP(int.msc, dims=1:100)
saveRDS(int.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.msc.rds"))
rm(obj.msc,int.msc)

#combine osteoblasts from all samples
obj.osteo <- list(objects[[5]],objects[[6]],objects[[11]],objects[[12]],objects[[17]],objects[[18]],objects[[23]],objects[[24]],
                  objects[[29]],objects[[30]],objects[[35]],objects[[36]],objects[[41]],objects[[42]])
obj.osteo.features <- SelectIntegrationFeatures(object.list=obj.osteo, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.osteo <- lapply(X=obj.osteo, FUN=function(x) {
    x <- ScaleData(x, features=obj.osteo.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.osteo.features, verbose=FALSE)
})
obj.osteo.anchors <- FindIntegrationAnchors(object.list=obj.osteo, normalization.method="LogNormalize", anchor.features=obj.osteo.features,
                                           reference=reference_dataset, reduction="rpca")
int.osteo <- IntegrateData(anchorset=obj.osteo.anchors, normalization.method="LogNormalize")
int.osteo <- ScaleData(int.osteo, verbose=FALSE)
int.osteo <- RunPCA(object=int.osteo, npcs=100, verbose=FALSE)
pva <- int.osteo@reductions$pca@stdev^2/int.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.osteo <- RunUMAP(int.osteo, dims=1:100)
saveRDS(int.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.osteo.rds"))
rm(obj.osteo,int.osteo)

```

```{r integrate scRNA data - separate species but keep all cell types together}

#BASHSCRIPT: rpca.l_spps.sh

##!/bin/bash
##SBATCH --job-name=rpca.l_spps
##SBATCH --output=rpca.l_spps.out
##SBATCH --error=rpca.l_spps.err
##SBATCH --time=03:00:00
##SBATCH --partition=bigmem2
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=12800
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_spps.R rpca.l_spps.out

#RSCRIPT: rpca.l_spps.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Define head directory containing scRNA data
dir_scrna <- paste0(dir,'scRNA/cellranger-data-full')

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps.log.rds"))

#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
objects.features <- SelectIntegrationFeatures(object.list=objects, nfeatures=1000)
reference_dataset <- 1 #make humans the reference
objects <- lapply(X=objects, FUN=function(x) {
    x <- ScaleData(x, features=objects.features, verbose=FALSE)
    x <- RunPCA(x, features=objects.features, verbose=FALSE)
})
objects.anchors <- FindIntegrationAnchors(object.list=objects, normalization.method="LogNormalize", anchor.features=objects.features,
                                          reference=reference_dataset, reduction="rpca")

#Integrate datasets
integrate <- IntegrateData(anchorset=objects.anchors, normalization.method="LogNormalize")
integrate <- ScaleData(integrate, verbose=FALSE)

#Reduce dimensionality
integrate <- RunPCA(object=integrate, npcs=100, verbose=FALSE)
pva <- integrate@reductions$pca@stdev^2/integrate@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
integrate <- RunUMAP(integrate, dims=1:100)

#Save data
saveRDS(integrate, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps.log.int.rds"))
rm(objects,integrate)

```

```{r integrate scRNA data - separate human-chimp pairs but keep all cell types together}

#BASHSCRIPT: rpca.l_pair.sh

##!/bin/bash
##SBATCH --job-name=rpca.l_pair
##SBATCH --output=rpca.l_pair.out
##SBATCH --error=rpca.l_pair.err
##SBATCH --time=03:00:00
##SBATCH --partition=broadwl
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=6400
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_pair.R rpca.l_pair.out

#RSCRIPT: rpca.l_pair.R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair.log.rds"))

#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
objects.features <- SelectIntegrationFeatures(object.list=objects, nfeatures=1000)
reference_dataset <- 2 #make H1C1r2 the references
objects <- lapply(X=objects, FUN=function(x) {
    x <- ScaleData(x, features=objects.features, verbose=FALSE)
    x <- RunPCA(x, features=objects.features, verbose=FALSE)
})
objects.anchors <- FindIntegrationAnchors(object.list=objects, normalization.method="LogNormalize", anchor.features=objects.features,
                                          reference=reference_dataset, reduction="rpca")

#Integrate datasets
integrate <- IntegrateData(anchorset=objects.anchors, normalization.method="LogNormalize")
integrate <- ScaleData(integrate, verbose=FALSE)

#Reduce dimensionality
integrate <- RunPCA(object=integrate, npcs=100, verbose=FALSE)
pva <- integrate@reductions$pca@stdev^2/integrate@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
integrate <- RunUMAP(integrate, dims=1:91)

#Save data
saveRDS(integrate, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair.log.int.rds"))
rm(objects,integrate)

```

```{r integrate scRNA data - separate individuals but keep all cell types together}

#BASHSCRIPT: rpca.l_indv.sh (or rpca.filter.l_indv.sh / rpca.filterMT.l_indv.sh)

##!/bin/bash
##SBATCH --job-name=rpca.l_indv
##SBATCH --output=rpca.l_indv.out
##SBATCH --error=rpca.l_indv.err
##SBATCH --time=03:00:00
##SBATCH --partition=bigmem2
##SBATCH --nodes=1
##SBATCH --ntasks-per-node=8
##SBATCH --mem-per-cpu=12800
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R CMD BATCH --no-save --no-restore rpca.l_indv.R rpca.l_indv.out

#RSCRIPT: rpca.l_indv.R (or rpca.filter.l_indv.R / rpca.filterMT.l_indv.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.rds"))

#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
objects.features <- SelectIntegrationFeatures(object.list=objects, nfeatures=1000)
reference_dataset <- c(2,9) #make H1-r2 and C1-r2 the references
objects <- lapply(X=objects, FUN=function(x) {
    x <- ScaleData(x, features=objects.features, verbose=FALSE)
    x <- RunPCA(x, features=objects.features, verbose=FALSE)
})
objects.anchors <- FindIntegrationAnchors(object.list=objects, normalization.method="LogNormalize", anchor.features=objects.features,
                                          reference=reference_dataset, reduction="rpca")

#Integrate datasets
integrate <- IntegrateData(anchorset=objects.anchors, normalization.method="LogNormalize")
integrate <- ScaleData(integrate, verbose=FALSE)

#Reduce dimensionality
integrate <- RunPCA(object=integrate, npcs=100, verbose=FALSE)
pva <- integrate@reductions$pca@stdev^2/integrate@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
integrate <- RunUMAP(integrate, dims=1:100)

#Save data
saveRDS(integrate, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.int.rds"))
rm(objects,integrate)

```

```{r examine batch effects - merged data vs. integrated data}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read and Visualize data (batch of interest: species, Pair, Individual, Cell.Type, Replicate, MSCmedia)

#combine ipscs from all samples
data.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.ipsc.rds"))             #merge
spps.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.ipsc.rds")) #integrate across species
pair.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.ipsc.rds")) #integrate across human-chimp pairs
indv.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.ipsc.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.ipsc,group.by="species"),
                        DimPlot(data.ipsc,group.by="Pair"),
                        DimPlot(data.ipsc,group.by="Individual"),
                        DimPlot(data.ipsc,group.by="Cell.Type"),
                        DimPlot(data.ipsc,group.by="Replicate"),
                        DimPlot(data.ipsc,group.by="MSCmedia"),

                        DimPlot(spps.ipsc,group.by="species"),
                        DimPlot(spps.ipsc,group.by="Pair"),
                        DimPlot(spps.ipsc,group.by="Individual"),
                        DimPlot(spps.ipsc,group.by="Cell.Type"),
                        DimPlot(spps.ipsc,group.by="Replicate"),
                        DimPlot(spps.ipsc,group.by="MSCmedia"),
                        
                        DimPlot(pair.ipsc,group.by="species"),
                        DimPlot(pair.ipsc,group.by="Pair"),
                        DimPlot(pair.ipsc,group.by="Individual"),
                        DimPlot(pair.ipsc,group.by="Cell.Type"),
                        DimPlot(pair.ipsc,group.by="Replicate"),
                        DimPlot(pair.ipsc,group.by="MSCmedia"),
                        
                        DimPlot(indv.ipsc,group.by="species"),
                        DimPlot(indv.ipsc,group.by="Pair"),
                        DimPlot(indv.ipsc,group.by="Individual"),
                        DimPlot(indv.ipsc,group.by="Cell.Type"),
                        DimPlot(indv.ipsc,group.by="Replicate"),
                        DimPlot(indv.ipsc,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.ipsc,group.by="species",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Collection",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Samples",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.ipsc,group.by="species",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.ipsc,group.by="species",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.ipsc,group.by="species",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.ipsc,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.ipsc,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.ipsc, spps.ipsc, pair.ipsc, indv.ipsc)

#data.ipsc:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.ipsc:   species integrated / batches separated  (PC/UMAP separates species)
#pair.ipsc:   species separated / batches integrated  (PC/UMAP separates species)
#indv.ipsc:   species and batches integrated
#NOTES:       UMI and gene counts are higher in humans than chimps


#combine mscs from all samples
data.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.msc.rds"))             #merge
spps.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.msc.rds")) #integrate across species
pair.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.msc.rds")) #integrate across human-chimp pairs
indv.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.msc.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.msc,group.by="species"),
                        DimPlot(data.msc,group.by="Pair"),
                        DimPlot(data.msc,group.by="Individual"),
                        DimPlot(data.msc,group.by="Cell.Type"),
                        DimPlot(data.msc,group.by="Replicate"),
                        DimPlot(data.msc,group.by="MSCmedia"),

                        DimPlot(spps.msc,group.by="species"),
                        DimPlot(spps.msc,group.by="Pair"),
                        DimPlot(spps.msc,group.by="Individual"),
                        DimPlot(spps.msc,group.by="Cell.Type"),
                        DimPlot(spps.msc,group.by="Replicate"),
                        DimPlot(spps.msc,group.by="MSCmedia"),
                        
                        DimPlot(pair.msc,group.by="species"),
                        DimPlot(pair.msc,group.by="Pair"),
                        DimPlot(pair.msc,group.by="Individual"),
                        DimPlot(pair.msc,group.by="Cell.Type"),
                        DimPlot(pair.msc,group.by="Replicate"),
                        DimPlot(pair.msc,group.by="MSCmedia"),
                        
                        DimPlot(indv.msc,group.by="species"),
                        DimPlot(indv.msc,group.by="Pair"),
                        DimPlot(indv.msc,group.by="Individual"),
                        DimPlot(indv.msc,group.by="Cell.Type"),
                        DimPlot(indv.msc,group.by="Replicate"),
                        DimPlot(indv.msc,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.msc,group.by="species",reduction='pca'),
                        DimPlot(data.msc,group.by="Collection",reduction='pca'),
                        DimPlot(data.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.msc,group.by="Samples",reduction='pca'),
                        DimPlot(data.msc,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.msc,group.by="species",reduction='pca'),
                        DimPlot(spps.msc,group.by="Pair",reduction='pca'),
                        DimPlot(spps.msc,group.by="Individual",reduction='pca'),
                        DimPlot(spps.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.msc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.msc,group.by="species",reduction='pca'),
                        DimPlot(pair.msc,group.by="Pair",reduction='pca'),
                        DimPlot(pair.msc,group.by="Individual",reduction='pca'),
                        DimPlot(pair.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.msc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.msc,group.by="species",reduction='pca'),
                        DimPlot(indv.msc,group.by="Pair",reduction='pca'),
                        DimPlot(indv.msc,group.by="Individual",reduction='pca'),
                        DimPlot(indv.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.msc,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.msc,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.msc,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.msc, spps.msc, pair.msc, indv.msc)

#data.msc:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.msc:   species integrated / batches separated  (PC/UMAP separates species)
#pair.msc:   species separated / batches integrated  (PC/UMAP separates species)
#indv.msc:   species and batches integrated
#NOTES:      UMI and gene counts are higher in humans than chimps


#combine osteoblasts from all samples
data.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.osteo.rds"))             #merge
spps.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.osteo.rds")) #integrate across species
pair.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.osteo.rds")) #integrate across human-chimp pairs
indv.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.osteo.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.osteo,group.by="species"),
                        DimPlot(data.osteo,group.by="Pair"),
                        DimPlot(data.osteo,group.by="Individual"),
                        DimPlot(data.osteo,group.by="Cell.Type"),
                        DimPlot(data.osteo,group.by="Replicate"),
                        DimPlot(data.osteo,group.by="MSCmedia"),

                        DimPlot(spps.osteo,group.by="species"),
                        DimPlot(spps.osteo,group.by="Pair"),
                        DimPlot(spps.osteo,group.by="Individual"),
                        DimPlot(spps.osteo,group.by="Cell.Type"),
                        DimPlot(spps.osteo,group.by="Replicate"),
                        DimPlot(spps.osteo,group.by="MSCmedia"),
                        
                        DimPlot(pair.osteo,group.by="species"),
                        DimPlot(pair.osteo,group.by="Pair"),
                        DimPlot(pair.osteo,group.by="Individual"),
                        DimPlot(pair.osteo,group.by="Cell.Type"),
                        DimPlot(pair.osteo,group.by="Replicate"),
                        DimPlot(pair.osteo,group.by="MSCmedia"),
                        
                        DimPlot(indv.osteo,group.by="species"),
                        DimPlot(indv.osteo,group.by="Pair"),
                        DimPlot(indv.osteo,group.by="Individual"),
                        DimPlot(indv.osteo,group.by="Cell.Type"),
                        DimPlot(indv.osteo,group.by="Replicate"),
                        DimPlot(indv.osteo,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.osteo,group.by="species",reduction='pca'),
                        DimPlot(data.osteo,group.by="Collection",reduction='pca'),
                        DimPlot(data.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.osteo,group.by="Samples",reduction='pca'),
                        DimPlot(data.osteo,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.osteo,group.by="species",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.osteo,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.osteo,group.by="species",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.osteo,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.osteo,group.by="species",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.osteo,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.osteo,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.osteo,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.osteo, spps.osteo, pair.osteo, indv.osteo)

#data.osteo:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.osteo:   species integrated / batches separated  (PC/UMAP separates species)
#pair.osteo:   species separated / batches integrated  (PC/UMAP separates species)
#indv.osteo:   species and batches integrated
#NOTES:        UMI and gene counts are higher in humans than chimps


#combine all cells from all samples
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.rds"))        #merge
spps <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps.log.int.rds")) #integrate across species
pair <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair.log.int.rds")) #integrate across human-chimp pairs
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.int.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data,group.by="species"),
                        DimPlot(data,group.by="Pair"),
                        DimPlot(data,group.by="Individual"),
                        DimPlot(data,group.by="Cell.Type"),
                        DimPlot(data,group.by="Replicate"),
                        DimPlot(data,group.by="MSCmedia"),

                        DimPlot(spps,group.by="species"),
                        DimPlot(spps,group.by="Pair"),
                        DimPlot(spps,group.by="Individual"),
                        DimPlot(spps,group.by="Cell.Type"),
                        DimPlot(spps,group.by="Replicate"),
                        DimPlot(spps,group.by="MSCmedia"),
                        
                        DimPlot(pair,group.by="species"),
                        DimPlot(pair,group.by="Pair"),
                        DimPlot(pair,group.by="Individual"),
                        DimPlot(pair,group.by="Cell.Type"),
                        DimPlot(pair,group.by="Replicate"),
                        DimPlot(pair,group.by="MSCmedia"),
                        
                        DimPlot(indv,group.by="species"),
                        DimPlot(indv,group.by="Pair"),
                        DimPlot(indv,group.by="Individual"),
                        DimPlot(indv,group.by="Cell.Type"),
                        DimPlot(indv,group.by="Replicate"),
                        DimPlot(indv,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data,group.by="species",reduction='pca'),
                        DimPlot(data,group.by="Collection",reduction='pca'),
                        DimPlot(data,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data,group.by="Samples",reduction='pca'),
                        DimPlot(data,group.by="Replicate",reduction='pca'),

                        DimPlot(spps,group.by="species",reduction='pca'),
                        DimPlot(spps,group.by="Pair",reduction='pca'),
                        DimPlot(spps,group.by="Individual",reduction='pca'),
                        DimPlot(spps,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps,group.by="Replicate",reduction='pca'),
                        DimPlot(spps,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair,group.by="species",reduction='pca'),
                        DimPlot(pair,group.by="Pair",reduction='pca'),
                        DimPlot(pair,group.by="Individual",reduction='pca'),
                        DimPlot(pair,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair,group.by="Replicate",reduction='pca'),
                        DimPlot(pair,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv,group.by="species",reduction='pca'),
                        DimPlot(indv,group.by="Pair",reduction='pca'),
                        DimPlot(indv,group.by="Individual",reduction='pca'),
                        DimPlot(indv,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv,group.by="Replicate",reduction='pca'),
                        DimPlot(indv,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data,'nCount_RNA',group.by='species'),
                        VlnPlot(spps,'nCount_RNA',group.by='species'),
                        VlnPlot(pair,'nCount_RNA',group.by='species'),
                        VlnPlot(indv,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv,'nFeature_RNA',group.by='species'),
                        VlnPlot(int,'nFeature_RNA',group.by='species')), ncol=4)

rm(data, spps, pair, indv)

#data:   species + cell type + batches separated             (PC/UMAP separates species, PC/UMAP separates cell types, PC/UMAP separates batches)
#spps:   cell type + batches separated / species integrated  (PC/UMAP separates cell types, PC/UMAP separates batches)
#pair:   species + cell type separated / batches integrated  (PC/UMAP separates species, PC/UMAP separates cell types)
#indv:   cell type separated / species + batches integrated  (PC/UMAP separates cell types)
#NOTES:  UMI and gene counts are higher in humans than chimps

```
