<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-data-cormotif-tot-cnts</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-data-cormotif-tot-cnts</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">July 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtreefd98145b0ad57351357e829de8f6f7ef4bd3261dtargetblankfd98145a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/fd98145b0ad57351357e829de8f6f7ef4bd3261d" target="_blank">fd98145</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtreefd98145b0ad57351357e829de8f6f7ef4bd3261dtargetblankfd98145a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/fd98145b0ad57351357e829de8f6f7ef4bd3261d" target="_blank">fd98145</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

Unstaged changes:
    Modified:   analysis/10x-data-6-cormotif-tot-cnts.Rmd
    Modified:   analysis/10x-data-6-cormotif-tot.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/license.Rmd
    Modified:   output/data-qc-prefilter.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-data-6-cormotif-tot-cnts.Rmd</code>) and HTML (<code>docs/10x-data-6-cormotif-tot-cnts.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e62c53703ed251e27c4d23397131fd218013a635/analysis/10x-data-6-cormotif-tot-cnts.Rmd" target="_blank">e62c537</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-05
</td>
<td>
updated analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---correlation-motifs" class="section level1">
<h1>10X Data Cellranger Analysis - Correlation Motifs</h1>
<p>Test for differential expression correlation motifs between different cell classifications in humans and chimpanzees.</p>
<p>Standard Cormotif (see 10x-data-6-cormotif-tot):</p>
<ul>
<li>calculates t statistics from running limma on log-cpm data</li>
</ul>
<p>Instead of doing this, here we used a modified version of Cormotif that Kenneth prepared:</p>
<ul>
<li>adds a call to voom before running limma so that limma can use vooms precision weights</li>
<li>then uses these t statistics for the rest of the analysis</li>
</ul>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(Cormotif)
library(gplots)
library(ComplexHeatmap)

source(&quot;./code/CormotifCounts.R&quot;)</code></pre>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate &lt;- paste0(data@meta.data$Individual,&quot;.&quot;,data@meta.data$Replicate)</code></pre>
<p>Define genes of interest: all except mitochondrial genes and ribosomal genes</p>
<pre class="r"><code>genes &lt;- rownames(data@assays$RNA@counts)
genes.mito &lt;- c(&quot;MT-ATP6&quot;,&quot;MT-ATP8&quot;,&quot;MT-CO1&quot;,&quot;MT-CO2&quot;,&quot;MT-CO3&quot;,&quot;MT-CYB&quot;,&quot;MT-ND1&quot;,&quot;MT-ND2&quot;,&quot;MT-ND3&quot;,&quot;MT-ND4&quot;,&quot;MT-ND4L&quot;,&quot;MT-ND5&quot;)
genes.ribo &lt;- grep(&#39;^RP&#39;,genes,value=T)
genes.no.mito.ribo &lt;- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)</code></pre>
</div>
<div id="run-cormotif-to-identify-de-genes" class="section level2">
<h2>Run Cormotif to Identify DE Genes</h2>
<p>Cormotif is a Bayesian clustering approach that identifies common temporal expression patterns (referred to as “correlation motifs”) that best fit the given data.</p>
<p>Cormotif helps to overcome issues of incomplete power affecting naïve pairwise DE comparisons and helps to account for dependency in data coming from different time points (stages of differentiation, etc). This joint modeling technique leverages expression information shared across time points to identify correlation motifs.</p>
<p>NOTE: In this modified version of Cormotif, we use raw counts instead of normalized log2 transformed cpm.</p>
<p>Plots of Interest:</p>
<ul>
<li>probability that motif k is differentially expressed in study d (bestmotif$motif.q)</li>
<li>posterior probabilities for each gene to be differentially expressed in each study for the best fitted model (bestmotif$p.post)</li>
</ul>
<p>Assemble normalized gene counts and run cormotif. <a href="https://www.biostars.org/p/317701/" class="uri">https://www.biostars.org/p/317701/</a></p>
</div>
<div id="cormotif-using-pseudobulk-data" class="section level2">
<h2>Cormotif using Pseudobulk Data</h2>
<p>Pseudobulk: Species vs. Stages * each time point separately + genes examined: 11579 + best fit model: 2 motifs + unique DE genes: 3064 in Time 0, 80 in Time 1, 0 in Time 2 * 3 time points and 4 comparisons + genes examined: 11579 + best fit model: 4 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 83.38% (9655 genes) + unique chimp DE genes: 11 in t0-t1, 0 in t1-t2 + unique human DE genes: 0 in t0-t1, 229 in t1-t2 * 3 time points and 4 comparisons + genes examined: 11579 + best fit model: 5 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 73.37% (8729 genes) + unique chimp DE genes: 607 in Time 0, 0 in Time 1, 0 in Time 2 + unique human DE genes: 68 in Time 0, 2 in Time 1, 60 in Time 2</p>
<p>Pseudobulk: Species vs. General Clusters (res=0.05) * each cluster separately + genes examined: 10742 + best fit model: 5 motifs + unique DE genes: 1019 in iPSC.c1, 0 in iPSC.c2, 0 in iPSC.c3, 270 in MSC.c1, 0 in Osteogenic.c1, 0 in Osteogenic.c2 * 3 clusters and 4 comparisons (A) + genes examined: 10742 + best fit model: 4 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 92.99% (9989 genes) + unique chimp DE genes: 15 in iPSC.c1-MSC.c1, 0 in MSC.c1-Osteogenic.c1 + unique human DE genes: 0 in iPSC.c1-MSC.c1, 89 in MSC.c1-Osteogenic.c1 * 3 clusters and 4 comparisons (B) + genes examined: 10742 + best fit model: 4 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 71.24% (7653 genes) + unique chimp DE genes: 7 in iPSC.c1-MSC.c1, 0 in MSC.c1-Osteogenic.c2 + unique human DE genes: 0 in iPSC.c1-MSC.c1, 300 in MSC.c1-Osteogenic.c1 * 4 clusters and 8 comparisons + genes examined: 10742 + best fit model: 10 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 50.42% (5416 genes) + unique chimp DE genes: 16 in iPSC.c1-MSC.c1, 3 in MSC.c1-Osteogenic.c1, 0 in MSC.c1-Osteogenic.c2 + unique human DE genes: 0 in iPSC.c1-MSC.c1, 7 in MSC.c1-Osteogenic.c1, 207 in MSC.c1-Osteogenic.c2 * 3 clusters and 6 comparisons (A) + genes examined: 10742 + best fit model: 5 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 80.75% (8674 genes) + unique chimp DE genes: 564 in iPSC.c1, 0 in MSC.c1, 0 in Osteogenic.c1 + unique human DE genes: 243 in iPSC.c1, 0 in MSC.c1, 12 in Osteogenic.c1 * 3 clusters and 6 comparisons (B) + genes examined: 10742 + best fit model: 7 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 63.38% (6808 genes) + unique chimp DE genes: 321 in iPSC.c1, 0 in MSC.c1, 0 in Osteogenic.c1 + unique human DE genes: 32 in iPSC.c1, 17 in MSC.c1, 215 in Osteogenic.c1</p>
<p>Pseudobulk: Species vs. General Ad Hoc Classifications (thresh0) * each ad hoc classification separately + genes examined: 11824 + best fit model: 2 motifs + unique DE genes: 3171 in iPSCs, 2 in MSCs, 0 in Osteogenic * 3 ad hoc classifications and 4 comparisons + genes examined: 11824 + best fit model: 2 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 23.57% (2787 genes) + unique chimp DE genes: 395 in iPSC-MSC, 10 in MSC-Osteogenic + unique human DE genes: 0 in iPSC-MSC, 79 in MSC-Osteogenic * 3 ad hoc classifications and 6 comparisons + genes examined: 11824 + best fit model: 5 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 20.96% (1625 genes) + unique chimp DE genes: 173 in iPSC, 445 in MSC, 0 in Osteogenic + unique human DE genes: 145 in iPSC, 0 in MSC, 3 in Osteogenic</p>
<p>Pseudobulk: Species vs. Osteogenic Clusters (res=0.50) * each cluster separately + genes examined: 10671 + best fit model: 2 motifs + unique DE genes: 2 in O.c1, 0 in O.c2, 2 in O.c3, 0 in O.c4 * 4 osteogenic clusters and 6 comparisons + genes examined: 10671 + best fit model: 6 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 73.78% (7873 genes) + unique chimp DE genes: 0 in O.c1-O.c2, 23 in O.c2-O.c3, 0 in O.c3-O.c4 + unique human DE genes: 0 in O.c1-O.c2, 0 in O.c2-O.c3, 1199 in O.c3-O.c4</p>
<p>Pseudobulk: Species vs. Osteogenic Clusters (res=0.75) * each cluster separately + genes examined: 10397 + best fit model: 3 motifs + unique DE genes: 3 in O.c1, 0 in O.c2, 0 in O.c3, 5 in O.c4, 3 in O.c5, 0 in O.c6, 0 in O.c7, 0 in O.c8 * 8 osteogenic clusters and 14 comparisons + genes examined: 10397 + best fit model: 15 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 20.16% (2096 genes) + unique chimp DE genes: 0 in O.c1-O.c2, 0 in O.c2-O.c3, 0 in O.c3-O.c4, 0 in O.c4-O.c5, 183 in O.c5-O.c6, 44 in O.c6-O.c7, 10 in O.c7-O.c8 + unique human DE genes: 3 in O.c1-O.c2, 4 in O.c2-O.c3, 7 in O.c3-O.c4, 0 in O.c4-O.c5, 0 in O.c5-O.c6, 0 in O.c6-O.c7, 31 in O.c7-O.c8</p>
<p>Pseudobulk: Species vs. Osteogenic Clusters (res=0.50) (Time 2 only) * each cluster separately + genes examined: 10554 + best fit model: 2 motifs + unique DE genes: 0 in O.c1, 0 in O.c2, 0 in O.c3, 0 in O.c4 * 4 osteogenic clusters and 6 comparisons + genes examined: 10554 + best fit model: 4 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 84.32% (8899 genes) + unique chimp DE genes: 0 in O.c1-O.c2, 49 in O.c2-O.c3, 0 in O.c3-O.c4 + unique human DE genes: 0 in O.c1-O.c2, 0 in O.c2-O.c3, 1200 in O.c3-O.c4</p>
<p>Pseudobulk: Species vs. Osteogenic Clusters (res=0.75) (Time 2 only) * each cluster separately + genes examined: 10212 + best fit model: 3 motifs + unique DE genes: 0 in O.c1, 0 in O.c2, 0 in O.c3, 0 in O.c4, 2 in O.c5, 0 in O.c6, 0 in O.c7, 0 in O.c8 * 8 osteogenic clusters and 14 comparisons + genes examined: 10212 + best fit model: 15 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 10.00% (1021 genes) + unique chimp DE genes: 0 in O.c1-O.c2, 0 in O.c2-O.c3, 0 in O.c3-O.c4, 0 in O.c4-O.c5, 2 in O.c5-O.c6, 3257 in O.c6-O.c7, 0 in O.c7-O.c8 + unique human DE genes: 0 in O.c1-O.c2, 22 in O.c2-O.c3, 0 in O.c3-O.c4, 0 in O.c4-O.c5, 0 in O.c5-O.c6, 0 in O.c6-O.c7, 0 in O.c7-O.c8</p>
<p>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0) * each osteogenic ad hoc classification separately + genes examined: 10669 + best fit model: 2 motifs + unique DE genes: 0 in Pre, 0 in Obl, 0 in Emb, 2 in Min, 0 in Mat * 5 osteogenic ad hoc classifications and 8 comparisons + genes examined: 10669 + best fit model: 1 motif + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 100.00% (10669 genes) + unique chimp DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat + unique human DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat</p>
<p>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0) * each osteogenic ad hoc classification separately + genes examined: 9999 + best fit model: 3 motifs + unique DE genes: 7 in Pre, 0 in Obl, 0 in Emb, 0 in Min, 0 in Mat * 5 osteogenic ad hoc classifications and 8 comparisons + genes examined: 9999 + best fit model: 1 motif + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 99.90% (9989 genes) + unique chimp DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat + unique human DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat</p>
<p>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0) (Time 2 only) * each osteogenic ad hoc classification separately + genes examined: 10822 + best fit model: 2 motifs + unique DE genes: 0 in Pre, 0 in Obl, 0 in Emb, 108 in Min, 0 in Mat * 5 osteogenic ad hoc classifications and 8 comparisons + genes examined: 10822 + best fit model: 1 motif + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 0.00% (0 genes) + unique chimp DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat + unique human DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat</p>
<p>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0) (Time 2 only) * each osteogenic ad hoc classification separately + genes examined: 10535 + best fit model: 3 motifs + unique DE genes: 0 in Pre, 0 in Obl, 0 in Emb, 0 in Min, 0 in Mat * 5 osteogenic ad hoc classifications and 8 comparisons + genes examined: 10535 + best fit model: 1 motifs + conservation across stages (species difference in gene posterior probabilities &lt;= 0.3): 100.00% (10535 genes) + unique chimp DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat + unique human DE genes: 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat</p>
<div id="pseudobulk-species-vs.stages" class="section level3">
<h3>Pseudobulk: Species vs. Stages</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg &lt;- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Stage&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.stg)</code></pre>
<div id="each-time-point-separately" class="section level4">
<h4>each time point separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;.&quot;),
                      levels=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3),
                        c2 = c(4, 5, 6))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Stage=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
                           col=list(Stage=c(&quot;Time 0&quot;=&quot;pink&quot;,&quot;Time 1&quot;=&quot;mediumorchid&quot;,&quot;Time 2&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

# Examine human/chimp differences unique to each time point
prob.motifT0  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifT0)
head(prob.motifT0)
#boxplot(tmm[prob.motifT0[1],]~groupFactor)

prob.motifT1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifT1)
head(prob.motifT1)
#boxplot(tmm[prob.motifT1[1],]~groupFactor)

prob.motifT2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifT2)
head(prob.motifT2)
#boxplot(tmm[prob.motifT2[1],]~groupFactor)

# Examine human/chimp differences shared across time points
prob.motifT012  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifT012)
head(prob.motifT012)
#boxplot(tmm[prob.motifT012[1],]~groupFactor)

prob.motifT01  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifT01)
head(prob.motifT01)
#boxplot(tmm[prob.motifT01[1],]~groupFactor)

prob.motifT02  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifT02)
head(prob.motifT02)
#boxplot(tmm[prob.motifT02[1],]~groupFactor)

prob.motifT12  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifT12)
head(prob.motifT12)
#boxplot(tmm[prob.motifT12[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2)
rm(prob.motifT0,prob.motifT1,prob.motifT2)
rm(prob.motifT012,prob.motifT01,prob.motifT02,prob.motifT12)</code></pre>
</div>
<div id="time-points-and-4-comparisons" class="section level4">
<h4>3 time points and 4 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;.&quot;),
                      levels=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 4, 5),
                        c2 = c(2, 3, 5, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 0&quot;,&quot;Time 1&quot;),
                                       c(&quot;Time 1&quot;,&quot;Time 2&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;)),
                           col=list(Stage=c(&quot;Time 0&quot;=&quot;pink&quot;,&quot;Time 1&quot;=&quot;mediumorchid&quot;,&quot;Time 2&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.t0-t1&quot;,&quot;H.t1-t2&quot;,&quot;C.t0-t1&quot;,&quot;C.t1-t2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.t0-t1&quot;,&quot;H.t1-t2&quot;,&quot;C.t0-t1&quot;,&quot;C.t1-t2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9655 genes) -&gt; 83.38%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (7596 genes) -&gt; 65.601%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2264 genes) -&gt; 19.55%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)</code></pre>
</div>
<div id="time-points-and-6-comparisons" class="section level4">
<h4>3 time points and 6 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;.&quot;),
                      levels=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 1, 4, 5, 4),
                        c2 = c(2, 3, 3, 5, 6, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 0&quot;,&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 0&quot;),
                                       c(&quot;Time 1&quot;,&quot;Time 2&quot;,&quot;Time 2&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;,&quot;Time 2&quot;)),
                           col=list(Stage=c(&quot;Time 0&quot;=&quot;pink&quot;,&quot;Time 1&quot;=&quot;mediumorchid&quot;,&quot;Time 2&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.t0-t1&quot;,&quot;H.t1-t2&quot;,&quot;H.t0-t2&quot;,&quot;C.t0-t1&quot;,&quot;C.t1-t2&quot;,&quot;C.t0-t2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.t0-t1&quot;,&quot;H.t1-t2&quot;,&quot;H.t0-t2&quot;,&quot;C.t0-t1&quot;,&quot;C.t1-t2&quot;,&quot;C.t0-t2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (8496 genes) -&gt; 73.37%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (6580 genes) -&gt; 56.83%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2672 genes) -&gt; 23.08%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

prob.motif5  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif5)
head(prob.motif5)
#boxplot(tmm[prob.motif5[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-stage6.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4,prob.motif5)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.general-clusters-res0.05" class="section level3">
<h3>Pseudobulk: Species vs. General Clusters (res=0.05)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Cluster0.05==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$Cluster0.05[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)</code></pre>
<div id="each-cluster-separately" class="section level4">
<h4>each cluster separately</h4>
<p>Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, iPSC.c2, iPSC.c3, MSC.c1, Osteogenic.c1, and Osteogenic.c2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3,  4,  5,  6),
                        c2 = c(7, 8, 9, 10, 11, 12))

# Assess differences within each cluster
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;iPSC.c2&quot;=&quot;pink4&quot;,
                                              &quot;iPSC.c3&quot;=&quot;grey&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        column_split=c(1,1,1,2,2,2),
        #column_split=c(1,1,2,3,3,4),
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

prob.motif5  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif5)
head(prob.motif5)
#boxplot(tmm[prob.motif5[1],]~groupFactor)

# Examine human/chimp differences unique to each cluster
prob.motifI1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifI1)
head(prob.motifI1)
#boxplot(tmm[prob.motifI1[1],]~groupFactor)

prob.motifI2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifI2)
head(prob.motifI2)
#boxplot(tmm[prob.motifI2[1],]~groupFactor)

prob.motifI3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifI3)
head(prob.motifI3)
#boxplot(tmm[prob.motifI3[1],]~groupFactor)

prob.motifM1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifM1)
head(prob.motifM1)
#boxplot(tmm[prob.motifM1[1],]~groupFactor)

prob.motifO1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifI123M1O12  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifI123M1O12)
head(prob.motifI123M1O12)
#boxplot(tmm[prob.motifI123M1O12[1],]~groupFactor)

prob.motifI123M1     &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifI123M1)
head(prob.motifI123M1)
#boxplot(tmm[prob.motifI123M1[1],]~groupFactor)

prob.motifI123O12    &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifI123O12)
head(prob.motifI123O12)
#boxplot(tmm[prob.motifI123O12[1],]~groupFactor)

prob.motifM1O12     &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifM1O12)
head(prob.motifM1O12)
#boxplot(tmm[prob.motifM1O12[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4,prob.motif5)
rm(prob.motifI1,prob.motifI2,prob.motifI3,prob.motifM1,prob.motifO1,prob.motifO2)
rm(prob.motifI123M1O12,prob.motifI123M1,prob.motifI123O12,prob.motifM1O12)</code></pre>
</div>
<div id="clusters-and-4-comparisons-a" class="section level4">
<h4>3 clusters and 4 comparisons (A)</h4>
<p>iPSC.c2, iPSC.c3, and Osteogenic.c2 are excluded here.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c1 for each species)
compareID &lt;- data.frame(c1 = c(1, 4,  7, 10),
                        c2 = c(4, 5, 10, 11))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Clusters&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c1: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9989 genes) -&gt; 92.99%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (8994 genes) -&gt; 83.73%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (6261 genes) -&gt; 58.29%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)</code></pre>
</div>
<div id="clusters-and-4-comparisons-b" class="section level4">
<h4>3 clusters and 4 comparisons (B)</h4>
<p>iPSC.c2, iPSC.c3, and Osteogenic.c1 are excluded here.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c2 for each species)
compareID &lt;- data.frame(c1 = c(1, 4,  7, 10),
                        c2 = c(4, 6, 10, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c2&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Clusters&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7653 genes) -&gt; 71.24%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (6113 genes) -&gt; 56.91%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (1337 genes) -&gt; 12.45%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-cluster4b.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)</code></pre>
</div>
<div id="clusters-and-8-comparisons" class="section level4">
<h4>4 clusters and 8 comparisons</h4>
<p>Only iPSC.c2 and iPSC.c3 are excluded here.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, Osteogenic.c1, and Osteogenic.c2 for each species)
compareID &lt;- data.frame(c1 = c(1, 4, 4, 5,  7, 10, 10, 11),
                        c2 = c(4, 5, 6, 6, 10, 11, 12, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:20,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 10 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c2&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c2&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Clusters&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;H.Osteogenic.c1-Osteogenic.c2&quot;,
                         &quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;,&quot;C.Osteogenic.c1-Osteogenic.c2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;H.Osteogenic.c1-Osteogenic.c2&quot;,
                                             &quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;,&quot;C.Osteogenic.c1-Osteogenic.c2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, Osteogenic.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (5416 genes) -&gt; 50.42%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (5349 genes) -&gt; 49.80%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.1 (4604 genes) -&gt; 42.86%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp; (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)</code></pre>
</div>
<div id="clusters-and-6-comparisons-a" class="section level4">
<h4>3 clusters and 6 comparisons (A)</h4>
<p>iPSC.c2, iPSC.c3, and Osteogenic.c2 are excluded here.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c1 for each species)
compareID &lt;- data.frame(c1 = c(1, 4, 1,  7, 10,  7),
                        c2 = c(4, 5, 5, 10, 11, 11))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c1&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;H.iPSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-Osteogenic.c1&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c1&quot;,&quot;H.iPSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c1&quot;,&quot;C.iPSC.c1-Osteogenic.c1&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (8674 genes) -&gt; 80.75%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (7549 genes) -&gt; 70.28%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (5066 genes) -&gt; 47.16%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

prob.motif5  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif5)
head(prob.motif5)
#boxplot(tmm[prob.motif5[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-cluster6a.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4,prob.motif5)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)</code></pre>
</div>
<div id="clusters-and-6-comparisons-b" class="section level4">
<h4>3 clusters and 6 comparisons (B)</h4>
<p>iPSC.c2, iPSC.c3, and Osteogenic.c1 are excluded here.</p>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c2 for each species)
compareID &lt;- data.frame(c1 = c(1, 4, 1,  7, 10,  7),
                        c2 = c(4, 6, 6, 10, 12, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 7 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c2&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c2&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;H.iPSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-Osteogenic.c2&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC.c1-MSC.c1&quot;,&quot;H.MSC.c1-Osteogenic.c2&quot;,&quot;H.iPSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-MSC.c1&quot;,&quot;C.MSC.c1-Osteogenic.c2&quot;,&quot;C.iPSC.c1-Osteogenic.c2&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (6808 genes) -&gt; 63.38%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (5263 genes) -&gt; 48.99%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (1769 genes) -&gt; 16.47%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (skipped)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.general-ad-hoc-classifications-thresh0" class="section level3">
<h3>Pseudobulk: Species vs. General Ad Hoc Classifications (thresh0)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=AdHoc.Assign.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- as.character(dataSub@meta.data$AdHoc.Assign.thresh0[w][1])
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<div id="each-ad-hoc-classification-separately" class="section level4">
<h4>each ad hoc classification separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = &quot;_&quot;),
                      levels=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Human_other&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;,&quot;Chimp_other&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each ad hoc assignment)
compareID &lt;- data.frame(c1 = c(1, 2, 3),
                        c2 = c(5, 6, 7))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
                           col=list(AdHoc=c(&quot;iPSC&quot;=&quot;pink&quot;,&quot;MSC&quot;=&quot;mediumorchid&quot;,&quot;Osteogenic&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

# Examine human/chimp differences unique to each time point
prob.motifI  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifI)
head(prob.motifI)
#boxplot(tmm[prob.motifI[1],]~groupFactor)

prob.motifM  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifM)
head(prob.motifM)
#boxplot(tmm[prob.motifM[1],]~groupFactor)

prob.motifO  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifO)
head(prob.motifO)
#boxplot(tmm[prob.motifO[1],]~groupFactor)

# Examine human/chimp differences shared across time points
prob.motifIMO  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifIMO)
head(prob.motifIMO)
#boxplot(tmm[prob.motifIMO[1],]~groupFactor)

prob.motifIM  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5),])
length(prob.motifIM)
head(prob.motifIM)
#boxplot(tmm[prob.motifIM[1],]~groupFactor)

prob.motifIO  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifIO)
head(prob.motifIO)
#boxplot(tmm[prob.motifIO[1],]~groupFactor)

prob.motifMO  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5),])
length(prob.motifMO)
head(prob.motifMO)
#boxplot(tmm[prob.motifMO[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2)
rm(prob.motifI,prob.motifM,prob.motifO)
rm(prob.motifIMO,prob.motifIM,prob.motifIO,prob.motifMO)</code></pre>
</div>
<div id="ad-hoc-classifications-and-4-comparisons" class="section level4">
<h4>3 ad hoc classifications and 4 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = &quot;_&quot;),
                      levels=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Human_other&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;,&quot;Chimp_other&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 4, 5),
                        c2 = c(2, 3, 5, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;iPSC&quot;,&quot;MSC&quot;),
                                       c(&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)),
                           col=list(Stage=c(&quot;iPSC&quot;=&quot;pink&quot;,
                                            &quot;MSC&quot;=&quot;mediumorchid&quot;,
                                            &quot;Osteogenic&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC-MSC&quot;,&quot;H.MSC-Osteogenic&quot;,&quot;C.iPSC-MSC&quot;,&quot;C.MSC-Osteogenic&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC-MSC&quot;,&quot;H.MSC-Osteogenic&quot;,&quot;C.iPSC-MSC&quot;,&quot;C.MSC-Osteogenic&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (2787 genes) -&gt; 23.57%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (1717 genes) -&gt; 14.52%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 ( 651 genes) -&gt;  5.51%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)</code></pre>
</div>
<div id="ad-hoc-classifications-and-6-comparisons" class="section level4">
<h4>3 ad hoc classifications and 6 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = &quot;_&quot;),
                      levels=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Human_other&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;,&quot;Chimp_other&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 1, 4, 5, 4),
                        c2 = c(2, 3, 3, 5, 6, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;iPSC&quot;,&quot;iPSC&quot;,&quot;MSC&quot;,&quot;iPSC&quot;),
                                       c(&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;Osteogenic&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;Osteogenic&quot;)),
                           col=list(Stage=c(&quot;iPSC&quot;=&quot;pink&quot;,
                                            &quot;MSC&quot;=&quot;mediumorchid&quot;,
                                            &quot;Osteogenic&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.iPSC-MSC&quot;,&quot;H.MSC-Osteogenic&quot;,&quot;H.iPSC-Osteogenic&quot;,&quot;C.iPSC-MSC&quot;,&quot;C.MSC-Osteogenic&quot;,&quot;C.iPSC-Osteogenic&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.iPSC-MSC&quot;,&quot;H.MSC-Osteogenic&quot;,&quot;H.iPSC-Osteogenic&quot;,&quot;C.iPSC-MSC&quot;,&quot;C.MSC-Osteogenic&quot;,&quot;C.iPSC-Osteogenic&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (2478 genes) -&gt; 20.96%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (1480 genes) -&gt; 12.52%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 ( 557 genes) -&gt;  4.71%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp; (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp; (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

prob.motif5  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif5)
head(prob.motif5)
#boxplot(tmm[prob.motif5[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-adhoc6.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4,prob.motif5)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-clusters-res0.50" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Clusters (res=0.50)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstCluster0.50==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$OstCluster0.50[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)</code></pre>
<div id="each-cluster-separately-1" class="section level4">
<h4>each cluster separately</h4>
<p>Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.</p>
<pre class="r"><code># Compile group labels of each sample (clusters are off by 1)
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4),
                        c2 = c(5, 6, 7, 8))

# Assess differences within each cluster
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:20,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

# Examine human/chimp differences unique to each cluster
prob.motifO1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO1234  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO1234)
head(prob.motifO1234)
#boxplot(tmm[prob.motifO1234[1],]~groupFactor)

prob.motifO134  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO134)
head(prob.motifO134)
#boxplot(tmm[prob.motifO134[1],]~groupFactor)

prob.motifO13  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO13)
head(prob.motifO13)
#boxplot(tmm[prob.motifO13[1],]~groupFactor)

prob.motifO34  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO34)
head(prob.motifO34)
#boxplot(tmm[prob.motifO34[1],]~groupFactor)

prob.motifO14  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO14)
head(prob.motifO14)
#boxplot(tmm[prob.motifO14[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4)
rm(prob.motifO1234,prob.motifO134,prob.motifO13,prob.motifO34,prob.motifO14)</code></pre>
</div>
<div id="osteogenic-clusters-and-6-comparisons" class="section level4">
<h4>4 osteogenic clusters and 6 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 5, 6, 7),
                        c2 = c(2, 3, 4, 6, 7, 8))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 6 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,
                                           &quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;),
                                         c(&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                           &quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,
                         &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,
                                             &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7873 genes) -&gt; 73.78%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (6027 genes) -&gt; 56.48%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2790 genes) -&gt; 26.15%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

prob.motif4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5),])
length(prob.motif4)
head(prob.motif4)
#boxplot(tmm[prob.motif4[1],]~groupFactor)

prob.motif5  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif5)
head(prob.motif5)
#boxplot(tmm[prob.motif5[1],]~groupFactor)

prob.motif6  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5),])
length(prob.motif6)
head(prob.motif6)
#boxplot(tmm[prob.motif6[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&gt;0.5 &amp; gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&gt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&gt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4,prob.motif5,prob.motif6)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-clusters-res0.75" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Clusters (res=0.75)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                 &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstCluster0.75==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$OstCluster0.75[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)</code></pre>
<div id="each-cluster-separately-2" class="section level4">
<h4>each cluster separately</h4>
<p>Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.</p>
<pre class="r"><code># Compile group labels of each sample (clusters are off by 1)
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Human_Osteogenic.c5&quot;,&quot;Human_Osteogenic.c6&quot;,&quot;Human_Osteogenic.c7&quot;,&quot;Human_Osteogenic.c8&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c5&quot;,&quot;Chimp_Osteogenic.c6&quot;,&quot;Chimp_Osteogenic.c7&quot;,&quot;Chimp_Osteogenic.c8&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID &lt;- data.frame(c1 = c(1,  2,  3,  4,  5,  6,  7,  8),
                        c2 = c(9, 10, 11, 12, 13, 14, 15, 16))

# Assess differences within each cluster
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                     &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;,
                                              &quot;Osteogenic.c5&quot;=&quot;lightblue1&quot;,
                                              &quot;Osteogenic.c6&quot;=&quot;royalblue3&quot;,
                                              &quot;Osteogenic.c7&quot;=&quot;grey&quot;,
                                              &quot;Osteogenic.c8&quot;=&quot;black&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

# Examine human/chimp differences unique to each cluster
prob.motifO1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

prob.motifO5  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO5)
head(prob.motifO5)
#boxplot(tmm[prob.motifO5[1],]~groupFactor)

prob.motifO6  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO6)
head(prob.motifO6)
#boxplot(tmm[prob.motifO6[1],]~groupFactor)

prob.motifO7  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO7)
head(prob.motifO7)
#boxplot(tmm[prob.motifO7[1],]~groupFactor)

prob.motifO8  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifO8)
head(prob.motifO8)
#boxplot(tmm[prob.motifO8[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO12345678  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                            gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifO12345678)
head(prob.motifO12345678)
#boxplot(tmm[prob.motifO12345678[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3,prob.motif4)
rm(prob.motifO1,prob.motifO2,prob.motifO3)
rm(prob.motifO12345678)</code></pre>
</div>
<div id="osteogenic-clusters-and-14-comparisons" class="section level4">
<h4>8 osteogenic clusters and 14 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Human_Osteogenic.c5&quot;,&quot;Human_Osteogenic.c6&quot;,&quot;Human_Osteogenic.c7&quot;,&quot;Human_Osteogenic.c8&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c5&quot;,&quot;Chimp_Osteogenic.c6&quot;,&quot;Chimp_Osteogenic.c7&quot;,&quot;Chimp_Osteogenic.c8&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 5, 6, 7,  9, 10, 11, 12, 13, 14, 15),
                        c2 = c(2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:20,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 15 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,
                                         &quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;),
                                       c(&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;,
                                         &quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)),
                           col=list(AdHoc=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                            &quot;Osteogenic.c2&quot;=&quot;dodgerblue4&quot;,
                                            &quot;Osteogenic.c3&quot;=&quot;deepskyblue2&quot;,
                                            &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;,
                                            &quot;Osteogenic.c5&quot;=&quot;lightblue1&quot;,
                                            &quot;Osteogenic.c6&quot;=&quot;royalblue3&quot;,
                                            &quot;Osteogenic.c7&quot;=&quot;grey&quot;,
                                            &quot;Osteogenic.c8&quot;=&quot;black&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,&quot;H.O4-O5&quot;,&quot;H.O5-O6&quot;,&quot;H.O6-O7&quot;,&quot;H.O7-O8&quot;,
                         &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;,&quot;C.O4-O5&quot;,&quot;C.O5-O6&quot;,&quot;C.O6-O7&quot;,&quot;C.O7-O8&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,&quot;H.O4-O5&quot;,&quot;H.O5-O6&quot;,&quot;H.O6-O7&quot;,&quot;H.O7-O8&quot;,
                                             &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;,&quot;C.O4-O5&quot;,&quot;C.O5-O6&quot;,&quot;C.O6-O7&quot;,&quot;C.O7-O8&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (4033 genes) -&gt; 38.79%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (2308 genes) -&gt; 22.20%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.2), ]))/dim(dge$counts)[1]

#0.1 ( 858 genes) -&gt;  8.25%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[ ,9]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[ ,9]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED - too many)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&gt;0.5 &amp; gene.prob[ ,9]&gt;0.5 &amp; gene.prob[,10]&gt;0.5 &amp; gene.prob[,11]&gt;0.5 &amp;
                                    gene.prob[,12]&gt;0.5 &amp; gene.prob[,13]&gt;0.5 &amp; gene.prob[,14]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&gt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&gt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&gt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&gt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifC5 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&gt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC5)
head(prob.motifC5)
#boxplot(tmm[prob.motifC5[1],]~groupFactor)

prob.motifC6 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&gt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC6)
head(prob.motifC6)
#boxplot(tmm[prob.motifC6[1],]~groupFactor)

prob.motifC7 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&gt;0.5),])
length(prob.motifC7)
head(prob.motifC7)
#boxplot(tmm[prob.motifC7[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp; gene.prob[ ,4]&gt;0.5 &amp;
                                    gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&gt;0.5 &amp; gene.prob[ ,7]&gt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&gt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

prob.motifH5 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH5)
head(prob.motifH5)
#boxplot(tmm[prob.motifH5[1],]~groupFactor)

prob.motifH6 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&gt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH6)
head(prob.motifH6)
#boxplot(tmm[prob.motifH6[1],]~groupFactor)

prob.motifH7 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&gt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH7)
head(prob.motifH7)
#boxplot(tmm[prob.motifH7[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4,prob.motifH5,prob.motifH6,prob.motifH7)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4,prob.motifC5,prob.motifC6,prob.motifC7)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-clusters-res0.50-time-2-only" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Clusters (res=0.50) (Time 2 only)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstCluster0.50==j &amp; Stage==&quot;Time 2&quot;)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$OstCluster0.50[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)</code></pre>
<div id="each-cluster-separately-3" class="section level4">
<h4>each cluster separately</h4>
<p>Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.</p>
<pre class="r"><code># Compile group labels of each sample (clusters are off by 1)
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4),
                        c2 = c(5, 6, 7, 8))

# Assess differences within each cluster
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences unique to each cluster
prob.motifO1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO1234  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO1234)
head(prob.motifO1234)
#boxplot(tmm[prob.motifO1234[1],]~groupFactor)

prob.motifO134  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO134)
head(prob.motifO134)
#boxplot(tmm[prob.motifO134[1],]~groupFactor)

prob.motifO13  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5),])
length(prob.motifO13)
head(prob.motifO13)
#boxplot(tmm[prob.motifO13[1],]~groupFactor)

prob.motifO34  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO34)
head(prob.motifO34)
#boxplot(tmm[prob.motifO34[1],]~groupFactor)

prob.motifO14  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5),])
length(prob.motifO14)
head(prob.motifO14)
#boxplot(tmm[prob.motifO14[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4)
rm(prob.motifO1234,prob.motifO134,prob.motifO13,prob.motifO34,prob.motifO14)</code></pre>
</div>
<div id="osteogenic-clusters-and-6-comparisons-1" class="section level4">
<h4>4 osteogenic clusters and 6 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 5, 6, 7),
                        c2 = c(2, 3, 4, 6, 7, 8))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 7 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,
                                           &quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;),
                                         c(&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                           &quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,
                         &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,
                                             &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (8899 genes) -&gt; 84.32%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (7297 genes) -&gt; 69.14%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (3983 genes) -&gt; 37.74%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&gt;0.5 &amp; gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&gt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&gt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp;
                                    gene.prob[ ,4]&lt;0.5 &amp; gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.T2.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-clusters-res0.75-time-2-only" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Clusters (res=0.75) (Time 2 only)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                 &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstCluster0.75==j &amp; Stage==&quot;Time 2&quot;)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$OstCluster0.75[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)</code></pre>
<div id="each-cluster-separately-4" class="section level4">
<h4>each cluster separately</h4>
<p>Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.</p>
<pre class="r"><code># Compile group labels of each sample (clusters are off by 1)
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Human_Osteogenic.c5&quot;,&quot;Human_Osteogenic.c6&quot;,&quot;Human_Osteogenic.c7&quot;,&quot;Human_Osteogenic.c8&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c5&quot;,&quot;Chimp_Osteogenic.c6&quot;,&quot;Chimp_Osteogenic.c7&quot;,&quot;Chimp_Osteogenic.c8&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID &lt;- data.frame(c1 = c(1,  2,  3,  4,  5,  6,  7,  8),
                        c2 = c(9, 10, 11, 12, 13, 14, 15, 16))

# Assess differences within each cluster
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                     &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;,
                                              &quot;Osteogenic.c5&quot;=&quot;lightblue1&quot;,
                                              &quot;Osteogenic.c6&quot;=&quot;royalblue3&quot;,
                                              &quot;Osteogenic.c7&quot;=&quot;grey&quot;,
                                              &quot;Osteogenic.c8&quot;=&quot;black&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences unique to each cluster
prob.motifO1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

prob.motifO5  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO5)
head(prob.motifO5)
#boxplot(tmm[prob.motifO5[1],]~groupFactor)

prob.motifO6  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO6)
head(prob.motifO6)
#boxplot(tmm[prob.motifO6[1],]~groupFactor)

prob.motifO7  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifO7)
head(prob.motifO7)
#boxplot(tmm[prob.motifO7[1],]~groupFactor)

prob.motifO8  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifO8)
head(prob.motifO8)
#boxplot(tmm[prob.motifO8[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO12345678  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                            gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifO12345678)
head(prob.motifO12345678)
#boxplot(tmm[prob.motifO12345678[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4,prob.motifO5,prob.motifO6,prob.motifO7,prob.motifO8)
rm(prob.motifO12345678)</code></pre>
</div>
<div id="osteogenic-clusters-and-14-comparisons-1" class="section level4">
<h4>8 osteogenic clusters and 14 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Human_Osteogenic.c5&quot;,&quot;Human_Osteogenic.c6&quot;,&quot;Human_Osteogenic.c7&quot;,&quot;Human_Osteogenic.c8&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c5&quot;,&quot;Chimp_Osteogenic.c6&quot;,&quot;Chimp_Osteogenic.c7&quot;,&quot;Chimp_Osteogenic.c8&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 5, 6, 7,  9, 10, 11, 12, 13, 14, 15),
                        c2 = c(2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:20,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 15 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,
                                         &quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;),
                                       c(&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;,
                                         &quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                         &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;)),
                           col=list(AdHoc=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                            &quot;Osteogenic.c2&quot;=&quot;dodgerblue4&quot;,
                                            &quot;Osteogenic.c3&quot;=&quot;deepskyblue2&quot;,
                                            &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;,
                                            &quot;Osteogenic.c5&quot;=&quot;lightblue1&quot;,
                                            &quot;Osteogenic.c6&quot;=&quot;royalblue3&quot;,
                                            &quot;Osteogenic.c7&quot;=&quot;grey&quot;,
                                            &quot;Osteogenic.c8&quot;=&quot;black&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,&quot;H.O4-O5&quot;,&quot;H.O5-O6&quot;,&quot;H.O6-O7&quot;,&quot;H.O7-O8&quot;,
                         &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;,&quot;C.O4-O5&quot;,&quot;C.O5-O6&quot;,&quot;C.O6-O7&quot;,&quot;C.O7-O8&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.O1-O2&quot;,&quot;H.O2-O3&quot;,&quot;H.O3-O4&quot;,&quot;H.O4-O5&quot;,&quot;H.O5-O6&quot;,&quot;H.O6-O7&quot;,&quot;H.O7-O8&quot;,
                                             &quot;C.O1-O2&quot;,&quot;C.O2-O3&quot;,&quot;C.O3-O4&quot;,&quot;C.O4-O5&quot;,&quot;C.O5-O6&quot;,&quot;C.O6-O7&quot;,&quot;C.O7-O8&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 ( 671 genes) -&gt;  6.57%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (   4 genes) -&gt;  0.04%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[, 9]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (   0 genes) -&gt;   0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[ ,9]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[ ,9]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,10]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,11]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,5] - gene.prob[,12]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,6] - gene.prob[,13]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,7] - gene.prob[,14]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&gt;0.5 &amp; gene.prob[ ,9]&gt;0.5 &amp; gene.prob[,10]&gt;0.5 &amp; gene.prob[,11]&gt;0.5 &amp;
                                    gene.prob[,12]&gt;0.5 &amp; gene.prob[,13]&gt;0.5 &amp; gene.prob[,14]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&gt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&gt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&gt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&gt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifC5 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&gt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC5)
head(prob.motifC5)
#boxplot(tmm[prob.motifC5[1],]~groupFactor)

prob.motifC6 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&gt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifC6)
head(prob.motifC6)
#boxplot(tmm[prob.motifC6[1],]~groupFactor)

prob.motifC7 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&gt;0.5),])
length(prob.motifC7)
head(prob.motifC7)
#boxplot(tmm[prob.motifC7[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp; gene.prob[ ,4]&gt;0.5 &amp;
                                    gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&gt;0.5 &amp; gene.prob[ ,7]&gt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[ ,1]&gt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&gt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&gt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&gt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

prob.motifH5 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&gt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH5)
head(prob.motifH5)
#boxplot(tmm[prob.motifH5[1],]~groupFactor)

prob.motifH6 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&gt;0.5 &amp; gene.prob[ ,7]&lt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH6)
head(prob.motifH6)
#boxplot(tmm[prob.motifH6[1],]~groupFactor)

prob.motifH7 &lt;- rownames(gene.prob[(gene.prob[ ,1]&lt;0.5 &amp; gene.prob[ ,2]&lt;0.5 &amp; gene.prob[ ,3]&lt;0.5 &amp; gene.prob[ ,4]&lt;0.5 &amp;
                                    gene.prob[ ,5]&lt;0.5 &amp; gene.prob[ ,6]&lt;0.5 &amp; gene.prob[ ,7]&gt;0.5 &amp;
                                    gene.prob[ ,8]&lt;0.5 &amp; gene.prob[ ,9]&lt;0.5 &amp; gene.prob[,10]&lt;0.5 &amp; gene.prob[,11]&lt;0.5 &amp;
                                    gene.prob[,12]&lt;0.5 &amp; gene.prob[,13]&lt;0.5 &amp; gene.prob[,14]&lt;0.5),])
length(prob.motifH7)
head(prob.motifH7)
#boxplot(tmm[prob.motifH7[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4,prob.motifH5,prob.motifH6,prob.motifH7)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4,prob.motifC5,prob.motifC6,prob.motifC7)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-ad-hoc-classifications-4-genes-flexible-thresh0" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstAdHoc.Assign.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<div id="each-osteogenic-ad-hoc-classification-separately" class="section level4">
<h4>each osteogenic ad hoc classification separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID &lt;- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classifications-and-8-comparisons" class="section level4">
<h4>5 osteogenic ad hoc classifications and 8 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,
                                         &quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                                       c(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;,
                                         &quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10669 genes) -&gt;  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10669 genes) -&gt;  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (10669 genes) -&gt;  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene8.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-ad-hoc-classifications-4-genes-strict-thresh0" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstAdHoc.Assign.x.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.x.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<div id="each-osteogenic-ad-hoc-classification-separately-1" class="section level4">
<h4>each osteogenic ad hoc classification separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID &lt;- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

prob.motif2  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motif2)
head(prob.motif2)
#boxplot(tmm[prob.motif2[1],]~groupFactor)

prob.motif3  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motif3)
head(prob.motif3)
#boxplot(tmm[prob.motif3[1],]~groupFactor)

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4geneX.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1,prob.motif2,prob.motif3)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classifications-and-8-comparisons-1" class="section level4">
<h4>5 osteogenic ad hoc classifications and 8 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,
                                         &quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                                       c(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;,
                                         &quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9989 genes) -&gt; 99.90%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (   0 genes) -&gt;  0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (   0 genes) -&gt;  0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif
prob.motif1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motif1)
head(prob.motif1)
#boxplot(tmm[prob.motif1[1],]~groupFactor)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4geneX8.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motif1)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-ad-hoc-classifications-4-genes-flexible-thresh0-time-2-only" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0) (Time 2 only)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=(OstAdHoc.Assign.4gene.thresh0==j &amp; Stage==&quot;Time 2&quot;))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<div id="each-osteogenic-ad-hoc-classification-separately-2" class="section level4">
<h4>each osteogenic ad hoc classification separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID &lt;- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classifications-and-8-comparisons-2" class="section level4">
<h4>5 osteogenic ad hoc classifications and 8 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,
                                         &quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                                       c(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;,
                                         &quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10822 genes) -&gt; 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10822 genes) -&gt; 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (10822 genes) -&gt; 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
</div>
</div>
<div id="pseudobulk-species-vs.osteogenic-ad-hoc-classifications-4-genes-strict-thresh0-time-2-only" class="section level3">
<h3>Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0) (Time 2 only)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=(OstAdHoc.Assign.x.4gene.thresh0==j &amp; Stage==&quot;Time 2&quot;))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.x.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
##normalize data
#dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
##log2 transformed cpm
#tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<div id="each-osteogenic-ad-hoc-classification-separately-3" class="section level4">
<h4>each osteogenic ad hoc classification separately</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID &lt;- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&lt;0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp; gene.prob[,5]&gt;0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4geneX.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classifications-and-8-comparisons-3" class="section level4">
<h4>5 osteogenic ad hoc classifications and 8 comparisons</h4>
<pre class="r"><code># Compile group labels of each sample
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))
groupID &lt;- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID &lt;- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData &lt;- cormotiffit.counts(exprs = dge$counts,
                                   groupid = groupID,
                                   compid = compareID,
                                   K = 1:10,
                                   max.iter = 1000,
                                   BIC = FALSE)

#which BIC and AIC are lowest? - 1 motif
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,
                                               &quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           AdHoc=cbind(c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,
                                         &quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                                       c(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;,
                                         &quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob &lt;- cormotifData$bestmotif$p.post
rownames(gene.prob) &lt;- rownames(dge$counts)
colnames(gene.prob) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) &lt;- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) &lt;- c(&quot;H.pre-obl&quot;,&quot;H.obl-emb&quot;,&quot;H.emb-min&quot;,&quot;H.min-mat&quot;,&quot;C.pre-obl&quot;,&quot;C.obl-emb&quot;,&quot;C.emb-min&quot;,&quot;C.min-mat&quot;)
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram=&quot;none&quot;,
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c(&quot;#fef0d9&quot;,&quot;#fdcc8a&quot;,&quot;#fc8d59&quot;,&quot;#e34a33&quot;,&quot;#b30000&quot;),
          margins=c(3,0),
          trace=&quot;none&quot;,
          key.title=&quot;&quot;,
          key.xlab=&quot;posterior prob.&quot;,
          key.ylab=&quot;&quot;,
          #ylab=paste0(length(rownames(gene.prob)),&quot; genes&quot;),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10535 genes) -&gt; 100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10535 genes) -&gt; 100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.2) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.2), ]))/dim(dge$counts)[1]
#0.1 (    0 genes) -&gt;   0%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,2] - gene.prob[,6]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,3] - gene.prob[,7]) &lt;= 0.1) &amp;
                          (abs(gene.prob[,4] - gene.prob[,8]) &lt;= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&gt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&gt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&gt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&gt;0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 &lt;- rownames(gene.prob[(gene.prob[,1]&gt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&gt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&gt;0.5 &amp; gene.prob[,4]&lt;0.5 &amp;
                                    gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  &lt;- rownames(gene.prob[(gene.prob[,1]&lt;0.5 &amp; gene.prob[,2]&lt;0.5 &amp; gene.prob[,3]&lt;0.5 &amp; gene.prob[,4]&gt;0.5 &amp;
                                     gene.prob[,5]&lt;0.5 &amp; gene.prob[,6]&lt;0.5 &amp; gene.prob[,7]&lt;0.5 &amp; gene.prob[,8]&lt;0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)</code></pre>
<pre class="r"><code>saveRDS(cormotifData,&quot;./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4geneX8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)</code></pre>
<pre class="r"><code>rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)</code></pre>
<pre class="r"><code>rm(dge)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ComplexHeatmap_2.0.0        gplots_3.0.1.1             
 [3] Cormotif_1.30.0             affy_1.62.0                
 [5] scran_1.12.1                SingleCellExperiment_1.6.0 
 [7] SummarizedExperiment_1.14.1 DelayedArray_0.10.0        
 [9] BiocParallel_1.18.0         matrixStats_0.57.0         
[11] Biobase_2.44.0              GenomicRanges_1.36.0       
[13] GenomeInfoDb_1.20.0         IRanges_2.18.1             
[15] S4Vectors_0.22.1            BiocGenerics_0.30.0        
[17] edgeR_3.26.5                limma_3.40.6               
[19] RColorBrewer_1.1-2          colorspace_2.0-0           
[21] gbm_2.1.5                   ggplot2_3.3.2              
[23] stringr_1.4.0               stringi_1.4.3              
[25] dplyr_1.0.2                 Seurat_3.1.3               

loaded via a namespace (and not attached):
  [1] circlize_0.4.12          workflowr_1.6.2         
  [3] plyr_1.8.4               igraph_1.2.4.1          
  [5] lazyeval_0.2.2           splines_3.6.1           
  [7] listenv_0.7.0            scater_1.12.2           
  [9] digest_0.6.27            htmltools_0.5.0         
 [11] viridis_0.5.1            gdata_2.18.0            
 [13] magrittr_2.0.1           cluster_2.1.0           
 [15] ROCR_1.0-7               globals_0.12.4          
 [17] RcppParallel_5.0.2       rappdirs_0.3.1          
 [19] ggrepel_0.8.2            xfun_0.8                
 [21] crayon_1.3.4             RCurl_1.98-1.1          
 [23] jsonlite_1.7.1           survival_2.44-1.1       
 [25] zoo_1.8-6                ape_5.4-1               
 [27] glue_1.4.2               gtable_0.3.0            
 [29] zlibbioc_1.30.0          XVector_0.24.0          
 [31] leiden_0.3.1             GetoptLong_1.0.5        
 [33] BiocSingular_1.0.0       shape_1.4.4             
 [35] future.apply_1.3.0       scales_1.1.1            
 [37] bibtex_0.4.2             Rcpp_1.0.5              
 [39] metap_1.1                viridisLite_0.3.0       
 [41] clue_0.3-58              reticulate_1.16         
 [43] dqrng_0.2.1              rsvd_1.0.1              
 [45] preprocessCore_1.46.0    tsne_0.1-3              
 [47] htmlwidgets_1.5.2        httr_1.4.2              
 [49] ellipsis_0.3.1           ica_1.0-2               
 [51] pkgconfig_2.0.3          uwot_0.1.5              
 [53] locfit_1.5-9.1           dynamicTreeCut_1.63-1   
 [55] tidyselect_1.1.0         rlang_0.4.9             
 [57] reshape2_1.4.3           later_1.1.0.1           
 [59] munsell_0.5.0            tools_3.6.1             
 [61] generics_0.1.0           ggridges_0.5.1          
 [63] evaluate_0.14            yaml_2.2.1              
 [65] npsurv_0.4-0             knitr_1.23              
 [67] fs_1.3.1                 fitdistrplus_1.0-14     
 [69] caTools_1.17.1.2         purrr_0.3.4             
 [71] RANN_2.6.1               pbapply_1.4-0           
 [73] future_1.14.0            nlme_3.1-140            
 [75] whisker_0.3-2            compiler_3.6.1          
 [77] beeswarm_0.2.3           plotly_4.9.2.1          
 [79] png_0.1-7                lsei_1.2-0              
 [81] affyio_1.54.0            tibble_3.0.4            
 [83] statmod_1.4.32           lattice_0.20-38         
 [85] Matrix_1.2-18            vctrs_0.3.5             
 [87] pillar_1.4.7             lifecycle_0.2.0         
 [89] BiocManager_1.30.4       GlobalOptions_0.1.2     
 [91] Rdpack_0.11-0            lmtest_0.9-37           
 [93] RcppAnnoy_0.0.12         BiocNeighbors_1.2.0     
 [95] data.table_1.13.2        cowplot_1.1.0           
 [97] bitops_1.0-6             irlba_2.3.3             
 [99] gbRd_0.4-11              httpuv_1.5.1            
[101] R6_2.5.0                 promises_1.1.1          
[103] KernSmooth_2.23-15       gridExtra_2.3           
[105] vipor_0.4.5              codetools_0.2-16        
[107] MASS_7.3-51.4            gtools_3.8.1            
[109] rjson_0.2.20             rprojroot_2.0.2         
[111] withr_2.3.0              sctransform_0.2.0       
[113] GenomeInfoDbData_1.2.1   tidyr_1.1.2             
[115] rmarkdown_1.13           DelayedMatrixStats_1.6.1
[117] Rtsne_0.15               git2r_0.26.1            
[119] ggbeeswarm_0.6.0        </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
