<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-data-variance-tot</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-data-variance-tot</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">June 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.DE-signif.csv
    Untracked:  data/gom-data/topics10.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics15.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics2.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics20.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics30.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics8.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics9.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-data-7-variance-tot.Rmd</code>) and HTML (<code>docs/10x-data-7-variance-tot.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/analysis/10x-data-7-variance-tot.Rmd" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
additional docs updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/b27d03159139d65981b9cce9452c116e67870001/docs/10x-data-7-variance-tot.html" target="_blank">b27d031</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
docs updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/eeeab5a35b24f8f36709823ea3f92e50aa6b9288/analysis/10x-data-7-variance-tot.Rmd" target="_blank">eeeab5a</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
minor fixes to analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e62c53703ed251e27c4d23397131fd218013a635/analysis/10x-data-7-variance-tot.Rmd" target="_blank">e62c537</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-05
</td>
<td>
updated analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---variance-in-gene-expression" class="section level1">
<h1>10X Data Cellranger Analysis - Variance in Gene Expression</h1>
<p>Examine whether variance in gene expression differs across time points in humans and chimpanzees.</p>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(reshape2)</code></pre>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate &lt;- paste0(data@meta.data$Individual,&quot;.&quot;,data@meta.data$Replicate)</code></pre>
<p>Define genes of interest: all except mitochondrial genes and ribosomal genes.</p>
<pre class="r"><code>genes &lt;- rownames(data@assays$RNA@counts)
genes.mito &lt;- c(&quot;MT-ATP6&quot;,&quot;MT-ATP8&quot;,&quot;MT-CO1&quot;,&quot;MT-CO2&quot;,&quot;MT-CO3&quot;,&quot;MT-CYB&quot;,&quot;MT-ND1&quot;,&quot;MT-ND2&quot;,&quot;MT-ND3&quot;,&quot;MT-ND4&quot;,&quot;MT-ND4L&quot;,&quot;MT-ND5&quot;)
genes.ribo &lt;- grep(&#39;^RP&#39;,genes,value=T)
genes.no.mito.ribo &lt;- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)</code></pre>
<p>Decided to proceed with pseudobulk, but also looked as single-cell data.</p>
<div id="examine-variance-in-log2cpm-pseudobulk-resolution" class="section level3">
<h3>Examine variance in log2cpm (pseudobulk resolution)</h3>
<p>Can only look at variance across species, not individuals.</p>
<div id="pseudobulk-stages-vs.species" class="section level4">
<h4>Pseudobulk: Stages vs. Species</h4>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk
cell.subset &lt;- c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg &lt;- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Stage&quot;)
metadata &lt;- as.data.frame(metadata)

#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.stg)

# Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;.&quot;),
                      levels=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human.Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time0&quot;)],1, mean))
HC.mean[&quot;Human.Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time1&quot;)],1, mean))
HC.mean[&quot;Human.Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time2&quot;)],1, mean))
HC.mean[&quot;Chimp.Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time0&quot;)],1, mean))
HC.mean[&quot;Chimp.Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time1&quot;)],1, mean))
HC.mean[&quot;Chimp.Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;[.]&quot;)[[1]][1])
HC.mean.melt$stage &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;[.]&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;Mean Gene Expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-4-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.mean$Human.Time0,HC.mean$Human.Time1) #p=0.4101</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human.Time0 and HC.mean$Human.Time1
t = -0.82378, df = 23139, p-value = 0.4101
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.08731839  0.03564096
sample estimates:
mean of x mean of y 
 4.341204  4.367043 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human.Time1,HC.mean$Human.Time2) #p=0.8676</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human.Time1 and HC.mean$Human.Time2
t = -0.16668, df = 23155, p-value = 0.8676
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06741557  0.05684850
sample estimates:
mean of x mean of y 
 4.367043  4.372327 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp.Time0,HC.mean$Chimp.Time1) #p=0.02706</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp.Time0 and HC.mean$Chimp.Time1
t = -2.2109, df = 23146, p-value = 0.02706
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.133507893 -0.008027503
sample estimates:
mean of x mean of y 
 4.281600  4.352368 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp.Time1,HC.mean$Chimp.Time2) #p=0.9422</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp.Time1 and HC.mean$Chimp.Time2
t = -0.072513, df = 23152, p-value = 0.9422
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06396384  0.05939996
sample estimates:
mean of x mean of y 
 4.352368  4.354650 </code></pre>
<pre class="r"><code># Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human.Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time0&quot;)],1, var))
HC.var[&quot;Human.Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time1&quot;)],1, var))
HC.var[&quot;Human.Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human.Time2&quot;)],1, var))
HC.var[&quot;Chimp.Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time0&quot;)],1, var))
HC.var[&quot;Chimp.Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time1&quot;)],1, var))
HC.var[&quot;Chimp.Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp.Time2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human.Time0&quot;,&quot;Human.Time1&quot;,&quot;Human.Time2&quot;,&quot;Chimp.Time0&quot;,&quot;Chimp.Time1&quot;,&quot;Chimp.Time2&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;[.]&quot;)[[1]][1])
HC.var.melt$stage &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;[.]&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=stage)) +
  geom_boxplot(width=0.8, outlier.shape=NA) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;Log2 Variance in Gene Expression&#39;) +
  ylim(c(-10,5)) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))</code></pre>
<pre><code>Warning: Removed 31 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-4-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.var$Human.Time0,HC.var$Human.Time1) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human.Time0 and HC.var$Human.Time1
t = -19.726, df = 16555, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4249450 -0.3481284
sample estimates:
mean of x mean of y 
0.3356472 0.7221840 </code></pre>
<pre class="r"><code>t.test(HC.var$Human.Time1,HC.var$Human.Time2) #p=1.4e-13</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human.Time1 and HC.var$Human.Time2
t = -7.3991, df = 21431, p-value = 1.422e-13
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2767713 -0.1608435
sample estimates:
mean of x mean of y 
0.7221840 0.9409914 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp.Time0,HC.var$Chimp.Time1) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp.Time0 and HC.var$Chimp.Time1
t = -29.972, df = 12955, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.7375330 -0.6469856
sample estimates:
mean of x mean of y 
0.1769205 0.8691798 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp.Time1,HC.var$Chimp.Time2) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp.Time1 and HC.var$Chimp.Time2
t = -9.8419, df = 22297, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4177135 -0.2789663
sample estimates:
mean of x mean of y 
0.8691798 1.2175197 </code></pre>
</div>
<div id="pseudobulk-general-cluster-vs.species-res0.05" class="section level4">
<h4>Pseudobulk: General Cluster vs. Species (res=0.05)</h4>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk
cell.subset &lt;- c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Cluster0.05==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$Cluster0.05[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)

#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)

# Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;Human_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;Human_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;Human_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;Chimp_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                                    &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.8) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(HC.mean.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.mean$Human_iPSC.c1,HC.mean$Human_MSC.c1)       #p=0.8279</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_iPSC.c1 and HC.mean$Human_MSC.c1
t = -0.21742, df = 21345, p-value = 0.8279
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06527185  0.05223697
sample estimates:
mean of x mean of y 
 4.590393  4.596911 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_MSC.c1,HC.mean$Human_Osteogenic.c1) #p=0.537</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_MSC.c1 and HC.mean$Human_Osteogenic.c1
t = -0.61733, df = 21470, p-value = 0.537
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.07937701  0.04135274
sample estimates:
mean of x mean of y 
 4.596911  4.615923 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_iPSC.c1,HC.mean$Chimp_MSC.c1)       #p=0.1539</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_iPSC.c1 and HC.mean$Chimp_MSC.c1
t = -1.426, df = 21448, p-value = 0.1539
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.10249979  0.01616574
sample estimates:
mean of x mean of y 
 4.559411  4.602578 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_MSC.c1,HC.mean$Chimp_Osteogenic.c1) #p=0.5797</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_MSC.c1 and HC.mean$Chimp_Osteogenic.c1
t = 0.55387, df = 21482, p-value = 0.5797
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.04338882  0.07756885
sample estimates:
mean of x mean of y 
 4.602578  4.585488 </code></pre>
<pre class="r"><code># Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c1&quot;)],1, var))
HC.var[&quot;Human_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c2&quot;)],1, var))
HC.var[&quot;Human_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c3&quot;)],1, var))
HC.var[&quot;Human_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c1&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c2&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c3&quot;)],1, var))
HC.var[&quot;Chimp_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                    id.vars=&quot;gene&quot;,
                    measure.vars=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                                   &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<pre><code>Warning: Removed 191 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-3.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(HC.var.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-4">
Past versions of unnamed-chunk-5-4.png
</button>
</p>
<div id="fig-unnamed-chunk-5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-5-4.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.var$Human_iPSC.c1,HC.var$Human_MSC.c1)       #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_iPSC.c1 and HC.var$Human_MSC.c1
t = -19.212, df = 16190, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3480761 -0.2836279
sample estimates:
mean of x mean of y 
0.2782269 0.5940789 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_MSC.c1,HC.var$Human_Osteogenic.c1) #p=0.3952</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_MSC.c1 and HC.var$Human_Osteogenic.c1
t = -0.85027, df = 21232, p-value = 0.3952
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06134636  0.02422559
sample estimates:
mean of x mean of y 
0.5940789 0.6126393 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_iPSC.c1,HC.var$Chimp_MSC.c1)       #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_iPSC.c1 and HC.var$Chimp_MSC.c1
t = -27.179, df = 13293, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.5631779 -0.4874099
sample estimates:
mean of x mean of y 
0.1460980 0.6713919 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_MSC.c1,HC.var$Chimp_Osteogenic.c1) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_MSC.c1 and HC.var$Chimp_Osteogenic.c1
t = -10.536, df = 20193, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3732340 -0.2561484
sample estimates:
mean of x mean of y 
0.6713919 0.9860831 </code></pre>
</div>
<div id="pseudobulk-general-ad-hoc-classifications-vs.species-thresh0" class="section level4">
<h4>Pseudobulk: General Ad Hoc Classifications vs. Species (thresh0)</h4>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk
cell.subset &lt;- c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=AdHoc.Assign.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- as.character(dataSub@meta.data$AdHoc.Assign.thresh0[w][1])
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)

#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)

# Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = &quot;_&quot;),
                      levels=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Human_other&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;,&quot;Chimp_other&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC&quot;)],1, mean))
HC.mean[&quot;Human_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC&quot;)],1, mean))
HC.mean[&quot;Chimp_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$adhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-6-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.mean$Human_iPSC,HC.mean$Human_MSC)       #p=0.5387</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_iPSC and HC.mean$Human_MSC
t = -0.61473, df = 23585, p-value = 0.5387
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.08135831  0.04251005
sample estimates:
mean of x mean of y 
 4.345648  4.365072 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_MSC,HC.mean$Human_Osteogenic) #p=0.003242</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_MSC and HC.mean$Human_Osteogenic
t = 2.9441, df = 23643, p-value = 0.003242
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.03205075 0.15973692
sample estimates:
mean of x mean of y 
 4.365072  4.269178 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_iPSC,HC.mean$Chimp_MSC)       #p=0.07194</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_iPSC and HC.mean$Chimp_MSC
t = -1.7996, df = 23637, p-value = 0.07194
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.122088890  0.005212236
sample estimates:
mean of x mean of y 
 4.284175  4.342614 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_MSC,HC.mean$Chimp_Osteogenic) #p=0.001382</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_MSC and HC.mean$Chimp_Osteogenic
t = 3.1988, df = 23642, p-value = 0.001382
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.04089803 0.17032919
sample estimates:
mean of x mean of y 
 4.342614  4.237000 </code></pre>
<pre class="r"><code># Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC&quot;)],1, var))
HC.var[&quot;Human_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC&quot;)],1, var))
HC.var[&quot;Human_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic&quot;)],1, var))
HC.var[&quot;Chimp_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC&quot;)],1, var))
HC.var[&quot;Chimp_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$adhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=adhoc)) +
  geom_boxplot(width=0.8, outlier.shape=NA) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10)) +
  ylim(-10,5)</code></pre>
<pre><code>Warning: Removed 43 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-6-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.var$Human_iPSC,HC.var$Human_MSC)       #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_iPSC and HC.var$Human_MSC
t = -18.801, df = 17347, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4036140 -0.3274026
sample estimates:
mean of x mean of y 
0.3464545 0.7119628 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_MSC,HC.var$Human_Osteogenic) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_MSC and HC.var$Human_Osteogenic
t = -9.2531, df = 21620, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3312877 -0.2154687
sample estimates:
mean of x mean of y 
0.7119628 0.9853410 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_iPSC,HC.var$Chimp_MSC)       #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_iPSC and HC.var$Chimp_MSC
t = -29.194, df = 14319, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.7227418 -0.6317947
sample estimates:
mean of x mean of y 
0.1933120 0.8705803 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_MSC,HC.var$Chimp_Osteogenic) #p&lt;2.2e-16</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_MSC and HC.var$Chimp_Osteogenic
t = -11.603, df = 22057, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4945164 -0.3515873
sample estimates:
mean of x mean of y 
0.8705803 1.2936321 </code></pre>
</div>
<div id="pseudobulk-osteogenic-cluster-vs.species-res0.5" class="section level4">
<h4>Pseudobulk: Osteogenic Cluster vs. Species (res=0.5)</h4>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk
cell.subset &lt;- c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Cluster0.50==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt &lt;- dataSub@meta.data$Cluster0.50[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Cluster&quot;)
metadata &lt;- as.data.frame(metadata)

#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)

# Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c3&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c4&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c3&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c4&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                                    &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(HC.mean.melt %&gt;% filter(cluster %in% c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)), aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.mean$Human_Osteogenic.c1,HC.mean$Human_Osteogenic.c2) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_Osteogenic.c1 and HC.mean$Human_Osteogenic.c2
t = 97.435, df = 16339, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 4.039020 4.204864
sample estimates:
mean of x mean of y 
 4.692404  0.570462 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_Osteogenic.c2,HC.mean$Human_Osteogenic.c3) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_Osteogenic.c2 and HC.mean$Human_Osteogenic.c3
t = -102.28, df = 15978, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -4.37330 -4.20883
sample estimates:
mean of x mean of y 
 0.570462  4.861527 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_Osteogenic.c3,HC.mean$Human_Osteogenic.c4) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_Osteogenic.c3 and HC.mean$Human_Osteogenic.c4
t = 14.884, df = 20033, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.4075659 0.5311923
sample estimates:
mean of x mean of y 
 4.861527  4.392148 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_Osteogenic.c1,HC.mean$Chimp_Osteogenic.c2) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_Osteogenic.c1 and HC.mean$Chimp_Osteogenic.c2
t = 86.734, df = 16900, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 3.321569 3.475168
sample estimates:
mean of x mean of y 
 4.652062  1.253694 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_Osteogenic.c2,HC.mean$Chimp_Osteogenic.c3) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_Osteogenic.c2 and HC.mean$Chimp_Osteogenic.c3
t = -92.211, df = 16427, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -3.648568 -3.496684
sample estimates:
mean of x mean of y 
 1.253694  4.826320 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_Osteogenic.c3,HC.mean$Chimp_Osteogenic.c4) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_Osteogenic.c3 and HC.mean$Chimp_Osteogenic.c4
t = 17.231, df = 20016, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.4620796 0.5807028
sample estimates:
mean of x mean of y 
 4.826320  4.304928 </code></pre>
<pre class="r"><code># Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c3&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c4&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c3&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c4&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                                    &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<pre><code>Warning: Removed 1977 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-3">
Past versions of unnamed-chunk-7-3.png
</button>
</p>
<div id="fig-unnamed-chunk-7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-3.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(HC.var.melt %&gt;% filter(cluster %in% c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)), aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<pre><code>Warning: Removed 1977 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-4">
Past versions of unnamed-chunk-7-4.png
</button>
</p>
<div id="fig-unnamed-chunk-7-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-7-4.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.var$Human_Osteogenic.c1,HC.var$Human_Osteogenic.c2) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_Osteogenic.c1 and HC.var$Human_Osteogenic.c2
t = -111, df = 10792, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -27.31962 -26.37151
sample estimates:
 mean of x  mean of y 
 0.6448376 27.4904019 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_Osteogenic.c2,HC.var$Human_Osteogenic.c3) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_Osteogenic.c2 and HC.var$Human_Osteogenic.c3
t = 109.77, df = 10885, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 26.12821 27.07837
sample estimates:
 mean of x  mean of y 
27.4904019  0.8871117 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_Osteogenic.c3,HC.var$Human_Osteogenic.c4) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_Osteogenic.c3 and HC.var$Human_Osteogenic.c4
t = -18.302, df = 20025, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.8043289 -0.6487128
sample estimates:
mean of x mean of y 
0.8871117 1.6136326 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_Osteogenic.c1,HC.var$Chimp_Osteogenic.c2) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_Osteogenic.c1 and HC.var$Chimp_Osteogenic.c2
t = -159.85, df = 11775, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -18.20985 -17.76866
sample estimates:
 mean of x  mean of y 
 0.9532208 18.9424795 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_Osteogenic.c2,HC.var$Chimp_Osteogenic.c3) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_Osteogenic.c2 and HC.var$Chimp_Osteogenic.c3
t = 155.35, df = 12389, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 17.50065 17.94791
sample estimates:
mean of x mean of y 
18.942480  1.218199 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_Osteogenic.c3,HC.var$Chimp_Osteogenic.c4) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_Osteogenic.c3 and HC.var$Chimp_Osteogenic.c4
t = -27.679, df = 18727, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.654915 -1.436030
sample estimates:
mean of x mean of y 
 1.218199  2.763672 </code></pre>
</div>
<div id="pseudobulk-osteogenic-ad-hoc-classifications-vs.species-thresh0" class="section level4">
<h4>Pseudobulk: Osteogenic Ad Hoc Classifications vs. Species (thresh0)</h4>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstAdHoc.Assign.4gene.thresh0==j)
  #dataSub &lt;- subset(data,subset=(OstAdHoc.Assign.4gene.median==j &amp; Stage==&quot;Time 2&quot;))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)

#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)

# Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,&quot;Chimp_other&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_preosteoblast&quot;)],1, mean))
HC.mean[&quot;Human_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_embedding.osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_mineralizing.osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_maturing.osteocyte&quot;)],1, mean))
HC.mean[&quot;Chimp_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_preosteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_embedding.osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_mineralizing.osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_maturing.osteocyte&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                                    &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$ostadhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=ostadhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ost Ad Hoc&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-8-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.mean$Human_preosteoblast,HC.mean$Human_osteoblast)                     </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_preosteoblast and HC.mean$Human_osteoblast
t = 17.049, df = 21101, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.4880080 0.6147928
sample estimates:
mean of x mean of y 
 4.614774  4.063373 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_osteoblast,HC.mean$Human_embedding.osteoblast)              </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_osteoblast and HC.mean$Human_embedding.osteoblast
t = -17.374, df = 21096, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.6249819 -0.4982607
sample estimates:
mean of x mean of y 
 4.063373  4.624994 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_embedding.osteoblast,HC.mean$Human_mineralizing.osteoblast) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_embedding.osteoblast and HC.mean$Human_mineralizing.osteoblast
t = 13.183, df = 20937, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.3693490 0.4983577
sample estimates:
mean of x mean of y 
 4.624994  4.191141 </code></pre>
<pre class="r"><code>t.test(HC.mean$Human_mineralizing.osteoblast,HC.mean$Human_maturing.osteocyte)   </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Human_mineralizing.osteoblast and HC.mean$Human_maturing.osteocyte
t = -3.337, df = 21308, p-value = 0.0008485
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.18939712 -0.04923092
sample estimates:
mean of x mean of y 
 4.191141  4.310455 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_preosteoblast,HC.mean$Chimp_osteoblast)                     </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_preosteoblast and HC.mean$Chimp_osteoblast
t = -32.441, df = 20561, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.1019066 -0.9763385
sample estimates:
mean of x mean of y 
 3.621592  4.660714 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_osteoblast,HC.mean$Chimp_embedding.osteoblast)              </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_osteoblast and HC.mean$Chimp_embedding.osteoblast
t = 2.2339, df = 21313, p-value = 0.0255
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.008005171 0.122617826
sample estimates:
mean of x mean of y 
 4.660714  4.595403 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_embedding.osteoblast,HC.mean$Chimp_mineralizing.osteoblast) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_embedding.osteoblast and HC.mean$Chimp_mineralizing.osteoblast
t = -4.3789, df = 21250, p-value = 1.198e-05
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.1826369 -0.0696909
sample estimates:
mean of x mean of y 
 4.595403  4.721567 </code></pre>
<pre class="r"><code>t.test(HC.mean$Chimp_mineralizing.osteoblast,HC.mean$Chimp_maturing.osteocyte)   </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.mean$Chimp_mineralizing.osteoblast and HC.mean$Chimp_maturing.osteocyte
t = 19.517, df = 19831, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.5750826 0.7034921
sample estimates:
mean of x mean of y 
 4.721567  4.082279 </code></pre>
<pre class="r"><code># Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_preosteoblast&quot;)],1, var))
HC.var[&quot;Human_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_osteoblast&quot;)],1, var))
HC.var[&quot;Human_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_embedding.osteoblast&quot;)],1, var))
HC.var[&quot;Human_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_mineralizing.osteoblast&quot;)],1, var))
HC.var[&quot;Human_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_maturing.osteocyte&quot;)],1, var))
HC.var[&quot;Chimp_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_preosteoblast&quot;)],1, var))
HC.var[&quot;Chimp_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_embedding.osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_mineralizing.osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_maturing.osteocyte&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                    id.vars=&quot;gene&quot;,
                    measure.vars=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                                   &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$ostadhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=ostadhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ost Ad Hoc&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;skyblue&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-data-7-variance-tot.Rmd/unnamed-chunk-8-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(HC.var$Human_preosteoblast,HC.var$Human_osteoblast)                     </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_preosteoblast and HC.var$Human_osteoblast
t = -59.932, df = 17363, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -5.409670 -5.067023
sample estimates:
mean of x mean of y 
 2.141484  7.379831 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_osteoblast,HC.var$Human_embedding.osteoblast)              </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_osteoblast and HC.var$Human_embedding.osteoblast
t = 64.08, df = 16938, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 5.373206 5.712296
sample estimates:
mean of x mean of y 
 7.379831  1.837080 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_embedding.osteoblast,HC.var$Human_mineralizing.osteoblast) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_embedding.osteoblast and HC.var$Human_mineralizing.osteoblast
t = -42.954, df = 17438, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -3.747044 -3.419996
sample estimates:
mean of x mean of y 
  1.83708   5.42060 </code></pre>
<pre class="r"><code>t.test(HC.var$Human_mineralizing.osteoblast,HC.var$Human_maturing.osteocyte)   </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Human_mineralizing.osteoblast and HC.var$Human_maturing.osteocyte
t = 15.939, df = 20863, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 1.319410 1.689425
sample estimates:
mean of x mean of y 
 5.420600  3.916182 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_preosteoblast,HC.var$Chimp_osteoblast)                     </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_preosteoblast and HC.var$Chimp_osteoblast
t = 104.69, df = 14447, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 9.510712 9.873650
sample estimates:
mean of x mean of y 
11.170008  1.477827 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_osteoblast,HC.var$Chimp_embedding.osteoblast)              </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_osteoblast and HC.var$Chimp_embedding.osteoblast
t = -8.6606, df = 20881, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.5924185 -0.3737538
sample estimates:
mean of x mean of y 
 1.477827  1.960913 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_embedding.osteoblast,HC.var$Chimp_mineralizing.osteoblast) </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_embedding.osteoblast and HC.var$Chimp_mineralizing.osteoblast
t = 15.071, df = 19911, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.6958827 0.9039465
sample estimates:
mean of x mean of y 
 1.960913  1.160998 </code></pre>
<pre class="r"><code>t.test(HC.var$Chimp_mineralizing.osteoblast,HC.var$Chimp_maturing.osteocyte)   </code></pre>
<pre><code>
    Welch Two Sample t-test

data:  HC.var$Chimp_mineralizing.osteoblast and HC.var$Chimp_maturing.osteocyte
t = -54.469, df = 14248, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -4.722295 -4.394228
sample estimates:
mean of x mean of y 
 1.160998  5.719260 </code></pre>
</div>
<div id="pseudobulk-osteogenic-ad-hoc-classifications-vs.species-thresh0-in-t2" class="section level4">
<h4>Pseudobulk: Osteogenic Ad Hoc Classifications vs. Species (thresh0 in t2)</h4>
</div>
</div>
<div id="examine-variance-in-log2cpm-single-cell-resolution" class="section level3">
<h3>Examine variance in log2cpm (single-cell resolution)</h3>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object
counts &lt;- as.matrix(GetAssayData(data, assay=&quot;RNA&quot;, slot=&quot;counts&quot;))
metadata &lt;- data@meta.data[,c(&quot;Species&quot;,
                              &quot;Individual&quot;,
                              &quot;Stage&quot;,
                              &quot;Cluster0.05&quot;,
                              &quot;AdHoc.Assign.thresh0&quot;,
                              &quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                              &quot;OstAdHoc.Assign.x.4gene.thresh0&quot;)]
metadata &lt;- metadata[colnames(counts),]
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes (filter genes across full dataset, pch=0.2)
dge$counts &lt;- dge$counts[rowSums(dge$counts!=0)&gt;=(0.2*dim(dge$counts)[2]),]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)</code></pre>
<div id="sc-stages-vs.species" class="section level4">
<h4>SC: Stages vs. Species</h4>
<pre class="r"><code># Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;_&quot;),
                      levels=c(&quot;Human_Time0&quot;,&quot;Human_Time1&quot;,&quot;Human_Time2&quot;,&quot;Chimp_Time0&quot;,&quot;Chimp_Time1&quot;,&quot;Chimp_Time2&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time0&quot;)],1, mean))
HC.mean[&quot;Human_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time1&quot;)],1, mean))
HC.mean[&quot;Human_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time2&quot;)],1, mean))
HC.mean[&quot;Chimp_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time0&quot;)],1, mean))
HC.mean[&quot;Chimp_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time1&quot;)],1, mean))
HC.mean[&quot;Chimp_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_Time0&quot;,&quot;Human_Time1&quot;,&quot;Human_Time2&quot;,&quot;Chimp_Time0&quot;,&quot;Chimp_Time1&quot;,&quot;Chimp_Time2&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$stage &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$Human_Time0,HC.mean$Human_Time1) 
t.test(HC.mean$Human_Time1,HC.mean$Human_Time2) 
t.test(HC.mean$Chimp_Time0,HC.mean$Chimp_Time1) 
t.test(HC.mean$Chimp_Time1,HC.mean$Chimp_Time2) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time0&quot;)],1, var))
HC.var[&quot;Human_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time1&quot;)],1, var))
HC.var[&quot;Human_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Time2&quot;)],1, var))
HC.var[&quot;Chimp_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time0&quot;)],1, var))
HC.var[&quot;Chimp_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time1&quot;)],1, var))
HC.var[&quot;Chimp_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Time2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_Time0&quot;,&quot;Human_Time1&quot;,&quot;Human_Time2&quot;,&quot;Chimp_Time0&quot;,&quot;Chimp_Time1&quot;,&quot;Chimp_Time2&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$stage &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=value, fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$Human_Time0,HC.var$Human_Time1) 
t.test(HC.var$Human_Time1,HC.var$Human_Time2) 
t.test(HC.var$Chimp_Time0,HC.var$Chimp_Time1) 
t.test(HC.var$Chimp_Time1,HC.var$Chimp_Time2) </code></pre>
</div>
<div id="sc-stages-vs.individuals" class="section level4">
<h4>SC: Stages vs. Individuals</h4>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;_&quot;),
                      levels=c(&quot;H1_Time0&quot;,&quot;H1_Time1&quot;,&quot;H1_Time2&quot;,
                               &quot;H2_Time0&quot;,&quot;H2_Time1&quot;,&quot;H2_Time2&quot;,
                               &quot;H3_Time0&quot;,&quot;H3_Time1&quot;,&quot;H3_Time2&quot;,
                               &quot;H4_Time0&quot;,&quot;H4_Time1&quot;,&quot;H4_Time2&quot;,
                               &quot;H5_Time0&quot;,&quot;H5_Time1&quot;,&quot;H5_Time2&quot;,
                               &quot;H6_Time0&quot;,&quot;H6_Time1&quot;,&quot;H6_Time2&quot;,
                               &quot;C1_Time0&quot;,&quot;C1_Time1&quot;,&quot;C1_Time2&quot;,
                               &quot;C2_Time0&quot;,&quot;C2_Time1&quot;,&quot;C2_Time2&quot;,
                               &quot;C3_Time0&quot;,&quot;C3_Time1&quot;,&quot;C3_Time2&quot;,
                               &quot;C4_Time0&quot;,&quot;C4_Time1&quot;,&quot;C4_Time2&quot;,
                               &quot;C5_Time0&quot;,&quot;C5_Time1&quot;,&quot;C5_Time2&quot;,
                               &quot;C6_Time0&quot;,&quot;C6_Time1&quot;,&quot;C6_Time2&quot;))

# Calculate log2cpm means within individuals - generally mean increases in time 1 (except C4, C5, H5) and decreases in time 2
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time0&quot;)],1, mean))
HC.mean[&quot;H1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time1&quot;)],1, mean))
HC.mean[&quot;H1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time2&quot;)],1, mean))
HC.mean[&quot;H2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time0&quot;)],1, mean))
HC.mean[&quot;H2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time1&quot;)],1, mean))
HC.mean[&quot;H2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time2&quot;)],1, mean))
HC.mean[&quot;H3_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time0&quot;)],1, mean))
HC.mean[&quot;H3_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time1&quot;)],1, mean))
HC.mean[&quot;H3_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time2&quot;)],1, mean))
HC.mean[&quot;H4_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time0&quot;)],1, mean))
HC.mean[&quot;H4_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time1&quot;)],1, mean))
HC.mean[&quot;H4_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time2&quot;)],1, mean))
HC.mean[&quot;H5_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time0&quot;)],1, mean))
HC.mean[&quot;H5_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time1&quot;)],1, mean))
HC.mean[&quot;H5_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time2&quot;)],1, mean))
HC.mean[&quot;H6_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time0&quot;)],1, mean))
HC.mean[&quot;H6_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time1&quot;)],1, mean))
HC.mean[&quot;H6_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time2&quot;)],1, mean))
HC.mean[&quot;C1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time0&quot;)],1, mean))
HC.mean[&quot;C1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time1&quot;)],1, mean))
HC.mean[&quot;C1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time2&quot;)],1, mean))
HC.mean[&quot;C2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time0&quot;)],1, mean))
HC.mean[&quot;C2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time1&quot;)],1, mean))
HC.mean[&quot;C2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time2&quot;)],1, mean))
HC.mean[&quot;C3_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time0&quot;)],1, mean))
HC.mean[&quot;C3_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time1&quot;)],1, mean))
HC.mean[&quot;C3_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time2&quot;)],1, mean))
HC.mean[&quot;C4_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time0&quot;)],1, mean))
HC.mean[&quot;C4_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time1&quot;)],1, mean))
HC.mean[&quot;C4_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time2&quot;)],1, mean))
HC.mean[&quot;C5_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time0&quot;)],1, mean))
HC.mean[&quot;C5_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time1&quot;)],1, mean))
HC.mean[&quot;C5_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time2&quot;)],1, mean))
HC.mean[&quot;C6_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time0&quot;)],1, mean))
HC.mean[&quot;C6_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time1&quot;)],1, mean))
HC.mean[&quot;C6_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_Time0&quot;,&quot;H1_Time1&quot;,&quot;H1_Time2&quot;,
                                    &quot;H2_Time0&quot;,&quot;H2_Time1&quot;,&quot;H2_Time2&quot;,
                                    &quot;H3_Time0&quot;,&quot;H3_Time1&quot;,&quot;H3_Time2&quot;,
                                    &quot;H4_Time0&quot;,&quot;H4_Time1&quot;,&quot;H4_Time2&quot;,
                                    &quot;H5_Time0&quot;,&quot;H5_Time1&quot;,&quot;H5_Time2&quot;,
                                    &quot;H6_Time0&quot;,&quot;H6_Time1&quot;,&quot;H6_Time2&quot;,
                                    &quot;C1_Time0&quot;,&quot;C1_Time1&quot;,&quot;C1_Time2&quot;,
                                    &quot;C2_Time0&quot;,&quot;C2_Time1&quot;,&quot;C2_Time2&quot;,
                                    &quot;C3_Time0&quot;,&quot;C3_Time1&quot;,&quot;C3_Time2&quot;,
                                    &quot;C4_Time0&quot;,&quot;C4_Time1&quot;,&quot;C4_Time2&quot;,
                                    &quot;C5_Time0&quot;,&quot;C5_Time1&quot;,&quot;C5_Time2&quot;,
                                    &quot;C6_Time0&quot;,&quot;C6_Time1&quot;,&quot;C6_Time2&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$stage &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1_Time0,HC.mean$H1_Time1)
t.test(HC.mean$H1_Time1,HC.mean$H1_Time2)
t.test(HC.mean$H2_Time0,HC.mean$H2_Time1)
t.test(HC.mean$H2_Time1,HC.mean$H2_Time2)
t.test(HC.mean$H3_Time0,HC.mean$H3_Time1)
t.test(HC.mean$H3_Time1,HC.mean$H3_Time2)
t.test(HC.mean$H4_Time0,HC.mean$H4_Time1)
t.test(HC.mean$H4_Time1,HC.mean$H4_Time2)
t.test(HC.mean$H5_Time0,HC.mean$H5_Time1)
t.test(HC.mean$H5_Time1,HC.mean$H5_Time2)
t.test(HC.mean$H6_Time0,HC.mean$H6_Time1)
t.test(HC.mean$H6_Time1,HC.mean$H6_Time2)
t.test(HC.mean$C1_Time0,HC.mean$C1_Time1)
t.test(HC.mean$C1_Time1,HC.mean$C1_Time2)
t.test(HC.mean$C2_Time0,HC.mean$C2_Time1)
t.test(HC.mean$C2_Time1,HC.mean$C2_Time2)
t.test(HC.mean$C3_Time0,HC.mean$C3_Time1)
t.test(HC.mean$C3_Time1,HC.mean$C3_Time2)
t.test(HC.mean$C4_Time0,HC.mean$C4_Time1)
t.test(HC.mean$C4_Time1,HC.mean$C4_Time2)
t.test(HC.mean$C5_Time0,HC.mean$C5_Time1)
t.test(HC.mean$C5_Time1,HC.mean$C5_Time2)
t.test(HC.mean$C6_Time0,HC.mean$C6_Time1)
t.test(HC.mean$C6_Time1,HC.mean$C6_Time2)

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time0&quot;)],1, var))
HC.var[&quot;H1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time1&quot;)],1, var))
HC.var[&quot;H1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Time2&quot;)],1, var))
HC.var[&quot;H2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time0&quot;)],1, var))
HC.var[&quot;H2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time1&quot;)],1, var))
HC.var[&quot;H2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Time2&quot;)],1, var))
HC.var[&quot;H3_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time0&quot;)],1, var))
HC.var[&quot;H3_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time1&quot;)],1, var))
HC.var[&quot;H3_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Time2&quot;)],1, var))
HC.var[&quot;H4_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time0&quot;)],1, var))
HC.var[&quot;H4_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time1&quot;)],1, var))
HC.var[&quot;H4_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Time2&quot;)],1, var))
HC.var[&quot;H5_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time0&quot;)],1, var))
HC.var[&quot;H5_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time1&quot;)],1, var))
HC.var[&quot;H5_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Time2&quot;)],1, var))
HC.var[&quot;H6_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time0&quot;)],1, var))
HC.var[&quot;H6_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time1&quot;)],1, var))
HC.var[&quot;H6_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Time2&quot;)],1, var))
HC.var[&quot;C1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time0&quot;)],1, var))
HC.var[&quot;C1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time1&quot;)],1, var))
HC.var[&quot;C1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Time2&quot;)],1, var))
HC.var[&quot;C2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time0&quot;)],1, var))
HC.var[&quot;C2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time1&quot;)],1, var))
HC.var[&quot;C2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Time2&quot;)],1, var))
HC.var[&quot;C3_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time0&quot;)],1, var))
HC.var[&quot;C3_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time1&quot;)],1, var))
HC.var[&quot;C3_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Time2&quot;)],1, var))
HC.var[&quot;C4_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time0&quot;)],1, var))
HC.var[&quot;C4_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time1&quot;)],1, var))
HC.var[&quot;C4_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Time2&quot;)],1, var))
HC.var[&quot;C5_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time0&quot;)],1, var))
HC.var[&quot;C5_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time1&quot;)],1, var))
HC.var[&quot;C5_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Time2&quot;)],1, var))
HC.var[&quot;C6_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time0&quot;)],1, var))
HC.var[&quot;C6_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time1&quot;)],1, var))
HC.var[&quot;C6_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Time2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_Time0&quot;,&quot;H1_Time1&quot;,&quot;H1_Time2&quot;,
                                    &quot;H2_Time0&quot;,&quot;H2_Time1&quot;,&quot;H2_Time2&quot;,
                                    &quot;H3_Time0&quot;,&quot;H3_Time1&quot;,&quot;H3_Time2&quot;,
                                    &quot;H4_Time0&quot;,&quot;H4_Time1&quot;,&quot;H4_Time2&quot;,
                                    &quot;H5_Time0&quot;,&quot;H5_Time1&quot;,&quot;H5_Time2&quot;,
                                    &quot;H6_Time0&quot;,&quot;H6_Time1&quot;,&quot;H6_Time2&quot;,
                                    &quot;C1_Time0&quot;,&quot;C1_Time1&quot;,&quot;C1_Time2&quot;,
                                    &quot;C2_Time0&quot;,&quot;C2_Time1&quot;,&quot;C2_Time2&quot;,
                                    &quot;C3_Time0&quot;,&quot;C3_Time1&quot;,&quot;C3_Time2&quot;,
                                    &quot;C4_Time0&quot;,&quot;C4_Time1&quot;,&quot;C4_Time2&quot;,
                                    &quot;C5_Time0&quot;,&quot;C5_Time1&quot;,&quot;C5_Time2&quot;,
                                    &quot;C6_Time0&quot;,&quot;C6_Time1&quot;,&quot;C6_Time2&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$stage &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=log2(value), fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1_Time0,HC.var$H1_Time1)
t.test(HC.var$H1_Time1,HC.var$H1_Time2)
t.test(HC.var$H2_Time0,HC.var$H2_Time1)
t.test(HC.var$H2_Time1,HC.var$H2_Time2)
t.test(HC.var$H3_Time0,HC.var$H3_Time1)
t.test(HC.var$H3_Time1,HC.var$H3_Time2)
t.test(HC.var$H4_Time0,HC.var$H4_Time1)
t.test(HC.var$H4_Time1,HC.var$H4_Time2)
t.test(HC.var$H5_Time0,HC.var$H5_Time1)
t.test(HC.var$H5_Time1,HC.var$H5_Time2)
t.test(HC.var$H6_Time0,HC.var$H6_Time1)
t.test(HC.var$H6_Time1,HC.var$H6_Time2)
t.test(HC.var$C1_Time0,HC.var$C1_Time1)
t.test(HC.var$C1_Time1,HC.var$C1_Time2)
t.test(HC.var$C2_Time0,HC.var$C2_Time1)
t.test(HC.var$C2_Time1,HC.var$C2_Time2)
t.test(HC.var$C3_Time0,HC.var$C3_Time1)
t.test(HC.var$C3_Time1,HC.var$C3_Time2)
t.test(HC.var$C4_Time0,HC.var$C4_Time1)
t.test(HC.var$C4_Time1,HC.var$C4_Time2)
t.test(HC.var$C5_Time0,HC.var$C5_Time1)
t.test(HC.var$C5_Time1,HC.var$C5_Time2)
t.test(HC.var$C6_Time0,HC.var$C6_Time1)
t.test(HC.var$C6_Time1,HC.var$C6_Time2)</code></pre>
</div>
<div id="sc-stages-vs.individual.replicates" class="section level4">
<h4>SC: Stages vs. Individual.Replicates</h4>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual.Replicate,sapply(dge$samples$Stage,function(x)sub(&quot; &quot;,&quot;&quot;,x)),sep = &quot;_&quot;),
                      levels=c(&quot;H1.1_Time0&quot;,&quot;H1.1_Time1&quot;,&quot;H1.1_Time2&quot;,
                               &quot;H1.2_Time0&quot;,&quot;H1.2_Time1&quot;,&quot;H1.2_Time2&quot;,
                               &quot;H2.1_Time0&quot;,&quot;H2.1_Time1&quot;,&quot;H2.1_Time2&quot;,
                               &quot;H3.1_Time0&quot;,&quot;H3.1_Time1&quot;,&quot;H3.1_Time2&quot;,
                               &quot;H4.1_Time0&quot;,&quot;H4.1_Time1&quot;,&quot;H4.1_Time2&quot;,
                               &quot;H5.1_Time0&quot;,&quot;H5.1_Time1&quot;,&quot;H5.1_Time2&quot;,
                               &quot;H6.1_Time0&quot;,&quot;H6.1_Time1&quot;,&quot;H6.1_Time2&quot;,
                               &quot;C1.1_Time0&quot;,&quot;C1.1_Time1&quot;,&quot;C1.1_Time2&quot;,
                               &quot;C1.2_Time0&quot;,&quot;C1.2_Time1&quot;,&quot;C1.2_Time2&quot;,
                               &quot;C2.1_Time0&quot;,&quot;C2.1_Time1&quot;,&quot;C2.1_Time2&quot;,
                               &quot;C3.1_Time0&quot;,&quot;C3.1_Time1&quot;,&quot;C3.1_Time2&quot;,
                               &quot;C4.1_Time0&quot;,&quot;C4.1_Time1&quot;,&quot;C4.1_Time2&quot;,
                               &quot;C5.1_Time0&quot;,&quot;C5.1_Time1&quot;,&quot;C5.1_Time2&quot;,
                               &quot;C6.1_Time0&quot;,&quot;C6.1_Time1&quot;,&quot;C6.1_Time2&quot;))

# Calculate log2cpm means within individuals - generally mean increases in time 1 (except C4.1, C5.1, H5.1) and decreases in time 2
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time0&quot;)],1, mean))
HC.mean[&quot;H1.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time1&quot;)],1, mean))
HC.mean[&quot;H1.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time2&quot;)],1, mean))
HC.mean[&quot;H1.2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time0&quot;)],1, mean))
HC.mean[&quot;H1.2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time1&quot;)],1, mean))
HC.mean[&quot;H1.2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time2&quot;)],1, mean))
HC.mean[&quot;H2.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time0&quot;)],1, mean))
HC.mean[&quot;H2.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time1&quot;)],1, mean))
HC.mean[&quot;H2.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time2&quot;)],1, mean))
HC.mean[&quot;H3.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time0&quot;)],1, mean))
HC.mean[&quot;H3.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time1&quot;)],1, mean))
HC.mean[&quot;H3.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time2&quot;)],1, mean))
HC.mean[&quot;H4.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time0&quot;)],1, mean))
HC.mean[&quot;H4.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time1&quot;)],1, mean))
HC.mean[&quot;H4.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time2&quot;)],1, mean))
HC.mean[&quot;H5.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time0&quot;)],1, mean))
HC.mean[&quot;H5.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time1&quot;)],1, mean))
HC.mean[&quot;H5.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time2&quot;)],1, mean))
HC.mean[&quot;H6.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time0&quot;)],1, mean))
HC.mean[&quot;H6.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time1&quot;)],1, mean))
HC.mean[&quot;H6.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time2&quot;)],1, mean))
HC.mean[&quot;C1.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time0&quot;)],1, mean))
HC.mean[&quot;C1.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time1&quot;)],1, mean))
HC.mean[&quot;C1.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time2&quot;)],1, mean))
HC.mean[&quot;C1.2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time0&quot;)],1, mean))
HC.mean[&quot;C1.2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time1&quot;)],1, mean))
HC.mean[&quot;C1.2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time2&quot;)],1, mean))
HC.mean[&quot;C2.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time0&quot;)],1, mean))
HC.mean[&quot;C2.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time1&quot;)],1, mean))
HC.mean[&quot;C2.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time2&quot;)],1, mean))
HC.mean[&quot;C3.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time0&quot;)],1, mean))
HC.mean[&quot;C3.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time1&quot;)],1, mean))
HC.mean[&quot;C3.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time2&quot;)],1, mean))
HC.mean[&quot;C4.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time0&quot;)],1, mean))
HC.mean[&quot;C4.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time1&quot;)],1, mean))
HC.mean[&quot;C4.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time2&quot;)],1, mean))
HC.mean[&quot;C5.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time0&quot;)],1, mean))
HC.mean[&quot;C5.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time1&quot;)],1, mean))
HC.mean[&quot;C5.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time2&quot;)],1, mean))
HC.mean[&quot;C6.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time0&quot;)],1, mean))
HC.mean[&quot;C6.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time1&quot;)],1, mean))
HC.mean[&quot;C6.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_Time0&quot;,&quot;H1.1_Time1&quot;,&quot;H1.1_Time2&quot;,
                                    &quot;H1.2_Time0&quot;,&quot;H1.2_Time1&quot;,&quot;H1.2_Time2&quot;,
                                    &quot;H2.1_Time0&quot;,&quot;H2.1_Time1&quot;,&quot;H2.1_Time2&quot;,
                                    &quot;H3.1_Time0&quot;,&quot;H3.1_Time1&quot;,&quot;H3.1_Time2&quot;,
                                    &quot;H4.1_Time0&quot;,&quot;H4.1_Time1&quot;,&quot;H4.1_Time2&quot;,
                                    &quot;H5.1_Time0&quot;,&quot;H5.1_Time1&quot;,&quot;H5.1_Time2&quot;,
                                    &quot;H6.1_Time0&quot;,&quot;H6.1_Time1&quot;,&quot;H6.1_Time2&quot;,
                                    &quot;C1.1_Time0&quot;,&quot;C1.1_Time1&quot;,&quot;C1.1_Time2&quot;,
                                    &quot;C1.2_Time0&quot;,&quot;C1.2_Time1&quot;,&quot;C1.2_Time2&quot;,
                                    &quot;C2.1_Time0&quot;,&quot;C2.1_Time1&quot;,&quot;C2.1_Time2&quot;,
                                    &quot;C3.1_Time0&quot;,&quot;C3.1_Time1&quot;,&quot;C3.1_Time2&quot;,
                                    &quot;C4.1_Time0&quot;,&quot;C4.1_Time1&quot;,&quot;C4.1_Time2&quot;,
                                    &quot;C5.1_Time0&quot;,&quot;C5.1_Time1&quot;,&quot;C5.1_Time2&quot;,
                                    &quot;C6.1_Time0&quot;,&quot;C6.1_Time1&quot;,&quot;C6.1_Time2&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$stage &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1.1_Time0,HC.mean$H1.1_Time1)
t.test(HC.mean$H1.1_Time1,HC.mean$H1.1_Time2)
t.test(HC.mean$H1.2_Time0,HC.mean$H1.2_Time1)
t.test(HC.mean$H1.2_Time1,HC.mean$H1.2_Time2)
t.test(HC.mean$H2.1_Time0,HC.mean$H2.1_Time1)
t.test(HC.mean$H2.1_Time1,HC.mean$H2.1_Time2)
t.test(HC.mean$H3.1_Time0,HC.mean$H3.1_Time1)
t.test(HC.mean$H3.1_Time1,HC.mean$H3.1_Time2)
t.test(HC.mean$H4.1_Time0,HC.mean$H4.1_Time1)
t.test(HC.mean$H4.1_Time1,HC.mean$H4.1_Time2)
t.test(HC.mean$H5.1_Time0,HC.mean$H5.1_Time1)
t.test(HC.mean$H5.1_Time1,HC.mean$H5.1_Time2)
t.test(HC.mean$H6.1_Time0,HC.mean$H6.1_Time1)
t.test(HC.mean$H6.1_Time1,HC.mean$H6.1_Time2)
t.test(HC.mean$C1.1_Time0,HC.mean$C1.1_Time1)
t.test(HC.mean$C1.1_Time1,HC.mean$C1.1_Time2)
t.test(HC.mean$C1.2_Time0,HC.mean$C1.2_Time1)
t.test(HC.mean$C1.2_Time1,HC.mean$C1.2_Time2)
t.test(HC.mean$C2.1_Time0,HC.mean$C2.1_Time1)
t.test(HC.mean$C2.1_Time1,HC.mean$C2.1_Time2)
t.test(HC.mean$C3.1_Time0,HC.mean$C3.1_Time1)
t.test(HC.mean$C3.1_Time1,HC.mean$C3.1_Time2)
t.test(HC.mean$C4.1_Time0,HC.mean$C4.1_Time1)
t.test(HC.mean$C4.1_Time1,HC.mean$C4.1_Time2)
t.test(HC.mean$C5.1_Time0,HC.mean$C5.1_Time1)
t.test(HC.mean$C5.1_Time1,HC.mean$C5.1_Time2)
t.test(HC.mean$C6.1_Time0,HC.mean$C6.1_Time1)
t.test(HC.mean$C6.1_Time1,HC.mean$C6.1_Time2)

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time0&quot;)],1, var))
HC.var[&quot;H1.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time1&quot;)],1, var))
HC.var[&quot;H1.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Time2&quot;)],1, var))
HC.var[&quot;H1.2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time0&quot;)],1, var))
HC.var[&quot;H1.2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time1&quot;)],1, var))
HC.var[&quot;H1.2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Time2&quot;)],1, var))
HC.var[&quot;H2.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time0&quot;)],1, var))
HC.var[&quot;H2.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time1&quot;)],1, var))
HC.var[&quot;H2.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Time2&quot;)],1, var))
HC.var[&quot;H3.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time0&quot;)],1, var))
HC.var[&quot;H3.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time1&quot;)],1, var))
HC.var[&quot;H3.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Time2&quot;)],1, var))
HC.var[&quot;H4.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time0&quot;)],1, var))
HC.var[&quot;H4.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time1&quot;)],1, var))
HC.var[&quot;H4.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Time2&quot;)],1, var))
HC.var[&quot;H5.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time0&quot;)],1, var))
HC.var[&quot;H5.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time1&quot;)],1, var))
HC.var[&quot;H5.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Time2&quot;)],1, var))
HC.var[&quot;H6.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time0&quot;)],1, var))
HC.var[&quot;H6.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time1&quot;)],1, var))
HC.var[&quot;H6.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Time2&quot;)],1, var))
HC.var[&quot;C1.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time0&quot;)],1, var))
HC.var[&quot;C1.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time1&quot;)],1, var))
HC.var[&quot;C1.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Time2&quot;)],1, var))
HC.var[&quot;C1.2_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time0&quot;)],1, var))
HC.var[&quot;C1.2_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time1&quot;)],1, var))
HC.var[&quot;C1.2_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Time2&quot;)],1, var))
HC.var[&quot;C2.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time0&quot;)],1, var))
HC.var[&quot;C2.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time1&quot;)],1, var))
HC.var[&quot;C2.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Time2&quot;)],1, var))
HC.var[&quot;C3.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time0&quot;)],1, var))
HC.var[&quot;C3.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time1&quot;)],1, var))
HC.var[&quot;C3.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Time2&quot;)],1, var))
HC.var[&quot;C4.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time0&quot;)],1, var))
HC.var[&quot;C4.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time1&quot;)],1, var))
HC.var[&quot;C4.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Time2&quot;)],1, var))
HC.var[&quot;C5.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time0&quot;)],1, var))
HC.var[&quot;C5.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time1&quot;)],1, var))
HC.var[&quot;C5.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Time2&quot;)],1, var))
HC.var[&quot;C6.1_Time0&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time0&quot;)],1, var))
HC.var[&quot;C6.1_Time1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time1&quot;)],1, var))
HC.var[&quot;C6.1_Time2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Time2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_Time0&quot;,&quot;H1.1_Time1&quot;,&quot;H1.1_Time2&quot;,
                                    &quot;H1.2_Time0&quot;,&quot;H1.2_Time1&quot;,&quot;H1.2_Time2&quot;,
                                    &quot;H2.1_Time0&quot;,&quot;H2.1_Time1&quot;,&quot;H2.1_Time2&quot;,
                                    &quot;H3.1_Time0&quot;,&quot;H3.1_Time1&quot;,&quot;H3.1_Time2&quot;,
                                    &quot;H4.1_Time0&quot;,&quot;H4.1_Time1&quot;,&quot;H4.1_Time2&quot;,
                                    &quot;H5.1_Time0&quot;,&quot;H5.1_Time1&quot;,&quot;H5.1_Time2&quot;,
                                    &quot;H6.1_Time0&quot;,&quot;H6.1_Time1&quot;,&quot;H6.1_Time2&quot;,
                                    &quot;C1.1_Time0&quot;,&quot;C1.1_Time1&quot;,&quot;C1.1_Time2&quot;,
                                    &quot;C1.2_Time0&quot;,&quot;C1.2_Time1&quot;,&quot;C1.2_Time2&quot;,
                                    &quot;C2.1_Time0&quot;,&quot;C2.1_Time1&quot;,&quot;C2.1_Time2&quot;,
                                    &quot;C3.1_Time0&quot;,&quot;C3.1_Time1&quot;,&quot;C3.1_Time2&quot;,
                                    &quot;C4.1_Time0&quot;,&quot;C4.1_Time1&quot;,&quot;C4.1_Time2&quot;,
                                    &quot;C5.1_Time0&quot;,&quot;C5.1_Time1&quot;,&quot;C5.1_Time2&quot;,
                                    &quot;C6.1_Time0&quot;,&quot;C6.1_Time1&quot;,&quot;C6.1_Time2&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$stage &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=log2(value), fill=stage)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;, labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1.1_Time0,HC.var$H1.1_Time1)
t.test(HC.var$H1.1_Time1,HC.var$H1.1_Time2)
t.test(HC.var$H1.2_Time0,HC.var$H1.2_Time1)
t.test(HC.var$H1.2_Time1,HC.var$H1.2_Time2)
t.test(HC.var$H2.1_Time0,HC.var$H2.1_Time1)
t.test(HC.var$H2.1_Time1,HC.var$H2.1_Time2)
t.test(HC.var$H3.1_Time0,HC.var$H3.1_Time1)
t.test(HC.var$H3.1_Time1,HC.var$H3.1_Time2)
t.test(HC.var$H4.1_Time0,HC.var$H4.1_Time1)
t.test(HC.var$H4.1_Time1,HC.var$H4.1_Time2)
t.test(HC.var$H5.1_Time0,HC.var$H5.1_Time1)
t.test(HC.var$H5.1_Time1,HC.var$H5.1_Time2)
t.test(HC.var$H6.1_Time0,HC.var$H6.1_Time1)
t.test(HC.var$H6.1_Time1,HC.var$H6.1_Time2)
t.test(HC.var$C1.1_Time0,HC.var$C1.1_Time1)
t.test(HC.var$C1.1_Time1,HC.var$C1.1_Time2)
t.test(HC.var$C1.2_Time0,HC.var$C1.2_Time1)
t.test(HC.var$C1.2_Time1,HC.var$C1.2_Time2)
t.test(HC.var$C2.1_Time0,HC.var$C2.1_Time1)
t.test(HC.var$C2.1_Time1,HC.var$C2.1_Time2)
t.test(HC.var$C3.1_Time0,HC.var$C3.1_Time1)
t.test(HC.var$C3.1_Time1,HC.var$C3.1_Time2)
t.test(HC.var$C4.1_Time0,HC.var$C4.1_Time1)
t.test(HC.var$C4.1_Time1,HC.var$C4.1_Time2)
t.test(HC.var$C5.1_Time0,HC.var$C5.1_Time1)
t.test(HC.var$C5.1_Time1,HC.var$C5.1_Time2)
t.test(HC.var$C6.1_Time0,HC.var$C6.1_Time1)
t.test(HC.var$C6.1_Time1,HC.var$C6.1_Time2)</code></pre>
</div>
<div id="sc-general-clusters-vs.species-res0.05" class="section level4">
<h4>SC: General Clusters vs. Species (res=0.05)</h4>
<pre class="r"><code># Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                               &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;Human_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;Human_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;Human_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;Chimp_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                                    &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.mean.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$Human_iPSC.c1,HC.mean$Human_MSC.c1)       
t.test(HC.mean$Human_MSC.c1,HC.mean$Human_Osteogenic.c1) 
t.test(HC.mean$Chimp_iPSC.c1,HC.mean$Chimp_MSC.c1)       
t.test(HC.mean$Chimp_MSC.c1,HC.mean$Chimp_Osteogenic.c1) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c1&quot;)],1, var))
HC.var[&quot;Human_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c2&quot;)],1, var))
HC.var[&quot;Human_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC.c3&quot;)],1, var))
HC.var[&quot;Human_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c1&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c2&quot;)],1, var))
HC.var[&quot;Chimp_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC.c3&quot;)],1, var))
HC.var[&quot;Chimp_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                    id.vars=&quot;gene&quot;,
                    measure.vars=c(&quot;Human_iPSC.c1&quot;,&quot;Human_iPSC.c2&quot;,&quot;Human_iPSC.c3&quot;,&quot;Human_MSC.c1&quot;,&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,
                                   &quot;Chimp_iPSC.c1&quot;,&quot;Chimp_iPSC.c2&quot;,&quot;Chimp_iPSC.c3&quot;,&quot;Chimp_MSC.c1&quot;,&quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.var.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=species, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$Human_iPSC.c1,HC.var$Human_MSC.c1)       
t.test(HC.var$Human_MSC.c1,HC.var$Human_Osteogenic.c1) 
t.test(HC.var$Chimp_iPSC.c1,HC.var$Chimp_MSC.c1)       
t.test(HC.var$Chimp_MSC.c1,HC.var$Chimp_Osteogenic.c1) </code></pre>
</div>
</div>
<div id="sc-general-clusters-vs.individuals-res0.05" class="section level3">
<h3>SC: General Clusters vs. Individuals (res=0.05)</h3>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;H1_iPSC.c1&quot;,&quot;H1_iPSC.c2&quot;,&quot;H1_iPSC.c3&quot;,&quot;H1_MSC.c1&quot;,&quot;H1_Osteogenic.c1&quot;,&quot;H1_Osteogenic.c2&quot;,
                               &quot;H2_iPSC.c1&quot;,&quot;H2_iPSC.c2&quot;,&quot;H2_iPSC.c3&quot;,&quot;H2_MSC.c1&quot;,&quot;H2_Osteogenic.c1&quot;,&quot;H2_Osteogenic.c2&quot;,
                               &quot;H3_iPSC.c1&quot;,&quot;H3_iPSC.c2&quot;,&quot;H3_iPSC.c3&quot;,&quot;H3_MSC.c1&quot;,&quot;H3_Osteogenic.c1&quot;,&quot;H3_Osteogenic.c2&quot;,
                               &quot;H4_iPSC.c1&quot;,&quot;H4_iPSC.c2&quot;,&quot;H4_iPSC.c3&quot;,&quot;H4_MSC.c1&quot;,&quot;H4_Osteogenic.c1&quot;,&quot;H4_Osteogenic.c2&quot;,
                               &quot;H5_iPSC.c1&quot;,&quot;H5_iPSC.c2&quot;,&quot;H5_iPSC.c3&quot;,&quot;H5_MSC.c1&quot;,&quot;H5_Osteogenic.c1&quot;,&quot;H5_Osteogenic.c2&quot;,
                               &quot;H6_iPSC.c1&quot;,&quot;H6_iPSC.c2&quot;,&quot;H6_iPSC.c3&quot;,&quot;H6_MSC.c1&quot;,&quot;H6_Osteogenic.c1&quot;,&quot;H6_Osteogenic.c2&quot;,
                               &quot;C1_iPSC.c1&quot;,&quot;C1_iPSC.c2&quot;,&quot;C1_iPSC.c3&quot;,&quot;C1_MSC.c1&quot;,&quot;C1_Osteogenic.c1&quot;,&quot;C1_Osteogenic.c2&quot;,
                               &quot;C2_iPSC.c1&quot;,&quot;C2_iPSC.c2&quot;,&quot;C2_iPSC.c3&quot;,&quot;C2_MSC.c1&quot;,&quot;C2_Osteogenic.c1&quot;,&quot;C2_Osteogenic.c2&quot;,
                               &quot;C3_iPSC.c1&quot;,&quot;C3_iPSC.c2&quot;,&quot;C3_iPSC.c3&quot;,&quot;C3_MSC.c1&quot;,&quot;C3_Osteogenic.c1&quot;,&quot;C3_Osteogenic.c2&quot;,
                               &quot;C4_iPSC.c1&quot;,&quot;C4_iPSC.c2&quot;,&quot;C4_iPSC.c3&quot;,&quot;C4_MSC.c1&quot;,&quot;C4_Osteogenic.c1&quot;,&quot;C4_Osteogenic.c2&quot;,
                               &quot;C5_iPSC.c1&quot;,&quot;C5_iPSC.c2&quot;,&quot;C5_iPSC.c3&quot;,&quot;C5_MSC.c1&quot;,&quot;C5_Osteogenic.c1&quot;,&quot;C5_Osteogenic.c2&quot;,
                               &quot;C6_iPSC.c1&quot;,&quot;C6_iPSC.c2&quot;,&quot;C6_iPSC.c3&quot;,&quot;C6_MSC.c1&quot;,&quot;C6_Osteogenic.c1&quot;,&quot;C6_Osteogenic.c2&quot;))

# Calculate log2cpm means within individuals - generally mean decreases in MSC.c1 (except C1, C3, C6, H1, H3, H4, H6) and decreases in Osteogenic.c1 (except H1)
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H3_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H3_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H3_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H3_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H3_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H3_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H4_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H4_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H4_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H4_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H4_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H4_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H5_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H5_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H5_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H5_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H5_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H5_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H6_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H6_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H6_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H6_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H6_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H6_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C3_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C3_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C3_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C3_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C3_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C3_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C4_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C4_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C4_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C4_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C4_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C4_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C5_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C5_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C5_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C5_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C5_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C5_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C6_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C6_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C6_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C6_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C6_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C6_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic.c2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_iPSC.c1&quot;,&quot;H1_iPSC.c2&quot;,&quot;H1_iPSC.c3&quot;,&quot;H1_MSC.c1&quot;,&quot;H1_Osteogenic.c1&quot;,&quot;H1_Osteogenic.c2&quot;,
                                    &quot;H2_iPSC.c1&quot;,&quot;H2_iPSC.c2&quot;,&quot;H2_iPSC.c3&quot;,&quot;H2_MSC.c1&quot;,&quot;H2_Osteogenic.c1&quot;,&quot;H2_Osteogenic.c2&quot;,
                                    &quot;H3_iPSC.c1&quot;,&quot;H3_iPSC.c2&quot;,&quot;H3_iPSC.c3&quot;,&quot;H3_MSC.c1&quot;,&quot;H3_Osteogenic.c1&quot;,&quot;H3_Osteogenic.c2&quot;,
                                    &quot;H4_iPSC.c1&quot;,&quot;H4_iPSC.c2&quot;,&quot;H4_iPSC.c3&quot;,&quot;H4_MSC.c1&quot;,&quot;H4_Osteogenic.c1&quot;,&quot;H4_Osteogenic.c2&quot;,
                                    &quot;H5_iPSC.c1&quot;,&quot;H5_iPSC.c2&quot;,&quot;H5_iPSC.c3&quot;,&quot;H5_MSC.c1&quot;,&quot;H5_Osteogenic.c1&quot;,&quot;H5_Osteogenic.c2&quot;,
                                    &quot;H6_iPSC.c1&quot;,&quot;H6_iPSC.c2&quot;,&quot;H6_iPSC.c3&quot;,&quot;H6_MSC.c1&quot;,&quot;H6_Osteogenic.c1&quot;,&quot;H6_Osteogenic.c2&quot;,
                                    &quot;C1_iPSC.c1&quot;,&quot;C1_iPSC.c2&quot;,&quot;C1_iPSC.c3&quot;,&quot;C1_MSC.c1&quot;,&quot;C1_Osteogenic.c1&quot;,&quot;C1_Osteogenic.c2&quot;,
                                    &quot;C2_iPSC.c1&quot;,&quot;C2_iPSC.c2&quot;,&quot;C2_iPSC.c3&quot;,&quot;C2_MSC.c1&quot;,&quot;C2_Osteogenic.c1&quot;,&quot;C2_Osteogenic.c2&quot;,
                                    &quot;C3_iPSC.c1&quot;,&quot;C3_iPSC.c2&quot;,&quot;C3_iPSC.c3&quot;,&quot;C3_MSC.c1&quot;,&quot;C3_Osteogenic.c1&quot;,&quot;C3_Osteogenic.c2&quot;,
                                    &quot;C4_iPSC.c1&quot;,&quot;C4_iPSC.c2&quot;,&quot;C4_iPSC.c3&quot;,&quot;C4_MSC.c1&quot;,&quot;C4_Osteogenic.c1&quot;,&quot;C4_Osteogenic.c2&quot;,
                                    &quot;C5_iPSC.c1&quot;,&quot;C5_iPSC.c2&quot;,&quot;C5_iPSC.c3&quot;,&quot;C5_MSC.c1&quot;,&quot;C5_Osteogenic.c1&quot;,&quot;C5_Osteogenic.c2&quot;,
                                    &quot;C6_iPSC.c1&quot;,&quot;C6_iPSC.c2&quot;,&quot;C6_iPSC.c3&quot;,&quot;C6_MSC.c1&quot;,&quot;C6_Osteogenic.c1&quot;,&quot;C6_Osteogenic.c2&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.mean.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=individual, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1_iPSC.c1,HC.mean$H1_MSC.c1)       
t.test(HC.mean$H1_MSC.c1,HC.mean$H1_Osteogenic.c1) 
t.test(HC.mean$H2_iPSC.c1,HC.mean$H2_MSC.c1)       
t.test(HC.mean$H2_MSC.c1,HC.mean$H2_Osteogenic.c1) 
t.test(HC.mean$H3_iPSC.c1,HC.mean$H3_MSC.c1)       
t.test(HC.mean$H3_MSC.c1,HC.mean$H3_Osteogenic.c1) 
t.test(HC.mean$H4_iPSC.c1,HC.mean$H4_MSC.c1)       
t.test(HC.mean$H4_MSC.c1,HC.mean$H4_Osteogenic.c1) 
t.test(HC.mean$H5_iPSC.c1,HC.mean$H5_MSC.c1)       
t.test(HC.mean$H5_MSC.c1,HC.mean$H5_Osteogenic.c1) 
t.test(HC.mean$H6_iPSC.c1,HC.mean$H6_MSC.c1)       
t.test(HC.mean$H6_MSC.c1,HC.mean$H6_Osteogenic.c1) 
t.test(HC.mean$C1_iPSC.c1,HC.mean$C1_MSC.c1)       
t.test(HC.mean$C1_MSC.c1,HC.mean$C1_Osteogenic.c1) 
t.test(HC.mean$C2_iPSC.c1,HC.mean$C2_MSC.c1)       
t.test(HC.mean$C2_MSC.c1,HC.mean$C2_Osteogenic.c1) 
t.test(HC.mean$C3_iPSC.c1,HC.mean$C3_MSC.c1)       
t.test(HC.mean$C3_MSC.c1,HC.mean$C3_Osteogenic.c1) 
t.test(HC.mean$C4_iPSC.c1,HC.mean$C4_MSC.c1)       
t.test(HC.mean$C4_MSC.c1,HC.mean$C4_Osteogenic.c1) 
t.test(HC.mean$C5_iPSC.c1,HC.mean$C5_MSC.c1)       
t.test(HC.mean$C5_MSC.c1,HC.mean$C5_Osteogenic.c1) 
t.test(HC.mean$C6_iPSC.c1,HC.mean$C6_MSC.c1)       
t.test(HC.mean$C6_MSC.c1,HC.mean$C6_Osteogenic.c1) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_MSC.c1&quot;)],1, var))
HC.var[&quot;H1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c1&quot;)],1, var))
HC.var[&quot;H2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c2&quot;)],1, var))
HC.var[&quot;H2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC.c3&quot;)],1, var))
HC.var[&quot;H2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_MSC.c1&quot;)],1, var))
HC.var[&quot;H2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H3_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c1&quot;)],1, var))
HC.var[&quot;H3_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c2&quot;)],1, var))
HC.var[&quot;H3_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC.c3&quot;)],1, var))
HC.var[&quot;H3_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_MSC.c1&quot;)],1, var))
HC.var[&quot;H3_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H3_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H4_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c1&quot;)],1, var))
HC.var[&quot;H4_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c2&quot;)],1, var))
HC.var[&quot;H4_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC.c3&quot;)],1, var))
HC.var[&quot;H4_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_MSC.c1&quot;)],1, var))
HC.var[&quot;H4_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H4_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H5_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c1&quot;)],1, var))
HC.var[&quot;H5_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c2&quot;)],1, var))
HC.var[&quot;H5_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC.c3&quot;)],1, var))
HC.var[&quot;H5_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_MSC.c1&quot;)],1, var))
HC.var[&quot;H5_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H5_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H6_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c1&quot;)],1, var))
HC.var[&quot;H6_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c2&quot;)],1, var))
HC.var[&quot;H6_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC.c3&quot;)],1, var))
HC.var[&quot;H6_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_MSC.c1&quot;)],1, var))
HC.var[&quot;H6_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H6_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_MSC.c1&quot;)],1, var))
HC.var[&quot;C1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c1&quot;)],1, var))
HC.var[&quot;C2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c2&quot;)],1, var))
HC.var[&quot;C2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC.c3&quot;)],1, var))
HC.var[&quot;C2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_MSC.c1&quot;)],1, var))
HC.var[&quot;C2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C3_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c1&quot;)],1, var))
HC.var[&quot;C3_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c2&quot;)],1, var))
HC.var[&quot;C3_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC.c3&quot;)],1, var))
HC.var[&quot;C3_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_MSC.c1&quot;)],1, var))
HC.var[&quot;C3_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C3_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C4_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c1&quot;)],1, var))
HC.var[&quot;C4_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c2&quot;)],1, var))
HC.var[&quot;C4_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC.c3&quot;)],1, var))
HC.var[&quot;C4_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_MSC.c1&quot;)],1, var))
HC.var[&quot;C4_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C4_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C5_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c1&quot;)],1, var))
HC.var[&quot;C5_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c2&quot;)],1, var))
HC.var[&quot;C5_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC.c3&quot;)],1, var))
HC.var[&quot;C5_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_MSC.c1&quot;)],1, var))
HC.var[&quot;C5_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C5_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C6_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c1&quot;)],1, var))
HC.var[&quot;C6_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c2&quot;)],1, var))
HC.var[&quot;C6_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC.c3&quot;)],1, var))
HC.var[&quot;C6_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_MSC.c1&quot;)],1, var))
HC.var[&quot;C6_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C6_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic.c2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_iPSC.c1&quot;,&quot;H1_iPSC.c2&quot;,&quot;H1_iPSC.c3&quot;,&quot;H1_MSC.c1&quot;,&quot;H1_Osteogenic.c1&quot;,&quot;H1_Osteogenic.c2&quot;,
                                    &quot;H2_iPSC.c1&quot;,&quot;H2_iPSC.c2&quot;,&quot;H2_iPSC.c3&quot;,&quot;H2_MSC.c1&quot;,&quot;H2_Osteogenic.c1&quot;,&quot;H2_Osteogenic.c2&quot;,
                                    &quot;H3_iPSC.c1&quot;,&quot;H3_iPSC.c2&quot;,&quot;H3_iPSC.c3&quot;,&quot;H3_MSC.c1&quot;,&quot;H3_Osteogenic.c1&quot;,&quot;H3_Osteogenic.c2&quot;,
                                    &quot;H4_iPSC.c1&quot;,&quot;H4_iPSC.c2&quot;,&quot;H4_iPSC.c3&quot;,&quot;H4_MSC.c1&quot;,&quot;H4_Osteogenic.c1&quot;,&quot;H4_Osteogenic.c2&quot;,
                                    &quot;H5_iPSC.c1&quot;,&quot;H5_iPSC.c2&quot;,&quot;H5_iPSC.c3&quot;,&quot;H5_MSC.c1&quot;,&quot;H5_Osteogenic.c1&quot;,&quot;H5_Osteogenic.c2&quot;,
                                    &quot;H6_iPSC.c1&quot;,&quot;H6_iPSC.c2&quot;,&quot;H6_iPSC.c3&quot;,&quot;H6_MSC.c1&quot;,&quot;H6_Osteogenic.c1&quot;,&quot;H6_Osteogenic.c2&quot;,
                                    &quot;C1_iPSC.c1&quot;,&quot;C1_iPSC.c2&quot;,&quot;C1_iPSC.c3&quot;,&quot;C1_MSC.c1&quot;,&quot;C1_Osteogenic.c1&quot;,&quot;C1_Osteogenic.c2&quot;,
                                    &quot;C2_iPSC.c1&quot;,&quot;C2_iPSC.c2&quot;,&quot;C2_iPSC.c3&quot;,&quot;C2_MSC.c1&quot;,&quot;C2_Osteogenic.c1&quot;,&quot;C2_Osteogenic.c2&quot;,
                                    &quot;C3_iPSC.c1&quot;,&quot;C3_iPSC.c2&quot;,&quot;C3_iPSC.c3&quot;,&quot;C3_MSC.c1&quot;,&quot;C3_Osteogenic.c1&quot;,&quot;C3_Osteogenic.c2&quot;,
                                    &quot;C4_iPSC.c1&quot;,&quot;C4_iPSC.c2&quot;,&quot;C4_iPSC.c3&quot;,&quot;C4_MSC.c1&quot;,&quot;C4_Osteogenic.c1&quot;,&quot;C4_Osteogenic.c2&quot;,
                                    &quot;C5_iPSC.c1&quot;,&quot;C5_iPSC.c2&quot;,&quot;C5_iPSC.c3&quot;,&quot;C5_MSC.c1&quot;,&quot;C5_Osteogenic.c1&quot;,&quot;C5_Osteogenic.c2&quot;,
                                    &quot;C6_iPSC.c1&quot;,&quot;C6_iPSC.c2&quot;,&quot;C6_iPSC.c3&quot;,&quot;C6_MSC.c1&quot;,&quot;C6_Osteogenic.c1&quot;,&quot;C6_Osteogenic.c2&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.var.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=individual, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1_iPSC.c1,HC.var$H1_MSC.c1)       
t.test(HC.var$H1_MSC.c1,HC.var$H1_Osteogenic.c1) 
t.test(HC.var$H2_iPSC.c1,HC.var$H2_MSC.c1)       
t.test(HC.var$H2_MSC.c1,HC.var$H2_Osteogenic.c1) 
t.test(HC.var$H3_iPSC.c1,HC.var$H3_MSC.c1)       
t.test(HC.var$H3_MSC.c1,HC.var$H3_Osteogenic.c1) 
t.test(HC.var$H4_iPSC.c1,HC.var$H4_MSC.c1)       
t.test(HC.var$H4_MSC.c1,HC.var$H4_Osteogenic.c1) 
t.test(HC.var$H5_iPSC.c1,HC.var$H5_MSC.c1)       
t.test(HC.var$H5_MSC.c1,HC.var$H5_Osteogenic.c1) 
t.test(HC.var$H6_iPSC.c1,HC.var$H6_MSC.c1)       
t.test(HC.var$H6_MSC.c1,HC.var$H6_Osteogenic.c1) 
t.test(HC.var$C1_iPSC.c1,HC.var$C1_MSC.c1)       
t.test(HC.var$C1_MSC.c1,HC.var$C1_Osteogenic.c1) 
t.test(HC.var$C2_iPSC.c1,HC.var$C2_MSC.c1)       
t.test(HC.var$C2_MSC.c1,HC.var$C2_Osteogenic.c1) 
t.test(HC.var$C3_iPSC.c1,HC.var$C3_MSC.c1)       
t.test(HC.var$C3_MSC.c1,HC.var$C3_Osteogenic.c1) 
t.test(HC.var$C4_iPSC.c1,HC.var$C4_MSC.c1)       
t.test(HC.var$C4_MSC.c1,HC.var$C4_Osteogenic.c1) 
t.test(HC.var$C5_iPSC.c1,HC.var$C5_MSC.c1)       
t.test(HC.var$C5_MSC.c1,HC.var$C5_Osteogenic.c1) 
t.test(HC.var$C6_iPSC.c1,HC.var$C6_MSC.c1)       
t.test(HC.var$C6_MSC.c1,HC.var$C6_Osteogenic.c1) </code></pre>
<div id="sc-general-clusters-vs.individual.replicates-res0.05" class="section level4">
<h4>SC: General Clusters vs. Individual.Replicates (res=0.05)</h4>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual.Replicate,dge$samples$Cluster,sep=&quot;_&quot;),
                      levels=c(&quot;H1.1_iPSC.c1&quot;,&quot;H1.1_iPSC.c2&quot;,&quot;H1.1_iPSC.c3&quot;,&quot;H1.1_MSC.c1&quot;,&quot;H1.1_Osteogenic.c1&quot;,&quot;H1.1_Osteogenic.c2&quot;,
                               &quot;H1.2_iPSC.c1&quot;,&quot;H1.2_iPSC.c2&quot;,&quot;H1.2_iPSC.c3&quot;,&quot;H1.2_MSC.c1&quot;,&quot;H1.2_Osteogenic.c1&quot;,&quot;H1.2_Osteogenic.c2&quot;,
                               &quot;H2.1_iPSC.c1&quot;,&quot;H2.1_iPSC.c2&quot;,&quot;H2.1_iPSC.c3&quot;,&quot;H2.1_MSC.c1&quot;,&quot;H2.1_Osteogenic.c1&quot;,&quot;H2.1_Osteogenic.c2&quot;,
                               &quot;H3.1_iPSC.c1&quot;,&quot;H3.1_iPSC.c2&quot;,&quot;H3.1_iPSC.c3&quot;,&quot;H3.1_MSC.c1&quot;,&quot;H3.1_Osteogenic.c1&quot;,&quot;H3.1_Osteogenic.c2&quot;,
                               &quot;H4.1_iPSC.c1&quot;,&quot;H4.1_iPSC.c2&quot;,&quot;H4.1_iPSC.c3&quot;,&quot;H4.1_MSC.c1&quot;,&quot;H4.1_Osteogenic.c1&quot;,&quot;H4.1_Osteogenic.c2&quot;,
                               &quot;H5.1_iPSC.c1&quot;,&quot;H5.1_iPSC.c2&quot;,&quot;H5.1_iPSC.c3&quot;,&quot;H5.1_MSC.c1&quot;,&quot;H5.1_Osteogenic.c1&quot;,&quot;H5.1_Osteogenic.c2&quot;,
                               &quot;H6.1_iPSC.c1&quot;,&quot;H6.1_iPSC.c2&quot;,&quot;H6.1_iPSC.c3&quot;,&quot;H6.1_MSC.c1&quot;,&quot;H6.1_Osteogenic.c1&quot;,&quot;H6.1_Osteogenic.c2&quot;,
                               &quot;C1.1_iPSC.c1&quot;,&quot;C1.1_iPSC.c2&quot;,&quot;C1.1_iPSC.c3&quot;,&quot;C1.1_MSC.c1&quot;,&quot;C1.1_Osteogenic.c1&quot;,&quot;C1.1_Osteogenic.c2&quot;,
                               &quot;C1.2_iPSC.c1&quot;,&quot;C1.2_iPSC.c2&quot;,&quot;C1.2_iPSC.c3&quot;,&quot;C1.2_MSC.c1&quot;,&quot;C1.2_Osteogenic.c1&quot;,&quot;C1.2_Osteogenic.c2&quot;,
                               &quot;C2.1_iPSC.c1&quot;,&quot;C2.1_iPSC.c2&quot;,&quot;C2.1_iPSC.c3&quot;,&quot;C2.1_MSC.c1&quot;,&quot;C2.1_Osteogenic.c1&quot;,&quot;C2.1_Osteogenic.c2&quot;,
                               &quot;C3.1_iPSC.c1&quot;,&quot;C3.1_iPSC.c2&quot;,&quot;C3.1_iPSC.c3&quot;,&quot;C3.1_MSC.c1&quot;,&quot;C3.1_Osteogenic.c1&quot;,&quot;C3.1_Osteogenic.c2&quot;,
                               &quot;C4.1_iPSC.c1&quot;,&quot;C4.1_iPSC.c2&quot;,&quot;C4.1_iPSC.c3&quot;,&quot;C4.1_MSC.c1&quot;,&quot;C4.1_Osteogenic.c1&quot;,&quot;C4.1_Osteogenic.c2&quot;,
                               &quot;C5.1_iPSC.c1&quot;,&quot;C5.1_iPSC.c2&quot;,&quot;C5.1_iPSC.c3&quot;,&quot;C5.1_MSC.c1&quot;,&quot;C5.1_Osteogenic.c1&quot;,&quot;C5.1_Osteogenic.c2&quot;,
                               &quot;C6.1_iPSC.c1&quot;,&quot;C6.1_iPSC.c2&quot;,&quot;C6.1_iPSC.c3&quot;,&quot;C6.1_MSC.c1&quot;,&quot;C6.1_Osteogenic.c1&quot;,&quot;C6.1_Osteogenic.c2&quot;))

# Calculate log2cpm means within individuals - generally mean decreases in MSC.c1 (except C1.1, C3.1, C6.1, H1.1, H1.2, H3.1, H4.1, H6.1) and decreases in Osteogenic.c1 (except C1.1, H1.1, H1.2)
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H1.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H1.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H1.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H1.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H1.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H1.2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H1.2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H1.2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H1.2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H1.2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H1.2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H2.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H2.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H2.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H2.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H2.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H2.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H3.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H3.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H3.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H3.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H3.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H3.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H4.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H4.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H4.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H4.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H4.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H4.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H5.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H5.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H5.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H5.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H5.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H5.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;H6.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;H6.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;H6.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;H6.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;H6.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;H6.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C1.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C1.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C1.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C1.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C1.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C1.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C1.2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C1.2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C1.2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C1.2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C1.2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C1.2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C2.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C2.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C2.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C2.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C2.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C2.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C3.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C3.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C3.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C3.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C3.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C3.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C4.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C4.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C4.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C4.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C4.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C4.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C5.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C5.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C5.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C5.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C5.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C5.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;C6.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c1&quot;)],1, mean))
HC.mean[&quot;C6.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c2&quot;)],1, mean))
HC.mean[&quot;C6.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c3&quot;)],1, mean))
HC.mean[&quot;C6.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_MSC.c1&quot;)],1, mean))
HC.mean[&quot;C6.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;C6.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic.c2&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_iPSC.c1&quot;,&quot;H1.1_iPSC.c2&quot;,&quot;H1.1_iPSC.c3&quot;,&quot;H1.1_MSC.c1&quot;,&quot;H1.1_Osteogenic.c1&quot;,&quot;H1.1_Osteogenic.c2&quot;,
                                    &quot;H1.2_iPSC.c1&quot;,&quot;H1.2_iPSC.c2&quot;,&quot;H1.2_iPSC.c3&quot;,&quot;H1.2_MSC.c1&quot;,&quot;H1.2_Osteogenic.c1&quot;,&quot;H1.2_Osteogenic.c2&quot;,
                                    &quot;H2.1_iPSC.c1&quot;,&quot;H2.1_iPSC.c2&quot;,&quot;H2.1_iPSC.c3&quot;,&quot;H2.1_MSC.c1&quot;,&quot;H2.1_Osteogenic.c1&quot;,&quot;H2.1_Osteogenic.c2&quot;,
                                    &quot;H3.1_iPSC.c1&quot;,&quot;H3.1_iPSC.c2&quot;,&quot;H3.1_iPSC.c3&quot;,&quot;H3.1_MSC.c1&quot;,&quot;H3.1_Osteogenic.c1&quot;,&quot;H3.1_Osteogenic.c2&quot;,
                                    &quot;H4.1_iPSC.c1&quot;,&quot;H4.1_iPSC.c2&quot;,&quot;H4.1_iPSC.c3&quot;,&quot;H4.1_MSC.c1&quot;,&quot;H4.1_Osteogenic.c1&quot;,&quot;H4.1_Osteogenic.c2&quot;,
                                    &quot;H5.1_iPSC.c1&quot;,&quot;H5.1_iPSC.c2&quot;,&quot;H5.1_iPSC.c3&quot;,&quot;H5.1_MSC.c1&quot;,&quot;H5.1_Osteogenic.c1&quot;,&quot;H5.1_Osteogenic.c2&quot;,
                                    &quot;H6.1_iPSC.c1&quot;,&quot;H6.1_iPSC.c2&quot;,&quot;H6.1_iPSC.c3&quot;,&quot;H6.1_MSC.c1&quot;,&quot;H6.1_Osteogenic.c1&quot;,&quot;H6.1_Osteogenic.c2&quot;,
                                    &quot;C1.1_iPSC.c1&quot;,&quot;C1.1_iPSC.c2&quot;,&quot;C1.1_iPSC.c3&quot;,&quot;C1.1_MSC.c1&quot;,&quot;C1.1_Osteogenic.c1&quot;,&quot;C1.1_Osteogenic.c2&quot;,
                                    &quot;C1.2_iPSC.c1&quot;,&quot;C1.2_iPSC.c2&quot;,&quot;C1.2_iPSC.c3&quot;,&quot;C1.2_MSC.c1&quot;,&quot;C1.2_Osteogenic.c1&quot;,&quot;C1.2_Osteogenic.c2&quot;,
                                    &quot;C2.1_iPSC.c1&quot;,&quot;C2.1_iPSC.c2&quot;,&quot;C2.1_iPSC.c3&quot;,&quot;C2.1_MSC.c1&quot;,&quot;C2.1_Osteogenic.c1&quot;,&quot;C2.1_Osteogenic.c2&quot;,
                                    &quot;C3.1_iPSC.c1&quot;,&quot;C3.1_iPSC.c2&quot;,&quot;C3.1_iPSC.c3&quot;,&quot;C3.1_MSC.c1&quot;,&quot;C3.1_Osteogenic.c1&quot;,&quot;C3.1_Osteogenic.c2&quot;,
                                    &quot;C4.1_iPSC.c1&quot;,&quot;C4.1_iPSC.c2&quot;,&quot;C4.1_iPSC.c3&quot;,&quot;C4.1_MSC.c1&quot;,&quot;C4.1_Osteogenic.c1&quot;,&quot;C4.1_Osteogenic.c2&quot;,
                                    &quot;C5.1_iPSC.c1&quot;,&quot;C5.1_iPSC.c2&quot;,&quot;C5.1_iPSC.c3&quot;,&quot;C5.1_MSC.c1&quot;,&quot;C5.1_Osteogenic.c1&quot;,&quot;C5.1_Osteogenic.c2&quot;,
                                    &quot;C6.1_iPSC.c1&quot;,&quot;C6.1_iPSC.c2&quot;,&quot;C6.1_iPSC.c3&quot;,&quot;C6.1_MSC.c1&quot;,&quot;C6.1_Osteogenic.c1&quot;,&quot;C6.1_Osteogenic.c2&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.mean.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=individual, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1.1_iPSC.c1,HC.mean$H1.1_MSC.c1)       
t.test(HC.mean$H1.1_MSC.c1,HC.mean$H1.1_Osteogenic.c1) 
t.test(HC.mean$H1.2_iPSC.c1,HC.mean$H1.2_MSC.c1)       
t.test(HC.mean$H1.2_MSC.c1,HC.mean$H1.2_Osteogenic.c1) 
t.test(HC.mean$H2.1_iPSC.c1,HC.mean$H2.1_MSC.c1)       
t.test(HC.mean$H2.1_MSC.c1,HC.mean$H2.1_Osteogenic.c1) 
t.test(HC.mean$H3.1_iPSC.c1,HC.mean$H3.1_MSC.c1)       
t.test(HC.mean$H3.1_MSC.c1,HC.mean$H3.1_Osteogenic.c1) 
t.test(HC.mean$H4.1_iPSC.c1,HC.mean$H4.1_MSC.c1)       
t.test(HC.mean$H4.1_MSC.c1,HC.mean$H4.1_Osteogenic.c1) 
t.test(HC.mean$H5.1_iPSC.c1,HC.mean$H5.1_MSC.c1)       
t.test(HC.mean$H5.1_MSC.c1,HC.mean$H5.1_Osteogenic.c1) 
t.test(HC.mean$H6.1_iPSC.c1,HC.mean$H6.1_MSC.c1)       
t.test(HC.mean$H6.1_MSC.c1,HC.mean$H6.1_Osteogenic.c1) 
t.test(HC.mean$C1.1_iPSC.c1,HC.mean$C1.1_MSC.c1)       
t.test(HC.mean$C1.1_MSC.c1,HC.mean$C1.1_Osteogenic.c1) 
t.test(HC.mean$C1.2_iPSC.c1,HC.mean$C1.2_MSC.c1)       
t.test(HC.mean$C1.2_MSC.c1,HC.mean$C1.2_Osteogenic.c1) 
t.test(HC.mean$C2.1_iPSC.c1,HC.mean$C2.1_MSC.c1)       
t.test(HC.mean$C2.1_MSC.c1,HC.mean$C2.1_Osteogenic.c1) 
t.test(HC.mean$C3.1_iPSC.c1,HC.mean$C3.1_MSC.c1)       
t.test(HC.mean$C3.1_MSC.c1,HC.mean$C3.1_Osteogenic.c1) 
t.test(HC.mean$C4.1_iPSC.c1,HC.mean$C4.1_MSC.c1)       
t.test(HC.mean$C4.1_MSC.c1,HC.mean$C4.1_Osteogenic.c1) 
t.test(HC.mean$C5.1_iPSC.c1,HC.mean$C5.1_MSC.c1)       
t.test(HC.mean$C5.1_MSC.c1,HC.mean$C5.1_Osteogenic.c1) 
t.test(HC.mean$C6.1_iPSC.c1,HC.mean$C6.1_MSC.c1)       
t.test(HC.mean$C6.1_MSC.c1,HC.mean$C6.1_Osteogenic.c1) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H1.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H1.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H1.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H1.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H1.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H1.2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c1&quot;)],1, var))
HC.var[&quot;H1.2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c2&quot;)],1, var))
HC.var[&quot;H1.2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC.c3&quot;)],1, var))
HC.var[&quot;H1.2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_MSC.c1&quot;)],1, var))
HC.var[&quot;H1.2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H1.2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H2.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H2.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H2.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H2.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H2.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H2.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H3.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H3.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H3.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H3.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H3.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H3.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H4.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H4.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H4.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H4.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H4.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H4.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H5.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H5.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H5.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H5.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H5.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H5.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;H6.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;H6.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;H6.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;H6.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_MSC.c1&quot;)],1, var))
HC.var[&quot;H6.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;H6.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C1.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C1.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C1.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C1.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C1.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C1.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C1.2_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c1&quot;)],1, var))
HC.var[&quot;C1.2_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c2&quot;)],1, var))
HC.var[&quot;C1.2_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC.c3&quot;)],1, var))
HC.var[&quot;C1.2_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_MSC.c1&quot;)],1, var))
HC.var[&quot;C1.2_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C1.2_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C2.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C2.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C2.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C2.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C2.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C2.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C3.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C3.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C3.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C3.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C3.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C3.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C4.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C4.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C4.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C4.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C4.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C4.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C5.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C5.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C5.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C5.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C5.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C5.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;C6.1_iPSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c1&quot;)],1, var))
HC.var[&quot;C6.1_iPSC.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c2&quot;)],1, var))
HC.var[&quot;C6.1_iPSC.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC.c3&quot;)],1, var))
HC.var[&quot;C6.1_MSC.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_MSC.c1&quot;)],1, var))
HC.var[&quot;C6.1_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;C6.1_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic.c2&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_iPSC.c1&quot;,&quot;H1.1_iPSC.c2&quot;,&quot;H1.1_iPSC.c3&quot;,&quot;H1.1_MSC.c1&quot;,&quot;H1.1_Osteogenic.c1&quot;,&quot;H1.1_Osteogenic.c2&quot;,
                                    &quot;H1.2_iPSC.c1&quot;,&quot;H1.2_iPSC.c2&quot;,&quot;H1.2_iPSC.c3&quot;,&quot;H1.2_MSC.c1&quot;,&quot;H1.2_Osteogenic.c1&quot;,&quot;H1.2_Osteogenic.c2&quot;,
                                    &quot;H2.1_iPSC.c1&quot;,&quot;H2.1_iPSC.c2&quot;,&quot;H2.1_iPSC.c3&quot;,&quot;H2.1_MSC.c1&quot;,&quot;H2.1_Osteogenic.c1&quot;,&quot;H2.1_Osteogenic.c2&quot;,
                                    &quot;H3.1_iPSC.c1&quot;,&quot;H3.1_iPSC.c2&quot;,&quot;H3.1_iPSC.c3&quot;,&quot;H3.1_MSC.c1&quot;,&quot;H3.1_Osteogenic.c1&quot;,&quot;H3.1_Osteogenic.c2&quot;,
                                    &quot;H4.1_iPSC.c1&quot;,&quot;H4.1_iPSC.c2&quot;,&quot;H4.1_iPSC.c3&quot;,&quot;H4.1_MSC.c1&quot;,&quot;H4.1_Osteogenic.c1&quot;,&quot;H4.1_Osteogenic.c2&quot;,
                                    &quot;H5.1_iPSC.c1&quot;,&quot;H5.1_iPSC.c2&quot;,&quot;H5.1_iPSC.c3&quot;,&quot;H5.1_MSC.c1&quot;,&quot;H5.1_Osteogenic.c1&quot;,&quot;H5.1_Osteogenic.c2&quot;,
                                    &quot;H6.1_iPSC.c1&quot;,&quot;H6.1_iPSC.c2&quot;,&quot;H6.1_iPSC.c3&quot;,&quot;H6.1_MSC.c1&quot;,&quot;H6.1_Osteogenic.c1&quot;,&quot;H6.1_Osteogenic.c2&quot;,
                                    &quot;C1.1_iPSC.c1&quot;,&quot;C1.1_iPSC.c2&quot;,&quot;C1.1_iPSC.c3&quot;,&quot;C1.1_MSC.c1&quot;,&quot;C1.1_Osteogenic.c1&quot;,&quot;C1.1_Osteogenic.c2&quot;,
                                    &quot;C1.2_iPSC.c1&quot;,&quot;C1.2_iPSC.c2&quot;,&quot;C1.2_iPSC.c3&quot;,&quot;C1.2_MSC.c1&quot;,&quot;C1.2_Osteogenic.c1&quot;,&quot;C1.2_Osteogenic.c2&quot;,
                                    &quot;C2.1_iPSC.c1&quot;,&quot;C2.1_iPSC.c2&quot;,&quot;C2.1_iPSC.c3&quot;,&quot;C2.1_MSC.c1&quot;,&quot;C2.1_Osteogenic.c1&quot;,&quot;C2.1_Osteogenic.c2&quot;,
                                    &quot;C3.1_iPSC.c1&quot;,&quot;C3.1_iPSC.c2&quot;,&quot;C3.1_iPSC.c3&quot;,&quot;C3.1_MSC.c1&quot;,&quot;C3.1_Osteogenic.c1&quot;,&quot;C3.1_Osteogenic.c2&quot;,
                                    &quot;C4.1_iPSC.c1&quot;,&quot;C4.1_iPSC.c2&quot;,&quot;C4.1_iPSC.c3&quot;,&quot;C4.1_MSC.c1&quot;,&quot;C4.1_Osteogenic.c1&quot;,&quot;C4.1_Osteogenic.c2&quot;,
                                    &quot;C5.1_iPSC.c1&quot;,&quot;C5.1_iPSC.c2&quot;,&quot;C5.1_iPSC.c3&quot;,&quot;C5.1_MSC.c1&quot;,&quot;C5.1_Osteogenic.c1&quot;,&quot;C5.1_Osteogenic.c2&quot;,
                                    &quot;C6.1_iPSC.c1&quot;,&quot;C6.1_iPSC.c2&quot;,&quot;C6.1_iPSC.c3&quot;,&quot;C6.1_MSC.c1&quot;,&quot;C6.1_Osteogenic.c1&quot;,&quot;C6.1_Osteogenic.c2&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(HC.var.melt %&gt;% filter(cluster %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)), aes(x=individual, y=log2(value), fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1.1_iPSC.c1,HC.var$H1.1_MSC.c1)       
t.test(HC.var$H1.1_MSC.c1,HC.var$H1.1_Osteogenic.c1) 
t.test(HC.var$H1.2_iPSC.c1,HC.var$H1.2_MSC.c1)       
t.test(HC.var$H1.2_MSC.c1,HC.var$H1.2_Osteogenic.c1) 
t.test(HC.var$H2.1_iPSC.c1,HC.var$H2.1_MSC.c1)       
t.test(HC.var$H2.1_MSC.c1,HC.var$H2.1_Osteogenic.c1) 
t.test(HC.var$H3.1_iPSC.c1,HC.var$H3.1_MSC.c1)       
t.test(HC.var$H3.1_MSC.c1,HC.var$H3.1_Osteogenic.c1) 
t.test(HC.var$H4.1_iPSC.c1,HC.var$H4.1_MSC.c1)       
t.test(HC.var$H4.1_MSC.c1,HC.var$H4.1_Osteogenic.c1) 
t.test(HC.var$H5.1_iPSC.c1,HC.var$H5.1_MSC.c1)       
t.test(HC.var$H5.1_MSC.c1,HC.var$H5.1_Osteogenic.c1) 
t.test(HC.var$H6.1_iPSC.c1,HC.var$H6.1_MSC.c1)       
t.test(HC.var$H6.1_MSC.c1,HC.var$H6.1_Osteogenic.c1) 
t.test(HC.var$C1.1_iPSC.c1,HC.var$C1.1_MSC.c1)       
t.test(HC.var$C1.1_MSC.c1,HC.var$C1.1_Osteogenic.c1) 
t.test(HC.var$C1.2_iPSC.c1,HC.var$C1.2_MSC.c1)       
t.test(HC.var$C1.2_MSC.c1,HC.var$C1.2_Osteogenic.c1) 
t.test(HC.var$C2.1_iPSC.c1,HC.var$C2.1_MSC.c1)       
t.test(HC.var$C2.1_MSC.c1,HC.var$C2.1_Osteogenic.c1) 
t.test(HC.var$C3.1_iPSC.c1,HC.var$C3.1_MSC.c1)       
t.test(HC.var$C3.1_MSC.c1,HC.var$C3.1_Osteogenic.c1) 
t.test(HC.var$C4.1_iPSC.c1,HC.var$C4.1_MSC.c1)       
t.test(HC.var$C4.1_MSC.c1,HC.var$C4.1_Osteogenic.c1) 
t.test(HC.var$C5.1_iPSC.c1,HC.var$C5.1_MSC.c1)       
t.test(HC.var$C5.1_MSC.c1,HC.var$C5.1_Osteogenic.c1) 
t.test(HC.var$C6.1_iPSC.c1,HC.var$C6.1_MSC.c1)       
t.test(HC.var$C6.1_MSC.c1,HC.var$C6.1_Osteogenic.c1) </code></pre>
</div>
<div id="sc-general-ad-hoc-classifications-vs.species-thresh0" class="section level4">
<h4>SC: General Ad Hoc Classifications vs. Species (thresh0)</h4>
<pre class="r"><code># Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$AdHoc.Assign.thresh0,sep = &quot;_&quot;),
                      levels=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Human_other&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;,&quot;Chimp_other&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC&quot;)],1, mean))
HC.mean[&quot;Human_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic&quot;)],1, mean))
HC.mean[&quot;Chimp_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC&quot;)],1, mean))
HC.mean[&quot;Chimp_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$adhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$Human_iPSC,HC.mean$Human_MSC)       
t.test(HC.mean$Human_MSC,HC.mean$Human_Osteogenic) 
t.test(HC.mean$Chimp_iPSC,HC.mean$Chimp_MSC)       
t.test(HC.mean$Chimp_MSC,HC.mean$Chimp_Osteogenic) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_iPSC&quot;)],1, var))
HC.var[&quot;Human_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_MSC&quot;)],1, var))
HC.var[&quot;Human_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic&quot;)],1, var))
HC.var[&quot;Chimp_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_iPSC&quot;)],1, var))
HC.var[&quot;Chimp_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_MSC&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,id.vars=&quot;gene&quot;,measure.vars=c(&quot;Human_iPSC&quot;,&quot;Human_MSC&quot;,&quot;Human_Osteogenic&quot;,&quot;Chimp_iPSC&quot;,&quot;Chimp_MSC&quot;,&quot;Chimp_Osteogenic&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$adhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$Human_iPSC,HC.var$Human_MSC)       
t.test(HC.var$Human_MSC,HC.var$Human_Osteogenic) 
t.test(HC.var$Chimp_iPSC,HC.var$Chimp_MSC)       
t.test(HC.var$Chimp_MSC,HC.var$Chimp_Osteogenic) </code></pre>
</div>
</div>
<div id="sc-general-ad-hoc-classifications-vs.individuals-thresh0" class="section level3">
<h3>SC: General Ad Hoc Classifications vs. Individuals (thresh0)</h3>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual,dge$samples$AdHoc.Assign.thresh0,sep = &quot;_&quot;),
                      levels=c(&quot;H1_iPSC&quot;,&quot;H1_MSC&quot;,&quot;H1_Osteogenic&quot;,
                               &quot;H2_iPSC&quot;,&quot;H2_MSC&quot;,&quot;H2_Osteogenic&quot;,
                               &quot;H3_iPSC&quot;,&quot;H3_MSC&quot;,&quot;H3_Osteogenic&quot;,
                               &quot;H4_iPSC&quot;,&quot;H4_MSC&quot;,&quot;H4_Osteogenic&quot;,
                               &quot;H5_iPSC&quot;,&quot;H5_MSC&quot;,&quot;H5_Osteogenic&quot;,
                               &quot;H6_iPSC&quot;,&quot;H6_MSC&quot;,&quot;H6_Osteogenic&quot;,
                               &quot;C1_iPSC&quot;,&quot;C1_MSC&quot;,&quot;C1_Osteogenic&quot;,
                               &quot;C2_iPSC&quot;,&quot;C2_MSC&quot;,&quot;C2_Osteogenic&quot;,
                               &quot;C3_iPSC&quot;,&quot;C3_MSC&quot;,&quot;C3_Osteogenic&quot;,
                               &quot;C4_iPSC&quot;,&quot;C4_MSC&quot;,&quot;C4_Osteogenic&quot;,
                               &quot;C5_iPSC&quot;,&quot;C5_MSC&quot;,&quot;C5_Osteogenic&quot;,
                               &quot;C6_iPSC&quot;,&quot;C6_MSC&quot;,&quot;C6_Osteogenic&quot;))

# Calculate log2cpm means within individuals - generally mean increases in MSCs (except C4, C5, H5, H6) and decreases in Osteogenic cells (except H1)
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC&quot;)],1, mean))
HC.mean[&quot;H1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_MSC&quot;)],1, mean))
HC.mean[&quot;H1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC&quot;)],1, mean))
HC.mean[&quot;H2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_MSC&quot;)],1, mean))
HC.mean[&quot;H2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H3_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC&quot;)],1, mean))
HC.mean[&quot;H3_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_MSC&quot;)],1, mean))
HC.mean[&quot;H3_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H4_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC&quot;)],1, mean))
HC.mean[&quot;H4_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_MSC&quot;)],1, mean))
HC.mean[&quot;H4_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H5_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC&quot;)],1, mean))
HC.mean[&quot;H5_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_MSC&quot;)],1, mean))
HC.mean[&quot;H5_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H6_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC&quot;)],1, mean))
HC.mean[&quot;H6_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_MSC&quot;)],1, mean))
HC.mean[&quot;H6_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC&quot;)],1, mean))
HC.mean[&quot;C1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_MSC&quot;)],1, mean))
HC.mean[&quot;C1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC&quot;)],1, mean))
HC.mean[&quot;C2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_MSC&quot;)],1, mean))
HC.mean[&quot;C2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C3_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC&quot;)],1, mean))
HC.mean[&quot;C3_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_MSC&quot;)],1, mean))
HC.mean[&quot;C3_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C4_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC&quot;)],1, mean))
HC.mean[&quot;C4_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_MSC&quot;)],1, mean))
HC.mean[&quot;C4_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C5_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC&quot;)],1, mean))
HC.mean[&quot;C5_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_MSC&quot;)],1, mean))
HC.mean[&quot;C5_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C6_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC&quot;)],1, mean))
HC.mean[&quot;C6_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_MSC&quot;)],1, mean))
HC.mean[&quot;C6_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_iPSC&quot;,&quot;H1_MSC&quot;,&quot;H1_Osteogenic&quot;,
                                    &quot;H2_iPSC&quot;,&quot;H2_MSC&quot;,&quot;H2_Osteogenic&quot;,
                                    &quot;H3_iPSC&quot;,&quot;H3_MSC&quot;,&quot;H3_Osteogenic&quot;,
                                    &quot;H4_iPSC&quot;,&quot;H4_MSC&quot;,&quot;H4_Osteogenic&quot;,
                                    &quot;H5_iPSC&quot;,&quot;H5_MSC&quot;,&quot;H5_Osteogenic&quot;,
                                    &quot;H6_iPSC&quot;,&quot;H6_MSC&quot;,&quot;H6_Osteogenic&quot;,
                                    &quot;C1_iPSC&quot;,&quot;C1_MSC&quot;,&quot;C1_Osteogenic&quot;,
                                    &quot;C2_iPSC&quot;,&quot;C2_MSC&quot;,&quot;C2_Osteogenic&quot;,
                                    &quot;C3_iPSC&quot;,&quot;C3_MSC&quot;,&quot;C3_Osteogenic&quot;,
                                    &quot;C4_iPSC&quot;,&quot;C4_MSC&quot;,&quot;C4_Osteogenic&quot;,
                                    &quot;C5_iPSC&quot;,&quot;C5_MSC&quot;,&quot;C5_Osteogenic&quot;,
                                    &quot;C6_iPSC&quot;,&quot;C6_MSC&quot;,&quot;C6_Osteogenic&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$adhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1_iPSC,HC.mean$H1_MSC)       
t.test(HC.mean$H1_MSC,HC.mean$H1_Osteogenic) 
t.test(HC.mean$H2_iPSC,HC.mean$H2_MSC)       
t.test(HC.mean$H2_MSC,HC.mean$H2_Osteogenic) 
t.test(HC.mean$H3_iPSC,HC.mean$H3_MSC)       
t.test(HC.mean$H3_MSC,HC.mean$H3_Osteogenic) 
t.test(HC.mean$H4_iPSC,HC.mean$H4_MSC)       
t.test(HC.mean$H4_MSC,HC.mean$H4_Osteogenic) 
t.test(HC.mean$H5_iPSC,HC.mean$H5_MSC)       
t.test(HC.mean$H5_MSC,HC.mean$H5_Osteogenic) 
t.test(HC.mean$H6_iPSC,HC.mean$H6_MSC)       
t.test(HC.mean$H6_MSC,HC.mean$H6_Osteogenic) 
t.test(HC.mean$C1_iPSC,HC.mean$C1_MSC)       
t.test(HC.mean$C1_MSC,HC.mean$C1_Osteogenic) 
t.test(HC.mean$C2_iPSC,HC.mean$C2_MSC)       
t.test(HC.mean$C2_MSC,HC.mean$C2_Osteogenic) 
t.test(HC.mean$C3_iPSC,HC.mean$C3_MSC)       
t.test(HC.mean$C3_MSC,HC.mean$C3_Osteogenic) 
t.test(HC.mean$C4_iPSC,HC.mean$C4_MSC)       
t.test(HC.mean$C4_MSC,HC.mean$C4_Osteogenic) 
t.test(HC.mean$C5_iPSC,HC.mean$C5_MSC)       
t.test(HC.mean$C5_MSC,HC.mean$C5_Osteogenic) 
t.test(HC.mean$C6_iPSC,HC.mean$C6_MSC)       
t.test(HC.mean$C6_MSC,HC.mean$C6_Osteogenic) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_iPSC&quot;)],1, var))
HC.var[&quot;H1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_MSC&quot;)],1, var))
HC.var[&quot;H1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1_Osteogenic&quot;)],1, var))
HC.var[&quot;H2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_iPSC&quot;)],1, var))
HC.var[&quot;H2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_MSC&quot;)],1, var))
HC.var[&quot;H2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2_Osteogenic&quot;)],1, var))
HC.var[&quot;H3_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_iPSC&quot;)],1, var))
HC.var[&quot;H3_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_MSC&quot;)],1, var))
HC.var[&quot;H3_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3_Osteogenic&quot;)],1, var))
HC.var[&quot;H4_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_iPSC&quot;)],1, var))
HC.var[&quot;H4_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_MSC&quot;)],1, var))
HC.var[&quot;H4_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4_Osteogenic&quot;)],1, var))
HC.var[&quot;H5_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_iPSC&quot;)],1, var))
HC.var[&quot;H5_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_MSC&quot;)],1, var))
HC.var[&quot;H5_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5_Osteogenic&quot;)],1, var))
HC.var[&quot;H6_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_iPSC&quot;)],1, var))
HC.var[&quot;H6_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_MSC&quot;)],1, var))
HC.var[&quot;H6_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6_Osteogenic&quot;)],1, var))
HC.var[&quot;C1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_iPSC&quot;)],1, var))
HC.var[&quot;C1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_MSC&quot;)],1, var))
HC.var[&quot;C1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1_Osteogenic&quot;)],1, var))
HC.var[&quot;C2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_iPSC&quot;)],1, var))
HC.var[&quot;C2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_MSC&quot;)],1, var))
HC.var[&quot;C2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2_Osteogenic&quot;)],1, var))
HC.var[&quot;C3_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_iPSC&quot;)],1, var))
HC.var[&quot;C3_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_MSC&quot;)],1, var))
HC.var[&quot;C3_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3_Osteogenic&quot;)],1, var))
HC.var[&quot;C4_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_iPSC&quot;)],1, var))
HC.var[&quot;C4_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_MSC&quot;)],1, var))
HC.var[&quot;C4_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4_Osteogenic&quot;)],1, var))
HC.var[&quot;C5_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_iPSC&quot;)],1, var))
HC.var[&quot;C5_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_MSC&quot;)],1, var))
HC.var[&quot;C5_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5_Osteogenic&quot;)],1, var))
HC.var[&quot;C6_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_iPSC&quot;)],1, var))
HC.var[&quot;C6_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_MSC&quot;)],1, var))
HC.var[&quot;C6_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6_Osteogenic&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1_iPSC&quot;,&quot;H1_MSC&quot;,&quot;H1_Osteogenic&quot;,
                                    &quot;H2_iPSC&quot;,&quot;H2_MSC&quot;,&quot;H2_Osteogenic&quot;,
                                    &quot;H3_iPSC&quot;,&quot;H3_MSC&quot;,&quot;H3_Osteogenic&quot;,
                                    &quot;H4_iPSC&quot;,&quot;H4_MSC&quot;,&quot;H4_Osteogenic&quot;,
                                    &quot;H5_iPSC&quot;,&quot;H5_MSC&quot;,&quot;H5_Osteogenic&quot;,
                                    &quot;H6_iPSC&quot;,&quot;H6_MSC&quot;,&quot;H6_Osteogenic&quot;,
                                    &quot;C1_iPSC&quot;,&quot;C1_MSC&quot;,&quot;C1_Osteogenic&quot;,
                                    &quot;C2_iPSC&quot;,&quot;C2_MSC&quot;,&quot;C2_Osteogenic&quot;,
                                    &quot;C3_iPSC&quot;,&quot;C3_MSC&quot;,&quot;C3_Osteogenic&quot;,
                                    &quot;C4_iPSC&quot;,&quot;C4_MSC&quot;,&quot;C4_Osteogenic&quot;,
                                    &quot;C5_iPSC&quot;,&quot;C5_MSC&quot;,&quot;C5_Osteogenic&quot;,
                                    &quot;C6_iPSC&quot;,&quot;C6_MSC&quot;,&quot;C6_Osteogenic&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$adhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1_iPSC,HC.var$H1_MSC)       
t.test(HC.var$H1_MSC,HC.var$H1_Osteogenic) 
t.test(HC.var$H2_iPSC,HC.var$H2_MSC)       
t.test(HC.var$H2_MSC,HC.var$H2_Osteogenic) 
t.test(HC.var$H3_iPSC,HC.var$H3_MSC)       
t.test(HC.var$H3_MSC,HC.var$H3_Osteogenic) 
t.test(HC.var$H4_iPSC,HC.var$H4_MSC)       
t.test(HC.var$H4_MSC,HC.var$H4_Osteogenic) 
t.test(HC.var$H5_iPSC,HC.var$H5_MSC)       
t.test(HC.var$H5_MSC,HC.var$H5_Osteogenic) 
t.test(HC.var$H6_iPSC,HC.var$H6_MSC)       
t.test(HC.var$H6_MSC,HC.var$H6_Osteogenic) 
t.test(HC.var$C1_iPSC,HC.var$C1_MSC)       
t.test(HC.var$C1_MSC,HC.var$C1_Osteogenic) 
t.test(HC.var$C2_iPSC,HC.var$C2_MSC)       
t.test(HC.var$C2_MSC,HC.var$C2_Osteogenic) 
t.test(HC.var$C3_iPSC,HC.var$C3_MSC)       
t.test(HC.var$C3_MSC,HC.var$C3_Osteogenic) 
t.test(HC.var$C4_iPSC,HC.var$C4_MSC)       
t.test(HC.var$C4_MSC,HC.var$C4_Osteogenic) 
t.test(HC.var$C5_iPSC,HC.var$C5_MSC)       
t.test(HC.var$C5_MSC,HC.var$C5_Osteogenic) 
t.test(HC.var$C6_iPSC,HC.var$C6_MSC)       
t.test(HC.var$C6_MSC,HC.var$C6_Osteogenic) </code></pre>
</div>
<div id="sc-general-ad-hoc-classifications-vs.individual.replicates-thresh0" class="section level3">
<h3>SC: General Ad Hoc Classifications vs. Individual.Replicates (thresh0)</h3>
<pre class="r"><code># Compile group labels of each individual
groupFactor &lt;- factor(paste(dge$samples$Individual.Replicate,dge$samples$AdHoc.Assign.thresh0,sep = &quot;_&quot;),
                      levels=c(&quot;H1.1_iPSC&quot;,&quot;H1.1_MSC&quot;,&quot;H1.1_Osteogenic&quot;,
                               &quot;H1.2_iPSC&quot;,&quot;H1.2_MSC&quot;,&quot;H1.2_Osteogenic&quot;,
                               &quot;H2.1_iPSC&quot;,&quot;H2.1_MSC&quot;,&quot;H2.1_Osteogenic&quot;,
                               &quot;H3.1_iPSC&quot;,&quot;H3.1_MSC&quot;,&quot;H3.1_Osteogenic&quot;,
                               &quot;H4.1_iPSC&quot;,&quot;H4.1_MSC&quot;,&quot;H4.1_Osteogenic&quot;,
                               &quot;H5.1_iPSC&quot;,&quot;H5.1_MSC&quot;,&quot;H5.1_Osteogenic&quot;,
                               &quot;H6.1_iPSC&quot;,&quot;H6.1_MSC&quot;,&quot;H6.1_Osteogenic&quot;,
                               &quot;C1.1_iPSC&quot;,&quot;C1.1_MSC&quot;,&quot;C1.1_Osteogenic&quot;,
                               &quot;C1.2_iPSC&quot;,&quot;C1.2_MSC&quot;,&quot;C1.2_Osteogenic&quot;,
                               &quot;C2.1_iPSC&quot;,&quot;C2.1_MSC&quot;,&quot;C2.1_Osteogenic&quot;,
                               &quot;C3.1_iPSC&quot;,&quot;C3.1_MSC&quot;,&quot;C3.1_Osteogenic&quot;,
                               &quot;C4.1_iPSC&quot;,&quot;C4.1_MSC&quot;,&quot;C4.1_Osteogenic&quot;,
                               &quot;C5.1_iPSC&quot;,&quot;C5.1_MSC&quot;,&quot;C5.1_Osteogenic&quot;,
                               &quot;C6.1_iPSC&quot;,&quot;C6.1_MSC&quot;,&quot;C6.1_Osteogenic&quot;))

# Calculate log2cpm means within individuals - generally mean increases in MSCs (except C4.1, C5.1, H5.1, H6.1) and decreases in Osteogenic cells (except H1.2)
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;H1.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H1.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_MSC&quot;)],1, mean))
HC.mean[&quot;H1.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H1.2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC&quot;)],1, mean))
HC.mean[&quot;H1.2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_MSC&quot;)],1, mean))
HC.mean[&quot;H1.2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H2.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H2.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_MSC&quot;)],1, mean))
HC.mean[&quot;H2.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H3.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H3.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_MSC&quot;)],1, mean))
HC.mean[&quot;H3.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H4.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H4.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_MSC&quot;)],1, mean))
HC.mean[&quot;H4.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H5.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H5.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_MSC&quot;)],1, mean))
HC.mean[&quot;H5.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;H6.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC&quot;)],1, mean))
HC.mean[&quot;H6.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_MSC&quot;)],1, mean))
HC.mean[&quot;H6.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C1.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C1.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_MSC&quot;)],1, mean))
HC.mean[&quot;C1.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C1.2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC&quot;)],1, mean))
HC.mean[&quot;C1.2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_MSC&quot;)],1, mean))
HC.mean[&quot;C1.2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C2.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C2.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_MSC&quot;)],1, mean))
HC.mean[&quot;C2.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C3.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C3.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_MSC&quot;)],1, mean))
HC.mean[&quot;C3.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C4.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C4.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_MSC&quot;)],1, mean))
HC.mean[&quot;C4.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C5.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C5.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_MSC&quot;)],1, mean))
HC.mean[&quot;C5.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic&quot;)],1, mean))
HC.mean[&quot;C6.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC&quot;)],1, mean))
HC.mean[&quot;C6.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_MSC&quot;)],1, mean))
HC.mean[&quot;C6.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_iPSC&quot;,&quot;H1.1_MSC&quot;,&quot;H1.1_Osteogenic&quot;,
                                    &quot;H1.2_iPSC&quot;,&quot;H1.2_MSC&quot;,&quot;H1.2_Osteogenic&quot;,
                                    &quot;H2.1_iPSC&quot;,&quot;H2.1_MSC&quot;,&quot;H2.1_Osteogenic&quot;,
                                    &quot;H3.1_iPSC&quot;,&quot;H3.1_MSC&quot;,&quot;H3.1_Osteogenic&quot;,
                                    &quot;H4.1_iPSC&quot;,&quot;H4.1_MSC&quot;,&quot;H4.1_Osteogenic&quot;,
                                    &quot;H5.1_iPSC&quot;,&quot;H5.1_MSC&quot;,&quot;H5.1_Osteogenic&quot;,
                                    &quot;H6.1_iPSC&quot;,&quot;H6.1_MSC&quot;,&quot;H6.1_Osteogenic&quot;,
                                    &quot;C1.1_iPSC&quot;,&quot;C1.1_MSC&quot;,&quot;C1.1_Osteogenic&quot;,
                                    &quot;C1.2_iPSC&quot;,&quot;C1.2_MSC&quot;,&quot;C1.2_Osteogenic&quot;,
                                    &quot;C2.1_iPSC&quot;,&quot;C2.1_MSC&quot;,&quot;C2.1_Osteogenic&quot;,
                                    &quot;C3.1_iPSC&quot;,&quot;C3.1_MSC&quot;,&quot;C3.1_Osteogenic&quot;,
                                    &quot;C4.1_iPSC&quot;,&quot;C4.1_MSC&quot;,&quot;C4.1_Osteogenic&quot;,
                                    &quot;C5.1_iPSC&quot;,&quot;C5.1_MSC&quot;,&quot;C5.1_Osteogenic&quot;,
                                    &quot;C6.1_iPSC&quot;,&quot;C6.1_MSC&quot;,&quot;C6.1_Osteogenic&quot;))
HC.mean.melt$individual &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$adhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=individual, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$H1.1_iPSC,HC.mean$H1.1_MSC)       
t.test(HC.mean$H1.1_MSC,HC.mean$H1.1_Osteogenic) 
t.test(HC.mean$H1.2_iPSC,HC.mean$H1.2_MSC)       
t.test(HC.mean$H1.2_MSC,HC.mean$H1.2_Osteogenic) 
t.test(HC.mean$H2.1_iPSC,HC.mean$H2.1_MSC)       
t.test(HC.mean$H2.1_MSC,HC.mean$H2.1_Osteogenic) 
t.test(HC.mean$H3.1_iPSC,HC.mean$H3.1_MSC)       
t.test(HC.mean$H3.1_MSC,HC.mean$H3.1_Osteogenic) 
t.test(HC.mean$H4.1_iPSC,HC.mean$H4.1_MSC)       
t.test(HC.mean$H4.1_MSC,HC.mean$H4.1_Osteogenic) 
t.test(HC.mean$H5.1_iPSC,HC.mean$H5.1_MSC)       
t.test(HC.mean$H5.1_MSC,HC.mean$H5.1_Osteogenic) 
t.test(HC.mean$H6.1_iPSC,HC.mean$H6.1_MSC)       
t.test(HC.mean$H6.1_MSC,HC.mean$H6.1_Osteogenic) 
t.test(HC.mean$C1.1_iPSC,HC.mean$C1.1_MSC)       
t.test(HC.mean$C1.1_MSC,HC.mean$C1.1_Osteogenic) 
t.test(HC.mean$C1.2_iPSC,HC.mean$C1.2_MSC)       
t.test(HC.mean$C1.2_MSC,HC.mean$C1.2_Osteogenic) 
t.test(HC.mean$C2.1_iPSC,HC.mean$C2.1_MSC)       
t.test(HC.mean$C2.1_MSC,HC.mean$C2.1_Osteogenic) 
t.test(HC.mean$C3.1_iPSC,HC.mean$C3.1_MSC)       
t.test(HC.mean$C3.1_MSC,HC.mean$C3.1_Osteogenic) 
t.test(HC.mean$C4.1_iPSC,HC.mean$C4.1_MSC)       
t.test(HC.mean$C4.1_MSC,HC.mean$C4.1_Osteogenic) 
t.test(HC.mean$C5.1_iPSC,HC.mean$C5.1_MSC)       
t.test(HC.mean$C5.1_MSC,HC.mean$C5.1_Osteogenic) 
t.test(HC.mean$C6.1_iPSC,HC.mean$C6.1_MSC)       
t.test(HC.mean$C6.1_MSC,HC.mean$C6.1_Osteogenic) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;H1.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_iPSC&quot;)],1, var))
HC.var[&quot;H1.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_MSC&quot;)],1, var))
HC.var[&quot;H1.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.1_Osteogenic&quot;)],1, var))
HC.var[&quot;H1.2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_iPSC&quot;)],1, var))
HC.var[&quot;H1.2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_MSC&quot;)],1, var))
HC.var[&quot;H1.2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H1.2_Osteogenic&quot;)],1, var))
HC.var[&quot;H2.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_iPSC&quot;)],1, var))
HC.var[&quot;H2.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_MSC&quot;)],1, var))
HC.var[&quot;H2.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H2.1_Osteogenic&quot;)],1, var))
HC.var[&quot;H3.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_iPSC&quot;)],1, var))
HC.var[&quot;H3.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_MSC&quot;)],1, var))
HC.var[&quot;H3.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H3.1_Osteogenic&quot;)],1, var))
HC.var[&quot;H4.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_iPSC&quot;)],1, var))
HC.var[&quot;H4.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_MSC&quot;)],1, var))
HC.var[&quot;H4.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H4.1_Osteogenic&quot;)],1, var))
HC.var[&quot;H5.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_iPSC&quot;)],1, var))
HC.var[&quot;H5.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_MSC&quot;)],1, var))
HC.var[&quot;H5.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H5.1_Osteogenic&quot;)],1, var))
HC.var[&quot;H6.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_iPSC&quot;)],1, var))
HC.var[&quot;H6.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_MSC&quot;)],1, var))
HC.var[&quot;H6.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;H6.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C1.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_iPSC&quot;)],1, var))
HC.var[&quot;C1.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_MSC&quot;)],1, var))
HC.var[&quot;C1.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C1.2_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_iPSC&quot;)],1, var))
HC.var[&quot;C1.2_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_MSC&quot;)],1, var))
HC.var[&quot;C1.2_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C1.2_Osteogenic&quot;)],1, var))
HC.var[&quot;C2.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_iPSC&quot;)],1, var))
HC.var[&quot;C2.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_MSC&quot;)],1, var))
HC.var[&quot;C2.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C2.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C3.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_iPSC&quot;)],1, var))
HC.var[&quot;C3.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_MSC&quot;)],1, var))
HC.var[&quot;C3.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C3.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C4.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_iPSC&quot;)],1, var))
HC.var[&quot;C4.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_MSC&quot;)],1, var))
HC.var[&quot;C4.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C4.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C5.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_iPSC&quot;)],1, var))
HC.var[&quot;C5.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_MSC&quot;)],1, var))
HC.var[&quot;C5.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C5.1_Osteogenic&quot;)],1, var))
HC.var[&quot;C6.1_iPSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_iPSC&quot;)],1, var))
HC.var[&quot;C6.1_MSC&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_MSC&quot;)],1, var))
HC.var[&quot;C6.1_Osteogenic&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;C6.1_Osteogenic&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;H1.1_iPSC&quot;,&quot;H1.1_MSC&quot;,&quot;H1.1_Osteogenic&quot;,
                                    &quot;H1.2_iPSC&quot;,&quot;H1.2_MSC&quot;,&quot;H1.2_Osteogenic&quot;,
                                    &quot;H2.1_iPSC&quot;,&quot;H2.1_MSC&quot;,&quot;H2.1_Osteogenic&quot;,
                                    &quot;H3.1_iPSC&quot;,&quot;H3.1_MSC&quot;,&quot;H3.1_Osteogenic&quot;,
                                    &quot;H4.1_iPSC&quot;,&quot;H4.1_MSC&quot;,&quot;H4.1_Osteogenic&quot;,
                                    &quot;H5.1_iPSC&quot;,&quot;H5.1_MSC&quot;,&quot;H5.1_Osteogenic&quot;,
                                    &quot;H6.1_iPSC&quot;,&quot;H6.1_MSC&quot;,&quot;H6.1_Osteogenic&quot;,
                                    &quot;C1.1_iPSC&quot;,&quot;C1.1_MSC&quot;,&quot;C1.1_Osteogenic&quot;,
                                    &quot;C1.2_iPSC&quot;,&quot;C1.2_MSC&quot;,&quot;C1.2_Osteogenic&quot;,
                                    &quot;C2.1_iPSC&quot;,&quot;C2.1_MSC&quot;,&quot;C2.1_Osteogenic&quot;,
                                    &quot;C3.1_iPSC&quot;,&quot;C3.1_MSC&quot;,&quot;C3.1_Osteogenic&quot;,
                                    &quot;C4.1_iPSC&quot;,&quot;C4.1_MSC&quot;,&quot;C4.1_Osteogenic&quot;,
                                    &quot;C5.1_iPSC&quot;,&quot;C5.1_MSC&quot;,&quot;C5.1_Osteogenic&quot;,
                                    &quot;C6.1_iPSC&quot;,&quot;C6.1_MSC&quot;,&quot;C6.1_Osteogenic&quot;))
HC.var.melt$individual &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$adhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=individual, y=value, fill=adhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Individual&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;, labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;), values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$H1.1_iPSC,HC.var$H1.1_MSC)       
t.test(HC.var$H1.1_MSC,HC.var$H1.1_Osteogenic) 
t.test(HC.var$H1.2_iPSC,HC.var$H1.2_MSC)       
t.test(HC.var$H1.2_MSC,HC.var$H1.2_Osteogenic) 
t.test(HC.var$H2.1_iPSC,HC.var$H2.1_MSC)       
t.test(HC.var$H2.1_MSC,HC.var$H2.1_Osteogenic) 
t.test(HC.var$H3.1_iPSC,HC.var$H3.1_MSC)       
t.test(HC.var$H3.1_MSC,HC.var$H3.1_Osteogenic) 
t.test(HC.var$H4.1_iPSC,HC.var$H4.1_MSC)       
t.test(HC.var$H4.1_MSC,HC.var$H4.1_Osteogenic) 
t.test(HC.var$H5.1_iPSC,HC.var$H5.1_MSC)       
t.test(HC.var$H5.1_MSC,HC.var$H5.1_Osteogenic) 
t.test(HC.var$H6.1_iPSC,HC.var$H6.1_MSC)       
t.test(HC.var$H6.1_MSC,HC.var$H6.1_Osteogenic) 
t.test(HC.var$C1.1_iPSC,HC.var$C1.1_MSC)       
t.test(HC.var$C1.1_MSC,HC.var$C1.1_Osteogenic) 
t.test(HC.var$C1.2_iPSC,HC.var$C1.2_MSC)       
t.test(HC.var$C1.2_MSC,HC.var$C1.2_Osteogenic) 
t.test(HC.var$C2.1_iPSC,HC.var$C2.1_MSC)       
t.test(HC.var$C2.1_MSC,HC.var$C2.1_Osteogenic) 
t.test(HC.var$C3.1_iPSC,HC.var$C3.1_MSC)       
t.test(HC.var$C3.1_MSC,HC.var$C3.1_Osteogenic) 
t.test(HC.var$C4.1_iPSC,HC.var$C4.1_MSC)       
t.test(HC.var$C4.1_MSC,HC.var$C4.1_Osteogenic) 
t.test(HC.var$C5.1_iPSC,HC.var$C5.1_MSC)       
t.test(HC.var$C5.1_MSC,HC.var$C5.1_Osteogenic) 
t.test(HC.var$C6.1_iPSC,HC.var$C6.1_MSC)       
t.test(HC.var$C6.1_MSC,HC.var$C6.1_Osteogenic) </code></pre>
<div id="sc-osteogenic-clusters-vs.species-res0.5" class="section level4">
<h4>SC: Osteogenic Clusters vs. Species (res=0.5)</h4>
<pre class="r"><code># Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,dge$samples$Cluster0.5,sep=&quot;_&quot;),
                      levels=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                               &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c3&quot;)],1, mean))
HC.mean[&quot;Human_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c4&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c3&quot;)],1, mean))
HC.mean[&quot;Chimp_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c4&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                                    &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$cluster &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$Human_Osteogenic.c1,HC.mean$Human_Osteogenic.c2) 
t.test(HC.mean$Human_Osteogenic.c2,HC.mean$Human_Osteogenic.c3) 
t.test(HC.mean$Human_Osteogenic.c3,HC.mean$Human_Osteogenic.c4) 
t.test(HC.mean$Chimp_Osteogenic.c1,HC.mean$Chimp_Osteogenic.c2) 
t.test(HC.mean$Chimp_Osteogenic.c2,HC.mean$Chimp_Osteogenic.c3) 
t.test(HC.mean$Chimp_Osteogenic.c3,HC.mean$Chimp_Osteogenic.c4) 

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c3&quot;)],1, var))
HC.var[&quot;Human_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_Osteogenic.c4&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c1&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c1&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c2&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c2&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c3&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c3&quot;)],1, var))
HC.var[&quot;Chimp_Osteogenic.c4&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_Osteogenic.c4&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_Osteogenic.c1&quot;,&quot;Human_Osteogenic.c2&quot;,&quot;Human_Osteogenic.c3&quot;,&quot;Human_Osteogenic.c4&quot;,
                                    &quot;Chimp_Osteogenic.c1&quot;,&quot;Chimp_Osteogenic.c2&quot;,&quot;Chimp_Osteogenic.c3&quot;,&quot;Chimp_Osteogenic.c4&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$cluster &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=value, fill=cluster)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance gene expression&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$Human_Osteogenic.c1,HC.var$Human_Osteogenic.c2) 
t.test(HC.var$Human_Osteogenic.c2,HC.var$Human_Osteogenic.c3) 
t.test(HC.var$Human_Osteogenic.c3,HC.var$Human_Osteogenic.c4) 
t.test(HC.var$Chimp_Osteogenic.c1,HC.var$Chimp_Osteogenic.c2) 
t.test(HC.var$Chimp_Osteogenic.c2,HC.var$Chimp_Osteogenic.c3) 
t.test(HC.var$Chimp_Osteogenic.c3,HC.var$Chimp_Osteogenic.c4) </code></pre>
</div>
<div id="sc-osteogenic-clusters-vs.individuals-res0.5" class="section level4">
<h4>SC: Osteogenic Clusters vs. Individuals (res=0.5)</h4>
</div>
<div id="sc-osteogenic-ad-hoc-classifications-vs.species-thresh0" class="section level4">
<h4>SC: Osteogenic Ad Hoc Classifications vs. Species (thresh0)</h4>
<pre class="r"><code># Compile group labels of each species
groupFactor &lt;- factor(paste(dge$samples$Species,sapply(dge$samples$OstAdHoc.Assign.4gene.thresh0,function(x)sub(&quot; &quot;,&quot;.&quot;,x)),sep=&quot;_&quot;),
                      levels=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                               &quot;Human_unknown.osteogenic&quot;,&quot;Human_not.osteogenic&quot;,
                               &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;,
                               &quot;Chimp_unknown.osteogenic&quot;,&quot;Chimp_not.osteogenic&quot;))

# Calculate log2cpm means within species
HC.mean &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.mean[&quot;gene&quot;] &lt;- rownames(tmm)
HC.mean[&quot;Human_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_preosteoblast&quot;)],1, mean))
HC.mean[&quot;Human_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_embedding.osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_mineralizing.osteoblast&quot;)],1, mean))
HC.mean[&quot;Human_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_maturing.osteocyte&quot;)],1, mean))
HC.mean[&quot;Chimp_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_preosteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_embedding.osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_mineralizing.osteoblast&quot;)],1, mean))
HC.mean[&quot;Chimp_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_maturing.osteocyte&quot;)],1, mean))
HC.mean.melt &lt;- melt(HC.mean,
                     id.vars=&quot;gene&quot;,
                     measure.vars=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                                    &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;))
HC.mean.melt$species &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.mean.melt$ostadhoc &lt;- sapply(HC.mean.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.mean.melt, aes(x=species, y=value, fill=ostadhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 mean gene expression&#39;) +
  scale_fill_manual(name=&quot;Ost Ad Hoc&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;skyblue&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.mean$Human_preosteoblast,HC.mean$Human_osteoblast)                     
t.test(HC.mean$Human_osteoblast,HC.mean$Human_embedding.osteoblast)              
t.test(HC.mean$Human_embedding.osteoblast,HC.mean$Human_mineralizing.osteoblast) 
t.test(HC.mean$Human_mineralizing.osteoblast,HC.mean$Human_maturing.osteocyte)   
t.test(HC.mean$Chimp_preosteoblast,HC.mean$Chimp_osteoblast)                     
t.test(HC.mean$Chimp_osteoblast,HC.mean$Chimp_embedding.osteoblast)              
t.test(HC.mean$Chimp_embedding.osteoblast,HC.mean$Chimp_mineralizing.osteoblast) 
t.test(HC.mean$Chimp_mineralizing.osteoblast,HC.mean$Chimp_maturing.osteocyte)   

# Calculate log2cpm variances within species
HC.var &lt;- data.frame(matrix(nrow=dim(tmm)[1],ncol=0))
HC.var[&quot;gene&quot;] &lt;- rownames(tmm)
HC.var[&quot;Human_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_preosteoblast&quot;)],1, var))
HC.var[&quot;Human_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_osteoblast&quot;)],1, var))
HC.var[&quot;Human_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_embedding.osteoblast&quot;)],1, var))
HC.var[&quot;Human_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_mineralizing.osteoblast&quot;)],1, var))
HC.var[&quot;Human_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Human_maturing.osteocyte&quot;)],1, var))
HC.var[&quot;Chimp_preosteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_preosteoblast&quot;)],1, var))
HC.var[&quot;Chimp_osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_embedding.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_embedding.osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_mineralizing.osteoblast&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_mineralizing.osteoblast&quot;)],1, var))
HC.var[&quot;Chimp_maturing.osteocyte&quot;] &lt;- as.data.frame(apply(tmm[,which(groupFactor==&quot;Chimp_maturing.osteocyte&quot;)],1, var))
HC.var.melt &lt;- melt(HC.var,
                    id.vars=&quot;gene&quot;,
                    measure.vars=c(&quot;Human_preosteoblast&quot;,&quot;Human_osteoblast&quot;,&quot;Human_embedding.osteoblast&quot;,&quot;Human_mineralizing.osteoblast&quot;,&quot;Human_maturing.osteocyte&quot;,
                                   &quot;Chimp_preosteoblast&quot;,&quot;Chimp_osteoblast&quot;,&quot;Chimp_embedding.osteoblast&quot;,&quot;Chimp_mineralizing.osteoblast&quot;,&quot;Chimp_maturing.osteocyte&quot;))
HC.var.melt$species &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
HC.var.melt$ostadhoc &lt;- sapply(HC.var.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])

ggplot(HC.var.melt, aes(x=species, y=value, fill=ostadhoc)) +
  geom_boxplot(width=0.4) +
  xlab(&quot;Species&quot;) +
  ylab(&#39;log2 variance in gene expression&#39;) +
  scale_fill_manual(name=&quot;Ost Ad Hoc&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;MSC.c1&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;skyblue&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
t.test(HC.var$Human_preosteoblast,HC.var$Human_osteoblast)                     
t.test(HC.var$Human_osteoblast,HC.var$Human_embedding.osteoblast)              
t.test(HC.var$Human_embedding.osteoblast,HC.var$Human_mineralizing.osteoblast) 
t.test(HC.var$Human_mineralizing.osteoblast,HC.var$Human_maturing.osteocyte)   
t.test(HC.var$Chimp_preosteoblast,HC.var$Chimp_osteoblast)                     
t.test(HC.var$Chimp_osteoblast,HC.var$Chimp_embedding.osteoblast)              
t.test(HC.var$Chimp_embedding.osteoblast,HC.var$Chimp_mineralizing.osteoblast) 
t.test(HC.var$Chimp_mineralizing.osteoblast,HC.var$Chimp_maturing.osteocyte)   </code></pre>
</div>
<div id="sc-osteogenic-ad-hoc-classifications-vs.individuals-thresh0" class="section level4">
<h4>SC: Osteogenic Ad Hoc Classifications vs. Individuals (thresh0)</h4>
</div>
<div id="sc-osteogenic-ad-hoc-classifications-vs.individual.replicates-thresh0" class="section level4">
<h4>SC: Osteogenic Ad Hoc Classifications vs. Individual.Replicates (thresh0)</h4>
</div>
<div id="sc-osteogenic-ad-hoc-classifications-vs.individuals-thresh0-in-t2" class="section level4">
<h4>SC: Osteogenic Ad Hoc Classifications vs. Individuals (thresh0 in t2)</h4>
</div>
<div id="sc-osteogenic-ad-hoc-classifications-vs.individual.replicates-thresh0-in-t2" class="section level4">
<h4>SC: Osteogenic Ad Hoc Classifications vs. Individual.Replicates (thresh0 in t2)</h4>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] reshape2_1.4.3              scran_1.12.1               
 [3] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
 [5] DelayedArray_0.10.0         BiocParallel_1.18.0        
 [7] matrixStats_0.57.0          Biobase_2.44.0             
 [9] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[11] IRanges_2.18.1              S4Vectors_0.22.1           
[13] BiocGenerics_0.30.0         edgeR_3.26.5               
[15] limma_3.40.6                RColorBrewer_1.1-2         
[17] colorspace_2.0-0            gbm_2.1.5                  
[19] ggplot2_3.3.2               stringr_1.4.0              
[21] stringi_1.4.3               dplyr_1.0.2                
[23] Seurat_3.1.3               

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0         Rtsne_0.15              
  [3] ellipsis_0.3.1           ggridges_0.5.1          
  [5] dynamicTreeCut_1.63-1    rprojroot_2.0.2         
  [7] XVector_0.24.0           BiocNeighbors_1.2.0     
  [9] fs_1.3.1                 farver_2.0.3            
 [11] leiden_0.3.1             listenv_0.7.0           
 [13] npsurv_0.4-0             ggrepel_0.8.2           
 [15] codetools_0.2-16         splines_3.6.1           
 [17] lsei_1.2-0               scater_1.12.2           
 [19] knitr_1.23               jsonlite_1.7.1          
 [21] workflowr_1.6.2          ica_1.0-2               
 [23] cluster_2.1.0            png_0.1-7               
 [25] uwot_0.1.5               sctransform_0.2.0       
 [27] compiler_3.6.1           httr_1.4.2              
 [29] dqrng_0.2.1              Matrix_1.2-18           
 [31] lazyeval_0.2.2           BiocSingular_1.0.0      
 [33] later_1.1.0.1            htmltools_0.5.0         
 [35] tools_3.6.1              rsvd_1.0.1              
 [37] igraph_1.2.4.1           gtable_0.3.0            
 [39] glue_1.4.2               GenomeInfoDbData_1.2.1  
 [41] RANN_2.6.1               rappdirs_0.3.1          
 [43] Rcpp_1.0.5               vctrs_0.3.5             
 [45] gdata_2.18.0             ape_5.4-1               
 [47] nlme_3.1-140             DelayedMatrixStats_1.6.1
 [49] gbRd_0.4-11              lmtest_0.9-37           
 [51] xfun_0.8                 globals_0.12.4          
 [53] lifecycle_0.2.0          irlba_2.3.3             
 [55] gtools_3.8.1             statmod_1.4.32          
 [57] future_1.14.0            zlibbioc_1.30.0         
 [59] MASS_7.3-51.4            zoo_1.8-6               
 [61] scales_1.1.1             promises_1.1.1          
 [63] yaml_2.2.1               reticulate_1.16         
 [65] pbapply_1.4-0            gridExtra_2.3           
 [67] caTools_1.17.1.2         bibtex_0.4.2            
 [69] Rdpack_0.11-0            rlang_0.4.9             
 [71] pkgconfig_2.0.3          bitops_1.0-6            
 [73] evaluate_0.14            lattice_0.20-38         
 [75] ROCR_1.0-7               purrr_0.3.4             
 [77] labeling_0.4.2           htmlwidgets_1.5.2       
 [79] cowplot_1.1.0            tidyselect_1.1.0        
 [81] RcppAnnoy_0.0.12         plyr_1.8.4              
 [83] magrittr_2.0.1           R6_2.5.0                
 [85] gplots_3.0.1.1           generics_0.1.0          
 [87] pillar_1.4.7             whisker_0.3-2           
 [89] withr_2.3.0              fitdistrplus_1.0-14     
 [91] survival_2.44-1.1        RCurl_1.98-1.1          
 [93] tibble_3.0.4             future.apply_1.3.0      
 [95] tsne_0.1-3               crayon_1.3.4            
 [97] KernSmooth_2.23-15       plotly_4.9.2.1          
 [99] rmarkdown_1.13           viridis_0.5.1           
[101] locfit_1.5-9.1           data.table_1.13.2       
[103] git2r_0.26.1             metap_1.1               
[105] digest_0.6.27            tidyr_1.1.2             
[107] httpuv_1.5.1             RcppParallel_5.0.2      
[109] munsell_0.5.0            beeswarm_0.2.3          
[111] viridisLite_0.3.0        vipor_0.4.5             </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
