<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-data-classification</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-data-classification</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">May 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree325664d831fec2328643e91fec129a88f26fde72targetblank325664da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/325664d831fec2328643e91fec129a88f26fde72" target="_blank">325664d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree325664d831fec2328643e91fec129a88f26fde72targetblank325664da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/325664d831fec2328643e91fec129a88f26fde72" target="_blank">325664d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/10x-plot-enrichment-gom.Rmd
    Untracked:  analysis/TODO_10x-data-4-classification-t2.Rmd
    Untracked:  analysis/TODO_10x-data-4-classification.Rmd
    Untracked:  analysis/TODO_10x-data-4-reclassification-tot.Rmd
    Untracked:  analysis/TODO_10x-data-5-diffexp.Rmd
    Untracked:  analysis/TODO_10x-data-6-conservexp.Rmd
    Untracked:  analysis/TODO_10x-data-6-cormotif-t2.Rmd
    Untracked:  analysis/TODO_10x-data-6-cormotif-tot.Rmd
    Untracked:  analysis/TODO_10x-data-6-variance-t2.Rmd
    Untracked:  analysis/TODO_10x-data-6-variance-tot.Rmd
    Untracked:  analysis/TODO_10x-data-7-enrichment.Rmd
    Untracked:  analysis/TODO_10x-data-8-batch-correlation.Rmd
    Untracked:  analysis/TODO_10x-plot-diffexp.Rmd
    Untracked:  analysis/TODO_10x-plot-replicate-correlation-t2.Rmd
    Untracked:  analysis/TODO_10x-plot-replicate-correlation-tot.Rmd
    Untracked:  analysis/TODO_10x-plot-utf1-tac3.Rmd
    Untracked:  analysis/TODO_10x-view-batch-effects.Rmd
    Untracked:  analysis/TODO_10x-view-data-metrics.Rmd
    Untracked:  analysis/TODO_10x-view-species-assignments.Rmd
    Untracked:  analysis/TODO_10x-view-validation-data.Rmd
    Untracked:  analysis/v2/
    Untracked:  code/FindMarkers.R
    Untracked:  code/FindMarkers.sh
    Untracked:  code/FindMarkers_0.1.err
    Untracked:  code/FindMarkers_0.1.out
    Untracked:  code/FindMarkers_0.25.err
    Untracked:  code/FindMarkers_0.25.out
    Untracked:  code/FindMarkers_0.5.err
    Untracked:  code/FindMarkers_0.5.out
    Untracked:  code/FindMarkers_0.75.err
    Untracked:  code/FindMarkers_0.75.out
    Untracked:  code/FindMarkers_1.err
    Untracked:  code/FindMarkers_1.out
    Untracked:  code/GoM.R
    Untracked:  code/GoM.sh
    Untracked:  code/fastTopics.R
    Untracked:  code/fastTopics.sh
    Untracked:  code/fastTopicsBC.R
    Untracked:  code/fastTopicsBC.sh
    Untracked:  code/fastTopicsBC_3_tot.err
    Untracked:  code/fastTopicsBC_3_tot.out
    Untracked:  code/fastTopicsBC_4_tot.err
    Untracked:  code/fastTopicsBC_4_tot.out
    Untracked:  code/fastTopicsBC_5_tot.err
    Untracked:  code/fastTopicsBC_5_tot.out
    Untracked:  code/fastTopicsBC_6_tot.err
    Untracked:  code/fastTopicsBC_6_tot.out
    Untracked:  code/fastTopicsBC_7_tot.err
    Untracked:  code/fastTopicsBC_7_tot.out
    Untracked:  code/fastTopicsSpp.R
    Untracked:  code/fastTopicsSpp.sh
    Untracked:  code/fastTopicsSppBC.R
    Untracked:  code/fastTopicsSppBC.sh
    Untracked:  code/fastTopicsSppBC_3_tot.err
    Untracked:  code/fastTopicsSppBC_3_tot.out
    Untracked:  code/fastTopicsSppBC_4_tot.err
    Untracked:  code/fastTopicsSppBC_4_tot.out
    Untracked:  code/fastTopicsSppBC_5_tot.err
    Untracked:  code/fastTopicsSppBC_5_tot.out
    Untracked:  code/fastTopicsSppBC_6_tot.err
    Untracked:  code/fastTopicsSppBC_6_tot.out
    Untracked:  code/fastTopicsSppBC_7_tot.err
    Untracked:  code/fastTopicsSppBC_7_tot.out
    Untracked:  code/fastTopicsSpp_3_tot.err
    Untracked:  code/fastTopicsSpp_3_tot.out
    Untracked:  code/fastTopicsSpp_4_tot.err
    Untracked:  code/fastTopicsSpp_4_tot.out
    Untracked:  code/fastTopicsSpp_5_tot.err
    Untracked:  code/fastTopicsSpp_5_tot.out
    Untracked:  code/fastTopicsSpp_6_tot.err
    Untracked:  code/fastTopicsSpp_6_tot.out
    Untracked:  code/fastTopicsSpp_7_tot.err
    Untracked:  code/fastTopicsSpp_7_tot.out
    Untracked:  code/fastTopics_3_tot.err
    Untracked:  code/fastTopics_3_tot.out
    Untracked:  code/fastTopics_4_tot.err
    Untracked:  code/fastTopics_4_tot.out
    Untracked:  code/fastTopics_5_tot.err
    Untracked:  code/fastTopics_5_tot.out
    Untracked:  code/fastTopics_6_tot.err
    Untracked:  code/fastTopics_6_tot.out
    Untracked:  code/fastTopics_7_tot.err
    Untracked:  code/fastTopics_7_tot.out
    Untracked:  code/v2/
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/alizarin-red-quant-total.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/
    Untracked:  data/facs-MSC-collection-summary-sample-avg.csv
    Untracked:  data/facs-MSC-collection-summary-species-avg.csv
    Untracked:  data/facs-MSC-collection-summary-total.csv
    Untracked:  data/gom-data/
    Untracked:  data/oil-red-quant-total.csv
    Untracked:  data/scrna-batch-reformat.csv
    Untracked:  data/scrna-batch.csv
    Untracked:  output/data-qc-post-subsets-batchcor.pdf
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterC.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL.pdf
    Untracked:  output/data-qc-prefilter.pdf
    Untracked:  output/markers0.05-I.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05-M.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05-O.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.1.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.25.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.5.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers1.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-I.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-M.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-O.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/v1/
    Untracked:  output/v2/

Unstaged changes:
    Deleted:    analysis/10x-data-4-classification-t2.Rmd
    Modified:   analysis/10x-data-4-classification-tot.Rmd
    Deleted:    analysis/10x-data-4-classification.Rmd
    Deleted:    analysis/10x-data-5-diffexp.Rmd
    Deleted:    analysis/10x-data-6-conservexp.Rmd
    Deleted:    analysis/10x-data-6-cormotif-t2.Rmd
    Deleted:    analysis/10x-data-6-cormotif-tot.Rmd
    Deleted:    analysis/10x-data-6-variance-t2.Rmd
    Deleted:    analysis/10x-data-6-variance-tot.Rmd
    Deleted:    analysis/10x-data-7-enrichment.Rmd
    Deleted:    analysis/10x-data-8-batch-correlation.Rmd
    Modified:   analysis/10x-plot-classification-adhoc.Rmd
    Modified:   analysis/10x-plot-classification.Rmd
    Deleted:    analysis/10x-plot-diffexp.Rmd
    Deleted:    analysis/10x-plot-replicate-correlation-t2.Rmd
    Deleted:    analysis/10x-plot-replicate-correlation-tot.Rmd
    Deleted:    analysis/10x-plot-utf1-tac3.Rmd
    Deleted:    analysis/10x-view-batch-effects.Rmd
    Deleted:    analysis/10x-view-data-metrics.Rmd
    Deleted:    analysis/10x-view-species-assignments.Rmd
    Deleted:    analysis/10x-view-validation-data.Rmd
    Deleted:    analysis/asbmr2020-plots.Rmd
    Deleted:    analysis/ashg2020-plots.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/temp-code.Rmd
    Deleted:    analysis/v1/index.Rmd
    Deleted:    analysis/v1/license.Rmd
    Modified:   code/DEdream.R
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_FALSE.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_stage.out
    Modified:   code/DEdreamsubset.R
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_stage.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-data-4-classification-tot.Rmd</code>) and HTML (<code>docs/10x-data-4-classification-tot.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/325664d831fec2328643e91fec129a88f26fde72/analysis/10x-data-4-classification-tot.Rmd" target="_blank">325664d</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-05-18
</td>
<td>
fixed ad hoc gene 4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/d0f47b3d5ec28955e5891be68ad9cddd4e8e4087/analysis/10x-data-4-classification-tot.Rmd" target="_blank">d0f47b3</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-05-13
</td>
<td>
updated classifications
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e6ab6ebde72ba993a4e3964d0bc4bb696872e3e4/analysis/10x-data-4-classification-tot.Rmd" target="_blank">e6ab6eb</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-10-05
</td>
<td>
updated cormotif
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/46b9c5ac4448371b8aa6cdf24a41d4516100d316/analysis/10x-data-4-classification-tot.Rmd" target="_blank">46b9c5a</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-28
</td>
<td>
Updated utf1-tac3 code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/d0731f9fd132fcf8d193d7967db1081128618482/analysis/10x-data-4-classification-tot.Rmd" target="_blank">d0731f9</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-26
</td>
<td>
Finalized cell assignment scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/83481d142300f9872471ed1da959f9036e9e4428/analysis/10x-data-4-classification-tot.Rmd" target="_blank">83481d1</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-26
</td>
<td>
Updated tot cell assignments
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2b6d922b2dc3306305ced978a114fdf67d52c1d8/analysis/10x-data-4-classification-tot.Rmd" target="_blank">2b6d922</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-23
</td>
<td>
Updated cell classifications for t2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/0373ccf854f59faa7e436b79cc680b8a362b00d8/analysis/10x-data-4-classification-tot.Rmd" target="_blank">0373ccf</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-22
</td>
<td>
Updated variance and cormotif analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/750156c90d27b7c7b9b88b5a993c5ae749b9e2d8/analysis/10x-data-4-classification-tot.Rmd" target="_blank">750156c</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-09-20
</td>
<td>
Updated DE analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---cell-classification" class="section level1">
<h1>10X Data Cellranger Analysis - Cell Classification</h1>
<p>Define and assign cell type classifications. Three methods considered:</p>
<ul>
<li>unsupervised clustering (Seurat)</li>
<li>unsupervised topic modeling (Grades of Membership analysis)</li>
<li>ad hoc assignments (using known marker gene expression patterns)</li>
</ul>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(pheatmap)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)
library(fastTopics)</code></pre>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;
data &lt;- readRDS(scrna)</code></pre>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)

data</code></pre>
<pre><code>An object of class Seurat 
37859 features across 101000 samples within 2 assays 
Active assay: integrated (18482 features)
 1 other assay present: RNA
 2 dimensional reductions calculated: pca, umap</code></pre>
<pre class="r"><code>head(data@meta.data)</code></pre>
<pre><code>                         orig.ident nCount_RNA nFeature_RNA Species
H1.I_AAACCCAAGCACCCAC SeuratProject      20121         4426   Human
H1.I_AAACCCACACAAGTTC SeuratProject      44970         6385   Human
H1.I_AAACCCAGTTATGGTC SeuratProject      24631         4920   Human
H1.I_AAACCCATCGTGGAAG SeuratProject      33783         5070   Human
H1.I_AAACCCATCTCGAGTA SeuratProject      16457         3885   Human
H1.I_AAACGAAAGGATCATA SeuratProject      27102         5232   Human
                      percent.mt Collection Cell.Type  Pair Individual
H1.I_AAACCCAAGCACCCAC   5.109090          1      iPSC H1+C1         H1
H1.I_AAACCCACACAAGTTC   7.011341          1      iPSC H1+C1         H1
H1.I_AAACCCAGTTATGGTC   5.736673          1      iPSC H1+C1         H1
H1.I_AAACCCATCGTGGAAG   6.772637          1      iPSC H1+C1         H1
H1.I_AAACCCATCTCGAGTA   5.231816          1      iPSC H1+C1         H1
H1.I_AAACGAAAGGATCATA   6.254151          1      iPSC H1+C1         H1
                      Replicate MSCmedia Sample Sex Age DateCollection
H1.I_AAACCCAAGCACCCAC         1       3B   H1-I   M  22       6/3/2019
H1.I_AAACCCACACAAGTTC         1       3B   H1-I   M  22       6/3/2019
H1.I_AAACCCAGTTATGGTC         1       3B   H1-I   M  22       6/3/2019
H1.I_AAACCCATCGTGGAAG         1       3B   H1-I   M  22       6/3/2019
H1.I_AAACCCATCTCGAGTA         1       3B   H1-I   M  22       6/3/2019
H1.I_AAACGAAAGGATCATA         1       3B   H1-I   M  22       6/3/2019
                      Number6W PrewashCount PrewashViability PostwashCount
H1.I_AAACCCAAGCACCCAC        3      9228000             0.89         7e+05
H1.I_AAACCCACACAAGTTC        3      9228000             0.89         7e+05
H1.I_AAACCCAGTTATGGTC        3      9228000             0.89         7e+05
H1.I_AAACCCATCGTGGAAG        3      9228000             0.89         7e+05
H1.I_AAACCCATCTCGAGTA        3      9228000             0.89         7e+05
H1.I_AAACGAAAGGATCATA        3      9228000             0.89         7e+05
                      PostwashViability PostmixCount PostmixViability
H1.I_AAACCCAAGCACCCAC               0.9       650000              0.9
H1.I_AAACCCACACAAGTTC               0.9       650000              0.9
H1.I_AAACCCAGTTATGGTC               0.9       650000              0.9
H1.I_AAACCCATCGTGGAAG               0.9       650000              0.9
H1.I_AAACCCATCTCGAGTA               0.9       650000              0.9
H1.I_AAACGAAAGGATCATA               0.9       650000              0.9
                      DatecDNA DateLibrary SequencingBatch CD90_Control
H1.I_AAACCCAAGCACCCAC 6/3/2019   6/10/2019               1       0.0021
H1.I_AAACCCACACAAGTTC 6/3/2019   6/10/2019               1       0.0021
H1.I_AAACCCAGTTATGGTC 6/3/2019   6/10/2019               1       0.0021
H1.I_AAACCCATCGTGGAAG 6/3/2019   6/10/2019               1       0.0021
H1.I_AAACCCATCTCGAGTA 6/3/2019   6/10/2019               1       0.0021
H1.I_AAACGAAAGGATCATA 6/3/2019   6/10/2019               1       0.0021
                      CD73_Control CD105_Control
H1.I_AAACCCAAGCACCCAC       0.0025   0.001166667
H1.I_AAACCCACACAAGTTC       0.0025   0.001166667
H1.I_AAACCCAGTTATGGTC       0.0025   0.001166667
H1.I_AAACCCATCGTGGAAG       0.0025   0.001166667
H1.I_AAACCCATCTCGAGTA       0.0025   0.001166667
H1.I_AAACGAAAGGATCATA       0.0025   0.001166667
                      CD45.CD34.CD11b.CD19.HLA.DR_Control   CD90
H1.I_AAACCCAAGCACCCAC                              0.0017 0.6572
H1.I_AAACCCACACAAGTTC                              0.0017 0.6572
H1.I_AAACCCAGTTATGGTC                              0.0017 0.6572
H1.I_AAACCCATCGTGGAAG                              0.0017 0.6572
H1.I_AAACCCATCTCGAGTA                              0.0017 0.6572
H1.I_AAACGAAAGGATCATA                              0.0017 0.6572
                             CD73        CD105 CD45.CD34.CD11b.CD19.HLA.DR
H1.I_AAACCCAAGCACCCAC 0.001266667 0.0007333333                 0.001966667
H1.I_AAACCCACACAAGTTC 0.001266667 0.0007333333                 0.001966667
H1.I_AAACCCAGTTATGGTC 0.001266667 0.0007333333                 0.001966667
H1.I_AAACCCATCGTGGAAG 0.001266667 0.0007333333                 0.001966667
H1.I_AAACCCATCTCGAGTA 0.001266667 0.0007333333                 0.001966667
H1.I_AAACGAAAGGATCATA 0.001266667 0.0007333333                 0.001966667
                      Alizarin OilRed  Stage Complexity      S.Score
H1.I_AAACCCAAGCACCCAC       NA     NA Time 0  0.6576382 -0.079583719
H1.I_AAACCCACACAAGTTC       NA     NA Time 0  0.8477620 -0.051582337
H1.I_AAACCCAGTTATGGTC       NA     NA Time 0  0.6995169  0.006639814
H1.I_AAACCCATCGTGGAAG       NA     NA Time 0  0.8236903  0.095840891
H1.I_AAACCCATCTCGAGTA       NA     NA Time 0  0.6269596  0.113270570
H1.I_AAACGAAAGGATCATA       NA     NA Time 0  0.7143336 -0.067822836
                         G2M.Score Phase integrated_snn_res.1
H1.I_AAACCCAAGCACCCAC -0.053363880    G1                    8
H1.I_AAACCCACACAAGTTC -0.067659885    G1                   17
H1.I_AAACCCAGTTATGGTC  0.202169108   G2M                    4
H1.I_AAACCCATCGTGGAAG -0.196044816     S                   11
H1.I_AAACCCATCTCGAGTA -0.180149719     S                    1
H1.I_AAACGAAAGGATCATA  0.004518702   G2M                    4
                      seurat_clusters integrated_snn_res.0.75
H1.I_AAACCCAAGCACCCAC               1                       1
H1.I_AAACCCACACAAGTTC               1                       7
H1.I_AAACCCAGTTATGGTC               1                       3
H1.I_AAACCCATCGTGGAAG               1                       7
H1.I_AAACCCATCTCGAGTA               1                       1
H1.I_AAACGAAAGGATCATA               1                       3
                      integrated_snn_res.0.5 integrated_snn_res.0.25
H1.I_AAACCCAAGCACCCAC                      2                       2
H1.I_AAACCCACACAAGTTC                      9                       3
H1.I_AAACCCAGTTATGGTC                      3                       3
H1.I_AAACCCATCGTGGAAG                      9                       2
H1.I_AAACCCATCTCGAGTA                      2                       2
H1.I_AAACGAAAGGATCATA                      3                       3
                      integrated_snn_res.0.1 integrated_snn_res.0.05
H1.I_AAACCCAAGCACCCAC                      0                       1
H1.I_AAACCCACACAAGTTC                      0                       1
H1.I_AAACCCAGTTATGGTC                      0                       1
H1.I_AAACCCATCGTGGAAG                      0                       1
H1.I_AAACCCATCTCGAGTA                      0                       1
H1.I_AAACGAAAGGATCATA                      0                       1
                      Cluster0.05 Cluster0.25 Cluster0.50
H1.I_AAACCCAAGCACCCAC     iPSC.c1     iPSC.c1     iPSC.c1
H1.I_AAACCCACACAAGTTC     iPSC.c1     iPSC.c2     iPSC.c3
H1.I_AAACCCAGTTATGGTC     iPSC.c1     iPSC.c2     iPSC.c2
H1.I_AAACCCATCGTGGAAG     iPSC.c1     iPSC.c1     iPSC.c3
H1.I_AAACCCATCTCGAGTA     iPSC.c1     iPSC.c1     iPSC.c1
H1.I_AAACGAAAGGATCATA     iPSC.c1     iPSC.c2     iPSC.c2
                      AdHoc.Assign.thresh0 AdHoc.Assign.threshN0.01
H1.I_AAACCCAAGCACCCAC                 iPSC                     iPSC
H1.I_AAACCCACACAAGTTC                 iPSC                     iPSC
H1.I_AAACCCAGTTATGGTC                 iPSC                     iPSC
H1.I_AAACCCATCGTGGAAG                 iPSC                     iPSC
H1.I_AAACCCATCTCGAGTA                 iPSC                     iPSC
H1.I_AAACGAAAGGATCATA                 iPSC                     iPSC
                      AdHoc.Assign.threshN0.05 AdHoc.Assign.threshN0.1
H1.I_AAACCCAAGCACCCAC                    other                   other
H1.I_AAACCCACACAAGTTC                     iPSC                    iPSC
H1.I_AAACCCAGTTATGGTC                     iPSC                    iPSC
H1.I_AAACCCATCGTGGAAG                     iPSC                    iPSC
H1.I_AAACCCATCTCGAGTA                     iPSC                    iPSC
H1.I_AAACGAAAGGATCATA                     iPSC                    iPSC
                      AdHoc.Assign.thresh0.01 AdHoc.Assign.thresh0.05
H1.I_AAACCCAAGCACCCAC                    iPSC                    iPSC
H1.I_AAACCCACACAAGTTC                    iPSC                    iPSC
H1.I_AAACCCAGTTATGGTC                    iPSC                    iPSC
H1.I_AAACCCATCGTGGAAG                    iPSC                    iPSC
H1.I_AAACCCATCTCGAGTA                    iPSC                    iPSC
H1.I_AAACGAAAGGATCATA                    iPSC                    iPSC
                      AdHoc.Assign.thresh0.1
H1.I_AAACCCAAGCACCCAC                   iPSC
H1.I_AAACCCACACAAGTTC                   iPSC
H1.I_AAACCCAGTTATGGTC                   iPSC
H1.I_AAACCCATCGTGGAAG                   iPSC
H1.I_AAACCCATCTCGAGTA                   iPSC
H1.I_AAACGAAAGGATCATA                   iPSC
                      OstAdHoc.Assign.x.8gene.thresh0
H1.I_AAACCCAAGCACCCAC                  not osteogenic
H1.I_AAACCCACACAAGTTC                  not osteogenic
H1.I_AAACCCAGTTATGGTC                  not osteogenic
H1.I_AAACCCATCGTGGAAG                  not osteogenic
H1.I_AAACCCATCTCGAGTA                  not osteogenic
H1.I_AAACGAAAGGATCATA                  not osteogenic
                      OstAdHoc.Assign.8gene.thresh0
H1.I_AAACCCAAGCACCCAC                not osteogenic
H1.I_AAACCCACACAAGTTC                not osteogenic
H1.I_AAACCCAGTTATGGTC                not osteogenic
H1.I_AAACCCATCGTGGAAG                not osteogenic
H1.I_AAACCCATCTCGAGTA                not osteogenic
H1.I_AAACGAAAGGATCATA                not osteogenic
                      OstAdHoc.Assign.x.8gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                      not osteogenic
H1.I_AAACCCACACAAGTTC                      not osteogenic
H1.I_AAACCCAGTTATGGTC                      not osteogenic
H1.I_AAACCCATCGTGGAAG                      not osteogenic
H1.I_AAACCCATCTCGAGTA                      not osteogenic
H1.I_AAACGAAAGGATCATA                      not osteogenic
                      OstAdHoc.Assign.8gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                    not osteogenic
H1.I_AAACCCACACAAGTTC                    not osteogenic
H1.I_AAACCCAGTTATGGTC                    not osteogenic
H1.I_AAACCCATCGTGGAAG                    not osteogenic
H1.I_AAACCCATCTCGAGTA                    not osteogenic
H1.I_AAACGAAAGGATCATA                    not osteogenic
                      OstAdHoc.Assign.x.8gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                     not osteogenic
H1.I_AAACCCACACAAGTTC                     not osteogenic
H1.I_AAACCCAGTTATGGTC                     not osteogenic
H1.I_AAACCCATCGTGGAAG                     not osteogenic
H1.I_AAACCCATCTCGAGTA                     not osteogenic
H1.I_AAACGAAAGGATCATA                     not osteogenic
                      OstAdHoc.Assign.8gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                   not osteogenic
H1.I_AAACCCACACAAGTTC                   not osteogenic
H1.I_AAACCCAGTTATGGTC                   not osteogenic
H1.I_AAACCCATCGTGGAAG                   not osteogenic
H1.I_AAACCCATCTCGAGTA                   not osteogenic
H1.I_AAACGAAAGGATCATA                   not osteogenic
                      OstAdHoc.Assign.x.6gene.thresh0
H1.I_AAACCCAAGCACCCAC                  not osteogenic
H1.I_AAACCCACACAAGTTC                  not osteogenic
H1.I_AAACCCAGTTATGGTC                  not osteogenic
H1.I_AAACCCATCGTGGAAG                  not osteogenic
H1.I_AAACCCATCTCGAGTA                  not osteogenic
H1.I_AAACGAAAGGATCATA                  not osteogenic
                      OstAdHoc.Assign.6gene.thresh0
H1.I_AAACCCAAGCACCCAC                not osteogenic
H1.I_AAACCCACACAAGTTC                not osteogenic
H1.I_AAACCCAGTTATGGTC                not osteogenic
H1.I_AAACCCATCGTGGAAG                not osteogenic
H1.I_AAACCCATCTCGAGTA                not osteogenic
H1.I_AAACGAAAGGATCATA                not osteogenic
                      OstAdHoc.Assign.x.6gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                      not osteogenic
H1.I_AAACCCACACAAGTTC                      not osteogenic
H1.I_AAACCCAGTTATGGTC                      not osteogenic
H1.I_AAACCCATCGTGGAAG                      not osteogenic
H1.I_AAACCCATCTCGAGTA                      not osteogenic
H1.I_AAACGAAAGGATCATA                      not osteogenic
                      OstAdHoc.Assign.6gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                    not osteogenic
H1.I_AAACCCACACAAGTTC                    not osteogenic
H1.I_AAACCCAGTTATGGTC                    not osteogenic
H1.I_AAACCCATCGTGGAAG                    not osteogenic
H1.I_AAACCCATCTCGAGTA                    not osteogenic
H1.I_AAACGAAAGGATCATA                    not osteogenic
                      OstAdHoc.Assign.x.6gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                     not osteogenic
H1.I_AAACCCACACAAGTTC                     not osteogenic
H1.I_AAACCCAGTTATGGTC                     not osteogenic
H1.I_AAACCCATCGTGGAAG                     not osteogenic
H1.I_AAACCCATCTCGAGTA                     not osteogenic
H1.I_AAACGAAAGGATCATA                     not osteogenic
                      OstAdHoc.Assign.6gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                   not osteogenic
H1.I_AAACCCACACAAGTTC                   not osteogenic
H1.I_AAACCCAGTTATGGTC                   not osteogenic
H1.I_AAACCCATCGTGGAAG                   not osteogenic
H1.I_AAACCCATCTCGAGTA                   not osteogenic
H1.I_AAACGAAAGGATCATA                   not osteogenic
                      OstAdHoc.Assign.x.4gene.thresh0
H1.I_AAACCCAAGCACCCAC                  not osteogenic
H1.I_AAACCCACACAAGTTC                  not osteogenic
H1.I_AAACCCAGTTATGGTC                  not osteogenic
H1.I_AAACCCATCGTGGAAG                  not osteogenic
H1.I_AAACCCATCTCGAGTA                  not osteogenic
H1.I_AAACGAAAGGATCATA                  not osteogenic
                      OstAdHoc.Assign.4gene.thresh0
H1.I_AAACCCAAGCACCCAC                not osteogenic
H1.I_AAACCCACACAAGTTC                not osteogenic
H1.I_AAACCCAGTTATGGTC                not osteogenic
H1.I_AAACCCATCGTGGAAG                not osteogenic
H1.I_AAACCCATCTCGAGTA                not osteogenic
H1.I_AAACGAAAGGATCATA                not osteogenic
                      OstAdHoc.Assign.x.4gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                      not osteogenic
H1.I_AAACCCACACAAGTTC                      not osteogenic
H1.I_AAACCCAGTTATGGTC                      not osteogenic
H1.I_AAACCCATCGTGGAAG                      not osteogenic
H1.I_AAACCCATCTCGAGTA                      not osteogenic
H1.I_AAACGAAAGGATCATA                      not osteogenic
                      OstAdHoc.Assign.4gene.threshN0.01
H1.I_AAACCCAAGCACCCAC                    not osteogenic
H1.I_AAACCCACACAAGTTC                    not osteogenic
H1.I_AAACCCAGTTATGGTC                    not osteogenic
H1.I_AAACCCATCGTGGAAG                    not osteogenic
H1.I_AAACCCATCTCGAGTA                    not osteogenic
H1.I_AAACGAAAGGATCATA                    not osteogenic
                      OstAdHoc.Assign.x.4gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                     not osteogenic
H1.I_AAACCCACACAAGTTC                     not osteogenic
H1.I_AAACCCAGTTATGGTC                     not osteogenic
H1.I_AAACCCATCGTGGAAG                     not osteogenic
H1.I_AAACCCATCTCGAGTA                     not osteogenic
H1.I_AAACGAAAGGATCATA                     not osteogenic
                      OstAdHoc.Assign.4gene.thresh0.01
H1.I_AAACCCAAGCACCCAC                   not osteogenic
H1.I_AAACCCACACAAGTTC                   not osteogenic
H1.I_AAACCCAGTTATGGTC                   not osteogenic
H1.I_AAACCCATCGTGGAAG                   not osteogenic
H1.I_AAACCCATCTCGAGTA                   not osteogenic
H1.I_AAACGAAAGGATCATA                   not osteogenic
                      Individual.Replicate
H1.I_AAACCCAAGCACCCAC                 H1.1
H1.I_AAACCCACACAAGTTC                 H1.1
H1.I_AAACCCAGTTATGGTC                 H1.1
H1.I_AAACCCATCGTGGAAG                 H1.1
H1.I_AAACCCATCTCGAGTA                 H1.1
H1.I_AAACGAAAGGATCATA                 H1.1</code></pre>
<pre class="r"><code>DimPlot(data,
        group.by=&quot;Stage&quot;,
        reduction=&quot;umap&quot;,
        cols=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="seurat-clustering" class="section level2">
<h2>Seurat Clustering</h2>
<pre><code>                           tot.intNo0.reg</code></pre>
<p>data, k=40, res=1.00 clusters: 26 data, k=40, res=0.75 clusters: 20 data, k=40, res=0.50 clusters: 16 data, k=40, res=0.25 clusters: 12 data, k=40, res=0.10 clusters: 8 data, k=40, res=0.05 clusters: 6</p>
<pre class="r"><code>#use dims that explaim more than 0.1% of variance
pva &lt;- data@reductions$pca@stdev^2/data@reductions$pca@misc$total.variance
ndim &lt;- length(which(pva&gt;=0.001))
print(ndim)

#Identify Clusters (increased resolution identifies greater number of clusters)
data &lt;- FindNeighbors(data, k.param=40, dims=1:ndim)
data &lt;- FindClusters(data, resolution=1.00)
data &lt;- FindClusters(data, resolution=0.75)
data &lt;- FindClusters(data, resolution=0.50)
data &lt;- FindClusters(data, resolution=0.25)
data &lt;- FindClusters(data, resolution=0.10)
data &lt;- FindClusters(data, resolution=0.05)

rm(pva,ndim)</code></pre>
<pre class="r"><code>saveRDS(data, file=&quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;)</code></pre>
<pre class="r"><code>#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)</code></pre>
<p>Check the number of cells per cluster.</p>
<pre class="r"><code>table(data@meta.data$integrated_snn_res.1)
table(data@meta.data$integrated_snn_res.0.75)
table(data@meta.data$integrated_snn_res.0.5)
table(data@meta.data$integrated_snn_res.0.25)
table(data@meta.data$integrated_snn_res.0.1)
table(data@meta.data$integrated_snn_res.0.05)
#table(data@meta.data$seurat_clusters)</code></pre>
<p>Calculate numbers of cells in each cluster (resolution=0.05) given particular parameters.</p>
<pre><code>                      cluster 0 | cluster 1 | cluster 2 | cluster 3 | cluster 4 | cluster 5 |</code></pre>
<p>tot.intNo0.reg.res.0.05 36057 | 31196 | 21950 | 7422 | 3697 | 678 | Human: 16990 | 14372 | 8751 | 4990 | 1859 | 448 | Chimp: 19067 | 16824 | 13199 | 2432 | 1838 | 230 | Time 0: 6 | 31194 | 8 | 6 | 3696 | 678 | Time 1: 35236 | 2 | 707 | 1282 | 1 | 0 | Time 2: 815 | 0 | 21235 | 6134 | 0 | 0 | Human Time 0: 6 | 14371 | 8 | 5 | 1858 | 448 | Human Time 1: 16821 | 1 | 574 | 924 | 1 | 0 | Human Time 2: 163 | 0 | 8169 | 4061 | 0 | 0 | Chimp Time 0: 0 | 16823 | 0 | 1 | 1838 | 230 | Chimp Time 1: 18415 | 1 | 133 | 358 | 0 | 0 | Chimp Time 2: 652 | 0 | 13066 | 2073 | 0 | 0 |</p>
<pre class="r"><code>table(data@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=Species==&quot;Human&quot;)@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=Species==&quot;Chimp&quot;)@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=Stage==&quot;Time 0&quot;)@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=Stage==&quot;Time 1&quot;)@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=Stage==&quot;Time 2&quot;)@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$integrated_snn_res.0.05)
table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$integrated_snn_res.0.05)</code></pre>
<p>Visualize clusters (see 10x-plot-calssification-clusters.Rmd for figures)</p>
<pre class="r"><code>x &lt;- &quot;integrated_snn_res.0.05&quot;
#x &lt;- &quot;integrated_snn_res.0.25&quot;
#x &lt;- &quot;integrated_snn_res.0.5&quot;

CombinePlots(plots=list((DimPlot(data, group.by=x, reduction=&quot;pca&quot;, dims=c(1,2)) + labs(x=&quot;PC1&quot;,y=&quot;PC2&quot;)),
                        (DimPlot(data, group.by=x, reduction=&quot;pca&quot;, dims=c(3,4)) + labs(x=&quot;PC3&quot;,y=&quot;PC4&quot;)),
                        (DimPlot(data, group.by=x, reduction=&quot;pca&quot;, dims=c(5,6)) + labs(x=&quot;PC5&quot;,y=&quot;PC6&quot;)),
                        (DimPlot(data, group.by=x, reduction=&quot;pca&quot;, dims=c(7,8)) + labs(x=&quot;PC7&quot;,y=&quot;PC8&quot;)),
                        (DimPlot(data, group.by=x, reduction=&quot;pca&quot;, dims=c(9,10)) + labs(x=&quot;PC9&quot;,y=&quot;PC10&quot;))), ncol=5)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(data, group.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;), split.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;), split.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))),
             ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;), split.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;), split.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;), split.by=x, reduction=&quot;umap&quot;) + labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))),
             ncol=1)</code></pre>
<pre><code>Warning: Graphs cannot be vertically aligned unless the axis parameter is
set. Placing graphs unaligned.</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Define new metadata label for clusters (resolution=0.05).</p>
<pre class="r"><code>#resolution=0.05
data &lt;- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name=&quot;Cluster0.05&quot;)
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;1&quot;)] &lt;- &quot;iPSC.c1&quot;
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;4&quot;)] &lt;- &quot;iPSC.c2&quot;
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;5&quot;)] &lt;- &quot;iPSC.c3&quot;
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;0&quot;)] &lt;- &quot;MSC.c1&quot;
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;2&quot;)] &lt;- &quot;Osteogenic.c1&quot;
data$Cluster0.05[which(data$integrated_snn_res.0.05==&quot;3&quot;)] &lt;- &quot;Osteogenic.c2&quot;

#resolution=0.25
data &lt;- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name=&quot;Cluster0.25&quot;)
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;2&quot;)]  &lt;- &quot;iPSC.c1&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;3&quot;)]  &lt;- &quot;iPSC.c2&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;9&quot;)]  &lt;- &quot;iPSC.c3&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;7&quot;)]  &lt;- &quot;iPSC.c4&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;10&quot;)] &lt;- &quot;iPSC.c5&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;11&quot;)] &lt;- &quot;iPSC.c6&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;0&quot;)]  &lt;- &quot;MSC.c1&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;4&quot;)]  &lt;- &quot;MSC.c2&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;8&quot;)]  &lt;- &quot;MSC.c3&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;1&quot;)]  &lt;- &quot;Osteogenic.c1&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;6&quot;)]  &lt;- &quot;Osteogenic.c2&quot;
data$Cluster0.25[which(data$integrated_snn_res.0.25==&quot;5&quot;)]  &lt;- &quot;Osteogenic.c3&quot;

#resolution=0.50
data &lt;- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name=&quot;Cluster0.50&quot;)
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;2&quot;)]  &lt;- &quot;iPSC.c1&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;3&quot;)]  &lt;- &quot;iPSC.c2&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;9&quot;)]  &lt;- &quot;iPSC.c3&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;12&quot;)] &lt;- &quot;iPSC.c4&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;11&quot;)] &lt;- &quot;iPSC.c5&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;10&quot;)] &lt;- &quot;iPSC.c6&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;14&quot;)] &lt;- &quot;iPSC.c7&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;15&quot;)] &lt;- &quot;iPSC.c8&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;1&quot;)]  &lt;- &quot;MSC.c1&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;4&quot;)]  &lt;- &quot;MSC.c2&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;7&quot;)]  &lt;- &quot;MSC.c3&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;5&quot;)]  &lt;- &quot;MSC.c4&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;0&quot;)]  &lt;- &quot;Osteogenic.c1&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;13&quot;)] &lt;- &quot;Osteogenic.c2&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;8&quot;)]  &lt;- &quot;Osteogenic.c3&quot;
data$Cluster0.50[which(data$integrated_snn_res.0.5==&quot;6&quot;)]  &lt;- &quot;Osteogenic.c4&quot;</code></pre>
<pre class="r"><code>saveRDS(data, file=&quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;)</code></pre>
<p>Make nicer plots of clusters: resolution=0.05</p>
<pre class="r"><code>y &lt;- &quot;Cluster0.05&quot;
#c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)

DimPlot(data,
        group.by=y,
        reduction=&quot;umap&quot;,
        cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(1,2),
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC1&quot;,y=&quot;PC2&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(3,4),
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC3&quot;,y=&quot;PC4&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(5,6),
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC5&quot;,y=&quot;PC6&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(7,8),
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC7&quot;,y=&quot;PC8&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(9,10),
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC9&quot;,y=&quot;PC10&quot;))), ncol=5)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=2)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=3)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make nicer plots of clusters: resolution=0.25</p>
<pre class="r"><code>y &lt;- &quot;Cluster0.25&quot;
#c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)

DimPlot(data,
        group.by=y,
        reduction=&quot;umap&quot;,
        cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(1,2),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC1&quot;,y=&quot;PC2&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(3,4),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC3&quot;,y=&quot;PC4&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(5,6),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC5&quot;,y=&quot;PC6&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(7,8),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC7&quot;,y=&quot;PC8&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(9,10),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC9&quot;,y=&quot;PC10&quot;))), ncol=5)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=2)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=3)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make nicer plots of clusters: resolution=0.50</p>
<pre class="r"><code>y &lt;- &quot;Cluster0.50&quot;
#c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,&quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)

DimPlot(data,
        group.by=y,
        reduction=&quot;umap&quot;,
        cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
               &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
               &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(1,2),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC1&quot;,y=&quot;PC2&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(3,4),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC3&quot;,y=&quot;PC4&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(5,6),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC5&quot;,y=&quot;PC6&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(7,8),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC7&quot;,y=&quot;PC8&quot;)),
                        (DimPlot(data,
                                 group.by=y,
                                 reduction=&quot;pca&quot;,
                                 dims=c(9,10),
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;PC9&quot;,y=&quot;PC10&quot;))), ncol=5)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=2)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;palevioletred1&quot;,&quot;palevioletred3&quot;,&quot;palevioletred4&quot;,&quot;pink3&quot;,&quot;pink4&quot;,&quot;grey40&quot;,&quot;grey&quot;,
                                        &quot;mediumorchid&quot;,
                                        &quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink3&quot;,&quot;pink4&quot;,
                                        &quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=y,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;mediumorchid&quot;,&quot;orchid2&quot;,&quot;orchid4&quot;,&quot;purple&quot;,
                                        &quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=3)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Identify markers defining each cluster (resolution=0.05).</p>
<pre class="r"><code>#Find markers for every cluster compared to all remaining cells, report only the positive ones
markers &lt;- FindAllMarkers(data, only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25)
markers %&gt;% group_by(cluster) %&gt;% top_n(n=5, wt=avg_logFC)
top5 &lt;- markers %&gt;% group_by(cluster) %&gt;% top_n(n=5, wt=avg_logFC)
#DoHeatmap(data, group.by=&quot;Cluster0.05&quot;, features=top5$gene) + NoLegend()
saveRDS(markers, file=&quot;./output/markers0.05.data.filterC.log.indv-cell.intNo0.reg-tot..rds&quot;)

VlnPlot(data, features=top5$gene[1:5],   group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 0 (MSC.c1): MMP1, LGALS1, TAGLN2, VIM, TPM1
VlnPlot(data, features=top5$gene[6:10],  group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 1 (iPSC.c1): CD24, APOE, POU5F1, L1TD1, GAL
VlnPlot(data, features=top5$gene[11:15], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 2 (Osteogenic.c1): NUPR1, COL1A1, CTSD, POSTN, COL1A2
VlnPlot(data, features=top5$gene[16:20], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 3 (Osteogenic.c2): CCND1, MTRNR2L8, NUPR1, TIMP1, CCND2
VlnPlot(data, features=top5$gene[21:25], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 4 (iPSC.c2): TUBB2B, POU5F1, CRABP1, HMGA1, HMGB2
VlnPlot(data, features=top5$gene[26:30], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #cluster 5 (iPSC.c3): tmsb15a, TTYH1, DLK1, PTN, FABP7

#Find markers for groups of clusters to all remaining cells
markers.i &lt;- FindMarkers(data, ident.1=c(1,4,5), ident.2=c(0,2,3), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #ipsc
markers.m &lt;- FindMarkers(data, ident.1=c(0), ident.2=c(1,2,3,4,5), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #msc
markers.o &lt;- FindMarkers(data, ident.1=c(2,3), ident.2=c(0,1,4,5), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #osteo
saveRDS(markers.i, file=&quot;./output/markers0.05-I.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)
saveRDS(markers.m, file=&quot;./output/markers0.05-M.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)
saveRDS(markers.o, file=&quot;./output/markers0.05-O.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

VlnPlot(data, features=rownames(markers.i)[1:5], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #CD24, POU5F1, APOE, TUBB2B, L1TD1
VlnPlot(data, features=rownames(markers.m)[1:5], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #MMP1, LGALS1, TAGLN2, VIM, TPM1
VlnPlot(data, features=rownames(markers.o)[1:5], group.by=&quot;Cluster0.05&quot;, slot=&quot;data&quot;, pt.size=0, ncol=5) #NUPR1, TIMP1, IIFI6, GFBP4, CTSD</code></pre>
<pre class="bash"><code>#use cluster to identify genes defining each cluster
cd /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/code/
sbatch ./FindMarkers.sh 0.1
sbatch ./FindMarkers.sh 0.25
sbatch ./FindMarkers.sh 0.5
sbatch ./FindMarkers.sh 0.75
sbatch ./FindMarkers.sh 1</code></pre>
</div>
<div id="grades-of-membership-topic-modeling---unsupervised" class="section level2">
<h2>Grades of Membership (Topic Modeling - unsupervised)</h2>
<p>Alternative way to look at data structure. A topic model explains each sample (an expression profile) as a mixture of “gene programs” that are estimated from the data. In a topic modeling analysis, the gene expression programs are the topics. Overall, the topic modeling analysis retains the benefits of the clustering analysis while also revealing additional substructure that cannot be identified by standard clustering methods. We are interested in identifying which topics show major differences across groups of cells (what genes/pathways are enriched in these topics).</p>
<ul>
<li>fastTopics</li>
<li>This analytical method implements fast optimization algorithms that fit topic models and Poisson non-negative matrix factorizations to scRNA-seq count data. The methods exploit the close relationship between topic modeling and Poisson non-negative matrix factorization (NMF).</li>
<li>fit_poisson_nmf computes maximum-likelihood estimates of the Poisson NMF parameters and stores these parameters into a loadings matrix (fit<span class="math inline">\(L) and a factors matrix (fit\)</span>F).</li>
<li>poisson2multinom recovers the multinomial topic model as a reparameterization of the Poisson NMF model</li>
</ul>
<p>Reference: <a href="https://github.com/stephenslab/fastTopics" class="uri">https://github.com/stephenslab/fastTopics</a></p>
<pre class="r"><code>#Load libraries
library(Seurat)
library(CountClust)
library(classtpx)
library(fastTopics)

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)

#Prepare data (unintegrated count data)
counts &lt;- as.matrix(data@assays$RNA@counts) #do not use integrated data
counts &lt;- counts[rowSums(counts&gt;0)&gt;0,]      #remove genes with 0 counts across cells
counts &lt;- t(as.matrix(counts))              #fix matrix orientation (barcodes x features)
rm(data)

#Batch correct counts (collection replicates only, not species)
#labelBatch1 &lt;- as.factor(paste0(data@meta.data$Pair,data@meta.data$Replicate)[which(rownames(data@meta.data) %in% rownames(counts))])
#countsBC1 &lt;- BatchCorrectedCounts(counts, labelBatch1, use_parallel=FALSE)
#rm(data)

#Fit fastTopics model and save data
fit &lt;- fit_poisson_nmf(counts, k=7, numiter=100)
saveRDS(fit, file=&quot;./data/gom-data/topics7.log.indv-cell.int19k.reg-tot.rds&quot;)
#fit &lt;- fit_poisson_nmf(countsBC1, k=3, numiter=100)
#saveRDS(fit, file=&quot;./data/gom-data/topics3.log.indv-cell.int19k.reg-tot.bc1.rds&quot;)</code></pre>
<pre class="bash"><code>#easties to run these scripts on the cluster
cd /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/code/
sbatch ./fastTopics.sh 3 tot
sbatch ./fastTopics.sh 4 tot
sbatch ./fastTopics.sh 5 tot
sbatch ./fastTopics.sh 6 tot
sbatch ./fastTopics.sh 7 tot
#sbatch ./fastTopics.sh 10 tot
#sbatch ./fastTopics.sh 15 tot
#sbatch ./fastTopics.sh 20 tot
#sbatch ./fastTopics.sh 30 tot
#sbatch ./fastTopics.sh 50 tot
sbatch ./fastTopicsSpp.sh 3 tot
sbatch ./fastTopicsSpp.sh 4 tot
sbatch ./fastTopicsSpp.sh 5 tot
sbatch ./fastTopicsSpp.sh 6 tot
sbatch ./fastTopicsSpp.sh 7 tot
#sbatch ./fastTopicsSpp.sh 10 tot
#sbatch ./fastTopicsSpp.sh 15 tot
#sbatch ./fastTopicsSpp.sh 20 tot
#sbatch ./fastTopicsSpp.sh 30 tot
#sbatch ./fastTopicsSpp.sh 50 tot
sbatch ./fastTopicsBC.sh 3 tot
sbatch ./fastTopicsBC.sh 4 tot
sbatch ./fastTopicsBC.sh 5 tot
sbatch ./fastTopicsBC.sh 6 tot
sbatch ./fastTopicsBC.sh 7 tot
#sbatch ./fastTopicsBC.sh 10 tot
#sbatch ./fastTopicsBC.sh 15 tot
#sbatch ./fastTopicsBC.sh 20 tot
#sbatch ./fastTopicsBC.sh 30 tot
#sbatch ./fastTopicsBC.sh 50 tot
sbatch ./fastTopicsSppBC.sh 3 tot
sbatch ./fastTopicsSppBC.sh 4 tot
sbatch ./fastTopicsSppBC.sh 5 tot
sbatch ./fastTopicsSppBC.sh 6 tot
sbatch ./fastTopicsSppBC.sh 7 tot
#sbatch ./fastTopicsSppBC.sh 10 tot
#sbatch ./fastTopicsSppBC.sh 15 tot
#sbatch ./fastTopicsSppBC.sh 20 tot
#sbatch ./fastTopicsSppBC.sh 30 tot
#sbatch ./fastTopicsSppBC.sh 50 tot</code></pre>
<pre class="r"><code>#GoM Functions

plot_structure_plot &lt;- function(omega, labels){
     annotation &lt;- data.frame(sample_id=paste0(&quot;X&quot;, c(1:NROW(omega))), tissue_label=labels)
     rownames(omega) &lt;- annotation$sample_id
     StructureGGplot(omega=omega, annotation=annotation, order_sample=TRUE,
                     axis_tick=list(axis_ticks_length = .1,
                                    axis_ticks_lwd_y = .1,
                                    axis_ticks_lwd_x = .1,
                                    axis_label_size = 7,
                                    axis_label_face = &quot;bold&quot;))
     }

make_topic_plots &lt;- function(omega, labels, angle) {
  i=1
  omega &lt;- as.data.frame(omega)
  colNames &lt;- names(omega)
  omega &lt;- cbind(omega,labels)
  names(omega)
  p &lt;- list()
  for (k in colNames) {
    p[[i]] &lt;- ggplot(data=omega, mapping=aes(x=labels)) +
      geom_violin(mapping=aes_string(y=k),width=1,fill=&quot;lightgrey&quot;,color=&quot;lightgrey&quot;) +
      geom_boxplot(mapping=aes_string(y=k),width=0.1,outlier.shape=NA, alpha=0.2) +
      labs(y=paste0(&quot;Topic &quot;, k, &quot; value&quot;), x=paste0(&quot;&quot;)) +
      theme_classic() +
      theme(axis.text.x=element_text(angle=angle, hjust=0, vjust=0))
    i=i+1
  }
  do.call(grid.arrange,c(p,ncol=1))
}

annotate_clusters &lt;- function(theta_mat){
  top_features &lt;- ExtractTopFeatures(theta_mat, top_features=100, method=&quot;poisson&quot;, options=&quot;min&quot;)
  gene_vector &lt;- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(theta_mat)[top_features[[1]][x,]]))
  scores_vector &lt;- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}</code></pre>
<p>Assess fastTopics results (see 10x-plot-calssification-gom.Rmd for figures)</p>
<pre class="r"><code>#Check BIC

numclust &lt;- 3
#numclust &lt;- &quot;3.human&quot;
#numclust &lt;- &quot;3.chimp&quot;

#Read fastTopics data
fit &lt;- readRDS(paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;))
#fit &lt;- readRDS(paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds&quot;))

#Compute omega (weights, cells x k) and theta (topics, genes x k) (need to add up to 1)
l &lt;- fit$L
f &lt;- fit$F
omega &lt;- sweep(l, MARGIN=2, colSums(f), `*`)
scale &lt;- rowSums(omega)
omega &lt;- omega / scale
theta &lt;- f/colSums(f)

#BIC
loglik &lt;- fit$progress$loglik[length(fit$progress$loglik)]
BIC &lt;- -2*loglik + nrow(theta)*(ncol(theta)-1)*log(nrow(omega))
BIC</code></pre>
<pre class="r"><code>#Plot BIC

par(mfrow=c(1,1))

#fastTopics BIC
x=c(3,4,5,6,7)
y=c(2385198270,2347325326,2313214572,2281539581,2255267597)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fastTopics.Human BIC
x=c(3,4,5,6,7)
y=c(1098763998,1081234940,1064314595,1051699136,1037815943)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fastTopics.Chimp BIC
x=c(3,4,5,6,7)
y=c(1212496514,1183947130,1169079411,1154808262,1145154735)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fastTopics BIC (batch corrected)
x=c(3,4,5,6,7)
y=c(2340110918,2301627330,2268542187,2243574970,2222862911)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fastTopics.Human BIC (batch corrected)
x=c(3,4,5,6,7)
y=c(1068104529,1051742121,1036639381,1029192440,1022890111)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fastTopics.Chimp BIC (batch corrected)
x=c(3,4,5,6,7)
y=c(1168907169,1152256850,1141247764,1132319037,1123799212)
plot(x,y,xlab=&quot;Topics (k)&quot;,ylab=&quot;BIC&quot;)
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-21-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Examine structure plots and gene contributions to topics

numclust &lt;- 3
#numclust &lt;- &quot;3.human&quot;
#numclust &lt;- &quot;3.chimp&quot;

colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;) #k=3
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue&quot;) #k=4, data=tot,tot.bc1,c.tot,c.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;) #k=4, data=h.tot
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;grey&quot;) #k=4, data=h.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue&quot;) #k=5, data=tot,tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;,&quot;grey&quot;) #k=5, data=h.tot
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;lightgrey&quot;,&quot;darkgrey&quot;) #k=5, data=h.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;) #k=5, data=c.tot
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue&quot;,&quot;grey&quot;) #k=5, data=c.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;) #k=6, data=tot
#colclust &lt;- c(&quot;pink&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue&quot;,&quot;grey&quot;) #k=6, data=tot.bc1,h.tot
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;,&quot;lightgrey&quot;,&quot;darkgrey&quot;) #k=6, data=h.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;,&quot;grey&quot;) #k=6, data=c.tot,c.tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;pink4&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;,&quot;grey&quot;) #k7, data=tot,tot.bc1
#colclust &lt;- c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;orchid4&quot;,&quot;dodgerblue1&quot;,&quot;dodgerblue4&quot;,&quot;lightgrey&quot;,&quot;darkgrey&quot;) #k7, data=h.tot,h.tot.bc1,c.tot,c.tot.bc1

#Read fastTopics data
fit &lt;- readRDS(paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;))
#fit &lt;- readRDS(paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds&quot;))

#Check fitting procedure
quickplot(x=1:length(fit$progress$loglik),
          y=fit$progress$loglik,
          geom=c(&quot;point&quot;,&quot;line&quot;),
          xlab=&quot;iteration&quot;,
          ylab=&quot;loglik&quot;)

#Compute omega (weights, cells x k) and theta (topics, genes x k) (need to add up to 1)
l &lt;- fit$L
f &lt;- fit$F
omega &lt;- sweep(l, MARGIN=2, colSums(f), `*`)
scale &lt;- rowSums(omega)
omega &lt;- omega / scale
theta &lt;- f/colSums(f)

#BIC
loglik &lt;- fit$progress$loglik[length(fit$progress$loglik)]
BIC &lt;- -2*loglik + nrow(theta)*(ncol(theta)-1)*log(nrow(omega))
BIC

#Recover the multinomial topic model
fit.multinom &lt;- poisson2multinom(fit)
range(rowSums(fit.multinom$L))
range(colSums(fit.multinom$F))

#Topic annotations
genes &lt;- annotate_clusters(theta_mat=theta)
genes[[1]][,1:10]
saveRDS(genes, file=paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.100genes.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;))
#saveRDS(genes, file=paste0(&quot;./data/gom-data/topics&quot;,numclust,&quot;.100genes.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds&quot;))

#Create labels
labels &lt;- sapply(strsplit(rownames(omega),&quot;_&quot;), `[`, 1)
annotation1 &lt;- data.frame(sample_id=paste0(&quot;X&quot;, c(1:NROW(omega))), tissue_label=as.factor(labels))
celltype &lt;- labels
celltype[celltype %in% c(&quot;C1.I&quot;,&quot;C1r2.I&quot;,&quot;C2.I&quot;,&quot;C3.I&quot;,&quot;C4.I&quot;,&quot;C5.I&quot;,&quot;C6.I&quot;)] &lt;- &quot;Chimp.iPSC&quot;
celltype[celltype %in% c(&quot;C1.M&quot;,&quot;C1r2.M&quot;,&quot;C2.M&quot;,&quot;C3.M&quot;,&quot;C4.M&quot;,&quot;C5.M&quot;,&quot;C6.M&quot;)] &lt;- &quot;Chimp.MSC&quot;
celltype[celltype %in% c(&quot;C1.O&quot;,&quot;C1r2.O&quot;,&quot;C2.O&quot;,&quot;C3.O&quot;,&quot;C4.O&quot;,&quot;C5.O&quot;,&quot;C6.O&quot;)] &lt;- &quot;Chimp.Osteogenic&quot;
celltype[celltype %in% c(&quot;H1.I&quot;,&quot;H1r2.I&quot;,&quot;H2.I&quot;,&quot;H3.I&quot;,&quot;H4.I&quot;,&quot;H5.I&quot;,&quot;H6.I&quot;)] &lt;- &quot;Human.iPSC&quot;
celltype[celltype %in% c(&quot;H1.M&quot;,&quot;H1r2.M&quot;,&quot;H2.M&quot;,&quot;H3.M&quot;,&quot;H4.M&quot;,&quot;H5.M&quot;,&quot;H6.M&quot;)] &lt;- &quot;Human.MSC&quot;
celltype[celltype %in% c(&quot;H1.O&quot;,&quot;H1r2.O&quot;,&quot;H2.O&quot;,&quot;H3.O&quot;,&quot;H4.O&quot;,&quot;H5.O&quot;,&quot;H6.O&quot;)] &lt;- &quot;Human.Osteogenic&quot;
annotation2 &lt;- data.frame(sample_id=paste0(&quot;X&quot;, c(1:NROW(omega))), tissue_label=as.factor(celltype))

#Reorder k
omega &lt;- omega[,c(3,1,2)] #k=3, data=tot,tot.bc1,h.tot,c.tot,c.tot.bc1
#omega &lt;- omega[,c(3,2,1)] #k=3, data=h.tot.bc1
#omega &lt;- omega[,c(4,2,3,1)] #k=4, data=tot,tot.bc1
#omega &lt;- omega[,c(4,3,2,1)] #k=4, data=h.tot
#omega &lt;- omega[,c(1,4,3,2)] #k=4, data=h.tot.bc1
#omega &lt;- omega[,c(4,2,3,1)] #k=4, data=c.tot
#omega &lt;- omega[,c(4,3,1,2)] #k=4, data=c.tot.bc1
#omega &lt;- omega[,c(2,1,5,3,4)] #k=5, data=tot
#omega &lt;- omega[,c(4,3,2,5,1)] #k=5, data=tot.bc1
#omega &lt;- omega[,c(1,5,3,4,2)] #k=5, data=h.tot
#omega &lt;- omega[,c(3,5,1,2,4)] #k=5, data=h.tot.bc1
#omega &lt;- omega[,c(5,4,1,2,3)] #k=5, data=c.tot
#omega &lt;- omega[,c(4,5,1,3,2)] #k=5, data=c.tot.bc1
#omega &lt;- omega[,c(4,2,5,3,6,1)] #k=6, data=tot
#omega &lt;- omega[,c(3,2,5,6,4,1)] #k=6, data=tot.bc1
#omega &lt;- omega[,c(6,3,5,4,2,1)] #k=6, data=h.tot
#omega &lt;- omega[,c(5,6,1,2,4,3)] #k=6, data=h.tot.bc1
#omega &lt;- omega[,c(6,1,5,3,4,2)] #k=6, data=c.tot
#omega &lt;- omega[,c(6,5,2,4,1,3)] #k=6, data=c.tot.bc1
#omega &lt;- omega[,c(6,7,5,3,2,1,4)] #k=7, data=tot
#omega &lt;- omega[,c(6,7,4,2,5,3,1)] #k=7, data=tot.bc1
#omega &lt;- omega[,c(3,7,4,5,1,6,2)] #k=7, data=h.tot
#omega &lt;- omega[,c(6,5,1,2,3,7,4)] #k=7, data=h.tot.bc1
#omega &lt;- omega[,c(1,5,3,7,6,4,2)] #k=7, data=c.tot
#omega &lt;- omega[,c(5,7,3,2,6,4,1)] #k=7, data=c.tot.bc1

#Plot of topic membership
make_topic_plots(omega=omega, labels=celltype, angle=-20)

#Structure plots
rownames(omega) &lt;- annotation2$sample_id
CountClust::StructureGGplot(omega=omega,
                annotation=annotation2,
                palette=colclust,
                yaxis_label=&quot;Species&quot;,
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face=&quot;bold&quot;))

#Heatmap of topic weights (omega)
annotation &lt;- data.frame(row.names=paste0(&quot;X&quot;, c(1:NROW(omega))), labels=as.factor(celltype))
rownames(omega) &lt;- rownames(annotation)
pheatmap(omega,
         cluster_cols=FALSE,
         cluster_rows=FALSE,
         annotation_row=annotation,
         show_rownames=FALSE,
         main=&quot;Heatmap of topic weights&quot;)</code></pre>
<pre class="r"><code>saveRDS(data, file=&quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;)</code></pre>
</div>
<div id="ad-hoc-assignments" class="section level2">
<h2>Ad Hoc Assignments</h2>
<p>Several different candidate genes and thresholds were examined (see 10x-plot-classification-adhoc.Rmd for more details).</p>
<p>Candidate Genes</p>
<ul>
<li><p>Time 0</p></li>
<li>diagnostic: POU5F1 (OCT4), TDGF1, SOX2, EPCAM</li>
<li>additional: NANOG, ALPL, STAT3, PROM1 (CD133), PODXL (TRA-1-60R), LIN28A, ZFP42, UTF1, MYC, BMP4, FGF2, TGFB1, DNMT3B, SALL4</li>
<li><p>if needed: FOXD3, FUT4 (SSEA-1), KLF2, KLF4, KLF5, INHBA (Activin A), LIF, PRDM14</p></li>
<li><p>Time 1</p></li>
<li>diagnostic: THY1 (CD90), NT5E (CD73), ENG (CD105), CD44</li>
<li>additional: ANPEP (CD13), NGFR (CD271), ITGB1 (CD29), NCAM1 (CD56)</li>
<li>if needed: CD9, CD14, MME (CD10), ITGAL (CD11A), ITGAX (CD11C)*, CD36, ITGA3 (CD49C), ITGA4 (CD49D), ITGAV (CD51), ICAM1 (CD54), CD55, CD59, TFRC (CD71), VCAM1 (CD106), KIT (CD117), TNFRSF1A (CD120a), PDGFRB (CD140B), CDH5 (CD144), MCAM (CD146), ALCAM (CD166), ERBB2 (CD340)</li>
<li><p>negative controls: ITGAM (CD11B), CD14, PECAM1 (CD31), CD34, PTPRC (CD45), CD79A, HLA-DRA (HLA-DR)</p></li>
<li><p>Time 2</p></li>
<li>diagnostic: RUNX2, BGLAP, CAPG, [DMP1], [PHEX], MEPE, [SOST], [HYOU1]</li>
<li>additional: COL1A1, COL1A2, ALPL, TNFRSF11B (OPG), TNFSF11 (RANKL), CSF1 (M-CSF), DKK1, SFRP1, PTGES2, PTGS2, SPP1 (OPN), SPARC (ONT)</li>
<li>if needed: E11 (not in ortho-exon)</li>
<li>markers that differentiated BM osteoblasts in Anastasia Tikhonova’s research (ASBMR2020): COL11A2, SPARC, SOX9, COMP, CHAD, COL10A1, EDIL3, MMP14, OSTN, COL12A1, ANGPTL2, COL16A1</li>
<li><p>markers used to differentiate different stages of osteogenic differentiation from Jialiang Wang’s research (ASBMR2020): BGLAP=preosteoblast, TNC=intermediate, PDPN+FBLN7=osteocytes</p></li>
<li><p>Chondrogenic</p></li>
<li><p>diagnostic: ACAN, COL2A1, SOX9, COL10A1, COL11A1, SOX5, SOX6</p></li>
<li><p>Adipogenic</p></li>
<li><p>diagnostic: FABP4, PPARG, EBF2, PDGFRA, FGF10, FGF16, FGF19, SLC7A10, SLC36A2, MYF5, P2RX5</p></li>
</ul>
<p>Thresholds</p>
<ul>
<li>scaled integrated data (with confounding variables regressed out)</li>
<li>log normalized data</li>
<li>rescaled integrated gene expression data to range from 0 to 1</li>
<li>rescaled log normalized gene expression data to range from 0 to 1</li>
<li>non-scaled integrated data (with confounding variables regressed out)</li>
</ul>
<p>However, the best gene combinations and thresholds decided on were (see 10x-plot-classification-adhoc.Rmd for more details):</p>
<ul>
<li>General Ad Hoc Assignments
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>for good measure run a couple other assignments at different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)</li>
<li>ips = +POU5F1,+TDGF,+SOX2,+EPCAM</li>
<li>msc = +THY1,+NT5E,+ENG,+CD44</li>
<li>ost = -POU5F1,+COL1A1,+COL1A2,+ALPL</li>
</ul></li>
<li>Osteogenic Ad Hoc Assignements
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>for good measure run a couple other assignments at different cutoffs but not as many as for the general ad hoc assignments (thresh-0.01,thresh0.01)</li>
<li>in addition to osteogenic specific marker genes, also use general iPSC and osteogenic marker genes</li>
<li>8 genes - most consistent with literature even though it is the most conservative criteria (RUNX2,BGLAP,CAPG,DMP1,PHEX,MEPE,SOST,HYOU1)</li>
<li>6A genes - intermediate number of genes that removes a highly expressed gene in all cell types (CAPG) and a heat-shock protein (HYOU1) (RUNX2,BGLAP,DMP1,PHEX,MEPE,SOST)</li>
<li>4 genes - simplest gene combination (RUNX2,BGLAP,PHEX,MEPE)</li>
</ul></li>
</ul>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)</code></pre>
<pre class="r"><code>#Isolate genes of interest

genes &lt;- c(&quot;POU5F1&quot;,&quot;TDGF1&quot;,&quot;SOX2&quot;,&quot;EPCAM&quot;,
           &quot;THY1&quot;,&quot;NT5E&quot;,&quot;ENG&quot;,&quot;CD44&quot;,
           &quot;RUNX2&quot;,&quot;BGLAP&quot;,&quot;CAPG&quot;,&quot;DMP1&quot;,&quot;PHEX&quot;,&quot;MEPE&quot;,&quot;SOST&quot;,&quot;HYOU1&quot;,
           &quot;COL1A1&quot;,&quot;COL1A2&quot;,&quot;ALPL&quot;,
           &quot;ACAN&quot;,&quot;COL2A1&quot;,&quot;SOX9&quot;,
           &quot;FABP4&quot;,&quot;PPARG&quot;,&quot;EBF2&quot;,
           &quot;TNC&quot;,&quot;PDPN&quot;,&quot;FBLN7&quot;)</code></pre>
<p>Define function for setting ad hoc thresholds.</p>
<pre class="r"><code>#Define functions for setting ad hoc thresholds

chk.thres.manual &lt;- function(data, genes, thres) {
  assignments &lt;- data
  i=1
  while(i &lt;= length(genes)) {
    assignments[,genes[i]] &lt;- ifelse(data[,genes[i]] &gt; thres[i], 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.mean &lt;- function(data, genes, cutoff) {
  assignments &lt;- data
  i=1
  while(i &lt;= length(genes)) {
    if(is.na(cutoff)) { thres &lt;- mean(data[,genes[i]]) }
    else { thres &lt;- mean(data[,genes[i]][data[,genes[i]]&lt;cutoff]) }
    assignments[,genes[i]] &lt;- ifelse(data[,genes[i]] &gt; thres, 1, 0)
    i=i+1
  }
  return(assignments)
}</code></pre>
<p>Define functions for visualizing ad hoc assignments.</p>
<pre class="r"><code>#Define functions for making upsetr plots

plot.upset.HCcellcompare &lt;- function(data, genes, ymax) {
  plot.ls &lt;- list(
    T0.Human = upset(data[which(data$Stage==&quot;Time 0&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage==&quot;Time 1&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage==&quot;Time 2&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage==&quot;Time 0&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage==&quot;Time 1&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage==&quot;Time 2&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE, sets=genes, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit(&#39;arrange&#39;, name=v)
    vp &lt;- grid.grab()
    plot.ls[[v]] &lt;- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}

plot.upset.sub.HCcellcompare &lt;- function(data, genes, intersections, ymax) {
  plot.ls &lt;- list(
    T0.Human = upset(data[which(data$Stage==&quot;Time 0&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage==&quot;Time 1&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage==&quot;Time 2&quot; &amp; data$Species==&quot;Human&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage==&quot;Time 0&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage==&quot;Time 1&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage==&quot;Time 2&quot; &amp; data$Species==&quot;Chimp&quot;),genes],
                     nintersects=NA, empty.intersections=&quot;on&quot;, keep.order=TRUE, order.by=&quot;degree&quot;, decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit(&#39;arrange&#39;, name=v)
    vp &lt;- grid.grab()
    plot.ls[[v]] &lt;- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}</code></pre>
<p>Calculate if genes reach expression thresholds (see 10x-plot-classification-adhoc.Rmd for more details).</p>
<pre class="r"><code>#Scaled integrated data (with confounding variables regressed out)
tot.genes &lt;- cbind(as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes,]))),data@meta.data[,c(&quot;Species&quot;,&quot;Stage&quot;,&quot;Sample&quot;)])

assign.threshN0.1 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.1,length(genes))))
assign.threshN0.05 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.05,length(genes))))
assign.threshN0.01 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.01,length(genes))))
assign.thresh0 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(0,length(genes))))
assign.thresh0.01 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(0.01,length(genes))))
assign.thresh0.05 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(0.05,length(genes))))
assign.thresh0.1 &lt;- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,length(genes))))

assign.mean &lt;- chk.thres.mean(tot.genes, genes, cutoff=NA)</code></pre>
<div id="ad-hoc-assignment-general" class="section level3">
<h3>Ad Hoc Assignment (general)</h3>
<p>Assign cell type labels based on general ad hoc cutoffs.</p>
<p>Cutoff scheme:</p>
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>for good measure run a couple other assignments at different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)</li>
<li>ips = +POU5F1,+TDGF,+SOX2,+EPCAM</li>
<li>msc = +THY1,+NT5E,+ENG,+CD44</li>
<li>ost = -POU5F1,+COL1A1,+COL1A2,+ALPL</li>
</ul>
<p>To make this cutoff scheme work, the coded cutoff actually needs to be:</p>
<ul>
<li>ips = +POU5F1 [or] +TDGF [or] +SOX2 [or] +EPCAM [&amp;] -NT5E,-ENG,-CD44</li>
<li>msc = +THY1 [or] +NT5E [or] +ENG [or] +CD44 [&amp;] -POU5F1,-TDGF,-SOX2,-EPCAM,-ALPL</li>
<li>ost = +ALPL [&amp;] -POU5F1,-TDGF,-SOX2,-EPCAM</li>
</ul>
<p>Cutoff scheme described in words:</p>
<ul>
<li>ips = must express 1 or more pluripotent markers (POU5F1,TDGF1,SOX2,EPCAM) but not express any mesenchymal markers (NT5E,ENG,CD44)</li>
<li>msc = must express 1 or more mesenchymal markers (THY1,NT5E,ENG,CD44) but do not express any pluripotent markers (POU5F1,TDGF1,SOX2,EPCAM) or osteogenic markers (ALPL)</li>
<li>ost = must express osteogenic markers (ALPL) but do not express any pluripotent markers (POU5F1,TDGF1,SOX2,EPCAM)</li>
</ul>
<pre class="r"><code>hoc.assign &lt;- function(assignments) {
  assignments$assign &lt;- &quot;other&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if((assignments$POU5F1[i] == 1 &amp; assignments$NT5E[i] == 0 &amp; assignments$ENG[i] == 0 &amp; assignments$CD44[i] == 0) |
       (assignments$TDGF1[i]  == 1 &amp; assignments$NT5E[i] == 0 &amp; assignments$ENG[i] == 0 &amp; assignments$CD44[i] == 0) |
       (assignments$SOX2[i]   == 1 &amp; assignments$NT5E[i] == 0 &amp; assignments$ENG[i] == 0 &amp; assignments$CD44[i] == 0) |
       (assignments$EPCAM[i]  == 1 &amp; assignments$NT5E[i] == 0 &amp; assignments$ENG[i] == 0 &amp; assignments$CD44[i] == 0))
      {assignments$assign[i] &lt;- &quot;iPSC&quot;}
    if((assignments$THY1[i] == 1 &amp; assignments$POU5F1[i] == 0 &amp; assignments$TDGF1[i] == 0 &amp; assignments$SOX2[i] == 0 &amp; assignments$EPCAM[i] == 0 &amp; assignments$ALPL[i] == 0) |
       (assignments$NT5E[i] == 1 &amp; assignments$POU5F1[i] == 0 &amp; assignments$TDGF1[i] == 0 &amp; assignments$SOX2[i] == 0 &amp; assignments$EPCAM[i] == 0 &amp; assignments$ALPL[i] == 0) |
       (assignments$ENG[i]  == 1 &amp; assignments$POU5F1[i] == 0 &amp; assignments$TDGF1[i] == 0 &amp; assignments$SOX2[i] == 0 &amp; assignments$EPCAM[i] == 0 &amp; assignments$ALPL[i] == 0) |
       (assignments$CD44[i] == 1 &amp; assignments$POU5F1[i] == 0 &amp; assignments$TDGF1[i] == 0 &amp; assignments$SOX2[i] == 0 &amp; assignments$EPCAM[i] == 0 &amp; assignments$ALPL[i] == 0))
      {assignments$assign[i] &lt;- &quot;MSC&quot;}
    if((assignments$ALPL[i] == 1 &amp; assignments$POU5F1[i] == 0 &amp; assignments$TDGF1[i] == 0 &amp; assignments$SOX2[i] == 0 &amp; assignments$EPCAM[i] == 0))
      {assignments$assign[i] &lt;- &quot;Osteogenic&quot;}
    i=i+1
  }
 return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}</code></pre>
<pre class="r"><code>#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0)
ad.hoc.assign &lt;- hoc.assign(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.thresh0 &lt;- factor(data@meta.data$AdHoc.Assign.thresh0,
                                              levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.01)
ad.hoc.assign &lt;- hoc.assign(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.threshN0.01 &lt;- factor(data@meta.data$AdHoc.Assign.threshN0.01,
                                                  levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.05)
ad.hoc.assign &lt;- hoc.assign(assign.threshN0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.threshN0.05 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.threshN0.05 &lt;- factor(data@meta.data$AdHoc.Assign.threshN0.05,
                                                  levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.1)
ad.hoc.assign &lt;- hoc.assign(assign.threshN0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.threshN0.1 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.threshN0.1 &lt;- factor(data@meta.data$AdHoc.Assign.threshN0.1,
                                                 levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.01)
ad.hoc.assign &lt;- hoc.assign(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.thresh0.01 &lt;- factor(data@meta.data$AdHoc.Assign.thresh0.01,
                                                 levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.05)
ad.hoc.assign &lt;- hoc.assign(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.05 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.thresh0.05 &lt;- factor(data@meta.data$AdHoc.Assign.thresh0.05,
                                                 levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.1)
ad.hoc.assign &lt;- hoc.assign(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.1 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$AdHoc.Assign.thresh0.1 &lt;- factor(data@meta.data$AdHoc.Assign.thresh0.1,
                                                levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;other&quot;))</code></pre>
<p>Make nice plots of ad hoc assignments (thresh0).</p>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Human&quot;)@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
     14827      21629       6286       4668 </code></pre>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Chimp&quot;)@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
     17355      21945      11289       3001 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 0&quot;)@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
     32144         19          9       3416 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 1&quot;)@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
        20      34982        978       1248 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 2&quot;)@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
        18       8573      16588       3005 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
     14804         15          9       1868 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
        13      16655        853        800 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
        10       4959       5424       2000 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
     17340          4          0       1548 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
         7      18327        125        448 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$AdHoc.Assign.thresh0)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic      other 
         8       3614      11164       1005 </code></pre>
<pre class="r"><code>ggplot(data@meta.data, aes(x=AdHoc.Assign.thresh0, fill=Species)) +
  geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;) +
  scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data@meta.data, aes(x=AdHoc.Assign.thresh0, fill=Species)) +
  geom_bar(mapping = aes(x=AdHoc.Assign.thresh0, y=..prop.., group=Species), stat=&quot;count&quot;, position=&quot;dodge&quot;) +
  scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data@meta.data, aes(x=AdHoc.Assign.thresh0, fill=Individual)) +
  geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data@meta.data, aes(x=AdHoc.Assign.thresh0, fill=Individual)) +
  geom_bar(mapping = aes(x=AdHoc.Assign.thresh0, y=..prop.., group=Individual), stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(data,
        group.by=&quot;Species&quot;,
        split.by=&quot;AdHoc.Assign.thresh0&quot;,
        reduction=&quot;umap&quot;,
        cols=c(&quot;grey&quot;,&quot;black&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;AdHoc.Assign.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;AdHoc.Assign.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;AdHoc.Assign.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;pink&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;AdHoc.Assign.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;mediumorchid&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;AdHoc.Assign.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-31-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(data, file=&quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;)</code></pre>
</div>
<div id="ad-hoc-assignment-osteogenic" class="section level3">
<h3>Ad Hoc Assignment (osteogenic)</h3>
<p>Assign cell type labels based on ad hoc cutoffs.</p>
<p>Cutoff scheme:</p>
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>for good measure run a couple other assignments at different cutoffs but not as many as for the general ad hoc assignments (thresh-0.01,thresh0.01)</li>
<li>8 genes - most consistent with literature even though it is the most conservative criteria
<ul>
<li>pre = +RUNX2,-BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>obl = +RUNX2,+BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>emb = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,-MEPE,-SOST,-HYOU1 &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>min = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,-SOST,-HYOU1 &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>mat = -RUNX2,-BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,+SOST,+HYOU1 &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
</ul></li>
<li>6A genes - intermediate number of genes that removes a highly expressed gene in all cell types (CAPG) and a heat-shock protein (HYOU1)
<ul>
<li>pre = +RUNX2,-BGLAP,-DMP1,-PHEX,-MEPE,-SOST &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>obl = +RUNX2,+BGLAP,-DMP1,-PHEX,-MEPE,-SOST &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>emb = -RUNX2,+BGLAP,+DMP1,+PHEX,-MEPE,-SOST &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>min = -RUNX2,+BGLAP,+DMP1,+PHEX,+MEPE,-SOST &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>mat = -RUNX2,-BGLAP,+DMP1,+PHEX,+MEPE,+SOST &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
</ul></li>
<li>4 genes - simplest gene combination
<ul>
<li>pre = +RUNX2,-BGLAP,-PHEX,-MEPE &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>obl = +RUNX2,+BGLAP,-PHEX,-MEPE &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>emb = -RUNX2,+BGLAP,+PHEX,-MEPE &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>min = -RUNX2,+BGLAP,+PHEX,+MEPE &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
<li>mat = -RUNX2,-BGLAP,+PHEX,+MEPE &amp; -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL</li>
</ul></li>
</ul>
<p>This cutoff scheme should work as is, but closely related gene expression patterns are also defined as x.osteoblast, x.embedding osteoblast, x.mineralizing osteoblast, and x.maturing osteocyte:</p>
<ul>
<li>8 genes
<ul>
<li>x.obl = +BGLAP [and] -CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1</li>
<li>x.emb = +CAPG [or] +DMP1 [or] +PHEX [and] -MEPE,-SOST,-HYOU1</li>
<li>x.min = +MEPE [and] -SOST,-HYOU1</li>
<li>x.mat = +SOST [or] +HYOU1</li>
</ul></li>
<li>6A genes
<ul>
<li>x.obl = +BGLAP [and] -DMP1,-PHEX,-MEPE,-SOST</li>
<li>x.emb = +DMP1 [or] +PHEX [and] -MEPE,-SOST</li>
<li>x.min = +MEPE [and] -SOST</li>
<li>x.mat = +SOST</li>
</ul></li>
<li>4 genes
<ul>
<li>x.obl = +BGLAP [and] -PHEX,-MEPE</li>
<li>x.emb = +PHEX [and] -MEPE</li>
<li>x.min = +MEPE</li>
<li>x.mat = +MEPE [and] -RUNX2,-BGLAP</li>
</ul></li>
</ul>
<p>Assignments using 8 genes</p>
<pre class="r"><code>#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes)

hoc.assign.x.8 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #lenient criteria
      if((assignments$SOST[i] ==1) |
         (assignments$HYOU1[i]==1)) {
        assignments$assign[i] &lt;- &quot;x.maturing osteocyte&quot;
      }
      if(assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) {
        assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;
      }
      if((assignments$CAPG[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) |
         (assignments$DMP[i] ==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) |
         (assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0)) {
        assignments$assign[i] &lt;- &quot;x.embedding osteoblast&quot;
      }
      if(assignments$BGLAP[i]==1 &amp; assignments$CAPG[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) {
        assignments$assign[i] &lt;- &quot;x.osteoblast&quot;
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==1 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==1 &amp; assignments$SOST[i] ==1 &amp; assignments$HYOU1[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.8 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #lenient criteria
      if((assignments$SOST[i] ==1) |
         (assignments$HYOU1[i]==1)) {
        assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;
      }
      if(assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) {
        assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;
      }
      if((assignments$CAPG[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) |
         (assignments$DMP[i] ==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) |
         (assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0)) {
        assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;
      }
      if(assignments$BGLAP[i]==1 &amp; assignments$CAPG[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0 &amp; assignments$HYOU1[i]==0) {
        assignments$assign[i] &lt;- &quot;osteoblast&quot;
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==0 &amp; assignments$DMP[i]  ==0 &amp; assignments$PHEX[i] ==0 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==0 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==1 &amp; assignments$SOST[i] ==0 &amp; assignments$HYOU1[i]==0) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp;
         assignments$CAPG[i] ==1 &amp; assignments$DMP[i]  ==1 &amp; assignments$PHEX[i] ==1 &amp;
         assignments$MEPE[i] ==1 &amp; assignments$SOST[i] ==1 &amp; assignments$HYOU1[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}</code></pre>
<pre class="r"><code>#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, thresh0)
ad.hoc.assign &lt;- hoc.assign.x.8(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.8gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.8gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.8gene.thresh0,
                                                         levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                  &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                  &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.8(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.8gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.8gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.8gene.thresh0,
                                                       levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, threshN0.01)
ad.hoc.assign &lt;- hoc.assign.x.8(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01,
                                                             levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                      &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                      &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.8(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.8gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.8gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.8gene.threshN0.01,
                                                           levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                    &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, thresh0.01)
ad.hoc.assign &lt;- hoc.assign.x.8(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01,
                                                            levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                     &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                     &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.8(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.8gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.8gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.8gene.thresh0.01,
                                                          levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                   &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))</code></pre>
<p>Assignments using 6A genes</p>
<pre class="r"><code>#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes)

hoc.assign.x.6 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #lenient criteria
      if(assignments$SOST[i]==1) {
        assignments$assign[i] &lt;- &quot;x.maturing osteocyte&quot;
      }
      if(assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0) {
        assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;
      }
      if((assignments$DMP[i] ==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) |
         (assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0)) {
        assignments$assign[i] &lt;- &quot;x.embedding osteoblast&quot;
      }
      if(assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {
        assignments$assign[i] &lt;- &quot;x.osteoblast&quot;
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1 &amp; assignments$SOST[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.6 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #lenient criteria
      if(assignments$SOST[i]==1) {
        assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;
      }
      if(assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0) {
        assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;
      }
      if((assignments$DMP[i] ==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) |
         (assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0)) {
        assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;
      }
      if(assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {
        assignments$assign[i] &lt;- &quot;osteoblast&quot;
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1 &amp; assignments$SOST[i]==0) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$DMP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1 &amp; assignments$SOST[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}</code></pre>
<pre class="r"><code>#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, thresh0)
ad.hoc.assign &lt;- hoc.assign.x.6(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.6gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.6gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.6gene.thresh0,
                                                         levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                  &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                  &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.6(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.6gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.6gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.6gene.thresh0,
                                                       levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, threshN0.01)
ad.hoc.assign &lt;- hoc.assign.x.6(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01,
                                                             levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                      &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                      &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.6(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.6gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.6gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.6gene.threshN0.01,
                                                           levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                    &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, thresh0.01)
ad.hoc.assign &lt;- hoc.assign.x.6(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01,
                                                            levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                     &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                     &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.6(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.6gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.6gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.6gene.thresh0.01,
                                                          levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                   &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))</code></pre>
<p>Assignments using 4 genes</p>
<pre class="r"><code>#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes)

hoc.assign.x.4 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
      #lenient criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;x.osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;x.embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;x.embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;x.embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;x.maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.4 &lt;- function(assignments) {
  assignments$assign &lt;- &quot;not osteogenic&quot;
  i=1
  while(i &lt;= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 &amp; assignments$POU5F1[i]==0 &amp; assignments$TDGF1[i]==0 &amp; assignments$SOX2[i]==0 &amp; assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;unknown osteogenic&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;preosteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
      #lenient criteria
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==0) {assignments$assign[i] &lt;- &quot;embedding osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==1 &amp; assignments$BGLAP[i]==1 &amp; assignments$PHEX[i]==1 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;mineralizing osteoblast&quot;}
      if(assignments$RUNX2[i]==0 &amp; assignments$BGLAP[i]==0 &amp; assignments$PHEX[i]==0 &amp; assignments$MEPE[i]==1) {assignments$assign[i] &lt;- &quot;maturing osteocyte&quot;}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}</code></pre>
<pre class="r"><code>#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, thresh0)
ad.hoc.assign &lt;- hoc.assign.x.4(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.4gene.thresh0,
                                                         levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                  &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                  &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.4(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4gene.thresh0 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4gene.thresh0 &lt;- factor(data@meta.data$OstAdHoc.Assign.4gene.thresh0,
                                                       levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, threshN0.01)
ad.hoc.assign &lt;- hoc.assign.x.4(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01,
                                                             levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                      &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                      &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.4(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4gene.threshN0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4gene.threshN0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.4gene.threshN0.01,
                                                           levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                    &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, thresh0.01)
ad.hoc.assign &lt;- hoc.assign.x.4(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01,
                                                            levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                     &quot;x.osteoblast&quot;,&quot;x.embedding osteoblast&quot;,&quot;x.mineralizing osteoblast&quot;,&quot;x.maturing osteocyte&quot;,
                                                                     &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))
ad.hoc.assign &lt;- hoc.assign.4(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4gene.thresh0.01 &lt;- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4gene.thresh0.01 &lt;- factor(data@meta.data$OstAdHoc.Assign.4gene.thresh0.01,
                                                          levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,
                                                                   &quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;,&quot;other&quot;))</code></pre>
<p>Make nicer plots of osteogenic ad hoc assignments (8 gene, thresh0).</p>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Human&quot;)@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    124                      90                    1706 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   1957                    2304                     105 
         not osteogenic                   other 
                  41124                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Chimp&quot;)@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    149                     282                    4310 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   2725                    3726                      97 
         not osteogenic                   other 
                  42301                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 0&quot;)@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       1 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      2                       3                       3 
         not osteogenic                   other 
                  35579                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 1&quot;)@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                     19                       4                     212 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                    261                     441                      41 
         not osteogenic                   other 
                  36250                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 2&quot;)@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    254                     368                    5803 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   4419                    5586                     158 
         not osteogenic                   other 
                  11596                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       1 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      2                       3                       3 
         not osteogenic                   other 
                  16687                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                     19                       4                     198 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                    225                     369                      38 
         not osteogenic                   other 
                  17468                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    105                      86                    1507 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   1730                    1932                      64 
         not osteogenic                   other 
                   6969                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      0                       0                       0 
         not osteogenic                   other 
                  18892                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                      14 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                     36                      72                       3 
         not osteogenic                   other 
                  18782                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.8gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    149                     282                    4296 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   2689                    3654                      94 
         not osteogenic                   other 
                   4627                       0 </code></pre>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.8gene.thresh0)), aes(x=OstAdHoc.Assign.8gene.thresh0, fill=Species)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.8gene.thresh0)), aes(x=OstAdHoc.Assign.8gene.thresh0, fill=Species)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.8gene.thresh0, y=..prop.., group=Species), stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.8gene.thresh0)), aes(x=OstAdHoc.Assign.8gene.thresh0, fill=Individual)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.8gene.thresh0)), aes(x=OstAdHoc.Assign.8gene.thresh0, fill=Individual)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.8gene.thresh0, y=..prop.., group=Individual), stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(data,
        group.by=&quot;Species&quot;,
        split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
        reduction=&quot;umap&quot;,
        cols=c(&quot;grey&quot;,&quot;black&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<pre><code>Warning: Graphs cannot be vertically aligned unless the axis parameter is
set. Placing graphs unaligned.</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(DimPlot(data,
              group.by=&quot;Stage&quot;,
              split.by=&quot;OstAdHoc.Assign.8gene.thresh0&quot;,
              reduction=&quot;umap&quot;,
              cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
        labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-39-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make nicer plots of osteogenic ad hoc assignments (6 gene, thresh0).</p>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Human&quot;)@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    807                     672                    1328 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   2820                      52                     607 
         not osteogenic                   other 
                  41124                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Chimp&quot;)@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    702                     992                    5149 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   3913                     141                     392 
         not osteogenic                   other 
                  42301                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 0&quot;)@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      3                       2                       4 
         not osteogenic                   other 
                  35579                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 1&quot;)@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                     90                      12                     202 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                    443                      55                     176 
         not osteogenic                   other 
                  36250                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 2&quot;)@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                   1419                    1652                    6275 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   6287                     136                     819 
         not osteogenic                   other 
                  11596                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      3                       2                       4 
         not osteogenic                   other 
                  16687                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                     90                      11                     179 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                    392                      13                     168 
         not osteogenic                   other 
                  17468                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    717                     661                    1149 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   2425                      37                     435 
         not osteogenic                   other 
                   6969                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      0                       0                       0 
         not osteogenic                   other 
                  18892                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       1                      23 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                     51                      42                       8 
         not osteogenic                   other 
                  18782                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.6gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    702                     991                    5126 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   3862                      99                     384 
         not osteogenic                   other 
                   4627                       0 </code></pre>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.6gene.thresh0)), aes(x=OstAdHoc.Assign.6gene.thresh0, fill=Species)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.6gene.thresh0)), aes(x=OstAdHoc.Assign.6gene.thresh0, fill=Species)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.6gene.thresh0, y=..prop.., group=Species), stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.6gene.thresh0)), aes(x=OstAdHoc.Assign.6gene.thresh0, fill=Individual)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.6gene.thresh0)), aes(x=OstAdHoc.Assign.6gene.thresh0, fill=Individual)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.6gene.thresh0, y=..prop.., group=Individual), stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(data,
        group.by=&quot;Species&quot;,
        split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
        reduction=&quot;umap&quot;,
        cols=c(&quot;grey&quot;,&quot;black&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<pre><code>Warning: Graphs cannot be vertically aligned unless the axis parameter is
set. Placing graphs unaligned.</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(DimPlot(data,
              group.by=&quot;Stage&quot;,
              split.by=&quot;OstAdHoc.Assign.6gene.thresh0&quot;,
              reduction=&quot;umap&quot;,
              cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
        labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-40-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make nicer plots of osteogenic ad hoc assignments (4 gene, thresh0).</p>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Human&quot;)@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                   1014                     838                     675 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   1178                    1691                     890 
         not osteogenic                   other 
                  41124                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Species==&quot;Chimp&quot;)@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    990                    3040                    2344 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   3295                     752                     868 
         not osteogenic                   other 
                  42301                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 0&quot;)@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      0                       5                       4 
         not osteogenic                   other 
                  35579                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 1&quot;)@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    157                      15                     123 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                    103                     392                     188 
         not osteogenic                   other 
                  36250                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=Stage==&quot;Time 2&quot;)@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                   1847                    3863                    2896 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   4370                    2046                    1566 
         not osteogenic                   other 
                  11596                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      0                       5                       4 
         not osteogenic                   other 
                  16687                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    156                      13                     102 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                     89                     313                     180 
         not osteogenic                   other 
                  17468                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Human&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    858                     825                     573 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   1089                    1373                     706 
         not osteogenic                   other 
                   6969                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 0&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      0                       0                       0 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                      0                       0                       0 
         not osteogenic                   other 
                  18892                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 1&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                      1                       2                      21 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                     14                      79                       8 
         not osteogenic                   other 
                  18782                       0 </code></pre>
<pre class="r"><code>table(subset(data,subset=(Species==&quot;Chimp&quot; &amp; Stage==&quot;Time 2&quot;))@meta.data$OstAdHoc.Assign.4gene.thresh0)</code></pre>
<pre><code>
          preosteoblast              osteoblast    embedding osteoblast 
                    989                    3038                    2323 
mineralizing osteoblast      maturing osteocyte      unknown osteogenic 
                   3281                     673                     860 
         not osteogenic                   other 
                   4627                       0 </code></pre>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.4gene.thresh0)), aes(x=OstAdHoc.Assign.4gene.thresh0, fill=Species)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.4gene.thresh0)), aes(x=OstAdHoc.Assign.4gene.thresh0, fill=Species)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.4gene.thresh0, y=..prop.., group=Species), stat=&quot;count&quot;, position=&quot;dodge&quot;) +
      scale_fill_manual(values=c(&quot;grey&quot;,&quot;black&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.4gene.thresh0)), aes(x=OstAdHoc.Assign.4gene.thresh0, fill=Individual)) +
      geom_bar(stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(subset(data@meta.data, !is.na(OstAdHoc.Assign.4gene.thresh0)), aes(x=OstAdHoc.Assign.4gene.thresh0, fill=Individual)) +
      geom_bar(mapping = aes(x=OstAdHoc.Assign.4gene.thresh0, y=..prop.., group=Individual), stat=&quot;count&quot;, position=&quot;dodge&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(data,
        group.by=&quot;Species&quot;,
        split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
        reduction=&quot;umap&quot;,
        cols=c(&quot;grey&quot;,&quot;black&quot;)) +
  labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Species==&quot;Human&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Species==&quot;Chimp&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CombinePlots(plots=list((DimPlot(subset(data,subset=Stage==&quot;Time 0&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 1&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;lightgrey&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;)),
                        (DimPlot(subset(data,subset=Stage==&quot;Time 2&quot;),
                                 group.by=&quot;Stage&quot;,
                                 split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
                                 reduction=&quot;umap&quot;,
                                 cols=c(&quot;dodgerblue&quot;)) +
                           labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))), ncol=1)</code></pre>
<pre><code>Warning: Graphs cannot be vertically aligned unless the axis parameter is
set. Placing graphs unaligned.</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(DimPlot(data,
              group.by=&quot;Stage&quot;,
              split.by=&quot;OstAdHoc.Assign.4gene.thresh0&quot;,
              reduction=&quot;umap&quot;,
              cols=c(&quot;lightgrey&quot;,&quot;lightgrey&quot;,&quot;dodgerblue&quot;)) +
        labs(x=&quot;UMAP1&quot;,y=&quot;UMAP2&quot;))</code></pre>
<p><img src="figure/10x-data-4-classification-tot.Rmd/unnamed-chunk-41-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(data, file=&quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] fastTopics_0.3-185 classtpx_0.99.0    slam_0.1-45       
 [4] CountClust_1.12.0  reshape2_1.4.3     UpSetR_1.4.0      
 [7] tidyr_1.1.2        RColorBrewer_1.1-2 colorspace_2.0-0  
[10] gbm_2.1.5          pheatmap_1.0.12    ggplot2_3.3.2     
[13] stringr_1.4.0      stringi_1.4.3      dplyr_1.0.2       
[16] Seurat_3.1.3      

loaded via a namespace (and not attached):
  [1] Rtsne_0.15          ellipsis_0.3.1      modeltools_0.2-22  
  [4] ggridges_0.5.1      rprojroot_2.0.2     fs_1.3.1           
  [7] farver_2.0.3        leiden_0.3.1        listenv_0.7.0      
 [10] npsurv_0.4-0        MatrixModels_0.4-1  ggrepel_0.8.2      
 [13] flexmix_2.3-15      codetools_0.2-16    splines_3.6.1      
 [16] lsei_1.2-0          knitr_1.23          jsonlite_1.7.1     
 [19] workflowr_1.6.2     mcmc_0.9-7          ica_1.0-2          
 [22] cluster_2.1.0       png_0.1-7           uwot_0.1.5         
 [25] sctransform_0.2.0   compiler_3.6.1      httr_1.4.2         
 [28] Matrix_1.2-18       lazyeval_0.2.2      limma_3.40.6       
 [31] later_1.1.0.1       prettyunits_1.1.1   quantreg_5.75      
 [34] htmltools_0.5.0     tools_3.6.1         rsvd_1.0.1         
 [37] igraph_1.2.4.1      coda_0.19-4         gtable_0.3.0       
 [40] glue_1.4.2          RANN_2.6.1          rappdirs_0.3.1     
 [43] Rcpp_1.0.5          vctrs_0.3.5         gdata_2.18.0       
 [46] ape_5.4-1           nlme_3.1-140        conquer_1.0.2      
 [49] gbRd_0.4-11         lmtest_0.9-37       xfun_0.8           
 [52] globals_0.12.4      lifecycle_0.2.0     irlba_2.3.3        
 [55] gtools_3.8.1        future_1.14.0       MASS_7.3-51.4      
 [58] zoo_1.8-6           scales_1.1.1        hms_0.5.3          
 [61] promises_1.1.1      parallel_3.6.1      SparseM_1.78       
 [64] yaml_2.2.1          reticulate_1.16     pbapply_1.4-0      
 [67] gridExtra_2.3       SQUAREM_2017.10-1   permute_0.9-5      
 [70] caTools_1.17.1.2    bibtex_0.4.2        matrixStats_0.57.0 
 [73] Rdpack_0.11-0       rlang_0.4.9         pkgconfig_2.0.3    
 [76] bitops_1.0-6        evaluate_0.14       lattice_0.20-38    
 [79] ROCR_1.0-7          purrr_0.3.4         labeling_0.4.2     
 [82] htmlwidgets_1.5.2   maptpx_1.9-7        cowplot_1.1.0      
 [85] tidyselect_1.1.0    RcppAnnoy_0.0.12    plyr_1.8.4         
 [88] magrittr_2.0.1      R6_2.5.0            gplots_3.0.1.1     
 [91] generics_0.1.0      picante_1.8         mgcv_1.8-28        
 [94] pillar_1.4.7        whisker_0.3-2       withr_2.3.0        
 [97] fitdistrplus_1.0-14 survival_2.44-1.1   nnet_7.3-12        
[100] tibble_3.0.4        future.apply_1.3.0  tsne_0.1-3         
[103] crayon_1.3.4        KernSmooth_2.23-15  plotly_4.9.2.1     
[106] rmarkdown_1.13      progress_1.2.2      data.table_1.13.2  
[109] vegan_2.5-5         git2r_0.26.1        metap_1.1          
[112] digest_0.6.27       httpuv_1.5.1        MCMCpack_1.4-9     
[115] RcppParallel_5.0.2  stats4_3.6.1        munsell_0.5.0      
[118] viridisLite_0.3.0   quadprog_1.5-8     </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
