<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-plot-diffexp</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-plot-diffexp</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">July 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.DE-signif.csv
    Untracked:  data/gom-data/topics10.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics15.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics2.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics20.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics30.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics8.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics9.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-plot-diffexp.Rmd</code>) and HTML (<code>docs/10x-plot-diffexp.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/analysis/10x-plot-diffexp.Rmd" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
<td>
updated enrich overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/535b9e22ebc8ab063d62637e164396414bce3022/docs/10x-plot-diffexp.html" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
<td>
updated enrich overlap
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/fb816c55da607ebfdd87caca43c792b61b8450d8/analysis/10x-plot-diffexp.Rmd" target="_blank">fb816c5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-10
</td>
<td>
updated GO enrichment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/fb816c55da607ebfdd87caca43c792b61b8450d8/docs/10x-plot-diffexp.html" target="_blank">fb816c5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-10
</td>
<td>
updated GO enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/analysis/10x-plot-diffexp.Rmd" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
fixed enrichment plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/10x-plot-diffexp.html" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
fixed enrichment plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/b97d07b7d6e8928f1c68741df8773f74d6cbf4c9/analysis/10x-plot-diffexp.Rmd" target="_blank">b97d07b</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-03
</td>
<td>
checked pairwise DE
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/analysis/10x-plot-diffexp.Rmd" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
additional docs updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/b27d03159139d65981b9cce9452c116e67870001/docs/10x-plot-diffexp.html" target="_blank">b27d031</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
docs updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/eeeab5a35b24f8f36709823ea3f92e50aa6b9288/analysis/10x-plot-diffexp.Rmd" target="_blank">eeeab5a</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
minor fixes to analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e54f13853433f82c3dd33d33f0140d3210999ea9/analysis/10x-plot-diffexp.Rmd" target="_blank">e54f138</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-08
</td>
<td>
fixed analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e62c53703ed251e27c4d23397131fd218013a635/analysis/10x-plot-diffexp.Rmd" target="_blank">e62c537</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-05
</td>
<td>
updated analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/d4afda66409b590e4e2c147b459254c8ee44093f/analysis/10x-plot-diffexp.Rmd" target="_blank">d4afda6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-10-06
</td>
<td>
updated variance code and sc vs. pseudo
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---differential-expression" class="section level1">
<h1>10X Data Cellranger Analysis - Differential Expression</h1>
<p>View results of pairwise differential expression tests between humans and chimpanzees within different cell classifications.</p>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ggsignif)</code></pre>
<pre class="r"><code>#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna &lt;- &quot;./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds&quot;
data &lt;- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate &lt;- paste0(data@meta.data$Individual,&quot;.&quot;,data@meta.data$Replicate)</code></pre>
<pre class="r"><code>genes &lt;- rownames(data@assays$RNA@counts)
genes.mito &lt;- c(&quot;MT-ATP6&quot;,&quot;MT-ATP8&quot;,&quot;MT-CO1&quot;,&quot;MT-CO2&quot;,&quot;MT-CO3&quot;,&quot;MT-CYB&quot;,&quot;MT-ND1&quot;,&quot;MT-ND2&quot;,&quot;MT-ND3&quot;,&quot;MT-ND4&quot;,&quot;MT-ND4L&quot;,&quot;MT-ND5&quot;)
genes.ribo &lt;- grep(&#39;^RP&#39;,genes,value=T)
genes.no.mito.ribo &lt;- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)</code></pre>
<pre class="r"><code>#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich &lt;- function(totGenes, deGenes, ontology) {
  ref.df &lt;- bitr(totGenes, fromType=&quot;SYMBOL&quot;, toType=c(&quot;ENSEMBL&quot;,&quot;ENTREZID&quot;), OrgDb=org.Hs.eg.db)
  gene.df &lt;- bitr(deGenes, fromType=&quot;SYMBOL&quot;, toType=c(&quot;ENSEMBL&quot;,&quot;ENTREZID&quot;), OrgDb=org.Hs.eg.db)
  ego &lt;- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = &#39;ENSEMBL&#39;,
                  ont           = ontology,
                  pAdjustMethod = &quot;BH&quot;,
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot &lt;- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split=&#39;ONTOLOGY&#39;) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale=&quot;free&quot;) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option=&quot;C&quot;) +
          theme_bw() +
          labs(title=title, x=&quot;Gene Ratio&quot;, y=&quot;&quot;))
}

#Define function to plot GO functional enrichment results (specific ontology only)
goEnPlot.spec &lt;- function(goTot, title, ontology) {
  print(ggplot(goTot[which(goTot$ONTOLOGY==ontology),], aes(x=GeneRatio, y=Description, color=p.adjust, size=Count)) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option=&quot;C&quot;) +
          theme_bw() +
          labs(title=title, x=&quot;Gene Ratio&quot;, y=&quot;&quot;))
}</code></pre>
<div id="load-external-gene-datasets" class="section level2">
<h2>Load External Gene Datasets</h2>
<p>Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.</p>
<p>Genes previously identified as human and chimpanzee differences:</p>
<ul>
<li>Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
<ul>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Blake et al. 2020 Genome Research (Supplemental Table S4)
<ul>
<li>DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each tissue type</li>
</ul></li>
<li>Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
<ul>
<li>DE genes in parental iPSC lines (n = 7655)</li>
<li>DE genes in hybrid iPSC lines (n = 6009)</li>
<li>DE genes in parental CNCC lines (n = 2223)</li>
<li>DE genes in hybrid CNCC lines (n = 3612)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
<ul>
<li>DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)</li>
<li>AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)</li>
<li>(human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)</li>
<li>(chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)</li>
<li>Note: only DMRs that overlapped promotor regions of genes were included</li>
<li>Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels</li>
</ul></li>
</ul>
<p>General skeleton-related and skeletal phenotype-related genes:</p>
<ul>
<li>Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
<ul>
<li>hip geometry GWAS loci (n = 10)</li>
</ul></li>
<li>Styrkarsdottir et al. 2019 Nature Communications (Table 1)
<ul>
<li>DXA bone area GWAS loci (n = 21 associated with 20 genes)</li>
</ul></li>
<li>Yengo et al. 2018 Human Molecular Genetics
<ul>
<li>height GWAS loci (n = 3290) (data provided at <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files" class="uri">https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files</a>)</li>
<li>height GWAS loci with associated eQTLs (n = 610) (list not provided)</li>
<li>Note: only used unique gene symbols associated with height GWAS loci (n = 1500)</li>
</ul></li>
</ul>
<p>Genes associated with skeletal diseases (OA + OP):</p>
<ul>
<li>Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
<ul>
<li>genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)</li>
</ul></li>
<li>Morris et al. 2019 Nature Genetics (Supplemental Table 2)
<ul>
<li>OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)</li>
</ul></li>
</ul>
<p>Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:</p>
<ul>
<li>Gallego Romero et al. 2015 eLife (Supplementary File 3)
<ul>
<li>DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR &lt; 1% out of 12171 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>REMOVED because subsequent studies (Blake et al. 2018 &amp; Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm &amp; CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear</li>
</ul></li>
<li>Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type</li>
</ul></li>
<li>Blake et al. 2020 Genome Research (Supplemental Table S4)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type</li>
</ul></li>
<li>Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type</li>
</ul></li>
<li>Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
<ul>
<li>Skeleton-related genes (n = 1790)</li>
<li>Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)</li>
<li>REMOVED because it is unclear where these genes came from to interpreting biology is difficult</li>
</ul></li>
</ul>
<p>Some external gene datasets were considered but not examined:</p>
<ul>
<li>Ward et al. 2019 eLife
<ul>
<li>DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR &lt; 10% out of 11974 genes)</li>
</ul></li>
<li>Shibata et al. 2012 PLOS Genetics
<ul>
<li>increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Edsall et al. 2019 Genome Biology and Evolution
<ul>
<li>increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
<ul>
<li>increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Prescott et al. 2015 Cell (Supplemental)
<ul>
<li>top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)</li>
<li>top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Joganic et al. 2017 American Journal of Physical Anthropology
<ul>
<li>genetics of craniofacial skeleton in baboons</li>
</ul></li>
<li>Cibrián Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
<ul>
<li>robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)</li>
</ul></li>
<li>Tachmazidou et al. 2019 Nature Genetics
<ul>
<li>OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])</li>
<li>OA loci (Supplementary Table 2)</li>
<li>established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)</li>
<li>Note: used loci compiled by AH</li>
<li>(OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)</li>
<li>(OA.estab = gene symbol(s) nearest to established OA loci)</li>
<li>(OA.broad = genes within 500kb of OA loci)</li>
</ul></li>
</ul>
<pre class="r"><code>#Define function for testing significant enrichment of gene sets

genesetEnrich &lt;- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative=&quot;greater&quot;))

}

genesetEnrich.data &lt;- function(InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test and output data
  data &lt;- data.frame(&quot;Source.DEGenes&quot;=InternalDataSource,
                     &quot;Source.GenesOfInterest&quot;=ExternalDataSource,
                     &quot;DE.Interest&quot;=DE.Interest,
                     &quot;NonDE.Interest&quot;=NonDE.Interest,
                     &quot;DE.NotInterest&quot;=DE.NotInterest,
                     &quot;NonDE.NotInterest&quot;=NonDE.NotInterest,
                     &quot;p.value&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$p.value,
                     &quot;odds.ratio&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$estimate,
                     row.names=NULL)
  return(data)

}</code></pre>
<pre class="r"><code>#Load external datasets (manually currated from papers)
gallegoromero2015    &lt;- read.csv(&quot;./data/external-gene-sets/GallegoRomero2015eLife.csv&quot;)
blake2018            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2018GenBio.csv&quot;)
blake2020            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2020GenRes.csv&quot;)
gokhman2021          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2021NatGen.csv&quot;)
gokhman2020          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm.csv&quot;)
gokhman2020.skeleton &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv&quot;)
hsu2019              &lt;- read.csv(&quot;./data/external-gene-sets/Hsu2019JBoneMinRes.csv&quot;)
styrkarsdottir2019   &lt;- read.csv(&quot;./data/external-gene-sets/Styrkarsdottir2019NatComm.csv&quot;)
yengo2018            &lt;- read.csv(&quot;./data/external-gene-sets/Yengo2018HumMolGen.csv&quot;)
steinberg2020        &lt;- read.csv(&quot;./data/external-gene-sets/Steinberg2020biorxiv.csv&quot;)
morris2019           &lt;- read.csv(&quot;./data/external-gene-sets/Morris2019NatGen.csv&quot;)
#cibrianuhalte2017    &lt;- read.csv(&quot;./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv&quot;)
#tachmazidou2019      &lt;- read.csv(&quot;./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv&quot;)</code></pre>
<pre class="r"><code>#genes previously identified as human and chimpanzee differences

#genesExternal &lt;- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                                blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==1 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                              blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==1 &amp;
                                              blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                              blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==1 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==1 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                               gokhman2021$DE.Hybrid.CNCCs==0)]   #hybrid.iPSCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal &lt;- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal &lt;- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal &lt;- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal &lt;- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal &lt;- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS</code></pre>
</div>
<div id="examine-output-of-de-analyses" class="section level2">
<h2>Examine output of DE analyses</h2>
<pre class="r"><code>#Load DE results

compare1  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.rds&quot;)
compare2  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.rds&quot;)
compare3  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhoc.rds&quot;)
compare4  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.rds&quot;)
compare5  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.75.rds&quot;)
compare6  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.T2.rds&quot;)
compare7  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.75.T2.rds&quot;)
compare8  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.rds&quot;)
compare9  &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocX.rds&quot;)
compare10 &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.T2.rds&quot;)
compare11 &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocX.T2.rds&quot;)

DEdata &lt;- rbind(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9,compare10,compare11)
colnames(DEdata)[5] &lt;- &quot;FDR&quot;
DEdata &lt;- as_tibble(DEdata)

rm(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9,compare10,compare11)</code></pre>
<pre class="r"><code>#Isolate significant DE genes

DEdataSig &lt;- DEdata %&gt;% filter(FDR&lt;=0.01)
#DEdataSig &lt;- DEdata %&gt;% filter(FDR&lt;=0.05)
#DEdataSig &lt;- DEdata %&gt;% filter(abs(logFC)&gt;=1.2 &amp; FDR&lt;=0.01)</code></pre>
</div>
<div id="examine-numbers-overlap-and-logfc-of-pairwise-de-genes" class="section level2">
<h2>Examine Numbers, Overlap, and logFC of Pairwise DE Genes</h2>
<div id="stages-of-differentiation-at-collection" class="section level3">
<h3>Stages of Differentiation at Collection</h3>
<p>DE Genes Counts</p>
<ul>
<li>Time 0: (FDR&lt;0.01) = 2098 / (FDR&lt;0.05) = 4002</li>
<li>Time 1: (FDR&lt;0.01) = 904 / (FDR&lt;0.05) = 2441</li>
<li>Time 2: (FDR&lt;0.01) = 446 / (FDR&lt;0.05) = 1603</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>Time 0 vs. Time 1: (FDR&lt;0.01): p = 3.3e-11 / (FDR&lt;0.05): p &lt; 2.2e-16 **</li>
<li>Time 1 vs. Time 2: (FDR&lt;0.01): p = 8.2e-06 / (FDR&lt;0.05): p = 3.3e-15 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
Time 0 Time 1 Time 2 
  2098    904    446 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;,
                    labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-10-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;Time.0&quot;),
                         list(&quot;Time.1&quot;),
                         list(&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.1&quot;),
                         list(&quot;Time.1&quot;,&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-10-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fill=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;,
                    labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10)) +
  geom_signif(comparisons = list(c(&quot;Time 0&quot;,&quot;Time 1&quot;),
                                 c(&quot;Time 1&quot;,&quot;Time 2&quot;)), 
              test=&quot;t.test&quot;,
              test.args=list(alternative=&quot;two.sided&quot;),
              map_signif_level=TRUE)

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Time 0&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Time 1&quot;)])) #p=3.3e-11</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Time 0&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Time 1&quot;)])
t = -6.6865, df = 1358.4, p-value = 3.329e-11
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4016591 -0.2194388
sample estimates:
mean of x mean of y 
 1.337391  1.647940 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Time 1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Time 2&quot;)])) #p=8.2e-06</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Time 1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Time 2&quot;)])
t = -4.4922, df = 750.75, p-value = 8.16e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.5375910 -0.2106169
sample estimates:
mean of x mean of y 
 1.647940  2.022044 </code></pre>
<pre class="r"><code>DEdataCells &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 0&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 0&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: Time 0&quot;,
       ylim=c(0,10),
       xlim=c(0,4.75))
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 1&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 1&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: Time 1&quot;,
       ylim=c(0,8.5),
       xlim=c(0,4.5))
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 2&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;Time 2&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: Time 2&quot;)
abline(0,1)</code></pre>
</div>
<div id="general-clusters-res0.05" class="section level3">
<h3>General Clusters (res=0.05)</h3>
<p>DE Genes</p>
<ul>
<li>FDR &lt; 0.01
<ul>
<li>iPSC.c1: (FDR&lt;0.01) = 2155 / (FDR&lt;0.05) = 4093</li>
<li>iPSC.c2: (FDR&lt;0.01) = 413 / (FDR&lt;0.05) = 1168</li>
<li>iPSC.c3: (FDR&lt;0.01) = 0 / (FDR&lt;0.05) = 81</li>
<li>MSC.c1: (FDR&lt;0.01) = 936 / (FDR&lt;0.05) = 2494</li>
<li>Osteogenic.c1: (FDR&lt;0.01) = 534 / (FDR&lt;0.05) = 1807</li>
<li>Osteogenic.c2: (FDR&lt;0.01) = 209 / (FDR&lt;0.05) = 1120</li>
</ul></li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>iPSC.c1 vs. MSC.c1: (FDR&lt;0.01): p = 5.3e-11 / (FDR&lt;0.05): p &lt; 2.2e-16 **</li>
<li>MSC.c1 vs. Osteogenic.c1: (FDR&lt;0.01): p = 8.9e-06 / (FDR&lt;0.05): p = 8.1e-12 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;cluster&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
      iPSC.c1       iPSC.c2       iPSC.c3        MSC.c1 Osteogenic.c1 
         2155           413             0           936           534 
Osteogenic.c2 
          209 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                    values=c(&quot;pink&quot;,&quot;pink3&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-12-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;iPSC.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC.c1&quot;)]))       #p=5.3e-11</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;iPSC.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC.c1&quot;)])
t = -6.6148, df = 1411.8, p-value = 5.266e-11
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3919930 -0.2126755
sample estimates:
mean of x mean of y 
 1.332429  1.634763 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)])) #p=8.9e-06</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)])
t = -4.4652, df = 960.74, p-value = 8.947e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4911190 -0.1912309
sample estimates:
mean of x mean of y 
 1.634763  1.975938 </code></pre>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;cluster&quot;) %&gt;%
    dplyr::filter(cell.subset %in% c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)) %&gt;% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;iPSC.c1&quot;) %&gt;% dplyr::select(gene), &quot;iPSC.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;MSC.c1&quot;) %&gt;% dplyr::select(gene), &quot;MSC.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;iPSC.c1&quot;),
                         list(&quot;MSC.c1&quot;),
                         list(&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;),
                         list(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-13-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;iPSC.c1&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;MSC.c1&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fill=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;iPSC.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC.c1&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;iPSC.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC.c1&quot;)])
t = -6.6148, df = 1411.8, p-value = 5.266e-11
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3919930 -0.2126755
sample estimates:
mean of x mean of y 
 1.332429  1.634763 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)])
t = -4.4652, df = 960.74, p-value = 8.947e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4911190 -0.1912309
sample estimates:
mean of x mean of y 
 1.634763  1.975938 </code></pre>
</div>
<div id="general-ad-hoc-classification-of-cells-scaled-integrated-data-threshold0" class="section level3">
<h3>General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)</h3>
<p>DE Genes</p>
<ul>
<li>iPSC (FDR&lt;0.01) = 2128 / (FDR&lt;0.05) = 4009</li>
<li>MSC (FDR&lt;0.01) = 963 / (FDR&lt;0.05) = 2554</li>
<li>Osteogenic (FDR&lt;0.01) = 613 / (FDR&lt;0.05) = 2005</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>iPSC vs. MSC: (FDR&lt;0.01): p = 4.3e-09 / (FDR&lt;0.05): p &lt; 2.2e-16 **</li>
<li>MSC vs. Osteogenic: (FDR&lt;0.01): p = 1.8e-06 / (FDR&lt;0.05): p = 4.0e-12 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;adhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic 
      2128        963        613 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;,
                    labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;iPSC&quot;) %&gt;% dplyr::select(gene), &quot;iPSC&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;MSC&quot;) %&gt;% dplyr::select(gene), &quot;MSC&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;iPSC&quot;),
                         list(&quot;MSC&quot;),
                         list(&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;MSC&quot;),
                         list(&quot;MSC&quot;,&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-14-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;iPSC&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;MSC&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic&quot;) %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fill=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Ad Hoc&quot;,
                    labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;iPSC&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC&quot;)]))       #p=4.344e-09</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;iPSC&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC&quot;)])
t = -5.9059, df = 1477.3, p-value = 4.344e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3550223 -0.1779900
sample estimates:
mean of x mean of y 
 1.330313  1.596820 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;MSC&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic&quot;)])) #p=1.753e-06</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;MSC&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic&quot;)])
t = -4.8048, df = 1155.8, p-value = 1.753e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4828623 -0.2028534
sample estimates:
mean of x mean of y 
 1.596820  1.939677 </code></pre>
</div>
<div id="osteogenic-clusters-res0.50" class="section level3">
<h3>Osteogenic Clusters (res=0.50)</h3>
<p>DE Genes</p>
<ul>
<li>Osteogenic.c1 (FDR&lt;0.01) = 643 / (FDR&lt;0.05) = 2057</li>
<li>Osteogenic.c2 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 9</li>
<li>Osteogenic.c3 (FDR&lt;0.01) = 428 / (FDR&lt;0.05) = 1562</li>
<li>Osteogenic.c4 (FDR&lt;0.01) = 217 / (FDR&lt;0.05) = 1087</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>Osteogenic.c1 vs. Osteogenic.c3: (FDR&lt;0.01): p = 0.00311 / (FDR&lt;0.05): p = 0.00518 **</li>
<li>Osteogenic.c3 vs. Osteogenic.c4: (FDR&lt;0.01): p = 0.01908 / (FDR&lt;0.05): p = 0.12870 *</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.50&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4 
          643             1           428           217 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-15-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)])
t = -2.9648, df = 884.66, p-value = 0.00311
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.44910781 -0.09134334
sample estimates:
mean of x mean of y 
 1.824694  2.094919 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)])
t = -2.3518, df = 487.27, p-value = 0.01908
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.49281806 -0.04417443
sample estimates:
mean of x mean of y 
 2.094919  2.363415 </code></pre>
</div>
<div id="osteogenic-clusters-res0.75" class="section level3">
<h3>Osteogenic Clusters (res=0.75)</h3>
<p>DE Genes</p>
<ul>
<li>Osteogenic.c1 (FDR&lt;0.01) = 364 / (FDR&lt;0.05) = 1478</li>
<li>Osteogenic.c2 (FDR&lt;0.01) = 433 / (FDR&lt;0.05) = 1773</li>
<li>Osteogenic.c3 (FDR&lt;0.01) = 427 / (FDR&lt;0.05) = 1493</li>
<li>Osteogenic.c4 (FDR&lt;0.01) = 196 / (FDR&lt;0.05) = 1123</li>
<li>Osteogenic.c5 (FDR&lt;0.01) = 271 / (FDR&lt;0.05) = 1151</li>
<li>Osteogenic.c6 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 2</li>
<li>Osteogenic.c7 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 1</li>
<li>Osteogenic.c8 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 3</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>Osteogenic.c1 vs. Osteogenic.c2: (FDR&lt;0.01): p = 0.91230 / (FDR&lt;0.05): p = 0.83610</li>
<li>Osteogenic.c2 vs. Osteogenic.c3: (FDR&lt;0.01): p = 3.5e-07 / (FDR&lt;0.05): p = 1.2e-10 **</li>
<li>Osteogenic.c3 vs. Osteogenic.c4: (FDR&lt;0.01): p = 0.12800 / (FDR&lt;0.05): p = 0.00086 *</li>
<li>Osteogenic.c4 vs. Osteogenic.c5: (FDR&lt;0.01): p = 0.05663 / (FDR&lt;0.05): p = 4.7e-05 *</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.75&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                       &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4 Osteogenic.c5 
          364           433           427           196           271 
Osteogenic.c6 Osteogenic.c7 Osteogenic.c8 
            1             1             1 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                    values=c(&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;deepskyblue2&quot;,&quot;navyblue&quot;,&quot;lightblue1&quot;,&quot;royalblue3&quot;,&quot;grey&quot;,&quot;black&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c5&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c5&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c6&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c6&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c7&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c7&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c8&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c8&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-16-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code># Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                    values=c(&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;deepskyblue2&quot;,&quot;navyblue&quot;,&quot;lightblue1&quot;,&quot;royalblue3&quot;,&quot;grey&quot;,&quot;black&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c2&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c2&quot;)])
t = 0.11014, df = 761.94, p-value = 0.9123
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.1909239  0.2136217
sample estimates:
mean of x mean of y 
 2.019983  2.008634 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c2&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c2&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)])
t = -5.1351, df = 835.23, p-value = 3.512e-07
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.7474087 -0.3340397
sample estimates:
mean of x mean of y 
 2.008634  2.549358 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)])
t = 1.5247, df = 471.75, p-value = 0.128
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.05412092  0.42896212
sample estimates:
mean of x mean of y 
 2.549358  2.361937 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c5&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c5&quot;)])
t = 1.9111, df = 448.66, p-value = 0.05663
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.007047765  0.504235590
sample estimates:
mean of x mean of y 
 2.361937  2.113343 </code></pre>
</div>
<div id="osteogenic-clusters-res0.50-time-2-only" class="section level3">
<h3>Osteogenic Clusters (res=0.50) (Time 2 only)</h3>
<p>DE Genes</p>
<ul>
<li>Osteogenic.c1 (FDR&lt;0.01) = 545 / (FDR&lt;0.05) = 1796</li>
<li>Osteogenic.c2 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 9</li>
<li>Osteogenic.c3 (FDR&lt;0.01) = 388 / (FDR&lt;0.05) = 1513</li>
<li>Osteogenic.c4 (FDR&lt;0.01) = 79 / (FDR&lt;0.05) = 406</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>Osteogenic.c1 vs. Osteogenic.c3: (FDR&lt;0.01): p = 0.02433 / (FDR&lt;0.05): p = 0.07446 *</li>
<li>Osteogenic.c3 vs. Osteogenic.c4: (FDR&lt;0.01): p = 2.7e-06 / (FDR&lt;0.05): p = 9.2e-13 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.50.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4 
          545             1           388            79 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-2">
Past versions of unnamed-chunk-17-2.png
</button>
</p>
<div id="fig-unnamed-chunk-17-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-17-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;dodgerblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)])
t = -2.2561, df = 822.7, p-value = 0.02433
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.41453988 -0.02880913
sample estimates:
mean of x mean of y 
 1.893534  2.115209 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)])
t = -4.9447, df = 112.35, p-value = 2.69e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.2714769 -0.5440245
sample estimates:
mean of x mean of y 
 2.115209  3.022959 </code></pre>
</div>
<div id="osteogenic-clusters-res0.75-time-2-only" class="section level3">
<h3>Osteogenic Clusters (res=0.75) (Time 2 only)</h3>
<p>DE Genes</p>
<ul>
<li>Osteogenic.c1 (FDR&lt;0.01) = 210 / (FDR&lt;0.05) = 833</li>
<li>Osteogenic.c2 (FDR&lt;0.01) = 369 / (FDR&lt;0.05) = 1397</li>
<li>Osteogenic.c3 (FDR&lt;0.01) = 322 / (FDR&lt;0.05) = 1041</li>
<li>Osteogenic.c4 (FDR&lt;0.01) = 54 / (FDR&lt;0.05) = 387</li>
<li>Osteogenic.c5 (FDR&lt;0.01) = 268 / (FDR&lt;0.05) = 1078</li>
<li>Osteogenic.c6 (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 2</li>
<li>Osteogenic.c7 (FDR&lt;0.01) = 0 / (FDR&lt;0.05) = 0</li>
<li>Osteogenic.c8 (FDR&lt;0.01) = 0 / (FDR&lt;0.05) = 1</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>Osteogenic.c1 vs. Osteogenic.c2: (FDR&lt;0.01): p = 0.10230 / (FDR&lt;0.05): p = 0.11000</li>
<li>Osteogenic.c2 vs. Osteogenic.c3: (FDR&lt;0.01): p = 4.7e-06 / (FDR&lt;0.05): p = 6.3e-10 **</li>
<li>Osteogenic.c3 vs. Osteogenic.c4: (FDR&lt;0.01): p = 0.02699 / (FDR&lt;0.05): p = 0.30090 *</li>
<li>Osteogenic.c4 vs. Osteogenic.c5: (FDR&lt;0.01): p = 1.7e-05 / (FDR&lt;0.05): p = 3.5e-15 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.75.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                       &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4 Osteogenic.c5 
          210           369           322            54           268 
Osteogenic.c6 Osteogenic.c7 Osteogenic.c8 
            1             0             0 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                    values=c(&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;deepskyblue2&quot;,&quot;navyblue&quot;,&quot;lightblue1&quot;,&quot;royalblue3&quot;,&quot;grey&quot;,&quot;black&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c5&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c5&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c6&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c6&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-2">
Past versions of unnamed-chunk-18-2.png
</button>
</p>
<div id="fig-unnamed-chunk-18-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-18-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code># Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                             &quot;Osteogenic.c5&quot;,&quot;Osteogenic.c6&quot;,&quot;Osteogenic.c7&quot;,&quot;Osteogenic.c8&quot;),
                    values=c(&quot;dodgerblue&quot;,&quot;dodgerblue4&quot;,&quot;deepskyblue2&quot;,&quot;navyblue&quot;,&quot;lightblue1&quot;,&quot;royalblue3&quot;,&quot;grey&quot;,&quot;black&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c1&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c2&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c1&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c2&quot;)])
t = 1.6377, df = 404.73, p-value = 0.1023
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.0443984  0.4875747
sample estimates:
mean of x mean of y 
 2.319012  2.097424 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c2&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c2&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)])
t = -4.6182, df = 638.35, p-value = 4.682e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.8082571 -0.3259775
sample estimates:
mean of x mean of y 
 2.097424  2.664541 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c3&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c3&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)])
t = -2.2544, df = 77.613, p-value = 0.02699
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.96560675 -0.05991129
sample estimates:
mean of x mean of y 
 2.664541  3.177300 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c4&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;Osteogenic.c5&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c4&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;Osteogenic.c5&quot;)])
t = 4.6, df = 75.088, p-value = 1.68e-05
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.5882768 1.4869737
sample estimates:
mean of x mean of y 
 3.177300  2.139675 </code></pre>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)</h3>
<p>DE Genes</p>
<ul>
<li>pre (FDR&lt;0.01) = 144 / (FDR&lt;0.05) = 657</li>
<li>obl (FDR&lt;0.01) = 242 / (FDR&lt;0.05) = 996</li>
<li>emb (FDR&lt;0.01) = 142 / (FDR&lt;0.05) = 772</li>
<li>min (FDR&lt;0.01) = 429 / (FDR&lt;0.05) = 1569</li>
<li>mat (FDR&lt;0.01) = 115 / (FDR&lt;0.05) = 761</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>pre vs. obl: (FDR&lt;0.01): p = 0.71640 / (FDR&lt;0.05): p = 0.87480</li>
<li>obl vs. emb: (FDR&lt;0.01): p = 0.47320 / (FDR&lt;0.05): p = 0.55990</li>
<li>emb vs. min: (FDR&lt;0.01): p = 0.02644 / (FDR&lt;0.05): p = 0.00020 **</li>
<li>min vs. mat: (FDR&lt;0.01): p = 0.01655 / (FDR&lt;0.05): p = 2.4e-06 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                              labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                     144                      242                      142 
mineralizing\nosteoblast      maturing\nosteocyte 
                     429                      115 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-2">
Past versions of unnamed-chunk-19-2.png
</button>
</p>
<div id="fig-unnamed-chunk-19-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-3">
Past versions of unnamed-chunk-19-3.png
</button>
</p>
<div id="fig-unnamed-chunk-19-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-3.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-19-3.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;preosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;preosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)])
t = 0.36363, df = 312.57, p-value = 0.7164
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2658654  0.3864117
sample estimates:
mean of x mean of y 
 2.471183  2.410909 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)])
t = -0.71832, df = 279.34, p-value = 0.4732
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4821089  0.2243244
sample estimates:
mean of x mean of y 
 2.410909  2.539802 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)])
t = 2.2341, df = 229.19, p-value = 0.02644
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.0437625 0.6977508
sample estimates:
mean of x mean of y 
 2.539802  2.169045 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;maturing\nosteocyte&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;maturing\nosteocyte&quot;)])
t = -2.4181, df = 189.6, p-value = 0.01655
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.71680694 -0.07274374
sample estimates:
mean of x mean of y 
 2.169045  2.563820 </code></pre>
<pre class="r"><code>DEdataCells &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;preosteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;preosteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: preosteoblast&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: osteoblast&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;embedding osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;embedding osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: embedding osteoblast&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;mineralizing osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;mineralizing osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: mineralizing osteoblast&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;maturing osteocyte&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;maturing osteocyte&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: maturing osteocyte&quot;)
abline(0,1)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)</h3>
<p>DE Genes</p>
<ul>
<li>pre (FDR&lt;0.01) = 144 / (FDR&lt;0.05) = 657</li>
<li>obl (FDR&lt;0.01) = 134 / (FDR&lt;0.05) = 680</li>
<li>emb (FDR&lt;0.01) = 36 / (FDR&lt;0.05) = 229</li>
<li>min (FDR&lt;0.01) = 57 / (FDR&lt;0.05) = 138</li>
<li>mat (FDR&lt;0.01) = 1 / (FDR&lt;0.05) = 6</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>pre vs. obl: (FDR&lt;0.01): p = 0.71240 / (FDR&lt;0.05): p = 0.37970</li>
<li>obl vs. emb: (FDR&lt;0.01): p = 0.01529 / (FDR&lt;0.05): p = 4.5e-08 **</li>
<li>emb vs. min: (FDR&lt;0.01): p = 0.00481 / (FDR&lt;0.05): p &lt; 2.2e-16 **</li>
<li>min vs. mat: (FDR&lt;0.01): p = n/a / (FDR&lt;0.05): p = 0.84870</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhocX&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                              labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                     144                      134                       36 
mineralizing\nosteoblast      maturing\nosteocyte 
                      57                        1 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-2">
Past versions of unnamed-chunk-21-2.png
</button>
</p>
<div id="fig-unnamed-chunk-21-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-2.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-3">
Past versions of unnamed-chunk-21-3.png
</button>
</p>
<div id="fig-unnamed-chunk-21-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-3.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-21-3.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;preosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;preosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)])
t = -0.36897, df = 269.42, p-value = 0.7124
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4538181  0.3105669
sample estimates:
mean of x mean of y 
 2.471183  2.542808 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)])
t = -2.507, df = 52.79, p-value = 0.01529
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.4972704 -0.1662361
sample estimates:
mean of x mean of y 
 2.542808  3.374561 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)])
t = -2.9387, df = 54.909, p-value = 0.004812
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.6703897 -0.3158286
sample estimates:
mean of x mean of y 
 3.374561  4.367671 </code></pre>
<pre class="r"><code>#t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;maturing\nosteocyte&quot;)]))</code></pre>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible-time-2-only" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)</h3>
<p>DE Genes</p>
<ul>
<li>pre (FDR&lt;0.01) = 121 / (FDR&lt;0.05) = 552</li>
<li>obl (FDR&lt;0.01) = 235 / (FDR&lt;0.05) = 951</li>
<li>emb (FDR&lt;0.01) = 74 / (FDR&lt;0.05) = 502</li>
<li>min (FDR&lt;0.01) = 241 / (FDR&lt;0.05) = 935</li>
<li>mat (FDR&lt;0.01) = 34 / (FDR&lt;0.05) = 378</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>pre vs. obl: (FDR&lt;0.01): p = 0.51260 / (FDR&lt;0.05): p = 0.75750</li>
<li>obl vs. emb: (FDR&lt;0.01): p = 0.03799 / (FDR&lt;0.05): p = 0.14660 *</li>
<li>emb vs. min: (FDR&lt;0.01): p = 0.00677 / (FDR&lt;0.05): p = 0.00106 **</li>
<li>min vs. mat: (FDR&lt;0.01): p = 0.00015 / (FDR&lt;0.05): p = 1.3e-07 **</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                              labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                     121                      235                       74 
mineralizing\nosteoblast      maturing\nosteocyte 
                     241                       34 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-2">
Past versions of unnamed-chunk-22-2.png
</button>
</p>
<div id="fig-unnamed-chunk-22-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-3">
Past versions of unnamed-chunk-22-3.png
</button>
</p>
<div id="fig-unnamed-chunk-22-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-22-3.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;preosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;preosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)])
t = 0.65571, df = 246.69, p-value = 0.5126
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2367875  0.4731277
sample estimates:
mean of x mean of y 
 2.539761  2.421591 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)])
t = -2.1007, df = 108.44, p-value = 0.03799
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.0072183 -0.0292538
sample estimates:
mean of x mean of y 
 2.421591  2.939827 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)])
t = 2.7622, df = 106.41, p-value = 0.006765
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.1913473 1.1643923
sample estimates:
mean of x mean of y 
 2.939827  2.261957 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;maturing\nosteocyte&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;maturing\nosteocyte&quot;)])
t = -4.1758, df = 42.273, p-value = 0.0001453
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.8680418 -0.6509123
sample estimates:
mean of x mean of y 
 2.261957  3.521434 </code></pre>
<pre class="r"><code>DEdataCells &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;preosteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;preosteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: preosteoblast (t2)&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: osteoblast (t2)&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;embedding osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;embedding osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: embedding osteoblast (t2)&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;mineralizing osteoblast&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;mineralizing osteoblast&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: mineralizing osteoblast (t2)&quot;)
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;maturing osteocyte&quot;)]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset==&quot;maturing osteocyte&quot;)]),
       xlab=&quot;-log10(Uniform)&quot;,
       ylab=&quot;-log10(FDR)&quot;,
       main=&quot;QQ Plot: maturing osteocyte (t2)&quot;)
abline(0,1)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict-time-2-only" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)</h3>
<p>DE Genes</p>
<ul>
<li>pre (FDR&lt;0.01) = 121 / (FDR&lt;0.05) = 552</li>
<li>obl (FDR&lt;0.01) = 121 / (FDR&lt;0.05) = 624</li>
<li>emb (FDR&lt;0.01) = 36 / (FDR&lt;0.05) = 239</li>
<li>min (FDR&lt;0.01) = 57 / (FDR&lt;0.05) = 144</li>
<li>mat (FDR&lt;0.01) = 0 / (FDR&lt;0.05) = 0</li>
</ul>
<p>Comparison in Absolute LogFC of DE Genes</p>
<ul>
<li>pre vs. obl: (FDR&lt;0.01): p = 0.86120 / (FDR&lt;0.05): p = 0.49660</li>
<li>obl vs. emb: (FDR&lt;0.01): p = 0.03134 / (FDR&lt;0.05): p = 3.8e-07 **</li>
<li>emb vs. min: (FDR&lt;0.01): p = 0.00306 / (FDR&lt;0.05): p &lt; 2.2e-16 **</li>
<li>min vs. mat: (FDR&lt;0.01): p = n/a / (FDR&lt;0.05): p = n/a</li>
</ul>
<pre class="r"><code>DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhocX.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                              levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                              labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

# Number of DE genes
table(DEgenes$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                     121                      121                       36 
mineralizing\nosteoblast      maturing\nosteocyte 
                      57                        0 </code></pre>
<pre class="r"><code>ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Overlap of DE genes
combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                        data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1)
                        ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>combined[is.na(combined)] &lt;- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-2">
Past versions of unnamed-chunk-24-2.png
</button>
</p>
<div id="fig-unnamed-chunk-24-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upset(combined,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      sets=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-3">
Past versions of unnamed-chunk-24-3.png
</button>
</p>
<div id="fig-unnamed-chunk-24-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-24-3.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>#v &lt;- venn.diagram(list(pull(DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene)),
#                       pull(DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene))),
#                  category.names = c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fill=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;),
#                  fontfamily=&quot;sans&quot;,
#                  cat.fontfamily=&quot;sans&quot;)
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab(&#39;Absolute Log2 Fold Change in DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;preosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;preosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)])
t = -0.17507, df = 238.6, p-value = 0.8612
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4591056  0.3841657
sample estimates:
mean of x mean of y 
 2.539761  2.577231 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;osteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;osteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)])
t = -2.2078, df = 56.382, p-value = 0.03134
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.40590426 -0.06841196
sample estimates:
mean of x mean of y 
 2.577231  3.314389 </code></pre>
<pre class="r"><code>t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;embedding\nosteoblast&quot;)]),abs(DEgenes$logFC[which(DEgenes$cell.subset==&quot;mineralizing\nosteoblast&quot;)]))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;embedding\nosteoblast&quot;)]) and abs(DEgenes$logFC[which(DEgenes$cell.subset == &quot;mineralizing\nosteoblast&quot;)])
t = -3.1001, df = 54.333, p-value = 0.003064
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.6983632 -0.3644671
sample estimates:
mean of x mean of y 
 3.314389  4.345804 </code></pre>
</div>
</div>
<div id="examine-interesting-de-genes-and-assess-functionalgeneset-enrichment" class="section level2">
<h2>Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment</h2>
<div id="stages-of-differentiation-at-collection-1" class="section level3">
<h3>Stages of Differentiation at Collection</h3>
<p>Pairwise DE Genes</p>
<ul>
<li>Shared across Stages: (FDR&lt;0.01) = 150 / (FDR&lt;0.05) = 530</li>
<li>Shared across Times 0 &amp; 1: (FDR&lt;0.01) = 250 / (FDR&lt;0.05) = 637</li>
<li>Shared across Times 1 &amp; 2: (FDR&lt;0.01) = 108 / (FDR&lt;0.05) = 393</li>
<li>Unique to Time 0: (FDR&lt;0.01) = 1648 / (FDR&lt;0.05) = 2640</li>
<li>Unique to Time 1: (FDR&lt;0.01) = 396 / (FDR&lt;0.05) = 881</li>
<li>Unique to Time 2: (FDR&lt;0.01) = 138 / (FDR&lt;0.05) = 485</li>
<li>Shared across Stages: (FDR&lt;0.01, tested in all) = 150 / (FDR&lt;0.05) = 530</li>
<li>Shared across Times 0 &amp; 1: (FDR&lt;0.01, tested in all) = 248 / (FDR&lt;0.05) = 633</li>
<li>Shared across Times 1 &amp; 2: (FDR&lt;0.01, tested in all) = 104 / (FDR&lt;0.05) = 364</li>
<li>Unique to Time 0: (FDR&lt;0.01, tested in all) = 1297 / (FDR&lt;0.05, tested in all) = 2012</li>
<li>Unique to Time 1: (FDR&lt;0.01, tested in all) = 369 / (FDR&lt;0.05, tested in all) = 805</li>
<li>Unique to Time 2: (FDR&lt;0.01, tested in all) = 115 / (FDR&lt;0.05, tested in all) = 415</li>
</ul>
<pre class="r"><code>#Pairwise DE genes and cell subset overlap
DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboGenes[is.na(comboGenes)] &lt;- 0

#Pairwise DE genes Shared across All Stages
share.stg &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(Time.0==1,
                                      Time.1==1,
                                      Time.2==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))</code></pre>
<pre><code>[1] 150</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across Time 0 and Time 1
share.stg.t01 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Time.0==1,
                                          Time.1==1,
                                          Time.2==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))</code></pre>
<pre><code>[1] 250</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across Time 1 and Time 2
share.stg.t12 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Time.0==0,
                                          Time.1==1,
                                          Time.2==1) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))</code></pre>
<pre><code>[1] 108</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Time 0
uniq.stg.t0 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;Time 0&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(Time.0==1,
                                        Time.1==0,
                                        Time.2==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)</code></pre>
<pre><code>[1] 1648</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Time 1
uniq.stg.t1 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;Time 1&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(Time.0==0,
                                        Time.1==1,
                                        Time.2==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)</code></pre>
<pre><code>[1] 396</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Time 2
uniq.stg.t2 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;Time 2&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(Time.0==0,
                                        Time.1==0,
                                        Time.2==1) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)</code></pre>
<pre><code>[1] 138</code></pre>
<pre class="r"><code>#Isolate genes tested in all cell classifications
DEtests &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboTests[is.na(comboTests)] &lt;- 0
tests &lt;- pull(comboTests %&gt;%
                filter(Time.0==1,
                       Time.1==1,
                       Time.2==1) %&gt;%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg &lt;- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 &lt;- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 &lt;- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 &lt;- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 &lt;- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 &lt;- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))</code></pre>
<pre><code>[1] 150</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t01$gene))</code></pre>
<pre><code>[1] 248</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t12$gene))</code></pre>
<pre><code>[1] 104</code></pre>
<pre class="r"><code>length(uniqX.stg.t0$gene)</code></pre>
<pre><code>[1] 1297</code></pre>
<pre class="r"><code>length(uniqX.stg.t1$gene)</code></pre>
<pre><code>[1] 369</code></pre>
<pre class="r"><code>length(uniqX.stg.t2$gene)</code></pre>
<pre><code>[1] 115</code></pre>
<pre class="r"><code>#Save DE gene lists
write.csv(tests,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-tests.csv&quot;)
write.csv(shareX.stg,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.csv&quot;)
write.csv(shareX.stg.t01,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.t01.csv&quot;)
write.csv(shareX.stg.t12,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.t12.csv&quot;)
write.csv(uniqX.stg.t0,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t0.csv&quot;)
write.csv(uniqX.stg.t1,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t1.csv&quot;)
write.csv(uniqX.stg.t2,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t2.csv&quot;)</code></pre>
<p>Get psuedobulk data for viewing gene expression differences.</p>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,j)
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg &lt;- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;Stage&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
##boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.stg)</code></pre>
<p>Look at gene expression levels of interesting DE genes.</p>
<pre class="r"><code>#Reformat pseudobulk data for plotting
tmm$gene &lt;- rownames(tmm)
tmm.melt &lt;- melt(tmm,id.vars=&quot;gene&quot;,measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep &lt;- sapply(tmm.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
tmm.melt$stage &lt;- sapply(tmm.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])
tmm.melt$species &lt;- ifelse(grepl(&quot;H&quot;,tmm.melt$ind.rep),&quot;Human&quot;,&quot;Chimp&quot;)

#Plot gene expression levels
pdf(file=&quot;./output/de-shareX.stg-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-shareX.stg.t01-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg.t01$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-shareX.stg.t12-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg.t12$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.stg.t0-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t0$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.stg.t1-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t1$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.stg.t2-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t2$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Stages (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared across Time 0 &amp; Time 1 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Time 0 &amp; Time 1&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 0 &amp; Time 1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 0 &amp; Time 1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 0 &amp; Time 1&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared across Time 1 &amp; Time 2 (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in Time 1 &amp; Time 2&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 1 &amp; Time 2&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 1 &amp; Time 2&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Time 1 &amp; Time 2&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Time 0 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Time 0&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 0&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 0&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 0&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Time 1 (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Time 1&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 1&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Time 2 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Time 2&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 2&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 2&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Time 2&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test.</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;GallegoRomero2015-DE.iPSCs&quot;,gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list(&quot;Blake2018-DE.iPSCs&quot;,blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list(&quot;Blake2018-DE.iPSC-endoderm-day1&quot;,blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list(&quot;Blake2018-DE.iPSC-endoderm-day2&quot;,blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list(&quot;Blake2018-DE.iPSC-endoderm-day3&quot;,blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list(&quot;Blake2018-DE.iPSCs-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list(&quot;Blake2020-DE.heart&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list(&quot;Blake2020-DE.kidney&quot;,blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list(&quot;Blake2020-DE.liver&quot;,blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list(&quot;Blake2020-DE.lung&quot;,blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list(&quot;Blake2020-DE.heart-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list(&quot;Blake2020-DE.kidney-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list(&quot;Blake2020-DE.liver-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list(&quot;Blake2020-DE.lung-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list(&quot;Gokhman2021-DE.Parental.iPSCs&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list(&quot;Gokhman2021-DE.Parental.CNCCs&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list(&quot;Gokhman2021-DE.Parental.iPSCs-UNIQUE&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-UNIQUE&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list(&quot;Gokhman2021-DE.Parental.CNCCs-UNIQUE&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-UNIQUE&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Gokhman2020-skeletal.genes&quot;,gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All Time 0 DE Genes
  print(&quot;All Time 0 DE Genes&quot;)
  tempDE &lt;- pull(DEdataSig %&gt;%
                  dplyr::filter(cell.assign==&quot;stage&quot;,cell.subset==&quot;Time 0&quot;) %&gt;%
                  dplyr::select(gene))
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 0&quot; &amp;
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  #All Time 1 DE Genes
  print(&quot;All Time 1 DE Genes&quot;)
  tempDE &lt;- pull(DEdataSig %&gt;%
                  dplyr::filter(cell.assign==&quot;stage&quot;,cell.subset==&quot;Time 1&quot;) %&gt;%
                  dplyr::select(gene))
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 1&quot; &amp;
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  #All Time 2 DE Genes
  print(&quot;All Time 2 DE Genes&quot;)
  tempDE &lt;- pull(DEdataSig %&gt;%
                  dplyr::filter(cell.assign==&quot;stage&quot;,cell.subset==&quot;Time 2&quot;) %&gt;%
                  dplyr::select(gene))
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 2&quot; &amp;
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Unique to Time 0
  print(&quot;Unique to Time 0&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 0&quot; &amp;
                                  DEdata$gene %in% uniq.stg.t0$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t0$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 1
  print(&quot;Unique to Time 1&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 1&quot; &amp;
                                  DEdata$gene %in% uniq.stg.t1$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t1$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 2
  print(&quot;Unique to Time 2&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.subset==&quot;Time 2&quot; &amp;
                                  DEdata$gene %in% uniq.stg.t2$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t2$gene,
                NonDEGenes=nonDEgenes)
  
  #Shared in All Stages (tested in all classifications)
  print(&quot;Shared in All Stages, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg$gene,
                NonDEGenes=nonDEgenes)
  #Shared in Time 0 &amp; Time 1 (tested in all classifications)
  print(&quot;Shared in Time 0 &amp; Time 1, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg.t01$gene,
                NonDEGenes=nonDEgenes)
  #Shared in Time 1 &amp; Time 2 (tested in all classifications)
  print(&quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg.t12$gene,
                NonDEGenes=nonDEgenes)  
  
  #Unique to Time 0 (tested in all classifications)
  print(&quot;Unique to Time 0, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t0$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 1 (tested in all classifications)
  print(&quot;Unique to Time 1, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t1$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 2 (tested in all classifications)
  print(&quot;Unique to Time 2, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t2$gene,
                NonDEGenes=nonDEgenes)

}</code></pre>
<p>Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared in All Stages (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 0 &amp; Time 1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 0 &amp; Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 1 &amp; Time 2 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 1 &amp; Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to Time 0 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 0&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 2 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-32-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-32-1.png" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-33-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-2">
Past versions of unnamed-chunk-33-2.png
</button>
</p>
<div id="fig-unnamed-chunk-33-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-33-2.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>              Source.DEGenes               Source.GenesOfInterest
58          Unique to Time 0   Gokhman2021-DE.Hybrid.iPSCs-Unique
52          Unique to Time 0 Gokhman2021-DE.Parental.iPSCs-Unique
4           Unique to Time 0            Blake2018-DE.iPSCs-Unique
69 Shared in Time 1 &amp; Time 2   Gokhman2021-DE.Hybrid.CNCCs-Unique
34          Unique to Time 0           Blake2020-DE.kidney-Unique
77          Unique to Time 1                 Gokhman2020-DMR.bone
72          Unique to Time 2   Gokhman2021-DE.Hybrid.CNCCs-Unique
38 Shared in Time 0 &amp; Time 1            Blake2020-DE.liver-Unique
   DE.Interest      p.value  GeneRatio
58         422 7.504825e-18 0.32536623
52         634 1.863256e-15 0.48882035
4           75 1.091163e-06 0.05782575
69          19 1.663308e-03 0.18269231
34          28 7.509800e-03 0.02158828
77          15 1.285847e-02 0.04065041
72          16 4.430631e-02 0.13913043
38          16 4.444194e-02 0.06451613</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.enrichment.csv&quot;)</code></pre>
<p>Determine whether EVC2, NFIX, XYLT1, ACAN, COL2A1, and SOX9 overlap with DE genes:</p>
<pre class="r"><code>geneList &lt;- c(&quot;EVC2&quot;,&quot;NFIX&quot;,&quot;XYLT1&quot;,&quot;ACAN&quot;,&quot;COL2A1&quot;,&quot;SOX9&quot;)

#Shared in All Stages (tested in all classifications)
print(&quot;Shared in All Stages, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Shared in All Stages, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.stg$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Shared in Time 0 &amp; Time 1 (tested in all classifications)
print(&quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;)</code></pre>
<pre><code>[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.stg.t01$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Shared in Time 1 &amp; Time 2 (tested in all classifications)
print(&quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.stg.t12$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to Time 0 (tested in all classifications)
print(&quot;Unique to Time 0 (tested in all classifications)&quot;)</code></pre>
<pre><code>[1] &quot;Unique to Time 0 (tested in all classifications)&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.stg.t0$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to Time 1 (tested in all classifications)
print(&quot;Unique to Time 1 (tested in all classifications)&quot;)</code></pre>
<pre><code>[1] &quot;Unique to Time 1 (tested in all classifications)&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.stg.t1$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to Time 2 (tested in all classifications)
print(&quot;Unique to Time 2 (tested in all classifications)&quot;)</code></pre>
<pre><code>[1] &quot;Unique to Time 2 (tested in all classifications)&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.stg.t2$gene))</code></pre>
<pre><code>character(0)</code></pre>
<p>Determine which skeletal-phenotype-related genes overlap with DE genes:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared in All Stages (tested in all classifications)
  print(&quot;Shared in All Stages, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.stg$gene)))

  #Shared in Time 0 &amp; Time 1 (tested in all classifications)
  print(&quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.stg.t01$gene)))
  
  #Shared in Time 1 &amp; Time 2 (tested in all classifications)
  print(&quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.stg.t12$gene)))

  #Unique to Time 0 (tested in all classifications)
  print(&quot;Unique to Time 0 (tested in all classifications)&quot;)
  print(intersect(genesExternal[[2]],uniqX.stg.t0$gene))

  #Unique to Time 1 (tested in all classifications)
  print(&quot;Unique to Time 1 (tested in all classifications)&quot;)
  print(intersect(genesExternal[[2]],uniqX.stg.t1$gene))
  
  #Unique to Time 2 (tested in all classifications)
  print(&quot;Unique to Time 2 (tested in all classifications)&quot;)
  print(intersect(genesExternal[[2]],uniqX.stg.t2$gene))

}</code></pre>
<pre><code>[1] &quot;Hsu2019-GWAS.hip.geometry&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
character(0)
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
[1] &quot;PPP6R3&quot;
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
character(0)
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Styrkarsdottir2019-GWAS.bone.area&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
character(0)
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
character(0)
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
[1] &quot;BCKDHB&quot;  &quot;COL11A1&quot;
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Yengo2018-GWAS.height&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
 [1] &quot;SH3YL1&quot;   &quot;SDCCAG8&quot;  &quot;PBLD&quot;     &quot;DDX59&quot;    &quot;POLR1B&quot;   &quot;ZSCAN9&quot;  
 [7] &quot;HAUS2&quot;    &quot;ABT1&quot;     &quot;CTSF&quot;     &quot;C16orf95&quot;
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
 [1] &quot;MYO9A&quot;   &quot;LHPP&quot;    &quot;MORN1&quot;   &quot;SSH2&quot;    &quot;MYO1F&quot;   &quot;TRIM41&quot;  &quot;SKI&quot;    
 [8] &quot;ZYG11B&quot;  &quot;QSOX2&quot;   &quot;SPINDOC&quot; &quot;SUCLG2&quot;  &quot;AATK&quot;    &quot;RAD51B&quot;  &quot;ZKSCAN3&quot;
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
[1] &quot;DKK3&quot;   &quot;TXNL4B&quot; &quot;VANGL2&quot; &quot;FEZ2&quot;   &quot;NELFA&quot; 
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
  [1] &quot;ICA1&quot;     &quot;RBM6&quot;     &quot;TTC27&quot;    &quot;PRDM11&quot;   &quot;SLC45A4&quot;  &quot;PARP12&quot;  
  [7] &quot;COL11A1&quot;  &quot;SRBD1&quot;    &quot;TGFBR3&quot;   &quot;ELP1&quot;     &quot;PPP2R3A&quot;  &quot;ICAM3&quot;   
 [13] &quot;FGFR1&quot;    &quot;TULP3&quot;    &quot;CDC14B&quot;   &quot;ERC1&quot;     &quot;BCKDHB&quot;   &quot;DIS3&quot;    
 [19] &quot;NOX4&quot;     &quot;DOCK9&quot;    &quot;KIF16B&quot;   &quot;FUS&quot;      &quot;P3H2&quot;     &quot;DPYSL2&quot;  
 [25] &quot;MAPK1&quot;    &quot;KHNYN&quot;    &quot;FNDC3A&quot;   &quot;CAB39L&quot;   &quot;AAGAB&quot;    &quot;TRPS1&quot;   
 [31] &quot;PLOD3&quot;    &quot;PDLIM1&quot;   &quot;SUFU&quot;     &quot;PITRM1&quot;   &quot;CDKN1B&quot;   &quot;CHMP3&quot;   
 [37] &quot;UXS1&quot;     &quot;BIRC6&quot;    &quot;COX7A2L&quot;  &quot;TGFBI&quot;    &quot;MTERF4&quot;   &quot;ITPR2&quot;   
 [43] &quot;METTL8&quot;   &quot;MOCS1&quot;    &quot;RRBP1&quot;    &quot;MAP2K2&quot;   &quot;ABHD8&quot;    &quot;MTERF1&quot;  
 [49] &quot;LOXL1&quot;    &quot;SLC44A2&quot;  &quot;XPO7&quot;     &quot;ZNF304&quot;   &quot;ZSWIM3&quot;   &quot;CTIF&quot;    
 [55] &quot;CCNH&quot;     &quot;USP37&quot;    &quot;ALDH1L2&quot;  &quot;RCBTB1&quot;   &quot;CALCOCO2&quot; &quot;DCAF7&quot;   
 [61] &quot;YAP1&quot;     &quot;SEMA6D&quot;   &quot;MFSD13A&quot;  &quot;RANBP10&quot;  &quot;NFIC&quot;     &quot;COL6A2&quot;  
 [67] &quot;TTC13&quot;    &quot;DAAM2&quot;    &quot;NIPSNAP2&quot; &quot;PROSER2&quot;  &quot;PTPRJ&quot;    &quot;ADAMTS12&quot;
 [73] &quot;WWC2&quot;     &quot;PDE3B&quot;    &quot;DAB2&quot;     &quot;CFDP1&quot;    &quot;JAZF1&quot;    &quot;SORBS2&quot;  
 [79] &quot;PLCD3&quot;    &quot;DDR2&quot;     &quot;DPY30&quot;    &quot;PRRC1&quot;    &quot;DLC1&quot;     &quot;ZCCHC24&quot; 
 [85] &quot;REEP3&quot;    &quot;NDRG2&quot;    &quot;TK2&quot;      &quot;SPRYD3&quot;   &quot;TPST1&quot;    &quot;PDCD6IP&quot; 
 [91] &quot;SMAD1&quot;    &quot;CYTL1&quot;    &quot;SPSB1&quot;    &quot;GPHN&quot;     &quot;FGGY&quot;     &quot;PEAK1&quot;   
 [97] &quot;FBXW8&quot;    &quot;PHYKPL&quot;   &quot;PPP2R2D&quot;  &quot;MSRA&quot;     &quot;SLC38A9&quot;  &quot;PCGF5&quot;   
[103] &quot;ARSJ&quot;     &quot;ACBD4&quot;    &quot;CEP63&quot;    &quot;RNLS&quot;     &quot;WWOX&quot;     &quot;COL14A1&quot; 
[109] &quot;SRC&quot;      &quot;FAM114A1&quot; &quot;SPG7&quot;     &quot;GRK6&quot;     &quot;MAN1A2&quot;   &quot;PAPSS2&quot;  
[115] &quot;COLGALT2&quot; &quot;GRK5&quot;     &quot;SGMS1&quot;    &quot;HMGN1&quot;    &quot;NCKIPSD&quot;  &quot;FNTB&quot;    
[121] &quot;CUX1&quot;    
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
 [1] &quot;PIK3C2A&quot; &quot;IKZF2&quot;   &quot;CYFIP2&quot;  &quot;EFHC1&quot;   &quot;HSCB&quot;    &quot;PXMP4&quot;   &quot;CEP192&quot; 
 [8] &quot;GLI3&quot;    &quot;WNT3&quot;    &quot;TFG&quot;     &quot;DHCR24&quot;  &quot;ATL2&quot;    &quot;NCOA5&quot;   &quot;SYNE1&quot;  
[15] &quot;NMT1&quot;    &quot;UGGT1&quot;   &quot;SLC16A3&quot; &quot;PIGK&quot;    &quot;ASXL2&quot;   &quot;NEK7&quot;    &quot;VTI1A&quot;  
[22] &quot;CMIP&quot;    &quot;VPS8&quot;    &quot;HLCS&quot;    &quot;ADCY9&quot;   &quot;PCOLCE2&quot; &quot;PSMD2&quot;   &quot;ZNF784&quot; 
[29] &quot;HMCES&quot;   &quot;RNFT1&quot;   &quot;RAB40C&quot; 
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
[1] &quot;USP36&quot;  &quot;RFX2&quot;   &quot;NUDT6&quot;  &quot;LCORL&quot;  &quot;POLR2A&quot;
[1] &quot;Steinberg2020-GWAS.OA&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
[1] &quot;EXOSC4&quot;  &quot;PGAM5&quot;   &quot;SDR39U1&quot;
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
[1] &quot;PID1&quot;   &quot;TRIM47&quot;
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
[1] &quot;GALE&quot; &quot;PYGL&quot;
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
 [1] &quot;AEBP1&quot;   &quot;ASAH1&quot;   &quot;ATAD3B&quot;  &quot;C1GALT1&quot; &quot;CA5B&quot;    &quot;CCDC124&quot; &quot;CPT1A&quot;  
 [8] &quot;CPT2&quot;    &quot;CYP27A1&quot; &quot;DDR2&quot;    &quot;DDX19A&quot;  &quot;DERA&quot;    &quot;EXD2&quot;    &quot;GGCT&quot;   
[15] &quot;GGH&quot;     &quot;GNPDA1&quot;  &quot;GPHN&quot;    &quot;GRN&quot;     &quot;GSTZ1&quot;   &quot;GULP1&quot;   &quot;HNRNPA3&quot;
[22] &quot;HSPA4L&quot;  &quot;HTRA3&quot;   &quot;ISG20&quot;   &quot;MAP1B&quot;   &quot;MYBBP1A&quot; &quot;MYO6&quot;    &quot;NFIA&quot;   
[29] &quot;NUDT15&quot;  &quot;NXT1&quot;    &quot;ODF2&quot;    &quot;OPLAH&quot;   &quot;OSBPL3&quot;  &quot;PDLIM1&quot;  &quot;PHKA1&quot;  
[36] &quot;PLCG1&quot;   &quot;POSTN&quot;   &quot;PRPF4&quot;   &quot;PTGIS&quot;   &quot;S100A4&quot;  &quot;SMYD5&quot;   &quot;SNX21&quot;  
[43] &quot;SPECC1&quot;  &quot;TGFBI&quot;   &quot;TIMM10&quot;  &quot;TOMM34&quot;  &quot;TYMP&quot;   
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
 [1] &quot;COL5A1&quot; &quot;CRLF1&quot;  &quot;CYP1B1&quot; &quot;EPS8L2&quot; &quot;IRS1&quot;   &quot;MPDZ&quot;   &quot;MRPS26&quot;
 [8] &quot;NAMPT&quot;  &quot;PDLIM4&quot; &quot;RHOC&quot;   &quot;TBL2&quot;   &quot;UNK&quot;   
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
[1] &quot;BCL2L13&quot; &quot;IGFBP7&quot; 
[1] &quot;Morris2020-GWAS.OP&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
[1] &quot;PSEN2&quot;   &quot;HLA-A&quot;   &quot;PCNX3&quot;   &quot;DHRS12&quot;  &quot;SLC27A5&quot;
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
[1] &quot;ITGA1&quot;  &quot;LDB3&quot;   &quot;RAD51B&quot; &quot;ZNF75A&quot; &quot;EEF2K&quot;  &quot;PPP6R2&quot;
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
[1] &quot;FAM234A&quot;
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
 [1] &quot;AURKAIP1&quot;  &quot;NFIA&quot;      &quot;TGFBR3&quot;    &quot;COL11A1&quot;   &quot;TMEM18&quot;   
 [6] &quot;TRIB2&quot;     &quot;CYBRD1&quot;    &quot;MLPH&quot;      &quot;FBXL2&quot;     &quot;PRKAR2A&quot;  
[11] &quot;RBM5&quot;      &quot;WWC2&quot;      &quot;DAB2&quot;      &quot;SLC38A9&quot;   &quot;FAM174A&quot;  
[16] &quot;HSPA4&quot;     &quot;TRIM27&quot;    &quot;DAAM2&quot;     &quot;JAZF1&quot;     &quot;EPDR1&quot;    
[21] &quot;STARD3NL&quot;  &quot;SEMA3A&quot;    &quot;ZNF800&quot;    &quot;DLC1&quot;      &quot;RUNX1T1&quot;  
[26] &quot;TRPS1&quot;     &quot;GAS1&quot;      &quot;FBXW5&quot;     &quot;ZNF438&quot;    &quot;ZCCHC24&quot;  
[31] &quot;PAPSS2&quot;    &quot;CCDC34&quot;    &quot;PTPRJ&quot;     &quot;MYRF&quot;      &quot;YAP1&quot;     
[36] &quot;ITPR2&quot;     &quot;FAR2&quot;      &quot;ARF3&quot;      &quot;ATF7&quot;      &quot;CDK2AP1&quot;  
[41] &quot;DOCK9&quot;     &quot;BAZ1A&quot;     &quot;SAV1&quot;      &quot;FBLN5&quot;     &quot;SEMA6D&quot;   
[46] &quot;RAB11FIP3&quot; &quot;SPG7&quot;      &quot;ABR&quot;       &quot;ACSF2&quot;     &quot;POLG2&quot;    
[51] &quot;LLGL2&quot;     &quot;RFNG&quot;      &quot;NFIC&quot;      &quot;MAP2K2&quot;    &quot;TRPC4AP&quot;  
[56] &quot;KREMEN1&quot;  
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
 [1] &quot;ASXL2&quot;   &quot;IKZF2&quot;   &quot;IRS1&quot;    &quot;PCOLCE2&quot; &quot;ANAPC4&quot;  &quot;SOBP&quot;    &quot;SYNE1&quot;  
 [8] &quot;MPDZ&quot;    &quot;CNTLN&quot;   &quot;VTI1A&quot;   &quot;CADM1&quot;   &quot;SPATA13&quot; &quot;ADCY9&quot;   &quot;CMIP&quot;   
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
[1] &quot;PKDCC&quot;  &quot;ACTR2&quot;  &quot;LCORL&quot;  &quot;PTS&quot;    &quot;EMP2&quot;   &quot;POLR2A&quot; &quot;USP36&quot;  &quot;RFX2&quot;  </code></pre>
</div>
<div id="general-clusters-res0.05-1" class="section level3">
<h3>General Clusters (res=0.05)</h3>
<p>Pairwise DE Genes</p>
<pre class="r"><code># DE genes and cell subset overlap
DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;cluster&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)
comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;iPSC.c1&quot;)       %&gt;% dplyr::select(gene), &quot;iPSC.c1&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;MSC.c1&quot;)        %&gt;% dplyr::select(gene), &quot;MSC.c1&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboGenes[is.na(comboGenes)] &lt;- 0

#Pairwise DE genes Shared across All Clusters (subset)
share.stg &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(iPSC.c1==1,
                                      MSC.c1==1,
                                      Osteogenic.c1==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))</code></pre>
<pre><code>[1] 168</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across iPSC.c1 and MSC.c1
share.stg.t01 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(iPSC.c1==1,
                                          MSC.c1==1,
                                          Osteogenic.c1==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))</code></pre>
<pre><code>[1] 242</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across MSC.c1 and Osteogenic.c1
share.stg.t12 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(iPSC.c1==0,
                                          MSC.c1==1,
                                          Osteogenic.c1==1) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to iPSC.c1
uniq.stg.t0 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;iPSC.c1&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC.c1==1,
                                        MSC.c1==0,
                                        Osteogenic.c1==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)</code></pre>
<pre><code>[1] 1689</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to MSC.c1
uniq.stg.t1 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;MSC.c1&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC.c1==0,
                                        MSC.c1==1,
                                        Osteogenic.c1==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)</code></pre>
<pre><code>[1] 396</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Osteogenic.c1
uniq.stg.t2 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;Osteogenic.c1&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC.c1==0,
                                        MSC.c1==0,
                                        Osteogenic.c1==1) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)</code></pre>
<pre><code>[1] 180</code></pre>
<pre class="r"><code>#Isolate genes tested in all cell classifications
DEtests &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;cluster&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;iPSC.c1&quot;) %&gt;% dplyr::select(gene), &quot;iPSC.c1&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;MSC.c1&quot;) %&gt;% dplyr::select(gene), &quot;MSC.c1&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboTests[is.na(comboTests)] &lt;- 0
tests &lt;- pull(comboTests %&gt;%
                filter(iPSC.c1==1,
                       MSC.c1==1,
                       Osteogenic.c1==1) %&gt;%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg &lt;- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 &lt;- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 &lt;- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 &lt;- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 &lt;- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 &lt;- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))</code></pre>
<pre><code>[1] 168</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t01$gene))</code></pre>
<pre><code>[1] 241</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t12$gene))</code></pre>
<pre><code>[1] 125</code></pre>
<pre class="r"><code>length(uniqX.stg.t0$gene)</code></pre>
<pre><code>[1] 1330</code></pre>
<pre class="r"><code>length(uniqX.stg.t1$gene)</code></pre>
<pre><code>[1] 372</code></pre>
<pre class="r"><code>length(uniqX.stg.t2$gene)</code></pre>
<pre><code>[1] 154</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Clusters (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in iPSC.c1 and MSC.c1 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in MSC.c1 and Osteogenic.c1 (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to iPSC.c1 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to iPSC.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC.c1&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to MSC.c1 (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to MSC.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC.c1&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Osteogenic.c1 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteogenic.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x</code></pre>
<p>Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared in All Clusters (subset) (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in iPSC.c1 &amp; MSC.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in MSC.c1 &amp; Osteogenic.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to iPSC.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to iPSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to MSC.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to MSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteo.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-41-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-41-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-42-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-42-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-2">
Past versions of unnamed-chunk-42-2.png
</button>
</p>
<div id="fig-unnamed-chunk-42-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-42-2.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                Source.DEGenes               Source.GenesOfInterest
52           Unique to iPSC.c1 Gokhman2021-DE.Parental.iPSCs-Unique
58           Unique to iPSC.c1   Gokhman2021-DE.Hybrid.iPSCs-Unique
4            Unique to iPSC.c1            Blake2018-DE.iPSCs-Unique
77            Unique to MSC.c1                 Gokhman2020-DMR.bone
69 Shared in MSC.c1 &amp; Osteo.c1   Gokhman2021-DE.Hybrid.CNCCs-Unique
72          Unique to Osteo.c1   Gokhman2021-DE.Hybrid.CNCCs-Unique
41            Unique to MSC.c1            Blake2020-DE.liver-Unique
34           Unique to iPSC.c1           Blake2020-DE.kidney-Unique
   DE.Interest      p.value  GeneRatio
52         654 9.168015e-17 0.49172932
58         421 2.650484e-15 0.31654135
4           70 6.257791e-05 0.05263158
77          16 6.077858e-03 0.04301075
69          19 1.245325e-02 0.15200000
72          22 1.517409e-02 0.14285714
41          24 1.722759e-02 0.06451613
34          27 1.823693e-02 0.02030075</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.enrichment.csv&quot;)</code></pre>
</div>
<div id="general-ad-hoc-classification-of-cells-scaled-integrated-data-threshold0-1" class="section level3">
<h3>General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)</h3>
<p>Pairwise DE Genes</p>
<pre class="r"><code># DE genes and cell subset overlap
DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;adhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)
comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;iPSC&quot;)       %&gt;% dplyr::select(gene), &quot;iPSC&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;MSC&quot;)        %&gt;% dplyr::select(gene), &quot;MSC&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboGenes[is.na(comboGenes)] &lt;- 0

#Pairwise DE genes Shared across All Ad Hoc
share.stg &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(iPSC==1,
                                      MSC==1,
                                      Osteogenic==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))</code></pre>
<pre><code>[1] 208</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across iPSC and MSC
share.stg.t01 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(iPSC==1,
                                          MSC==1,
                                          Osteogenic==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))</code></pre>
<pre><code>[1] 223</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across MSC and Osteogenic
share.stg.t12 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(iPSC==0,
                                          MSC==1,
                                          Osteogenic==1) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))</code></pre>
<pre><code>[1] 170</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to iPSC
uniq.stg.t0 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;iPSC&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC==1,
                                        MSC==0,
                                        Osteogenic==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)</code></pre>
<pre><code>[1] 1651</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to MSC
uniq.stg.t1 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;MSC&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC==0,
                                        MSC==1,
                                        Osteogenic==0) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)</code></pre>
<pre><code>[1] 362</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Osteogenic
uniq.stg.t2 &lt;- DEgenes %&gt;% 
                 filter(cell.subset==&quot;Osteogenic&quot;,
                        gene %in%
                          pull(comboGenes %&gt;%
                                 filter(iPSC==0,
                                        MSC==0,
                                        Osteogenic==1) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)</code></pre>
<pre><code>[1] 189</code></pre>
<pre class="r"><code>#Isolate genes tested in all cell classifications
DEtests &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;adhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;iPSC&quot;) %&gt;% dplyr::select(gene), &quot;iPSC&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;MSC&quot;) %&gt;% dplyr::select(gene), &quot;MSC&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboTests[is.na(comboTests)] &lt;- 0
tests &lt;- pull(comboTests %&gt;%
                filter(iPSC==1,
                       MSC==1,
                       Osteogenic==1) %&gt;%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg &lt;- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 &lt;- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 &lt;- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 &lt;- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 &lt;- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 &lt;- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))</code></pre>
<pre><code>[1] 208</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t01$gene))</code></pre>
<pre><code>[1] 220</code></pre>
<pre class="r"><code>length(unique(shareX.stg.t12$gene))</code></pre>
<pre><code>[1] 163</code></pre>
<pre class="r"><code>length(uniqX.stg.t0$gene)</code></pre>
<pre><code>[1] 1308</code></pre>
<pre class="r"><code>length(uniqX.stg.t1$gene)</code></pre>
<pre><code>[1] 333</code></pre>
<pre class="r"><code>length(uniqX.stg.t2$gene)</code></pre>
<pre><code>[1] 169</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Ad Hoc (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in iPSC and MSC (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in iPSC &amp; MSC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC &amp; MSC&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC &amp; MSC&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in iPSC &amp; MSC&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in MSC and Osteogenic (FDR&lt;0.01, tested in all classifications)
#goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in MSC &amp; Osteo&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC &amp; Osteo&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC &amp; Osteo&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in MSC &amp; Osteo&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to iPSC (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to iPSC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to iPSC&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to MSC (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to MSC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to MSC&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Osteogenic (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteogenic&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x</code></pre>
<p>Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared in All Stages (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 0 &amp; Time 1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 0 &amp; Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 1 &amp; Time 2 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 1 &amp; Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to Time 0 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 0&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 2 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-48-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-48-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-49-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-49-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-2">
Past versions of unnamed-chunk-49-2.png
</button>
</p>
<div id="fig-unnamed-chunk-49-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-49-2.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>              Source.DEGenes               Source.GenesOfInterest
58          Unique to Time 0   Gokhman2021-DE.Hybrid.iPSCs-Unique
52          Unique to Time 0 Gokhman2021-DE.Parental.iPSCs-Unique
4           Unique to Time 0            Blake2018-DE.iPSCs-Unique
69 Shared in Time 1 &amp; Time 2   Gokhman2021-DE.Hybrid.CNCCs-Unique
77          Unique to Time 1                 Gokhman2020-DMR.bone
34          Unique to Time 0           Blake2020-DE.kidney-Unique
45 Shared in Time 1 &amp; Time 2             Blake2020-DE.lung-Unique
   DE.Interest      p.value  GeneRatio
58         426 1.084307e-17 0.32568807
52         638 2.589981e-15 0.48776758
4           78 1.731428e-07 0.05963303
69          27 1.045023e-03 0.16564417
77          15 5.230131e-03 0.04504505
34          27 1.361525e-02 0.02064220
45           6 4.576746e-02 0.03680982</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhoc.enrichment.csv&quot;)</code></pre>
</div>
<div id="osteogenic-clusters-res0.50-1" class="section level3">
<h3>Osteogenic Clusters (res=0.50)</h3>
<p>Pairwise DE Genes</p>
<pre class="r"><code># DE genes and cell subset overlap
DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.50&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)
comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboGenes[is.na(comboGenes)] &lt;- 0

#Pairwise DE genes Shared across O.c1, O.c3, O.c4
share.o1o3o4 &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Osteogenic.c1==1,
                                          Osteogenic.c2==0,
                                          Osteogenic.c3==1,
                                          Osteogenic.c4==1) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.o1o3o4$gene))</code></pre>
<pre><code>[1] 104</code></pre>
<pre class="r"><code># DE Genes Unique to Osteogenic.c1
Uniq.50o1 &lt;- DEgenes %&gt;% 
               filter(cell.subset==&quot;Osteogenic.c1&quot;,
                      gene %in%
                        pull(comboGenes %&gt;%
                               filter(Osteogenic.c1==1,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==0) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o1$gene))</code></pre>
<pre><code>[1] 297</code></pre>
<pre class="r"><code># DE Genes Unique to Osteogenic.c2
Uniq.50o2 &lt;- DEgenes %&gt;% 
               filter(cell.subset==&quot;Osteogenic.c2&quot;,
                      gene %in%
                        pull(comboGenes %&gt;%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==1,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==0) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o2$gene))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># DE Genes Unique to Osteogenic.c3
Uniq.50o3 &lt;- DEgenes %&gt;% 
               filter(cell.subset==&quot;Osteogenic.c3&quot;,
                      gene %in%
                        pull(comboGenes %&gt;%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==1,
                                      Osteogenic.c4==0) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o3$gene))</code></pre>
<pre><code>[1] 120</code></pre>
<pre class="r"><code># DE Genes Unique to Osteogenic.c4
Uniq.50o4 &lt;- DEgenes %&gt;% 
               filter(cell.subset==&quot;Osteogenic.c4&quot;,
                      gene %in%
                        pull(comboGenes %&gt;%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o4$gene))</code></pre>
<pre><code>[1] 55</code></pre>
<pre class="r"><code>#Isolate genes tested in all cell classifications
DEtests &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;ostcluster0.50&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic.c1&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c1&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic.c2&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c2&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic.c3&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c3&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Osteogenic.c4&quot;) %&gt;% dplyr::select(gene), &quot;Osteogenic.c4&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboTests[is.na(comboTests)] &lt;- 0
tests &lt;- pull(comboTests %&gt;%
                filter(Osteogenic.c1==1,
                       Osteogenic.c2==1,
                       Osteogenic.c3==1,
                       Osteogenic.c4==1) %&gt;%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.o1o3o4 &lt;- share.o1o3o4[which(share.o1o3o4$gene %in% tests),]
UniqX.50o1 &lt;- Uniq.50o1[which(Uniq.50o1$gene %in% tests),]
UniqX.50o2 &lt;- Uniq.50o2[which(Uniq.50o2$gene %in% tests),]
UniqX.50o3 &lt;- Uniq.50o3[which(Uniq.50o3$gene %in% tests),]
UniqX.50o4 &lt;- Uniq.50o4[which(Uniq.50o4$gene %in% tests),]
length(unique(shareX.o1o3o4$gene))</code></pre>
<pre><code>[1] 76</code></pre>
<pre class="r"><code>length(UniqX.50o1$gene)</code></pre>
<pre><code>[1] 205</code></pre>
<pre class="r"><code>length(UniqX.50o2$gene)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(UniqX.50o3$gene)</code></pre>
<pre><code>[1] 73</code></pre>
<pre class="r"><code>length(UniqX.50o4$gene)</code></pre>
<pre><code>[1] 41</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Osteogenic Clusters (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.o1o3o4)$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in Osteo.c1c3c4&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Osteo.c1c3c4&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Osteo.c1c3c4&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Osteo.c1c3c4&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Osteogenic.c1 (FDR&lt;0.01, tested in all classifications) - no enrichments
#goTot &lt;- goEnrich(totGenes=tests, deGenes=UniqX.50o1$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Osteogenic.c1&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c1&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x

#Unique to Osteogenic.c3 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=UniqX.50o3$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteogenic.c3&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c3&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c3&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c3&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Osteogenic.c4 (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=UniqX.50o4$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Osteogenic.c4&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c4&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c4&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteogenic.c4&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x</code></pre>
<p>Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared in All Clusters (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% shareX.o1o3o4$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Osteo.c1c3c4&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.o1o3o4$gene),
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic.c1 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% UniqX.50o1$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteogenic.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c3 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% UniqX.50o3$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteogenic.c3&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o3$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c4 (tested in all classifications)
  nonDEgenes &lt;- tests[which(tests %in% UniqX.50o4$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteogenic.c4&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o4$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-55-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-55-1">
Past versions of unnamed-chunk-55-1.png
</button>
</p>
<div id="fig-unnamed-chunk-55-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-55-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-55-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-56-1">
Past versions of unnamed-chunk-56-1.png
</button>
</p>
<div id="fig-unnamed-chunk-56-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-56-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-56-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-56-2">
Past versions of unnamed-chunk-56-2.png
</button>
</p>
<div id="fig-unnamed-chunk-56-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-56-2.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>[1] Source.DEGenes         Source.GenesOfInterest DE.Interest           
[4] p.value                GeneRatio             
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.enrichment.csv&quot;)</code></pre>
</div>
<div id="osteogenic-clusters-res0.75-1" class="section level3">
<h3>Osteogenic Clusters (res=0.75)</h3>
</div>
<div id="osteogenic-clusters-res0.50-time-2-only-1" class="section level3">
<h3>Osteogenic Clusters (res=0.50) (Time 2 only)</h3>
</div>
<div id="osteogenic-clusters-res0.75-time-2-only-1" class="section level3">
<h3>Osteogenic Clusters (res=0.75) (Time 2 only)</h3>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible-1" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)</h3>
<p>Pairwise DE Genes</p>
<ul>
<li>Shared across all ostadhoc classifications: (FDR&lt;0.01) = 25 / (FDR&lt;0.05) = 193</li>
<li>Shared across pre+obl+emb+min: (FDR&lt;0.01) = 22 / (FDR&lt;0.05) = 75</li>
<li>Shared across obl+emb+min: (FDR&lt;0.01) = 17 / (FDR&lt;0.05) = 70</li>
<li>Unique to preosteoblasts: (FDR&lt;0.01) = 33 / (FDR&lt;0.05) = 80</li>
<li>Unique to osteoblasts: (FDR&lt;0.01) = 80 / (FDR&lt;0.05) = 250</li>
<li>Unique to embedding osteoblasts: (FDR&lt;0.01) = 33 / (FDR&lt;0.05) = 99</li>
<li>Unique to mineralizing osteoblasts: (FDR&lt;0.01) = 233 / (FDR&lt;0.05) = 560</li>
<li>Unique to maturing osteocytes: (FDR&lt;0.01) = 36 / (FDR&lt;0.05) = 148</li>
<li>Shared across all ostadhoc classifications: (FDR&lt;0.01, tested in all) = 25 / (FDR&lt;0.05, tested in all) = 193</li>
<li>Shared across pre+obl+emb+min: (FDR&lt;0.01, tested in all) = 22 / (FDR&lt;0.05, tested in all) = 75</li>
<li>Shared across obl+emb+min: (FDR&lt;0.01, tested in all) = 17 / (FDR&lt;0.05, tested in all) = 68</li>
<li>Unique to preosteoblasts: (FDR&lt;0.01, tested in all) = 33 / (FDR&lt;0.05, tested in all) = 77</li>
<li>Unique to osteoblasts: (FDR&lt;0.01, tested in all) = 73 / (FDR&lt;0.05, tested in all) = 226</li>
<li>Unique to embedding osteoblasts: (FDR&lt;0.01, tested in all) = 26 / (FDR&lt;0.05, tested in all) = 86</li>
<li>Unique to mineralizing osteoblasts: (FDR&lt;0.01, tested in all) = 229 / (FDR&lt;0.05, tested in all) = 533</li>
<li>Unique to maturing osteocytes: (FDR&lt;0.01, tested in all) = 36 / (FDR&lt;0.05, tested in all) = 146</li>
</ul>
<p>Note: Too few DE genes shared across obl+emb+min+mat (n=3) and emb+min+mat (n=4), so these were not examined further. Instead, DE genes shared across pre+obl+emb+min (n=22) and obl+emb+min (n=17) were considered.</p>
<pre class="r"><code>#Pairwise DE genes and cell subset overlap
DEgenes &lt;-
  DEdataSig %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)
comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;embedding osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;) %&gt;% dplyr::select(gene), &quot;maturing osteocyte&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboGenes[is.na(comboGenes)] &lt;- 0

#Pairwise DE genes Shared across All OstAdHoc Classifications
share.oad &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(preosteoblast==1,
                                      osteoblast==1,
                                      embedding.osteoblast==1,
                                      mineralizing.osteoblast==1,
                                      maturing.osteocyte==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.oad$gene))</code></pre>
<pre><code>[1] 25</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across obl+emb+min+mat
share.oad.poem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==1,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.oad.poem$gene))</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code>#Pairwise DE genes Shared across emb+min+mat
share.oad.oem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==0,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.oad.oem$gene))</code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Preosteoblasts
uniq.oad.pre &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==1,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.pre$gene)</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Osteoblasts
uniq.oad.obl &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==1,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.obl$gene)</code></pre>
<pre><code>[1] 80</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Embedding Osteoblasts
uniq.oad.emb &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==1,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.emb$gene)</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Mineralizing Osteoblasts
uniq.oad.min &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==1,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.min$gene)</code></pre>
<pre><code>[1] 233</code></pre>
<pre class="r"><code>#Pairwise DE genes Unique to Maturing Osteocytes
uniq.oad.mat &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==1) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.mat$gene)</code></pre>
<pre><code>[1] 36</code></pre>
<pre class="r"><code>#Isolate genes tested in all cell classifications
DEtests &lt;-
  DEdata %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;preosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;embedding osteoblast&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing osteoblast&quot;=1),
                          data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;) %&gt;% dplyr::select(gene), &quot;maturing osteocyte&quot;=1)
                          ), full_join)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>comboTests[is.na(comboTests)] &lt;- 0
tests &lt;- pull(comboTests %&gt;%
                filter(preosteoblast==1,
                       osteoblast==1,
                       embedding.osteoblast==1,
                       mineralizing.osteoblast==1,
                       maturing.osteocyte==1) %&gt;%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.oad &lt;- share.oad[which(share.oad$gene %in% tests),]
shareX.oad.poem &lt;- share.oad.poem[which(share.oad.poem$gene %in% tests),]
shareX.oad.oem &lt;- share.oad.oem[which(share.oad.oem$gene %in% tests),]
uniqX.oad.pre &lt;- uniq.oad.pre[which(uniq.oad.pre$gene %in% tests),]
uniqX.oad.obl &lt;- uniq.oad.obl[which(uniq.oad.obl$gene %in% tests),]
uniqX.oad.emb &lt;- uniq.oad.emb[which(uniq.oad.emb$gene %in% tests),]
uniqX.oad.min &lt;- uniq.oad.min[which(uniq.oad.min$gene %in% tests),]
uniqX.oad.mat &lt;- uniq.oad.mat[which(uniq.oad.mat$gene %in% tests),]
length(unique(shareX.oad$gene))</code></pre>
<pre><code>[1] 25</code></pre>
<pre class="r"><code>length(unique(shareX.oad.poem$gene))</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code>length(unique(shareX.oad.oem$gene))</code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code>length(uniqX.oad.pre$gene)</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>length(uniqX.oad.obl$gene)</code></pre>
<pre><code>[1] 73</code></pre>
<pre class="r"><code>length(uniqX.oad.emb$gene)</code></pre>
<pre><code>[1] 26</code></pre>
<pre class="r"><code>length(uniqX.oad.min$gene)</code></pre>
<pre><code>[1] 229</code></pre>
<pre class="r"><code>length(uniqX.oad.mat$gene)</code></pre>
<pre><code>[1] 36</code></pre>
<pre class="r"><code>#Save DE gene lists
write.csv(tests,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-tests.csv&quot;)
write.csv(shareX.oad,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.csv&quot;)
write.csv(shareX.oad.poem,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.poem.csv&quot;)
write.csv(shareX.oad.oem,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.oem.csv&quot;)
write.csv(uniqX.oad.pre,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.pre.csv&quot;)
write.csv(uniqX.oad.obl,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.obl.csv&quot;)
write.csv(uniqX.oad.emb,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.emb.csv&quot;)
write.csv(uniqX.oad.min,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.min.csv&quot;)
write.csv(uniqX.oad.mat,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.mat.csv&quot;)</code></pre>
<p>Get psuedobulk data for viewing gene expression differences.</p>
<pre class="r"><code># Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)&gt;0)
cell.subset &lt;- c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;,&quot;unknown osteogenic&quot;,&quot;not osteogenic&quot;)
counts &lt;- c()
metadata &lt;- c()
labels &lt;- c()
for(j in cell.subset) {
  dataSub &lt;- subset(data,subset=OstAdHoc.Assign.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab &lt;- paste0(i,&quot;_&quot;,sub(&quot; &quot;,&quot;.&quot;,j))
    w &lt;- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)&gt;0) {
      x.spp &lt;- dataSub@meta.data$Species[w][1]
      x.col &lt;- dataSub@meta.data$Collection[w][1]
      x.ind &lt;- dataSub@meta.data$Individual[w][1]
      x.rep &lt;- dataSub@meta.data$Replicate[w][1]
      x.idr &lt;- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh &lt;- sub(&quot; &quot;,&quot;.&quot;,as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt &lt;- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt &lt;- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts &lt;- cbind(counts, x.cnt)
      metadata &lt;- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels &lt;- c(labels, x.lab)
    }
  }
}
colnames(counts) &lt;- labels
rownames(metadata) &lt;- labels
colnames(metadata) &lt;- c(&quot;Sample&quot;,&quot;Species&quot;,&quot;Collection&quot;,&quot;Individual&quot;,&quot;Replicate&quot;,&quot;Individual.Replicate&quot;,&quot;AdHoc&quot;)
metadata &lt;- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts &lt;- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge &lt;- DGEList(counts)
meta_dge &lt;- dge$samples[,c(&quot;lib.size&quot;,&quot;norm.factors&quot;)]
meta_dge &lt;- cbind(meta_dge, metadata)
dge$samples &lt;- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep &lt;- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))&gt;0
dge$counts &lt;- dge$counts[keep,]
#normalize data
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
#log2 transformed cpm
tmm &lt;- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
##boxplot(tmm,xlab=&quot;&quot;,ylab=&quot;Log2 counts per million&quot;,las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)</code></pre>
<p>Look at gene expression levels of interesting DE genes.</p>
<pre class="r"><code>#Reformat pseudobulk data for plotting
tmm$gene &lt;- rownames(tmm)
tmm.melt &lt;- melt(tmm,id.vars=&quot;gene&quot;,measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep &lt;- sapply(tmm.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][1])
tmm.melt$stage &lt;- sapply(tmm.melt$variable,function(x)str_split(x,&quot;_&quot;)[[1]][2])
tmm.melt$species &lt;- ifelse(grepl(&quot;H&quot;,tmm.melt$ind.rep),&quot;Human&quot;,&quot;Chimp&quot;)

#Plot gene expression levels
pdf(file=&quot;./output/de-shareX.oad-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-shareX.oad.poem-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad.poem$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-shareX.oad.oem-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad.oem$gene)) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.oad.pre-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.pre$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.oad.obl-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.obl$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.oad.emb-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.emb$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.oad.min-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.min$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file=&quot;./output/de-uniqX.oad.mat-fdr0.01-pseudo.pdf&quot;, onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.mat$gene) {
  p &lt;- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y=&quot;Genes Expression&quot;, x=&quot;&quot;) +
         scale_fill_manual(name=&quot;&quot;, values=c(&quot;lightgrey&quot;,&quot;black&quot;)) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across All OstAdHoc Classifications (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.oad)$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in All&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in Pre, Obl, Emb, and Min (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.oad.poem)$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Pre/Obl/Emb/Min&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Pre/Obl/Emb/Min&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Pre/Obl/Emb/Min&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Pre/Obl/Emb/Min&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Shared in Obl, Emb, and Min (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=unique(shareX.oad.oem)$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Obl/Emb/Min&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Obl/Emb/Min&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Obl/Emb/Min&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Shared in Obl/Emb/Min&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Preosteoblasts (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.oad.pre$gene, ontology=&quot;ALL&quot;)
goEnPlot.spec(goTot,title=&quot;Unique to Preosteoblasts&quot;,ontology=&quot;BP&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Preosteoblasts&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Preosteoblasts&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Preosteoblasts&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Preosteoblasts&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Osteoblasts (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.oad.obl$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteoblasts&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteoblasts&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteoblasts&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Osteoblasts&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Embedding Osteoblasts (FDR&lt;0.01, tested in all classifications) - no enrichment
#goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.oad.emb$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Embedding Osteoblasts&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Embedding Osteoblasts&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Embedding Osteoblasts&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Embedding Osteoblasts&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Mineralizing Osteoblasts (FDR&lt;0.01, tested in all classifications)
goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.oad.min$gene, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Mineralizing Osteoblasts&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Mineralizing Osteoblasts&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Mineralizing Osteoblasts&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Mineralizing Osteoblasts&quot;,ontology=&quot;MF&quot;)
x &lt;- goTot@result
x

#Unique to Maturing Osteocytes (FDR&lt;0.01, tested in all classifications) - no enrichment
#goTot &lt;- goEnrich(totGenes=tests, deGenes=uniqX.oad.mat$gene, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Maturing Osteocytes&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Maturing Osteocytes&quot;,ontology=&quot;BP&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Maturing Osteocytes&quot;,ontology=&quot;CC&quot;)
#goEnPlot.spec(goTot,title=&quot;Unique to Maturing Osteocytes&quot;,ontology=&quot;MF&quot;)
#x &lt;- goTot@result
#x</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test.</p>
<pre class="r"><code>#Enrichment of external gene sets in cormotif DE gene sets

listsExternal &lt;- list(

  list(&quot;GallegoRomero2015-DE.iPSCs&quot;,gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list(&quot;Blake2018-DE.iPSCs&quot;,blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list(&quot;Blake2018-DE.iPSC-endoderm-day1&quot;,blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list(&quot;Blake2018-DE.iPSC-endoderm-day2&quot;,blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list(&quot;Blake2018-DE.iPSC-endoderm-day3&quot;,blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list(&quot;Blake2018-DE.iPSCs-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-UNIQUE&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list(&quot;Blake2020-DE.heart&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list(&quot;Blake2020-DE.kidney&quot;,blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list(&quot;Blake2020-DE.liver&quot;,blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list(&quot;Blake2020-DE.lung&quot;,blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list(&quot;Blake2020-DE.heart-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list(&quot;Blake2020-DE.kidney-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list(&quot;Blake2020-DE.liver-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list(&quot;Blake2020-DE.lung-UNIQUE&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list(&quot;Gokhman2021-DE.Parental.iPSCs&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list(&quot;Gokhman2021-DE.Parental.CNCCs&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list(&quot;Gokhman2021-DE.Parental.iPSCs-UNIQUE&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-UNIQUE&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list(&quot;Gokhman2021-DE.Parental.CNCCs-UNIQUE&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-UNIQUE&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Gokhman2020-skeletal.genes&quot;,gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Unique to preosteoblasts
  print(&quot;Unique to preosteoblasts&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.assign==&quot;ostadhoc&quot; &amp;
                                  DEdata$cell.subset==&quot;preosteoblast&quot; &amp;
                                  DEdata$gene %in% uniq.oad.pre$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.pre$gene,
                NonDEGenes=nonDEgenes)
  #Unique to osteoblasts
  print(&quot;Unique to osteoblasts&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.assign==&quot;ostadhoc&quot; &amp;
                                  DEdata$cell.subset==&quot;osteoblast&quot; &amp;
                                  DEdata$gene %in% uniq.oad.obl$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.obl$gene,
                NonDEGenes=nonDEgenes)
  #Unique to embedding osteoblasts
  print(&quot;Unique to embedding osteoblasts&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.assign==&quot;ostadhoc&quot; &amp;
                                  DEdata$cell.subset==&quot;embedding osteoblast&quot; &amp;
                                  DEdata$gene %in% uniq.oad.emb$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.emb$gene,
                NonDEGenes=nonDEgenes)
  #Unique to mineralizing osteoblasts
  print(&quot;Unique to mineralizing osteoblasts&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.assign==&quot;ostadhoc&quot; &amp;
                                  DEdata$cell.subset==&quot;mineralizing osteoblast&quot; &amp;
                                  DEdata$gene %in% uniq.oad.min$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.min$gene,
                NonDEGenes=nonDEgenes)
  #Unique to maturing osteocytes
  print(&quot;Unique to maturing osteocytes&quot;)
  nonDEgenes &lt;- DEdata$gene[which(DEdata$cell.assign==&quot;ostadhoc&quot; &amp;
                                  DEdata$cell.subset==&quot;maturing osteocyte&quot; &amp;
                                  DEdata$gene %in% uniq.oad.mat$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.mat$gene,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications, tested in all classifications
  print(&quot;Shared across all ostadhoc classifications, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad$gene,
                NonDEGenes=nonDEgenes)
  #Shared across pre+obl+emb+min, tested in all classifications
  print(&quot;Shared across pre+obl+emb+min, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad.poem$gene,
                NonDEGenes=nonDEgenes)
  #Shared across obl+emb+min, tested in all classifications
  print(&quot;Shared across obl+emb+min, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad.oem$gene,
                NonDEGenes=nonDEgenes)
  
  #Unique to preosteoblasts, tested in all classifications
  print(&quot;Unique to preosteoblasts, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.pre$gene,
                NonDEGenes=nonDEgenes)
  #Unique to osteoblasts, tested in all classifications
  print(&quot;Unique to osteoblasts, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.obl$gene,
                NonDEGenes=nonDEgenes)
  #Unique to embedding osteoblasts, tested in all classifications
  print(&quot;Unique to embedding osteoblasts, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.emb$gene,
                NonDEGenes=nonDEgenes)
  #Unique to mineralizing osteoblasts, tested in all classifications
  print(&quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.min$gene,
                NonDEGenes=nonDEgenes)
  #Unique to maturing osteocytes, tested in all classifications
  print(&quot;Unique to maturing osteocytes, tested in all classifications&quot;)
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.mat$gene,
                NonDEGenes=nonDEgenes)

}</code></pre>
<p>Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared across pre+obl+emb+min, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Pre/Obl/Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad.poem$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared across obl+emb+min, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Obl/Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad.oem$gene),
                                       NonDEGenes=nonDEgenes))
  
  #Unique to preosteoblasts, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Preosteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.pre$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to osteoblasts, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.obl$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to embedding osteoblasts, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Embedding Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.emb$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to mineralizing osteoblasts, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Mineralizing Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.min$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to maturing osteocytes, tested in all classifications
  nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Maturing Osteocytes&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.mat$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-66-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-66-1">
Past versions of unnamed-chunk-66-1.png
</button>
</p>
<div id="fig-unnamed-chunk-66-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-66-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/2f46ddd62bd0a5f9b758038a09abc6e266f4ac05/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-66-1.png" target="_blank">2f46ddd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-67-1.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-diffexp.Rmd/unnamed-chunk-67-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-2">
Past versions of unnamed-chunk-67-2.png
</button>
</p>
<div id="fig-unnamed-chunk-67-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/535b9e22ebc8ab063d62637e164396414bce3022/docs/figure/10x-plot-diffexp.Rmd/unnamed-chunk-67-2.png" target="_blank">535b9e2</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                    Source.DEGenes               Source.GenesOfInterest
45           Unique to Osteoblasts           Blake2020-DE.kidney-Unique
46 Unique to Embedding Osteoblasts           Blake2020-DE.kidney-Unique
85           Unique to Osteoblasts Gokhman2021-DE.Parental.CNCCs-Unique
84        Unique to Preosteoblasts Gokhman2021-DE.Parental.CNCCs-Unique
   DE.Interest     p.value  GeneRatio
45           5 0.003014934 0.06849315
46           3 0.004997557 0.11538462
85           7 0.017801145 0.09589041
84           4 0.032136850 0.12121212</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.enrichment.csv&quot;)</code></pre>
<p>Determine whether EVC2, NFIX, XYLT1, ACAN, COL2A1, and SOX9 overlap with DE genes:</p>
<pre class="r"><code>geneList &lt;- c(&quot;EVC2&quot;,&quot;NFIX&quot;,&quot;XYLT1&quot;,&quot;ACAN&quot;,&quot;COL2A1&quot;,&quot;SOX9&quot;)

#Shared across all ostadhoc classifications, tested in all classifications
print(&quot;Shared across all ostadhoc classifications, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.oad$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Shared across pre+obl+emb+min, tested in all classifications
print(&quot;Shared across pre+obl+emb+min, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.oad.poem$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Shared across obl+emb+min, tested in all classifications
print(&quot;Shared across obl+emb+min, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,unique(shareX.oad.oem$gene)))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to preosteoblasts, tested in all classifications
print(&quot;Unique to preosteoblasts, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.oad.pre$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to osteoblasts, tested in all classifications
print(&quot;Unique to osteoblasts, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Unique to osteoblasts, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.oad.obl$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to embedding osteoblasts, tested in all classifications
print(&quot;Unique to embedding osteoblasts, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.oad.emb$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to mineralizing osteoblasts, tested in all classifications
print(&quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.oad.min$gene))</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>#Unique to maturing osteocytes, tested in all classifications
print(&quot;Unique to maturing osteocytes, tested in all classifications&quot;)</code></pre>
<pre><code>[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;</code></pre>
<pre class="r"><code>print(intersect(geneList,uniqX.oad.mat$gene))</code></pre>
<pre><code>character(0)</code></pre>
<p>Determine which skeletal-phenotype-related genes overlap with DE genes:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print(&quot;Shared across all ostadhoc classifications, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.oad$gene)))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print(&quot;Shared across pre+obl+emb+min, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.oad.poem$gene)))
  
  #Shared across obl+emb+min, tested in all classifications
  print(&quot;Shared across obl+emb+min, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],unique(shareX.oad.oem$gene)))

  #Unique to preosteoblasts, tested in all classifications
  print(&quot;Unique to preosteoblasts, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],uniqX.oad.pre$gene))
  
  #Unique to osteoblasts, tested in all classifications
  print(&quot;Unique to osteoblasts, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],uniqX.oad.obl$gene))

  #Unique to embedding osteoblasts, tested in all classifications
  print(&quot;Unique to embedding osteoblasts, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],uniqX.oad.emb$gene))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print(&quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],uniqX.oad.min$gene))
  
  #Unique to maturing osteocytes, tested in all classifications
  print(&quot;Unique to maturing osteocytes, tested in all classifications&quot;)
  print(intersect(genesExternal[[2]],uniqX.oad.mat$gene))

}</code></pre>
<pre><code>[1] &quot;Hsu2019-GWAS.hip.geometry&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
character(0)
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Styrkarsdottir2019-GWAS.bone.area&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
character(0)
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Yengo2018-GWAS.height&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
[1] &quot;CTSF&quot;
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
[1] &quot;SDCCAG8&quot;
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
[1] &quot;HAUS2&quot;
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
[1] &quot;BCL11A&quot; &quot;SPATA5&quot; &quot;VTI1A&quot; 
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
[1] &quot;PARN&quot;    &quot;COLEC12&quot;
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
[1] &quot;ICAM3&quot; &quot;PXMP4&quot; &quot;CCND1&quot;
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
 [1] &quot;ADAMTS6&quot; &quot;USP36&quot;   &quot;FUS&quot;     &quot;TXNL4B&quot;  &quot;RANBP10&quot; &quot;TUFT1&quot;   &quot;CNIH3&quot;  
 [8] &quot;PIK3R1&quot;  &quot;ABT1&quot;    &quot;BEND6&quot;   &quot;DDAH1&quot;   &quot;SKI&quot;     &quot;TAOK1&quot;   &quot;ADCY9&quot;  
[15] &quot;ZCCHC24&quot; &quot;NBEA&quot;    &quot;RAD51B&quot;  &quot;UBE2L3&quot;  &quot;THBS2&quot;   &quot;VEPH1&quot;  
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
[1] &quot;ZNF784&quot;
[1] &quot;Steinberg2020-GWAS.OA&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
[1] &quot;EXOSC4&quot;
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
[1] &quot;CFH&quot;    &quot;DCPS&quot;   &quot;SEC31A&quot;
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
 [1] &quot;DDAH1&quot;    &quot;DDX19A&quot;   &quot;DGKA&quot;     &quot;FABP5&quot;    &quot;GALE&quot;     &quot;MGST1&quot;   
 [7] &quot;SERPINB8&quot; &quot;SH3KBP1&quot;  &quot;SMYD5&quot;    &quot;TBL2&quot;    
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
[1] &quot;GPT2&quot;
[1] &quot;Morris2020-GWAS.OP&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
[1] &quot;HLA-A&quot;
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
[1] &quot;SLC27A5&quot;
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
[1] &quot;BCL11A&quot;  &quot;FAM174A&quot; &quot;VTI1A&quot;  
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
 [1] &quot;HHAT&quot;    &quot;ANAPC4&quot;  &quot;ADAMTS6&quot; &quot;TRMT11&quot;  &quot;ZCCHC24&quot; &quot;RAD51B&quot;  &quot;ADCY9&quot;  
 [8] &quot;EEF2K&quot;   &quot;USP36&quot;   &quot;UBE2L3&quot;  &quot;LIF&quot;    
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)</code></pre>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict-1" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)</h3>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible-time-2-only-1" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)</h3>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict-time-2-only-1" class="section level3">
<h3>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)</h3>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ggsignif_0.5.0              VennDiagram_1.6.20         
 [3] futile.logger_1.4.3         viridis_0.5.1              
 [5] viridisLite_0.3.0           enrichplot_1.4.0           
 [7] Rgraphviz_2.28.0            graph_1.62.0               
 [9] org.Hs.eg.db_3.8.2          AnnotationDbi_1.46.0       
[11] clusterProfiler_3.12.0      UpSetR_1.4.0               
[13] purrr_0.3.4                 scran_1.12.1               
[15] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
[17] DelayedArray_0.10.0         BiocParallel_1.18.0        
[19] matrixStats_0.57.0          Biobase_2.44.0             
[21] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[23] IRanges_2.18.1              S4Vectors_0.22.1           
[25] BiocGenerics_0.30.0         edgeR_3.26.5               
[27] limma_3.40.6                RColorBrewer_1.1-2         
[29] colorspace_2.0-0            gbm_2.1.5                  
[31] ggplot2_3.3.2               stringr_1.4.0              
[33] stringi_1.4.3               dplyr_1.0.2                
[35] Seurat_3.1.3               

loaded via a namespace (and not attached):
  [1] reticulate_1.16          tidyselect_1.1.0        
  [3] RSQLite_2.1.1            htmlwidgets_1.5.2       
  [5] Rtsne_0.15               munsell_0.5.0           
  [7] codetools_0.2-16         ica_1.0-2               
  [9] statmod_1.4.32           future_1.14.0           
 [11] withr_2.3.0              GOSemSim_2.10.0         
 [13] knitr_1.23               ROCR_1.0-7              
 [15] DOSE_3.10.2              gbRd_0.4-11             
 [17] listenv_0.7.0            labeling_0.4.2          
 [19] Rdpack_0.11-0            git2r_0.26.1            
 [21] urltools_1.7.3           GenomeInfoDbData_1.2.1  
 [23] polyclip_1.10-0          bit64_0.9-7             
 [25] farver_2.0.3             rprojroot_2.0.2         
 [27] vctrs_0.3.5              generics_0.1.0          
 [29] lambda.r_1.2.3           xfun_0.8                
 [31] R6_2.5.0                 ggbeeswarm_0.6.0        
 [33] rsvd_1.0.1               locfit_1.5-9.1          
 [35] gridGraphics_0.4-1       bitops_1.0-6            
 [37] fgsea_1.15.1             promises_1.1.1          
 [39] scales_1.1.1             ggraph_1.0.2            
 [41] beeswarm_0.2.3           gtable_0.3.0            
 [43] npsurv_0.4-0             globals_0.12.4          
 [45] workflowr_1.6.2          rlang_0.4.9             
 [47] splines_3.6.1            lazyeval_0.2.2          
 [49] europepmc_0.3            yaml_2.2.1              
 [51] reshape2_1.4.3           httpuv_1.5.1            
 [53] qvalue_2.16.0            tools_3.6.1             
 [55] ggplotify_0.0.3          ellipsis_0.3.1          
 [57] gplots_3.0.1.1           dynamicTreeCut_1.63-1   
 [59] ggridges_0.5.1           Rcpp_1.0.5              
 [61] plyr_1.8.4               progress_1.2.2          
 [63] zlibbioc_1.30.0          RCurl_1.98-1.1          
 [65] prettyunits_1.1.1        pbapply_1.4-0           
 [67] cowplot_1.1.0            zoo_1.8-6               
 [69] ggrepel_0.8.2            cluster_2.1.0           
 [71] fs_1.3.1                 magrittr_2.0.1          
 [73] futile.options_1.0.1     data.table_1.13.2       
 [75] DO.db_2.9                lmtest_0.9-37           
 [77] triebeard_0.3.0          RANN_2.6.1              
 [79] whisker_0.3-2            fitdistrplus_1.0-14     
 [81] hms_0.5.3                lsei_1.2-0              
 [83] evaluate_0.14            gridExtra_2.3           
 [85] compiler_3.6.1           scater_1.12.2           
 [87] tibble_3.0.4             KernSmooth_2.23-15      
 [89] crayon_1.3.4             htmltools_0.5.0         
 [91] later_1.1.0.1            tidyr_1.1.2             
 [93] RcppParallel_5.0.2       DBI_1.1.0               
 [95] formatR_1.7              tweenr_1.0.1            
 [97] MASS_7.3-51.4            rappdirs_0.3.1          
 [99] Matrix_1.2-18            gdata_2.18.0            
[101] metap_1.1                igraph_1.2.4.1          
[103] pkgconfig_2.0.3          rvcheck_0.1.3           
[105] plotly_4.9.2.1           xml2_1.3.2              
[107] vipor_0.4.5              dqrng_0.2.1             
[109] XVector_0.24.0           bibtex_0.4.2            
[111] digest_0.6.27            sctransform_0.2.0       
[113] RcppAnnoy_0.0.12         tsne_0.1-3              
[115] rmarkdown_1.13           leiden_0.3.1            
[117] fastmatch_1.1-0          uwot_0.1.5              
[119] DelayedMatrixStats_1.6.1 gtools_3.8.1            
[121] lifecycle_0.2.0          nlme_3.1-140            
[123] jsonlite_1.7.1           BiocNeighbors_1.2.0     
[125] pillar_1.4.7             lattice_0.20-38         
[127] httr_1.4.2               survival_2.44-1.1       
[129] GO.db_3.8.2              glue_1.4.2              
[131] png_0.1-7                bit_1.1-14              
[133] ggforce_0.2.2            blob_1.2.0              
[135] BiocSingular_1.0.0       caTools_1.17.1.2        
[137] memoise_1.1.0            irlba_2.3.3             
[139] future.apply_1.3.0       ape_5.4-1               </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
