<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-plot-diffexp-thresh</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-plot-diffexp-thresh</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">September 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree7fa17e693a3e8238a3d607166efdf3c0bc59648etargetblank7fa17e6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/7fa17e693a3e8238a3d607166efdf3c0bc59648e" target="_blank">7fa17e6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree7fa17e693a3e8238a3d607166efdf3c0bc59648etargetblank7fa17e6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/7fa17e693a3e8238a3d607166efdf3c0bc59648e" target="_blank">7fa17e6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhoc.enrichment.csv
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.enrichment.csv
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.enrichment.csv
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.enrichment.csv
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.enrichment.csv
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

Unstaged changes:
    Modified:   analysis/10x-plot-cormotif-cnts-thresh_v1.Rmd
    Modified:   analysis/10x-plot-cormotif-cnts-thresh_v2.Rmd
    Modified:   analysis/10x-plot-cormotif-cnts_v2.Rmd
    Modified:   analysis/10x-plot-diffexp.Rmd
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Modified:   data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-plot-diffexp-thresh.Rmd</code>) and HTML (<code>docs/10x-plot-diffexp-thresh.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/analysis/10x-plot-diffexp-thresh.Rmd" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
updated threshold code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/10x-plot-diffexp-thresh.html" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
updated threshold code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/analysis/10x-plot-diffexp-thresh.Rmd" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
<td>
checked DE thresholds
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/10x-plot-diffexp-thresh.html" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
<td>
checked DE thresholds
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---differential-expression-different-thresholds" class="section level1">
<h1>10X Data Cellranger Analysis - Differential Expression (different thresholds)</h1>
<p>View results of pairwise differential expression tests between humans and chimpanzees within stages of differentiation using different FDR thresholds.</p>
<ul>
<li>within stages of differentiation</li>
<li>within stages of osteogenesis</li>
<li>within stages of osteogenesis (T2 only)</li>
</ul>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ggsignif)</code></pre>
<div id="load-external-gene-datasets" class="section level2">
<h2>Load External Gene Datasets</h2>
<p>Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.</p>
<p>Genes previously identified as human and chimpanzee differences:</p>
<ul>
<li>Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
<ul>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Blake et al. 2020 Genome Research (Supplemental Table S4)
<ul>
<li>DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each tissue type</li>
</ul></li>
<li>Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
<ul>
<li>DE genes in parental iPSC lines (n = 7655)</li>
<li>DE genes in hybrid iPSC lines (n = 6009)</li>
<li>DE genes in parental CNCC lines (n = 2223)</li>
<li>DE genes in hybrid CNCC lines (n = 3612)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
<ul>
<li>DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)</li>
<li>AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)</li>
<li>(human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)</li>
<li>(chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)</li>
<li>Note: only DMRs that overlapped promotor regions of genes were included</li>
<li>Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels</li>
</ul></li>
</ul>
<p>General skeleton-related and skeletal phenotype-related genes:</p>
<ul>
<li>Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
<ul>
<li>hip geometry GWAS loci (n = 10)</li>
</ul></li>
<li>Styrkarsdottir et al. 2019 Nature Communications (Table 1)
<ul>
<li>DXA bone area GWAS loci (n = 21 associated with 20 genes)</li>
</ul></li>
<li>Yengo et al. 2018 Human Molecular Genetics
<ul>
<li>height GWAS loci (n = 3290) (data provided at <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files" class="uri">https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files</a>)</li>
<li>height GWAS loci with associated eQTLs (n = 610) (list not provided)</li>
<li>Note: only used unique gene symbols associated with height GWAS loci (n = 1500)</li>
</ul></li>
</ul>
<p>Genes associated with skeletal diseases (OA + OP):</p>
<ul>
<li>Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
<ul>
<li>genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)</li>
</ul></li>
<li>Morris et al. 2019 Nature Genetics (Supplemental Table 2)
<ul>
<li>OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)</li>
</ul></li>
</ul>
<pre class="r"><code>#Define function for testing significant enrichment of gene sets

genesetEnrich &lt;- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative=&quot;greater&quot;))

}

genesetEnrich.data &lt;- function(InternalDataCutoff, InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test and output data
  data &lt;- data.frame(&quot;FDR.Cutoff&quot;=InternalDataCutoff,
                     &quot;Source.DEGenes&quot;=InternalDataSource,
                     &quot;Source.GenesOfInterest&quot;=ExternalDataSource,
                     &quot;DE.Interest&quot;=DE.Interest,
                     &quot;NonDE.Interest&quot;=NonDE.Interest,
                     &quot;DE.NotInterest&quot;=DE.NotInterest,
                     &quot;NonDE.NotInterest&quot;=NonDE.NotInterest,
                     &quot;p.value&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$p.value,
                     &quot;odds.ratio&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$estimate,
                     row.names=NULL)
  return(data)

}</code></pre>
<pre class="r"><code>#Load external datasets (manually currated from papers)
gallegoromero2015    &lt;- read.csv(&quot;./data/external-gene-sets/GallegoRomero2015eLife.csv&quot;)
blake2018            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2018GenBio.csv&quot;)
blake2020            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2020GenRes.csv&quot;)
gokhman2021          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2021NatGen.csv&quot;)
gokhman2020          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm.csv&quot;)
gokhman2020.skeleton &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv&quot;)
hsu2019              &lt;- read.csv(&quot;./data/external-gene-sets/Hsu2019JBoneMinRes.csv&quot;)
styrkarsdottir2019   &lt;- read.csv(&quot;./data/external-gene-sets/Styrkarsdottir2019NatComm.csv&quot;)
yengo2018            &lt;- read.csv(&quot;./data/external-gene-sets/Yengo2018HumMolGen.csv&quot;)
steinberg2020        &lt;- read.csv(&quot;./data/external-gene-sets/Steinberg2020biorxiv.csv&quot;)
morris2019           &lt;- read.csv(&quot;./data/external-gene-sets/Morris2019NatGen.csv&quot;)
#cibrianuhalte2017    &lt;- read.csv(&quot;./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv&quot;)
#tachmazidou2019      &lt;- read.csv(&quot;./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv&quot;)</code></pre>
</div>
<div id="load-output-of-de-analyses" class="section level2">
<h2>Load output of DE analyses</h2>
<pre class="r"><code>#Load DE results

compare1 &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.rds&quot;)
compare2 &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.rds&quot;)
compare3 &lt;- readRDS(&quot;./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.T2.rds&quot;)

DEdata &lt;- rbind(compare1,compare2,compare3)
colnames(DEdata)[5] &lt;- &quot;FDR&quot;
DEdata &lt;- as_tibble(DEdata)

rm(compare1,compare2,compare3)</code></pre>
</div>
<div id="define-fdr-thresholds" class="section level2">
<h2>Define FDR Thresholds</h2>
<p>Define different FDR thresholds (0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.01, 0.001).</p>
<pre class="r"><code>cutoffs &lt;- c(0.50, 0.40, 0.30, 0.20, 0.10, 0.05, 0.01, 0.001)</code></pre>
</div>
<div id="stages-of-differentiation-at-collection" class="section level2">
<h2>Stages of Differentiation at Collection</h2>
<p>Plot number of DE genes for each cutoff.</p>
<pre class="r"><code>x=1

while (x &lt;= length(cutoffs)) {
  
  #cutoff
  print(paste0(&quot;Posterior Probability Cutoff: &quot;,cutoffs[x]))

  DEgenes &lt;-
    DEdata %&gt;%
    filter(FDR&lt;=cutoffs[x]) %&gt;%
    dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

  #number of DE genes
  print(table(DEgenes$cell.subset))
  print(ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
          geom_bar() +
          ylab(&#39;Number of DE Genes&#39;) +
          xlab(&#39;&#39;) +
          scale_fill_manual(name=&quot;Stage&quot;,
                            labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
                            values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
          theme_classic() +
          theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12)))

  #overlap of DE genes
  combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                          ), full_join)
  combined[is.na(combined)] &lt;- 0
  print(upset(combined,
              keep.order=TRUE,
              empty.intersections=&quot;on&quot;,
              intersections=list(list(&quot;Time.0&quot;),
                                 list(&quot;Time.1&quot;),
                                 list(&quot;Time.2&quot;),
                                 list(&quot;Time.0&quot;,&quot;Time.1&quot;),
                                 list(&quot;Time.1&quot;,&quot;Time.2&quot;),
                                 list(&quot;Time.0&quot;,&quot;Time.2&quot;),
                                 list(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;)),
              text.scale=2))

  x=x+1

}</code></pre>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.5&quot;

Time 0 Time 1 Time 2 
  8999   7846   7143 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.4&quot;

Time 0 Time 1 Time 2 
  8316   7015   6312 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-4">
Past versions of unnamed-chunk-6-4.png
</button>
</p>
<div id="fig-unnamed-chunk-6-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.3&quot;

Time 0 Time 1 Time 2 
  7477   6107   5305 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-5">
Past versions of unnamed-chunk-6-5.png
</button>
</p>
<div id="fig-unnamed-chunk-6-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-6">
Past versions of unnamed-chunk-6-6.png
</button>
</p>
<div id="fig-unnamed-chunk-6-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.2&quot;

Time 0 Time 1 Time 2 
  6464   5062   4097 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-7">
Past versions of unnamed-chunk-6-7.png
</button>
</p>
<div id="fig-unnamed-chunk-6-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-8">
Past versions of unnamed-chunk-6-8.png
</button>
</p>
<div id="fig-unnamed-chunk-6-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.1&quot;

Time 0 Time 1 Time 2 
  5133   3529   2604 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-9">
Past versions of unnamed-chunk-6-9.png
</button>
</p>
<div id="fig-unnamed-chunk-6-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-10">
Past versions of unnamed-chunk-6-10.png
</button>
</p>
<div id="fig-unnamed-chunk-6-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-10.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.05&quot;

Time 0 Time 1 Time 2 
  4002   2441   1603 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-11">
Past versions of unnamed-chunk-6-11.png
</button>
</p>
<div id="fig-unnamed-chunk-6-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-11.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-12">
Past versions of unnamed-chunk-6-12.png
</button>
</p>
<div id="fig-unnamed-chunk-6-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-12.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.01&quot;

Time 0 Time 1 Time 2 
  2098    904    446 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-13">
Past versions of unnamed-chunk-6-13.png
</button>
</p>
<div id="fig-unnamed-chunk-6-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-13.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-14">
Past versions of unnamed-chunk-6-14.png
</button>
</p>
<div id="fig-unnamed-chunk-6-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-14.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.001&quot;

Time 0 Time 1 Time 2 
   697    196     81 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-15">
Past versions of unnamed-chunk-6-15.png
</button>
</p>
<div id="fig-unnamed-chunk-6-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-15.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-16">
Past versions of unnamed-chunk-6-16.png
</button>
</p>
<div id="fig-unnamed-chunk-6-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-6-16.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;FDR.Cutoff&quot;=double(),
                       &quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

x=1

while (x &lt;= length(cutoffs)) {
  
  for (genesExternal in listsExternal) {
    
    #Pairwise DE genes and cell subset overlap
    DEgenes &lt;-
      DEdata %&gt;%
      filter(FDR&lt;=cutoffs[x]) %&gt;%
      dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
      dplyr::select(cell.subset,gene,logFC,FDR)
    comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                              ), full_join)
    comboGenes[is.na(comboGenes)] &lt;- 0
    
    #Isolate genes tested in all cell classifications
    DEtests &lt;-
      DEdata %&gt;%
        dplyr::filter(cell.assign==&quot;stage&quot;) %&gt;%
        dplyr::select(cell.subset,gene,logFC,FDR)
    comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 0&quot;) %&gt;% dplyr::select(gene), &quot;Time.0&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 1&quot;) %&gt;% dplyr::select(gene), &quot;Time.1&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;Time 2&quot;) %&gt;% dplyr::select(gene), &quot;Time.2&quot;=1)
                              ), full_join)
    comboTests[is.na(comboTests)] &lt;- 0
    tests &lt;- pull(comboTests %&gt;%
                    filter(Time.0==1,
                           Time.1==1,
                           Time.2==1) %&gt;%
                    dplyr::select(gene))

    #Shared in All Stages (tested in all classifications)
    share.stg &lt;- DEgenes %&gt;% 
                 filter(gene %in%
                          pull(comboGenes %&gt;%
                                 filter(Time.0==1,
                                        Time.1==1,
                                        Time.2==1) %&gt;%
                                 dplyr::select(gene))) %&gt;% 
                 arrange(gene)
    shareX.stg &lt;- share.stg[which(share.stg$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.stg$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in All&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.stg$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared in Time 0 &amp; Time 1 (tested in all classifications)
    share.stg.t01 &lt;- DEgenes %&gt;% 
                     filter(gene %in%
                              pull(comboGenes %&gt;%
                                     filter(Time.0==1,
                                            Time.1==1,
                                            Time.2==0) %&gt;%
                                     dplyr::select(gene))) %&gt;% 
                     arrange(gene)
    shareX.stg.t01 &lt;- share.stg.t01[which(share.stg.t01$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Time 0 &amp; Time 1&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.stg.t01$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared in Time 1 &amp; Time 2 (tested in all classifications)
    share.stg.t12 &lt;- DEgenes %&gt;% 
                     filter(gene %in%
                              pull(comboGenes %&gt;%
                                     filter(Time.0==0,
                                            Time.1==1,
                                            Time.2==1) %&gt;%
                                     dplyr::select(gene))) %&gt;% 
                     arrange(gene)
    shareX.stg.t12 &lt;- share.stg.t12[which(share.stg.t12$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Time 1 &amp; Time 2&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.stg.t12$gene),
                                         NonDEGenes=nonDEgenes))  
    
    #Unique to Time 0 (tested in all classifications)
    uniq.stg.t0 &lt;- DEgenes %&gt;% 
                   filter(cell.subset==&quot;Time 0&quot;,
                          gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Time.0==1,
                                          Time.1==0,
                                          Time.2==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    uniqX.stg.t0 &lt;- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Time 0&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.stg.t0$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to Time 1 (tested in all classifications)
    uniq.stg.t1 &lt;- DEgenes %&gt;% 
                   filter(cell.subset==&quot;Time 1&quot;,
                          gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Time.0==0,
                                          Time.1==1,
                                          Time.2==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    uniqX.stg.t1 &lt;- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Time 1&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.stg.t1$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to Time 2 (tested in all classifications)
    uniq.stg.t2 &lt;- DEgenes %&gt;% 
                   filter(cell.subset==&quot;Time 2&quot;,
                          gene %in%
                            pull(comboGenes %&gt;%
                                   filter(Time.0==0,
                                          Time.1==0,
                                          Time.2==1) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    uniqX.stg.t2 &lt;- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Time 2&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.stg.t2$gene,
                                         NonDEGenes=nonDEgenes))
  
  }
  
  x=x+1

}</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Add p-value color variable
DEenrich$p.value.color &lt;- &quot;nonsignificant&quot;
DEenrich$p.value.color[which(DEenrich$p.value&lt;0.05)] &lt;- &quot;significant&quot;
DEenrich$p.value.color &lt;- as.factor(DEenrich$p.value.color)</code></pre>
<p>Check how the gene ratio and p value change across posterior probability cutoffs for different external gene datasets.</p>
<pre class="r"><code>#aiming for a cutoff at which only Unique to Time 0 are enriched
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-3">
Past versions of unnamed-chunk-8-3.png
</button>
</p>
<div id="fig-unnamed-chunk-8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-4">
Past versions of unnamed-chunk-8-4.png
</button>
</p>
<div id="fig-unnamed-chunk-8-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-4.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-5">
Past versions of unnamed-chunk-8-5.png
</button>
</p>
<div id="fig-unnamed-chunk-8-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-5.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-6">
Past versions of unnamed-chunk-8-6.png
</button>
</p>
<div id="fig-unnamed-chunk-8-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-6.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-7">
Past versions of unnamed-chunk-8-7.png
</button>
</p>
<div id="fig-unnamed-chunk-8-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-7.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-8">
Past versions of unnamed-chunk-8-8.png
</button>
</p>
<div id="fig-unnamed-chunk-8-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-8.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-9">
Past versions of unnamed-chunk-8-9.png
</button>
</p>
<div id="fig-unnamed-chunk-8-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-9.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#no prior expectations
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-10">
Past versions of unnamed-chunk-8-10.png
</button>
</p>
<div id="fig-unnamed-chunk-8-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-10.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-11">
Past versions of unnamed-chunk-8-11.png
</button>
</p>
<div id="fig-unnamed-chunk-8-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-11.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-12">
Past versions of unnamed-chunk-8-12.png
</button>
</p>
<div id="fig-unnamed-chunk-8-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-8-12.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)</h2>
<p>Plot number of DE genes for each cutoff.</p>
<pre class="r"><code>x=1

while (x &lt;= length(cutoffs)) {
  
  #cutoff
  print(paste0(&quot;Posterior Probability Cutoff: &quot;,cutoffs[x]))

  DEgenes &lt;-
    DEdata %&gt;%
    filter(FDR&lt;=cutoffs[x]) %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

  DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                                levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                                labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

  #number of DE genes
  print(table(DEgenes$cell.subset))
  print(ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
          geom_bar() +
          ylab(&#39;Number of DE Genes&#39;) +
          xlab(&#39;&#39;) +
          scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                            labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                            values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
          theme_classic() +
          theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10)))

  #overlap of DE genes
  combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                          data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                        ), full_join)
  combined[is.na(combined)] &lt;- 0
  print(upset(combined,
              keep.order=TRUE,
              empty.intersections=&quot;on&quot;))

  x=x+1

}</code></pre>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.5&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    5972                     6362                     6010 
mineralizing\nosteoblast      maturing\nosteocyte 
                    6842                     6250 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.4&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    4959                     5375                     5111 
mineralizing\nosteoblast      maturing\nosteocyte 
                    6012                     5218 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-4">
Past versions of unnamed-chunk-9-4.png
</button>
</p>
<div id="fig-unnamed-chunk-9-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.3&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    3934                     4354                     4043 
mineralizing\nosteoblast      maturing\nosteocyte 
                    5083                     4121 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-5">
Past versions of unnamed-chunk-9-5.png
</button>
</p>
<div id="fig-unnamed-chunk-9-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-6">
Past versions of unnamed-chunk-9-6.png
</button>
</p>
<div id="fig-unnamed-chunk-9-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.2&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    2728                     3224                     2874 
mineralizing\nosteoblast      maturing\nosteocyte 
                    3950                     2957 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-7">
Past versions of unnamed-chunk-9-7.png
</button>
</p>
<div id="fig-unnamed-chunk-9-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-8">
Past versions of unnamed-chunk-9-8.png
</button>
</p>
<div id="fig-unnamed-chunk-9-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.1&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    1410                     1858                     1534 
mineralizing\nosteoblast      maturing\nosteocyte 
                    2619                     1578 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-9">
Past versions of unnamed-chunk-9-9.png
</button>
</p>
<div id="fig-unnamed-chunk-9-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-10">
Past versions of unnamed-chunk-9-10.png
</button>
</p>
<div id="fig-unnamed-chunk-9-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-10.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.05&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                     657                      996                      772 
mineralizing\nosteoblast      maturing\nosteocyte 
                    1569                      761 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-11">
Past versions of unnamed-chunk-9-11.png
</button>
</p>
<div id="fig-unnamed-chunk-9-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-11.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-12">
Past versions of unnamed-chunk-9-12.png
</button>
</p>
<div id="fig-unnamed-chunk-9-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-12.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.01&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                     144                      242                      142 
mineralizing\nosteoblast      maturing\nosteocyte 
                     429                      115 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-13">
Past versions of unnamed-chunk-9-13.png
</button>
</p>
<div id="fig-unnamed-chunk-9-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-13.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-14">
Past versions of unnamed-chunk-9-14.png
</button>
</p>
<div id="fig-unnamed-chunk-9-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-14.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.001&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                      17                       27                       26 
mineralizing\nosteoblast      maturing\nosteocyte 
                      69                        9 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-15">
Past versions of unnamed-chunk-9-15.png
</button>
</p>
<div id="fig-unnamed-chunk-9-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-15.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-16">
Past versions of unnamed-chunk-9-16.png
</button>
</p>
<div id="fig-unnamed-chunk-9-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-9-16.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;FDR.Cutoff&quot;=double(),
                       &quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())


x=1

while (x &lt;= length(cutoffs)) {
  
  for (genesExternal in listsExternal) {
    
    #Pairwise DE genes and cell subset overlap
    DEgenes &lt;-
      DEdata %&gt;%
      filter(FDR&lt;=cutoffs[x]) %&gt;%
      dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
      dplyr::select(cell.subset,gene,logFC)
    comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                              ), full_join)
    comboGenes[is.na(comboGenes)] &lt;- 0
    
    #Isolate genes tested in all cell classifications
    DEtests &lt;-
      DEdata %&gt;%
        dplyr::filter(cell.assign==&quot;ostadhoc&quot;) %&gt;%
        dplyr::select(cell.subset,gene,logFC,FDR)
    comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                              data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                              ), full_join)
    comboTests[is.na(comboTests)] &lt;- 0
    tests &lt;- pull(comboTests %&gt;%
                    filter(preosteoblast==1,
                           osteoblast==1,
                           embedding.osteoblast==1,
                           mineralizing.osteoblast==1,
                           maturing.osteocyte==1) %&gt;%
                    dplyr::select(gene))

    #Shared across all ostadhoc classifications, tested in all classifications
    share.oad &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(preosteoblast==1,
                                      osteoblast==1,
                                      embedding.osteoblast==1,
                                      mineralizing.osteoblast==1,
                                      maturing.osteocyte==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
    shareX.oad &lt;- share.oad[which(share.oad$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in All&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared across pre+obl+emb+min, tested in all classifications
    share.oad.poem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==1,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    shareX.oad.poem &lt;- share.oad.poem[which(share.oad.poem$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Pre/Obl/Emb/Min&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad.poem$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared across obl+emb+min, tested in all classifications
    share.oad.oem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==0,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    shareX.oad.oem &lt;- share.oad.oem[which(share.oad.oem$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Obl/Emb/Min&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad.oem$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Unique to preosteoblasts, tested in all classifications
    uniq.oad.pre &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==1,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.pre &lt;- uniq.oad.pre[which(uniq.oad.pre$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Preosteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.pre$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to osteoblasts, tested in all classifications
    uniq.oad.obl &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==1,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.obl &lt;- uniq.oad.obl[which(uniq.oad.obl$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.obl$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to embedding osteoblasts, tested in all classifications
    uniq.oad.emb &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==1,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.emb &lt;- uniq.oad.emb[which(uniq.oad.emb$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Embedding Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.emb$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to mineralizing osteoblasts, tested in all classifications
    uniq.oad.min &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==1,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.min &lt;- uniq.oad.min[which(uniq.oad.min$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Mineralizing Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.min$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to maturing osteocytes, tested in all classifications
    uniq.oad.mat &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==1) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.mat &lt;- uniq.oad.mat[which(uniq.oad.mat$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Maturing Osteocytes&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.mat$gene,
                                         NonDEGenes=nonDEgenes))
  
  }
  
  x=x+1
  
}</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Add p-value color variable
DEenrich$p.value.color &lt;- &quot;nonsignificant&quot;
DEenrich$p.value.color[which(DEenrich$p.value&lt;0.05)] &lt;- &quot;significant&quot;
DEenrich$p.value.color &lt;- as.factor(DEenrich$p.value.color)</code></pre>
<p>Check how the gene ratio and p value change across posterior probability cutoffs for different external gene datasets.</p>
<pre class="r"><code>#aiming for a cutoff at which only Unique to Time 0 are enriched
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-4">
Past versions of unnamed-chunk-11-4.png
</button>
</p>
<div id="fig-unnamed-chunk-11-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-4.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-5">
Past versions of unnamed-chunk-11-5.png
</button>
</p>
<div id="fig-unnamed-chunk-11-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-5.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-6">
Past versions of unnamed-chunk-11-6.png
</button>
</p>
<div id="fig-unnamed-chunk-11-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-6.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-7">
Past versions of unnamed-chunk-11-7.png
</button>
</p>
<div id="fig-unnamed-chunk-11-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-7.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-8">
Past versions of unnamed-chunk-11-8.png
</button>
</p>
<div id="fig-unnamed-chunk-11-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-8.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-9">
Past versions of unnamed-chunk-11-9.png
</button>
</p>
<div id="fig-unnamed-chunk-11-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-9.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#no prior expectations
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-10">
Past versions of unnamed-chunk-11-10.png
</button>
</p>
<div id="fig-unnamed-chunk-11-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-10.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-11">
Past versions of unnamed-chunk-11-11.png
</button>
</p>
<div id="fig-unnamed-chunk-11-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-11.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-12">
Past versions of unnamed-chunk-11-12.png
</button>
</p>
<div id="fig-unnamed-chunk-11-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-11-12.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible-time-2-only" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)</h2>
<p>Plot number of DE genes for each cutoff.</p>
<pre class="r"><code>x=1

while (x &lt;= length(cutoffs)) {
  
  #cutoff
  print(paste0(&quot;Posterior Probability Cutoff: &quot;,cutoffs[x]))

  DEgenes &lt;-
    DEdata %&gt;%
    filter(FDR&lt;=cutoffs[x]) %&gt;%
    dplyr::filter(cell.assign==&quot;ostadhoc.T2&quot;) %&gt;%
    dplyr::select(cell.subset,gene,logFC)

  DEgenes$cell.subset &lt;- factor(DEgenes$cell.subset,
                                levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                                labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))

  #number of DE genes
  print(table(DEgenes$cell.subset))
  print(ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
          geom_bar() +
          ylab(&#39;Number of DE Genes&#39;) +
          xlab(&#39;&#39;) +
          scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                            labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                            values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
          theme_classic() +
          theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10)))

  #overlap of DE genes
  if(cutoffs[x]==0.001) {
      combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1)
                        ), full_join)
      combined[is.na(combined)] &lt;- 0
      combined$maturing.osteocyte &lt;- rep(0, dim(combined)[1])
  }
  else {
      combined &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)            %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)               %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding\nosteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing\nosteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                              data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing\nosteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                        ), full_join)
      combined[is.na(combined)] &lt;- 0
  }
  print(upset(combined,
              keep.order=TRUE,
              empty.intersections=&quot;on&quot;))
  print(upset(combined,
              keep.order=TRUE,
              empty.intersections=&quot;on&quot;))

  x=x+1

}</code></pre>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.5&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    5724                     6304                     5423 
mineralizing\nosteoblast      maturing\nosteocyte 
                    6038                     5428 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-3">
Past versions of unnamed-chunk-12-3.png
</button>
</p>
<div id="fig-unnamed-chunk-12-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.4&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    4737                     5337                     4374 
mineralizing\nosteoblast      maturing\nosteocyte 
                    5048                     4334 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-4">
Past versions of unnamed-chunk-12-4.png
</button>
</p>
<div id="fig-unnamed-chunk-12-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-5">
Past versions of unnamed-chunk-12-5.png
</button>
</p>
<div id="fig-unnamed-chunk-12-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-6">
Past versions of unnamed-chunk-12-6.png
</button>
</p>
<div id="fig-unnamed-chunk-12-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.3&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    3661                     4297                     3367 
mineralizing\nosteoblast      maturing\nosteocyte 
                    4103                     3182 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-7">
Past versions of unnamed-chunk-12-7.png
</button>
</p>
<div id="fig-unnamed-chunk-12-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-8">
Past versions of unnamed-chunk-12-8.png
</button>
</p>
<div id="fig-unnamed-chunk-12-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-9">
Past versions of unnamed-chunk-12-9.png
</button>
</p>
<div id="fig-unnamed-chunk-12-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.2&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    2385                     3159                     2209 
mineralizing\nosteoblast      maturing\nosteocyte 
                    3023                     1992 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-10">
Past versions of unnamed-chunk-12-10.png
</button>
</p>
<div id="fig-unnamed-chunk-12-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-10.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-11">
Past versions of unnamed-chunk-12-11.png
</button>
</p>
<div id="fig-unnamed-chunk-12-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-11.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-12">
Past versions of unnamed-chunk-12-12.png
</button>
</p>
<div id="fig-unnamed-chunk-12-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-12.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.1&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                    1177                     1826                     1052 
mineralizing\nosteoblast      maturing\nosteocyte 
                    1644                      878 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-13">
Past versions of unnamed-chunk-12-13.png
</button>
</p>
<div id="fig-unnamed-chunk-12-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-13.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-14">
Past versions of unnamed-chunk-12-14.png
</button>
</p>
<div id="fig-unnamed-chunk-12-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-14.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-15">
Past versions of unnamed-chunk-12-15.png
</button>
</p>
<div id="fig-unnamed-chunk-12-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-15.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.05&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                     552                      951                      502 
mineralizing\nosteoblast      maturing\nosteocyte 
                     935                      378 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-16">
Past versions of unnamed-chunk-12-16.png
</button>
</p>
<div id="fig-unnamed-chunk-12-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-16.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-17">
Past versions of unnamed-chunk-12-17.png
</button>
</p>
<div id="fig-unnamed-chunk-12-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-17.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-18">
Past versions of unnamed-chunk-12-18.png
</button>
</p>
<div id="fig-unnamed-chunk-12-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-18.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.01&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                     121                      235                       74 
mineralizing\nosteoblast      maturing\nosteocyte 
                     241                       34 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-19">
Past versions of unnamed-chunk-12-19.png
</button>
</p>
<div id="fig-unnamed-chunk-12-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-19.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-20">
Past versions of unnamed-chunk-12-20.png
</button>
</p>
<div id="fig-unnamed-chunk-12-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-20.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-21">
Past versions of unnamed-chunk-12-21.png
</button>
</p>
<div id="fig-unnamed-chunk-12-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-21.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Posterior Probability Cutoff: 0.001&quot;

           preosteoblast               osteoblast    embedding\nosteoblast 
                      15                       25                       14 
mineralizing\nosteoblast      maturing\nosteocyte 
                      23                        0 </code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-22">
Past versions of unnamed-chunk-12-22.png
</button>
</p>
<div id="fig-unnamed-chunk-12-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-22.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-23">
Past versions of unnamed-chunk-12-23.png
</button>
</p>
<div id="fig-unnamed-chunk-12-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-23.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-24">
Past versions of unnamed-chunk-12-24.png
</button>
</p>
<div id="fig-unnamed-chunk-12-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-12-24.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;FDR.Cutoff&quot;=double(),
                       &quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())


x=1

while (x &lt;= length(cutoffs)) {
  
  for (genesExternal in listsExternal) {
    
    #Pairwise DE genes and cell subset overlap
    DEgenes &lt;-
      DEdata %&gt;%
      filter(FDR&lt;=cutoffs[x]) %&gt;%
      dplyr::filter(cell.assign==&quot;ostadhoc.T2&quot;) %&gt;%
      dplyr::select(cell.subset,gene,logFC)
    if(cutoffs[x]==0.001) {
      comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1)
                                ), full_join)
      comboGenes[is.na(comboGenes)] &lt;- 0
      comboGenes$maturing.osteocyte &lt;- rep(0, dim(comboGenes)[1])
    }
    else {
      comboGenes &lt;- reduce(list(data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                                data.frame(gene=DEgenes %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                                ), full_join)
      comboGenes[is.na(comboGenes)] &lt;- 0
    }

    #Isolate genes tested in all cell classifications
    DEtests &lt;-
      DEdata %&gt;%
      dplyr::filter(cell.assign==&quot;ostadhoc.T2&quot;) %&gt;%
      dplyr::select(cell.subset,gene,logFC,FDR)
    if(cutoffs[x]==0.001) {
      comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1)
                                ), full_join)
      comboTests[is.na(comboTests)] &lt;- 0
      comboTests$maturing.osteocyte &lt;- rep(0, dim(comboTests)[1])
    }
    else {
      comboTests &lt;- reduce(list(data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;preosteoblast&quot;)           %&gt;% dplyr::select(gene), &quot;preosteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;osteoblast&quot;)              %&gt;% dplyr::select(gene), &quot;osteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;embedding osteoblast&quot;)    %&gt;% dplyr::select(gene), &quot;embedding.osteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;mineralizing osteoblast&quot;) %&gt;% dplyr::select(gene), &quot;mineralizing.osteoblast&quot;=1),
                                data.frame(gene=DEtests %&gt;% filter(cell.subset==&quot;maturing osteocyte&quot;)      %&gt;% dplyr::select(gene), &quot;maturing.osteocyte&quot;=1)
                                ), full_join)
      comboTests[is.na(comboTests)] &lt;- 0
    }
    tests &lt;- pull(comboTests %&gt;%
                    filter(preosteoblast==1,
                           osteoblast==1,
                           embedding.osteoblast==1,
                           mineralizing.osteoblast==1,
                           maturing.osteocyte==1) %&gt;%
                    dplyr::select(gene))

    #Shared across all ostadhoc classifications, tested in all classifications
    share.oad &lt;- DEgenes %&gt;% 
               filter(gene %in%
                        pull(comboGenes %&gt;%
                               filter(preosteoblast==1,
                                      osteoblast==1,
                                      embedding.osteoblast==1,
                                      mineralizing.osteoblast==1,
                                      maturing.osteocyte==1) %&gt;%
                               dplyr::select(gene))) %&gt;% 
               arrange(gene)
    shareX.oad &lt;- share.oad[which(share.oad$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in All&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared across pre+obl+emb+min, tested in all classifications
    share.oad.poem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==1,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    shareX.oad.poem &lt;- share.oad.poem[which(share.oad.poem$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Pre/Obl/Emb/Min&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad.poem$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Shared across obl+emb+min, tested in all classifications
    share.oad.oem &lt;- DEgenes %&gt;% 
                   filter(gene %in%
                            pull(comboGenes %&gt;%
                                   filter(preosteoblast==0,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %&gt;%
                                   dplyr::select(gene))) %&gt;% 
                   arrange(gene)
    shareX.oad.oem &lt;- share.oad.oem[which(share.oad.oem$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Shared in Obl/Emb/Min&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=unique(shareX.oad.oem$gene),
                                         NonDEGenes=nonDEgenes))
    
    #Unique to preosteoblasts, tested in all classifications
    uniq.oad.pre &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==1,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.pre &lt;- uniq.oad.pre[which(uniq.oad.pre$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Preosteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.pre$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to osteoblasts, tested in all classifications
    uniq.oad.obl &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==1,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.obl &lt;- uniq.oad.obl[which(uniq.oad.obl$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.obl$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to embedding osteoblasts, tested in all classifications
    uniq.oad.emb &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==1,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.emb &lt;- uniq.oad.emb[which(uniq.oad.emb$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Embedding Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.emb$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to mineralizing osteoblasts, tested in all classifications
    uniq.oad.min &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==1,
                                         maturing.osteocyte==0) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.min &lt;- uniq.oad.min[which(uniq.oad.min$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Mineralizing Osteoblasts&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.min$gene,
                                         NonDEGenes=nonDEgenes))
    
    #Unique to maturing osteocytes, tested in all classifications
    uniq.oad.mat &lt;- DEgenes %&gt;% 
                  filter(gene %in%
                           pull(comboGenes %&gt;%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==1) %&gt;%
                                  dplyr::select(gene))) %&gt;% 
                  arrange(gene)
    uniqX.oad.mat &lt;- uniq.oad.mat[which(uniq.oad.mat$gene %in% tests),]
    nonDEgenes &lt;- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
    DEenrich &lt;- rbind(DEenrich,
                      genesetEnrich.data(InternalDataCutoff=cutoffs[x],
                                         InternalDataSource=&quot;Unique to Maturing Osteocytes&quot;,
                                         ExternalDataSource=genesExternal[[1]],
                                         GenesOfInterest=genesExternal[[2]],
                                         DEgenes=uniqX.oad.mat$gene,
                                         NonDEGenes=nonDEgenes))
  
  }
  
  x=x+1
  
}</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Add p-value color variable
DEenrich$p.value.color &lt;- &quot;nonsignificant&quot;
DEenrich$p.value.color[which(DEenrich$p.value&lt;0.05)] &lt;- &quot;significant&quot;
DEenrich$p.value.color &lt;- as.factor(DEenrich$p.value.color)</code></pre>
<p>Check how the gene ratio and p value change across posterior probability cutoffs for different external gene datasets.</p>
<pre class="r"><code>#aiming for a cutoff at which only Unique to Time 0 are enriched
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-1.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-2.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-3.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-4">
Past versions of unnamed-chunk-14-4.png
</button>
</p>
<div id="fig-unnamed-chunk-14-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-4.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-4.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-5">
Past versions of unnamed-chunk-14-5.png
</button>
</p>
<div id="fig-unnamed-chunk-14-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-5.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-5.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-6">
Past versions of unnamed-chunk-14-6.png
</button>
</p>
<div id="fig-unnamed-chunk-14-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-6.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-6.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#aiming for a cutoff which minimizes enrichments across cell types
tempSource &lt;- c(&quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-7">
Past versions of unnamed-chunk-14-7.png
</button>
</p>
<div id="fig-unnamed-chunk-14-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-7.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-7.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-8">
Past versions of unnamed-chunk-14-8.png
</button>
</p>
<div id="fig-unnamed-chunk-14-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-8.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-8.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-9">
Past versions of unnamed-chunk-14-9.png
</button>
</p>
<div id="fig-unnamed-chunk-14-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-9.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1c012e52cf0c550665750d98f5bce1a68e06adff/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-9.png" target="_blank">1c012e5</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#no prior expectations
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=p.value, color=p.value.color)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes) +
  scale_color_manual(name=&quot;P-Value&quot;,
                    values=c(&quot;black&quot;,&quot;red&quot;)) +
  geom_hline(yintercept=0.05, color=&quot;black&quot;)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-10">
Past versions of unnamed-chunk-14-10.png
</button>
</p>
<div id="fig-unnamed-chunk-14-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-10.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=GeneRatio)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-11">
Past versions of unnamed-chunk-14-11.png
</button>
</p>
<div id="fig-unnamed-chunk-14-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-11.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=FDR.Cutoff, y=DE.Interest)) +
  geom_point() +
  facet_grid(Source.GenesOfInterest ~ Source.DEGenes)</code></pre>
<p><img src="figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-12">
Past versions of unnamed-chunk-14-12.png
</button>
</p>
<div id="fig-unnamed-chunk-14-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/figure/10x-plot-diffexp-thresh.Rmd/unnamed-chunk-14-12.png" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>Best Pairwise DE FDR Cutoffs</p>
<ul>
<li>Stages of Differentiation: 0.01</li>
<li>Stages of Osteogenesis: 0.05 or 0.20 or 0.30 &gt; 0.10 &gt; 0.01</li>
<li>Stages of Osteogenesis (Time 2 only): 0.01 or 0.05 or 0.20 &gt; 0.10 or 0.50</li>
<li>OVERALL: 0.01 (0.05 is slightly better for osteogenesis)</li>
<li>(0.001 not included because it reduces DE gene count substantially)</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ggsignif_0.5.0              VennDiagram_1.6.20         
 [3] futile.logger_1.4.3         viridis_0.5.1              
 [5] viridisLite_0.3.0           enrichplot_1.4.0           
 [7] Rgraphviz_2.28.0            graph_1.62.0               
 [9] org.Hs.eg.db_3.8.2          AnnotationDbi_1.46.0       
[11] clusterProfiler_3.12.0      UpSetR_1.4.0               
[13] purrr_0.3.4                 scran_1.12.1               
[15] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
[17] DelayedArray_0.10.0         BiocParallel_1.18.0        
[19] matrixStats_0.57.0          Biobase_2.44.0             
[21] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[23] IRanges_2.18.1              S4Vectors_0.22.1           
[25] BiocGenerics_0.30.0         edgeR_3.26.5               
[27] limma_3.40.6                RColorBrewer_1.1-2         
[29] colorspace_2.0-0            gbm_2.1.5                  
[31] ggplot2_3.3.2               stringr_1.4.0              
[33] stringi_1.4.3               dplyr_1.0.2                
[35] Seurat_3.1.3               

loaded via a namespace (and not attached):
  [1] reticulate_1.16          tidyselect_1.1.0        
  [3] RSQLite_2.1.1            htmlwidgets_1.5.2       
  [5] Rtsne_0.15               munsell_0.5.0           
  [7] codetools_0.2-16         ica_1.0-2               
  [9] statmod_1.4.32           future_1.14.0           
 [11] withr_2.3.0              GOSemSim_2.10.0         
 [13] knitr_1.23               ROCR_1.0-7              
 [15] DOSE_3.10.2              gbRd_0.4-11             
 [17] listenv_0.7.0            labeling_0.4.2          
 [19] Rdpack_0.11-0            git2r_0.26.1            
 [21] urltools_1.7.3           GenomeInfoDbData_1.2.1  
 [23] polyclip_1.10-0          bit64_0.9-7             
 [25] farver_2.0.3             rprojroot_2.0.2         
 [27] vctrs_0.3.5              generics_0.1.0          
 [29] lambda.r_1.2.3           xfun_0.8                
 [31] R6_2.5.0                 ggbeeswarm_0.6.0        
 [33] rsvd_1.0.1               locfit_1.5-9.1          
 [35] gridGraphics_0.4-1       bitops_1.0-6            
 [37] fgsea_1.15.1             promises_1.1.1          
 [39] scales_1.1.1             ggraph_1.0.2            
 [41] beeswarm_0.2.3           gtable_0.3.0            
 [43] npsurv_0.4-0             globals_0.12.4          
 [45] workflowr_1.6.2          rlang_0.4.9             
 [47] splines_3.6.1            lazyeval_0.2.2          
 [49] europepmc_0.3            yaml_2.2.1              
 [51] reshape2_1.4.3           httpuv_1.5.1            
 [53] qvalue_2.16.0            tools_3.6.1             
 [55] ggplotify_0.0.3          ellipsis_0.3.1          
 [57] gplots_3.0.1.1           dynamicTreeCut_1.63-1   
 [59] ggridges_0.5.1           Rcpp_1.0.5              
 [61] plyr_1.8.4               progress_1.2.2          
 [63] zlibbioc_1.30.0          RCurl_1.98-1.1          
 [65] prettyunits_1.1.1        pbapply_1.4-0           
 [67] cowplot_1.1.0            zoo_1.8-6               
 [69] ggrepel_0.8.2            cluster_2.1.0           
 [71] fs_1.3.1                 magrittr_2.0.1          
 [73] futile.options_1.0.1     data.table_1.13.2       
 [75] DO.db_2.9                lmtest_0.9-37           
 [77] triebeard_0.3.0          RANN_2.6.1              
 [79] whisker_0.3-2            fitdistrplus_1.0-14     
 [81] hms_0.5.3                lsei_1.2-0              
 [83] evaluate_0.14            gridExtra_2.3           
 [85] compiler_3.6.1           scater_1.12.2           
 [87] tibble_3.0.4             KernSmooth_2.23-15      
 [89] crayon_1.3.4             htmltools_0.5.0         
 [91] later_1.1.0.1            tidyr_1.1.2             
 [93] RcppParallel_5.0.2       DBI_1.1.0               
 [95] formatR_1.7              tweenr_1.0.1            
 [97] MASS_7.3-51.4            rappdirs_0.3.1          
 [99] Matrix_1.2-18            gdata_2.18.0            
[101] metap_1.1                igraph_1.2.4.1          
[103] pkgconfig_2.0.3          rvcheck_0.1.3           
[105] plotly_4.9.2.1           xml2_1.3.2              
[107] vipor_0.4.5              dqrng_0.2.1             
[109] XVector_0.24.0           bibtex_0.4.2            
[111] digest_0.6.27            sctransform_0.2.0       
[113] RcppAnnoy_0.0.12         tsne_0.1-3              
[115] rmarkdown_1.13           leiden_0.3.1            
[117] fastmatch_1.1-0          uwot_0.1.5              
[119] DelayedMatrixStats_1.6.1 gtools_3.8.1            
[121] lifecycle_0.2.0          nlme_3.1-140            
[123] jsonlite_1.7.1           BiocNeighbors_1.2.0     
[125] pillar_1.4.7             lattice_0.20-38         
[127] httr_1.4.2               survival_2.44-1.1       
[129] GO.db_3.8.2              glue_1.4.2              
[131] png_0.1-7                bit_1.1-14              
[133] ggforce_0.2.2            blob_1.2.0              
[135] BiocSingular_1.0.0       caTools_1.17.1.2        
[137] memoise_1.1.0            irlba_2.3.3             
[139] future.apply_1.3.0       ape_5.4-1               </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
