<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-view-data-metrics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-view-data-metrics</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">May 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtreeb242b7302e9ad2aafb5015b35d4496659dcb67fctargetblankb242b73a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/b242b7302e9ad2aafb5015b35d4496659dcb67fc" target="_blank">b242b73</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtreeb242b7302e9ad2aafb5015b35d4496659dcb67fctargetblankb242b73a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/b242b7302e9ad2aafb5015b35d4496659dcb67fc" target="_blank">b242b73</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/10x-plot-enrichment-gom.Rmd
    Untracked:  analysis/TODO_10x-data-4-classification-t2.Rmd
    Untracked:  analysis/TODO_10x-data-4-classification.Rmd
    Untracked:  analysis/TODO_10x-data-4-reclassification-tot.Rmd
    Untracked:  analysis/TODO_10x-data-5-diffexp.Rmd
    Untracked:  analysis/TODO_10x-data-6-conservexp.Rmd
    Untracked:  analysis/TODO_10x-data-6-cormotif-t2.Rmd
    Untracked:  analysis/TODO_10x-data-6-cormotif-tot.Rmd
    Untracked:  analysis/TODO_10x-data-6-variance-t2.Rmd
    Untracked:  analysis/TODO_10x-data-6-variance-tot.Rmd
    Untracked:  analysis/TODO_10x-data-7-enrichment.Rmd
    Untracked:  analysis/TODO_10x-data-8-batch-correlation.Rmd
    Untracked:  analysis/TODO_10x-plot-diffexp.Rmd
    Untracked:  analysis/TODO_10x-plot-replicate-correlation-t2.Rmd
    Untracked:  analysis/TODO_10x-plot-utf1-tac3.Rmd
    Untracked:  analysis/TODO_10x-view-batch-effects.Rmd
    Untracked:  analysis/TODO_10x-view-species-assignments.Rmd
    Untracked:  analysis/TODO_10x-view-validation-data.Rmd
    Untracked:  analysis/v2/
    Untracked:  code/FindMarkers.R
    Untracked:  code/FindMarkers.sh
    Untracked:  code/FindMarkers_0.1.err
    Untracked:  code/FindMarkers_0.1.out
    Untracked:  code/FindMarkers_0.25.err
    Untracked:  code/FindMarkers_0.25.out
    Untracked:  code/FindMarkers_0.5.err
    Untracked:  code/FindMarkers_0.5.out
    Untracked:  code/FindMarkers_0.75.err
    Untracked:  code/FindMarkers_0.75.out
    Untracked:  code/FindMarkers_1.err
    Untracked:  code/FindMarkers_1.out
    Untracked:  code/GoM.R
    Untracked:  code/GoM.sh
    Untracked:  code/fastTopics.R
    Untracked:  code/fastTopics.sh
    Untracked:  code/fastTopicsBC.R
    Untracked:  code/fastTopicsBC.sh
    Untracked:  code/fastTopicsBC_3_tot.err
    Untracked:  code/fastTopicsBC_3_tot.out
    Untracked:  code/fastTopicsBC_4_tot.err
    Untracked:  code/fastTopicsBC_4_tot.out
    Untracked:  code/fastTopicsBC_5_tot.err
    Untracked:  code/fastTopicsBC_5_tot.out
    Untracked:  code/fastTopicsBC_6_tot.err
    Untracked:  code/fastTopicsBC_6_tot.out
    Untracked:  code/fastTopicsBC_7_tot.err
    Untracked:  code/fastTopicsBC_7_tot.out
    Untracked:  code/fastTopicsSpp.R
    Untracked:  code/fastTopicsSpp.sh
    Untracked:  code/fastTopicsSppBC.R
    Untracked:  code/fastTopicsSppBC.sh
    Untracked:  code/fastTopicsSppBC_3_tot.err
    Untracked:  code/fastTopicsSppBC_3_tot.out
    Untracked:  code/fastTopicsSppBC_4_tot.err
    Untracked:  code/fastTopicsSppBC_4_tot.out
    Untracked:  code/fastTopicsSppBC_5_tot.err
    Untracked:  code/fastTopicsSppBC_5_tot.out
    Untracked:  code/fastTopicsSppBC_6_tot.err
    Untracked:  code/fastTopicsSppBC_6_tot.out
    Untracked:  code/fastTopicsSppBC_7_tot.err
    Untracked:  code/fastTopicsSppBC_7_tot.out
    Untracked:  code/fastTopicsSpp_3_tot.err
    Untracked:  code/fastTopicsSpp_3_tot.out
    Untracked:  code/fastTopicsSpp_4_tot.err
    Untracked:  code/fastTopicsSpp_4_tot.out
    Untracked:  code/fastTopicsSpp_5_tot.err
    Untracked:  code/fastTopicsSpp_5_tot.out
    Untracked:  code/fastTopicsSpp_6_tot.err
    Untracked:  code/fastTopicsSpp_6_tot.out
    Untracked:  code/fastTopicsSpp_7_tot.err
    Untracked:  code/fastTopicsSpp_7_tot.out
    Untracked:  code/fastTopics_3_tot.err
    Untracked:  code/fastTopics_3_tot.out
    Untracked:  code/fastTopics_4_tot.err
    Untracked:  code/fastTopics_4_tot.out
    Untracked:  code/fastTopics_5_tot.err
    Untracked:  code/fastTopics_5_tot.out
    Untracked:  code/fastTopics_6_tot.err
    Untracked:  code/fastTopics_6_tot.out
    Untracked:  code/fastTopics_7_tot.err
    Untracked:  code/fastTopics_7_tot.out
    Untracked:  code/v2/
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/alizarin-red-quant-total.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/
    Untracked:  data/facs-MSC-collection-summary-sample-avg.csv
    Untracked:  data/facs-MSC-collection-summary-species-avg.csv
    Untracked:  data/facs-MSC-collection-summary-total.csv
    Untracked:  data/gom-data/
    Untracked:  data/oil-red-quant-total.csv
    Untracked:  data/scrna-batch-reformat.csv
    Untracked:  data/scrna-batch.csv
    Untracked:  output/data-qc-post-subsets-batchcor.pdf
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchcor.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterC.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchcor.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL.pdf
    Untracked:  output/data-qc-prefilter.pdf
    Untracked:  output/markers0.05-I.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05-M.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05-O.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.05.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.1.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.25.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.5.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markers1.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-I.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-M.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/markersStage-O.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  output/v1/
    Untracked:  output/v2/

Unstaged changes:
    Deleted:    analysis/10x-data-4-classification-t2.Rmd
    Deleted:    analysis/10x-data-4-classification.Rmd
    Deleted:    analysis/10x-data-5-diffexp.Rmd
    Deleted:    analysis/10x-data-6-conservexp.Rmd
    Deleted:    analysis/10x-data-6-cormotif-t2.Rmd
    Deleted:    analysis/10x-data-6-cormotif-tot.Rmd
    Deleted:    analysis/10x-data-6-variance-t2.Rmd
    Deleted:    analysis/10x-data-6-variance-tot.Rmd
    Deleted:    analysis/10x-data-7-enrichment.Rmd
    Deleted:    analysis/10x-data-8-batch-correlation.Rmd
    Deleted:    analysis/10x-plot-diffexp.Rmd
    Deleted:    analysis/10x-plot-replicate-correlation-t2.Rmd
    Deleted:    analysis/10x-plot-utf1-tac3.Rmd
    Deleted:    analysis/10x-view-batch-effects.Rmd
    Modified:   analysis/10x-view-data-metrics.Rmd
    Deleted:    analysis/10x-view-species-assignments.Rmd
    Deleted:    analysis/10x-view-validation-data.Rmd
    Deleted:    analysis/asbmr2020-plots.Rmd
    Deleted:    analysis/ashg2020-plots.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/temp-code.Rmd
    Deleted:    analysis/v1/index.Rmd
    Deleted:    analysis/v1/license.Rmd
    Modified:   code/DEdream.R
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml2.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml3.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml4.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep1.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_FALSE.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.keepRep2.filter.pch20.fml5.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.pseudo.nofilter.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml1.nofc.bh_stage.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_adhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_adhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_cluster.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_cluster.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_ostadhoc.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_ostadhoc.out
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_stage.err
    Deleted:    code/DEdream_totcell.subNum.filter.pch20.fml5.nofc.bh_stage.out
    Modified:   code/DEdreamsubset.R
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.keepRep1.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.keepRep2.filter.pch20.fc.bh_stage.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_adhoc.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_adhoc.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_cluster.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_cluster.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_ostadhoc.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_ostadhoc.out
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_stage.err
    Deleted:    code/DEedgeR_totcell.subNum.filter.pch20.fc.bh_stage.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/10x-view-data-metrics.Rmd</code>) and HTML (<code>docs/10x-view-data-metrics.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/f918f2b3e5531a109d05b2b42ffc8af09ab4fed5/analysis/10x-view-data-metrics.Rmd" target="_blank">f918f2b</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-08-12
</td>
<td>
Starting ortho-exon v2 reanalysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/6e47142cc90e74a675056855982ae89bce514004/analysis/10x-view-data-metrics.Rmd" target="_blank">6e47142</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-05-20
</td>
<td>
newfiles
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="view-10x-data-cellranger-metrics" class="section level1">
<h1>View 10X Data Cellranger Metrics</h1>
<p>Data being examined comes from 6 humans and 6 chimpanzees.</p>
<ul>
<li>GHO-1: H1-I + C1-I (iPSCs from H23555 and C8861)</li>
<li>GHO-2: H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)</li>
<li>GHO-4: H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)</li>
<li>HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)</li>
<li>HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)</li>
<li>HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)</li>
<li>HOU-1: H2-I + C2-I (iPSCs from H20157 and C3647)</li>
<li>HOU-2: H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)</li>
<li>HOU-4: H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)</li>
<li>HOU-9: H3-I + C3-I (iPSCs from H28126 and C3649)</li>
<li>HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)</li>
<li>HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)</li>
<li>HOU-5: H4-I + C4-I (iPSCs from H28834 and C40210)</li>
<li>HOU-6: H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)</li>
<li>HOU-8: H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)</li>
<li>HOU-17: H5-I + C5-I (iPSCs from H21792 and C40280)</li>
<li>HOU-19: H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)</li>
<li>HOU-21: H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)</li>
<li>HOU-18: H6-I + C6-I (iPSCs from H20961 and C3624)</li>
<li>HOU-20: H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)</li>
<li>HOU-22: H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)</li>
</ul>
<p>Collections are classified by the stage of differentiation at which cells were collected:</p>
<ul>
<li>Time 0 = iPSCs</li>
<li>Time 1 = iPSC-derived MSCs</li>
<li>Time 2 = iPSC-MSC-derived osteoblasts</li>
</ul>
<pre class="r"><code>library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)</code></pre>
<pre class="r"><code>#Load batch info
batch &lt;- read.csv(file=&#39;./data/scrna-batch.csv&#39;, header=TRUE, sep=&quot;,&quot;)

#Define methods of alignment
align &lt;- c(&quot;HumanOrthoV2&quot;,&quot;ChimpOrthoV2&quot;)

#Load metrics data from full data
metrics &lt;- data.frame()
for (j in align) {
  i &lt;- 1
  while (i &lt;= length(batch$Sample_Name_at_Core)) {
    print(i)
    #Define directory containing data from one collection
    dir_data &lt;- paste0(&#39;./../scRNA/cellranger-data-full/cellranger-data-&#39;,j,&#39;/&#39;,batch$Sample_Name_at_Core[i],&#39;/outs&#39;)
    #Load summary metrics
    temp &lt;- read.csv(file=paste0(dir_data,&#39;/metrics_summary.csv&#39;))
    #Process metrics
    temp[,c(1:4,19:20)] &lt;- lapply(temp[,c(1:4,19:20)], function(x) as.numeric(gsub(&quot;,&quot;,&quot;&quot;,as.character(x))))
    temp[,c(5:18)] &lt;- lapply(temp[,c(5:18)], function(x) as.numeric(gsub(&quot;%&quot;,&quot;&quot;,as.character(x)))/100)
    temp$Collection.Name &lt;- batch$Sample_Name_at_Core[i]
    temp$Alignment.Type &lt;- j
    #Add data to dataframe
    metrics &lt;- rbind(metrics,temp)
    i &lt;- i+1
  }
}
rm(dir_data,temp,i,j)

#Fix metrics details so that plotting is easier
metrics$Collection.Name &lt;- factor(metrics$Collection.Name,
                                  levels=c(&quot;GHO-1&quot;,&quot;GHO-2&quot;,&quot;GHO-4&quot;,
                                           &quot;HOU-10&quot;,&quot;HOU-12&quot;,&quot;HOU-16&quot;,
                                           &quot;HOU-1&quot;,&quot;HOU-2&quot;,&quot;HOU-4&quot;,
                                           &quot;HOU-9&quot;,&quot;HOU-11&quot;,&quot;HOU-15&quot;,
                                           &quot;HOU-5&quot;,&quot;HOU-6&quot;,&quot;HOU-8&quot;,
                                           &quot;HOU-17&quot;,&quot;HOU-19&quot;,&quot;HOU-21&quot;,
                                           &quot;HOU-18&quot;,&quot;HOU-20&quot;,&quot;HOU-22&quot;))
metrics$reads_prop &lt;- metrics$Number.of.Reads / metrics$Number.of.Reads
metrics$reads_mapped_to_genome &lt;- metrics$Number.of.Reads * metrics$Reads.Mapped.to.Genome
metrics$reads_mapped_confidently_to_genome &lt;- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Genome
metrics$reads_mapped_confidently_to_exonic_regions &lt;- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Exonic.Regions
metrics$reads_mapped_confidently_to_transcriptome &lt;- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Transcriptome</code></pre>
<div id="visualize-read-alignment-metrics" class="section level2">
<h2>Visualize read alignment metrics:</h2>
<pre class="r"><code>#total number of reads
ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Number.of.Reads)) +
  geom_col(position=&quot;dodge&quot;) +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title=&quot;Number of Reads Sequenced&quot;,y=&quot;Number of Reads&quot;,x=&quot;Collection&quot;) +
  ylim(0,300000000)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#percent sequence saturation
ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Sequencing.Saturation)) +
  geom_col(position=&quot;dodge&quot;) +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title=&quot;Sequencing Saturation&quot;,y=&quot;Sequencing Saturation&quot;,x=&quot;Collection&quot;) +
  ylim(0,1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#mean reads per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Reads per Cell (hg38)&quot;,y=&quot;Mean Reads per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,110000)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Reads per Cell (panTro6)&quot;,y=&quot;Mean Reads per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,110000))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#median genes per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Genes per Cell (ortho-exon hg38)&quot;,y=&quot;Median Genes per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,6500)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Genes per Cell (ortho-exon panTro6)&quot;,y=&quot;Median Genes per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,6500))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#median UMI per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;UMIs per Cell (ortho-exon hg38)&quot;,y=&quot;Median UMI Counts per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,31000)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;UMIs per Cell (ortho-exon panTro6)&quot;,y=&quot;Median UMI Counts per Cell&quot;,x=&quot;Collection&quot;) +
                           ylim(0,31000))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#proportion of Q30 bases in barcode, rna read, and umi
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: Barcode (ortho-exon hg38)&quot;,y=&quot;Q30 Bases in Barcode&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: UMI (ortho-exon hg38)&quot;,y=&quot;Q30 Bases in UMI&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: RNA (ortho-exon hg38)&quot;,y=&quot;Q30 Bases in RNA Read&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: Barcode (ortho-exon panTro6)&quot;,y=&quot;Q30 Bases in Barcode&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: UMI (ortho-exon panTro6)&quot;,y=&quot;Q30 Bases in UMI&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion Q30: RNA (ortho-exon panTro6)&quot;,y=&quot;Q30 Bases in RNA Read&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1))), ncol=3)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;)) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position=&quot;dodge&quot;, alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position=&quot;dodge&quot;, alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position=&quot;dodge&quot;, alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position=&quot;dodge&quot;, alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position=&quot;dodge&quot;, alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion of Mapped Reads (ortho-exon hg38)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;)) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position=&quot;dodge&quot;, alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position=&quot;dodge&quot;, alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position=&quot;dodge&quot;, alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position=&quot;dodge&quot;, alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position=&quot;dodge&quot;, alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Proportion of Mapped Reads (ortho-exon panTro6)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;)) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position=&quot;dodge&quot;, alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Numbers of Mapped Reads (ortho-exon hg38)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
                           ylim(0,300000000)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;)) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position=&quot;dodge&quot;, alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Numbers of Mapped Reads (ortho-exon panTro6)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
                           ylim(0,300000000))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#reads mapped to intergenic regions, intronic regions, and antisense to gene
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Reads Mapped to Intergenic Regions (ortho-exon hg38)&quot;,
                                y=&quot;Reads Mapped to Intergenic Regions&quot;,
                                x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position=&quot;dodge&quot;) +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title=&quot;Reads Mapped to Intronic Regions (ortho-exon hg38)&quot;,
                                 y=&quot;Reads Mapped to Intronic Regions&quot;,
                                 x=&quot;Collection&quot;) +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position=&quot;dodge&quot;) +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title=&quot;Reads Mapped to Intergenic Regions (ortho-exon hg38)&quot;,
                                 y=&quot;Reads Mapped to Intergenic Regions&quot;,
                                 x=&quot;Collection&quot;) +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                            geom_col(position=&quot;dodge&quot;) +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title=&quot;Reads Mapped to Intergenic Regions (ortho-exon panTro6)&quot;,
                                 y=&quot;Reads Mapped to Intergenic Regions&quot;,
                                 x=&quot;Collection&quot;) +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position=&quot;dodge&quot;) +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title=&quot;Reads Mapped to Intronic Regions (ortho-exon panTro6)&quot;,
                                 y=&quot;Reads Mapped to Intronic Regions&quot;,
                                 x=&quot;Collection&quot;) +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position=&quot;dodge&quot;) +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title=&quot;Reads Mapped to Intergenic Regions (ortho-exon panTro6)&quot;,
                                 y=&quot;Reads Mapped to Intergenic Regions&quot;,
                                 x=&quot;Collection&quot;) +
                            ylim(0,1))), ncol=3)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#fraction of reads in cells
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Fraction of Reads in Cells (ortho-exon hg38)&quot;,y=&quot;Fraction of Reads in Cells&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position=&quot;dodge&quot;) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title=&quot;Fraction of Reads in Cells (ortho-exon panTro6)&quot;,y=&quot;Fraction of Reads in Cells&quot;,x=&quot;Collection&quot;) +
                           ylim(0,1))), ncol=1)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Compare alignment proportions
mean(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;)$Reads.Mapped.Confidently.to.Transcriptome) #65.17%</code></pre>
<pre><code>[1] 0.6517143</code></pre>
<pre class="r"><code>mean(subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;)$Reads.Mapped.Confidently.to.Transcriptome) #63.41%</code></pre>
<pre><code>[1] 0.6341429</code></pre>
<pre class="r"><code>#Summarize metrics in plots for figures
metrics.h &lt;- subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;)
metrics.h$Collection &lt;- c(&quot;H1-I/C1-I&quot;,&quot;H1-M/C1-M&quot;,&quot;H1-O/C1-O&quot;,
                          &quot;H1-r2-I/C1-r2-I&quot;,&quot;H1-r2-M/C1-r2-M&quot;,&quot;H1-r2-O/C1-r2-O&quot;,
                          &quot;H2-I/C2-I&quot;,&quot;H2-M/C2-M&quot;,&quot;H2-O/C2-O&quot;,
                          &quot;H3-I/C3-I&quot;,&quot;H3-M/C3-M&quot;,&quot;H3-O/C3-O&quot;,
                          &quot;H4-I/C4-I&quot;,&quot;H4-M/C4-M&quot;,&quot;H4-O/C4-O&quot;,
                          &quot;H5-I/C5-I&quot;,&quot;H5-M/C5-M&quot;,&quot;H5-O/C5-O&quot;,
                          &quot;H6-I/C6-I&quot;,&quot;H6-M/C6-M&quot;,&quot;H6-O/C6-O&quot;)
metrics.h$Collection &lt;- factor(metrics.h$Collection,
                               levels=c(&quot;H1-I/C1-I&quot;,&quot;H1-M/C1-M&quot;,&quot;H1-O/C1-O&quot;,
                                        &quot;H1-r2-I/C1-r2-I&quot;,&quot;H1-r2-M/C1-r2-M&quot;,&quot;H1-r2-O/C1-r2-O&quot;,
                                        &quot;H2-I/C2-I&quot;,&quot;H2-M/C2-M&quot;,&quot;H2-O/C2-O&quot;,
                                        &quot;H3-I/C3-I&quot;,&quot;H3-M/C3-M&quot;,&quot;H3-O/C3-O&quot;,
                                        &quot;H4-I/C4-I&quot;,&quot;H4-M/C4-M&quot;,&quot;H4-O/C4-O&quot;,
                                        &quot;H5-I/C5-I&quot;,&quot;H5-M/C5-M&quot;,&quot;H5-O/C5-O&quot;,
                                        &quot;H6-I/C6-I&quot;,&quot;H6-M/C6-M&quot;,&quot;H6-O/C6-O&quot;))
metrics.c &lt;- subset(metrics,Alignment.Type==&#39;ChimpOrthoV2&#39;)
metrics.c$Collection &lt;- c(&quot;H1-I/C1-I&quot;,&quot;H1-M/C1-M&quot;,&quot;H1-O/C1-O&quot;,
                          &quot;H1-r2-I/C1-r2-I&quot;,&quot;H1-r2-M/C1-r2-M&quot;,&quot;H1-r2-O/C1-r2-O&quot;,
                          &quot;H2-I/C2-I&quot;,&quot;H2-M/C2-M&quot;,&quot;H2-O/C2-O&quot;,
                          &quot;H3-I/C3-I&quot;,&quot;H3-M/C3-M&quot;,&quot;H3-O/C3-O&quot;,
                          &quot;H4-I/C4-I&quot;,&quot;H4-M/C4-M&quot;,&quot;H4-O/C4-O&quot;,
                          &quot;H5-I/C5-I&quot;,&quot;H5-M/C5-M&quot;,&quot;H5-O/C5-O&quot;,
                          &quot;H6-I/C6-I&quot;,&quot;H6-M/C6-M&quot;,&quot;H6-O/C6-O&quot;)
metrics.c$Collection &lt;- factor(metrics.c$Collection,
                               levels=c(&quot;H1-I/C1-I&quot;,&quot;H1-M/C1-M&quot;,&quot;H1-O/C1-O&quot;,
                                        &quot;H1-r2-I/C1-r2-I&quot;,&quot;H1-r2-M/C1-r2-M&quot;,&quot;H1-r2-O/C1-r2-O&quot;,
                                        &quot;H2-I/C2-I&quot;,&quot;H2-M/C2-M&quot;,&quot;H2-O/C2-O&quot;,
                                        &quot;H3-I/C3-I&quot;,&quot;H3-M/C3-M&quot;,&quot;H3-O/C3-O&quot;,
                                        &quot;H4-I/C4-I&quot;,&quot;H4-M/C4-M&quot;,&quot;H4-O/C4-O&quot;,
                                        &quot;H5-I/C5-I&quot;,&quot;H5-M/C5-M&quot;,&quot;H5-O/C5-O&quot;,
                                        &quot;H6-I/C6-I&quot;,&quot;H6-M/C6-M&quot;,&quot;H6-O/C6-O&quot;))</code></pre>
<pre class="r"><code>ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position=&quot;dodge&quot;, alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title=&quot;Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
  ylim(0,300000000) +
  scale_x_discrete(labels=c(&quot;GHO-1&quot;=&quot;H1+C1\nTime 0&quot;,
                            &quot;GHO-2&quot;=&quot;H1+C1\nTime 1&quot;,
                            &quot;GHO-4&quot;=&quot;H1+C1\nTime 2&quot;,
                            &quot;HOU-10&quot;=&quot;H1r2+C1r2\nTime 0&quot;,
                            &quot;HOU-12&quot;=&quot;H1r2+C1r2\nTime 1&quot;,
                            &quot;HOU-16&quot;=&quot;H1r2+C1r2\nTime 2&quot;,
                            &quot;HOU-1&quot;=&quot;H2+C2\nTime 0&quot;,
                            &quot;HOU-2&quot;=&quot;H2+C2\nTime 1&quot;,
                            &quot;HOU-4&quot;=&quot;H2+C2\nTime 2&quot;,
                            &quot;HOU-9&quot;=&quot;H3+C3\nTime 0&quot;,
                            &quot;HOU-11&quot;=&quot;H3+C3\nTime 1&quot;,
                            &quot;HOU-15&quot;=&quot;H3+C3\nTime 2&quot;,
                            &quot;HOU-5&quot;=&quot;H4+C4\nTime 0&quot;,
                            &quot;HOU-6&quot;=&quot;H4+C4\nTime 1&quot;,
                            &quot;HOU-8&quot;=&quot;H4+C4\nTime 2&quot;,
                            &quot;HOU-17&quot;=&quot;H5+C5\nTime 0&quot;,
                            &quot;HOU-19&quot;=&quot;H5+C5\nTime 1&quot;,
                            &quot;HOU-21&quot;=&quot;H5+C5\nTime 2&quot;,
                            &quot;HOU-18&quot;=&quot;H6+C6\nTime 0&quot;,
                            &quot;HOU-20&quot;=&quot;H6+C6\nTime 1&quot;,
                            &quot;HOU-22&quot;=&quot;H6+C6\nTime 2&quot;))</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(metrics.c) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position=&quot;dodge&quot;, alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title=&quot;Numbers of Mapped Reads (Genome: panTro6, Annotation: ortho-exon)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
  ylim(0,300000000) +
  scale_x_discrete(labels=c(&quot;GHO-1&quot;=&quot;H1+C1\nTime 0&quot;,
                            &quot;GHO-2&quot;=&quot;H1+C1\nTime 1&quot;,
                            &quot;GHO-4&quot;=&quot;H1+C1\nTime 2&quot;,
                            &quot;HOU-10&quot;=&quot;H1r2+C1r2\nTime 0&quot;,
                            &quot;HOU-12&quot;=&quot;H1r2+C1r2\nTime 1&quot;,
                            &quot;HOU-16&quot;=&quot;H1r2+C1r2\nTime 2&quot;,
                            &quot;HOU-1&quot;=&quot;H2+C2\nTime 0&quot;,
                            &quot;HOU-2&quot;=&quot;H2+C2\nTime 1&quot;,
                            &quot;HOU-4&quot;=&quot;H2+C2\nTime 2&quot;,
                            &quot;HOU-9&quot;=&quot;H3+C3\nTime 0&quot;,
                            &quot;HOU-11&quot;=&quot;H3+C3\nTime 1&quot;,
                            &quot;HOU-15&quot;=&quot;H3+C3\nTime 2&quot;,
                            &quot;HOU-5&quot;=&quot;H4+C4\nTime 0&quot;,
                            &quot;HOU-6&quot;=&quot;H4+C4\nTime 1&quot;,
                            &quot;HOU-8&quot;=&quot;H4+C4\nTime 2&quot;,
                            &quot;HOU-17&quot;=&quot;H5+C5\nTime 0&quot;,
                            &quot;HOU-19&quot;=&quot;H5+C5\nTime 1&quot;,
                            &quot;HOU-21&quot;=&quot;H5+C5\nTime 2&quot;,
                            &quot;HOU-18&quot;=&quot;H6+C6\nTime 0&quot;,
                            &quot;HOU-20&quot;=&quot;H6+C6\nTime 1&quot;,
                            &quot;HOU-22&quot;=&quot;H6+C6\nTime 2&quot;))</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title=&quot;Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;) +
  ylim(0,300000000) +
  scale_x_discrete(labels=c(&quot;GHO-1&quot;=&quot;H1+C1\nTime 0&quot;,
                            &quot;GHO-2&quot;=&quot;H1+C1\nTime 1&quot;,
                            &quot;GHO-4&quot;=&quot;H1+C1\nTime 2&quot;,
                            &quot;HOU-10&quot;=&quot;H1r2+C1r2\nTime 0&quot;,
                            &quot;HOU-12&quot;=&quot;H1r2+C1r2\nTime 1&quot;,
                            &quot;HOU-16&quot;=&quot;H1r2+C1r2\nTime 2&quot;,
                            &quot;HOU-1&quot;=&quot;H2+C2\nTime 0&quot;,
                            &quot;HOU-2&quot;=&quot;H2+C2\nTime 1&quot;,
                            &quot;HOU-4&quot;=&quot;H2+C2\nTime 2&quot;,
                            &quot;HOU-9&quot;=&quot;H3+C3\nTime 0&quot;,
                            &quot;HOU-11&quot;=&quot;H3+C3\nTime 1&quot;,
                            &quot;HOU-15&quot;=&quot;H3+C3\nTime 2&quot;,
                            &quot;HOU-5&quot;=&quot;H4+C4\nTime 0&quot;,
                            &quot;HOU-6&quot;=&quot;H4+C4\nTime 1&quot;,
                            &quot;HOU-8&quot;=&quot;H4+C4\nTime 2&quot;,
                            &quot;HOU-17&quot;=&quot;H5+C5\nTime 0&quot;,
                            &quot;HOU-19&quot;=&quot;H5+C5\nTime 1&quot;,
                            &quot;HOU-21&quot;=&quot;H5+C5\nTime 2&quot;,
                            &quot;HOU-18&quot;=&quot;H6+C6\nTime 0&quot;,
                            &quot;HOU-20&quot;=&quot;H6+C6\nTime 1&quot;,
                            &quot;HOU-22&quot;=&quot;H6+C6\nTime 2&quot;))</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>metrics$Stage &lt;- factor(rep(c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),7),
                        levels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;))

ggplot(subset(metrics,Alignment.Type==&#39;HumanOrthoV2&#39;)) +
                                geom_col(aes(x=Stage, y=Number.of.Reads), position=&quot;dodge&quot;, alpha=0.2) +
                                geom_col(aes(x=Stage, y=reads_mapped_to_genome), position=&quot;dodge&quot;, alpha=0.3) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_exonic_regions), position=&quot;dodge&quot;, alpha=0.6) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_transcriptome), position=&quot;dodge&quot;, alpha=1) +
                                theme(axis.text.x=element_text(size=6)) +
                                labs(title=&quot;Numbers of Mapped Reads (ortho-exon hg38)&quot;,y=&quot;Mapped Reads&quot;,x=&quot;Collection&quot;)</code></pre>
<p><img src="figure/10x-view-data-metrics.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<ul>
<li>Number of total reads sequenced varied slightly across collections</li>
<li>Number of reads that were retained (confidently mapped to the transcriptome) were similar across collections</li>
<li>Number of reads that were retained (confidently mapped to the transcriptome) were similar whether using hg38 or panTro6 reference</li>
<li>10X collections from Time 2 generally had higher sequencing saturation than from Time 0 or Time 1</li>
<li>Collection HOU-15 (H3+C3 Time 2) has higher reads/cell, UMIs/cell, and genes/per cell</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RColorBrewer_1.1-2 colorspace_2.0-0   ggplot2_3.3.2     
[4] stringr_1.4.0      stringi_1.4.3      dplyr_1.0.2       
[7] Seurat_3.1.3      

loaded via a namespace (and not attached):
 [1] tsne_0.1-3          nlme_3.1-140        bitops_1.0-6       
 [4] fs_1.3.1            RcppAnnoy_0.0.12    httr_1.4.2         
 [7] rprojroot_2.0.2     sctransform_0.2.0   tools_3.6.1        
[10] R6_2.5.0            irlba_2.3.3         KernSmooth_2.23-15 
[13] uwot_0.1.5          lazyeval_0.2.2      withr_2.3.0        
[16] npsurv_0.4-0        gridExtra_2.3       tidyselect_1.1.0   
[19] compiler_3.6.1      git2r_0.26.1        plotly_4.9.2.1     
[22] labeling_0.4.2      caTools_1.17.1.2    scales_1.1.1       
[25] lmtest_0.9-37       ggridges_0.5.1      pbapply_1.4-0      
[28] rappdirs_0.3.1      digest_0.6.27       rmarkdown_1.13     
[31] pkgconfig_2.0.3     htmltools_0.5.0     bibtex_0.4.2       
[34] htmlwidgets_1.5.2   rlang_0.4.9         farver_2.0.3       
[37] generics_0.1.0      zoo_1.8-6           jsonlite_1.7.1     
[40] ica_1.0-2           gtools_3.8.1        magrittr_2.0.1     
[43] Matrix_1.2-18       Rcpp_1.0.5          munsell_0.5.0      
[46] ape_5.4-1           reticulate_1.16     lifecycle_0.2.0    
[49] whisker_0.3-2       yaml_2.2.1          gbRd_0.4-11        
[52] MASS_7.3-51.4       gplots_3.0.1.1      Rtsne_0.15         
[55] plyr_1.8.4          grid_3.6.1          parallel_3.6.1     
[58] gdata_2.18.0        listenv_0.7.0       promises_1.1.1     
[61] ggrepel_0.8.2       crayon_1.3.4        lattice_0.20-38    
[64] cowplot_1.1.0       splines_3.6.1       knitr_1.23         
[67] pillar_1.4.7        igraph_1.2.4.1      reshape2_1.4.3     
[70] future.apply_1.3.0  codetools_0.2-16    leiden_0.3.1       
[73] glue_1.4.2          evaluate_0.14       lsei_1.2-0         
[76] metap_1.1           RcppParallel_5.0.2  data.table_1.13.2  
[79] png_0.1-7           vctrs_0.3.5         httpuv_1.5.1       
[82] Rdpack_0.11-0       gtable_0.3.0        RANN_2.6.1         
[85] purrr_0.3.4         tidyr_1.1.2         future_1.14.0      
[88] xfun_0.8            rsvd_1.0.1          later_1.1.0.1      
[91] survival_2.44-1.1   viridisLite_0.3.0   tibble_3.0.4       
[94] workflowr_1.6.2     cluster_2.1.0       globals_0.12.4     
[97] fitdistrplus_1.0-14 ellipsis_0.3.1      ROCR_1.0-7         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
