<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Genevieve Housman" />


<title>10x-plot-cormotif-cnts-v2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">10x-plot-cormotif-cnts-v2</h1>
<h4 class="author">Genevieve Housman</h4>
<h4 class="date">September 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190719code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190719)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190719code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190719)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree1c012e52cf0c550665750d98f5bce1a68e06adfftargetblank1c012e5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/1c012e52cf0c550665750d98f5bce1a68e06adff" target="_blank">1c012e5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree1c012e52cf0c550665750d98f5bce1a68e06adfftargetblank1c012e5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/1c012e52cf0c550665750d98f5bce1a68e06adff" target="_blank">1c012e5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/10x-data-6-cormotif-tot-cnts_v1.Rmd
    Untracked:  analysis/10x-data-6-cormotif-tot-cnts_v2.Rmd
    Untracked:  analysis/10x-plot-cormotif-cnts-thresh_v1.Rmd
    Untracked:  analysis/10x-plot-cormotif-cnts-thresh_v2.Rmd
    Untracked:  analysis/10x-plot-cormotif-cnts_v1.Rmd
    Untracked:  analysis/10x-plot-cormotif-cnts_v2.Rmd
    Untracked:  code/CormotifCounts_v1.R
    Untracked:  code/CormotifCounts_v2.R
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc6.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster6a.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.poem.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.oad.min.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oem.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oemm.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.T2.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t01.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t12.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t0.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t1.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t2.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage6.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

Unstaged changes:
    Deleted:    analysis/10x-data-6-cormotif-tot-cnts.Rmd
    Deleted:    analysis/10x-plot-cormotif-cnts-thresh.Rmd
    Deleted:    analysis/10x-plot-cormotif-cnts.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    code/CormotifCounts.R
    Modified:   code/DEdream.R
    Modified:   code/DEdream.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="x-data-cellranger-analysis---correlation-motifs" class="section level1">
<h1>10X Data Cellranger Analysis - Correlation Motifs</h1>
<p>View results of correlation motifs tests between humans and chimpanzees within different cell classifications.</p>
<p>Use cutoff of posterior probability &gt; 0.65 as this threshold optimizes expected gene set enrichments (see 10x-plot-cormotif-cnts-thresh_v2.Rmd).</p>
<pre class="r"><code>#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ComplexHeatmap)
library(Cormotif)</code></pre>
<pre class="r"><code>#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich &lt;- function(totGenes, deGenes, ontology) {
  ref.df &lt;- bitr(totGenes, fromType=&quot;SYMBOL&quot;, toType=c(&quot;ENSEMBL&quot;,&quot;ENTREZID&quot;), OrgDb=org.Hs.eg.db)
  gene.df &lt;- bitr(deGenes, fromType=&quot;SYMBOL&quot;, toType=c(&quot;ENSEMBL&quot;,&quot;ENTREZID&quot;), OrgDb=org.Hs.eg.db)
  ego &lt;- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = &#39;ENSEMBL&#39;,
                  ont           = ontology,
                  pAdjustMethod = &quot;BH&quot;,
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot &lt;- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split=&#39;ONTOLOGY&#39;) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale=&quot;free&quot;) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option=&quot;C&quot;) +
          theme_bw() +
          labs(title=title, x=&quot;Gene Ratio&quot;, y=&quot;&quot;))
}

#Define function to plot GO functional enrichment results (specific ontology only)
goEnPlot.spec &lt;- function(goTot, title, ontology) {
  print(ggplot(goTot[which(goTot$ONTOLOGY==ontology),], aes(x=GeneRatio, y=Description, color=p.adjust, size=Count)) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option=&quot;C&quot;) +
          theme_bw() +
          labs(title=title, x=&quot;Gene Ratio&quot;, y=&quot;&quot;))
}</code></pre>
<div id="load-external-gene-datasets" class="section level2">
<h2>Load External Gene Datasets</h2>
<p>Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.</p>
<p>Genes previously identified as human and chimpanzee differences:</p>
<ul>
<li>Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
<ul>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR &lt; 1% out of 10304 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Blake et al. 2020 Genome Research (Supplemental Table S4)
<ul>
<li>DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR &lt; 1% out of 12184 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>Note: only considered DE genes unique to each tissue type</li>
</ul></li>
<li>Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
<ul>
<li>DE genes in parental iPSC lines (n = 7655)</li>
<li>DE genes in hybrid iPSC lines (n = 6009)</li>
<li>DE genes in parental CNCC lines (n = 2223)</li>
<li>DE genes in hybrid CNCC lines (n = 3612)</li>
<li>Note: only considered DE genes unique to each cell type</li>
</ul></li>
<li>Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
<ul>
<li>DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)</li>
<li>AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)</li>
<li>(human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)</li>
<li>(chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)</li>
<li>Note: only DMRs that overlapped promotor regions of genes were included</li>
<li>Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels</li>
</ul></li>
</ul>
<p>General skeleton-related and skeletal phenotype-related genes:</p>
<ul>
<li>Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
<ul>
<li>hip geometry GWAS loci (n = 10)</li>
</ul></li>
<li>Styrkarsdottir et al. 2019 Nature Communications (Table 1)
<ul>
<li>DXA bone area GWAS loci (n = 21 associated with 20 genes)</li>
</ul></li>
<li>Yengo et al. 2018 Human Molecular Genetics
<ul>
<li>height GWAS loci (n = 3290) (data provided at <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files" class="uri">https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files</a>)</li>
<li>height GWAS loci with associated eQTLs (n = 610) (list not provided)</li>
<li>Note: only used unique gene symbols associated with height GWAS loci (n = 1500)</li>
</ul></li>
</ul>
<p>Genes associated with skeletal diseases (OA + OP):</p>
<ul>
<li>Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
<ul>
<li>genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)</li>
</ul></li>
<li>Morris et al. 2019 Nature Genetics (Supplemental Table 2)
<ul>
<li>OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)</li>
</ul></li>
</ul>
<p>Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:</p>
<ul>
<li>Gallego Romero et al. 2015 eLife (Supplementary File 3)
<ul>
<li>DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR &lt; 1% out of 12171 genes)</li>
<li>Note: only included genes with a known gene symbol (retreived from <a href="https://www.biotools.fr/human/ensembl_symbol_converter" class="uri">https://www.biotools.fr/human/ensembl_symbol_converter</a>)</li>
<li>REMOVED because subsequent studies (Blake et al. 2018 &amp; Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm &amp; CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear</li>
</ul></li>
<li>Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type</li>
</ul></li>
<li>Blake et al. 2020 Genome Research (Supplemental Table S4)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type</li>
</ul></li>
<li>Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
<ul>
<li>REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type</li>
</ul></li>
<li>Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
<ul>
<li>Skeleton-related genes (n = 1790)</li>
<li>Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)</li>
<li>REMOVED because it is unclear where these genes came from to interpreting biology is difficult</li>
</ul></li>
</ul>
<p>Some external gene datasets were considered but not examined:</p>
<ul>
<li>Ward et al. 2019 eLife
<ul>
<li>DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR &lt; 10% out of 11974 genes)</li>
</ul></li>
<li>Shibata et al. 2012 PLOS Genetics
<ul>
<li>increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Edsall et al. 2019 Genome Biology and Evolution
<ul>
<li>increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
<ul>
<li>increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)</li>
<li>increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Prescott et al. 2015 Cell (Supplemental)
<ul>
<li>top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)</li>
<li>top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)</li>
</ul></li>
<li>Joganic et al. 2017 American Journal of Physical Anthropology
<ul>
<li>genetics of craniofacial skeleton in baboons</li>
</ul></li>
<li>Cibrián Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
<ul>
<li>robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)</li>
</ul></li>
<li>Tachmazidou et al. 2019 Nature Genetics
<ul>
<li>OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])</li>
<li>OA loci (Supplementary Table 2)</li>
<li>established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)</li>
<li>Note: used loci compiled by AH</li>
<li>(OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)</li>
<li>(OA.estab = gene symbol(s) nearest to established OA loci)</li>
<li>(OA.broad = genes within 500kb of OA loci)</li>
</ul></li>
</ul>
<pre class="r"><code>#Define function for testing significant enrichment of gene sets

genesetEnrich &lt;- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative=&quot;greater&quot;))

}

genesetEnrich.data &lt;- function(InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       &lt;- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    &lt;- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    &lt;- length(DEgenes) - DE.Interest
  NonDE.NotInterest &lt;- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable &lt;- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) &lt;- c(&quot;DE Genes&quot;,&quot;Non DE Genes&quot;)
  rownames(ContingencyTable) &lt;- c(&quot;Genes of Interest&quot;,&quot;Not Genes of Interest&quot;)
  
  #perform test and output data
  data &lt;- data.frame(&quot;Source.DEGenes&quot;=InternalDataSource,
                     &quot;Source.GenesOfInterest&quot;=ExternalDataSource,
                     &quot;DE.Interest&quot;=DE.Interest,
                     &quot;NonDE.Interest&quot;=NonDE.Interest,
                     &quot;DE.NotInterest&quot;=DE.NotInterest,
                     &quot;NonDE.NotInterest&quot;=NonDE.NotInterest,
                     &quot;p.value&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$p.value,
                     &quot;odds.ratio&quot;=fisher.test(ContingencyTable, alternative=&quot;greater&quot;)$estimate,
                     row.names=NULL)
  return(data)

}</code></pre>
<pre class="r"><code>#Load external datasets (manually currated from papers)
gallegoromero2015    &lt;- read.csv(&quot;./data/external-gene-sets/GallegoRomero2015eLife.csv&quot;)
blake2018            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2018GenBio.csv&quot;)
blake2020            &lt;- read.csv(&quot;./data/external-gene-sets/Blake2020GenRes.csv&quot;)
gokhman2021          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2021NatGen.csv&quot;)
gokhman2020          &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm.csv&quot;)
gokhman2020.skeleton &lt;- read.csv(&quot;./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv&quot;)
hsu2019              &lt;- read.csv(&quot;./data/external-gene-sets/Hsu2019JBoneMinRes.csv&quot;)
styrkarsdottir2019   &lt;- read.csv(&quot;./data/external-gene-sets/Styrkarsdottir2019NatComm.csv&quot;)
yengo2018            &lt;- read.csv(&quot;./data/external-gene-sets/Yengo2018HumMolGen.csv&quot;)
steinberg2020        &lt;- read.csv(&quot;./data/external-gene-sets/Steinberg2020biorxiv.csv&quot;)
morris2019           &lt;- read.csv(&quot;./data/external-gene-sets/Morris2019NatGen.csv&quot;)
#cibrianuhalte2017    &lt;- read.csv(&quot;./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv&quot;)
#tachmazidou2019      &lt;- read.csv(&quot;./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv&quot;)</code></pre>
<pre class="r"><code>#genes previously identified as human and chimpanzee differences

#genesExternal &lt;- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                                blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==1 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                              blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==1 &amp;
                                              blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal &lt;- blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                      blake2018$DE.Day1==0 &amp;
                                                blake2018$DE.Day2==0 &amp;
                                              blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==1 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==1 &amp;
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal &lt;- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                      blake2020$DE.HvC_Kidney==0 &amp;
                                      blake2020$DE.HvC_Liver==0 &amp;
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                               gokhman2021$DE.Hybrid.CNCCs==0)]), #hybrid.iPSCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal &lt;- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal &lt;- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal &lt;- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal &lt;- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal &lt;- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal &lt;- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal &lt;- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal &lt;- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal &lt;- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal &lt;- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS</code></pre>
</div>
<div id="stages-of-differentiation-at-collection" class="section level2">
<h2>Stages of Differentiation at Collection</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Stage=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;),
                           col=list(Stage=c(&quot;Time 0&quot;=&quot;pink&quot;,&quot;Time 1&quot;=&quot;mediumorchid&quot;,&quot;Time 2&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>            Time.0     Time.1     Time.2
TSPAN6   0.9739985 0.22338930 0.17245593
DPM1     0.9149139 0.18103761 0.14232067
SCYL3    0.9805232 0.99609302 0.98895592
C1orf112 0.2346324 0.08926467 0.06555109
CFH      0.7258897 0.99864021 0.99997850
FUCA2    0.9609797 0.63640543 0.58140404</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         Time.0 Time.1 Time.2
TSPAN6        1      0      0
DPM1          1      0      0
SCYL3         1      1      1
C1orf112      0      0      0
CFH           1      1      1
FUCA2         1      0      0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;Time.0&quot;,length(which(dataUpset$Time.0==1))),
                                       rep(&quot;Time.1&quot;,length(which(dataUpset$Time.1==1))),
                                       rep(&quot;Time.2&quot;,length(which(dataUpset$Time.2==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$Time.0==1)],
                                rownames(dataUpset)[which(dataUpset$Time.1==1)],
                                rownames(dataUpset)[which(dataUpset$Time.2==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$Time.0==1),&quot;Time.0&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.1==1),&quot;Time.1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.2==1),&quot;Time.2&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;),
                               labels=c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;,
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
Time 0 Time 1 Time 2 
  6822   4523   4020 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;Time.0&quot;),
                         list(&quot;Time.1&quot;),
                         list(&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.1&quot;),
                         list(&quot;Time.1&quot;,&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.2&quot;),
                         list(&quot;Time.0&quot;,&quot;Time.1&quot;,&quot;Time.2&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Save DE gene lists
write.csv(rownames(dataUpset),&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                    dataUpset$Time.1==1 &amp;
                                    dataUpset$Time.2==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                    dataUpset$Time.1==1 &amp;
                                    dataUpset$Time.2==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t01.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                    dataUpset$Time.1==1 &amp;
                                    dataUpset$Time.2==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t12.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                    dataUpset$Time.1==0 &amp;
                                    dataUpset$Time.2==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t0.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                    dataUpset$Time.1==1 &amp;
                                    dataUpset$Time.2==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t1.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                    dataUpset$Time.1==0 &amp;
                                    dataUpset$Time.2==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t2.csv&quot;)</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across t0+t1+t2
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across t0+t1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Time 0 &amp; Time 1&quot;)

#Shared across t1+t2
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Time 1 &amp; Time 2&quot;)

#Unique to Time 0
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==0 &amp;
                                      dataUpset$Time.2==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Time 0&quot;)

#Unique to Time 1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Time 1&quot;)

#Unique to Time 2
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
#                                      dataUpset$Time.1==0 &amp;
#                                      dataUpset$Time.2==1)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Time 2&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 0 &amp; Time 1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 0 &amp; Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 1 &amp; Time 2
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Time 1 &amp; Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to Time 0
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==0 &amp;
                                      dataUpset$Time.2==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 0&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 2
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==0 &amp;
                                      dataUpset$Time.2==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #All to Time 0
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Time 0&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  #All to Time 1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.1==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Time 1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  #All to Time 2
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.2==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Time 2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  }

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in All&quot;,
               &quot;Shared in Time 0 &amp; Time 1&quot;,
               &quot;Shared in Time 1 &amp; Time 2&quot;,
               &quot;Unique to Time 0&quot;,
               &quot;Unique to Time 1&quot;,
               &quot;Unique to Time 2&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.004)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in All&quot;,
               &quot;Shared in Time 0 &amp; Time 1&quot;,
               &quot;Shared in Time 1 &amp; Time 2&quot;,
               &quot;Unique to Time 0&quot;,
               &quot;Unique to Time 1&quot;,
               &quot;Unique to Time 2&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.001)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Time 0&quot;,
               &quot;All Time 1&quot;,
               &quot;All Time 2&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Time 0&quot;,
               &quot;All Time 1&quot;,
               &quot;All Time 2&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>               Source.DEGenes                 Source.GenesOfInterest
76           Unique to Time 0   Gokhman2021-DE.Parental.iPSCs-Unique
85           Unique to Time 0     Gokhman2021-DE.Hybrid.iPSCs-Unique
34                 All Time 0 Blake2018-DE.iPSC-endoderm-day3-Unique
25                 All Time 0 Blake2018-DE.iPSC-endoderm-day2-Unique
106                All Time 0     Gokhman2021-DE.Hybrid.CNCCs-Unique
149          Unique to Time 1                  Steinberg2020-GWAS.OA
4            Unique to Time 0              Blake2018-DE.iPSCs-Unique
156 Shared in Time 1 &amp; Time 2                     Morris2020-GWAS.OP
36                 All Time 2 Blake2018-DE.iPSC-endoderm-day3-Unique
35                 All Time 1 Blake2018-DE.iPSC-endoderm-day3-Unique
49           Unique to Time 0             Blake2020-DE.kidney-Unique
16                 All Time 0 Blake2018-DE.iPSC-endoderm-day1-Unique
26                 All Time 1 Blake2018-DE.iPSC-endoderm-day2-Unique
139          Unique to Time 0                  Yengo2018-GWAS.height
73              Shared in All   Gokhman2021-DE.Parental.iPSCs-Unique
27                 All Time 2 Blake2018-DE.iPSC-endoderm-day2-Unique
    DE.Interest      p.value    GeneRatio
76         1245 5.551655e-16 2.594983e-03
85          771 2.755307e-12 1.607013e-03
34          378 7.616807e-07 7.878742e-04
25          145 5.532021e-04 3.022269e-04
106         484 6.470021e-04 1.008813e-03
149          13 1.730636e-03 2.709620e-05
4           116 1.929970e-03 2.417815e-04
156          24 2.789214e-03 5.002376e-05
36          532 4.709355e-03 1.108860e-03
35          499 5.516113e-03 1.040077e-03
49           50 5.840379e-03 1.042162e-04
16          125 1.075423e-02 2.605404e-04
26          193 1.203896e-02 4.022744e-04
139         299 1.375239e-02 6.232127e-04
73         1490 1.599783e-02 3.105642e-03
27          203 2.238809e-02 4.231176e-04</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
<p>Determine which skeletal-phenotype-related genes overlap with DE genes:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared in All Stages (tested in all classifications)
  print(&quot;Shared in All Stages, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Shared in Time 0 &amp; Time 1 (tested in all classifications)
  print(&quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared in Time 1 &amp; Time 2 (tested in all classifications)
  print(&quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 0 (tested in all classifications)
  print(&quot;Unique to Time 0 (tested in all classifications)&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==1 &amp;
                                      dataUpset$Time.1==0 &amp;
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 1 (tested in all classifications)
  print(&quot;Unique to Time 1 (tested in all classifications)&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==1 &amp;
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to Time 2 (tested in all classifications)
  print(&quot;Unique to Time 2 (tested in all classifications)&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Time.0==0 &amp;
                                      dataUpset$Time.1==0 &amp;
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

}</code></pre>
<pre><code>[1] &quot;Hsu2019-GWAS.hip.geometry&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
[1] &quot;PPP6R3&quot; &quot;GAL&quot;   
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
character(0)
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Styrkarsdottir2019-GWAS.bone.area&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
[1] &quot;BCKDHB&quot;  &quot;COL11A1&quot; &quot;SOX9&quot;    &quot;CTDSP2&quot; 
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
character(0)
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
[1] &quot;DYM&quot;    &quot;SH3GL3&quot; &quot;HHIP&quot;  
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
character(0)
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Yengo2018-GWAS.height&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
  [1] &quot;PLXND1&quot;     &quot;PTBP1&quot;      &quot;PHF20&quot;      &quot;IKZF2&quot;      &quot;IFT88&quot;     
  [6] &quot;GAB2&quot;       &quot;SH3YL1&quot;     &quot;PHLPP2&quot;     &quot;HDAC9&quot;      &quot;ADAMTS6&quot;   
 [11] &quot;DKK3&quot;       &quot;ARID4B&quot;     &quot;SDCCAG8&quot;    &quot;ATP9A&quot;      &quot;CUL1&quot;      
 [16] &quot;CYFIP2&quot;     &quot;USP36&quot;      &quot;COL11A1&quot;    &quot;MYO9A&quot;      &quot;PFKP&quot;      
 [21] &quot;PRKCZ&quot;      &quot;SRBD1&quot;      &quot;TGFBR3&quot;     &quot;ELP1&quot;       &quot;ARHGAP10&quot;  
 [26] &quot;ADGRL1&quot;     &quot;P4HA2&quot;      &quot;NSF&quot;        &quot;EED&quot;        &quot;FOSL2&quot;     
 [31] &quot;ICAM3&quot;      &quot;POLD3&quot;      &quot;CAPZB&quot;      &quot;FGFR1&quot;      &quot;FBLN1&quot;     
 [36] &quot;TULP3&quot;      &quot;PPP2R5C&quot;    &quot;DLGAP4&quot;     &quot;BCKDHB&quot;     &quot;NCOA1&quot;     
 [41] &quot;NOX4&quot;       &quot;PGS1&quot;       &quot;NID2&quot;       &quot;GNAS&quot;       &quot;RFX2&quot;      
 [46] &quot;ALG6&quot;       &quot;DOCK9&quot;      &quot;KIZ&quot;        &quot;ANAPC5&quot;     &quot;KIF16B&quot;    
 [51] &quot;FUS&quot;        &quot;P3H2&quot;       &quot;G2E3&quot;       &quot;TGFB2&quot;      &quot;EFHC1&quot;     
 [56] &quot;HIRA&quot;       &quot;PIK3IP1&quot;    &quot;HSCB&quot;       &quot;MYH9&quot;       &quot;KHNYN&quot;     
 [61] &quot;TRIP11&quot;     &quot;NFATC4&quot;     &quot;JAG1&quot;       &quot;PXMP4&quot;      &quot;CEP192&quot;    
 [66] &quot;CAB39L&quot;     &quot;CRISPLD2&quot;   &quot;BCKDK&quot;      &quot;AAGAB&quot;      &quot;TRPS1&quot;     
 [71] &quot;LIG1&quot;       &quot;PLEKHA4&quot;    &quot;CDK6&quot;       &quot;LFNG&quot;       &quot;GLCCI1&quot;    
 [76] &quot;AHR&quot;        &quot;GLI3&quot;       &quot;KANK1&quot;      &quot;PDLIM1&quot;     &quot;SUFU&quot;      
 [81] &quot;LHPP&quot;       &quot;SH3PXD2A&quot;   &quot;CCDC6&quot;      &quot;ZMIZ1&quot;      &quot;PBLD&quot;      
 [86] &quot;WNT3&quot;       &quot;UTP6&quot;       &quot;FAM20A&quot;     &quot;MTCH2&quot;      &quot;CCND1&quot;     
 [91] &quot;CHORDC1&quot;    &quot;SLC38A1&quot;    &quot;UST&quot;        &quot;FBXO9&quot;      &quot;GPLD1&quot;     
 [96] &quot;ADGRG6&quot;     &quot;TFEB&quot;       &quot;GMDS&quot;       &quot;FAF2&quot;       &quot;PDE4D&quot;     
[101] &quot;ERGIC1&quot;     &quot;STC2&quot;       &quot;TFG&quot;        &quot;SCAP&quot;       &quot;ACTR3&quot;     
[106] &quot;FN1&quot;        &quot;BIRC6&quot;      &quot;COX7A2L&quot;    &quot;THADA&quot;      &quot;DHCR24&quot;    
[111] &quot;MORN1&quot;      &quot;CEP104&quot;     &quot;ERRFI1&quot;     &quot;MFAP2&quot;      &quot;SLC2A1&quot;    
[116] &quot;STAG1&quot;      &quot;DDX59&quot;      &quot;FOXO3&quot;      &quot;NPC2&quot;       &quot;ATL2&quot;      
[121] &quot;BCL11A&quot;     &quot;WASHC3&quot;     &quot;ZRANB3&quot;     &quot;MTERF4&quot;     &quot;ITPR2&quot;     
[126] &quot;LRP1&quot;       &quot;METTL8&quot;     &quot;EXOSC9&quot;     &quot;NCOA5&quot;      &quot;CDKN1A&quot;    
[131] &quot;POLR1B&quot;     &quot;RRBP1&quot;      &quot;BMP2&quot;       &quot;HIVEP3&quot;     &quot;FGFRL1&quot;    
[136] &quot;LOXL1&quot;      &quot;TXNDC17&quot;    &quot;XPO7&quot;       &quot;PXDN&quot;       &quot;CLIP1&quot;     
[141] &quot;SYNE1&quot;      &quot;NFATC1&quot;     &quot;RARA&quot;       &quot;CHD1L&quot;      &quot;ZNF304&quot;    
[146] &quot;MATN3&quot;      &quot;RAP1GAP2&quot;   &quot;SERPINF1&quot;   &quot;ZSWIM3&quot;     &quot;IPO8&quot;      
[151] &quot;RRAS2&quot;      &quot;LOXL2&quot;      &quot;CTIF&quot;       &quot;CCNH&quot;       &quot;SOX5&quot;      
[156] &quot;CLOCK&quot;      &quot;GGACT&quot;      &quot;ARHGAP32&quot;   &quot;SRPK2&quot;      &quot;DYSF&quot;      
[161] &quot;USP37&quot;      &quot;ALDH1L2&quot;    &quot;FLNB&quot;       &quot;RCBTB1&quot;     &quot;PHF11&quot;     
[166] &quot;IGF2BP3&quot;    &quot;CALCOCO2&quot;   &quot;TRA2B&quot;      &quot;SCN2A&quot;      &quot;UGGT1&quot;     
[171] &quot;TXN&quot;        &quot;TLN1&quot;       &quot;ZSCAN9&quot;     &quot;YAP1&quot;       &quot;HAUS2&quot;     
[176] &quot;CENPO&quot;      &quot;KIF11&quot;      &quot;ARHGAP24&quot;   &quot;FAM13A&quot;     &quot;FGF2&quot;      
[181] &quot;SLC39A8&quot;    &quot;RGS3&quot;       &quot;TAPBPL&quot;     &quot;SLC7A1&quot;     &quot;RB1&quot;       
[186] &quot;PELI2&quot;      &quot;FANCI&quot;      &quot;PARN&quot;       &quot;TXNL4B&quot;     &quot;ZFHX3&quot;     
[191] &quot;UTP4&quot;       &quot;RANBP10&quot;    &quot;SSH2&quot;       &quot;SLC16A3&quot;    &quot;NFIC&quot;      
[196] &quot;URB1&quot;       &quot;SAE1&quot;       &quot;MYO1F&quot;      &quot;PIGK&quot;       &quot;MGST3&quot;     
[201] &quot;TUFT1&quot;      &quot;ADAMTSL4&quot;   &quot;MCL1&quot;       &quot;TTC13&quot;      &quot;CNIH3&quot;     
[206] &quot;VPS54&quot;      &quot;ASXL2&quot;      &quot;BOC&quot;        &quot;PIK3R1&quot;     &quot;PAM&quot;       
[211] &quot;ATG12&quot;      &quot;TRIM41&quot;     &quot;ABT1&quot;       &quot;DAAM2&quot;      &quot;CREB5&quot;     
[216] &quot;GSN&quot;        &quot;PAMR1&quot;      &quot;SERPINH1&quot;   &quot;MPP7&quot;       &quot;LATS2&quot;     
[221] &quot;VTI1A&quot;      &quot;ZNF827&quot;     &quot;FLI1&quot;       &quot;BEND6&quot;      &quot;PTPN14&quot;    
[226] &quot;CCDC50&quot;     &quot;ASAP1&quot;      &quot;CFDP1&quot;      &quot;JAZF1&quot;      &quot;DDAH1&quot;     
[231] &quot;ANKH&quot;       &quot;SORBS2&quot;     &quot;NUP205&quot;     &quot;MAP3K7CL&quot;   &quot;VPS8&quot;      
[236] &quot;SKI&quot;        &quot;PTPDC1&quot;     &quot;COLEC12&quot;    &quot;TSPAN33&quot;    &quot;HLCS&quot;      
[241] &quot;KALRN&quot;      &quot;TAOK1&quot;      &quot;SIK3&quot;       &quot;ADAR&quot;       &quot;ADCY9&quot;     
[246] &quot;ZYG11B&quot;     &quot;PLPP3&quot;      &quot;VANGL2&quot;     &quot;ACTG2&quot;      &quot;SCNM1&quot;     
[251] &quot;COL6A3&quot;     &quot;PCOLCE2&quot;    &quot;TOPBP1&quot;     &quot;SLBP&quot;       &quot;DLC1&quot;      
[256] &quot;ZCCHC24&quot;    &quot;NUDT5&quot;      &quot;QSOX2&quot;      &quot;ATP9B&quot;      &quot;XRRA1&quot;     
[261] &quot;TK2&quot;        &quot;STX3&quot;       &quot;WDR81&quot;      &quot;SPINDOC&quot;    &quot;UBTD2&quot;     
[266] &quot;ABHD15&quot;     &quot;ROR2&quot;       &quot;CRADD&quot;      &quot;TPST1&quot;      &quot;PDCD6IP&quot;   
[271] &quot;SMAD1&quot;      &quot;ELOVL6&quot;     &quot;CYTL1&quot;      &quot;NUDT6&quot;      &quot;FEZ2&quot;      
[276] &quot;CXXC5&quot;      &quot;GPHN&quot;       &quot;TLN2&quot;       &quot;SUCLG2&quot;     &quot;FGGY&quot;      
[281] &quot;AFF1&quot;       &quot;NBEA&quot;       &quot;SH3RF3&quot;     &quot;RELA&quot;       &quot;GOLIM4&quot;    
[286] &quot;XXYLT1&quot;     &quot;CTSF&quot;       &quot;ADCY6&quot;      &quot;MRPL11&quot;     &quot;RSRC1&quot;     
[291] &quot;FBXW8&quot;      &quot;PSMD2&quot;      &quot;PHYKPL&quot;     &quot;PPP2R2D&quot;    &quot;SLC38A9&quot;   
[296] &quot;ZNF518A&quot;    &quot;LCORL&quot;      &quot;PLEC&quot;       &quot;NT5DC1&quot;     &quot;ARID3B&quot;    
[301] &quot;ZNF784&quot;     &quot;CCDC66&quot;     &quot;PITPNB&quot;     &quot;POLR2A&quot;     &quot;AATK&quot;      
[306] &quot;TDRKH&quot;      &quot;IST1&quot;       &quot;RAD51B&quot;     &quot;PAPPA&quot;      &quot;CEP63&quot;     
[311] &quot;KMT5A&quot;      &quot;ST6GALNAC3&quot; &quot;NELFA&quot;      &quot;PRKG1&quot;      &quot;UBE2L3&quot;    
[316] &quot;THBS2&quot;      &quot;PPARA&quot;      &quot;IFT140&quot;     &quot;COL14A1&quot;    &quot;RNFT1&quot;     
[321] &quot;ZKSCAN3&quot;    &quot;ZNF471&quot;     &quot;AFAP1&quot;      &quot;SRC&quot;        &quot;SLC22A4&quot;   
[326] &quot;VEPH1&quot;      &quot;RAB40C&quot;     &quot;CDC42SE1&quot;   &quot;FAM114A1&quot;   &quot;MCMBP&quot;     
[331] &quot;MYO1C&quot;      &quot;GRK6&quot;       &quot;PRIM1&quot;      &quot;MAN1A2&quot;     &quot;HSD17B11&quot;  
[336] &quot;NCOA6&quot;      &quot;PAPSS2&quot;     &quot;PNP&quot;        &quot;GRK5&quot;       &quot;SGMS1&quot;     
[341] &quot;CRYZL1&quot;     &quot;NCKIPSD&quot;    &quot;PLEKHM1&quot;    &quot;C16orf95&quot;  
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
 [1] &quot;ICA1&quot;    &quot;ETV1&quot;    &quot;PIK3C2A&quot; &quot;TNC&quot;     &quot;CDK14&quot;   &quot;SMARCD3&quot; &quot;DIS3&quot;   
 [8] &quot;COL16A1&quot; &quot;PRTFDC1&quot; &quot;DIDO1&quot;   &quot;VWA8&quot;    &quot;CAV2&quot;    &quot;PLOD3&quot;   &quot;C5&quot;     
[15] &quot;PDE10A&quot;  &quot;DROSHA&quot;  &quot;MLLT1&quot;   &quot;SAP130&quot;  &quot;SEMA6D&quot;  &quot;KIF21A&quot;  &quot;SCAPER&quot; 
[22] &quot;CDH11&quot;   &quot;CTDSPL&quot;  &quot;INPP5E&quot;  &quot;CACUL1&quot;  &quot;CNKSR3&quot;  &quot;NPR2&quot;    &quot;NOL9&quot;   
[29] &quot;MEAF6&quot;   &quot;TEX264&quot;  &quot;CENPX&quot;   &quot;CENPP&quot;   &quot;ZNF585A&quot; &quot;ZGPAT&quot;   &quot;KANK2&quot;  
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
 [1] &quot;SBNO2&quot;    &quot;MARK3&quot;    &quot;REST&quot;     &quot;FAM234B&quot;  &quot;TNRC6A&quot;   &quot;NALCN&quot;   
 [7] &quot;FUBP3&quot;    &quot;KLF3&quot;     &quot;CBLB&quot;     &quot;RTN4&quot;     &quot;EFEMP1&quot;   &quot;MKLN1&quot;   
[13] &quot;NREP&quot;     &quot;TNS3&quot;     &quot;NMT1&quot;     &quot;CDK5RAP2&quot; &quot;TSTD2&quot;    &quot;TLE3&quot;    
[19] &quot;GAREM1&quot;   &quot;PTPRF&quot;    &quot;GNA12&quot;    &quot;TCF7L2&quot;   &quot;MBNL1&quot;    &quot;CMIP&quot;    
[25] &quot;ELMO1&quot;    &quot;WBP1L&quot;    &quot;NDUFV1&quot;   &quot;LUZP1&quot;    &quot;RHOD&quot;     &quot;PDIK1L&quot;  
[31] &quot;HMCES&quot;    &quot;PTCH1&quot;    &quot;SUPT3H&quot;  
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
  [1] &quot;RBM6&quot;       &quot;TSPAN9&quot;     &quot;MAP4K5&quot;     &quot;TTC27&quot;      &quot;PHLDB1&quot;    
  [6] &quot;PRDM11&quot;     &quot;SLC45A4&quot;    &quot;BTN3A1&quot;     &quot;TIMP2&quot;      &quot;SNX29&quot;     
 [11] &quot;RSF1&quot;       &quot;FAM120A&quot;    &quot;H6PD&quot;       &quot;LTBP1&quot;      &quot;HEXB&quot;      
 [16] &quot;PTPRN&quot;      &quot;PUM2&quot;       &quot;TRAF3IP2&quot;   &quot;ATG5&quot;       &quot;PARP12&quot;    
 [21] &quot;TNPO3&quot;      &quot;TBC1D1&quot;     &quot;ATXN3&quot;      &quot;NAV3&quot;       &quot;RORA&quot;      
 [26] &quot;NEDD4&quot;      &quot;PPP2R3A&quot;    &quot;ZNF532&quot;     &quot;FNDC3B&quot;     &quot;TMEM131&quot;   
 [31] &quot;SIRT6&quot;      &quot;SLC1A3&quot;     &quot;SESN1&quot;      &quot;CDC14B&quot;     &quot;CADPS2&quot;    
 [36] &quot;ERC1&quot;       &quot;PDS5B&quot;      &quot;OXCT1&quot;      &quot;ITPKC&quot;      &quot;ADAMTS2&quot;   
 [41] &quot;DNMT3B&quot;     &quot;MAPKAPK5&quot;   &quot;LTBP4&quot;      &quot;RGS17&quot;      &quot;HNRNPC&quot;    
 [46] &quot;DPYSL2&quot;     &quot;FKBP5&quot;      &quot;MYO9B&quot;      &quot;MAPK1&quot;      &quot;CENPM&quot;     
 [51] &quot;TCF20&quot;      &quot;TTLL12&quot;     &quot;TAB1&quot;       &quot;ACO2&quot;       &quot;ITPK1&quot;     
 [56] &quot;CEP128&quot;     &quot;DICER1&quot;     &quot;MYBL2&quot;      &quot;FNDC3A&quot;     &quot;ZNF423&quot;    
 [61] &quot;AXIN1&quot;      &quot;PIEZO1&quot;     &quot;CPPED1&quot;     &quot;KNOP1&quot;      &quot;MTFMT&quot;     
 [66] &quot;KCTD9&quot;      &quot;DOT1L&quot;      &quot;HNRNPUL1&quot;   &quot;ITGB8&quot;      &quot;GRB10&quot;     
 [71] &quot;SUSD1&quot;      &quot;PITRM1&quot;     &quot;MED13&quot;      &quot;BLMH&quot;       &quot;YWHAE&quot;     
 [76] &quot;DCUN1D4&quot;    &quot;JHY&quot;        &quot;CDKN1B&quot;     &quot;PHC1&quot;       &quot;SASH1&quot;     
 [81] &quot;LOX&quot;        &quot;ARSB&quot;       &quot;FGF1&quot;       &quot;CISH&quot;       &quot;FOXP1&quot;     
 [86] &quot;GLS&quot;        &quot;EHBP1&quot;      &quot;CHMP3&quot;      &quot;UXS1&quot;       &quot;HDLBP&quot;     
 [91] &quot;AAK1&quot;       &quot;ARID1A&quot;     &quot;FILIP1&quot;     &quot;LDAH&quot;       &quot;CCND2&quot;     
 [96] &quot;DENND1A&quot;    &quot;LTBP2&quot;      &quot;DNMT3A&quot;     &quot;ARAP3&quot;      &quot;SMAD9&quot;     
[101] &quot;TGFBI&quot;      &quot;NCAPH&quot;      &quot;RBBP6&quot;      &quot;BBS9&quot;       &quot;RASSF8&quot;    
[106] &quot;MOCS1&quot;      &quot;CEP250&quot;     &quot;PCNX4&quot;      &quot;HSPA2&quot;      &quot;MAP2K2&quot;    
[111] &quot;ABHD8&quot;      &quot;SMARCA4&quot;    &quot;MTERF1&quot;     &quot;SLC44A2&quot;    &quot;KIF1A&quot;     
[116] &quot;LSM4&quot;       &quot;TBC1D5&quot;     &quot;GFPT2&quot;      &quot;EXOC4&quot;      &quot;CHSY1&quot;     
[121] &quot;PRMT7&quot;      &quot;NASP&quot;       &quot;PATJ&quot;       &quot;CNDP2&quot;      &quot;SBF2&quot;      
[126] &quot;IAH1&quot;       &quot;NAV1&quot;       &quot;CABLES1&quot;    &quot;PUM1&quot;       &quot;ETS1&quot;      
[131] &quot;PRR5L&quot;      &quot;REPS1&quot;      &quot;LAMC1&quot;      &quot;DCAF7&quot;      &quot;DAB2IP&quot;    
[136] &quot;MFSD13A&quot;    &quot;CPNE8&quot;      &quot;WDFY2&quot;      &quot;CUL4A&quot;      &quot;UBE2Q2&quot;    
[141] &quot;IGF1R&quot;      &quot;ARRDC4&quot;     &quot;BBS4&quot;       &quot;PML&quot;        &quot;ZNF710&quot;    
[146] &quot;FURIN&quot;      &quot;MARVELD3&quot;   &quot;GALNS&quot;      &quot;TP53&quot;       &quot;TBCD&quot;      
[151] &quot;DYM&quot;        &quot;COL6A2&quot;     &quot;ADAMTS10&quot;   &quot;NOSIP&quot;      &quot;PRUNE1&quot;    
[156] &quot;ZNF687&quot;     &quot;SEMA6C&quot;     &quot;INTS3&quot;      &quot;PDIA6&quot;      &quot;CAMKMT&quot;    
[161] &quot;WDPCP&quot;      &quot;RHBDD1&quot;     &quot;VGLL4&quot;      &quot;IL17RD&quot;     &quot;NIPSNAP2&quot;  
[166] &quot;TLK2&quot;       &quot;UHRF2&quot;      &quot;PROSER2&quot;    &quot;PDCD11&quot;     &quot;PTPRJ&quot;     
[171] &quot;NCAM1&quot;      &quot;HMGA2&quot;      &quot;THRB&quot;       &quot;NPAS3&quot;      &quot;ADAMTS12&quot;  
[176] &quot;CCDC3&quot;      &quot;WWC2&quot;       &quot;RABGAP1L&quot;   &quot;SPC25&quot;      &quot;PDE3B&quot;     
[181] &quot;PLEKHH2&quot;    &quot;NADK2&quot;      &quot;PTPRK&quot;      &quot;PLOD2&quot;      &quot;DAB2&quot;      
[186] &quot;BCL2L11&quot;    &quot;RBMS1&quot;      &quot;LPCAT1&quot;     &quot;CEP112&quot;     &quot;PIEZO2&quot;    
[191] &quot;FMN2&quot;       &quot;RMND1&quot;      &quot;ADAMTS3&quot;    &quot;SFXN2&quot;      &quot;RBPMS&quot;     
[196] &quot;DDX19B&quot;     &quot;SPPL3&quot;      &quot;AGAP1&quot;      &quot;FANCC&quot;      &quot;CDC42SE2&quot;  
[201] &quot;IGF2BP1&quot;    &quot;SLC25A44&quot;   &quot;LMNA&quot;       &quot;DBF4B&quot;      &quot;PLCD3&quot;     
[206] &quot;SELENON&quot;    &quot;DDR2&quot;       &quot;PPP1R21&quot;    &quot;DPY30&quot;      &quot;CLHC1&quot;     
[211] &quot;FRZB&quot;       &quot;CDC42EP3&quot;   &quot;CLASP2&quot;     &quot;HHIP&quot;       &quot;ABCE1&quot;     
[216] &quot;PRRC1&quot;      &quot;STRBP&quot;      &quot;SLC16A9&quot;    &quot;REEP3&quot;      &quot;NDRG2&quot;     
[221] &quot;ZNF219&quot;     &quot;TTC7B&quot;      &quot;STXBP4&quot;     &quot;HSP90B1&quot;    &quot;AP1G1&quot;     
[226] &quot;SPRYD3&quot;     &quot;PXK&quot;        &quot;GDNF&quot;       &quot;VAMP5&quot;      &quot;ZNF608&quot;    
[231] &quot;PTPN9&quot;      &quot;ZEB2&quot;       &quot;APLF&quot;       &quot;LINGO1&quot;     &quot;ALCAM&quot;     
[236] &quot;ZNF318&quot;     &quot;TBCA&quot;       &quot;SPSB1&quot;      &quot;BPTF&quot;       &quot;THOP1&quot;     
[241] &quot;SP3&quot;        &quot;FAM222B&quot;    &quot;PEAK1&quot;      &quot;MSRB3&quot;      &quot;MSRA&quot;      
[246] &quot;POLE&quot;       &quot;GRB2&quot;       &quot;RNF212&quot;     &quot;PDXDC1&quot;     &quot;PCGF5&quot;     
[251] &quot;ARSJ&quot;       &quot;ACBD4&quot;      &quot;PLAG1&quot;      &quot;FIGN&quot;       &quot;TTC3&quot;      
[256] &quot;JRKL&quot;       &quot;TM2D3&quot;      &quot;PLA2G6&quot;     &quot;ZFP1&quot;       &quot;RNLS&quot;      
[261] &quot;PRR16&quot;      &quot;WWOX&quot;       &quot;HEXIM1&quot;     &quot;TEAD1&quot;      &quot;AKR1C1&quot;    
[266] &quot;THSD4&quot;      &quot;LCOR&quot;       &quot;XPNPEP3&quot;    &quot;LAMA2&quot;      &quot;TECPR2&quot;    
[271] &quot;NF1&quot;        &quot;COL27A1&quot;    &quot;FAM3C&quot;      &quot;BLM&quot;        &quot;ATG7&quot;      
[276] &quot;SPG7&quot;       &quot;DNM3&quot;       &quot;ZNF273&quot;     &quot;CD2AP&quot;      &quot;LPAR1&quot;     
[281] &quot;TXNRD1&quot;     &quot;TLK1&quot;       &quot;ZNF652&quot;     &quot;COLGALT2&quot;   &quot;MTOR&quot;      
[286] &quot;MAP3K3&quot;     &quot;L3MBTL3&quot;    &quot;SPIRE2&quot;     &quot;LGR4&quot;       &quot;HMGN1&quot;     
[291] &quot;VGLL3&quot;      &quot;HOMEZ&quot;      &quot;ZNF345&quot;     &quot;AP002495.1&quot; &quot;FNTB&quot;      
[296] &quot;CUX1&quot;       &quot;UHRF1&quot;      &quot;PIP4K2B&quot;    &quot;TYW1B&quot;     
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
 [1] &quot;DYRK4&quot;   &quot;EPB41L1&quot; &quot;PITPNM2&quot; &quot;PAPLN&quot;   &quot;FBXW4&quot;   &quot;MAPK10&quot;  &quot;DDX6&quot;   
 [8] &quot;KTN1&quot;    &quot;TBX3&quot;    &quot;AK3&quot;     &quot;ZNF462&quot;  &quot;NEK7&quot;    &quot;FER&quot;     &quot;GRPEL2&quot; 
[15] &quot;SAMD12&quot;  &quot;COL15A1&quot;
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Steinberg2020-GWAS.OA&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
  [1] &quot;ARHGAP17&quot;   &quot;ARSA&quot;       &quot;ASAH1&quot;      &quot;ATAD3B&quot;     &quot;ATP6AP1&quot;   
  [6] &quot;BCL2L13&quot;    &quot;C1R&quot;        &quot;CASP4&quot;      &quot;CCDC50&quot;     &quot;CFH&quot;       
 [11] &quot;CHDH&quot;       &quot;CLIC3&quot;      &quot;CMBL&quot;       &quot;COL5A1&quot;     &quot;CRISPLD1&quot;  
 [16] &quot;CRLF1&quot;      &quot;CSTB&quot;       &quot;CYLD&quot;       &quot;CYP1B1&quot;     &quot;DCPS&quot;      
 [21] &quot;DDAH1&quot;      &quot;DDX19A&quot;     &quot;DGKA&quot;       &quot;ENO1&quot;       &quot;EPS8L2&quot;    
 [26] &quot;ERP44&quot;      &quot;EXOSC4&quot;     &quot;FABP5&quot;      &quot;FBLN1&quot;      &quot;FDFT1&quot;     
 [31] &quot;FN1&quot;        &quot;FSTL1&quot;      &quot;GADD45GIP1&quot; &quot;GALE&quot;       &quot;GALNT16&quot;   
 [36] &quot;GGCT&quot;       &quot;GGH&quot;        &quot;GNPDA1&quot;     &quot;GPHN&quot;       &quot;GULP1&quot;     
 [41] &quot;HSPA4L&quot;     &quot;IGFBP7&quot;     &quot;INTS10&quot;     &quot;IRS1&quot;       &quot;KPNA3&quot;     
 [46] &quot;LIMA1&quot;      &quot;MAP1A&quot;      &quot;MAP1B&quot;      &quot;METTL7A&quot;    &quot;MFAP4&quot;     
 [51] &quot;MGST1&quot;      &quot;MMP19&quot;      &quot;MPDZ&quot;       &quot;MPP6&quot;       &quot;MRPS26&quot;    
 [56] &quot;MRPS27&quot;     &quot;NAMPT&quot;      &quot;NDRG1&quot;      &quot;NDUFB8&quot;     &quot;NFIA&quot;      
 [61] &quot;NME1&quot;       &quot;NPC2&quot;       &quot;NQO1&quot;       &quot;NRBF2&quot;      &quot;ODF2&quot;      
 [66] &quot;OSBPL1A&quot;    &quot;OSBPL3&quot;     &quot;PAK1IP1&quot;    &quot;PDK3&quot;       &quot;PDLIM1&quot;    
 [71] &quot;PDLIM4&quot;     &quot;PEX5&quot;       &quot;PFAS&quot;       &quot;PFKL&quot;       &quot;PGAM5&quot;     
 [76] &quot;PGRMC1&quot;     &quot;PID1&quot;       &quot;PNP&quot;        &quot;POSTN&quot;      &quot;PPFIA1&quot;    
 [81] &quot;PTGFRN&quot;     &quot;PYGL&quot;       &quot;RARRES2&quot;    &quot;RCAN1&quot;      &quot;RFC3&quot;      
 [86] &quot;RHOC&quot;       &quot;RRAS2&quot;      &quot;S100A4&quot;     &quot;SDR39U1&quot;    &quot;SEC31A&quot;    
 [91] &quot;SERPINB8&quot;   &quot;SH3KBP1&quot;    &quot;SMS&quot;        &quot;SMYD5&quot;      &quot;SNX21&quot;     
 [96] &quot;SPECC1&quot;     &quot;STK4&quot;       &quot;STRAP&quot;      &quot;TBL2&quot;       &quot;THBS3&quot;     
[101] &quot;THOC3&quot;      &quot;TRA2B&quot;      &quot;TRAFD1&quot;     &quot;TRIM47&quot;     &quot;UCHL1&quot;     
[106] &quot;VAV2&quot;       &quot;YARS2&quot;      &quot;ZFHX3&quot;     
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
 [1] &quot;C1GALT1&quot; &quot;CPT2&quot;    &quot;GTF2F2&quot;  &quot;KANK2&quot;   &quot;MTAP&quot;    &quot;MYO6&quot;    &quot;NEDD4L&quot; 
 [8] &quot;NPR2&quot;    &quot;OPLAH&quot;   &quot;P4HA3&quot;   &quot;PKN1&quot;    &quot;TYMP&quot;   
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
[1] &quot;AKR1B1&quot;  &quot;CCPG1&quot;   &quot;EFEMP1&quot;  &quot;ELMO1&quot;   &quot;FDPS&quot;    &quot;PBXIP1&quot;  &quot;SORBS1&quot; 
[8] &quot;TNFAIP6&quot;
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
 [1] &quot;ABCE1&quot;    &quot;ACLY&quot;     &quot;ACSL4&quot;    &quot;ADPRH&quot;    &quot;AEBP1&quot;    &quot;AKR1C1&quot;  
 [7] &quot;AMPD2&quot;    &quot;C17orf58&quot; &quot;CA5B&quot;     &quot;CASP3&quot;    &quot;CCDC124&quot;  &quot;CHMP2A&quot;  
[13] &quot;CMAS&quot;     &quot;COL4A1&quot;   &quot;COL4A2&quot;   &quot;COPG1&quot;    &quot;CPT1A&quot;    &quot;CTBS&quot;    
[19] &quot;CTSA&quot;     &quot;CYP27A1&quot;  &quot;DDR2&quot;     &quot;DDX21&quot;    &quot;DERA&quot;     &quot;EARS2&quot;   
[25] &quot;EPB41&quot;    &quot;EXD2&quot;     &quot;FARSB&quot;    &quot;FGF1&quot;     &quot;FRZB&quot;     &quot;FTL&quot;     
[31] &quot;GFPT2&quot;    &quot;GNPNAT1&quot;  &quot;GPT2&quot;     &quot;GRN&quot;      &quot;GSTZ1&quot;    &quot;HINT3&quot;   
[37] &quot;HNRNPA3&quot;  &quot;HNRNPD&quot;   &quot;HTRA3&quot;    &quot;ISG20&quot;    &quot;ITSN1&quot;    &quot;LAMC1&quot;   
[43] &quot;LMNB2&quot;    &quot;LTBP2&quot;    &quot;LTBP3&quot;    &quot;MYBBP1A&quot;  &quot;MYOF&quot;     &quot;NANS&quot;    
[49] &quot;NMT2&quot;     &quot;NOVA1&quot;    &quot;NUDT15&quot;   &quot;NXT1&quot;     &quot;PDCD11&quot;   &quot;PHKA1&quot;   
[55] &quot;PLCG1&quot;    &quot;PPIA&quot;     &quot;PRMT5&quot;    &quot;PRPF4&quot;    &quot;PRPS1&quot;    &quot;PTGIS&quot;   
[61] &quot;PTPRM&quot;    &quot;PXN&quot;      &quot;RSPO2&quot;    &quot;RTN3&quot;     &quot;SEC13&quot;    &quot;SEMA3C&quot;  
[67] &quot;SH3D19&quot;   &quot;SH3PXD2B&quot; &quot;SLC39A14&quot; &quot;SNCA&quot;     &quot;SNUPN&quot;    &quot;SNX5&quot;    
[73] &quot;SRI&quot;      &quot;STOML2&quot;   &quot;TGFBI&quot;    &quot;TIMM10&quot;   &quot;TIMP2&quot;    &quot;TOMM34&quot;  
[79] &quot;TP53BP1&quot;  &quot;UPP1&quot;     &quot;XRCC6&quot;   
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
 [1] &quot;BAIAP2L1&quot; &quot;COL15A1&quot;  &quot;CPSF4&quot;    &quot;FAP&quot;      &quot;GCLM&quot;     &quot;GPX8&quot;    
 [7] &quot;KATNAL1&quot;  &quot;LYPLA2&quot;   &quot;SCFD2&quot;    &quot;SDHB&quot;     &quot;TBC1D4&quot;   &quot;UNK&quot;     
[13] &quot;VDAC1&quot;   
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)
[1] &quot;Morris2020-GWAS.OP&quot;
[1] &quot;Shared in All Stages, tested in all classifications&quot;
  [1] &quot;PEX14&quot;     &quot;FBLIM1&quot;    &quot;CAPZB&quot;     &quot;ALPL&quot;      &quot;SF3A3&quot;    
  [6] &quot;CTPS1&quot;     &quot;HIVEP3&quot;    &quot;PLPP3&quot;     &quot;NFIA&quot;      &quot;NEGR1&quot;    
 [11] &quot;LPAR3&quot;     &quot;PKN2&quot;      &quot;TGFBR3&quot;    &quot;COL11A1&quot;   &quot;THBS3&quot;    
 [16] &quot;HHAT&quot;      &quot;TGFB2&quot;     &quot;PSEN2&quot;     &quot;IRF2BP2&quot;   &quot;ASXL2&quot;    
 [21] &quot;PKDCC&quot;     &quot;THADA&quot;     &quot;MSH6&quot;      &quot;BCL11A&quot;    &quot;ACTR2&quot;    
 [26] &quot;MEIS1&quot;     &quot;SH3RF3&quot;    &quot;ANAPC1&quot;    &quot;MERTK&quot;     &quot;CYBRD1&quot;   
 [31] &quot;FZD7&quot;      &quot;IKZF2&quot;     &quot;IRS1&quot;      &quot;COL6A3&quot;    &quot;MLPH&quot;     
 [36] &quot;HDAC4&quot;     &quot;TMEM43&quot;    &quot;PRKAR2A&quot;   &quot;RBM5&quot;      &quot;ITGB5&quot;    
 [41] &quot;PCOLCE2&quot;   &quot;DLG1&quot;      &quot;PCGF3&quot;     &quot;FGFRL1&quot;    &quot;LCORL&quot;    
 [46] &quot;ANAPC4&quot;    &quot;ARHGAP24&quot;  &quot;AFF1&quot;      &quot;ZNF827&quot;    &quot;FAT1&quot;     
 [51] &quot;ITGA1&quot;     &quot;SLC38A9&quot;   &quot;ADAMTS6&quot;   &quot;FAM174A&quot;   &quot;STK10&quot;    
 [56] &quot;GMDS&quot;      &quot;HLA-A&quot;     &quot;DAAM2&quot;     &quot;DST&quot;       &quot;GJA1&quot;     
 [61] &quot;TRMT11&quot;    &quot;AKAP12&quot;    &quot;SYNE1&quot;     &quot;SNX13&quot;     &quot;JAZF1&quot;    
 [66] &quot;CREB5&quot;     &quot;SEMA3E&quot;    &quot;SEMA3A&quot;    &quot;CDK6&quot;      &quot;SEM1&quot;     
 [71] &quot;KIAA1549&quot;  &quot;DLC1&quot;      &quot;MMP16&quot;     &quot;RUNX1T1&quot;   &quot;TRPS1&quot;    
 [76] &quot;MPDZ&quot;      &quot;CNTLN&quot;     &quot;PTPDC1&quot;    &quot;FBXW5&quot;     &quot;RAB18&quot;    
 [81] &quot;MPP7&quot;      &quot;BICC1&quot;     &quot;CCDC6&quot;     &quot;ZCCHC24&quot;   &quot;LDB3&quot;     
 [86] &quot;PAPSS2&quot;    &quot;VTI1A&quot;     &quot;MGMT&quot;      &quot;PCNX3&quot;     &quot;TPCN2&quot;    
 [91] &quot;PRSS23&quot;    &quot;YAP1&quot;      &quot;PTS&quot;       &quot;CADM1&quot;     &quot;CLMP&quot;     
 [96] &quot;FLI1&quot;      &quot;SOX5&quot;      &quot;SSPN&quot;      &quot;ITPR2&quot;     &quot;FAR2&quot;     
[101] &quot;METTL7A&quot;   &quot;ATF7&quot;      &quot;PPM1H&quot;     &quot;TMTC2&quot;     &quot;CRADD&quot;    
[106] &quot;SPATA13&quot;   &quot;DHRS12&quot;    &quot;KLF12&quot;     &quot;DOCK9&quot;     &quot;SAV1&quot;     
[111] &quot;BMP4&quot;      &quot;RHOJ&quot;      &quot;ZBTB25&quot;    &quot;RAD51B&quot;    &quot;TMED10&quot;   
[116] &quot;FBLN5&quot;     &quot;FAM234A&quot;   &quot;RAB11FIP3&quot; &quot;ZNF75A&quot;    &quot;CREBBP&quot;   
[121] &quot;ADCY9&quot;     &quot;EMP2&quot;      &quot;EEF2K&quot;     &quot;CYLD&quot;      &quot;ZFHX3&quot;    
[126] &quot;COTL1&quot;     &quot;POLR2A&quot;    &quot;GAS7&quot;      &quot;NUFIP2&quot;    &quot;STAT5A&quot;   
[131] &quot;DCAKD&quot;     &quot;PGS1&quot;      &quot;USP36&quot;     &quot;RFNG&quot;      &quot;FHOD3&quot;    
[136] &quot;NFATC1&quot;    &quot;PLPPR3&quot;    &quot;NFIC&quot;      &quot;RFX2&quot;      &quot;RHPN2&quot;    
[141] &quot;GPATCH1&quot;   &quot;SLC27A5&quot;   &quot;BMP2&quot;      &quot;SLX4IP&quot;    &quot;JAG1&quot;     
[146] &quot;UBE2L3&quot;    &quot;LIF&quot;       &quot;MYH9&quot;      &quot;RANGAP1&quot;   &quot;CELSR1&quot;   
[151] &quot;PPP6R2&quot;   
[1] &quot;Shared in Time 0 &amp; Time 1 (tested in all classifications)&quot;
 [1] &quot;AURKAIP1&quot; &quot;WNT2B&quot;    &quot;PLEKHM3&quot;  &quot;CDH6&quot;     &quot;SOBP&quot;     &quot;STARD3NL&quot;
 [7] &quot;SMARCD3&quot;  &quot;NPR2&quot;     &quot;ASRGL1&quot;   &quot;C2CD2L&quot;   &quot;C12orf66&quot; &quot;GPC6&quot;    
[13] &quot;SEMA6D&quot;   &quot;PKD1&quot;     &quot;FTO&quot;      &quot;JUP&quot;      &quot;NAGLU&quot;    &quot;ISYNA1&quot;  
[19] &quot;ZBED4&quot;   
[1] &quot;Shared in Time 1 &amp; Time 2, tested in all classifications&quot;
 [1] &quot;CTTNBP2NL&quot; &quot;UBE2E3&quot;    &quot;MBNL1&quot;     &quot;REST&quot;      &quot;MSMO1&quot;    
 [6] &quot;SUPT3H&quot;    &quot;GTF2I&quot;     &quot;ACHE&quot;      &quot;TSPAN12&quot;   &quot;MKLN1&quot;    
[11] &quot;PTCH1&quot;     &quot;TCF7L2&quot;    &quot;SOX6&quot;      &quot;SORL1&quot;     &quot;DCN&quot;      
[16] &quot;MARK3&quot;     &quot;TLE3&quot;      &quot;ZNF689&quot;    &quot;SALL1&quot;     &quot;CMIP&quot;     
[21] &quot;HS3ST3B1&quot;  &quot;MMD&quot;       &quot;VAPA&quot;      &quot;SBNO2&quot;    
[1] &quot;Unique to Time 0 (tested in all classifications)&quot;
  [1] &quot;KLHL17&quot;   &quot;ARID1A&quot;   &quot;LRRC8C&quot;   &quot;HIPK1&quot;    &quot;DNM3&quot;     &quot;NAV1&quot;    
  [7] &quot;FMN2&quot;     &quot;TMEM18&quot;   &quot;TRIB2&quot;    &quot;DNMT3A&quot;   &quot;PRKD3&quot;    &quot;BCL2L11&quot; 
 [13] &quot;FMNL2&quot;    &quot;FRZB&quot;     &quot;HECW2&quot;    &quot;SATB2&quot;    &quot;PPARG&quot;    &quot;THRB&quot;    
 [19] &quot;FBXL2&quot;    &quot;PXK&quot;      &quot;RNF212&quot;   &quot;RAB28&quot;    &quot;RBPJ&quot;     &quot;SMARCAD1&quot;
 [25] &quot;HHIP&quot;     &quot;TRIM2&quot;    &quot;WWC2&quot;     &quot;SLC1A3&quot;   &quot;DAB2&quot;     &quot;PELO&quot;    
 [31] &quot;FCHO2&quot;    &quot;ARSB&quot;     &quot;XRCC4&quot;    &quot;HSPA4&quot;    &quot;FOXC1&quot;    &quot;TRIM27&quot;  
 [37] &quot;ZNF318&quot;   &quot;VEGFA&quot;    &quot;SLC29A1&quot;  &quot;LRRC1&quot;    &quot;COL21A1&quot;  &quot;L3MBTL3&quot; 
 [43] &quot;EPDR1&quot;    &quot;GRB10&quot;    &quot;SEMA3D&quot;   &quot;FAM3C&quot;    &quot;ZNF800&quot;   &quot;ZNF783&quot;  
 [49] &quot;SLC4A2&quot;   &quot;PLAG1&quot;    &quot;RSPO2&quot;    &quot;DEPTOR&quot;   &quot;GAS1&quot;     &quot;KLF4&quot;    
 [55] &quot;DENND1A&quot;  &quot;ZNF438&quot;   &quot;ERLIN1&quot;   &quot;NT5C2&quot;    &quot;DUSP5&quot;    &quot;TEAD1&quot;   
 [61] &quot;CCDC34&quot;   &quot;LGR4&quot;     &quot;ARL14EP&quot;  &quot;PTPRJ&quot;    &quot;MYRF&quot;     &quot;AHNAK&quot;   
 [67] &quot;LTBP3&quot;    &quot;NCAM1&quot;    &quot;PHLDB1&quot;   &quot;PKNOX2&quot;   &quot;TSPAN9&quot;   &quot;EMP1&quot;    
 [73] &quot;RASSF8&quot;   &quot;ARF3&quot;     &quot;MSRB3&quot;    &quot;HMGA2&quot;    &quot;NAV3&quot;     &quot;CDK2AP1&quot; 
 [79] &quot;SMAD9&quot;    &quot;BAZ1A&quot;    &quot;MAP4K5&quot;   &quot;DICER1&quot;   &quot;CYFIP1&quot;   &quot;EHD4&quot;    
 [85] &quot;IGDCC4&quot;   &quot;PPCDC&quot;    &quot;IQGAP1&quot;   &quot;AXIN1&quot;    &quot;RSL1D1&quot;   &quot;SNX29&quot;   
 [91] &quot;SMG1&quot;     &quot;ZNF423&quot;   &quot;PIEZO1&quot;   &quot;SPG7&quot;     &quot;NXN&quot;      &quot;ABR&quot;     
 [97] &quot;NAA38&quot;    &quot;ACSF2&quot;    &quot;SPAG9&quot;    &quot;BCAS3&quot;    &quot;TLK2&quot;     &quot;POLG2&quot;   
[103] &quot;LLGL2&quot;    &quot;TIMP2&quot;    &quot;FAM210A&quot;  &quot;CABLES1&quot;  &quot;SETBP1&quot;   &quot;BCL2&quot;    
[109] &quot;DOT1L&quot;    &quot;MAP2K2&quot;   &quot;AXL&quot;      &quot;PRKD2&quot;    &quot;TRPC4AP&quot;  &quot;KREMEN1&quot; 
[115] &quot;MICALL1&quot;  &quot;FAM227A&quot; 
[1] &quot;Unique to Time 1 (tested in all classifications)&quot;
[1] &quot;CAMSAP2&quot; &quot;FBXW4&quot;   &quot;INPP5A&quot;  &quot;TBX3&quot;    &quot;MMP2&quot;    &quot;RGS9&quot;   
[1] &quot;Unique to Time 2 (tested in all classifications)&quot;
character(0)</code></pre>
</div>
<div id="examine-conservation-of-differentiation-specific-de-genes" class="section level3">
<h3>Examine Conservation of Differentiation-Specific DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;Time 0&quot;,&quot;Time 1&quot;,&quot;Time 0&quot;,&quot;Time 1&quot;),
                                       c(&quot;Time 1&quot;,&quot;Time 2&quot;,&quot;Time 1&quot;,&quot;Time 2&quot;)),
                           col=list(Stage=c(&quot;Time 0&quot;=&quot;pink&quot;,&quot;Time 1&quot;=&quot;mediumorchid&quot;,&quot;Time 2&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#subset data
gene.prob &lt;- cormotifData$bestmotif$p.post

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10327 genes) -&gt; 89.19%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))</code></pre>
<pre><code>[1] 10327</code></pre>
<pre class="r"><code>length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/11579</code></pre>
<pre><code>[1] 0.8918732</code></pre>
<pre class="r"><code># Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code> [1] &quot;AMACR&quot;   &quot;ARRDC1&quot;  &quot;BOC&quot;     &quot;C4orf47&quot; &quot;DNAJC28&quot; &quot;DPF2&quot;    &quot;FARS2&quot;  
 [8] &quot;FSIP1&quot;   &quot;GNPAT&quot;   &quot;GPATCH2&quot; &quot;GPX4&quot;    &quot;METTL23&quot; &quot;PSD3&quot;    &quot;SPTLC1&quot; 
[15] &quot;TRA2A&quot;   &quot;TUSC2&quot;   &quot;UBXN1&quot;   &quot;USPL1&quot;   &quot;WDR36&quot;   &quot;ZNF512&quot; </code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 266</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>  [1] &quot;ACER3&quot;    &quot;ACTR6&quot;    &quot;ACYP1&quot;    &quot;ADPRHL1&quot;  &quot;AGL&quot;      &quot;AIG1&quot;    
  [7] &quot;AKR1C1&quot;   &quot;ALKBH7&quot;   &quot;AMBRA1&quot;   &quot;ANGPT1&quot;   &quot;ANKRD40&quot;  &quot;ANO7&quot;    
 [13] &quot;APBA3&quot;    &quot;APP&quot;      &quot;ARHGDIG&quot;  &quot;ARL6&quot;     &quot;ARNTL&quot;    &quot;ARPC5L&quot;  
 [19] &quot;ASNSD1&quot;   &quot;ATP11C&quot;   &quot;ATP1B1&quot;   &quot;ATP1B3&quot;   &quot;ATR&quot;      &quot;ATXN7L3&quot; 
 [25] &quot;BBS7&quot;     &quot;BRCC3&quot;    &quot;BRD4&quot;     &quot;BRMS1&quot;    &quot;BTF3&quot;     &quot;BZW1&quot;    
 [31] &quot;C16orf74&quot; &quot;C1R&quot;      &quot;C1RL&quot;     &quot;CBFA2T2&quot;  &quot;CDC23&quot;    &quot;CDCP1&quot;   
 [37] &quot;CDK2&quot;     &quot;CEBPG&quot;    &quot;CEPT1&quot;    &quot;CHCHD2&quot;   &quot;CHML&quot;     &quot;CLDN12&quot;  
 [43] &quot;CMC2&quot;     &quot;CNIH4&quot;    &quot;COG2&quot;     &quot;COL8A2&quot;   &quot;COPS5&quot;    &quot;CPM&quot;     
 [49] &quot;CRNKL1&quot;   &quot;CSDE1&quot;    &quot;CTNNB1&quot;   &quot;CTSK&quot;     &quot;CWC15&quot;    &quot;CWC25&quot;   
 [55] &quot;DCAF12&quot;   &quot;DDB2&quot;     &quot;DDX17&quot;    &quot;DDX19A&quot;   &quot;DDX56&quot;    &quot;DHPS&quot;    
 [61] &quot;DHRS12&quot;   &quot;DLG4&quot;     &quot;DOCK9&quot;    &quot;DPM2&quot;     &quot;DPP4&quot;     &quot;DUSP19&quot;  
 [67] &quot;DUSP4&quot;    &quot;ECHS1&quot;    &quot;EIF2AK3&quot;  &quot;EIF3G&quot;    &quot;EIF6&quot;     &quot;EPSTI1&quot;  
 [73] &quot;ERCC3&quot;    &quot;EXOC4&quot;    &quot;FAM149B1&quot; &quot;FAM78A&quot;   &quot;FCF1&quot;     &quot;FTCDNL1&quot; 
 [79] &quot;FUCA2&quot;    &quot;GANC&quot;     &quot;GAS1&quot;     &quot;GCN1&quot;     &quot;GDPGP1&quot;   &quot;GFM1&quot;    
 [85] &quot;GGT7&quot;     &quot;GNA13&quot;    &quot;GOLGA8B&quot;  &quot;GPN2&quot;     &quot;GRSF1&quot;    &quot;GSK3A&quot;   
 [91] &quot;H3C6&quot;     &quot;HCN3&quot;     &quot;HMGN4&quot;    &quot;HSPA1L&quot;   &quot;HUWE1&quot;    &quot;IDE&quot;     
 [97] &quot;IFT27&quot;    &quot;IGDCC4&quot;   &quot;IGF1R&quot;    &quot;IL20RB&quot;   &quot;IMPACT&quot;   &quot;INSYN1&quot;  
[103] &quot;INTS6L&quot;   &quot;ITGB5&quot;    &quot;ITPA&quot;     &quot;KCNAB3&quot;   &quot;KCNJ14&quot;   &quot;KIAA1217&quot;
[109] &quot;KIF16B&quot;   &quot;KLC4&quot;     &quot;KLHL18&quot;   &quot;KRT17&quot;    &quot;LAMA2&quot;    &quot;LEAP2&quot;   
[115] &quot;LRIG3&quot;    &quot;LRRC42&quot;   &quot;LRRC75A&quot;  &quot;LSM10&quot;    &quot;LUC7L2&quot;   &quot;MAP3K11&quot; 
[121] &quot;MBD3&quot;     &quot;MCCC1&quot;    &quot;MLLT1&quot;    &quot;MPLKIP&quot;   &quot;MROH6&quot;    &quot;MRPL27&quot;  
[127] &quot;MRPL54&quot;   &quot;NAB1&quot;     &quot;NAPB&quot;     &quot;NATD1&quot;    &quot;NDUFA10&quot;  &quot;NDUFB7&quot;  
[133] &quot;NHLRC2&quot;   &quot;NOL10&quot;    &quot;NOL7&quot;     &quot;NOSIP&quot;    &quot;NPIPA1&quot;   &quot;NT5C3B&quot;  
[139] &quot;NTNG1&quot;    &quot;NTNG2&quot;    &quot;OGFOD2&quot;   &quot;OTULIN&quot;   &quot;P2RY11&quot;   &quot;PARP16&quot;  
[145] &quot;PATL1&quot;    &quot;PC&quot;       &quot;PCYT2&quot;    &quot;PDCD7&quot;    &quot;PDE4B&quot;    &quot;PHF10&quot;   
[151] &quot;PKIG&quot;     &quot;PLPP6&quot;    &quot;PMS2&quot;     &quot;PNPLA6&quot;   &quot;PODNL1&quot;   &quot;POLE3&quot;   
[157] &quot;PPHLN1&quot;   &quot;PPIL6&quot;    &quot;PPIP5K1&quot;  &quot;PPP6C&quot;    &quot;PRKAB1&quot;   &quot;PRKAB2&quot;  
[163] &quot;PROSER3&quot;  &quot;PRRT3&quot;    &quot;PSMA5&quot;    &quot;PSMA6&quot;    &quot;PSMD12&quot;   &quot;PSMD6&quot;   
[169] &quot;PSME3IP1&quot; &quot;PTK2B&quot;    &quot;PTPDC1&quot;   &quot;PTPN9&quot;    &quot;PTPRK&quot;    &quot;RAB33B&quot;  
[175] &quot;RAB40C&quot;   &quot;RAB8A&quot;    &quot;RAF1&quot;     &quot;RBBP9&quot;    &quot;RBM43&quot;    &quot;RHBDD1&quot;  
[181] &quot;RNF139&quot;   &quot;RNF40&quot;    &quot;RORA&quot;     &quot;RTN2&quot;     &quot;SEMA3B&quot;   &quot;SEMA3E&quot;  
[187] &quot;SENP2&quot;    &quot;SESN3&quot;    &quot;SH3GL1&quot;   &quot;SIRT3&quot;    &quot;SKIV2L&quot;   &quot;SLC12A2&quot; 
[193] &quot;SLC22A17&quot; &quot;SLC23A2&quot;  &quot;SLC25A11&quot; &quot;SLC3A2&quot;   &quot;SLF2&quot;     &quot;SMIM10&quot;  
[199] &quot;SMIM12&quot;   &quot;SNAPC5&quot;   &quot;SNRNP48&quot;  &quot;SNW1&quot;     &quot;SNX8&quot;     &quot;SPIN1&quot;   
[205] &quot;STK11IP&quot;  &quot;STK4&quot;     &quot;STT3B&quot;    &quot;SYCE1L&quot;   &quot;SYVN1&quot;    &quot;TASOR2&quot;  
[211] &quot;TBC1D32&quot;  &quot;TCEANC&quot;   &quot;TIGD2&quot;    &quot;TM6SF2&quot;   &quot;TMEM120B&quot; &quot;TMEM128&quot; 
[217] &quot;TMEM14C&quot;  &quot;TMEM35A&quot;  &quot;TRAK1&quot;    &quot;TRANK1&quot;   &quot;TRAPPC5&quot;  &quot;TRIM41&quot;  
[223] &quot;TTBK2&quot;    &quot;TUBE1&quot;    &quot;UBAC2&quot;    &quot;UBE2D3&quot;   &quot;UBE2J2&quot;   &quot;UBQLN1&quot;  
[229] &quot;UBXN11&quot;   &quot;UQCRC1&quot;   &quot;URI1&quot;     &quot;URM1&quot;     &quot;USP12&quot;    &quot;USP14&quot;   
[235] &quot;USP33&quot;    &quot;WDFY3&quot;    &quot;WDR55&quot;    &quot;WDR61&quot;    &quot;XPOT&quot;     &quot;YJEFN3&quot;  
[241] &quot;YTHDC2&quot;   &quot;ZBED8&quot;    &quot;ZBTB26&quot;   &quot;ZBTB42&quot;   &quot;ZBTB5&quot;    &quot;ZFAT&quot;    
[247] &quot;ZFP64&quot;    &quot;ZGPAT&quot;    &quot;ZNF133&quot;   &quot;ZNF155&quot;   &quot;ZNF304&quot;   &quot;ZNF451&quot;  
[253] &quot;ZNF460&quot;   &quot;ZNF461&quot;   &quot;ZNF513&quot;   &quot;ZNF517&quot;   &quot;ZNF526&quot;   &quot;ZNF528&quot;  
[259] &quot;ZNF529&quot;   &quot;ZNF554&quot;   &quot;ZNF609&quot;   &quot;ZNF706&quot;   &quot;ZNF780A&quot;  &quot;ZSCAN18&quot; 
[265] &quot;ZSWIM1&quot;   &quot;ZXDB&quot;    </code></pre>
<pre class="r"><code>#DE genes
plotMotif(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="general-clusters-res0.05" class="section level2">
<h2>General Clusters (res=0.05)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-1" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;iPSC.c2&quot;=&quot;pink4&quot;,
                                              &quot;iPSC.c3&quot;=&quot;grey&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        column_split=c(1,1,1,2,2,2),
        #column_split=c(1,1,2,3,3,4),
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>           iPSC.c1    iPSC.c2    iPSC.c3     MSC.c1 Osteogenic.c1
TSPAN6   0.9241894 0.42201360 0.16257817 0.25032570    0.05409602
DPM1     0.9993051 0.99836738 0.51975348 0.36893409    0.29099082
SCYL3    0.9481345 0.76077131 0.74163925 0.99891637    0.99485048
C1orf112 0.1111052 0.03081154 0.02063854 0.13922949    0.09239193
FUCA2    0.9817726 0.80463759 0.82386380 0.91163771    0.87631578
GCLC     0.5135217 0.45481567 0.06148321 0.08281718    0.01539041
         Osteogenic.c2
TSPAN6      0.04593441
DPM1        0.29182608
SCYL3       0.98897590
C1orf112    0.08624761
FUCA2       0.87477182
GCLC        0.01320334</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         iPSC.c1 iPSC.c2 iPSC.c3 MSC.c1 Osteogenic.c1 Osteogenic.c2
TSPAN6         1       0       0      0             0             0
DPM1           1       1       0      0             0             0
SCYL3          1       1       1      1             1             1
C1orf112       0       0       0      0             0             0
FUCA2          1       1       1      1             1             1
GCLC           0       0       0      0             0             0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;iPSC.c1&quot;,length(which(dataUpset$iPSC.c1==1))),
                                       rep(&quot;MSC.c1&quot;,length(which(dataUpset$MSC.c1==1))),
                                       rep(&quot;Osteogenic.c1&quot;,length(which(dataUpset$Osteogenic.c1==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),&quot;iPSC.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),&quot;MSC.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),&quot;Osteogenic.c1&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                               labels=c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
      iPSC.c1        MSC.c1 Osteogenic.c1 
         5168          4535          3585 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;iPSC.c1&quot;),
                         list(&quot;MSC.c1&quot;),
                         list(&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;),
                         list(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;Osteogenic.c1&quot;),
                         list(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>           iPSC.c1    iPSC.c2    iPSC.c3     MSC.c1 Osteogenic.c1
TSPAN6   0.9241894 0.42201360 0.16257817 0.25032570    0.05409602
DPM1     0.9993051 0.99836738 0.51975348 0.36893409    0.29099082
SCYL3    0.9481345 0.76077131 0.74163925 0.99891637    0.99485048
C1orf112 0.1111052 0.03081154 0.02063854 0.13922949    0.09239193
FUCA2    0.9817726 0.80463759 0.82386380 0.91163771    0.87631578
GCLC     0.5135217 0.45481567 0.06148321 0.08281718    0.01539041
         Osteogenic.c2
TSPAN6      0.04593441
DPM1        0.29182608
SCYL3       0.98897590
C1orf112    0.08624761
FUCA2       0.87477182
GCLC        0.01320334</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         iPSC.c1 iPSC.c2 iPSC.c3 MSC.c1 Osteogenic.c1 Osteogenic.c2
TSPAN6         1       0       0      0             0             0
DPM1           1       1       0      0             0             0
SCYL3          1       1       1      1             1             1
C1orf112       0       0       0      0             0             0
FUCA2          1       1       1      1             1             1
GCLC           0       0       0      0             0             0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;iPSC.c1&quot;,length(which(dataUpset$iPSC.c1==1))),
                                       rep(&quot;iPSC.c2&quot;,length(which(dataUpset$iPSC.c2==1))),
                                       rep(&quot;iPSC.c3&quot;,length(which(dataUpset$iPSC.c3==1))),
                                       rep(&quot;MSC.c1&quot;,length(which(dataUpset$MSC.c1==1))),
                                       rep(&quot;Osteogenic.c1&quot;,length(which(dataUpset$Osteogenic.c1==1))),
                                       rep(&quot;Osteogenic.c2&quot;,length(which(dataUpset$Osteogenic.c2==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c2==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c3==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),&quot;iPSC.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c2==1),&quot;iPSC.c2&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c3==1),&quot;iPSC.c3&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),&quot;MSC.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),&quot;Osteogenic.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),&quot;Osteogenic.c2&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;),
                               labels=c(&quot;iPSC.c1&quot;,&quot;iPSC.c2&quot;,&quot;iPSC.c3&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Cluster&quot;,
                    values=c(&quot;pink&quot;,&quot;pink4&quot;,&quot;grey&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;,&quot;navyblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
      iPSC.c1       iPSC.c2       iPSC.c3        MSC.c1 Osteogenic.c1 
         5168          2771          1295          4535          3585 
Osteogenic.c2 
         3507 </code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across ipsc.c1+msc.c1+osteo.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across ipsc.c1+msc.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in iPSC.c1 0 &amp; MSC.c1&quot;)

#Shared across msc.c1+osteo.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;)

#Unique to ipsc.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==0 &amp;
                                      dataUpset$Osteogenic.c1==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to iPSC.c1&quot;)

#Unique to msc.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to MSC.c1&quot;)

#Unique to osteo.c1
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==0 &amp;
                                      dataUpset$Osteogenic.c1==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteo.c1&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSC.c1 &amp; MSC.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in iPSC.c1 &amp; MSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSC.c1 &amp; Osteogenic.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in MSC.c1 &amp; Osteo.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSC.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &amp;
                                      dataUpset$MSC.c1==0 &amp;
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to iPSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSC.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==1 &amp;
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to MSC.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &amp;
                                      dataUpset$MSC.c1==0 &amp;
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteo.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                Source.DEGenes                 Source.GenesOfInterest
52           Unique to iPSC.c1   Gokhman2021-DE.Parental.iPSCs-Unique
58           Unique to iPSC.c1     Gokhman2021-DE.Hybrid.iPSCs-Unique
4            Unique to iPSC.c1              Blake2018-DE.iPSCs-Unique
49               Shared in All   Gokhman2021-DE.Parental.iPSCs-Unique
69 Shared in MSC.c1 &amp; Osteo.c1     Gokhman2021-DE.Hybrid.CNCCs-Unique
21 Shared in MSC.c1 &amp; Osteo.c1 Blake2018-DE.iPSC-endoderm-day3-Unique
56  Shared in iPSC.c1 &amp; MSC.c1     Gokhman2021-DE.Hybrid.iPSCs-Unique
34           Unique to iPSC.c1             Blake2020-DE.kidney-Unique
31               Shared in All             Blake2020-DE.kidney-Unique
11            Unique to MSC.c1 Blake2018-DE.iPSC-endoderm-day1-Unique
2   Shared in iPSC.c1 &amp; MSC.c1              Blake2018-DE.iPSCs-Unique
94           Unique to iPSC.c1                  Yengo2018-GWAS.height
40           Unique to iPSC.c1              Blake2020-DE.liver-Unique
50  Shared in iPSC.c1 &amp; MSC.c1   Gokhman2021-DE.Parental.iPSCs-Unique
65            Unique to MSC.c1   Gokhman2021-DE.Parental.CNCCs-Unique
95            Unique to MSC.c1                  Yengo2018-GWAS.height
63 Shared in MSC.c1 &amp; Osteo.c1   Gokhman2021-DE.Parental.CNCCs-Unique
26  Shared in iPSC.c1 &amp; MSC.c1              Blake2020-DE.heart-Unique
   DE.Interest      p.value    GeneRatio
52        1005 2.271920e-14 0.0083110053
58         634 6.329748e-13 0.0052429625
4          105 6.280806e-05 0.0008683140
49         972 1.063912e-03 0.0080381066
69         144 1.075772e-03 0.0011908306
21         110 3.576944e-03 0.0009096623
56         181 3.644160e-03 0.0014968079
34          42 5.276529e-03 0.0003473256
31          44 5.926854e-03 0.0003638649
11          13 2.280827e-02 0.0001075055
2           32 2.492682e-02 0.0002646290
94         241 2.521414e-02 0.0019929873
40         105 2.650611e-02 0.0008683140
50         278 3.292507e-02 0.0022989646
65          17 3.976434e-02 0.0001405842
95          39 4.207645e-02 0.0003225166
63          56 4.649727e-02 0.0004631008
26          81 4.812627e-02 0.0006698422</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
</div>
<div id="examine-conservation-of-differentiation-specific-de-genes-1" class="section level3">
<h3>Examine Conservation of Differentiation-Specific DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;,&quot;iPSC.c1&quot;,&quot;MSC.c1&quot;),
                                         c(&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;,&quot;MSC.c1&quot;,&quot;Osteogenic.c1&quot;)),
                           col=list(Cluster=c(&quot;iPSC.c1&quot;=&quot;pink&quot;,
                                              &quot;MSC.c1&quot;=&quot;mediumorchid&quot;,
                                              &quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Clusters&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-27-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#subset data
gene.prob &lt;- cormotifData$bestmotif$p.post

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c1: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9122 genes) -&gt; 84.92%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))</code></pre>
<pre><code>[1] 9122</code></pre>
<pre class="r"><code>length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/10742</code></pre>
<pre><code>[1] 0.8491901</code></pre>
<pre class="r"><code># Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code> [1] &quot;ARRDC1&quot;   &quot;BAX&quot;      &quot;C4orf47&quot;  &quot;CHMP4C&quot;   &quot;CLDND1&quot;   &quot;CLIC3&quot;   
 [7] &quot;COQ5&quot;     &quot;DAGLB&quot;    &quot;DDX24&quot;    &quot;DNAJC21&quot;  &quot;EYA3&quot;     &quot;FARS2&quot;   
[13] &quot;FLT1&quot;     &quot;GNPAT&quot;    &quot;HDAC9&quot;    &quot;KIAA0586&quot; &quot;MIEF1&quot;    &quot;MT1E&quot;    
[19] &quot;SESTD1&quot;   &quot;TFB2M&quot;    &quot;TMEM250&quot;  &quot;ZNF512&quot;  </code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 265</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>  [1] &quot;ABHD12&quot;   &quot;ACER3&quot;    &quot;ACYP1&quot;    &quot;AGL&quot;      &quot;AIG1&quot;     &quot;AKR1C1&quot;  
  [7] &quot;ALKBH7&quot;   &quot;ANKRD40&quot;  &quot;ANO7&quot;     &quot;AP3S2&quot;    &quot;APBA3&quot;    &quot;APLF&quot;    
 [13] &quot;APP&quot;      &quot;ARHGDIG&quot;  &quot;ARPC5L&quot;   &quot;ARSK&quot;     &quot;ASNSD1&quot;   &quot;ATP11C&quot;  
 [19] &quot;ATP13A1&quot;  &quot;ATP1B1&quot;   &quot;ATP1B3&quot;   &quot;ATP9B&quot;    &quot;ATXN7L3&quot;  &quot;BRCC3&quot;   
 [25] &quot;BRD4&quot;     &quot;BRMS1&quot;    &quot;BRPF3&quot;    &quot;BTF3&quot;     &quot;BZW1&quot;     &quot;C16orf74&quot;
 [31] &quot;C1orf21&quot;  &quot;C1RL&quot;     &quot;CALM1&quot;    &quot;CBFA2T2&quot;  &quot;CBWD3&quot;    &quot;CCND2&quot;   
 [37] &quot;CDC23&quot;    &quot;CDK14&quot;    &quot;CEBPG&quot;    &quot;CHCHD2&quot;   &quot;CMC2&quot;     &quot;COG2&quot;    
 [43] &quot;COQ4&quot;     &quot;CTNNB1&quot;   &quot;CTSK&quot;     &quot;CWC15&quot;    &quot;DCAF12&quot;   &quot;DCLRE1B&quot; 
 [49] &quot;DDB2&quot;     &quot;DDX17&quot;    &quot;DDX19A&quot;   &quot;DHPS&quot;     &quot;DHRS12&quot;   &quot;DLG4&quot;    
 [55] &quot;DOCK9&quot;    &quot;DPY19L3&quot;  &quot;DUSP19&quot;   &quot;ECHS1&quot;    &quot;EIF2AK3&quot;  &quot;EIF3D&quot;   
 [61] &quot;EIF3G&quot;    &quot;EIF6&quot;     &quot;EPC2&quot;     &quot;ERCC3&quot;    &quot;ERLIN2&quot;   &quot;EXOC4&quot;   
 [67] &quot;FAM149B1&quot; &quot;FAM160A2&quot; &quot;FAM200A&quot;  &quot;FCF1&quot;     &quot;FKBP1C&quot;   &quot;FUCA2&quot;   
 [73] &quot;GANC&quot;     &quot;GAS1&quot;     &quot;GCC2&quot;     &quot;GCN1&quot;     &quot;GDPGP1&quot;   &quot;GFM1&quot;    
 [79] &quot;GGT7&quot;     &quot;GNA13&quot;    &quot;GOLGA8B&quot;  &quot;GPI&quot;      &quot;GPN2&quot;     &quot;GRSF1&quot;   
 [85] &quot;GSK3A&quot;    &quot;GTF3C3&quot;   &quot;GTPBP10&quot;  &quot;H3C6&quot;     &quot;HDAC6&quot;    &quot;HMGN4&quot;   
 [91] &quot;IDE&quot;      &quot;IFT27&quot;    &quot;IGDCC4&quot;   &quot;IGF1R&quot;    &quot;IL20RB&quot;   &quot;ILRUN&quot;   
 [97] &quot;IMPACT&quot;   &quot;INTS6L&quot;   &quot;ITGB5&quot;    &quot;ITPA&quot;     &quot;KDM5C&quot;    &quot;KHDC4&quot;   
[103] &quot;KIAA1217&quot; &quot;KIF16B&quot;   &quot;KIF9&quot;     &quot;KLC4&quot;     &quot;KLHL18&quot;   &quot;LAMA2&quot;   
[109] &quot;LRRC75A&quot;  &quot;LRWD1&quot;    &quot;LUC7L2&quot;   &quot;MALSU1&quot;   &quot;MAMDC4&quot;   &quot;MAP3K11&quot; 
[115] &quot;MAP4K4&quot;   &quot;MAPK7&quot;    &quot;MBD3&quot;     &quot;MCCC1&quot;    &quot;MESD&quot;     &quot;MIA3&quot;    
[121] &quot;MINPP1&quot;   &quot;MLLT1&quot;    &quot;MPLKIP&quot;   &quot;MRFAP1&quot;   &quot;MRPL27&quot;   &quot;MRPL52&quot;  
[127] &quot;MRPL54&quot;   &quot;MSH3&quot;     &quot;MYC&quot;      &quot;MYNN&quot;     &quot;NAA35&quot;    &quot;NAA60&quot;   
[133] &quot;NAB1&quot;     &quot;NABP2&quot;    &quot;NAPB&quot;     &quot;NBPF3&quot;    &quot;NDUFB11&quot;  &quot;NDUFB7&quot;  
[139] &quot;NHLRC2&quot;   &quot;NME5&quot;     &quot;NOL7&quot;     &quot;NOSIP&quot;    &quot;NR2C2&quot;    &quot;NRDE2&quot;   
[145] &quot;NT5C3B&quot;   &quot;NTNG2&quot;    &quot;NUDT6&quot;    &quot;OGFOD2&quot;   &quot;PACS2&quot;    &quot;PATL1&quot;   
[151] &quot;PC&quot;       &quot;PCYT2&quot;    &quot;PDE4B&quot;    &quot;PKIG&quot;     &quot;PMS2&quot;     &quot;PNPLA6&quot;  
[157] &quot;POLE3&quot;    &quot;PPM1A&quot;    &quot;PPP2R5E&quot;  &quot;PPP6C&quot;    &quot;PROSER3&quot;  &quot;PRRT3&quot;   
[163] &quot;PSMA5&quot;    &quot;PSMA6&quot;    &quot;PSMB1&quot;    &quot;PSMC1&quot;    &quot;PSMC4&quot;    &quot;PSMD6&quot;   
[169] &quot;PSME3IP1&quot; &quot;PTPDC1&quot;   &quot;PTPN9&quot;    &quot;PTPRK&quot;    &quot;PTRHD1&quot;   &quot;RAB33B&quot;  
[175] &quot;RAB8A&quot;    &quot;RASSF4&quot;   &quot;RBBP9&quot;    &quot;RBM39&quot;    &quot;RHBDD1&quot;   &quot;RNF139&quot;  
[181] &quot;RNF40&quot;    &quot;RORA&quot;     &quot;RSAD1&quot;    &quot;RTN2&quot;     &quot;SELENON&quot;  &quot;SEMA3B&quot;  
[187] &quot;SENP2&quot;    &quot;SESN3&quot;    &quot;SETX&quot;     &quot;SFXN1&quot;    &quot;SKIV2L&quot;   &quot;SLC12A2&quot; 
[193] &quot;SLC22A17&quot; &quot;SLC23A2&quot;  &quot;SLC25A11&quot; &quot;SLC30A6&quot;  &quot;SLC3A2&quot;   &quot;SLF2&quot;    
[199] &quot;SMIM10&quot;   &quot;SMIM12&quot;   &quot;SNAPC5&quot;   &quot;SPIN1&quot;    &quot;SPTSSA&quot;   &quot;ST6GAL2&quot; 
[205] &quot;STAMBP&quot;   &quot;STK25&quot;    &quot;STK40&quot;    &quot;STT3B&quot;    &quot;SYCE1L&quot;   &quot;SYVN1&quot;   
[211] &quot;TALDO1&quot;   &quot;TASOR2&quot;   &quot;TBC1D32&quot;  &quot;TCF12&quot;    &quot;THAP7&quot;    &quot;TIGD2&quot;   
[217] &quot;TLN2&quot;     &quot;TM6SF2&quot;   &quot;TMEM120B&quot; &quot;TMEM128&quot;  &quot;TOMM20&quot;   &quot;TRAK1&quot;   
[223] &quot;TRANK1&quot;   &quot;TRIM41&quot;   &quot;TRPC1&quot;    &quot;TRUB1&quot;    &quot;TTBK2&quot;    &quot;TUBE1&quot;   
[229] &quot;TXNL1&quot;    &quot;UBAC2&quot;    &quot;UBE2J2&quot;   &quot;UBXN11&quot;   &quot;UQCRC1&quot;   &quot;URI1&quot;    
[235] &quot;USP12&quot;    &quot;USP14&quot;    &quot;WDR55&quot;    &quot;WDR61&quot;    &quot;XPOT&quot;     &quot;YJEFN3&quot;  
[241] &quot;YTHDC2&quot;   &quot;ZBED8&quot;    &quot;ZBTB5&quot;    &quot;ZBTB6&quot;    &quot;ZDHHC16&quot;  &quot;ZGPAT&quot;   
[247] &quot;ZNF155&quot;   &quot;ZNF174&quot;   &quot;ZNF189&quot;   &quot;ZNF211&quot;   &quot;ZNF304&quot;   &quot;ZNF460&quot;  
[253] &quot;ZNF461&quot;   &quot;ZNF500&quot;   &quot;ZNF513&quot;   &quot;ZNF514&quot;   &quot;ZNF526&quot;   &quot;ZNF527&quot;  
[259] &quot;ZNF529&quot;   &quot;ZNF609&quot;   &quot;ZNF7&quot;     &quot;ZNF706&quot;   &quot;ZRSR2&quot;    &quot;ZSWIM1&quot;  
[265] &quot;ZXDB&quot;    </code></pre>
</div>
</div>
<div id="general-ad-hoc-classification-of-cells-scaled-integrated-data-threshold0" class="section level2">
<h2>General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-2" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
                           col=list(AdHoc=c(&quot;iPSC&quot;=&quot;pink&quot;,&quot;MSC&quot;=&quot;mediumorchid&quot;,&quot;Osteogenic&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>              iPSC       MSC Osteogenic
TSPAN6   0.9738628 0.2309320  0.2292592
DPM1     0.9036413 0.1238028  0.1223380
SCYL3    0.9736890 0.9977407  0.9976999
C1orf112 0.4363738 0.5317194  0.5328557
CFH      0.9557079 0.9999952  0.9999986
FUCA2    0.9261341 0.5671191  0.5648067</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         iPSC MSC Osteogenic
TSPAN6      1   0          0
DPM1        1   0          0
SCYL3       1   1          1
C1orf112    0   0          0
CFH         1   1          1
FUCA2       1   0          0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;iPSC&quot;,length(which(dataUpset$iPSC==1))),
                                       rep(&quot;MSC&quot;,length(which(dataUpset$MSC==1))),
                                       rep(&quot;Osteogenic&quot;,length(which(dataUpset$Osteogenic==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$iPSC==1)],
                                rownames(dataUpset)[which(dataUpset$MSC==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC==1),&quot;iPSC&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC==1),&quot;MSC&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic==1),&quot;Osteogenic&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;),
                               labels=c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;,
                    values=c(&quot;pink&quot;,&quot;mediumorchid&quot;,&quot;dodgerblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
      iPSC        MSC Osteogenic 
      6629       4567       4553 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;iPSC&quot;),
                         list(&quot;MSC&quot;),
                         list(&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;MSC&quot;),
                         list(&quot;MSC&quot;,&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;Osteogenic&quot;),
                         list(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-29-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across ips+msc+ost
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across ips+msc
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
#                                      dataUpset$MSC==1 &amp;
#                                      dataUpset$Osteogenic==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in iPSC &amp; MSC&quot;)

#Shared across msc+ost
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in MSC &amp; Osteo&quot;)

#Unique to ips
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
                                      dataUpset$MSC==0 &amp;
                                      dataUpset$Osteogenic==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to iPSC&quot;)

#Unique to msc
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to MSC&quot;)

#Unique to ost
subGenes &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==0 &amp;
                                      dataUpset$Osteogenic==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteo&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across All Ad Hoc
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSCs &amp; MSCs
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in iPSCs &amp; MSCs&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSCs &amp; Osteogenic Cells
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in MSCs &amp; Osteogenic&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSCs
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==1 &amp;
                                      dataUpset$MSC==0 &amp;
                                      dataUpset$Osteogenic==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to iPSCs&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSCs
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==1 &amp;
                                      dataUpset$Osteogenic==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to MSCs&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic Cells
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$iPSC==0 &amp;
                                      dataUpset$MSC==0 &amp;
                                      dataUpset$Osteogenic==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to Osteogenic&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                Source.DEGenes                 Source.GenesOfInterest
52             Unique to iPSCs   Gokhman2021-DE.Parental.iPSCs-Unique
58             Unique to iPSCs     Gokhman2021-DE.Hybrid.iPSCs-Unique
4              Unique to iPSCs              Blake2018-DE.iPSCs-Unique
49               Shared in All   Gokhman2021-DE.Parental.iPSCs-Unique
94             Unique to iPSCs                  Yengo2018-GWAS.height
69 Shared in MSCs &amp; Osteogenic     Gokhman2021-DE.Hybrid.CNCCs-Unique
21 Shared in MSCs &amp; Osteogenic Blake2018-DE.iPSC-endoderm-day3-Unique
55               Shared in All     Gokhman2021-DE.Hybrid.iPSCs-Unique
63 Shared in MSCs &amp; Osteogenic   Gokhman2021-DE.Parental.CNCCs-Unique
75 Shared in MSCs &amp; Osteogenic                   Gokhman2020-DMR.bone
34             Unique to iPSCs             Blake2020-DE.kidney-Unique
   DE.Interest      p.value    GeneRatio
52        1285 8.802480e-16 0.0096107820
58         806 1.582363e-13 0.0060282415
4          126 1.221148e-04 0.0009423802
49        1519 1.984788e-03 0.0113609167
94         316 2.442023e-03 0.0023634297
69          94 6.154620e-03 0.0007030455
21          67 2.195034e-02 0.0005011069
55         909 2.451646e-02 0.0067985999
63          41 3.651237e-02 0.0003066475
75          26 3.717676e-02 0.0001944594
34          47 3.834983e-02 0.0003515228</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
</div>
<div id="examine-conservation-of-differentiation-specific-de-genes-2" class="section level3">
<h3>Examine Conservation of Differentiation-Specific DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Stage=cbind(c(&quot;iPSC&quot;,&quot;MSC&quot;,&quot;iPSC&quot;,&quot;MSC&quot;),
                                       c(&quot;MSC&quot;,&quot;Osteogenic&quot;,&quot;MSC&quot;,&quot;Osteogenic&quot;)),
                           col=list(Stage=c(&quot;iPSC&quot;=&quot;pink&quot;,
                                            &quot;MSC&quot;=&quot;mediumorchid&quot;,
                                            &quot;Osteogenic&quot;=&quot;dodgerblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-36-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#subset data
gene.prob &lt;- cormotifData$bestmotif$p.post

# Test degree of conservation for ad hoc assignments: posterior probability difference between the humans and the chimps within a motif 
#0.3 (3450 genes) -&gt; 29.18%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))</code></pre>
<pre><code>[1] 3450</code></pre>
<pre class="r"><code>length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) &lt;= 0.3) &amp; (abs(gene.prob[,2] - gene.prob[,4]) &lt;= 0.3), ]))/11824</code></pre>
<pre><code>[1] 0.2917794</code></pre>
<pre class="r"><code># Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 14</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code> [1] &quot;BRD4&quot;    &quot;C2orf92&quot; &quot;C5orf51&quot; &quot;DOCK9&quot;   &quot;FMC1&quot;    &quot;FNTA&quot;    &quot;GUCD1&quot;  
 [8] &quot;JAG1&quot;    &quot;MYH10&quot;   &quot;NEDD1&quot;   &quot;PELI3&quot;   &quot;RBMX2&quot;   &quot;SYCE1L&quot;  &quot;TOB2&quot;   </code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 453</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&gt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>  [1] &quot;ABAT&quot;       &quot;ACADSB&quot;     &quot;ACCS&quot;       &quot;ADA&quot;        &quot;ADAP2&quot;     
  [6] &quot;ADIPOR2&quot;    &quot;ADM&quot;        &quot;AGER&quot;       &quot;AIG1&quot;       &quot;AIMP1&quot;     
 [11] &quot;AK5&quot;        &quot;AKAP12&quot;     &quot;AKR1C1&quot;     &quot;AL445524.2&quot; &quot;AMBRA1&quot;    
 [16] &quot;AMMECR1&quot;    &quot;AMN1&quot;       &quot;AMZ2&quot;       &quot;ANGPT2&quot;     &quot;ANKRD17&quot;   
 [21] &quot;ANO4&quot;       &quot;APLF&quot;       &quot;APOBEC3G&quot;   &quot;AQP3&quot;       &quot;ARHGDIG&quot;   
 [26] &quot;ARHGEF37&quot;   &quot;ARL14EP&quot;    &quot;ARMC7&quot;      &quot;ATE1&quot;       &quot;ATP2B1&quot;    
 [31] &quot;ATP5MGL&quot;    &quot;ATP6V0E2&quot;   &quot;ATP6V1G1&quot;   &quot;ATP9B&quot;      &quot;BAG6&quot;      
 [36] &quot;BAZ1B&quot;      &quot;BBC3&quot;       &quot;BCKDK&quot;      &quot;BCL2L13&quot;    &quot;BLOC1S4&quot;   
 [41] &quot;BMERB1&quot;     &quot;BMF&quot;        &quot;BMP6&quot;       &quot;BRCC3&quot;      &quot;BRF1&quot;      
 [46] &quot;BRF2&quot;       &quot;BUD23&quot;      &quot;C12orf29&quot;   &quot;C16orf91&quot;   &quot;C17orf67&quot;  
 [51] &quot;C1orf159&quot;   &quot;C2CD2&quot;      &quot;C8orf82&quot;    &quot;C9orf78&quot;    &quot;CACNA1C&quot;   
 [56] &quot;CACTIN&quot;     &quot;CAPN5&quot;      &quot;CBWD3&quot;      &quot;CCDC102B&quot;   &quot;CCDC25&quot;    
 [61] &quot;CCDC3&quot;      &quot;CCDC90B&quot;    &quot;CD276&quot;      &quot;CDK8&quot;       &quot;CDKL2&quot;     
 [66] &quot;CEP44&quot;      &quot;CERK&quot;       &quot;CGGBP1&quot;     &quot;CHCHD2&quot;     &quot;CHML&quot;      
 [71] &quot;CHMP4C&quot;     &quot;CISD1&quot;      &quot;CLN3&quot;       &quot;CLN8&quot;       &quot;CMBL&quot;      
 [76] &quot;CNIH3&quot;      &quot;COG2&quot;       &quot;COL21A1&quot;    &quot;COLGALT2&quot;   &quot;COP1&quot;      
 [81] &quot;CORO7&quot;      &quot;CPNE7&quot;      &quot;CPT1C&quot;      &quot;CRYBB2&quot;     &quot;CSDE1&quot;     
 [86] &quot;CTHRC1&quot;     &quot;CTNND1&quot;     &quot;CUL3&quot;       &quot;CWC15&quot;      &quot;CXorf38&quot;   
 [91] &quot;CXorf56&quot;    &quot;DAB2IP&quot;     &quot;DDB1&quot;       &quot;DDT&quot;        &quot;DDX19A&quot;    
 [96] &quot;DEAF1&quot;      &quot;DEDD&quot;       &quot;DELE1&quot;      &quot;DGAT2&quot;      &quot;DHRSX&quot;     
[101] &quot;DICER1&quot;     &quot;DIPK2A&quot;     &quot;DNAAF4&quot;     &quot;DNAJB6&quot;     &quot;DNER&quot;      
[106] &quot;DNM3&quot;       &quot;DPY19L3&quot;    &quot;DUSP12&quot;     &quot;DUSP19&quot;     &quot;E4F1&quot;      
[111] &quot;ECPAS&quot;      &quot;EDC3&quot;       &quot;EGFL6&quot;      &quot;EHBP1&quot;      &quot;EID2&quot;      
[116] &quot;EIF2A&quot;      &quot;EIF4A2&quot;     &quot;ELOVL5&quot;     &quot;EMC2&quot;       &quot;ENTPD6&quot;    
[121] &quot;EPDR1&quot;      &quot;ERCC3&quot;      &quot;ETFDH&quot;      &quot;ETNK2&quot;      &quot;EXOC5&quot;     
[126] &quot;FAM104B&quot;    &quot;FAM200A&quot;    &quot;FAM241A&quot;    &quot;FAM86B1&quot;    &quot;FASTKD5&quot;   
[131] &quot;FBXO22&quot;     &quot;FEZ1&quot;       &quot;FGD6&quot;       &quot;FGF2&quot;       &quot;FGGY&quot;      
[136] &quot;FKBP11&quot;     &quot;FKBP1C&quot;     &quot;FMNL2&quot;      &quot;FN3K&quot;       &quot;FOS&quot;       
[141] &quot;FRG1&quot;       &quot;FUCA2&quot;      &quot;G0S2&quot;       &quot;GALNT16&quot;    &quot;GEMIN8&quot;    
[146] &quot;GGA1&quot;       &quot;GJA1&quot;       &quot;GK&quot;         &quot;GON7&quot;       &quot;GPR173&quot;    
[151] &quot;GPR82&quot;      &quot;GPRASP1&quot;    &quot;GPRC5C&quot;     &quot;GSTM4&quot;      &quot;GTF3C3&quot;    
[156] &quot;GTPBP10&quot;    &quot;GUCY1B1&quot;    &quot;H2AW&quot;       &quot;H4C5&quot;       &quot;HACD4&quot;     
[161] &quot;HEBP2&quot;      &quot;HK2&quot;        &quot;HLA-A&quot;      &quot;HOMEZ&quot;      &quot;HSD17B2&quot;   
[166] &quot;HSPA1L&quot;     &quot;HUS1&quot;       &quot;ID4&quot;        &quot;IFNAR2&quot;     &quot;IFT27&quot;     
[171] &quot;IGF1R&quot;      &quot;IL18&quot;       &quot;IL27RA&quot;     &quot;INA&quot;        &quot;INPP5E&quot;    
[176] &quot;INPP5F&quot;     &quot;INTS6&quot;      &quot;IPO8&quot;       &quot;IQCG&quot;       &quot;IRF2BPL&quot;   
[181] &quot;JMJD8&quot;      &quot;KAZN&quot;       &quot;KIAA0895L&quot;  &quot;KIAA1324L&quot;  &quot;KLHDC2&quot;    
[186] &quot;KLHL15&quot;     &quot;KLHL17&quot;     &quot;KLHL18&quot;     &quot;KMT2D&quot;      &quot;KPNA6&quot;     
[191] &quot;KRT7&quot;       &quot;LANCL2&quot;     &quot;LAP3&quot;       &quot;LARP4B&quot;     &quot;LIN54&quot;     
[196] &quot;LPIN3&quot;      &quot;LRRC1&quot;      &quot;LRRC3&quot;      &quot;LUC7L2&quot;     &quot;LUZP2&quot;     
[201] &quot;M6PR&quot;       &quot;MAD2L1BP&quot;   &quot;MAGIX&quot;      &quot;MAMDC2&quot;     &quot;MAP2K7&quot;    
[206] &quot;MAPK3&quot;      &quot;MAT2B&quot;      &quot;MBNL3&quot;      &quot;MCAT&quot;       &quot;MCCC1&quot;     
[211] &quot;MCM3AP&quot;     &quot;MCTP2&quot;      &quot;MEA1&quot;       &quot;MED12&quot;      &quot;MED7&quot;      
[216] &quot;MESD&quot;       &quot;MESP1&quot;      &quot;METTL6&quot;     &quot;MFAP3L&quot;     &quot;MIF4GD&quot;    
[221] &quot;MINDY2&quot;     &quot;MLEC&quot;       &quot;MMADHC&quot;     &quot;MOCOS&quot;      &quot;MORN1&quot;     
[226] &quot;MPDU1&quot;      &quot;MRPL35&quot;     &quot;MRPS11&quot;     &quot;MSH3&quot;       &quot;MTHFD1L&quot;   
[231] &quot;MTHFD2&quot;     &quot;MTIF3&quot;      &quot;MXD3&quot;       &quot;NAA35&quot;      &quot;NAALAD2&quot;   
[236] &quot;NAGA&quot;       &quot;NBPF3&quot;      &quot;NCLN&quot;       &quot;NDUFA10&quot;    &quot;NDUFB11&quot;   
[241] &quot;NEK11&quot;      &quot;NEO1&quot;       &quot;NGLY1&quot;      &quot;NINJ1&quot;      &quot;NIPBL&quot;     
[246] &quot;NOL10&quot;      &quot;NOP10&quot;      &quot;NPIPB15&quot;    &quot;NPPB&quot;       &quot;NR2C2&quot;     
[251] &quot;NRBP1&quot;      &quot;NSMCE1&quot;     &quot;NT5DC3&quot;     &quot;NUDT6&quot;      &quot;NUFIP2&quot;    
[256] &quot;NUP58&quot;      &quot;OGFOD1&quot;     &quot;OSCAR&quot;      &quot;PABPC4&quot;     &quot;PABPC4L&quot;   
[261] &quot;PAFAH1B2&quot;   &quot;PAPOLA&quot;     &quot;PCDH17&quot;     &quot;PDHX&quot;       &quot;PDIK1L&quot;    
[266] &quot;PDK1&quot;       &quot;PDRG1&quot;      &quot;PDS5A&quot;      &quot;PFKM&quot;       &quot;PGAP6&quot;     
[271] &quot;PGM2L1&quot;     &quot;PHF23&quot;      &quot;PIAS1&quot;      &quot;PIGBOS1&quot;    &quot;PIGF&quot;      
[276] &quot;PIGU&quot;       &quot;PIK3C2B&quot;    &quot;PITPNB&quot;     &quot;PLIN2&quot;      &quot;PMVK&quot;      
[281] &quot;PNMA2&quot;      &quot;POLDIP2&quot;    &quot;POPDC2&quot;     &quot;PRAG1&quot;      &quot;PRELID3B&quot;  
[286] &quot;PRLR&quot;       &quot;PROCR&quot;      &quot;PROSER3&quot;    &quot;PRPF6&quot;      &quot;PRR3&quot;      
[291] &quot;PSEN2&quot;      &quot;PSME4&quot;      &quot;PTGIS&quot;      &quot;PTK2&quot;       &quot;PTK2B&quot;     
[296] &quot;PTPDC1&quot;     &quot;PTPN3&quot;      &quot;QSOX2&quot;      &quot;RABL3&quot;      &quot;RAD51B&quot;    
[301] &quot;RAD51C&quot;     &quot;RALGAPA1&quot;   &quot;RAMAC&quot;      &quot;RASL11A&quot;    &quot;RBM33&quot;     
[306] &quot;RBM43&quot;      &quot;RBM48&quot;      &quot;RC3H1&quot;      &quot;REL&quot;        &quot;REST&quot;      
[311] &quot;REV1&quot;       &quot;RFLNA&quot;      &quot;RFX5&quot;       &quot;RGS7&quot;       &quot;RHBDD1&quot;    
[316] &quot;RHOU&quot;       &quot;RIOK2&quot;      &quot;RNF115&quot;     &quot;RNF145&quot;     &quot;RNF32&quot;     
[321] &quot;RNF41&quot;      &quot;RRN3&quot;       &quot;RSAD1&quot;      &quot;RUNDC1&quot;     &quot;SAP30BP&quot;   
[326] &quot;SART1&quot;      &quot;SBF2&quot;       &quot;SCAND1&quot;     &quot;SCRN2&quot;      &quot;SDC1&quot;      
[331] &quot;SDHAF2&quot;     &quot;SDHAF4&quot;     &quot;SEMA3E&quot;     &quot;SENP3&quot;      &quot;SEPTIN9&quot;   
[336] &quot;SERP2&quot;      &quot;SERTAD4&quot;    &quot;SETD1B&quot;     &quot;SETX&quot;       &quot;SFXN1&quot;     
[341] &quot;SH2D4A&quot;     &quot;SIPA1&quot;      &quot;SLC12A8&quot;    &quot;SLC30A9&quot;    &quot;SLC35A4&quot;   
[346] &quot;SLC36A4&quot;    &quot;SLC3A2&quot;     &quot;SLC9A3&quot;     &quot;SLC9A5&quot;     &quot;SMARCC2&quot;   
[351] &quot;SNRNP35&quot;    &quot;SNW1&quot;       &quot;SNX16&quot;      &quot;SORBS2&quot;     &quot;SPEN&quot;      
[356] &quot;SPRED1&quot;     &quot;SRC&quot;        &quot;SRSF5&quot;      &quot;SS18L1&quot;     &quot;STARD10&quot;   
[361] &quot;STK11&quot;      &quot;STK25&quot;      &quot;STK4&quot;       &quot;STMN2&quot;      &quot;STT3B&quot;     
[366] &quot;SUMO3&quot;      &quot;SYF2&quot;       &quot;TACSTD2&quot;    &quot;TAF5L&quot;      &quot;TAFA2&quot;     
[371] &quot;TALDO1&quot;     &quot;TCEAL8&quot;     &quot;TCF12&quot;      &quot;TEF&quot;        &quot;TEFM&quot;      
[376] &quot;TEK&quot;        &quot;TENT4A&quot;     &quot;TGDS&quot;       &quot;THY1&quot;       &quot;TIGD5&quot;     
[381] &quot;TIMMDC1&quot;    &quot;TLCD3A&quot;     &quot;TLN2&quot;       &quot;TM7SF3&quot;     &quot;TMED8&quot;     
[386] &quot;TMEM102&quot;    &quot;TMEM107&quot;    &quot;TMEM126A&quot;   &quot;TMEM159&quot;    &quot;TMEM164&quot;   
[391] &quot;TMEM218&quot;    &quot;TMEM231&quot;    &quot;TMEM241&quot;    &quot;TMEM70&quot;     &quot;TMTC2&quot;     
[396] &quot;TNPO1&quot;      &quot;TRAPPC10&quot;   &quot;TRPV2&quot;      &quot;TTC12&quot;      &quot;TUG1&quot;      
[401] &quot;UBE2J2&quot;     &quot;UBP1&quot;       &quot;UBXN2A&quot;     &quot;UBXN8&quot;      &quot;UCKL1&quot;     
[406] &quot;UNC45A&quot;     &quot;URM1&quot;       &quot;USP36&quot;      &quot;UVSSA&quot;      &quot;VARS1&quot;     
[411] &quot;VAV2&quot;       &quot;VHL&quot;        &quot;VPS13A&quot;     &quot;VPS33B&quot;     &quot;VPS37C&quot;    
[416] &quot;VRK3&quot;       &quot;WAPL&quot;       &quot;WARS2&quot;      &quot;WASL&quot;       &quot;WDR82&quot;     
[421] &quot;WNT3&quot;       &quot;XYLB&quot;       &quot;YWHAZ&quot;      &quot;ZBED3&quot;      &quot;ZBED5&quot;     
[426] &quot;ZBED8&quot;      &quot;ZBTB33&quot;     &quot;ZBTB8OS&quot;    &quot;ZFAND5&quot;     &quot;ZMYM5&quot;     
[431] &quot;ZNF222&quot;     &quot;ZNF280D&quot;    &quot;ZNF283&quot;     &quot;ZNF302&quot;     &quot;ZNF329&quot;    
[436] &quot;ZNF382&quot;     &quot;ZNF404&quot;     &quot;ZNF408&quot;     &quot;ZNF415&quot;     &quot;ZNF512&quot;    
[441] &quot;ZNF517&quot;     &quot;ZNF554&quot;     &quot;ZNF587&quot;     &quot;ZNF611&quot;     &quot;ZNF629&quot;    
[446] &quot;ZNF658&quot;     &quot;ZNF7&quot;       &quot;ZNF721&quot;     &quot;ZNF740&quot;     &quot;ZNF765&quot;    
[451] &quot;ZNF839&quot;     &quot;ZSCAN9&quot;     &quot;ZSWIM1&quot;    </code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code>[1] 23</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&gt;0.65),]))</code></pre>
<pre><code> [1] &quot;AP3S2&quot;   &quot;BST1&quot;    &quot;CASKIN2&quot; &quot;DNM1&quot;    &quot;GCLM&quot;    &quot;LIMCH1&quot;  &quot;LRFN4&quot;  
 [8] &quot;MACROD1&quot; &quot;NELFE&quot;   &quot;OTUB2&quot;   &quot;PFKFB4&quot;  &quot;PITPNM2&quot; &quot;PLCE1&quot;   &quot;PNPLA6&quot; 
[15] &quot;POLR2K&quot;  &quot;POMK&quot;    &quot;PTPA&quot;    &quot;SLC1A1&quot;  &quot;STARD5&quot;  &quot;THAP8&quot;   &quot;TP53BP2&quot;
[22] &quot;TUSC2&quot;   &quot;ZNF684&quot; </code></pre>
<pre class="r"><code>#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&gt;0.65 &amp; gene.prob[,2]&lt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>[1] 239</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[,1]&lt;0.65 &amp; gene.prob[,2]&gt;0.65 &amp; gene.prob[,3]&lt;0.65 &amp; gene.prob[,4]&lt;0.65),]))</code></pre>
<pre><code>  [1] &quot;AAR2&quot;        &quot;ABHD12&quot;      &quot;ACYP1&quot;       &quot;AFF1&quot;        &quot;AGAP5&quot;      
  [6] &quot;AHSA1&quot;       &quot;ANGPTL1&quot;     &quot;ANKRD39&quot;     &quot;ARMC8&quot;       &quot;ATP5F1C&quot;    
 [11] &quot;ATP5PB&quot;      &quot;BBS10&quot;       &quot;BRMS1&quot;       &quot;C12orf76&quot;    &quot;C1RL&quot;       
 [16] &quot;CARD8&quot;       &quot;CBFA2T2&quot;     &quot;CBX7&quot;        &quot;CCDC121&quot;     &quot;CCDC43&quot;     
 [21] &quot;CCDC78&quot;      &quot;CCND2&quot;       &quot;CD320&quot;       &quot;CDC37L1&quot;     &quot;CDK5RAP2&quot;   
 [26] &quot;CEP120&quot;      &quot;CEP55&quot;       &quot;CHD1L&quot;       &quot;CIAO3&quot;       &quot;CINP&quot;       
 [31] &quot;CLDN15&quot;      &quot;CLEC2D&quot;      &quot;CLK1&quot;        &quot;CLPX&quot;        &quot;CMC2&quot;       
 [36] &quot;CMTR2&quot;       &quot;COL7A1&quot;      &quot;COPS4&quot;       &quot;COQ4&quot;        &quot;COX7A2&quot;     
 [41] &quot;CSNK2B&quot;      &quot;CYP2U1&quot;      &quot;DCUN1D1&quot;     &quot;DDX56&quot;       &quot;DHPS&quot;       
 [46] &quot;DHX40&quot;       &quot;DLG4&quot;        &quot;DNHD1&quot;       &quot;DPM2&quot;        &quot;DTX3&quot;       
 [51] &quot;EBNA1BP2&quot;    &quot;ECHS1&quot;       &quot;EFL1&quot;        &quot;EGR2&quot;        &quot;EIF2AK3&quot;    
 [56] &quot;EIF2B3&quot;      &quot;EIF3D&quot;       &quot;EIF3G&quot;       &quot;EIF4E2&quot;      &quot;EIF5&quot;       
 [61] &quot;EIF5AL1&quot;     &quot;EIF6&quot;        &quot;EMC8&quot;        &quot;EPM2A&quot;       &quot;FBXW4&quot;      
 [66] &quot;FCF1&quot;        &quot;FHL3&quot;        &quot;FKBP5&quot;       &quot;FLCN&quot;        &quot;FPGS&quot;       
 [71] &quot;FTCDNL1&quot;     &quot;GAB1&quot;        &quot;GADD45GIP1&quot;  &quot;GLIS2&quot;       &quot;GNA13&quot;      
 [76] &quot;GNPNAT1&quot;     &quot;GPN2&quot;        &quot;GRSF1&quot;       &quot;GSK3A&quot;       &quot;HCCS&quot;       
 [81] &quot;HCFC1R1&quot;     &quot;ILRUN&quot;       &quot;IP6K1&quot;       &quot;ITPA&quot;        &quot;KATNBL1&quot;    
 [86] &quot;KCNJ14&quot;      &quot;KIAA0319L&quot;   &quot;KIAA1841&quot;    &quot;KIF27&quot;       &quot;KIF9&quot;       
 [91] &quot;KLF16&quot;       &quot;KPNA5&quot;       &quot;LIME1&quot;       &quot;MADD&quot;        &quot;MAMDC4&quot;     
 [96] &quot;MANBAL&quot;      &quot;MAP4K3&quot;      &quot;MAST3&quot;       &quot;MATN3&quot;       &quot;MBD3&quot;       
[101] &quot;METTL26&quot;     &quot;MPHOSPH10&quot;   &quot;MPLKIP&quot;      &quot;MRPL18&quot;      &quot;MRPL19&quot;     
[106] &quot;MRPL54&quot;      &quot;MRPS10&quot;      &quot;MRPS25&quot;      &quot;MTCH1&quot;       &quot;NAPB&quot;       
[111] &quot;NAPEPLD&quot;     &quot;NATD1&quot;       &quot;NCOA5&quot;       &quot;NDUFB10&quot;     &quot;NDUFB7&quot;     
[116] &quot;NDUFV1&quot;      &quot;NIPA2&quot;       &quot;NOSIP&quot;       &quot;NPIPA1&quot;      &quot;NTNG2&quot;      
[121] &quot;OGFOD2&quot;      &quot;OMA1&quot;        &quot;P2RY11&quot;      &quot;PACS2&quot;       &quot;PAPLN&quot;      
[126] &quot;PAQR6&quot;       &quot;PARK7&quot;       &quot;PCMT1&quot;       &quot;PER1&quot;        &quot;PFKFB2&quot;     
[131] &quot;PFN1&quot;        &quot;PGK1&quot;        &quot;PHF10&quot;       &quot;PKIG&quot;        &quot;PLAG1&quot;      
[136] &quot;PLXDC2&quot;      &quot;PMS2&quot;        &quot;PNMA8B&quot;      &quot;POLE3&quot;       &quot;POT1&quot;       
[141] &quot;PPIP5K1&quot;     &quot;PPM1D&quot;       &quot;PPP1R14B&quot;    &quot;PPP4C&quot;       &quot;PRDM11&quot;     
[146] &quot;PRR7&quot;        &quot;PRRC2B&quot;      &quot;PRRT3&quot;       &quot;PSMA1&quot;       &quot;PSMA7&quot;      
[151] &quot;PSMB1&quot;       &quot;PSMB3&quot;       &quot;PSMB5&quot;       &quot;PSMC1&quot;       &quot;PSMC4&quot;      
[156] &quot;PSMD13&quot;      &quot;PSMD6&quot;       &quot;PSMD7&quot;       &quot;PSME2&quot;       &quot;PTRHD1&quot;     
[161] &quot;PWP1&quot;        &quot;RAB8A&quot;       &quot;RABEPK&quot;      &quot;RANBP17&quot;     &quot;RBAK-RBAKDN&quot;
[166] &quot;REEP4&quot;       &quot;RFT1&quot;        &quot;RNF139&quot;      &quot;RRP36&quot;       &quot;SDF2L1&quot;     
[171] &quot;SEMA3B&quot;      &quot;SENP6&quot;       &quot;SEPSECS&quot;     &quot;SERAC1&quot;      &quot;SERINC2&quot;    
[176] &quot;SH3D19&quot;      &quot;SH3GL1&quot;      &quot;SLC22A17&quot;    &quot;SLC23A2&quot;     &quot;SLC25A11&quot;   
[181] &quot;SLC35E4&quot;     &quot;SMAD4&quot;       &quot;SNAPC2&quot;      &quot;SNF8&quot;        &quot;SNORC&quot;      
[186] &quot;SP1&quot;         &quot;SRP72&quot;       &quot;STAT5A&quot;      &quot;SYVN1&quot;       &quot;SZRD1&quot;      
[191] &quot;TASOR2&quot;      &quot;TBC1D15&quot;     &quot;TCEANC&quot;      &quot;TCF7L2&quot;      &quot;TMEM81&quot;     
[196] &quot;TMEM94&quot;      &quot;TMTC1&quot;       &quot;TNFAIP8L1&quot;   &quot;TRAK1&quot;       &quot;TRIP10&quot;     
[201] &quot;TRMT6&quot;       &quot;TSTD2&quot;       &quot;TTC8&quot;        &quot;TUBA1B&quot;      &quot;TUBB4B&quot;     
[206] &quot;UBB&quot;         &quot;UBE2F&quot;       &quot;UBE2L3&quot;      &quot;UCHL3&quot;       &quot;UPF1&quot;       
[211] &quot;UQCRC2&quot;      &quot;URI1&quot;        &quot;USP14&quot;       &quot;USP48&quot;       &quot;VCP&quot;        
[216] &quot;VDAC1&quot;       &quot;VDAC2&quot;       &quot;VPS4A&quot;       &quot;WBP1&quot;        &quot;WDFY1&quot;      
[221] &quot;WDFY3&quot;       &quot;WDSUB1&quot;      &quot;XAB2&quot;        &quot;XPOT&quot;        &quot;YWHAB&quot;      
[226] &quot;ZBED9&quot;       &quot;ZBTB34&quot;      &quot;ZBTB40&quot;      &quot;ZBTB49&quot;      &quot;ZC3H15&quot;     
[231] &quot;ZCCHC14&quot;     &quot;ZGLP1&quot;       &quot;ZNF133&quot;      &quot;ZNF181&quot;      &quot;ZNF263&quot;     
[236] &quot;ZNF28&quot;       &quot;ZNF451&quot;      &quot;ZNF526&quot;      &quot;ZNF768&quot;     </code></pre>
</div>
</div>
<div id="osteogenic-clusters-res0.50" class="section level2">
<h2>Osteogenic Clusters (res=0.50)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-3" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Cluster=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Species&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>         Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4
TSPAN6      0.01783566   0.001036920    0.01757839    0.01649149
DPM1        0.18970104   0.011974676    0.18963006    0.18924365
SCYL3       0.99402479   0.032418715    0.99365900    0.99046553
C1orf112    0.04315819   0.005199112    0.04284710    0.04116710
CFH         0.99999929   0.620340996    0.99999996    0.99999405
FUCA2       0.89895778   0.253253305    0.89925441    0.89637429</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4
TSPAN6               0             0             0             0
DPM1                 0             0             0             0
SCYL3                1             0             1             1
C1orf112             0             0             0             0
CFH                  1             0             1             1
FUCA2                1             0             1             1</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;Osteogenic.c1&quot;,length(which(dataUpset$Osteogenic.c1==1))),
                                       rep(&quot;Osteogenic.c2&quot;,length(which(dataUpset$Osteogenic.c2==1))),
                                       rep(&quot;Osteogenic.c3&quot;,length(which(dataUpset$Osteogenic.c3==1))),
                                       rep(&quot;Osteogenic.c4&quot;,length(which(dataUpset$Osteogenic.c4==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),&quot;Osteogenic.c1&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),&quot;Osteogenic.c2&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c3==1),&quot;Osteogenic.c3&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c4==1),&quot;Osteogenic.c4&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                               labels=c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Stage&quot;,
                    values=c(&quot;dodgerblue&quot;,&quot;deepskyblue2&quot;,&quot;dodgerblue4&quot;,&quot;navyblue&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
Osteogenic.c1 Osteogenic.c2 Osteogenic.c3 Osteogenic.c4 
         3847            91          3846          3813 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-38-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=&quot;on&quot;,
      intersections=list(list(&quot;Osteogenic.c1&quot;),
                         list(&quot;Osteogenic.c3&quot;),
                         list(&quot;Osteogenic.c4&quot;),
                         list(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c3&quot;),
                         list(&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;),
                         list(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c4&quot;),
                         list(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-38-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Osteogenic Clusters
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
#                                      dataUpset$Osteogenic.c2==1 &amp;
#                                      dataUpset$Osteogenic.c3==1 &amp;
#                                      dataUpset$Osteogenic.c4==1)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across O.c1 + O.c3 + O.c4
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
                                      dataUpset$Osteogenic.c2==0 &amp;
                                      dataUpset$Osteogenic.c3==1 &amp;
                                      dataUpset$Osteogenic.c4==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Osteo.c1 &amp; c3 &amp; c4&quot;)

#Shared across O.c1 + O.c3
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
                                      dataUpset$Osteogenic.c2==0 &amp;
                                      dataUpset$Osteogenic.c3==1 &amp;
                                      dataUpset$Osteogenic.c4==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in Osteo.c1 &amp; c3&quot;)

#Unique to O.c1
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
#                                      dataUpset$Osteogenic.c2==0 &amp;
#                                      dataUpset$Osteogenic.c3==0 &amp;
#                                      dataUpset$Osteogenic.c4==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Osteo.c1&quot;)

#Unique to O.c2
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==0 &amp;
#                                      dataUpset$Osteogenic.c2==1 &amp;
#                                      dataUpset$Osteogenic.c3==0 &amp;
#                                      dataUpset$Osteogenic.c4==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Osteo.c2&quot;)

#Unique to O.c3
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==0 &amp;
#                                      dataUpset$Osteogenic.c2==0 &amp;
#                                      dataUpset$Osteogenic.c3==1 &amp;
#                                      dataUpset$Osteogenic.c4==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Osteo.c3&quot;)

#Unique to O.c4
subGenes &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==0 &amp;
                                      dataUpset$Osteogenic.c2==0 &amp;
                                      dataUpset$Osteogenic.c3==0 &amp;
                                      dataUpset$Osteogenic.c4==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Osteo.c4&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across all ostclusters
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
                                      dataUpset$Osteogenic.c2==1 &amp;
                                      dataUpset$Osteogenic.c3==1 &amp;
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across O.c1, O.c3, O.c4
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
                                      dataUpset$Osteogenic.c2==0 &amp;
                                      dataUpset$Osteogenic.c3==1 &amp;
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Osteogenic.c1/c3/c4&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across O.c1, O.c3
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &amp;
                                      dataUpset$Osteogenic.c2==0 &amp;
                                      dataUpset$Osteogenic.c3==1 &amp;
                                      dataUpset$Osteogenic.c4==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Osteogenic.c1/c3&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #All Osteogenic.c1
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteogenic.c1&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c2
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteogenic.c2&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c3
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteogenic.c3&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c4
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteogenic.c4&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
ggplot(DEenrich[which(DEenrich$Source.GenesOfInterest %in% tempSource),], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-42-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- DEenrich[which(DEenrich$p.value&lt;0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                  Source.DEGenes Source.GenesOfInterest DE.Interest
91             All Osteogenic.c4   Gokhman2020-DMR.bone         111
90             All Osteogenic.c3   Gokhman2020-DMR.bone         110
88             All Osteogenic.c1   Gokhman2020-DMR.bone         110
86 Shared in Osteogenic.c1/c3/c4   Gokhman2020-DMR.bone         107
       p.value    GeneRatio
91 0.000543912 0.0003993438
90 0.001216205 0.0003957461
88 0.001228827 0.0003957461
86 0.001238548 0.0003849530</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
</div>
<div id="examine-conservation-of-differentiation-specific-de-genes-3" class="section level3">
<h3>Examine Conservation of Differentiation-Specific DE Genes</h3>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(Species=anno_text(c(&quot;Human&quot;,&quot;Human&quot;,&quot;Human&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;,&quot;Chimp&quot;),
                                             rot=360,
                                             just=&quot;center&quot;,
                                             location=0.5),
                           Cluster=cbind(c(&quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,
                                           &quot;Osteogenic.c1&quot;,&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;),
                                         c(&quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;,
                                           &quot;Osteogenic.c2&quot;,&quot;Osteogenic.c3&quot;,&quot;Osteogenic.c4&quot;)),
                           col=list(Cluster=c(&quot;Osteogenic.c1&quot;=&quot;dodgerblue&quot;,
                                              &quot;Osteogenic.c2&quot;=&quot;deepskyblue2&quot;,
                                              &quot;Osteogenic.c3&quot;=&quot;dodgerblue4&quot;,
                                              &quot;Osteogenic.c4&quot;=&quot;navyblue&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Stages&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-45-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#subset data
gene.prob &lt;- cormotifData$bestmotif$p.post

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7929 genes) -&gt; 74.30%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))</code></pre>
<pre><code>[1] 7929</code></pre>
<pre class="r"><code>length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,2] - gene.prob[,5]) &lt;= 0.3) &amp;
                          (abs(gene.prob[,3] - gene.prob[,6]) &lt;= 0.3), ]))/10671</code></pre>
<pre><code>[1] 0.7430419</code></pre>
<pre class="r"><code># Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&gt;0.65 &amp; gene.prob[ ,5]&gt;0.65 &amp; gene.prob[ ,6]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&gt;0.65 &amp; gene.prob[ ,5]&gt;0.65 &amp; gene.prob[ ,6]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between O.c1 and O.c2 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&gt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&gt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between O.c2 and O.c3 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&gt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 53</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&gt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code> [1] &quot;ADAMTSL5&quot; &quot;ANGPT1&quot;   &quot;C19orf33&quot; &quot;C8orf76&quot;  &quot;CAMKK2&quot;   &quot;CCDC126&quot; 
 [7] &quot;CRTC3&quot;    &quot;CSK&quot;      &quot;CXCL1&quot;    &quot;DBP&quot;      &quot;DDR2&quot;     &quot;DOCK9&quot;   
[13] &quot;EXOC4&quot;    &quot;FAM241B&quot;  &quot;FIG4&quot;     &quot;FKBPL&quot;    &quot;FLYWCH1&quot;  &quot;FNBP1&quot;   
[19] &quot;FUCA1&quot;    &quot;GFM2&quot;     &quot;IGSF8&quot;    &quot;ITGA4&quot;    &quot;KBTBD3&quot;   &quot;LRATD2&quot;  
[25] &quot;LRFN4&quot;    &quot;MESP1&quot;    &quot;MID1IP1&quot;  &quot;MOB3A&quot;    &quot;MPRIP&quot;    &quot;MTIF2&quot;   
[31] &quot;NBL1&quot;     &quot;NFE2L1&quot;   &quot;PCDH18&quot;   &quot;PEX11G&quot;   &quot;PICK1&quot;    &quot;PIP4K2B&quot; 
[37] &quot;PSMB9&quot;    &quot;RAB38&quot;    &quot;RAB3IL1&quot;  &quot;RIN1&quot;     &quot;RXRA&quot;     &quot;SBF1&quot;    
[43] &quot;SLC25A35&quot; &quot;SLC43A2&quot;  &quot;SOD2&quot;     &quot;SPIRE1&quot;   &quot;SPOCK1&quot;   &quot;ST3GAL2&quot; 
[49] &quot;TFB2M&quot;    &quot;THRAP3&quot;   &quot;TRMT10A&quot;  &quot;USP7&quot;     &quot;VAMP8&quot;   </code></pre>
<pre class="r"><code>#DE genes between O.c3 and O.c4 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&gt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&gt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]&gt;0.65 &amp; gene.prob[ ,2]&gt;0.65 &amp; gene.prob[ ,3]&gt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&gt;0.65 &amp; gene.prob[ ,2]&gt;0.65 &amp; gene.prob[ ,3]&gt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between O.c1 and O.c2 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]&gt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&gt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between O.c2 and O.c3 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&gt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&gt;0.65 &amp; gene.prob[ ,3]&lt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#DE genes between O.c3 and O.c4 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&gt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>[1] 681</code></pre>
<pre class="r"><code>sort(rownames(gene.prob[(gene.prob[ ,1]&lt;0.65 &amp; gene.prob[ ,2]&lt;0.65 &amp; gene.prob[ ,3]&gt;0.65 &amp; gene.prob[ ,4]&lt;0.65 &amp; gene.prob[ ,5]&lt;0.65 &amp; gene.prob[ ,6]&lt;0.65),]))</code></pre>
<pre><code>  [1] &quot;AATK&quot;         &quot;ABCB8&quot;        &quot;ABCB9&quot;        &quot;ABCG2&quot;       
  [5] &quot;ABRAXAS1&quot;     &quot;ACADM&quot;        &quot;ADAP1&quot;        &quot;ADGRA2&quot;      
  [9] &quot;ADSL&quot;         &quot;AGGF1&quot;        &quot;AHNAK2&quot;       &quot;AIG1&quot;        
 [13] &quot;AK3&quot;          &quot;AKT3&quot;         &quot;AL049629.2&quot;   &quot;AL121758.1&quot;  
 [17] &quot;ALAS1&quot;        &quot;ALG13&quot;        &quot;ALKBH5&quot;       &quot;AMIGO2&quot;      
 [21] &quot;AMOTL1&quot;       &quot;ANK2&quot;         &quot;ANKRD13A&quot;     &quot;ANKRD13C&quot;    
 [25] &quot;ANKRD13D&quot;     &quot;AP1B1&quot;        &quot;APEH&quot;         &quot;APH1B&quot;       
 [29] &quot;APMAP&quot;        &quot;APOBEC3F&quot;     &quot;ARHGAP26&quot;     &quot;ARHGAP31&quot;    
 [33] &quot;ARHGEF10L&quot;    &quot;ARIH2OS&quot;      &quot;ARL13B&quot;       &quot;ARMC1&quot;       
 [37] &quot;ARMC6&quot;        &quot;ARMCX3&quot;       &quot;ARMCX6&quot;       &quot;ASAP2&quot;       
 [41] &quot;ASCC2&quot;        &quot;ATF7&quot;         &quot;ATG4B&quot;        &quot;ATN1&quot;        
 [45] &quot;ATP10A&quot;       &quot;ATP1B3&quot;       &quot;ATP23&quot;        &quot;ATP6V0B&quot;     
 [49] &quot;ATP7B&quot;        &quot;AUTS2&quot;        &quot;B2M&quot;          &quot;B3GAT3&quot;      
 [53] &quot;B3GNT9&quot;       &quot;BACH1&quot;        &quot;BAG5&quot;         &quot;BASP1&quot;       
 [57] &quot;BCAP31&quot;       &quot;BCAR3&quot;        &quot;BCAS3&quot;        &quot;BCDIN3D&quot;     
 [61] &quot;BCL11A&quot;       &quot;BLOC1S4&quot;      &quot;BLOC1S5&quot;      &quot;BLOC1S6&quot;     
 [65] &quot;BMP4&quot;         &quot;BRCA1&quot;        &quot;BRWD1&quot;        &quot;C11orf54&quot;    
 [69] &quot;C17orf58&quot;     &quot;C19orf12&quot;     &quot;C1orf21&quot;      &quot;C1orf56&quot;     
 [73] &quot;C1QTNF6&quot;      &quot;C21orf91&quot;     &quot;C2CD3&quot;        &quot;C2orf69&quot;     
 [77] &quot;C2orf74&quot;      &quot;C7orf26&quot;      &quot;C8orf33&quot;      &quot;CAMK2D&quot;      
 [81] &quot;CAMSAP1&quot;      &quot;CASD1&quot;        &quot;CBR3&quot;         &quot;CCDC171&quot;     
 [85] &quot;CCDC174&quot;      &quot;CCDC191&quot;      &quot;CCDC22&quot;       &quot;CCDC88B&quot;     
 [89] &quot;CCDC92&quot;       &quot;CCHCR1&quot;       &quot;CCNE1&quot;        &quot;CCNG2&quot;       
 [93] &quot;CD151&quot;        &quot;CD3EAP&quot;       &quot;CD9&quot;          &quot;CDK16&quot;       
 [97] &quot;CDKL5&quot;        &quot;CDKN1B&quot;       &quot;CEMIP2&quot;       &quot;CENPBD1&quot;     
[101] &quot;CENPJ&quot;        &quot;CEP120&quot;       &quot;CEP78&quot;        &quot;CERS6&quot;       
[105] &quot;CES1&quot;         &quot;CHD1L&quot;        &quot;CHIC2&quot;        &quot;CHML&quot;        
[109] &quot;CHST14&quot;       &quot;CIAO1&quot;        &quot;CIR1&quot;         &quot;CITED2&quot;      
[113] &quot;CLBA1&quot;        &quot;CLCF1&quot;        &quot;CLIC4&quot;        &quot;CLINT1&quot;      
[117] &quot;CLIP4&quot;        &quot;CLP1&quot;         &quot;CMTM4&quot;        &quot;CNEP1R1&quot;     
[121] &quot;CNTLN&quot;        &quot;COL4A2&quot;       &quot;COQ10A&quot;       &quot;COQ6&quot;        
[125] &quot;CPSF4&quot;        &quot;CREB1&quot;        &quot;CRELD2&quot;       &quot;CRNKL1&quot;      
[129] &quot;CRYZL1&quot;       &quot;CSNK1G3&quot;      &quot;CSTF1&quot;        &quot;CTBS&quot;        
[133] &quot;CTTNBP2NL&quot;    &quot;CWC22&quot;        &quot;CWC25&quot;        &quot;CYBA&quot;        
[137] &quot;CYP27A1&quot;      &quot;CYP2R1&quot;       &quot;CYP2U1&quot;       &quot;CYP4V2&quot;      
[141] &quot;CYTH3&quot;        &quot;DAB2IP&quot;       &quot;DARS1&quot;        &quot;DARS2&quot;       
[145] &quot;DCTN4&quot;        &quot;DCUN1D4&quot;      &quot;DDX18&quot;        &quot;DECR2&quot;       
[149] &quot;DENND2B&quot;      &quot;DHCR24&quot;       &quot;DHFR&quot;         &quot;DHX38&quot;       
[153] &quot;DHX8&quot;         &quot;DKK1&quot;         &quot;DNAAF2&quot;       &quot;DNM1&quot;        
[157] &quot;DPM3&quot;         &quot;DSG2&quot;         &quot;DUS4L-BCAP29&quot; &quot;DUSP1&quot;       
[161] &quot;DUSP11&quot;       &quot;DUSP28&quot;       &quot;DUSP8&quot;        &quot;DVL2&quot;        
[165] &quot;DYNC2LI1&quot;     &quot;EBP&quot;          &quot;EDA2R&quot;        &quot;EEF1AKNMT&quot;   
[169] &quot;EFCAB7&quot;       &quot;EFL1&quot;         &quot;EGFL7&quot;        &quot;EHD2&quot;        
[173] &quot;EIF1B&quot;        &quot;ELP3&quot;         &quot;EMC7&quot;         &quot;ENC1&quot;        
[177] &quot;EPB41L1&quot;      &quot;EPC2&quot;         &quot;EPN2&quot;         &quot;EPS8&quot;        
[181] &quot;ERRFI1&quot;       &quot;ETV5&quot;         &quot;EXO5&quot;         &quot;FAM227B&quot;     
[185] &quot;FAM43A&quot;       &quot;FAM76A&quot;       &quot;FARP1&quot;        &quot;FAXDC2&quot;      
[189] &quot;FBXO21&quot;       &quot;FCGRT&quot;        &quot;FGD6&quot;         &quot;FGFRL1&quot;      
[193] &quot;FIGNL1&quot;       &quot;FILIP1L&quot;      &quot;FLI1&quot;         &quot;FLNA&quot;        
[197] &quot;FLOT2&quot;        &quot;G3BP2&quot;        &quot;GAB2&quot;         &quot;GADD45B&quot;     
[201] &quot;GALNT5&quot;       &quot;GAREM1&quot;       &quot;GAS8&quot;         &quot;GATB&quot;        
[205] &quot;GBP1&quot;         &quot;GCFC2&quot;        &quot;GDF6&quot;         &quot;GFOD2&quot;       
[209] &quot;GGH&quot;          &quot;GIGYF2&quot;       &quot;GLA&quot;          &quot;GLMN&quot;        
[213] &quot;GLRB&quot;         &quot;GNPTG&quot;        &quot;GOLIM4&quot;       &quot;GOLPH3&quot;      
[217] &quot;GOLPH3L&quot;      &quot;GORAB&quot;        &quot;GPAA1&quot;        &quot;GPATCH1&quot;     
[221] &quot;GPATCH3&quot;      &quot;GPC4&quot;         &quot;GPR63&quot;        &quot;GPT2&quot;        
[225] &quot;GPX3&quot;         &quot;GRASP&quot;        &quot;GSDME&quot;        &quot;GSR&quot;         
[229] &quot;GTF2A1&quot;       &quot;GTF2E1&quot;       &quot;GTF2F2&quot;       &quot;GTPBP4&quot;      
[233] &quot;GXYLT1&quot;       &quot;GZF1&quot;         &quot;HBP1&quot;         &quot;HDX&quot;         
[237] &quot;HGH1&quot;         &quot;HILPDA&quot;       &quot;HIVEP3&quot;       &quot;HMCN1&quot;       
[241] &quot;HPS1&quot;         &quot;HPS6&quot;         &quot;HS3ST3A1&quot;     &quot;HS3ST3B1&quot;    
[245] &quot;HSD17B1&quot;      &quot;HSD17B14&quot;     &quot;HSF2&quot;         &quot;HSPA12A&quot;     
[249] &quot;ICA1L&quot;        &quot;ID2&quot;          &quot;IDE&quot;          &quot;IER5L&quot;       
[253] &quot;IFFO2&quot;        &quot;IFT80&quot;        &quot;IGDCC4&quot;       &quot;IGFBP5&quot;      
[257] &quot;IK&quot;           &quot;IL11&quot;         &quot;IL1RAP&quot;       &quot;ILRUN&quot;       
[261] &quot;INAFM2&quot;       &quot;INKA2&quot;        &quot;INSIG1&quot;       &quot;IRF2BP1&quot;     
[265] &quot;IRS1&quot;         &quot;ISG20&quot;        &quot;ITGA3&quot;        &quot;ITPKB&quot;       
[269] &quot;ITPKC&quot;        &quot;IWS1&quot;         &quot;JUNB&quot;         &quot;KAT14&quot;       
[273] &quot;KAT8&quot;         &quot;KCNMA1&quot;       &quot;KCTD5&quot;        &quot;KEAP1&quot;       
[277] &quot;KIAA0100&quot;     &quot;KIAA0825&quot;     &quot;KIAA1671&quot;     &quot;KIFC3&quot;       
[281] &quot;KLF3&quot;         &quot;KLHL26&quot;       &quot;KNOP1&quot;        &quot;KPNB1&quot;       
[285] &quot;KRBOX4&quot;       &quot;KRT19&quot;        &quot;LAPTM4B&quot;      &quot;LIN52&quot;       
[289] &quot;LONP2&quot;        &quot;LRBA&quot;         &quot;LRIF1&quot;        &quot;LRIG1&quot;       
[293] &quot;LRP10&quot;        &quot;LRWD1&quot;        &quot;LSM11&quot;        &quot;LUZP1&quot;       
[297] &quot;LY96&quot;         &quot;LYSMD4&quot;       &quot;LZTS1&quot;        &quot;MADCAM1&quot;     
[301] &quot;MAGED2&quot;       &quot;MAMDC4&quot;       &quot;MAML1&quot;        &quot;MAML3&quot;       
[305] &quot;MAP3K3&quot;       &quot;MAPK11&quot;       &quot;MAPK12&quot;       &quot;MAPKAPK2&quot;    
[309] &quot;MARCHF8&quot;      &quot;MAT2B&quot;        &quot;MBIP&quot;         &quot;MCM8&quot;        
[313] &quot;MED24&quot;        &quot;MED28&quot;        &quot;MED9&quot;         &quot;MEGF8&quot;       
[317] &quot;MEGF9&quot;        &quot;MEIS1&quot;        &quot;MICU3&quot;        &quot;MINK1&quot;       
[321] &quot;MINPP1&quot;       &quot;MLEC&quot;         &quot;MPHOSPH6&quot;     &quot;MPP1&quot;        
[325] &quot;MPP5&quot;         &quot;MRPL49&quot;       &quot;MRS2&quot;         &quot;MTA2&quot;        
[329] &quot;MTAP&quot;         &quot;MTCL1&quot;        &quot;MTERF2&quot;       &quot;MTERF4&quot;      
[333] &quot;MTHFSD&quot;       &quot;MVB12B&quot;       &quot;MVD&quot;          &quot;MVK&quot;         
[337] &quot;MXRA5&quot;        &quot;MYPOP&quot;        &quot;N6AMT1&quot;       &quot;NAA30&quot;       
[341] &quot;NANP&quot;         &quot;NAV3&quot;         &quot;NBPF1&quot;        &quot;NCAPD3&quot;      
[345] &quot;NCOA7&quot;        &quot;NDEL1&quot;        &quot;NDN&quot;          &quot;NDUFAF7&quot;     
[349] &quot;NEK3&quot;         &quot;NEK7&quot;         &quot;NGDN&quot;         &quot;NLRX1&quot;       
[353] &quot;NMT2&quot;         &quot;NNT&quot;          &quot;NOL11&quot;        &quot;NOL12&quot;       
[357] &quot;NPRL3&quot;        &quot;NR1D2&quot;        &quot;NR2F2&quot;        &quot;NRIP3&quot;       
[361] &quot;NRP2&quot;         &quot;NSRP1&quot;        &quot;NSUN2&quot;        &quot;NSUN7&quot;       
[365] &quot;NUAK2&quot;        &quot;NUCB2&quot;        &quot;NUDCD3&quot;       &quot;NUP50&quot;       
[369] &quot;NUPR1&quot;        &quot;NXT1&quot;         &quot;OAT&quot;          &quot;OBSL1&quot;       
[373] &quot;OGDH&quot;         &quot;OGFR&quot;         &quot;OGG1&quot;         &quot;ORAI1&quot;       
[377] &quot;OSGEPL1&quot;      &quot;OTUD1&quot;        &quot;OXA1L&quot;        &quot;P3H4&quot;        
[381] &quot;PAF1&quot;         &quot;PAFAH2&quot;       &quot;PALLD&quot;        &quot;PARG&quot;        
[385] &quot;PBLD&quot;         &quot;PBX2&quot;         &quot;PCGF3&quot;        &quot;PCP4&quot;        
[389] &quot;PCTP&quot;         &quot;PDCD7&quot;        &quot;PDIK1L&quot;       &quot;PDLIM3&quot;      
[393] &quot;PELI1&quot;        &quot;PELP1&quot;        &quot;PEX13&quot;        &quot;PFKFB2&quot;      
[397] &quot;PFKFB4&quot;       &quot;PGAP6&quot;        &quot;PHF11&quot;        &quot;PIGQ&quot;        
[401] &quot;PIGS&quot;         &quot;PIGT&quot;         &quot;PITPNM1&quot;      &quot;PKIA&quot;        
[405] &quot;PLAA&quot;         &quot;PLAUR&quot;        &quot;PLCXD1&quot;       &quot;PLK1&quot;        
[409] &quot;PLPPR2&quot;       &quot;PLXNB1&quot;       &quot;PLXND1&quot;       &quot;PODXL2&quot;      
[413] &quot;POLD3&quot;        &quot;POLG2&quot;        &quot;POLI&quot;         &quot;POMGNT1&quot;     
[417] &quot;PON2&quot;         &quot;PPIB&quot;         &quot;PPP1R12C&quot;     &quot;PPP1R26&quot;     
[421] &quot;PPP1R8&quot;       &quot;PPP2R2A&quot;      &quot;PPP3CB&quot;       &quot;PPP5C&quot;       
[425] &quot;PPP6R1&quot;       &quot;PRICKLE3&quot;     &quot;PRKAR2B&quot;      &quot;PRKCI&quot;       
[429] &quot;PRKDC&quot;        &quot;PRKRIP1&quot;      &quot;PROCA1&quot;       &quot;PROS1&quot;       
[433] &quot;PROSER2&quot;      &quot;PRPF18&quot;       &quot;PRPF19&quot;       &quot;PRXL2A&quot;      
[437] &quot;PSD4&quot;         &quot;PSENEN&quot;       &quot;PSKH1&quot;        &quot;PSMC6&quot;       
[441] &quot;PSORS1C1&quot;     &quot;PTPN13&quot;       &quot;PTPN18&quot;       &quot;PWP1&quot;        
[445] &quot;PYGO2&quot;        &quot;R3HDM1&quot;       &quot;RAB14&quot;        &quot;RAB30&quot;       
[449] &quot;RAB6B&quot;        &quot;RABGAP1L&quot;     &quot;RAD52&quot;        &quot;RARA&quot;        
[453] &quot;RARS1&quot;        &quot;RASL10B&quot;      &quot;RBBP9&quot;        &quot;RBM19&quot;       
[457] &quot;RBM22&quot;        &quot;RBM4B&quot;        &quot;RBP1&quot;         &quot;RCCD1&quot;       
[461] &quot;RDH5&quot;         &quot;RELB&quot;         &quot;RGP1&quot;         &quot;RHBDD2&quot;      
[465] &quot;RHOB&quot;         &quot;RIMS2&quot;        &quot;RING1&quot;        &quot;RIOX1&quot;       
[469] &quot;RNASEH1&quot;      &quot;RND3&quot;         &quot;RNF215&quot;       &quot;RNF34&quot;       
[473] &quot;ROR1&quot;         &quot;RRAGB&quot;        &quot;RRAS2&quot;        &quot;RSRC1&quot;       
[477] &quot;RTN2&quot;         &quot;RUNX2&quot;        &quot;RWDD2B&quot;       &quot;RYK&quot;         
[481] &quot;SACS&quot;         &quot;SAMD14&quot;       &quot;SAMD4A&quot;       &quot;SAT1&quot;        
[485] &quot;SATB1&quot;        &quot;SCFD1&quot;        &quot;SCYL2&quot;        &quot;SDE2&quot;        
[489] &quot;SDR39U1&quot;      &quot;SELENOO&quot;      &quot;SELENOS&quot;      &quot;SERINC5&quot;     
[493] &quot;SESN1&quot;        &quot;SETD6&quot;        &quot;SETDB2&quot;       &quot;SGCB&quot;        
[497] &quot;SGO2&quot;         &quot;SGTB&quot;         &quot;SH3BP1&quot;       &quot;SHMT2&quot;       
[501] &quot;SIX4&quot;         &quot;SLAIN2&quot;       &quot;SLC25A23&quot;     &quot;SLC29A3&quot;     
[505] &quot;SLC2A8&quot;       &quot;SLC31A1&quot;      &quot;SLC38A6&quot;      &quot;SLC41A3&quot;     
[509] &quot;SLC43A3&quot;      &quot;SLC45A4&quot;      &quot;SLC4A1AP&quot;     &quot;SLC4A2&quot;      
[513] &quot;SLC66A2&quot;      &quot;SLC6A9&quot;       &quot;SLC7A2&quot;       &quot;SLC7A5&quot;      
[517] &quot;SMAD3&quot;        &quot;SMARCA4&quot;      &quot;SMARCD1&quot;      &quot;SMARCE1&quot;     
[521] &quot;SMOX&quot;         &quot;SNCA&quot;         &quot;SNRPA&quot;        &quot;SNTB2&quot;       
[525] &quot;SNX1&quot;         &quot;SNX29&quot;        &quot;SORBS2&quot;       &quot;SOX6&quot;        
[529] &quot;SPATA13&quot;      &quot;SPATA2&quot;       &quot;SPATA20&quot;      &quot;SPATA5&quot;      
[533] &quot;SPATS2L&quot;      &quot;SPTLC1&quot;       &quot;SRPRA&quot;        &quot;SRRD&quot;        
[537] &quot;SSC5D&quot;        &quot;SSH3&quot;         &quot;SSRP1&quot;        &quot;ST6GAL1&quot;     
[541] &quot;STAU2&quot;        &quot;STX12&quot;        &quot;STXBP1&quot;       &quot;STXBP6&quot;      
[545] &quot;STYXL1&quot;       &quot;SWAP70&quot;       &quot;TACC2&quot;        &quot;TACO1&quot;       
[549] &quot;TAF2&quot;         &quot;TAP2&quot;         &quot;TARBP2&quot;       &quot;TBC1D1&quot;      
[553] &quot;TBC1D9B&quot;      &quot;TBL3&quot;         &quot;TBX19&quot;        &quot;TCF7L1&quot;      
[557] &quot;TCTN3&quot;        &quot;TEAD3&quot;        &quot;TECPR2&quot;       &quot;TEFM&quot;        
[561] &quot;TELO2&quot;        &quot;TENM4&quot;        &quot;TENT5A&quot;       &quot;TEX10&quot;       
[565] &quot;TEX261&quot;       &quot;TGFB2&quot;        &quot;THAP1&quot;        &quot;THAP5&quot;       
[569] &quot;THBS2&quot;        &quot;TKFC&quot;         &quot;TM2D1&quot;        &quot;TM9SF1&quot;      
[573] &quot;TMEM135&quot;      &quot;TMEM171&quot;      &quot;TMEM182&quot;      &quot;TMEM192&quot;     
[577] &quot;TMEM238&quot;      &quot;TMEM240&quot;      &quot;TMEM59&quot;       &quot;TNPO3&quot;       
[581] &quot;TOE1&quot;         &quot;TP53BP2&quot;      &quot;TPD52&quot;        &quot;TRIM35&quot;      
[585] &quot;TRIM59&quot;       &quot;TRIM62&quot;       &quot;TRIM8&quot;        &quot;TRMT5&quot;       
[589] &quot;TRNT1&quot;        &quot;TRUB2&quot;        &quot;TSEN2&quot;        &quot;TSNAX&quot;       
[593] &quot;TSPAN2&quot;       &quot;TSPAN3&quot;       &quot;TTC33&quot;        &quot;TTF1&quot;        
[597] &quot;TUSC3&quot;        &quot;TXNDC12&quot;      &quot;UBALD2&quot;       &quot;UBE2G1&quot;      
[601] &quot;UBN1&quot;         &quot;UBP1&quot;         &quot;UBXN2B&quot;       &quot;UHRF1BP1&quot;    
[605] &quot;UHRF1BP1L&quot;    &quot;USP10&quot;        &quot;USP18&quot;        &quot;USP27X&quot;      
[609] &quot;USP5&quot;         &quot;UXS1&quot;         &quot;VASH1&quot;        &quot;VASP&quot;        
[613] &quot;VLDLR&quot;        &quot;VOPP1&quot;        &quot;VPS16&quot;        &quot;WBP2&quot;        
[617] &quot;WBP4&quot;         &quot;WDFY1&quot;        &quot;WDFY2&quot;        &quot;WDR20&quot;       
[621] &quot;WDR44&quot;        &quot;WFDC3&quot;        &quot;WNK3&quot;         &quot;XPR1&quot;        
[625] &quot;YAF2&quot;         &quot;YIPF2&quot;        &quot;YLPM1&quot;        &quot;YTHDF2&quot;      
[629] &quot;ZBED8&quot;        &quot;ZBTB14&quot;       &quot;ZBTB45&quot;       &quot;ZDHHC24&quot;     
[633] &quot;ZFP82&quot;        &quot;ZFP91&quot;        &quot;ZHX2&quot;         &quot;ZMAT3&quot;       
[637] &quot;ZNF134&quot;       &quot;ZNF14&quot;        &quot;ZNF16&quot;        &quot;ZNF17&quot;       
[641] &quot;ZNF18&quot;        &quot;ZNF195&quot;       &quot;ZNF25&quot;        &quot;ZNF250&quot;      
[645] &quot;ZNF267&quot;       &quot;ZNF275&quot;       &quot;ZNF324&quot;       &quot;ZNF329&quot;      
[649] &quot;ZNF34&quot;        &quot;ZNF347&quot;       &quot;ZNF354A&quot;      &quot;ZNF404&quot;      
[653] &quot;ZNF416&quot;       &quot;ZNF419&quot;       &quot;ZNF429&quot;       &quot;ZNF460&quot;      
[657] &quot;ZNF462&quot;       &quot;ZNF473&quot;       &quot;ZNF503&quot;       &quot;ZNF577&quot;      
[661] &quot;ZNF579&quot;       &quot;ZNF616&quot;       &quot;ZNF639&quot;       &quot;ZNF649&quot;      
[665] &quot;ZNF658&quot;       &quot;ZNF668&quot;       &quot;ZNF670&quot;       &quot;ZNF688&quot;      
[669] &quot;ZNF689&quot;       &quot;ZNF69&quot;        &quot;ZNF691&quot;       &quot;ZNF721&quot;      
[673] &quot;ZNF74&quot;        &quot;ZNF772&quot;       &quot;ZNF773&quot;       &quot;ZNF786&quot;      
[677] &quot;ZNF83&quot;        &quot;ZNF830&quot;       &quot;ZNF85&quot;        &quot;ZSWIM1&quot;      
[681] &quot;ZWILCH&quot;      </code></pre>
</div>
</div>
<div id="osteogenic-clusters-res0.75" class="section level2">
<h2>Osteogenic Clusters (res=0.75)</h2>
</div>
<div id="osteogenic-clusters-res0.50-time-2-only" class="section level2">
<h2>Osteogenic Clusters (res=0.50) (Time 2 only)</h2>
</div>
<div id="osteogenic-clusters-res0.75-time-2-only" class="section level2">
<h2>Osteogenic Clusters (res=0.75) (Time 2 only)</h2>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-4" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<p>Note: Too few DE genes shared across pre+obl+emb+min and obl+emb+min, so these were not examined further. Instead, DE genes shared across obl+emb+min+mat and emb+min+mat were considered.</p>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-46-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>         preosteoblast   osteoblast embedding.osteoblast
TSPAN6    0.0006489377 0.0006960817         0.0006822098
DPM1      0.0035664664 0.0035585244         0.0034761267
SCYL3     0.9753720890 0.9747084584         0.9747431673
C1orf112  0.0718263741 0.0726219490         0.0716897071
CFH       0.9999999006 0.9999998968         0.9999961486
FUCA2     0.4989726432 0.4988131169         0.4994976697
         mineralizing.osteoblast maturing.osteocyte
TSPAN6               0.001299618       0.0006609366
DPM1                 0.004570101       0.0035415208
SCYL3                0.975044765       0.9747888069
C1orf112             0.082613472       0.0716748727
CFH                  0.999999672       0.9999970634
FUCA2                0.500642294       0.4998680344</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         preosteoblast osteoblast embedding.osteoblast
TSPAN6               0          0                    0
DPM1                 0          0                    0
SCYL3                1          1                    1
C1orf112             0          0                    0
CFH                  1          1                    1
FUCA2                0          0                    0
         mineralizing.osteoblast maturing.osteocyte
TSPAN6                         0                  0
DPM1                           0                  0
SCYL3                          1                  1
C1orf112                       0                  0
CFH                            1                  1
FUCA2                          0                  0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;preosteoblast&quot;,length(which(dataUpset$preosteoblast==1))),
                                       rep(&quot;osteoblast&quot;,length(which(dataUpset$osteoblast==1))),
                                       rep(&quot;embedding.osteoblast&quot;,length(which(dataUpset$embedding.osteoblast==1))),
                                       rep(&quot;mineralizing.osteoblast&quot;,length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep(&quot;maturing.osteocyte&quot;,length(which(dataUpset$maturing.osteocyte==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),&quot;preosteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),&quot;osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),&quot;embedding.osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),&quot;mineralizing.osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),&quot;maturing.osteocyte&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                               labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                    3287                     3287                     3289 
mineralizing\nosteoblast      maturing\nosteocyte 
                    3299                     3287 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-47-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-47-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-47-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Save DE gene lists
write.csv(rownames(dataUpset),&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                    dataUpset$osteoblast==1 &amp;
                                    dataUpset$embedding.osteoblast==1 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                    dataUpset$osteoblast==1 &amp;
                                    dataUpset$embedding.osteoblast==1 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oemm.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                    dataUpset$osteoblast==1 &amp;
                                    dataUpset$embedding.osteoblast==1 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oem.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv&quot;)</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Osteogenic Ad Hoc Classifications
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across pre+obl+emb+min
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in pre+obl+emb+min&quot;)

#Shared across obl+emb+min+mat
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in obl+emb+min+mat&quot;)

#Shared across emb+min
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in emb+min&quot;)

#Unique to Mineralizing Osteoblasts
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Unique to Min&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across pre+obl+emb+min
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in\nPre/Obl/Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across obl+emb+min+mat
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in\nObl/Emb/Min/Mat&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across emb+min
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to mineralizing osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to\nMineralizing Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across all ostadhoc classifications
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All preosteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Preostoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All embedding osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Embedding Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All mineralizing osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Mineralizing Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All maturing osteocytes
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Maturing Osteocytes&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- &quot;Shared in All&quot;
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.004)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- &quot;Shared in All&quot;
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.001)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-52-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in\nPre/Obl/Emb/Min&quot;,
               &quot;Shared in\nObl/Emb/Min/Mat&quot;,
               &quot;Shared in Emb/Min&quot;,
               &quot;Unique to\nMineralizing Osteoblasts&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.004)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-53-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in\nPre/Obl/Emb/Min&quot;,
               &quot;Shared in\nObl/Emb/Min/Mat&quot;,
               &quot;Shared in Emb/Min&quot;,
               &quot;Unique to\nMineralizing Osteoblasts&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.001)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-53-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Preostoblasts&quot;,
               &quot;All Ostoblasts&quot;,
               &quot;All Embedding Osteoblasts&quot;,
               &quot;All Mineralizing Osteoblasts&quot;,
               &quot;All Maturing Osteocytes&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Preostoblasts&quot;,
               &quot;All Ostoblasts&quot;,
               &quot;All Embedding Osteoblasts&quot;,
               &quot;All Mineralizing Osteoblasts&quot;,
               &quot;All Maturing Osteocytes&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-54-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- unique(DEenrich[which(DEenrich$p.value&lt;0.1),c(1:3,7,9)])
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                         Source.DEGenes
133                       Shared in All
139                   All Preostoblasts
140                     All Osteoblasts
143             All Maturing Osteocytes
141           All Embedding Osteoblasts
142        All Mineralizing Osteoblasts
111                       Shared in All
117                   All Preostoblasts
118                     All Osteoblasts
121             All Maturing Osteocytes
119           All Embedding Osteoblasts
120        All Mineralizing Osteoblasts
192 Unique to\nMineralizing Osteoblasts
78                        Shared in All
56                        Shared in All
84                    All Preostoblasts
85                      All Osteoblasts
88              All Maturing Osteocytes
62                    All Preostoblasts
63                      All Osteoblasts
66              All Maturing Osteocytes
86            All Embedding Osteoblasts
64            All Embedding Osteoblasts
65         All Mineralizing Osteoblasts
87         All Mineralizing Osteoblasts
123          Shared in\nPre/Obl/Emb/Min
                  Source.GenesOfInterest DE.Interest    p.value
133                 Gokhman2020-DMR.bone          89 0.01297356
139                 Gokhman2020-DMR.bone          89 0.01319012
140                 Gokhman2020-DMR.bone          89 0.01319012
143                 Gokhman2020-DMR.bone          89 0.01319012
141                 Gokhman2020-DMR.bone          89 0.01340974
142                 Gokhman2020-DMR.bone          89 0.01455498
111 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03407072
117 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03470621
118 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03470621
121 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03470621
119 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03535125
120 Gokhman2021-DE.Parental.CNCCs-Unique         139 0.03872298
192                   Morris2020-GWAS.OP           2 0.07129261
78              Blake2020-DE.lung-Unique          64 0.07864350
56            Blake2020-DE.kidney-Unique          53 0.07889256
84              Blake2020-DE.lung-Unique          64 0.07946087
85              Blake2020-DE.lung-Unique          64 0.07946087
88              Blake2020-DE.lung-Unique          64 0.07946087
62            Blake2020-DE.kidney-Unique          53 0.07963180
63            Blake2020-DE.kidney-Unique          53 0.07963180
66            Blake2020-DE.kidney-Unique          53 0.07963180
86              Blake2020-DE.lung-Unique          64 0.08028445
64            Blake2020-DE.kidney-Unique          53 0.08037609
65            Blake2020-DE.kidney-Unique          53 0.08417383
87              Blake2020-DE.lung-Unique          64 0.08449636
123   Gokhman2021-DE.Hybrid.CNCCs-Unique           1 0.08998032
       GeneRatio
133 2.146772e-04
139 2.146772e-04
140 2.146772e-04
143 2.146772e-04
141 2.146772e-04
142 2.146772e-04
111 3.352823e-04
117 3.352823e-04
118 3.352823e-04
121 3.352823e-04
119 3.352823e-04
120 3.352823e-04
192 4.824206e-06
78  1.543746e-04
56  1.278415e-04
84  1.543746e-04
85  1.543746e-04
88  1.543746e-04
62  1.278415e-04
63  1.278415e-04
66  1.278415e-04
86  1.543746e-04
64  1.278415e-04
65  1.278415e-04
87  1.543746e-04
123 2.412103e-06</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
<p>Determine which skeletal-phenotype-related genes overlap with DE genes:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print(&quot;Shared across all ostadhoc classifications, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print(&quot;Shared across pre+obl+emb+min, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across obl+emb+min, tested in all classifications
  print(&quot;Shared across obl+emb+min, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to preosteoblasts, tested in all classifications
  print(&quot;Unique to preosteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to osteoblasts, tested in all classifications
  print(&quot;Unique to osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to embedding osteoblasts, tested in all classifications
  print(&quot;Unique to embedding osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print(&quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to maturing osteocytes, tested in all classifications
  print(&quot;Unique to maturing osteocytes, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))

}</code></pre>
<pre><code>[1] &quot;Hsu2019-GWAS.hip.geometry&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
[1] &quot;PPP6R3&quot;
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Styrkarsdottir2019-GWAS.bone.area&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
[1] &quot;COL11A1&quot; &quot;SOX9&quot;    &quot;CTDSP2&quot; 
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Yengo2018-GWAS.height&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
  [1] &quot;PLXND1&quot;     &quot;PHTF2&quot;      &quot;GAB2&quot;       &quot;SH3YL1&quot;     &quot;HDAC9&quot;     
  [6] &quot;ADAMTS6&quot;    &quot;LTBP1&quot;      &quot;DKK3&quot;       &quot;SDCCAG8&quot;    &quot;CUL1&quot;      
 [11] &quot;CYFIP2&quot;     &quot;USP36&quot;      &quot;PARP12&quot;     &quot;COL11A1&quot;    &quot;BICRA&quot;     
 [16] &quot;MYO9A&quot;      &quot;PFKP&quot;       &quot;NAV3&quot;       &quot;TGFBR3&quot;     &quot;ARHGAP10&quot;  
 [21] &quot;P4HA2&quot;      &quot;IGF2BP2&quot;    &quot;EED&quot;        &quot;MARK3&quot;      &quot;FOSL2&quot;     
 [26] &quot;ICAM3&quot;      &quot;POLD3&quot;      &quot;CAPZB&quot;      &quot;FGFR1&quot;      &quot;TULP3&quot;     
 [31] &quot;PPP2R5C&quot;    &quot;DLGAP4&quot;     &quot;NCOA1&quot;      &quot;CPNE3&quot;      &quot;NOX4&quot;      
 [36] &quot;PGS1&quot;       &quot;GNAS&quot;       &quot;RFX2&quot;       &quot;DOCK9&quot;      &quot;C20orf194&quot; 
 [41] &quot;KIZ&quot;        &quot;FUS&quot;        &quot;P3H2&quot;       &quot;PITPNM2&quot;    &quot;G2E3&quot;      
 [46] &quot;PHGDH&quot;      &quot;TGFB2&quot;      &quot;FKBP5&quot;      &quot;EFHC1&quot;      &quot;HIRA&quot;      
 [51] &quot;PIK3IP1&quot;    &quot;HSCB&quot;       &quot;MYH9&quot;       &quot;KHNYN&quot;      &quot;NFATC4&quot;    
 [56] &quot;PXMP4&quot;      &quot;NALCN&quot;      &quot;CAB39L&quot;     &quot;AXIN1&quot;      &quot;CRISPLD2&quot;  
 [61] &quot;TRPS1&quot;      &quot;PLEKHA4&quot;    &quot;ELL&quot;        &quot;CDK6&quot;       &quot;GLCCI1&quot;    
 [66] &quot;GLI3&quot;       &quot;KANK1&quot;      &quot;FUBP3&quot;      &quot;GLIS3&quot;      &quot;SH3PXD2A&quot;  
 [71] &quot;ZMIZ1&quot;      &quot;PBLD&quot;       &quot;UTP6&quot;       &quot;EZH1&quot;       &quot;FAM20A&quot;    
 [76] &quot;KLF3&quot;       &quot;MTCH2&quot;      &quot;CCND1&quot;      &quot;SLC38A1&quot;    &quot;UST&quot;       
 [81] &quot;TFEB&quot;       &quot;CCND3&quot;      &quot;GMDS&quot;       &quot;FGF1&quot;       &quot;ERGIC1&quot;    
 [86] &quot;STC2&quot;       &quot;TFG&quot;        &quot;CBLB&quot;       &quot;SCAP&quot;       &quot;ACTR3&quot;     
 [91] &quot;RTN4&quot;       &quot;EFEMP1&quot;     &quot;FN1&quot;        &quot;GLS&quot;        &quot;UXS1&quot;      
 [96] &quot;DHCR24&quot;     &quot;CAPZA1&quot;     &quot;MFAP2&quot;      &quot;SLC2A1&quot;     &quot;STAG1&quot;     
[101] &quot;DDX59&quot;      &quot;FOXO3&quot;      &quot;NPC2&quot;       &quot;BCL11A&quot;     &quot;HOXB3&quot;     
[106] &quot;WASHC3&quot;     &quot;METTL8&quot;     &quot;EXOSC9&quot;     &quot;CDKN1A&quot;     &quot;POLR1B&quot;    
[111] &quot;BMP2&quot;       &quot;IRF3&quot;       &quot;HIVEP3&quot;     &quot;FGFRL1&quot;     &quot;MKLN1&quot;     
[116] &quot;CALU&quot;       &quot;LOXL1&quot;      &quot;TXNDC17&quot;    &quot;PXDN&quot;       &quot;SYNE1&quot;     
[121] &quot;NFATC1&quot;     &quot;RARA&quot;       &quot;CHD1L&quot;      &quot;MATN3&quot;      &quot;ZSWIM3&quot;    
[126] &quot;IPO8&quot;       &quot;RRAS2&quot;      &quot;LOXL2&quot;      &quot;CTIF&quot;       &quot;CCNH&quot;      
[131] &quot;SOX5&quot;       &quot;GGACT&quot;      &quot;NREP&quot;       &quot;TBX3&quot;       &quot;SRPK2&quot;     
[136] &quot;DYSF&quot;       &quot;USP37&quot;      &quot;ALDH1L2&quot;    &quot;FLNB&quot;       &quot;PHF11&quot;     
[141] &quot;IGF2BP3&quot;    &quot;CALCOCO2&quot;   &quot;CDK5RAP2&quot;   &quot;TLN1&quot;       &quot;ZSCAN9&quot;    
[146] &quot;YAP1&quot;       &quot;HAUS2&quot;      &quot;SEMA6D&quot;     &quot;TET1&quot;       &quot;PARP9&quot;     
[151] &quot;ARHGAP24&quot;   &quot;FAM13A&quot;     &quot;FGF2&quot;       &quot;SLC39A8&quot;    &quot;RGS3&quot;      
[156] &quot;ETV6&quot;       &quot;SLC7A1&quot;     &quot;PELI2&quot;      &quot;TLE3&quot;       &quot;PARN&quot;      
[161] &quot;TXNL4B&quot;     &quot;CDH11&quot;      &quot;RANBP10&quot;    &quot;SSH2&quot;       &quot;GAREM1&quot;    
[166] &quot;SLC16A3&quot;    &quot;NFIC&quot;       &quot;URB1&quot;       &quot;SAE1&quot;       &quot;MYO1F&quot;     
[171] &quot;PTPRF&quot;      &quot;TUFT1&quot;      &quot;TTC13&quot;      &quot;CNIH3&quot;      &quot;PLEKHA6&quot;   
[176] &quot;VPS54&quot;      &quot;ASXL2&quot;      &quot;PTPRG&quot;      &quot;BOC&quot;        &quot;PDGFC&quot;     
[181] &quot;PIK3R1&quot;     &quot;PAM&quot;        &quot;TRIM41&quot;     &quot;ABT1&quot;       &quot;DAAM2&quot;     
[186] &quot;CREB5&quot;      &quot;NIPSNAP2&quot;   &quot;CSGALNACT1&quot; &quot;GSN&quot;        &quot;TCF7L2&quot;    
[191] &quot;ADAM12&quot;     &quot;PAMR1&quot;      &quot;SERPINH1&quot;   &quot;LATS2&quot;      &quot;VTI1A&quot;     
[196] &quot;ZNF827&quot;     &quot;FLI1&quot;       &quot;BEND6&quot;      &quot;PTPN14&quot;     &quot;MBNL1&quot;     
[201] &quot;BCL2L11&quot;    &quot;ASAP1&quot;      &quot;CMIP&quot;       &quot;DDAH1&quot;      &quot;SORBS2&quot;    
[206] &quot;MAP3K7CL&quot;   &quot;SKI&quot;        &quot;COLEC12&quot;    &quot;TAOK1&quot;      &quot;ADAR&quot;      
[211] &quot;ADCY9&quot;      &quot;PLPP3&quot;      &quot;VANGL2&quot;     &quot;FRZB&quot;       &quot;ACTG2&quot;     
[216] &quot;COL6A3&quot;     &quot;PCOLCE2&quot;    &quot;TOPBP1&quot;     &quot;SLBP&quot;       &quot;CREBRF&quot;    
[221] &quot;ZCCHC24&quot;    &quot;NUDT5&quot;      &quot;E2F7&quot;       &quot;XRRA1&quot;      &quot;WDR81&quot;     
[226] &quot;NDUFV1&quot;     &quot;SPINDOC&quot;    &quot;UBTD2&quot;      &quot;ABHD15&quot;     &quot;ZNF608&quot;    
[231] &quot;ROR2&quot;       &quot;LUZP1&quot;      &quot;CENPX&quot;      &quot;TPST1&quot;      &quot;ALCAM&quot;     
[236] &quot;SMAD1&quot;      &quot;ELOVL6&quot;     &quot;CYTL1&quot;      &quot;NUDT6&quot;      &quot;FEZ2&quot;      
[241] &quot;PRKCE&quot;      &quot;PDE7B&quot;      &quot;TLN2&quot;       &quot;SUCLG2&quot;     &quot;FGGY&quot;      
[246] &quot;AFF1&quot;       &quot;NBEA&quot;       &quot;SH3RF3&quot;     &quot;RHOD&quot;       &quot;EHBP1L1&quot;   
[251] &quot;XXYLT1&quot;     &quot;CTSF&quot;       &quot;MSRB3&quot;      &quot;MRPL11&quot;     &quot;RSRC1&quot;     
[256] &quot;PSMD2&quot;      &quot;SLC38A9&quot;    &quot;LCORL&quot;      &quot;PLEC&quot;       &quot;NT5DC1&quot;    
[261] &quot;RFLNA&quot;      &quot;ARID3B&quot;     &quot;ZNF784&quot;     &quot;CCDC66&quot;     &quot;PITPNB&quot;    
[266] &quot;POLR2A&quot;     &quot;AATK&quot;       &quot;PLAG1&quot;      &quot;ZBTB20&quot;     &quot;IST1&quot;      
[271] &quot;RAD51B&quot;     &quot;PAPPA&quot;      &quot;CEP63&quot;      &quot;HMCES&quot;      &quot;KMT5A&quot;     
[276] &quot;ST6GALNAC3&quot; &quot;NELFA&quot;      &quot;PRKG1&quot;      &quot;UBE2L3&quot;     &quot;PTCH1&quot;     
[281] &quot;THBS2&quot;      &quot;PPARA&quot;      &quot;IFT140&quot;     &quot;COL14A1&quot;    &quot;TDRD7&quot;     
[286] &quot;ZNF471&quot;     &quot;SUPT3H&quot;     &quot;AFAP1&quot;      &quot;TLE1&quot;       &quot;SLC22A4&quot;   
[291] &quot;VEPH1&quot;      &quot;CDC42SE1&quot;   &quot;FAM114A1&quot;   &quot;MYO1C&quot;      &quot;GRK6&quot;      
[296] &quot;MAN1A2&quot;     &quot;HSD17B11&quot;   &quot;NCOA6&quot;      &quot;PNP&quot;        &quot;GRK5&quot;      
[301] &quot;FNIP1&quot;      &quot;PLEKHM1&quot;    &quot;C16orf95&quot;   &quot;UHRF1&quot;     
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
[1] &quot;SRC&quot;
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Steinberg2020-GWAS.OA&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
 [1] &quot;ACSL4&quot;      &quot;ARSA&quot;       &quot;ATAD3B&quot;     &quot;ATP6AP1&quot;    &quot;BCL2L13&quot;   
 [6] &quot;C1R&quot;        &quot;CAPG&quot;       &quot;CASP4&quot;      &quot;CCDC86&quot;     &quot;CCPG1&quot;     
[11] &quot;CFH&quot;        &quot;COL4A1&quot;     &quot;COL4A2&quot;     &quot;COPG1&quot;      &quot;CYLD&quot;      
[16] &quot;DCPS&quot;       &quot;DDAH1&quot;      &quot;DGKA&quot;       &quot;EFEMP1&quot;     &quot;ENO1&quot;      
[21] &quot;EPS8L2&quot;     &quot;EXOSC4&quot;     &quot;FABP5&quot;      &quot;FGF1&quot;       &quot;FN1&quot;       
[26] &quot;FRZB&quot;       &quot;FSTL1&quot;      &quot;GADD45GIP1&quot; &quot;GALE&quot;       &quot;GGCT&quot;      
[31] &quot;GGH&quot;        &quot;GNPDA1&quot;     &quot;GPT2&quot;       &quot;GULP1&quot;      &quot;IGFBP7&quot;    
[36] &quot;IRAK4&quot;      &quot;IRS1&quot;       &quot;LIMA1&quot;      &quot;MAP1A&quot;      &quot;MAP1B&quot;     
[41] &quot;METTL7A&quot;    &quot;MGST1&quot;      &quot;MPDZ&quot;       &quot;MPP6&quot;       &quot;MPRIP&quot;     
[46] &quot;MRPS26&quot;     &quot;MRPS27&quot;     &quot;MYOF&quot;       &quot;NAMPT&quot;      &quot;NDRG1&quot;     
[51] &quot;NDUFB8&quot;     &quot;NME1&quot;       &quot;NOVA1&quot;      &quot;NPC2&quot;       &quot;NQO1&quot;      
[56] &quot;NRBF2&quot;      &quot;P4HA3&quot;      &quot;PBXIP1&quot;     &quot;PDLIM4&quot;     &quot;PEX5&quot;      
[61] &quot;PFKL&quot;       &quot;PGAM5&quot;      &quot;PGRMC1&quot;     &quot;PID1&quot;       &quot;PNP&quot;       
[66] &quot;POLR2C&quot;     &quot;POSTN&quot;      &quot;PPFIA1&quot;     &quot;PPIA&quot;       &quot;PYGL&quot;      
[71] &quot;RANBP1&quot;     &quot;RARRES2&quot;    &quot;RCAN1&quot;      &quot;RFC3&quot;       &quot;RHOC&quot;      
[76] &quot;RRAGC&quot;      &quot;RRAS2&quot;      &quot;RRP9&quot;       &quot;SDR39U1&quot;    &quot;SEC31A&quot;    
[81] &quot;SERPINB8&quot;   &quot;SH3KBP1&quot;    &quot;SMS&quot;        &quot;SMYD5&quot;      &quot;SNTB1&quot;     
[86] &quot;SORBS1&quot;     &quot;SSR3&quot;       &quot;STK4&quot;       &quot;STRAP&quot;      &quot;TBL2&quot;      
[91] &quot;THBS3&quot;      &quot;THOC3&quot;      &quot;TNFAIP6&quot;    &quot;TRAFD1&quot;     &quot;TRIM47&quot;    
[96] &quot;UCHL1&quot;      &quot;VAV2&quot;       &quot;YARS2&quot;     
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Morris2020-GWAS.OP&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
  [1] &quot;PEX14&quot;     &quot;CAPZB&quot;     &quot;ALPL&quot;      &quot;CLIC4&quot;     &quot;HIVEP3&quot;   
  [6] &quot;PLPP3&quot;     &quot;NEGR1&quot;     &quot;PKN2&quot;      &quot;TGFBR3&quot;    &quot;COL11A1&quot;  
 [11] &quot;CTTNBP2NL&quot; &quot;THBS3&quot;     &quot;HHAT&quot;      &quot;TGFB2&quot;     &quot;PSEN2&quot;    
 [16] &quot;ASXL2&quot;     &quot;SLC8A1&quot;    &quot;PKDCC&quot;     &quot;ZFP36L2&quot;   &quot;PRKCE&quot;    
 [21] &quot;MSH6&quot;      &quot;BCL11A&quot;    &quot;ACTR2&quot;     &quot;MEIS1&quot;     &quot;SH3RF3&quot;   
 [26] &quot;BCL2L11&quot;   &quot;MERTK&quot;     &quot;CYBRD1&quot;    &quot;UBE2E3&quot;    &quot;FRZB&quot;     
 [31] &quot;HECW2&quot;     &quot;FZD7&quot;      &quot;IRS1&quot;      &quot;COL6A3&quot;    &quot;MLPH&quot;     
 [36] &quot;HDAC4&quot;     &quot;PPARG&quot;     &quot;TMEM43&quot;    &quot;TGFBR2&quot;    &quot;PRKAR2A&quot;  
 [41] &quot;ITGB5&quot;     &quot;PCOLCE2&quot;   &quot;MBNL1&quot;     &quot;IGF2BP2&quot;   &quot;DLG1&quot;     
 [46] &quot;FGFRL1&quot;    &quot;LCORL&quot;     &quot;ANAPC4&quot;    &quot;ARHGAP24&quot;  &quot;AFF1&quot;     
 [51] &quot;ZNF827&quot;    &quot;PDGFC&quot;     &quot;MSMO1&quot;     &quot;FAT1&quot;      &quot;ITGA1&quot;    
 [56] &quot;SLC38A9&quot;   &quot;ADAMTS6&quot;   &quot;FAM174A&quot;   &quot;FOXC1&quot;     &quot;GMDS&quot;     
 [61] &quot;HLA-A&quot;     &quot;DAAM2&quot;     &quot;VEGFA&quot;     &quot;SUPT3H&quot;    &quot;DST&quot;      
 [66] &quot;GJA1&quot;      &quot;TRMT11&quot;    &quot;PDE7B&quot;     &quot;SYNE1&quot;     &quot;CREB5&quot;    
 [71] &quot;GTF2I&quot;     &quot;SEMA3A&quot;    &quot;CDK6&quot;      &quot;TSPAN12&quot;   &quot;MKLN1&quot;    
 [76] &quot;KIAA1549&quot;  &quot;PLAG1&quot;     &quot;ZBTB10&quot;    &quot;MMP16&quot;     &quot;TRPS1&quot;    
 [81] &quot;MPDZ&quot;      &quot;CNTLN&quot;     &quot;KLF9&quot;      &quot;PTCH1&quot;     &quot;BICC1&quot;    
 [86] &quot;ZCCHC24&quot;   &quot;VTI1A&quot;     &quot;TCF7L2&quot;    &quot;ADAM12&quot;    &quot;MGMT&quot;     
 [91] &quot;SOX6&quot;      &quot;CCDC34&quot;    &quot;PCNX3&quot;     &quot;FAM168A&quot;   &quot;PRSS23&quot;   
 [96] &quot;YAP1&quot;      &quot;PTS&quot;       &quot;CLMP&quot;      &quot;FLI1&quot;      &quot;EMP1&quot;     
[101] &quot;SOX5&quot;      &quot;SSPN&quot;      &quot;METTL7A&quot;   &quot;ATF7&quot;      &quot;MSRB3&quot;    
[106] &quot;NAV3&quot;      &quot;DCN&quot;       &quot;TBX3&quot;      &quot;SPATA13&quot;   &quot;FOXO1&quot;    
[111] &quot;DHRS12&quot;    &quot;KLF12&quot;     &quot;DOCK9&quot;     &quot;SAV1&quot;      &quot;BMP4&quot;     
[116] &quot;RHOJ&quot;      &quot;ZBTB25&quot;    &quot;RAD51B&quot;    &quot;MARK3&quot;     &quot;SEMA6D&quot;   
[121] &quot;TLE3&quot;      &quot;PPCDC&quot;     &quot;FAM234A&quot;   &quot;AXIN1&quot;     &quot;RAB11FIP3&quot;
[126] &quot;ADCY9&quot;     &quot;EMP2&quot;      &quot;CYLD&quot;      &quot;SALL1&quot;     &quot;CMIP&quot;     
[131] &quot;ZNF469&quot;    &quot;POLR2A&quot;    &quot;GAS7&quot;      &quot;HS3ST3B1&quot;  &quot;NUFIP2&quot;   
[136] &quot;STAT5A&quot;    &quot;HDAC5&quot;     &quot;DCAKD&quot;     &quot;ACSF2&quot;     &quot;PGS1&quot;     
[141] &quot;USP36&quot;     &quot;RFNG&quot;      &quot;VAPA&quot;      &quot;FHOD3&quot;     &quot;NFATC1&quot;   
[146] &quot;PLPPR3&quot;    &quot;NFIC&quot;      &quot;RFX2&quot;      &quot;ARRDC2&quot;    &quot;RHPN2&quot;    
[151] &quot;GPATCH1&quot;   &quot;SLC27A5&quot;   &quot;BMP2&quot;      &quot;SLX4IP&quot;    &quot;UBE2L3&quot;   
[156] &quot;LIF&quot;       &quot;MYH9&quot;     
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
[1] &quot;EEF2K&quot;  &quot;PPP6R2&quot;
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)</code></pre>
</div>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-5" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
</div>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-flexible-time-2-only" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)</h2>
<div id="examine-numbers-and-overlap-of-cormotif-de-genes-functionalgeneset-enrichment-of-de-genes-6" class="section level3">
<h3>Examine Numbers and Overlap of Cormotif DE Genes &amp; Functional/GeneSet Enrichment of DE Genes</h3>
<p>Note: Too few DE genes shared across obl+emb+min+mat, so these were not examined further. Instead, DE genes shared across pre+obl+emb+min+mat and unique to min were considered.</p>
<pre class="r"><code>#load data
cormotifData &lt;- readRDS(&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds&quot;)

#plot BIC/AIC comparisons
plotIC(cormotifData)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot the probability that each motif is differentially expressed in each comparison
HAtop &lt;- HeatmapAnnotation(AdHoc=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                           col=list(AdHoc=c(&quot;preosteoblast&quot;=&quot;lightblue1&quot;,
                                            &quot;osteoblast&quot;=&quot;deepskyblue&quot;,
                                            &quot;embedding.osteoblast&quot;=&quot;dodgerblue2&quot;,
                                            &quot;mineralizing.osteoblast&quot;=&quot;royalblue3&quot;,
                                            &quot;maturing.osteocyte&quot;=&quot;royalblue4&quot;)),
                           gp=gpar(col=&quot;black&quot;),
                           gap=unit(5,&quot;points&quot;),
                           annotation_name_side=&quot;left&quot;)
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,&quot;cm&quot;),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name=&quot;DE Prob.&quot;,
        rect_gp=gpar(col=&quot;black&quot;,lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title=&quot;Comparisons across Osteogenic Ad Hoc&quot;,
        row_title=&quot;Motif&quot;,
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-58-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#obtain DE pattern at 0.65 posterior distribution cutoff and reformat for plotting
head(cormotifData$bestmotif$p.post)</code></pre>
<pre><code>         preosteoblast   osteoblast embedding.osteoblast
TSPAN6    0.0002163807 0.0002545543         0.0002100975
DPM1      0.0055254203 0.0055215923         0.0054020513
SCYL3     0.5664610004 0.5645863463         0.5639585951
C1orf112  0.0466909326 0.0471770929         0.0465462044
CFH       0.9999999595 0.9999999410         0.9999973886
FUCA2     0.5159252541 0.5159704985         0.5162887683
         mineralizing.osteoblast maturing.osteocyte
TSPAN6               0.002383143       0.0001961325
DPM1                 0.016335014       0.0051130986
SCYL3                0.568043224       0.5447572417
C1orf112             0.080970985       0.0457752101
CFH                  0.999999895       0.9999949862
FUCA2                0.520729816       0.5155295263</code></pre>
<pre class="r"><code>dataUpset &lt;- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset&gt;0.65] &lt;- 1
dataUpset[dataUpset&lt;=0.65] &lt;- 0
head(dataUpset)</code></pre>
<pre><code>         preosteoblast osteoblast embedding.osteoblast
TSPAN6               0          0                    0
DPM1                 0          0                    0
SCYL3                0          0                    0
C1orf112             0          0                    0
CFH                  1          1                    1
FUCA2                0          0                    0
         mineralizing.osteoblast maturing.osteocyte
TSPAN6                         0                  0
DPM1                           0                  0
SCYL3                          0                  0
C1orf112                       0                  0
CFH                            1                  1
FUCA2                          0                  0</code></pre>
<pre class="r"><code>dataPlot &lt;- data.frame(&quot;cell.subset&quot;=c(rep(&quot;preosteoblast&quot;,length(which(dataUpset$preosteoblast==1))),
                                       rep(&quot;osteoblast&quot;,length(which(dataUpset$osteoblast==1))),
                                       rep(&quot;embedding.osteoblast&quot;,length(which(dataUpset$embedding.osteoblast==1))),
                                       rep(&quot;mineralizing.osteoblast&quot;,length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep(&quot;maturing.osteocyte&quot;,length(which(dataUpset$maturing.osteocyte==1)))),
                       &quot;gene&quot;=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       &quot;post.p&quot;=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),&quot;preosteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),&quot;osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),&quot;embedding.osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),&quot;mineralizing.osteoblast&quot;],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),&quot;maturing.osteocyte&quot;]),
                       row.names=NULL)

dataPlot$cell.subset &lt;- factor(dataPlot$cell.subset,
                               levels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                               labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding\nosteoblast&quot;,&quot;mineralizing\nosteoblast&quot;,&quot;maturing\nosteocyte&quot;))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab(&#39;Number of DE Genes&#39;) +
  xlab(&#39;&#39;) +
  scale_fill_manual(name=&quot;Osteogenic Stage&quot;,
                    labels=c(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding osteoblast&quot;,&quot;mineralizing osteoblast&quot;,&quot;maturing osteocyte&quot;),
                    values=c(&quot;lightblue1&quot;,&quot;deepskyblue&quot;,&quot;dodgerblue2&quot;,&quot;royalblue3&quot;,&quot;royalblue4&quot;)) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#total number of DE genes identified
table(dataPlot$cell.subset)</code></pre>
<pre><code>
           preosteoblast               osteoblast    embedding\nosteoblast 
                    2857                     2857                     2856 
mineralizing\nosteoblast      maturing\nosteocyte 
                    2896                     2827 </code></pre>
<pre class="r"><code>#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-59-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-59-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list(&quot;preosteoblast&quot;),
                         list(&quot;osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;),
                         list(&quot;mineralizing.osteoblast&quot;),
                         list(&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;),
                         list(&quot;preosteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;),
                         list(&quot;preosteoblast&quot;,&quot;osteoblast&quot;,&quot;embedding.osteoblast&quot;,&quot;mineralizing.osteoblast&quot;,&quot;maturing.osteocyte&quot;)),
      text.scale=2)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-59-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Save DE gene lists
write.csv(rownames(dataUpset),&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                    dataUpset$osteoblast==1 &amp;
                                    dataUpset$embedding.osteoblast==1 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                    dataUpset$osteoblast==1 &amp;
                                    dataUpset$embedding.osteoblast==1 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.poem.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                    dataUpset$osteoblast==0 &amp;
                                    dataUpset$embedding.osteoblast==0 &amp;
                                    dataUpset$mineralizing.osteoblast==1 &amp;
                                    dataUpset$maturing.osteocyte==0)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.oad.min.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv&quot;)
write.csv(rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)],
          &quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv&quot;)</code></pre>
<p>Assess GO functional enrichment of interesting DE genes.</p>
<pre class="r"><code>#Shared across Osteogenic Ad Hoc Classifications
subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
totGenes &lt;- rownames(dataUpset)
goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title=&quot;Shared in All&quot;)

#Shared across pre+obl+emb+min
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
#                                      dataUpset$osteoblast==1 &amp;
#                                      dataUpset$embedding.osteoblast==1 &amp;
#                                      dataUpset$mineralizing.osteoblast==1 &amp;
#                                      dataUpset$maturing.osteocyte==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in pre+obl+emb+min&quot;)

#Shared across obl+emb+min+mat
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
#                                      dataUpset$osteoblast==1 &amp;
#                                      dataUpset$embedding.osteoblast==1 &amp;
#                                      dataUpset$mineralizing.osteoblast==1 &amp;
#                                      dataUpset$maturing.osteocyte==1)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in obl+emb+min+mat&quot;)

#Shared across emb+min
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
#                                      dataUpset$osteoblast==0 &amp;
#                                      dataUpset$embedding.osteoblast==1 &amp;
#                                      dataUpset$mineralizing.osteoblast==1 &amp;
#                                      dataUpset$maturing.osteocyte==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Shared in emb+min&quot;)

#Unique to Mineralizing Osteoblasts
#subGenes &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
#                                      dataUpset$osteoblast==0 &amp;
#                                      dataUpset$embedding.osteoblast==0 &amp;
#                                      dataUpset$mineralizing.osteoblast==1 &amp;
#                                      dataUpset$maturing.osteocyte==0)]
#totGenes &lt;- rownames(dataUpset)
#goTot &lt;- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology=&quot;ALL&quot;)
#goTot@result$Description &lt;- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title=&quot;Unique to Min&quot;)</code></pre>
<p>Test for significant enrichment of external gene sets in DE gene sets using Fisher’s exact test. Make enrichment plot:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Blake2018-DE.iPSCs-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==1 &amp;
                                                        blake2018$DE.Day1==0 &amp;
                                                                blake2018$DE.Day2==0 &amp;
                                                                blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==1 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==1 &amp;
                                                                           blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list(&quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,blake2018$gene[which(blake2018$DE.Day0==0 &amp;
                                                                     blake2018$DE.Day1==0 &amp;
                                                                             blake2018$DE.Day2==0 &amp;
                                                                           blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list(&quot;Blake2020-DE.heart-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==1 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==0 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list(&quot;Blake2020-DE.kidney-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                         blake2020$DE.HvC_Kidney==1 &amp;
                                                         blake2020$DE.HvC_Liver==0 &amp;
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list(&quot;Blake2020-DE.liver-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                        blake2020$DE.HvC_Kidney==0 &amp;
                                                        blake2020$DE.HvC_Liver==1 &amp;
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list(&quot;Blake2020-DE.lung-Unique&quot;,blake2020$gene[which(blake2020$DE.HvC_Heart==0 &amp;
                                                       blake2020$DE.HvC_Kidney==0 &amp;
                                                       blake2020$DE.HvC_Liver==0 &amp;
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list(&quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list(&quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &amp;
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list(&quot;Gokhman2020-DMR.bone&quot;,gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich &lt;- data.frame(&quot;Source.DEGenes&quot;=character(),
                       &quot;Source.GenesOfInterest&quot;=character(),
                       &quot;DE.Interest&quot;=integer(),
                       &quot;NonDE.Interest&quot;=integer(),
                       &quot;DE.NotInterest&quot;=integer(),
                       &quot;NonDE.NotInterest&quot;=integer(),
                       &quot;p.value&quot;=double(),
                       &quot;odds.ratio&quot;=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across pre+obl+emb+min
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in\nPre/Obl/Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across obl+emb+min+mat
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in\nObl/Emb/Min/Mat&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across emb+min
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in Emb/Min&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to mineralizing osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Unique to\nMineralizing Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across all ostadhoc classifications
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;Shared in All&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All preosteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Preostoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All embedding osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Embedding Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All mineralizing osteoblasts
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Mineralizing Osteoblasts&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All maturing osteocytes
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes &lt;- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich &lt;- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource=&quot;All Maturing Osteocytes&quot;,
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes &lt;- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest &lt;- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio &lt;- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)</code></pre>
<pre class="r"><code>#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-63-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- &quot;Shared in All&quot;
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.004)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-64-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- &quot;Shared in All&quot;
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.001)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-64-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in\nPre/Obl/Emb/Min&quot;,
               &quot;Shared in\nObl/Emb/Min/Mat&quot;,
               &quot;Shared in Emb/Min&quot;,
               &quot;Unique to\nMineralizing Osteoblasts&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.004)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;Shared in\nPre/Obl/Emb/Min&quot;,
               &quot;Shared in\nObl/Emb/Min/Mat&quot;,
               &quot;Shared in Emb/Min&quot;,
               &quot;Unique to\nMineralizing Osteoblasts&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.001)) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-65-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Preostoblasts&quot;,
               &quot;All Ostoblasts&quot;,
               &quot;All Embedding Osteoblasts&quot;,
               &quot;All Mineralizing Osteoblasts&quot;,
               &quot;All Maturing Osteocytes&quot;)
tempSource &lt;- c(&quot;Blake2018-DE.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Parental.iPSCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.iPSCs-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day1-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day2-Unique&quot;,
                &quot;Blake2018-DE.iPSC-endoderm-day3-Unique&quot;,
                &quot;Blake2020-DE.heart-Unique&quot;,
                &quot;Blake2020-DE.kidney-Unique&quot;,
                &quot;Blake2020-DE.liver-Unique&quot;,
                &quot;Blake2020-DE.lung-Unique&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-66-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tempGenes &lt;- c(&quot;All Preostoblasts&quot;,
               &quot;All Ostoblasts&quot;,
               &quot;All Embedding Osteoblasts&quot;,
               &quot;All Mineralizing Osteoblasts&quot;,
               &quot;All Maturing Osteocytes&quot;)
tempSource &lt;- c(&quot;Gokhman2021-DE.Parental.CNCCs-Unique&quot;,
                &quot;Gokhman2021-DE.Hybrid.CNCCs-Unique&quot;,
                &quot;Gokhman2020-DMR.bone&quot;,
                &quot;Hsu2019-GWAS.hip.geometry&quot;,
                &quot;Styrkarsdottir2019-GWAS.bone.area&quot;,
                &quot;Yengo2018-GWAS.height&quot;,
                &quot;Steinberg2020-GWAS.OA&quot;,
                &quot;Morris2020-GWAS.OP&quot;)
index &lt;- which(DEenrich$Source.DEGenes %in% tempGenes &amp; DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option=&quot;C&quot;, limits=c(0,0.05)) +
          theme_bw() +
          labs(x=&quot;Gene Ratio&quot;, y=&quot;&quot;) +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)</code></pre>
<p><img src="figure/10x-plot-cormotif-cnts_v2.Rmd/unnamed-chunk-66-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Print p-values for significant enrichments:</p>
<pre class="r"><code>tbl &lt;- unique(DEenrich[which(DEenrich$p.value&lt;0.1),c(1:3,7,9)])
tbl[order(tbl$p.value),]</code></pre>
<pre><code>                         Source.DEGenes
122                       Shared in All
132             All Maturing Osteocytes
131        All Mineralizing Osteoblasts
141           All Embedding Osteoblasts
139                   All Preostoblasts
140                     All Osteoblasts
130           All Embedding Osteoblasts
128                   All Preostoblasts
129                     All Osteoblasts
142        All Mineralizing Osteoblasts
133                       Shared in All
143             All Maturing Osteocytes
159 Unique to\nMineralizing Osteoblasts
119           All Embedding Osteoblasts
117                   All Preostoblasts
118                     All Osteoblasts
120        All Mineralizing Osteoblasts
111                       Shared in All
121             All Maturing Osteocytes
112          Shared in\nPre/Obl/Emb/Min
64            All Embedding Osteoblasts
62                    All Preostoblasts
63                      All Osteoblasts
                  Source.GenesOfInterest DE.Interest     p.value
122   Gokhman2021-DE.Hybrid.CNCCs-Unique         287 0.006256676
132   Gokhman2021-DE.Hybrid.CNCCs-Unique         287 0.006380971
131   Gokhman2021-DE.Hybrid.CNCCs-Unique         293 0.006984314
141                 Gokhman2020-DMR.bone          81 0.007195806
139                 Gokhman2020-DMR.bone          81 0.007263642
140                 Gokhman2020-DMR.bone          81 0.007263642
130   Gokhman2021-DE.Hybrid.CNCCs-Unique         288 0.009062571
128   Gokhman2021-DE.Hybrid.CNCCs-Unique         288 0.009233971
129   Gokhman2021-DE.Hybrid.CNCCs-Unique         288 0.009233971
142                 Gokhman2020-DMR.bone          81 0.010377269
133                 Gokhman2020-DMR.bone          79 0.011724884
143                 Gokhman2020-DMR.bone          79 0.011829061
159    Styrkarsdottir2019-GWAS.bone.area           1 0.019798623
119 Gokhman2021-DE.Parental.CNCCs-Unique         124 0.029421614
117 Gokhman2021-DE.Parental.CNCCs-Unique         124 0.029715791
118 Gokhman2021-DE.Parental.CNCCs-Unique         124 0.029715791
120 Gokhman2021-DE.Parental.CNCCs-Unique         125 0.033905913
111 Gokhman2021-DE.Parental.CNCCs-Unique         121 0.046005506
121 Gokhman2021-DE.Parental.CNCCs-Unique         121 0.046430073
112 Gokhman2021-DE.Parental.CNCCs-Unique           3 0.085514972
64            Blake2020-DE.kidney-Unique          46 0.086765333
62            Blake2020-DE.kidney-Unique          46 0.087176198
63            Blake2020-DE.kidney-Unique          46 0.087176198
       GeneRatio
122 7.968238e-04
132 7.968238e-04
131 8.134821e-04
141 2.248876e-04
139 2.248876e-04
140 2.248876e-04
130 7.996002e-04
128 7.996002e-04
129 7.996002e-04
142 2.248876e-04
133 2.193348e-04
143 2.193348e-04
159 2.776390e-06
119 3.442723e-04
117 3.442723e-04
118 3.442723e-04
120 3.470487e-04
111 3.359431e-04
121 3.359431e-04
112 8.329169e-06
64  1.277139e-04
62  1.277139e-04
63  1.277139e-04</code></pre>
<p>Save enrichment results:</p>
<pre class="r"><code>#Save enrichment results
write.csv(DEenrich,&quot;./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.enrichment.csv&quot;)</code></pre>
<p>Determine which skeletal-phenotype-related genes overlap with DE genes:</p>
<pre class="r"><code>#Enrichment of external gene sets in DE gene sets

listsExternal &lt;- list(

  list(&quot;Hsu2019-GWAS.hip.geometry&quot;,hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list(&quot;Styrkarsdottir2019-GWAS.bone.area&quot;,styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list(&quot;Yengo2018-GWAS.height&quot;,yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list(&quot;Steinberg2020-GWAS.OA&quot;,steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list(&quot;Morris2020-GWAS.OP&quot;,morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print(&quot;Shared across all ostadhoc classifications, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print(&quot;Shared across pre+obl+emb+min, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across obl+emb+min, tested in all classifications
  print(&quot;Shared across obl+emb+min, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to preosteoblasts, tested in all classifications
  print(&quot;Unique to preosteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to osteoblasts, tested in all classifications
  print(&quot;Unique to osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==1 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to embedding osteoblasts, tested in all classifications
  print(&quot;Unique to embedding osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==1 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print(&quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==1 &amp;
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to maturing osteocytes, tested in all classifications
  print(&quot;Unique to maturing osteocytes, tested in all classifications&quot;)
  tempDE &lt;- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &amp;
                                      dataUpset$osteoblast==0 &amp;
                                      dataUpset$embedding.osteoblast==0 &amp;
                                      dataUpset$mineralizing.osteoblast==0 &amp;
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))

}</code></pre>
<pre><code>[1] &quot;Hsu2019-GWAS.hip.geometry&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
character(0)
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Styrkarsdottir2019-GWAS.bone.area&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
character(0)
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
[1] &quot;CTDSP2&quot;
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Yengo2018-GWAS.height&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
  [1] &quot;PLXND1&quot;     &quot;PHTF2&quot;      &quot;GAB2&quot;       &quot;SH3YL1&quot;     &quot;HDAC9&quot;     
  [6] &quot;ADAMTS6&quot;    &quot;DKK3&quot;       &quot;SDCCAG8&quot;    &quot;CUL1&quot;       &quot;CYFIP2&quot;    
 [11] &quot;USP36&quot;      &quot;BICRA&quot;      &quot;MYO9A&quot;      &quot;NAV3&quot;       &quot;TGFBR3&quot;    
 [16] &quot;ARHGAP10&quot;   &quot;P4HA2&quot;      &quot;IGF2BP2&quot;    &quot;EED&quot;        &quot;MARK3&quot;     
 [21] &quot;FOSL2&quot;      &quot;ICAM3&quot;      &quot;POLD3&quot;      &quot;CAPZB&quot;      &quot;FGFR1&quot;     
 [26] &quot;TULP3&quot;      &quot;PPP2R5C&quot;    &quot;NCOA1&quot;      &quot;CPNE3&quot;      &quot;NOX4&quot;      
 [31] &quot;PGS1&quot;       &quot;GNAS&quot;       &quot;RFX2&quot;       &quot;DOCK9&quot;      &quot;C20orf194&quot; 
 [36] &quot;KIF16B&quot;     &quot;FUS&quot;        &quot;P3H2&quot;       &quot;PITPNM2&quot;    &quot;G2E3&quot;      
 [41] &quot;PHGDH&quot;      &quot;TGFB2&quot;      &quot;FKBP5&quot;      &quot;EFHC1&quot;      &quot;HIRA&quot;      
 [46] &quot;PIK3IP1&quot;    &quot;HSCB&quot;       &quot;MYH9&quot;       &quot;KHNYN&quot;      &quot;NFATC4&quot;    
 [51] &quot;PXMP4&quot;      &quot;NALCN&quot;      &quot;CAB39L&quot;     &quot;AXIN1&quot;      &quot;CRISPLD2&quot;  
 [56] &quot;BCKDK&quot;      &quot;TRPS1&quot;      &quot;PLEKHA4&quot;    &quot;ELL&quot;        &quot;CDK6&quot;      
 [61] &quot;AHR&quot;        &quot;GLI3&quot;       &quot;KANK1&quot;      &quot;FUBP3&quot;      &quot;GLIS3&quot;     
 [66] &quot;SH3PXD2A&quot;   &quot;ZMIZ1&quot;      &quot;PBLD&quot;       &quot;EZH1&quot;       &quot;FAM20A&quot;    
 [71] &quot;KLF3&quot;       &quot;CCND1&quot;      &quot;SLC38A1&quot;    &quot;UST&quot;        &quot;TFEB&quot;      
 [76] &quot;CCND3&quot;      &quot;GMDS&quot;       &quot;FGF1&quot;       &quot;ERGIC1&quot;     &quot;STC2&quot;      
 [81] &quot;CBLB&quot;       &quot;SCAP&quot;       &quot;EFEMP1&quot;     &quot;FN1&quot;        &quot;GLS&quot;       
 [86] &quot;UXS1&quot;       &quot;DHCR24&quot;     &quot;MFAP2&quot;      &quot;SLC2A1&quot;     &quot;STAG1&quot;     
 [91] &quot;DDX59&quot;      &quot;FOXO3&quot;      &quot;NPC2&quot;       &quot;BCL11A&quot;     &quot;HOXB3&quot;     
 [96] &quot;WASHC3&quot;     &quot;METTL8&quot;     &quot;CDKN1A&quot;     &quot;POLR1B&quot;     &quot;BMP2&quot;      
[101] &quot;IRF3&quot;       &quot;HIVEP3&quot;     &quot;FGFRL1&quot;     &quot;CALU&quot;       &quot;PXDN&quot;      
[106] &quot;LSP1&quot;       &quot;SYNE1&quot;      &quot;NFATC1&quot;     &quot;RARA&quot;       &quot;CHD1L&quot;     
[111] &quot;MATN3&quot;      &quot;ZSWIM3&quot;     &quot;IPO8&quot;       &quot;LOXL2&quot;      &quot;CTIF&quot;      
[116] &quot;CCNH&quot;       &quot;SOX5&quot;       &quot;GGACT&quot;      &quot;NREP&quot;       &quot;SRPK2&quot;     
[121] &quot;DYSF&quot;       &quot;ALDH1L2&quot;    &quot;FLNB&quot;       &quot;PHF11&quot;      &quot;TNS3&quot;      
[126] &quot;IGF2BP3&quot;    &quot;SCN2A&quot;      &quot;CDK5RAP2&quot;   &quot;TSTD2&quot;      &quot;TLN1&quot;      
[131] &quot;ZSCAN9&quot;     &quot;HAUS2&quot;      &quot;PARP9&quot;      &quot;ARHGAP24&quot;   &quot;FGF2&quot;      
[136] &quot;SLC39A8&quot;    &quot;RGS3&quot;       &quot;ETV6&quot;       &quot;TAPBPL&quot;     &quot;SLC7A1&quot;    
[141] &quot;PELI2&quot;      &quot;TLE3&quot;       &quot;PARN&quot;       &quot;TXNL4B&quot;     &quot;CDH11&quot;     
[146] &quot;RANBP10&quot;    &quot;SSH2&quot;       &quot;SLC16A3&quot;    &quot;NFIC&quot;       &quot;URB1&quot;      
[151] &quot;SAE1&quot;       &quot;MYO1F&quot;      &quot;PTPRF&quot;      &quot;TUFT1&quot;      &quot;TTC13&quot;     
[156] &quot;CNIH3&quot;      &quot;ASXL2&quot;      &quot;PTPRG&quot;      &quot;BOC&quot;        &quot;PDGFC&quot;     
[161] &quot;PIK3R1&quot;     &quot;PAM&quot;        &quot;TRIM41&quot;     &quot;ABT1&quot;       &quot;DAAM2&quot;     
[166] &quot;CREB5&quot;      &quot;NIPSNAP2&quot;   &quot;CSGALNACT1&quot; &quot;TCF7L2&quot;     &quot;ADAM12&quot;    
[171] &quot;PAMR1&quot;      &quot;SERPINH1&quot;   &quot;LATS2&quot;      &quot;VTI1A&quot;      &quot;FLI1&quot;      
[176] &quot;BEND6&quot;      &quot;PTPN14&quot;     &quot;BCL2L11&quot;    &quot;ASAP1&quot;      &quot;CMIP&quot;      
[181] &quot;DDAH1&quot;      &quot;ANKH&quot;       &quot;SORBS2&quot;     &quot;PIEZO2&quot;     &quot;MAP3K7CL&quot;  
[186] &quot;SKI&quot;        &quot;COLEC12&quot;    &quot;TAOK1&quot;      &quot;ADAR&quot;       &quot;ADCY9&quot;     
[191] &quot;PLPP3&quot;      &quot;VANGL2&quot;     &quot;FRZB&quot;       &quot;ACTG2&quot;      &quot;COL6A3&quot;    
[196] &quot;PCOLCE2&quot;    &quot;TOPBP1&quot;     &quot;SLBP&quot;       &quot;CREBRF&quot;     &quot;ZCCHC24&quot;   
[201] &quot;XRRA1&quot;      &quot;WDR81&quot;      &quot;NDUFV1&quot;     &quot;SPINDOC&quot;    &quot;UBTD2&quot;     
[206] &quot;ABHD15&quot;     &quot;ZNF608&quot;     &quot;ROR2&quot;       &quot;LUZP1&quot;      &quot;TPST1&quot;     
[211] &quot;ALCAM&quot;      &quot;SMAD1&quot;      &quot;ELOVL6&quot;     &quot;CYTL1&quot;      &quot;NUDT6&quot;     
[216] &quot;FEZ2&quot;       &quot;PRKCE&quot;      &quot;PDE7B&quot;      &quot;TLN2&quot;       &quot;SUCLG2&quot;    
[221] &quot;AFF1&quot;       &quot;NBEA&quot;       &quot;SH3RF3&quot;     &quot;RHOD&quot;       &quot;EHBP1L1&quot;   
[226] &quot;XXYLT1&quot;     &quot;CTSF&quot;       &quot;MRPL11&quot;     &quot;RSRC1&quot;      &quot;PSMD2&quot;     
[231] &quot;PHYKPL&quot;     &quot;SLC38A9&quot;    &quot;ZNF518A&quot;    &quot;LCORL&quot;      &quot;PLEC&quot;      
[236] &quot;NT5DC1&quot;     &quot;RFLNA&quot;      &quot;ARID3B&quot;     &quot;ZNF784&quot;     &quot;CCDC66&quot;    
[241] &quot;PITPNB&quot;     &quot;POLR2A&quot;     &quot;ZBTB20&quot;     &quot;RAD51B&quot;     &quot;PAPPA&quot;     
[246] &quot;CEP63&quot;      &quot;KMT5A&quot;      &quot;NELFA&quot;      &quot;PTCH1&quot;      &quot;THBS2&quot;     
[251] &quot;PPARA&quot;      &quot;COL14A1&quot;    &quot;ZNF471&quot;     &quot;AFAP1&quot;      &quot;TLE1&quot;      
[256] &quot;SRC&quot;        &quot;SLC22A4&quot;    &quot;VEPH1&quot;      &quot;CDC42SE1&quot;   &quot;FAM114A1&quot;  
[261] &quot;MYO1C&quot;      &quot;MAN1A2&quot;     &quot;HSD17B11&quot;   &quot;NCOA6&quot;      &quot;ANKRD13B&quot;  
[266] &quot;PNP&quot;        &quot;GRK5&quot;       &quot;PLEKHM1&quot;    &quot;C16orf95&quot;   &quot;UHRF1&quot;     
[271] &quot;PCGF2&quot;     
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
[1] &quot;PRKG1&quot;
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
[1] &quot;EXOSC9&quot; &quot;FAM13A&quot;
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Steinberg2020-GWAS.OA&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
 [1] &quot;ACSL4&quot;    &quot;APOD&quot;     &quot;ARHGAP17&quot; &quot;ARSA&quot;     &quot;ATP6AP1&quot;  &quot;BCL2L13&quot; 
 [7] &quot;CASP4&quot;    &quot;CCDC86&quot;   &quot;CCPG1&quot;    &quot;CFH&quot;      &quot;COL4A1&quot;   &quot;COL4A2&quot;  
[13] &quot;COPG1&quot;    &quot;CTSD&quot;     &quot;DCPS&quot;     &quot;DDAH1&quot;    &quot;DGKA&quot;     &quot;EFEMP1&quot;  
[19] &quot;ENO1&quot;     &quot;EPS8L2&quot;   &quot;EXOSC4&quot;   &quot;FABP5&quot;    &quot;FGF1&quot;     &quot;FN1&quot;     
[25] &quot;FRZB&quot;     &quot;FSTL1&quot;    &quot;GALE&quot;     &quot;GGCT&quot;     &quot;GGH&quot;      &quot;GNPDA1&quot;  
[31] &quot;GPT2&quot;     &quot;GULP1&quot;    &quot;IGFBP7&quot;   &quot;IRAK4&quot;    &quot;IRS1&quot;     &quot;KCNMA1&quot;  
[37] &quot;MAP1B&quot;    &quot;METTL7A&quot;  &quot;MGST1&quot;    &quot;MPDZ&quot;     &quot;MPP6&quot;     &quot;MPRIP&quot;   
[43] &quot;MRPS26&quot;   &quot;MRPS27&quot;   &quot;MYOF&quot;     &quot;NAMPT&quot;    &quot;NDRG1&quot;    &quot;NME1&quot;    
[49] &quot;NOVA1&quot;    &quot;NPC2&quot;     &quot;NQO1&quot;     &quot;NRBF2&quot;    &quot;P4HA3&quot;    &quot;PDLIM4&quot;  
[55] &quot;PFKL&quot;     &quot;PGAM5&quot;    &quot;PID1&quot;     &quot;PNP&quot;      &quot;POLR2C&quot;   &quot;POSTN&quot;   
[61] &quot;PPFIA1&quot;   &quot;PYCR1&quot;    &quot;PYGL&quot;     &quot;RARRES2&quot;  &quot;RCAN1&quot;    &quot;RFC3&quot;    
[67] &quot;RHOC&quot;     &quot;RRAGC&quot;    &quot;RRP9&quot;     &quot;SDR39U1&quot;  &quot;SEC31A&quot;   &quot;SERPINB8&quot;
[73] &quot;SH3KBP1&quot;  &quot;SMYD5&quot;    &quot;SNX21&quot;    &quot;SORBS1&quot;   &quot;SSR3&quot;     &quot;STK4&quot;    
[79] &quot;TBL2&quot;     &quot;THBS3&quot;    &quot;THOC3&quot;    &quot;THOC5&quot;    &quot;TRAFD1&quot;   &quot;TRIM47&quot;  
[85] &quot;UCHL1&quot;    &quot;VAV2&quot;     &quot;YARS2&quot;   
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
[1] &quot;PBXIP1&quot;
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)
[1] &quot;Morris2020-GWAS.OP&quot;
[1] &quot;Shared across all ostadhoc classifications, tested in all classifications&quot;
  [1] &quot;PHF13&quot;     &quot;PEX14&quot;     &quot;CAPZB&quot;     &quot;ALPL&quot;      &quot;CLIC4&quot;    
  [6] &quot;HIVEP3&quot;    &quot;PLPP3&quot;     &quot;NEGR1&quot;     &quot;PKN2&quot;      &quot;TGFBR3&quot;   
 [11] &quot;THBS3&quot;     &quot;HHAT&quot;      &quot;TGFB2&quot;     &quot;PSEN2&quot;     &quot;ASXL2&quot;    
 [16] &quot;SLC8A1&quot;    &quot;PKDCC&quot;     &quot;PRKCE&quot;     &quot;MSH6&quot;      &quot;BCL11A&quot;   
 [21] &quot;ACTR2&quot;     &quot;MEIS1&quot;     &quot;SH3RF3&quot;    &quot;BCL2L11&quot;   &quot;MERTK&quot;    
 [26] &quot;FRZB&quot;      &quot;IRS1&quot;      &quot;COL6A3&quot;    &quot;MLPH&quot;      &quot;HDAC4&quot;    
 [31] &quot;PPARG&quot;     &quot;TMEM43&quot;    &quot;TGFBR2&quot;    &quot;PRKAR2A&quot;   &quot;ITGB5&quot;    
 [36] &quot;PCOLCE2&quot;   &quot;IGF2BP2&quot;   &quot;DLG1&quot;      &quot;FGFRL1&quot;    &quot;LCORL&quot;    
 [41] &quot;ANAPC4&quot;    &quot;ARHGAP24&quot;  &quot;AFF1&quot;      &quot;PDGFC&quot;     &quot;MSMO1&quot;    
 [46] &quot;FAT1&quot;      &quot;ITGA1&quot;     &quot;SLC38A9&quot;   &quot;ADAMTS6&quot;   &quot;FAM174A&quot;  
 [51] &quot;FOXC1&quot;     &quot;GMDS&quot;      &quot;HLA-A&quot;     &quot;DAAM2&quot;     &quot;VEGFA&quot;    
 [56] &quot;GJA1&quot;      &quot;TRMT11&quot;    &quot;PDE7B&quot;     &quot;AKAP12&quot;    &quot;SYNE1&quot;    
 [61] &quot;CREB5&quot;     &quot;GTF2I&quot;     &quot;SEMA3E&quot;    &quot;SEMA3A&quot;    &quot;CDK6&quot;     
 [66] &quot;TSPAN12&quot;   &quot;KIAA1549&quot;  &quot;MMP16&quot;     &quot;TRPS1&quot;     &quot;MPDZ&quot;     
 [71] &quot;CNTLN&quot;     &quot;KLF9&quot;      &quot;PTCH1&quot;     &quot;BICC1&quot;     &quot;KCNMA1&quot;   
 [76] &quot;ZCCHC24&quot;   &quot;VTI1A&quot;     &quot;TCF7L2&quot;    &quot;ADAM12&quot;    &quot;MGMT&quot;     
 [81] &quot;SOX6&quot;      &quot;CCDC34&quot;    &quot;PCNX3&quot;     &quot;FAM168A&quot;   &quot;PRSS23&quot;   
 [86] &quot;PTS&quot;       &quot;CLMP&quot;      &quot;FLI1&quot;      &quot;CACNA1C&quot;   &quot;EMP1&quot;     
 [91] &quot;SOX5&quot;      &quot;SSPN&quot;      &quot;METTL7A&quot;   &quot;ATF7&quot;      &quot;NAV3&quot;     
 [96] &quot;DCN&quot;       &quot;SPATA13&quot;   &quot;FOXO1&quot;     &quot;DHRS12&quot;    &quot;KLF12&quot;    
[101] &quot;DOCK9&quot;     &quot;BAZ1A&quot;     &quot;SAV1&quot;      &quot;RHOJ&quot;      &quot;ZBTB25&quot;   
[106] &quot;RAD51B&quot;    &quot;MARK3&quot;     &quot;TLE3&quot;      &quot;PPCDC&quot;     &quot;FAM234A&quot;  
[111] &quot;AXIN1&quot;     &quot;RAB11FIP3&quot; &quot;ADCY9&quot;     &quot;EMP2&quot;      &quot;SALL1&quot;    
[116] &quot;CMIP&quot;      &quot;ZNF469&quot;    &quot;POLR2A&quot;    &quot;GAS7&quot;      &quot;HS3ST3B1&quot; 
[121] &quot;NUFIP2&quot;    &quot;STAT5A&quot;    &quot;HDAC5&quot;     &quot;DCAKD&quot;     &quot;ACSF2&quot;    
[126] &quot;PGS1&quot;      &quot;USP36&quot;     &quot;RFNG&quot;      &quot;VAPA&quot;      &quot;FHOD3&quot;    
[131] &quot;NFATC1&quot;    &quot;PLPPR3&quot;    &quot;NFIC&quot;      &quot;RFX2&quot;      &quot;ARRDC2&quot;   
[136] &quot;GPATCH1&quot;   &quot;SLC27A5&quot;   &quot;BMP2&quot;      &quot;LIF&quot;       &quot;MYH9&quot;     
[1] &quot;Shared across pre+obl+emb+min, tested in all classifications&quot;
[1] &quot;HECW2&quot;
[1] &quot;Shared across obl+emb+min, tested in all classifications&quot;
character(0)
[1] &quot;Unique to preosteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to embedding osteoblasts, tested in all classifications&quot;
character(0)
[1] &quot;Unique to mineralizing osteoblasts, tested in all classifications&quot;
[1] &quot;EEF2K&quot;
[1] &quot;Unique to maturing osteocytes, tested in all classifications&quot;
character(0)</code></pre>
</div>
</div>
<div id="osteogenic-ad-hoc-classification-of-cells-scaled-integrated-data-thresh0-4gene-strict-time-2-only" class="section level2">
<h2>Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] Cormotif_1.30.0             affy_1.62.0                
 [3] ComplexHeatmap_2.0.0        VennDiagram_1.6.20         
 [5] futile.logger_1.4.3         viridis_0.5.1              
 [7] viridisLite_0.3.0           enrichplot_1.4.0           
 [9] Rgraphviz_2.28.0            graph_1.62.0               
[11] org.Hs.eg.db_3.8.2          AnnotationDbi_1.46.0       
[13] clusterProfiler_3.12.0      UpSetR_1.4.0               
[15] purrr_0.3.4                 scran_1.12.1               
[17] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
[19] DelayedArray_0.10.0         BiocParallel_1.18.0        
[21] matrixStats_0.57.0          Biobase_2.44.0             
[23] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[25] IRanges_2.18.1              S4Vectors_0.22.1           
[27] BiocGenerics_0.30.0         edgeR_3.26.5               
[29] limma_3.40.6                RColorBrewer_1.1-2         
[31] colorspace_2.0-0            gbm_2.1.5                  
[33] ggplot2_3.3.2               stringr_1.4.0              
[35] stringi_1.4.3               dplyr_1.0.2                
[37] Seurat_3.1.3               

loaded via a namespace (and not attached):
  [1] reticulate_1.16          tidyselect_1.1.0        
  [3] RSQLite_2.1.1            htmlwidgets_1.5.2       
  [5] Rtsne_0.15               munsell_0.5.0           
  [7] preprocessCore_1.46.0    codetools_0.2-16        
  [9] ica_1.0-2                statmod_1.4.32          
 [11] future_1.14.0            withr_2.3.0             
 [13] GOSemSim_2.10.0          knitr_1.23              
 [15] ROCR_1.0-7               DOSE_3.10.2             
 [17] gbRd_0.4-11              listenv_0.7.0           
 [19] labeling_0.4.2           Rdpack_0.11-0           
 [21] git2r_0.26.1             urltools_1.7.3          
 [23] GenomeInfoDbData_1.2.1   polyclip_1.10-0         
 [25] bit64_0.9-7              farver_2.0.3            
 [27] rprojroot_2.0.2          vctrs_0.3.5             
 [29] generics_0.1.0           lambda.r_1.2.3          
 [31] xfun_0.8                 R6_2.5.0                
 [33] clue_0.3-58              ggbeeswarm_0.6.0        
 [35] rsvd_1.0.1               locfit_1.5-9.1          
 [37] gridGraphics_0.4-1       bitops_1.0-6            
 [39] fgsea_1.15.1             promises_1.1.1          
 [41] scales_1.1.1             ggraph_1.0.2            
 [43] beeswarm_0.2.3           gtable_0.3.0            
 [45] npsurv_0.4-0             globals_0.12.4          
 [47] workflowr_1.6.2          rlang_0.4.9             
 [49] GlobalOptions_0.1.2      splines_3.6.1           
 [51] lazyeval_0.2.2           europepmc_0.3           
 [53] BiocManager_1.30.4       yaml_2.2.1              
 [55] reshape2_1.4.3           httpuv_1.5.1            
 [57] qvalue_2.16.0            tools_3.6.1             
 [59] ggplotify_0.0.3          affyio_1.54.0           
 [61] ellipsis_0.3.1           gplots_3.0.1.1          
 [63] dynamicTreeCut_1.63-1    ggridges_0.5.1          
 [65] Rcpp_1.0.5               plyr_1.8.4              
 [67] progress_1.2.2           zlibbioc_1.30.0         
 [69] RCurl_1.98-1.1           prettyunits_1.1.1       
 [71] GetoptLong_1.0.5         pbapply_1.4-0           
 [73] cowplot_1.1.0            zoo_1.8-6               
 [75] ggrepel_0.8.2            cluster_2.1.0           
 [77] fs_1.3.1                 magrittr_2.0.1          
 [79] futile.options_1.0.1     data.table_1.13.2       
 [81] DO.db_2.9                circlize_0.4.12         
 [83] lmtest_0.9-37            triebeard_0.3.0         
 [85] RANN_2.6.1               fitdistrplus_1.0-14     
 [87] hms_0.5.3                lsei_1.2-0              
 [89] evaluate_0.14            shape_1.4.4             
 [91] gridExtra_2.3            compiler_3.6.1          
 [93] scater_1.12.2            tibble_3.0.4            
 [95] KernSmooth_2.23-15       crayon_1.3.4            
 [97] htmltools_0.5.0          later_1.1.0.1           
 [99] tidyr_1.1.2              RcppParallel_5.0.2      
[101] DBI_1.1.0                formatR_1.7             
[103] tweenr_1.0.1             MASS_7.3-51.4           
[105] rappdirs_0.3.1           Matrix_1.2-18           
[107] gdata_2.18.0             metap_1.1               
[109] igraph_1.2.4.1           pkgconfig_2.0.3         
[111] rvcheck_0.1.3            plotly_4.9.2.1          
[113] xml2_1.3.2               vipor_0.4.5             
[115] dqrng_0.2.1              XVector_0.24.0          
[117] bibtex_0.4.2             digest_0.6.27           
[119] sctransform_0.2.0        RcppAnnoy_0.0.12        
[121] tsne_0.1-3               rmarkdown_1.13          
[123] leiden_0.3.1             fastmatch_1.1-0         
[125] uwot_0.1.5               DelayedMatrixStats_1.6.1
[127] gtools_3.8.1             rjson_0.2.20            
[129] lifecycle_0.2.0          nlme_3.1-140            
[131] jsonlite_1.7.1           BiocNeighbors_1.2.0     
[133] pillar_1.4.7             lattice_0.20-38         
[135] httr_1.4.2               survival_2.44-1.1       
[137] GO.db_3.8.2              glue_1.4.2              
[139] png_0.1-7                bit_1.1-14              
[141] ggforce_0.2.2            blob_1.2.0              
[143] BiocSingular_1.0.0       caTools_1.17.1.2        
[145] memoise_1.1.0            irlba_2.3.3             
[147] future.apply_1.3.0       ape_5.4-1               </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
