<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">human-chimp-skeletal-scRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>human-chimp-skeletal-scRNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomghousmanhumanchimpskeletalscRNAtree744bb3c332ca60f1246dd9b4887a0721871208a0targetblank744bb3ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/tree/744bb3c332ca60f1246dd9b4887a0721871208a0" target="_blank">744bb3c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/HumanPrimaryCellAtlasData-SingleR
    Untracked:  data/HumanPrimaryCellAtlasDetails-SingleR.csv
    Untracked:  data/HumanPrimaryCellAtlasLabels-SingleR.csv
    Untracked:  data/cell_atlas_ref_panel
    Untracked:  data/cellranger-data-full/
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot-DE-signif.csv
    Untracked:  data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/cormotif-data/cormotif.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds
    Untracked:  data/corrmatrix.ost
    Untracked:  data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.DE-signif.csv
    Untracked:  data/gom-data/topics10.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics15.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics2.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics20.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics30.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics8.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/topics9.data.filterC.log.indv-cell.intNo0.reg-tot.bc1.rds
    Untracked:  data/gom-data/v2/gom10.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.100genes.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom3.log.indv-cell.int19k.reg-tot.rds
    Untracked:  data/gom-data/v2/gom4.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom5.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom6.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom7.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom8.log.indv-cell.int19k.reg-t2.rds
    Untracked:  data/gom-data/v2/gom9.log.indv-cell.int19k.reg-t2.rds
    Untracked:  output/data-qc-post-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterC-subsets-merge-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-intNo0.reg-batchviz.pdf
    Untracked:  output/data-qc-postfilterL-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-int19k.reg-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets-merge-batchviz.pdf
    Untracked:  output/v2/data-qc-postfilter-subsets.reg-batchviz.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/744bb3c332ca60f1246dd9b4887a0721871208a0/analysis/index.Rmd" target="_blank">744bb3c</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2022-01-10
</td>
<td>
webpage fix
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/29538f8ca381f692484b02ec81e487c87431176a/analysis/index.Rmd" target="_blank">29538f8</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2022-01-10
</td>
<td>
webpage edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/29538f8ca381f692484b02ec81e487c87431176a/docs/index.html" target="_blank">29538f8</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2022-01-10
</td>
<td>
webpage edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/d837dd63e7ff6c2a962ba9a25eae146e494c6fe4/analysis/index.Rmd" target="_blank">d837dd6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-07
</td>
<td>
de gene overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/d837dd63e7ff6c2a962ba9a25eae146e494c6fe4/docs/index.html" target="_blank">d837dd6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-07
</td>
<td>
de gene overlap
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7fa17e693a3e8238a3d607166efdf3c0bc59648e/analysis/index.Rmd" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
updated threshold code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/7fa17e693a3e8238a3d607166efdf3c0bc59648e/docs/index.html" target="_blank">7fa17e6</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
updated threshold code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/de2afe484d8dd8020aabbe827591a36112cc1a85/analysis/index.Rmd" target="_blank">de2afe4</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
cormotif v2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/de2afe484d8dd8020aabbe827591a36112cc1a85/docs/index.html" target="_blank">de2afe4</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-06
</td>
<td>
cormotif v2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/7bf1028abf0c2cebed997cdc3c4bdae7409c0e1f/analysis/index.Rmd" target="_blank">7bf1028</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-02
</td>
<td>
fixed index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/7bf1028abf0c2cebed997cdc3c4bdae7409c0e1f/docs/index.html" target="_blank">7bf1028</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-09-02
</td>
<td>
fixed index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/929e6fd25913cf3c2a0c5f4f3ada99888213b1cb/docs/index.html" target="_blank">929e6fd</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-26
</td>
<td>
fixed website links
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/3690374fcf65f937c8923fc893956f208d4089b1/analysis/index.Rmd" target="_blank">3690374</a>
</td>
<td>
GitHub
</td>
<td>
2021-08-26
</td>
<td>
Update index.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/de4fe24f33ddb85682a26b2a7a029dc33517961f/analysis/index.Rmd" target="_blank">de4fe24</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-26
</td>
<td>
index edit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/e1794eefe63f6c92d42386d2efe14c17ab52b87a/docs/index.html" target="_blank">e1794ee</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-10
</td>
<td>
additional docs updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/e62c53703ed251e27c4d23397131fd218013a635/analysis/index.Rmd" target="_blank">e62c537</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-08-05
</td>
<td>
updated analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/9ba9279ef2b3fe2df1025b69d41bc94e1bbfa886/docs/index.html" target="_blank">9ba9279</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-06-01
</td>
<td>
initial docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/3a934e43a10effeb1a91c9f494d51edc28059d6f/analysis/index.Rmd" target="_blank">3a934e4</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-05-31
</td>
<td>
updated index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/a478a3928ffb9607519522f7c91784f21b612a6e/docs/index.html" target="_blank">a478a39</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-02-11
</td>
<td>
redo site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/eef3584cbd46225af06c2b9ffac1c97bf7a25397/analysis/index.Rmd" target="_blank">eef3584</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2021-02-11
</td>
<td>
rearranging files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/f918f2b3e5531a109d05b2b42ffc8af09ab4fed5/analysis/index.Rmd" target="_blank">f918f2b</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2020-08-12
</td>
<td>
Starting ortho-exon v2 reanalysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ghousman/human-chimp-skeletal-scRNA/09f445a517cdd1610147c54873d888f838bffe74/docs/index.html" target="_blank">09f445a</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2019-07-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/b69614ccb59291199916da09d82f6d671ff7eb1b/analysis/index.Rmd" target="_blank">b69614c</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2019-07-19
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ghousman/human-chimp-skeletal-scRNA/blob/1f85ad394f45212ee6df60d3c60e0dc4980ab31f/analysis/index.Rmd" target="_blank">1f85ad3</a>
</td>
<td>
Genevieve Housman
</td>
<td>
2019-07-19
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Genevieve Housman</p>
<div id="workflow-and-analysis-of-my-human-and-chimpanzee-comparative-skeletal-single-cell-gene-expression-project." class="section level1">
<h1>Workflow and analysis of my human and chimpanzee comparative skeletal single-cell gene expression project.</h1>
<p>iPSCs from 6 human and 6 chimpanzees were differentiated into mesenchymal stem cells and subsequently into osteogenic cells. Single cell RNAseq from each major stage of differentiation (iPSCs, MSCs, and osteogenic cells) were collected using the 10x platform for all samples, as well as 1 replicate human cell line and 1 replicate chimpanzee cell line.</p>
<p>If you are running the files in this repository to recreate the analyses, running them in the order they appear on this list is recommended.</p>
<div id="single-cell-data-processing" class="section level2">
<h2>Single-cell data processing</h2>
<ul>
<li><p><a href="10x-data-1-process.html">Pre-processing of raw 10x fastq files into count matrices and assigning species of origin</a></p>
<ul>
<li><p><a href="10x-data-0-process-test.html">Analyses justifying method the used to assign species of origin</a></p></li>
<li><p><a href="10x-plot-cellcalling.html">Visualized cell calls</a></p></li>
</ul></li>
<li><p><a href="10x-data-2-qc.html">QC of count matricies and barcode filtering</a></p>
<ul>
<li>Decided to proceed with a conservative threshold: umi per cell &gt; 1000 | genes per cell &gt; 700 | percent mito &lt; 25</li>
</ul></li>
<li><p><a href="10x-data-3-integrate.html">Data integration</a></p>
<ul>
<li>Decided to proceed with data integrated across all genes with non-zero UMI counts and with nCount_RNA and percent.mt regressed out during data scaling following integration</li>
</ul></li>
</ul>
</div>
<div id="cell-classification" class="section level2">
<h2>Cell classification</h2>
<ul>
<li><p><a href="10x-data-4-classification-tot.html">Classified cells using unsupervised clustering, topic modeling grades of membership, and ad hoc assignments</a></p>
<ul>
<li><p><a href="10x-plot-classification-clusters.html">Visualized results of unsupervised clustering</a></p></li>
<li><p><a href="10x-plot-classification-gom.html">Visualized results of topic modeling grades of membership</a></p></li>
<li><p><a href="10x-plot-classification-adhoc.html">Visualized results of ad hoc assignments</a></p></li>
<li><p><a href="10x-plot-classification.html">Made finalized plots, examined functional enrichment, and evaluated pseudobulk correlations between cell classifications</a></p></li>
</ul></li>
<li><p>For General Unsupervised Clustering: Decided to proceed with cluster resolutions 0.05. Also examined 0.25 and 0.50 further.</p></li>
<li><p>For Grades of Membership: Decided to proceed with total dataset batch corrected (individual, replicate) with k=3, k=4, k=5, k=6, and k=7.</p></li>
<li><p>For General Ad Hoc: Decided to proceed with the following:</p>
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>ips = +POU5F1,+TDGF,+SOX2,+EPCAM</li>
<li>msc = +THY1,+NT5E,+ENG,+CD44</li>
<li>ost = -POU5F1,+COL1A1,+COL1A2,+ALPL</li>
<li>also examined assignments at a couple different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)</li>
</ul></li>
<li><p>For Osteogenic Unsupervised Clustering: Decided to proceed with the following:</p>
<ul>
<li>cluster resolutions 0.50 (4 osteogenic clusters)</li>
<li>cluster resolutions 0.75 (8 osteogenic clusters)</li>
</ul></li>
<li><p>For Osteogenic Ad Hoc: Decided to proceed with the following:</p>
<ul>
<li>thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)</li>
<li>assignments based on 4 genes (both strict and flexible classifications)</li>
<li>also examined assignments at a couple different cutoffs (thresh-0.01,thresh0.01)</li>
<li>also examined assignments based on 8 genes (strict and flexible) and 6 genes (strict and flexible)</li>
</ul></li>
</ul>
</div>
<div id="differential-expression" class="section level2">
<h2>Differential expression</h2>
<ul>
<li><p>Performed:</p>
<ul>
<li><p><a href="10x-data-5-diffexp.html">Pairwise DE tests</a> &amp; <a href="10x-plot-diffexp.html">Visualized results</a></p></li>
<li><p><a href="10x-data-6-cormotif-tot-cnts_v2.html">Cormotif DE tests</a> &amp; <a href="10x-plot-cormotif-cnts_v2.html">Visualized results</a></p></li>
<li><p>Examined gene set enrichments to determine optimal <a href="10x-plot-diffexp-thresh.html">FDR cutoff</a> (FDR &lt; 0.01) and <a href="10x-plot-cormotif-cnts-thresh_v2.html">posterior probability cutoff</a> (posterior probability &gt; 0.65)</p></li>
<li><p><a href="10x-plot-de-overlap.html">Evaluated overlap of Pairwise DE genes and Cormotif DE genes</a></p></li>
</ul></li>
<li><p><a href="10x-data-7-variance-tot.html">Examined variance in gene expression</a></p></li>
</ul>
</div>
<div id="enrichment-analyses" class="section level2">
<h2>Enrichment analyses</h2>
<ul>
<li><p>Examined enrichment of GO categories and external gene sets in DE genes identified using:</p>
<ul>
<li><p><a href="10x-plot-diffexp.html">Pairwise tests</a></p></li>
<li><p><a href="10x-plot-cormotif-cnts_v2.html">Cormotif</a></p></li>
</ul></li>
</ul>
</div>
<div id="standard-validation-analyses" class="section level2">
<h2>Standard validation analyses</h2>
<ul>
<li><a href="10x-view-validation-data.html">Analyzed flow cytometry, staining, and qPCR data</a></li>
</ul>
</div>
<div id="additional-analyses-and-visualizations" class="section level2">
<h2>Additional analyses and visualizations</h2>
<ul>
<li><p><a href="10x-view-batch-effects.html">Viewed correlation of batch effects</a></p></li>
<li><p><a href="10x-view-data-metrics.html">Viewed 10x sequencing data metrics</a></p></li>
<li><p><a href="10x-view-species-assignments.html">Viewed species assignments</a></p></li>
<li><p><a href="10x-view-data-subsets.html">Viewed 10x data subset by stage of differentiation</a></p></li>
<li><p><a href="10x-plot-replicate-correlation-tot.html">Analyzed technical replicate correlations</a></p></li>
<li><p><a href="paper-revisions.html">Analyses performed in response to paper revisions</a></p></li>
</ul>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
