---
title: "10x-data-enrichment"
author: "Genevieve Housman"
date: "September 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Differential Expression

Test for differential expression between humans and chimpanzees within different cell classifications.


```{r load libraries}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)

```

### Examine output of DE analyses in more detail

```{r}

#Load DE results
#compare1 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.stage.rds")
#compare2 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.cluster.rds")
#compare3 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.adhoc.rds")
#compare4 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.ostadhoc.rds")
compare1 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.stage.rds")
compare2 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.cluster.rds")
compare3 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.adhoc.rds")
compare4 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.ostadhoc.rds")
DEdata <- rbind(compare1,compare2,compare3,compare4)
colnames(DEdata)[5] <- "FDR"
DEdata <- as_tibble(DEdata)
rm(compare1,compare2,compare3,compare4)

```

```{r, eval=FALSE}

#TEMPORARY - fix gene names

compare1 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.stage.rds")
compare1$gene <- rownames(compare1)
compare1$geneCompare0 <- compare1$gene %in% rownames(data@assays$integrated)
table(compare1$geneCompare0)
compare1$geneNew1 <- compare1$gene
compare1$geneNew1[which(compare1$geneCompare0==FALSE)] <- substr(compare1$gene[which(compare1$geneCompare0==FALSE)],1,nchar(compare1$gene[which(compare1$geneCompare0==FALSE)])-1)
compare1$geneCompare1 <- compare1$geneNew1 %in% rownames(data@assays$integrated)
table(compare1$geneCompare1)
compare1$geneNew2 <- compare1$geneNew1
compare1$geneNew2[which(compare1$geneCompare1==FALSE)] <- substr(compare1$geneNew1[which(compare1$geneCompare1==FALSE)],1,nchar(compare1$geneNew1[which(compare1$geneCompare1==FALSE)])-1)
compare1$geneCompare2 <- compare1$geneNew2 %in% rownames(data@assays$integrated)
table(compare1$geneCompare2)
compare1$gene <- compare1$geneNew2

compare2 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.cluster.rds")
compare2$gene <- rownames(compare2)
compare2$geneCompare0 <- compare2$gene %in% rownames(data@assays$integrated)
table(compare2$geneCompare0)
compare2$geneNew1 <- compare2$gene
compare2$geneNew1[which(compare2$geneCompare0==FALSE)] <- substr(compare2$gene[which(compare2$geneCompare0==FALSE)],1,nchar(compare2$gene[which(compare2$geneCompare0==FALSE)])-1)
compare2$genecompare2 <- compare2$geneNew1 %in% rownames(data@assays$integrated)
table(compare2$genecompare2)
compare2$geneNew2 <- compare2$geneNew1
compare2$geneNew2[which(compare2$genecompare2==FALSE)] <- substr(compare2$geneNew1[which(compare2$genecompare2==FALSE)],1,nchar(compare2$geneNew1[which(compare2$genecompare2==FALSE)])-1)
compare2$geneCompare2 <- compare2$geneNew2 %in% rownames(data@assays$integrated)
table(compare2$geneCompare2)
compare2$gene <- compare2$geneNew2

compare3 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.adhoc.rds")
compare3$gene <- rownames(compare3)
compare3$geneCompare0 <- compare3$gene %in% rownames(data@assays$integrated)
table(compare3$geneCompare0)
compare3$geneNew1 <- compare3$gene
compare3$geneNew1[which(compare3$geneCompare0==FALSE)] <- substr(compare3$gene[which(compare3$geneCompare0==FALSE)],1,nchar(compare3$gene[which(compare3$geneCompare0==FALSE)])-1)
compare3$genecompare3 <- compare3$geneNew1 %in% rownames(data@assays$integrated)
table(compare3$genecompare3)
compare3$geneNew2 <- compare3$geneNew1
compare3$geneNew2[which(compare3$genecompare3==FALSE)] <- substr(compare3$geneNew1[which(compare3$genecompare3==FALSE)],1,nchar(compare3$geneNew1[which(compare3$genecompare3==FALSE)])-1)
compare3$geneCompare2 <- compare3$geneNew2 %in% rownames(data@assays$integrated)
table(compare3$geneCompare2)
compare3$gene <- compare3$geneNew2

compare4 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.ostadhoc.rds")
compare4$gene <- rownames(compare4)
compare4$geneCompare0 <- compare4$gene %in% rownames(data@assays$integrated)
table(compare4$geneCompare0)
compare4$geneNew1 <- compare4$gene
compare4$geneNew1[which(compare4$geneCompare0==FALSE)] <- substr(compare4$gene[which(compare4$geneCompare0==FALSE)],1,nchar(compare4$gene[which(compare4$geneCompare0==FALSE)])-1)
compare4$genecompare4 <- compare4$geneNew1 %in% rownames(data@assays$integrated)
table(compare4$genecompare4)
compare4$gene <- compare4$geneNew1

DEdata <- rbind(compare1[,1:15],compare2[,1:15],compare3[,1:15],compare4[,1:15])
DEdata <- as_tibble(DEdata)
rm(compare1,compare2,compare3,compare4)

##########

```

```{r}

#Isolate significant DE genes
DEdataSig <- DEdata %>% filter(FDR<=0.01)
#DEdataSig <- DEdata %>% filter(abs(logFC)>=1.2 & FDR<=0.01)

```

```{r}

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

```

```{r}

#cell assignment = stage
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                        data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                        data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0","Time.1","Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.0","Time.2"),
                         list("Time.1","Time.2"),
                         list("Time.0"),
                         list("Time.1"),
                         list("Time.2")))

#GO enrichment of significant DE genes
go.t0 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                  deGenes = pull(DEdata %>% filter(logFC>=1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                  ontology = "ALL")
go.t0.h <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    deGenes = pull(DEdata %>% filter(logFC>=1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    ontology = "ALL")
go.t0.c <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    deGenes = pull(DEdata %>% filter(logFC<=-1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    ontology = "ALL")
go.t0@result$Description = str_wrap(go.t0@result$Description, width=30)
go.t0.h@result$Description = str_wrap(go.t0.h@result$Description, width=30)
go.t0.c@result$Description = str_wrap(go.t0.c@result$Description, width=30)

#head(summary(go.t0))
#barplot(go.t0, drop=TRUE, showCategory=10, title='DE Genes GO Analysis')
#dotplot(go.t0, x="GeneRatio", color="p.adjust", showCategory=5, size=NULL, split="ONTOLOGY", font.size=12, title="") + facet_grid(ONTOLOGY~., scale="free")
#cnetplot(go.t0, showCategory=5, foldChange=pull(DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene, logFC)))
#plotGOgraph(go.t0, firstSigNodes=5, useInfo="all", sigForAll=TRUE, useFullNames=TRUE)
#upsetplot(go.t0)
#heatplot(go.t0)
#emapplot(go.t0, showCategory=5, color="p.adjust")

p1 <- ggplot(go.t0, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="All DE Genes")

p2 <- ggplot(go.t0.h, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="Upregulated in Humans")

p3 <- ggplot(go.t0.c, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="Upregulated in Chimps")

CombinePlots(list(p1,p2,p3),ncol=3,legend="right")

#GO enrichment of cell type specific significant DE genes
go.share <- goEnrich(totGenes = intersect(pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                                          intersect(pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)),
                                                    pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)))),
                     deGenes = combined$gene[which(rowSums(combined[,2:4])==3)],
                     ontology = "ALL")
go.share@result$Description = str_wrap(go.share@result$Description, width=30)

go.uniq.t0 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,2]==1 & rowSums(combined[,3:4])==0)],
                       ontology = "ALL")
go.uniq.t0@result$Description = str_wrap(go.uniq.t0@result$Description, width=30)

go.uniq.t1 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,3]==1 & rowSums(combined[,c(2,4)])==0)],
                       ontology = "ALL")
go.uniq.t1@result$Description = str_wrap(go.uniq.t1@result$Description, width=30)

go.uniq.t2 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,4]==1 & rowSums(combined[,2:3])==0)],
                       ontology = "ALL")
go.uniq.t2@result$Description = str_wrap(go.uniq.t2@result$Description, width=30)

p0 <- ggplot(go.share, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Shared\nacross Stages") +
        ylab(label="")

p1 <- ggplot(go.uniq.t0, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 0") +
        ylab(label="")

p2 <- ggplot(go.uniq.t1, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 1") +
        ylab(label="")

p3 <- ggplot(go.uniq.t2, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 2") +
        ylab(label="")

CombinePlots(list(p1,p2,p3),ncol=3,legend="right")

```

```{r}

#cell assignment = ad hoc
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c2") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c3") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="MSC.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c2") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1_iPSC.c2_iPSC.c3") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1_Osteoblast.c2") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1") %>% select(gene), "iPSC.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c2") %>% select(gene), "iPSC.c2"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c3") %>% select(gene), "iPSC.c3"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="MSC.c1") %>% select(gene), "MSC.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1") %>% select(gene), "Osteoblast.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c2") %>% select(gene), "Osteoblast.c2"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1_iPSC.c2_iPSC.c3") %>% select(gene), "iPSC.c1.c2.c3"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1_Osteoblast.c2") %>% select(gene), "Osteoblast.c1.c2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1.c2.c3","MSC.c1","Osteoblast.c1.c2"),
                         list("iPSC.c1.c2.c3","MSC.c1"),
                         list("iPSC.c1.c2.c3","Osteoblast.c1.c2"),
                         list("MSC.c1","Osteoblast.c1.c2"),
                         list("iPSC.c1.c2.c3"),
                         list("MSC.c1"),
                         list("Osteoblast.c1.c2")))

upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteoblast.c1","Osteoblast.c2"),
      intersections=list(list("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteoblast.c1","Osteoblast.c2"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3","Osteoblast.c1","Osteoblast.c2"),
                         list("MSC.c1","Osteoblast.c1","Osteoblast.c2"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3"),
                         #list("iPSC.c1","iPSC.c2"),
                         #list("iPSC.c1","iPSC.c3"),
                         #list("iPSC.c2","iPSC.c3"),
                         list("iPSC.c1"),
                         list("iPSC.c2"),
                         list("iPSC.c3"),
                         list("MSC.c1"),
                         list("Osteoblast.c1","Osteoblast.c2"),
                         list("Osteoblast.c1"),
                         list("Osteoblast.c2"))
                         )

```

```{r}

#cell assignment = ad hoc
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="iPSC") %>% select(gene)
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="MSC") %>% select(gene)
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="Osteoblast") %>% select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="iPSC") %>% dplyr::select(gene), "iPSC"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="MSC") %>% dplyr::select(gene), "MSC"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="Osteoblast") %>% dplyr::select(gene), "Osteoblast"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC","MSC","Osteoblast"),
                         list("iPSC","MSC"),
                         list("iPSC","Osteoblast"),
                         list("MSC","Osteoblast"),
                         list("iPSC"),
                         list("MSC"),
                         list("Osteoblast")))

```

```{r}

#cell assignment = ost ad hoc
DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene), "pre.osteo"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene), "emb.osteo"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("pre.osteo","emb.osteo"),
                         list("pre.osteo"),
                         list("emb.osteo")))

#GO enrichment of cell type specific significant DE genes
go.share <- goEnrich(totGenes = intersect(pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)),
                                          pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene))),
                        deGenes = combined$gene[which(rowSums(combined[,2:3])==2)],
                        ontology = "ALL")
go.share@result$Description = str_wrap(go.share@result$Description, width=30)

go.uniq.pre <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)),
                        deGenes = combined$gene[which(combined[,2]==1 & combined[,3]==0)],
                        ontology = "ALL")
go.uniq.pre@result$Description = str_wrap(go.uniq.pre@result$Description, width=30)

go.uniq.emb <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene)),
                        deGenes = combined$gene[which(combined[,3]==1 & combined[,2]==0)],
                        ontology = "ALL")
go.uniq.emb@result$Description = str_wrap(go.uniq.emb@result$Description, width=30)

p1 <- ggplot(go.share, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Shared in\nosteogenic cells") +
        ylab(label="")

p2 <- ggplot(go.uniq.pre, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique to\npreosteoblasts") +
        ylab(label="")

p3 <- ggplot(go.uniq.emb, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique to\nembedding osteoblasts") +
        ylab(label="")

CombinePlots(list(p1,p2,p3),ncol=3,legend="right")


```





```{r}

#Define functions for functional enrichment assessment in differentially expressed genes
goCCenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "CC",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
goBPenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "BP",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
funcEnrich <- function(data, markers) {
  genes <- structure(rep(1, length(rownames(data@assays$RNA@data))), names=rownames(data@assays$RNA@data))
  x=1
  while (x <= length(rownames(markers))) {
    genes[which(names(genes)==rownames(markers)[x])] <- markers$p_val_adj[x]
    x=x+1
  }
  go_data <- new("topGOdata", ontology="BP", allGenes=genes, geneSel=function(allScore){return(allScore < 0.01)},
                 nodeSize=5, annotationFun=annFUN.org, mapping="org.Hs.eg.db", ID="symbol")
  go_test <- runTest(go_data, algorithm="weight01", statistic="fisher")
  go_table <- GenTable(go_data, weightFisher=go_test, orderBy="weightFisher", ranksOf="weightFisher", topNodes=sum(score(go_test) < .01))
  return(go_table)
}

```

```{r}

#Examine differentially expressed genes using differentiation stage at collection
compare.ips <- readRDS("./data/cellranger-data-full/de.w-compare.ips.rds")
compare.msc <- readRDS("./data/cellranger-data-full/de.w-compare.msc.rds")
compare.ost <- readRDS("./data/cellranger-data-full/de.w-compare.ost.rds")
dim(compare.ips)
dim(compare.msc)
dim(compare.ost)
table(compare.ips$p_val_adj<0.05)
table(compare.msc$p_val_adj<0.05)
table(compare.ost$p_val_adj<0.05)
head(compare.ips, n=5)
head(compare.msc, n=5)
head(compare.ost, n=5)
compare.ips$gene <- rownames(compare.ips)
compare.msc$gene <- rownames(compare.msc)
compare.ost$gene <- rownames(compare.ost)
fracRibo(compare.ips)
fracRibo(compare.msc)
fracRibo(compare.ost)
p.ips <- goCCenrich(indv,compare.ips)
p.msc <- goCCenrich(indv,compare.msc)
p.ost <- goCCenrich(indv,compare.ost)
p.ips[[3]]
p.msc[[3]]
p.ost[[3]]
#go.ips <- funcEnrich(indv, compare.ips)
#go.msc <- funcEnrich(indv, compare.msc)
#go.ost <- funcEnrich(indv, compare.ost)
#dim(go.ips)
#dim(go.msc)
#dim(go.ost)
#head(go.ips, n=5)
#head(go.msc, n=5)
#head(go.ost, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


#Examine differentially expressed genes using cluster definitions
compare.i <- readRDS("./data/cellranger-data-full/de.w-compare.i.rds")
compare.m <- readRDS("./data/cellranger-data-full/de.w-compare.m.rds")
compare.o <- readRDS("./data/cellranger-data-full/de.w-compare.o.rds")
compare.i1 <- readRDS("./data/cellranger-data-full/de.w-compare.i1.rds")
compare.i2 <- readRDS("./data/cellranger-data-full/de.w-compare.i2.rds")
compare.i3 <- readRDS("./data/cellranger-data-full/de.w-compare.i3.rds")
compare.o1 <- readRDS("./data/cellranger-data-full/de.w-compare.o1.rds")
compare.o2 <- readRDS("./data/cellranger-data-full/de.w-compare.o2.rds")
compare.o3 <- readRDS("./data/cellranger-data-full/de.w-compare.o3.rds")
dim(compare.i)
dim(compare.m)
dim(compare.o) 
dim(compare.i1)
dim(compare.i2)
dim(compare.i3)
dim(compare.o1)
dim(compare.o2)
dim(compare.o3)
table(compare.i$p_val_adj<0.05)
table(compare.m$p_val_adj<0.05)
table(compare.o$p_val_adj<0.05)
table(compare.i1$p_val_adj<0.05)
table(compare.i2$p_val_adj<0.05)
table(compare.i3$p_val_adj<0.05)
table(compare.o1$p_val_adj<0.05)
table(compare.o2$p_val_adj<0.05)
table(compare.o3$p_val_adj<0.05)
head(compare.i, n=5)
head(compare.m, n=5)
head(compare.o, n=5)
head(compare.i1, n=5)
head(compare.i2, n=5)
head(compare.i3, n=5)
head(compare.o1, n=5)
head(compare.o2, n=5)
head(compare.o3, n=5)
compare.i$gene <- rownames(compare.i)
compare.m$gene <- rownames(compare.m)
compare.o$gene <- rownames(compare.o)
compare.i1$gene <- rownames(compare.i1)
compare.i2$gene <- rownames(compare.i2)
compare.i3$gene <- rownames(compare.i3)
compare.o1$gene <- rownames(compare.o1)
compare.o2$gene <- rownames(compare.o2)
compare.o3$gene <- rownames(compare.o3)
fracRibo(compare.i)
fracRibo(compare.m)
fracRibo(compare.o)
fracRibo(compare.i1)
fracRibo(compare.i2)
fracRibo(compare.i3)
fracRibo(compare.o1)
fracRibo(compare.o2)
fracRibo(compare.o3)
p.i <- goCCenrich(indv,compare.i)
p.m <- goCCenrich(indv,compare.m)
p.o <- goCCenrich(indv,compare.o)
p.i1 <- goCCenrich(indv,compare.i1)
p.i2 <- goCCenrich(indv,compare.i2)
p.i3 <- goCCenrich(indv,compare.i3)
p.o1 <- goCCenrich(indv,compare.o1)
p.o2 <- goCCenrich(indv,compare.o2)
p.o3 <- goCCenrich(indv,compare.o3)
p.i[[3]]
p.m[[3]]
p.o[[3]]
p.i1[[3]]
p.i2[[3]]
p.i3[[3]]
p.o1[[3]]
p.o2[[3]]
p.o3[[3]]
#go.i <- funcEnrich(indv, compare.i)
#go.m <- funcEnrich(indv, compare.m)
#go.o <- funcEnrich(indv, compare.o)
#go.i1 <- funcEnrich(indv, compare.i1)
#go.i2 <- funcEnrich(indv, compare.i2)
#go.i3 <- funcEnrich(indv, compare.i3)
#go.o1 <- funcEnrich(indv, compare.o1)
#go.o2 <- funcEnrich(indv, compare.o2)
#go.o3 <- funcEnrich(indv, compare.o3)
#dim(go.i)
#dim(go.m)
#dim(go.o)
#dim(go.i1)
#dim(go.i2)
#dim(go.i3)
#dim(go.o1)
#dim(go.o2)
#dim(go.o3)
#head(go.i, n=5)
#head(go.m, n=5)
#head(go.o, n=5)
#head(go.i1, n=5)
#head(go.i2, n=5)
#head(go.i3, n=5)
#head(go.o1, n=5)
#head(go.o2, n=5)
#head(go.o3, n=5)
compare.tot2 <- intersect(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                          intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                    rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #shared DE genes
compare.i.tot <- setdiff(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],c(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to i
compare.m.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to m
compare.o.tot <- setdiff(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.m)[which(compare.m$p_val_adj<0.05)])) #DE unique to o
compare.tot3 <- intersect(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],
                          intersect(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                    intersect(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                              intersect(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                        intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                  intersect(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                            intersect(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                      rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)]))))))) #shared DE
compare.i1.tot <- setdiff(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],c(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i1
compare.i2.tot <- setdiff(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i2
compare.i3.tot <- setdiff(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i3
compare.m1.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                 rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                 rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                 rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                 rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                 rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to m1
compare.o1.tot <- setdiff(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o1
compare.o2.tot <- setdiff(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o2
compare.o3.tot <- setdiff(rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)])) #DE unique to o3
length(compare.tot2)
length(compare.i.tot)
length(compare.m.tot)
length(compare.o.tot)
length(compare.tot3)
length(compare.i1.tot)
length(compare.i2.tot)
length(compare.i3.tot)
length(compare.m1.tot)
length(compare.o1.tot)
length(compare.o2.tot)
length(compare.o3.tot)
head(compare.i[which(rownames(compare.i)%in%compare.tot2),], n=5)
head(compare.i[which(rownames(compare.i)%in%compare.i.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m.tot),], n=5)
head(compare.o[which(rownames(compare.o)%in%compare.o.tot),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.tot3),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),], n=5)
head(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),], n=5)
head(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m1.tot),], n=5)
head(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),], n=5)
head(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),], n=5)
head(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),], n=5)
fracRibo(compare.i[which(rownames(compare.i)%in%compare.tot2),])
fracRibo(compare.i[which(rownames(compare.i)%in%compare.i.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m.tot),])
fracRibo(compare.o[which(rownames(compare.o)%in%compare.o.tot),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
fracRibo(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
fracRibo(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
fracRibo(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
fracRibo(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
fracRibo(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2 <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goCCenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goCCenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goCCenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goCCenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goCCenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goCCenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
p.tot2 <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goBPenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goBPenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goBPenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goBPenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goBPenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goBPenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.i$gene[which(compare.i$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o$gene[which(compare.o$p_val_adj<0.05)])),
                    Other=c((length(compare.i$gene[which(compare.i$p_val_adj<0.05)])-length(compare.i.tot)-length(compare.tot2)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m.tot)-length(compare.tot2)),
                            (length(compare.o$gene[which(compare.o$p_val_adj<0.05)])-length(compare.o.tot)-length(compare.tot2))),
                    Shared=c(rep(length(compare.tot2),3)),
                    Unique=c(length(compare.i.tot),
                             length(compare.m.tot),
                             length(compare.o.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))
DE.df <- data.frame(cell=(c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3")),
                    totalDEgenes=c(length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)]),
                                   length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)]),
                                   length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)]),
                                   length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)]),
                                   length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])),
                    Other=c((length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)])-length(compare.i1.tot)-length(compare.tot3)),
                            (length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)])-length(compare.i2.tot)-length(compare.tot3)),
                            (length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)])-length(compare.i3.tot)-length(compare.tot3)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m1.tot)-length(compare.tot3)),
                            (length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)])-length(compare.o1.tot)-length(compare.tot3)),
                            (length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)])-length(compare.o2.tot)-length(compare.tot3)),
                            (length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])-length(compare.o3.tot)-length(compare.tot3))),
                    Shared=c(rep(length(compare.tot3),7)),
                    Unique=c(length(compare.i1.tot),
                             length(compare.i2.tot),
                             length(compare.i3.tot),
                             length(compare.m1.tot),
                             length(compare.o1.tot),
                             length(compare.o2.tot),
                             length(compare.o3.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[8:28,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))



#Find differentially expressed genes using ad hoc definitions
compare.hoci  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="iPSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoci, file="./data/cellranger-data-full/de-compare.hoci.rds")
compare.hocm  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="MSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hocm, file="./data/cellranger-data-full/de-compare.hocm.rds")
compare.hoco  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="Osteoblast"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoco, file="./data/cellranger-data-full/de-compare.hoco.rds")

compare.hoci <- readRDS("./data/cellranger-data-full/de.w-compare.hoci.rds")
compare.hocm <- readRDS("./data/cellranger-data-full/de.w-compare.hocm.rds")
compare.hoco <- readRDS("./data/cellranger-data-full/de.w-compare.hoco.rds")
dim(compare.hoci)
dim(compare.hocm)
dim(compare.hoco)
table(compare.hoci$p_val_adj<0.05)
table(compare.hocm$p_val_adj<0.05)
table(compare.hoco$p_val_adj<0.05)
head(compare.hoci, n=5)
head(compare.hocm, n=5)
head(compare.hoco, n=5)
compare.hoci$gene <- rownames(compare.hoci)
compare.hocm$gene <- rownames(compare.hocm)
compare.hoco$gene <- rownames(compare.hoco)
fracRibo(compare.hoci)
fracRibo(compare.hocm)
fracRibo(compare.hoco)
p.hoci <- goCCenrich(indv,compare.hoci)
p.hocm <- goCCenrich(indv,compare.hocm)
p.hoco <- goCCenrich(indv,compare.hoco)
p.hoci[[3]]
p.hocm[[3]]
p.hoco[[3]]
#go.hoci <- funcEnrich(indv, compare.hoci)
#go.hocm <- funcEnrich(indv, compare.hocm)
#go.hoco <- funcEnrich(indv, compare.hoco)
#dim(go.hoci)
#dim(go.hocm)
#dim(go.hoco)
#head(go.hoci, n=5)
#head(go.hocm, n=5)
#head(go.hoco, n=5)
compare.tot5 <- intersect(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],intersect(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                                rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)]))
compare.hoci.tot <- setdiff(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],c(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hoci
compare.hocm.tot <- setdiff(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hocm
compare.hoco.tot <- setdiff(rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)])) #DE hoco
length(compare.tot5)
length(compare.hoci.tot)
length(compare.hocm.tot)
length(compare.hoco.tot)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),], n=5)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),], n=5)
head(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),], n=5)
head(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),], n=5)
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
fracRibo(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
fracRibo(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5 <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goCCenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goCCenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
p.tot5 <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goBPenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goBPenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)]),
                                   length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)]),
                                   length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])),
                    Other=c((length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)])-length(compare.hoci.tot)-length(compare.tot5)),
                            (length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)])-length(compare.hocm.tot)-length(compare.tot5)),
                            (length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])-length(compare.hoco.tot)-length(compare.tot5))),
                    Shared=c(rep(length(compare.tot5),3)),
                    Unique=c(length(compare.hoci.tot),
                             length(compare.hocm.tot),
                             length(compare.hoco.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

#Examine differentially expressed genes using osteogenic ad hoc definitions
compare.pre <- readRDS("./data/cellranger-data-full/de.w-compare.pre.rds")
compare.obl <- readRDS("./data/cellranger-data-full/de.w-compare.obl.rds")
compare.emb <- readRDS("./data/cellranger-data-full/de.w-compare.emb.rds")
dim(compare.pre)
dim(compare.obl)
dim(compare.emb)
table(compare.pre$p_val_adj<0.05)
table(compare.obl$p_val_adj<0.05)
table(compare.emb$p_val_adj<0.05)
head(compare.pre, n=5)
head(compare.obl, n=5)
head(compare.emb, n=5)
compare.pre$gene <- rownames(compare.pre)
compare.obl$gene <- rownames(compare.obl)
compare.emb$gene <- rownames(compare.emb)
fracRibo(compare.pre)
fracRibo(compare.obl)
fracRibo(compare.emb)
p.pre <- goCCenrich(indv,compare.pre)
p.obl <- goCCenrich(indv,compare.obl)
p.emb <- goCCenrich(indv,compare.emb)
p.pre[[3]]
p.obl[[3]]
p.emb[[3]]
#go.pre <- funcEnrich(indv, compare.pre)
#go.obl <- funcEnrich(indv, compare.obl)
#go.emb <- funcEnrich(indv, compare.emb)
#dim(go.pre)
#dim(go.obl)
#dim(go.emb)
#head(go.pre, n=5)
#head(go.obl, n=5)
#head(go.emb, n=5)
compare.tot6 <- intersect(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                          intersect(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                    rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #shared DE genes
compare.pre.tot <- setdiff(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)], c(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                                                                       rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE pre
compare.obl.tot <- setdiff(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE obl
compare.emb.tot <- setdiff(rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)])) #DE emb
length(compare.tot6)
length(compare.pre.tot)
length(compare.obl.tot)
length(compare.emb.tot)
head(compare.pre[which(rownames(compare.pre)%in%compare.tot6),], n=5)
head(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),], n=5)
head(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),], n=5)
head(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),], n=5)
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
fracRibo(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
fracRibo(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6 <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goCCenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goCCenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
p.tot6 <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goBPenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goBPenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
DE.df <- data.frame(cell=(c("preosteoblast","osteoblast","embedding osteoblast")),
                    totalDEgenes=c(length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)]),
                                   length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)]),
                                   length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])),
                    Other=c((length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)])-length(compare.pre.tot)-length(compare.tot6)),
                            (length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)])-length(compare.obl.tot)-length(compare.tot6)),
                            (length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])-length(compare.emb.tot)-length(compare.tot6))),
                    Shared=c(rep(length(compare.tot6),3)),
                    Unique=c(length(compare.pre.tot),
                             length(compare.obl.tot),
                             length(compare.emb.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("preosteoblast","osteoblast","embedding osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

#Examine differentially expressed genes using grade of membership definitions
compare.gomi <- readRDS("./data/cellranger-data-full/de.w-compare.gomi.rds")
compare.gomm <- readRDS("./data/cellranger-data-full/de.w-compare.gomm.rds")
compare.gomo <- readRDS("./data/cellranger-data-full/de.w-compare.gomo.rds")
dim(compare.gomi)
dim(compare.gomm)
dim(compare.gomo)
table(compare.gomi$p_val_adj<0.05)
table(compare.gomm$p_val_adj<0.05)
table(compare.gomo$p_val_adj<0.05)
head(compare.gomi, n=5)
head(compare.gomm, n=5)
head(compare.gomo, n=5)
compare.gomi$gene <- rownames(compare.gomi)
compare.gomm$gene <- rownames(compare.gomm)
compare.gomo$gene <- rownames(compare.gomo)
fracRibo(compare.gomi)
fracRibo(compare.gomm)
fracRibo(compare.gomo)
p.gomi <- goCCenrich(indv,compare.gomi)
p.gomm <- goCCenrich(indv,compare.gomm)
p.gomo <- goCCenrich(indv,compare.gomo)
p.gomi[[3]]
p.gomm[[3]]
p.gomo[[3]]
#go.gomi <- funcEnrich(indv, compare.gomi)
#go.gomm <- funcEnrich(indv, compare.gomm)
#go.gomo <- funcEnrich(indv, compare.gomo)
#dim(go.gomi)
#dim(go.gomm)
#dim(go.gomo)
#head(go.gomi, n=5)
#head(go.gomm, n=5)
#head(go.gomo, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.o$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

```

```{r NEED TO FIX FOR MY DATA}

ContigencyTable <- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)
rownames(ContigencyTable) <- c("Chimp eGene", "Not Chimp eGene")
colnames(ContigencyTable) <- c("Human eGene", "Not human eGene")
#what is qval threshold for human eGene classification in this contigency table
print(GtexHeartEgenes %>% top_n(-HumanTopN, qval) %>% top_n(1, qval) %>% pull(qval))
#Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
ContigencyTable
#One-sided Fisher test for greater overlap than expected by chance
fisher.test(ContigencyTable, alternative="greater")

#The above contingency table and one-sided fisher test indicated a greater-than-chance overlap between the sets of eGenes in chimp and human.

```


