---
title: "10x-plot-diffexp"
author: "Genevieve Housman"
date: "July 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Differential Expression

View results of pairwise differential expression tests between humans and chimpanzees within different cell classifications.

```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ggsignif)

```

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate <- paste0(data@meta.data$Individual,".",data@meta.data$Replicate)

```

```{r}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

```{r}

#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot <- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale="free") +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

#Define function to plot GO functional enrichment results (specific ontology only)
goEnPlot.spec <- function(goTot, title, ontology) {
  print(ggplot(goTot[which(goTot$ONTOLOGY==ontology),], aes(x=GeneRatio, y=Description, color=p.adjust, size=Count)) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

```

## Load External Gene Datasets

Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.

Genes previously identified as human and chimpanzee differences:

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR < 1% out of 10304 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each cell type
    
* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR < 1% out of 12184 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each tissue type

* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + DE genes in parental iPSC lines (n = 7655)
    + DE genes in hybrid iPSC lines (n = 6009)
    + DE genes in parental CNCC lines (n = 2223)
    + DE genes in hybrid CNCC lines (n = 3612)
    + Note: only considered DE genes unique to each cell type
    
* Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
    + DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)
    + AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)
    + (human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)
    + (chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)
    + Note: only DMRs that overlapped promotor regions of genes were included
    + Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels

General skeleton-related and skeletal phenotype-related genes:

* Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
    + hip geometry GWAS loci (n = 10)

* Styrkarsdottir et al. 2019 Nature Communications (Table 1)
    + DXA bone area GWAS loci (n = 21 associated with 20 genes)

* Yengo et al. 2018 Human Molecular Genetics
    + height GWAS loci (n = 3290) (data provided at https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
    + height GWAS loci with associated eQTLs (n = 610) (list not provided)
    + Note: only used unique gene symbols associated with height GWAS loci (n = 1500)

Genes associated with skeletal diseases (OA + OP):

* Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
    + genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)
    
* Morris et al. 2019 Nature Genetics (Supplemental Table 2)
    + OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)
    
Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:

* Gallego Romero et al. 2015 eLife (Supplementary File 3)
    + DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR < 1% out of 12171 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + REMOVED because subsequent studies (Blake et al. 2018 & Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm & CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type
    
* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
    + Skeleton-related genes (n = 1790)
    + Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)
    + REMOVED because it is unclear where these genes came from to interpreting biology is difficult

Some external gene datasets were considered but not examined:

* Ward et al. 2019 eLife
    +  DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR < 10% out of 11974 genes)

* Shibata et al. 2012 PLOS Genetics
    + increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)

* Edsall et al. 2019 Genome Biology and Evolution
    + increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)

* Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
    + increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)

* Prescott et al. 2015 Cell (Supplemental)
    + top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)
    + top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)

* Joganic et al. 2017 American Journal of Physical Anthropology
    + genetics of craniofacial skeleton in baboons

* CibriÃ¡n Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
    + robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)

* Tachmazidou et al. 2019 Nature Genetics
    + OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])
    + OA loci (Supplementary Table 2)
    + established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)
    + Note: used loci compiled by AH
    + (OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)
    + (OA.estab = gene symbol(s) nearest to established OA loci)
    + (OA.broad = genes within 500kb of OA loci)

```{r}

#Define function for testing significant enrichment of gene sets

genesetEnrich <- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative="greater"))

}

genesetEnrich.data <- function(InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test and output data
  data <- data.frame("Source.DEGenes"=InternalDataSource,
                     "Source.GenesOfInterest"=ExternalDataSource,
                     "DE.Interest"=DE.Interest,
                     "NonDE.Interest"=NonDE.Interest,
                     "DE.NotInterest"=DE.NotInterest,
                     "NonDE.NotInterest"=NonDE.NotInterest,
                     "p.value"=fisher.test(ContingencyTable, alternative="greater")$p.value,
                     "odds.ratio"=fisher.test(ContingencyTable, alternative="greater")$estimate,
                     row.names=NULL)
  return(data)

}

```

```{r}

#Load external datasets (manually currated from papers)
gallegoromero2015    <- read.csv("./data/external-gene-sets/GallegoRomero2015eLife.csv")
blake2018            <- read.csv("./data/external-gene-sets/Blake2018GenBio.csv")
blake2020            <- read.csv("./data/external-gene-sets/Blake2020GenRes.csv")
gokhman2021          <- read.csv("./data/external-gene-sets/Gokhman2021NatGen.csv")
gokhman2020          <- read.csv("./data/external-gene-sets/Gokhman2020NatComm.csv")
gokhman2020.skeleton <- read.csv("./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv")
hsu2019              <- read.csv("./data/external-gene-sets/Hsu2019JBoneMinRes.csv")
styrkarsdottir2019   <- read.csv("./data/external-gene-sets/Styrkarsdottir2019NatComm.csv")
yengo2018            <- read.csv("./data/external-gene-sets/Yengo2018HumMolGen.csv")
steinberg2020        <- read.csv("./data/external-gene-sets/Steinberg2020biorxiv.csv")
morris2019           <- read.csv("./data/external-gene-sets/Morris2019NatGen.csv")
#cibrianuhalte2017    <- read.csv("./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv")
#tachmazidou2019      <- read.csv("./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv")

```

```{r, eval=FALSE}

#genes previously identified as human and chimpanzee differences

#genesExternal <- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal <- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal <- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal <- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1 &
                                      blake2018$DE.Day1==0 &
                            					blake2018$DE.Day2==0 &
                            					blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==1 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==1 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==1 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==1 &
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                               gokhman2021$DE.Hybrid.CNCCs==0)]   #hybrid.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal <- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal <- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal <- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal <- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal <- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal <- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS

```

## Examine output of DE analyses

```{r}

#Load DE results

compare1  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.rds")
compare2  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.rds")
compare3  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhoc.rds")
compare4  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.rds")
compare5  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.75.rds")
compare6  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.T2.rds")
compare7  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.75.T2.rds")
compare8  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.rds")
compare9  <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocX.rds")
compare10 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.T2.rds")
compare11 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocX.T2.rds")

DEdata <- rbind(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9,compare10,compare11)
colnames(DEdata)[5] <- "FDR"
DEdata <- as_tibble(DEdata)

rm(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9,compare10,compare11)

```

```{r}

#Isolate significant DE genes

DEdataSig <- DEdata %>% filter(FDR<=0.01)
#DEdataSig <- DEdata %>% filter(FDR<=0.05)
#DEdataSig <- DEdata %>% filter(abs(logFC)>=1.2 & FDR<=0.01)

```

## Examine Numbers, Overlap, and logFC of Pairwise DE Genes

### Stages of Differentiation at Collection

DE Genes Counts

* Time 0: (FDR<0.01) = 2098 / (FDR<0.05) = 4002
* Time 1: (FDR<0.01) =  904 / (FDR<0.05) = 2441
* Time 2: (FDR<0.01) =  446 / (FDR<0.05) = 1603 

Comparison in Absolute LogFC of DE Genes

* Time 0 vs. Time 1: (FDR<0.01): p = 3.3e-11 / (FDR<0.05): p < 2.2e-16 **
* Time 1 vs. Time 2: (FDR<0.01): p = 8.2e-06 / (FDR<0.05): p = 3.3e-15 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    labels=c("Time 0","Time 1","Time 2"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0"),
                         list("Time.1"),
                         list("Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.1","Time.2"),
                         list("Time.0","Time.2"),
                         list("Time.0","Time.1","Time.2")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))),
#                  category.names = c("Time 0","Time 1","Time 2"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("pink","mediumorchid","dodgerblue"),
#                  fill=c("pink","mediumorchid","dodgerblue"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    labels=c("Time 0","Time 1","Time 2"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10)) +
  geom_signif(comparisons = list(c("Time 0","Time 1"),
                                 c("Time 1","Time 2")), 
              test="t.test",
              test.args=list(alternative="two.sided"),
              map_signif_level=TRUE)

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 0")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 1")])) #p=3.3e-11
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 2")])) #p=8.2e-06

```

```{r, eval=FALSE}

DEdataCells <-
  DEdata %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 0")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 0")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: Time 0",
       ylim=c(0,10),
       xlim=c(0,4.75))
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 1")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 1")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: Time 1",
       ylim=c(0,8.5),
       xlim=c(0,4.5))
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 2")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="Time 2")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: Time 2")
abline(0,1)

```

### General Clusters (res=0.05)

DE Genes

* FDR < 0.01
    + iPSC.c1:       (FDR<0.01) = 2155 / (FDR<0.05) = 4093
    + iPSC.c2:       (FDR<0.01) =  413 / (FDR<0.05) = 1168
    + iPSC.c3:       (FDR<0.01) =    0 / (FDR<0.05) =   81
    + MSC.c1:        (FDR<0.01) =  936 / (FDR<0.05) = 2494
    + Osteogenic.c1: (FDR<0.01) =  534 / (FDR<0.05) = 1807
    + Osteogenic.c2: (FDR<0.01) =  209 / (FDR<0.05) = 1120

Comparison in Absolute LogFC of DE Genes

* iPSC.c1 vs. MSC.c1:        (FDR<0.01): p = 5.3e-11 / (FDR<0.05): p < 2.2e-16 **
* MSC.c1  vs. Osteogenic.c1: (FDR<0.01): p = 8.9e-06 / (FDR<0.05): p = 8.1e-12 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                    values=c("pink","pink3","mediumorchid","dodgerblue","dodgerblue4")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                    values=c("pink","pink3","mediumorchid","dodgerblue","dodgerblue4")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="iPSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]))       #p=5.3e-11
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")])) #p=8.9e-06

```

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset %in% c("iPSC.c1","MSC.c1","Osteogenic.c1")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC.c1") %>% dplyr::select(gene), "iPSC.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="MSC.c1") %>% dplyr::select(gene), "MSC.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1"),
                         list("MSC.c1"),
                         list("Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1"),
                         list("MSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1","Osteogenic.c1")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="iPSC.c1") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="MSC.c1") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene))),
#                  category.names = c("iPSC.c1","MSC.c1","Osteogenic.c1"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("pink","mediumorchid","dodgerblue"),
#                  fill=c("pink","mediumorchid","dodgerblue"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="iPSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]))

```

### General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)

DE Genes

* iPSC       (FDR<0.01) = 2128 / (FDR<0.05) = 4009
* MSC        (FDR<0.01) =  963 / (FDR<0.05) = 2554
* Osteogenic (FDR<0.01) =  613 / (FDR<0.05) = 2005
    
Comparison in Absolute LogFC of DE Genes

* iPSC vs. MSC:        (FDR<0.01): p = 4.3e-09 / (FDR<0.05): p < 2.2e-16 **
* MSC  vs. Osteogenic: (FDR<0.01): p = 1.8e-06 / (FDR<0.05): p = 4.0e-12 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="adhoc") %>%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Ad Hoc",
                    labels=c("iPSC","MSC","Osteogenic"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC") %>% dplyr::select(gene), "iPSC"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="MSC") %>% dplyr::select(gene), "MSC"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene), "Osteogenic"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC"),
                         list("MSC"),
                         list("Osteogenic"),
                         list("iPSC","MSC"),
                         list("MSC","Osteogenic"),
                         list("iPSC","Osteogenic"),
                         list("iPSC","MSC","Osteogenic")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="iPSC") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="MSC") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene))),
#                  category.names = c("iPSC","MSC","Osteogenic"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("pink","mediumorchid","dodgerblue"),
#                  fill=c("pink","mediumorchid","dodgerblue"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Ad Hoc",
                    labels=c("iPSC","MSC","Osteogenic"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="iPSC")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC")]))       #p=4.344e-09
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic")])) #p=1.753e-06

```

### Osteogenic Clusters (res=0.50)

DE Genes

* Osteogenic.c1 (FDR<0.01) = 643 / (FDR<0.05) = 2057
* Osteogenic.c2 (FDR<0.01) =   1 / (FDR<0.05) =    9
* Osteogenic.c3 (FDR<0.01) = 428 / (FDR<0.05) = 1562
* Osteogenic.c4 (FDR<0.01) = 217 / (FDR<0.05) = 1087

Comparison in Absolute LogFC of DE Genes

* Osteogenic.c1 vs. Osteogenic.c3: (FDR<0.01): p = 0.00311 / (FDR<0.05): p = 0.00518 **
* Osteogenic.c3 vs. Osteogenic.c4: (FDR<0.01): p = 0.01908 / (FDR<0.05): p = 0.12870 *

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostcluster0.50") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene))),
#                  category.names = c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]))

```

### Osteogenic Clusters (res=0.75)

DE Genes

* Osteogenic.c1 (FDR<0.01) =  364 / (FDR<0.05) = 1478
* Osteogenic.c2 (FDR<0.01) =  433 / (FDR<0.05) = 1773
* Osteogenic.c3 (FDR<0.01) =  427 / (FDR<0.05) = 1493
* Osteogenic.c4 (FDR<0.01) =  196 / (FDR<0.05) = 1123
* Osteogenic.c5 (FDR<0.01) =  271 / (FDR<0.05) = 1151
* Osteogenic.c6 (FDR<0.01) =    1 / (FDR<0.05) =    2
* Osteogenic.c7 (FDR<0.01) =    1 / (FDR<0.05) =    1
* Osteogenic.c8 (FDR<0.01) =    1 / (FDR<0.05) =    3

Comparison in Absolute LogFC of DE Genes

* Osteogenic.c1 vs. Osteogenic.c2: (FDR<0.01): p = 0.91230 / (FDR<0.05): p = 0.83610
* Osteogenic.c2 vs. Osteogenic.c3: (FDR<0.01): p = 3.5e-07 / (FDR<0.05): p = 1.2e-10 **
* Osteogenic.c3 vs. Osteogenic.c4: (FDR<0.01): p = 0.12800 / (FDR<0.05): p = 0.00086  *
* Osteogenic.c4 vs. Osteogenic.c5: (FDR<0.01): p = 0.05663 / (FDR<0.05): p = 4.7e-05  *

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostcluster0.75") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                       "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                    values=c("dodgerblue","dodgerblue4","deepskyblue2","navyblue","lightblue1","royalblue3","grey","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene), "Osteogenic.c5"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c6") %>% dplyr::select(gene), "Osteogenic.c6"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c7") %>% dplyr::select(gene), "Osteogenic.c7"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c8") %>% dplyr::select(gene), "Osteogenic.c8"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"))
dev.off()

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                    values=c("dodgerblue","dodgerblue4","deepskyblue2","navyblue","lightblue1","royalblue3","grey","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c5")]))

```

### Osteogenic Clusters (res=0.50) (Time 2 only)

DE Genes

* Osteogenic.c1 (FDR<0.01) = 545 / (FDR<0.05) = 1796
* Osteogenic.c2 (FDR<0.01) =   1 / (FDR<0.05) =    9
* Osteogenic.c3 (FDR<0.01) = 388 / (FDR<0.05) = 1513
* Osteogenic.c4 (FDR<0.01) =  79 / (FDR<0.05) =  406

Comparison in Absolute LogFC of DE Genes

* Osteogenic.c1 vs. Osteogenic.c3: (FDR<0.01): p = 0.02433 / (FDR<0.05): p = 0.07446 *
* Osteogenic.c3 vs. Osteogenic.c4: (FDR<0.01): p = 2.7e-06 / (FDR<0.05): p = 9.2e-13 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostcluster0.50.T2") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene))),
#                  category.names = c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]))

```

### Osteogenic Clusters (res=0.75) (Time 2 only)

DE Genes

* Osteogenic.c1 (FDR<0.01) =  210 / (FDR<0.05) =  833
* Osteogenic.c2 (FDR<0.01) =  369 / (FDR<0.05) = 1397
* Osteogenic.c3 (FDR<0.01) =  322 / (FDR<0.05) = 1041
* Osteogenic.c4 (FDR<0.01) =   54 / (FDR<0.05) =  387
* Osteogenic.c5 (FDR<0.01) =  268 / (FDR<0.05) = 1078
* Osteogenic.c6 (FDR<0.01) =    1 / (FDR<0.05) =    2
* Osteogenic.c7 (FDR<0.01) =    0 / (FDR<0.05) =    0
* Osteogenic.c8 (FDR<0.01) =    0 / (FDR<0.05) =    1

Comparison in Absolute LogFC of DE Genes

* Osteogenic.c1 vs. Osteogenic.c2: (FDR<0.01): p = 0.10230 / (FDR<0.05): p = 0.11000
* Osteogenic.c2 vs. Osteogenic.c3: (FDR<0.01): p = 4.7e-06 / (FDR<0.05): p = 6.3e-10 **
* Osteogenic.c3 vs. Osteogenic.c4: (FDR<0.01): p = 0.02699 / (FDR<0.05): p = 0.30090 *
* Osteogenic.c4 vs. Osteogenic.c5: (FDR<0.01): p = 1.7e-05 / (FDR<0.05): p = 3.5e-15 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostcluster0.75.T2") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                       "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                    values=c("dodgerblue","dodgerblue4","deepskyblue2","navyblue","lightblue1","royalblue3","grey","black")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene), "Osteogenic.c5"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c6") %>% dplyr::select(gene), "Osteogenic.c6"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
             "Osteogenic.c5","Osteogenic.c6"))
dev.off()

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                    values=c("dodgerblue","dodgerblue4","deepskyblue2","navyblue","lightblue1","royalblue3","grey","black")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c4")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c5")]))

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)

DE Genes

* pre (FDR<0.01) =  144 / (FDR<0.05) =  657
* obl (FDR<0.01) =  242 / (FDR<0.05) =  996
* emb (FDR<0.01) =  142 / (FDR<0.05) =  772
* min (FDR<0.01) =  429 / (FDR<0.05) = 1569
* mat (FDR<0.01) =  115 / (FDR<0.05) =  761

Comparison in Absolute LogFC of DE Genes

* pre vs. obl: (FDR<0.01): p = 0.71640 / (FDR<0.05): p = 0.87480
* obl vs. emb: (FDR<0.01): p = 0.47320 / (FDR<0.05): p = 0.55990
* emb vs. min: (FDR<0.01): p = 0.02644 / (FDR<0.05): p = 0.00020 **
* min vs. mat: (FDR<0.01): p = 0.01655 / (FDR<0.05): p = 2.4e-06 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostadhoc") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
#                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))

```

```{r, eval=FALSE}

DEdataCells <-
  DEdata %>%
    dplyr::filter(cell.assign=="ostadhoc") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="preosteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="preosteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: preosteoblast")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: osteoblast")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="embedding osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="embedding osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: embedding osteoblast")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="mineralizing osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="mineralizing osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: mineralizing osteoblast")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="maturing osteocyte")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="maturing osteocyte")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: maturing osteocyte")
abline(0,1)

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)

DE Genes

* pre (FDR<0.01) =  144 / (FDR<0.05) =  657
* obl (FDR<0.01) =  134 / (FDR<0.05) =  680
* emb (FDR<0.01) =   36 / (FDR<0.05) =  229
* min (FDR<0.01) =   57 / (FDR<0.05) =  138
* mat (FDR<0.01) =    1 / (FDR<0.05) =    6

Comparison in Absolute LogFC of DE Genes

* pre vs. obl: (FDR<0.01): p = 0.71240 / (FDR<0.05): p = 0.37970
* obl vs. emb: (FDR<0.01): p = 0.01529 / (FDR<0.05): p = 4.5e-08 **
* emb vs. min: (FDR<0.01): p = 0.00481 / (FDR<0.05): p < 2.2e-16 **
* min vs. mat: (FDR<0.01): p = n/a     / (FDR<0.05): p = 0.84870

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostadhocX") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
#                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]))
#t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)

DE Genes

* pre (FDR<0.01) =  121 / (FDR<0.05) =  552
* obl (FDR<0.01) =  235 / (FDR<0.05) =  951
* emb (FDR<0.01) =   74 / (FDR<0.05) =  502
* min (FDR<0.01) =  241 / (FDR<0.05) =  935
* mat (FDR<0.01) =   34 / (FDR<0.05) =  378

Comparison in Absolute LogFC of DE Genes

* pre vs. obl: (FDR<0.01): p = 0.51260 / (FDR<0.05): p = 0.75750
* obl vs. emb: (FDR<0.01): p = 0.03799 / (FDR<0.05): p = 0.14660 *
* emb vs. min: (FDR<0.01): p = 0.00677 / (FDR<0.05): p = 0.00106 **
* min vs. mat: (FDR<0.01): p = 0.00015 / (FDR<0.05): p = 1.3e-07 **

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostadhoc.T2") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
#                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))

```

```{r, eval=FALSE}

DEdataCells <-
  DEdata %>%
    dplyr::filter(cell.assign=="ostadhoc.T2") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)

#FDR Distribution
ggplot(DEdataCells, aes(x=FDR)) +
  geom_histogram() +
  facet_grid(~cell.subset)

#Volcano plot
ggplot(DEdataCells, aes(x=logFC,y=-log10(FDR))) +
  geom_point() +
  facet_grid(~cell.subset)

#qqplot
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="preosteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="preosteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: preosteoblast (t2)")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: osteoblast (t2)")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="embedding osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="embedding osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: embedding osteoblast (t2)")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="mineralizing osteoblast")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="mineralizing osteoblast")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: mineralizing osteoblast (t2)")
abline(0,1)
qqplot(-log10(runif(length(DEdataCells$FDR[which(DEdataCells$cell.subset=="maturing osteocyte")]))),
       -log10(DEdataCells$FDR[which(DEdataCells$cell.subset=="maturing osteocyte")]),
       xlab="-log10(Uniform)",
       ylab="-log10(FDR)",
       main="QQ Plot: maturing osteocyte (t2)")
abline(0,1)

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)

DE Genes

* pre (FDR<0.01) =  121 / (FDR<0.05) =  552
* obl (FDR<0.01) =  121 / (FDR<0.05) =  624
* emb (FDR<0.01) =   36 / (FDR<0.05) =  239
* min (FDR<0.01) =   57 / (FDR<0.05) =  144
* mat (FDR<0.01) =    0 / (FDR<0.05) =    0

Comparison in Absolute LogFC of DE Genes

* pre vs. obl: (FDR<0.01): p = 0.86120 / (FDR<0.05): p = 0.49660
* obl vs. emb: (FDR<0.01): p = 0.03134 / (FDR<0.05): p = 3.8e-07 **
* emb vs. min: (FDR<0.01): p = 0.00306 / (FDR<0.05): p < 2.2e-16 **
* min vs. mat: (FDR<0.01): p = n/a     / (FDR<0.05): p = n/a

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostadhocX.T2") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
table(DEgenes$cell.subset)
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast")),
      text.scale=2)
dev.off()
#v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
#                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
#                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
#                  filename=NULL,
#                  scaled=TRUE,
#                  euler.d=TRUE,
#                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
#                  fontfamily="sans",
#                  cat.fontfamily="sans")
#grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  scale_x_discrete(drop=FALSE) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]))

```

## Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

### Stages of Differentiation at Collection

Pairwise DE Genes

* Shared across Stages:      (FDR<0.01)                =  150 / (FDR<0.05)                =  530
* Shared across Times 0 & 1: (FDR<0.01)                =  250 / (FDR<0.05)                =  637
* Shared across Times 1 & 2: (FDR<0.01)                =  108 / (FDR<0.05)                =  393
* Unique to Time 0:          (FDR<0.01)                = 1648 / (FDR<0.05)                = 2640
* Unique to Time 1:          (FDR<0.01)                =  396 / (FDR<0.05)                =  881
* Unique to Time 2:          (FDR<0.01)                =  138 / (FDR<0.05)                =  485
* Shared across Stages:      (FDR<0.01, tested in all) =  150 / (FDR<0.05)                =  530
* Shared across Times 0 & 1: (FDR<0.01, tested in all) =  248 / (FDR<0.05)                =  633
* Shared across Times 1 & 2: (FDR<0.01, tested in all) =  104 / (FDR<0.05)                =  364
* Unique to Time 0:          (FDR<0.01, tested in all) = 1297 / (FDR<0.05, tested in all) = 2012
* Unique to Time 1:          (FDR<0.01, tested in all) =  369 / (FDR<0.05, tested in all) =  805
* Unique to Time 2:          (FDR<0.01, tested in all) =  115 / (FDR<0.05, tested in all) =  415

```{r}

#Pairwise DE genes and cell subset overlap
DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboGenes <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                          ), full_join)
comboGenes[is.na(comboGenes)] <- 0

#Pairwise DE genes Shared across All Stages
share.stg <- DEgenes %>% 
               filter(gene %in%
                        pull(comboGenes %>%
                               filter(Time.0==1,
                                      Time.1==1,
                                      Time.2==1) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))

#Pairwise DE genes Shared across Time 0 and Time 1
share.stg.t01 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(Time.0==1,
                                          Time.1==1,
                                          Time.2==0) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))

#Pairwise DE genes Shared across Time 1 and Time 2
share.stg.t12 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(Time.0==0,
                                          Time.1==1,
                                          Time.2==1) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))

#Pairwise DE genes Unique to Time 0
uniq.stg.t0 <- DEgenes %>% 
                 filter(cell.subset=="Time 0",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(Time.0==1,
                                        Time.1==0,
                                        Time.2==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)

#Pairwise DE genes Unique to Time 1
uniq.stg.t1 <- DEgenes %>% 
                 filter(cell.subset=="Time 1",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(Time.0==0,
                                        Time.1==1,
                                        Time.2==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)

#Pairwise DE genes Unique to Time 2
uniq.stg.t2 <- DEgenes %>% 
                 filter(cell.subset=="Time 2",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(Time.0==0,
                                        Time.1==0,
                                        Time.2==1) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)

#Isolate genes tested in all cell classifications
DEtests <-
  DEdata %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests <- reduce(list(data.frame(gene=DEtests %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                          ), full_join)
comboTests[is.na(comboTests)] <- 0
tests <- pull(comboTests %>%
                filter(Time.0==1,
                       Time.1==1,
                       Time.2==1) %>%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg <- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 <- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 <- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 <- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 <- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 <- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))
length(unique(shareX.stg.t01$gene))
length(unique(shareX.stg.t12$gene))
length(uniqX.stg.t0$gene)
length(uniqX.stg.t1$gene)
length(uniqX.stg.t2$gene)

```

```{r, eval=FALSE}

#Save DE gene lists
write.csv(tests,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-tests.csv")
write.csv(shareX.stg,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.csv")
write.csv(shareX.stg.t01,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.t01.csv")
write.csv(shareX.stg.t12,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-shareX.stg.t12.csv")
write.csv(uniqX.stg.t0,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t0.csv")
write.csv(uniqX.stg.t1,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t1.csv")
write.csv(uniqX.stg.t2,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.DE-fdr0.01-uniqX.stg.t2.csv")

```

Get psuedobulk data for viewing gene expression differences.

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Time 0","Time 1","Time 2")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg <- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Stage")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
#log2 transformed cpm
tmm <- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
##boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.stg)

```

Look at gene expression levels of interesting DE genes.

```{r, eval=FALSE}

#Reformat pseudobulk data for plotting
tmm$gene <- rownames(tmm)
tmm.melt <- melt(tmm,id.vars="gene",measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][1])
tmm.melt$stage <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][2])
tmm.melt$species <- ifelse(grepl("H",tmm.melt$ind.rep),"Human","Chimp")

#Plot gene expression levels
pdf(file="./output/de-shareX.stg-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-shareX.stg.t01-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg.t01$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-shareX.stg.t12-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.stg.t12$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.stg.t0-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t0$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.stg.t1-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t1$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.stg.t2-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.stg.t2$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Stages (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")
#goEnPlot.spec(goTot,title="Shared in All",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in All",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in All",ontology="MF")

#Shared across Time 0 & Time 1 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in Time 0 & Time 1")
#goEnPlot.spec(goTot,title="Shared in Time 0 & Time 1",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in Time 0 & Time 1",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in Time 0 & Time 1",ontology="MF")

#Shared across Time 1 & Time 2 (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in Time 1 & Time 2")
#goEnPlot.spec(goTot,title="Shared in Time 1 & Time 2",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in Time 1 & Time 2",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in Time 1 & Time 2",ontology="MF")

#Unique to Time 0 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Time 0")
#goEnPlot.spec(goTot,title="Unique to Time 0",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Time 0",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Time 0",ontology="MF")

#Unique to Time 1 (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to Time 1")
#goEnPlot.spec(goTot,title="Unique to Time 1",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Time 1",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Time 1",ontology="MF")

#Unique to Time 2 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Time 2")
#goEnPlot.spec(goTot,title="Unique to Time 2",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Time 2",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Time 2",ontology="MF")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

```{r, eval=FALSE}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All Time 0 DE Genes
  print("All Time 0 DE Genes")
  tempDE <- pull(DEdataSig %>%
                  dplyr::filter(cell.assign=="stage",cell.subset=="Time 0") %>%
                  dplyr::select(gene))
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 0" &
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  #All Time 1 DE Genes
  print("All Time 1 DE Genes")
  tempDE <- pull(DEdataSig %>%
                  dplyr::filter(cell.assign=="stage",cell.subset=="Time 1") %>%
                  dplyr::select(gene))
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 1" &
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  #All Time 2 DE Genes
  print("All Time 2 DE Genes")
  tempDE <- pull(DEdataSig %>%
                  dplyr::filter(cell.assign=="stage",cell.subset=="Time 2") %>%
                  dplyr::select(gene))
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 2" &
                                  DEdata$gene %in% tempDE == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Unique to Time 0
  print("Unique to Time 0")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 0" &
                                  DEdata$gene %in% uniq.stg.t0$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t0$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 1
  print("Unique to Time 1")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 1" &
                                  DEdata$gene %in% uniq.stg.t1$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t1$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 2
  print("Unique to Time 2")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.subset=="Time 2" &
                                  DEdata$gene %in% uniq.stg.t2$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.stg.t2$gene,
                NonDEGenes=nonDEgenes)
  
  #Shared in All Stages (tested in all classifications)
  print("Shared in All Stages, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.stg$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg$gene,
                NonDEGenes=nonDEgenes)
  #Shared in Time 0 & Time 1 (tested in all classifications)
  print("Shared in Time 0 & Time 1, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg.t01$gene,
                NonDEGenes=nonDEgenes)
  #Shared in Time 1 & Time 2 (tested in all classifications)
  print("Shared in Time 1 & Time 2, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.stg.t12$gene,
                NonDEGenes=nonDEgenes)  
  
  #Unique to Time 0 (tested in all classifications)
  print("Unique to Time 0, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t0$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 1 (tested in all classifications)
  print("Unique to Time 1, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t1$gene,
                NonDEGenes=nonDEgenes)
  #Unique to Time 2 (tested in all classifications)
  print("Unique to Time 2, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.stg.t2$gene,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared in All Stages (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 0 & Time 1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 0 & Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 1 & Time 2 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 1 & Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to Time 0 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 0",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 2 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)

```

```{r}

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)

```

```{r}

tempSource <- c("Blake2018-DE.iPSCs-Unique",
                "Gokhman2021-DE.Parental.iPSCs-Unique",
                "Gokhman2021-DE.Hybrid.iPSCs-Unique",
                "Blake2018-DE.iPSC-endoderm-day1-Unique",
                "Blake2018-DE.iPSC-endoderm-day2-Unique",
                "Blake2018-DE.iPSC-endoderm-day3-Unique",
                "Blake2020-DE.heart-Unique",
                "Blake2020-DE.kidney-Unique",
                "Blake2020-DE.liver-Unique",
                "Blake2020-DE.lung-Unique")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)

tempSource <- c("Gokhman2021-DE.Parental.CNCCs-Unique",
                "Gokhman2021-DE.Hybrid.CNCCs-Unique",
                "Gokhman2020-DMR.bone",
                "Hsu2019-GWAS.hip.geometry",
                "Styrkarsdottir2019-GWAS.bone.area",
                "Yengo2018-GWAS.height",
                "Steinberg2020-GWAS.OA",
                "Morris2020-GWAS.OP")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)

```

Print p-values for significant enrichments:

```{r}

tbl <- DEenrich[which(DEenrich$p.value<0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]

```

Save enrichment results:

```{r}

#Save enrichment results
write.csv(DEenrich,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.enrichment.csv")

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared in All Stages (tested in all classifications)
  print("Shared in All Stages, tested in all classifications")
  print(intersect(genesExternal[[2]],unique(shareX.stg$gene)))

  #Shared in Time 0 & Time 1 (tested in all classifications)
  print("Shared in Time 0 & Time 1 (tested in all classifications)")
  print(intersect(genesExternal[[2]],unique(shareX.stg.t01$gene)))
  
  #Shared in Time 1 & Time 2 (tested in all classifications)
  print("Shared in Time 1 & Time 2, tested in all classifications")
  print(intersect(genesExternal[[2]],unique(shareX.stg.t12$gene)))

  #Unique to Time 0 (tested in all classifications)
  print("Unique to Time 0 (tested in all classifications)")
  print(intersect(genesExternal[[2]],uniqX.stg.t0$gene))

  #Unique to Time 1 (tested in all classifications)
  print("Unique to Time 1 (tested in all classifications)")
  print(intersect(genesExternal[[2]],uniqX.stg.t1$gene))
  
  #Unique to Time 2 (tested in all classifications)
  print("Unique to Time 2 (tested in all classifications)")
  print(intersect(genesExternal[[2]],uniqX.stg.t2$gene))

}

```

### General Clusters (res=0.05)

Pairwise DE Genes

```{r}

# DE genes and cell subset overlap
DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::select(cell.subset,gene,logFC)
comboGenes <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC.c1")       %>% dplyr::select(gene), "iPSC.c1"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="MSC.c1")        %>% dplyr::select(gene), "MSC.c1"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1)
                          ), full_join)
comboGenes[is.na(comboGenes)] <- 0

#Pairwise DE genes Shared across All Clusters (subset)
share.stg <- DEgenes %>% 
               filter(gene %in%
                        pull(comboGenes %>%
                               filter(iPSC.c1==1,
                                      MSC.c1==1,
                                      Osteogenic.c1==1) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))

#Pairwise DE genes Shared across iPSC.c1 and MSC.c1
share.stg.t01 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(iPSC.c1==1,
                                          MSC.c1==1,
                                          Osteogenic.c1==0) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))

#Pairwise DE genes Shared across MSC.c1 and Osteogenic.c1
share.stg.t12 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(iPSC.c1==0,
                                          MSC.c1==1,
                                          Osteogenic.c1==1) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))

#Pairwise DE genes Unique to iPSC.c1
uniq.stg.t0 <- DEgenes %>% 
                 filter(cell.subset=="iPSC.c1",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC.c1==1,
                                        MSC.c1==0,
                                        Osteogenic.c1==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)

#Pairwise DE genes Unique to MSC.c1
uniq.stg.t1 <- DEgenes %>% 
                 filter(cell.subset=="MSC.c1",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC.c1==0,
                                        MSC.c1==1,
                                        Osteogenic.c1==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)

#Pairwise DE genes Unique to Osteogenic.c1
uniq.stg.t2 <- DEgenes %>% 
                 filter(cell.subset=="Osteogenic.c1",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC.c1==0,
                                        MSC.c1==0,
                                        Osteogenic.c1==1) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)

#Isolate genes tested in all cell classifications
DEtests <-
  DEdata %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests <- reduce(list(data.frame(gene=DEtests %>% filter(cell.subset=="iPSC.c1") %>% dplyr::select(gene), "iPSC.c1"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="MSC.c1") %>% dplyr::select(gene), "MSC.c1"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1)
                          ), full_join)
comboTests[is.na(comboTests)] <- 0
tests <- pull(comboTests %>%
                filter(iPSC.c1==1,
                       MSC.c1==1,
                       Osteogenic.c1==1) %>%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg <- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 <- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 <- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 <- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 <- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 <- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))
length(unique(shareX.stg.t01$gene))
length(unique(shareX.stg.t12$gene))
length(uniqX.stg.t0$gene)
length(uniqX.stg.t1$gene)
length(uniqX.stg.t2$gene)

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Clusters (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")
#goEnPlot.spec(goTot,title="Shared in All",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in All",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in All",ontology="MF")

#Shared in iPSC.c1 and MSC.c1 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in iPSC.c1 & MSC.c1")
#goEnPlot.spec(goTot,title="Shared in iPSC.c1 & MSC.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in iPSC.c1 & MSC.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in iPSC.c1 & MSC.c1",ontology="MF")

#Shared in MSC.c1 and Osteogenic.c1 (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in MSC.c1 & Osteo.c1")
#goEnPlot.spec(goTot,title="Shared in MSC.c1 & Osteo.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in MSC.c1 & Osteo.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in MSC.c1 & Osteo.c1",ontology="MF")

#Unique to iPSC.c1 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to iPSC.c1")
#goEnPlot.spec(goTot,title="Unique to iPSC.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to iPSC.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to iPSC.c1",ontology="MF")

#Unique to MSC.c1 (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to MSC.c1")
#goEnPlot.spec(goTot,title="Unique to MSC.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to MSC.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to MSC.c1",ontology="MF")

#Unique to Osteogenic.c1 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Osteogenic.c1")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="MF")

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared in All Clusters (subset) (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in iPSC.c1 & MSC.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in iPSC.c1 & MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in MSC.c1 & Osteogenic.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in MSC.c1 & Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to iPSC.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to iPSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to MSC.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)

```

```{r}

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)

```

```{r}

tempSource <- c("Blake2018-DE.iPSCs-Unique",
                "Gokhman2021-DE.Parental.iPSCs-Unique",
                "Gokhman2021-DE.Hybrid.iPSCs-Unique",
                "Blake2018-DE.iPSC-endoderm-day1-Unique",
                "Blake2018-DE.iPSC-endoderm-day2-Unique",
                "Blake2018-DE.iPSC-endoderm-day3-Unique",
                "Blake2020-DE.heart-Unique",
                "Blake2020-DE.kidney-Unique",
                "Blake2020-DE.liver-Unique",
                "Blake2020-DE.lung-Unique")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)

tempSource <- c("Gokhman2021-DE.Parental.CNCCs-Unique",
                "Gokhman2021-DE.Hybrid.CNCCs-Unique",
                "Gokhman2020-DMR.bone",
                "Hsu2019-GWAS.hip.geometry",
                "Styrkarsdottir2019-GWAS.bone.area",
                "Yengo2018-GWAS.height",
                "Steinberg2020-GWAS.OA",
                "Morris2020-GWAS.OP")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)

```

Print p-values for significant enrichments:

```{r}

tbl <- DEenrich[which(DEenrich$p.value<0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]

```

Save enrichment results:

```{r}

#Save enrichment results
write.csv(DEenrich,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.enrichment.csv")

```

### General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)

Pairwise DE Genes

```{r}

# DE genes and cell subset overlap
DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="adhoc") %>%
    dplyr::select(cell.subset,gene,logFC)
comboGenes <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC")       %>% dplyr::select(gene), "iPSC"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="MSC")        %>% dplyr::select(gene), "MSC"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene), "Osteogenic"=1)
                          ), full_join)
comboGenes[is.na(comboGenes)] <- 0

#Pairwise DE genes Shared across All Ad Hoc
share.stg <- DEgenes %>% 
               filter(gene %in%
                        pull(comboGenes %>%
                               filter(iPSC==1,
                                      MSC==1,
                                      Osteogenic==1) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.stg$gene))

#Pairwise DE genes Shared across iPSC and MSC
share.stg.t01 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(iPSC==1,
                                          MSC==1,
                                          Osteogenic==0) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t01$gene))

#Pairwise DE genes Shared across MSC and Osteogenic
share.stg.t12 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(iPSC==0,
                                          MSC==1,
                                          Osteogenic==1) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.stg.t12$gene))

#Pairwise DE genes Unique to iPSC
uniq.stg.t0 <- DEgenes %>% 
                 filter(cell.subset=="iPSC",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC==1,
                                        MSC==0,
                                        Osteogenic==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t0$gene)

#Pairwise DE genes Unique to MSC
uniq.stg.t1 <- DEgenes %>% 
                 filter(cell.subset=="MSC",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC==0,
                                        MSC==1,
                                        Osteogenic==0) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t1$gene)

#Pairwise DE genes Unique to Osteogenic
uniq.stg.t2 <- DEgenes %>% 
                 filter(cell.subset=="Osteogenic",
                        gene %in%
                          pull(comboGenes %>%
                                 filter(iPSC==0,
                                        MSC==0,
                                        Osteogenic==1) %>%
                                 dplyr::select(gene))) %>% 
                 arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.stg.t2$gene)

#Isolate genes tested in all cell classifications
DEtests <-
  DEdata %>%
    dplyr::filter(cell.assign=="adhoc") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests <- reduce(list(data.frame(gene=DEtests %>% filter(cell.subset=="iPSC") %>% dplyr::select(gene), "iPSC"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="MSC") %>% dplyr::select(gene), "MSC"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene), "Osteogenic"=1)
                          ), full_join)
comboTests[is.na(comboTests)] <- 0
tests <- pull(comboTests %>%
                filter(iPSC==1,
                       MSC==1,
                       Osteogenic==1) %>%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.stg <- share.stg[which(share.stg$gene %in% tests),]
shareX.stg.t01 <- share.stg.t01[which(share.stg.t01$gene %in% tests),]
shareX.stg.t12 <- share.stg.t12[which(share.stg.t12$gene %in% tests),]
uniqX.stg.t0 <- uniq.stg.t0[which(uniq.stg.t0$gene %in% tests),]
uniqX.stg.t1 <- uniq.stg.t1[which(uniq.stg.t1$gene %in% tests),]
uniqX.stg.t2 <- uniq.stg.t2[which(uniq.stg.t2$gene %in% tests),]
length(unique(shareX.stg$gene))
length(unique(shareX.stg.t01$gene))
length(unique(shareX.stg.t12$gene))
length(uniqX.stg.t0$gene)
length(uniqX.stg.t1$gene)
length(uniqX.stg.t2$gene)

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Ad Hoc (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")
#goEnPlot.spec(goTot,title="Shared in All",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in All",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in All",ontology="MF")

#Shared in iPSC and MSC (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t01$gene), ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in iPSC & MSC")
#goEnPlot.spec(goTot,title="Shared in iPSC & MSC",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in iPSC & MSC",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in iPSC & MSC",ontology="MF")

#Shared in MSC and Osteogenic (FDR<0.01, tested in all classifications)
#goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.stg.t12$gene), ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in MSC & Osteo")
#goEnPlot.spec(goTot,title="Shared in MSC & Osteo",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in MSC & Osteo",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in MSC & Osteo",ontology="MF")

#Unique to iPSC (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t0$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to iPSC")
#goEnPlot.spec(goTot,title="Unique to iPSC",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to iPSC",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to iPSC",ontology="MF")

#Unique to MSC (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t1$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to MSC")
#goEnPlot.spec(goTot,title="Unique to MSC",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to MSC",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to MSC",ontology="MF")

#Unique to Osteogenic (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.stg.t2$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Osteogenic")
#goEnPlot.spec(goTot,title="Unique to Osteogenic",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteogenic",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteogenic",ontology="MF")

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared in All Stages (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 0 & Time 1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t01$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 0 & Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t01$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared in Time 1 & Time 2 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.stg.t12$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 1 & Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.stg.t12$gene),
                                       NonDEGenes=nonDEgenes))  
  
  #Unique to Time 0 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t0$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 0",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t0$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t1$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Time 2 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% uniqX.stg.t2$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.stg.t2$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)

```

```{r}

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)

```

```{r}

tempSource <- c("Blake2018-DE.iPSCs-Unique",
                "Gokhman2021-DE.Parental.iPSCs-Unique",
                "Gokhman2021-DE.Hybrid.iPSCs-Unique",
                "Blake2018-DE.iPSC-endoderm-day1-Unique",
                "Blake2018-DE.iPSC-endoderm-day2-Unique",
                "Blake2018-DE.iPSC-endoderm-day3-Unique",
                "Blake2020-DE.heart-Unique",
                "Blake2020-DE.kidney-Unique",
                "Blake2020-DE.liver-Unique",
                "Blake2020-DE.lung-Unique")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)

tempSource <- c("Gokhman2021-DE.Parental.CNCCs-Unique",
                "Gokhman2021-DE.Hybrid.CNCCs-Unique",
                "Gokhman2020-DMR.bone",
                "Hsu2019-GWAS.hip.geometry",
                "Styrkarsdottir2019-GWAS.bone.area",
                "Yengo2018-GWAS.height",
                "Steinberg2020-GWAS.OA",
                "Morris2020-GWAS.OP")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)

```

Print p-values for significant enrichments:

```{r}

tbl <- DEenrich[which(DEenrich$p.value<0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]

```

Save enrichment results:

```{r}

#Save enrichment results
write.csv(DEenrich,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhoc.enrichment.csv")

```

### Osteogenic Clusters (res=0.50)

Pairwise DE Genes

```{r}

# DE genes and cell subset overlap
DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostcluster0.50") %>%
    dplyr::select(cell.subset,gene,logFC)
comboGenes <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1)
                          ), full_join)
comboGenes[is.na(comboGenes)] <- 0

#Pairwise DE genes Shared across O.c1, O.c3, O.c4
share.o1o3o4 <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(Osteogenic.c1==1,
                                          Osteogenic.c2==0,
                                          Osteogenic.c3==1,
                                          Osteogenic.c4==1) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.o1o3o4$gene))

# DE Genes Unique to Osteogenic.c1
Uniq.50o1 <- DEgenes %>% 
               filter(cell.subset=="Osteogenic.c1",
                      gene %in%
                        pull(comboGenes %>%
                               filter(Osteogenic.c1==1,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==0) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o1$gene))

# DE Genes Unique to Osteogenic.c2
Uniq.50o2 <- DEgenes %>% 
               filter(cell.subset=="Osteogenic.c2",
                      gene %in%
                        pull(comboGenes %>%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==1,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==0) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o2$gene))

# DE Genes Unique to Osteogenic.c3
Uniq.50o3 <- DEgenes %>% 
               filter(cell.subset=="Osteogenic.c3",
                      gene %in%
                        pull(comboGenes %>%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==1,
                                      Osteogenic.c4==0) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o3$gene))

# DE Genes Unique to Osteogenic.c4
Uniq.50o4 <- DEgenes %>% 
               filter(cell.subset=="Osteogenic.c4",
                      gene %in%
                        pull(comboGenes %>%
                               filter(Osteogenic.c1==0,
                                      Osteogenic.c2==0,
                                      Osteogenic.c3==0,
                                      Osteogenic.c4==1) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(Uniq.50o4$gene))

#Isolate genes tested in all cell classifications
DEtests <-
  DEdata %>%
    dplyr::filter(cell.assign=="ostcluster0.50") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests <- reduce(list(data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="Osteogenic.c4") %>% dplyr::select(gene), "Osteogenic.c4"=1)
                          ), full_join)
comboTests[is.na(comboTests)] <- 0
tests <- pull(comboTests %>%
                filter(Osteogenic.c1==1,
                       Osteogenic.c2==1,
                       Osteogenic.c3==1,
                       Osteogenic.c4==1) %>%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.o1o3o4 <- share.o1o3o4[which(share.o1o3o4$gene %in% tests),]
UniqX.50o1 <- Uniq.50o1[which(Uniq.50o1$gene %in% tests),]
UniqX.50o2 <- Uniq.50o2[which(Uniq.50o2$gene %in% tests),]
UniqX.50o3 <- Uniq.50o3[which(Uniq.50o3$gene %in% tests),]
UniqX.50o4 <- Uniq.50o4[which(Uniq.50o4$gene %in% tests),]
length(unique(shareX.o1o3o4$gene))
length(UniqX.50o1$gene)
length(UniqX.50o2$gene)
length(UniqX.50o3$gene)
length(UniqX.50o4$gene)

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Osteogenic Clusters (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.o1o3o4)$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in Osteo.c1c3c4")
#goEnPlot.spec(goTot,title="Shared in Osteo.c1c3c4",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in Osteo.c1c3c4",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in Osteo.c1c3c4",ontology="MF")

#Unique to Osteogenic.c1 (FDR<0.01, tested in all classifications) - no enrichments
#goTot <- goEnrich(totGenes=tests, deGenes=UniqX.50o1$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to Osteogenic.c1")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c1",ontology="MF")

#Unique to Osteogenic.c3 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=UniqX.50o3$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Osteogenic.c3")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c3",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c3",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c3",ontology="MF")

#Unique to Osteogenic.c4 (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=UniqX.50o4$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Osteogenic.c4")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c4",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c4",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteogenic.c4",ontology="MF")

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared in All Clusters (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% shareX.o1o3o4$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Osteo.c1c3c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.o1o3o4$gene),
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic.c1 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% UniqX.50o1$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteogenic.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o1$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c3 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% UniqX.50o3$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteogenic.c3",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o3$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to Osteogenic.c4 (tested in all classifications)
  nonDEgenes <- tests[which(tests %in% UniqX.50o4$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteogenic.c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=UniqX.50o4$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)

```

```{r}

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)

```

```{r}

tempSource <- c("Blake2018-DE.iPSCs-Unique",
                "Gokhman2021-DE.Parental.iPSCs-Unique",
                "Gokhman2021-DE.Hybrid.iPSCs-Unique",
                "Blake2018-DE.iPSC-endoderm-day1-Unique",
                "Blake2018-DE.iPSC-endoderm-day2-Unique",
                "Blake2018-DE.iPSC-endoderm-day3-Unique",
                "Blake2020-DE.heart-Unique",
                "Blake2020-DE.kidney-Unique",
                "Blake2020-DE.liver-Unique",
                "Blake2020-DE.lung-Unique")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)

tempSource <- c("Gokhman2021-DE.Parental.CNCCs-Unique",
                "Gokhman2021-DE.Hybrid.CNCCs-Unique",
                "Gokhman2020-DMR.bone",
                "Hsu2019-GWAS.hip.geometry",
                "Styrkarsdottir2019-GWAS.bone.area",
                "Yengo2018-GWAS.height",
                "Steinberg2020-GWAS.OA",
                "Morris2020-GWAS.OP")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)

```

Print p-values for significant enrichments:

```{r}

tbl <- DEenrich[which(DEenrich$p.value<0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]

```

Save enrichment results:

```{r}

#Save enrichment results
write.csv(DEenrich,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostcluster0.50.enrichment.csv")

```

### Osteogenic Clusters (res=0.75)
### Osteogenic Clusters (res=0.50) (Time 2 only)
### Osteogenic Clusters (res=0.75) (Time 2 only)

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)

Pairwise DE Genes

* Shared across all ostadhoc classifications: (FDR<0.01)                =  25 / (FDR<0.05)                = 193 
* Shared across pre+obl+emb+min:              (FDR<0.01)                =  22 / (FDR<0.05)                =  75
* Shared across obl+emb+min:                  (FDR<0.01)                =  17 / (FDR<0.05)                =  70
* Unique to preosteoblasts:                   (FDR<0.01)                =  33 / (FDR<0.05)                =  80
* Unique to osteoblasts:                      (FDR<0.01)                =  80 / (FDR<0.05)                = 250
* Unique to embedding osteoblasts:            (FDR<0.01)                =  33 / (FDR<0.05)                =  99
* Unique to mineralizing osteoblasts:         (FDR<0.01)                = 233 / (FDR<0.05)                = 560
* Unique to maturing osteocytes:              (FDR<0.01)                =  36 / (FDR<0.05)                = 148
* Shared across all ostadhoc classifications: (FDR<0.01, tested in all) =  25 / (FDR<0.05, tested in all) = 193
* Shared across pre+obl+emb+min:              (FDR<0.01, tested in all) =  22 / (FDR<0.05, tested in all) =  75
* Shared across obl+emb+min:                  (FDR<0.01, tested in all) =  17 / (FDR<0.05, tested in all) =  68
* Unique to preosteoblasts:                   (FDR<0.01, tested in all) =  33 / (FDR<0.05, tested in all) =  77
* Unique to osteoblasts:                      (FDR<0.01, tested in all) =  73 / (FDR<0.05, tested in all) = 226
* Unique to embedding osteoblasts:            (FDR<0.01, tested in all) =  26 / (FDR<0.05, tested in all) =  86
* Unique to mineralizing osteoblasts:         (FDR<0.01, tested in all) = 229 / (FDR<0.05, tested in all) = 533
* Unique to maturing osteocytes:              (FDR<0.01, tested in all) =  36 / (FDR<0.05, tested in all) = 146

Note: Too few DE genes shared across obl+emb+min+mat (n=3) and emb+min+mat (n=4), so these were not examined further. Instead, DE genes shared across pre+obl+emb+min (n=22) and obl+emb+min (n=17) were considered.

```{r}

#Pairwise DE genes and cell subset overlap
DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="ostadhoc") %>%
    dplyr::select(cell.subset,gene,logFC)
comboGenes <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast") %>% dplyr::select(gene), "preosteoblast"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast") %>% dplyr::select(gene), "osteoblast"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="embedding osteoblast") %>% dplyr::select(gene), "embedding osteoblast"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing osteoblast") %>% dplyr::select(gene), "mineralizing osteoblast"=1),
                          data.frame(gene=DEgenes %>% filter(cell.subset=="maturing osteocyte") %>% dplyr::select(gene), "maturing osteocyte"=1)
                          ), full_join)
comboGenes[is.na(comboGenes)] <- 0

#Pairwise DE genes Shared across All OstAdHoc Classifications
share.oad <- DEgenes %>% 
               filter(gene %in%
                        pull(comboGenes %>%
                               filter(preosteoblast==1,
                                      osteoblast==1,
                                      embedding.osteoblast==1,
                                      mineralizing.osteoblast==1,
                                      maturing.osteocyte==1) %>%
                               dplyr::select(gene))) %>% 
               arrange(gene)
               #arrange(desc(abs(logFC)))
length(unique(share.oad$gene))

#Pairwise DE genes Shared across obl+emb+min+mat
share.oad.poem <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(preosteoblast==1,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.oad.poem$gene))

#Pairwise DE genes Shared across emb+min+mat
share.oad.oem <- DEgenes %>% 
                   filter(gene %in%
                            pull(comboGenes %>%
                                   filter(preosteoblast==0,
                                          osteoblast==1,
                                          embedding.osteoblast==1,
                                          mineralizing.osteoblast==1,
                                          maturing.osteocyte==0) %>%
                                   dplyr::select(gene))) %>% 
                   arrange(gene)
                   #arrange(desc(abs(logFC)))
length(unique(share.oad.oem$gene))

#Pairwise DE genes Unique to Preosteoblasts
uniq.oad.pre <- DEgenes %>% 
                  filter(gene %in%
                           pull(comboGenes %>%
                                  filter(preosteoblast==1,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %>%
                                  dplyr::select(gene))) %>% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.pre$gene)

#Pairwise DE genes Unique to Osteoblasts
uniq.oad.obl <- DEgenes %>% 
                  filter(gene %in%
                           pull(comboGenes %>%
                                  filter(preosteoblast==0,
                                         osteoblast==1,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %>%
                                  dplyr::select(gene))) %>% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.obl$gene)

#Pairwise DE genes Unique to Embedding Osteoblasts
uniq.oad.emb <- DEgenes %>% 
                  filter(gene %in%
                           pull(comboGenes %>%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==1,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==0) %>%
                                  dplyr::select(gene))) %>% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.emb$gene)

#Pairwise DE genes Unique to Mineralizing Osteoblasts
uniq.oad.min <- DEgenes %>% 
                  filter(gene %in%
                           pull(comboGenes %>%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==1,
                                         maturing.osteocyte==0) %>%
                                  dplyr::select(gene))) %>% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.min$gene)

#Pairwise DE genes Unique to Maturing Osteocytes
uniq.oad.mat <- DEgenes %>% 
                  filter(gene %in%
                           pull(comboGenes %>%
                                  filter(preosteoblast==0,
                                         osteoblast==0,
                                         embedding.osteoblast==0,
                                         mineralizing.osteoblast==0,
                                         maturing.osteocyte==1) %>%
                                  dplyr::select(gene))) %>% 
                  arrange(gene)
                 #arrange(desc(abs(logFC)))
length(uniq.oad.mat$gene)

#Isolate genes tested in all cell classifications
DEtests <-
  DEdata %>%
    dplyr::filter(cell.assign=="ostadhoc") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
comboTests <- reduce(list(data.frame(gene=DEtests %>% filter(cell.subset=="preosteoblast") %>% dplyr::select(gene), "preosteoblast"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="osteoblast") %>% dplyr::select(gene), "osteoblast"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="embedding osteoblast") %>% dplyr::select(gene), "embedding osteoblast"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="mineralizing osteoblast") %>% dplyr::select(gene), "mineralizing osteoblast"=1),
                          data.frame(gene=DEtests %>% filter(cell.subset=="maturing osteocyte") %>% dplyr::select(gene), "maturing osteocyte"=1)
                          ), full_join)
comboTests[is.na(comboTests)] <- 0
tests <- pull(comboTests %>%
                filter(preosteoblast==1,
                       osteoblast==1,
                       embedding.osteoblast==1,
                       mineralizing.osteoblast==1,
                       maturing.osteocyte==1) %>%
                dplyr::select(gene))

#Deterime which unique DE genes were tested in all cell classifications
shareX.oad <- share.oad[which(share.oad$gene %in% tests),]
shareX.oad.poem <- share.oad.poem[which(share.oad.poem$gene %in% tests),]
shareX.oad.oem <- share.oad.oem[which(share.oad.oem$gene %in% tests),]
uniqX.oad.pre <- uniq.oad.pre[which(uniq.oad.pre$gene %in% tests),]
uniqX.oad.obl <- uniq.oad.obl[which(uniq.oad.obl$gene %in% tests),]
uniqX.oad.emb <- uniq.oad.emb[which(uniq.oad.emb$gene %in% tests),]
uniqX.oad.min <- uniq.oad.min[which(uniq.oad.min$gene %in% tests),]
uniqX.oad.mat <- uniq.oad.mat[which(uniq.oad.mat$gene %in% tests),]
length(unique(shareX.oad$gene))
length(unique(shareX.oad.poem$gene))
length(unique(shareX.oad.oem$gene))
length(uniqX.oad.pre$gene)
length(uniqX.oad.obl$gene)
length(uniqX.oad.emb$gene)
length(uniqX.oad.min$gene)
length(uniqX.oad.mat$gene)

```

```{r, eval=FALSE}

#Save DE gene lists
write.csv(tests,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-tests.csv")
write.csv(shareX.oad,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.csv")
write.csv(shareX.oad.poem,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.poem.csv")
write.csv(shareX.oad.oem,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-shareX.oad.oem.csv")
write.csv(uniqX.oad.pre,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.pre.csv")
write.csv(uniqX.oad.obl,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.obl.csv")
write.csv(uniqX.oad.emb,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.emb.csv")
write.csv(uniqX.oad.min,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.min.csv")
write.csv(uniqX.oad.mat,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.DE-fdr0.01-uniqX.oad.mat.csv")

```

Get psuedobulk data for viewing gene expression differences.

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstAdHoc.Assign.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
#log2 transformed cpm
tmm <- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
##boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

Look at gene expression levels of interesting DE genes.

```{r, eval=FALSE}

#Reformat pseudobulk data for plotting
tmm$gene <- rownames(tmm)
tmm.melt <- melt(tmm,id.vars="gene",measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][1])
tmm.melt$stage <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][2])
tmm.melt$species <- ifelse(grepl("H",tmm.melt$ind.rep),"Human","Chimp")

#Plot gene expression levels
pdf(file="./output/de-shareX.oad-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-shareX.oad.poem-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad.poem$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-shareX.oad.oem-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in unique(shareX.oad.oem$gene)) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.oad.pre-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.pre$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.oad.obl-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.obl$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.oad.emb-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.emb$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.oad.min-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.min$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-uniqX.oad.mat-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in uniqX.oad.mat$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across All OstAdHoc Classifications (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.oad)$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")
#goEnPlot.spec(goTot,title="Shared in All",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in All",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in All",ontology="MF")

#Shared in Pre, Obl, Emb, and Min (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.oad.poem)$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in Pre/Obl/Emb/Min")
#goEnPlot.spec(goTot,title="Shared in Pre/Obl/Emb/Min",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in Pre/Obl/Emb/Min",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in Pre/Obl/Emb/Min",ontology="MF")

#Shared in Obl, Emb, and Min (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=unique(shareX.oad.oem)$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in Obl/Emb/Min")
#goEnPlot.spec(goTot,title="Shared in Obl/Emb/Min",ontology="BP")
#goEnPlot.spec(goTot,title="Shared in Obl/Emb/Min",ontology="CC")
#goEnPlot.spec(goTot,title="Shared in Obl/Emb/Min",ontology="MF")

#Unique to Preosteoblasts (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.oad.pre$gene, ontology="ALL")
goEnPlot.spec(goTot,title="Unique to Preosteoblasts",ontology="BP")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Preosteoblasts")
#goEnPlot.spec(goTot,title="Unique to Preosteoblasts",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Preosteoblasts",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Preosteoblasts",ontology="MF")

#Unique to Osteoblasts (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.oad.obl$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Osteoblasts")
#goEnPlot.spec(goTot,title="Unique to Osteoblasts",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Osteoblasts",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Osteoblasts",ontology="MF")

#Unique to Embedding Osteoblasts (FDR<0.01, tested in all classifications) - no enrichment
#goTot <- goEnrich(totGenes=tests, deGenes=uniqX.oad.emb$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to Embedding Osteoblasts")
#goEnPlot.spec(goTot,title="Unique to Embedding Osteoblasts",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Embedding Osteoblasts",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Embedding Osteoblasts",ontology="MF")

#Unique to Mineralizing Osteoblasts (FDR<0.01, tested in all classifications)
goTot <- goEnrich(totGenes=tests, deGenes=uniqX.oad.min$gene, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Mineralizing Osteoblasts")
#goEnPlot.spec(goTot,title="Unique to Mineralizing Osteoblasts",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Mineralizing Osteoblasts",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Mineralizing Osteoblasts",ontology="MF")

#Unique to Maturing Osteocytes (FDR<0.01, tested in all classifications) - no enrichment
#goTot <- goEnrich(totGenes=tests, deGenes=uniqX.oad.mat$gene, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Unique to Maturing Osteocytes")
#goEnPlot.spec(goTot,title="Unique to Maturing Osteocytes",ontology="BP")
#goEnPlot.spec(goTot,title="Unique to Maturing Osteocytes",ontology="CC")
#goEnPlot.spec(goTot,title="Unique to Maturing Osteocytes",ontology="MF")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

```{r, eval=FALSE}

#Enrichment of external gene sets in cormotif DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Unique to preosteoblasts
  print("Unique to preosteoblasts")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.assign=="ostadhoc" &
                                  DEdata$cell.subset=="preosteoblast" &
                                  DEdata$gene %in% uniq.oad.pre$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.pre$gene,
                NonDEGenes=nonDEgenes)
  #Unique to osteoblasts
  print("Unique to osteoblasts")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.assign=="ostadhoc" &
                                  DEdata$cell.subset=="osteoblast" &
                                  DEdata$gene %in% uniq.oad.obl$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.obl$gene,
                NonDEGenes=nonDEgenes)
  #Unique to embedding osteoblasts
  print("Unique to embedding osteoblasts")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.assign=="ostadhoc" &
                                  DEdata$cell.subset=="embedding osteoblast" &
                                  DEdata$gene %in% uniq.oad.emb$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.emb$gene,
                NonDEGenes=nonDEgenes)
  #Unique to mineralizing osteoblasts
  print("Unique to mineralizing osteoblasts")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.assign=="ostadhoc" &
                                  DEdata$cell.subset=="mineralizing osteoblast" &
                                  DEdata$gene %in% uniq.oad.min$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.min$gene,
                NonDEGenes=nonDEgenes)
  #Unique to maturing osteocytes
  print("Unique to maturing osteocytes")
  nonDEgenes <- DEdata$gene[which(DEdata$cell.assign=="ostadhoc" &
                                  DEdata$cell.subset=="maturing osteocyte" &
                                  DEdata$gene %in% uniq.oad.mat$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniq.oad.mat$gene,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications, tested in all classifications
  print("Shared across all ostadhoc classifications, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.oad$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad$gene,
                NonDEGenes=nonDEgenes)
  #Shared across pre+obl+emb+min, tested in all classifications
  print("Shared across pre+obl+emb+min, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad.poem$gene,
                NonDEGenes=nonDEgenes)
  #Shared across obl+emb+min, tested in all classifications
  print("Shared across obl+emb+min, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=shareX.oad.oem$gene,
                NonDEGenes=nonDEgenes)
  
  #Unique to preosteoblasts, tested in all classifications
  print("Unique to preosteoblasts, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.pre$gene,
                NonDEGenes=nonDEgenes)
  #Unique to osteoblasts, tested in all classifications
  print("Unique to osteoblasts, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.obl$gene,
                NonDEGenes=nonDEgenes)
  #Unique to embedding osteoblasts, tested in all classifications
  print("Unique to embedding osteoblasts, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.emb$gene,
                NonDEGenes=nonDEgenes)
  #Unique to mineralizing osteoblasts, tested in all classifications
  print("Unique to mineralizing osteoblasts, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.min$gene,
                NonDEGenes=nonDEgenes)
  #Unique to maturing osteocytes, tested in all classifications
  print("Unique to maturing osteocytes, tested in all classifications")
  nonDEgenes <- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=uniqX.oad.mat$gene,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications, tested in all classifications
  nonDEgenes <- tests[which(tests %in% shareX.oad$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared across pre+obl+emb+min, tested in all classifications
  nonDEgenes <- tests[which(tests %in% shareX.oad.poem$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Pre/Obl/Emb/Min",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad.poem$gene),
                                       NonDEGenes=nonDEgenes))
  #Shared across obl+emb+min, tested in all classifications
  nonDEgenes <- tests[which(tests %in% shareX.oad.oem$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Obl/Emb/Min",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=unique(shareX.oad.oem$gene),
                                       NonDEGenes=nonDEgenes))
  
  #Unique to preosteoblasts, tested in all classifications
  nonDEgenes <- tests[which(tests %in% uniqX.oad.pre$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Preosteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.pre$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to osteoblasts, tested in all classifications
  nonDEgenes <- tests[which(tests %in% uniqX.oad.obl$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.obl$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to embedding osteoblasts, tested in all classifications
  nonDEgenes <- tests[which(tests %in% uniqX.oad.emb$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Embedding Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.emb$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to mineralizing osteoblasts, tested in all classifications
  nonDEgenes <- tests[which(tests %in% uniqX.oad.min$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Mineralizing Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.min$gene,
                                       NonDEGenes=nonDEgenes))
  #Unique to maturing osteocytes, tested in all classifications
  nonDEgenes <- tests[which(tests %in% uniqX.oad.mat$gene == FALSE)]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Maturing Osteocytes",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=uniqX.oad.mat$gene,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / (DEenrich$DE.Interest + DEenrich$DE.NotInterest)

```

```{r}

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.6) +
          scale_y_discrete(limits=rev)

```

```{r}

tempSource <- c("Blake2018-DE.iPSCs-Unique",
                "Gokhman2021-DE.Parental.iPSCs-Unique",
                "Gokhman2021-DE.Hybrid.iPSCs-Unique",
                "Blake2018-DE.iPSC-endoderm-day1-Unique",
                "Blake2018-DE.iPSC-endoderm-day2-Unique",
                "Blake2018-DE.iPSC-endoderm-day3-Unique",
                "Blake2020-DE.heart-Unique",
                "Blake2020-DE.kidney-Unique",
                "Blake2020-DE.liver-Unique",
                "Blake2020-DE.lung-Unique")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.6)) +
          scale_y_discrete(limits=rev)

tempSource <- c("Gokhman2021-DE.Parental.CNCCs-Unique",
                "Gokhman2021-DE.Hybrid.CNCCs-Unique",
                "Gokhman2020-DMR.bone",
                "Hsu2019-GWAS.hip.geometry",
                "Styrkarsdottir2019-GWAS.bone.area",
                "Yengo2018-GWAS.height",
                "Steinberg2020-GWAS.OA",
                "Morris2020-GWAS.OP")
index <- which(DEenrich$Source.GenesOfInterest %in% tempSource)
ggplot(DEenrich[index,], aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          scale_x_continuous(guide=guide_axis(angle=30), limits=c(0,0.2)) +
          scale_y_discrete(limits=rev)

```

Print p-values for significant enrichments:

```{r}

tbl <- DEenrich[which(DEenrich$p.value<0.05),c(1:3,7,9)]
tbl[order(tbl$p.value),]

```

Save enrichment results:

```{r}

#Save enrichment results
write.csv(DEenrich,"./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhoc.enrichment.csv")

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print("Shared across all ostadhoc classifications, tested in all classifications")
  print(intersect(genesExternal[[2]],unique(shareX.oad$gene)))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print("Shared across pre+obl+emb+min, tested in all classifications")
  print(intersect(genesExternal[[2]],unique(shareX.oad.poem$gene)))
  
  #Shared across obl+emb+min, tested in all classifications
  print("Shared across obl+emb+min, tested in all classifications")
  print(intersect(genesExternal[[2]],unique(shareX.oad.oem$gene)))

  #Unique to preosteoblasts, tested in all classifications
  print("Unique to preosteoblasts, tested in all classifications")
  print(intersect(genesExternal[[2]],uniqX.oad.pre$gene))
  
  #Unique to osteoblasts, tested in all classifications
  print("Unique to osteoblasts, tested in all classifications")
  print(intersect(genesExternal[[2]],uniqX.oad.obl$gene))

  #Unique to embedding osteoblasts, tested in all classifications
  print("Unique to embedding osteoblasts, tested in all classifications")
  print(intersect(genesExternal[[2]],uniqX.oad.emb$gene))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print("Unique to mineralizing osteoblasts, tested in all classifications")
  print(intersect(genesExternal[[2]],uniqX.oad.min$gene))
  
  #Unique to maturing osteocytes, tested in all classifications
  print("Unique to maturing osteocytes, tested in all classifications")
  print(intersect(genesExternal[[2]],uniqX.oad.mat$gene))

}

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)
### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)
### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)
