---
title: "10x-data-process-test"
author: "Genevieve Housman"
date: "February 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
```

# 10X Data Cellranger Analysis - Data Processing Test

Using cellranger to assign cells in human/chimp mixtures is somewhat problematic. Here, I test how well cells are assigned using cell ranger when all cells are known to be human cells.

Human-Only Data
(samples collected for Anthony's pilot eQTL project + external data)

* /project2/gilad/anthonyhung/Projects/OAStrain_project/YriMultiplex/fastq/
  + YG-AH-2S-ANT-1 containing a mixture of 3 humans
* Species-specific maturation profiles of human, chimpanzee and bonobo neural cells
  + Human_NPC_10X (scRNA-seq): GSE124706 - GSM3543690 - SRX5212502 - SRR8403264
  + https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124706
  + ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR840/004/SRR8403264/SRR8403264.fastq.gz

Chimp-Only Data
(external data)

* Species-specific maturation profiles of human, chimpanzee and bonobo neural cells
  + Chimp_NPC_10X (scRNA-seq): GSE124706 - GSM3543691 - SRX5212503 - SRR8403265
  + https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124706
  + ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR840/005/SRR8403265/SRR8403265.fastq.gz

References

* https://www.imm.ox.ac.uk/files/ccb/downloading_fastq_geo
* https://hbctraining.github.io/Accessing_public_genomic_data/lessons/accessing_public_experimental_data.html

### Download external data

```{bash, eval=FALSE}
cd /project2/gilad/ghousman/external-data
module load sra_toolkit/2.8

#chimp data
mkdir SRR8403265
cd SRR8403265/
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR840/005/SRR8403265/SRR8403265.fastq.gz
fastq-dump --split-files SRR8403265
#rename and zip files to conform to cellranger specifications
mv SRR8403265_1.fastq SRR8403265_S1_L001_R1_001.fastq
mv SRR8403265_2.fastq SRR8403265_S1_L001_R2_001.fastq
mv SRR8403265_3.fastq SRR8403265_S1_L001_I1_001.fastq
gzip *.fastq
#check file names
zcat SRR8403265_S1_L001_R1_001.fastq.gz | head -n 10
zcat SRR8403265_S1_L001_R2_001.fastq.gz | head -n 10
zcat SRR8403265_S1_L001_I1_001.fastq.gz | head -n 10
rm SRR8403265.fastq.gz

#human data
mkdir SRR8403264
cd SRR8403264/
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR840/004/SRR8403264/SRR8403264.fastq.gz
fastq-dump --split-files SRR8403264
#rename and zip files to conform to cellranger specifications
mv SRR8403264_1.fastq SRR8403264_S1_L001_R1_001.fastq
mv SRR8403264_2.fastq SRR8403264_S1_L001_R2_001.fastq
mv SRR8403264_3.fastq SRR8403264_S1_L001_I1_001.fastq
gzip *.fastq
#check file names
zcat SRR8403264_S1_L001_R1_001.fastq.gz | head -n 10
zcat SRR8403264_S1_L001_R2_001.fastq.gz | head -n 10
zcat SRR8403264_S1_L001_I1_001.fastq.gz | head -n 10
rm SRR8403264.fastq.gz
```

### Process fastq files

Fastq files were processed into count matrices (genes x cells) and cell types assigned using cellranger 3.1.0.

The cellranger reference was made using the hg38 and panTro6 genomes and ortho-exon v2 annotation (completed by KB):

* combined_genome_index: /project2/gilad/kenneth/References/HumanChimpCellranger/HumanOrthoV2_and_ChimpOrthoV2

Notes on what cellranger does to compute count matrices:

* genome alignment
	 + aligner (STAR) peforms splicing-aware alignment of reads to genome
	 + transcript annotation (GTF) buckets reads into exonic/intronic/intergenic and whether reads align (confidently) to genome
	 + [read = exonic if >=50% of it intersects an exon]
	 + [read = intronic if it is non-exonic and intersects an intron]
	 + [read = intergenic in all other cases]
* MAPQ adjustment (MAPQ 255)
   + for reads that align to a single exonic locus and to 1+ non-exonic loci, the exonic locus is prioritized and read is considered to be confidently mapped to exonic locus
* transcriptome alignment
	 + aligns exonic reads to annotated transcripts and looks for compatibility
	 + read is mapped to transcriptome if: read that is compatible with exons of annotated transcript and aligned to same strand
	 + read is uniquely (confidently) mapped to transcriptome if: read is compatible with a single gene annotation
	 + [uniquely (confidently) mapped reads are only ones considered for UMI counting]
* UMI counting
	 + before counting UMIs, corrections for sequencing errors in UMI sequences are attempted
	 + reads confidently mapped to transcriptome are grouped according to shared barcodes, UMIs, and gene annotations
	 + filter 1: if two groups of reads have same barcode and gene, but UMIs differ by a single base (Hamming distance 1 apart), then one UMI was likely introduced by substitution error in sequencing, and UMI of less-supported read group is corrected to UMI with higher support
	 + reads confidently mapped to transcriptome are grouped according to shared barcodes, UMIs (possibly corrected), and gene annotations
	 + filter 2: if two or more groups of reads have same barcode and UMI, but different gene annotations, gene annotation with most supporting reads is kept for UMI counting and other read groups are discarded (ties for maximal read support: all read groups are discarded as gene cannot be confidently assigned)
	 + each observed barcode-UMI-gene combination is recorded as UMI count in unfiltered feature-barcode matrix (number of reads supporting each counted UMI also recorded in molecule info file)
* calling cell barcodes
	 + [improved cell-calling algorithm that better identifies populations of low RNA content cells, especially when low RNA content cells are mixed into population of high RNA content cells, based on EmptyDrops method by Lun et al. 2018]
	 + step 1:
		  - uses cutoff based on total UMI counts of each barcode to identify cells - identifies primary mode of high RNA content cells
		  - (takes as input expected number of recovered cells, N)
		  - (let m be 99th percentile of top N barcodes by total UMI counts)
		  - (all barcodes whose total UMI counts exceed m/10 are called as cells)
	 + step 2:
		  - uses RNA profile of each remaining barcode to determine if it is "empty" or cell containing partition - captures low RNA content cells whose total UMI counts may be similar to empty GEMs
		  - (creates background model of RNA profiles from selected barcodes - multinomial distribution over genes - using Simple Good-Turing smoothing to provide a non-zero model estimate for genes not observed in representative empty GEM set)
		  - (RNA profile of each barcode not called as cell in step 1 is compared to background model)
		  - (barcodes whose RNA profile strongly disagrees with background model are added to set of positive cell calls)
		  - (to remedy cases in which set of barcodes called as cells does not match desired set of barcodes based on visual inspection: re-run count/reanalyze with --force-cells option, select desired barcodes from raw feature-barcode matrix in downstream analysis, or select custom barcodes in reanalyze using --barcodes)
* estimating multiplet rates
	 + cell-associated barcodes are classified as species1 or species2 based on which genome has more total UMI counts for that barcode
	 + cell-associated barcodes with total UMI counts that exceed the 10th percentile of the distributions for both species1 and species2 are classified as multiplets
	 + computes inferred multiplet rate by estimating the total number of multiplets (including (species1, species1) and (species2, species2))
		  - estimates via ML the total number of multiplet GEMs from the observed multiplets and the inferred ratio of species1 to species2 cells
		  - if ratio is 1:1, the inferred multiplet rate is approximately twice the observed (species1, species2) multiplets

```{bash, eval=FALSE}
cd /project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2

sbatch ./../../../human-chimp-skeletal-scRNA/code/submit-cellranger-count-multispp.sh YG-AH-2S-ANT-1 /project2/gilad/kenneth/References/HumanChimpCellranger/cellranger3.1/HumanOrthoV2_and_ChimpOrthoV2 /project2/gilad/anthonyhung/Projects/OAStrain_project/YriMultiplex/fastq/

sbatch ./../../../human-chimp-skeletal-scRNA/code/submit-cellranger-count-multispp.sh SRR8403265 /project2/gilad/kenneth/References/HumanChimpCellranger/cellranger3.1/HumanOrthoV2_and_ChimpOrthoV2 /project2/gilad/ghousman/external-data/SRR8403265

sbatch ./../../../human-chimp-skeletal-scRNA/code/submit-cellranger-count-multispp.sh SRR8403264 /project2/gilad/kenneth/References/HumanChimpCellranger/cellranger3.1/HumanOrthoV2_and_ChimpOrthoV2 /project2/gilad/ghousman/external-data/SRR8403264
```

```{r}

library(ggplot2)
library(gridExtra)

```

### Examine human-only data (YG-AH-2S-ANT-1)

```{r}

#load data
dir_data <- paste0('./../scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2/YG-AH-2S-ANT-1/outs/analysis/')
temp <- read.csv(file=paste0(dir_data,'gem_classification.csv'))

#make 10x calls a factor
temp$call <- factor(temp$call, levels=c("ChimpOrthoV2","HumanOrthoV2","Multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$call)

#maximum number of UMIs/cell that aligned to each genome
max(temp$ChimpOrthoV2)
max(temp$HumanOrthoV2)
max_umi <- max(c(temp$ChimpOrthoV2,temp$HumanOrthoV2))

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=call)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="CellRanger Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=call, fill=call)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,3100) +
        theme(legend.position="none") +
        labs(x="CellRanger Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

#calculate total number of umis
temp$TotalOrthoV2 <- temp$HumanOrthoV2 + temp$ChimpOrthoV2
#calculate ratio of human umis to total umis
temp$RatioOrthoV2 <- temp$HumanOrthoV2 / temp$TotalOrthoV2

#mean ratio
mean(temp$RatioOrthoV2)

#human RatioOrthoV2 cutoff
cutoff <- min(temp$RatioOrthoV2)
round(cutoff,2)

ggplot(temp, aes(x=RatioOrthoV2)) +
       geom_histogram(binwidth=0.01, color="black", fill="white", alpha=0.2) +
       xlim(0,1) +
       geom_vline(xintercept=cutoff, color="red") +
       labs(x="Ratio of Human UMIs to Total UMIs", y="Frequency")

#assign new human/chimp cells
temp$NEWcall <- "multiplet"
temp$NEWcall[which(temp$RatioOrthoV2>=cutoff)] <- "human"
temp$NEWcall[which(temp$RatioOrthoV2<=0.1)] <- "chimp"
temp$NEWcall <- factor(temp$NEWcall, levels=c("chimp","human","multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$NEWcall)

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=NEWcall)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="New Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=NEWcall, fill=NEWcall)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,3100) +
        theme(legend.position="none") +
        labs(x="New Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

```

### Examine chimp-only data (SRR8403265)

```{r}

#load data
dir_data <- paste0('./../scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2/SRR8403265/outs/analysis/')
temp <- read.csv(file=paste0(dir_data,'gem_classification.csv'))

#make 10x calls a factor
temp$call <- factor(temp$call, levels=c("ChimpOrthoV2","HumanOrthoV2","Multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$call)

#maximum number of UMIs/cell that aligned to each genome
max(temp$ChimpOrthoV2)
max(temp$HumanOrthoV2)
max_umi <- max(c(temp$ChimpOrthoV2,temp$HumanOrthoV2))

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=call)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="CellRanger Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=call, fill=call)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,5900) +
        theme(legend.position="none") +
        labs(x="CellRanger Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

#calculate total number of umis
temp$TotalOrthoV2 <- temp$HumanOrthoV2 + temp$ChimpOrthoV2
#calculate ratio of human umis to total umis
temp$RatioOrthoV2 <- temp$HumanOrthoV2 / temp$TotalOrthoV2

#mean ratio
mean(temp$RatioOrthoV2)

#chimp RatioOrthoV2 cutoff
cutoff <- max(temp$RatioOrthoV2)
round(cutoff,2)

ggplot(temp, aes(x=RatioOrthoV2)) +
       geom_histogram(binwidth=0.01, color="black", fill="white", alpha=0.2) +
       xlim(0,1) +
       geom_vline(xintercept=cutoff, color="blue") +
       labs(x="Ratio of Human UMIs to Total UMIs", y="Frequency")

#assign new human/chimp cells
temp$NEWcall <- "multiplet"
temp$NEWcall[which(temp$RatioOrthoV2>=0.1)] <- "human"
temp$NEWcall[which(temp$RatioOrthoV2<=cutoff)] <- "chimp"
temp$NEWcall <- factor(temp$NEWcall, levels=c("chimp","human","multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$NEWcall)

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=NEWcall)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="New Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=NEWcall, fill=NEWcall)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,5900) +
        theme(legend.position="none") +
        labs(x="New Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

```

### Examine human-only data (SRR8403264)

```{r}

#load data
dir_data <- paste0('./../scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2/SRR8403264/outs/analysis/')
temp <- read.csv(file=paste0(dir_data,'gem_classification.csv'))

#make 10x calls a factor
temp$call <- factor(temp$call, levels=c("ChimpOrthoV2","HumanOrthoV2","Multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$call)

#maximum number of UMIs/cell that aligned to each genome
max(temp$ChimpOrthoV2)
max(temp$HumanOrthoV2)
max_umi <- max(c(temp$ChimpOrthoV2,temp$HumanOrthoV2))

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=call)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="CellRanger Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=call, fill=call)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,7300) +
        theme(legend.position="none") +
        labs(x="CellRanger Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

#calculate total number of umis
temp$TotalOrthoV2 <- temp$HumanOrthoV2 + temp$ChimpOrthoV2
#calculate ratio of human umis to total umis
temp$RatioOrthoV2 <- temp$HumanOrthoV2 / temp$TotalOrthoV2

#mean ratio
mean(temp$RatioOrthoV2)

#human RatioOrthoV2 cutoff
cutoff <- min(temp$RatioOrthoV2)
round(cutoff,2)

ggplot(temp, aes(x=RatioOrthoV2)) +
       geom_histogram(binwidth=0.01, color="black", fill="white", alpha=0.2) +
       xlim(0,1) +
       geom_vline(xintercept=cutoff, color="red") +
       labs(x="Ratio of Human UMIs to Total UMIs", y="Frequency")

#assign new human/chimp cells
temp$NEWcall <- "multiplet"
temp$NEWcall[which(temp$RatioOrthoV2>=cutoff)] <- "human"
temp$NEWcall[which(temp$RatioOrthoV2<=0.1)] <- "chimp"
temp$NEWcall <- factor(temp$NEWcall, levels=c("chimp","human","multiplet"), labels=c("chimp","human","multiplet"))

#number of species assignments
table(temp$NEWcall)

p1 <- ggplot(temp, aes(x=HumanOrthoV2, y=ChimpOrthoV2, color=NEWcall)) +
        geom_point() +
        scale_color_manual(drop=FALSE, values=c("blue","red","grey")) +
        lims(x=c(0,max_umi), y=c(0,max_umi)) +
        labs(x="UMI Aligned to Human Genome", y="UMI Aligned to Chimp Genome", color="New Calls") +
        theme(legend.position=c(0.8,0.8))

p2 <- ggplot(temp, aes(x=NEWcall, fill=NEWcall)) +
        geom_bar() +
        scale_x_discrete(drop=FALSE) +
        scale_fill_manual(drop=FALSE, values=c("blue","red","grey")) +
        ylim(0,7300) +
        theme(legend.position="none") +
        labs(x="New Calls", y="Frequency")

grid.arrange(p1,p2, ncol=2, widths=c(2,1))

```

### Conclusions

Cellranger is really bad at human/chimp assignments, so using a cutoff based on the ratio of human-aligned UMIs per cells vs. the total number of aligned UMIs per cells (bars near 0 are likely chimp cells and bars near 1 are likely human cells) seems like a better option.

Chimp reads in human droplets and human reads in chimp droplets may come from two things

* chance alignment of reads to either the chimp or human genome
* background coming from lysed cells

The first thing is empirically measured here. Based on the human-only and chimp-only datasets, on average ~3% of human reads align to the chimp genome, and ~4% of chimp reads align to the human genome. However, a ~6% cutoff was needed to correctly assign all human cells, and a ~9% cutoff was needed to correctly assign all chimp cells. On top of this, we expect that some mixed species reads will come from lysed cells. Some batches may have more lysed cell read backgrounds than others. Nevertheless, we propose that using a 10% cutoff for human and chimp cells allows some flexibility for background reads while remaining close to the cutoffs calculated here. 

