---
title: "10x-data-process"
author: "Genevieve Housman"
date: "2/2/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Data Processing

Compile 10X data using seurat.

Data being examined comes from 6 humans and 6 chimpanzees.
- GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
- GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
- GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
- HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
- HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
- HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
- HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)
- HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
- HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
- HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)
- HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
- HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
- HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)
- HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
- HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
- HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)
- HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
- HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
- HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)

Each dataset being evaluated comprises gene count data that were seperately calculated using both the hg38 human genome and the panTro6 chimpanzee genome, as well as the corresponding ortho-exon annotation in cellranger. Species-specific assignments were determined using human-chimpanzee genome combination with ortho-exon annotation.

```{bash}

#easties to run these scripts on the cluster
sinteractive --mem=24g --time=12:00:00 --partition=broadwl
module load R/3.5.1
R

```

```{r load libraries}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

```{r define data location}

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Define head directory containing full scRNA data
dir_scrna <- paste0(dir,'scRNA/cellranger-data-full')

```

```{r assess species assignments}

#Load species assignments
spp <- data.frame()
i <- 1
while (i <= length(batch$Sample_Name_at_Core)) {
  print(i)
  #Define directory containing data from one collection
  dir_data <- paste0(dir_scrna,'/cellranger-ortho-hg38-and-ortho-panTro6/',batch$Sample_Name_at_Core[i],'/outs/analysis/')
  #Load summary metrics
  temp <- read.csv(file=paste0(dir_data,'gem_classification.csv'))
  #Process metrics
  temp$barcode <- sapply(temp$barcode, gsub, pattern="-1", replacement="")
  temp$call <- sapply(temp$call, gsub, pattern="ortho.panTro6", replacement="Chimp")
  temp$call <- sapply(temp$call, gsub, pattern="ortho.hg38", replacement="Human")
  temp$Collection.Name <- batch$Sample_Name_at_Core[i]
  #Add data to dataframe
  spp <- rbind(spp,temp)
  i <- i+1
}

```

```{r process scRNA data - isolate human and chimp cells}

#Isolate human and chimp cells for each collection
i <- 1
while (i <= length(batch$Sample_Name_at_Core)) {

  print(paste0("Processing ",batch$Sample_Name_at_Core[i]))
  
  #(0)define directory
    dir_h.data <- paste0(dir_scrna,'/cellranger-ortho-hg38/',batch$Sample_Name_at_Core[i],'/outs')
    dir_c.data <- paste0(dir_scrna,'/cellranger-ortho-panTro6/',batch$Sample_Name_at_Core[i],'/outs')
    
  #(1)load data
    h.data <- Read10X(data=paste0(dir_h.data,"/raw_feature_bc_matrix"))
    c.data <- Read10X(data=paste0(dir_c.data,"/raw_feature_bc_matrix"))
    
  #(2)create seurat objects
    h.dat <- CreateSeuratObject(counts=h.data) #CHECK IF I NEED TO READD PROJECT=IPSC/MSC/OSTEO
    c.dat <- CreateSeuratObject(counts=c.data) #CHECK IF I NEED TO READD PROJECT=IPSC/MSC/OSTEO
    
  #(3)view details of seurat objects
    h.dat                       #19399 features, 6794880 samples
    c.dat                       #19399 features, 6794880 samples
    
  #(4)clear raw data from workspace
    rm(h.data,c.data)
  
  #(5)add species assignments to data
    dir_data <- paste0(dir_scrna,'/cellranger-ortho-hg38-and-ortho-panTro6/',batch$Sample_Name_at_Core[i],'/outs/analysis/')
    #Load summary metrics
    temp <- read.csv(file=paste0(dir_data,'gem_classification.csv'))
    #Process metrics
    temp$barcode <- sapply(temp$barcode, gsub, pattern="-1", replacement="")
    temp$call <- sapply(temp$call, gsub, pattern="ortho.panTro6", replacement="Chimp")
    temp$call <- sapply(temp$call, gsub, pattern="ortho.hg38", replacement="Human")
    temp$Collection.Name <- batch$Sample_Name_at_Core[i]
      
    m <- match(rownames(h.dat@meta.data),temp$barcode)
    if(any(is.na(m))) cat("Not all barcodes are in data.\n")
    h.dat <- AddMetaData(h.dat, temp[m,]$call, col.name="species")
  
    m <- match(rownames(c.dat@meta.data),temp$barcode)
    if(any(is.na(m))) cat("Not all barcodes are in data.\n")
    c.dat <- AddMetaData(c.dat, temp[m,]$call, col.name="species")

  #(6)view details of species assignments - sppo
    table(h.dat@meta.data$species, useNA="always")
    table(c.dat@meta.data$species, useNA="always")
  
  #(7)subset data to only include human and chimp cells/genes - sppo
    h.dat.spp <- subset(h.dat, subset=species=="Human")
    c.dat.spp <- subset(c.dat, subset=species=="Chimp")
 
  #(8)create merged object
    h.c.dat.spp <- merge(h.dat.spp,c.dat.spp)
    saveRDS(h.c.dat.spp, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/h.c.dat.spp.",batch$Sample_Name_at_Core[i],".rds"))

  #(9)remove unnecessary variables
    rm(h.dat,c.dat,h.dat.spp,c.dat.spp)
    rm(dir_h.data,dir_c.data)

  i <- i+1
  
}

```

```{r process scRNA data - add relevant metadata}

#Read in files
data <- list()
for (i in 1:length(batch$Sample_Name_at_Core)) {
  print(i)
  data[[i]] <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/h.c.dat.spp.",batch$Sample_Name_at_Core[i],".rds"))
}

#Add additional details to list
for (i in 1:length(data)) { data[[i]]$Collection <- i }
for (i in c(1,4,7,10,13,16,19)) { data[[i]]$Cell.Type <- "iPSC" }
for (i in c(2,5,8,11,14,17,20)) { data[[i]]$Cell.Type <- "MSC" }
for (i in c(3,6,9,12,15,18,21)) { data[[i]]$Cell.Type <- "Osteoblast" }
for (i in 1:6) { data[[i]]$Pair <- "H1+C1" }
for (i in 7:9) { data[[i]]$Pair <- "H2+C2" }
for (i in 10:12) { data[[i]]$Pair <- "H3+C3" }
for (i in 13:15) { data[[i]]$Pair <- "H4+C4" }
for (i in 16:18) { data[[i]]$Pair <- "H5+C5" }
for (i in 19:21) { data[[i]]$Pair <- "H6+C6" }
for (i in 1:length(data)) { data[[i]]$Individual <- NA }
for (i in 1:6) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H1" }
for (i in 1:6) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C1" }
for (i in 7:9) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H2" }
for (i in 7:9) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C2" }
for (i in 10:12) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H3" }
for (i in 10:12) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C3" }
for (i in 13:15) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H4" }
for (i in 13:15) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C4" }
for (i in 16:18) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H5" }
for (i in 16:18) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C5" }
for (i in 19:21) { data[[i]]$Individual[which(data[[i]]$species=="Human")] <- "H6" }
for (i in 19:21) { data[[i]]$Individual[which(data[[i]]$species=="Chimp")] <- "C6" }
for (i in c(1:3,7:21)) { data[[i]]$Replicate <- "1" }
for (i in 4:6) { data[[i]]$Replicate <- "2" }
for (i in c(1:3,7:12)) { data[[i]]$MSCmedia <- "3B" }
for (i in c(4:6,13:21)) { data[[i]]$MSCmedia <- "2B" }

#Identify mitochondrial genes (should be 13 genes total)
#which(sapply(data[[i]]@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="MT-")==1)
#MT-ND1   19387
#MT-ND2   19388
#MT-CO1   19389
#MT-CO2   19390
#MT-ATP8  19391
#MT-ATP6  19392
#MT-CO3   19393
#MT-ND3   19394
#MT-ND4L  19395
#MT-ND4   19396
#MT-ND5   19397
#MT-ND6   19398
#MT-CYB   19399
mito <- c("MT-ND1","MT-ND2","MT-CO1","MT-CO2","MT-ATP8","MT-ATP6","MT-CO3","MT-ND3","MT-ND4L","MT-ND4","MT-ND5","MT-ND6","MT-CYB")
for (i in 1:length(data)) { data[[i]]$percent.mt <- PercentageFeatureSet(data[[i]], features=mito) }

#Save data
for (i in 1:length(batch$Sample_Name_at_Core)) {
  print(i)
  saveRDS(data[[i]], file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/h.c.dat.spp.",batch$Sample_Name_at_Core[i],".rds"))
}

#Remove unnecessary variables
rm(data,mito)

```

```{r process scRNA data - filter out mitochondrial genes from data}

#Read in files
data <- list()
for (i in 1:length(batch$Sample_Name_at_Core)) {
  print(i)
  data[[i]] <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/h.c.dat.spp.",batch$Sample_Name_at_Core[i],".rds"))
}

#Remove mitochondrial genes
mito <- c("MT-ND1","MT-ND2","MT-CO1","MT-CO2","MT-ATP8","MT-ATP6","MT-CO3","MT-ND3","MT-ND4L","MT-ND4","MT-ND5","MT-ND6","MT-CYB")
objects <- list()
for (i in 1:length(data))
{
  print(i)
  genes <- which(!(data[[i]]@assays$RNA@counts@Dimnames[[1]] %in% mito))
  obj <- subset(data[[i]], features=genes)
  objects[[i]] <- CreateSeuratObject(obj@assays$RNA@counts, meta.data=obj@meta.data)
}
rm(data,obj)

#Save data
saveRDS(objects, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.rds"))

```
