---
title: "10x-view-species-assignments"
author: "Genevieve Housman"
date: "May 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# View 10X Data Cellranger Species Assignments

Data being examined come from 6 humans and 6 chimpanzees.

* GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
* GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
* GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)
* HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
* HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
* HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)
* HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
* HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
* HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)
* HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
* HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
* HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)
* HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
* HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
* HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)
* HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
* HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
* HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)
* HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
* HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
* HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)

Collections are classified by the stage of differentiation at which cells were collected:

* Time 0 = iPSCs
* Time 1 = iPSC-derived MSCs
* Time 2 = iPSC-MSC-derived osteoblasts

```{r load libraries, message=FALSE}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

```{r load species assignments, results="hide"}

#Load batch info
batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")

#Load species assignments
spp <- data.frame()
i <- 1
while (i <= length(batch$Sample_Name_at_Core)) {
  print(i)
  #Define directory containing data from one collection
  dir_data <- paste0('./../scRNA/cellranger-data-full/cellranger-data-HumanOrthoV2_and_ChimpOrthoV2/',batch$Sample_Name_at_Core[i],'/outs/analysis/')
  #Load summary metrics from one collection
  temp <- read.csv(file=paste0(dir_data,'gem_classification_GHedit.csv'))
  #Process metrics
  temp$barcode <- sapply(temp$barcode, gsub, pattern="-1", replacement="")
  temp$NEWcall <- sapply(temp$NEWcall, gsub, pattern="Human", replacement="Human")
  temp$NEWcall <- sapply(temp$NEWcall, gsub, pattern="Chimp", replacement="Chimp")
  temp$Collection.Name <- batch$Sample_Name_at_Core[i]
  #Add data to dataframe
  spp <- rbind(spp,temp)
  i <- i+1
}
rm(dir_data,temp)

#Fix species assignment details so that plotting is easier
spp$Collection.Name <- factor(spp$Collection.Name,
                                  levels=c("GHO-1","GHO-2","GHO-4",
                                           "HOU-10","HOU-12","HOU-16",
                                           "HOU-1","HOU-2","HOU-4",
                                           "HOU-9","HOU-11","HOU-15",
                                           "HOU-5","HOU-6","HOU-8",
                                           "HOU-17","HOU-19","HOU-21",
                                           "HOU-18","HOU-20","HOU-22"))
stage = c()
for (j in spp$Collection.Name) {
  if (j=="GHO-1"|j=="HOU-10"|j=="HOU-1"|j=="HOU-9"|j=="HOU-5"|j=="HOU-17"|j=="HOU-18") {
    stage = c(stage,"Time 0")
  }
  if (j=="GHO-2"|j=="HOU-12"|j=="HOU-2"|j=="HOU-11"|j=="HOU-6"|j=="HOU-19"|j=="HOU-20") {
    stage = c(stage,"Time 1")
  }
  if (j=="GHO-4"|j=="HOU-16"|j=="HOU-4"|j=="HOU-15"|j=="HOU-8"|j=="HOU-21"|j=="HOU-22") {
    stage = c(stage,"Time 2")
  }
}
spp$Stage <- stage
spp$Stage <- factor(spp$Stage, levels=c("Time 0","Time 1","Time 2"))

rm(stage,i,j)

```

```{r number of cells assigned to each species, results="hide"}

#Examine species assignment data
table(spp[spp$Collection.Name=="GHO-1",]$NEWcall)
table(spp[spp$Collection.Name=="GHO-2",]$NEWcall)
table(spp[spp$Collection.Name=="GHO-4",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-10",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-12",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-16",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-1",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-2",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-4",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-9",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-11",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-15",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-5",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-6",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-8",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-17",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-19",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-21",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-18",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-20",]$NEWcall)
table(spp[spp$Collection.Name=="HOU-22",]$NEWcall)

table(spp$Stage)
table(spp[spp$Stage=="Time 0",]$NEWcall)
table(spp[spp$Stage=="Time 1",]$NEWcall)
table(spp[spp$Stage=="Time 2",]$NEWcall)
table(spp$NEWcall)

```

## Visualize species assignment counts

* numbers of human and chimpanzee cells are approximately 50/50 across all collections
* cells collected at Time 2 (osteogenic) tend to have lower counts than cells collected at Time 0 (iPSCs) and Time 1 (MSCs)

```{r}

#total number of species assignments per collection
ggplot(spp, aes(x=Collection.Name, fill=NEWcall)) +
  geom_bar() +
  labs(title="Alignment to Human Ortho-Exon") +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)",y="Number of Cells",x="Collection") +
  scale_fill_grey(start=0.8, end=0.2) +
  theme_classic() +
  ylim(0,7500) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

```{r}

#proportions of species assignments per collection
ggplot(spp, aes(x=Collection.Name, fill=NEWcall)) +
  geom_bar(position="fill") +
  labs(title="Alignment to Human Ortho-Exon") +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)",y="Proportion of Cells",x="Collection") +
  scale_fill_grey(start=0.8, end=0.2) +
  theme_classic() +
  ylim(0,1) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

## Species assigments with multiplets removed

```{r}

#total number of species assignments per collection with multiplets removed
ggplot(subset(spp,NEWcall!="Multiplet"), aes(x=Collection.Name, fill=NEWcall)) +
  geom_bar() +
  labs(title="Alignment to Human Ortho-Exon") +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), legend.text=element_text(size=12)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)", y="Number of Cells", x="Collection") +
  scale_fill_grey(start=0.7, end=0.3) +
  ylim(0,7500) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

```{r}

#proportions of species assignments per collection with multiplets removed
ggplot(subset(spp,NEWcall!="Multiplet"), aes(x=Collection.Name, fill=NEWcall)) +
  geom_bar(position="fill") +
  labs(title="Alignment to Human Ortho-Exon") +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)",y="Proportion of Cells",x="Collection") +
  scale_fill_grey(start=0.7, end=0.3) +
  theme_classic() +
  ylim(0,1) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

## Species assignments of cells collected at different stages of differentiation

```{r}

#proportion of species assignments for different stages of differentiation with multiplets removed
ggplot(subset(spp,NEWcall!="Multiplet"), aes(x=Stage, fill=NEWcall)) +
  geom_bar(position="fill") +
  labs(title="Alignment to Human Ortho-Exon") +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12), legend.text=element_text(size=12)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)",y="Proportion of Cells",x="Cell Type") +
  scale_fill_grey(start=0.7, end=0.3, name="Species Assignment") +
  coord_flip()

```

```{r}

#total number of species assignments for different stages of differentiation with multiplets removed
ggplot(subset(spp,NEWcall!="Multiplet"), aes(x=Stage, fill=NEWcall)) +
  geom_bar(position="dodge") +
  labs(title="Alignment to Human Ortho-Exon") +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12), legend.text=element_text(size=12)) +
  labs(title="Species Assignments (Genomes: hg38 + panTro6 Annotations: ortho-exon)",y="Number of Cells",x="Cell Type") +
  scale_fill_grey(start=0.7, end=0.3, name="Species Assignment")


```

## Conclusions

* Most 10X collection has ~50% human cells and ~50% chimp cells
* Cells from each stage of differentiation (across individuals) are ~50% human and ~50% chimp
* 10X collections from Time 2 generally had fewer total barcodes (fewer singlets, more multiplets) than Time 0 or Time 1
* Some variation in the number of cells across collections
* Collection HOU-15 (H3+C3 Time 2) has many fewer cells than other collections
