---
title: "10x-plot-cormotif"
author: "Genevieve Housman"
date: "June 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Correlation Motifs

View results of correlation motifs tests between humans and chimpanzees within different cell classifications.

```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ComplexHeatmap)

```

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate <- paste0(data@meta.data$Individual,".",data@meta.data$Replicate)

```

```{r}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

```{r}

#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot <- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale="free") +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

```

## Load External Gene Datasets

Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.

Genes previously identified as human and chimpanzee differences:

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR < 1% out of 10304 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each cell type
    
* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR < 1% out of 12184 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each tissue type

* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + DE genes in parental iPSC lines (n = 7655)
    + DE genes in hybrid iPSC lines (n = 6009)
    + DE genes in parental CNCC lines (n = 2223)
    + DE genes in hybrid CNCC lines (n = 3612)
    + Note: only considered DE genes unique to each cell type
    
* Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
    + DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)
    + AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)
    + (human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)
    + (chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)
    + Note: only DMRs that overlapped promotor regions of genes were included
    + Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels

General skeleton-related and skeletal phenotype-related genes:

* Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
    + hip geometry GWAS loci (n = 10)

* Styrkarsdottir et al. 2019 Nature Communications (Table 1)
    + DXA bone area GWAS loci (n = 21 associated with 20 genes)

* Yengo et al. 2018 Human Molecular Genetics
    + height GWAS loci (n = 3290) (data provided at https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
    + height GWAS loci with associated eQTLs (n = 610) (list not provided)
    + Note: only used unique gene symbols associated with height GWAS loci (n = 1500)

Genes associated with skeletal diseases (OA + OP):

* Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
    + genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)
    
* Morris et al. 2019 Nature Genetics (Supplemental Table 2)
    + OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)
    
Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:

* Gallego Romero et al. 2015 eLife (Supplementary File 3)
    + DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR < 1% out of 12171 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + REMOVED because subsequent studies (Blake et al. 2018 & Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm & CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type
    
* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
    + Skeleton-related genes (n = 1790)
    + Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)
    + REMOVED because it is unclear where these genes came from to interpreting biology is difficult

Some external gene datasets were considered but not examined:

* Ward et al. 2019 eLife
    +  DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR < 10% out of 11974 genes)

* Shibata et al. 2012 PLOS Genetics
    + increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)

* Edsall et al. 2019 Genome Biology and Evolution
    + increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)

* Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
    + increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)

* Prescott et al. 2015 Cell (Supplemental)
    + top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)
    + top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)

* Joganic et al. 2017 American Journal of Physical Anthropology
    + genetics of craniofacial skeleton in baboons

* CibriÃ¡n Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
    + robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)

* Tachmazidou et al. 2019 Nature Genetics
    + OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])
    + OA loci (Supplementary Table 2)
    + established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)
    + Note: used loci compiled by AH
    + (OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)
    + (OA.estab = gene symbol(s) nearest to established OA loci)
    + (OA.broad = genes within 500kb of OA loci)

```{r}

#Define function for testing significant enrichment of gene sets

genesetEnrich <- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative="greater"))

}

```

```{r}

#Load external datasets (manually currated from papers)
gallegoromero2015    <- read.csv("./data/external-gene-sets/GallegoRomero2015eLife.csv")
blake2018            <- read.csv("./data/external-gene-sets/Blake2018GenBio.csv")
blake2020            <- read.csv("./data/external-gene-sets/Blake2020GenRes.csv")
gokhman2021          <- read.csv("./data/external-gene-sets/Gokhman2021NatGen.csv")
gokhman2020          <- read.csv("./data/external-gene-sets/Gokhman2020NatComm.csv")
gokhman2020.skeleton <- read.csv("./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv")
hsu2019              <- read.csv("./data/external-gene-sets/Hsu2019JBoneMinRes.csv")
styrkarsdottir2019   <- read.csv("./data/external-gene-sets/Styrkarsdottir2019NatComm.csv")
yengo2018            <- read.csv("./data/external-gene-sets/Yengo2018HumMolGen.csv")
steinberg2020        <- read.csv("./data/external-gene-sets/Steinberg2020biorxiv.csv")
morris2019           <- read.csv("./data/external-gene-sets/Morris2019NatGen.csv")
#cibrianuhalte2017    <- read.csv("./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv")
#tachmazidou2019      <- read.csv("./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv")

```

```{r, eval=FALSE}

#genes previously identified as human and chimpanzee differences

#genesExternal <- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal <- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal <- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal <- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1 &
                                      blake2018$DE.Day1==0 &
                            					blake2018$DE.Day2==0 &
                            					blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==1 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==1 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==1 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==1 &
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                               gokhman2021$DE.Hybrid.CNCCs==0)]), #hybrid.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal <- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal <- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal <- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal <- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal <- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal <- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS

```

## Stages of Differentiation at Collection
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Stage=c("Time 0","Time 1","Time 2"),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("Time.0",length(which(dataUpset$Time.0==1))),
                                       rep("Time.1",length(which(dataUpset$Time.1==1))),
                                       rep("Time.2",length(which(dataUpset$Time.2==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$Time.0==1)],
                                rownames(dataUpset)[which(dataUpset$Time.1==1)],
                                rownames(dataUpset)[which(dataUpset$Time.2==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$Time.0==1),"Time.0"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.1==1),"Time.1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.2==1),"Time.2"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("Time.0","Time.1","Time.2"),
                               labels=c("Time 0","Time 1","Time 2"))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(rowSums(dataUpset)>=1)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0"),
                         list("Time.1"),
                         list("Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.1","Time.2"),
                         list("Time.0","Time.2"),
                         list("Time.0","Time.1","Time.2")),
      text.scale=2)
```

Assess GO functional enrichment of interesting DE genes.

```{r}

#Shared across t0+t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across All Stages")

#Shared across t0+t1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 0 & Time 1")

#Shared across t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 1 & Time 2")

#Unique to Time 0
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 0")

#Unique to Time 1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 1")

#Unique to Time 2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 2")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

* Enrichment of GallegoRomero2015-DE.iPSCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 ( 985 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 441 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.54550 (  21 genes intersect) / (post.p>0.95): p = 0.00202 (  50 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 ( 269 genes intersect) / (post.p>0.95): p = 0.99990 ( 229 genes intersect)
    + Unique to Time 0              (post.p>0.5): p < 2.2e-16 ( 906 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 691 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.96900 (  21 genes intersect) / (post.p>0.95): p = 0.86390 (  48 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.66340 (   1 genes intersect) / (post.p>0.95): p = 0.58920 (   2 genes intersect)

* Enrichment of Blake2018-DE.iPSCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 ( 823 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 368 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.56370 (  17 genes intersect) / (post.p>0.95): p = 0.00345 (  42 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 ( 208 genes intersect) / (post.p>0.95): p = 1.00000 ( 160 genes intersect) 
    + Unique to Time 0              (post.p>0.5): p < 2.2e-16 ( 709 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 544 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.88020 (  19 genes intersect) / (post.p>0.95): p = 0.61570 (  43 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.57590 (   1 genes intersect) / (post.p>0.95): p = 0.46290 (   2 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1:
    + Shared across All Stages      (post.p>0.5): p = 6.9e-15 ( 740 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 344 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.35080 (  19 genes intersect) / (post.p>0.95): p = 0.07173 (  36 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.99990 ( 250 genes intersect) / (post.p>0.95): p = 0.99830 ( 193 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 2.1e-06 ( 609 genes intersect) / (post.p>0.95): p = 3.5e-16 ( 454 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.67210 (  22 genes intersect) / (post.p>0.95): p = 0.41620 (  46 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.57690 (   1 genes intersect) / (post.p>0.95): p = 0.46430 (   2 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2:
    + Shared across All Stages      (post.p>0.5): p = 8.6e-11 ( 752 genes intersect) / (post.p>0.95): p = 2.2e-10 ( 326 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.18350 (  22 genes intersect) / (post.p>0.95): p = 0.25010 (  34 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.95570 ( 297 genes intersect) / (post.p>0.95): p = 0.93120 ( 224 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.01417 ( 597 genes intersect) / (post.p>0.95): p = 1.1e-06 ( 423 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.70300 (  23 genes intersect) / (post.p>0.95): p = 0.10100 (  55 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.60150 (   1 genes intersect) / (post.p>0.95): p = 0.84120 (   5 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3:
    + Shared across All Stages      (post.p>0.5): p = 7.9e-07 ( 784 genes intersect) / (post.p>0.95): p = 1.6e-07 ( 336 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.43790 (  21 genes intersect) / (post.p>0.95): p = 0.39710 (  35 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.98030 ( 322 genes intersect) / (post.p>0.95): p = 0.97340 ( 241 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.18810 ( 627 genes intersect) / (post.p>0.95): p = 0.00048 ( 436 genes intersect) *
    + Unique to Time 1              (post.p>0.5): p = 0.64760 (  26 genes intersect) / (post.p>0.95): p = 0.12400 (  59 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.87150 (   1 genes intersect)

* Enrichment of Blake2018-DE.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.00206 ( 102 genes intersect) / (post.p>0.95): p = 0.04534 (  40 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.67090 (   2 genes intersect) / (post.p>0.95): p = 0.33200 (   5 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.95440 (  31 genes intersect) / (post.p>0.95): p = 0.93950 (  23 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 2.1e-05 ( 102 genes intersect) / (post.p>0.95): p = 4.2e-05 (  70 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.82270 (   2 genes intersect) / (post.p>0.95): p = 0.70610 (   5 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.99060 (  39 genes intersect) / (post.p>0.95): p = 0.97620 (  13 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.45910 (   2 genes intersect) / (post.p>0.95): p = 0.92680 (   1 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.03225 (  37 genes intersect) / (post.p>0.95): p = 0.17670 (  25 genes intersect) *
    + Unique to Time 0              (post.p>0.5): p = 0.98910 (  34 genes intersect) / (post.p>0.95): p = 0.98970 (  19 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.62540 (   2 genes intersect) / (post.p>0.95): p = 0.56660 (   4 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.99990 (  35 genes intersect) / (post.p>0.95): p = 0.99990 (   8 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.78100 (   2 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.37890 (  32 genes intersect) / (post.p>0.95): p = 0.41630 (  24 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.87970 (  45 genes intersect) / (post.p>0.95): p = 0.99790 (  19 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.67910 (   2 genes intersect) / (post.p>0.95): p = 0.27740 (   6 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 1.00000 ( 103 genes intersect) / (post.p>0.95): p = 1.00000 (  25 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.99110 (   1 genes intersect) / (post.p>0.95): p = 0.99960 (   1 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.10630 (  91 genes intersect) / (post.p>0.95): p = 0.33730 (  64 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.99560 ( 113 genes intersect) / (post.p>0.95): p = 0.99990 (  58 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.27980 (   8 genes intersect) / (post.p>0.95): p = 0.12840 (  16 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Blake2020-DE.heart:
    + Shared across All Stages      (post.p>0.5): p = 0.09067 ( 371 genes intersect) / (post.p>0.95): p = 2.5e-06 ( 185 genes intersect) *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.57390 (  10 genes intersect) / (post.p>0.95): p = 0.81680 (  14 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.54980 ( 178 genes intersect) / (post.p>0.95): p = 0.26770 ( 142 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.04948 ( 335 genes intersect) / (post.p>0.95): p = 0.54970 ( 194 genes intersect) *
    + Unique to Time 1              (post.p>0.5): p = 0.64150 (  13 genes intersect) / (post.p>0.95): p = 0.46570 (  27 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.21700 (   2 genes intersect)
* Enrichment of Blake2020-DE.kidney:
    + Shared across All Stages      (post.p>0.5): p = 1.0e-09 ( 172 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 104 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.64650 (   5 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.51990 (  58 genes intersect) / (post.p>0.95): p = 0.04485 (  55 genes intersect) *
    + Unique to Time 0              (post.p>0.5): p = 0.20990 ( 108 genes intersect) / (post.p>0.95): p = 0.42150 (  65 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.46690 (   5 genes intersect) / (post.p>0.95): p = 0.75080 (   7 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.25410 (   1 genes intersect)
* Enrichment of Blake2020-DE.liver:
    + Shared across All Stages      (post.p>0.5): p = 4.9e-05 ( 255 genes intersect) / (post.p>0.95): p = 2.8e-10 ( 135 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.94530 (   3 genes intersect) / (post.p>0.95): p = 0.67410 (   9 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.53110 ( 105 genes intersect) / (post.p>0.95): p = 0.01175 (  99 genes intersect) *
    + Unique to Time 0              (post.p>0.5): p = 0.02525 ( 206 genes intersect) / (post.p>0.95): p = 0.27380 ( 121 genes intersect) *
    + Unique to Time 1              (post.p>0.5): p = 0.29560 (  10 genes intersect) / (post.p>0.95): p = 0.95150 (  10 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.08893 (   2 genes intersect)
* Enrichment of Blake2020-DE.lung:
    + Shared across All Stages      (post.p>0.5): p = 3.1e-07 ( 166 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 104 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.96940 (   1 genes intersect) / (post.p>0.95): p = 0.82320 (   4 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.39730 (  62 genes intersect) / (post.p>0.95): p = 0.07050 (  55 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.95460 (  89 genes intersect) / (post.p>0.95): p = 0.77370 (  60 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.17900 (   7 genes intersect) / (post.p>0.95): p = 0.51390 (   9 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.14030 (   1 genes intersect) / (post.p>0.95): p = 0.26100 (   1 genes intersect)

* Enrichment of Blake2020-DE.heart-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.97770 ( 212 genes intersect) / (post.p>0.95): p = 0.65290 (  89 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.38450 (   8 genes intersect) / (post.p>0.95): p = 0.72450 (  10 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.27010 ( 128 genes intersect) / (post.p>0.95): p = 0.79530 (  85 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.03310 ( 234 genes intersect) / (post.p>0.95): p = 0.45330 ( 134 genes intersect) *
    + Unique to Time 1              (post.p>0.5): p = 0.84180 (   7 genes intersect) / (post.p>0.95): p = 0.24160 (  21 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.46840 (   1 genes intersect)
* Enrichment of Blake2020-DE.kidney-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.06654 (  39 genes intersect) / (post.p>0.95): p = 0.42520 (  13 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.44400 (   2 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.93010 (  11 genes intersect) / (post.p>0.95): p = 0.31000 (  14 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.00224 (  42 genes intersect) / (post.p>0.95): p = 0.00083 (  31 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.71030 (   1 genes intersect) / (post.p>0.95): p = 0.67830 (   2 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.liver-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.59360 (  93 genes intersect) / (post.p>0.95): p = 0.30350 (  40 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.93990 (   1 genes intersect) / (post.p>0.95): p = 0.67730 (   4 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.13660 (  56 genes intersect) / (post.p>0.95): p = 0.06113 (  46 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.02878 ( 100 genes intersect) / (post.p>0.95): p = 0.04260 (  65 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.72870 (   3 genes intersect) / (post.p>0.95): p = 0.92820 (   4 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.lung-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.24690 (  43 genes intersect) / (post.p>0.95): p = 0.07406 (  21 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.67980 (   1 genes intersect) / (post.p>0.95): p = 0.56300 (   2 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.70550 (  18 genes intersect) / (post.p>0.95): p = 0.89260 (  11 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.55850 (  34 genes intersect) / (post.p>0.95): p = 0.32470 (  24 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.78840 (   1 genes intersect) / (post.p>0.95): p = 0.55860 (   3 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.04818 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect) *

* Enrichment of Gokhman2021-DE.Parental.iPSCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 (1414 genes intersect) / (post.p>0.95): p = 2.3e-16 ( 570 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.26040 (  37 genes intersect) / (post.p>0.95): p = 0.01154 (  69 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.99970 ( 541 genes intersect) / (post.p>0.95): p = 0.90140 ( 432 genes intersect)
    + Unique to Time 0              (post.p>0.5): p < 2.2e-16 (1293 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 912 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.46700 (  47 genes intersect) / (post.p>0.95): p = 0.66380 (  85 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.48570 (   2 genes intersect) / (post.p>0.95): p = 0.10060 (   5 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 (1156 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 492 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.05183 (  34 genes intersect) / (post.p>0.95): p = 9.5e-05 (  65 genes intersect) **
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 ( 403 genes intersect) / (post.p>0.95): p = 0.80390 ( 346 genes intersect)
    + Unique to Time 0              (post.p>0.5): p < 2.2e-16 (1032 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 734 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.74480 (  34 genes intersect) / (post.p>0.95): p = 0.91560 (  61 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.77260 (   1 genes intersect) / (post.p>0.95): p = 0.16510 (   4 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 ( 504 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 236 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.10590 (  14 genes intersect) / (post.p>0.95): p = 0.01017 (  26 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.88470 ( 160 genes intersect) / (post.p>0.95): p = 0.04257 ( 149 genes intersect) *
    + Unique to Time 0              (post.p>0.5): p = 3.9e-05 ( 359 genes intersect) / (post.p>0.95): p = 1.2e-11 ( 237 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.93310 (   9 genes intersect) / (post.p>0.95): p = 0.79610 (  22 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.36950 (   1 genes intersect) / (post.p>0.95): p = 0.04131 (   3 genes intersect) *
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs:
    + Shared across All Stages      (post.p>0.5): p = 2.9e-16 ( 751 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 338 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.26420 (  20 genes intersect) / (post.p>0.95): p = 0.05090 (  37 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.47460 ( 306 genes intersect) / (post.p>0.95): p = 0.10970 ( 246 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.02355 ( 564 genes intersect) / (post.p>0.95): p = 0.00011 ( 388 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.88840 (  19 genes intersect) / (post.p>0.95): p = 0.43200 (  46 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.15680 (   2 genes intersect) / (post.p>0.95): p = 0.03784 (   4 genes intersect) *

* Enrichment of Gokhman2021-DE.Parental.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 7.2e-05 ( 998 genes intersect) / (post.p>0.95): p = 0.33250 ( 362 genes intersect) *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.41070 (  28 genes intersect) / (post.p>0.95): p = 0.33910 (  47 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.94570 ( 444 genes intersect) / (post.p>0.95): p = 0.91870 ( 335 genes intersect)
    + Unique to Time 0              (post.p>0.5): p < 2.2e-16 ( 989 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 673 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.13330 (  42 genes intersect) / (post.p>0.95): p = 0.22680 (  74 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.76840 (   1 genes intersect) / (post.p>0.95): p = 0.42640 (   3 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.00811 ( 592 genes intersect) / (post.p>0.95): p = 0.64530 ( 208 genes intersect) *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.05762 (  22 genes intersect) / (post.p>0.95): p = 0.13310 (  32 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.99810 ( 241 genes intersect) / (post.p>0.95): p = 0.83530 ( 200 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 8.2e-16 ( 628 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 444 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.50610 (  22 genes intersect) / (post.p>0.95): p = 0.78860 (  37 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.79220 (   1 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.61840 (  88 genes intersect) / (post.p>0.95): p = 0.91230 (  28 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.12040 (   5 genes intersect) / (post.p>0.95): p = 0.63940 (   4 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.00609 (  63 genes intersect) / (post.p>0.95): p = 0.00217 (  52 genes intersect) **
    + Unique to Time 0              (post.p>0.5): p = 0.99960 (  55 genes intersect) / (post.p>0.95): p = 0.99960 (  30 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.47990 (   4 genes intersect) / (post.p>0.95): p = 0.07609 (  11 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.47990 (   0 genes intersect) / (post.p>0.95): p = 0.20700 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.99310 ( 187 genes intersect) / (post.p>0.95): p = 0.99990 (  54 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.29270 (   8 genes intersect) / (post.p>0.95): p = 0.99470 (   4 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.00046 ( 144 genes intersect) / (post.p>0.95): p = 0.03262 ( 100 genes intersect) **
    + Unique to Time 0              (post.p>0.5): p = 0.99740 ( 160 genes intersect) / (post.p>0.95): p = 0.99250 (  98 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.76630 (   7 genes intersect) / (post.p>0.95): p = 0.08818 (  22 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.24960 (   1 genes intersect) / (post.p>0.95): p = 0.43700 (   1 genes intersect)

* Enrichment of Gokhman2020-DMR.bone:
    + Shared across All Stages      (post.p>0.5): p = 0.04100 (  65 genes intersect) / (post.p>0.95): p = 0.58950 (  20 genes intersect) *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.45910 (   2 genes intersect) / (post.p>0.95): p = 0.47690 (   3 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.83200 (  23 genes intersect) / (post.p>0.95): p = 0.08631 (  27 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.34320 (  50 genes intersect) / (post.p>0.95): p = 0.04395 (  39 genes intersect) *
    + Unique to Time 1              (post.p>0.5): p = 0.88180 (   1 genes intersect) / (post.p>0.95): p = 0.56660 (   4 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2020-skeletal.genes:
    + Shared across All Stages      (post.p>0.5): p = 0.79210 ( 246 genes intersect) / (post.p>0.95): p = 0.83840 (  91 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.03156 (  13 genes intersect) / (post.p>0.95): p = 0.47420 (  13 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.44830 ( 133 genes intersect) / (post.p>0.95): p = 0.60920 (  97 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.01845 ( 255 genes intersect) / (post.p>0.95): p = 0.36920 ( 147 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.13300 (  14 genes intersect) / (post.p>0.95): p = 0.28190 (  22 genes intersect) *
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.49620 (   1 genes intersect)
* Enrichment of Hsu2019-GWAS.hip.geometry:
    + Shared across All Stages      (post.p>0.5): p = 0.43760 (   2 genes intersect) / (post.p>0.95): p = 0.44030 (   1 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.57300 (   1 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Styrkarsdottir2019-GWAS.bone.area:
    + Shared across All Stages      (post.p>0.5): p = 0.21460 (   3 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.58850 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.44240 (   2 genes intersect) / (post.p>0.95): p = 0.23070 (   2 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Yengo2018-GWAS.height:
    + Shared across All Stages      (post.p>0.5): p = 0.96750 ( 208 genes intersect) / (post.p>0.95): p = 0.99590 (  68 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.35560 (   8 genes intersect) / (post.p>0.95): p = 0.69200 (  10 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.71940 ( 113 genes intersect) / (post.p>0.95): p = 0.99540 (  68 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.05573 ( 225 genes intersect) / (post.p>0.95): p = 0.73220 ( 123 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.30480 (  11 genes intersect) / (post.p>0.95): p = 0.14330 (  22 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 0.26430 (   1 genes intersect) / (post.p>0.95): p = 0.45880 (   1 genes intersect)

* Enrichment of Steinberg2020-GWAS.OA:
    + Shared across All Stages      (post.p>0.5): p = 0.94640 (  62 genes intersect) / (post.p>0.95): p = 0.96780 (  20 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.63420 (   2 genes intersect) / (post.p>0.95): p = 0.69510 (   3 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.42880 (  39 genes intersect) / (post.p>0.95): p = 0.77670 (  25 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.52850 (  65 genes intersect) / (post.p>0.95): p = 0.17460 (  47 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.00853 (   8 genes intersect) / (post.p>0.95): p = 0.64850 (   5 genes intersect) *
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.17180 (   1 genes intersect)
* Enrichment of Morris2020-GWAS.OP:
    + Shared across All Stages      (post.p>0.5): p = 0.61740 ( 102 genes intersect) / (post.p>0.95): p = 0.99770 (  25 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.18500 (   5 genes intersect) / (post.p>0.95): p = 0.24050 (   7 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.71140 (  50 genes intersect) / (post.p>0.95): p = 0.54330 (  40 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.39810 (  95 genes intersect) / (post.p>0.95): p = 0.11770 (  67 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.23070 (   6 genes intersect) / (post.p>0.95): p = 0.38000 (   9 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

```{r, eval=FALSE}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #Shared across All Stages
  print("Shared across All Stages")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 0 & Time 1
  print("Shared across Time 0 & Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 1 & Time 2
  print("Shared across Time 1 & Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Unique to Time 0 (FDR<0.01)
  print("Unique to Time 0")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 1 (FDR<0.01)
  print("Unique to Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 2 (FDR<0.01)
  print("Unique to Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```


## General Clusters (res=0.05)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## Osteogenic Clusters (res=0.50)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## Osteogenic Clusters (res=0.75)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## Osteogenic Clusters (res=0.50) (Time 2 only)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## Osteogenic Clusters (res=0.75) (Time 2 only)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

## Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

#### each osteogenic ad hoc classification separately

Note: Too few DE genes shared across pre+obl+emb+min and obl+emb+min, so these were not examined further. Instead, DE genes shared across obl+emb+min+mat and emb+min+mat were considered.

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("preosteoblast",length(which(dataUpset$preosteoblast==1))),
                                       rep("osteoblast",length(which(dataUpset$osteoblast==1))),
                                       rep("embedding.osteoblast",length(which(dataUpset$embedding.osteoblast==1))),
                                       rep("mineralizing.osteoblast",length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep("maturing.osteocyte",length(which(dataUpset$maturing.osteocyte==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),"preosteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),"osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),"embedding.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),"mineralizing.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),"maturing.osteocyte"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                               labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(rowSums(dataUpset)>=1)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)

```

Assess GO functional enrichment of interesting DE genes.

```{r}

#Shared across Osteogenic Ad Hoc Classifications
subGenes <- rownames(dataUpset)[which(rowSums(dataUpset)==5)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Osteogenic Ad Hoc Classifications")

#Shared across pre+emb+min+mat
subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across 4 Osteogenic Ad Hoc Classifications (A)")

#Shared across obl+emb+min+mat
subGenes <- rownames(dataUpset)[which(dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across 4 Osteogenic Ad Hoc Classifications (B)")

#Shared across obl+emb+min
subGenes <- rownames(dataUpset)[which(dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across 3 Osteogenic Ad Hoc Classifications (A)")

#Shared across emb+min+mat
subGenes <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across 3 Osteogenic Ad Hoc Classifications (B)")

#Shared between Embedding and Mineralizing Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared between Embedding and Mineralizing Osteoblasts")

#Unique to Embedding Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Osteogenic Ad Hoc Classifications")

#Unique to Mineralizing Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Osteogenic Ad Hoc Classifications")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

* Enrichment of GallegoRomero2015-DE.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 0.02101 ( 461 genes intersect) / (post.p>0.95): p = 0.09819 ( 144 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 0.00080 ( 537 genes intersect) / (post.p>0.95): p = 0.02235 ( 209 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00105 ( 696 genes intersect) / (post.p>0.95): p = 0.00141 ( 461 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00142 ( 693 genes intersect) / (post.p>0.95): p = 0.00172 ( 454 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.00163 ( 670 genes intersect) / (post.p>0.95): p = 0.01099 ( 379 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.02296 ( 416 genes intersect) / (post.p>0.95): p = 0.05478 ( 102 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.01166 ( 107 genes intersect) / (post.p>0.95): p = 0.17200 (  82 genes intersect) *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.33410 ( 102 genes intersect) / (post.p>0.95): p = 0.04062 ( 156 genes intersect)  *

* Enrichment of Blake2018-DE.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 0.31750 ( 367 genes intersect) / (post.p>0.95): p = 0.86690 ( 101 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.09657 ( 427 genes intersect) / (post.p>0.95): p = 0.49290 ( 157 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.03639 ( 568 genes intersect) / (post.p>0.95): p = 0.02125 ( 378 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.03621 ( 567 genes intersect) / (post.p>0.95): p = 0.01534 ( 375 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.02033 ( 553 genes intersect) / (post.p>0.95): p = 0.07699 ( 310 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.28350 ( 332 genes intersect) / (post.p>0.95): p = 0.89700 (  66 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.04028 (  88 genes intersect) / (post.p>0.95): p = 0.28990 (  67 genes intersect) *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.02666 (  98 genes intersect) / (post.p>0.95): p = 0.00253 ( 143 genes intersect) **
* Enrichment of Blake2018-DE.iPSC-endoderm-day1:
    + All preosteoblasts                         (post.p>0.5): p = 0.20420 ( 374 genes intersect) / (post.p>0.95): p = 0.68520 ( 107 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.22490 ( 420 genes intersect) / (post.p>0.95): p = 0.66660 ( 153 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.05419 ( 567 genes intersect) / (post.p>0.95): p = 0.01023 ( 384 genes intersect)  *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.05394 ( 566 genes intersect) / (post.p>0.95): p = 0.01012 ( 379 genes intersect)  *
    + All maturing osteocytes                    (post.p>0.5): p = 0.03559 ( 551 genes intersect) / (post.p>0.95): p = 0.01736 ( 321 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.34240 ( 331 genes intersect) / (post.p>0.95): p = 0.81380 (  69 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.21960 (  81 genes intersect) / (post.p>0.95): p = 0.58830 (  62 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.05118 (  96 genes intersect) / (post.p>0.95): p = 4.9e-05 ( 154 genes intersect)  *
* Enrichment of Blake2018-DE.iPSC-endoderm-day2:
    + All preosteoblasts                         (post.p>0.5): p = 0.39880 ( 384 genes intersect) / (post.p>0.95): p = 0.74810 ( 111 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.11010 ( 449 genes intersect) / (post.p>0.95): p = 0.05360 ( 183 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.09385 ( 590 genes intersect) / (post.p>0.95): p = 0.06606 ( 390 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.09302 ( 589 genes intersect) / (post.p>0.95): p = 0.05631 ( 386 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.10880 ( 568 genes intersect) / (post.p>0.95): p = 0.03619 ( 332 genes intersect)  *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.36490 ( 347 genes intersect) / (post.p>0.95): p = 0.76610 (  74 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.04526 (  92 genes intersect) / (post.p>0.95): p = 0.01062 (  83 genes intersect) **
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.28290 (  92 genes intersect) / (post.p>0.95): p = 0.03924 ( 140 genes intersect)  *
* Enrichment of Blake2018-DE.iPSC-endoderm-day3:
    + All preosteoblasts                         (post.p>0.5): p = 0.31670 ( 423 genes intersect) / (post.p>0.95): p = 0.62200 ( 125 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.25320 ( 480 genes intersect) / (post.p>0.95): p = 0.02372 ( 203 genes intersect)  *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.04234 ( 651 genes intersect) / (post.p>0.95): p = 0.00817 ( 439 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.04633 ( 649 genes intersect) / (post.p>0.95): p = 0.00413 ( 437 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.05617 ( 626 genes intersect) / (post.p>0.95): p = 0.00999 ( 369 genes intersect)  *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.41390 ( 377 genes intersect) / (post.p>0.95): p = 0.59100 (  85 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.19970 (  93 genes intersect) / (post.p>0.95): p = 0.01843 (  88 genes intersect)  *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.03897 ( 110 genes intersect) / (post.p>0.95): p = 0.00856 ( 158 genes intersect) **

* Enrichment of Blake2018-DE.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.80340 (  42 genes intersect) / (post.p>0.95): p = 0.78210 (  12 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.79290 (  48 genes intersect) / (post.p>0.95): p = 0.91990 (  15 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.80470 (  64 genes intersect) / (post.p>0.95): p = 0.60530 (  44 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.79970 (  64 genes intersect) / (post.p>0.95): p = 0.56560 (  44 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.74580 (  63 genes intersect) / (post.p>0.95): p = 0.77090 (  34 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.86840 (  36 genes intersect) / (post.p>0.95): p = 0.76510 (   8 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.26760 (  12 genes intersect) / (post.p>0.95): p = 0.96650 (   4 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.75350 (   9 genes intersect) / (post.p>0.95): p = 0.26180 (  18 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.28400 (  36 genes intersect) / (post.p>0.95): p = 0.41780 (  11 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.58380 (  36 genes intersect) / (post.p>0.95): p = 0.66770 (  13 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.43190 (  50 genes intersect) / (post.p>0.95): p = 0.16420 (  37 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.42600 (  50 genes intersect) / (post.p>0.95): p = 0.14360 (  37 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.38470 (  49 genes intersect) / (post.p>0.95): p = 0.13780 (  32 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.47500 (  30 genes intersect) / (post.p>0.95): p = 0.23850 (   9 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.67320 (   6 genes intersect) / (post.p>0.95): p = 0.92760 (   3 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.62330 (   7 genes intersect) / (post.p>0.95): p = 0.01916 (  18 genes intersect)  *
* Enrichment of Blake2018-DE.iPSC-endoderm-day2-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.94530 (  28 genes intersect) / (post.p>0.95): p = 0.78620 (   9 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.92680 (  33 genes intersect) / (post.p>0.95): p = 0.71350 (  14 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.94760 (  44 genes intersect) / (post.p>0.95): p = 0.99150 (  23 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.94600 (  44 genes intersect) / (post.p>0.95): p = 0.98940 (  23 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.93260 (  43 genes intersect) / (post.p>0.95): p = 0.91000 (  23 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.93980 (  25 genes intersect) / (post.p>0.95): p = 0.63050 (   7 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.62700 (   7 genes intersect) / (post.p>0.95): p = 0.44620 (   7 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.59170 (   8 genes intersect) / (post.p>0.95): p = 0.95220 (   7 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.49520 (  94 genes intersect) / (post.p>0.95): p = 0.50670 (  29 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.62660 ( 103 genes intersect) / (post.p>0.95): p = 0.31380 (  44 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.41140 ( 142 genes intersect) / (post.p>0.95): p = 0.62190 (  88 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.43940 ( 141 genes intersect) / (post.p>0.95): p = 0.56510 (  88 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.37360 ( 138 genes intersect) / (post.p>0.95): p = 0.72750 (  71 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.47580 (  85 genes intersect) / (post.p>0.95): p = 0.48270 (  20 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.74890 (  17 genes intersect) / (post.p>0.95): p = 0.37010 (  18 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.12690 (  27 genes intersect) / (post.p>0.95): p = 0.86940 (  25 genes intersect)

* Enrichment of Blake2020-DE.heart:
    + All preosteoblasts                         (post.p>0.5): p = 0.10480 ( 234 genes intersect) / (post.p>0.95): p = 0.56640 (  66 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.16630 ( 259 genes intersect) / (post.p>0.95): p = 0.13050 ( 105 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00082 ( 372 genes intersect) / (post.p>0.95): p = 0.07024 ( 229 genes intersect) *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00088 ( 371 genes intersect) / (post.p>0.95): p = 0.08113 ( 225 genes intersect) *
    + All maturing osteocytes                    (post.p>0.5): p = 0.00039 ( 363 genes intersect) / (post.p>0.95): p = 0.05927 ( 194 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.20860 ( 206 genes intersect) / (post.p>0.95): p = 0.29420 (  49 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.34260 (  48 genes intersect) / (post.p>0.95): p = 0.16460 (  44 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 3.5e-06 (  81 genes intersect) / (post.p>0.95): p = 0.03453 (  85 genes intersect) **
* Enrichment of Blake2020-DE.kidney:
    + All preosteoblasts                         (post.p>0.5): p = 4.2e-07 ( 111 genes intersect) / (post.p>0.95): p = 0.00608 (  34 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 1.7e-05 ( 115 genes intersect) / (post.p>0.95): p = 0.00074 (  49 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 3.2e-10 ( 164 genes intersect) / (post.p>0.95): p = 5.6e-10 ( 118 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 2.7e-10 ( 164 genes intersect) / (post.p>0.95): p = 4.9e-10 ( 117 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 2.4e-10 ( 160 genes intersect) / (post.p>0.95): p = 1.8e-09 ( 102 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.6e-05 (  96 genes intersect) / (post.p>0.95): p = 0.02203 (  24 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.39330 (  16 genes intersect) / (post.p>0.95): p = 0.04068 (  19 genes intersect)  *
    + Shared across emb+min+mat                  (post.p>0.5): p = 7.6e-05 (  33 genes intersect) / (post.p>0.95): p = 1.3e-07 (  50 genes intersect) **
* Enrichment of Blake2020-DE.liver:
    + All preosteoblasts                         (post.p>0.5): p = 3.3e-06 ( 174 genes intersect) / (post.p>0.95): p = 0.01807 (  52 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.00035 ( 180 genes intersect) / (post.p>0.95): p = 0.00144 (  77 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 4.4e-08 ( 254 genes intersect) / (post.p>0.95): p = 2.0e-07 ( 175 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 5.8e-08 ( 253 genes intersect) / (post.p>0.95): p = 1.2e-07 ( 174 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.4e-08 ( 249 genes intersect) / (post.p>0.95): p = 2.7e-06 ( 146 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00016 ( 150 genes intersect) / (post.p>0.95): p = 0.07291 (  34 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.46900 (  27 genes intersect) / (post.p>0.95): p = 0.02132 (  32 genes intersect)  *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.00033 (  48 genes intersect) / (post.p>0.95): p = 0.00020 (  64 genes intersect) **
* Enrichment of Blake2020-DE.lung:
    + All preosteoblasts                         (post.p>0.5): p = 0.00039 (  99 genes intersect) / (post.p>0.95): p = 0.03360 (  31 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.00073 ( 108 genes intersect) / (post.p>0.95): p = 0.00801 (  45 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 2.6e-05 ( 145 genes intersect) / (post.p>0.95): p = 2.4e-05 ( 102 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 3.7e-05 ( 144 genes intersect) / (post.p>0.95): p = 3.8e-05 ( 100 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.5e-05 ( 142 genes intersect) / (post.p>0.95): p = 0.00023 (  84 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00053 (  90 genes intersect) / (post.p>0.95): p = 0.07104 (  21 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.31770 (  17 genes intersect) / (post.p>0.95): p = 0.12390 (  17 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.01374 (  26 genes intersect) / (post.p>0.95): p = 0.00332 (  37 genes intersect) **
    
* Enrichment of Blake2020-DE.heart-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.94720 ( 131 genes intersect) / (post.p>0.95): p = 0.89830 (  38 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.81670 ( 157 genes intersect) / (post.p>0.95): p = 0.85490 (  57 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.54470 ( 219 genes intersect) / (post.p>0.95): p = 0.97860 ( 122 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.56300 ( 218 genes intersect) / (post.p>0.95): p = 0.98380 ( 119 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.48620 ( 213 genes intersect) / (post.p>0.95): p = 0.95690 ( 103 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.93460 ( 118 genes intersect) / (post.p>0.95): p = 0.65920 (  29 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.17160 (  36 genes intersect) / (post.p>0.95): p = 0.81690 (  22 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.01895 (  46 genes intersect) / (post.p>0.95): p = 0.79730 (  43 genes intersect) *
* Enrichment of Blake2020-DE.kidney-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.09862 (  25 genes intersect) / (post.p>0.95): p = 0.84690 (   4 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.05881 (  29 genes intersect) / (post.p>0.95): p = 0.21460 (  11 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.05186 (  37 genes intersect) / (post.p>0.95): p = 0.02743 (  27 genes intersect)  *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.05062 (  37 genes intersect) / (post.p>0.95): p = 0.02336 (  27 genes intersect)  *
    + All maturing osteocytes                    (post.p>0.5): p = 0.04932 (  36 genes intersect) / (post.p>0.95): p = 0.05667 (  22 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.09214 (  23 genes intersect) / (post.p>0.95): p = 0.56970 (   4 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.36750 (   5 genes intersect) / (post.p>0.95): p = 0.24560 (   5 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.27830 (   6 genes intersect) / (post.p>0.95): p = 0.00947 (  13 genes intersect)  *
* Enrichment of Blake2020-DE.liver-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.17530 (  64 genes intersect) / (post.p>0.95): p = 0.39210 (  19 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.20020 (  71 genes intersect) / (post.p>0.95): p = 0.28100 (  28 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.14800 (  94 genes intersect) / (post.p>0.95): p = 0.17250 (  62 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.17250 (  93 genes intersect) / (post.p>0.95): p = 0.14460 (  62 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.11800 (  92 genes intersect) / (post.p>0.95): p = 0.18980 (  52 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.21490 (  57 genes intersect) / (post.p>0.95): p = 0.40940 (  13 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.40460 (  13 genes intersect) / (post.p>0.95): p = 0.41410 (  11 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.33040 (  15 genes intersect) / (post.p>0.95): p = 0.16520 (  23 genes intersect)
* Enrichment of Blake2020-DE.lung-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.32610 (  26 genes intersect) / (post.p>0.95): p = 0.59070 (   7 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.20260 (  31 genes intersect) / (post.p>0.95): p = 0.32980 (  12 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.61430 (  34 genes intersect) / (post.p>0.95): p = 0.51230 (  23 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.68110 (  33 genes intersect) / (post.p>0.95): p = 0.57430 (  22 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.59690 (  33 genes intersect) / (post.p>0.95): p = 0.63320 (  18 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.28920 (  24 genes intersect) / (post.p>0.95): p = 0.36700 (   6 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.21810 (   7 genes intersect) / (post.p>0.95): p = 0.39520 (   5 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.78200 (   6 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 6.6e-05 ( 754 genes intersect) / (post.p>0.95): p = 0.09084 ( 225 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 9.9e-05 ( 844 genes intersect) / (post.p>0.95): p = 7.4e-05 ( 345 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 1.9e-05 (1108 genes intersect) / (post.p>0.95): p = 0.00111 ( 716 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 2.9e-05 (1104 genes intersect) / (post.p>0.95): p = 0.00082 ( 708 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.7e-05 (1070 genes intersect) / (post.p>0.95): p = 0.00714 ( 593 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 6.6e-05 ( 682 genes intersect) / (post.p>0.95): p = 0.00520 ( 165 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.31200 ( 147 genes intersect) / (post.p>0.95): p = 0.02304 ( 136 genes intersect)  *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.08654 ( 170 genes intersect) / (post.p>0.95): p = 0.14750 ( 234 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 7.6e-06 ( 616 genes intersect) / (post.p>0.95): p = 0.00435 ( 193 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 5.2e-06 ( 691 genes intersect) / (post.p>0.95): p = 0.00014 ( 279 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 1.4e-06 ( 901 genes intersect) / (post.p>0.95): p = 1.0e-05 ( 595 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 1.6e-06 ( 899 genes intersect) / (post.p>0.95): p = 1.1e-05 ( 587 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.1e-06 ( 872 genes intersect) / (post.p>0.95): p = 0.00016 ( 493 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.4e-05 ( 556 genes intersect) / (post.p>0.95): p = 0.01139 ( 132 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.10660 ( 123 genes intersect) / (post.p>0.95): p = 0.00948 ( 113 genes intersect)  *
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.16610 ( 133 genes intersect) / (post.p>0.95): p = 0.10100 ( 189 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs:
    + All preosteoblasts                         (post.p>0.5): p = 4.9e-06 ( 251 genes intersect) / (post.p>0.95): p = 2.1e-05 (  91 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 1.5e-08 ( 294 genes intersect) / (post.p>0.95): p = 5.8e-05 ( 119 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 7.7e-08 ( 368 genes intersect) / (post.p>0.95): p = 2.7e-06 ( 245 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 8.9e-08 ( 367 genes intersect) / (post.p>0.95): p = 1.3e-06 ( 244 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.4e-07 ( 355 genes intersect) / (post.p>0.95): p = 2.6e-05 ( 204 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.0e-05 ( 227 genes intersect) / (post.p>0.95): p = 0.00176 (  59 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.00097 (  60 genes intersect) / (post.p>0.95): p = 0.00316 (  50 genes intersect) **
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.57870 (  44 genes intersect) / (post.p>0.95): p = 0.48630 (  64 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs:
    + All preosteoblasts                         (post.p>0.5): p = 2.1e-05 ( 412 genes intersect) / (post.p>0.95): p = 0.00017 ( 140 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 5.1e-07 ( 473 genes intersect) / (post.p>0.95): p = 3.8e-06 ( 200 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 4.7e-07 ( 608 genes intersect) / (post.p>0.95): p = 2.6e-06 ( 406 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 4.6e-07 ( 607 genes intersect) / (post.p>0.95): p = 1.4e-06 ( 403 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 9.0e-07 ( 586 genes intersect) / (post.p>0.95): p = 2.0e-05 ( 339 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 2.2e-05 ( 374 genes intersect) / (post.p>0.95): p = 0.00017 ( 100 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.01023 (  90 genes intersect) / (post.p>0.95): p = 0.01146 (  77 genes intersect) **
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.31870 (  84 genes intersect) / (post.p>0.95): p = 0.14720 ( 123 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.09878 ( 566 genes intersect) / (post.p>0.95): p = 0.89540 ( 155 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.39050 ( 619 genes intersect) / (post.p>0.95): p = 0.22250 ( 246 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.24520 ( 824 genes intersect) / (post.p>0.95): p = 0.43270 ( 528 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.26860 ( 821 genes intersect) / (post.p>0.95): p = 0.44870 ( 520 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.21400 ( 797 genes intersect) / (post.p>0.95): p = 0.56670 ( 436 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.09260 ( 511 genes intersect) / (post.p>0.95): p = 0.38330 ( 116 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.95910 (  99 genes intersect) / (post.p>0.95): p = 0.52510 (  95 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.21240 ( 132 genes intersect) / (post.p>0.95): p = 0.15540 ( 187 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.07992 ( 341 genes intersect) / (post.p>0.95): p = 0.39210 ( 101 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.29050 ( 370 genes intersect) / (post.p>0.95): p = 0.20450 ( 148 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.29920 ( 480 genes intersect) / (post.p>0.95): p = 0.27120 ( 318 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.30100 ( 485 genes intersect) / (post.p>0.95): p = 0.31030 ( 312 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.25780 ( 471 genes intersect) / (post.p>0.95): p = 0.30700 ( 265 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.11550 ( 305 genes intersect) / (post.p>0.95): p = 0.66310 (  64 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.86660 (  59 genes intersect) / (post.p>0.95): p = 0.07198 (  66 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.64540 (  71 genes intersect) / (post.p>0.95): p = 0.72040 (  99 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.06695 (  63 genes intersect) / (post.p>0.95): p = 0.12910 (  21 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.09266 (  69 genes intersect) / (post.p>0.95): p = 0.76330 (  20 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.24290 (  84 genes intersect) / (post.p>0.95): p = 0.18290 (  57 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.23690 (  84 genes intersect) / (post.p>0.95): p = 0.19340 (  56 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.24850 (  81 genes intersect) / (post.p>0.95): p = 0.23860 (  47 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.09611 (  56 genes intersect) / (post.p>0.95): p = 0.65680 (  10 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.40530 (  12 genes intersect) / (post.p>0.95): p = 0.57090 (   9 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.98290 (   6 genes intersect) / (post.p>0.95): p = 0.53610 (  17 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.17480 ( 137 genes intersect) / (post.p>0.95): p = 0.07688 (  48 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.22330 ( 152 genes intersect) / (post.p>0.95): p = 0.03080 (  69 genes intersect)  *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.39380 ( 193 genes intersect) / (post.p>0.95): p = 0.27720 ( 129 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.38210 ( 193 genes intersect) / (post.p>0.95): p = 0.25270 ( 128 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.43420 ( 185 genes intersect) / (post.p>0.95): p = 0.18930 ( 111 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.19690 ( 123 genes intersect) / (post.p>0.95): p = 0.14880 (  32 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.56070 (  26 genes intersect) / (post.p>0.95): p = 0.05533 (  30 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.93910 (  22 genes intersect) / (post.p>0.95): p = 0.93070 (  33 genes intersect)

* Enrichment of Gokhman2020-DMR.bone:
    + All preosteoblasts                         (post.p>0.5): p = 0.01739 (  43 genes intersect) / (post.p>0.95): p = 0.10120 (  14 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 0.00929 (  49 genes intersect) / (post.p>0.95): p = 0.08786 (  19 genes intersect) *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00818 (  62 genes intersect) / (post.p>0.95): p = 0.05397 (  39 genes intersect) *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00784 (  62 genes intersect) / (post.p>0.95): p = 0.06566 (  38 genes intersect) *
    + All maturing osteocytes                    (post.p>0.5): p = 0.01321 (  59 genes intersect) / (post.p>0.95): p = 0.01737 (  36 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00479 (  42 genes intersect) / (post.p>0.95): p = 0.20430 (   9 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.63360 (   6 genes intersect) / (post.p>0.95): p = 0.09773 (   9 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.18160 (  10 genes intersect) / (post.p>0.95): p = 0.21490 (  13 genes intersect)

* Enrichment of Gokhman2020-skeletal.genes:
    + All preosteoblasts                         (post.p>0.5): p = 0.41900 ( 158 genes intersect) / (post.p>0.95): p = 0.74210 (  44 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.32550 ( 181 genes intersect) / (post.p>0.95): p = 0.74890 (  63 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.40350 ( 235 genes intersect) / (post.p>0.95): p = 0.75400 ( 143 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.39030 ( 235 genes intersect) / (post.p>0.95): p = 0.78150 ( 140 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.33580 ( 229 genes intersect) / (post.p>0.95): p = 0.95510 ( 109 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.50540 ( 140 genes intersect) / (post.p>0.95): p = 0.55280 (  32 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.26220 (  36 genes intersect) / (post.p>0.95): p = 0.98970 (  17 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.57180 (  35 genes intersect) / (post.p>0.95): p = 0.60500 (  49 genes intersect)
* Enrichment of Hsu2019-GWAS.hip.geometry:
    + All preosteoblasts                         (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.73260 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.73180 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.71820 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.16900 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Styrkarsdottir2019-GWAS.bone.area:
    + All preosteoblasts                         (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.73260 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.73180 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.71820 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.16900 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Yengo2018-GWAS.height:
    + All preosteoblasts                         (post.p>0.5): p = 0.83660 ( 131 genes intersect) / (post.p>0.95): p = 0.89980 (  36 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.77620 ( 151 genes intersect) / (post.p>0.95): p = 0.59120 (  60 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.89270 ( 195 genes intersect) / (post.p>0.95): p = 0.94560 ( 120 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.90170 ( 194 genes intersect) / (post.p>0.95): p = 0.94780 ( 118 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.92930 ( 185 genes intersect) / (post.p>0.95): p = 0.87570 ( 103 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.72770 ( 121 genes intersect) / (post.p>0.95): p = 0.76970 (  26 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.76380 (  26 genes intersect) / (post.p>0.95): p = 0.04923 (  33 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.81500 (  28 genes intersect) / (post.p>0.95): p = 0.96590 (  35 genes intersect)

* Enrichment of Steinberg2020-GWAS.OA:
    + All preosteoblasts                         (post.p>0.5): p = 0.24980 (  51 genes intersect) / (post.p>0.95): p = 0.07878 (  20 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.08312 (  62 genes intersect) / (post.p>0.95): p = 0.15640 (  25 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.18860 (  76 genes intersect) / (post.p>0.95): p = 0.44380 (  46 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.22140 (  75 genes intersect) / (post.p>0.95): p = 0.46780 (  45 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.13730 (  75 genes intersect) / (post.p>0.95): p = 0.18190 (  43 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.20800 (  47 genes intersect) / (post.p>0.95): p = 0.03289 (  16 genes intersect)  *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.05896 (  15 genes intersect) / (post.p>0.95): p = 0.56560 (   8 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.74160 (   9 genes intersect) / (post.p>0.95): p = 0.54090 (  15 genes intersect)
* Enrichment of Morris2020-GWAS.OP:
    + All preosteoblasts                         (post.p>0.5): p = 0.28410 (  68 genes intersect) / (post.p>0.95): p = 0.30780 (  22 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.25810 (  77 genes intersect) / (post.p>0.95): p = 0.49810 (  28 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.27640 ( 100 genes intersect) / (post.p>0.95): p = 0.86140 (  54 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.26930 ( 100 genes intersect) / (post.p>0.95): p = 0.83380 (  54 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.22290 (  98 genes intersect) / (post.p>0.95): p = 0.64950 (  49 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.21440 (  63 genes intersect) / (post.p>0.95): p = 0.44960 (  14 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 0.44450 (  14 genes intersect) / (post.p>0.95): p = 0.22070 (  14 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 0.38370 (  16 genes intersect) / (post.p>0.95): p = 0.97490 (  14 genes intersect)

```{r, eval=FALSE}

#Enrichment of external gene sets in cormotif DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All preosteoblasts
  print("All preosteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All osteoblasts
  print("All osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All embedding osteoblasts
  print("All embedding osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All mineralizing osteoblasts
  print("All mineralizing osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All maturing osteocytes
  print("All maturing osteocytes")
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications
  print("Shared across all ostadhoc classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across obl+emb+min+mat
  print("Shared across obl+emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across emb+min+mat
  print("Shared across emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```



### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)
### Examine Numbers, Overlap, and logFC of Cormotif DE Genes
### Examine Interesting DE Genes and Assess Functional/GeneSet Enrichment

