---
title: "10x-data-integrate"
author: "Genevieve Housman"
date: "August 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Integrate Data

For each sample, normalization, variance stabilization, and regression of unwanted variation were performed. Following this, samples were integrated using shared highly variable genes.

For the first...

Another question was how should the data be subsetted before normalization. The following subsets were assessed:

* separate species and cell types (n=6) (EX: Human iPSCs)
* separate human-chimp pairs and cell types (n=21) (EX: H1+C1 iPSCs)
* separate individuals and cell types (n=42) (EX: H1 iPSCs)
* separate species but keep all cell types together (n=2) (EX: Human iPSCs+MSCs+Osteoblasts)
* separate human-chimp pairs but keep all cell types together (n=7) (EX: H1+C1 iPSCs+MSCs+Osteoblasts)
* separate individuals but keep all cell types together (n=14) (EX: H1 iPSCs+MSCs+Osteoblasts)

What I used:

* separate individuals and cell types (n=42) (EX: H1 iPSCs)
* separate individuals but keep all cell types together

**Step 1 - Normalization**

In Seurat the main normalization methods are:

1. Log Normalization
    + Within a cell, gene counts are divided by the total counts, multiplied by a scale.factor (10000), and natural-log transformed using log1p.
2. SCTransform
    + Alternative workflow that perfoms log normalization, variable feature identification, and data scaling in one command.

Because data are more difficult to manipulate with SCTransform, standard log normalization is used here.

**Step 2 - Calculate Cell Cycle Score**

Cell cycle scores are assigned to cells based on G2/M and S phase markers in each cell. The calculation depends on average gene expression levels across cells in a seurat object, so merging or subsetting data can affect these scores. We calculate these scores within separate samples (n=42, each individual and cell type separated) rather than after different samples are merged together.

Possible Cell Cycle Assignments

* G1: beginning of interphase
* S: synthesis phase
* G2: end of interphase prior to mitosis

**Step 3 - Variance Stabilization**

Identifying variable genes depends on expression patterns across cells in a seurat object, so merging or subsetting data can affect which genes are identified as variable.

**Step 4 - Regression of Unwanted Variation**

We do not regress out unwanted variation in our data subsets.

**Step 5 - Data Integration**





**References**

* https://satijalab.org/seurat/v3.2/integration.html
* https://hbctraining.github.io/scRNA-seq/lessons/06_SC_SCT_and_integration.html
* https://hbctraining.github.io/scRNA-seq/lessons/cell_cycle_scoring.html
* See the following scripts for previous trial runs:
  + ./analysis/v1/v1-10x-data-normalize-log.Rmd
  + ./analysis/v1/v1-10x-data-normalize-sct.Rmd

```{r}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

### Load and Prepare Data

Prepare filtered 10X data by separating each individual and cell type into seperate datasets.

```{r}

#Load batch info
batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")

#Read in files
data <- readRDS("./data/cellranger-data-full/data.filter.rds")

#Load cell cycle markers (Tirosh et al., 2016 Science) with seurat
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

#Separate human and chimp data within each collection into separate objects
objects <- list()
idx <- 1
for (i in 1:length(batch$Sample_Name_at_Core))
{
  obj <- subset(data[[i]], Species=='Human')
  objects[[idx]] <- CreateSeuratObject(obj@assays$RNA@counts, meta.data=obj@meta.data)
  idx <- idx+1
  obj <- subset(data[[i]], Species=='Chimp')
  objects[[idx]] <- CreateSeuratObject(obj@assays$RNA@counts, meta.data=obj@meta.data)
  idx <- idx+1
}
rm(data,obj,idx)

```

### Normalize Data and Score Cell Cycle

Because cell cycle score depends on average gene expression levels across cells in a seurat object, it is probably best to calculate these scores within separate samples (n=42, each individual and cell type separated) rather than after different samples are merged together.

```{r}

#Perform log normalization and cell cycle scoring on each individual and cell type
for (i in 1:length(objects)) {
  print(i)
  objects[[i]] <- NormalizeData(objects[[i]], normalization.method="LogNormalize", scale.factor=10000, verbose=FALSE)
  objects[[i]] <- CellCycleScoring(objects[[i]], s.features=s.genes, g2m.features=g2m.genes, set.ident=FALSE)
}
rm(s.genes,g2m.genes,i)

```

### Merge Data into Different Subsets

Datasets include:

* cells separated by individual and cell type (n=42)
* cells separated by individual (n=14)
* all cells from Time 0 collections (n=1)
* all cells from Time 1 collections (n=1)
* all cells from Time 2 collections (n=1)
* all cells from all collections (n=1)

```{r}

#keep collections separated by individual and cell type (.log.indv-cell)
#objects

#merge T0, T1, and T2 collections within each individual (.log.indv)
objects[[43]] <- merge(objects[[1]], y=c(objects[[3]],objects[[5]]),add.cell.ids=c("H1.I","H1.M","H1.O"))
objects[[44]] <- merge(objects[[7]], y=c(objects[[9]],objects[[11]]),add.cell.ids=c("H1r2.I","H1r2.M","H1r2.O"))
objects[[45]] <- merge(objects[[13]], y=c(objects[[15]],objects[[17]]),add.cell.ids=c("H2.I","H2.M","H2.O"))
objects[[46]] <- merge(objects[[19]], y=c(objects[[21]],objects[[23]]),add.cell.ids=c("H3.I","H3.M","H3.O"))
objects[[47]] <- merge(objects[[25]], y=c(objects[[27]],objects[[29]]),add.cell.ids=c("H4.I","H4.M","H4.O"))
objects[[48]] <- merge(objects[[31]], y=c(objects[[33]],objects[[35]]),add.cell.ids=c("H5.I","H5.M","H5.O"))
objects[[49]] <- merge(objects[[37]], y=c(objects[[39]],objects[[41]]),add.cell.ids=c("H6.I","H6.M","H6.O"))
objects[[50]] <- merge(objects[[2]], y=c(objects[[4]],objects[[6]]),add.cell.ids=c("C1.I","C1.M","C1.O"))
objects[[51]] <- merge(objects[[8]], y=c(objects[[10]],objects[[12]]),add.cell.ids=c("C1r2.I","C1r2.M","C1r2.O"))
objects[[52]] <- merge(objects[[14]], y=c(objects[[16]],objects[[18]]),add.cell.ids=c("C2.I","C2.M","C2.O"))
objects[[53]] <- merge(objects[[20]], y=c(objects[[22]],objects[[24]]),add.cell.ids=c("C3.I","C3.M","C3.O"))
objects[[54]] <- merge(objects[[26]], y=c(objects[[28]],objects[[30]]),add.cell.ids=c("C4.I","C4.M","C4.O"))
objects[[55]] <- merge(objects[[32]], y=c(objects[[34]],objects[[36]]),add.cell.ids=c("C5.I","C5.M","C5.O"))
objects[[56]] <- merge(objects[[38]], y=c(objects[[40]],objects[[42]]),add.cell.ids=c("C6.I","C6.M","C6.O"))

#merge all T0 collections (.log.t0)
objects[[57]] <- merge(objects[[1]],
                        y=c(objects[[7]],objects[[13]],objects[[19]],objects[[25]],objects[[31]],objects[[37]],
                            objects[[2]],objects[[8]],objects[[14]],objects[[20]],objects[[26]],objects[[32]],objects[[38]]),
                        add.cell.ids=c("H1.I","H1r2.I","H2.I","H3.I","H4.I","H5.I","H6.I",
                                       "C1.I","C1r2.I","C2.I","C3.I","C4.I","C5.I","C6.I"))

#merge all T1 collections (.log.t1)
objects[[58]] <- merge(objects[[3]],
                        y=c(objects[[9]],objects[[15]],objects[[21]],objects[[27]],objects[[33]],objects[[39]],
                            objects[[4]],objects[[10]],objects[[16]],objects[[22]],objects[[28]],objects[[34]],objects[[40]]),
                        add.cell.ids=c("H1.M","H1r2.M","H2.M","H3.M","H4.M","H5.M","H6.M",
                                       "C1.M","C1r2.M","C2.M","C3.M","C4.M","C5.M","C6.M"))

#merge all T2 collections (.log.t2)
objects[[59]] <- merge(objects[[5]],
                        y=c(objects[[11]],objects[[17]],objects[[23]],objects[[29]],objects[[35]],objects[[41]],
                            objects[[6]],objects[[12]],objects[[18]],objects[[24]],objects[[30]],objects[[36]],objects[[42]]),
                        add.cell.ids=c("H1.O","H1r2.O","H2.O","H3.O","H4.O","H5.O","H6.O",
                                       "C1.O","C1r2.O","C2.O","C3.O","C4.O","C5.O","C6.O"))

#merge all collections (.log.tot)
objects[[60]] <- merge(objects[[1]],
                        y=c(objects[[7]],objects[[13]],objects[[19]],objects[[25]],objects[[31]],objects[[37]],
                            objects[[2]],objects[[8]],objects[[14]],objects[[20]],objects[[26]],objects[[32]],objects[[38]],
                            objects[[3]],objects[[9]],objects[[15]],objects[[21]],objects[[27]],objects[[33]],objects[[39]],
                            objects[[4]],objects[[10]],objects[[16]],objects[[22]],objects[[28]],objects[[34]],objects[[40]],
                            objects[[5]],objects[[11]],objects[[17]],objects[[23]],objects[[29]],objects[[35]],objects[[41]],
                            objects[[6]],objects[[12]],objects[[18]],objects[[24]],objects[[30]],objects[[36]],objects[[42]]),
                        add.cell.ids=c("H1.I","H1r2.I","H2.I","H3.I","H4.I","H5.I","H6.I",
                                       "C1.I","C1r2.I","C2.I","C3.I","C4.I","C5.I","C6.I",
                                       "H1.M","H1r2.M","H2.M","H3.M","H4.M","H5.M","H6.M",
                                       "C1.M","C1r2.M","C2.M","C3.M","C4.M","C5.M","C6.M",
                                       "H1.O","H1r2.O","H2.O","H3.O","H4.O","H5.O","H6.O",
                                       "C1.O","C1r2.O","C2.O","C3.O","C4.O","C5.O","C6.O"))

```

### Find Variable Features, Stabilize Variance, and Reduce Data Dimensionality

Which variable features are identified depends on which samples are included into the seurat object. This is why we first compile different sets of merged data before assessing variable feature in the merged dataset.

When scaling the data, no variables are regressed out at this time.

To QC each data subset, we also do dimensional reduction for each dataset

```{r}

#Identify variable feature for each individual and cell type
for (i in 1:length(objects)) {
  print(i)
  objects[[i]] <- FindVariableFeatures(objects[[i]], selection.method="vst", nfeatures=3000, verbose=FALSE)
  objects[[i]] <- ScaleData(objects[[i]], vars.to.regress=NULL, verbose=FALSE)
  objects[[i]] <- RunPCA(objects[[i]], npcs=100, verbose=FALSE)
  #keep all dims that explaim more than 0.1% of variance
  pva <- objects[[i]]@reductions$pca@stdev^2/objects[[i]]@reductions$pca@misc$total.variance
  ndim <- length(which(pva>=0.001))
  objects[[i]] <- RunUMAP(objects[[i]], dims=1:ndim)
}
rm(pva,ndim,i)

```

### Save Data Subsets

```{r}

#Save data
saveRDS(objects[1:42], file="./data/cellranger-data-full/data.filter.log.indv-cell.rds")
saveRDS(objects[43:56], file="./data/cellranger-data-full/data.filter.log.indv.rds")
saveRDS(objects[57], file="./data/cellranger-data-full/data.filter.log.merge-t0.rds")
saveRDS(objects[57], file="./data/cellranger-data-full/data.filter.log.merge-t1.rds")
saveRDS(objects[57], file="./data/cellranger-data-full/data.filter.log.merge-t2.rds")
saveRDS(objects[57], file="./data/cellranger-data-full/data.filter.log.merge-tot.rds")

```

### QC of Data Subsets

```{r cell cycle}


#plot cell cycle scores
CombinePlots(plots=list((DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(1,2)) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(3,4)) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(5,6)) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(7,8)) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(9,10)) +
                           labs(x="PC9",y="PC10"))), ncol=5)
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"),
                                 group.by="Phase",
                                 reduction="umap") +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"),
                                 group.by="Phase",
                                 reduction="umap") +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Phase=="G1"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Phase=="G2M"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Phase=="S"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=3)


```

### Data Integration

What I used:

* separate individuals and cells types
* separate individuals but keep all cell types together
* merge data without integration

...something about merging datasets using the intersect of the 6000 most variable genes in humans and chimpanzees

```{r integrate scRNA data - separate individuals and cells types}



#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/v1/data.indv-cell.log.rds"))

#Define list subset
#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
#Integrate datasets
#Reduce dimensionality
#Save data

#combine ipscs from all samples
obj.ipsc  <- list(objects[[1]],objects[[2]],objects[[7]],objects[[8]],objects[[13]],objects[[14]],objects[[19]],objects[[20]],
                  objects[[25]],objects[[26]],objects[[31]],objects[[32]],objects[[37]],objects[[38]])
obj.ipsc.features <- SelectIntegrationFeatures(object.list=obj.ipsc, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.ipsc <- lapply(X=obj.ipsc, FUN=function(x) {
    x <- ScaleData(x, features=obj.ipsc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.ipsc.features, verbose=FALSE)
})
obj.ipsc.anchors <- FindIntegrationAnchors(object.list=obj.ipsc, normalization.method="LogNormalize", anchor.features=obj.ipsc.features,
                                           reference=reference_dataset, reduction="rpca")
int.ipsc <- IntegrateData(anchorset=obj.ipsc.anchors, normalization.method="LogNormalize")
int.ipsc <- ScaleData(int.ipsc, verbose=FALSE)
int.ipsc <- RunPCA(object=int.ipsc, npcs=100, verbose=FALSE)
pva <- int.ipsc@reductions$pca@stdev^2/int.ipsc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.ipsc <- RunUMAP(int.ipsc, dims=1:100)
saveRDS(int.ipsc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.ipsc.rds"))
rm(obj.ipsc,int.ipsc)

#combine mscs from all samples
obj.msc   <- list(objects[[3]],objects[[4]],objects[[9]],objects[[10]],objects[[15]],objects[[16]],objects[[21]],objects[[22]],
                  objects[[27]],objects[[28]],objects[[33]],objects[[34]],objects[[39]],objects[[40]])
obj.msc.features <- SelectIntegrationFeatures(object.list=obj.msc, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.msc <- lapply(X=obj.msc, FUN=function(x) {
    x <- ScaleData(x, features=obj.msc.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.msc.features, verbose=FALSE)
})
obj.msc.anchors <- FindIntegrationAnchors(object.list=obj.msc, normalization.method="LogNormalize", anchor.features=obj.msc.features,
                                           reference=reference_dataset, reduction="rpca")
int.msc <- IntegrateData(anchorset=obj.msc.anchors, normalization.method="LogNormalize")
int.msc <- ScaleData(int.msc, verbose=FALSE)
int.msc <- RunPCA(object=int.msc, npcs=100, verbose=FALSE)
pva <- int.msc@reductions$pca@stdev^2/int.msc@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.msc <- RunUMAP(int.msc, dims=1:100)
saveRDS(int.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.msc.rds"))
rm(obj.msc,int.msc)

#combine osteoblasts from all samples
obj.osteo <- list(objects[[5]],objects[[6]],objects[[11]],objects[[12]],objects[[17]],objects[[18]],objects[[23]],objects[[24]],
                  objects[[29]],objects[[30]],objects[[35]],objects[[36]],objects[[41]],objects[[42]])
obj.osteo.features <- SelectIntegrationFeatures(object.list=obj.osteo, nfeatures=1000)
reference_dataset <- c(3:4) #H1C1-r2 differentiated the best
obj.osteo <- lapply(X=obj.osteo, FUN=function(x) {
    x <- ScaleData(x, features=obj.osteo.features, verbose=FALSE)
    x <- RunPCA(x, features=obj.osteo.features, verbose=FALSE)
})
obj.osteo.anchors <- FindIntegrationAnchors(object.list=obj.osteo, normalization.method="LogNormalize", anchor.features=obj.osteo.features,
                                           reference=reference_dataset, reduction="rpca")
int.osteo <- IntegrateData(anchorset=obj.osteo.anchors, normalization.method="LogNormalize")
int.osteo <- ScaleData(int.osteo, verbose=FALSE)
int.osteo <- RunPCA(object=int.osteo, npcs=100, verbose=FALSE)
pva <- int.osteo@reductions$pca@stdev^2/int.osteo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
int.osteo <- RunUMAP(int.osteo, dims=1:100)
saveRDS(int.osteo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.osteo.rds"))
rm(obj.osteo,int.osteo)

```

```{r integrate scRNA data - separate individuals but keep all cell types together}

#Load objects
objects <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.rds"))

#Select features for downstream integration (keeping at 1000 for now)
#Identify anchors (used references + RPCA reduction method) #ALL OTHER METHODS CRASH
objects.features <- SelectIntegrationFeatures(object.list=objects, nfeatures=1000)
reference_dataset <- c(2,9) #make H1-r2 and C1-r2 the references
objects <- lapply(X=objects, FUN=function(x) {
    x <- ScaleData(x, features=objects.features, verbose=FALSE)
    x <- RunPCA(x, features=objects.features, verbose=FALSE)
})
objects.anchors <- FindIntegrationAnchors(object.list=objects, normalization.method="LogNormalize", anchor.features=objects.features,
                                          reference=reference_dataset, reduction="rpca")

#Integrate datasets
integrate <- IntegrateData(anchorset=objects.anchors, normalization.method="LogNormalize")
integrate <- ScaleData(integrate, verbose=FALSE)

#Reduce dimensionality
integrate <- RunPCA(object=integrate, npcs=100, verbose=FALSE)
pva <- integrate@reductions$pca@stdev^2/integrate@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001)) #keep all dims that explaim more than 0.1% of variance
print(ndim)
integrate <- RunUMAP(integrate, dims=1:100)

#Save data
saveRDS(integrate, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.int.rds"))
rm(objects,integrate)

```

```{r examine batch effects - merged data vs. integrated data}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load batch info
batch <- read.csv(file=paste0(dir,'human-chimp-skeletal-scRNA/data/scrna-batch.csv'), header=TRUE, sep=",")

#Read and Visualize data (batch of interest: species, Pair, Individual, Cell.Type, Replicate, MSCmedia)

#combine ipscs from all samples
data.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.ipsc.rds"))             #merge
spps.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.ipsc.rds")) #integrate across species
pair.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.ipsc.rds")) #integrate across human-chimp pairs
indv.ipsc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.ipsc.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.ipsc,group.by="species"),
                        DimPlot(data.ipsc,group.by="Pair"),
                        DimPlot(data.ipsc,group.by="Individual"),
                        DimPlot(data.ipsc,group.by="Cell.Type"),
                        DimPlot(data.ipsc,group.by="Replicate"),
                        DimPlot(data.ipsc,group.by="MSCmedia"),

                        DimPlot(spps.ipsc,group.by="species"),
                        DimPlot(spps.ipsc,group.by="Pair"),
                        DimPlot(spps.ipsc,group.by="Individual"),
                        DimPlot(spps.ipsc,group.by="Cell.Type"),
                        DimPlot(spps.ipsc,group.by="Replicate"),
                        DimPlot(spps.ipsc,group.by="MSCmedia"),
                        
                        DimPlot(pair.ipsc,group.by="species"),
                        DimPlot(pair.ipsc,group.by="Pair"),
                        DimPlot(pair.ipsc,group.by="Individual"),
                        DimPlot(pair.ipsc,group.by="Cell.Type"),
                        DimPlot(pair.ipsc,group.by="Replicate"),
                        DimPlot(pair.ipsc,group.by="MSCmedia"),
                        
                        DimPlot(indv.ipsc,group.by="species"),
                        DimPlot(indv.ipsc,group.by="Pair"),
                        DimPlot(indv.ipsc,group.by="Individual"),
                        DimPlot(indv.ipsc,group.by="Cell.Type"),
                        DimPlot(indv.ipsc,group.by="Replicate"),
                        DimPlot(indv.ipsc,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.ipsc,group.by="species",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Collection",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Samples",reduction='pca'),
                        DimPlot(data.ipsc,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.ipsc,group.by="species",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.ipsc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.ipsc,group.by="species",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.ipsc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.ipsc,group.by="species",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Pair",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Individual",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.ipsc,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.ipsc,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.ipsc,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.ipsc,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.ipsc,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.ipsc, spps.ipsc, pair.ipsc, indv.ipsc)

#data.ipsc:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.ipsc:   species integrated / batches separated  (PC/UMAP separates species)
#pair.ipsc:   species separated / batches integrated  (PC/UMAP separates species)
#indv.ipsc:   species and batches integrated
#NOTES:       UMI and gene counts are higher in humans than chimps


#combine mscs from all samples
data.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.msc.rds"))             #merge
spps.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.msc.rds")) #integrate across species
pair.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.msc.rds")) #integrate across human-chimp pairs
indv.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.msc.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.msc,group.by="species"),
                        DimPlot(data.msc,group.by="Pair"),
                        DimPlot(data.msc,group.by="Individual"),
                        DimPlot(data.msc,group.by="Cell.Type"),
                        DimPlot(data.msc,group.by="Replicate"),
                        DimPlot(data.msc,group.by="MSCmedia"),

                        DimPlot(spps.msc,group.by="species"),
                        DimPlot(spps.msc,group.by="Pair"),
                        DimPlot(spps.msc,group.by="Individual"),
                        DimPlot(spps.msc,group.by="Cell.Type"),
                        DimPlot(spps.msc,group.by="Replicate"),
                        DimPlot(spps.msc,group.by="MSCmedia"),
                        
                        DimPlot(pair.msc,group.by="species"),
                        DimPlot(pair.msc,group.by="Pair"),
                        DimPlot(pair.msc,group.by="Individual"),
                        DimPlot(pair.msc,group.by="Cell.Type"),
                        DimPlot(pair.msc,group.by="Replicate"),
                        DimPlot(pair.msc,group.by="MSCmedia"),
                        
                        DimPlot(indv.msc,group.by="species"),
                        DimPlot(indv.msc,group.by="Pair"),
                        DimPlot(indv.msc,group.by="Individual"),
                        DimPlot(indv.msc,group.by="Cell.Type"),
                        DimPlot(indv.msc,group.by="Replicate"),
                        DimPlot(indv.msc,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.msc,group.by="species",reduction='pca'),
                        DimPlot(data.msc,group.by="Collection",reduction='pca'),
                        DimPlot(data.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.msc,group.by="Samples",reduction='pca'),
                        DimPlot(data.msc,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.msc,group.by="species",reduction='pca'),
                        DimPlot(spps.msc,group.by="Pair",reduction='pca'),
                        DimPlot(spps.msc,group.by="Individual",reduction='pca'),
                        DimPlot(spps.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.msc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.msc,group.by="species",reduction='pca'),
                        DimPlot(pair.msc,group.by="Pair",reduction='pca'),
                        DimPlot(pair.msc,group.by="Individual",reduction='pca'),
                        DimPlot(pair.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.msc,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.msc,group.by="species",reduction='pca'),
                        DimPlot(indv.msc,group.by="Pair",reduction='pca'),
                        DimPlot(indv.msc,group.by="Individual",reduction='pca'),
                        DimPlot(indv.msc,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.msc,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.msc,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.msc,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.msc,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.msc,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.msc,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.msc, spps.msc, pair.msc, indv.msc)

#data.msc:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.msc:   species integrated / batches separated  (PC/UMAP separates species)
#pair.msc:   species separated / batches integrated  (PC/UMAP separates species)
#indv.msc:   species and batches integrated
#NOTES:      UMI and gene counts are higher in humans than chimps


#combine osteoblasts from all samples
data.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.osteo.rds"))             #merge
spps.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps-cell.log.int.osteo.rds")) #integrate across species
pair.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair-cell.log.int.osteo.rds")) #integrate across human-chimp pairs
indv.osteo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv-cell.log.int.osteo.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data.osteo,group.by="species"),
                        DimPlot(data.osteo,group.by="Pair"),
                        DimPlot(data.osteo,group.by="Individual"),
                        DimPlot(data.osteo,group.by="Cell.Type"),
                        DimPlot(data.osteo,group.by="Replicate"),
                        DimPlot(data.osteo,group.by="MSCmedia"),

                        DimPlot(spps.osteo,group.by="species"),
                        DimPlot(spps.osteo,group.by="Pair"),
                        DimPlot(spps.osteo,group.by="Individual"),
                        DimPlot(spps.osteo,group.by="Cell.Type"),
                        DimPlot(spps.osteo,group.by="Replicate"),
                        DimPlot(spps.osteo,group.by="MSCmedia"),
                        
                        DimPlot(pair.osteo,group.by="species"),
                        DimPlot(pair.osteo,group.by="Pair"),
                        DimPlot(pair.osteo,group.by="Individual"),
                        DimPlot(pair.osteo,group.by="Cell.Type"),
                        DimPlot(pair.osteo,group.by="Replicate"),
                        DimPlot(pair.osteo,group.by="MSCmedia"),
                        
                        DimPlot(indv.osteo,group.by="species"),
                        DimPlot(indv.osteo,group.by="Pair"),
                        DimPlot(indv.osteo,group.by="Individual"),
                        DimPlot(indv.osteo,group.by="Cell.Type"),
                        DimPlot(indv.osteo,group.by="Replicate"),
                        DimPlot(indv.osteo,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data.osteo,group.by="species",reduction='pca'),
                        DimPlot(data.osteo,group.by="Collection",reduction='pca'),
                        DimPlot(data.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data.osteo,group.by="Samples",reduction='pca'),
                        DimPlot(data.osteo,group.by="Replicate",reduction='pca'),

                        DimPlot(spps.osteo,group.by="species",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(spps.osteo,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair.osteo,group.by="species",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(pair.osteo,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv.osteo,group.by="species",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Pair",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Individual",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv.osteo,group.by="Replicate",reduction='pca'),
                        DimPlot(indv.osteo,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(spps.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(pair.osteo,'nCount_RNA',group.by='species'),
                        VlnPlot(indv.osteo,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair.osteo,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv.osteo,'nFeature_RNA',group.by='species')), ncol=4)

rm(data.osteo, spps.osteo, pair.osteo, indv.osteo)

#data.osteo:   species and batches separated           (PC/UMAP separates species, PC/UMAP separates batches)
#spps.osteo:   species integrated / batches separated  (PC/UMAP separates species)
#pair.osteo:   species separated / batches integrated  (PC/UMAP separates species)
#indv.osteo:   species and batches integrated
#NOTES:        UMI and gene counts are higher in humans than chimps


#combine all cells from all samples
data <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.rds"))        #merge
spps <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.spps.log.int.rds")) #integrate across species
pair <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.pair.log.int.rds")) #integrate across human-chimp pairs
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.int.rds")) #integrate across individuals

CombinePlots(plots=list(DimPlot(data,group.by="species"),
                        DimPlot(data,group.by="Pair"),
                        DimPlot(data,group.by="Individual"),
                        DimPlot(data,group.by="Cell.Type"),
                        DimPlot(data,group.by="Replicate"),
                        DimPlot(data,group.by="MSCmedia"),

                        DimPlot(spps,group.by="species"),
                        DimPlot(spps,group.by="Pair"),
                        DimPlot(spps,group.by="Individual"),
                        DimPlot(spps,group.by="Cell.Type"),
                        DimPlot(spps,group.by="Replicate"),
                        DimPlot(spps,group.by="MSCmedia"),
                        
                        DimPlot(pair,group.by="species"),
                        DimPlot(pair,group.by="Pair"),
                        DimPlot(pair,group.by="Individual"),
                        DimPlot(pair,group.by="Cell.Type"),
                        DimPlot(pair,group.by="Replicate"),
                        DimPlot(pair,group.by="MSCmedia"),
                        
                        DimPlot(indv,group.by="species"),
                        DimPlot(indv,group.by="Pair"),
                        DimPlot(indv,group.by="Individual"),
                        DimPlot(indv,group.by="Cell.Type"),
                        DimPlot(indv,group.by="Replicate"),
                        DimPlot(indv,group.by="MSCmedia")), ncol=5)

CombinePlots(plots=list(DimPlot(data,group.by="species",reduction='pca'),
                        DimPlot(data,group.by="Collection",reduction='pca'),
                        DimPlot(data,group.by="Cell.Type",reduction='pca'),
                        DimPlot(data,group.by="Samples",reduction='pca'),
                        DimPlot(data,group.by="Replicate",reduction='pca'),

                        DimPlot(spps,group.by="species",reduction='pca'),
                        DimPlot(spps,group.by="Pair",reduction='pca'),
                        DimPlot(spps,group.by="Individual",reduction='pca'),
                        DimPlot(spps,group.by="Cell.Type",reduction='pca'),
                        DimPlot(spps,group.by="Replicate",reduction='pca'),
                        DimPlot(spps,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(pair,group.by="species",reduction='pca'),
                        DimPlot(pair,group.by="Pair",reduction='pca'),
                        DimPlot(pair,group.by="Individual",reduction='pca'),
                        DimPlot(pair,group.by="Cell.Type",reduction='pca'),
                        DimPlot(pair,group.by="Replicate",reduction='pca'),
                        DimPlot(pair,group.by="MSCmedia",reduction='pca'),
                        
                        DimPlot(indv,group.by="species",reduction='pca'),
                        DimPlot(indv,group.by="Pair",reduction='pca'),
                        DimPlot(indv,group.by="Individual",reduction='pca'),
                        DimPlot(indv,group.by="Cell.Type",reduction='pca'),
                        DimPlot(indv,group.by="Replicate",reduction='pca'),
                        DimPlot(indv,group.by="MSCmedia",reduction='pca')), ncol=6)

CombinePlots(plots=list(VlnPlot(data,'nCount_RNA',group.by='species'),
                        VlnPlot(spps,'nCount_RNA',group.by='species'),
                        VlnPlot(pair,'nCount_RNA',group.by='species'),
                        VlnPlot(indv,'nCount_RNA',group.by='species'),
                        
                        VlnPlot(data,'nFeature_RNA',group.by='species'),
                        VlnPlot(spps,'nFeature_RNA',group.by='species'),
                        VlnPlot(pair,'nFeature_RNA',group.by='species'),
                        VlnPlot(indv,'nFeature_RNA',group.by='species'),
                        VlnPlot(int,'nFeature_RNA',group.by='species')), ncol=4)

rm(data, spps, pair, indv)

#data:   species + cell type + batches separated             (PC/UMAP separates species, PC/UMAP separates cell types, PC/UMAP separates batches)
#spps:   cell type + batches separated / species integrated  (PC/UMAP separates cell types, PC/UMAP separates batches)
#pair:   species + cell type separated / batches integrated  (PC/UMAP separates species, PC/UMAP separates cell types)
#indv:   cell type separated / species + batches integrated  (PC/UMAP separates cell types)
#NOTES:  UMI and gene counts are higher in humans than chimps

```

