---
title: "10x-plot-cormotif-cnts"
author: "Genevieve Housman"
date: "July 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Correlation Motifs

View results of correlation motifs tests between humans and chimpanzees within different cell classifications.

```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ComplexHeatmap)

```

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate <- paste0(data@meta.data$Individual,".",data@meta.data$Replicate)

```

```{r}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

```{r}

#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot <- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale="free") +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

#Define function to plot GO functional enrichment results (specific ontology only)
goEnPlot.spec <- function(goTot, title, ontology) {
  print(ggplot(goTot[which(goTot$ONTOLOGY==ontology),], aes(x=GeneRatio, y=Description, color=p.adjust, size=Count)) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

```

## Load External Gene Datasets

Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.

Genes previously identified as human and chimpanzee differences:

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR < 1% out of 10304 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each cell type
    
* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR < 1% out of 12184 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each tissue type

* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + DE genes in parental iPSC lines (n = 7655)
    + DE genes in hybrid iPSC lines (n = 6009)
    + DE genes in parental CNCC lines (n = 2223)
    + DE genes in hybrid CNCC lines (n = 3612)
    + Note: only considered DE genes unique to each cell type
    
* Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
    + DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)
    + AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)
    + (human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)
    + (chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)
    + Note: only DMRs that overlapped promotor regions of genes were included
    + Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels

General skeleton-related and skeletal phenotype-related genes:

* Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
    + hip geometry GWAS loci (n = 10)

* Styrkarsdottir et al. 2019 Nature Communications (Table 1)
    + DXA bone area GWAS loci (n = 21 associated with 20 genes)

* Yengo et al. 2018 Human Molecular Genetics
    + height GWAS loci (n = 3290) (data provided at https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
    + height GWAS loci with associated eQTLs (n = 610) (list not provided)
    + Note: only used unique gene symbols associated with height GWAS loci (n = 1500)

Genes associated with skeletal diseases (OA + OP):

* Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
    + genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)
    
* Morris et al. 2019 Nature Genetics (Supplemental Table 2)
    + OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)
    
Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:

* Gallego Romero et al. 2015 eLife (Supplementary File 3)
    + DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR < 1% out of 12171 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + REMOVED because subsequent studies (Blake et al. 2018 & Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm & CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type
    
* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
    + Skeleton-related genes (n = 1790)
    + Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)
    + REMOVED because it is unclear where these genes came from to interpreting biology is difficult

Some external gene datasets were considered but not examined:

* Ward et al. 2019 eLife
    +  DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR < 10% out of 11974 genes)

* Shibata et al. 2012 PLOS Genetics
    + increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)

* Edsall et al. 2019 Genome Biology and Evolution
    + increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)

* Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
    + increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)

* Prescott et al. 2015 Cell (Supplemental)
    + top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)
    + top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)

* Joganic et al. 2017 American Journal of Physical Anthropology
    + genetics of craniofacial skeleton in baboons

* Cibrián Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
    + robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)

* Tachmazidou et al. 2019 Nature Genetics
    + OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])
    + OA loci (Supplementary Table 2)
    + established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)
    + Note: used loci compiled by AH
    + (OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)
    + (OA.estab = gene symbol(s) nearest to established OA loci)
    + (OA.broad = genes within 500kb of OA loci)

```{r}

#Define function for testing significant enrichment of gene sets

genesetEnrich <- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative="greater"))

}

genesetEnrich.data <- function(InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test and output data
  data <- data.frame("Source.DEGenes"=InternalDataSource,
                     "Source.GenesOfInterest"=ExternalDataSource,
                     "DE.Interest"=DE.Interest,
                     "NonDE.Interest"=NonDE.Interest,
                     "DE.NotInterest"=DE.NotInterest,
                     "NonDE.NotInterest"=NonDE.NotInterest,
                     "p.value"=fisher.test(ContingencyTable, alternative="greater")$p.value,
                     "odds.ratio"=fisher.test(ContingencyTable, alternative="greater")$estimate,
                     row.names=NULL)
  return(data)

}

```

```{r}

#Load external datasets (manually currated from papers)
gallegoromero2015    <- read.csv("./data/external-gene-sets/GallegoRomero2015eLife.csv")
blake2018            <- read.csv("./data/external-gene-sets/Blake2018GenBio.csv")
blake2020            <- read.csv("./data/external-gene-sets/Blake2020GenRes.csv")
gokhman2021          <- read.csv("./data/external-gene-sets/Gokhman2021NatGen.csv")
gokhman2020          <- read.csv("./data/external-gene-sets/Gokhman2020NatComm.csv")
gokhman2020.skeleton <- read.csv("./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv")
hsu2019              <- read.csv("./data/external-gene-sets/Hsu2019JBoneMinRes.csv")
styrkarsdottir2019   <- read.csv("./data/external-gene-sets/Styrkarsdottir2019NatComm.csv")
yengo2018            <- read.csv("./data/external-gene-sets/Yengo2018HumMolGen.csv")
steinberg2020        <- read.csv("./data/external-gene-sets/Steinberg2020biorxiv.csv")
morris2019           <- read.csv("./data/external-gene-sets/Morris2019NatGen.csv")
#cibrianuhalte2017    <- read.csv("./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv")
#tachmazidou2019      <- read.csv("./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv")

```

```{r, eval=FALSE}

#genes previously identified as human and chimpanzee differences

#genesExternal <- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal <- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal <- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal <- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1 &
                                      blake2018$DE.Day1==0 &
                            					blake2018$DE.Day2==0 &
                            					blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==1 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==1 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==1 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==1 &
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                               gokhman2021$DE.Hybrid.CNCCs==0)]), #hybrid.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal <- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal <- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal <- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal <- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal <- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal <- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS

```

## Stages of Differentiation at Collection

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Stage=c("Time 0","Time 1","Time 2"),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("Time.0",length(which(dataUpset$Time.0==1))),
                                       rep("Time.1",length(which(dataUpset$Time.1==1))),
                                       rep("Time.2",length(which(dataUpset$Time.2==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$Time.0==1)],
                                rownames(dataUpset)[which(dataUpset$Time.1==1)],
                                rownames(dataUpset)[which(dataUpset$Time.2==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$Time.0==1),"Time.0"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.1==1),"Time.1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.2==1),"Time.2"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("Time.0","Time.1","Time.2"),
                               labels=c("Time 0","Time 1","Time 2"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0"),
                         list("Time.1"),
                         list("Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.1","Time.2"),
                         list("Time.0","Time.2"),
                         list("Time.0","Time.1","Time.2")),
      text.scale=2)

#Save DE gene lists
write.csv(rownames(dataUpset),"./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t01.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t12.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==0 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t0.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t1.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==0 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t2.csv")

```

Assess GO functional enrichment of interesting DE genes.

```{r}

#Shared across t0+t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across All Stages")

#Shared across t0+t1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 0 & Time 1")

#Shared across t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 1 & Time 2")

#Unique to Time 0
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 0")

#Unique to Time 1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 1")

#Unique to Time 2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 2")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

* Enrichment of GallegoRomero2015-DE.iPSCs:
    + Shared across All Stages      (post.p>0.5): p = 3.4e-12 (1618 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 631 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.34110 (  77 genes intersect) / (post.p>0.95): p = 4.4e-06 ( 127 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.95840 (  10 genes intersect) / (post.p>0.95): p = 1.00000 ( 161 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 1.5e-15 (1107 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 991 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.99720 (  14 genes intersect) / (post.p>0.95): p = 0.99930 (  42 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.22150 (   2 genes intersect)

* Enrichment of Blake2018-DE.iPSCs:
    + Shared across All Stages      (post.p>0.5): p = 2.4e-11 (1336 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 506 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.00819 (  77 genes intersect) / (post.p>0.95): p = 7.8e-08 ( 115 genes intersect) **
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.32350 (  14 genes intersect) / (post.p>0.95): p = 1.00000 ( 126 genes intersect) 
    + Unique to Time 0              (post.p>0.5): p = 4.6e-06 ( 854 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 805 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.99810 (  10 genes intersect) / (post.p>0.95): p = 0.97540 (  39 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.57590 (   1 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1:
    + Shared across All Stages      (post.p>0.5): p = 2.4e-06 (1293 genes intersect) / (post.p>0.95): p = 2.4e-13 ( 481 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.08545 (  70 genes intersect) / (post.p>0.95): p = 3.7e-05 ( 105 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.32710 (  14 genes intersect) / (post.p>0.95): p = 0.99980 ( 143 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.03682 ( 801 genes intersect) / (post.p>0.95): p = 1.5e-13 ( 682 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.99530 (  11 genes intersect) / (post.p>0.95): p = 0.55010 (  50 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.57690 (   1 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2:
    + Shared across All Stages      (post.p>0.5): p = 3.3e-07 (1375 genes intersect) / (post.p>0.95): p = 1.0e-08 ( 478 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.51980 (  64 genes intersect) / (post.p>0.95): p = 0.00065 ( 104 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.41840 (  14 genes intersect) / (post.p>0.95): p = 0.98510 ( 168 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.36300 ( 817 genes intersect) / (post.p>0.95): p = 8.2e-06 ( 660 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 0.99510 (  12 genes intersect) / (post.p>0.95): p = 0.30280 (  57 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.60150 (   1 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3:
    + Shared across All Stages      (post.p>0.5): p = 3.3e-05 (1476 genes intersect) / (post.p>0.95): p = 3.2e-05 ( 491 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.36120 (  73 genes intersect) / (post.p>0.95): p = 0.01014 ( 105 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.57770 (  14 genes intersect) / (post.p>0.95): p = 0.99010 ( 184 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.82060 ( 868 genes intersect) / (post.p>0.95): p = 0.00025 ( 702 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 0.92240 (  18 genes intersect) / (post.p>0.95): p = 0.26330 (  63 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.20300 (   2 genes intersect)

* Enrichment of Blake2018-DE.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.30100 ( 163 genes intersect) / (post.p>0.95): p = 0.01686 (  63 genes intersect)  *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.40810 (   9 genes intersect) / (post.p>0.95): p = 0.28310 (  12 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.07801 (   4 genes intersect) / (post.p>0.95): p = 0.95360 (  17 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.00128 ( 128 genes intersect) / (post.p>0.95): p = 2.6e-06 ( 109 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.49630 (   3 genes intersect) / (post.p>0.95): p = 0.90580 (   4 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.65090 ( 104 genes intersect) / (post.p>0.95): p = 0.84250 (  28 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.63370 (   5 genes intersect) / (post.p>0.95): p = 0.96510 (   3 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.67070 (   1 genes intersect) / (post.p>0.95): p = 0.27500 (  19 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.99590 (  51 genes intersect) / (post.p>0.95): p = 0.99710 (  33 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.53750 (   2 genes intersect) / (post.p>0.95): p = 0.00575 (  11 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.99530 (  97 genes intersect) / (post.p>0.95): p = 0.99750 (  22 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.55430 (   6 genes intersect) / (post.p>0.95): p = 0.93980 (   4 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.34220 (   2 genes intersect) / (post.p>0.95): p = 0.53370 (  18 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.97470 (  62 genes intersect) / (post.p>0.95): p = 0.99750 (  37 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.86570 (   1 genes intersect) / (post.p>0.95): p = 0.12780 (   8 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.99890 ( 275 genes intersect) / (post.p>0.95): p = 0.99980 (  67 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.88230 (  12 genes intersect) / (post.p>0.95): p = 0.99550 (  10 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.40660 (   4 genes intersect) / (post.p>0.95): p = 0.34580 (  51 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.98280 ( 178 genes intersect) / (post.p>0.95): p = 0.99970 ( 110 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.00595 (  12 genes intersect) / (post.p>0.95): p = 0.25930 (  16 genes intersect) *
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.18520 (   0 genes intersect)

* Enrichment of Blake2020-DE.heart:
    + Shared across All Stages      (post.p>0.5): p = 0.31540 ( 711 genes intersect) / (post.p>0.95): p = 4.0e-07 ( 281 genes intersect)  *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.03854 (  46 genes intersect) / (post.p>0.95): p = 0.38770 (  46 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.74760 (   6 genes intersect) / (post.p>0.95): p = 0.82600 (  99 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.79150 ( 438 genes intersect) / (post.p>0.95): p = 0.68190 ( 316 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.96080 (   7 genes intersect) / (post.p>0.95): p = 0.43110 (  31 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.05864 (   2 genes intersect)
* Enrichment of Blake2020-DE.kidney:
    + Shared across All Stages      (post.p>0.5): p = 1.4e-07 ( 286 genes intersect) / (post.p>0.95): p = 2.5e-15 ( 136 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.89640 (   8 genes intersect) / (post.p>0.95): p = 0.34810 (  16 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.90920 (   1 genes intersect) / (post.p>0.95): p = 0.08367 (  43 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.98020 ( 126 genes intersect) / (post.p>0.95): p = 0.66690 ( 101 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.98020 (   1 genes intersect) / (post.p>0.95): p = 0.75330 (   8 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.13630 (   1 genes intersect)
* Enrichment of Blake2020-DE.liver:
    + Shared across All Stages      (post.p>0.5): p = 9.5e-05 ( 469 genes intersect) / (post.p>0.95): p = 1.1e-11 ( 199 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.62180 (  20 genes intersect) / (post.p>0.95): p = 0.51520 (  26 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.93380 (   2 genes intersect) / (post.p>0.95): p = 0.02722 (  78 genes intersect)  *
    + Unique to Time 0              (post.p>0.5): p = 0.66960 ( 260 genes intersect) / (post.p>0.95): p = 0.31310 ( 196 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.92450 (   4 genes intersect) / (post.p>0.95): p = 0.84400 (  14 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.02119 (   2 genes intersect)  *
* Enrichment of Blake2020-DE.lung:
    + Shared across All Stages      (post.p>0.5): p = 8.4e-08 ( 295 genes intersect) / (post.p>0.95): p = 8.4e-11 ( 126 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.84710 (   9 genes intersect) / (post.p>0.95): p = 0.21430 (  18 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.43520 (   3 genes intersect) / (post.p>0.95): p = 0.11840 (  43 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 1.00000 ( 111 genes intersect) / (post.p>0.95): p = 0.92750 (  95 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.76000 (   3 genes intersect) / (post.p>0.95): p = 0.07274 (  15 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Blake2020-DE.heart-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.97670 ( 445 genes intersect) / (post.p>0.95): p = 0.60980 ( 144 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.01604 (  35 genes intersect) / (post.p>0.95): p = 0.17660 (  35 genes intersect) *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.73520 (   4 genes intersect) / (post.p>0.95): p = 0.88230 (  64 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.54470 ( 305 genes intersect) / (post.p>0.95): p = 0.27320 ( 227 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.82340 (   6 genes intersect) / (post.p>0.95): p = 0.55250 (  20 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.kidney-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.47330 (  63 genes intersect) / (post.p>0.95): p = 0.20060 (  23 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.61500 (   3 genes intersect) / (post.p>0.95): p = 0.90420 (   2 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.61460 (   9 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.05849 (  49 genes intersect) / (post.p>0.95): p = 0.00267 (  43 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.liver-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.09113 ( 204 genes intersect) / (post.p>0.95): p = 0.27570 (  63 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.63070 (   9 genes intersect) / (post.p>0.95): p = 0.75420 (  10 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.00961 (  42 genes intersect)  *
    + Unique to Time 0              (post.p>0.5): p = 0.43230 ( 124 genes intersect) / (post.p>0.95): p = 0.01970 ( 105 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 0.83360 (   2 genes intersect) / (post.p>0.95): p = 0.90920 (   5 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.lung-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.19590 (  84 genes intersect) / (post.p>0.95): p = 0.52860 (  24 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.98210 (   1 genes intersect) / (post.p>0.95): p = 0.21470 (   7 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.19040 (   2 genes intersect) / (post.p>0.95): p = 0.52910 (  12 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.77170 (  46 genes intersect) / (post.p>0.95): p = 0.51210 (  36 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.14230 (   3 genes intersect) / (post.p>0.95): p = 0.23470 (   5 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs:
    + Shared across All Stages      (post.p>0.5): p = 4.7e-13 (2526 genes intersect) / (post.p>0.95): p = 2.4e-16 ( 865 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.08004 ( 130 genes intersect) / (post.p>0.95): p = 1.6e-07 ( 190 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.84410 (  21 genes intersect) / (post.p>0.95): p = 0.99990 ( 307 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 1.1e-10 (1654 genes intersect) / (post.p>0.95): p < 2.2e-16 (1403 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.79900 (  36 genes intersect) / (post.p>0.95): p = 0.99080 (  83 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.11790 (   3 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs:
    + Shared across All Stages      (post.p>0.5): p = 7.4e-12 (2031 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 727 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.87930 (  86 genes intersect) / (post.p>0.95): p = 2.1e-05 ( 151 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.23810 (  22 genes intersect) / (post.p>0.95): p = 0.99130 ( 254 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 4.2e-12 (1353 genes intersect) / (post.p>0.95): p < 2.2e-16 (1124 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.96300 (  24 genes intersect) / (post.p>0.95): p = 0.99920 (  58 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.05911 (   3 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs:
    + Shared across All Stages      (post.p>0.5): p < 2.2e-16 ( 849 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 349 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.98310 (  24 genes intersect) / (post.p>0.95): p = 0.02445 (  55 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.55930 (   7 genes intersect) / (post.p>0.95): p = 0.50590 ( 104 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.15450 ( 454 genes intersect) / (post.p>0.95): p = 2.5e-08 ( 395 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 0.99960 (   3 genes intersect) / (post.p>0.95): p = 0.59440 (  28 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.36950 (   1 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs:
    + Shared across All Stages      (post.p>0.5): p = 2.7e-10 (1337 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 506 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.91630 (  52 genes intersect) / (post.p>0.95): p = 0.04277 (  88 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.71490 (  11 genes intersect) / (post.p>0.95): p = 0.63240 ( 179 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.22030 ( 784 genes intersect) / (post.p>0.95): p = 0.00126 ( 605 genes intersect)  *
    + Unique to Time 1              (post.p>0.5): p = 0.95940 (  14 genes intersect) / (post.p>0.95): p = 0.84160 (  45 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.57910 (   1 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.06415 (1878 genes intersect) / (post.p>0.95): p = 0.33450 ( 574 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.00822 ( 112 genes intersect) / (post.p>0.95): p = 0.00022 ( 145 genes intersect) **
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.65690 (  18 genes intersect) / (post.p>0.95): p = 0.99750 ( 246 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 2.2e-07 (1300 genes intersect) / (post.p>0.95): p < 2.2e-16 (1064 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.27050 (  34 genes intersect) / (post.p>0.95): p = 0.97610 (  65 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.33180 (   2 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.07441 (1130 genes intersect) / (post.p>0.95): p = 0.61320 ( 334 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.68780 (  53 genes intersect) / (post.p>0.95): p = 0.02852 (  83 genes intersect)  *
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.22190 (  14 genes intersect) / (post.p>0.95): p = 0.82250 ( 158 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 2.7e-08 ( 816 genes intersect) / (post.p>0.95): p < 2.2e-16 ( 688 genes intersect) **
    + Unique to Time 1              (post.p>0.5): p = 0.58870 (  18 genes intersect) / (post.p>0.95): p = 0.98820 (  34 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.13470 (   2 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.02505 ( 201 genes intersect) / (post.p>0.95): p = 0.38570 (  58 genes intersect) *
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.90140 (   6 genes intersect) / (post.p>0.95): p = 0.69860 (  10 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.11400 (   4 genes intersect) / (post.p>0.95): p = 0.00245 (  43 genes intersect)  *
    + Unique to Time 0              (post.p>0.5): p = 0.96850 ( 100 genes intersect) / (post.p>0.95): p = 0.99980 (  56 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.95510 (   1 genes intersect) / (post.p>0.95): p = 0.23750 (  10 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs-UNIQUE:
    + Shared across All Stages      (post.p>0.5): p = 0.48330 ( 436 genes intersect) / (post.p>0.95): p = 0.98270 ( 113 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.78820 (  19 genes intersect) / (post.p>0.95): p = 0.94690 (  20 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.83720 (   3 genes intersect) / (post.p>0.95): p = 0.01865 (  83 genes intersect)  *
    + Unique to Time 0              (post.p>0.5): p = 0.99290 ( 247 genes intersect) / (post.p>0.95): p = 0.99580 ( 169 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.44850 (   8 genes intersect) / (post.p>0.95): p = 0.29750 (  21 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2020-DMR.bone:
    + Shared across All Stages      (post.p>0.5): p = 0.03807 ( 121 genes intersect) / (post.p>0.95): p = 0.01053 (  46 genes intersect) **
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.99600 (   1 genes intersect) / (post.p>0.95): p = 0.66020 (   6 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.09632 (   3 genes intersect) / (post.p>0.95): p = 0.66740 (  15 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.84200 (  62 genes intersect) / (post.p>0.95): p = 0.18810 (  55 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.26570 (   3 genes intersect) / (post.p>0.95): p = 0.66590 (   4 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2020-skeletal.genes:
    + Shared across All Stages      (post.p>0.5): p = 0.42210 ( 518 genes intersect) / (post.p>0.95): p = 0.87730 ( 146 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.17860 (  31 genes intersect) / (post.p>0.95): p = 0.11730 (  39 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.62140 (   5 genes intersect) / (post.p>0.95): p = 0.44980 (  80 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.42490 ( 334 genes intersect) / (post.p>0.95): p = 0.07436 ( 256 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.15020 (  12 genes intersect) / (post.p>0.95): p = 0.95860 (  15 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.29020 (   1 genes intersect)
* Enrichment of Hsu2019-GWAS.hip.geometry:
    + Shared across All Stages      (post.p>0.5): p = 0.85590 (   2 genes intersect) / (post.p>0.95): p = 0.61270 (   1 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.13750 (   1 genes intersect) / (post.p>0.95): p = 0.16690 (   1 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Styrkarsdottir2019-GWAS.bone.area:
    + Shared across All Stages      (post.p>0.5): p = 0.43260 (   4 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.18830 (   1 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.35820 (   3 genes intersect) / (post.p>0.95): p = 0.04686 (   4 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Yengo2018-GWAS.height:
    + Shared across All Stages      (post.p>0.5): p = 0.77330 ( 452 genes intersect) / (post.p>0.95): p = 0.99940 ( 110 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.32490 (  26 genes intersect) / (post.p>0.95): p = 0.10070 (  36 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.19480 (   7 genes intersect) / (post.p>0.95): p = 0.97050 (  57 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.07274 ( 319 genes intersect) / (post.p>0.95): p = 0.54720 ( 212 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.89980 (   5 genes intersect) / (post.p>0.95): p = 0.03463 (  28 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Steinberg2020-GWAS.OA:
    + Shared across All Stages      (post.p>0.5): p = 0.69870 ( 143 genes intersect) / (post.p>0.95): p = 0.89990 (  38 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.76380 (   6 genes intersect) / (post.p>0.95): p = 0.81980 (   7 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.78610 (   1 genes intersect) / (post.p>0.95): p = 0.66250 (  21 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.77480 (  89 genes intersect) / (post.p>0.95): p = 0.11600 (  77 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.10140 (   5 genes intersect) / (post.p>0.95): p = 0.28770 (   8 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Morris2020-GWAS.OP:
    + Shared across All Stages      (post.p>0.5): p = 0.40830 ( 212 genes intersect) / (post.p>0.95): p = 0.85880 (  57 genes intersect)
    + Shared across Time 0 & Time 1 (post.p>0.5): p = 0.91150 (   7 genes intersect) / (post.p>0.95): p = 0.43530 (  14 genes intersect)
    + Shared across Time 1 & Time 2 (post.p>0.5): p = 0.64010 (   2 genes intersect) / (post.p>0.95): p = 0.44850 (  33 genes intersect)
    + Unique to Time 0              (post.p>0.5): p = 0.65450 ( 131 genes intersect) / (post.p>0.95): p = 0.19560 ( 104 genes intersect)
    + Unique to Time 1              (post.p>0.5): p = 0.87190 (   2 genes intersect) / (post.p>0.95): p = 0.78890 (   7 genes intersect)
    + Unique to Time 2              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

```{r, eval=FALSE}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #Shared across All Stages
  print("Shared across All Stages")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 0 & Time 1
  print("Shared across Time 0 & Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 1 & Time 2
  print("Shared across Time 1 & Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Unique to Time 0
  print("Unique to Time 0")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 1
  print("Unique to Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 2
  print("Unique to Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 0 & Time 1
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 0 & Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 1 & Time 2
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 1 & Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to Time 0
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 0",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 1
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 2
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared in All Stages (tested in all classifications)
  print("Shared in All Stages, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Shared in Time 0 & Time 1 (tested in all classifications)
  print("Shared in Time 0 & Time 1 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared in Time 1 & Time 2 (tested in all classifications)
  print("Shared in Time 1 & Time 2, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 0 (tested in all classifications)
  print("Unique to Time 0 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 1 (tested in all classifications)
  print("Unique to Time 1 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to Time 2 (tested in all classifications)
  print("Unique to Time 2 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

}

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("Time 0","Time 1","Time 0","Time 1"),
                                       c("Time 1","Time 2","Time 1","Time 2")),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9655 genes) -> 83.38%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/11579

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

#DE genes
plotMotif(cormotifData)

```

## General Clusters (res=0.05)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Cluster=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "iPSC.c2"="pink4",
                                              "iPSC.c3"="grey",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        column_split=c(1,1,1,2,2,2),
        #column_split=c(1,1,2,3,3,4),
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))
```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC.c1",length(which(dataUpset$iPSC.c1==1))),
                                       rep("MSC.c1",length(which(dataUpset$MSC.c1==1))),
                                       rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),"iPSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),"MSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                               labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Cluster",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1"),
                         list("MSC.c1"),
                         list("Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1"),
                         list("MSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1","Osteogenic.c1")),
      text.scale=2)

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC.c1",length(which(dataUpset$iPSC.c1==1))),
                                       rep("iPSC.c2",length(which(dataUpset$iPSC.c2==1))),
                                       rep("iPSC.c3",length(which(dataUpset$iPSC.c3==1))),
                                       rep("MSC.c1",length(which(dataUpset$MSC.c1==1))),
                                       rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1))),
                                       rep("Osteogenic.c2",length(which(dataUpset$Osteogenic.c2==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c2==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c3==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),"iPSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c2==1),"iPSC.c2"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c3==1),"iPSC.c3"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),"MSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),"Osteogenic.c2"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                               labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Cluster",
                    values=c("pink","pink4","grey","mediumorchid","dodgerblue","navyblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSC.c1 & MSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in iPSC.c1 & MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSC.c1 & Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in MSC.c1 & Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==0 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to iPSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==0 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","MSC.c1"),
                                         c("MSC.c1","Osteogenic.c1","MSC.c1","Osteogenic.c1")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Clusters",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c1: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9989 genes) -> 92.99%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/10742

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

```

## General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("iPSC","MSC","Osteogenic"),
                           col=list(AdHoc=c("iPSC"="pink","MSC"="mediumorchid","Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC",length(which(dataUpset$iPSC==1))),
                                       rep("MSC",length(which(dataUpset$MSC==1))),
                                       rep("Osteogenic",length(which(dataUpset$Osteogenic==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC==1)],
                                rownames(dataUpset)[which(dataUpset$MSC==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC==1),"iPSC"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC==1),"MSC"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic==1),"Osteogenic"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC","MSC","Osteogenic"),
                               labels=c("iPSC","MSC","Osteogenic"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC"),
                         list("MSC"),
                         list("Osteogenic"),
                         list("iPSC","MSC"),
                         list("MSC","Osteogenic"),
                         list("iPSC","Osteogenic"),
                         list("iPSC","MSC","Osteogenic")),
      text.scale=2)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Ad Hoc
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSCs & MSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in iPSCs & MSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSCs & Osteogenic Cells
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in MSCs & Osteogenic",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==0 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to iPSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to MSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic Cells
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==0 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteogenic",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("iPSC","MSC","iPSC","MSC"),
                                       c("MSC","Osteogenic","MSC","Osteogenic")),
                           col=list(Stage=c("iPSC"="pink",
                                            "MSC"="mediumorchid",
                                            "Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for ad hoc assignments: posterior probability difference between the humans and the chimps within a motif 
#0.3 (2787 genes) -> 23.57%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/11824

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

```

## Osteogenic Clusters (res=0.50)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1))),
                                       rep("Osteogenic.c2",length(which(dataUpset$Osteogenic.c2==1))),
                                       rep("Osteogenic.c3",length(which(dataUpset$Osteogenic.c3==1))),
                                       rep("Osteogenic.c4",length(which(dataUpset$Osteogenic.c4==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),"Osteogenic.c2"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c3==1),"Osteogenic.c3"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c4==1),"Osteogenic.c4"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                               labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("dodgerblue","deepskyblue2","dodgerblue4","navyblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Osteogenic.c1"),
                         list("Osteogenic.c3"),
                         list("Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c3"),
                         list("Osteogenic.c3","Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c3","Osteogenic.c4")),
      text.scale=2)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostclusters
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &
                                      dataUpset$Osteogenic.c2==1 &
                                      dataUpset$Osteogenic.c3==1 &
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across O.c1, O.c3, O.c4
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &
                                      dataUpset$Osteogenic.c2==0 &
                                      dataUpset$Osteogenic.c3==1 &
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Osteogenic.c1/c3/c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))


  #All Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c2
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c3
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c3",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c4
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3",
                                           "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                                         c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                           "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7873 genes) -> 73.78%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/10671

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5),]))
#DE genes between O.c1 and O.c2 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c2 and O.c3 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c3 and O.c4 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c1 and O.c2 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c2 and O.c3 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c3 and O.c4 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))

```

## Osteogenic Clusters (res=0.75)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes

## Osteogenic Clusters (res=0.50) (Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes

## Osteogenic Clusters (res=0.75) (Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes

## Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

Note: Too few DE genes shared across pre+obl+emb+min and obl+emb+min, so these were not examined further. Instead, DE genes shared across obl+emb+min+mat and emb+min+mat were considered.

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("preosteoblast",length(which(dataUpset$preosteoblast==1))),
                                       rep("osteoblast",length(which(dataUpset$osteoblast==1))),
                                       rep("embedding.osteoblast",length(which(dataUpset$embedding.osteoblast==1))),
                                       rep("mineralizing.osteoblast",length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep("maturing.osteocyte",length(which(dataUpset$maturing.osteocyte==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),"preosteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),"osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),"embedding.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),"mineralizing.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),"maturing.osteocyte"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                               labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)

#Save DE gene lists
write.csv(rownames(dataUpset),"./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oemm.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oem.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv")
write.csv(rownames(dataUpset)[which(dataUpset$osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv")
write.csv(rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv")
write.csv(rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv")
write.csv(rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv")

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Osteogenic Ad Hoc Classifications
subGenes <- rownames(dataUpset)[which(rowSums(dataUpset)==5)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")

#Shared across obl+emb+min+mat
subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in obl+emb+min+mat")

#Shared across obl+emb+min
subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in obl+emb+min")

#Unique to Embedding Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Emb")

#Unique to Mineralizing Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Min")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

* Enrichment of GallegoRomero2015-DE.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 5.4e-05 (1163 genes intersect) / (post.p>0.95): p = 1.2e-05 ( 891 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 5.4e-05 (1163 genes intersect) / (post.p>0.95): p = 6.1e-06 ( 895 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 5.7e-05 (1163 genes intersect) / (post.p>0.95): p = 6.1e-06 ( 894 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 5.3e-05 (1164 genes intersect) / (post.p>0.95): p = 5.1e-06 ( 898 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 5.4e-05 (1163 genes intersect) / (post.p>0.95): p = 7.6e-06 ( 893 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 5.1e-05 (1163 genes intersect) / (post.p>0.95): p = 7.8e-06 ( 891 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.09158 (   2 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Blake2018-DE.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 0.00572 ( 958 genes intersect) / (post.p>0.95): p = 0.01866 ( 716 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.00572 ( 958 genes intersect) / (post.p>0.95): p = 0.01815 ( 717 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00572 ( 958 genes intersect) / (post.p>0.95): p = 0.01645 ( 717 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00555 ( 959 genes intersect) / (post.p>0.95): p = 0.01774 ( 719 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.00572 ( 958 genes intersect) / (post.p>0.95): p = 0.01645 ( 717 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00552 ( 958 genes intersect) / (post.p>0.95): p = 0.01532 ( 716 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.44390 (   1 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1:
    + All preosteoblasts                         (post.p>0.5): p = 0.04189 ( 945 genes intersect) / (post.p>0.95): p = 0.09426 ( 704 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 0.04189 ( 945 genes intersect) / (post.p>0.95): p = 0.09236 ( 705 genes intersect) *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.04299 ( 945 genes intersect) / (post.p>0.95): p = 0.08597 ( 705 genes intersect) *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.04095 ( 946 genes intersect) / (post.p>0.95): p = 0.09939 ( 706 genes intersect) *
    + All maturing osteocytes                    (post.p>0.5): p = 0.04189 ( 945 genes intersect) / (post.p>0.95): p = 0.09426 ( 704 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.04082 ( 945 genes intersect) / (post.p>0.95): p = 0.08961 ( 703 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.44560 (   1 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2:
    + All preosteoblasts                         (post.p>0.5): p = 0.04879 ( 991 genes intersect) / (post.p>0.95): p = 0.11710 ( 737 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 0.04879 ( 991 genes intersect) / (post.p>0.95): p = 0.11530 ( 738 genes intersect) *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.05007 ( 991 genes intersect) / (post.p>0.95): p = 0.10750 ( 738 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.04792 ( 992 genes intersect) / (post.p>0.95): p = 0.10490 ( 741 genes intersect) *
    + All maturing osteocytes                    (post.p>0.5): p = 0.04879 ( 991 genes intersect) / (post.p>0.95): p = 0.10750 ( 738 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.04754 ( 991 genes intersect) / (post.p>0.95): p = 0.10180 ( 737 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.46510 (   1 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3:
    + All preosteoblasts                         (post.p>0.5): p = 0.06938 (1077 genes intersect) / (post.p>0.95): p = 0.04688 ( 814 genes intersect)  *
    + All osteoblasts                            (post.p>0.5): p = 0.06938 (1077 genes intersect) / (post.p>0.95): p = 0.04182 ( 816 genes intersect)  *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.07117 (1077 genes intersect) / (post.p>0.95): p = 0.04229 ( 815 genes intersect)  *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.07486 (1077 genes intersect) / (post.p>0.95): p = 0.03059 ( 821 genes intersect)  *
    + All maturing osteocytes                    (post.p>0.5): p = 0.06938 (1077 genes intersect) / (post.p>0.95): p = 0.04229 ( 815 genes intersect)  *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.06762 (1077 genes intersect) / (post.p>0.95): p = 0.03892 ( 814 genes intersect)  *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.50110 (   1 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Blake2018-DE.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.29830 ( 123 genes intersect) / (post.p>0.95): p = 0.33070 (  92 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.29830 ( 123 genes intersect) / (post.p>0.95): p = 0.33520 (  92 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.29970 ( 123 genes intersect) / (post.p>0.95): p = 0.33070 (  92 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.30230 ( 123 genes intersect) / (post.p>0.95): p = 0.34590 (  92 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.29830 ( 123 genes intersect) / (post.p>0.95): p = 0.33070 (  92 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.29700 ( 123 genes intersect) / (post.p>0.95): p = 0.32170 (  92 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day1-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.57570 (  81 genes intersect) / (post.p>0.95): p = 0.47710 (  62 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.57570 (  81 genes intersect) / (post.p>0.95): p = 0.48120 (  62 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.57690 (  81 genes intersect) / (post.p>0.95): p = 0.47710 (  62 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.57940 (  81 genes intersect) / (post.p>0.95): p = 0.49080 (  62 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.57570 (  81 genes intersect) / (post.p>0.95): p = 0.53660 (  61 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.57440 (  81 genes intersect) / (post.p>0.95): p = 0.52830 (  61 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day2-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.96120 (  78 genes intersect) / (post.p>0.95): p = 0.93210 (  58 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.96120 (  78 genes intersect) / (post.p>0.95): p = 0.93350 (  58 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.96140 (  78 genes intersect) / (post.p>0.95): p = 0.93210 (  58 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.96200 (  78 genes intersect) / (post.p>0.95): p = 0.93660 (  58 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.96120 (  78 genes intersect) / (post.p>0.95): p = 0.93210 (  58 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.96090 (  78 genes intersect) / (post.p>0.95): p = 0.92920 (  58 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2018-DE.iPSC-endoderm-day3-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.87110 ( 222 genes intersect) / (post.p>0.95): p = 0.36130 ( 180 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.87110 ( 222 genes intersect) / (post.p>0.95): p = 0.33480 ( 181 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.87220 ( 222 genes intersect) / (post.p>0.95): p = 0.36130 ( 180 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.87450 ( 222 genes intersect) / (post.p>0.95): p = 0.31770 ( 182 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.87110 ( 222 genes intersect) / (post.p>0.95): p = 0.36130 ( 180 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.86990 ( 222 genes intersect) / (post.p>0.95): p = 0.34790 ( 180 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Blake2020-DE.heart:
    + All preosteoblasts                         (post.p>0.5): p = 0.34050 ( 555 genes intersect) / (post.p>0.95): p = 0.12210 ( 428 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.34050 ( 555 genes intersect) / (post.p>0.95): p = 0.12800 ( 428 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.32310 ( 556 genes intersect) / (post.p>0.95): p = 0.12210 ( 428 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.32940 ( 556 genes intersect) / (post.p>0.95): p = 0.14260 ( 428 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.34050 ( 555 genes intersect) / (post.p>0.95): p = 0.12210 ( 428 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.33730 ( 555 genes intersect) / (post.p>0.95): p = 0.11080 ( 428 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.kidney:
    + All preosteoblasts                         (post.p>0.5): p = 1.0e-09 ( 243 genes intersect) / (post.p>0.95): p = 1.4e-10 ( 197 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 1.0e-09 ( 243 genes intersect) / (post.p>0.95): p = 1.5e-10 ( 197 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 1.1e-09 ( 243 genes intersect) / (post.p>0.95): p = 1.4e-10 ( 197 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 1.1e-09 ( 243 genes intersect) / (post.p>0.95): p = 2.0e-10 ( 197 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.0e-09 ( 243 genes intersect) / (post.p>0.95): p = 1.4e-10 ( 197 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.0e-09 ( 243 genes intersect) / (post.p>0.95): p = 1.1e-10 ( 197 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.liver:
    + All preosteoblasts                         (post.p>0.5): p = 0.00066 ( 364 genes intersect) / (post.p>0.95): p = 3.6e-07 ( 303 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.00066 ( 364 genes intersect) / (post.p>0.95): p = 4.0e-07 ( 303 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00068 ( 364 genes intersect) / (post.p>0.95): p = 3.6e-07 ( 303 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00071 ( 364 genes intersect) / (post.p>0.95): p = 3.6e-07 ( 303 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.00066 ( 364 genes intersect) / (post.p>0.95): p = 3.6e-07 ( 303 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00065 ( 364 genes intersect) / (post.p>0.95): p = 2.9e-07 ( 303 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.lung:
    + All preosteoblasts                         (post.p>0.5): p = 3.0e-07 ( 235 genes intersect) / (post.p>0.95): p = 4.2e-06 ( 180 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 3.0e-07 ( 235 genes intersect) / (post.p>0.95): p = 4.5e-06 ( 180 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 3.1e-07 ( 235 genes intersect) / (post.p>0.95): p = 4.2e-06 ( 180 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 3.3e-07 ( 235 genes intersect) / (post.p>0.95): p = 5.4e-06 ( 180 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 3.0e-07 ( 235 genes intersect) / (post.p>0.95): p = 4.2e-06 ( 180 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 3.0e-07 ( 235 genes intersect) / (post.p>0.95): p = 3.6e-06 ( 180 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    
* Enrichment of Blake2020-DE.heart-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.99880 ( 327 genes intersect) / (post.p>0.95): p = 0.99610 ( 242 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.99880 ( 327 genes intersect) / (post.p>0.95): p = 0.99630 ( 242 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.99860 ( 328 genes intersect) / (post.p>0.95): p = 0.99610 ( 242 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.99860 ( 328 genes intersect) / (post.p>0.95): p = 0.99680 ( 242 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.99880 ( 327 genes intersect) / (post.p>0.95): p = 0.99610 ( 242 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.99880 ( 327 genes intersect) / (post.p>0.95): p = 0.99550 ( 242 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.kidney-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.22690 (  53 genes intersect) / (post.p>0.95): p = 0.14700 (  42 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.22690 (  53 genes intersect) / (post.p>0.95): p = 0.14890 (  42 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.22770 (  53 genes intersect) / (post.p>0.95): p = 0.14700 (  42 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.22910 (  53 genes intersect) / (post.p>0.95): p = 0.15330 (  42 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.22690 (  53 genes intersect) / (post.p>0.95): p = 0.14700 (  42 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.22620 (  53 genes intersect) / (post.p>0.95): p = 0.14340 (  42 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.liver-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.73570 ( 138 genes intersect) / (post.p>0.95): p = 0.31060 ( 112 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.73570 ( 138 genes intersect) / (post.p>0.95): p = 0.31550 ( 112 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.73710 ( 138 genes intersect) / (post.p>0.95): p = 0.31060 ( 112 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.73980 ( 138 genes intersect) / (post.p>0.95): p = 0.28760 ( 113 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.73570 ( 138 genes intersect) / (post.p>0.95): p = 0.31060 ( 112 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.73440 ( 138 genes intersect) / (post.p>0.95): p = 0.30090 ( 112 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Blake2020-DE.lung-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.20120 (  65 genes intersect) / (post.p>0.95): p = 0.47890 (  45 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.20120 (  65 genes intersect) / (post.p>0.95): p = 0.48240 (  45 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.20190 (  65 genes intersect) / (post.p>0.95): p = 0.47890 (  45 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.20350 (  65 genes intersect) / (post.p>0.95): p = 0.49060 (  45 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.20120 (  65 genes intersect) / (post.p>0.95): p = 0.47890 (  45 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.20040 (  65 genes intersect) / (post.p>0.95): p = 0.47200 (  45 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 1.2e-06 (1842 genes intersect) / (post.p>0.95): p = 3.1e-06 (1391 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 1.2e-06 (1842 genes intersect) / (post.p>0.95): p = 2.2e-06 (1394 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 1.4e-06 (1842 genes intersect) / (post.p>0.95): p = 2.5e-06 (1392 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 1.4e-06 (1843 genes intersect) / (post.p>0.95): p = 2.0e-06 (1398 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 1.0e-06 (1843 genes intersect) / (post.p>0.95): p = 2.5e-06 (1392 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 1.1e-06 (1842 genes intersect) / (post.p>0.95): p = 3.1e-06 (1388 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.23610 (   2 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs:
    + All preosteoblasts                         (post.p>0.5): p = 2.3e-06 (1471 genes intersect) / (post.p>0.95): p = 3.1e-07 (1126 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 2.3e-06 (1471 genes intersect) / (post.p>0.95): p = 2.0e-07 (1129 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 2.5e-06 (1471 genes intersect) / (post.p>0.95): p = 2.5e-07 (1127 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 2.9e-06 (1471 genes intersect) / (post.p>0.95): p = 1.5e-07 (1133 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 2.3e-06 (1471 genes intersect) / (post.p>0.95): p = 1.9e-07 (1128 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 2.6e-06 (1470 genes intersect) / (post.p>0.95): p = 3.6e-07 (1123 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.14680 (   2 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs:
    + All preosteoblasts                         (post.p>0.5): p = 3.4e-14 ( 620 genes intersect) / (post.p>0.95): p = 1.4e-14 ( 488 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 3.4e-14 ( 620 genes intersect) / (post.p>0.95): p = 1.7e-14 ( 488 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 3.6e-14 ( 620 genes intersect) / (post.p>0.95): p = 2.3e-14 ( 487 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 4.1e-14 ( 620 genes intersect) / (post.p>0.95): p = 6.9e-15 ( 491 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 3.4e-14 ( 620 genes intersect) / (post.p>0.95): p = 1.4e-14 ( 488 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 3.2e-14 ( 620 genes intersect) / (post.p>0.95): p = 1.5e-14 ( 487 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs:
    + All preosteoblasts                         (post.p>0.5): p = 5.6e-11 (1014 genes intersect) / (post.p>0.95): p = 8.0e-10 ( 773 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 5.6e-11 (1014 genes intersect) / (post.p>0.95): p = 1.4e-09 ( 772 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 6.0e-11 (1014 genes intersect) / (post.p>0.95): p = 1.5e-09 ( 771 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 7.1e-11 (1014 genes intersect) / (post.p>0.95): p = 7.1e-10 ( 776 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 4.1e-11 (1014 genes intersect) / (post.p>0.95): p = 2.0e-09 ( 770 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 5.2e-11 (1014 genes intersect) / (post.p>0.95): p = 3.2e-09 ( 767 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2021-DE.Parental.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.40970 (1373 genes intersect) / (post.p>0.95): p = 0.36300 (1029 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.40970 (1373 genes intersect) / (post.p>0.95): p = 0.33170 (1032 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.41600 (1373 genes intersect) / (post.p>0.95): p = 0.32910 (1031 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.41220 (1373 genes intersect) / (post.p>0.95): p = 0.36050 (1033 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.39330 (1373 genes intersect) / (post.p>0.95): p = 0.34590 (1030 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.40340 (1373 genes intersect) / (post.p>0.95): p = 0.35760 (1027 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.14790 (   2 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.iPSCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.44030 ( 808 genes intersect) / (post.p>0.95): p = 0.33840 ( 609 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.44030 ( 808 genes intersect) / (post.p>0.95): p = 0.27630 ( 613 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.44470 ( 808 genes intersect) / (post.p>0.95): p = 0.30030 ( 611 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.45350 ( 808 genes intersect) / (post.p>0.95): p = 0.30540 ( 613 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.45980 ( 807 genes intersect) / (post.p>0.95): p = 0.28200 ( 612 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.45540 ( 807 genes intersect) / (post.p>0.95): p = 0.31220 ( 609 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 0.05122 (   2 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Parental.CNCCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.02254 ( 151 genes intersect) / (post.p>0.95): p = 0.00089 ( 126 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.02254 ( 151 genes intersect) / (post.p>0.95): p = 0.00093 ( 126 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 0.02276 ( 151 genes intersect) / (post.p>0.95): p = 0.00089 ( 126 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.02320 ( 151 genes intersect) / (post.p>0.95): p = 0.00104 ( 126 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.02254 ( 151 genes intersect) / (post.p>0.95): p = 0.00089 ( 126 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.02232 ( 151 genes intersect) / (post.p>0.95): p = 0.00081 ( 126 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Gokhman2021-DE.Hybrid.CNCCs-UNIQUE:
    + All preosteoblasts                         (post.p>0.5): p = 0.01417 ( 351 genes intersect) / (post.p>0.95): p = 0.09632 ( 256 genes intersect) *
    + All osteoblasts                            (post.p>0.5): p = 0.01417 ( 351 genes intersect) / (post.p>0.95): p = 0.10000 ( 256 genes intersect) *
    + All embedding osteoblasts                  (post.p>0.5): p = 0.01440 ( 351 genes intersect) / (post.p>0.95): p = 0.11020 ( 255 genes intersect) *
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.01489 ( 351 genes intersect) / (post.p>0.95): p = 0.10900 ( 256 genes intersect) *
    + All maturing osteocytes                    (post.p>0.5): p = 0.01417 ( 351 genes intersect) / (post.p>0.95): p = 0.12540 ( 254 genes intersect) *
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.01393 ( 351 genes intersect) / (post.p>0.95): p = 0.13280 ( 253 genes intersect) *
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2020-DMR.bone:
    + All preosteoblasts                         (post.p>0.5): p = 0.00685 (  97 genes intersect) / (post.p>0.95): p = 0.01358 (  74 genes intersect) **
    + All osteoblasts                            (post.p>0.5): p = 0.00685 (  97 genes intersect) / (post.p>0.95): p = 0.01395 (  74 genes intersect) **
    + All embedding osteoblasts                  (post.p>0.5): p = 0.00691 (  97 genes intersect) / (post.p>0.95): p = 0.01358 (  74 genes intersect) **
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.00704 (  97 genes intersect) / (post.p>0.95): p = 0.01018 (  75 genes intersect) **
    + All maturing osteocytes                    (post.p>0.5): p = 0.00685 (  97 genes intersect) / (post.p>0.95): p = 0.01952 (  73 genes intersect) **
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.00679 (  97 genes intersect) / (post.p>0.95): p = 0.01856 (  73 genes intersect) **
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Gokhman2020-skeletal.genes:
    + All preosteoblasts                         (post.p>0.5): p = 0.42710 ( 394 genes intersect) / (post.p>0.95): p = 0.72260 ( 284 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.42710 ( 394 genes intersect) / (post.p>0.95): p = 0.75370 ( 283 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.42990 ( 394 genes intersect) / (post.p>0.95): p = 0.74620 ( 283 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.41000 ( 394 genes intersect) / (post.p>0.95): p = 0.72480 ( 285 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.42710 ( 394 genes intersect) / (post.p>0.95): p = 0.74620 ( 283 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.42430 ( 394 genes intersect) / (post.p>0.95): p = 0.75420 ( 282 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Hsu2019-GWAS.hip.geometry:
    + All preosteoblasts                         (post.p>0.5): p = 0.91210 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.91210 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.91220 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.91230 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.91210 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.91200 (   1 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Styrkarsdottir2019-GWAS.bone.area:
    + All preosteoblasts                         (post.p>0.5): p = 0.31920 (   3 genes intersect) / (post.p>0.95): p = 0.46310 (   2 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.31920 (   3 genes intersect) / (post.p>0.95): p = 0.46380 (   2 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.31930 (   3 genes intersect) / (post.p>0.95): p = 0.46310 (   2 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.31970 (   3 genes intersect) / (post.p>0.95): p = 0.46540 (   2 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.31920 (   3 genes intersect) / (post.p>0.95): p = 0.46310 (   2 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.31900 (   3 genes intersect) / (post.p>0.95): p = 0.46180 (   2 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Yengo2018-GWAS.height:
    + All preosteoblasts                         (post.p>0.5): p = 0.98530 ( 323 genes intersect) / (post.p>0.95): p = 0.98810 ( 235 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.98530 ( 323 genes intersect) / (post.p>0.95): p = 0.98870 ( 235 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.98560 ( 323 genes intersect) / (post.p>0.95): p = 0.98810 ( 235 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.98340 ( 323 genes intersect) / (post.p>0.95): p = 0.99020 ( 235 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.98260 ( 323 genes intersect) / (post.p>0.95): p = 0.99030 ( 234 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.98510 ( 323 genes intersect) / (post.p>0.95): p = 0.98900 ( 234 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

* Enrichment of Steinberg2020-GWAS.OA:
    + All preosteoblasts                         (post.p>0.5): p = 0.82440 ( 109 genes intersect) / (post.p>0.95): p = 0.95790 (  74 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.82440 ( 109 genes intersect) / (post.p>0.95): p = 0.95890 (  74 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.82530 ( 109 genes intersect) / (post.p>0.95): p = 0.95790 (  74 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.82730 ( 109 genes intersect) / (post.p>0.95): p = 0.94910 (  75 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.82440 ( 109 genes intersect) / (post.p>0.95): p = 0.95790 (  74 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.82340 ( 109 genes intersect) / (post.p>0.95): p = 0.95560 (  74 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
* Enrichment of Morris2020-GWAS.OP:
    + All preosteoblasts                         (post.p>0.5): p = 0.31080 ( 165 genes intersect) / (post.p>0.95): p = 0.11060 ( 131 genes intersect)
    + All osteoblasts                            (post.p>0.5): p = 0.31080 ( 165 genes intersect) / (post.p>0.95): p = 0.13520 ( 130 genes intersect)
    + All embedding osteoblasts                  (post.p>0.5): p = 0.31240 ( 165 genes intersect) / (post.p>0.95): p = 0.13200 ( 130 genes intersect)
    + All mineralizing osteoblasts               (post.p>0.5): p = 0.28140 ( 166 genes intersect) / (post.p>0.95): p = 0.12020 ( 131 genes intersect)
    + All maturing osteocytes                    (post.p>0.5): p = 0.31080 ( 165 genes intersect) / (post.p>0.95): p = 0.11060 ( 131 genes intersect)
    + Shared across all ostadhoc classifications (post.p>0.5): p = 0.30920 ( 165 genes intersect) / (post.p>0.95): p = 0.12580 ( 130 genes intersect)
    + Shared across obl+emb+min+mat              (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)
    + Shared across emb+min+mat                  (post.p>0.5): p = 1.00000 (   0 genes intersect) / (post.p>0.95): p = 1.00000 (   0 genes intersect)

```{r, eval=FALSE}

#Enrichment of external gene sets in cormotif DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All preosteoblasts
  print("All preosteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All osteoblasts
  print("All osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All embedding osteoblasts
  print("All embedding osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All mineralizing osteoblasts
  print("All mineralizing osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All maturing osteocytes
  print("All maturing osteocytes")
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications
  print("Shared across all ostadhoc classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across obl+emb+min+mat
  print("Shared across obl+emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across emb+min+mat
  print("Shared across emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across obl+emb+min+mat
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Obl/Emb/Min/Mat",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across emb+min+mat
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Emb/Min/Mat",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
                    
  #All preosteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Preostoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All embedding osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Embedding Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All mineralizing osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Mineralizing Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All maturing osteocytes
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Maturing Osteocytes",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print("Shared across all ostadhoc classifications, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print("Shared across pre+obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across obl+emb+min, tested in all classifications
  print("Shared across obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to preosteoblasts, tested in all classifications
  print("Unique to preosteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to osteoblasts, tested in all classifications
  print("Unique to osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to embedding osteoblasts, tested in all classifications
  print("Unique to embedding osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print("Unique to mineralizing osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to maturing osteocytes, tested in all classifications
  print("Unique to maturing osteocytes, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))

}

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
### Examine Conservation of Differentiation-Specific DE Genes
