---
title: "10x-test-data-full-cell-type-validation"
author: "Genevieve Housman"
date: "11/12/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Comparison

Loading, processing, and viewing of 10X data using seurat.

Data being examined comes from 1 human and 1 chimpanzee.
- GHO1 includes iPSCs from both species
- GHO2 inclues iPSC-derived MSCs from both species
- GHO3 includes iPSC-MSC-derived chondrocytes
- GHO4 includes iPSC-MSC-derived osteoblasts

Gene count data were processed using the human genome (h) or the chimpanzee genome (c), as well as the full annotation (full) for each.
Species-specific assignments were determined using the human-chimpanzee genome combination with ortho-exon annotation (sppo).

Thus, the final datasets examined, include:
- h.full.sppo
- c.full.sppo


```{r load libraries}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

```{r define data location}

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full'        #data from three lanes of sequencing

#Define sub-directories
dir_h.full.GHO1 <- paste0(dir,'/cellranger-runs-human/GHO-1/outs')
dir_h.full.GHO2 <- paste0(dir,'/cellranger-runs-human/GHO-2/outs')
dir_h.full.GHO3 <- paste0(dir,'/cellranger-runs-human/GHO-3/outs')
dir_h.full.GHO4 <- paste0(dir,'/cellranger-runs-human/GHO-4/outs')
dir_c.full.GHO1 <- paste0(dir,'/cellranger-runs-chimp/GHO-1/outs')
dir_c.full.GHO2 <- paste0(dir,'/cellranger-runs-chimp/GHO-2/outs')
dir_c.full.GHO3 <- paste0(dir,'/cellranger-runs-chimp/GHO-3/outs')
dir_c.full.GHO4 <- paste0(dir,'/cellranger-runs-chimp/GHO-4/outs')

dir_hc.orth.GHO1 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-1/outs')
dir_hc.orth.GHO2 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-2/outs')
dir_hc.orth.GHO3 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-3/outs')
dir_hc.orth.GHO4 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-4/outs')

#Check that directories contain barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz
#list.files(paste0(dir_hc.orth.GHO1,"/filtered_feature_bc_matrix"))
#list.files(paste0(dir_hc.orth.GHO1,"/raw_feature_bc_matrix"))

```

```{r load species assignments}

#Load species assignment (only possible for human-chimp combo reference)
hc.orth.spp1 <- read.csv(file=paste0(dir_hc.orth.GHO1,"/analysis/gem_classification.csv"))
hc.orth.spp2 <- read.csv(file=paste0(dir_hc.orth.GHO2,"/analysis/gem_classification.csv"))
hc.orth.spp3 <- read.csv(file=paste0(dir_hc.orth.GHO3,"/analysis/gem_classification.csv"))
hc.orth.spp4 <- read.csv(file=paste0(dir_hc.orth.GHO4,"/analysis/gem_classification.csv"))
hc.orth.spp1$barcode <- sapply(hc.orth.spp1$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp2$barcode <- sapply(hc.orth.spp2$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp3$barcode <- sapply(hc.orth.spp3$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp4$barcode <- sapply(hc.orth.spp4$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="ortho_", replacement="")

#Examine species assignment data    #FULL DATA
table(hc.orth.spp1$call)            #Chimp=3294, Human=2319, Multiplet=  87
table(hc.orth.spp2$call)            #Chimp=3149, Human=2563, Multiplet= 123
table(hc.orth.spp3$call)            #Chimp=1305, Human=2168, Multiplet= 114
table(hc.orth.spp4$call)            #Chimp=1805, Human=1580, Multiplet=  62

CombinePlots(plots=list((ggplot(hc.orth.spp1, aes(x=call))+geom_bar()+labs(title="iPSC-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp2, aes(x=call))+geom_bar()+labs(title="MSC-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp3, aes(x=call))+geom_bar()+labs(title="chondro-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp4, aes(x=call))+geom_bar()+labs(title="osteo-orth")+ylim(0,4500))), ncol=4)

```

```{r prepare scRNA data}

#Prepare data for h.full.sppo
#(1)load data
  h.full.GHO1_data <- Read10X(data=paste0(dir_h.full.GHO1,"/raw_feature_bc_matrix"))
  h.full.GHO2_data <- Read10X(data=paste0(dir_h.full.GHO2,"/raw_feature_bc_matrix"))
  h.full.GHO3_data <- Read10X(data=paste0(dir_h.full.GHO3,"/raw_feature_bc_matrix"))
  h.full.GHO4_data <- Read10X(data=paste0(dir_h.full.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  h.full.GHO1 <- CreateSeuratObject(counts=h.full.GHO1_data, project="iPSC")
  h.full.GHO2 <- CreateSeuratObject(counts=h.full.GHO2_data, project="MSC")
  h.full.GHO3 <- CreateSeuratObject(counts=h.full.GHO3_data, project="chondrocyte")
  h.full.GHO4 <- CreateSeuratObject(counts=h.full.GHO4_data, project="osteoblast")
#(3)view details of seurat objects
  h.full.GHO1                       #33538 features, 6794880 samples
  h.full.GHO2                       #33538 features, 6794880 samples
  h.full.GHO3                       #33538 features, 6794880 samples
  h.full.GHO4                       #33538 features, 6794880 samples
#(4)clear raw data from workspace
  rm(h.full.GHO1_data,h.full.GHO2_data,h.full.GHO3_data,h.full.GHO4_data)
#(5)add species assignments to data
  m <- match(rownames(h.full.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO1 <- AddMetaData(h.full.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO2 <- AddMetaData(h.full.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO3 <- AddMetaData(h.full.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO4 <- AddMetaData(h.full.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments
  table(h.full.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M= 87, NA=6789180
  table(h.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M=123, NA=6789045
  table(h.full.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M=114, NA=6791293
  table(h.full.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M= 62, NA=6791433
#(7)subset data to only include human cells
  h.full.sppo.GHO1 <- subset(h.full.GHO1, subset=species=="Human")
  h.full.sppo.GHO2 <- subset(h.full.GHO2, subset=species=="Human")
  h.full.sppo.GHO3 <- subset(h.full.GHO3, subset=species=="Human")
  h.full.sppo.GHO4 <- subset(h.full.GHO4, subset=species=="Human")
#(8)create merged object
  h.full.sppo <- merge(h.full.sppo.GHO1,c(h.full.sppo.GHO2,h.full.sppo.GHO3,h.full.sppo.GHO4),
                       add.cell.ids=c("iPSC","MSC","chondro","osteo"),
                       project="h.full.sppo")
  saveRDS(h.full.sppo, file = "../data/cellranger-data-full/h.full.sppo.rds")
#(9)clear excess data from workspace
  rm(h.full.GHO1,h.full.GHO2,h.full.GHO3,h.full.GHO4) 
  rm(h.full.sppo.GHO1,h.full.sppo.GHO2,h.full.sppo.GHO3,h.full.sppo.GHO4)

#Prepare data for c.full.sppo
#(1)load data
  c.full.GHO1_data <- Read10X(data=paste0(dir_c.full.GHO1,"/raw_feature_bc_matrix"))
  c.full.GHO2_data <- Read10X(data=paste0(dir_c.full.GHO2,"/raw_feature_bc_matrix"))
  c.full.GHO3_data <- Read10X(data=paste0(dir_c.full.GHO3,"/raw_feature_bc_matrix"))
  c.full.GHO4_data <- Read10X(data=paste0(dir_c.full.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  c.full.GHO1 <- CreateSeuratObject(counts=c.full.GHO1_data, project="iPSC")
  c.full.GHO2 <- CreateSeuratObject(counts=c.full.GHO2_data, project="MSC")
  c.full.GHO3 <- CreateSeuratObject(counts=c.full.GHO3_data, project="chondrocyte")
  c.full.GHO4 <- CreateSeuratObject(counts=c.full.GHO4_data, project="osteoblast")
#(3)view details of seurat objects
  c.full.GHO1                       #25312 features, 6794880 samples
  c.full.GHO2                       #25312 features, 6794880 samples
  c.full.GHO3                       #25312 features, 6794880 samples
  c.full.GHO4                       #25312 features, 6794880 samples
#(4)clear raw data from workspace
  rm(c.full.GHO1_data,c.full.GHO2_data,c.full.GHO3_data,c.full.GHO4_data)
#(5)add species assignments to data
  m <- match(rownames(c.full.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO1 <- AddMetaData(c.full.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO2 <- AddMetaData(c.full.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO3 <- AddMetaData(c.full.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO4 <- AddMetaData(c.full.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments
  table(c.full.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M= 87, NA=6789180
  table(c.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M=123, NA=6789045
  table(c.full.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M=114, NA=6791293
  table(c.full.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M= 62, NA=6791433
#(7)subset data to only include chimp cells
  c.full.sppo.GHO1 <- subset(c.full.GHO1, subset=species=="Chimp")
  c.full.sppo.GHO2 <- subset(c.full.GHO2, subset=species=="Chimp")
  c.full.sppo.GHO3 <- subset(c.full.GHO3, subset=species=="Chimp")
  c.full.sppo.GHO4 <- subset(c.full.GHO4, subset=species=="Chimp")
#(8)create merged object - sppo
  c.full.sppo <- merge(c.full.sppo.GHO1,c(c.full.sppo.GHO2,c.full.sppo.GHO3,c.full.sppo.GHO4),
                       add.cell.ids=c("iPSC","MSC","chondro","osteo"),
                       project="c.full.sppo")
  saveRDS(c.full.sppo, file = "../data/cellranger-data-full/c.full.sppo.rds")
#(9)clear excess data from workspace
  rm(c.full.GHO1,c.full.GHO2,c.full.GHO3,c.full.GHO4)
  rm(c.full.sppo.GHO1,c.full.sppo.GHO2,c.full.sppo.GHO3,c.full.sppo.GHO4)

rm(hc.orth.spp1,hc.orth.spp2,hc.orth.spp3,hc.orth.spp4)
rm(dir_h.full.GHO1,dir_h.full.GHO2,dir_h.full.GHO3,dir_h.full.GHO4)
rm(dir_c.full.GHO1,dir_c.full.GHO2,dir_c.full.GHO3,dir_c.full.GHO4)
rm(dir_hc.orth.GHO1,dir_hc.orth.GHO2,dir_hc.orth.GHO3,dir_hc.orth.GHO4)
rm(m)

```

```{r data meta information adjustments}

# Load data
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/c.full.sppo.rds"))

#Fix meta.data details so that plotting is easier later on
h.full.sppo@meta.data$species <- factor(h.full.sppo@meta.data$species, levels=c("Human","Chimp"))
c.full.sppo@meta.data$species <- factor(c.full.sppo@meta.data$species, levels=c("Human","Chimp"))
h.full.sppo@meta.data$orig.ident <- factor(h.full.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
c.full.sppo@meta.data$orig.ident <- factor(c.full.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))

#notes: colors for coordination with figures
#col4=c("pink","mediumorchid","gold","dodgerblue") #colors for cell types
#scale_fill_manual(values=col4)

```

```{r data qc prefiltering}

#Calculate the percentage of reads that map to the mitochondrial genome
h.full.sppo[["percent.mt"]] <- PercentageFeatureSet(h.full.sppo, pattern="^MT-")
c.full.sppo[["percent.mt"]] <- PercentageFeatureSet(c.full.sppo, pattern="^MT-")
#notes: 13 mito genes in human full annotation + 1 mito gene in chimp full annotation

```

```{r data qc prefiltering}

#Check and visualize numbers of cells per species
table(h.full.sppo@meta.data$species, useNA="always") #H=8630
table(c.full.sppo@meta.data$species, useNA="always") #C=9553

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="h.full.sppo")+ylim(0,10000)),
                        (ggplot(c.full.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="c.full.sppo")+ylim(0,10000))), ncol=2)

#Show QC metrics for the first 5 cells
#nFeature_RNA: number of unique genes detected in each cell
#nCount_RNA: total number of molecules (umi) detected within a cell
#percent.mt: percentage of reads that map to the mitochondrial genome
head(h.full.sppo@meta.data, 5)
head(c.full.sppo@meta.data, 5)

#Check and visualize QC metrics
#notes: examined nCount_RNA, nFeature_RNA, percent.mt
mean(h.full.sppo@meta.data$nCount_RNA)
mean(h.full.sppo@meta.data$nCount_RNA[which(h.full.sppo[['orig.ident']]=="iPSC")])
mean(h.full.sppo@meta.data$nCount_RNA[which(h.full.sppo[['orig.ident']]=="MSC")])
mean(h.full.sppo@meta.data$nCount_RNA[which(h.full.sppo[['orig.ident']]=="chondrocyte")])
mean(h.full.sppo@meta.data$nCount_RNA[which(h.full.sppo[['orig.ident']]=="osteoblast")])
mean(c.full.sppo@meta.data$nCount_RNA)
mean(c.full.sppo@meta.data$nCount_RNA[which(c.full.sppo[['orig.ident']]=="iPSC")])
mean(c.full.sppo@meta.data$nCount_RNA[which(c.full.sppo[['orig.ident']]=="MSC")])
mean(c.full.sppo@meta.data$nCount_RNA[which(c.full.sppo[['orig.ident']]=="chondrocyte")])
mean(c.full.sppo@meta.data$nCount_RNA[which(c.full.sppo[['orig.ident']]=="osteoblast")])

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,135000)),
                        (ggplot(c.full.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,135000))), ncol=2)
CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,135000)),
                        (ggplot(c.full.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,135000))), ncol=2)

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,9500)),
                        (ggplot(c.full.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,9500))), ncol=2)
CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,9500)),
                        (ggplot(c.full.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,9500))), ncol=2)

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,100)),
                        (ggplot(c.full.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,100))), ncol=2)
CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.full.sppo")+ylim(0,100)),
                        (ggplot(c.full.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="c.full.sppo")+ylim(0,100))), ncol=2)

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,100)+labs(title="h.fullsppo")),
                        (ggplot(c.full.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,100)+labs(title="c.full.sppo"))), ncol=2)

CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,9500)+labs(title="h.fullsppo")),
                        (ggplot(c.full.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,9500)+labs(title="c.full.sppo"))), ncol=2)

```

```{r save data temporarily}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
saveRDS(h.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.full.sppo.rds"))
saveRDS(c.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-c.full.sppo.rds"))

```

```{r load processed data}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-c.full.sppo.rds"))

```

```{r data qc}

#Visualize some qc metrics of data
# - number of reads per cell (count depth)
# - reads per genes
# - expressed genes per cell (complexity)
# - rarity of genes (cells expressing genes)

##old way of doing this
#counts_per_cell <- Matrix::colSums(h.full.sppo@assays$RNA@counts) #or use nCount_RNA
#counts_per_gene <- Matrix::rowSums(h.full.sppo@assays$RNA@counts)
#genes_per_cell <- Matrix::colSums(h.full.sppo@assays$RNA@counts>0) #or use nFeature_RNA
#cells_per_gene <- Matrix::rowSums(h.full.sppo@assays$RNA@counts>0)
#par(mfrow=c(2,2))
#hist(log10(counts_per_cell+1),main='counts per cell',col='wheat')
#hist(log10(counts_per_gene+1),main='counts per gene',col='wheat')
#hist(log10(genes_per_cell+1),main='genes per cell',col='wheat')
#hist(log10(cells_per_gene+1),main='cells per gene',col='wheat')
#par(mfrow=c(1,1))
#plot(counts_per_cell,genes_per_cell, log='xy',col='wheat')
#title('counts vs genes per cell')
#plot(sort(genes_per_cell), xlab='cell', log='y', main='genes per cell (ordered)')

##### ADD CELL TYPE INFORMATION #####

df_per_cell <- data.frame(
  species <- factor(rep(c("Human","Chimp"),times=c(length(colnames(h.full.sppo@assays$RNA@counts)),length(colnames(c.full.sppo@assays$RNA@counts)))),
                    levels=c("Human","Chimp")),
  orig.ident <- factor(c(as.character(h.full.sppo@meta.data$orig.ident),as.character(c.full.sppo@meta.data$orig.ident)),
                       levels=c("iPSC","MSC","chondrocyte","osteoblast")),
  counts_per_cell <- c(Matrix::colSums(h.full.sppo@assays$RNA@counts),Matrix::colSums(c.full.sppo@assays$RNA@counts)),
  genes_per_cell <- c(Matrix::colSums(h.full.sppo@assays$RNA@counts>0),Matrix::colSums(c.full.sppo@assays$RNA@counts>0))
)
colnames(df_per_cell) <- c("species","orig.ident","counts_per_cell","genes_per_cell")

df_per_gene <- data.frame(
  species <- factor(rep(c("Human","Chimp"),times=c(length(rownames(h.full.sppo@assays$RNA@counts)),length(rownames(c.full.sppo@assays$RNA@counts))))),
  counts_per_gene <- c(Matrix::rowSums(h.full.sppo@assays$RNA@counts),Matrix::rowSums(c.full.sppo@assays$RNA@counts)),
  cells_per_gene <- c(Matrix::rowSums(h.full.sppo@assays$RNA@counts>0),Matrix::rowSums(c.full.sppo@assays$RNA@counts>0))
)
colnames(df_per_gene) <- c("species","counts_per_gene","cells_per_gene")

#count depth (number of reads per cell)
ggplot(df_per_cell, aes(x=counts_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.1, position="identity", binwidth=1000) +
  facet_grid(orig.ident ~ .)
ggplot(df_per_cell, aes(x=counts_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.1, position="identity", binwidth=50) +
  xlim(0,4000) +
  geom_vline(xintercept=1600, linetype="dashed") +
  facet_grid(orig.ident ~ .)
ggplot(df_per_cell, aes(x=counts_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.5, position="identity", binwidth=1000) +
  geom_vline(xintercept=1600, linetype="dashed") +
  facet_grid(orig.ident ~ .)
#may be good to subset to minimum of 1600 reads or should thresholds be different across different cell types/species?

#complexity (expressed genes per cell)
ggplot(df_per_cell, aes(x=genes_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.1, position="identity", binwidth=100) +
  facet_grid(orig.ident ~ .)
ggplot(df_per_cell, aes(x=genes_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.1, position="identity", binwidth=25) +
  xlim(0,2000) +
  geom_vline(xintercept=800, linetype="dashed") +
  facet_grid(orig.ident ~ .)
ggplot(df_per_cell, aes(x=genes_per_cell, color=species)) +
  geom_histogram(fill="white", alpha=0.1, position="identity", binwidth=100) +
  geom_vline(xintercept=800, linetype="dashed") +
  facet_grid(orig.ident ~ .)
#may be good to subset to minimum of 800 genes or should thresholds be different across different cell types/species?

#check correlation between count depth and complexity
ggplot(df_per_cell, aes(x=counts_per_cell, y=genes_per_cell, color=species)) +
  geom_point(size=0.5,alpha=0.1)+labs(title="counts vs. genes per cell") +
  geom_vline(xintercept=1600, linetype="dashed") +
  geom_hline(yintercept=800, linetype="dashed") +
  facet_grid(orig.ident ~ .)

#investigate potential that cells are actually failed libraries (lower end outliers) or are cell doublets (higher end outliers)
CombinePlots(plots=list((ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "iPSC"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           labs(title="Human iPSCs", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "MSC"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           labs(title="Human MSCs", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "chondrocyte"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           labs(title="Human Chondrocytes", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "osteoblast"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           labs(title="Human Osteoblasts", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "iPSC"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           labs(title="Chimp iPSCs", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "MSC"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           labs(title="Chimp MSCs", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "chondrocyte"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           labs(title="Chimp Chondrocytes", x="Cell Rank", y="Reads per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "osteoblast"),
                                aes(x=1:length(counts_per_cell), y=sort(counts_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           labs(title="Chimp Osteoblasts", x="Cell Rank", y="Reads per Cell"))), ncol=4)

CombinePlots(plots=list((ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "iPSC"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Human iPSCs", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "MSC"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Human MSCs", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "chondrocyte"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Human Chondrocytes", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Human"), orig.ident %in% "osteoblast"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#F8766D") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Human Osteoblasts", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "iPSC"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Chimp iPSCs", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "MSC"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Chimp MSCs", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "chondrocyte"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Chimp Chondrocytes", x="Cell Rank", y="Genes per Cell")),
                        (ggplot(subset(subset(df_per_cell, species %in% "Chimp"), orig.ident %in% "osteoblast"),
                                aes(x=1:length(genes_per_cell), y=sort(genes_per_cell))) +
                           geom_point(size=0.5,alpha=0.5,color="#00BFC4") +
                           geom_hline(yintercept=800, linetype="dashed") +
                           labs(title="Chimp Osteoblasts", x="Cell Rank", y="Genes per Cell"))), ncol=4)

#reads per gene
ggplot(df_per_gene, aes(x=log(counts_per_gene+1), color=species)) +
  geom_histogram(fill="white", alpha=0.5, binwidth=0.25) +
  facet_grid(species ~ .)

#rarity of genes (cells expressing genes)
ggplot(df_per_gene, aes(x=log(cells_per_gene+1), color=species)) +
  geom_histogram(fill="white", alpha=0.5, binwidth=0.2) +
  facet_grid(species ~ .)

#Should I look into assessing expression of housekeeping genes?
#https://broadinstitute.github.io/2019_scWorkshop/data-wrangling-scrnaseq.html

```

```{r data filtering}

#Filter Barcodes

#library(scater)
#https://scrnaseq-course.cog.sanger.ac.uk/website/cleaning-the-expression-matrix.html
#https://broadinstitute.github.io/2019_scWorkshop/data-wrangling-scrnaseq.html#preprocessing-step-1-filter-out-low-quality-cells

#(1a) Filter based on number of read counts in each cell (count depth)
#(1b) Filter based on number of unique genes detected in each cell (complexity)
CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,10000)+labs(title="h.full.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(c.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,10000)+labs(title="c.full.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed"))), ncol=2)
#notes: no filtering on feature counts at this time
#test <- subset(h.full.sppo, subset=nFeature_RNA>800 && nCount_RNA>1600)

#(2) Filter basted on detectable genes
#detectable gene: >=1 UMI in >=2 cells (>=5 reads in >=2 cells)
#genes must be filtered after cell filtering since some genes may only be detected in poor quality cells
#notes: no filtering on genes at this time

#(3) Filter based on mitochondrial RNA content - CANNOT DO THIS FOR HUMAN-CHIMP COMPARISONS
#Rationale: cells lyse more easily than mitochondria, so high mito RNA content can signal cell lysis and RNA degradation (ideally percent.mt should be <5%)
CombinePlots(plots=list((ggplot(h.full.sppo@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,100)+labs(title="h.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(c.full.sppo@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,135000)+ylim(0,100)+labs(title="c.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed"))), ncol=2)
#notes: no filtering on percent mito at this time

```

```{r normalize data}

#Normalize data using default settings
h.full.sppo <- NormalizeData(h.full.sppo, normalization.method="LogNormalize", scale.factor=10000)
c.full.sppo <- NormalizeData(c.full.sppo, normalization.method="LogNormalize", scale.factor=10000)

#global-scaling normalization method that normalizes the gene expression measurements for each cell by the total expression, multiplies this by a scale factor, and log-transforms the result (simplest and most intuitive method)

```

```{r highly variable feature selection}

#Calculate subset of features that exhibit high cell-to-cell variation in dataset
h.full.sppo <- FindVariableFeatures(h.full.sppo, selection.method="vst", nfeatures=2000)
c.full.sppo <- FindVariableFeatures(c.full.sppo, selection.method="vst", nfeatures=2000)
#calculates average expression and dispersion for each gene, places genes into bins, and calculates z-score for dispersion within each bin; this helps control for the relationship between variability and average expression

# Identify 10 most highly variable genes
top10.h <- head(VariableFeatures(h.full.sppo), 10)
top10.c <- head(VariableFeatures(c.full.sppo), 10)

#Plot variable features with top 10 features labels
LabelPoints(plot=VariableFeaturePlot(h.full.sppo),points=top10.h, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(c.full.sppo),points=top10.c, repel=TRUE)

```

```{r scale data}

#Scale data with linear transformation such that:
#(1) expression of each gene is shifted so that mean expression across cells is 0
#(2) expression of each gene is scaled so that variance across cells is 1
#results of this are stored in pbmc[["RNA"]]@scale.data
h.full.sppo <- ScaleData(h.full.sppo, features=rownames(h.full.sppo))
c.full.sppo <- ScaleData(c.full.sppo, features=rownames(c.full.sppo))

```

```{r linear dimensional reduction}

#Perform PCA on scaled data
h.full.sppo <- RunPCA(h.full.sppo, npcs=100, features=VariableFeatures(object=h.full.sppo), verbose=FALSE)
c.full.sppo <- RunPCA(c.full.sppo, npcs=100, features=VariableFeatures(object=c.full.sppo), verbose=FALSE)

#Determine dimensionality of dataset
#seurat clusters cells using PCA scores - each PC represents a ‘metafeature’ that combines information across a correlated feature set, the top PCs represent a robust compression of the dataset - need to determine how many PCs to include

#OPTION 1 - Exploring PCs to determine relevant sources of heterogeneity
#Examine PCA results
print(h.full.sppo[["pca"]], dims=1:5, nfeatures=5)
print(c.full.sppo[["pca"]], dims=1:5, nfeatures=5)
#Visualize PCA results
VizDimLoadings(h.full.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(c.full.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
DimPlot(h.full.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(c.full.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
#pca plot for top 500 cells on pc1
DimHeatmap(h.full.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(c.full.sppo, dims=1, cells=500, balanced=TRUE)
#pca plot for top 500 cells on pc1 through pc9
DimHeatmap(h.full.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(c.full.sppo, dims=1:9, cells=500, balanced=TRUE)

#OPTION 2- JackStraw procedure: randomly permute subset of data (1% by default), rerun PCA constructing a null distribution of feature scores, repeat this procedure, identify significant PCs as those with strong enrichment of low p-value features
#h.full.sppo <- JackStraw(h.full.sppo, num.replicate=100)
#c.full.sppo <- JackStraw(c.full.sppo, num.replicate=100)
#h.full.sppo <- ScoreJackStraw(h.full.sppo, dims=1:20)
#c.full.sppo <- ScoreJackStraw(c.full.sppo, dims=1:20)
#visualize the distribution of p-values for each PC with a uniform distribution (dashed line)
#sign PCs show strong enrichment of features with low p-values (solid curve above dashed line)
#JackStrawPlot(h.full.sppo, dims=1:20)
#JackStrawPlot(c.full.sppo, dims=1:20)
#TOO COMPUTATIONALLY EXPENSIVE - DO NOT RUN

#OPTION 3 -Elbow plot procedure: ranking of PCs based on the percentage of variance explained by each one
ElbowPlot(h.full.sppo, ndims=100)
ElbowPlot(c.full.sppo, ndims=100)

#keep all dims that explaim more than 0.1% of variance
pva <- h.full.sppo@reductions$pca@stdev^2/h.full.sppo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001))
ElbowPlot(h.full.sppo, ndims=100) + geom_vline(xintercept=ndim, linetype="dashed", color="red")
ndim
pva <- c.full.sppo@reductions$pca@stdev^2/c.full.sppo@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001))
ElbowPlot(c.full.sppo, ndims=100) + geom_vline(xintercept=ndim, linetype="dashed", color="red")
ndim
#lets retain the first 52PCs for h.full.sppo and first 49PCs for c.full.spof

```

```{r clustering}

#Cluster cells using a distance metric based on previously identified PCs and partion cellular distance matrix into clusters by embedding cells in a graph structure (e.g. K-nearest neighbor graph) with edges drawn between cells with similar feature expression patterns and then dividing this graph into highly interconnected ‘quasi-cliques’/‘communities’

#(1)construct KNN graph based on euclidean distance in PCA space and refine edge weights between any two cells based on shared overlap in their local neighborhoods (Jaccard similarity). This step is performed using the FindNeighbors function, and takes as input the previously defined dimensionality of the dataset (first 20 PCs).
h.full.sppo <- FindNeighbors(h.full.sppo, dims=1:52)
c.full.sppo <- FindNeighbors(c.full.sppo, dims=1:49)

#(2)apply modularity optimization techniques to iteratively group cells together
h.full.sppo <- FindClusters(h.full.sppo, dims.use=1:52, resolution=0.13)
c.full.sppo <- FindClusters(c.full.sppo, dims.use=1:49, resolution=0.13)

#Look at cluster IDs of the first 5 cells
head(Idents(h.full.sppo),5)          #8 clusters
head(Idents(c.full.sppo),5)          #10 clusters

#Save data
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
saveRDS(h.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.full.sppo.rds"))
saveRDS(c.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-c.full.sppo.rds"))

```

```{r non-linear dimensional reduction}

#Run non-linear dimensional reduction (UMAP/tSNE)
#notes: run this in the terminal and then come back to rstudio

##Install umap
#module load python/3.7.0
#module load R/3.5.1
#R
#library(reticulate)
#use_python("/software/python-3.7.0-el7-x86_64/bin/python")
#py_config()
#py_discover_config()
#cd /home/ghousman
#wget https://github.com/lmcinnes/umap/archive/master.zip
#unzip master.zip
#rm master.zip
#cd umap-master
#export PYTHONUSERBASE=$HOME/.local/python
#mkdir -p $PYTHONUSERBASE
#pip3 install -r requirements.txt --user
#pip3 install --upgrade pip --user
#python setup.py install --user
##notes: package installed in /home/ghousman/.local/python/lib/python3.7/site-packages

#sinteractive --mem=4g
#module load python/3.7.0
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R
library(Seurat)
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.sppo.rds"))
h.full.sppo <- RunUMAP(h.full.sppo, dims=1:52)
saveRDS(h.full.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
rm(h.full.sppo)
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-c.full.sppo.rds"))
c.full.sppo <- RunUMAP(c.full.sppo, dims=1:49)
saveRDS(c.full.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))
rm(c.full.sppo)

```

```{r visualize non-linear dimensional reduction}

#Visualize non-linear dimensional reduction (UMAP/tSNE)
#notes: do this next part in rstudio

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))

#ggplot(as.data.frame(hc.orth.sppo.tot@reductions$umap@cell.embeddings), aes(UMAP_1, UMAP_2))+geom_point()
CombinePlots(plots=list(DimPlot(h.full.sppo, reduction="umap", label=TRUE),
                        DimPlot(c.full.sppo, reduction="umap", label=TRUE)), ncol=2)

CombinePlots(plots=list(DimPlot(h.full.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(c.full.sppo, reduction="umap", group.by="orig.ident")), ncol=2)

```

```{r visualize marker expression}

#notes: VlnPlot shows expression probability distributions across clusters
#notes: FeaturePlot shows feature expression on a UMAP, tSNE, or PCA plot
#notes: other plotting functions include RidgePlot, CellScatter, and DotPlot
#notes: DoHeatmap generates an expression heatmap for given cells and features

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))

#ipsc - all markers expressed in human and chimp iPSCs
FeaturePlot(h.full.sppo, features=c("NANOG","POU5F1","TDGF1"))
FeaturePlot(c.full.sppo, features=c("NANOG","POU5F1","TDGF1"))
VlnPlot(h.full.sppo, features=c("NANOG","POU5F1","TDGF1"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("NANOG","POU5F1","TDGF1"), group.by="orig.ident")
#mesoderm - no genes look good
FeaturePlot(h.full.sppo, features=c("MIXL1","TBXT","EOMES"))
FeaturePlot(c.full.sppo, features=c("MIXL1","TBXT","EOMES"))
VlnPlot(h.full.sppo, features=c("MIXL1","TBXT","EOMES"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("MIXL1","TBXT","EOMES"), group.by="orig.ident")
#msc - genes generally match expectations
FeaturePlot(h.full.sppo, features=c("THY1","ENG","NT5E","CD44","ITGB1","CD200","ALCAM","MCAM","NGFR")) #should be positive in msc
FeaturePlot(c.full.sppo, features=c("THY1","ENG","NT5E","CD44","ITGB1","CD200","ALCAM","MCAM","NGFR")) #should be positive in msc
VlnPlot(h.full.sppo, features=c("THY1","ENG","NT5E","CD44","ITGB1","CD200","ALCAM","MCAM","NGFR"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("THY1","ENG","NT5E","CD44","ITGB1","CD200","ALCAM","MCAM","NGFR"), group.by="orig.ident")
FeaturePlot(h.full.sppo, features=c("CD34","PTPRC","ITGAM","CD19","HLA-DRB1","CD14","CD79A")) #should be negative in msc
FeaturePlot(c.full.sppo, features=c("CD34","PTPRC","ITGAM","CD19","HLA-DRB1","CD14","CD79A")) #should be negative in msc
VlnPlot(h.full.sppo, features=c("CD34","PTPRC","ITGAM","CD19","HLA-DRB1","CD14","CD79A"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("CD34","PTPRC","ITGAM","CD19","HLA-DRB1","CD14","CD79A"), group.by="orig.ident")
#chondrocyte - SOX5/SOX6/SOX9/COL10A1 look okay; COL11A1 in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"))
FeaturePlot(c.full.sppo, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"))
VlnPlot(h.full.sppo, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"), group.by="orig.ident")
#osteoblast - RUNX2 looks okay; COL1A1 in MSCs/chondrocytes/osteoblasts; ALPL in iPSCs
FeaturePlot(h.full.sppo, features=c("COL1A1","BGLAP","RUNX2","ALPL","IBSP","SP7"))
FeaturePlot(c.full.sppo, features=c("COL1A1","BGLAP","RUNX2","ALPL","IBSP","SP7"))
VlnPlot(h.full.sppo, features=c("COL1A1","BGLAP","RUNX2","ALPL","IBSP","SP7"), group.by="orig.ident")
VlnPlot(c.full.sppo, features=c("COL1A1","BGLAP","RUNX2","ALPL","IBSP","SP7"), group.by="orig.ident")
#other - FABP4/TNNT2 in human chondrocytes; SOX2/EPCAM/MYL7 in iPSCs; GAPDH/GUSB/YWHAZ in iPSCs/MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features=c("FABP4","PPARG","SOX17","SOX1","SOX2","EPCAM","ALB","HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"))
FeaturePlot(c.full.sppo, features=c("FABP4","PPARG","SOX17","SOX1","SOX2","EPCAM","ALB","HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"))
VlnPlot(h.full.sppo, features=c("FABP4","PPARG","SOX17","SOX1","SOX2","EPCAM","ALB","HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"),
        group.by="orig.ident")
VlnPlot(h.full.sppo, features=c("FABP4","PPARG","SOX17","SOX1","SOX2","EPCAM","ALB","HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"),
        group.by="orig.ident")

FeaturePlot(h.full.sppo, features="CFL1")   #cofilin - in iPSCs/MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="RPSA")   #laminin receptor 1 - in iPSCs/MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="PPIA")   #cyclophilin A - in iPSCs/MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="SPARC")  #osteonectin - in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="LGALS1") #galectin 1 - in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="TGFB1")  #TGFb1 - in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.full.sppo, features="TGFB2")  #TGFb2 - in MSCs/osteoblasts
FeaturePlot(h.full.sppo, features="TGFB2")  #TGFb3 - in chondrocytes
FeaturePlot(h.full.sppo, features="SPP1")   #osteopontin - in iPSCs
FeaturePlot(h.full.sppo, features="MMP1")   #matrix metalloproteinase 2 - in MSCs
FeaturePlot(h.full.sppo, features="MMP2")   #matrix metalloproteinase 2 - in chondrocytes (human) / in iPSCs/MSCs/chondrocytes/osteoblasts (chimp)
#MMP3, MMP11, MMP14, MMP16, ID3 (inhibitor of DNA binding 3)

#pluripotency markers:  NANOG     ENSG00000111704
#                       POU5F1    ENSG00000204531
#                       TDGF1
#mesoderm markers:      MIXL1     ENSG00000185155
#                       TBXT      ENSG00000164458
#endomesoderm marker:   EOMES     ENSG00000163508
#chondrocyte markers:   ACAN      ENSG00000157766
#                       COL2A1    ENSG00000139219
#                       SOX9      ENSG00000125398
#                       COL10A1   ENSG00000123500
#                       COL11A1   ENSG00000060718
#                       SOX5      ENSG00000134532
#                       SOX6      ENSG00000110693
#osteoblast markers:    COL1A1    ENSG00000108821
#                       BGLAP     -----
#                       RUNX2     ENSG00000124813
#                       ALPL      ENSG00000162551
#                       IBSP      ENSG00000029559
#                       SP7       ENSG00000170374
#adipocyte markers:     FABP4     ENSG00000170323
#                       PPARG     ENSG00000132170
#endoderm marker:       SOX17     ENSG00000164736
#neuroectoderm marker:  SOX1      ENSG00000182968
#neuroepithelial:       SOX2      ENSG00000181449
#epithelial marker:     EPCAM     ENSG00000119888
#hepatocyte marker:     ALB       ENSG00000163631
#heart/NC marker:       HAND1     ENSG00000113196
#cardiomyocyte marker:  TNNT2     ENSG00000118194
#cariac muscle markers: MYL7      ENSG00000106631
#                       MYH6      ENSG00000197616
#neutral markers:       GAPDH     ENSG00000111640
#                       GUSB      -----
#                       YWHAZ     ENSG00000164924

#Should be positive in iPSC/MSC: THY1 (CD90)
#Should be positive in MSC: ENG (CD105), NT5E (CD73)
#Can also be positive in MSC: CD44, ITGB1 (CD29), CD200, ALCAM (CD166), MCAM (CD146), NGFR (CD271)
#Should be negative in MSCs: CD34, PTPRC (CD45), ITGAM (CD11b), CD19, HLA-DRB1 (HLA-DR)
#Should also be negative in MSCs: CD14, CD79A (CD79α)

```

Neural lineage markers
Neuroepithelial cells
SOX2
NES
NOTCH1
HES1
Schwann cell precursor
SOX10
GAP43
FABP7
MPZ
DHH
NGFR
Myelinating Schwann cells
SOX10
S100A1
EGR2
MPZ
MBP
Non myelinating Schwann cells
SOX10
S100A1
GAP43
NCAM1
NGFR
Radial Glia
VIM
PAX6
HES1
HES5
TNC
SLC1A3
Oligodendroctye precursors
PDGFRA
CSPG4
Oligodendroctyes
OLIG1
OLIG2
OLIG3
MOG
CLDN11
Astrocytes
GFAP
SLC1A3
SLC1A2
GLUL
S100B
ALDH1L1
Intermediate progenitors
EOMES
ASCL1
Immature Neurons
DCX
NEUROD1
TBR1
TUBB3
STMN1
Mature Neurons
MAP2
RBFOX3
NEFM
NEFH
DLG4
Glutaminergic
SLC17A7
SLC17A6
GRIN1
GRIN2B
GABAergic
SLC6A1
GABBR1
GABBR2
GAD1
GAD2
Dopaminergic
TH
SLC6A3
FOXA2
KCNJ6
NR4A2
LMX1B

Ectoderm markers
VIM
EMX2
CDH2
SOX11
TTR
MARCKS
FABP7
TMSB15A

Mesoderm Markers
Cardiomyoctyes precursors
MESP1
GATA4
GATA6
NKX2-5
NPPA
ISL1

Cardiac genes
TNNT2
MYH6
SIRPA
CD47
GJA1
DES
Skeletal muscle
PAX3
PAX7
MYF5
MYOD1
MYOG
MYF6
MYH3
MYH7
Endothelial cells
KDR
CD34
Adipocytes
SLC7A10
SLC36A2
P2RX5
MYF5
EBF2
PDGFRA
FGF10
FGF16
FGF19

Endoderm
Hepatocytes
ALB
HNF4A
AFP
Enterocytes
KRT18
EPCAM
MGAM
Globlet cells
KRT1
EGFR
ERBB2
ERBB3
Airway epithelium
SCGB3A2
SCGB3A2
SCGB3A1
TP63
KRT5
NKX2-1

```{r differential expression}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))

#Find differentially expressed features that define clusters (cluster biomarkers)

#notes: Seurat identifes positive and negative markers of a single cluster compared to all other cells by default
#notes: you can designate which groups of clusters vs. each other, or against all cells
#notes: FindAllMarkers automates this process (identifying positive and negative markers) for all clusters

#notes: min.pct argument requires a feature to be detected at a minimum percentage in either of the two groups of cells
#notes: thresh.test argument requires a feature to be differentially expressed (on average) by some amount between the two groups
#notes: max.cells.per.ident can be set to downsample each identity class to have no more cells than whatever this is set to (option to speed up computations)
#notes: test.use parameter can designate which of several tests for differential expression to use

#find all markers distinguishing iPSC clusters from other cell types
#h.full.sppo: iPSC clusters (clusters 0+7)
#c.full.sppo: iPSC clusters (clusters 0+8+9)
h.ipsc <- FindMarkers(h.full.sppo, ident.1=c(0,7), min.pct=0.5, test.use='wilcox') #use negbinom or MAST test instead?
c.ipsc <- FindMarkers(c.full.sppo, ident.1=c(0,8,9), min.pct=0.5, test.use='wilcox')
head(h.ipsc, n=5) #CD24, POU5F1, L1TD1, TUBB2B, ACAT2 (negbinom: CD24, TUBB2B, GAL, CLDN6, ACAT2 / MAST: CD24, POU5F1, L1TD1, TUBB2B, ACAT2)
head(c.ipsc, n=5) #CD24, POU5F1, L1TD1, MDK, SNRPN
dim(h.ipsc) #2403 genes total (negbinom: 3372 / MAST: 2403)
dim(c.ipsc) #1761 genes total
#find all markers distinguishing MSC clusters from other cell types
#h.full.sppo: MSC clusters (clusters 1+4)
#c.full.sppo: iPSC clusters (clusters 1+3)
h.msc <- FindMarkers(h.full.sppo, ident.1=c(1,4), min.pct=0.5, test.use='wilcox')
c.msc <- FindMarkers(c.full.sppo, ident.1=c(1,3), min.pct=0.5, test.use='wilcox')
head(h.msc, n=5) #RGS4, SERPINE1, MT2A, TAGLN2, LGALS1
head(c.msc, n=5) #CRYAB, IGFBP5, TPM1, LGALS1, TAGLN2
dim(h.msc) #840 genes total
dim(c.msc) #677 genes total
#find all markers distinguishing chondrocyte clusters from other cell types
#h.full.sppo: chondrocyte clusters (clusters 2) - excluding cluster 6
#c.full.sppo: chondrocyte clusters (clusters 2)
h.chondro <- FindMarkers(h.full.sppo, ident.1=2, min.pct=0.5, test.use='wilcox')
c.chondro <- FindMarkers(c.full.sppo, ident.1=2, min.pct=0.5, test.use='wilcox')
head(h.chondro, n=5) #POSTN, LUM, PTGDS, TIMP3, COL3A1
head(c.chondro, n=5) #LUM, COL3A1, COL11A1, POSTN, MFAP4
dim(h.chondro) #1496 genes total
dim(c.chondro) #1416 genes total
#find all markers distinguishing osteoblast clusters from other cell types
#h.full.sppo: osteoblast clusters (clusters 3+5) - excluding cluster 6
#c.full.sppo: osteoblast clusters (clusters 4+5+6)
h.osteo <- FindMarkers(h.full.sppo, ident.1=c(3,5), min.pct=0.5, test.use='wilcox')
c.osteo <- FindMarkers(c.full.sppo, ident.1=c(4,5,6), min.pct=0.5, test.use='wilcox')
head(h.osteo, n=5) #ACTA2, DSTN, ramp1, COL8A1, CCDC80
head(c.osteo, n=5) #MT2A, LGALS3, MME, S100A4, SPON2
dim(h.osteo) #981 genes total
dim(c.osteo) #834 genes total

#Visualize marker expression in more detail

FeaturePlot(h.full.sppo, features=c("CD24","POU5F1","L1TD1","TUBB2B","ACAT2")) #markers distinguishing iPSC clusters other cell types
FeaturePlot(h.full.sppo, features=c("RGS4","SERPINE1","MT2A","TAGLN2","LGALS1")) #markers distinguishing MSC clusters other cell types
FeaturePlot(h.full.sppo, features=c("ACTA2","ITGA11","DSTN","COL8A1","CCDC80")) #markers distinguishing osteoblasts clusters other cell types
FeaturePlot(h.full.sppo, features=c("LUM","POSTN","PTGDS","TIMP3","COL3A1")) #markers distinguishing chondrocytes clusters other cell types

FeaturePlot(c.full.sppo, features=c("CD24","POU5F1","L1TD1","MDK","SNRPN")) #markers distinguishing iPSC clusters other cell types
FeaturePlot(c.full.sppo, features=c("CRYAB","IGFBP5","TPM1","LGALS1","TAGLN2")) #markers distinguishing MSC clusters other cell types
FeaturePlot(c.full.sppo, features=c("MT2A","LGALS3","MME","S100A4","SPON2")) #markers distinguishing osteoblasts clusters other cell types
FeaturePlot(c.full.sppo, features=c("LUM","S100A4","COL3A1","COL1A1","MFAP4")) #markers distinguishing chondrocytes clusters other cell types

#Assess overlap of human and chimp cell type markers
ipsc = intersect(rownames(h.ipsc),rownames(c.ipsc)) #1204 genes
msc = intersect(rownames(h.msc),rownames(c.msc)) #332 genes
osteo = intersect(rownames(h.osteo),rownames(c.osteo)) #343 genes
chondro = intersect(rownames(h.chondro),rownames(c.chondro)) #609 genes

```

```{r cell atlas assignment}

###SCRIPTS GOTTEN FROM BINGQING XIE###

###compute correlation matrix between sample and cell lines in cellAtlas
###object: Seurat object
###cell.line: cell Atlas expression matrix
###CorMethod: correlation method to use
ComputeCorMat=function(object,cell.line,CorMethod = "pearson"){

	data=GetAssayData(object)
    data_norm <- apply(data, 2, function(x) x/sum(x))
    data_libsize <- apply(data_norm, 2, function(x) x * 1e+06)

    Data.use <- data_libsize

    gene.id <- rownames(Data.use)
    gene.ref <- rownames(cell.line)
    common <- intersect(gene.id, gene.ref)
    logxx <- apply(Data.use[common, ], 2, function(x) {
        log(x + 0.1)
    })
    selected.cell.line <- apply(cell.line[common, ], 2, function(x) {
        x - mean(x)
    })
    n <- ncol(logxx)
    m <- ncol(cell.line)
    cor.mat <- matrix(0, nrow = n, ncol = m)
    for (j in 1:m) {
        for (i in 1:n) {
            cor.mat[i, j] <- cor(logxx[, i], selected.cell.line[, 
                j], method = CorMethod)
        }
    }
    rownames(cor.mat) <- colnames(Data.use)
    colnames(cor.mat) <- colnames(cell.line)
	return(cor.mat)
}

###Assign clustered cell atlas as cell type to all cells in the sample
###object: seurat object
###cor.mat: correlation matrix
###K: number of clusters generated from cell atlas
###topCorrelation: number of highest correlated cell lines in cell atlas to be assigned, set to 1 to extract only the top correlated cell line
AssignClusteredLabel=function (object,cor.mat,K=10,topCorrelation=1, dist.method="euclidean", hclust.method="ward.D"){
	
	dist_mat <- dist(t(cor.mat), method = dist.method)
	hclust_avg_cellline <- hclust(dist_mat, method = hclust.method)
	#plot(hclust_avg_cellline)

	cut_cellline <- cutree(hclust_avg_cellline, k = K)
	cellline_abstract=sapply(strsplit(as.character(colnames(cor.mat)), "\\_"), "[[", 2)
	clusteredCellline=rbind(cut_cellline,cellline_abstract)
	nametable=rep('n',K)
	for(i in 1:K){
		indexy=as.character(i)
		nametable[i]=paste0(names( sort(table(clusteredCellline[2,clusteredCellline[1,]==indexy]),decreasing=TRUE)),collapse='_' )
	}
	assignm=apply(cor.mat,1,function(x){
		indexy=names(which.max(table(clusteredCellline[1,names(sort(x,decreasing=T)[1:topCorrelation])])))
		y=paste0(names( sort(table(clusteredCellline[2,clusteredCellline[1,]==indexy]),decreasing=TRUE)),collapse='_' )
		return(y)
	})
	object$clusteredCellAtlas=assignm
	return(object)
}

#load seurat data
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))

#load cell atlas data matrix (currated in Mengjie's lab)
#references: Mabbott et al. BMC Genomics 2013; Wu C et al. Nucleic acids research 2016
load("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cell_atlas_ref_panel")

#Compute the correlation matrix use a seurat object and cell line matrix as input
cor.mat.h=ComputeCorMat(h.full.sppo,cell.line)
cor.mat.c=ComputeCorMat(c.full.sppo,cell.line)

#Add a metadata column "clusteredCellAtlas" to the object
h.full.sppo <- AssignClusteredLabel(h.full.sppo,cor.mat.h)
c.full.sppo <- AssignClusteredLabel(c.full.sppo,cor.mat.c)

#Visualize assigned clusters
CombinePlots(plots=list(DimPlot(h.full.sppo, reduction="umap", label=TRUE),
                        DimPlot(c.full.sppo, reduction="umap", label=TRUE)), ncol=2)

CombinePlots(plots=list(DimPlot(h.full.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(c.full.sppo, reduction="umap", group.by="orig.ident")), ncol=2)

CombinePlots(plots=list(DimPlot(h.full.sppo, reduction="umap", group.by="clusteredCellAtlas"),
                        DimPlot(c.full.sppo, reduction="umap", group.by="clusteredCellAtlas")), ncol=2)

```

```{r extra plots}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.sppo.rds"))
c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-c.full.sppo.rds"))

#ipsc
FeaturePlot(h.full.sppo, features="NANOG", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="POU5F1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="TDGF1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(h.full.sppo, features=c("NANOG","POU5F1","TDGF1"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#msc
FeaturePlot(h.full.sppo, features="THY1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD90 - positive in iPSC+MSC
FeaturePlot(h.full.sppo, features="ENG", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD105 - positive in MSC
FeaturePlot(h.full.sppo, features="NT5E", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD73 - positive in MSC
FeaturePlot(h.full.sppo, features="CD44", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD44 - can be positive
VlnPlot(h.full.sppo, features=c("THY1","ENG","NT5E","CD44"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#chondrocyte
FeaturePlot(h.full.sppo, features="COL2A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="COL10A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="COL11A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="SOX5", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(h.full.sppo, features=c("COL2A1","COL10A1","COL11A1","SOX5"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#osteoblast
FeaturePlot(h.full.sppo, features="COL1A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="RUNX2", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="ALPL", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(h.full.sppo, features="BGLAP", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(h.full.sppo, features=c("COL1A1","RUNX2","ALPL","BGLAP"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)

#ipsc
FeaturePlot(c.full.sppo, features="NANOG", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="POU5F1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="TDGF1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(c.full.sppo, features=c("NANOG","POU5F1","TDGF1"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#msc
FeaturePlot(c.full.sppo, features="THY1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD90 - positive in iPSC+MSC
FeaturePlot(c.full.sppo, features="ENG", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD105 - positive in MSC
FeaturePlot(c.full.sppo, features="NT5E", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD73 - positive in MSC
FeaturePlot(c.full.sppo, features="CD44", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap") #CD44 - can be positive
VlnPlot(c.full.sppo, features=c("THY1","ENG","NT5E","CD44"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#chondrocyte
FeaturePlot(c.full.sppo, features="COL2A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="COL10A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="COL11A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="SOX5", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(c.full.sppo, features=c("COL2A1","COL10A1","COL11A1","SOX5"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)
#osteoblast
FeaturePlot(c.full.sppo, features="COL1A1", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="RUNX2", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="ALPL", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
FeaturePlot(c.full.sppo, features="BGLAP", cols=c("grey90","black"), pt.size=0.5, order=TRUE, reduction="umap")
VlnPlot(c.full.sppo, features=c("COL1A1","RUNX2","ALPL","BGLAP"), group.by="orig.ident", cols=c("pink","mediumorchid","gold","dodgerblue"), pt.size=0)

```

