---
title: "10x-plot-cormotif-cnts"
author: "Genevieve Housman"
date: "July 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Correlation Motifs

View results of correlation motifs tests between humans and chimpanzees within different cell classifications.

```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)
library(ComplexHeatmap)
library(Cormotif)

```

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate <- paste0(data@meta.data$Individual,".",data@meta.data$Replicate)

```

```{r}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

```{r}

#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

#Define function to plot GO functional enrichment results
goEnPlot <- function(goTot, title) {
  print(ggplot(goTot, aes(x=GeneRatio, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          facet_grid(ONTOLOGY~., scale="free") +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

#Define function to plot GO functional enrichment results (specific ontology only)
goEnPlot.spec <- function(goTot, title, ontology) {
  print(ggplot(goTot[which(goTot$ONTOLOGY==ontology),], aes(x=GeneRatio, y=Description, color=p.adjust, size=Count)) +
          geom_point() +
          theme(axis.text.x=element_text(angle=-40, hjust=0)) +
          scale_size_area(max_size=6) +
          scale_color_viridis(option="C") +
          theme_bw() +
          labs(title=title, x="Gene Ratio", y=""))
}

```

## Load External Gene Datasets

Several external gene datasets were manually compiled from papers to test for enrichment in interesting DE gene sets.

Genes previously identified as human and chimpanzee differences:

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 0 (n = 3267 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 1 (n = 3239 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 2 (n = 3477 DE genes at FDR < 1% out of 10304 genes)
    + DE genes in human and chimpanzee iPSC-derived endoderm day 3 (n = 3820 DE genes at FDR < 1% out of 10304 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each cell type
    
* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + DE genes in human and chimpanzee heart tissue (n = 2195 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee kidney tissue (n = 799 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee liver tissue (n = 1364 DE genes at FDR < 1% out of 12184 genes)
    + DE genes in human and chimpanzee lung tissue (n = 805 DE genes at FDR < 1% out of 12184 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + Note: only considered DE genes unique to each tissue type

* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + DE genes in parental iPSC lines (n = 7655)
    + DE genes in hybrid iPSC lines (n = 6009)
    + DE genes in parental CNCC lines (n = 2223)
    + DE genes in hybrid CNCC lines (n = 3612)
    + Note: only considered DE genes unique to each cell type
    
* Gokhman et al. 2020 Nature Communications (Supplementary Data 2)
    + DMRs identified between chimpanzees and archaic/modern humans in bone tissues (n = 2031 DMRs)
    + AMH-derived methylation patterns in bone tissues (n = 873 DMRs linked with 588 DMGs)
    + (human hypermethylated promotor regions: expect higher gene expression in chimps as compared to humans)
    + (chimp hypermethylated promotor regions: expect higher gene expression in humans as compared to chimps)
    + Note: only DMRs that overlapped promotor regions of genes were included
    + Note: hyper/hypomethylation determined by comparing average AMH methylation levels with average chimp methylation levels

General skeleton-related and skeletal phenotype-related genes:

* Hsu et al. 2019 Journal of Bone and Mineral Research (Table 1)
    + hip geometry GWAS loci (n = 10)

* Styrkarsdottir et al. 2019 Nature Communications (Table 1)
    + DXA bone area GWAS loci (n = 21 associated with 20 genes)

* Yengo et al. 2018 Human Molecular Genetics
    + height GWAS loci (n = 3290) (data provided at https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files)
    + height GWAS loci with associated eQTLs (n = 610) (list not provided)
    + Note: only used unique gene symbols associated with height GWAS loci (n = 1500)

Genes associated with skeletal diseases (OA + OP):

* Steinberg et al. 2020 bioRxiv (Supplementary Table 2)
    + genes with cross-omics (DE RNA + protein) evidence for involvement in OA progression (n = 409)
    
* Morris et al. 2019 Nature Genetics (Supplemental Table 2)
    + OP GWAS loci (n = 1103 conditionally independent signals associated with 863 genes)
    
Some external gene datasets were initially exmained in relation to DE genes identified across Stages and Ostadhoc Classifications (4 genes, flexible), but were subsequently removed from analyses for reasons specified below:

* Gallego Romero et al. 2015 eLife (Supplementary File 3)
    + DE genes in human and chimpanzee iPSCs (n = 4605 DE genes at FDR < 1% out of 12171 genes)
    + Note: only included genes with a known gene symbol (retreived from https://www.biotools.fr/human/ensembl_symbol_converter)
    + REMOVED because subsequent studies (Blake et al. 2018 & Gokhman et al. 2021) also examine human and chimpanzee iPSCs and because these studies find that some iPSC-related DE genes overlap with other cell types (endoderm & CNCCs, respectively) but since this study only examined iPSCs subsetting to unique DE gene is not clear

* Blake et al. 2018 Genome Biology (Supplementary Table S5A-D)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Blake et al. 2020 Genome Research (Supplemental Table S4)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each tissue type
    
* Gokhman et al. 2021 Nature Genetics (Supplementary Table 5)
    + REMOVED comparisons of all DE genes in favor of DE genes unique to each cell type

* Gokhman et al. 2020 Nature Communications (Supplementary Data 7)
    + Skeleton-related genes (n = 1790)
    + Note: it is unclear where these genes came from (maybe just genes associated with HPO skeletal divergent traits)
    + REMOVED because it is unclear where these genes came from to interpreting biology is difficult

Some external gene datasets were considered but not examined:

* Ward et al. 2019 eLife
    +  DE genes in human and chimpanzee iPSC-derived cardiomyocytes (condition A: 6378 DE genes at FDR < 10% out of 11974 genes)

* Shibata et al. 2012 PLOS Genetics
    + increased chromatin accessibility (DNase-I) in human iPSCs (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp iPSCs (expect higher gene expression in chimps as compared to humans)

* Edsall et al. 2019 Genome Biology and Evolution
    + increased chromatin accessibility (DNase-I) in human fibroblasts (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (DNase-I) in chimp fibroblasts (expect higher gene expression in chimps as compared to humans)

* Swain-Lenz et al. 2019 Genome Biology and Evolution (none of the supplements list the 732 human-increased, 782 human-decreased, 1012 chimp-increased, or 466 chimp-decreased states)
    + increased chromatin accessibility (ATAC-seq) in human adipose (expect higher gene expression in humans as compared to chimps)
    + increased chromatin accessibility (ATAC-seq) in chimp adipose (expect higher gene expression in chimps as compared to humans)

* Prescott et al. 2015 Cell (Supplemental)
    + top human-biased CNCC enhancers (n = 1000) (expect higher gene expression in humans as compared to chimps)
    + top chimp-biased CNCC enhancers (n = 1000) (expect higher gene expression in chimps as compared to humans)

* Joganic et al. 2017 American Journal of Physical Anthropology
    + genetics of craniofacial skeleton in baboons

* CibriÃ¡n Uhalte et al. 2017 Human Molecular Genetics (Figure 2)
    + robustly established OA genetic loci from GWAS and candidate gene-based approaches (n = 21 associated with 24 nearest genes)

* Tachmazidou et al. 2019 Nature Genetics
    + OA loci identified in study (Table 1) (n = 64 [52 novel + 12 previously identified])
    + OA loci (Supplementary Table 2)
    + established OA loci (Supplementary Table 4) (n = 34 associated with 42 genes)
    + Note: used loci compiled by AH
    + (OA.GWAS.eQTL = genes with OA GWAS signals and an eQTL signal in at least one tissue)
    + (OA.estab = gene symbol(s) nearest to established OA loci)
    + (OA.broad = genes within 500kb of OA loci)

```{r}

#Define function for testing significant enrichment of gene sets

genesetEnrich <- function(GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test
  print(ContingencyTable)
  print(fisher.test(ContingencyTable, alternative="greater"))

}

genesetEnrich.data <- function(InternalDataSource, ExternalDataSource, GenesOfInterest, DEgenes, NonDEGenes){
  
  #fill contingency table
  DE.Interest       <- length(intersect(GenesOfInterest, DEgenes))
  NonDE.Interest    <- length(intersect(GenesOfInterest, NonDEGenes))
  DE.NotInterest    <- length(DEgenes) - DE.Interest
  NonDE.NotInterest <- length(NonDEGenes) - NonDE.Interest
  
  ContingencyTable <- matrix(c(DE.Interest,
                               NonDE.Interest,
                               DE.NotInterest,
                               NonDE.NotInterest),
                             nrow=2,
                             ncol=2,
                             byrow=TRUE)
  colnames(ContingencyTable) <- c("DE Genes","Non DE Genes")
  rownames(ContingencyTable) <- c("Genes of Interest","Not Genes of Interest")
  
  #perform test and output data
  data <- data.frame("Source.DEGenes"=InternalDataSource,
                     "Source.GenesOfInterest"=ExternalDataSource,
                     "DE.Interest"=DE.Interest,
                     "NonDE.Interest"=NonDE.Interest,
                     "DE.NotInterest"=DE.NotInterest,
                     "NonDE.NotInterest"=NonDE.NotInterest,
                     "p.value"=fisher.test(ContingencyTable, alternative="greater")$p.value,
                     "odds.ratio"=fisher.test(ContingencyTable, alternative="greater")$estimate,
                     row.names=NULL)
  return(data)

}

```

```{r}

#Load external datasets (manually currated from papers)
gallegoromero2015    <- read.csv("./data/external-gene-sets/GallegoRomero2015eLife.csv")
blake2018            <- read.csv("./data/external-gene-sets/Blake2018GenBio.csv")
blake2020            <- read.csv("./data/external-gene-sets/Blake2020GenRes.csv")
gokhman2021          <- read.csv("./data/external-gene-sets/Gokhman2021NatGen.csv")
gokhman2020          <- read.csv("./data/external-gene-sets/Gokhman2020NatComm.csv")
gokhman2020.skeleton <- read.csv("./data/external-gene-sets/Gokhman2020NatComm_skeleton.csv")
hsu2019              <- read.csv("./data/external-gene-sets/Hsu2019JBoneMinRes.csv")
styrkarsdottir2019   <- read.csv("./data/external-gene-sets/Styrkarsdottir2019NatComm.csv")
yengo2018            <- read.csv("./data/external-gene-sets/Yengo2018HumMolGen.csv")
steinberg2020        <- read.csv("./data/external-gene-sets/Steinberg2020biorxiv.csv")
morris2019           <- read.csv("./data/external-gene-sets/Morris2019NatGen.csv")
#cibrianuhalte2017    <- read.csv("./data/external-gene-sets/CibrianUhalte2017HumMolGen.csv")
#tachmazidou2019      <- read.csv("./data/external-gene-sets/Tachmazidou2019NatGen_AH.csv")

```

```{r, eval=FALSE}

#genes previously identified as human and chimpanzee differences

#genesExternal <- gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]      #iPSCs

#genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1)]  #iPSCs
#genesExternal <- blake2018$gene[which(blake2018$DE.Day1==1)]  #iPSC-endoderm day1
#genesExternal <- blake2018$gene[which(blake2018$DE.Day2==1)]  #iPSC-endoderm day2
#genesExternal <- blake2018$gene[which(blake2018$DE.Day3==1)]  #iPSC-endoderm day3

genesExternal <- blake2018$gene[which(blake2018$DE.Day0==1 &
                                      blake2018$DE.Day1==0 &
                            					blake2018$DE.Day2==0 &
                            					blake2018$DE.Day3==0)]  #iPSCs-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==1 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day1-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==1 &
                            				  blake2018$DE.Day3==0)]  #iPSC-endoderm day2-UNIQUE
genesExternal <- blake2018$gene[which(blake2018$DE.Day0==0 &
                                      blake2018$DE.Day1==0 &
                            				 	blake2018$DE.Day2==0 &
                            				  blake2018$DE.Day3==1)]  #iPSC-endoderm day3-UNIQUE

#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1)]  #heart
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Kidney==1)] #kidney
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Liver==1)]  #liver
#genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Lung==1)]   #lung

genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #heart-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==1 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==0)]   #kidney-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==1 &
                                      blake2020$DE.HvC_Lung==0)]   #liver-UNIQUE
genesExternal <- blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                      blake2020$DE.HvC_Kidney==0 &
                                      blake2020$DE.HvC_Liver==0 &
                                      blake2020$DE.HvC_Lung==1)]   #lung-UNIQUE

#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)] #parental.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]   #hybrid.iPSCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs
#genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs

genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                               gokhman2021$DE.Parental.CNCCs==0)] #parental.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                               gokhman2021$DE.Hybrid.CNCCs==0)]), #hybrid.iPSCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                               gokhman2021$DE.Parental.CNCCs==1)] #parental.CNCCs-UNIQUE
genesExternal <- gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                               gokhman2021$DE.Hybrid.CNCCs==1)]   #hybrid.CNCCs-UNIQUE

#genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1)]       #chimp-specific bone DMRs
#genesExternal <- gokhman2020$Gene[which(gokhman2020$Human.DMR.Promotor==1)]       #human-specific bone DMRs
genesExternal <- gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                        gokhman2020$Human.DMR.Promotor==1)]       #all human-chimp DMRs

#general skeleton-related and skeletal phenotype-related genes

#genesExternal <- gokhman2020.skeleton$Skeleton.related.gene)                                                #skeleton-related
genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]                                   #hip geometry GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.length==1)]                            #femoral neck length GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.narrowest.width.of.femoral.neck==1)]                #femoral neck width GWAS
#genesExternal <- hsu2019$Nearby.gene[which(hsu2019$GWAS.femoral.neck.section.modulus==1)]                   #femoral neck area GWAS
genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]               #bone area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.lumbar.spine==1)]            #lumbar spine area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.total.hip==1)]               #total hip area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.intertrochanteric.shaft==1)] #intertrochanteric shaft area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.trochanter==1)]              #trochanter area GWAS
#genesExternal <- styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.femoral.neck==1)]            #femoral neck area GWAS
genesExternal <- yengo2018$gene[which(yengo2018$GWAS.Height==1)]                                            #height GWAS

#genes associated with skeletal diseases (OA + OP)

genesExternal <- steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]     #OA GWAS
#genesExternal <- cibrianuhalte2017$gene[which(cibrianuhalte2017$OA.gene==1)]  #OA GWAS
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.GWAS.eQTL==1)] #OA GWAS - overlapping eQTL
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.estab==1)]     #OA GWAS - previously established
#genesExternal <- tachmazidou2019$gene[which(tachmazidou2019$OA.broad==1)]     #OA GWAS - within 500kb of OA loci
genesExternal <- morris2019$unique.gene[which(morris2019$GWAS.BMD==1)]        #OP GWAS

```

## Stages of Differentiation at Collection

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Stage=c("Time 0","Time 1","Time 2"),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("Time.0",length(which(dataUpset$Time.0==1))),
                                       rep("Time.1",length(which(dataUpset$Time.1==1))),
                                       rep("Time.2",length(which(dataUpset$Time.2==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$Time.0==1)],
                                rownames(dataUpset)[which(dataUpset$Time.1==1)],
                                rownames(dataUpset)[which(dataUpset$Time.2==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$Time.0==1),"Time.0"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.1==1),"Time.1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Time.2==1),"Time.2"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("Time.0","Time.1","Time.2"),
                               labels=c("Time 0","Time 1","Time 2"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0"),
                         list("Time.1"),
                         list("Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.1","Time.2"),
                         list("Time.0","Time.2"),
                         list("Time.0","Time.1","Time.2")),
      text.scale=2)

```

```{r, eval=FALSE}

#Save DE gene lists
write.csv(rownames(dataUpset),"./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t01.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.stg.t12.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                    dataUpset$Time.1==0 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t0.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==1 &
                                    dataUpset$Time.2==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t1.csv")
write.csv(rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                    dataUpset$Time.1==0 &
                                    dataUpset$Time.2==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.stg.t2.csv")

```

Assess GO functional enrichment of interesting DE genes.

```{r}

#Shared across t0+t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across All Stages")

#Shared across t0+t1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 0 & Time 1")

#Shared across t1+t2
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Shared across Time 1 & Time 2")

#Unique to Time 0
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 0")

#Unique to Time 1
subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="DE Genes Uniqe to Time 1")

#Unique to Time 2
#subGenes <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
#                                      dataUpset$Time.1==0 &
#                                      dataUpset$Time.2==1)]
#totGenes <- rownames(dataUpset)
#goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="DE Genes Uniqe to Time 2")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

```{r, eval=FALSE}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #Shared across All Stages
  print("Shared across All Stages")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 0 & Time 1
  print("Shared across Time 0 & Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across Time 1 & Time 2
  print("Shared across Time 1 & Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Unique to Time 0
  print("Unique to Time 0")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 1
  print("Unique to Time 1")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Unique to Time 2
  print("Unique to Time 2")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 0 & Time 1
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 0 & Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across Time 1 & Time 2
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Time 1 & Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to Time 0
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 0",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 1
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Time 2
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Time 2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared in All Stages (tested in all classifications)
  print("Shared in All Stages, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Shared in Time 0 & Time 1 (tested in all classifications)
  print("Shared in Time 0 & Time 1 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared in Time 1 & Time 2 (tested in all classifications)
  print("Shared in Time 1 & Time 2, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 0 (tested in all classifications)
  print("Unique to Time 0 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==1 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to Time 1 (tested in all classifications)
  print("Unique to Time 1 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==1 &
                                      dataUpset$Time.2==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to Time 2 (tested in all classifications)
  print("Unique to Time 2 (tested in all classifications)")
  tempDE <- rownames(dataUpset)[which(dataUpset$Time.0==0 &
                                      dataUpset$Time.1==0 &
                                      dataUpset$Time.2==1)]
  print(intersect(genesExternal[[2]],tempDE))

}

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("Time 0","Time 1","Time 0","Time 1"),
                                       c("Time 1","Time 2","Time 1","Time 2")),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9655 genes) -> 83.38%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/11579

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

#DE genes
plotMotif(cormotifData)

```

## General Clusters (res=0.05)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Cluster=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "iPSC.c2"="pink4",
                                              "iPSC.c3"="grey",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        column_split=c(1,1,1,2,2,2),
        #column_split=c(1,1,2,3,3,4),
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))
```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC.c1",length(which(dataUpset$iPSC.c1==1))),
                                       rep("MSC.c1",length(which(dataUpset$MSC.c1==1))),
                                       rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),"iPSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),"MSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                               labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Cluster",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1"),
                         list("MSC.c1"),
                         list("Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1"),
                         list("MSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1","Osteogenic.c1")),
      text.scale=2)

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC.c1",length(which(dataUpset$iPSC.c1==1))),
                                       rep("iPSC.c2",length(which(dataUpset$iPSC.c2==1))),
                                       rep("iPSC.c3",length(which(dataUpset$iPSC.c3==1))),
                                       rep("MSC.c1",length(which(dataUpset$MSC.c1==1))),
                                       rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1))),
                                       rep("Osteogenic.c2",length(which(dataUpset$Osteogenic.c2==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c2==1)],
                                rownames(dataUpset)[which(dataUpset$iPSC.c3==1)],
                                rownames(dataUpset)[which(dataUpset$MSC.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c1==1),"iPSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c2==1),"iPSC.c2"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$iPSC.c3==1),"iPSC.c3"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC.c1==1),"MSC.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),"Osteogenic.c2"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                               labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Cluster",
                    values=c("pink","pink4","grey","mediumorchid","dodgerblue","navyblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Stages
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSC.c1 & MSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in iPSC.c1 & MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSC.c1 & Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in MSC.c1 & Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==1 &
                                      dataUpset$MSC.c1==0 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to iPSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSC.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==1 &
                                      dataUpset$Osteogenic.c1==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to MSC.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC.c1==0 &
                                      dataUpset$MSC.c1==0 &
                                      dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteo.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","MSC.c1"),
                                         c("MSC.c1","Osteogenic.c1","MSC.c1","Osteogenic.c1")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Clusters",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c1: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9989 genes) -> 92.99%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/10742

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

```

## General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("iPSC","MSC","Osteogenic"),
                           col=list(AdHoc=c("iPSC"="pink","MSC"="mediumorchid","Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("iPSC",length(which(dataUpset$iPSC==1))),
                                       rep("MSC",length(which(dataUpset$MSC==1))),
                                       rep("Osteogenic",length(which(dataUpset$Osteogenic==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$iPSC==1)],
                                rownames(dataUpset)[which(dataUpset$MSC==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$iPSC==1),"iPSC"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$MSC==1),"MSC"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic==1),"Osteogenic"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("iPSC","MSC","Osteogenic"),
                               labels=c("iPSC","MSC","Osteogenic"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC"),
                         list("MSC"),
                         list("Osteogenic"),
                         list("iPSC","MSC"),
                         list("MSC","Osteogenic"),
                         list("iPSC","Osteogenic"),
                         list("iPSC","MSC","Osteogenic")),
      text.scale=2)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across All Ad Hoc
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across iPSCs & MSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in iPSCs & MSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across MSCs & Osteogenic Cells
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in MSCs & Osteogenic",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Unique to iPSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==1 &
                                      dataUpset$MSC==0 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to iPSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to MSCs
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==1 &
                                      dataUpset$Osteogenic==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to MSCs",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to Osteogenic Cells
  tempDE <- rownames(dataUpset)[which(dataUpset$iPSC==0 &
                                      dataUpset$MSC==0 &
                                      dataUpset$Osteogenic==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Osteogenic",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("iPSC","MSC","iPSC","MSC"),
                                       c("MSC","Osteogenic","MSC","Osteogenic")),
                           col=list(Stage=c("iPSC"="pink",
                                            "MSC"="mediumorchid",
                                            "Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for ad hoc assignments: posterior probability difference between the humans and the chimps within a motif 
#0.3 (2787 genes) -> 23.57%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/11824

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),]))
#DE genes between t0 and t1 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to chimps
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t0 and t1 unique to humans
length(rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))
#DE genes between t1 and t2 unique to humans
length(rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),]))

```

## Osteogenic Clusters (res=0.50)

### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("Osteogenic.c1",length(which(dataUpset$Osteogenic.c1==1))),
                                       rep("Osteogenic.c2",length(which(dataUpset$Osteogenic.c2==1))),
                                       rep("Osteogenic.c3",length(which(dataUpset$Osteogenic.c3==1))),
                                       rep("Osteogenic.c4",length(which(dataUpset$Osteogenic.c4==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)],
                                rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c1==1),"Osteogenic.c1"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c2==1),"Osteogenic.c2"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c3==1),"Osteogenic.c3"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$Osteogenic.c4==1),"Osteogenic.c4"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                               labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"))

#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Stage",
                    values=c("dodgerblue","deepskyblue2","dodgerblue4","navyblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Osteogenic.c1"),
                         list("Osteogenic.c3"),
                         list("Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c3"),
                         list("Osteogenic.c3","Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c4"),
                         list("Osteogenic.c1","Osteogenic.c3","Osteogenic.c4")),
      text.scale=2)

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostclusters
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &
                                      dataUpset$Osteogenic.c2==1 &
                                      dataUpset$Osteogenic.c3==1 &
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across O.c1, O.c3, O.c4
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1 &
                                      dataUpset$Osteogenic.c2==0 &
                                      dataUpset$Osteogenic.c3==1 &
                                      dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Osteogenic.c1/c3/c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))


  #All Osteogenic.c1
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c1==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c1",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c2
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c2==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c2",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c3
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c3==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c3",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All Osteogenic.c4
  tempDE <- rownames(dataUpset)[which(dataUpset$Osteogenic.c4==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteogenic.c4",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

### Examine Conservation of Differentiation-Specific DE Genes

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3",
                                           "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                                         c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                           "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#subset data
gene.prob <- cormotifData$bestmotif$p.post

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7873 genes) -> 73.78%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/10671

# Examine human/chimp differences

#DE genes unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5),]))
#DE genes between O.c1 and O.c2 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c2 and O.c3 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c3 and O.c4 unique to chimps
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5),]))
#DE genes unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c1 and O.c2 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c2 and O.c3 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))
#DE genes between O.c3 and O.c4 unique to humans
length(rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),]))

```

## Osteogenic Clusters (res=0.75)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

## Osteogenic Clusters (res=0.50) (Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

## Osteogenic Clusters (res=0.75) (Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

## Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

Note: Too few DE genes shared across pre+obl+emb+min and obl+emb+min, so these were not examined further. Instead, DE genes shared across obl+emb+min+mat and emb+min+mat were considered.

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("preosteoblast",length(which(dataUpset$preosteoblast==1))),
                                       rep("osteoblast",length(which(dataUpset$osteoblast==1))),
                                       rep("embedding.osteoblast",length(which(dataUpset$embedding.osteoblast==1))),
                                       rep("mineralizing.osteoblast",length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep("maturing.osteocyte",length(which(dataUpset$maturing.osteocyte==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),"preosteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),"osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),"embedding.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),"mineralizing.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),"maturing.osteocyte"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                               labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)

```

```{r, eval=FALSE}

#Save DE gene lists
write.csv(rownames(dataUpset),"./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oemm.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.oem.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv")
write.csv(rownames(dataUpset)[which(dataUpset$osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv")
write.csv(rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv")
write.csv(rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv")
write.csv(rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv")

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Osteogenic Ad Hoc Classifications
subGenes <- rownames(dataUpset)[which(rowSums(dataUpset)==5)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")

#Shared across obl+emb+min+mat
#subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
#                                      dataUpset$osteoblast==1 &
#                                      dataUpset$embedding.osteoblast==1 &
#                                      dataUpset$mineralizing.osteoblast==1 &
#                                      dataUpset$maturing.osteocyte==1)]
#totGenes <- rownames(dataUpset)
#goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in obl+emb+min+mat")

#Shared across obl+emb+min
#subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
#                                      dataUpset$osteoblast==1 &
#                                      dataUpset$embedding.osteoblast==1 &
#                                      dataUpset$mineralizing.osteoblast==1 &
#                                      dataUpset$maturing.osteocyte==0)]
#totGenes <- rownames(dataUpset)
#goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in obl+emb+min")

#Unique to Embedding Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Emb")

#Unique to Mineralizing Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Min")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

```{r, eval=FALSE}

#Enrichment of external gene sets in cormotif DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All preosteoblasts
  print("All preosteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All osteoblasts
  print("All osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All embedding osteoblasts
  print("All embedding osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All mineralizing osteoblasts
  print("All mineralizing osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All maturing osteocytes
  print("All maturing osteocytes")
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications
  print("Shared across all ostadhoc classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across obl+emb+min+mat
  print("Shared across obl+emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across emb+min+mat
  print("Shared across emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across obl+emb+min+mat
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Obl/Emb/Min/Mat",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Shared across emb+min+mat
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Emb/Min/Mat",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
                    
  #All preosteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Preostoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All embedding osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Embedding Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All mineralizing osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Mineralizing Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All maturing osteocytes
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Maturing Osteocytes",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print("Shared across all ostadhoc classifications, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print("Shared across pre+obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across obl+emb+min, tested in all classifications
  print("Shared across obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to preosteoblasts, tested in all classifications
  print("Unique to preosteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to osteoblasts, tested in all classifications
  print("Unique to osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to embedding osteoblasts, tested in all classifications
  print("Unique to embedding osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print("Unique to mineralizing osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to maturing osteocytes, tested in all classifications
  print("Unique to maturing osteocytes, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))

}

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene flexible, Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes

Note: Too few DE genes shared across obl+emb+min+mat, so these were not examined further. Instead, DE genes shared across pre+obl+emb+min+mat and unique to min were considered.

```{r}

#load data
cormotifData <- readRDS("./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

#plot BIC/AIC comparisons
plotIC(cormotifData)

#plot the probability that each motif is differentially expressed in each comparison
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

```

```{r}

#obtain DE pattern at 0.5 posterior distribution cutoff and reformat for plotting
#0.5 cutoff is used in Wei et al. 2015 and Blake et al. 2018
head(cormotifData$bestmotif$p.post)

dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
dataUpset[dataUpset>0.5] <- 1
dataUpset[dataUpset<=0.5] <- 0
head(dataUpset)
#dataUpset <- as.data.frame(cormotifData$bestmotif$p.post)
#dataUpset[dataUpset>0.95] <- 1
#dataUpset[dataUpset<=0.95] <- 0
#head(dataUpset)

dataPlot <- data.frame("cell.subset"=c(rep("preosteoblast",length(which(dataUpset$preosteoblast==1))),
                                       rep("osteoblast",length(which(dataUpset$osteoblast==1))),
                                       rep("embedding.osteoblast",length(which(dataUpset$embedding.osteoblast==1))),
                                       rep("mineralizing.osteoblast",length(which(dataUpset$mineralizing.osteoblast==1))),
                                       rep("maturing.osteocyte",length(which(dataUpset$maturing.osteocyte==1)))),
                       "gene"=c(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
                                rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]),
                       "post.p"=c(cormotifData$bestmotif$p.post[which(dataUpset$preosteoblast==1),"preosteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$osteoblast==1),"osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$embedding.osteoblast==1),"embedding.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$mineralizing.osteoblast==1),"mineralizing.osteoblast"],
                                  cormotifData$bestmotif$p.post[which(dataUpset$maturing.osteocyte==1),"maturing.osteocyte"]),
                       row.names=NULL)

dataPlot$cell.subset <- factor(dataPlot$cell.subset,
                               levels=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                               labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))


#number of DE genes
ggplot(dataPlot, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  xlab('') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

#total number of DE genes identified
table(dataPlot$cell.subset)

#make upset plot of DE genes
upset(dataUpset,
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
upset(dataUpset,
      keep.order=TRUE,
      empty.intersections=TRUE,
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("preosteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)

```

```{r, eval=FALSE}

#Save DE gene lists
write.csv(rownames(dataUpset),"./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-tests.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                    dataUpset$osteoblast==1 &
                                    dataUpset$embedding.osteoblast==1 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-shareX.oad.poem.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                    dataUpset$osteoblast==0 &
                                    dataUpset$embedding.osteoblast==0 &
                                    dataUpset$mineralizing.osteoblast==1 &
                                    dataUpset$maturing.osteocyte==0)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-uniqX.oad.min.csv")
write.csv(rownames(dataUpset)[which(dataUpset$preosteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.pre.csv")
write.csv(rownames(dataUpset)[which(dataUpset$osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.obl.csv")
write.csv(rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.emb.csv")
write.csv(rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.min.csv")
write.csv(rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)],
          "./data/cormotif-data/cormotif.cnts.pseudo-ostadhoc.T2.data.filterC.log.indv-cell.intNo0.reg-tot.DE-post0.50-all.oad.mat.csv")

```

Assess GO functional enrichment of interesting DE genes.

```{r, eval=FALSE}

#Shared across Osteogenic Ad Hoc Classifications
subGenes <- rownames(dataUpset)[which(rowSums(dataUpset)==5)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Shared in All")

#Shared across pre+obl+emb+min
#subGenes <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
#                                      dataUpset$osteoblast==1 &
#                                      dataUpset$embedding.osteoblast==1 &
#                                      dataUpset$mineralizing.osteoblast==1 &
#                                      dataUpset$maturing.osteocyte==0)]
#totGenes <- rownames(dataUpset)
#goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
#goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
#goEnPlot(goTot,title="Shared in obl+emb+min+mat")

#Unique to Mineralizing Osteoblasts
subGenes <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
totGenes <- rownames(dataUpset)
goTot <- goEnrich(totGenes=totGenes, deGenes=subGenes, ontology="ALL")
goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
goEnPlot(goTot,title="Unique to Min")

```

Test for significant enrichment of external gene sets in DE gene sets using Fisher's exact test.

```{r, eval=FALSE}

#Enrichment of external gene sets in cormotif DE gene sets

listsExternal <- list(

  list("GallegoRomero2015-DE.iPSCs",gallegoromero2015$gene[which(gallegoromero2015$DE.iPSC==1)]), #iPSCs

  list("Blake2018-DE.iPSCs",blake2018$gene[which(blake2018$DE.Day0==1)]),              #iPSCs
  list("Blake2018-DE.iPSC-endoderm-day1",blake2018$gene[which(blake2018$DE.Day1==1)]), #iPSC-endoderm day1
  list("Blake2018-DE.iPSC-endoderm-day2",blake2018$gene[which(blake2018$DE.Day2==1)]), #iPSC-endoderm day2
  list("Blake2018-DE.iPSC-endoderm-day3",blake2018$gene[which(blake2018$DE.Day3==1)]), #iPSC-endoderm day3
  
  list("Blake2018-DE.iPSCs-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day1-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day2-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-UNIQUE
  list("Blake2018-DE.iPSC-endoderm-day3-UNIQUE",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-UNIQUE

  list("Blake2020-DE.heart",blake2020$gene[which(blake2020$DE.HvC_Heart==1)]),   #heart
  list("Blake2020-DE.kidney",blake2020$gene[which(blake2020$DE.HvC_Kidney==1)]), #kidney
  list("Blake2020-DE.liver",blake2020$gene[which(blake2020$DE.HvC_Liver==1)]),   #liver
  list("Blake2020-DE.lung",blake2020$gene[which(blake2020$DE.HvC_Lung==1)]),     #lung
  
  list("Blake2020-DE.heart-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-UNIQUE
  list("Blake2020-DE.kidney-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-UNIQUE
  list("Blake2020-DE.liver-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-UNIQUE
  list("Blake2020-DE.lung-UNIQUE",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-UNIQUE

  list("Gokhman2021-DE.Parental.iPSCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1)]), #parental.iPSCs
  list("Gokhman2021-DE.Hybrid.iPSCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1)]),   #hybrid.iPSCs
  list("Gokhman2021-DE.Parental.CNCCs",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs
  list("Gokhman2021-DE.Hybrid.CNCCs",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs
  
  list("Gokhman2021-DE.Parental.iPSCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.iPSCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-UNIQUE
  list("Gokhman2021-DE.Parental.CNCCs-UNIQUE",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-UNIQUE
  list("Gokhman2021-DE.Hybrid.CNCCs-UNIQUE",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-UNIQUE
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Gokhman2020-skeletal.genes",gokhman2020.skeleton$Skeleton.related.gene),                                          #skeleton-related
  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])
  
  #All preosteoblasts
  print("All preosteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All osteoblasts
  print("All osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All embedding osteoblasts
  print("All embedding osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All mineralizing osteoblasts
  print("All mineralizing osteoblasts")
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #All maturing osteocytes
  print("All maturing osteocytes")
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)
  
  #Shared across all ostadhoc classifications
  print("Shared across all ostadhoc classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across obl+emb+min+mat
  print("Shared across obl+emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

  #Shared across emb+min+mat
  print("Shared across emb+min+mat")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  genesetEnrich(GenesOfInterest=genesExternal[[2]],
                DEgenes=tempDE,
                NonDEGenes=nonDEgenes)

}

```

Make enrichment plot:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Blake2018-DE.iPSCs-Unique",blake2018$gene[which(blake2018$DE.Day0==1 &
                                                        blake2018$DE.Day1==0 &
                                              					blake2018$DE.Day2==0 &
                                              					blake2018$DE.Day3==0)]),              #iPSCs-Unique
  list("Blake2018-DE.iPSC-endoderm-day1-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==1 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day1-Unique
  list("Blake2018-DE.iPSC-endoderm-day2-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==1 &
                                                          				   blake2018$DE.Day3==0)]), #iPSC-endoderm day2-Unique
  list("Blake2018-DE.iPSC-endoderm-day3-Unique",blake2018$gene[which(blake2018$DE.Day0==0 &
                                                                     blake2018$DE.Day1==0 &
                                                          				 	 blake2018$DE.Day2==0 &
                                                          				   blake2018$DE.Day3==1)]), #iPSC-endoderm day3-Unique

  list("Blake2020-DE.heart-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==1 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==0 &
                                                        blake2020$DE.HvC_Lung==0)]),   #heart-Unique
  list("Blake2020-DE.kidney-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                         blake2020$DE.HvC_Kidney==1 &
                                                         blake2020$DE.HvC_Liver==0 &
                                                         blake2020$DE.HvC_Lung==0)]), #kidney-Unique
  list("Blake2020-DE.liver-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                        blake2020$DE.HvC_Kidney==0 &
                                                        blake2020$DE.HvC_Liver==1 &
                                                        blake2020$DE.HvC_Lung==0)]),   #liver-Unique
  list("Blake2020-DE.lung-Unique",blake2020$gene[which(blake2020$DE.HvC_Heart==0 &
                                                       blake2020$DE.HvC_Kidney==0 &
                                                       blake2020$DE.HvC_Liver==0 &
                                                       blake2020$DE.HvC_Lung==1)]),     #lung-Unique

  list("Gokhman2021-DE.Parental.iPSCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==1 &
                                                                            gokhman2021$DE.Parental.CNCCs==0)]), #parental.iPSCs-Unique
  list("Gokhman2021-DE.Hybrid.iPSCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==1 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==0)]),   #hybrid.iPSCs-Unique
  list("Gokhman2021-DE.Parental.CNCCs-Unique",gokhman2021$Gene.Symbol[which(gokhman2021$DE.Parental.iPSCs==0 &
                                                                            gokhman2021$DE.Parental.CNCCs==1)]), #parental.CNCCs-Unique
  list("Gokhman2021-DE.Hybrid.CNCCs-Unique",  gokhman2021$Gene.Symbol[which(gokhman2021$DE.Hybrid.iPSCs==0 &
                                                                            gokhman2021$DE.Hybrid.CNCCs==1)]),   #hybrid.CNCCs-Unique
  
  list("Gokhman2020-DMR.bone",gokhman2020$Gene[which(gokhman2020$Chimp.DMR.Promotor==1 |
                                                     gokhman2020$Human.DMR.Promotor==1)]), #all human-chimp DMRs

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

DEenrich <- data.frame("Source.DEGenes"=character(),
                       "Source.GenesOfInterest"=character(),
                       "DE.Interest"=integer(),
                       "NonDE.Interest"=integer(),
                       "DE.NotInterest"=integer(),
                       "NonDE.NotInterest"=integer(),
                       "p.value"=double(),
                       "odds.ratio"=double())

for (genesExternal in listsExternal) {

  #Shared across all ostadhoc classifications
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in All",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #Shared across pre+obl+emb+min
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Shared in Pre/Obl/Emb/Min",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

  #Unique to min
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="Unique to Min",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
                    
  #All preosteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Preostoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All embedding osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$embedding.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Embedding Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All mineralizing osteoblasts
  tempDE <- rownames(dataUpset)[which(dataUpset$mineralizing.osteoblast==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Mineralizing Osteoblasts",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))
  
  #All maturing osteocytes
  tempDE <- rownames(dataUpset)[which(dataUpset$maturing.osteocyte==1)]
  nonDEgenes <- rownames(dataUpset)[which(!(rownames(dataUpset) %in% tempDE))]
  DEenrich <- rbind(DEenrich,
                    genesetEnrich.data(InternalDataSource="All Maturing Osteocytes",
                                       ExternalDataSource=genesExternal[[1]],
                                       GenesOfInterest=genesExternal[[2]],
                                       DEgenes=tempDE,
                                       NonDEGenes=nonDEgenes))

}

#Reformat Data
DEenrich$Source.DEGenes <- as.factor(DEenrich$Source.DEGenes)
DEenrich$Source.GenesOfInterest <- as.factor(DEenrich$Source.GenesOfInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total DE Genes
DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$DE.NotInterest)

#Calculate Ratio of Overlapping DE Genes vs. Total Genes (do not do this)
#DEenrich$GeneRatio <- DEenrich$DE.Interest / sum(DEenrich$DE.Interest,DEenrich$NonDE.Interest,DEenrich$DE.NotInterest,DEenrich$NonDE.NotInterest)

#Plot Enrichments
ggplot(DEenrich, aes(x=GeneRatio, y=Source.GenesOfInterest, color=p.value, size=DE.Interest)) +
          geom_point() +
          facet_grid(~Source.DEGenes) +
          scale_size_area(max_size=10) +
          scale_color_viridis(option="C", limits=c(0,0.05)) +
          theme_bw() +
          labs(x="Gene Ratio", y="") +
          xlim(0,0.017) +
          scale_y_discrete(limits=rev)

```

Determine which skeletal-phenotype-related genes overlap with DE genes:

```{r}

#Enrichment of external gene sets in DE gene sets

listsExternal <- list(

  list("Hsu2019-GWAS.hip.geometry",hsu2019$Nearby.gene[which(hsu2019$GWAS.hip.geometry==1)]),                             #hip geometry GWAS
  list("Styrkarsdottir2019-GWAS.bone.area",styrkarsdottir2019$Closest.Gene[which(styrkarsdottir2019$GWAS.bone.area==1)]), #bone area GWAS
  list("Yengo2018-GWAS.height",yengo2018$gene[which(yengo2018$GWAS.Height==1)]),                                          #height GWAS

  list("Steinberg2020-GWAS.OA",steinberg2020$GeneName[which(steinberg2020$OA.omics==1)]), #OA GWAS
  list("Morris2020-GWAS.OP",morris2019$unique.gene[which(morris2019$GWAS.BMD==1)])        #OP GWAS

)

for (genesExternal in listsExternal) {
  
  print(genesExternal[[1]])

  #Shared across all ostadhoc classifications, tested in all classifications
  print("Shared across all ostadhoc classifications, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across pre+obl+emb+min, tested in all classifications
  print("Shared across pre+obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Shared across obl+emb+min, tested in all classifications
  print("Shared across obl+emb+min, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to preosteoblasts, tested in all classifications
  print("Unique to preosteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==1 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to osteoblasts, tested in all classifications
  print("Unique to osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==1 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))

  #Unique to embedding osteoblasts, tested in all classifications
  print("Unique to embedding osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==1 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to mineralizing osteoblasts, tested in all classifications
  print("Unique to mineralizing osteoblasts, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==1 &
                                      dataUpset$maturing.osteocyte==0)]
  print(intersect(genesExternal[[2]],tempDE))
  
  #Unique to maturing osteocytes, tested in all classifications
  print("Unique to maturing osteocytes, tested in all classifications")
  tempDE <- rownames(dataUpset)[which(dataUpset$preosteoblast==0 &
                                      dataUpset$osteoblast==0 &
                                      dataUpset$embedding.osteoblast==0 &
                                      dataUpset$mineralizing.osteoblast==0 &
                                      dataUpset$maturing.osteocyte==1)]
  print(intersect(genesExternal[[2]],tempDE))

}

```

### Osteogenic Ad Hoc Classification of Cells (scaled integrated data, thresh=0, 4gene strict, Time 2 only)
### Examine Numbers and Overlap of Cormotif DE Genes & Functional/GeneSet Enrichment of DE Genes
