---
title: "10x-plot-replicate-correlation-tot"
author: "Genevieve Housman"
date: "May 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Examine Data Reproducibility Across Technical Replicates

Replicate Correlation Findings:

* across stages of differentiation: Time 0 > Time 1 > Time 2
* across unsupervized clusters (res=0.05): iPSC.c1 ~ iPCS.c2 > iPSC.c3 > MSC.c1 > Osteogenic.c1 ~ Osteogenic.c2
* across general ad hoc assignments: iPSC > MSC > Osteogenic
* across osteogenic ad hoc assignments:
    + 4 genes ( strict ): preosteoblast ~ osteoblast ~ embedding osteoblast ~ mineralizing osteoblast ~ maturing osteocyte
    + 4 genes (flexible): 
* for osteogenic cells, osteogenic ad hoc assignments ~ general ad hoc assignments ~ unsupervised clusters > stages

```{r}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)
library(ggplot2)
library(cowplot)
theme_set(theme_cowplot())

```

Check correlations using tot dataset.

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

```

Log normalized single-cell data are compared across technical replicates. Log normalized data were processed using NormalizedData(). Feature counts within a cell were divided by the total counts for that cell, multiplied by 10000, and natural-log transformed (log1p).

## Stage of differentiation at collection

### Time 0 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9918615

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9878004

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9934719

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### Time 1 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9714808

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9826743

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9251

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### Time 2 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9449687

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9801599

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8719403

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

## Unsupervised clustering (res=0.05)

### iPSC.c1 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.991741

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9880903

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9935819

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### iPSC.c2 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.988097

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink4", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9842512

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9864289

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### iPSC.c3 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9649778

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="grey", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9253106

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="grey", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9550134

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="grey", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### MSC.c1 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9722715

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9826208

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9278316

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### Osteogenic.c1 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9358406

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9787584

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8725624

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### Osteogenic.c2 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9667704

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="navyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9825803

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="navyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster0.05=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8812231

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="navyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

## General Ad Hoc

### iPSC Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9919855

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9880406

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9934564

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### MSC Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9744742

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9856457

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9274509

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### Osteogenic Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9285725

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9790619

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.thresh0=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.878369

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

## Osteogenic Ad Hoc

### 4 genes (flexible)

#### Preosteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9495209

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9460934

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8345675

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8471942

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.874614

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8721406

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Embedding Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8580735

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately. (no H1-r2 cells are embedding osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8625412

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Mineralizing Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9047385

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9729178

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8747518

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Maturing Osteocyte Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.960972

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9557495

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8809453

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### 4 genes ( strict )

#### Preosteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9495209

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9460934

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8345675

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8340107

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately. (no H1-r2 cells are osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8646472

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Embedding Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8570428

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately. (no H1-r2 cells are embedding osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8629882

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Mineralizing Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.7837459

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately. (no H1-r2 cells are mineralizing osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately. (no C1-r1 cells are mineralizing osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Maturing Osteocyte Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8481444

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8955833

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.x.4gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.7896373

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### 6 genes (flexible)

#### Preosteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9481142

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9460934

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8083983

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8477528

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.874614

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8327554

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Embedding Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.866648

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately. (no H1-r2 cells are embedding osteoblasts)

```{r, eval=FALSE}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson")

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8715044

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Mineralizing Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.927584

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9775199

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8774862

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Maturing Osteocyte Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.7858132

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.6437781

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.6gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.6974061

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

### 8 genes (flexible)

#### Preosteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8510288

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8498634

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.607413

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="lightblue1", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8666882

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8600955

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8073713

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="deepskyblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Embedding Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8816764

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9027286

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8725887

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue2", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Mineralizing Osteoblast Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9263957

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9739093

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8830671

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue3", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

#### Maturing Osteocyte Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9291552

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9744883

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.8gene.thresh0=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8610427

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="royalblue4", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

rm(rep1,rep2,avg.rep1,avg.rep2,rep.avg)

```
