---
title: "10x-data-enrichment"
author: "Genevieve Housman"
date: "September 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Differential Expression

Test for differential expression between humans and chimpanzees within different cell classifications.


```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)

```

### Examine output of DE analyses in more detail

```{r}

#Load DE results
#compare1 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.stage.rds")
#compare2 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.cluster.rds")
#compare3 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.adhoc.rds")
#compare4 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.ostadhoc.rds")
compare1 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.stage.rds")
compare2 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.cluster.rds")
compare3 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.adhoc.rds")
compare4 <- readRDS("./data/de-data/DEdream.totcell.filter.pch20.fml1.nofc.bh.ostadhoc.rds")
DEdata <- rbind(compare1,compare2,compare3,compare4)
colnames(DEdata)[5] <- "FDR"
DEdata <- as_tibble(DEdata)
rm(compare1,compare2,compare3,compare4)

```

```{r, eval=FALSE}

#TEMPORARY - fix gene names

compare1 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.stage.rds")
compare1$gene <- rownames(compare1)
compare1$geneCompare0 <- compare1$gene %in% rownames(data@assays$integrated)
table(compare1$geneCompare0)
compare1$geneNew1 <- compare1$gene
compare1$geneNew1[which(compare1$geneCompare0==FALSE)] <- substr(compare1$gene[which(compare1$geneCompare0==FALSE)],1,nchar(compare1$gene[which(compare1$geneCompare0==FALSE)])-1)
compare1$geneCompare1 <- compare1$geneNew1 %in% rownames(data@assays$integrated)
table(compare1$geneCompare1)
compare1$geneNew2 <- compare1$geneNew1
compare1$geneNew2[which(compare1$geneCompare1==FALSE)] <- substr(compare1$geneNew1[which(compare1$geneCompare1==FALSE)],1,nchar(compare1$geneNew1[which(compare1$geneCompare1==FALSE)])-1)
compare1$geneCompare2 <- compare1$geneNew2 %in% rownames(data@assays$integrated)
table(compare1$geneCompare2)
compare1$gene <- compare1$geneNew2

compare2 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.cluster.rds")
compare2$gene <- rownames(compare2)
compare2$geneCompare0 <- compare2$gene %in% rownames(data@assays$integrated)
table(compare2$geneCompare0)
compare2$geneNew1 <- compare2$gene
compare2$geneNew1[which(compare2$geneCompare0==FALSE)] <- substr(compare2$gene[which(compare2$geneCompare0==FALSE)],1,nchar(compare2$gene[which(compare2$geneCompare0==FALSE)])-1)
compare2$genecompare2 <- compare2$geneNew1 %in% rownames(data@assays$integrated)
table(compare2$genecompare2)
compare2$geneNew2 <- compare2$geneNew1
compare2$geneNew2[which(compare2$genecompare2==FALSE)] <- substr(compare2$geneNew1[which(compare2$genecompare2==FALSE)],1,nchar(compare2$geneNew1[which(compare2$genecompare2==FALSE)])-1)
compare2$geneCompare2 <- compare2$geneNew2 %in% rownames(data@assays$integrated)
table(compare2$geneCompare2)
compare2$gene <- compare2$geneNew2

compare3 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.adhoc.rds")
compare3$gene <- rownames(compare3)
compare3$geneCompare0 <- compare3$gene %in% rownames(data@assays$integrated)
table(compare3$geneCompare0)
compare3$geneNew1 <- compare3$gene
compare3$geneNew1[which(compare3$geneCompare0==FALSE)] <- substr(compare3$gene[which(compare3$geneCompare0==FALSE)],1,nchar(compare3$gene[which(compare3$geneCompare0==FALSE)])-1)
compare3$genecompare3 <- compare3$geneNew1 %in% rownames(data@assays$integrated)
table(compare3$genecompare3)
compare3$geneNew2 <- compare3$geneNew1
compare3$geneNew2[which(compare3$genecompare3==FALSE)] <- substr(compare3$geneNew1[which(compare3$genecompare3==FALSE)],1,nchar(compare3$geneNew1[which(compare3$genecompare3==FALSE)])-1)
compare3$geneCompare2 <- compare3$geneNew2 %in% rownames(data@assays$integrated)
table(compare3$geneCompare2)
compare3$gene <- compare3$geneNew2

compare4 <- readRDS("./data/de-data/DEedgeR.totcell.filter.pch20.fc.bh.ostadhoc.rds")
compare4$gene <- rownames(compare4)
compare4$geneCompare0 <- compare4$gene %in% rownames(data@assays$integrated)
table(compare4$geneCompare0)
compare4$geneNew1 <- compare4$gene
compare4$geneNew1[which(compare4$geneCompare0==FALSE)] <- substr(compare4$gene[which(compare4$geneCompare0==FALSE)],1,nchar(compare4$gene[which(compare4$geneCompare0==FALSE)])-1)
compare4$genecompare4 <- compare4$geneNew1 %in% rownames(data@assays$integrated)
table(compare4$genecompare4)
compare4$gene <- compare4$geneNew1

DEdata <- rbind(compare1[,1:15],compare2[,1:15],compare3[,1:15],compare4[,1:15])
DEdata <- as_tibble(DEdata)
rm(compare1,compare2,compare3,compare4)

##########

```

```{r}

#Isolate significant DE genes
DEdataSig <- DEdata %>% filter(FDR<=0.01)
#DEdataSig <- DEdata %>% filter(abs(logFC)>=1.2 & FDR<=0.01)

```

```{r}

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

```

```{r}

#cell assignment = stage
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                        data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                        data.frame(gene=DEdataSig %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0","Time.1","Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.0","Time.2"),
                         list("Time.1","Time.2"),
                         list("Time.0"),
                         list("Time.1"),
                         list("Time.2")))

#GO enrichment of significant DE genes
go.t0 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                  deGenes = pull(DEdata %>% filter(logFC>=1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                  ontology = "ALL")
go.t0.h <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    deGenes = pull(DEdata %>% filter(logFC>=1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    ontology = "ALL")
go.t0.c <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    deGenes = pull(DEdata %>% filter(logFC<=-1.2 & FDR<=0.01) %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                    ontology = "ALL")
go.t0@result$Description = str_wrap(go.t0@result$Description, width=30)
go.t0.h@result$Description = str_wrap(go.t0.h@result$Description, width=30)
go.t0.c@result$Description = str_wrap(go.t0.c@result$Description, width=30)
p1 <- ggplot(go.t0, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="All DE Genes")
p2 <- ggplot(go.t0.h, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="Upregulated in Humans")
p3 <- ggplot(go.t0.c, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        theme(axis.text.x=element_text(angle=-40, hjust=0)) +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="Upregulated in Chimps")
CombinePlots(list(p1,p2,p3),ncol=3,legend="right")

#GO enrichment of cell type specific significant DE genes
go.share <- goEnrich(totGenes = intersect(pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                                          intersect(pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)),
                                                    pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)))),
                     deGenes = combined$gene[which(rowSums(combined[,2:4])==3)],
                     ontology = "ALL")
go.share@result$Description = str_wrap(go.share@result$Description, width=30)

go.uniq.t0 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 0") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,2]==1 & rowSums(combined[,3:4])==0)],
                       ontology = "ALL")
go.uniq.t0@result$Description = str_wrap(go.uniq.t0@result$Description, width=30)

go.uniq.t1 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 1") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,3]==1 & rowSums(combined[,c(2,4)])==0)],
                       ontology = "ALL")
go.uniq.t1@result$Description = str_wrap(go.uniq.t1@result$Description, width=30)

go.uniq.t2 <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="stage" & cell.subset=="Time 2") %>% dplyr::select(gene)),
                       deGenes = combined$gene[which(combined[,4]==1 & rowSums(combined[,2:3])==0)],
                       ontology = "ALL")
go.uniq.t2@result$Description = str_wrap(go.uniq.t2@result$Description, width=30)

p0 <- ggplot(go.share, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Shared\nacross Stages") +
        ylab(label="")

p1 <- ggplot(go.uniq.t0, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 0") +
        ylab(label="")

p2 <- ggplot(go.uniq.t1, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 1") +
        ylab(label="")

p3 <- ggplot(go.uniq.t2, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique\nto Time 2") +
        ylab(label="")

CombinePlots(list(p1,p2,p3),ncol=3,legend="right")

```

```{r}

#cell assignment = cluster
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c2") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c3") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="MSC.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c2") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1_iPSC.c2_iPSC.c3") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1_Osteoblast.c2") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1") %>% select(gene), "iPSC.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c2") %>% select(gene), "iPSC.c2"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c3") %>% select(gene), "iPSC.c3"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="MSC.c1") %>% select(gene), "MSC.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1") %>% select(gene), "Osteoblast.c1"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c2") %>% select(gene), "Osteoblast.c2"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="iPSC.c1_iPSC.c2_iPSC.c3") %>% select(gene), "iPSC.c1.c2.c3"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="cluster" & cell.subset=="Osteoblast.c1_Osteoblast.c2") %>% select(gene), "Osteoblast.c1.c2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1.c2.c3","MSC.c1","Osteoblast.c1.c2"),
                         list("iPSC.c1.c2.c3","MSC.c1"),
                         list("iPSC.c1.c2.c3","Osteoblast.c1.c2"),
                         list("MSC.c1","Osteoblast.c1.c2"),
                         list("iPSC.c1.c2.c3"),
                         list("MSC.c1"),
                         list("Osteoblast.c1.c2")))

upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteoblast.c1","Osteoblast.c2"),
      intersections=list(list("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteoblast.c1","Osteoblast.c2"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3","Osteoblast.c1","Osteoblast.c2"),
                         list("MSC.c1","Osteoblast.c1","Osteoblast.c2"),
                         list("iPSC.c1","iPSC.c2","iPSC.c3"),
                         #list("iPSC.c1","iPSC.c2"),
                         #list("iPSC.c1","iPSC.c3"),
                         #list("iPSC.c2","iPSC.c3"),
                         list("iPSC.c1"),
                         list("iPSC.c2"),
                         list("iPSC.c3"),
                         list("MSC.c1"),
                         list("Osteoblast.c1","Osteoblast.c2"),
                         list("Osteoblast.c1"),
                         list("Osteoblast.c2"))
                         )

```

```{r}

#cell assignment = ad hoc
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="iPSC") %>% select(gene)
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="MSC") %>% select(gene)
DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="Osteoblast") %>% select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="iPSC") %>% dplyr::select(gene), "iPSC"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="MSC") %>% dplyr::select(gene), "MSC"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="adhoc" & cell.subset=="Osteoblast") %>% dplyr::select(gene), "Osteoblast"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC","MSC","Osteoblast"),
                         list("iPSC","MSC"),
                         list("iPSC","Osteoblast"),
                         list("MSC","Osteoblast"),
                         list("iPSC"),
                         list("MSC"),
                         list("Osteoblast")))

```

```{r}

#cell assignment = ost ad hoc
DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)
DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene)

#assess overlap
combined <- reduce(list(data.frame(DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene), "pre.osteo"=1),
                        data.frame(DEdataSig %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene), "emb.osteo"=1)
                        ), full_join)
combined[is.na(combined)] <- 0

#upsetr plot
upset(combined,
      order.by="degree",
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("pre.osteo","emb.osteo"),
                         list("pre.osteo"),
                         list("emb.osteo")))

#GO enrichment of cell type specific significant DE genes
go.share <- goEnrich(totGenes = intersect(pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)),
                                          pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene))),
                        deGenes = combined$gene[which(rowSums(combined[,2:3])==2)],
                        ontology = "ALL")
go.share@result$Description = str_wrap(go.share@result$Description, width=30)

go.uniq.pre <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="preosteoblast") %>% dplyr::select(gene)),
                        deGenes = combined$gene[which(combined[,2]==1 & combined[,3]==0)],
                        ontology = "ALL")
go.uniq.pre@result$Description = str_wrap(go.uniq.pre@result$Description, width=30)

go.uniq.emb <- goEnrich(totGenes = pull(DEdata %>% filter(cell.assign=="ostadhoc" & cell.subset=="embedding osteoblast") %>% dplyr::select(gene)),
                        deGenes = combined$gene[which(combined[,3]==1 & combined[,2]==0)],
                        ontology = "ALL")
go.uniq.emb@result$Description = str_wrap(go.uniq.emb@result$Description, width=30)

p1 <- ggplot(go.share, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Shared in\nosteogenic cells") +
        ylab(label="")

p2 <- ggplot(go.uniq.pre, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique to\npreosteoblasts") +
        ylab(label="")

p3 <- ggplot(go.uniq.emb, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
        geom_point() +
        facet_grid(ONTOLOGY~., scale="free") +
        scale_size_area(max_size=6) +
        scale_color_viridis(option="C") +
        theme_bw() +
        labs(title="DE Genes Unique to\nembedding osteoblasts") +
        ylab(label="")

CombinePlots(list(p1,p2,p3),ncol=3,legend="right")


```




```{r NEED TO FIX FOR MY DATA}

ContigencyTable <- matrix( c( length(intersect(ChimpSigGenes,HumanSigGenes)),
                              length(intersect(HumanSigGenes,ChimpNonSigGenes)),
                              length(intersect(ChimpSigGenes,HumanNonSigGenes)),
                              length(intersect(ChimpNonSigGenes,HumanNonSigGenes))), 
                           nrow = 2)
rownames(ContigencyTable) <- c("Chimp eGene", "Not Chimp eGene")
colnames(ContigencyTable) <- c("Human eGene", "Not human eGene")
#what is qval threshold for human eGene classification in this contigency table
print(GtexHeartEgenes %>% top_n(-HumanTopN, qval) %>% top_n(1, qval) %>% pull(qval))
#Contigency table of one to one orthologs tested in both chimps and humans of whether significant in humans, or chimps, or both, or neither
ContigencyTable
#One-sided Fisher test for greater overlap than expected by chance
fisher.test(ContigencyTable, alternative="greater")

#The above contingency table and one-sided fisher test indicated a greater-than-chance overlap between the sets of eGenes in chimp and human.

```


