---
title: "10x-test-data-cellranger-compare"
author: "Genevieve Housman"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Comparison

Initial loading, processing, and viewing of 10X data using seurat.

Data being examined comes from 1 human and 1 chimpanzee.
- GHO1 includes iPSCs from both species
- GHO2 inclues iPSC-derived MSCs from both species
- GHO3 includes iPSC-MSC-derived chondrocytes
- GHO4 includes iPSC-MSC-derived osteoblasts

The prefix of each object refers to whether the gene count data were processed using the human genome (h), the chimpanzee genome (c), or the human-chimpanzee genome combination, as well as the full annotation (full), the ortho-exon annotation (orth), or a filtered version of the ortho-exon annotation (orfl).
Species-specific assignments were determined using either the human-chimpanzee genome combination with full annotation (sppf), the human-chimpanzee genome combination with ortho-exon annotation (sppo), or the human-chimpanzee genome combination with filtered ortho-exon annotation (spof).

Thus, the final datasets compared, include:
- hc.orth.sppo
- hc.orth.sppf
- hc.orth.spof
- hc.full.sppo
- hc.full.sppf
- hc.full.spof
- hc.orfl.sppo
- hc.orfl.sppf
- hc.orfl.spof
- h.orth.c.orth.sppo
- h.orth.c.orth.sppf
- h.orth.c.orth.spof
- h.full.c.full.sppo
- h.full.c.full.sppf
- h.full.c.full.spof
- h.orfl.c.orfl.sppo
- h.orfl.c.orfl.sppf
- h.orfl.c.orfl.spof

```{r load libraries}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

```{r define data location}

#Best data to use are: h.orfl.c.orfl.spof

#Data in raw_feature_bc_matrix contains information for all 6794880 known barcodes.
#Data in filtered_feature_bc_matrix contains information for just the bacodes cellranger thinks actually code for cell data.
#Data using full references contain more cells than data using ortho-exon references.
#Data using ortho-exon references have more even human-chimp mixtures and fewer multiplets than data using full references.
#Data using ortho-exon references have no mitochondrial genes.
#Data using human-chimpanzee genome combination drops many more read counts (which corresponds to fewer gene counts) than data using independent genome references. This is most apparent when grouping by species or by orig.ident*species.

#Define main directory
#dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-subset'       #subset of data from first lane of sequencing
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full'        #data from three lanes of sequencing

#Define sub-directories
dir_h.full.GHO1 <- paste0(dir,'/cellranger-runs-human/GHO-1/outs')
dir_h.full.GHO2 <- paste0(dir,'/cellranger-runs-human/GHO-2/outs')
dir_h.full.GHO3 <- paste0(dir,'/cellranger-runs-human/GHO-3/outs')
dir_h.full.GHO4 <- paste0(dir,'/cellranger-runs-human/GHO-4/outs')
dir_c.full.GHO1 <- paste0(dir,'/cellranger-runs-chimp/GHO-1/outs')
dir_c.full.GHO2 <- paste0(dir,'/cellranger-runs-chimp/GHO-2/outs')
dir_c.full.GHO3 <- paste0(dir,'/cellranger-runs-chimp/GHO-3/outs')
dir_c.full.GHO4 <- paste0(dir,'/cellranger-runs-chimp/GHO-4/outs')
dir_hc.full.GHO1 <- paste0(dir,'/cellranger-runs-human-chimp/GHO-1/outs')
dir_hc.full.GHO2 <- paste0(dir,'/cellranger-runs-human-chimp/GHO-2/outs')
dir_hc.full.GHO3 <- paste0(dir,'/cellranger-runs-human-chimp/GHO-3/outs')
dir_hc.full.GHO4 <- paste0(dir,'/cellranger-runs-human-chimp/GHO-4/outs')
dir_h.orth.GHO1 <- paste0(dir,'/cellranger-runs-ortho-human/GHO-1/outs')
dir_h.orth.GHO2 <- paste0(dir,'/cellranger-runs-ortho-human/GHO-2/outs')
dir_h.orth.GHO3 <- paste0(dir,'/cellranger-runs-ortho-human/GHO-3/outs')
dir_h.orth.GHO4 <- paste0(dir,'/cellranger-runs-ortho-human/GHO-4/outs')
dir_c.orth.GHO1 <- paste0(dir,'/cellranger-runs-ortho-chimp/GHO-1/outs')
dir_c.orth.GHO2 <- paste0(dir,'/cellranger-runs-ortho-chimp/GHO-2/outs')
dir_c.orth.GHO3 <- paste0(dir,'/cellranger-runs-ortho-chimp/GHO-3/outs')
dir_c.orth.GHO4 <- paste0(dir,'/cellranger-runs-ortho-chimp/GHO-4/outs')
dir_hc.orth.GHO1 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-1/outs')
dir_hc.orth.GHO2 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-2/outs')
dir_hc.orth.GHO3 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-3/outs')
dir_hc.orth.GHO4 <- paste0(dir,'/cellranger-runs-ortho-human-chimp/GHO-4/outs')
dir_h.orfl.GHO1 <- paste0(dir,'/cellranger-runs-ortho-filter-human/GHO-1/outs')
dir_h.orfl.GHO2 <- paste0(dir,'/cellranger-runs-ortho-filter-human/GHO-2/outs')
dir_h.orfl.GHO3 <- paste0(dir,'/cellranger-runs-ortho-filter-human/GHO-3/outs')
dir_h.orfl.GHO4 <- paste0(dir,'/cellranger-runs-ortho-filter-human/GHO-4/outs')
dir_c.orfl.GHO1 <- paste0(dir,'/cellranger-runs-ortho-filter-chimp/GHO-1/outs')
dir_c.orfl.GHO2 <- paste0(dir,'/cellranger-runs-ortho-filter-chimp/GHO-2/outs')
dir_c.orfl.GHO3 <- paste0(dir,'/cellranger-runs-ortho-filter-chimp/GHO-3/outs')
dir_c.orfl.GHO4 <- paste0(dir,'/cellranger-runs-ortho-filter-chimp/GHO-4/outs')
dir_hc.orfl.GHO1 <- paste0(dir,'/cellranger-runs-ortho-filter-human-chimp/GHO-1/outs')
dir_hc.orfl.GHO2 <- paste0(dir,'/cellranger-runs-ortho-filter-human-chimp/GHO-2/outs')
dir_hc.orfl.GHO3 <- paste0(dir,'/cellranger-runs-ortho-filter-human-chimp/GHO-3/outs')
dir_hc.orfl.GHO4 <- paste0(dir,'/cellranger-runs-ortho-filter-human-chimp/GHO-4/outs')

#Check that directories contain barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz
#list.files(paste0(dir_hc.orth.GHO1,"/filtered_feature_bc_matrix"))
#list.files(paste0(dir_hc.orth.GHO1,"/raw_feature_bc_matrix"))

```

```{r compile summary metrics}

#Load summary metrics
metrics.h.full.GHO1 <- read.csv(file=paste0(dir_h.full.GHO1,'/metrics_summary.csv'))
metrics.h.full.GHO2 <- read.csv(file=paste0(dir_h.full.GHO2,'/metrics_summary.csv'))
metrics.h.full.GHO3 <- read.csv(file=paste0(dir_h.full.GHO3,'/metrics_summary.csv'))
metrics.h.full.GHO4 <- read.csv(file=paste0(dir_h.full.GHO4,'/metrics_summary.csv'))
metrics.c.full.GHO1 <- read.csv(file=paste0(dir_c.full.GHO1,'/metrics_summary.csv'))
metrics.c.full.GHO2 <- read.csv(file=paste0(dir_c.full.GHO2,'/metrics_summary.csv'))
metrics.c.full.GHO3 <- read.csv(file=paste0(dir_c.full.GHO3,'/metrics_summary.csv'))
metrics.c.full.GHO4 <- read.csv(file=paste0(dir_c.full.GHO4,'/metrics_summary.csv'))
metrics.hc.full.GHO1 <- read.csv(file=paste0(dir_hc.full.GHO1,'/metrics_summary.csv'))
metrics.hc.full.GHO2 <- read.csv(file=paste0(dir_hc.full.GHO2,'/metrics_summary.csv'))
metrics.hc.full.GHO3 <- read.csv(file=paste0(dir_hc.full.GHO3,'/metrics_summary.csv'))
metrics.hc.full.GHO4 <- read.csv(file=paste0(dir_hc.full.GHO4,'/metrics_summary.csv'))
metrics.h.orth.GHO1 <- read.csv(file=paste0(dir_h.orth.GHO1,'/metrics_summary.csv'))
metrics.h.orth.GHO2 <- read.csv(file=paste0(dir_h.orth.GHO2,'/metrics_summary.csv'))
metrics.h.orth.GHO3 <- read.csv(file=paste0(dir_h.orth.GHO3,'/metrics_summary.csv'))
metrics.h.orth.GHO4 <- read.csv(file=paste0(dir_h.orth.GHO4,'/metrics_summary.csv'))
metrics.c.orth.GHO1 <- read.csv(file=paste0(dir_c.orth.GHO1,'/metrics_summary.csv'))
metrics.c.orth.GHO2 <- read.csv(file=paste0(dir_c.orth.GHO2,'/metrics_summary.csv'))
metrics.c.orth.GHO3 <- read.csv(file=paste0(dir_c.orth.GHO3,'/metrics_summary.csv'))
metrics.c.orth.GHO4 <- read.csv(file=paste0(dir_c.orth.GHO4,'/metrics_summary.csv'))
metrics.hc.orth.GHO1 <- read.csv(file=paste0(dir_hc.orth.GHO1,'/metrics_summary.csv'))
metrics.hc.orth.GHO2 <- read.csv(file=paste0(dir_hc.orth.GHO2,'/metrics_summary.csv'))
metrics.hc.orth.GHO3 <- read.csv(file=paste0(dir_hc.orth.GHO3,'/metrics_summary.csv'))
metrics.hc.orth.GHO4 <- read.csv(file=paste0(dir_hc.orth.GHO4,'/metrics_summary.csv'))
metrics.h.orfl.GHO1 <- read.csv(file=paste0(dir_h.orfl.GHO1,'/metrics_summary.csv'))
metrics.h.orfl.GHO2 <- read.csv(file=paste0(dir_h.orfl.GHO2,'/metrics_summary.csv'))
metrics.h.orfl.GHO3 <- read.csv(file=paste0(dir_h.orfl.GHO3,'/metrics_summary.csv'))
metrics.h.orfl.GHO4 <- read.csv(file=paste0(dir_h.orfl.GHO4,'/metrics_summary.csv'))
metrics.c.orfl.GHO1 <- read.csv(file=paste0(dir_c.orfl.GHO1,'/metrics_summary.csv'))
metrics.c.orfl.GHO2 <- read.csv(file=paste0(dir_c.orfl.GHO2,'/metrics_summary.csv'))
metrics.c.orfl.GHO3 <- read.csv(file=paste0(dir_c.orfl.GHO3,'/metrics_summary.csv'))
metrics.c.orfl.GHO4 <- read.csv(file=paste0(dir_c.orfl.GHO4,'/metrics_summary.csv'))
metrics.hc.orfl.GHO1 <- read.csv(file=paste0(dir_hc.orfl.GHO1,'/metrics_summary.csv'))
metrics.hc.orfl.GHO2 <- read.csv(file=paste0(dir_hc.orfl.GHO2,'/metrics_summary.csv'))
metrics.hc.orfl.GHO3 <- read.csv(file=paste0(dir_hc.orfl.GHO3,'/metrics_summary.csv'))
metrics.hc.orfl.GHO4 <- read.csv(file=paste0(dir_hc.orfl.GHO4,'/metrics_summary.csv'))

#Process data
#metrics.hc.full.GHO3$GRCh38.Median.Genes.per.Cell <- as.factor(metrics.hc.full.GHO3$GRCh38.Median.Genes.per.Cell)
#metrics.hc.full.GHO3$Pan.tro.3.0.Median.Genes.per.Cell <- as.factor(metrics.hc.full.GHO3$Pan.tro.3.0.Median.Genes.per.Cell)
#metrics.hc.full.GHO2$GEMs.with..1.Cell <- as.factor(metrics.hc.full.GHO2$GEMs.with..1.Cell)

metrics.h.full <- rbind(metrics.h.full.GHO1,metrics.h.full.GHO2,metrics.h.full.GHO3,metrics.h.full.GHO4)
metrics.c.full <- rbind(metrics.c.full.GHO1,metrics.c.full.GHO2,metrics.c.full.GHO3,metrics.c.full.GHO4)
metrics.hc.full <- rbind(metrics.hc.full.GHO1,metrics.hc.full.GHO2,metrics.hc.full.GHO3,metrics.hc.full.GHO4)
metrics.h.orth <- rbind(metrics.h.orth.GHO1,metrics.h.orth.GHO2,metrics.h.orth.GHO3,metrics.h.orth.GHO4)
metrics.c.orth <- rbind(metrics.c.orth.GHO1,metrics.c.orth.GHO2,metrics.c.orth.GHO3,metrics.c.orth.GHO4)
metrics.hc.orth <- rbind(metrics.hc.orth.GHO1,metrics.hc.orth.GHO2,metrics.hc.orth.GHO3,metrics.hc.orth.GHO4)
metrics.h.orfl <- rbind(metrics.h.orfl.GHO1,metrics.h.orfl.GHO2,metrics.h.orfl.GHO3,metrics.h.orfl.GHO4)
metrics.c.orfl <- rbind(metrics.c.orfl.GHO1,metrics.c.orfl.GHO2,metrics.c.orfl.GHO3,metrics.c.orfl.GHO4)
metrics.hc.orfl <- rbind(metrics.hc.orfl.GHO1,metrics.hc.orfl.GHO2,metrics.hc.orfl.GHO3,metrics.hc.orfl.GHO4)

metrics.h.full[,c(1:4,19:20)] <- lapply(metrics.h.full[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.full[,c(1:4,19:20)] <- lapply(metrics.c.full[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.hc.full[,c(1:3,29:30,34:41)] <- lapply(metrics.hc.full[,c(1:3,29:30,34:41)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.h.orth[,c(1:4,19:20)] <- lapply(metrics.h.orth[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.orth[,c(1:4,19:20)] <- lapply(metrics.c.orth[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.hc.orth[,c(1:3,29:30,34:41)] <- lapply(metrics.hc.orth[,c(1:3,29:30,34:41)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.h.orfl[,c(1:4,19:20)] <- lapply(metrics.h.orfl[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.orfl[,c(1:4,19:20)] <- lapply(metrics.c.orfl[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.hc.orfl[,c(1:3,29:30,34:41)] <- lapply(metrics.hc.orfl[,c(1:3,29:30,34:41)], function(x) as.numeric(gsub(",","",as.character(x))))

metrics.h.full[,c(5:18)] <- lapply(metrics.h.full[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.full[,c(5:18)] <- lapply(metrics.c.full[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.hc.full[,c(4:28,31:33,42:45)] <- lapply(metrics.hc.full[,c(4:28,31:33,42:45)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.h.orth[,c(5:18)] <- lapply(metrics.h.orth[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.orth[,c(5:18)] <- lapply(metrics.c.orth[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.hc.orth[,c(4:28,31:33,42:45)] <- lapply(metrics.hc.orth[,c(4:28,31:33,42:45)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.h.orfl[,c(5:18)] <- lapply(metrics.h.orfl[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.orfl[,c(5:18)] <- lapply(metrics.c.orfl[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.hc.orfl[,c(4:28,31:33,42:45)] <- lapply(metrics.hc.orfl[,c(4:28,31:33,42:45)], function(x) as.numeric(gsub("%","",as.character(x)))/100)

metrics.h.full$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.full$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.hc.full$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.h.orth$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.orth$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.hc.orth$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.h.orfl$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.orfl$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.hc.orfl$Sample <- c("GHO1","GHO2","GHO3","GHO4")

rm(metrics.h.full.GHO1,metrics.h.full.GHO2,metrics.h.full.GHO3,metrics.h.full.GHO4)
rm(metrics.c.full.GHO1,metrics.c.full.GHO2,metrics.c.full.GHO3,metrics.c.full.GHO4)
rm(metrics.hc.full.GHO1,metrics.hc.full.GHO2,metrics.hc.full.GHO3,metrics.hc.full.GHO4)
rm(metrics.h.orth.GHO1,metrics.h.orth.GHO2,metrics.h.orth.GHO3,metrics.h.orth.GHO4)
rm(metrics.c.orth.GHO1,metrics.c.orth.GHO2,metrics.c.orth.GHO3,metrics.c.orth.GHO4)
rm(metrics.hc.orth.GHO1,metrics.hc.orth.GHO2,metrics.hc.orth.GHO3,metrics.hc.orth.GHO4)
rm(metrics.h.orfl.GHO1,metrics.h.orfl.GHO2,metrics.h.orfl.GHO3,metrics.h.orfl.GHO4)
rm(metrics.c.orfl.GHO1,metrics.c.orfl.GHO2,metrics.c.orfl.GHO3,metrics.c.orfl.GHO4)
rm(metrics.hc.orfl.GHO1,metrics.hc.orfl.GHO2,metrics.hc.orfl.GHO3,metrics.hc.orfl.GHO4)


#Summarize Metrics
mean(metrics.h.full$Number.of.Reads)
mean(metrics.c.full$Number.of.Reads)
mean(metrics.hc.full$Number.of.Reads)
mean(metrics.h.orth$Number.of.Reads)
mean(metrics.c.orth$Number.of.Reads)
mean(metrics.hc.orth$Number.of.Reads)
mean(metrics.h.orfl$Number.of.Reads)
mean(metrics.c.orfl$Number.of.Reads)
mean(metrics.hc.orfl$Number.of.Reads)

mean(metrics.h.full$Sequencing.Saturation)
mean(metrics.c.full$Sequencing.Saturation)
mean(metrics.hc.full$Sequencing.Saturation)
mean(metrics.h.orth$Sequencing.Saturation)
mean(metrics.c.orth$Sequencing.Saturation)
mean(metrics.hc.orth$Sequencing.Saturation)
mean(metrics.h.orfl$Sequencing.Saturation)
mean(metrics.c.orfl$Sequencing.Saturation)
mean(metrics.hc.orfl$Sequencing.Saturation)

mean(metrics.h.full$Mean.Reads.per.Cell)
mean(metrics.c.full$Mean.Reads.per.Cell)
mean(metrics.hc.full$Mean.Reads.per.Cell)
mean(metrics.h.orth$Mean.Reads.per.Cell)
mean(metrics.c.orth$Mean.Reads.per.Cell)
mean(metrics.hc.orth$Mean.Reads.per.Cell)
mean(metrics.h.orfl$Mean.Reads.per.Cell)
mean(metrics.c.orfl$Mean.Reads.per.Cell)
mean(metrics.hc.orfl$Mean.Reads.per.Cell)

mean(metrics.h.full$Median.Genes.per.Cell)
mean(metrics.c.full$Median.Genes.per.Cell)
mean(metrics.hc.full$GRCh38.Median.Genes.per.Cell)
mean(metrics.hc.full$Pan.tro.3.0.Median.Genes.per.Cell)
mean(metrics.h.orth$Median.Genes.per.Cell)
mean(metrics.c.orth$Median.Genes.per.Cell)
mean(metrics.hc.orth$ortho.GRCh38.Median.Genes.per.Cell)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Median.Genes.per.Cell)
mean(metrics.h.orfl$Median.Genes.per.Cell)
mean(metrics.c.orfl$Median.Genes.per.Cell)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Median.Genes.per.Cell)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Median.Genes.per.Cell)

mean(metrics.h.full$Median.UMI.Counts.per.Cell)
mean(metrics.c.full$Median.UMI.Counts.per.Cell)
mean(metrics.hc.full$GRCh38.Median.UMI.Counts.per.Cell)
mean(metrics.hc.full$Pan.tro.3.0.Median.UMI.Counts.per.Cell)
mean(metrics.h.orth$Median.UMI.Counts.per.Cell)
mean(metrics.c.orth$Median.UMI.Counts.per.Cell)
mean(metrics.hc.orth$ortho.GRCh38.Median.UMI.Counts.per.Cell)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Median.UMI.Counts.per.Cell)
mean(metrics.h.orfl$Median.UMI.Counts.per.Cell)
mean(metrics.c.orfl$Median.UMI.Counts.per.Cell)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Median.UMI.Counts.per.Cell)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Median.UMI.Counts.per.Cell)

mean(metrics.h.full$Q30.Bases.in.Barcode)
mean(metrics.c.full$Q30.Bases.in.Barcode)
mean(metrics.hc.full$Q30.Bases.in.Barcode)
mean(metrics.h.orth$Q30.Bases.in.Barcode)
mean(metrics.c.orth$Q30.Bases.in.Barcode)
mean(metrics.hc.orth$Q30.Bases.in.Barcode)
mean(metrics.h.orfl$Q30.Bases.in.Barcode)
mean(metrics.c.orfl$Q30.Bases.in.Barcode)
mean(metrics.hc.orfl$Q30.Bases.in.Barcode)

mean(metrics.h.full$Q30.Bases.in.RNA.Read)
mean(metrics.c.full$Q30.Bases.in.RNA.Read)
mean(metrics.hc.full$Q30.Bases.in.RNA.Read)
mean(metrics.h.orth$Q30.Bases.in.RNA.Read)
mean(metrics.c.orth$Q30.Bases.in.RNA.Read)
mean(metrics.hc.orth$Q30.Bases.in.RNA.Read)
mean(metrics.h.orfl$Q30.Bases.in.RNA.Read)
mean(metrics.c.orfl$Q30.Bases.in.RNA.Read)
mean(metrics.hc.orfl$Q30.Bases.in.RNA.Read)

mean(metrics.h.full$Q30.Bases.in.UMI)
mean(metrics.c.full$Q30.Bases.in.UMI)
mean(metrics.hc.full$Q30.Bases.in.UMI)
mean(metrics.h.orth$Q30.Bases.in.UMI)
mean(metrics.c.orth$Q30.Bases.in.UMI)
mean(metrics.hc.orth$Q30.Bases.in.UMI)
mean(metrics.h.orfl$Q30.Bases.in.UMI)
mean(metrics.c.orfl$Q30.Bases.in.UMI)
mean(metrics.hc.orfl$Q30.Bases.in.UMI)

mean(metrics.h.full$Reads.Mapped.to.Genome)
mean(metrics.c.full$Reads.Mapped.to.Genome)
mean(metrics.hc.full$Reads.Mapped.to.Genome)
mean(metrics.h.orth$Reads.Mapped.to.Genome)
mean(metrics.c.orth$Reads.Mapped.to.Genome)
mean(metrics.hc.orth$Reads.Mapped.to.Genome)
mean(metrics.h.orfl$Reads.Mapped.to.Genome)
mean(metrics.c.orfl$Reads.Mapped.to.Genome)
mean(metrics.hc.orfl$Reads.Mapped.to.Genome)
mean(metrics.hc.full$GRCh38.Reads.Mapped.to.Genome)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.to.Genome)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.to.Genome)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.to.Genome)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.to.Genome)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.to.Genome)

mean(metrics.h.full$Reads.Mapped.Confidently.to.Genome)
mean(metrics.c.full$Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.full$Reads.Mapped.Confidently.to.Genome)
mean(metrics.h.orth$Reads.Mapped.Confidently.to.Genome)
mean(metrics.c.orth$Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orth$Reads.Mapped.Confidently.to.Genome)
mean(metrics.h.orfl$Reads.Mapped.Confidently.to.Genome)
mean(metrics.c.orfl$Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orfl$Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.full$GRCh38.Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome)

mean(metrics.h.full$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.c.full$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.full$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.h.orth$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.c.orth$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orth$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.h.orfl$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.c.orfl$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orfl$Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.full$GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions)

mean(metrics.h.full$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.c.full$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.full$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.h.orth$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.c.orth$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orth$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.h.orfl$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.c.orfl$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orfl$Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.full$GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions)

mean(metrics.h.full$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.c.full$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.full$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.h.orth$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.c.orth$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orth$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.h.orfl$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.c.orfl$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orfl$Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.full$GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions)

mean(metrics.h.full$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.c.full$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.full$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.h.orth$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.c.orth$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orth$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.h.orfl$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.c.orfl$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orfl$Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.full$GRCh38.Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orth$ortho.GRCh38.Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.full$Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome)

mean(metrics.h.full$Reads.Mapped.Antisense.to.Gene)
mean(metrics.c.full$Reads.Mapped.Antisense.to.Gene)
mean(metrics.hc.full$Reads.Mapped.Antisense.to.Gene)
mean(metrics.h.orth$Reads.Mapped.Antisense.to.Gene)
mean(metrics.c.orth$Reads.Mapped.Antisense.to.Gene)
mean(metrics.hc.orth$Reads.Mapped.Antisense.to.Gene)
mean(metrics.h.orfl$Reads.Mapped.Antisense.to.Gene)
mean(metrics.c.orfl$Reads.Mapped.Antisense.to.Gene)
mean(metrics.hc.orfl$Reads.Mapped.Antisense.to.Gene)

mean(metrics.h.full$Fraction.Reads.in.Cells)
mean(metrics.c.full$Fraction.Reads.in.Cells)
mean(metrics.hc.full$Fraction.Reads.in.Cells)
mean(metrics.h.orth$Fraction.Reads.in.Cells)
mean(metrics.c.orth$Fraction.Reads.in.Cells)
mean(metrics.hc.orth$Fraction.Reads.in.Cells)
mean(metrics.h.orfl$Fraction.Reads.in.Cells)
mean(metrics.c.orfl$Fraction.Reads.in.Cells)
mean(metrics.hc.orfl$Fraction.Reads.in.Cells)
mean(metrics.hc.full$GRCh38.Fraction.Reads.in.Cells)
mean(metrics.hc.orth$ortho.GRCh38.Fraction.Reads.in.Cells)
mean(metrics.hc.orfl$ortho.filter.GRCh38.Fraction.Reads.in.Cells)
mean(metrics.hc.full$Pan.tro.3.0.Fraction.Reads.in.Cells)
mean(metrics.hc.orth$ortho.Pan.tro.3.0.Fraction.Reads.in.Cells)
mean(metrics.hc.orfl$ortho.filter.Pan.tro.3.0.Fraction.Reads.in.Cells)

#notes: full data number of reads increased by 3x but sequence saturation only increased 10%
#FULL-DATA
#         Number.of.Reads  Sequencing.Saturation  Q30.Bases.in.Barcode  Q30.Bases.in.RNA.Read Q30.Bases.in.UMI
#h.full   282750845        0.36875                0.98525               0.73775               0.986
#c.full   282750845        0.36325                0.98525               0.73775               0.986               
#hc.full  282750845        0.33175                0.98525               0.73775               0.986
#h.orth   282750845        0.34925                0.98525               0.73775               0.986
#c.orth   282750845        0.34825                0.98525               0.73775               0.986
#hc.orth  282750845        0.31775                0.98525               0.73775               0.986
#h.orfl   282750845        0.35075                0.98525               0.73775               0.986
#c.orfl   282750845        0.35075                0.98525               0.73775               0.986
#hc.orfl  282750845        0.31925                0.98525               0.73775               0.986

#         Mean.Reads.per.Cell   Median.Genes.per.Cell   Median.UMI.Counts.per.Cell
#                               (GRCh38/Pan.tro.3.0)    (GRCh38/Pan.tro.3.0)
#h.full   44310.50               3440.00                13837.25
#c.full   47063.50               3078.75                12772.25
#hc.full  45528.75              (2198.75 / 1804.25)     (6962.25 / 5509.50)
#h.orth   58717.50               3003.25                 8891.25
#c.orth   55826.75               2882.00                 8434.50
#hc.orth  64930.75              (2259.50 / 1727.25)     (5859.25 / 3908.00)
#h.orfl   61638.00               3020.25                 9082.75
#c.orfl   61350.75               2996.75                 8965.50
#hc.orfl  65043.75              (2213.75 / 1693.00)     (5771.50 / 3859.25)

#                                                             h.full  c.full  hc.full h.orth  c.orth  hc.orth h.orfl  c.orfl  hc.orfl
#Reads.Mapped.to.Genome                                       0.91050 0.91525 0.92975 0.90625 0.91125 0.92550 0.90625 0.91125 0.92550
#GRCh38.Reads.Mapped.to.Genome                                ---     ---     0.71100 ---     ---     0.69475 ---     ---     0.69475
#Pan.tro.3.0.Reads.Mapped.to.Genome                           ---     ---     0.66850 ---     ---     0.66750 ---     ---     0.66750
#Reads.Mapped.Confidently.to.Genome                           0.88125 0.82400 0.50575 0.82500 0.76050 0.44475 0.82175 0.75650 0.44325
#GRCh38.Reads.Mapped.Confidently.to.Genome                    ---     ---     0.30350 ---     ---     0.25275 ---     ---     0.25175
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome               ---     ---     0.20225 ---     ---     0.19225 ---     ---     0.19175
#Reads.Mapped.Confidently.to.Intergenic.Regions               0.04900 0.13825 0.06525 0.46550 0.41025 0.27600 0.47150 0.41475 0.27900
#GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions        ---     ---     0.01900 ---     ---     0.15525 ---     ---     0.15725
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions   ---     ---     0.04625 ---     ---     0.12075 ---     ---     0.12175
#Reads.Mapped.Confidently.to.Intronic.Regions                 0.11650 0.11075 0.07600 0.00000 0.00000 0.00000 0.00000 0.00000 0.00000
#GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions          ---     ---     0.04175 ---     ---     0.00000 ---     ---     0.00000
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions     ---     ---     0.03375 ---     ---     0.00000 ---     ---     0.00000
#Reads.Mapped.Confidently.to.Exonic.Regions                   0.71575 0.57500 0.36475 0.35950 0.35025 0.16850 0.35075 0.34175 0.16425
#GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions            ---     ---     0.24275 ---     ---     0.09775 ---     ---     0.09425
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions       ---     ---     0.12225 ---     ---     0.07150 ---     ---     0.07000
#Reads.Mapped.Confidently.to.Transcriptome                    0.67750 0.54800 0.34900 0.31450 0.30950 0.15375 0.30635 0.30175 0.14950
#GRCh38.Reads.Mapped.Confidently.to.Transcriptome             ---     ---     0.23200 ---     ---     0.08875 ---     ---     0.08550
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome        ---     ---     0.11675 ---     ---     0.06500 ---     ---     0.06375
#Reads.Mapped.Antisense.to.Gene                               0.01100 0.00700 0.00500 0.00700 0.00650 0.00375 0.00650 0.00650 0.00375
#Fraction.Reads.in.Cells                                      0.90575 0.91025 0.89025 0.90775 0.91400 0.89950 0.90150 0.90225 0.89950
#GRCh38.Fraction.Reads.in.Cells                               ---     ---     0.80425 ---     ---     0.84925 ---     ---     0.85975
#Pan.tro.3.0.Fraction.Reads.in.Cells                          ---     ---     0.80475 ---     ---     0.84375 ---     ---     0.84825

#SUBSET DATA
#         Number.of.Reads  Sequencing.Saturation  Q30.Bases.in.Barcode  Q30.Bases.in.RNA.Read Q30.Bases.in.UMI
#h.full   97851860         0.24225                0.98725               0.746                 0.98725
#c.full   97851860         0.23900                0.98725               0.746                 0.98725               
#hc.full  97851860         0.22275                0.98725               0.746                 0.98725
#h.orth   97851860         0.23275                0.98725               0.746                 0.98725
#c.orth   97851860         0.23250                0.98725               0.746                 0.98725
#hc.orth  97851860         0.21600                0.98725               0.746                 0.98725

#         Mean.Reads.per.Cell   Median.Genes.per.Cell   Median.UMI.Counts.per.Cell
#                               (GRCh38/Pan.tro.3.0)    (GRCh38/Pan.tro.3.0)
#h.full   17508.75               2317.00                 6696.25
#c.full   18803.50               2056.75                 5978.25
#hc.full  18928.25              (1616.25 / 1081.75)     (3812.75 / 2461.50)
#h.orth   22705.50               1902.75                 4149.50
#c.orth   21687.75               1809.00                 3864.50
#hc.orth  24126.75              (1331.00 /  946.25)     (2667.25 / 1658.00)

#                                                             h.full  c.full  hc.full h.orth  c.orth  hc.orth
#Reads.Mapped.to.Genome                                       0.91950 0.92500 0.93950 0.92100 0.92100 0.93535
#GRCh38.Reads.Mapped.to.Genome                                ---     ---     0.71750 ---     ---     0.70100
#Pan.tro.3.0.Reads.Mapped.to.Genome                           ---     ---     0.67550 ---     ---     0.67500
#Reads.Mapped.Confidently.to.Genome                           0.88975 0.83225 0.51075 0.83300 0.76775 0.44925
#GRCh38.Reads.Mapped.Confidently.to.Genome                    ---     ---     0.30600 ---     ---     0.25500
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome               ---     ---     0.20450 ---     ---     0.19425
#Reads.Mapped.Confidently.to.Intergenic.Regions               0.04925 0.13975 0.06600 0.47000 0.41425 0.27875
#GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions        ---     ---     0.01900 ---     ---     0.15650
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions   ---     ---     0.04650 ---     ---     0.12225
#Reads.Mapped.Confidently.to.Intronic.Regions                 0.11750 0.11175 0.07650 0.00000 0.00000 0.00000
#GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions          ---     ---     0.04225 ---     ---     0.00000
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions     ---     ---     0.03400 ---     ---     0.00000
#Reads.Mapped.Confidently.to.Exonic.Regions                   0.72275 0.58075 0.36800 0.36275 0.35375 0.17025
#GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions            ---     ---     0.24450 ---     ---     0.09850
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions       ---     ---     0.12375 ---     ---     0.07200
#Reads.Mapped.Confidently.to.Transcriptome                    0.68350 0.55325 0.35200 0.31725 0.31225 0.15500
#GRCh38.Reads.Mapped.Confidently.to.Transcriptome             ---     ---     0.23425 ---     ---     0.08925
#Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome        ---     ---     0.11825 ---     ---     0.06575
#Reads.Mapped.Antisense.to.Gene                               0.01100 0.00700 0.00500 0.00700 0.00675 0.00375
#Fraction.Reads.in.Cells                                      0.90000 0.90150 0.87500 0.89800 0.90425 0.89325
#GRCh38.Fraction.Reads.in.Cells                               ---     ---     0.79275 ---     ---     0.84650
#Pan.tro.3.0.Fraction.Reads.in.Cells                          ---     ---     0.79900 ---     ---     0.84150


#Visualize summary metrics
#total number of reads
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="h.full")+ylim(0,300000000)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="c.full")+ylim(0,300000000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="hc.full")+ylim(0,300000000)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="h.orth")+ylim(0,300000000)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="c.orth")+ylim(0,300000000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="hc.orth")+ylim(0,300000000)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="h.orfl")+ylim(0,300000000)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="c.orfl")+ylim(0,300000000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Number.of.Reads))+geom_col()+labs(title="hc.orfl")+ylim(0,300000000))), ncol=3)

#percent sequence saturation
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="hc.orfl")+ylim(0,1))), ncol=3)

#mean reads per cell
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="h.full")+ylim(0,85000)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="c.full")+ylim(0,85000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="hc.full")+ylim(0,85000)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="h.orth")+ylim(0,85000)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="c.orth")+ylim(0,85000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,85000)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="h.orfl")+ylim(0,85000)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="c.orfl")+ylim(0,85000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,85000))), ncol=3)

#median genes per cell
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="h.full")+ylim(0,4500)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="c.full")+ylim(0,4500)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="h.orth")+ylim(0,4500)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="c.orth")+ylim(0,4500)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="h.orfl")+ylim(0,4500)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="c.orfl")+ylim(0,4500)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Median.Genes.per.Cell))+geom_col()+labs(title="hc.full")+ylim(0,4500)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Median.Genes.per.Cell))+geom_col()+labs(title="hc.full")+ylim(0,4500)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Median.Genes.per.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,4500)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Median.Genes.per.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,4500)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Median.Genes.per.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,4500)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Median.Genes.per.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,4500))),
             ncol=6)

#median UMI per cell
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="h.full")+ylim(0,21000)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="c.full")+ylim(0,21000)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="h.orth")+ylim(0,21000)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="c.orth")+ylim(0,21000)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="h.orfl")+ylim(0,21000)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="c.orfl")+ylim(0,21000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Median.UMI.Counts.per.Cell))+geom_col()+labs(title="hc.full")+ylim(0,21000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Median.UMI.Counts.per.Cell))+geom_col()+labs(title="hc.full")+ylim(0,21000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Median.UMI.Counts.per.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,21000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Median.UMI.Counts.per.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,21000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Median.UMI.Counts.per.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,21000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Median.UMI.Counts.per.Cell))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,21000))),
             ncol=6)

#proportion of Q30 bases in barcode, rna read, and umi
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Q30.Bases.in.Barcode))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Q30.Bases.in.RNA.Read))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Q30.Bases.in.UMI))+geom_col()+labs(title="hc.orfl")+ylim(0,1))), ncol=9)

#reads mapped and confidently mapped to genome
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="c.orfl")+ylim(0,1))), ncol=6)
CombinePlots(plots=list((ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.Confidently.to.Genome))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.to.Genome))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.to.Genome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.to.Genome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Genome))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Genome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Genome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1))),ncol=6)

#reads mapped to intergenic regions, intronic regions, exonic regions, transcriptome, and antisense to gene
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.h.full, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+geom_col()+labs(title="c.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Reads.Mapped.Antisense.to.Gene))+geom_col()+labs(title="c.orfl")+ylim(0,1))), ncol=5)
CombinePlots(plots=list((ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intergenic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Intronic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Exonic.Regions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Reads.Mapped.Confidently.to.Transcriptome))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1))),ncol=12)

#fraction of reads in cells
CombinePlots(plots=list((ggplot(metrics.h.full, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="h.full")+ylim(0,1)),
                        (ggplot(metrics.c.full, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="c.full")+ylim(0,1)),
                        (ggplot(metrics.h.orth, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="h.orth")+ylim(0,1)),
                        (ggplot(metrics.c.orth, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="c.orth")+ylim(0,1)),
                        (ggplot(metrics.h.orfl, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="h.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.orfl, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="c.orfl")+ylim(0,1))), ncol=2)
CombinePlots(plots=list((ggplot(metrics.hc.full, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Fraction.Reads.in.Cells))+
                           geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Fraction.Reads.in.Cells))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Fraction.Reads.in.Cells))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,1))), ncol=3)

#estimated number of cell partitions
CombinePlots(plots=list((ggplot(metrics.hc.full, aes(x=Sample, y=GRCh38.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.full")+ylim(0,6500)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Pan.tro.3.0.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.full")+ylim(0,6500)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.GRCh38.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,6500)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=ortho.Pan.tro.3.0.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.orth")+ylim(0,6500)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.GRCh38.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,6500)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=ortho.filter.Pan.tro.3.0.Estimated.Number.of.Cell.Partitions))+
                           geom_col()+labs(title="hc.orfl")+ylim(0,6500))), ncol=2)
CombinePlots(plots=list((ggplot(metrics.hc.full, aes(x=Sample, y=GEMs.with..0.Cells))+geom_col()+labs(title="hc.full")+ylim(0,9000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=GEMs.with..1.Cell))+geom_col()+labs(title="hc.full")+ylim(0,9000)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Fraction.GEMs.with..1.Cell))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Lower.Bound.))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.full, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Upper.Bound.))+geom_col()+labs(title="hc.full")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=GEMs.with..0.Cells))+geom_col()+labs(title="hc.orth")+ylim(0,9000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=GEMs.with..1.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,9000)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Fraction.GEMs.with..1.Cell))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Lower.Bound.))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orth, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Upper.Bound.))+geom_col()+labs(title="hc.orth")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=GEMs.with..0.Cells))+geom_col()+labs(title="hc.orfl")+ylim(0,9000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=GEMs.with..1.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,9000)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Fraction.GEMs.with..1.Cell))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Lower.Bound.))+geom_col()+labs(title="hc.orfl")+ylim(0,1)),
                        (ggplot(metrics.hc.orfl, aes(x=Sample, y=Fraction.GEMs.with..1.Cell..Upper.Bound.))+geom_col()+labs(title="hc.orfl")+ylim(0,1))),
             ncol=5)

rm(metrics.h.full,metrics.c.full,metrics.hc.full)
rm(metrics.h.orth,metrics.c.orth,metrics.hc.orth)
rm(metrics.h.orfl,metrics.c.orfl,metrics.hc.orfl)

```

```{r load species assignments}

#Load species assignment (only possible for human-chimp combo reference)
hc.full.spp1 <- read.csv(file=paste0(dir_hc.full.GHO1,"/analysis/gem_classification.csv"))
hc.full.spp2 <- read.csv(file=paste0(dir_hc.full.GHO2,"/analysis/gem_classification.csv"))
hc.full.spp3 <- read.csv(file=paste0(dir_hc.full.GHO3,"/analysis/gem_classification.csv"))
hc.full.spp4 <- read.csv(file=paste0(dir_hc.full.GHO4,"/analysis/gem_classification.csv"))
hc.full.spp1$barcode <- sapply(hc.full.spp1$barcode, gsub, pattern="-1", replacement="")
hc.full.spp2$barcode <- sapply(hc.full.spp2$barcode, gsub, pattern="-1", replacement="")
hc.full.spp3$barcode <- sapply(hc.full.spp3$barcode, gsub, pattern="-1", replacement="")
hc.full.spp4$barcode <- sapply(hc.full.spp4$barcode, gsub, pattern="-1", replacement="")
hc.full.spp1$call <- sapply(hc.full.spp1$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.full.spp2$call <- sapply(hc.full.spp2$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.full.spp3$call <- sapply(hc.full.spp3$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.full.spp4$call <- sapply(hc.full.spp4$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.full.spp1$call <- sapply(hc.full.spp1$call, gsub, pattern="GRCh38", replacement="Human")
hc.full.spp2$call <- sapply(hc.full.spp2$call, gsub, pattern="GRCh38", replacement="Human")
hc.full.spp3$call <- sapply(hc.full.spp3$call, gsub, pattern="GRCh38", replacement="Human")
hc.full.spp4$call <- sapply(hc.full.spp4$call, gsub, pattern="GRCh38", replacement="Human")

hc.orth.spp1 <- read.csv(file=paste0(dir_hc.orth.GHO1,"/analysis/gem_classification.csv"))
hc.orth.spp2 <- read.csv(file=paste0(dir_hc.orth.GHO2,"/analysis/gem_classification.csv"))
hc.orth.spp3 <- read.csv(file=paste0(dir_hc.orth.GHO3,"/analysis/gem_classification.csv"))
hc.orth.spp4 <- read.csv(file=paste0(dir_hc.orth.GHO4,"/analysis/gem_classification.csv"))
hc.orth.spp1$barcode <- sapply(hc.orth.spp1$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp2$barcode <- sapply(hc.orth.spp2$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp3$barcode <- sapply(hc.orth.spp3$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp4$barcode <- sapply(hc.orth.spp4$barcode, gsub, pattern="-1", replacement="")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="GRCh38", replacement="Human")
hc.orth.spp1$call <- sapply(hc.orth.spp1$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp2$call <- sapply(hc.orth.spp2$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp3$call <- sapply(hc.orth.spp3$call, gsub, pattern="ortho_", replacement="")
hc.orth.spp4$call <- sapply(hc.orth.spp4$call, gsub, pattern="ortho_", replacement="")

hc.orfl.spp1 <- read.csv(file=paste0(dir_hc.orfl.GHO1,"/analysis/gem_classification.csv"))
hc.orfl.spp2 <- read.csv(file=paste0(dir_hc.orfl.GHO2,"/analysis/gem_classification.csv"))
hc.orfl.spp3 <- read.csv(file=paste0(dir_hc.orfl.GHO3,"/analysis/gem_classification.csv"))
hc.orfl.spp4 <- read.csv(file=paste0(dir_hc.orfl.GHO4,"/analysis/gem_classification.csv"))
hc.orfl.spp1$barcode <- sapply(hc.orfl.spp1$barcode, gsub, pattern="-1", replacement="")
hc.orfl.spp2$barcode <- sapply(hc.orfl.spp2$barcode, gsub, pattern="-1", replacement="")
hc.orfl.spp3$barcode <- sapply(hc.orfl.spp3$barcode, gsub, pattern="-1", replacement="")
hc.orfl.spp4$barcode <- sapply(hc.orfl.spp4$barcode, gsub, pattern="-1", replacement="")
hc.orfl.spp1$call <- sapply(hc.orfl.spp1$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orfl.spp2$call <- sapply(hc.orfl.spp2$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orfl.spp3$call <- sapply(hc.orfl.spp3$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orfl.spp4$call <- sapply(hc.orfl.spp4$call, gsub, pattern="Pan_tro_3.0", replacement="Chimp")
hc.orfl.spp1$call <- sapply(hc.orfl.spp1$call, gsub, pattern="GRCh38", replacement="Human")
hc.orfl.spp2$call <- sapply(hc.orfl.spp2$call, gsub, pattern="GRCh38", replacement="Human")
hc.orfl.spp3$call <- sapply(hc.orfl.spp3$call, gsub, pattern="GRCh38", replacement="Human")
hc.orfl.spp4$call <- sapply(hc.orfl.spp4$call, gsub, pattern="GRCh38", replacement="Human")
hc.orfl.spp1$call <- sapply(hc.orfl.spp1$call, gsub, pattern="ortho_filter_", replacement="")
hc.orfl.spp2$call <- sapply(hc.orfl.spp2$call, gsub, pattern="ortho_filter_", replacement="")
hc.orfl.spp3$call <- sapply(hc.orfl.spp3$call, gsub, pattern="ortho_filter_", replacement="")
hc.orfl.spp4$call <- sapply(hc.orfl.spp4$call, gsub, pattern="ortho_filter_", replacement="")

#Examine species assignment data    #FULL DATA                                #SUBSET DATA
table(hc.full.spp1$call)            #Chimp=1837, Human=2949, Multiplet=1976   #Chimp=3281, Human=2400, Multiplet= 118
table(hc.full.spp2$call)            #Chimp=3241, Human=2629, Multiplet= 182   #Chimp=3104, Human=2543, Multiplet= 124
table(hc.full.spp3$call)            #Chimp= 931, Human=4347, Multiplet=2750   #Chimp= 792, Human=4109, Multiplet= 931
table(hc.full.spp4$call)            #Chimp= 807, Human=1952, Multiplet=2097   #Chimp= 823, Human=1824, Multiplet=1266
table(hc.orth.spp1$call)            #Chimp=3294, Human=2319, Multiplet=  87   #Chimp=3253, Human=2260, Multiplet=  77
table(hc.orth.spp2$call)            #Chimp=3149, Human=2563, Multiplet= 123   #Chimp=3025, Human=2484, Multiplet= 113
table(hc.orth.spp3$call)            #Chimp=1305, Human=2168, Multiplet= 114   #Chimp=1074, Human=2123, Multiplet=  56
table(hc.orth.spp4$call)            #Chimp=1805, Human=1580, Multiplet=  62   #Chimp=1697, Human=1400, Multiplet=  28
table(hc.orfl.spp1$call)            #Chimp=3296, Human=2318, Multiplet=  82   ---
table(hc.orfl.spp2$call)            #Chimp=3149, Human=2557, Multiplet= 122   ---
table(hc.orfl.spp3$call)            #Chimp=1290, Human=2166, Multiplet= 105   ---
table(hc.orfl.spp4$call)            #Chimp=1828, Human=1586, Multiplet=  42   ---

CombinePlots(plots=list((ggplot(hc.full.spp1, aes(x=call))+geom_bar()+labs(title="iPSC-full")+ylim(0,4500)),
                        (ggplot(hc.full.spp2, aes(x=call))+geom_bar()+labs(title="MSC-full")+ylim(0,4500)),
                        (ggplot(hc.full.spp3, aes(x=call))+geom_bar()+labs(title="chondro-full")+ylim(0,4500)),
                        (ggplot(hc.full.spp4, aes(x=call))+geom_bar()+labs(title="osteo-full")+ylim(0,4500)),
                        (ggplot(hc.orth.spp1, aes(x=call))+geom_bar()+labs(title="iPSC-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp2, aes(x=call))+geom_bar()+labs(title="MSC-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp3, aes(x=call))+geom_bar()+labs(title="chondro-orth")+ylim(0,4500)),
                        (ggplot(hc.orth.spp4, aes(x=call))+geom_bar()+labs(title="osteo-orth")+ylim(0,4500)),
                        (ggplot(hc.orfl.spp1, aes(x=call))+geom_bar()+labs(title="iPSC-orfl")+ylim(0,4500)),
                        (ggplot(hc.orfl.spp2, aes(x=call))+geom_bar()+labs(title="MSC-orfl")+ylim(0,4500)),
                        (ggplot(hc.orfl.spp3, aes(x=call))+geom_bar()+labs(title="chondro-orfl")+ylim(0,4500)),
                        (ggplot(hc.orfl.spp4, aes(x=call))+geom_bar()+labs(title="osteo-orfl")+ylim(0,4500))), ncol=4)

#Data using full references contain more cells than data using ortho-exon references.
#Data using ortho-exon references have more even human-chimp mixtures and fewer multiplets than data using full references.

```

```{r prepare scRNA data}


#raw data will have all cells for species assignment but filtered data may not
#use only raw data for now (can compare to filtered later if necessary)
#example: h.full.GHO3_data when compared to hc.full.spp3 - filtered data was missing 53 human + 67 chimp barcodes


#Steps for preparing data:
#(1)load data
#(2)create seurat objects
#(3)view details of seurat objects
#(4)clear raw data from workspace
#(5)add species assignments to data
#(6)view details of species assignments
#(7)keep some data that contains multiplets
#(7)subset data to only include human and chimp cells/genes
#(8)create merged object
#(9)clear excess data from workspace


#Prepare data for hc.orth.sppo
#(1)load data
  hc.orth.GHO1_data <- Read10X(data=paste0(dir_hc.orth.GHO1,"/filtered_feature_bc_matrix"))
  hc.orth.GHO2_data <- Read10X(data=paste0(dir_hc.orth.GHO2,"/filtered_feature_bc_matrix"))
  hc.orth.GHO3_data <- Read10X(data=paste0(dir_hc.orth.GHO3,"/filtered_feature_bc_matrix"))
  hc.orth.GHO4_data <- Read10X(data=paste0(dir_hc.orth.GHO4,"/filtered_feature_bc_matrix"))
#(2)create seurat objects
  hc.orth.GHO1 <- CreateSeuratObject(counts=hc.orth.GHO1_data, project="iPSC")
  hc.orth.GHO2 <- CreateSeuratObject(counts=hc.orth.GHO2_data, project="MSC")
  hc.orth.GHO3 <- CreateSeuratObject(counts=hc.orth.GHO3_data, project="chondrocyte")
  hc.orth.GHO4 <- CreateSeuratObject(counts=hc.orth.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                      #SUBSET DATA
  hc.orth.GHO1                      #88250 features, 5700 samples   #88250 features, 5590 samples
  hc.orth.GHO2                      #88250 features, 5835 samples   #88250 features, 5622 samples
  hc.orth.GHO3                      #88250 features, 3587 samples   #88250 features, 3253 samples
  hc.orth.GHO4                      #88250 features, 3447 samples   #88250 features, 3125 samples
#(4)clear raw data from workspace
  rm(hc.orth.GHO1_data,hc.orth.GHO2_data,hc.orth.GHO3_data,hc.orth.GHO4_data)
#(5)add species assignments to data
  m <- match(rownames(hc.orth.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO1 <- AddMetaData(hc.orth.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO2 <- AddMetaData(hc.orth.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO3 <- AddMetaData(hc.orth.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO4 <- AddMetaData(hc.orth.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments                   #FULL DATA                        #SUBSET DATA
  table(hc.orth.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=   0  #C=3253, H=2260, M=  77, NA=   0
  table(hc.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=   0  #C=3025, H=2484, M= 113, NA=   0
  table(hc.orth.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=   0  #C=1074, H=2123, M=  56, NA=   0
  table(hc.orth.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=   0  #C=1697, H=1400, M=  28, NA=   0
#(7)keep some data that contains multiplets
  hc.orth.sppo.GHO1.tot <- hc.orth.GHO1
  hc.orth.sppo.GHO2.tot <- hc.orth.GHO2
  hc.orth.sppo.GHO3.tot <- hc.orth.GHO3
  hc.orth.sppo.GHO4.tot <- hc.orth.GHO4
  hc.orth.sppo.tot <- merge(hc.orth.sppo.GHO1.tot,c(hc.orth.sppo.GHO2.tot,hc.orth.sppo.GHO3.tot,hc.orth.sppo.GHO4.tot),
                            add.cell.ids=c("iPSC","MSC","chondro","osteo"),
                            project="hc.orth.sppo")
  saveRDS(hc.orth.sppo.tot, file = "../data/cellranger-data-full/hc.orth.sppo.tot.rds")
  rm(hc.orth.sppo.GHO1.tot,hc.orth.sppo.GHO2.tot,hc.orth.sppo.GHO3.tot,hc.orth.sppo.GHO4.tot)
#(7)subset data to only include human and chimp cells/genes
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orth.sppo.GHO1.h <- subset(hc.orth.GHO1, features=c(1:44125), subset=species=="Human")
  hc.orth.sppo.GHO1.c <- subset(hc.orth.GHO1, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppo.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppo.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppo.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppo.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppo.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppo.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)
  h.gene=c()
  c.gene=c()
  hc.orth.sppo.GHO2.h <- subset(hc.orth.GHO2, features=c(1:44125), subset=species=="Human")
  hc.orth.sppo.GHO2.c <- subset(hc.orth.GHO2, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppo.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppo.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppo.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppo.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppo.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppo.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)
  h.gene=c()
  c.gene=c()
  hc.orth.sppo.GHO3.h <- subset(hc.orth.GHO3, features=c(1:44125), subset=species=="Human")
  hc.orth.sppo.GHO3.c <- subset(hc.orth.GHO3, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppo.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppo.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppo.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppo.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppo.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppo.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)
  h.gene=c()
  c.gene=c()
  hc.orth.sppo.GHO4.h <- subset(hc.orth.GHO4, features=c(1:44125), subset=species=="Human")
  hc.orth.sppo.GHO4.c <- subset(hc.orth.GHO4, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppo.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppo.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppo.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppo.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppo.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppo.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object
  hc.orth.sppo <- merge(hc.orth.sppo.GHO1.h,c(hc.orth.sppo.GHO1.c,
                        hc.orth.sppo.GHO2.h,hc.orth.sppo.GHO2.c,
                        hc.orth.sppo.GHO3.h,hc.orth.sppo.GHO3.c,
                        hc.orth.sppo.GHO4.h,hc.orth.sppo.GHO4.c),
                      add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                      project="hc.orth.sppo")
  saveRDS(hc.orth.sppo, file = "../data/cellranger-data-full/hc.orth.sppo.rds")
#(9)clear excess data from workspace
  rm(hc.orth.GHO1,hc.orth.GHO2,hc.orth.GHO3,hc.orth.GHO4)
  rm(hc.orth.sppo.GHO1.h,hc.orth.sppo.GHO2.h,hc.orth.sppo.GHO3.h,hc.orth.sppo.GHO4.h)
  rm(hc.orth.sppo.GHO1.c,hc.orth.sppo.GHO2.c,hc.orth.sppo.GHO3.c,hc.orth.sppo.GHO4.c)


#Prepare data for hc.orth.sppf & hc.orth.spof
#(1)load data
  hc.orth.GHO1_data <- Read10X(data=paste0(dir_hc.orth.GHO1,"/raw_feature_bc_matrix"))
  hc.orth.GHO2_data <- Read10X(data=paste0(dir_hc.orth.GHO2,"/raw_feature_bc_matrix"))
  hc.orth.GHO3_data <- Read10X(data=paste0(dir_hc.orth.GHO3,"/raw_feature_bc_matrix"))
  hc.orth.GHO4_data <- Read10X(data=paste0(dir_hc.orth.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  hc.orth.GHO1 <- CreateSeuratObject(counts=hc.orth.GHO1_data, project="iPSC")
  hc.orth.GHO2 <- CreateSeuratObject(counts=hc.orth.GHO2_data, project="MSC")
  hc.orth.GHO3 <- CreateSeuratObject(counts=hc.orth.GHO3_data, project="chondrocyte")
  hc.orth.GHO4 <- CreateSeuratObject(counts=hc.orth.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  hc.orth.GHO1                      #88250 features, 6794880 samples  #88250 features, 6794880 samples
  hc.orth.GHO2                      #88250 features, 6794880 samples  #88250 features, 6794880 samples
  hc.orth.GHO3                      #88250 features, 6794880 samples  #88250 features, 6794880 samples
  hc.orth.GHO4                      #88250 features, 6794880 samples  #88250 features, 6794880 samples
#(4)clear raw data from workspace
  rm(hc.orth.GHO1_data,hc.orth.GHO2_data,hc.orth.GHO3_data,hc.orth.GHO4_data)
#steps for hc.orth.sppf
#(5)add species assignments to data - sppf
  m <- match(rownames(hc.orth.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO1 <- AddMetaData(hc.orth.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO2 <- AddMetaData(hc.orth.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO3 <- AddMetaData(hc.orth.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO4 <- AddMetaData(hc.orth.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppf            #FULL DATA                          #SUBSET DATA
  table(hc.orth.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118 #C=3281, H=2400, M= 118, NA=6789081
  table(hc.orth.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828 #C=3104, H=2543, M= 124, NA=6789109
  table(hc.orth.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852 #C= 792, H=4109, M= 931, NA=6789048
  table(hc.orth.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6790024 #C= 823, H=1824, M=1266, NA=6790967
#(7)subset data to only include human and chimp cells/genes - sppf
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orth.sppf.GHO1.h <- subset(hc.orth.GHO1, features=c(1:44125), subset=species=="Human")
  hc.orth.sppf.GHO1.c <- subset(hc.orth.GHO1, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppf.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppf.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppf.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppf.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppf.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppf.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  hc.orth.sppf.GHO2.h <- subset(hc.orth.GHO2, features=c(1:44125), subset=species=="Human")
  hc.orth.sppf.GHO2.c <- subset(hc.orth.GHO2, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppf.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppf.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppf.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppf.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppf.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppf.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  hc.orth.sppf.GHO3.h <- subset(hc.orth.GHO3, features=c(1:44125), subset=species=="Human")
  hc.orth.sppf.GHO3.c <- subset(hc.orth.GHO3, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppf.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppf.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppf.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppf.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppf.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppf.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  hc.orth.sppf.GHO4.h <- subset(hc.orth.GHO4, features=c(1:44125), subset=species=="Human")
  hc.orth.sppf.GHO4.c <- subset(hc.orth.GHO4, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.sppf.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.sppf.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.sppf.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.sppf.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.sppf.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.sppf.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - sppf
  hc.orth.sppf <- merge(hc.orth.sppf.GHO1.h,c(hc.orth.sppf.GHO1.c,
                          hc.orth.sppf.GHO2.h,hc.orth.sppf.GHO2.c,
                          hc.orth.sppf.GHO3.h,hc.orth.sppf.GHO3.c,
                          hc.orth.sppf.GHO4.h,hc.orth.sppf.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.orth.sppf")
  saveRDS(hc.orth.sppf, file = "../data/cellranger-data-full/hc.orth.sppf.rds")
#steps for hc.orth.spof
#(5)add species assignments to data - spof
  m <- match(rownames(hc.orth.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO1 <- AddMetaData(hc.orth.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO2 <- AddMetaData(hc.orth.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO3 <- AddMetaData(hc.orth.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orth.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orth.GHO4 <- AddMetaData(hc.orth.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - spof            #FULL DATA                          #SUBSET DATA
  table(hc.orth.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184 ---
  table(hc.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052 ---
  table(hc.orth.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319 ---
  table(hc.orth.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424 ---
#(7)subset data to only include human and chimp cells/genes - spof
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orth.spof.GHO1.h <- subset(hc.orth.GHO1, features=c(1:44125), subset=species=="Human")
  hc.orth.spof.GHO1.c <- subset(hc.orth.GHO1, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.spof.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.spof.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.spof.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.spof.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.spof.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.spof.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  hc.orth.spof.GHO2.h <- subset(hc.orth.GHO2, features=c(1:44125), subset=species=="Human")
  hc.orth.spof.GHO2.c <- subset(hc.orth.GHO2, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.spof.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.spof.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.spof.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.spof.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.spof.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.spof.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  hc.orth.spof.GHO3.h <- subset(hc.orth.GHO3, features=c(1:44125), subset=species=="Human")
  hc.orth.spof.GHO3.c <- subset(hc.orth.GHO3, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.spof.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.spof.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.spof.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.spof.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.spof.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.spof.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  hc.orth.spof.GHO4.h <- subset(hc.orth.GHO4, features=c(1:44125), subset=species=="Human")
  hc.orth.spof.GHO4.c <- subset(hc.orth.GHO4, features=c(44126:88250), subset=species=="Chimp")
  for (i in rownames(hc.orth.spof.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orth.spof.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orth.spof.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orth.spof.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orth.spof.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orth.spof.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - spof
  hc.orth.spof <- merge(hc.orth.spof.GHO1.h,c(hc.orth.spof.GHO1.c,
                          hc.orth.spof.GHO2.h,hc.orth.spof.GHO2.c,
                          hc.orth.spof.GHO3.h,hc.orth.spof.GHO3.c,
                          hc.orth.spof.GHO4.h,hc.orth.spof.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.orth.spof")
  saveRDS(hc.orth.spof, file = "../data/cellranger-data-full/hc.orth.spof.rds")
#(9)clear excess data from workspace
  rm(hc.orth.GHO1,hc.orth.GHO2,hc.orth.GHO3,hc.orth.GHO4)
  rm(hc.orth.sppf.GHO1.h,hc.orth.sppf.GHO2.h,hc.orth.sppf.GHO3.h,hc.orth.sppf.GHO4.h)
  rm(hc.orth.sppf.GHO1.c,hc.orth.sppf.GHO2.c,hc.orth.sppf.GHO3.c,hc.orth.sppf.GHO4.c)
  rm(hc.orth.spof.GHO1.h,hc.orth.spof.GHO2.h,hc.orth.spof.GHO3.h,hc.orth.spof.GHO4.h)
  rm(hc.orth.spof.GHO1.c,hc.orth.spof.GHO2.c,hc.orth.spof.GHO3.c,hc.orth.spof.GHO4.c)


#Prepare data for hc.full.sppf
#(1)load data
  hc.full.GHO1_data <- Read10X(data=paste0(dir_hc.full.GHO1,"/filtered_feature_bc_matrix"))
  hc.full.GHO2_data <- Read10X(data=paste0(dir_hc.full.GHO2,"/filtered_feature_bc_matrix"))
  hc.full.GHO3_data <- Read10X(data=paste0(dir_hc.full.GHO3,"/filtered_feature_bc_matrix"))
  hc.full.GHO4_data <- Read10X(data=paste0(dir_hc.full.GHO4,"/filtered_feature_bc_matrix"))
#(2)create seurat objects
  hc.full.GHO1 <- CreateSeuratObject(counts=hc.full.GHO1_data, project="iPSC")
  hc.full.GHO2 <- CreateSeuratObject(counts=hc.full.GHO2_data, project="MSC")
  hc.full.GHO3 <- CreateSeuratObject(counts=hc.full.GHO3_data, project="chondrocyte")
  hc.full.GHO4 <- CreateSeuratObject(counts=hc.full.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                      #SUBSET DATA
  hc.full.GHO1                      #58850 features, 6762 samples   #58850 features, 5799 samples
  hc.full.GHO2                      #58850 features, 6052 samples   #58850 features, 5771 samples
  hc.full.GHO3                      #58850 features, 8028 samples   #58850 features, 5832 samples
  hc.full.GHO4                      #58850 features, 4856 samples   #58850 features, 3913 samples
#(4)clear raw data from workspace
  rm(hc.full.GHO1_data,hc.full.GHO2_data,hc.full.GHO3_data,hc.full.GHO4_data)
#(5)add species assignments to data
  m <- match(rownames(hc.full.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO1 <- AddMetaData(hc.full.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO2 <- AddMetaData(hc.full.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO3 <- AddMetaData(hc.full.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO4 <- AddMetaData(hc.full.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments                   #FULL DATA                        #SUBSET DATA
  table(hc.full.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=   0  #C=3281, H=2400, M= 118, NA=   0
  table(hc.full.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=   0  #C=3104, H=2543, M= 124, NA=   0
  table(hc.full.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=   0  #C= 792, H=4109, M= 931, NA=   0
  table(hc.full.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=   0  #C= 823, H=1824, M=1266, NA=   0
#(7)subset data to only include human and chimp cells/genes
  hc.full.sppf.GHO1.tot <- hc.full.GHO1 #keep some data that contains multiplets
  hc.full.sppf.GHO2.tot <- hc.full.GHO2
  hc.full.sppf.GHO3.tot <- hc.full.GHO3
  hc.full.sppf.GHO4.tot <- hc.full.GHO4
  hc.full.sppf.tot <- merge(hc.full.sppf.GHO1.tot,c(hc.full.sppf.GHO2.tot,hc.full.sppf.GHO3.tot,hc.full.sppf.GHO4.tot),
                            add.cell.ids=c("iPSC","MSC","chondro","osteo"),
                            project="hc.full.sppf")
  saveRDS(hc.full.sppf.tot, file = "../data/cellranger-data-full/hc.full.sppf.tot.rds")
  rm(hc.full.sppf.GHO1.tot,hc.full.sppf.GHO2.tot,hc.full.sppf.GHO3.tot,hc.full.sppf.GHO4.tot)
#(7)subset data to only include human and chimp cells/genes
#notes: 16542 genes overlap between humans + chimps (otherwise 33538 human genes + 25312 chimp genes = 42308 genes total)
# (GHO1)
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppf.GHO1.h <- subset(hc.full.GHO1, features=h, subset=species=="Human")
  hc.full.sppf.GHO1.c <- subset(hc.full.GHO1, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppf.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppf.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppf.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppf.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppf.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppf.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppf.GHO2.h <- subset(hc.full.GHO2, features=h, subset=species=="Human")
  hc.full.sppf.GHO2.c <- subset(hc.full.GHO2, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppf.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppf.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppf.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppf.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppf.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppf.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)   
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppf.GHO3.h <- subset(hc.full.GHO3, features=h, subset=species=="Human")
  hc.full.sppf.GHO3.c <- subset(hc.full.GHO3, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppf.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppf.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppf.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppf.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppf.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppf.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppf.GHO4.h <- subset(hc.full.GHO4, features=h, subset=species=="Human")
  hc.full.sppf.GHO4.c <- subset(hc.full.GHO4, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppf.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppf.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppf.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppf.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppf.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppf.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object  
  hc.full.sppf <- merge(hc.full.sppf.GHO1.h,c(hc.full.sppf.GHO1.c,
                          hc.full.sppf.GHO2.h,hc.full.sppf.GHO2.c,
                          hc.full.sppf.GHO3.h,hc.full.sppf.GHO3.c,
                          hc.full.sppf.GHO4.h,hc.full.sppf.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.full.sppf")
  saveRDS(hc.full.sppf, file = "../data/cellranger-data-full/hc.full.sppf.rds")
#(9)clear excess data from workspace
  rm(hc.full.GHO1,hc.full.GHO2,hc.full.GHO3,hc.full.GHO4)
  rm(hc.full.sppf.GHO1.h,hc.full.sppf.GHO2.h,hc.full.sppf.GHO3.h,hc.full.sppf.GHO4.h)
  rm(hc.full.sppf.GHO1.c,hc.full.sppf.GHO2.c,hc.full.sppf.GHO3.c,hc.full.sppf.GHO4.c)
  
  
#Prepare data for hc.full.sppo & hc.full.sppf
#(1)load data
  hc.full.GHO1_data <- Read10X(data=paste0(dir_hc.full.GHO1,"/raw_feature_bc_matrix"))
  hc.full.GHO2_data <- Read10X(data=paste0(dir_hc.full.GHO2,"/raw_feature_bc_matrix"))
  hc.full.GHO3_data <- Read10X(data=paste0(dir_hc.full.GHO3,"/raw_feature_bc_matrix"))
  hc.full.GHO4_data <- Read10X(data=paste0(dir_hc.full.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  hc.full.GHO1 <- CreateSeuratObject(counts=hc.full.GHO1_data, project="iPSC")
  hc.full.GHO2 <- CreateSeuratObject(counts=hc.full.GHO2_data, project="MSC")
  hc.full.GHO3 <- CreateSeuratObject(counts=hc.full.GHO3_data, project="chondrocyte")
  hc.full.GHO4 <- CreateSeuratObject(counts=hc.full.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  hc.full.GHO1                      #58850 features, 6794880 samples  #58850 features, 6794880 samples
  hc.full.GHO2                      #58850 features, 6794880 samples  #58850 features, 6794880 samples
  hc.full.GHO3                      #58850 features, 6794880 samples  #58850 features, 6794880 samples
  hc.full.GHO4                      #58850 features, 6794880 samples  #58850 features, 6794880 samples
#(4)clear raw data from workspace
  rm(hc.full.GHO1_data,hc.full.GHO2_data,hc.full.GHO3_data,hc.full.GHO4_data)
#steps for hc.full.sppo
#(5)add species assignments to data - sppo
  m <- match(rownames(hc.full.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO1 <- AddMetaData(hc.full.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO2 <- AddMetaData(hc.full.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO3 <- AddMetaData(hc.full.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO4 <- AddMetaData(hc.full.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppo            #FULL DATA                          #SUBSET DATA
  table(hc.full.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180 #C=3253, H=2260, M=  77, NA=6789290
  table(hc.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045 #C=3025, H=2484, M= 113, NA=6789258
  table(hc.full.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293 #C=1074, H=2123, M=  56, NA=6791627
  table(hc.full.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433 #C=1697, H=1400, M=  28, NA=6791755
#(7)subset data to only include human and chimp cells/genes - sppo
#notes: 16542 genes overlap between humans + chimps (otherwise 33538 human genes + 25312 chimp genes = 42308 genes total)
# (GHO1)
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppo.GHO1.h <- subset(hc.full.GHO1, features=h, subset=species=="Human")
  hc.full.sppo.GHO1.c <- subset(hc.full.GHO1, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppo.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppo.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppo.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppo.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppo.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppo.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppo.GHO2.h <- subset(hc.full.GHO2, features=h, subset=species=="Human")
  hc.full.sppo.GHO2.c <- subset(hc.full.GHO2, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppo.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppo.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppo.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppo.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppo.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppo.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppo.GHO3.h <- subset(hc.full.GHO3, features=h, subset=species=="Human")
  hc.full.sppo.GHO3.c <- subset(hc.full.GHO3, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppo.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppo.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppo.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppo.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppo.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppo.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.sppo.GHO4.h <- subset(hc.full.GHO4, features=h, subset=species=="Human")
  hc.full.sppo.GHO4.c <- subset(hc.full.GHO4, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.sppo.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.sppo.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.sppo.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.sppo.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.full.sppo.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.sppo.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - sppo
  hc.full.sppo <- merge(hc.full.sppo.GHO1.h,c(hc.full.sppo.GHO1.c,
                          hc.full.sppo.GHO2.h,hc.full.sppo.GHO2.c,
                          hc.full.sppo.GHO3.h,hc.full.sppo.GHO3.c,
                          hc.full.sppo.GHO4.h,hc.full.sppo.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.full.sppo")
  saveRDS(hc.full.sppo, file = "../data/cellranger-data-full/hc.full.sppo.rds")
#steps for hc.full.spof
#(5)add species assignments to data - spof
  m <- match(rownames(hc.full.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO1 <- AddMetaData(hc.full.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO2 <- AddMetaData(hc.full.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO3 <- AddMetaData(hc.full.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.full.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.full.GHO4 <- AddMetaData(hc.full.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - spof            #FULL DATA                          #SUBSET DATA
  table(hc.full.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184 ---
  table(hc.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052 ---
  table(hc.full.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319 ---
  table(hc.full.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424 ---
#(7)subset data to only include human and chimp cells/genes - spof
#notes: 16542 genes overlap between humans + chimps (otherwise 33538 human genes + 25312 chimp genes = 42308 genes total)
# (GHO1)
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO1@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.spof.GHO1.h <- subset(hc.full.GHO1, features=h, subset=species=="Human")
  hc.full.spof.GHO1.c <- subset(hc.full.GHO1, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.spof.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.spof.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.spof.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.spof.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.full.spof.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.spof.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO2@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.spof.GHO2.h <- subset(hc.full.GHO2, features=h, subset=species=="Human")
  hc.full.spof.GHO2.c <- subset(hc.full.GHO2, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.spof.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.spof.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.spof.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.spof.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.full.spof.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.spof.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO3@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.spof.GHO3.h <- subset(hc.full.GHO3, features=h, subset=species=="Human")
  hc.full.spof.GHO3.c <- subset(hc.full.GHO3, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.spof.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.spof.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.spof.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.spof.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.full.spof.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.spof.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[1:33538]){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.GHO4@assays$RNA@counts)[33539:58850]){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  h=which(h.gene %in% c.gene)
  c=which(c.gene %in% h.gene)
  m <- match(h.gene[h],c.gene[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  c=c+33538
  hc.full.spof.GHO4.h <- subset(hc.full.GHO4, features=h, subset=species=="Human")
  hc.full.spof.GHO4.c <- subset(hc.full.GHO4, features=c, subset=species=="Chimp")
  h.gene=c()
  c.gene=c()
  for (i in rownames(hc.full.spof.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
  for (i in rownames(hc.full.spof.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.full.spof.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.full.spof.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.full.spof.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.full.spof.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - spof
  hc.full.spof <- merge(hc.full.spof.GHO1.h,c(hc.full.spof.GHO1.c,
                          hc.full.spof.GHO2.h,hc.full.spof.GHO2.c,
                          hc.full.spof.GHO3.h,hc.full.spof.GHO3.c,
                          hc.full.spof.GHO4.h,hc.full.spof.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.full.spof")
  saveRDS(hc.full.spof, file = "../data/cellranger-data-full/hc.full.spof.rds")
#(9)clear excess data from workspace
  rm(hc.full.GHO1,hc.full.GHO2,hc.full.GHO3,hc.full.GHO4)
  rm(hc.full.sppo.GHO1.h,hc.full.sppo.GHO2.h,hc.full.sppo.GHO3.h,hc.full.sppo.GHO4.h)
  rm(hc.full.sppo.GHO1.c,hc.full.sppo.GHO2.c,hc.full.sppo.GHO3.c,hc.full.sppo.GHO4.c)
  rm(hc.full.spof.GHO1.h,hc.full.spof.GHO2.h,hc.full.spof.GHO3.h,hc.full.spof.GHO4.h)
  rm(hc.full.spof.GHO1.c,hc.full.spof.GHO2.c,hc.full.spof.GHO3.c,hc.full.spof.GHO4.c)


#Prepare data for hc.orfl.spof
#(1)load data
  hc.orfl.GHO1_data <- Read10X(data=paste0(dir_hc.orfl.GHO1,"/filtered_feature_bc_matrix"))
  hc.orfl.GHO2_data <- Read10X(data=paste0(dir_hc.orfl.GHO2,"/filtered_feature_bc_matrix"))
  hc.orfl.GHO3_data <- Read10X(data=paste0(dir_hc.orfl.GHO3,"/filtered_feature_bc_matrix"))
  hc.orfl.GHO4_data <- Read10X(data=paste0(dir_hc.orfl.GHO4,"/filtered_feature_bc_matrix"))
#(2)create seurat objects
  hc.orfl.GHO1 <- CreateSeuratObject(counts=hc.orfl.GHO1_data, project="iPSC")
  hc.orfl.GHO2 <- CreateSeuratObject(counts=hc.orfl.GHO2_data, project="MSC")
  hc.orfl.GHO3 <- CreateSeuratObject(counts=hc.orfl.GHO3_data, project="chondrocyte")
  hc.orfl.GHO4 <- CreateSeuratObject(counts=hc.orfl.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                      #SUBSET DATA
  hc.orfl.GHO1                      #57884 features, 5696 samples   ---
  hc.orfl.GHO2                      #57884 features, 5828 samples   ---
  hc.orfl.GHO3                      #57884 features, 3561 samples   ---
  hc.orfl.GHO4                      #57884 features, 3456 samples   ---
#(4)clear raw data from workspace
  rm(hc.orfl.GHO1_data,hc.orfl.GHO2_data,hc.orfl.GHO3_data,hc.orfl.GHO4_data)
#(5)add species assignments to data
  m <- match(rownames(hc.orfl.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO1 <- AddMetaData(hc.orfl.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO2 <- AddMetaData(hc.orfl.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO3 <- AddMetaData(hc.orfl.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO4 <- AddMetaData(hc.orfl.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments                   #FULL DATA                        #SUBSET DATA
  table(hc.orfl.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=   0  ---
  table(hc.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=   0  ---
  table(hc.orfl.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=   0  ---
  table(hc.orfl.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=   0  ---
#(7)keep some data that contains multiplets
  hc.orfl.spof.GHO1.tot <- hc.orfl.GHO1
  hc.orfl.spof.GHO2.tot <- hc.orfl.GHO2
  hc.orfl.spof.GHO3.tot <- hc.orfl.GHO3
  hc.orfl.spof.GHO4.tot <- hc.orfl.GHO4
  hc.orfl.spof.tot <- merge(hc.orfl.spof.GHO1.tot,c(hc.orfl.spof.GHO2.tot,hc.orfl.spof.GHO3.tot,hc.orfl.spof.GHO4.tot),
                            add.cell.ids=c("iPSC","MSC","chondro","osteo"),
                            project="hc.orfl.spof")
  saveRDS(hc.orfl.spof.tot, file = "../data/cellranger-data-full/hc.orfl.spof.tot.rds")
  rm(hc.orfl.spof.GHO1.tot,hc.orfl.spof.GHO2.tot,hc.orfl.spof.GHO3.tot,hc.orfl.spof.GHO4.tot)
#(7)subset data to only include human and chimp cells/genes
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orfl.spof.GHO1.h <- subset(hc.orfl.GHO1, features=c(1:28942), subset=species=="Human")
  hc.orfl.spof.GHO1.c <- subset(hc.orfl.GHO1, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.spof.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.spof.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.spof.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.spof.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.spof.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.spof.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)
  h.gene=c()
  c.gene=c()
  hc.orfl.spof.GHO2.h <- subset(hc.orfl.GHO2, features=c(1:28942), subset=species=="Human")
  hc.orfl.spof.GHO2.c <- subset(hc.orfl.GHO2, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.spof.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.spof.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.spof.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.spof.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.spof.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.spof.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)
  h.gene=c()
  c.gene=c()
  hc.orfl.spof.GHO3.h <- subset(hc.orfl.GHO3, features=c(1:28942), subset=species=="Human")
  hc.orfl.spof.GHO3.c <- subset(hc.orfl.GHO3, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.spof.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.spof.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.spof.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.spof.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.spof.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.spof.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)
  h.gene=c()
  c.gene=c()
  hc.orfl.spof.GHO4.h <- subset(hc.orfl.GHO4, features=c(1:28942), subset=species=="Human")
  hc.orfl.spof.GHO4.c <- subset(hc.orfl.GHO4, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.spof.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.spof.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.spof.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.spof.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.spof.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.spof.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object
  hc.orfl.spof <- merge(hc.orfl.spof.GHO1.h,c(hc.orfl.spof.GHO1.c,
                        hc.orfl.spof.GHO2.h,hc.orfl.spof.GHO2.c,
                        hc.orfl.spof.GHO3.h,hc.orfl.spof.GHO3.c,
                        hc.orfl.spof.GHO4.h,hc.orfl.spof.GHO4.c),
                      add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                      project="hc.orfl.spof")
  saveRDS(hc.orfl.spof, file = "../data/cellranger-data-full/hc.orfl.spof.rds")
#(9)clear excess data from workspace
  rm(hc.orfl.GHO1,hc.orfl.GHO2,hc.orfl.GHO3,hc.orfl.GHO4)
  rm(hc.orfl.spof.GHO1.h,hc.orfl.spof.GHO2.h,hc.orfl.spof.GHO3.h,hc.orfl.spof.GHO4.h)
  rm(hc.orfl.spof.GHO1.c,hc.orfl.spof.GHO2.c,hc.orfl.spof.GHO3.c,hc.orfl.spof.GHO4.c)

  
#Prepare data for hc.orfl.sppo & hc.orfl.sppf
#(1)load data
  hc.orfl.GHO1_data <- Read10X(data=paste0(dir_hc.orfl.GHO1,"/raw_feature_bc_matrix"))
  hc.orfl.GHO2_data <- Read10X(data=paste0(dir_hc.orfl.GHO2,"/raw_feature_bc_matrix"))
  hc.orfl.GHO3_data <- Read10X(data=paste0(dir_hc.orfl.GHO3,"/raw_feature_bc_matrix"))
  hc.orfl.GHO4_data <- Read10X(data=paste0(dir_hc.orfl.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  hc.orfl.GHO1 <- CreateSeuratObject(counts=hc.orfl.GHO1_data, project="iPSC")
  hc.orfl.GHO2 <- CreateSeuratObject(counts=hc.orfl.GHO2_data, project="MSC")
  hc.orfl.GHO3 <- CreateSeuratObject(counts=hc.orfl.GHO3_data, project="chondrocyte")
  hc.orfl.GHO4 <- CreateSeuratObject(counts=hc.orfl.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  hc.orfl.GHO1                      #57884 features, 6794880 samples  ---
  hc.orfl.GHO2                      #57884 features, 6794880 samples  ---
  hc.orfl.GHO3                      #57884 features, 6794880 samples  ---
  hc.orfl.GHO4                      #57884 features, 6794880 samples  ---
#(4)clear raw data from workspace
  rm(hc.orfl.GHO1_data,hc.orfl.GHO2_data,hc.orfl.GHO3_data,hc.orfl.GHO4_data)
#steps for hc.orfl.sppo
#(5)add species assignments to data - sppo
  m <- match(rownames(hc.orfl.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO1 <- AddMetaData(hc.orfl.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO2 <- AddMetaData(hc.orfl.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO3 <- AddMetaData(hc.orfl.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO4 <- AddMetaData(hc.orfl.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppo            #FULL DATA                          #SUBSET DATA
  table(hc.orfl.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180 ---
  table(hc.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045 ---
  table(hc.orfl.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293 ---
  table(hc.orfl.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433 ---
#(7)subset data to only include human and chimp cells/genes - sppo
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orfl.sppo.GHO1.h <- subset(hc.orfl.GHO1, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppo.GHO1.c <- subset(hc.orfl.GHO1, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppo.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppo.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppo.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppo.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppo.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppo.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppo.GHO2.h <- subset(hc.orfl.GHO2, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppo.GHO2.c <- subset(hc.orfl.GHO2, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppo.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppo.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppo.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppo.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppo.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppo.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppo.GHO3.h <- subset(hc.orfl.GHO3, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppo.GHO3.c <- subset(hc.orfl.GHO3, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppo.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppo.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppo.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppo.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppo.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppo.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppo.GHO4.h <- subset(hc.orfl.GHO4, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppo.GHO4.c <- subset(hc.orfl.GHO4, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppo.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppo.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppo.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppo.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppo.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppo.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - sppo
  hc.orfl.sppo <- merge(hc.orfl.sppo.GHO1.h,c(hc.orfl.sppo.GHO1.c,
                          hc.orfl.sppo.GHO2.h,hc.orfl.sppo.GHO2.c,
                          hc.orfl.sppo.GHO3.h,hc.orfl.sppo.GHO3.c,
                          hc.orfl.sppo.GHO4.h,hc.orfl.sppo.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.orfl.sppo")
  saveRDS(hc.orfl.sppo, file = "../data/cellranger-data-full/hc.orfl.sppo.rds")
#steps for hc.orfl.sppf
#(5)add species assignments to data - sppf
  m <- match(rownames(hc.orfl.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO1 <- AddMetaData(hc.orfl.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO2 <- AddMetaData(hc.orfl.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO3 <- AddMetaData(hc.orfl.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(hc.orfl.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  hc.orfl.GHO4 <- AddMetaData(hc.orfl.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppf            #FULL DATA                          #SUBSET DATA
  table(hc.orfl.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118 ---
  table(hc.orfl.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828 ---
  table(hc.orfl.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852 ---
  table(hc.orfl.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6790024 ---
#(7)subset data to only include human and chimp cells/genes - sppf
# (GHO1)
  h.gene=c()
  c.gene=c()
  hc.orfl.sppf.GHO1.h <- subset(hc.orfl.GHO1, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppf.GHO1.c <- subset(hc.orfl.GHO1, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppf.GHO1.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppf.GHO1.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppf.GHO1.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppf.GHO1.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppf.GHO1.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppf.GHO1.c@assays$RNA@data) <- c.gene
# (GHO2)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppf.GHO2.h <- subset(hc.orfl.GHO2, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppf.GHO2.c <- subset(hc.orfl.GHO2, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppf.GHO2.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppf.GHO2.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppf.GHO2.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppf.GHO2.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppf.GHO2.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppf.GHO2.c@assays$RNA@data) <- c.gene
# (GHO3)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppf.GHO3.h <- subset(hc.orfl.GHO3, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppf.GHO3.c <- subset(hc.orfl.GHO3, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppf.GHO3.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppf.GHO3.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppf.GHO3.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppf.GHO3.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppf.GHO3.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppf.GHO3.c@assays$RNA@data) <- c.gene
# (GHO4)  
  h.gene=c()
  c.gene=c()
  hc.orfl.sppf.GHO4.h <- subset(hc.orfl.GHO4, features=c(1:28942), subset=species=="Human")
  hc.orfl.sppf.GHO4.c <- subset(hc.orfl.GHO4, features=c(28943:57884), subset=species=="Chimp")
  for (i in rownames(hc.orfl.sppf.GHO4.h@assays$RNA@counts)){ h.gene=c(h.gene,strsplit(i,"ortho-filter-GRCh38------")[[1]][2]) }
  for (i in rownames(hc.orfl.sppf.GHO4.c@assays$RNA@counts)){ c.gene=c(c.gene,strsplit(i,"ortho-filter-Pan-tro-3.0-")[[1]][2]) }
  m <- match(h.gene,c.gene)
  if(any(is.na(m))) cat("Not all genes match.\n")
  rownames(hc.orfl.sppf.GHO4.h@assays$RNA@counts) <- h.gene
  rownames(hc.orfl.sppf.GHO4.h@assays$RNA@data) <- h.gene
  rownames(hc.orfl.sppf.GHO4.c@assays$RNA@counts) <- c.gene
  rownames(hc.orfl.sppf.GHO4.c@assays$RNA@data) <- c.gene
#(8)create merged object - sppf
  hc.orfl.sppf <- merge(hc.orfl.sppf.GHO1.h,c(hc.orfl.sppf.GHO1.c,
                          hc.orfl.sppf.GHO2.h,hc.orfl.sppf.GHO2.c,
                          hc.orfl.sppf.GHO3.h,hc.orfl.sppf.GHO3.c,
                          hc.orfl.sppf.GHO4.h,hc.orfl.sppf.GHO4.c),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="hc.orfl.sppf")
  saveRDS(hc.orfl.sppf, file = "../data/cellranger-data-full/hc.orfl.sppf.rds")
#(9)clear excess data from workspace
  rm(hc.orfl.GHO1,hc.orfl.GHO2,hc.orfl.GHO3,hc.orfl.GHO4)
  rm(hc.orfl.sppo.GHO1.h,hc.orfl.sppo.GHO2.h,hc.orfl.sppo.GHO3.h,hc.orfl.sppo.GHO4.h)
  rm(hc.orfl.sppo.GHO1.c,hc.orfl.sppo.GHO2.c,hc.orfl.sppo.GHO3.c,hc.orfl.sppo.GHO4.c)
  rm(hc.orfl.sppf.GHO1.h,hc.orfl.sppf.GHO2.h,hc.orfl.sppf.GHO3.h,hc.orfl.sppf.GHO4.h)
  rm(hc.orfl.sppf.GHO1.c,hc.orfl.sppf.GHO2.c,hc.orfl.sppf.GHO3.c,hc.orfl.sppf.GHO4.c)


#Prepare data for h.orth.c.ortho.sppo & h.orth.c.orth.sppf & h.orth.c.orth.spof
#(1)load data
  h.orth.GHO1_data <- Read10X(data=paste0(dir_h.orth.GHO1,"/raw_feature_bc_matrix"))
  h.orth.GHO2_data <- Read10X(data=paste0(dir_h.orth.GHO2,"/raw_feature_bc_matrix"))
  h.orth.GHO3_data <- Read10X(data=paste0(dir_h.orth.GHO3,"/raw_feature_bc_matrix"))
  h.orth.GHO4_data <- Read10X(data=paste0(dir_h.orth.GHO4,"/raw_feature_bc_matrix"))
  c.orth.GHO1_data <- Read10X(data=paste0(dir_c.orth.GHO1,"/raw_feature_bc_matrix"))
  c.orth.GHO2_data <- Read10X(data=paste0(dir_c.orth.GHO2,"/raw_feature_bc_matrix"))
  c.orth.GHO3_data <- Read10X(data=paste0(dir_c.orth.GHO3,"/raw_feature_bc_matrix"))
  c.orth.GHO4_data <- Read10X(data=paste0(dir_c.orth.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  h.orth.GHO1 <- CreateSeuratObject(counts=h.orth.GHO1_data, project="iPSC")
  h.orth.GHO2 <- CreateSeuratObject(counts=h.orth.GHO2_data, project="MSC")
  h.orth.GHO3 <- CreateSeuratObject(counts=h.orth.GHO3_data, project="chondrocyte")
  h.orth.GHO4 <- CreateSeuratObject(counts=h.orth.GHO4_data, project="osteoblast")
  c.orth.GHO1 <- CreateSeuratObject(counts=c.orth.GHO1_data, project="iPSC")
  c.orth.GHO2 <- CreateSeuratObject(counts=c.orth.GHO2_data, project="MSC")
  c.orth.GHO3 <- CreateSeuratObject(counts=c.orth.GHO3_data, project="chondrocyte")
  c.orth.GHO4 <- CreateSeuratObject(counts=c.orth.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  h.orth.GHO1                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  h.orth.GHO2                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  h.orth.GHO3                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  h.orth.GHO4                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  c.orth.GHO1                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  c.orth.GHO2                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  c.orth.GHO3                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
  c.orth.GHO4                       #44125 features, 6794880 samples  #44125 features, 6794880 samples
#(4)clear raw data from workspace
  rm(h.orth.GHO1_data,h.orth.GHO2_data,h.orth.GHO3_data,h.orth.GHO4_data)
  rm(c.orth.GHO1_data,c.orth.GHO2_data,c.orth.GHO3_data,c.orth.GHO4_data)
#steps for h.orth.c.orth.sppo
#(5)add species assignments to data - sppo
  m <- match(rownames(h.orth.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO1 <- AddMetaData(h.orth.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO2 <- AddMetaData(h.orth.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO3 <- AddMetaData(h.orth.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO4 <- AddMetaData(h.orth.GHO4, hc.orth.spp4[m,]$call, col.name="species")
    m <- match(rownames(c.orth.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO1 <- AddMetaData(c.orth.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO2 <- AddMetaData(c.orth.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO3 <- AddMetaData(c.orth.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO4 <- AddMetaData(c.orth.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppo           #FULL DATA                           #SUBSET DATA
  table(h.orth.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(h.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(h.orth.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(h.orth.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
  table(c.orth.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(c.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(c.orth.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(c.orth.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
#(7)subset data to only include human and chimp cells/genes - sppo
# (GHO1)
  h.orth.sppo.GHO1 <- subset(h.orth.GHO1, subset=species=="Human")
  c.orth.sppo.GHO1 <- subset(c.orth.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orth.sppo.GHO2 <- subset(h.orth.GHO2, subset=species=="Human")
  c.orth.sppo.GHO2 <- subset(c.orth.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orth.sppo.GHO3 <- subset(h.orth.GHO3, subset=species=="Human")
  c.orth.sppo.GHO3 <- subset(c.orth.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orth.sppo.GHO4 <- subset(h.orth.GHO4, subset=species=="Human")
  c.orth.sppo.GHO4 <- subset(c.orth.GHO4, subset=species=="Chimp")
#(8)create merged object - sppo
  h.orth.c.orth.sppo <- merge(h.orth.sppo.GHO1,c(c.orth.sppo.GHO1,
                          h.orth.sppo.GHO2,c.orth.sppo.GHO2,
                          h.orth.sppo.GHO3,c.orth.sppo.GHO3,
                          h.orth.sppo.GHO4,c.orth.sppo.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orth.c.orth.sppo")
  saveRDS(h.orth.c.orth.sppo, file = "../data/cellranger-data-full/h.orth.c.orth.sppo.rds")
#steps for h.orth.c.orth.sppf
#(5)add species assignments to data - sppf
  m <- match(rownames(h.orth.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO1 <- AddMetaData(h.orth.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO2 <- AddMetaData(h.orth.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO3 <- AddMetaData(h.orth.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO4 <- AddMetaData(h.orth.GHO4, hc.full.spp4[m,]$call, col.name="species")
    m <- match(rownames(c.orth.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO1 <- AddMetaData(c.orth.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO2 <- AddMetaData(c.orth.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO3 <- AddMetaData(c.orth.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO4 <- AddMetaData(c.orth.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppf           #FULL DATA                           #SUBSET DATA
  table(h.orth.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(h.orth.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(h.orth.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(h.orth.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
  table(c.orth.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(c.orth.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(c.orth.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(c.orth.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
#(7)subset data to only include human and chimp cells/genes - sppf
# (GHO1)
  h.orth.sppf.GHO1 <- subset(h.orth.GHO1, subset=species=="Human")
  c.orth.sppf.GHO1 <- subset(c.orth.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orth.sppf.GHO2 <- subset(h.orth.GHO2, subset=species=="Human")
  c.orth.sppf.GHO2 <- subset(c.orth.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orth.sppf.GHO3 <- subset(h.orth.GHO3, subset=species=="Human")
  c.orth.sppf.GHO3 <- subset(c.orth.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orth.sppf.GHO4 <- subset(h.orth.GHO4, subset=species=="Human")
  c.orth.sppf.GHO4 <- subset(c.orth.GHO4, subset=species=="Chimp")
#(8)create merged object - sppf
  h.orth.c.orth.sppf <- merge(h.orth.sppf.GHO1,c(c.orth.sppf.GHO1,
                          h.orth.sppf.GHO2,c.orth.sppf.GHO2,
                          h.orth.sppf.GHO3,c.orth.sppf.GHO3,
                          h.orth.sppf.GHO4,c.orth.sppf.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orth.c.orth.sppf")
  saveRDS(h.orth.c.orth.sppf, file = "../data/cellranger-data-full/h.orth.c.orth.sppf.rds")
#steps for h.orth.c.orth.spof
#(5)add species assignments to data - spof
  m <- match(rownames(h.orth.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO1 <- AddMetaData(h.orth.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO2 <- AddMetaData(h.orth.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO3 <- AddMetaData(h.orth.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orth.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orth.GHO4 <- AddMetaData(h.orth.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO1 <- AddMetaData(c.orth.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO2 <- AddMetaData(c.orth.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO3 <- AddMetaData(c.orth.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orth.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orth.GHO4 <- AddMetaData(c.orth.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - spof           #FULL DATA                           #SUBSET DATA
  table(h.orth.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(h.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(h.orth.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(h.orth.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
  table(c.orth.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(c.orth.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(c.orth.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(c.orth.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
#(7)subset data to only include human and chimp cells/genes - spof
# (GHO1)
  h.orth.spof.GHO1 <- subset(h.orth.GHO1, subset=species=="Human")
  c.orth.spof.GHO1 <- subset(c.orth.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orth.spof.GHO2 <- subset(h.orth.GHO2, subset=species=="Human")
  c.orth.spof.GHO2 <- subset(c.orth.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orth.spof.GHO3 <- subset(h.orth.GHO3, subset=species=="Human")
  c.orth.spof.GHO3 <- subset(c.orth.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orth.spof.GHO4 <- subset(h.orth.GHO4, subset=species=="Human")
  c.orth.spof.GHO4 <- subset(c.orth.GHO4, subset=species=="Chimp")
#(8)create merged object - spof
  h.orth.c.orth.spof <- merge(h.orth.spof.GHO1,c(c.orth.spof.GHO1,
                          h.orth.spof.GHO2,c.orth.spof.GHO2,
                          h.orth.spof.GHO3,c.orth.spof.GHO3,
                          h.orth.spof.GHO4,c.orth.spof.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orth.c.orth.spof")
  saveRDS(h.orth.c.orth.spof, file = "../data/cellranger-data-full/h.orth.c.orth.spof.rds")
#(9)clear excess data from workspace
  rm(h.orth.GHO1,h.orth.GHO2,h.orth.GHO3,h.orth.GHO4) 
  rm(c.orth.GHO1,c.orth.GHO2,c.orth.GHO3,c.orth.GHO4)
  rm(h.orth.sppo.GHO1,h.orth.sppo.GHO2,h.orth.sppo.GHO3,h.orth.sppo.GHO4)
  rm(c.orth.sppo.GHO1,c.orth.sppo.GHO2,c.orth.sppo.GHO3,c.orth.sppo.GHO4)
  rm(h.orth.sppf.GHO1,h.orth.sppf.GHO2,h.orth.sppf.GHO3,h.orth.sppf.GHO4)
  rm(c.orth.sppf.GHO1,c.orth.sppf.GHO2,c.orth.sppf.GHO3,c.orth.sppf.GHO4)
  rm(h.orth.spof.GHO1,h.orth.spof.GHO2,h.orth.spof.GHO3,h.orth.spof.GHO4)
  rm(c.orth.spof.GHO1,c.orth.spof.GHO2,c.orth.spof.GHO3,c.orth.spof.GHO4)

  
#Prepare data for h.full.c.full.sppo & h.full.c.full.sppf & h.full.c.full.spof
#(1)load data
  h.full.GHO1_data <- Read10X(data=paste0(dir_h.full.GHO1,"/raw_feature_bc_matrix"))
  h.full.GHO2_data <- Read10X(data=paste0(dir_h.full.GHO2,"/raw_feature_bc_matrix"))
  h.full.GHO3_data <- Read10X(data=paste0(dir_h.full.GHO3,"/raw_feature_bc_matrix"))
  h.full.GHO4_data <- Read10X(data=paste0(dir_h.full.GHO4,"/raw_feature_bc_matrix"))
  c.full.GHO1_data <- Read10X(data=paste0(dir_c.full.GHO1,"/raw_feature_bc_matrix"))
  c.full.GHO2_data <- Read10X(data=paste0(dir_c.full.GHO2,"/raw_feature_bc_matrix"))
  c.full.GHO3_data <- Read10X(data=paste0(dir_c.full.GHO3,"/raw_feature_bc_matrix"))
  c.full.GHO4_data <- Read10X(data=paste0(dir_c.full.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  h.full.GHO1 <- CreateSeuratObject(counts=h.full.GHO1_data, project="iPSC")
  h.full.GHO2 <- CreateSeuratObject(counts=h.full.GHO2_data, project="MSC")
  h.full.GHO3 <- CreateSeuratObject(counts=h.full.GHO3_data, project="chondrocyte")
  h.full.GHO4 <- CreateSeuratObject(counts=h.full.GHO4_data, project="osteoblast")
  c.full.GHO1 <- CreateSeuratObject(counts=c.full.GHO1_data, project="iPSC")
  c.full.GHO2 <- CreateSeuratObject(counts=c.full.GHO2_data, project="MSC")
  c.full.GHO3 <- CreateSeuratObject(counts=c.full.GHO3_data, project="chondrocyte")
  c.full.GHO4 <- CreateSeuratObject(counts=c.full.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  h.full.GHO1                       #33538 features, 6794880 samples  #33538 features, 6794880 samples
  h.full.GHO2                       #33538 features, 6794880 samples  #33538 features, 6794880 samples
  h.full.GHO3                       #33538 features, 6794880 samples  #33538 features, 6794880 samples
  h.full.GHO4                       #33538 features, 6794880 samples  #33538 features, 6794880 samples
  c.full.GHO1                       #25312 features, 6794880 samples  #25312 features, 6794880 samples
  c.full.GHO2                       #25312 features, 6794880 samples  #25312 features, 6794880 samples
  c.full.GHO3                       #25312 features, 6794880 samples  #25312 features, 6794880 samples
  c.full.GHO4                       #25312 features, 6794880 samples  #25312 features, 6794880 samples
#(4)clear raw data from workspace
  rm(h.full.GHO1_data,h.full.GHO2_data,h.full.GHO3_data,h.full.GHO4_data)
  rm(c.full.GHO1_data,c.full.GHO2_data,c.full.GHO3_data,c.full.GHO4_data)
#steps for h.full.c.full.sppo
#(5)add species assignments to data - sppo
  m <- match(rownames(h.full.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO1 <- AddMetaData(h.full.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO2 <- AddMetaData(h.full.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO3 <- AddMetaData(h.full.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO4 <- AddMetaData(h.full.GHO4, hc.orth.spp4[m,]$call, col.name="species")
    m <- match(rownames(c.full.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO1 <- AddMetaData(c.full.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO2 <- AddMetaData(c.full.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO3 <- AddMetaData(c.full.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO4 <- AddMetaData(c.full.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppo           #FULL DATA                           #SUBSET DATA
  table(h.full.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(h.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(h.full.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(h.full.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
  table(c.full.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(c.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(c.full.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(c.full.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
#(7)subset data to only include human and chimp cells/genes - sppo
#notes: 16542 genes overlap between humans + chimps (otherwise 33538 human genes + 25312 chimp genes = 42308 genes total)
# (GHO1)
  h=which(rownames(h.full.GHO1@assays$RNA@counts) %in% rownames(c.full.GHO1@assays$RNA@counts))
  c=which(rownames(c.full.GHO1@assays$RNA@counts) %in% rownames(h.full.GHO1@assays$RNA@counts))
  m <- match(rownames(h.full.GHO1@assays$RNA@counts)[h],rownames(c.full.GHO1@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppo.GHO1 <- subset(h.full.GHO1, features=h, subset=species=="Human")
  c.full.sppo.GHO1 <- subset(c.full.GHO1, features=c, subset=species=="Chimp")
# (GHO2)  
  h=which(rownames(h.full.GHO2@assays$RNA@counts) %in% rownames(c.full.GHO2@assays$RNA@counts))
  c=which(rownames(c.full.GHO2@assays$RNA@counts) %in% rownames(h.full.GHO2@assays$RNA@counts))
  m <- match(rownames(h.full.GHO2@assays$RNA@counts)[h],rownames(c.full.GHO2@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppo.GHO2 <- subset(h.full.GHO2, features=h, subset=species=="Human")
  c.full.sppo.GHO2 <- subset(c.full.GHO2, features=c, subset=species=="Chimp")
# (GHO3)  
  h=which(rownames(h.full.GHO3@assays$RNA@counts) %in% rownames(c.full.GHO3@assays$RNA@counts))
  c=which(rownames(c.full.GHO3@assays$RNA@counts) %in% rownames(h.full.GHO3@assays$RNA@counts))
  m <- match(rownames(h.full.GHO3@assays$RNA@counts)[h],rownames(c.full.GHO3@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppo.GHO3 <- subset(h.full.GHO3, features=h, subset=species=="Human")
  c.full.sppo.GHO3 <- subset(c.full.GHO3, features=c, subset=species=="Chimp")
# (GHO4)  
  h=which(rownames(h.full.GHO4@assays$RNA@counts) %in% rownames(c.full.GHO4@assays$RNA@counts))
  c=which(rownames(c.full.GHO4@assays$RNA@counts) %in% rownames(h.full.GHO4@assays$RNA@counts))
  m <- match(rownames(h.full.GHO4@assays$RNA@counts)[h],rownames(c.full.GHO4@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppo.GHO4 <- subset(h.full.GHO4, features=h, subset=species=="Human")
  c.full.sppo.GHO4 <- subset(c.full.GHO4, features=c, subset=species=="Chimp")
#(8)create merged object - sppo
  h.full.c.full.sppo <- merge(h.full.sppo.GHO1,c(c.full.sppo.GHO1,
                          h.full.sppo.GHO2,c.full.sppo.GHO2,
                          h.full.sppo.GHO3,c.full.sppo.GHO3,
                          h.full.sppo.GHO4,c.full.sppo.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.full.c.full.sppo")
  saveRDS(h.full.c.full.sppo, file = "../data/cellranger-data-full/h.full.c.full.sppo.rds")
#steps for h.full.c.full.sppf
#(5)add species assignments to data - sppf
  m <- match(rownames(h.full.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO1 <- AddMetaData(h.full.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO2 <- AddMetaData(h.full.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO3 <- AddMetaData(h.full.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO4 <- AddMetaData(h.full.GHO4, hc.full.spp4[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO1 <- AddMetaData(c.full.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO2 <- AddMetaData(c.full.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO3 <- AddMetaData(c.full.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO4 <- AddMetaData(c.full.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppf           #FULL DATA                           #SUBSET DATA
  table(h.full.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(h.full.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(h.full.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(h.full.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
  table(c.full.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(c.full.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(c.full.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(c.full.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
#(7)subset data to only include human and chimp cells/genes - sppf
# (GHO1)
  h=which(rownames(h.full.GHO1@assays$RNA@counts) %in% rownames(c.full.GHO1@assays$RNA@counts))
  c=which(rownames(c.full.GHO1@assays$RNA@counts) %in% rownames(h.full.GHO1@assays$RNA@counts))
  m <- match(rownames(h.full.GHO1@assays$RNA@counts)[h],rownames(c.full.GHO1@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppf.GHO1 <- subset(h.full.GHO1, features=h, subset=species=="Human")
  c.full.sppf.GHO1 <- subset(c.full.GHO1, features=c, subset=species=="Chimp")
# (GHO2)  
  h=which(rownames(h.full.GHO2@assays$RNA@counts) %in% rownames(c.full.GHO2@assays$RNA@counts))
  c=which(rownames(c.full.GHO2@assays$RNA@counts) %in% rownames(h.full.GHO2@assays$RNA@counts))
  m <- match(rownames(h.full.GHO2@assays$RNA@counts)[h],rownames(c.full.GHO2@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppf.GHO2 <- subset(h.full.GHO2, features=h, subset=species=="Human")
  c.full.sppf.GHO2 <- subset(c.full.GHO2, features=c, subset=species=="Chimp")
# (GHO3)  
  h=which(rownames(h.full.GHO3@assays$RNA@counts) %in% rownames(c.full.GHO3@assays$RNA@counts))
  c=which(rownames(c.full.GHO3@assays$RNA@counts) %in% rownames(h.full.GHO3@assays$RNA@counts))
  m <- match(rownames(h.full.GHO3@assays$RNA@counts)[h],rownames(c.full.GHO3@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppf.GHO3 <- subset(h.full.GHO3, features=h, subset=species=="Human")
  c.full.sppf.GHO3 <- subset(c.full.GHO3, features=c, subset=species=="Chimp")
# (GHO4)  
  h=which(rownames(h.full.GHO4@assays$RNA@counts) %in% rownames(c.full.GHO4@assays$RNA@counts))
  c=which(rownames(c.full.GHO4@assays$RNA@counts) %in% rownames(h.full.GHO4@assays$RNA@counts))
  m <- match(rownames(h.full.GHO4@assays$RNA@counts)[h],rownames(c.full.GHO4@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.sppf.GHO4 <- subset(h.full.GHO4, features=h, subset=species=="Human")
  c.full.sppf.GHO4 <- subset(c.full.GHO4, features=c, subset=species=="Chimp")
#(8)create merged object - sppf
  h.full.c.full.sppf <- merge(h.full.sppf.GHO1,c(c.full.sppf.GHO1,
                          h.full.sppf.GHO2,c.full.sppf.GHO2,
                          h.full.sppf.GHO3,c.full.sppf.GHO3,
                          h.full.sppf.GHO4,c.full.sppf.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.full.c.full.sppf")
  saveRDS(h.full.c.full.sppf, file = "../data/cellranger-data-full/h.full.c.full.sppf.rds")
#steps for h.full.c.full.spof
#(5)add species assignments to data - spof
  m <- match(rownames(h.full.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO1 <- AddMetaData(h.full.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO2 <- AddMetaData(h.full.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO3 <- AddMetaData(h.full.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.full.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.full.GHO4 <- AddMetaData(h.full.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO1 <- AddMetaData(c.full.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO2 <- AddMetaData(c.full.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO3 <- AddMetaData(c.full.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.full.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.full.GHO4 <- AddMetaData(c.full.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - spof           #FULL DATA                           #SUBSET DATA
  table(h.full.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(h.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(h.full.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(h.full.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
  table(c.full.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(c.full.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(c.full.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(c.full.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
#(7)subset data to only include human and chimp cells/genes - spof
# (GHO1)
  h=which(rownames(h.full.GHO1@assays$RNA@counts) %in% rownames(c.full.GHO1@assays$RNA@counts))
  c=which(rownames(c.full.GHO1@assays$RNA@counts) %in% rownames(h.full.GHO1@assays$RNA@counts))
  m <- match(rownames(h.full.GHO1@assays$RNA@counts)[h],rownames(c.full.GHO1@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.spof.GHO1 <- subset(h.full.GHO1, features=h, subset=species=="Human")
  c.full.spof.GHO1 <- subset(c.full.GHO1, features=c, subset=species=="Chimp")
# (GHO2)  
  h=which(rownames(h.full.GHO2@assays$RNA@counts) %in% rownames(c.full.GHO2@assays$RNA@counts))
  c=which(rownames(c.full.GHO2@assays$RNA@counts) %in% rownames(h.full.GHO2@assays$RNA@counts))
  m <- match(rownames(h.full.GHO2@assays$RNA@counts)[h],rownames(c.full.GHO2@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.spof.GHO2 <- subset(h.full.GHO2, features=h, subset=species=="Human")
  c.full.spof.GHO2 <- subset(c.full.GHO2, features=c, subset=species=="Chimp")
# (GHO3)  
  h=which(rownames(h.full.GHO3@assays$RNA@counts) %in% rownames(c.full.GHO3@assays$RNA@counts))
  c=which(rownames(c.full.GHO3@assays$RNA@counts) %in% rownames(h.full.GHO3@assays$RNA@counts))
  m <- match(rownames(h.full.GHO3@assays$RNA@counts)[h],rownames(c.full.GHO3@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.spof.GHO3 <- subset(h.full.GHO3, features=h, subset=species=="Human")
  c.full.spof.GHO3 <- subset(c.full.GHO3, features=c, subset=species=="Chimp")
# (GHO4)  
  h=which(rownames(h.full.GHO4@assays$RNA@counts) %in% rownames(c.full.GHO4@assays$RNA@counts))
  c=which(rownames(c.full.GHO4@assays$RNA@counts) %in% rownames(h.full.GHO4@assays$RNA@counts))
  m <- match(rownames(h.full.GHO4@assays$RNA@counts)[h],rownames(c.full.GHO4@assays$RNA@counts)[c])
  if(any(is.na(m))) cat("Not all genes match.\n")
  h.full.spof.GHO4 <- subset(h.full.GHO4, features=h, subset=species=="Human")
  c.full.spof.GHO4 <- subset(c.full.GHO4, features=c, subset=species=="Chimp")
#(8)create merged object - spof
  h.full.c.full.spof <- merge(h.full.spof.GHO1,c(c.full.spof.GHO1,
                          h.full.spof.GHO2,c.full.spof.GHO2,
                          h.full.spof.GHO3,c.full.spof.GHO3,
                          h.full.spof.GHO4,c.full.spof.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.full.c.full.spof")
  saveRDS(h.full.c.full.spof, file = "../data/cellranger-data-full/h.full.c.full.spof.rds")
#(9)clear excess data from workspace
  rm(h.full.GHO1,h.full.GHO2,h.full.GHO3,h.full.GHO4) 
  rm(c.full.GHO1,c.full.GHO2,c.full.GHO3,c.full.GHO4)
  rm(h.full.sppo.GHO1,h.full.sppo.GHO2,h.full.sppo.GHO3,h.full.sppo.GHO4)
  rm(c.full.sppo.GHO1,c.full.sppo.GHO2,c.full.sppo.GHO3,c.full.sppo.GHO4)
  rm(h.full.sppf.GHO1,h.full.sppf.GHO2,h.full.sppf.GHO3,h.full.sppf.GHO4)
  rm(c.full.sppf.GHO1,c.full.sppf.GHO2,c.full.sppf.GHO3,c.full.sppf.GHO4)
  rm(h.full.spof.GHO1,h.full.spof.GHO2,h.full.spof.GHO3,h.full.spof.GHO4)
  rm(c.full.spof.GHO1,c.full.spof.GHO2,c.full.spof.GHO3,c.full.spof.GHO4)
  

#Prepare data for h.orfl.c.orfl.sppo & h.orfl.c.orfl.sppf & h.orfl.c.orfl.spof
#(1)load data
  h.orfl.GHO1_data <- Read10X(data=paste0(dir_h.orfl.GHO1,"/raw_feature_bc_matrix"))
  h.orfl.GHO2_data <- Read10X(data=paste0(dir_h.orfl.GHO2,"/raw_feature_bc_matrix"))
  h.orfl.GHO3_data <- Read10X(data=paste0(dir_h.orfl.GHO3,"/raw_feature_bc_matrix"))
  h.orfl.GHO4_data <- Read10X(data=paste0(dir_h.orfl.GHO4,"/raw_feature_bc_matrix"))
  c.orfl.GHO1_data <- Read10X(data=paste0(dir_c.orfl.GHO1,"/raw_feature_bc_matrix"))
  c.orfl.GHO2_data <- Read10X(data=paste0(dir_c.orfl.GHO2,"/raw_feature_bc_matrix"))
  c.orfl.GHO3_data <- Read10X(data=paste0(dir_c.orfl.GHO3,"/raw_feature_bc_matrix"))
  c.orfl.GHO4_data <- Read10X(data=paste0(dir_c.orfl.GHO4,"/raw_feature_bc_matrix"))
#(2)create seurat objects
  h.orfl.GHO1 <- CreateSeuratObject(counts=h.orfl.GHO1_data, project="iPSC")
  h.orfl.GHO2 <- CreateSeuratObject(counts=h.orfl.GHO2_data, project="MSC")
  h.orfl.GHO3 <- CreateSeuratObject(counts=h.orfl.GHO3_data, project="chondrocyte")
  h.orfl.GHO4 <- CreateSeuratObject(counts=h.orfl.GHO4_data, project="osteoblast")
  c.orfl.GHO1 <- CreateSeuratObject(counts=c.orfl.GHO1_data, project="iPSC")
  c.orfl.GHO2 <- CreateSeuratObject(counts=c.orfl.GHO2_data, project="MSC")
  c.orfl.GHO3 <- CreateSeuratObject(counts=c.orfl.GHO3_data, project="chondrocyte")
  c.orfl.GHO4 <- CreateSeuratObject(counts=c.orfl.GHO4_data, project="osteoblast")
#(3)view details of seurat objects  #FULL DATA                        #SUBSET DATA
  h.orfl.GHO1                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  h.orfl.GHO2                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  h.orfl.GHO3                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  h.orfl.GHO4                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  c.orfl.GHO1                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  c.orfl.GHO2                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  c.orfl.GHO3                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
  c.orfl.GHO4                       #28942 features, 6794880 samples  #44125 features, 6794880 samples
#(4)clear raw data from workspace
  rm(h.orfl.GHO1_data,h.orfl.GHO2_data,h.orfl.GHO3_data,h.orfl.GHO4_data)
  rm(c.orfl.GHO1_data,c.orfl.GHO2_data,c.orfl.GHO3_data,c.orfl.GHO4_data)
#steps for h.orfl.c.orfl.sppo
#(5)add species assignments to data - sppo
  m <- match(rownames(h.orfl.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO1 <- AddMetaData(h.orfl.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO2 <- AddMetaData(h.orfl.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO3 <- AddMetaData(h.orfl.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO4 <- AddMetaData(h.orfl.GHO4, hc.orth.spp4[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO1@meta.data),hc.orth.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO1 <- AddMetaData(c.orfl.GHO1, hc.orth.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO2@meta.data),hc.orth.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO2 <- AddMetaData(c.orfl.GHO2, hc.orth.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO3@meta.data),hc.orth.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO3 <- AddMetaData(c.orfl.GHO3, hc.orth.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO4@meta.data),hc.orth.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO4 <- AddMetaData(c.orfl.GHO4, hc.orth.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppo           #FULL DATA                           #SUBSET DATA
  table(h.orfl.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(h.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(h.orfl.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(h.orfl.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
  table(c.orfl.GHO1@meta.data$species, useNA="always")   #C=3294, H=2319, M=  87, NA=6789180  #C=3253, H=2260, M=  77, NA=6789290
  table(c.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2563, M= 123, NA=6789045  #C=3025, H=2484, M= 113, NA=6789258
  table(c.orfl.GHO3@meta.data$species, useNA="always")   #C=1305, H=2168, M= 114, NA=6791293  #C=1074, H=2123, M=  56, NA=6791627
  table(c.orfl.GHO4@meta.data$species, useNA="always")   #C=1805, H=1580, M=  62, NA=6791433  #C=1697, H=1400, M=  28, NA=6791755
#(7)subset data to only include human and chimp cells/genes - sppo
# (GHO1)
  h.orfl.sppo.GHO1 <- subset(h.orfl.GHO1, subset=species=="Human")
  c.orfl.sppo.GHO1 <- subset(c.orfl.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orfl.sppo.GHO2 <- subset(h.orfl.GHO2, subset=species=="Human")
  c.orfl.sppo.GHO2 <- subset(c.orfl.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orfl.sppo.GHO3 <- subset(h.orfl.GHO3, subset=species=="Human")
  c.orfl.sppo.GHO3 <- subset(c.orfl.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orfl.sppo.GHO4 <- subset(h.orfl.GHO4, subset=species=="Human")
  c.orfl.sppo.GHO4 <- subset(c.orfl.GHO4, subset=species=="Chimp")
#(8)create merged object - sppo
  h.orfl.c.orfl.sppo <- merge(h.orfl.sppo.GHO1,c(c.orfl.sppo.GHO1,
                          h.orfl.sppo.GHO2,c.orfl.sppo.GHO2,
                          h.orfl.sppo.GHO3,c.orfl.sppo.GHO3,
                          h.orfl.sppo.GHO4,c.orfl.sppo.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orfl.c.orfl.sppo")
  saveRDS(h.orfl.c.orfl.sppo, file = "../data/cellranger-data-full/h.orfl.c.orfl.sppo.rds")
#steps for h.orfl.c.orfl.sppf
#(5)add species assignments to data - sppf
  m <- match(rownames(h.orfl.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO1 <- AddMetaData(h.orfl.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO2 <- AddMetaData(h.orfl.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO3 <- AddMetaData(h.orfl.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO4 <- AddMetaData(h.orfl.GHO4, hc.full.spp4[m,]$call, col.name="species")
    m <- match(rownames(c.orfl.GHO1@meta.data),hc.full.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO1 <- AddMetaData(c.orfl.GHO1, hc.full.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO2@meta.data),hc.full.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO2 <- AddMetaData(c.orfl.GHO2, hc.full.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO3@meta.data),hc.full.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO3 <- AddMetaData(c.orfl.GHO3, hc.full.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO4@meta.data),hc.full.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO4 <- AddMetaData(c.orfl.GHO4, hc.full.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - sppf           #FULL DATA                           #SUBSET DATA
  table(h.orfl.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(h.orfl.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(h.orfl.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(h.orfl.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
  table(c.orfl.GHO1@meta.data$species, useNA="always")   #C=1837, H=2949, M=1976, NA=6788118  #C=3281, H=2400, M= 118, NA=6789081
  table(c.orfl.GHO2@meta.data$species, useNA="always")   #C=3241, H=2629, M= 182, NA=6788828  #C=3104, H=2543, M= 124, NA=6789109
  table(c.orfl.GHO3@meta.data$species, useNA="always")   #C= 931, H=4347, M=2750, NA=6786852  #C= 792, H=4109, M= 931, NA=6789048
  table(c.orfl.GHO4@meta.data$species, useNA="always")   #C= 807, H=1952, M=2097, NA=6788118  #C= 823, H=1824, M=1266, NA=6790967
#(7)subset data to only include human and chimp cells/genes - sppf
# (GHO1)
  h.orfl.sppf.GHO1 <- subset(h.orfl.GHO1, subset=species=="Human")
  c.orfl.sppf.GHO1 <- subset(c.orfl.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orfl.sppf.GHO2 <- subset(h.orfl.GHO2, subset=species=="Human")
  c.orfl.sppf.GHO2 <- subset(c.orfl.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orfl.sppf.GHO3 <- subset(h.orfl.GHO3, subset=species=="Human")
  c.orfl.sppf.GHO3 <- subset(c.orfl.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orfl.sppf.GHO4 <- subset(h.orfl.GHO4, subset=species=="Human")
  c.orfl.sppf.GHO4 <- subset(c.orfl.GHO4, subset=species=="Chimp")
#(8)create merged object - sppf
  h.orfl.c.orfl.sppf <- merge(h.orfl.sppf.GHO1,c(c.orfl.sppf.GHO1,
                          h.orfl.sppf.GHO2,c.orfl.sppf.GHO2,
                          h.orfl.sppf.GHO3,c.orfl.sppf.GHO3,
                          h.orfl.sppf.GHO4,c.orfl.sppf.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orfl.c.orfl.sppf")
  saveRDS(h.orfl.c.orfl.sppf, file = "../data/cellranger-data-full/h.orfl.c.orfl.sppf.rds")
#steps for h.orfl.c.orfl.spof
#(5)add species assignments to data - spof
  m <- match(rownames(h.orfl.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO1 <- AddMetaData(h.orfl.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO2 <- AddMetaData(h.orfl.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO3 <- AddMetaData(h.orfl.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(h.orfl.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  h.orfl.GHO4 <- AddMetaData(h.orfl.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO1@meta.data),hc.orfl.spp1$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO1 <- AddMetaData(c.orfl.GHO1, hc.orfl.spp1[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO2@meta.data),hc.orfl.spp2$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO2 <- AddMetaData(c.orfl.GHO2, hc.orfl.spp2[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO3@meta.data),hc.orfl.spp3$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO3 <- AddMetaData(c.orfl.GHO3, hc.orfl.spp3[m,]$call, col.name="species")
  m <- match(rownames(c.orfl.GHO4@meta.data),hc.orfl.spp4$barcode)
  if(any(is.na(m))) cat("Not all barcodes are in data.\n")
  c.orfl.GHO4 <- AddMetaData(c.orfl.GHO4, hc.orfl.spp4[m,]$call, col.name="species")
#(6)view details of species assignments - spof           #FULL DATA                           #SUBSET DATA
  table(h.orfl.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(h.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(h.orfl.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(h.orfl.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
  table(c.orfl.GHO1@meta.data$species, useNA="always")   #C=3296, H=2318, M=  82, NA=6789184  ---
  table(c.orfl.GHO2@meta.data$species, useNA="always")   #C=3149, H=2557, M= 122, NA=6789052  ---
  table(c.orfl.GHO3@meta.data$species, useNA="always")   #C=1290, H=2166, M= 105, NA=6791319  ---
  table(c.orfl.GHO4@meta.data$species, useNA="always")   #C=1828, H=1586, M=  42, NA=6791424  ---
#(7)subset data to only include human and chimp cells/genes - spof
# (GHO1)
  h.orfl.spof.GHO1 <- subset(h.orfl.GHO1, subset=species=="Human")
  c.orfl.spof.GHO1 <- subset(c.orfl.GHO1, subset=species=="Chimp")
# (GHO2)  
  h.orfl.spof.GHO2 <- subset(h.orfl.GHO2, subset=species=="Human")
  c.orfl.spof.GHO2 <- subset(c.orfl.GHO2, subset=species=="Chimp")
# (GHO3)  
  h.orfl.spof.GHO3 <- subset(h.orfl.GHO3, subset=species=="Human")
  c.orfl.spof.GHO3 <- subset(c.orfl.GHO3, subset=species=="Chimp")
# (GHO4)  
  h.orfl.spof.GHO4 <- subset(h.orfl.GHO4, subset=species=="Human")
  c.orfl.spof.GHO4 <- subset(c.orfl.GHO4, subset=species=="Chimp")
#(8)create merged object - spof
  h.orfl.c.orfl.spof <- merge(h.orfl.spof.GHO1,c(c.orfl.spof.GHO1,
                          h.orfl.spof.GHO2,c.orfl.spof.GHO2,
                          h.orfl.spof.GHO3,c.orfl.spof.GHO3,
                          h.orfl.spof.GHO4,c.orfl.spof.GHO4),
                        add.cell.ids=c("iPSC","iPSC","MSC","MSC","chondro","chondro","osteo","osteo"),
                        project="h.orfl.c.orfl.spof")
  saveRDS(h.orfl.c.orfl.spof, file = "../data/cellranger-data-full/h.orfl.c.orfl.spof.rds")
#(9)clear excess data from workspace
  rm(h.orfl.GHO1,h.orfl.GHO2,h.orfl.GHO3,h.orfl.GHO4) 
  rm(c.orfl.GHO1,c.orfl.GHO2,c.orfl.GHO3,c.orfl.GHO4)
  rm(h.orfl.sppo.GHO1,h.orfl.sppo.GHO2,h.orfl.sppo.GHO3,h.orfl.sppo.GHO4)
  rm(c.orfl.sppo.GHO1,c.orfl.sppo.GHO2,c.orfl.sppo.GHO3,c.orfl.sppo.GHO4)
  rm(h.orfl.sppf.GHO1,h.orfl.sppf.GHO2,h.orfl.sppf.GHO3,h.orfl.sppf.GHO4)
  rm(c.orfl.sppf.GHO1,c.orfl.sppf.GHO2,c.orfl.sppf.GHO3,c.orfl.sppf.GHO4)
  rm(h.orfl.spof.GHO1,h.orfl.spof.GHO2,h.orfl.spof.GHO3,h.orfl.spof.GHO4)
  rm(c.orfl.spof.GHO1,c.orfl.spof.GHO2,c.orfl.spof.GHO3,c.orfl.spof.GHO4)


#Clear additional excess data from workspace
rm(dir_h.full.GHO1,dir_h.full.GHO2,dir_h.full.GHO3,dir_h.full.GHO4)
rm(dir_c.full.GHO1,dir_c.full.GHO2,dir_c.full.GHO3,dir_c.full.GHO4)
rm(dir_hc.full.GHO1,dir_hc.full.GHO2,dir_hc.full.GHO3,dir_hc.full.GHO4)
rm(dir_h.orth.GHO1,dir_h.orth.GHO2,dir_h.orth.GHO3,dir_h.orth.GHO4)
rm(dir_c.orth.GHO1,dir_c.orth.GHO2,dir_c.orth.GHO3,dir_c.orth.GHO4)
rm(dir_hc.orth.GHO1,dir_hc.orth.GHO2,dir_hc.orth.GHO3,dir_hc.orth.GHO4)
rm(dir_h.orfl.GHO1,dir_h.orfl.GHO2,dir_h.orfl.GHO3,dir_h.orfl.GHO4)
rm(dir_c.orfl.GHO1,dir_c.orfl.GHO2,dir_c.orfl.GHO3,dir_c.orfl.GHO4)
rm(dir_hc.orfl.GHO1,dir_hc.orfl.GHO2,dir_hc.orfl.GHO3,dir_hc.orfl.GHO4)
rm(hc.orth.spp1,hc.orth.spp2,hc.orth.spp3,hc.orth.spp4)
rm(hc.full.spp1,hc.full.spp2,hc.full.spp3,hc.full.spp4)
rm(hc.orfl.spp1,hc.orfl.spp2,hc.orfl.spp3,hc.orfl.spp4)
rm(m,i,h,c,h.gene,c.gene)


#View details of final datasets to be compared  #FULL DATA                      #SUBSET DATA
hc.orfl.spof.tot                                #57884 features, 18541 samples  ---
hc.orth.sppo.tot                                #88250 features, 18569 samples  #88250 features, 17590 samples
hc.full.sppf.tot                                #58850 features, 25698 samples  #58850 features, 21315 samples
hc.orfl.spof                                    #28942 features, 18190 samples  ---
hc.orfl.sppo                                    #28942 features, 18183 samples  ---
hc.orfl.sppf                                    #28942 features, 18693 samples  ---
hc.orth.spof                                    #44125 features, 18190 samples  ---
hc.orth.sppo                                    #44125 features, 18183 samples  #44125 features, 17316 samples
hc.orth.sppf                                    #44125 features, 18693 samples  #44125 features, 18876 samples
hc.full.spof                                    #16542 features, 18190 samples  ---
hc.full.sppo                                    #16542 features, 18183 samples  #16542 features, 17316 samples
hc.full.sppf                                    #16542 features, 18693 samples  #16542 features, 18876 samples
h.orfl.c.orfl.spof                              #28942 features, 18190 samples  ---
h.orfl.c.orfl.sppo                              #28942 features, 18183 samples  ---
h.orfl.c.orfl.sppf                              #28942 features, 18693 samples  ---
h.orth.c.orth.spof                              #44125 features, 18190 samples  ---
h.orth.c.orth.sppo                              #44125 features, 18183 samples  #44125 features, 17316 samples
h.orth.c.orth.sppf                              #44125 features, 18693 samples  #44125 features, 18876 samples
h.full.c.full.spof                              #16542 features, 18190 samples  ---
h.full.c.full.sppo                              #16542 features, 18183 samples  #16542 features, 17316 samples
h.full.c.full.sppf                              #16542 features, 18693 samples  #16542 features, 18876 samples

#notes: details on datasets not filtered using sppo or sppf                     #SUBSET DATA
#h.full.GHO                                                                     #33538 features, 23234 samples
#c.full.GHO                                                                     #25312 features, 21795 samples
#hc.full.GHO                                                                    #58850 features, 21315 samples
#h.orth.GHO                                                                     #44125 features, 18360 samples
#c.orth.GHO                                                                     #44125 features, 19029 samples
#hc.orth.GHO                                                                    #88250 features, 17590 samples

```

```{r data meta information adjustments}

# Load data

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
hc.orfl.spof.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orfl.spof.tot.rds"))
hc.orth.sppo.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orth.sppo.tot.rds"))
hc.full.sppf.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.full.sppf.tot.rds"))
hc.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orfl.spof.rds"))
hc.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orfl.sppo.rds"))
hc.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orfl.sppf.rds"))
hc.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orth.spof.rds"))
hc.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orth.sppo.rds"))
hc.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.orth.sppf.rds"))
hc.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.full.spof.rds"))
hc.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.full.sppo.rds"))
hc.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/hc.full.sppf.rds"))
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orfl.c.orfl.spof.rds"))
h.orfl.c.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orfl.c.orfl.sppo.rds"))
h.orfl.c.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orfl.c.orfl.sppf.rds"))
h.orth.c.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orth.c.orth.spof.rds"))
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orth.c.orth.sppo.rds"))
h.orth.c.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/h.orth.c.orth.sppf.rds"))
h.full.c.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/h.full.c.full.spof.rds"))
h.full.c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/h.full.c.full.sppo.rds"))
h.full.c.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/h.full.c.full.sppf.rds"))


#Fix meta.data details so that plotting is easier later on

hc.orfl.spof.tot@meta.data$species <- factor(hc.orfl.spof.tot@meta.data$species, levels=c("Human","Chimp","Multiplet"))
hc.orth.sppo.tot@meta.data$species <- factor(hc.orth.sppo.tot@meta.data$species, levels=c("Human","Chimp","Multiplet"))
hc.full.sppf.tot@meta.data$species <- factor(hc.full.sppf.tot@meta.data$species, levels=c("Human","Chimp","Multiplet"))
hc.orfl.spof@meta.data$species <- factor(hc.orfl.spof@meta.data$species, levels=c("Human","Chimp"))
hc.orfl.sppo@meta.data$species <- factor(hc.orfl.sppo@meta.data$species, levels=c("Human","Chimp"))
hc.orfl.sppf@meta.data$species <- factor(hc.orfl.sppf@meta.data$species, levels=c("Human","Chimp"))
hc.orth.spof@meta.data$species <- factor(hc.orth.spof@meta.data$species, levels=c("Human","Chimp"))
hc.orth.sppo@meta.data$species <- factor(hc.orth.sppo@meta.data$species, levels=c("Human","Chimp"))
hc.orth.sppf@meta.data$species <- factor(hc.orth.sppf@meta.data$species, levels=c("Human","Chimp"))
hc.full.spof@meta.data$species <- factor(hc.full.spof@meta.data$species, levels=c("Human","Chimp"))
hc.full.sppo@meta.data$species <- factor(hc.full.sppo@meta.data$species, levels=c("Human","Chimp"))
hc.full.sppf@meta.data$species <- factor(hc.full.sppf@meta.data$species, levels=c("Human","Chimp"))
h.orfl.c.orfl.spof@meta.data$species <- factor(h.orfl.c.orfl.spof@meta.data$species, levels=c("Human","Chimp"))
h.orfl.c.orfl.sppo@meta.data$species <- factor(h.orfl.c.orfl.sppo@meta.data$species, levels=c("Human","Chimp"))
h.orfl.c.orfl.sppf@meta.data$species <- factor(h.orfl.c.orfl.sppf@meta.data$species, levels=c("Human","Chimp"))
h.orth.c.orth.spof@meta.data$species <- factor(h.orth.c.orth.spof@meta.data$species, levels=c("Human","Chimp"))
h.orth.c.orth.sppo@meta.data$species <- factor(h.orth.c.orth.sppo@meta.data$species, levels=c("Human","Chimp"))
h.orth.c.orth.sppf@meta.data$species <- factor(h.orth.c.orth.sppf@meta.data$species, levels=c("Human","Chimp"))
h.full.c.full.spof@meta.data$species <- factor(h.full.c.full.spof@meta.data$species, levels=c("Human","Chimp"))
h.full.c.full.sppo@meta.data$species <- factor(h.full.c.full.sppo@meta.data$species, levels=c("Human","Chimp"))
h.full.c.full.sppf@meta.data$species <- factor(h.full.c.full.sppf@meta.data$species, levels=c("Human","Chimp"))

hc.orfl.spof.tot@meta.data$orig.ident <- factor(hc.orfl.spof.tot@meta.data$orig.ident ,levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orth.sppo.tot@meta.data$orig.ident <- factor(hc.orth.sppo.tot@meta.data$orig.ident ,levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.full.sppf.tot@meta.data$orig.ident <- factor(hc.full.sppf.tot@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orfl.spof@meta.data$orig.ident <- factor(hc.orfl.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orfl.sppo@meta.data$orig.ident <- factor(hc.orfl.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orfl.sppf@meta.data$orig.ident <- factor(hc.orfl.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orth.spof@meta.data$orig.ident <- factor(hc.orth.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orth.sppo@meta.data$orig.ident <- factor(hc.orth.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.orth.sppf@meta.data$orig.ident <- factor(hc.orth.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.full.spof@meta.data$orig.ident <- factor(hc.full.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.full.sppo@meta.data$orig.ident <- factor(hc.full.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
hc.full.sppf@meta.data$orig.ident <- factor(hc.full.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orfl.c.orfl.spof@meta.data$orig.ident <- factor(h.orfl.c.orfl.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orfl.c.orfl.sppo@meta.data$orig.ident <- factor(h.orfl.c.orfl.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orfl.c.orfl.sppf@meta.data$orig.ident <- factor(h.orfl.c.orfl.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orth.c.orth.spof@meta.data$orig.ident <- factor(h.orth.c.orth.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orth.c.orth.sppo@meta.data$orig.ident <- factor(h.orth.c.orth.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.orth.c.orth.sppf@meta.data$orig.ident <- factor(h.orth.c.orth.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.full.c.full.spof@meta.data$orig.ident <- factor(h.full.c.full.spof@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.full.c.full.sppo@meta.data$orig.ident <- factor(h.full.c.full.sppo@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))
h.full.c.full.sppf@meta.data$orig.ident <- factor(h.full.c.full.sppf@meta.data$orig.ident, levels=c("iPSC","MSC","chondrocyte","osteoblast"))

#notes: colors for coordination with figures
#col4=c("pink","mediumorchid","gold","dodgerblue") #colors for cell types
#scale_fill_manual(values=col4)

```

```{r data qc prefiltering}

#Check which features are mitochondral genes

#notes: 0 mito genes in ortho-filter annotations
table(sapply(hc.orfl.spof.tot@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="-MT-"))
table(sapply(hc.orfl.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(hc.orfl.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(hc.orfl.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orfl.c.orfl.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orfl.c.orfl.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orfl.c.orfl.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
#notes: 0 mito genes in ortho annotations
table(sapply(hc.orth.sppo.tot@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="-MT-"))
table(sapply(hc.orth.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(hc.orth.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(hc.orth.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orth.c.orth.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orth.c.orth.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
table(sapply(h.orth.c.orth.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))

#notes: 13 mito genes in human full annotation + 1 mito gene in chimp full annotation
table(sapply(hc.full.sppf.tot@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="-MT-"))
which(sapply(hc.full.sppf.tot@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="-MT-")==1)
#GRCh38------MT-ND1   33497
#GRCh38------MT-ND2   33498
#GRCh38------MT-CO1   33499
#GRCh38------MT-CO2   33500
#GRCh38------MT-ATP8  33501
#GRCh38------MT-ATP6  33502
#GRCh38------MT-CO3   33503
#GRCh38------MT-ND3   33504
#GRCh38------MT-ND4L  33505
#GRCh38------MT-ND4   33506
#GRCh38------MT-ND5   33507
#GRCh38------MT-ND6   33508
#GRCh38------MT-CYB   33509
#Pan-tro-3.0-MT-ND2   57145

#notes: 1 mito gene that overlaps between human full annotation + chimp full annotation
table(sapply(hc.full.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(hc.full.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542
table(sapply(hc.full.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(hc.full.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542
table(sapply(hc.full.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(hc.full.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542
table(sapply(h.full.c.full.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(h.full.c.full.spof@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542 
table(sapply(h.full.c.full.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(h.full.c.full.sppo@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542 
table(sapply(h.full.c.full.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-"))
which(sapply(h.full.c.full.sppf@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
#MT-ND2 16542 


#Calculate the percentage of reads that map to the mitochondrial genome

hc.orfl.spof.tot[["percent.mt"]] <- PercentageFeatureSet(hc.orfl.spof.tot, pattern="-MT-")     #no mito
hc.orth.sppo.tot[["percent.mt"]] <- PercentageFeatureSet(hc.orth.sppo.tot, pattern="-MT-")     #no mito
hc.full.sppf.tot[["percent.mt"]] <- PercentageFeatureSet(hc.full.sppf.tot, pattern="-MT-ND2")  #13+1 mito genes but only calculating on 1
hc.orfl.spof[["percent.mt"]] <- PercentageFeatureSet(hc.orfl.spof, pattern="^MT-")             #no mito
hc.orfl.sppo[["percent.mt"]] <- PercentageFeatureSet(hc.orfl.sppo, pattern="^MT-")             #no mito
hc.orfl.sppf[["percent.mt"]] <- PercentageFeatureSet(hc.orfl.sppf, pattern="^MT-")             #no mito
hc.orth.spof[["percent.mt"]] <- PercentageFeatureSet(hc.orth.spof, pattern="^MT-")             #no mito
hc.orth.sppo[["percent.mt"]] <- PercentageFeatureSet(hc.orth.sppo, pattern="^MT-")             #no mito
hc.orth.sppf[["percent.mt"]] <- PercentageFeatureSet(hc.orth.sppf, pattern="^MT-")             #no mito
hc.full.spof[["percent.mt"]] <- PercentageFeatureSet(hc.full.spof, pattern="^MT-")             #1 mito gene
hc.full.sppo[["percent.mt"]] <- PercentageFeatureSet(hc.full.sppo, pattern="^MT-")             #1 mito gene
hc.full.sppf[["percent.mt"]] <- PercentageFeatureSet(hc.full.sppf, pattern="^MT-")             #1 mito gene
h.orfl.c.orfl.spof[["percent.mt"]] <- PercentageFeatureSet(h.orfl.c.orfl.spof, pattern="^MT-") #no mito
h.orfl.c.orfl.sppo[["percent.mt"]] <- PercentageFeatureSet(h.orfl.c.orfl.sppo, pattern="^MT-") #no mito
h.orfl.c.orfl.sppf[["percent.mt"]] <- PercentageFeatureSet(h.orfl.c.orfl.sppf, pattern="^MT-") #no mito
h.orth.c.orth.spof[["percent.mt"]] <- PercentageFeatureSet(h.orth.c.orth.spof, pattern="^MT-") #no mito
h.orth.c.orth.sppo[["percent.mt"]] <- PercentageFeatureSet(h.orth.c.orth.sppo, pattern="^MT-") #no mito
h.orth.c.orth.sppf[["percent.mt"]] <- PercentageFeatureSet(h.orth.c.orth.sppf, pattern="^MT-") #no mito
h.full.c.full.spof[["percent.mt"]] <- PercentageFeatureSet(h.full.c.full.spof, pattern="^MT-") #1 mito gene
h.full.c.full.sppo[["percent.mt"]] <- PercentageFeatureSet(h.full.c.full.sppo, pattern="^MT-") #1 mito gene
h.full.c.full.sppf[["percent.mt"]] <- PercentageFeatureSet(h.full.c.full.sppf, pattern="^MT-") #1 mito gene

```

```{r data qc prefiltering}

#Check and visualize numbers of cells per species
                                                            #FULL DATA                          #SUBSET DATA
table(hc.orfl.spof.tot@meta.data$species, useNA="always")   #H= 8267, C=9563, Mult= 351, NA=0   ---
table(hc.orth.sppo.tot@meta.data$species, useNA="always")   #H= 8630, C=9553, Mult= 386, NA=0   #H= 8267, C=9049, Mult= 274, NA=0
table(hc.full.sppf.tot@meta.data$species, useNA="always")   #H=11877, C=6816, Mult=7005, NA=0   #H=10876, C=8000, Mult=2439, NA=0
table(hc.orfl.spof@meta.data$species, useNA="always")       #H= 8267, C=9563, Mult=  na, NA=0   ---
table(hc.orfl.sppo@meta.data$species, useNA="always")       #H= 8630, C=9553, Mult=  na, NA=0   ---
table(hc.orfl.sppf@meta.data$species, useNA="always")       #H=11877, C=6816, Mult=  na, NA=0   ---
table(hc.orth.spof@meta.data$species, useNA="always")       #H= 8267, C=9563, Mult=  na, NA=0   ---
table(hc.orth.sppo@meta.data$species, useNA="always")       #H= 8630, C=9553, Mult=  na, NA=0   #H= 8267, C=9049, Mult= 274, NA=0
table(hc.orth.sppf@meta.data$species, useNA="always")       #H=11877, C=6816, Mult=  na, NA=0   #H=10876, C=8000, Mult=2439, NA=0
table(hc.full.spof@meta.data$species, useNA="always")       #H= 8267, C=9563, Mult=  na, NA=0   ---
table(hc.full.sppo@meta.data$species, useNA="always")       #H= 8630, C=9553, Mult=  na, NA=0   #H= 8267, C=9049, Mult= 274, NA=0
table(hc.full.sppf@meta.data$species, useNA="always")       #H=11877, C=6816, Mult=  na, NA=0   #H=10876, C=8000, Mult=2439, NA=0
table(h.orfl.c.orfl.spof@meta.data$species, useNA="always") #H= 8267, C=9563, Mult=  na, NA=0   ---
table(h.orfl.c.orfl.sppo@meta.data$species, useNA="always") #H= 8630, C=9553, Mult=  na, NA=0   ---
table(h.orfl.c.orfl.sppf@meta.data$species, useNA="always") #H=11877, C=6816, Mult=  na, NA=0   ---
table(h.orth.c.orth.spof@meta.data$species, useNA="always") #H= 8267, C=9563, Mult=  na, NA=0   ---
table(h.orth.c.orth.sppo@meta.data$species, useNA="always") #H= 8630, C=9553, Mult=  na, NA=0   #H= 8267, C=9049, Mult= 274, NA=0
table(h.orth.c.orth.sppf@meta.data$species, useNA="always") #H=11877, C=6816, Mult=  na, NA=0   #H=10876, C=8000, Mult=2439, NA=0
table(h.full.c.full.spof@meta.data$species, useNA="always") #H= 8267, C=9563, Mult=  na, NA=0   ---
table(h.full.c.full.sppo@meta.data$species, useNA="always") #H= 8630, C=9553, Mult=  na, NA=0   #H= 8267, C=9049, Mult= 274, NA=0
table(h.full.c.full.sppf@meta.data$species, useNA="always") #H=11877, C=6816, Mult=  na, NA=0   #H=10876, C=8000, Mult=2439, NA=0

#notes: details on datasets not filtered using sppo or sppf                                     #SUBSET DATA
#h.full.GHO                                                                                     #H=10769, C=7745, Mult=2439, NA=2281
#c.full.GHO                                                                                     #H=10520, C=7604, Mult=2439, NA=1232
#hc.full.GHO                                                                                    #H=10876, C=8000, Mult=2439, NA=   0
#h.orth.GHO                                                                                     #H= 8253, C=8877, Mult= 274, NA= 956
#c.orth.GHO                                                                                     #H= 8253, C=8877, Mult= 274, NA=1627
#hc.orth.GHO                                                                                    #H= 8257, C=9049, Mult= 274, NA=   0

CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orfl.spof")+ylim(0,12000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orth.spof")+ylim(0,12000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species))+geom_bar()+labs(title="hc.full.spof")+ylim(0,12000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species))+geom_bar()+labs(title="h.orfl.c.orfl.spof")+ylim(0,12000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species))+geom_bar()+labs(title="h.orth.c.orth.spof")+ylim(0,12000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species))+geom_bar()+labs(title="h.full.c.full.spof")+ylim(0,12000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orfl.spof.tot")+ylim(0,21000)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orfl.sppo")+ylim(0,12000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orth.sppo")+ylim(0,12000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="hc.full.sppo")+ylim(0,12000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,12000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="h.orth.c.orth.sppo")+ylim(0,12000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species))+geom_bar()+labs(title="h.full.c.full.sppo")+ylim(0,12000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orth.sppo.tot")+ylim(0,12000)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orfl.sppf")+ylim(0,12000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="hc.orth.sppf")+ylim(0,12000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="hc.full.sppf")+ylim(0,12000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,12000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="h.orth.c.orth.sppf")+ylim(0,12000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species))+geom_bar()+labs(title="h.full.c.full.sppf")+ylim(0,12000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species))+geom_bar()+labs(title="hc.full.sppf.tot")+ylim(0,12000))), ncol=7)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.orfl.spof\nhc.orth.spof\nhc.full.spof\nh.orfl.c.orfl.spof\nh.orth.c.orth.spof\nh.full.c.full.spof")+ylim(0,5000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.orfl.spof.tot")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.orfl.sppo\nhc.orth.sppo\nhc.full.sppo\nh.orfl.c.orfl.sppo\nh.orth.c.orth.sppo\nh.full.c.full.sppo")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.orth.sppo.tot")+ylim(0,5000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.orfl.sppf\nhc.orth.sppf\nhc.full.sppf\nh.orfl.c.orfl.sppf\nh.orth.c.orth.sppf\nh.full.c.full.sppf")+ylim(0,5000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species, fill=orig.ident))+geom_bar(position=position_dodge())+
                           labs(title="hc.full.sppf.tot")+ylim(0,5000))), ncol=2)
#notes: slightly more human cells with sppf filter (but this is primarily the case for chondrocytes)
#notes: more even species mixture with spof and sppo filters - comparable cell numbers


#Show QC metrics for the first 5 cells

#nFeature_RNA: number of unique genes detected in each cell
#nCount_RNA: total number of molecules (umi) detected within a cell
#percent.mt: percentage of reads that map to the mitochondrial genome
head(hc.orfl.spof.tot@meta.data, 5)
head(hc.orth.sppo.tot@meta.data, 5)
head(hc.full.sppf.tot@meta.data, 5)
head(hc.orfl.spof@meta.data, 5)
head(hc.orfl.sppo@meta.data, 5)
head(hc.orfl.sppf@meta.data, 5)
head(hc.orth.spof@meta.data, 5)
head(hc.orth.sppo@meta.data, 5)
head(hc.orth.sppf@meta.data, 5)
head(hc.full.spof@meta.data, 5)
head(hc.full.sppo@meta.data, 5)
head(hc.full.sppf@meta.data, 5)
head(h.orfl.c.orfl.spof@meta.data, 5)
head(h.orfl.c.orfl.sppo@meta.data, 5)
head(h.orfl.c.orfl.sppf@meta.data, 5)
head(h.orth.c.orth.spof@meta.data, 5)
head(h.orth.c.orth.sppo@meta.data, 5)
head(h.orth.c.orth.sppf@meta.data, 5)
head(h.full.c.full.spof@meta.data, 5)
head(h.full.c.full.sppo@meta.data, 5)
head(h.full.c.full.sppf@meta.data, 5)


#Check and visualize QC metrics

#notes: examined nCount_RNA, nFeature_RNA, percent.mt
mean(hc.orfl.spof.tot@meta.data$percent.mt)
mean(hc.orfl.spof.tot@meta.data$percent.mt[which(hc.orfl.spof.tot[['species']]=="Human")])
mean(hc.orfl.spof.tot@meta.data$percent.mt[which(hc.orfl.spof.tot[['species']]=="Chimp")])
mean(hc.orfl.spof.tot@meta.data$percent.mt[which(hc.orfl.spof.tot[['species']]=="Multiplet")])
mean(hc.orfl.spof.tot@meta.data$percent.mt[which(is.na(hc.orfl.spof.tot[['species']]))])
mean(hc.orth.sppo.tot@meta.data$percent.mt)
mean(hc.orth.sppo.tot@meta.data$percent.mt[which(hc.orth.sppo.tot[['species']]=="Human")])
mean(hc.orth.sppo.tot@meta.data$percent.mt[which(hc.orth.sppo.tot[['species']]=="Chimp")])
mean(hc.orth.sppo.tot@meta.data$percent.mt[which(hc.orth.sppo.tot[['species']]=="Multiplet")])
mean(hc.orth.sppo.tot@meta.data$percent.mt[which(is.na(hc.orth.sppo.tot[['species']]))])
mean(hc.full.sppf.tot@meta.data$percent.mt)
mean(hc.full.sppf.tot@meta.data$percent.mt[which(hc.full.sppf.tot[['species']]=="Human")])
mean(hc.full.sppf.tot@meta.data$percent.mt[which(hc.full.sppf.tot[['species']]=="Chimp")])
mean(hc.full.sppf.tot@meta.data$percent.mt[which(hc.full.sppf.tot[['species']]=="Multiplet")])
mean(hc.full.sppf.tot@meta.data$percent.mt[which(is.na(hc.full.sppf.tot[['species']]))])
mean(hc.orfl.spof@meta.data$percent.mt)
mean(hc.orfl.spof@meta.data$percent.mt[which(hc.orfl.spof[['species']]=="Human")])
mean(hc.orfl.spof@meta.data$percent.mt[which(hc.orfl.spof[['species']]=="Chimp")])
mean(hc.orfl.sppo@meta.data$percent.mt)
mean(hc.orfl.sppo@meta.data$percent.mt[which(hc.orfl.sppo[['species']]=="Human")])
mean(hc.orfl.sppo@meta.data$percent.mt[which(hc.orfl.sppo[['species']]=="Chimp")])
mean(hc.orfl.sppf@meta.data$percent.mt)
mean(hc.orfl.sppf@meta.data$percent.mt[which(hc.orfl.sppf[['species']]=="Human")])
mean(hc.orfl.sppf@meta.data$percent.mt[which(hc.orfl.sppf[['species']]=="Chimp")])
mean(hc.orth.spof@meta.data$percent.mt)
mean(hc.orth.spof@meta.data$percent.mt[which(hc.orth.spof[['species']]=="Human")])
mean(hc.orth.spof@meta.data$percent.mt[which(hc.orth.spof[['species']]=="Chimp")])
mean(hc.orth.sppo@meta.data$percent.mt)
mean(hc.orth.sppo@meta.data$percent.mt[which(hc.orth.sppo[['species']]=="Human")])
mean(hc.orth.sppo@meta.data$percent.mt[which(hc.orth.sppo[['species']]=="Chimp")])
mean(hc.orth.sppf@meta.data$percent.mt)
mean(hc.orth.sppf@meta.data$percent.mt[which(hc.orth.sppf[['species']]=="Human")])
mean(hc.orth.sppf@meta.data$percent.mt[which(hc.orth.sppf[['species']]=="Chimp")])
mean(hc.full.spof@meta.data$percent.mt)
mean(hc.full.spof@meta.data$percent.mt[which(hc.full.spof[['species']]=="Human")])
mean(hc.full.spof@meta.data$percent.mt[which(hc.full.spof[['species']]=="Chimp")])
mean(hc.full.sppo@meta.data$percent.mt)
mean(hc.full.sppo@meta.data$percent.mt[which(hc.full.sppo[['species']]=="Human")])
mean(hc.full.sppo@meta.data$percent.mt[which(hc.full.sppo[['species']]=="Chimp")])
mean(hc.full.sppf@meta.data$percent.mt)
mean(hc.full.sppf@meta.data$percent.mt[which(hc.full.sppf[['species']]=="Human")])
mean(hc.full.sppf@meta.data$percent.mt[which(hc.full.sppf[['species']]=="Chimp")])
mean(h.orfl.c.orfl.spof@meta.data$percent.mt)
mean(h.orfl.c.orfl.spof@meta.data$percent.mt[which(h.orfl.c.orfl.spof[['species']]=="Human")])
mean(h.orfl.c.orfl.spof@meta.data$percent.mt[which(h.orfl.c.orfl.spof[['species']]=="Chimp")])
mean(h.orfl.c.orfl.sppo@meta.data$percent.mt)
mean(h.orfl.c.orfl.sppo@meta.data$percent.mt[which(h.orfl.c.orfl.sppo[['species']]=="Human")])
mean(h.orfl.c.orfl.sppo@meta.data$percent.mt[which(h.orfl.c.orfl.sppo[['species']]=="Chimp")])
mean(h.orfl.c.orfl.sppf@meta.data$percent.mt)
mean(h.orfl.c.orfl.sppf@meta.data$percent.mt[which(h.orfl.c.orfl.sppf[['species']]=="Human")])
mean(h.orfl.c.orfl.sppf@meta.data$percent.mt[which(h.orfl.c.orfl.sppf[['species']]=="Chimp")])
mean(h.orth.c.orth.spof@meta.data$percent.mt)
mean(h.orth.c.orth.spof@meta.data$percent.mt[which(h.orth.c.orth.spof[['species']]=="Human")])
mean(h.orth.c.orth.spof@meta.data$percent.mt[which(h.orth.c.orth.spof[['species']]=="Chimp")])
mean(h.orth.c.orth.sppo@meta.data$percent.mt)
mean(h.orth.c.orth.sppo@meta.data$percent.mt[which(h.orth.c.orth.sppo[['species']]=="Human")])
mean(h.orth.c.orth.sppo@meta.data$percent.mt[which(h.orth.c.orth.sppo[['species']]=="Chimp")])
mean(h.orth.c.orth.sppf@meta.data$percent.mt)
mean(h.orth.c.orth.sppf@meta.data$percent.mt[which(h.orth.c.orth.sppf[['species']]=="Human")])
mean(h.orth.c.orth.sppf@meta.data$percent.mt[which(h.orth.c.orth.sppf[['species']]=="Chimp")])
mean(h.full.c.full.spof@meta.data$percent.mt)
mean(h.full.c.full.spof@meta.data$percent.mt[which(h.full.c.full.spof[['species']]=="Human")])
mean(h.full.c.full.spof@meta.data$percent.mt[which(h.full.c.full.spof[['species']]=="Chimp")])
mean(h.full.c.full.sppo@meta.data$percent.mt)
mean(h.full.c.full.sppo@meta.data$percent.mt[which(h.full.c.full.sppo[['species']]=="Human")])
mean(h.full.c.full.sppo@meta.data$percent.mt[which(h.full.c.full.sppo[['species']]=="Chimp")])
mean(h.full.c.full.sppf@meta.data$percent.mt)
mean(h.full.c.full.sppf@meta.data$percent.mt[which(h.full.c.full.sppf[['species']]=="Human")])
mean(h.full.c.full.sppf@meta.data$percent.mt[which(h.full.c.full.sppf[['species']]=="Chimp")])

mean(h.orth.c.orth.sppo@meta.data$nCount_RNA)
mean(h.orth.c.orth.sppo@meta.data$nCount_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="iPSC")])
mean(h.orth.c.orth.sppo@meta.data$nCount_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="MSC")])
mean(h.orth.c.orth.sppo@meta.data$nCount_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="chondrocyte")])
mean(h.orth.c.orth.sppo@meta.data$nCount_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="osteoblast")])

mean(h.orth.c.orth.sppo@meta.data$nFeature_RNA)
mean(h.orth.c.orth.sppo@meta.data$nFeature_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="iPSC")])
mean(h.orth.c.orth.sppo@meta.data$nFeature_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="MSC")])
mean(h.orth.c.orth.sppo@meta.data$nFeature_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="chondrocyte")])
mean(h.orth.c.orth.sppo@meta.data$nFeature_RNA[which(h.orth.c.orth.sppo[['orig.ident']]=="osteoblast")])

#notes: full data umi counts and gene counts have doubled as compared to subset data


#FULL DATA
#                   nCount_RNA  human-nCount_RNA  chimp-nCount_RNA  multiplet-nCount_RNA  iPSC-nCount_RNA MSC-nCount_RNA  chondro-nCount_RNA  osteo-nCount_RNA
#hc.orfl.spof.tot    5457.740    6302.898          4572.984           8790.356
#hc.orth.sppo.tot    5612.830    6476.851          4669.163           9650.047
#hc.full.sppf.tot    8875.286    7979.846          5601.706          13578.770
#hc.orfl.spof        5143.125    6067.656          4309.084                NaN
#hc.orfl.sppo        5118.742    6057.872          4270.349                NaN
#hc.orfl.sppf        2995.719    3273.116          2512.351                NaN
#hc.orth.spof        5242.445    6218.968          4361.502                NaN
#hc.orth.sppo        5217.790    6208.940          4322.405                NaN
#hc.orth.sppf        3062.156    3360.144          2542.905                NaN
#hc.full.spof        8051.487   10384.330          5946.979                NaN
#hc.full.sppo        8019.489   10366.850          5898.931                NaN
#hc.full.sppf        4965.488    5718.165          3653.935                NaN
#h.orfl.c.orfl.spof 10575.160   12167.850          9138.366                NaN            8896.104        10407.250       9840.240            14360.830
#h.orfl.c.orfl.sppo 10525.410   12148.820          9058.855                NaN
#h.orfl.c.orfl.sppf  6257.233    6703.269          5480.009                NaN
#h.orth.c.orth.spof 10805.770   12437.780          9333.501                NaN
#h.orth.c.orth.sppo 10755.070   12418.320          9252.530                NaN            9159.721        10603.350       9940.126            14492.640
#h.orth.c.orth.sppf  6400.214    6860.904          5597.452                NaN
#h.full.c.full.spof 17044.930   20310.190         14099.270                NaN
#h.full.c.full.sppo 16974.240   20278.600         13989.150                NaN
#h.full.c.full.sppf 10569.240   11550.600          8859.207                NaN

#                   nFeature_RNA  human-nFeature_RNA  chimp-nFeature_RNA  mult-nFeature_RNA  iPSC-nFeature MSC-nFeature  chondro-nFeature  osteo-nFeature
#hc.orfl.spof.tot   2139.842      2315.419            1930.934            3516.182
#hc.orth.sppo.tot   2211.913      2391.449            1990.905            3667.583
#hc.full.sppf.tot   2617.230      2312.469            1991.840            3742.470
#hc.orfl.spof       1956.312      2163.289            1769.593                 NaN
#hc.orfl.sppo       1951.154      2160.750            1761.809                 NaN
#hc.orfl.sppf       1247.427      1303.382            1149.924                 NaN
#hc.orth.spof       2003.154      2217.882            1809.444                 NaN
#hc.orth.sppo       1997.870      2215.289            1801.459                 NaN
#hc.orth.sppf       1277.694      1336.683            1174.906                 NaN
#hc.full.spof       2441.793      3000.438            1937.827                 NaN
#hc.full.sppo       2436.593      2997.191            1930.159                 NaN
#hc.full.sppf       1654.247      1856.294            1302.176                 NaN
#h.orfl.c.orfl.spof 3110.086      3353.502            2890.495                 NaN          3067.454        3036.859      2890.354          3525.015
#h.orfl.c.orfl.sppo 3102.956      3350.221            2879.582                 NaN
#h.orfl.c.orfl.sppf 2046.536      2102.474            1949.064                 NaN
#h.orth.c.orth.spof 3209.078      3459.292            2983.353                 NaN
#h.orth.c.orth.sppo 3201.680      3455.898            2972.024                 NaN          3175.148        3130.352      2950.764          3623.477
#h.orth.c.orth.sppf 2110.994      2168.313            2011.114                 NaN
#h.full.c.full.spof 3623.759      4128.418            3168.496                 NaN
#h.full.c.full.sppo 3616.742      4124.657            3157.902                 NaN
#h.full.c.full.sppf 2494.201      2654.268            2215.282                 NaN

#                   percent.mt  human-percent.mt  chimp-percent.mt  multiplet-percent.mt
#hc.orfl.spof.tot            0            0                0                 0
#hc.orth.sppo.tot            0            0                0                 0
#hc.full.sppf.tot   0.1930822    0.33117160       0.05163908        0.09657800
#hc.orfl.spof                0            0                0               NaN
#hc.orfl.sppo                0            0                0               NaN
#hc.orfl.sppf                0            0                0               NaN
#hc.orth.spof                0            0                0               NaN
#hc.orth.sppo                0            0                0               NaN
#hc.orth.sppf                0            0                0               NaN
#hc.full.spof       0.20743790   0.38987290       0.04285910               NaN
#hc.full.sppo       0.20804850   0.39064420       0.04309509               NaN
#hc.full.sppf       1.28609200   1.90989000       0.19911270               NaN
#h.orfl.c.orfl.spof          0            0                0               NaN
#h.orfl.c.orfl.sppo          0            0                0               NaN
#h.orfl.c.orfl.sppf          0            0                0               NaN
#h.orth.c.orth.spof          0            0                0               NaN
#h.orth.c.orth.sppo          0            0                0               NaN
#h.orth.c.orth.sppf          0            0                0               NaN
#h.full.c.full.spof 0.11365390   0.21196590       0.02496430               NaN
#h.full.c.full.sppo 0.11409380   0.21250350       0.02519230               NaN
#h.full.c.full.sppf 0.67146510   1.00485500       0.09052633               NaN


#SUBSET DATA
#                   nCount_RNA  human-nCount_RNA  chimp-nCount_RNA  multiplet-nCount_RNA  na-nCount_RNA
#hc.orth.sppo.tot   2371.906    2761.382          1975.289           3719.281                  NaN
#hc.full.sppf.tot   4302.386    4847.341          2935.121           6356.990                  NaN
#hc.orth.sppo       2222.459    2649.121          1832.667                NaN                  NaN
#hc.orth.sppf       1753.557    2071.651          1321.109                NaN                  NaN
#hc.full.sppo       3448.402    4474.880          2510.631                NaN                  NaN
#hc.full.sppf       2818.474    3503.228          1887.550                NaN                  NaN
#h.orth.c.orth.sppo 4696.994    5446.770          4012.013                NaN                  NaN
#h.orth.c.orth.sppf 3713.048    4268.583          2957.798                NaN                  NaN
#h.full.c.full.sppo 7558.132    9084.743          6163.449                NaN                  NaN
#h.full.c.full.sppf 6108.546    7117.030          4737.513                NaN                  NaN
##h.full.GHO        7710.547    8875.383          6521.622          12879.370             721.2218
##c.full.GHO        6707.277    7068.757          5808.117          10980.450             710.6510
##hc.full.GHO       4302.386    4847.341          2935.121           6356.990                  NaN
##h.orth.GHO        4573.710    5453.747          4064.034           7458.208             882.3912
##c.orth.GHO        4377.796    5303.141          4069.787           7342.996             864.7339
##hc.orth.GHO       2371.906    2761.382          1975.289           3719.281                  NaN

#                   nFeature_RNA  human-nFeature_RNA  chimp-nFeature_RNA  multiplet-nFeature_RNA  na-nFeature_RNA
#hc.orth.sppo.tot   1237.1900     1363.685            1097.2760           2041.358                     NaN
#hc.full.sppf.tot   1723.9860     1817.574            1382.2910           2427.428                     NaN
#hc.orth.sppo       1134.9290     1278.377            1003.8780                NaN                     NaN
#hc.orth.sppf        927.1996     1020.998             799.6812                NaN                     NaN
#hc.full.sppo       1467.8140     1836.270            1131.1990                NaN                     NaN
#hc.full.sppf       1240.8530     1475.426             921.9515                NaN                     NaN
#h.orth.c.orth.sppo 1978.9640     2174.450            1800.3710                NaN                     NaN
#h.orth.c.orth.sppf 1633.5160     1755.244            1468.0260                NaN                     NaN
#h.full.c.full.sppo 2388.2110     2766.840            2042.3030                NaN                     NaN
#h.full.c.full.sppf 2018.0380     2248.221            1705.1050                NaN                     NaN
##h.full.GHO        2340.7200     2567.073            2244.9320           3438.305                423.6944
##c.full.GHO        2074.7470     2162.264            1954.8820           2897.855                437.7427
##hc.full.GHO       1723.9860     1817.574            1382.2910           2427.428                     NaN
##h.orth.GHO        1932.3710     2176.849            1822.2290           2782.157                600.9958
##c.orth.GHO        1874.8310     2154.300            1822.8130           2768.128                590.5218
##hc.orth.GHO       1237.1900     1363.685            1097.2760           2041.358                     NaN

#                   percent.mt  human-percent.mt  chimp-percent.mt  multiplet-percent.mt  na-percent.mt
#hc.orth.sppo.tot            0            0                0                 0                   NaN
#hc.full.sppf.tot   5.74569800  10.54831000       0.52054250        1.46853100                   NaN
#hc.orth.sppo                0            0                0               NaN                   NaN
#hc.orth.sppf                0            0                0               NaN                   NaN
#hc.full.sppo       0.19451480   0.36719180       0.03676023               NaN                   NaN
#hc.full.sppf       0.59853400   0.97902040       0.08126272               NaN                   NaN
#h.orth.c.orth.sppo          0            0                0               NaN                   NaN
#h.orth.c.orth.sppf          0            0                0               NaN                   NaN
#h.full.c.full.sppo 0.10142190   0.19061180       0.01993968               NaN                   NaN
#h.full.c.full.sppf 0.31886820   0.52362000       0.04050812               NaN                   NaN
##h.full.GHO        5.66381000   6.49458900       3.13214300        2.39591800            13.8319400
##c.full.GHO        0.08204637   0.13405240       0.01832258        0.02768889             0.1388891
##hc.full.GHO       5.74569800  10.54831000       0.52054250        1.46853100                   NaN
##h.orth.GHO                 0            0                0                 0                     0
##c.orth.GHO                 0            0                0                 0                     0
##hc.orth.GHO                0            0                0                 0                   NaN


#VlnPlot(hc.orth.sppo, features="nCount_RNA", pt.size=0.1, group.by="species", y.max=60000)
#VlnPlot(hc.orth.sppo, features="nCount_RNA", pt.size=0.1, group.by="orig.ident", y.max=60000)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,120000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,120000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,120000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,120000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,120000)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,120000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,120000)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,120000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,120000))), ncol=7)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,120000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,120000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,120000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,120000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,120000)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,120000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,120000)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,120000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,120000))), ncol=7)
#notes: number of umi per cell increases: hc.orfl~hc.orth < hc.full < h.orfl.c.orfl~h.orth.c.orth < h.full.c.full (regardless of spof/sppo/sppf filter)
#notes: number of umi per cell increases: sppf < spof~sppo (regardless of orfl/orth/full dataset)
#notes: humans have more umi per cell than chimps
#notes: number of umi per cell increases: iPSC < MSC < chondrocyte < osteocyte

#VlnPlot(hc.orth.sppo, features="nFeature_RNA", pt.size=0.1, group.by="species", y.max=8000)
#VlnPlot(hc.orth.sppo, features="nFeature_RNA", pt.size=0.1, group.by="orig.ident", y.max=8000)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,5000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,5000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,5000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,5000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,5000)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,5000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,5000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,5000)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,5000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,5000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,5000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,5000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,5000))), ncol=7)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,5000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,5000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,5000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,5000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,5000)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,5000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,5000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,5000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,5000)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,5000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,5000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,5000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,5000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,5000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,5000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,5000))), ncol=7)
#notes: number of genes per cell increases: hc.orfl~hc.orth < hc.full < h.orfl.c.orfl~h.orth.c.orth < h.full.c.full (regardless of spof/sppo/sppf filter)
#notes: number of genes per cell increases: sppf < spof~sppo (regardless of orfl/orth/full dataset)
#notes: humans have more genes per cell than chimps
#notes: number of genes per cell increases: iPSC < MSC < chondrocyte < osteocyte

#VlnPlot(hc.orth.sppo, features="percent.mt", pt.size=0.1, group.by="species", y.max=100)
#VlnPlot(hc.orth.sppo, features="percent.mt", pt.size=0.1, group.by="orig.ident", y.max=100)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,100)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,100)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,100)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,100)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,100)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,100)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,100)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,100)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,100)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,100)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,100)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,100)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,100)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,100)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,100))), ncol=7)
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orlf.spof")+ylim(0,100)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.spof")+ylim(0,100)),
                        (ggplot(hc.full.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.full.spof")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.spof")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.spof")+ylim(0,100)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.spof")+ylim(0,100)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.spof.tot")+ylim(0,100)),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.sppo")+ylim(0,100)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppo")+ylim(0,100)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppo")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.sppo")+ylim(0,100)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.sppo")+ylim(0,100)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppo.tot")+ylim(0,100)),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orfl.sppf")+ylim(0,100)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.orth.sppf")+ylim(0,100)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppf")+ylim(0,100)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,100)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.orth.c.orth.sppf")+ylim(0,100)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="h.full.c.full.sppf")+ylim(0,100)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=orig.ident,y=percent.mt))+geom_boxplot()+labs(title="hc.full.sppf.tot")+ylim(0,100))), ncol=7)
#notes: percent mito decreases: hc.full > h.full.c.full (regardless of sppo/sppf filter)
#notes: percent mito is lower with spof~sppo filters than sppf filter
#notes: humans have higher percent mito than chimps
#notes: percent mito increases: iPSC < MSC < osteocyte < chondrocyte
#notes: with the hc.full.sppf percent mito is high in iPSC + MSC + chondrocyte / lower osteocyte

#FeatureScatter(hc.orth.sppo,feature1="nCount_RNA",feature2="percent.mt",pt.size=0.1,group.by="species")
#FeatureScatter(hc.orth.sppo,feature1="nCount_RNA",feature2="percent.mt",pt.size=0.1,group.by="orig.ident")
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orlf.spof")),
                        (ggplot(hc.orth.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orth.spof")),
                        (ggplot(hc.full.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.spof")),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orfl.c.orfl.spof")),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orth.c.orth.spof")),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.spof")),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orfl.spof.tot")),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orfl.sppo")),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orth.sppo")),
                        (ggplot(hc.full.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppo")),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orfl.c.orfl.sppo")),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orth.c.orth.sppo")),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppo")),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orth.sppo.tot")),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orfl.sppf")),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.orth.sppf")),
                        (ggplot(hc.full.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orfl.c.orfl.sppf")),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.orth.c.orth.sppf")),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=nCount_RNA,y=percent.mt,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf.tot"))), ncol=7)

#FeatureScatter(hc.orth.sppo,feature1="nCount_RNA",feature2="nFeature_RNA",pt.size=0.1,group.by="species")
#FeatureScatter(hc.orth.sppo,feature1="nCount_RNA",feature2="nFeature_RNA",pt.size=0.1,group.by="orig.ident")
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orlf.spof")),
                        (ggplot(hc.orth.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orth.spof")),
                        (ggplot(hc.full.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.full.spof")),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orfl.c.orfl.spof")),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orth.c.orth.spof")),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.full.c.full.spof")),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orfl.spof.tot")),
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orfl.sppo")),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orth.sppo")),
                        (ggplot(hc.full.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.full.sppo")),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orfl.c.orfl.sppo")),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orth.c.orth.sppo")),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.full.c.full.sppo")),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orth.sppo.tot")),
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orfl.sppf")),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.orth.sppf")),
                        (ggplot(hc.full.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.full.sppf")),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orfl.c.orfl.sppf")),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.orth.c.orth.sppf")),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="h.full.c.full.sppf")),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=nCount_RNA,y=nFeature_RNA,color=orig.ident,shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,5000)+labs(title="hc.full.sppf.tot"))), ncol=7)


#Overall, using the h.orfl.c.orfl.spof dataset for downstream analyses makes the most sense.
#Total Number of Features Examined = 28942 (While this number of features is less than the 44215 features for the sppo datasets, the filtering of features to only include protein_coding, lincRNA, antisense, gene, and pseudogene features is the method that cellranger recommends. Additionally, this number of features is greater than the 16542 features for the full datasets. Further, since the spof datasets contain orthologous genes as opposed to overlapping gene names, this method is a more theoretically sound starting place for comparative analyses.)
#Total Number of Cells Examined = 18190 (This number of cells is slightly higher than the 18183 for the sppo filter and slightly lower than the 18693 for the sppf filter. Most of this differences in cell numbers is isolated to human chondrocytes. Nevertheless, this mixture of human and chimp cells [H=8267,C=9563], as well as that for the sppo filter [H=8630,C=9553], is more even  than that using the sppf filter [H=11877,C=6816].)
#UMI and Genes per Cell (This has the second highest raw count of umi per cell and genes per cell. The h.orth.c.orth datasets have comparable values, and the h.full.c.full datasets have slightly higher values. However, these data have filtering and orthology issues, respectively.)
#Percentage of Mitochondrial Genes (Using the orfl data, as well as the orth data, means that there is no information regarding mitochondrial gene expression. Conversely, such information is available for the full data. Nevertheless, the spof filter, as well as the sppo filter, does seem to retain cells will lower percent.mt that is more consistent across cell types than the sppf filter [when examining the hc.full and h.full.c.full datasets].)
#Some other general trends to be aware of (that are common across datasets) include:
# - number of umi/genes per cell increases across species: chimpanzee < human
# - number of umi/genes per cell increases across cell type: iPSC < MSC < chondrocyte < osteocyte
# - percent.mt increases across species: chimpanzee < human
# - percent.mt generally increases across cell type: iPSC < MSC < osteocyte < chondrocyte

```

```{r save data temporarily}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
saveRDS(hc.orfl.spof.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.tot.rds"))
saveRDS(hc.orth.sppo.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.tot.rds"))
saveRDS(hc.full.sppf.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.tot.rds"))
saveRDS(hc.orfl.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.rds"))
saveRDS(hc.orfl.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppo.rds"))
saveRDS(hc.orfl.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppf.rds"))
saveRDS(hc.orth.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.spof.rds"))
saveRDS(hc.orth.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.rds"))
saveRDS(hc.orth.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppf.rds"))
saveRDS(hc.full.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.spof.rds"))
saveRDS(hc.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppo.rds"))
saveRDS(hc.full.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.rds"))
saveRDS(h.orfl.c.orfl.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
saveRDS(h.orfl.c.orfl.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppo.rds"))
saveRDS(h.orfl.c.orfl.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppf.rds"))
saveRDS(h.orth.c.orth.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.spof.rds"))
saveRDS(h.orth.c.orth.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
saveRDS(h.orth.c.orth.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppf.rds"))
saveRDS(h.full.c.full.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.spof.rds"))
saveRDS(h.full.c.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
saveRDS(h.full.c.full.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppf.rds"))

```


```{r sequencing metrics}

#Comparison of full dataset (3 lanes of sequencing) vs. subset dataset (1 lane of sequencing) in h.orth.c.orth.sppo dataset:
#(as well as data I will actually end up using -> h.orfl.c.orfl.spof)

dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA'
metrics.h.data.actu.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-human/GHO-1/outs/metrics_summary.csv'))
metrics.h.data.actu.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-human/GHO-2/outs/metrics_summary.csv'))
metrics.h.data.actu.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-human/GHO-3/outs/metrics_summary.csv'))
metrics.h.data.actu.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-human/GHO-4/outs/metrics_summary.csv'))
metrics.c.data.actu.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-chimp/GHO-1/outs/metrics_summary.csv'))
metrics.c.data.actu.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-chimp/GHO-2/outs/metrics_summary.csv'))
metrics.c.data.actu.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-chimp/GHO-3/outs/metrics_summary.csv'))
metrics.c.data.actu.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-filter-chimp/GHO-4/outs/metrics_summary.csv'))
metrics.h.data.actu <- rbind(metrics.h.data.actu.GHO1,metrics.h.data.actu.GHO2,metrics.h.data.actu.GHO3,metrics.h.data.actu.GHO4)
metrics.c.data.actu <- rbind(metrics.c.data.actu.GHO1,metrics.c.data.actu.GHO2,metrics.c.data.actu.GHO3,metrics.c.data.actu.GHO4)
metrics.h.data.actu[,c(1:4,19:20)] <- lapply(metrics.h.data.actu[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.data.actu[,c(1:4,19:20)] <- lapply(metrics.c.data.actu[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.h.data.actu[,c(5:18)] <- lapply(metrics.h.data.actu[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.data.actu[,c(5:18)] <- lapply(metrics.c.data.actu[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.h.data.actu$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.data.actu$Sample <- c("GHO1","GHO2","GHO3","GHO4")
rm(metrics.h.data.actu.GHO1,metrics.h.data.actu.GHO2,metrics.h.data.actu.GHO3,metrics.h.data.actu.GHO4)
rm(metrics.c.data.actu.GHO1,metrics.c.data.actu.GHO2,metrics.c.data.actu.GHO3,metrics.c.data.actu.GHO4)
metrics.h.data.full.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-human/GHO-1/outs/metrics_summary.csv'))
metrics.h.data.full.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-human/GHO-2/outs/metrics_summary.csv'))
metrics.h.data.full.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-human/GHO-3/outs/metrics_summary.csv'))
metrics.h.data.full.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-human/GHO-4/outs/metrics_summary.csv'))
metrics.c.data.full.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-chimp/GHO-1/outs/metrics_summary.csv'))
metrics.c.data.full.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-chimp/GHO-2/outs/metrics_summary.csv'))
metrics.c.data.full.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-chimp/GHO-3/outs/metrics_summary.csv'))
metrics.c.data.full.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-full/cellranger-runs-ortho-chimp/GHO-4/outs/metrics_summary.csv'))
metrics.h.data.full <- rbind(metrics.h.data.full.GHO1,metrics.h.data.full.GHO2,metrics.h.data.full.GHO3,metrics.h.data.full.GHO4)
metrics.c.data.full <- rbind(metrics.c.data.full.GHO1,metrics.c.data.full.GHO2,metrics.c.data.full.GHO3,metrics.c.data.full.GHO4)
metrics.h.data.full[,c(1:4,19:20)] <- lapply(metrics.h.data.full[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.data.full[,c(1:4,19:20)] <- lapply(metrics.c.data.full[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.h.data.full[,c(5:18)] <- lapply(metrics.h.data.full[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.data.full[,c(5:18)] <- lapply(metrics.c.data.full[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.h.data.full$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.data.full$Sample <- c("GHO1","GHO2","GHO3","GHO4")
rm(metrics.h.data.full.GHO1,metrics.h.data.full.GHO2,metrics.h.data.full.GHO3,metrics.h.data.full.GHO4)
rm(metrics.c.data.full.GHO1,metrics.c.data.full.GHO2,metrics.c.data.full.GHO3,metrics.c.data.full.GHO4)
metrics.h.data.subs.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-human/GHO-1/outs/metrics_summary.csv'))
metrics.h.data.subs.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-human/GHO-2/outs/metrics_summary.csv'))
metrics.h.data.subs.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-human/GHO-3/outs/metrics_summary.csv'))
metrics.h.data.subs.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-human/GHO-4/outs/metrics_summary.csv'))
metrics.c.data.subs.GHO1 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-chimp/GHO-1/outs/metrics_summary.csv'))
metrics.c.data.subs.GHO2 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-chimp/GHO-2/outs/metrics_summary.csv'))
metrics.c.data.subs.GHO3 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-chimp/GHO-3/outs/metrics_summary.csv'))
metrics.c.data.subs.GHO4 <- read.csv(file=paste0(dir,'/cellranger-data-subset/cellranger-runs-ortho-chimp/GHO-4/outs/metrics_summary.csv'))
metrics.h.data.subs <- rbind(metrics.h.data.subs.GHO1,metrics.h.data.subs.GHO2,metrics.h.data.subs.GHO3,metrics.h.data.subs.GHO4)
metrics.c.data.subs <- rbind(metrics.c.data.subs.GHO1,metrics.c.data.subs.GHO2,metrics.c.data.subs.GHO3,metrics.c.data.subs.GHO4)
metrics.h.data.subs[,c(1:4,18:19)] <- lapply(metrics.h.data.subs[,c(1:4,18:19)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.c.data.subs[,c(1:4,18:19)] <- lapply(metrics.c.data.subs[,c(1:4,18:19)], function(x) as.numeric(gsub(",","",as.character(x))))
metrics.h.data.subs[,c(5:17)] <- lapply(metrics.h.data.subs[,c(5:17)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.c.data.subs[,c(5:17)] <- lapply(metrics.c.data.subs[,c(5:17)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
metrics.h.data.subs$Sample <- c("GHO1","GHO2","GHO3","GHO4")
metrics.c.data.subs$Sample <- c("GHO1","GHO2","GHO3","GHO4")
rm(metrics.h.data.subs.GHO1,metrics.h.data.subs.GHO2,metrics.h.data.subs.GHO3,metrics.h.data.subs.GHO4)
rm(metrics.c.data.subs.GHO1,metrics.c.data.subs.GHO2,metrics.c.data.subs.GHO3,metrics.c.data.subs.GHO4)

#number of reads sequenced (sequence saturation):    actu = 282750845 (0.35075)   vs.   full = 282750845 (0.34925)    vs.   subset = 97851860 (0.23275)

#median reads per cell more than doubled:    actu: h.orfl = 61638.00      vs.   full: h.orth = 58717.50      vs.   subset: h.orth = 22705.50
#                                                  c.orfl = 61350.75                  c.orth = 55826.75                    c.orth = 21687.75
#median genes per cell increased by 1000:    actu: h.orfl =  3020.25      vs.   full: h.orth =  3003.25      vs.   subset: h.orth =  1902.75
#                                                  c.orfl =  2996.75                  c.orth =  2882.00                    c.orth =  1809.00
#median umi per cell more than doubled:      actu: h.orfl =  9082.75      vs.   full: h.orth =  8891.25      vs.   subset: h.orth =  4149.50
#                                                  c.orfl =  8965.50                  c.orth =  8434.50                    c.orth =  3864.50
CombinePlots(plots=list((ggplot(metrics.h.data.actu, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Actual Data\nh.orfl")+ylim(0,1)),
                        (ggplot(metrics.c.data.actu, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Actual Data\nc.orfl")+ylim(0,1)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Actual Data\nh.orfl")+ylim(0,80000)),
                        (ggplot(metrics.c.data.actu, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Actual Data\nc.orfl")+ylim(0,80000)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Actual Data\nh.orfl")+ylim(0,3800)),
                        (ggplot(metrics.c.data.actu, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Actual Data\nc.orfl")+ylim(0,3800)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Actual Data\nh.orfl")+ylim(0,13100)),
                        (ggplot(metrics.c.data.actu, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Actual Data\nc.orfl")+ylim(0,13100)),
                        (ggplot(metrics.h.data.full, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Full\nh.orth")+ylim(0,1)),
                        (ggplot(metrics.c.data.full, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Full\nc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.data.full, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Full\nh.orth")+ylim(0,80000)),
                        (ggplot(metrics.c.data.full, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Full\nc.orth")+ylim(0,80000)),
                        (ggplot(metrics.h.data.full, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Full\nh.orth")+ylim(0,3800)),
                        (ggplot(metrics.c.data.full, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Full\nc.orth")+ylim(0,3800)),
                        (ggplot(metrics.h.data.full, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Full\nh.orth")+ylim(0,13100)),
                        (ggplot(metrics.c.data.full, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Full\nc.orth")+ylim(0,13100)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Subset\nh.orth")+ylim(0,1)),
                        (ggplot(metrics.c.data.subs, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Subset\nc.orth")+ylim(0,1)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Subset\nh.orth")+ylim(0,80000)),
                        (ggplot(metrics.c.data.subs, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Subset\nc.orth")+ylim(0,80000)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Subset\nh.orth")+ylim(0,3800)),
                        (ggplot(metrics.c.data.subs, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Subset\nc.orth")+ylim(0,3800)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Subset\nh.orth")+ylim(0,13100)),
                        (ggplot(metrics.c.data.subs, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Subset\nc.orth")+ylim(0,13100))),
             ncol=8) #saved as compare.readstats

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
data.actu <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
data.full <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
data.subs <- readRDS(paste0(dir,"/data/cellranger-data-subset/x-h.orth.c.orth.sppo.rds"))
                     
#median(data.actu@meta.data$nCount_RNA)
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['species']]=="Human")])
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['species']]=="Chimp")])
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['orig.ident']]=="iPSC")])
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['orig.ident']]=="MSC")])
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['orig.ident']]=="chondrocyte")])
#median(data.actu@meta.data$nCount_RNA[which(data.actu[['orig.ident']]=="osteoblast")])

#average number of umi per cell more than doubled:   actu = 10575               vs.        full = 10525               vs.        subset =  4696
#                                                    (H:12167 / C:9138)                    (H:12148 / C:9058)                    (H:5446 / C:4012)
#                                                    (I:8896 / M:10407 / C:9840 / O:14360) (I:9159 / M:10603 / C:9940 / O:14492) (I:3640 / M:4294 / C:5140 / O:6835)
#median number of umi per cell more than doubled:    actu = 9240                vs.        full =  9422               vs.        subset =  3945
#                                                    (H:11509 / C:7930)                    (H:11770 / C:8101)                    (H:4904 / C:3342)
#                                                    (I:7729 / M:9803 / C:8189 / O:13332)  (I:7954 / M: 9990 / C:8232 / O:13540) (I:3144 / M:4027 / C:4818 / O:6231)
CombinePlots(plots=list((ggplot(data.actu@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="Actual Data\nh.orfl.c.orfl.spof")+ylim(0,75000)),
                        (ggplot(data.full@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="Full Data\nh.orth.c.orth.sppo")+ylim(0,75000)),
                        (ggplot(data.subs@meta.data, aes(x=species,y=nCount_RNA))+geom_boxplot()+labs(title="Data Subset\nh.orth.c.orth.sppo")+ylim(0,75000)),
                        (ggplot(data.actu@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="Actual Data\nh.orfl.c.orfl.spof")+ylim(0,75000)),
                        (ggplot(data.full@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="Full Data\nh.orth.c.orth.sppo")+ylim(0,75000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident,y=nCount_RNA))+geom_boxplot()+labs(title="Data Subset\nh.orth.c.orth.sppo")+ylim(0,75000))),
             ncol=3) #saved as compare.nCount

#average number of genes per cell increased by 1000: actu = 3110                vs.        full = 3102                vs.        subset =  1978
#                                                    (H:3353 / C:2890)                     (H:3350 / C:2879)                     (H:2174 / C:1800)
#                                                    (I:3067 / M:3036 / C:2890 / O:3525)   (I:3175 / M:3130 / C:2950 / O:3623)   (I:1800 / M:1845 / C:2047 / O:2462)
#median number of genes per cell increased by 1000:  actu = 3107                vs.        full = 3204                vs.        subset =  1872
#                                                    (H:3444 / C:2874)                     (H:3553 / C:2961)                     (H:2114 / C:1691)
#                                                    (I:2945 / M:3073 / C:2913 / O:3761)   (I:3047 / M:3169 / C:2938 / O:3870)   (I:1683 / M:1822 / C:2135 / O:2495)
CombinePlots(plots=list((ggplot(data.actu@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="Actual Data\nh.orfl.c.orfl.spof")+ylim(0,8000)),
                        (ggplot(data.full@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="Full Data\nh.orth.c.orth.sppo")+ylim(0,8000)),
                        (ggplot(data.subs@meta.data, aes(x=species,y=nFeature_RNA))+geom_boxplot()+labs(title="Data Subset\nh.orth.c.orth.sppo")+ylim(0,8000)),
                        (ggplot(data.actu@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="Actual Data\nh.orfl.c.orfl.spof")+ylim(0,8000)),
                        (ggplot(data.full@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="Full Data\nh.orth.c.orth.sppo")+ylim(0,8000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident,y=nFeature_RNA))+geom_boxplot()+labs(title="Data Subset\nh.orth.c.orth.sppo")+ylim(0,8000))),
             ncol=3) #saved as compare.nFeature

#number of cells only increased slightly: actual = 17830            vs.         full = 18183              vs.         subset = 17316
#                                         (H:8627 / C:9563)                     (H:8630 / C:9553)                     (H:8267 / C:9049)
#                                         (I:5614 / M:5706 / C:3456 / O:3414)   (I:5613 / M:5712 / C:3473 / O:3385)   (I:5513 / M:5509 / C:3197 / O:3097)
CombinePlots(plots=list((ggplot(data.actu@meta.data, aes(x=orig.ident, fill=species))+geom_bar()+labs(title="Actual Data\nh.orfl.c.orfl.spof")+ylim(0,6000)),
                        (ggplot(data.full@meta.data, aes(x=orig.ident, fill=species))+geom_bar()+labs(title="Full Data\nh.orth.c.orth.sppo")+ylim(0,6000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident, fill=species))+geom_bar()+labs(title="Data Subset\nh.orth.c.orth.sppo")+ylim(0,6000))),
             ncol=3) #saved as compare.cellcount

#summary plots
CombinePlots(plots=list((ggplot(metrics.h.data.actu, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Full")+ylim(0,1)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Full")+ylim(0,80000)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Full")+ylim(0,3800)),
                        (ggplot(metrics.h.data.actu, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Full")+ylim(0,13100)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Sequencing.Saturation))+geom_col()+labs(title="Data Subset")+ylim(0,1)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Mean.Reads.per.Cell))+geom_col()+labs(title="Data Subset")+ylim(0,80000)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Median.Genes.per.Cell))+geom_col()+labs(title="Data Subset")+ylim(0,3800)),
                        (ggplot(metrics.h.data.subs, aes(x=Sample, y=Median.UMI.Counts.per.Cell))+geom_col()+labs(title="Data Subset")+ylim(0,13100))),
             ncol=4) #saved as compare.summary1
CombinePlots(plots=list((ggplot(data.actu@meta.data, aes(x=orig.ident,y=nCount_RNA,fill=species))+geom_boxplot()+labs(title="Data Full")+ylim(0,75000)),
                        (ggplot(data.actu@meta.data, aes(x=orig.ident,y=nFeature_RNA,fill=species))+geom_boxplot()+labs(title="Data Full")+ylim(0,8000)),
                        (ggplot(data.actu@meta.data, aes(x=orig.ident, fill=species))+geom_bar()+labs(title="Data Full")+ylim(0,6000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident,y=nCount_RNA,fill=species))+geom_boxplot()+labs(title="Data Subset")+ylim(0,75000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident,y=nFeature_RNA,fill=species))+geom_boxplot()+labs(title="Data Subset")+ylim(0,8000)),
                        (ggplot(data.subs@meta.data, aes(x=orig.ident, fill=species))+geom_bar()+labs(title="Data Subset")+ylim(0,6000))),
             ncol=3) #saved as compare.summary2

rm(data.actu,data.full,data.subs)

#notes: 10X recommends sequencing 50,000 reads/cell (minimum of 20,000 reads/cell)
#notes: 50,000 reads/cell may only have 30-50% saturation (cellranger) but this still may be sufficient for clustering
#notes: 50,000 reads/cell corresponds to >1000 genes/cell (median) and >3500 UMI/cell (median)
#https://assets.ctfassets.net/an68im79xiti/6gDArDPBTOg4IIkYEO2Sis/803be2286bba5ca67f353e6baf68d276/CG000148_10x_Technical_Note_Resolving_Cell_Types_as_Function_of_Read_Depth_Cell_Number_RevA.pdf

#notes: both actual data (h.orfl.c.orfl.spof) and full data (h.orth.c.orth.sppo) meet these criteria
#       --> moving forward each collection should have ~3 lanes of sequencing

```

```{r load processed data}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
hc.orfl.spof.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.tot.rds"))
hc.orth.sppo.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.tot.rds"))
hc.full.sppf.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.tot.rds"))
hc.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.rds"))
hc.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppo.rds"))
hc.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppf.rds"))
hc.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.spof.rds"))
hc.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.rds"))
hc.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppf.rds"))
hc.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.spof.rds"))
hc.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppo.rds"))
hc.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.rds"))
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
h.orfl.c.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppo.rds"))
h.orfl.c.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppf.rds"))
h.orth.c.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.spof.rds"))
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
h.orth.c.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppf.rds"))
h.full.c.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.spof.rds"))
h.full.c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
h.full.c.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppf.rds"))

```

```{r data filtering}

#Filter Barcodes

#(1) Filter based on mitochondrial RNA content: Cells lyse more easily than mitochondria, so especially high mitochondrial RNA content is a potential sign of cell lysis and RNA degradation. Ideally, percent.mt should be less than 5%.

#notes: only full datasets can be filtered using this criteria
CombinePlots(plots=list((ggplot(hc.full.spof@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(hc.full.sppo@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppo")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(hc.full.sppf@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf.tot")+
                           geom_hline(yintercept=7.5, linetype="dashed"))), ncol=4)

#notes: hc.full.sppf dataset - most high mito counts are in human iPSC, human MSC, and human chondrocyte cells
CombinePlots(plots=list((ggplot(subset(hc.full.sppf@meta.data, orig.ident=="iPSC" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="MSC" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="chondrocyte" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="osteoblast" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="iPSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="MSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="chondrocyte" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(hc.full.sppf@meta.data, orig.ident=="osteoblast" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed"))), ncol=4)

#notes: h.full.c.full.sppf dataset - most high mito counts are in human iPSC, human MSC, and human chondrocyte cells
CombinePlots(plots=list((ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="iPSC" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=25, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="MSC" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=12, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="chondrocyte" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=12, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="osteoblast" & species=="Human"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=12, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="iPSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="MSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="chondrocyte" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed")),
                        (ggplot(subset(h.full.c.full.sppf@meta.data, orig.ident=="osteoblast" & species=="Chimp"),
                                aes(x=nCount_RNA, y=percent.mt, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,100)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=7.5, linetype="dashed"))), ncol=4)

#notes: no filtering on percent mito at this time


#(2) Filter based on number of unique genes detected in each cell

#notes: all datasets can be filtered using this criteria
CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.orth.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.full.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.spof.tot")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.sppo")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppo.tot")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.full.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.sppf")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppf.tot")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed"))), ncol=7)

#notes: h.orfl.c.orfl.spof dataset - high feature counts across human osteoblasts and human chondrocytes
CombinePlots(plots=list((ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="iPSC" & species=="Human"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="MSC" & species=="Human"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="chondrocyte" & species=="Human"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="osteoblast" & species=="Human"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="iPSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="MSC" & species=="Chimp"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="chondrocyte" & species=="Chimp"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed")),
                        (ggplot(subset(h.orfl.c.orfl.spof@meta.data, orig.ident=="osteoblast" & species=="Chimp"),
                                aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                           geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")+
                           geom_hline(yintercept=c(200,4000), linetype="dashed"))), ncol=4)

#notes: no filtering on feature counts at this time

##notes: just exclude extremely low feature counts for right now
##notes: this only removes features from the hc.orth.sppo.tot, hc.orth.sppo, hc.orth.sppf, and h.orth.c.orth.sppf datasets
#hc.orfl.spof.tot <- subset(hc.orfl.spof.tot, subset=nFeature_RNA>200)
#hc.orth.sppo.tot <- subset(hc.orth.sppo.tot, subset=nFeature_RNA>200)
#hc.full.sppf.tot <- subset(hc.full.sppf.tot, subset=nFeature_RNA>200)
#hc.orfl.spof <- subset(hc.orfl.spof, subset=nFeature_RNA>200)
#hc.orfl.sppo <- subset(hc.orfl.sppo, subset=nFeature_RNA>200)
#hc.orfl.sppf <- subset(hc.orfl.sppf, subset=nFeature_RNA>200)
#hc.orth.spof <- subset(hc.orth.spof, subset=nFeature_RNA>200)
#hc.orth.sppo <- subset(hc.orth.sppo, subset=nFeature_RNA>200)
#hc.orth.sppf <- subset(hc.orth.sppf, subset=nFeature_RNA>200)
#hc.full.spof <- subset(hc.full.spof, subset=nFeature_RNA>200)
#hc.full.sppo <- subset(hc.full.sppo, subset=nFeature_RNA>200)
#hc.full.sppf <- subset(hc.full.sppf, subset=nFeature_RNA>200)
#h.orfl.c.orfl.spof <- subset(h.orfl.c.orfl.spof, subset=nFeature_RNA>200)
#h.orfl.c.orfl.sppo <- subset(h.orfl.c.orfl.sppo, subset=nFeature_RNA>200)
#h.orfl.c.orfl.sppf <- subset(h.orfl.c.orfl.sppf, subset=nFeature_RNA>200)
#h.orth.c.orth.spof <- subset(h.orth.c.orth.spof, subset=nFeature_RNA>200)
#h.orth.c.orth.sppo <- subset(h.orth.c.orth.sppo, subset=nFeature_RNA>200)
#h.orth.c.orth.sppf <- subset(h.orth.c.orth.sppf, subset=nFeature_RNA>200)
#h.full.c.full.spof <- subset(h.full.c.full.spof, subset=nFeature_RNA>200)
#h.full.c.full.sppo <- subset(h.full.c.full.sppo, subset=nFeature_RNA>200)
#h.full.c.full.sppf <- subset(h.full.c.full.sppf, subset=nFeature_RNA>200)

```

```{r data qc}

# Visualize QC metrics

CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.spof")+ylim(0,120000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.spof")+ylim(0,120000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.spof")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.spof")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.spof")+ylim(0,120000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.spof")+ylim(0,120000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.spof.tot")+ylim(0,120000)),
                        
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppo")+ylim(0,120000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppo")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.sppo")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.sppo")+ylim(0,120000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppo.tot")+ylim(0,120000)),
                        
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.sppf")+ylim(0,120000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppf")+ylim(0,120000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,120000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.sppf")+ylim(0,120000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.sppf")+ylim(0,120000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species,y=nCount_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppf.tot")+ylim(0,120000))), ncol=7)

CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.spof")+ylim(0,10000)),
                        (ggplot(hc.orth.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.spof")+ylim(0,10000)),
                        (ggplot(hc.full.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.spof")+ylim(0,10000)),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.spof")+ylim(0,10000)),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.spof")+ylim(0,10000)),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.spof")+ylim(0,10000)),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.spof.tot")+ylim(0,10000)),
                        
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.sppo")+ylim(0,10000)),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppo")+ylim(0,10000)),
                        (ggplot(hc.full.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppo")+ylim(0,10000)),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.sppo")+ylim(0,10000)),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.sppo")+ylim(0,10000)),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.sppo")+ylim(0,10000)),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppo.tot")+ylim(0,10000)),
                        
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orfl.sppf")+ylim(0,10000)),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.orth.sppf")+ylim(0,10000)),
                        (ggplot(hc.full.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppf")+ylim(0,10000)),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orfl.c.orfl.sppf")+ylim(0,10000)),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.orth.c.orth.sppf")+ylim(0,10000)),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="h.full.c.full.sppf")+ylim(0,10000)),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=species,y=nFeature_RNA,fill=orig.ident))+
                           geom_violin()+labs(title="hc.full.sppf.tot")+ylim(0,10000))), ncol=7)

CombinePlots(plots=list((ggplot(hc.orfl.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.spof")),
                        (ggplot(hc.orth.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.spof")),
                        (ggplot(hc.full.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.spof")),
                        (ggplot(h.orfl.c.orfl.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.spof")),
                        (ggplot(h.orth.c.orth.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.spof")),
                        (ggplot(h.full.c.full.spof@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.spof")),
                        (ggplot(hc.orfl.spof.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.spof.tot")),
                        
                        (ggplot(hc.orfl.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.sppo")),
                        (ggplot(hc.orth.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppo")),
                        (ggplot(hc.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppo")),
                        (ggplot(h.orfl.c.orfl.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orfl.c.orfl.sppo")),
                        (ggplot(h.orth.c.orth.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.sppo")),
                        (ggplot(h.full.c.full.sppo@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.sppo")),
                        (ggplot(hc.orth.sppo.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppo.tot")),
                        
                        (ggplot(hc.orfl.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orfl.sppf")),
                        (ggplot(hc.orth.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.orth.sppf")),
                        (ggplot(hc.full.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppf")),
                        (ggplot(h.orfl.c.orfl.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.sppf")),
                        (ggplot(h.orth.c.orth.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.orth.c.orth.sppf")),
                        (ggplot(h.full.c.full.sppf@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="h.full.c.full.sppf")),
                        (ggplot(hc.full.sppf.tot@meta.data, aes(x=nCount_RNA, y=nFeature_RNA, color=orig.ident, shape=species))+
                          geom_point(size=2,alpha=0.3)+xlim(0,120000)+ylim(0,10000)+labs(title="hc.full.sppf.tot"))), ncol=7)

```

```{r load processed data}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
hc.orfl.spof.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.tot.rds"))
hc.orth.sppo.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.tot.rds"))
hc.full.sppf.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.tot.rds"))
hc.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.rds"))
hc.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppo.rds"))
hc.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppf.rds"))
hc.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.spof.rds"))
hc.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.rds"))
hc.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppf.rds"))
hc.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.spof.rds"))
hc.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppo.rds"))
hc.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.rds"))
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
h.orfl.c.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppo.rds"))
h.orfl.c.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppf.rds"))
h.orth.c.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.spof.rds"))
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
h.orth.c.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppf.rds"))
h.full.c.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.spof.rds"))
h.full.c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
h.full.c.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppf.rds"))

```

```{r normalize data}

#Normalize data using default settings
hc.orfl.spof.tot <- NormalizeData(hc.orfl.spof.tot, normalization.method="LogNormalize", scale.factor=10000)
hc.orth.sppo.tot <- NormalizeData(hc.orth.sppo.tot, normalization.method="LogNormalize", scale.factor=10000)
hc.full.sppf.tot <- NormalizeData(hc.full.sppf.tot, normalization.method="LogNormalize", scale.factor=10000)
hc.orfl.spof <- NormalizeData(hc.orfl.spof, normalization.method="LogNormalize", scale.factor=10000)
hc.orfl.sppo <- NormalizeData(hc.orfl.sppo, normalization.method="LogNormalize", scale.factor=10000)
hc.orfl.sppf <- NormalizeData(hc.orfl.sppf, normalization.method="LogNormalize", scale.factor=10000)
hc.orth.spof <- NormalizeData(hc.orth.spof, normalization.method="LogNormalize", scale.factor=10000)
hc.orth.sppo <- NormalizeData(hc.orth.sppo, normalization.method="LogNormalize", scale.factor=10000)
hc.orth.sppf <- NormalizeData(hc.orth.sppf, normalization.method="LogNormalize", scale.factor=10000)
hc.full.spof <- NormalizeData(hc.full.spof, normalization.method="LogNormalize", scale.factor=10000)
hc.full.sppo <- NormalizeData(hc.full.sppo, normalization.method="LogNormalize", scale.factor=10000)
hc.full.sppf <- NormalizeData(hc.full.sppf, normalization.method="LogNormalize", scale.factor=10000)
h.orfl.c.orfl.spof <- NormalizeData(h.orfl.c.orfl.spof, normalization.method="LogNormalize", scale.factor=10000)
h.orfl.c.orfl.sppo <- NormalizeData(h.orfl.c.orfl.sppo, normalization.method="LogNormalize", scale.factor=10000)
h.orfl.c.orfl.sppf <- NormalizeData(h.orfl.c.orfl.sppf, normalization.method="LogNormalize", scale.factor=10000)
h.orth.c.orth.spof <- NormalizeData(h.orth.c.orth.spof, normalization.method="LogNormalize", scale.factor=10000)
h.orth.c.orth.sppo <- NormalizeData(h.orth.c.orth.sppo, normalization.method="LogNormalize", scale.factor=10000)
h.orth.c.orth.sppf <- NormalizeData(h.orth.c.orth.sppf, normalization.method="LogNormalize", scale.factor=10000)
h.full.c.full.spof <- NormalizeData(h.full.c.full.spof, normalization.method="LogNormalize", scale.factor=10000)
h.full.c.full.sppo <- NormalizeData(h.full.c.full.sppo, normalization.method="LogNormalize", scale.factor=10000)
h.full.c.full.sppf <- NormalizeData(h.full.c.full.sppf, normalization.method="LogNormalize", scale.factor=10000)

```

```{r highly variable feature selection}

#Calculate subset of features that exhibit high cell-to-cell variation in dataset
hc.orfl.spof.tot <- FindVariableFeatures(hc.orfl.spof.tot, selection.method="vst", nfeatures=2000)
hc.orth.sppo.tot <- FindVariableFeatures(hc.orth.sppo.tot, selection.method="vst", nfeatures=2000)
hc.full.sppf.tot <- FindVariableFeatures(hc.full.sppf.tot, selection.method="vst", nfeatures=2000)
hc.orfl.spof <- FindVariableFeatures(hc.orfl.spof, selection.method="vst", nfeatures=2000)
hc.orfl.sppo <- FindVariableFeatures(hc.orfl.sppo, selection.method="vst", nfeatures=2000)
hc.orfl.sppf <- FindVariableFeatures(hc.orfl.sppf, selection.method="vst", nfeatures=2000)
hc.orth.spof <- FindVariableFeatures(hc.orth.spof, selection.method="vst", nfeatures=2000)
hc.orth.sppo <- FindVariableFeatures(hc.orth.sppo, selection.method="vst", nfeatures=2000)
hc.orth.sppf <- FindVariableFeatures(hc.orth.sppf, selection.method="vst", nfeatures=2000)
hc.full.spof <- FindVariableFeatures(hc.full.spof, selection.method="vst", nfeatures=2000)
hc.full.sppo <- FindVariableFeatures(hc.full.sppo, selection.method="vst", nfeatures=2000)
hc.full.sppf <- FindVariableFeatures(hc.full.sppf, selection.method="vst", nfeatures=2000)
h.orfl.c.orfl.spof <- FindVariableFeatures(h.orfl.c.orfl.spof, selection.method="vst", nfeatures=2000)
h.orfl.c.orfl.sppo <- FindVariableFeatures(h.orfl.c.orfl.sppo, selection.method="vst", nfeatures=2000)
h.orfl.c.orfl.sppf <- FindVariableFeatures(h.orfl.c.orfl.sppf, selection.method="vst", nfeatures=2000)
h.orth.c.orth.spof <- FindVariableFeatures(h.orth.c.orth.spof, selection.method="vst", nfeatures=2000)
h.orth.c.orth.sppo <- FindVariableFeatures(h.orth.c.orth.sppo, selection.method="vst", nfeatures=2000)
h.orth.c.orth.sppf <- FindVariableFeatures(h.orth.c.orth.sppf, selection.method="vst", nfeatures=2000)
h.full.c.full.spof <- FindVariableFeatures(h.full.c.full.spof, selection.method="vst", nfeatures=2000)
h.full.c.full.sppo <- FindVariableFeatures(h.full.c.full.sppo, selection.method="vst", nfeatures=2000)
h.full.c.full.sppf <- FindVariableFeatures(h.full.c.full.sppf, selection.method="vst", nfeatures=2000)


# Identify 10 most highly variable genes
top10.hc.orfl.spof.tot <- head(VariableFeatures(hc.orfl.spof.tot), 10)
top10.hc.orth.sppo.tot <- head(VariableFeatures(hc.orth.sppo.tot), 10)
top10.hc.full.sppf.tot <- head(VariableFeatures(hc.full.sppf.tot), 10)
top10.hc.orfl.spof <- head(VariableFeatures(hc.orfl.spof), 10)
top10.hc.orfl.sppo <- head(VariableFeatures(hc.orfl.sppo), 10)
top10.hc.orfl.sppf <- head(VariableFeatures(hc.orfl.sppf), 10)
top10.hc.orth.spof <- head(VariableFeatures(hc.orth.spof), 10)
top10.hc.orth.sppo <- head(VariableFeatures(hc.orth.sppo), 10)
top10.hc.orth.sppf <- head(VariableFeatures(hc.orth.sppf), 10)
top10.hc.full.spof <- head(VariableFeatures(hc.full.spof), 10)
top10.hc.full.sppo <- head(VariableFeatures(hc.full.sppo), 10)
top10.hc.full.sppf <- head(VariableFeatures(hc.full.sppf), 10)
top10.h.orfl.c.orfl.spof <- head(VariableFeatures(h.orfl.c.orfl.spof), 10)
top10.h.orfl.c.orfl.sppo <- head(VariableFeatures(h.orfl.c.orfl.sppo), 10)
top10.h.orfl.c.orfl.sppf <- head(VariableFeatures(h.orfl.c.orfl.sppf), 10)
top10.h.orth.c.orth.spof <- head(VariableFeatures(h.orth.c.orth.spof), 10)
top10.h.orth.c.orth.sppo <- head(VariableFeatures(h.orth.c.orth.sppo), 10)
top10.h.orth.c.orth.sppf <- head(VariableFeatures(h.orth.c.orth.sppf), 10)
top10.h.full.c.full.spof <- head(VariableFeatures(h.full.c.full.spof), 10)
top10.h.full.c.full.sppo <- head(VariableFeatures(h.full.c.full.sppo), 10)
top10.h.full.c.full.sppf <- head(VariableFeatures(h.full.c.full.sppf), 10)

##NO LONGER NECESSARY
##Determine what the top 10 variable genes are for the orth datasets
#dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
#setwd(paste0(dir,"/analysis"))
#h.gtf <- read.delim(file="../../../references/human-chimp-ortho-exons/Human_orthoexon_extended_info.txt")
#c.gtf <- read.delim(file="../../../references/human-chimp-ortho-exons/Chimp_orthoexon_extended_info_edit.txt")
#h.gtf <- unique(h.gtf[,c(1,3,4)])
#c.gtf <- unique(c.gtf[,c(1,3,4)])
#gene=c()
#gene2=c()
#for (i in hc.orth.sppo.tot@assays$RNA@var.features) {
#  if (grepl("GRCh38------",i)) { gene=c(gene,strsplit(i,"GRCh38------")[[1]][2]) }
#  else { gene=c(gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
#   }
#for (i in gene) { gene2=c(gene2,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene2], 10)
#gene=c()
#for (i in hc.orth.spof@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)
#gene=c()
#for (i in hc.orth.sppo@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)
#gene=c()
#for (i in hc.orth.sppf@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)
#gene=c()
#for (i in h.orth.c.orth.spof@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)
#gene=c()
#for (i in h.orth.c.orth.sppo@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)
#gene=c()
#for (i in h.orth.c.orth.sppf@assays$RNA@var.features) { gene=c(gene,which(h.gtf$ensg %in% i)) }
#head(h.gtf$gene_name[gene], 10)

#FULL DATA
#top10.hc.orfl.spof.tot: ortho-filter-Pan-tro-3.0-MGP, ortho-filter-GRCh38------MGP, ortho-filter-GRCh38------CTHRC1, ortho-filter-Pan-tro-3.0-RP11-865I6.2,
#                        ortho-filter-Pan-tro-3.0-IGFBP5, ortho-filter-GRCh38------TTR, ortho-filter-GRCh38------CSRP3, ortho-filter-Pan-tro-3.0-MT2A,
#                        ortho-filter-Pan-tro-3.0-MMP1, ortho-filter-Pan-tro-3.0-CXCL1
#top10.hc.orth.sppo.tot: ortho-Pan-tro-3.0-MGP, ortho-GRCh38------MGP, ortho-GRCh38------CTHRC1, ortho-Pan-tro-3.0-RP11-865I6.2, ortho-Pan-tro-3.0-IGFBP5,
#                        ortho-GRCh38------TTR, ortho-Pan-tro-3.0-MT1L, ortho-GRCh38------CSRP3, ortho-Pan-tro-3.0-MT2A, ortho-Pan-tro-3.0-MMP1
#top10.hc.full.sppf.tot: Pan-tro-3.0-MGP, GRCh38------MGP, GRCh38------CTHRC1, Pan-tro-3.0-IGFBP5, Pan-tro-3.0-PF4V1, GRCh38------TTR, GRCh38------CSRP3,
#                        Pan-tro-3.0-MT2A, Pan-tro-3.0-MMP1, Pan-tro-3.0-CXCL1
#top10.hc.orfl.spof: MGP, CTHRC1, TTR, SPON2, RP11-865I6.2, KISS1, PTGDS, CSRP3, CXCL1, NPPB
#top10.hc.orfl.sppo: MGP, CTHRC1, TTR, RP11-865I6.2, SPON2, KISS1, PTGDS, CSRP3, CXCL1, NPPB
#top10.hc.orfl.sppf: MGP, TTR, SCG2, CSRP3, CTHRC1, KISS1, RP11-865I6.2, NPPB, CYTL1, ANGPTL4
#top10.hc.orth.spof: MGP, CTHRC1, TTR, RP11-865I6.2 SPON2, KISS1, PTGDS, CSRP3, CXCL1, NPPB
#top10.hc.orth.sppo: MGP, CTHRC1, TTR, RP11-865I6.2 SPON2, KISS1, PTGDS, CSRP3, CXCL1, NPPB
#top10.hc.orth.sppf: MGP, TTR, SCG2, CSRP3, CTHRC1, KISS1, RP11-865I6.2 NPPB, CYTL1, ANGPTL4
#top10.hc.full.spof: MGP, CTHRC1, PF4V1, TTR, SCG2, SPON2, TIMP3, KISS1, CSRP3, CXCL1
#top10.hc.full.sppo: MGP, CTHRC1, PF4V1, TTR, SCG2, SPON2, TIMP3, KISS1, CSRP3, CXCL1
#top10.hc.full.sppf: MGP, SCG2, TTR, CSRP3, CTHRC1, KISS1, NPPB, CYTL1, ANGPTL4, DLK1
#top10.h.orfl.c.orfl.spof: MGP, CTHRC1, RP11-865I6.2, CXCL1, CCL2, MMP1, PTGDS, SCG2, TTR, SPON2
#top10.h.orfl.c.orfl.sppo: MGP, CTHRC1, RP11-865I6.2, CXCL1, MMP1, PTGDS, CCL2, SCG2, TTR, SPON2
#top10.h.orfl.c.orfl.sppf: SCG2, MGP, RP11-865I6.2, TTR, NPPB, CTHRC1, KISS1, CSRP3, DLK1, MMP1
#top10.h.full.c.full.spof: MGP, CTHRC1, TTR, SCG2, PF4V1, TIMP3, CXCL1, MMP1, NPPB, CCL2
#top10.h.full.c.full.sppo: MGP, CTHRC1, TTR, SCG2, PF4V1, TIMP3, CXCL1, MMP1, NPPB, CCL2
#top10.h.full.c.full.sppf: MGP, SCG2, TTR, NPPB, CSRP3, CTHRC1, KISS1, DLK1, MMP1, CYTL1

#SUBSET DATA
#top10.hc.orth.sppo.tot: ortho-Pan-tro-3.0-MGP, ortho-GRCh38------MGP, ortho-GRCh38------CTHRC1, ortho-Pan-tro-3.0-IGFBP5,
#                        ortho-Pan-tro-3.0-MT2A, ortho-Pan-tro-3.0-RP11-865I6.2, ortho-GRCh38------TTR, ortho-Pan-tro-3.0-MT1L,
#                        ortho-GRCh38------NPPB, ortho-Pan-tro-3.0-SPON2
#top10.hc.full.sppf.tot: Pan-tro-3.0-MGP, GRCh38------MGP, GRCh38------CTHRC1, Pan-tro-3.0-IGFBP5, Pan-tro-3.0-MT2A, Pan-tro-3.0-PF4V1,
#                        GRCh38------TTR, GRCh38------PTGDS, GRCh38------TIMP3, Pan-tro-3.0-MMP1
#top10.hc.orth.sppo: MGP, CTHRC1, SPON2, TTR, RP11-865I6.2, KISS1, PTGDS, NPPB, S100A4, IGFBP5
#top10.hc.orth.sppf: MGP, CTHRC1, TTR, NPPB, KISS1, S100A4, SCG2, PTGDS, RP11-865I6.2, CYTL1
#top10.hc.full.sppo: MGP, CTHRC1, PF4V1, TIMP3, SPON2, TTR, SCG2, KISS1, IGFBP5, NPPB
#top10.hc.full.sppf: MGP, CTHRC1, TTR, PF4V1, TIMP3, SCG2, S100A4, NPPB, KISS1, IGFBP5
#top10.h.orth.c.orth.sppo: MGP, CTHRC1, RP11-865I6.2, MMP1, CXCL1, CCL2, PTGDS, SPON2, NPPB, SCG2
#top10.h.orth.c.orth.sppf: MGP, CTHRC1, RP11-865I6.2, MMP1, CXCL1, CCL2, PTGDS, SPON2, NPPB, SCG2
#top10.h.full.c.full.sppo: MGP, CTHRC1, TIMP3, MMP1, TTR, NPPB, PF4V1, SCG2, CXCL1, SPON2
#top10.h.full.c.full.sppf: MGP, CTHRC1, TIMP3, MMP1, TTR, NPPB, PF4V1, SCG2, CXCL1, SPON2

#Plot variable features with top 10 features labels
LabelPoints(plot=VariableFeaturePlot(hc.orfl.spof),points=top10.hc.orfl.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orfl.sppf),points=top10.hc.orfl.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orth.spof),points=top10.hc.orth.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orfl.c.orfl.spof),points=top10.h.orfl.c.orfl.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orth.c.orth.spof),points=top10.h.orth.c.orth.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.full.c.full.spof),points=top10.h.full.c.full.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orfl.spof.tot),points=top10.hc.orfl.spof.tot, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orfl.sppo),points=top10.hc.orfl.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orth.sppo),points=top10.hc.orth.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.full.sppo),points=top10.hc.full.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orfl.c.orfl.sppo),points=top10.h.orfl.c.orfl.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orth.c.orth.sppo),points=top10.h.orth.c.orth.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.full.c.full.sppo),points=top10.h.full.c.full.sppo, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orth.sppo.tot),points=top10.hc.orth.sppo.tot, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.full.spof),points=top10.hc.full.spof, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.orth.sppf),points=top10.hc.orth.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.full.sppf),points=top10.hc.full.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orfl.c.orfl.sppf),points=top10.h.orfl.c.orfl.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.orth.c.orth.sppf),points=top10.h.orth.c.orth.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(h.full.c.full.sppf),points=top10.h.full.c.full.sppf, repel=TRUE)
LabelPoints(plot=VariableFeaturePlot(hc.full.sppf.tot),points=top10.hc.full.sppf.tot, repel=TRUE)

```

```{r scale data}

#Scale data with linear transformation such that:
#(1) expression of each gene is shifted so that mean expression across cells is 0
#(2) expression of each gene is scaled so that variance across cells is 1
#results of this are stored in pbmc[["RNA"]]@scale.data

hc.orfl.spof.tot <- ScaleData(hc.orfl.spof.tot, features=rownames(hc.orfl.spof.tot))
hc.orth.sppo.tot <- ScaleData(hc.orth.sppo.tot, features=rownames(hc.orth.sppo.tot))
hc.full.sppf.tot <- ScaleData(hc.full.sppf.tot, features=rownames(hc.full.sppf.tot))
hc.orfl.spof <- ScaleData(hc.orfl.spof, features=rownames(hc.orfl.spof))
hc.orfl.sppo <- ScaleData(hc.orfl.sppo, features=rownames(hc.orfl.sppo))
hc.orfl.sppf <- ScaleData(hc.orfl.sppf, features=rownames(hc.orfl.sppf))
hc.orth.spof <- ScaleData(hc.orth.spof, features=rownames(hc.orth.spof))
hc.orth.sppo <- ScaleData(hc.orth.sppo, features=rownames(hc.orth.sppo))
hc.orth.sppf <- ScaleData(hc.orth.sppf, features=rownames(hc.orth.sppf))
hc.full.spof <- ScaleData(hc.full.spof, features=rownames(hc.full.spof))
hc.full.sppo <- ScaleData(hc.full.sppo, features=rownames(hc.full.sppo))
hc.full.sppf <- ScaleData(hc.full.sppf, features=rownames(hc.full.sppf))
h.orfl.c.orfl.spof <- ScaleData(h.orfl.c.orfl.spof, features=rownames(h.orfl.c.orfl.spof))
h.orfl.c.orfl.sppo <- ScaleData(h.orfl.c.orfl.sppo, features=rownames(h.orfl.c.orfl.sppo))
h.orfl.c.orfl.sppf <- ScaleData(h.orfl.c.orfl.sppf, features=rownames(h.orfl.c.orfl.sppo))
h.orth.c.orth.spof <- ScaleData(h.orth.c.orth.spof, features=rownames(h.orth.c.orth.spof))
h.orth.c.orth.sppo <- ScaleData(h.orth.c.orth.sppo, features=rownames(h.orth.c.orth.sppo))
h.orth.c.orth.sppf <- ScaleData(h.orth.c.orth.sppf, features=rownames(h.orth.c.orth.sppf))
h.full.c.full.spof <- ScaleData(h.full.c.full.spof, features=rownames(h.full.c.full.spof))
h.full.c.full.sppo <- ScaleData(h.full.c.full.sppo, features=rownames(h.full.c.full.sppo))
h.full.c.full.sppf <- ScaleData(h.full.c.full.sppf, features=rownames(h.full.c.full.sppf))

```

```{r linear dimensional reduction}

#Perform PCA on scaled data
hc.orfl.spof.tot <- RunPCA(hc.orfl.spof.tot, features=VariableFeatures(object=hc.orfl.spof.tot), verbose=FALSE)
hc.orth.sppo.tot <- RunPCA(hc.orth.sppo.tot, features=VariableFeatures(object=hc.orth.sppo.tot), verbose=FALSE)
hc.full.sppf.tot <- RunPCA(hc.full.sppf.tot, features=VariableFeatures(object=hc.full.sppf.tot), verbose=FALSE)
hc.orfl.spof <- RunPCA(hc.orfl.spof, features=VariableFeatures(object=hc.orfl.spof), verbose=FALSE)
hc.orfl.sppo <- RunPCA(hc.orfl.sppo, features=VariableFeatures(object=hc.orfl.sppo), verbose=FALSE)
hc.orfl.sppf <- RunPCA(hc.orfl.sppf, features=VariableFeatures(object=hc.orfl.sppf), verbose=FALSE)
hc.orth.spof <- RunPCA(hc.orth.spof, features=VariableFeatures(object=hc.orth.spof), verbose=FALSE)
hc.orth.sppo <- RunPCA(hc.orth.sppo, features=VariableFeatures(object=hc.orth.sppo), verbose=FALSE)
hc.orth.sppf <- RunPCA(hc.orth.sppf, features=VariableFeatures(object=hc.orth.sppf), verbose=FALSE)
hc.full.spof <- RunPCA(hc.full.spof, features=VariableFeatures(object=hc.full.spof), verbose=FALSE)
hc.full.sppo <- RunPCA(hc.full.sppo, features=VariableFeatures(object=hc.full.sppo), verbose=FALSE)
hc.full.sppf <- RunPCA(hc.full.sppf, features=VariableFeatures(object=hc.full.sppf), verbose=FALSE)
h.orfl.c.orfl.spof <- RunPCA(h.orfl.c.orfl.spof, features=VariableFeatures(object=h.orfl.c.orfl.spof), verbose=FALSE)
h.orfl.c.orfl.sppo <- RunPCA(h.orfl.c.orfl.sppo, features=VariableFeatures(object=h.orfl.c.orfl.sppo), verbose=FALSE)
h.orfl.c.orfl.sppf <- RunPCA(h.orfl.c.orfl.sppf, features=VariableFeatures(object=h.orfl.c.orfl.sppf), verbose=FALSE)
h.orth.c.orth.spof <- RunPCA(h.orth.c.orth.spof, features=VariableFeatures(object=h.orth.c.orth.spof), verbose=FALSE)
h.orth.c.orth.sppo <- RunPCA(h.orth.c.orth.sppo, features=VariableFeatures(object=h.orth.c.orth.sppo), verbose=FALSE)
h.orth.c.orth.sppf <- RunPCA(h.orth.c.orth.sppf, features=VariableFeatures(object=h.orth.c.orth.sppf), verbose=FALSE)
h.full.c.full.spof <- RunPCA(h.full.c.full.spof, features=VariableFeatures(object=h.full.c.full.spof), verbose=FALSE)
h.full.c.full.sppo <- RunPCA(h.full.c.full.sppo, features=VariableFeatures(object=h.full.c.full.sppo), verbose=FALSE)
h.full.c.full.sppf <- RunPCA(h.full.c.full.sppf, features=VariableFeatures(object=h.full.c.full.sppf), verbose=FALSE)

#Determine dimensionality of dataset
#seurat clusters cells using PCA scores - each PC represents a ‘metafeature’ that combines information across a correlated feature set, the top PCs represent a robust compression of the dataset - need to determine how many PCs to include

#OPTION 1 - Exploring PCs to determine relevant sources of heterogeneity

#Examine PCA results
print(hc.orfl.spof.tot[["pca"]], dims=1:5, nfeatures=5)
print(hc.orth.sppo.tot[["pca"]], dims=1:5, nfeatures=5)
print(hc.full.sppf.tot[["pca"]], dims=1:5, nfeatures=5)
print(hc.orfl.spof[["pca"]], dims=1:5, nfeatures=5)
print(hc.orfl.sppo[["pca"]], dims=1:5, nfeatures=5)
print(hc.orfl.sppf[["pca"]], dims=1:5, nfeatures=5)
print(hc.orth.spof[["pca"]], dims=1:5, nfeatures=5)
print(hc.orth.sppo[["pca"]], dims=1:5, nfeatures=5)
print(hc.orth.sppf[["pca"]], dims=1:5, nfeatures=5)
print(hc.full.spof[["pca"]], dims=1:5, nfeatures=5)
print(hc.full.sppo[["pca"]], dims=1:5, nfeatures=5)
print(hc.full.sppf[["pca"]], dims=1:5, nfeatures=5)
print(h.orfl.c.orfl.spof[["pca"]], dims=1:5, nfeatures=5)
print(h.orfl.c.orfl.sppo[["pca"]], dims=1:5, nfeatures=5)
print(h.orfl.c.orfl.sppf[["pca"]], dims=1:5, nfeatures=5)
print(h.orth.c.orth.spof[["pca"]], dims=1:5, nfeatures=5)
print(h.orth.c.orth.sppo[["pca"]], dims=1:5, nfeatures=5)
print(h.orth.c.orth.sppf[["pca"]], dims=1:5, nfeatures=5)
print(h.full.c.full.spof[["pca"]], dims=1:5, nfeatures=5)
print(h.full.c.full.sppo[["pca"]], dims=1:5, nfeatures=5)
print(h.full.c.full.sppf[["pca"]], dims=1:5, nfeatures=5)

#Visualize PCA results
VizDimLoadings(hc.orfl.spof.tot, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orth.sppo.tot, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.full.sppf.tot, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orfl.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orfl.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orfl.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orth.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orth.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.orth.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.full.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.full.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(hc.full.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orfl.c.orfl.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orfl.c.orfl.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orfl.c.orfl.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orth.c.orth.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orth.c.orth.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.orth.c.orth.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.full.c.full.spof, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.full.c.full.sppo, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)
VizDimLoadings(h.full.c.full.sppf, dims=1:2, nfeatures=30, reduction="pca", balanced=TRUE)

DimPlot(hc.orfl.spof.tot, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orth.sppo.tot, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.full.sppf.tot, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orfl.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orfl.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orfl.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orth.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orth.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.orth.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.full.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.full.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(hc.full.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orfl.c.orfl.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orfl.c.orfl.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orfl.c.orfl.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orth.c.orth.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orth.c.orth.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.orth.c.orth.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.full.c.full.spof, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.full.c.full.sppo, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)
DimPlot(h.full.c.full.sppf, reduction="pca", group.by="orig.ident", label=TRUE, repel=TRUE, label.size=5)

#pca plot for top 500 cells on pc1
DimHeatmap(hc.orfl.spof.tot, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppo.tot, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppf.tot, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.sppf, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppf, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.full.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppf, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.sppf, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.sppf, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.spof, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.sppo, dims=1, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.sppf, dims=1, cells=500, balanced=TRUE)

#pca plot for top 500 cells on pc1 through pc9
DimHeatmap(hc.orfl.spof.tot, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppo.tot, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppf.tot, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orfl.sppf, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.orth.sppf, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.full.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(hc.full.sppf, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orfl.c.orfl.sppf, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.orth.c.orth.sppf, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.spof, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.sppo, dims=1:9, cells=500, balanced=TRUE)
DimHeatmap(h.full.c.full.sppf, dims=1:9, cells=500, balanced=TRUE)

#OPTION 2- JackStraw procedure: randomly permute subset of data (1% by default), rerun PCA constructing a null distribution of feature scores, repeat this procedure, identify significant PCs as those with strong enrichment of low p-value features
hc.orfl.spof.tot <- JackStraw(hc.orfl.spof.tot, num.replicate=100)
hc.orth.sppo.tot <- JackStraw(hc.orth.sppo.tot, num.replicate=100)
hc.full.sppf.tot <- JackStraw(hc.full.sppf.tot, num.replicate=100)
hc.orfl.spof <- JackStraw(hc.orfl.spof, num.replicate=100)
hc.orfl.sppo <- JackStraw(hc.orfl.sppo, num.replicate=100)
hc.orfl.sppf <- JackStraw(hc.orfl.sppf, num.replicate=100)
hc.orth.spof <- JackStraw(hc.orth.spof, num.replicate=100)
hc.orth.sppo <- JackStraw(hc.orth.sppo, num.replicate=100)
hc.orth.sppf <- JackStraw(hc.orth.sppf, num.replicate=100)
hc.full.spof <- JackStraw(hc.full.spof, num.replicate=100)
hc.full.sppo <- JackStraw(hc.full.sppo, num.replicate=100)
hc.full.sppf <- JackStraw(hc.full.sppf, num.replicate=100)
h.orfl.c.orfl.spof <- JackStraw(h.orfl.c.orfl.spof, num.replicate=100)
h.orfl.c.orfl.sppo <- JackStraw(h.orfl.c.orfl.sppo, num.replicate=100)
h.orfl.c.orfl.sppf <- JackStraw(h.orfl.c.orfl.sppf, num.replicate=100)
h.orth.c.orth.spof <- JackStraw(h.orth.c.orth.spof, num.replicate=100)
h.orth.c.orth.sppo <- JackStraw(h.orth.c.orth.sppo, num.replicate=100)
h.orth.c.orth.sppf <- JackStraw(h.orth.c.orth.sppf, num.replicate=100)
h.full.c.full.spof <- JackStraw(h.full.c.full.spof, num.replicate=100)
h.full.c.full.sppo <- JackStraw(h.full.c.full.sppo, num.replicate=100)
h.full.c.full.sppf <- JackStraw(h.full.c.full.sppf, num.replicate=100)

hc.orfl.spof.tot <- ScoreJackStraw(hc.orfl.spof.tot, dims=1:20)
hc.orth.sppo.tot <- ScoreJackStraw(hc.orth.sppo.tot, dims=1:20)
hc.full.sppf.tot <- ScoreJackStraw(hc.full.sppf.tot, dims=1:20)
hc.orfl.spof <- ScoreJackStraw(hc.orfl.spof, dims=1:20)
hc.orfl.sppo <- ScoreJackStraw(hc.orfl.sppo, dims=1:20)
hc.orfl.sppf <- ScoreJackStraw(hc.orfl.sppf, dims=1:20)
hc.orth.spof <- ScoreJackStraw(hc.orth.spof, dims=1:20)
hc.orth.sppo <- ScoreJackStraw(hc.orth.sppo, dims=1:20)
hc.orth.sppf <- ScoreJackStraw(hc.orth.sppf, dims=1:20)
hc.full.spof <- ScoreJackStraw(hc.full.spof, dims=1:20)
hc.full.sppo <- ScoreJackStraw(hc.full.sppo, dims=1:20)
hc.full.sppf <- ScoreJackStraw(hc.full.sppf, dims=1:20)
h.orfl.c.orfl.spof <- ScoreJackStraw(h.orfl.c.orfl.spof, dims=1:20)
h.orfl.c.orfl.sppo <- ScoreJackStraw(h.orfl.c.orfl.sppo, dims=1:20)
h.orfl.c.orfl.sppf <- ScoreJackStraw(h.orfl.c.orfl.sppf, dims=1:20)
h.orth.c.orth.spof <- ScoreJackStraw(h.orth.c.orth.spof, dims=1:20)
h.orth.c.orth.sppo <- ScoreJackStraw(h.orth.c.orth.sppo, dims=1:20)
h.orth.c.orth.sppf <- ScoreJackStraw(h.orth.c.orth.sppf, dims=1:20)
h.full.c.full.spof <- ScoreJackStraw(h.full.c.full.spof, dims=1:20)
h.full.c.full.sppo <- ScoreJackStraw(h.full.c.full.sppo, dims=1:20)
h.full.c.full.sppf <- ScoreJackStraw(h.full.c.full.sppf, dims=1:20)

#visualize the distribution of p-values for each PC with a uniform distribution (dashed line)
#sign PCs show strong enrichment of features with low p-values (solid curve above dashed line)
JackStrawPlot(hc.orfl.spof.tot, dims=1:20)
JackStrawPlot(hc.orth.sppo.tot, dims=1:20)
JackStrawPlot(hc.full.sppf.tot, dims=1:20)
JackStrawPlot(hc.orfl.spof, dims=1:20)
JackStrawPlot(hc.orfl.sppo, dims=1:20)
JackStrawPlot(hc.orfl.sppf, dims=1:20)
JackStrawPlot(hc.orth.spof, dims=1:20)
JackStrawPlot(hc.orth.sppo, dims=1:20)
JackStrawPlot(hc.orth.sppf, dims=1:20)
JackStrawPlot(hc.full.spof, dims=1:20)
JackStrawPlot(hc.full.sppo, dims=1:20)
JackStrawPlot(hc.full.sppf, dims=1:20)
JackStrawPlot(h.orfl.c.orfl.spof, dims=1:20)
JackStrawPlot(h.orfl.c.orfl.sppo, dims=1:20)
JackStrawPlot(h.orfl.c.orfl.sppf, dims=1:20)
JackStrawPlot(h.orth.c.orth.spof, dims=1:20)
JackStrawPlot(h.orth.c.orth.sppo, dims=1:20)
JackStrawPlot(h.orth.c.orth.sppf, dims=1:20)
JackStrawPlot(h.full.c.full.spof, dims=1:20)
JackStrawPlot(h.full.c.full.sppo, dims=1:20)
JackStrawPlot(h.full.c.full.sppf, dims=1:20)

#OPTION 3 -Elbow plot procedure: ranking of PCs based on the percentage of variance explained by each one
ElbowPlot(hc.orfl.spof.tot)
ElbowPlot(hc.orth.sppo.tot)
ElbowPlot(hc.full.sppf.tot)
ElbowPlot(hc.orfl.spof)
ElbowPlot(hc.orfl.sppo)
ElbowPlot(hc.orfl.sppf)
ElbowPlot(hc.orth.spof)
ElbowPlot(hc.orth.sppo)
ElbowPlot(hc.orth.sppf)
ElbowPlot(hc.full.spof)
ElbowPlot(hc.full.sppo)
ElbowPlot(hc.full.sppf)
ElbowPlot(h.orfl.c.orfl.spof)
ElbowPlot(h.orfl.c.orfl.sppo)
ElbowPlot(h.orfl.c.orfl.sppf)
ElbowPlot(h.orth.c.orth.spof)
ElbowPlot(h.orth.c.orth.sppo)
ElbowPlot(h.orth.c.orth.sppf)
ElbowPlot(h.full.c.full.spof)
ElbowPlot(h.full.c.full.sppo)
ElbowPlot(h.full.c.full.sppf)

#lets retain the first 20PCs for all datasets

```

```{r clustering}

#Cluster cells using a distance metric based on previously identified PCs and partion cellular distance matrix into clusters by embedding cells in a graph structure (e.g. K-nearest neighbor graph) with edges drawn between cells with similar feature expression patterns and then dividing this graph into highly interconnected ‘quasi-cliques’/‘communities’

#(1)construct KNN graph based on euclidean distance in PCA space and refine edge weights between any two cells based on shared overlap in their local neighborhoods (Jaccard similarity). This step is performed using the FindNeighbors function, and takes as input the previously defined dimensionality of the dataset (first 20 PCs).
hc.orfl.spof.tot <- FindNeighbors(hc.orfl.spof.tot, dims=1:20)
hc.orth.sppo.tot <- FindNeighbors(hc.orth.sppo.tot, dims=1:20)
hc.full.sppf.tot <- FindNeighbors(hc.full.sppf.tot, dims=1:20)
hc.orfl.spof <- FindNeighbors(hc.orfl.spof, dims=1:20)
hc.orfl.sppo <- FindNeighbors(hc.orfl.sppo, dims=1:20)
hc.orfl.sppf <- FindNeighbors(hc.orfl.sppf, dims=1:20)
hc.orth.spof <- FindNeighbors(hc.orth.spof, dims=1:20)
hc.orth.sppo <- FindNeighbors(hc.orth.sppo, dims=1:20)
hc.orth.sppf <- FindNeighbors(hc.orth.sppf, dims=1:20)
hc.full.spof <- FindNeighbors(hc.full.spof, dims=1:20)
hc.full.sppo <- FindNeighbors(hc.full.sppo, dims=1:20)
hc.full.sppf <- FindNeighbors(hc.full.sppf, dims=1:20)
h.orfl.c.orfl.spof <- FindNeighbors(h.orfl.c.orfl.spof, dims=1:20)
h.orfl.c.orfl.sppo <- FindNeighbors(h.orfl.c.orfl.sppo, dims=1:20)
h.orfl.c.orfl.sppf <- FindNeighbors(h.orfl.c.orfl.sppf, dims=1:20)
h.orth.c.orth.spof <- FindNeighbors(h.orth.c.orth.spof, dims=1:20)
h.orth.c.orth.sppo <- FindNeighbors(h.orth.c.orth.sppo, dims=1:20)
h.orth.c.orth.sppf <- FindNeighbors(h.orth.c.orth.sppf, dims=1:20)
h.full.c.full.spof <- FindNeighbors(h.full.c.full.spof, dims=1:20)
h.full.c.full.sppo <- FindNeighbors(h.full.c.full.sppo, dims=1:20)
h.full.c.full.sppf <- FindNeighbors(h.full.c.full.sppf, dims=1:20)

#(2)apply modularity optimization techniques to iteratively group cells together
hc.orfl.spof.tot <- FindClusters(hc.orfl.spof.tot, resolution=0.5)
hc.orth.sppo.tot <- FindClusters(hc.orth.sppo.tot, resolution=0.5)
hc.full.sppf.tot <- FindClusters(hc.full.sppf.tot, resolution=0.5)
hc.orfl.spof <- FindClusters(hc.orfl.spof, resolution=0.5)
hc.orfl.sppo <- FindClusters(hc.orfl.sppo, resolution=0.5)
hc.orfl.sppf <- FindClusters(hc.orfl.sppf, resolution=0.5)
hc.orth.spof <- FindClusters(hc.orth.spof, resolution=0.5)
hc.orth.sppo <- FindClusters(hc.orth.sppo, resolution=0.5)
hc.orth.sppf <- FindClusters(hc.orth.sppf, resolution=0.5)
hc.full.spof <- FindClusters(hc.full.spof, resolution=0.5)
hc.full.sppo <- FindClusters(hc.full.sppo, resolution=0.5)
hc.full.sppf <- FindClusters(hc.full.sppf, resolution=0.5)
h.orfl.c.orfl.spof <- FindClusters(h.orfl.c.orfl.spof, resolution=0.5)
h.orfl.c.orfl.sppo <- FindClusters(h.orfl.c.orfl.sppo, resolution=0.5)
h.orfl.c.orfl.sppf <- FindClusters(h.orfl.c.orfl.sppf, resolution=0.5)
h.orth.c.orth.spof <- FindClusters(h.orth.c.orth.spof, resolution=0.5)
h.orth.c.orth.sppo <- FindClusters(h.orth.c.orth.sppo, resolution=0.5)
h.orth.c.orth.sppf <- FindClusters(h.orth.c.orth.sppf, resolution=0.5)
h.full.c.full.spof <- FindClusters(h.full.c.full.spof, resolution=0.5)
h.full.c.full.sppo <- FindClusters(h.full.c.full.sppo, resolution=0.5)
h.full.c.full.sppf <- FindClusters(h.full.c.full.sppf, resolution=0.5)

#Look at cluster IDs of the first 5 cells   #FULL DATA    SUBSET DATA
head(Idents(hc.orfl.spof.tot),5)            #19 clusters
head(Idents(hc.orth.sppo.tot),5)            #19 clusters  17 clusters
head(Idents(hc.full.sppf.tot),5)            #24 clusters  22 clusters
head(Idents(hc.orfl.spof),5)                #21 clusters
head(Idents(hc.orfl.sppo),5)                #21 clusters
head(Idents(hc.orfl.sppf),5)                #19 clusters
head(Idents(hc.orth.spof),5)                #21 clusters
head(Idents(hc.orth.sppo),5)                #20 clusters  15 clusters
head(Idents(hc.orth.sppf),5)                #19 clusters  15 clusters
head(Idents(hc.full.spof),5)                #22 clusters
head(Idents(hc.full.sppo),5)                #22 clusters  18 clusters
head(Idents(hc.full.sppf),5)                #19 clusters  17 clusters
head(Idents(h.orfl.c.orfl.spof),5)          #22 clusters
head(Idents(h.orfl.c.orfl.sppo),5)          #21 clusters
head(Idents(h.orfl.c.orfl.sppf),5)          #19 clusters
head(Idents(h.orth.c.orth.spof),5)          #21 clusters
head(Idents(h.orth.c.orth.sppo),5)          #21 clusters  18 clusters
head(Idents(h.orth.c.orth.sppf),5)          #20 clusters  18 clusters
head(Idents(h.full.c.full.spof),5)          #24 clusters
head(Idents(h.full.c.full.sppo),5)          #24 clusters  22 clusters
head(Idents(h.full.c.full.sppf),5)          #22 clusters  22 clusters

#Save data
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
saveRDS(hc.orfl.spof.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.tot.rds"))
saveRDS(hc.orth.sppo.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.tot.rds"))
saveRDS(hc.full.sppf.tot, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.tot.rds"))
saveRDS(hc.orfl.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.rds"))
saveRDS(hc.orfl.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppo.rds"))
saveRDS(hc.orfl.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppf.rds"))
saveRDS(hc.orth.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.spof.rds"))
saveRDS(hc.orth.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.rds"))
saveRDS(hc.orth.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppf.rds"))
saveRDS(hc.full.spof, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.spof.rds"))
saveRDS(hc.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppo.rds"))
saveRDS(hc.full.sppf, file=paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.rds"))
saveRDS(h.orfl.c.orfl.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
saveRDS(h.orfl.c.orfl.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppo.rds"))
saveRDS(h.orfl.c.orfl.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppf.rds"))
saveRDS(h.orth.c.orth.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.spof.rds"))
saveRDS(h.orth.c.orth.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
saveRDS(h.orth.c.orth.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppf.rds"))
saveRDS(h.full.c.full.spof, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.spof.rds"))
saveRDS(h.full.c.full.sppo, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
saveRDS(h.full.c.full.sppf, file=paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppf.rds"))

```

```{r non-linear dimensional reduction}

#Run non-linear dimensional reduction (UMAP/tSNE)
#notes: run this in the terminal and then come back to rstudio

##Install umap
#module load python/3.7.0
#module load R/3.5.1
#R
#library(reticulate)
#use_python("/software/python-3.7.0-el7-x86_64/bin/python")
#py_config()
#py_discover_config()
#cd /home/ghousman
#wget https://github.com/lmcinnes/umap/archive/master.zip
#unzip master.zip
#rm master.zip
#cd umap-master
#export PYTHONUSERBASE=$HOME/.local/python
#mkdir -p $PYTHONUSERBASE
#pip3 install -r requirements.txt --user
#pip3 install --upgrade pip --user
#python setup.py install --user
##notes: package installed in /home/ghousman/.local/python/lib/python3.7/site-packages

#sinteractive --mem=4g
#module load python/3.7.0
#module load R/3.5.1
#export PYTHONUSERBASE=$HOME/.local/python
#R
library(Seurat)
dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
hc.orfl.spof.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.tot.rds"))
hc.orfl.spof.tot <- RunUMAP(hc.orfl.spof.tot, dims=1:20)
saveRDS(hc.orfl.spof.tot, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.spof.tot.rds"))
rm(hc.orfl.spof.tot)
hc.orth.sppo.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.tot.rds"))
hc.orth.sppo.tot <- RunUMAP(hc.orth.sppo.tot, dims=1:20)
saveRDS(hc.orth.sppo.tot, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppo.tot.rds"))
rm(hc.orth.sppo.tot)
hc.full.sppf.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.tot.rds"))
hc.full.sppf.tot <- RunUMAP(hc.full.sppf.tot, dims=1:20)
saveRDS(hc.full.sppf.tot, file = paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppf.tot.rds"))
rm(hc.full.sppf.tot)
hc.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.spof.rds"))
hc.orfl.spof <- RunUMAP(hc.orfl.spof, dims=1:20)
saveRDS(hc.orfl.spof, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.spof.rds"))
rm(hc.orfl.spof)
hc.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppo.rds"))
hc.orfl.sppo <- RunUMAP(hc.orfl.sppo, dims=1:20)
saveRDS(hc.orfl.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.sppo.rds"))
rm(hc.orfl.sppo)
hc.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orfl.sppf.rds"))
hc.orfl.sppf <- RunUMAP(hc.orfl.sppf, dims=1:20)
saveRDS(hc.orfl.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.sppf.rds"))
rm(hc.orfl.sppf)
hc.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.spof.rds"))
hc.orth.spof <- RunUMAP(hc.orth.spof, dims=1:20)
saveRDS(hc.orth.spof, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orth.spof.rds"))
rm(hc.orth.spof)
hc.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppo.rds"))
hc.orth.sppo <- RunUMAP(hc.orth.sppo, dims=1:20)
saveRDS(hc.orth.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppo.rds"))
rm(hc.orth.sppo)
hc.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.orth.sppf.rds"))
hc.orth.sppf <- RunUMAP(hc.orth.sppf, dims=1:20)
saveRDS(hc.orth.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppf.rds"))
rm(hc.orth.sppf)
hc.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.spof.rds"))
hc.full.spof <- RunUMAP(hc.full.spof, dims=1:20)
saveRDS(hc.full.spof, file = paste0(dir,"/data/cellranger-data-full/xx-hc.full.spof.rds"))
rm(hc.full.spof)
hc.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppo.rds"))
hc.full.sppo <- RunUMAP(hc.full.sppo, dims=1:20)
saveRDS(hc.full.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppo.rds"))
rm(hc.full.sppo)
hc.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-hc.full.sppf.rds"))
hc.full.sppf <- RunUMAP(hc.full.sppf, dims=1:20)
saveRDS(hc.full.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppf.rds"))
rm(hc.full.sppf)
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.spof.rds"))
h.orfl.c.orfl.spof <- RunUMAP(h.orfl.c.orfl.spof, dims=1:20)
saveRDS(h.orfl.c.orfl.spof, file = paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.spof.rds"))
rm(h.orfl.c.orfl.spof)
h.orfl.c.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppo.rds"))
h.orfl.c.orfl.sppo <- RunUMAP(h.orfl.c.orfl.sppo, dims=1:20)
saveRDS(h.orfl.c.orfl.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.sppo.rds"))
rm(h.orfl.c.orfl.sppo)
h.orfl.c.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orfl.c.orfl.sppf.rds"))
h.orfl.c.orfl.sppf <- RunUMAP(h.orfl.c.orfl.sppf, dims=1:20)
saveRDS(h.orfl.c.orfl.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.sppf.rds"))
rm(h.orfl.c.orfl.sppf)
h.orth.c.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.spof.rds"))
h.orth.c.orth.spof <- RunUMAP(h.orth.c.orth.spof, dims=1:20)
saveRDS(h.orth.c.orth.spof, file = paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.spof.rds"))
rm(h.orth.c.orth.spof)
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppo.rds"))
h.orth.c.orth.sppo <- RunUMAP(h.orth.c.orth.sppo, dims=1:20)
saveRDS(h.orth.c.orth.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.sppo.rds"))
rm(h.orth.c.orth.sppo)
h.orth.c.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.orth.c.orth.sppf.rds"))
h.orth.c.orth.sppf <- RunUMAP(h.orth.c.orth.sppf, dims=1:20)
saveRDS(h.orth.c.orth.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.sppf.rds"))
rm(h.orth.c.orth.sppf)
h.full.c.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.spof.rds"))
h.full.c.full.spof <- RunUMAP(h.full.c.full.spof, dims=1:20)
saveRDS(h.full.c.full.spof, file = paste0(dir,"/data/cellranger-data-full/xx-h.full.c.full.spof.rds"))
rm(h.full.c.full.spof)
h.full.c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
h.full.c.full.sppo <- RunUMAP(h.full.c.full.sppo, dims=1:20)
saveRDS(h.full.c.full.sppo, file = paste0(dir,"/data/cellranger-data-full/xx-h.full.c.full.sppo.rds"))
rm(h.full.c.full.sppo)
h.full.c.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/x-h.full.c.full.sppo.rds"))
h.full.c.full.sppf <- RunUMAP(h.full.c.full.sppf, dims=1:20)
saveRDS(h.full.c.full.sppf, file = paste0(dir,"/data/cellranger-data-full/xx-hh.full.c.full.sppf.rds"))
rm(h.full.c.full.sppf)

```

```{r visualize non-linear dimensional reduction}

#Visualize non-linear dimensional reduction (UMAP/tSNE)
#notes: do this next part in rstudio

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
hc.orfl.spof.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.spof.tot.rds"))
hc.orth.sppo.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppo.tot.rds"))
hc.full.sppf.tot <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppf.tot.rds"))
hc.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.spof.rds"))
hc.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.sppo.rds"))
hc.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orfl.sppf.rds"))
hc.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orth.spof.rds"))
hc.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppo.rds"))
hc.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.orth.sppf.rds"))
hc.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.full.spof.rds"))
hc.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppo.rds"))
hc.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-hc.full.sppf.rds"))
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.spof.rds"))
h.orfl.c.orfl.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.sppo.rds"))
h.orfl.c.orfl.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.sppf.rds"))
h.orth.c.orth.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.spof.rds"))
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.sppo.rds"))
h.orth.c.orth.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.sppf.rds"))
h.full.c.full.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.c.full.spof.rds"))
h.full.c.full.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.c.full.sppo.rds"))
h.full.c.full.sppf <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.full.c.full.sppf.rds"))

#ggplot(as.data.frame(hc.orth.sppo.tot@reductions$umap@cell.embeddings), aes(UMAP_1, UMAP_2))+geom_point()
CombinePlots(plots=list(DimPlot(hc.orfl.spof, reduction="umap", label=TRUE),
                        DimPlot(hc.orth.spof, reduction="umap", label=TRUE),
                        DimPlot(hc.full.spof, reduction="umap", label=TRUE),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", label=TRUE),
                        DimPlot(h.orth.c.orth.spof, reduction="umap", label=TRUE),
                        DimPlot(h.full.c.full.spof, reduction="umap", label=TRUE),
                        DimPlot(hc.orfl.spof.tot, reduction="umap", label=TRUE),
                        
                        DimPlot(hc.orfl.sppo, reduction="umap", label=TRUE),
                        DimPlot(hc.orth.sppo, reduction="umap", label=TRUE),
                        DimPlot(hc.full.sppo, reduction="umap", label=TRUE),
                        DimPlot(h.orfl.c.orfl.sppo, reduction="umap", label=TRUE),
                        DimPlot(h.orth.c.orth.sppo, reduction="umap", label=TRUE),
                        DimPlot(h.full.c.full.sppo, reduction="umap", label=TRUE),
                        DimPlot(hc.orth.sppo.tot, reduction="umap", label=TRUE),
                        
                        DimPlot(hc.orfl.sppf, reduction="umap", label=TRUE),
                        DimPlot(hc.orth.sppf, reduction="umap", label=TRUE),
                        DimPlot(hc.full.sppf, reduction="umap", label=TRUE),
                        DimPlot(h.orfl.c.orfl.sppf, reduction="umap", label=TRUE),
                        DimPlot(h.orth.c.orth.sppf, reduction="umap", label=TRUE),
                        DimPlot(h.full.c.full.sppf, reduction="umap", label=TRUE),
                        DimPlot(hc.full.sppf.tot, reduction="umap", label=TRUE)), ncol=7)

CombinePlots(plots=list(DimPlot(hc.orfl.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.orth.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.full.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orth.c.orth.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.full.c.full.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.orfl.spof.tot, reduction="umap", group.by="orig.ident"),
                        
                        DimPlot(hc.orfl.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.orth.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.full.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orfl.c.orfl.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orth.c.orth.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.full.c.full.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.orth.sppo.tot, reduction="umap", group.by="orig.ident"),
                        
                        DimPlot(hc.orfl.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.orth.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.full.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orfl.c.orfl.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orth.c.orth.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.full.c.full.sppf, reduction="umap", group.by="orig.ident"),
                        DimPlot(hc.full.sppf.tot, reduction="umap", group.by="orig.ident")), ncol=7)

CombinePlots(plots=list(DimPlot(hc.orfl.spof, reduction="umap", group.by="species"),
                        DimPlot(hc.orth.spof, reduction="umap", group.by="species"),
                        DimPlot(hc.full.spof, reduction="umap", group.by="species"),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="species"),
                        DimPlot(h.orth.c.orth.spof, reduction="umap", group.by="species"),
                        DimPlot(h.full.c.full.spof, reduction="umap", group.by="species"),
                        DimPlot(hc.orfl.spof.tot, reduction="umap", group.by="species"),
                        
                        DimPlot(hc.orfl.sppo, reduction="umap", group.by="species"),
                        DimPlot(hc.orth.sppo, reduction="umap", group.by="species"),
                        DimPlot(hc.full.sppo, reduction="umap", group.by="species"),
                        DimPlot(h.orfl.c.orfl.sppo, reduction="umap", group.by="species"),
                        DimPlot(h.orth.c.orth.sppo, reduction="umap", group.by="species"),
                        DimPlot(h.full.c.full.sppo, reduction="umap", group.by="species"),
                        DimPlot(hc.orth.sppo.tot, reduction="umap", group.by="species"),
                        
                        DimPlot(hc.orfl.sppf, reduction="umap", group.by="species"),
                        DimPlot(hc.orth.sppf, reduction="umap", group.by="species"),
                        DimPlot(hc.full.sppf, reduction="umap", group.by="species"),
                        DimPlot(h.orfl.c.orfl.sppf, reduction="umap", group.by="species"),
                        DimPlot(h.orth.c.orth.sppf, reduction="umap", group.by="species"),
                        DimPlot(h.full.c.full.sppf, reduction="umap", group.by="species"),
                        DimPlot(hc.full.sppf.tot, reduction="umap", group.by="species")), ncol=7)

#summary plot
CombinePlots(plots=list(DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="species")), ncol=2)

```

```{r differential expression}

dir <- "/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA"
h.orfl.c.orfl.spof <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orfl.c.orfl.spof.rds"))
h.orth.c.orth.sppo <- readRDS(paste0(dir,"/data/cellranger-data-full/xx-h.orth.c.orth.sppo.rds"))

#Find differentially expressed features that define clusters (cluster biomarkers)

#notes: Seurat identifes positive and negative markers of a single cluster compared to all other cells by default
#notes: you can designate which groups of clusters vs. each other, or against all cells
#notes: FindAllMarkers automates this process (identifying positive and negative markers) for all clusters

#notes: min.pct argument requires a feature to be detected at a minimum percentage in either of the two groups of cells
#notes: thresh.test argument requires a feature to be differentially expressed (on average) by some amount between the two groups
#notes: max.cells.per.ident can be set to downsample each identity class to have no more cells than whatever this is set to (option to speed up computations)
#notes: test.use parameter can designate which of several tests for differential expression to use

#example: find all markers of cluster 1
c1.h.orfl.c.orfl.spof <- FindMarkers(h.orfl.c.orfl.spof, ident.1=1, min.pct=0.25)
head(c1.h.orfl.c.orfl.spof, n=5) #RGS4, SRGN, ANKRD1, MT2A, DKK1

#example: find all markers distinguishing clusters 0 and 16 from clusters 2 and 9
c0.c16v.c2.c9.h.orfl.c.orfl.spof <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(0,16), ident.2=c(2,9), min.pct=0.25)
head(c0.c16v.c2.c9.h.orfl.c.orfl.spof, n=5) #L1TD1, UTF1, MDK, APOE, SFRP1

#example: use the ROC test to return the ‘classification power’ for any individual marker (ranging from 0=random to 1=perfect)
c1.h.orfl.c.orfl.spof <- FindMarkers(h.orfl.c.orfl.spof, ident.1=1, logfc.threshold=0.25, test.use="roc", only.pos=TRUE)
head(c1.h.orfl.c.orfl.spof, n=5) #RGS4, SRGN, RHOC, SH3BGRL3


#find all markers distinguishing iPSC clusters between humans and chimps
#h.orfl.c.orfl.spof: humans (clusters 3+8+20) and chimps (clusters 0+18+19)
#h.orth.c.orth.sppo: humans (clusters 2+8+19) and chimps (clusters 0+18+20)
ipsc.compare <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(3,8,20), ident.2=c(0,18,19), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
ipsc.compare2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(2,8,19), ident.2=c(0,18,20), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
head(ipsc.compare, n=5) #SLC25A6, RPL28, FLNA, ACTC1, CLEC2D (607 genes total)
head(ipsc.compare2, n=5) #SLC25A6, ESRG, RPL28, FLNA, ACTC1 (641 genes total)
#find all markers distinguishing MSC clusters between humans and chimps
#h.orfl.c.orfl.spof: humans (clusters 1+16) and chimps (clusters 2+6+12)
#h.orth.c.orth.sppo: humans (clusters 1) and chimps (clusters 3+6+12+15)
msc.compare <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(1,16), ident.2=c(2,6,12), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
msc.compare2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(1), ident.2=c(3,6,12,15), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
head(msc.compare, n=5) #B2M, SRGN, RGS4, FLNA, MT2A (905 genes total)
head(msc.compare2, n=5) #RGS4, B2M, FLNA, SRGN, MT2A (929 genes total)
#find all markers distinguishing osteoblast clusters between humans and chimps
#h.orfl.c.orfl.spof: humans (clusters 4+15) and chimps (clusters 10+11+14+17)
#h.orth.c.orth.sppo: humans (clusters 4+13+17) and chimps (clusters 10+11+16)
osteo.compare <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(4,15), ident.2=c(10,11,14,17), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
osteo.compare2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(4,13,17), ident.2=c(10,11,16), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
head(osteo.compare, n=5) #B2M, FLNA, CRYAB, SRGN, KDELR2 (1123 genes total)
head(osteo.compare2, n=5) #FLNA, B2M, CRYAB, ANKRD1, SRGN (1281 genes total)
#find all markers distinguishing osteoblast clusters between humans and chimps
#h.orfl.c.orfl.spof: humans (clusters 5+7) and chimps (clusters 9+13+21)
#h.orth.c.orth.sppo: humans (clusters 5+7) and chimps (clusters 9+14)
chondro.compare <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(5,7), ident.2=c(9,13,21), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
chondro.compare2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(5,7), ident.2=c(9,14), logfc.threshold=0.25, test.use="wilcox", min.pct=0.5)
head(chondro.compare, n=5) #B2M, KDELR2, SLC25A6, DKK2, SPINT2 (699 genes total)
head(chondro.compare2, n=5) #B2M, KDELR2, SLC25A6, DKK2, FLNA (761 genes total)
#find all markers distinguishing humans and chimps across all cell types
total.compare <- intersect(rownames(ipsc.compare),intersect(rownames(msc.compare),intersect(rownames(osteo.compare),rownames(chondro.compare)))) #82 genes total
total.compare2 <- intersect(rownames(ipsc.compare2),intersect(rownames(msc.compare2),intersect(rownames(osteo.compare2),rownames(chondro.compare2)))) #89 genes total

#find all markers distinguishing iPSC clusters from other cell types
#h.orfl.c.orfl.spof: iPSC clusters (clusters 0+3+8+18+19+20)
#h.orth.c.orth.sppo: iPSC clusters (clusters 0+2+8+18+19+20)
ipsc.unique <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(0,3,8,18,19,20), min.pct=0.5)
ipsc.unique2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(0,2,8,18,19,20), min.pct=0.5)
head(ipsc.unique, n=5) #L1TD1, SFRP1, SEPHS1, APOE, EPCAM (2714 genes total)
head(ipsc.unique2, n=5) #L1TD1, SFRP1, SEPHS1, APOE, EPCAM (2725 genes total)
#find all markers distinguishing MSC clusters from other cell types
#h.orfl.c.orfl.spof: MSC clusters(clusters 1+2+6+12+16)
#h.orth.c.orth.sppo: MSC clusters (clusters 1+3+6+12+15)
msc.unique <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(1,2,6,12,16), min.pct=0.5)
msc.unique2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(1,3,6,12,15), min.pct=0.5)
head(msc.unique, n=5) #MMP1, CRYAB, RGS4, TAGLN2, CYTL1 (990 genes total)
head(msc.unique2, n=5) #MMP1, CRYAB, RGS4, TAGLN2, SERPINE1 (990 genes total)
#find all markers distinguishing osteoblast clusters from other cell types
#h.orfl.c.orfl.spof: osteoblast clusters (clusters 4+10+11+14+15+17)
#h.orth.c.orth.sppo: osteoblast clusters (clusters 4+10+11+13+16+17)
osteo.unique <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(4,10,11,14,15,17), min.pct=0.5)
osteo.unique2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(4,10,11,13,16,17), min.pct=0.5)
head(osteo.unique, n=5) #MGP, MALAT1, SPON2, GDF15, LGALS3 (1209 genes total)
head(osteo.unique2, n=5) #MGP, MALAT1, GDF15, SPON2, LGALS3 (1212 genes total)
#find all markers distinguishing chondrocyte clusters from other cell types
#h.orfl.c.orfl.spof: chondrocyte clusters (clusters 5+7+9+13+21) from other cell types
#h.orth.c.orth.sppo: chondrocyte clusters (clusters 5+7+9+14) from other cell types
chondro.unique <- FindMarkers(h.orfl.c.orfl.spof, ident.1=c(5,7,9,13,21), min.pct=0.5)
chondro.unique2 <- FindMarkers(h.orth.c.orth.sppo, ident.1=c(5,7,9,14), min.pct=0.5)
head(chondro.unique, n=5) #LUM, POSTN, COL3A1, CTSK, PTGDS (1968 genes total)
head(chondro.unique2, n=5) #LUM, POSTN, COL3A1, CTSK, PTGDS (1938 genes total)

#example: find markers for every cluster compared to all remaining cells, report only the positive ones
markers.h.orfl.c.orfl.spof <- FindAllMarkers(h.orfl.c.orfl.spof, only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25)
markers.h.orfl.c.orfl.spof  %>% group_by(cluster) %>% top_n(n=2, wt=avg_logFC) #SFRP1, UTF1, RGS4, SRGN, CYP181, CYTL1, MGST1, SLC25A6, GDF15, COL8A1

```

Need to decide which of these to choose for downstream analyses:

hc.orth.sppo.tot
hc.orth.spof
hc.orth.sppo
hc.orth.sppf
h.orth.c.orth.spof
h.orth.c.orth.sppo --- choice #2 (?)
h.orth.c.orth.sppf

hc.orfl.spof.tot
hc.orfl.spof
hc.orfl.sppo
hc.orfl.sppf
h.orfl.c.orfl.spof --- choice #1 (?)
h.orfl.c.orfl.sppo
h.orfl.c.orfl.sppf

```{r visualize marker expression}

#notes: VlnPlot shows expression probability distributions across clusters
#notes: FeaturePlot shows feature expression on a UMAP, tSNE, or PCA plot
#notes: other plotting functions include RidgePlot, CellScatter, and DotPlot
#notes: DoHeatmap generates an expression heatmap for given cells and features


#Compare h.orth.c.orth.sppo to h.orfl.c.orfl.spof

CombinePlots(plots=list(DimPlot(h.orth.c.orth.sppo, reduction="umap", label=TRUE),
                        DimPlot(h.orth.c.orth.sppo, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orth.c.orth.sppo, reduction="umap", group.by="species"),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", label=TRUE),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="orig.ident"),
                        DimPlot(h.orfl.c.orfl.spof, reduction="umap", group.by="species")), ncol=3)

#ipsc - NANOG not expressed in any cells, POU5F1 primarlity in iPSCs (but not many)
FeaturePlot(h.orth.c.orth.sppo, features=c("NANOG","POU5F1"))
FeaturePlot(h.orfl.c.orfl.spof, features=c("NANOG","POU5F1"))
#msc - no genes look good (MIXL1 and T in iPSCs and MSCs)
FeaturePlot(h.orth.c.orth.sppo, features=c("MIXL1","T"))
FeaturePlot(h.orfl.c.orfl.spof, features=c("MIXL1","T"))
#chondrocyte - COL11A1 looks the best
FeaturePlot(h.orth.c.orth.sppo, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"))
FeaturePlot(h.orfl.c.orfl.spof, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6"))
#osteoblast - COL1A1 in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.orth.c.orth.sppo, features=c("COL1A1","RUNX2","ALPL","IBSP", "SP7"))
FeaturePlot(h.orfl.c.orfl.spof, features=c("COL1A1","RUNX2","ALPL","IBSP", "SP7"))
#other - FABP4 in human chondrocytes, PPARG in chimp osteoblasts, TNNT2 in chondrocytes, SOX2 and EPCAM in iPSCs, YWHAZ in all cells
FeaturePlot(h.orth.c.orth.sppo, features=c("FABP4","PPARG","EOMES","SOX17","SOX1","SOX2","EPCAM","ALB",
                                           "HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"))
FeaturePlot(h.orfl.c.orfl.spof, features=c("FABP4","PPARG","EOMES","SOX17","SOX1","SOX2","EPCAM","ALB",
                                           "HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"))


#Visualize marker expression in more detail

#VlnPlot(h.orfl.c.orfl.spof, features=c("SLC25A6","RPL28","FLNA","ACTC1","CLEC2D"), slot="counts", log=TRUE) #plot raw counts
VlnPlot(h.orfl.c.orfl.spof, features=c("SLC25A6","RPL28","FLNA","ACTC1","CLEC2D")) #plot normalized data

FeaturePlot(h.orfl.c.orfl.spof, features=c("SLC25A6","RPL28","FLNA","ACTC1","CLEC2D")) #markers distinguishing iPSC clusters between humans and chimps
FeaturePlot(h.orfl.c.orfl.spof, features=c("B2M","SRGN","RGS4","FLNA","MT2A")) #markers distinguishing MSC clusters between humans and chimps
FeaturePlot(h.orfl.c.orfl.spof, features=c("B2M","FLNA","CRYAB","SRGN","KDELR2")) #markers distinguishing osteoblast clusters between humans and chimps
FeaturePlot(h.orfl.c.orfl.spof, features=c("B2M","KDELR2","SLC25A6","DKK2","SPINT2")) #markers distinguishing chondrocyte clusters between humans and chimps

FeaturePlot(h.orfl.c.orfl.spof, features=c("L1TD1","SFRP1","SEPHS1","APOE","EPCAM")) #markers distinguishing iPSC clusters other cell types
FeaturePlot(h.orfl.c.orfl.spof, features=c("MMP1","CRYAB","RGS4","TAGLN2","CYTL1")) #markers distinguishing MSC clusters other cell types
FeaturePlot(h.orfl.c.orfl.spof, features=c("MGP","MALAT1","SPON2","GDF15","LGALS3")) #markers distinguishing osteoblasts clusters other cell types
FeaturePlot(h.orfl.c.orfl.spof, features=c("LUM","POSTN","COL3A1","CTSK","PTGDS")) #markers distinguishing chondrocytes clusters other cell types

FeaturePlot(h.orfl.c.orfl.spof, features=c("NANOG","POU5F1"))                                         #ipsc - no genes look good
FeaturePlot(h.orfl.c.orfl.spof, features=c("MIXL1","T"))                                              #msc - no genes look good
FeaturePlot(h.orfl.c.orfl.spof, features=c("ACAN","COL2A1","SOX9","COL10A1","COL11A1","SOX5","SOX6")) #chondrocyte - COL11A1 looks the best
FeaturePlot(h.orfl.c.orfl.spof, features=c("COL1A1","RUNX2","ALPL","IBSP", "SP7"))                    #osteoblast - COL1A1 in MSCs/chondrocytes/osteoblasts
FeaturePlot(h.orfl.c.orfl.spof, features=c("FABP4","PPARG","EOMES","SOX17","SOX1","SOX2","EPCAM","ALB",
                                           "HAND1","TNNT2","MYL7","MYH6","GAPDH","GUSB","YWHAZ"))     #other - EPCAM is in iPSCs & YWHAZ in all cells


FeaturePlot(hc.orth.sppo, features="RUNX2") 

#pluripotency markers:  NANOG     ENSG00000111704
#                       POU5F1    ENSG00000204531
#mesoderm markers:      MIXL1     ENSG00000185155
#                       T         ENSG00000164458
#chondrocyte markers:   ACAN      ENSG00000157766
#                       COL2A1    ENSG00000139219
#                       SOX9      ENSG00000125398
#                       COL10A1   ENSG00000123500
#                       COL11A1   ENSG00000060718
#                       SOX5      ENSG00000134532
#                       SOX6      ENSG00000110693
#osteoblast markers:    COL1A1    ENSG00000108821
#                       BGLAP     -----
#                       RUNX2     ENSG00000124813
#                       ALPL      ENSG00000162551
#                       IBSP      ENSG00000029559
#                       SP7       ENSG00000170374
#adipocyte markers:     FABP4     ENSG00000170323
#                       PPARG     ENSG00000132170
#endomesoderm marker:   EOMES     ENSG00000163508
#endoderm marker:       SOX17     ENSG00000164736
#neuroectoderm marker:  SOX1      ENSG00000182968
#neuroepithelial:       SOX2      ENSG00000181449
#epithelial marker:     EPCAM     ENSG00000119888
#hepatocyte marker:     ALB       ENSG00000163631
#heart/NC marker:       HAND1     ENSG00000113196
#cardiomyocyte marker:  TNNT2     ENSG00000118194
#cariac muscle markers: MYL7      ENSG00000106631
#                       MYH6      ENSG00000197616
#neutral markers:       GAPDH     ENSG00000111640
#                       GUSB      -----
#                       YWHAZ     ENSG00000164924

top10 <- markers.h.orfl.c.orfl.spof %>% group_by(cluster) %>% top_n(n=10, wt=avg_logFC)
DoHeatmap(h.orfl.c.orfl.spof, features=top10$gene) + NoLegend()

```

Pluripotency markers
POU5F1
NANOG
TDGF1

Primative streak markers
TBXT
EOMES
MIXL1

Neural lineage markers
Neuroepithelial cells
SOX2
NES
NOTCH1
HES1
Schwann cell precursor
SOX10
GAP43
FABP7
MPZ
DHH
NGFR
Myelinating Schwann cells
SOX10
S100A1
EGR2
MPZ
MBP
Non myelinating Schwann cells
SOX10
S100A1
GAP43
NCAM1
NGFR
Radial Glia
VIM
PAX6
HES1
HES5
TNC
SLC1A3
Oligodendroctye precursors
PDGFRA
CSPG4
Oligodendroctyes
OLIG1
OLIG2
OLIG3
MOG
CLDN11
Astrocytes
GFAP
SLC1A3
SLC1A2
GLUL
S100B
ALDH1L1
Intermediate progenitors
EOMES
ASCL1
Immature Neurons
DCX
NEUROD1
TBR1
TUBB3
STMN1
Mature Neurons
MAP2
RBFOX3
NEFM
NEFH
DLG4
Glutaminergic
SLC17A7
SLC17A6
GRIN1
GRIN2B
GABAergic
SLC6A1
GABBR1
GABBR2
GAD1
GAD2
Dopaminergic
TH
SLC6A3
FOXA2
KCNJ6
NR4A2
LMX1B

Ectoderm markers
VIM
EMX2
CDH2
SOX11
TTR
MARCKS
FABP7
TMSB15A

Mesoderm Markers
Cardiomyoctyes precursors
MESP1
GATA4
GATA6
NKX2-5
NPPA
ISL1

Cardiac genes
TNNT2
MYH6
SIRPA
CD47
GJA1
DES
Skeletal muscle
PAX3
PAX7
MYF5
MYOD1
MYOG
MYF6
MYH3
MYH7
Endothelial cells
KDR
CD34
Adipocytes
SLC7A10
SLC36A2
P2RX5
MYF5
EBF2
PDGFRA
FGF10
FGF16
FGF19

Endoderm
Hepatocytes
ALB
HNF4A
AFP
Enterocytes
KRT18
EPCAM
MGAM
Globlet cells
KRT1
EGFR
ERBB2
ERBB3
Airway epithelium
SCGB3A2
SCGB3A2
SCGB3A1
TP63
KRT5
NKX2-1

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r data qc check on mito genes}

### OLD CODE (AUGUST 5, 2019) - RUN BEFORE CURRENT SPECIES/CELL/GENE FILTERED DATASETS ###

#Look into potential mitochondrial gene discrepancies
#notes on: filtered_feature_bc_matrix [full ref] / filtered_feature_bc_matrix [ortho-exon ref]

which(sapply(h.GHO@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
which(sapply(c.GHO@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="^MT-")==1)
which(sapply(GHO@assays$RNA@counts@Dimnames[[1]], stri_count_regex, pattern="-MT-")==1)

###should percent.mito be calculated within or between species?###
#option 1: calculate human with 13 vs. chimp with 1
#option 2: calculate human and chimp with 1 overlaping gene (MT-ND2, GRCh38------MT-ND2, Pan-tro-3.0-MT-ND2)
#option 3: don't calculate at all (only option available for ortho-exon data)

#does sum of GRCH38 & Pan-tro-3.0 genes in GHO equal number of genes in h.GHO for human cells / genes in c.GHO for chimp cells

#ipsc example - chimp+human do add up to chimp alone
h.GHO@meta.data$species[5:8]
c.GHO@meta.data$species[5:8]
GHO@meta.data$species[5:8]
h.GHO@assays$RNA@counts[c(33498),5:8]
c.GHO@assays$RNA@counts[c(23607),5:8]
GHO@assays$RNA@counts[c(33498,57145),5:8]

#msc example - multiplet
h.GHO@meta.data$species[6018:6021]
c.GHO@meta.data$species[5770:5773]
GHO@meta.data$species[5814:5817]
h.GHO@assays$RNA@counts[c(33498),6018:6021]
c.GHO@assays$RNA@counts[c(23607),5770:5773]
GHO@assays$RNA@counts[c(33498,57145),5814:5817]

#chondro example - chimp+human do not add up to chimp alone
h.GHO@meta.data$species[c(11846,11847,11851,11852)]
c.GHO@meta.data$species[c(11531,11532,11534,11535)]
GHO@meta.data$species[11575:11578]
h.GHO@assays$RNA@counts[c(33498),c(11846,11847,11849,11852)]
c.GHO@assays$RNA@counts[c(23607),c(11531,11532,11534,11535)]
GHO@assays$RNA@counts[c(33498,57145),11575:11578]

#osteo example - multiplet
h.GHO@meta.data$species[c(19074,19076,19077,19078)]
c.GHO@meta.data$species[c(18086,18087,18088,18089)]
GHO@meta.data$species[17432:17435]
h.GHO@assays$RNA@counts[c(33498),c(19074,19076,19077,19078)]
c.GHO@assays$RNA@counts[c(23607),c(18086,18087,18088,18089)]
GHO@assays$RNA@counts[c(33498,57145),17432:17435]

#more systematic way of comparing chimp+human gene count issues for mito gene overlap
h.GHO       #33538 features, 6003 iPSC + 5838 MSC + 7194 chondro + 4199 osteo
c.GHO       #25312 features, 5755 iPSC + 5771 MSC + 6528 chondro + 3741 osteo
GHO         #58850 features, 5799 iPSC + 5771 MSC + 5832 chondro + 3913 osteo

tbl.GHO <- as.matrix(GetAssayData(GHO, slot="counts"))
chk <- (tbl.GHO[33498,] >= 1) & (tbl.GHO[57145,] >= 1)
j <- which(chk==TRUE) #1072 cells

barcode=c()
type=c()
for (i in rownames(GHO@meta.data)[j]){
  barcode=c(barcode,strsplit(i,"_")[[1]][2])
  type=c(type,strsplit(i,"_")[[1]][1])
}

hc.prob = data.frame("cell"=rownames(GHO@meta.data)[j],
                     "barcode"=barcode,
                     "type"=type,
                     "spp"=GHO@meta.data$species[j],
                     "orig-index"=j,
                     "h_gene"=GHO@assays$RNA@counts[33498,j],
                     "c_gene"=GHO@assays$RNA@counts[57145,j],
                     row.names=NULL)
hc.prob$hc_sum = rowSums(hc.prob[,6:7])

w=1
k=c() #1064 total
h.barcode=c()
h.type=c()
h.spp=c()
h.index=c()
h.gene=c()
while (w <= length(rownames(h.GHO@meta.data))){
  if (strsplit(rownames(h.GHO@meta.data)[w],"_")[[1]][2] %in% hc.prob$barcode) {
    k=c(k,w)
    h.barcode=c(h.barcode,strsplit(rownames(h.GHO@meta.data)[w],"_")[[1]][2])
    h.type=c(h.type,strsplit(rownames(h.GHO@meta.data)[w],"_")[[1]][1])
    h.spp=c(h.spp,h.GHO@meta.data$species[w])
    h.index=c(h.index,length(k))
    h.gene=c(h.gene,h.GHO@assays$RNA@counts[33498,w])
  }
  w=w+1
}
h.prob = data.frame("h.barcode"=h.barcode,"h.type"=h.type,"h.spp"=h.spp,"h.index"=h.index,"h.gene"=h.gene,row.names=NULL)
w=1
h.cell=c()
while (w <= length(rownames(h.prob))){
  if (h.prob$h.type[w]=="h.iPSC")    { h.cell=c(h.cell,paste0("iPSC_",h.prob$h.barcode[w])) }
  if (h.prob$h.type[w]=="h.MSC")     { h.cell=c(h.cell,paste0("MSC_",h.prob$h.barcode[w])) }
  if (h.prob$h.type[w]=="h.chondro") { h.cell=c(h.cell,paste0("chondro_",h.prob$h.barcode[w])) }
  if (h.prob$h.type[w]=="h.osteo")   { h.cell=c(h.cell,paste0("osteo_",h.prob$h.barcode[w])) }
  w=w+1
}
h.prob$h.cell = h.cell
rm(h.barcode,h.type,h.spp,h.index,h.gene,h.cell,w,k)

z=1
m=c() #1059 total
c.barcode=c()
c.type=c()
c.spp=c()
c.index=c()
c.gene=c()
while (z <= length(rownames(c.GHO@meta.data))){
  if (strsplit(rownames(c.GHO@meta.data)[z],"_")[[1]][2] %in% hc.prob$barcode) {
    m=c(m,z)
    c.barcode=c(c.barcode,strsplit(rownames(c.GHO@meta.data)[z],"_")[[1]][2])
    c.type=c(c.type,strsplit(rownames(c.GHO@meta.data)[z],"_")[[1]][1])
    c.spp=c(c.spp,c.GHO@meta.data$species[z])
    c.index=c(c.index,length(m))
    c.gene=c(c.gene,c.GHO@assays$RNA@counts[23607,z])
  }
  z=z+1
}
c.prob = data.frame("c.barcode"=c.barcode,"c.type"=c.type,"c.spp"=c.spp,"c.index"=c.index,"c.gene"=c.gene,row.names=NULL)
z=1
c.cell=c()
while (z <= length(rownames(c.prob))){
  if (c.prob$c.type[z]=="c.iPSC")    { c.cell=c(c.cell,paste0("iPSC_",c.prob$c.barcode[z])) }
  if (c.prob$c.type[z]=="c.MSC")     { c.cell=c(c.cell,paste0("MSC_",c.prob$c.barcode[z])) }
  if (c.prob$c.type[z]=="c.chondro") { c.cell=c(c.cell,paste0("chondro_",c.prob$c.barcode[z])) }
  if (c.prob$c.type[z]=="c.osteo")   { c.cell=c(c.cell,paste0("osteo_",c.prob$c.barcode[z])) }
  z=z+1
}
c.prob$c.cell = c.cell
rm(c.barcode,c.type,c.spp,c.index,c.gene,c.cell,z,m)

tot.prob <- merge(x=hc.prob, y=h.prob, by.x="cell", by.y="h.cell", all.x=TRUE)
tot.prob <- merge(x=tot.prob, y=c.prob, by.x="cell", by.y="c.cell", all.x=TRUE)

match_spp=rep(NA, length(rownames(tot.prob))) #F=353, T=206, NA=513
match_sum=rep(NA, length(rownames(tot.prob))) #F=330, T=229, NA=523
z=1
while (z < length(rownames(tot.prob))) {
  if (tot.prob$spp[z]=="Human" & is.na(tot.prob$h.gene[z])==FALSE) {
    if (tot.prob$h_gene[z]==tot.prob$h.gene[z]) { match_spp[z]=TRUE }
    if (tot.prob$h_gene[z]!=tot.prob$h.gene[z]) { match_spp[z]=FALSE }
    if (tot.prob$hc_sum[z]==tot.prob$h.gene[z]) { match_sum[z]=TRUE }
    if (tot.prob$hc_sum[z]!=tot.prob$h.gene[z]) { match_sum[z]=FALSE }
  }
  if (tot.prob$spp[z]=="Chimp" & is.na(tot.prob$c.gene[z])==FALSE) {
    if (tot.prob$c_gene[z]==tot.prob$c.gene[z]) { match_spp[z]=TRUE }
    if (tot.prob$c_gene[z]!=tot.prob$c.gene[z]) { match_spp[z]=FALSE }
    if (tot.prob$hc_sum[z]==tot.prob$c.gene[z]) { match_sum[z]=TRUE }
    if (tot.prob$hc_sum[z]!=tot.prob$c.gene[z]) { match_sum[z]=FALSE }
  }
  z=z+1
}
tot.prob$match_spp=match_spp
tot.prob$match_sum=match_sum
rm(match_spp,match_sum)

table(tot.prob$match_spp[which(tot.prob$spp=="Chimp")],useNA="always")
table(tot.prob$match_sum[which(tot.prob$spp=="Chimp")],useNA="always")
table(tot.prob$match_spp[which(tot.prob$spp=="Human")],useNA="always")
table(tot.prob$match_sum[which(tot.prob$spp=="Human")],useNA="always")
#no clear answer for human+chimp mito gene count issues


#####
#more systematic way to look across all genes (not just mito genes)???
h.x = which(rownames(h.GHO@assays$RNA@counts) %in% rownames(c.GHO@assays$RNA@counts)) #16542 genes overlap
c.x = which(rownames(c.GHO@assays$RNA@counts) %in% rownames(h.GHO@assays$RNA@counts))
#####


#systematic way to look across all genes (not just mito genes) for ortho-exon datasets
ortho.h.GHO #44125 features, 5597 iPSC + 5741 MSC + 3794 chondro + 3228 osteo
ortho.c.GHO #44125 features, 5608 iPSC + 5750 MSC + 4389 chondro + 3282 osteo
ortho.GHO   #88250 features, 5590 iPSC + 5622 MSC + 3253 chondro + 3125 osteo

h.x = which(rownames(ortho.h.GHO@assays$RNA@counts) %in% rownames(ortho.c.GHO@assays$RNA@counts)) #44125 genes overlap
c.x = which(rownames(ortho.c.GHO@assays$RNA@counts) %in% rownames(ortho.h.GHO@assays$RNA@counts))

h.gene=c()
for (i in rownames(ortho.GHO@assays$RNA@counts)[1:44125]){ h.gene=c(h.gene,strsplit(i,"ortho-GRCh38------")[[1]][2]) }
c.gene=c()
for (i in rownames(ortho.GHO@assays$RNA@counts)[44126:88250]){ c.gene=c(h.gene,strsplit(i,"ortho-Pan-tro-3.0-")[[1]][2]) }

tbl.ortho.GHO <- as.matrix(GetAssayData(ortho.GHO, slot="counts"))
check <- (tbl.ortho.GHO[1:44125,] >= 1) & (tbl.ortho.GHO[44126:88250,] >= 1)
rownames(check) <- h.gene
check.index <- which(rowSums(check)>1) #rows (genes) with >= 1 human gene and >= 1 chimp gene
table(check[check.index,]) #63004 instances of >= 1 human gene and >= 1 chimp gene

#separate human/chimp cells
c.index <- which(ortho.GHO@meta.data$species=="Chimp")
h.index <- which(ortho.GHO@meta.data$species=="Human")

#separate appropriate genes and cells in h.ortho.GHO and c.ortho.GHO
h.barcode <- c()
h.type <- c()
for (i in colnames(ortho.h.GHO@assays$RNA@counts)) {
  h.barcode=c(h.barcode,strsplit(i,"_")[[1]][2])
  h.type=c(h.type,strsplit(i,"_")[[1]][1])
}
h.type2=c()
for (i in h.type) {
  if (i=="h.iPSC")    { h.type2=c(h.type2,"iPSC") }
  if (i=="h.MSC")     { h.type2=c(h.type2,"MSC") }
  if (i=="h.chondro") { h.type2=c(h.type2,"chondro") }
  if (i=="h.osteo")   { h.type2=c(h.type2,"osteo") }
}
z=1
h.cell=c()
while (z <= length(h.type2)){
  h.cell=c(h.cell,paste0(paste0(h.type2[z],"_",h.barcode[z])))
  z=z+1
}
tbl.ortho.h.GHO <- as.matrix(GetAssayData(ortho.h.GHO, slot="counts"))
colnames(tbl.ortho.h.GHO) <- h.cell
rm(h.barcode,h.type,h.type2,h.cell,z)

c.barcode <- c()
c.type <- c()
for (i in colnames(ortho.c.GHO@assays$RNA@counts)) {
  c.barcode=c(c.barcode,strsplit(i,"_")[[1]][2])
  c.type=c(c.type,strsplit(i,"_")[[1]][1])
}
c.type2=c()
for (i in c.type) {
  if (i=="c.iPSC")    { c.type2=c(c.type2,"iPSC") }
  if (i=="c.MSC")     { c.type2=c(c.type2,"MSC") }
  if (i=="c.chondro") { c.type2=c(c.type2,"chondro") }
  if (i=="c.osteo")   { c.type2=c(c.type2,"osteo") }
}
z=1
c.cell=c()
while (z <= length(c.type2)){
  c.cell=c(c.cell,paste0(paste0(c.type2[z],"_",c.barcode[z])))
  z=z+1
}
tbl.ortho.c.GHO <- as.matrix(GetAssayData(ortho.c.GHO, slot="counts"))
colnames(tbl.ortho.c.GHO) <- c.cell
rm(c.barcode,c.type,c.type2,c.cell,z)

c.index.ortho.h.GHO <- which(colnames(tbl.ortho.h.GHO) %in% colnames(tbl.ortho.GHO)[c.index]) #c.index.ortho.h.GHO=8877
h.index.ortho.h.GHO <- which(colnames(tbl.ortho.h.GHO) %in% colnames(tbl.ortho.GHO)[h.index]) #h.index.ortho.h.GHO=8253
c.index.ortho.c.GHO <- which(colnames(tbl.ortho.c.GHO) %in% colnames(tbl.ortho.GHO)[c.index]) #c.index.ortho.c.GHO=8946
h.index.ortho.c.GHO <- which(colnames(tbl.ortho.c.GHO) %in% colnames(tbl.ortho.GHO)[h.index]) #h.index.ortho.c.GHO=8260

c.index.ortho.GHO.h <- which(colnames(tbl.ortho.GHO) %in% colnames(tbl.ortho.h.GHO)[c.index.ortho.h.GHO]) #c.index.ortho.GHO.h=8877
h.index.ortho.GHO.h <- which(colnames(tbl.ortho.GHO) %in% colnames(tbl.ortho.h.GHO)[h.index.ortho.h.GHO]) #h.index.ortho.GHO.h=8253
c.index.ortho.GHO.c <- which(colnames(tbl.ortho.GHO) %in% colnames(tbl.ortho.c.GHO)[c.index.ortho.c.GHO]) #c.index.ortho.GHO.c=8946
h.index.ortho.GHO.c <- which(colnames(tbl.ortho.GHO) %in% colnames(tbl.ortho.c.GHO)[h.index.ortho.c.GHO]) #h.index.ortho.GHO.c=8260

#check if gene totals = human/chimp gene counts
tbl.ortho.h.GHO[,c.index.ortho.h.GHO] #ortho.h.GHO with only human genes and with only chimp cells
tbl.ortho.h.GHO[,h.index.ortho.h.GHO] #ortho.h.GHO with only human genes and with only human cells
tbl.ortho.c.GHO[,c.index.ortho.c.GHO] #ortho.c.GHO with only chimp genes and with only chimp cells
tbl.ortho.c.GHO[,h.index.ortho.c.GHO] #ortho.c.GHO with only chimp genes and with only human cells
check[,c.index.ortho.GHO.h] #ortho.GHO comparison of human genes vs chimp genes with only chimp cells (overlap with ortho.h.GHO)
check[,c.index.ortho.GHO.c] #ortho.GHO comparison of human genes vs chimp genes with only chimp cells (overlap with ortho.c.GHO)
check[,h.index.ortho.GHO.h] #ortho.GHO comparison of human genes vs chimp genes with only human cells (overlap with ortho.h.GHO)
check[,h.index.ortho.GHO.c] #ortho.GHO comparison of human genes vs chimp genes with only human cells (overlap with ortho.c.GHO)
tbl.ortho.GHO[1:44125,c.index.ortho.GHO.h] #ortho.GHO with only human genes and with only chimp cells (overlap with ortho.h.GHO)
tbl.ortho.GHO[1:44125,c.index.ortho.GHO.c] #ortho.GHO with only human genes and with only chimp cells (overlap with ortho.c.GHO)
tbl.ortho.GHO[1:44125,h.index.ortho.GHO.h] #ortho.GHO with only human genes and with only human cells (overlap with ortho.h.GHO)
tbl.ortho.GHO[1:44125,h.index.ortho.GHO.c] #ortho.GHO with only human genes and with only human cells (overlap with ortho.c.GHO)
tbl.ortho.GHO[44126:88250,c.index.ortho.GHO.h] #ortho.GHO with only chimp genes and with only chimp cells (overlap with ortho.h.GHO)
tbl.ortho.GHO[44126:88250,c.index.ortho.GHO.c] #ortho.GHO with only chimp genes and with only chimp cells (overlap with ortho.c.GHO)
tbl.ortho.GHO[44126:88250,h.index.ortho.GHO.h] #ortho.GHO with only chimp genes and with only human cells (overlap with ortho.h.GHO)
tbl.ortho.GHO[44126:88250,h.index.ortho.GHO.c] #ortho.GHO with only chimp genes and with only human cells (overlap with ortho.c.GHO)

sumcheck <- (tbl.ortho.GHO[1:44125,] + tbl.ortho.GHO[44126:88250,])
rownames(sumcheck) <- rownames(check)

check.index #genes with >= 1 human gene and >= 1 chimp gene: 1967

#for genes with >= 1 human gene and >= 1 chimp gene, does the sum of genes equal number of genes in h.ortho.GHO or c.ortho.GHO
dim(sumcheck[check.index,h.index.ortho.GHO.h])
dim(tbl.ortho.h.GHO[check.index,h.index.ortho.h.GHO])
test <- sumcheck[check.index,c.index.ortho.GHO.h] != tbl.ortho.h.GHO[check.index,c.index.ortho.h.GHO] #some unequal positions
test <- sumcheck[check.index,h.index.ortho.GHO.h] != tbl.ortho.h.GHO[check.index,h.index.ortho.h.GHO] #some unequal positions
test <- sumcheck[check.index,c.index.ortho.GHO.c] != tbl.ortho.c.GHO[check.index,c.index.ortho.c.GHO] #some unequal positions
test <- sumcheck[check.index,h.index.ortho.GHO.c] != tbl.ortho.c.GHO[check.index,h.index.ortho.c.GHO] #some unequal positions
length(which(rowSums(test)>1)) #how many rows with true results
table(test[1,])
test.index <- which(rowSums(test)>1)
#no - the sum of genes does not equal the genes in h.ortho.GHO or c.ortho.GHO

#plan moving forward - just use human-chimp species assignments from combo reference and discard non-corresponding human-chimp genes

```

```{r downsampling attempts}

##### DOWNSAMPLING

#cellranger seq. saturation = fraction of confidently mapped, valid cell-barcode, valid UMI reads that are non-unique (match existing cell-barcode/UMI/gene combo)
#sequencing saturation = 1 - (n_deduped_reads / n_reads), where
#n_deduped_reads = Number of unique (valid cell-barcode, valid UMI, gene) combinations among confidently mapped reads. 
#n_reads = Total number of confidently mapped, valid cell-barcode, valid UMI reads.
#(measure of the fraction of library complexity that was captured during sequencing)
#(1 minus the ratio of the number of UMIs to the number of reads, per cell for resampled and untargeted cells in the original scRNA-seq library or after resampling)

#sinteractive  --partition=broadwl
#module load samtools/1.9
#cd /project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full/cellranger-runs-ortho-human/GHO-1/outs
#mkdir downsampled
#samtools view -bs 42.1 possorted_genome_bam.bam > downsampled/downsampled_0.1.bam
##subsample 10 percent mapped reads with 42 as the seed for the random number generator
#samtools view -c possorted_genome_bam.bam
##get the total number of reads of a BAM file (may include unmapped and duplicated multi-aligned reads) 
#samtools view -c -F 260 possorted_genome_bam.bam
##counting only mapped (primary aligned) reads

h.orth.GHO1 <- '/cellranger-runs-ortho-human/GHO-1/outs'
h.orth.GHO2 <- '/cellranger-runs-ortho-human/GHO-2/outs'
h.orth.GHO3 <- '/cellranger-runs-ortho-human/GHO-3/outs'
h.orth.GHO4 <- '/cellranger-runs-ortho-human/GHO-4/outs'
c.orth.GHO1 <- '/cellranger-runs-ortho-chimp/GHO-1/outs'
c.orth.GHO2 <- '/cellranger-runs-ortho-chimp/GHO-2/outs'
c.orth.GHO3 <- '/cellranger-runs-ortho-chimp/GHO-3/outs'
c.orth.GHO4 <- '/cellranger-runs-ortho-chimp/GHO-4/outs'

h.orfl.GHO1 <- '/cellranger-runs-ortho-filter-human/GHO-1/outs'
h.orfl.GHO2 <- '/cellranger-runs-ortho-filter-human/GHO-2/outs'
h.orfl.GHO3 <- '/cellranger-runs-ortho-filter-human/GHO-3/outs'
h.orfl.GHO4 <- '/cellranger-runs-ortho-filter-human/GHO-4/outs'
c.orfl.GHO1 <- '/cellranger-runs-ortho-filter-chimp/GHO-1/outs'
c.orfl.GHO2 <- '/cellranger-runs-ortho-filter-chimp/GHO-2/outs'
c.orfl.GHO3 <- '/cellranger-runs-ortho-filter-chimp/GHO-3/outs'
c.orfl.GHO4 <- '/cellranger-runs-ortho-filter-chimp/GHO-4/outs'

dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full'        #data from three lanes of sequencing
metrics.h.orfl.GHO1 <- read.csv(file=paste0(dir,'/cellranger-runs-ortho-human/GHO-1/outs/metrics_summary.csv'))
head(metrics.h.orfl.GHO1)

#Run downsampling tests to assess sequencing saturation
#notes: run this in the terminal and then come back to rstudio

##Install DropletUtils
#module load R/3.6.1
#R
#BiocManager::install("DropletUtils")

#Load libraries
library(DropletUtils)
library(Matrix)

#Define main directory
#dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-subset'      #subset of data from first lane of sequencing
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/scRNA/cellranger-data-full'        #data from three lanes of sequencing

#Check content of directory
list.files(paste0(dir,"/cellranger-runs-ortho-human/GHO-1/outs"))

#Define Variables
down = c()
umi = c()
reads = c()

#Read 10X data - molecule information
mol.data <- read10xMolInfo(paste0(dir,"/cellranger-runs-ortho-filter-human/GHO-1/outs/molecule_info.h5"))
length(mol.data$data$reads)                            #total umi in data   = 56165642
sum(mol.data$data$reads)                               #total reads in data = 75145550
1-length(mol.data$data$reads)/sum(mol.data$data$reads) #saturation          = 0.2525753

down = c(down,1)
umi = c(umi,length(mol.data$data$reads))
reads = c(reads,sum(mol.data$data$reads))

#Downsampling - molecule information
#set.seed(100)
#down1.0 <- downsampleReads(paste0(dir,"/cellranger-runs-ortho-filter-human/GHO-1/outs/molecule_info.h5"), prop=1)
#sum(down1.0)                                #total umi in data   = 56165642
#sum(mol.data$data$reads)*1                  #total reads in data = 75145550
#1-sum(down1.0)/(sum(mol.data$data$reads)*1) #saturation          = 0.2525753
prop = c(0.9,0.8,0.7,0.6,0.5,0.4,0.3,0.2,0.1)
for (i in prop) {
  sub <- downsampleReads(paste0(dir,"/cellranger-runs-ortho-filter-human/GHO-1/outs/molecule_info.h5"), prop=i)
  down = c(down,i)
  umi = c(umi,sum(sub))
  reads = c(reads,sum(mol.data$data$reads)*i)
}
satu = 1-umi/reads

#h.orfl.GHO1
#down = c(1.0, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1)
#umi = c(56165642, 51842066, 47291066, 42501225, 37446925, 32103096, 26449551, 20446874, 14066728, 7265532)
#reads = c(75145550, 67630995, 60116440, 52601885, 45087330, 37572775, 30058220, 22543665, 15029110, 7514555)
#satu = c(0.25257528, 0.23345700, 0.21334221, 0.19202088, 0.16945792, 0.14557559, 0.12005598, 0.09301021, 0.06403453, 0.03313876)

```

```{r data gene annotation adjustments}

### DO NOT RUN THIS RIGHT NOW
### THERE IS AN ISSUE WITH CHANGING GENE NAMES WHEN TRYING TO CALCULATE VARIABLE FEATURES

#Replace ensembl gene id with corresponding gene name - ortho annotations

#notes: this also double checks whether any mito genes are in ortho annotations
#notes: no genes included from mitochondrial genome so no mito genes in ortho annotations
h.gtf <- read.delim(file="../../../references/human-chimp-ortho-exons/Human_orthoexon_extended_info.txt")
c.gtf <- read.delim(file="../../../references/human-chimp-ortho-exons/Chimp_orthoexon_extended_info_edit.txt")
h.gtf <- unique(h.gtf[,c(1,3,4)])
c.gtf <- unique(c.gtf[,c(1,3,4)])
#table(h.gtf$chr[which(sapply(h.gtf$gene_name, stri_count_regex, pattern="^MT")==1)])

h.gene=c()
c.gene=c()
for (i in rownames(hc.orth.sppo.tot@assays$RNA@counts)[1:44125]) { h.gene=c(h.gene,strsplit(i,"GRCh38------")[[1]][2]) }
for (i in rownames(hc.orth.sppo.tot@assays$RNA@counts)[44126:88250]) { c.gene=c(c.gene,strsplit(i,"Pan-tro-3.0-")[[1]][2]) }
h.gene2=c()
c.gene2=c()
for (i in h.gene) { h.gene2=c(h.gene2,paste0("GRCh38------",h.gtf$gene_name[which(h.gtf$ensg %in% i)])) }
for (i in c.gene) { c.gene2=c(c.gene2,paste0("GRCh38------",c.gtf$gene_name[which(c.gtf$ensg %in% i)])) }
rownames(hc.orth.sppo.tot@assays$RNA@counts) <- c(h.gene2,c.gene2)
rownames(hc.orth.sppo.tot@assays$RNA@data) <- c(h.gene2,c.gene2)

gene=c()
for (i in rownames(hc.orth.sppo@assays$RNA@counts)) { gene=c(gene,as.character(h.gtf$gene_name[which(h.gtf$ensg %in% i)])) }
rownames(hc.orth.sppo@assays$RNA@counts) <- gene
rownames(hc.orth.sppo@assays$RNA@data) <- gene

gene=c()
for (i in rownames(hc.orth.sppf@assays$RNA@counts)) { gene=c(gene,as.character(h.gtf$gene_name[which(h.gtf$ensg %in% i)])) }
rownames(hc.orth.sppf@assays$RNA@counts) <- gene
rownames(hc.orth.sppf@assays$RNA@data) <- gene

gene=c()
for (i in rownames(h.orth.c.orth.sppo@assays$RNA@counts)) { gene=c(gene,as.character(h.gtf$gene_name[which(h.gtf$ensg %in% i)])) }
rownames(h.orth.c.orth.sppo@assays$RNA@counts) <- gene
rownames(h.orth.c.orth.sppo@assays$RNA@data) <- gene

gene=c()
for (i in rownames(h.orth.c.orth.sppf@assays$RNA@counts)) { gene=c(gene,as.character(h.gtf$gene_name[which(h.gtf$ensg %in% i)])) }
rownames(h.orth.c.orth.sppf@assays$RNA@counts) <- gene
rownames(h.orth.c.orth.sppf@assays$RNA@data) <- gene

rm(h.gene,h.gene2,c.gene,c.gene2,gene,i)


#See how many genes overlap between ortho and full datasets
rownames(hc.orth.sppo@assays$RNA@data) #44125
rownames(hc.full.sppo@assays$RNA@data) #16542
orth <- which(rownames(hc.orth.sppo@assays$RNA@data) %in% rownames(hc.full.sppo@assays$RNA@data)) #15399
full <- which(rownames(hc.full.sppo@assays$RNA@data) %in% rownames(hc.orth.sppo@assays$RNA@data)) #15388
length(unique(rownames(hc.orth.sppo@assays$RNA@data)[orth])) #15388
length(unique(rownames(hc.full.sppo@assays$RNA@data)[full])) #15388
rownames(hc.orth.sppo@assays$RNA@data)[orth][which(duplicated(rownames(hc.orth.sppo@assays$RNA@data)[orth])==TRUE)]
#genes that are duplicated in hc.orth.spoo: BTBD8, RGS5, SDHD, SPATA13, SFTA3, ELFN2, MATR3, RABGEF1, MAL2, LYNX1, LYNX1
#indices in hc.orth.sppo@assays$RNA@data are: 1712, 2766, 8113, 11154, 12577, 27941, 34573, 38262, 40655, 40936, 40937
x=c(1712, 2766, 8113, 11154, 12577, 27941, 34573, 38262, 40655, 40936, 40937)
hc.orth.sppo@assays$RNA@data[x,1:3]

```