---
title: "10x-data-diffexp"
author: "Genevieve Housman"
date: "SEptember 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Differential Expression

Test for differential expression between humans and chimpanzees within different cell classifications.


```{r load libraries}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(limma)
library(variancePartition)

```

```{r}

#Load data
scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k-tot.assign.rds" #integrate across individuals - total (15k genes)
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int-tot.assign.rds" #integrate across individuals - total
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int-t0.assign.rds" #integrate across individuals - Time 0
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int-t1.assign.rds" #integrate across individuals - Time 1
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int-t2.assign.rds" #integrate across individuals - Time 2
data <- readRDS(scrna)

data

```

Define genes of interest: all except mitochondrial genes and ribosomal genes

```{r}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

### Identify differentially expressed genes

edgeRQLF (Han et al., 2015)
https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_edgeRQLFDetRate.R

* uses negative binomial model-based method to determine DE genes
* model: weighted trimmed mean of the log expression ratios used to normalize the sequencing depth and gene length between the samples
    + expression data fit to a negative binomial model (mean μ and variance ν have relationship of ν  = μ + αμ2, and α is the dispersion factor)
    + dispersion factor estimated by combining:
        - common dispersion across all the genes (estimated by a likelihood function), and
        - gene-specific dispersion (estimated by the empirical Bayes method)
* de: exact test with FDR control is used to determine DE genes


Tests Adjusting Different edgeR Function Parameters
[ using data subsetted to just cells assigned to Cluster iPSC.c3 ]
[ model.matrix(~Collection+cdr+nCount_RNA+Phase+Species) ]

*  797/ 5065: filterByExpr(min.count=0.05), glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BH")
*  841/ 5065: filterByExpr(min.count=0.05), glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=FALSE), decideTests(adjust.method="BH")
*  805/ 5065: filterByExpr(min.count=0.05), glmQLFit(robust=FALSE), glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BH")
* 1084/18137: [kept all genes],             glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BH")
*  804/ 5057: filterByExpr(min.count=0.25), glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BH")
*  616/ 2792: filterByExpr(min.count=0.5),  glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BH")
*  512/ 5065: filterByExpr(min.count=0.05), glmQLFit(robust=TRUE),  glmQLFTest(poisson.bound=TRUE),  decideTests(adjust.method="BY")

Tests Adjusting Model Designs
[ using data subsetted to just cells assigned to Cluster iPSC.c3 ]
[ filterByExpr(min.count=0.05), glmQLFit(robust=TRUE) ]

* 2621/ 5065: model.matrix(~Species)
* ----/ 5065: model.matrix(~Sample+Species)
    + Sample and Species are completely confounded
* 2012/ 5065: model.matrix(~DateCollection+Species)
*  961/ 5065: model.matrix(~Collection+Species)
* 2085/ 5065: model.matrix(~SequencingBatch+Species)
* 2237/ 5065: model.matrix(~nCount_RNA+Species)
* 2633/ 5065: model.matrix(~percent.mt+Species)
* 2551/ 5065: model.matrix(~Phase+Species)
* 2010/ 5065: model.matrix(~DateCollection+nCount_RNA+Species)
*  823/ 5065: model.matrix(~Collection+nCount_RNA+Species)
* ----/ 5065: model.matrix(~DateCollection+SequencingBatch+nCount_RNA+Species)
    + DateCollection and SequencingBatch completely confounded
* ----/ 5065: model.matrix(~Collection+SequencingBatch+nCount_RNA+Species)
    + Collection and SequencingBatch are confounded
*  811/ 5065: model.matrix(~Collection+nCount_RNA+Phase+Species)
*  797/ 5065: model.matrix(~Collection+nCount_RNA+percent.mt+Phase+Species)
*  696/ 5065: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species)
    + USE THIS MODEL FOR NOW
*  696/ 5065: model.matrix(~Collection_SequencingBatch+cdr+nCount_RNA+percent.mt+Phase+Species)
*  672 / 5065: model.matrix(~Collection+cdr+lib.size+nCount_RNA+percent.mt+Phase+Species)
*  709 / 5065: model.matrix(~Collection+cdr+lib.size+percent.mt+Phase+Species)

*  704/ 5065: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+S.Score+G2M+Score+Species)
*  730/ 5065: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Diff.S.G2M+Species)

Tests Adjusting which Samples are Included
[ using data subsetted to just cells assigned to Cluster iPSC.c3 ]
[ filterByExpr(min.count=0.05), glmQLFit(robust=TRUE) ]

*  696/ 5065: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species), all individuals
*  629/ 7307: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species), no H1-I or C1-I
*  694/ 5196: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species), no H1-I-r2 or C1-I-r2
*  365/ 3677: model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species), all individuals randomly subset to 200 human and 200 chimp cells


HOW TO DEAL WITH REPEAT SAMPLES?
NEED TO CHECK MIN.COUNT THRESHOLD TO SEE HOW MANY GENES ARE KEPT FOR EACH DATA SUBSET!

```{r}
dataSub <- subset(data,subset=Cluster=="iPSC.c3")

#dataSub <- subset(dataSub,subset=Sample!="H1-I")
#dataSub <- subset(dataSub,subset=Sample!="C1-I")
#dataSub <- subset(dataSub,subset=Sample!="H1-I-r2")
#dataSub <- subset(dataSub,subset=Sample!="C1-I-r2")

#human.cells <- rownames(dataSub@meta.data)[which(dataSub@meta.data$Species=="Human")]
#chimp.cells <- rownames(dataSub@meta.data)[which(dataSub@meta.data$Species=="Chimp")]
#h.cell.sub <- human.cells[sample(1:length(human.cells), 200, replace=FALSE)]
#c.cell.sub <- chimp.cells[sample(1:length(chimp.cells), 200, replace=FALSE)]
#dataSub <- subset(dataSub,cells=c(h.cell.sub,c.cell.sub))
```

```{r}

#Define EDGER Function - THIS IS NOT THE MOST UP TO DATE VERSION
#REFERENCE DEedgeR.R

runEDGER <- function(dataSub, genes.no.mito.ribo) {

  #count matrix
  counts <- as.matrix(GetAssayData(dataSub, assay="RNA", slot="counts"))

  #remove mitochodrial and ribosomal genes
  counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
  
  #metadata
  metadata <- dataSub@meta.data[,c("Species","Collection","nCount_RNA","percent.mt","Phase")]
  #metadata <- dataSub@meta.data[,c("Species","Collection","nCount_RNA","percent.mt","S.Score","G2M.Score","Phase","Sample","Individual","DateCollection","SequencingBatch")]
  metadata <- metadata[colnames(counts),]
  
  #make edgeR object
  dge <- DGEList(counts)
  meta_dge <- dge$samples[,c("lib.size","norm.factors")]
  meta_dge <- cbind(meta_dge, metadata)
  dge$samples <- meta_dge
  rm(dataSub,counts,metadata,meta_dge)
  
  #filter genes
  #dge$counts <- dge$counts[rowSums(dge$counts!=0)>=(0.2*dim(dge$counts)[2]),]
  keep <- filterByExpr(dge, group=dge$samples$Species, min.count=0.05, min.total.count=15)
  table(keep)
  dge <- dge[keep, , keep.lib.sizes=FALSE]
  rm(keep)
  
  #normalize data
  dge <- calcNormFactors(dge, method="TMM")
  summary(dge$samples$norm.factors)
  
  #calculate cellular detection rate
  cdr <- scale(colMeans(dge$count > 0))
  
  #design matrix
  dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
  dge$samples$Phase <- as.factor(dge$samples$Phase)
  dge$samples$Species <- as.factor(as.character(dge$samples$Species))
  dge$samples$Diff.S.G2M <- dge$samples$S.Score - dge$samples$G2M.Score
  #dge$samples$Sample <- as.factor(as.character(dge$samples$Sample))
  #dge$samples$Individual <- as.factor(as.character(dge$samples$Individual))
  #dge$samples$DateCollection <- as.factor(as.character(dge$samples$DateCollection))
  #dge$samples$SequencingBatch <- as.factor(as.numeric(dge$samples$SequencingBatch))
  #dge$samples$Ind_SeqBatch <- paste0(dge$samples$Individual,"_",dge$samples$SequencingBatch)
  #dge$samples$Ind_SeqBatch <- as.factor(dge$samples$Ind_SeqBatch)
  design <- model.matrix(~Collection+cdr+nCount_RNA+percent.mt+Phase+Species, data=dge$samples)
  
  dge <- estimateDisp(dge, design=design)
  print(paste0("common dispersion: ",dge$common.dispersion))

  #QLF model fit (quasi-likelihood ratio test) - reflects uncertainty in estimating dispersion for each gene
  fit <- glmQLFit(dge, design=design, robust=TRUE)
  #head(fit$coefficients)
  gof(fit,plot=TRUE)

  #differential expression
  qlf <- glmQLFTest(fit, coef=dim(design)[2], poisson.bound=TRUE) #detect DE genes between species
  
  #assess output
  tt <- topTags(qlf, n=Inf, adjust.method="BH", p.value=1)
  print("All DE Genes")
  print(dim(tt$table))
  print(table(tt$table$FDR<0.01))
  print(summary(decideTests(qlf, adjust.method="BH", p.value=0.01)))
  
  #pvalue distribution
  hist(tt$table$PValue)
  hist(tt$table$FDR)
  hist(tt$table$logFC)
  
  #qqplot
  exp.pval<-(rank(tt$table$FDR, ties.method="first")+.5)/(length(tt$table$FDR)+1)
  plot(-log10(exp.pval), -log10(sort(tt$table$FDR)))
  abline(0,1)

  return(tt)

}

```

```{bash, eval=FALSE}

#run DEedgeR.sh and DEedgeR.R on cluster

cd /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/code/

sbatch ./DEedgeR.sh totcell.filter.pch20.fc.bh TRUE pch 0.2 TRUE BH stage
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.bh TRUE pch 0.2 TRUE BH cluster
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.bh TRUE pch 0.2 TRUE BH adhoc
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.bh TRUE pch 0.2 TRUE BH ostadhoc

sbatch ./DEedgeR.sh totcell.filter.pch20.fc.by TRUE pch 0.2 TRUE BY stage
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.by TRUE pch 0.2 TRUE BY cluster
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.by TRUE pch 0.2 TRUE BY adhoc
sbatch ./DEedgeR.sh totcell.filter.pch20.fc.by TRUE pch 0.2 TRUE BY ostadhoc

#previous tests
#sbatch ./DEedgeR.sh totcell.filter.mincount1.nofc.bh TRUE min.count 0.1 FALSE BH ostadhoc
#sbatch ./DEedgeR.sh totcell.filter.mincount1.nofc.by TRUE min.count 0.1 FALSE BY ostadhoc
#sbatch ./DEedgeR.sh totcell.filter.mincount1.fc.bh TRUE min.count 0.1 TRUE BH ostadhoc
#sbatch ./DEedgeR.sh totcell.filter.mincount1.fc.by TRUE min.count 0.1 TRUE BY ostadhoc
#sbatch ./DEedgeR.sh totcell.filter.pch20.nofc.bh TRUE pch 0.2 TRUE BH ostadhoc
#sbatch ./DEedgeR.sh totcell.filter.pch20.nofc.by TRUE pch 0.2 TRUE BY ostadhoc
#sbatch ./DEedgeR.fc.sh totcell.filter.pch20.fc.bh TRUE pch 0.2 TRUE BH ostadhoc
#sbatch ./DEedgeR.fc.sh totcell.filter.pch20.fc.by TRUE pch 0.2 TRUE BY ostadhoc




#OLD CODE BELOW...
#ONCE SATIFIED - DELETE CODE, OLD SCRIPTS, AND OUTPUT FILES

sbatch ./DEedgeR.sh totcell.filter.mincount1 TRUE min.count 0.1 ostadhoc
sbatch ./DEedgeR.sh totcell.filter.pch20 TRUE pch 0.2 ostadhoc
sbatch ./DEedgeR.fc.sh totcell.filter.mincount1 TRUE min.count 0.1 ostadhoc
sbatch ./DEedgeR.fc.sh totcell.filter.pch20 TRUE pch 0.2 ostadhoc

#RERUNN THESE WITH BY INSTEAD OF BH
sbatch ./DEedgeR.sh totcell.filter.mincount1.BY TRUE min.count 0.1 ostadhoc
sbatch ./DEedgeR.sh totcell.filter.pch20.BY TRUE pch 0.2 ostadhoc
sbatch ./DEedgeR.fc.sh totcell.filter.mincount1.BY TRUE min.count 0.1 ostadhoc
sbatch ./DEedgeR.fc.sh totcell.filter.pch20.BY TRUE pch 0.2 ostadhoc

#sbatch ./DEedgeR.sh totcell.filter.mincount05 TRUE min.count 0.05 stage  #TESTED
#sbatch ./DEedgeR.sh totcell.filter.mincount1 TRUE min.count 0.1 stage    #TESTED
#sbatch ./DEedgeR.sh totcell.filter.pch20 TRUE pch 0.2 stage              #
#sbatch ./DEedgeR.sh totcell.nofilter FALSE 0 0 stage                     #TESTED
#sbatch ./DEedgeR.fc.sh totcell.filtermincount1 TRUE min.count 0.1 stage  #
#sbatch ./DEedgeR.fc.sh totcell.filter.pch20 TRUE pch 0.2 stage           #
#sbatch ./DEedgeR.fc.sh totcell.nofilter FALSE 0 0 stage                  #

sbatch ./DEedgeR.sh totcell.filter05 TRUE 0.05 stage
sbatch ./DEedgeR.sh totcell.filter05 TRUE 0.05 cluster
sbatch ./DEedgeR.sh totcell.filter05 TRUE 0.05 adhoc
sbatch ./DEedgeR.sh totcell.filter05 TRUE 0.05 ostadhoc

sbatch ./DEedgeR.sh totcell.filter1 TRUE 0.1 stage
sbatch ./DEedgeR.sh totcell.filter1 TRUE 0.1 cluster
sbatch ./DEedgeR.sh totcell.filter1 TRUE 0.1 adhoc
sbatch ./DEedgeR.sh totcell.filter1 TRUE 0.1 ostadhoc

sbatch ./DEedgeR.sh totcell.nofilter FALSE 0 stage
sbatch ./DEedgeR.sh totcell.nofilter FALSE 0 cluster
sbatch ./DEedgeR.sh totcell.nofilter FALSE 0 adhoc
sbatch ./DEedgeR.sh totcell.nofilter FALSE 0 ostadhoc

```

Limma Voom Method (Dream)
https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_voomlimma.R
https://www.bioconductor.org/packages/devel/bioc/vignettes/variancePartition/inst/doc/dream.html#:~:text=1%3A1000%2C%5D-,Limma%20Analysis,the%20same%20across%20all%20genes.

Tests Adjusting Model Designs
[ using data subsetted to just cells assigned to Cluster iPSC.c3 ]
[ no gene filtering ]

* 10030/ 19221: voom+lmFit;                 model.matrix(~Collection+nCount_RNA+percent.mt+Phase+Species)
*   868/ 19221: voomWithDreamWeights+dream; model.matrix(~Collection+nCount_RNA+percent.mt+Phase+(1|Individual)+Species)
*   869/ 19221: voomWithDreamWeights+dream; model.matrix(~Collection+nCount_RNA+percent.mt+Phase+(1|Individual)+(1|Replicate)+Species)
*   658/ 19221: voomWithDreamWeights+dream; model.matrix(~nCount_RNA+percent.mt+Phase+(1|Collection)+(1|Individual)+(1|Replicate)+Species)

```{r}

runDream <- function(dataSub, cell.assign, cell.subset, genes.no.mito.ribo, filter.arg, min.count) {
  
  #count matrix
  counts <- as.matrix(GetAssayData(dataSub, assay="RNA", slot="counts"))

  #remove mitochodrial and ribosomal genes
  counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]

  #metadata
  metadata <- dataSub@meta.data[,c("Species","Collection","Individual","Replicate","nCount_RNA","percent.mt","Phase")]
  metadata <- metadata[colnames(counts),]
  
  #make edgeR object
  dge <- DGEList(counts)
  meta_dge <- dge$samples[,c("lib.size","norm.factors")]
  meta_dge <- cbind(meta_dge, metadata)
  dge$samples <- meta_dge
  rm(dataSub,counts,metadata,meta_dge)
  
  #filter genes
  if (filter.arg==TRUE){
    keep <- filterByExpr(dge, group=dge$samples$Species, min.count=min.count, min.total.count=15)
    table(keep)
    dge <- dge[keep, , keep.lib.sizes=FALSE]
    rm(keep)
  }
  
  #normalize data
  dge <- calcNormFactors(dge, method="TMM")
  summary(dge$samples$norm.factors)
  
  #design matrix
  dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
  dge$samples$Phase <- as.factor(dge$samples$Phase)
  dge$samples$Individual <- as.factor(dge$samples$Individual)
  dge$samples$Replicate <- as.factor(as.character(dge$samples$Replicate))
  dge$samples$Species <- as.factor(dge$samples$Species)
  model <- "~Collection+nCount_RNA+percent.mt+Phase+(1|Individual)+(1|Replicate)+Species"
  formula <- ~Collection+nCount_RNA+percent.mt+Phase+(1|Individual)+(1|Replicate)+Species
  
  #estimate weights using linear mixed model of dream (voom)
  vobjDream = voomWithDreamWeights(dge, formula, dge$samples)
  
  #define contrasts
  L = getContrast(vobjDream, formula, dge$samples, coefficient="SpeciesHuman")
  plotContrasts(L)

  #fit dream model on each gene
  fitmm = dream(vobjDream, formula, dge$samples, L)
  fitmm = dream(vobjDream, formula, dge$samples)

  #assess differential expression output for particular contrast
  tt <- topTable(fitmm, n=Inf, coef='SpeciesHuman', adjust.method="BH", p.value=1)
  print("All DE Genes")
  print(dim(tt))
  print(table(tt$adj.P.Val<0.01))
  print(summary(decideTests(fitmm, adjust.method="BH", p.value=0.01)))
  print(summary(decideTests(fitmm, adjust.method="BH", p.value=0.01, lfc=abs(1.2))))
  
  #assess variance partitioning
  formula.x = ~nCount_RNA+percent.mt+(1|Phase)+(1|Collection)+(1|Individual)+(1|Replicate)+(1|Species)
  vp = fitExtractVarPartModel(vobjDream, formula.x, dge$samples)
  plotVarPart(sortCols(vp))
  
  #add details to output
  tt$data <- rep(scrna, dim(tt$table)[1])
  tt$cell.assign <- rep(cell.assign, dim(tt$table)[1])
  tt$cell.subset <- rep(cell.subset, dim(tt$table)[1])
  tt$gene.filter <- rep(filter.arg, dim(tt$table)[1])
  tt$gene.filter.args <- rep(paste0("min.count=",min.count), dim(tt$table)[1])
  tt$model <- rep(model, dim(tt$table)[1])
  tt$comparison <- rep(tt$comparison, dim(tt$table)[1])
  tt$test <- rep(tt$test, dim(tt$table)[1])
  tt$adjust.method <- rep(tt$adjust.method, dim(tt$table)[1])

  return(tt)

}

```

```{r}

compare <- readRDS("./data/de-data/DEedgeR.fc.totcell.filter.pch20.BY.ostadhoc.rds")

table(compare$cell.subset)
table(compare$cell.subset[which(compare$FDR<0.01)])
table(compare$cell.subset[which(compare$FDR<0.01 & abs(compare$logFC)>1.2)])

compareSub <- compare[which(compare$cell.subset=="embedding osteoblast"),]
hist(compareSub$logFC)
hist(compareSub$FDR)
plot(compareSub$logFC,-log2(compareSub$FDR))

```

```{r}
variancePartition::dream(~Collection+nCount_RNA+percent.mt+Phase+(1|Individual)+(1|Replicate)+Species)
```



Examine output of DE analyses

```{r}

#cell assignment = stage
compare.stage <- readRDS("./data/de-data/DEedgeR.totcell.filter05.stage.rds")
table(compare.stage$cell.subset)
table(compare.stage$cell.subset[which(compare.stage$FDR<0.01)])
table(compare.stage$cell.subset[which(compare.stage$FDR<0.01 & abs(compare.stage$logFC)>2)])
#shared DE genes
compare.stage.tot <- intersect(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),]),
                               intersect(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 1"),]),
                                         rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 2"),])))
#DE genes unique to each stage
compare.stage.ips <- setdiff(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),]),
                             c(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 1"),]),
                               rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 2"),]))) 
compare.stage.msc <- setdiff(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 1"),]),
                             c(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),]),
                               rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 2"),]))) 
compare.stage.ost <- setdiff(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 2"),]),
                             c(rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),]),
                               rownames(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 1"),])))

DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),]),
                                   length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 1"),]),
                                   length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 2"),])),
                    Other=c((length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),])-
                               length(compare.stage.ips)-
                               length(compare.stage.tot)),
                            (length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),])-
                               length(compare.stage.msc)-
                               length(compare.stage.tot)),
                            (length(compare.stage[which(compare.stage$FDR<0.01 & compare.stage$cell.subset=="Time 0"),])-
                               length(compare.stage.ost)-
                               length(compare.stage.tot))),
                    Shared=c(rep(length(compare.stage.tot),3)),
                    Unique=c(length(compare.stage.ips),
                             length(compare.stage.msc),
                             length(compare.stage.ost)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))



#cell assignment = cluster
compare.cluster <- readRDS("./data/de-data/DEedgeR.totcell.filter05.cluster.rds")
table(compare.cluster$cell.subset)
table(compare.cluster$cell.subset[which(compare.cluster$FDR<0.01)])
table(compare.cluster$cell.subset[which(compare.cluster$FDR<0.01 & abs(compare.cluster$logFC)>2)])

#cell assignment = ad hoc
compare.adhoc <- readRDS("./data/de-data/DEedgeR.totcell.filter05.adhoc.rds")
table(compare.adhoc$cell.subset)
table(compare.adhoc$cell.subset[which(compare.adhoc$FDR<0.01)])
table(compare.adhoc$cell.subset[which(compare.adhoc$FDR<0.01 & abs(compare.adhoc$logFC)>2)])

#cell assignment = ost ad hoc
compare.ostadhoc <- readRDS("./data/de-data/DEedgeR.totcell.filter05.ostadhoc.rds")
table(compare.ostadhoc$cell.subset)
table(compare.ostadhoc$cell.subset[which(compare.ostadhoc$FDR<0.01)])
table(compare.ostadhoc$cell.subset[which(compare.ostadhoc$FDR<0.01 & abs(compare.ostadhoc$logFC)>2)])

```



Assess functional enrichment of DE genes

```{r}

#Define functions for functional enrichment assessment in differentially expressed genes
goCCenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "CC",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
goBPenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "BP",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
funcEnrich <- function(data, markers) {
  genes <- structure(rep(1, length(rownames(data@assays$RNA@data))), names=rownames(data@assays$RNA@data))
  x=1
  while (x <= length(rownames(markers))) {
    genes[which(names(genes)==rownames(markers)[x])] <- markers$p_val_adj[x]
    x=x+1
  }
  go_data <- new("topGOdata", ontology="BP", allGenes=genes, geneSel=function(allScore){return(allScore < 0.01)},
                 nodeSize=5, annotationFun=annFUN.org, mapping="org.Hs.eg.db", ID="symbol")
  go_test <- runTest(go_data, algorithm="weight01", statistic="fisher")
  go_table <- GenTable(go_data, weightFisher=go_test, orderBy="weightFisher", ranksOf="weightFisher", topNodes=sum(score(go_test) < .01))
  return(go_table)
}

```

```{r}

#Examine differentially expressed genes using differentiation stage at collection
compare.ips <- readRDS("./data/cellranger-data-full/de.w-compare.ips.rds")
compare.msc <- readRDS("./data/cellranger-data-full/de.w-compare.msc.rds")
compare.ost <- readRDS("./data/cellranger-data-full/de.w-compare.ost.rds")
dim(compare.ips)
dim(compare.msc)
dim(compare.ost)
table(compare.ips$p_val_adj<0.05)
table(compare.msc$p_val_adj<0.05)
table(compare.ost$p_val_adj<0.05)
head(compare.ips, n=5)
head(compare.msc, n=5)
head(compare.ost, n=5)
compare.ips$gene <- rownames(compare.ips)
compare.msc$gene <- rownames(compare.msc)
compare.ost$gene <- rownames(compare.ost)
fracRibo(compare.ips)
fracRibo(compare.msc)
fracRibo(compare.ost)
p.ips <- goCCenrich(indv,compare.ips)
p.msc <- goCCenrich(indv,compare.msc)
p.ost <- goCCenrich(indv,compare.ost)
p.ips[[3]]
p.msc[[3]]
p.ost[[3]]
#go.ips <- funcEnrich(indv, compare.ips)
#go.msc <- funcEnrich(indv, compare.msc)
#go.ost <- funcEnrich(indv, compare.ost)
#dim(go.ips)
#dim(go.msc)
#dim(go.ost)
#head(go.ips, n=5)
#head(go.msc, n=5)
#head(go.ost, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


#Examine differentially expressed genes using cluster definitions
compare.i <- readRDS("./data/cellranger-data-full/de.w-compare.i.rds")
compare.m <- readRDS("./data/cellranger-data-full/de.w-compare.m.rds")
compare.o <- readRDS("./data/cellranger-data-full/de.w-compare.o.rds")
compare.i1 <- readRDS("./data/cellranger-data-full/de.w-compare.i1.rds")
compare.i2 <- readRDS("./data/cellranger-data-full/de.w-compare.i2.rds")
compare.i3 <- readRDS("./data/cellranger-data-full/de.w-compare.i3.rds")
compare.o1 <- readRDS("./data/cellranger-data-full/de.w-compare.o1.rds")
compare.o2 <- readRDS("./data/cellranger-data-full/de.w-compare.o2.rds")
compare.o3 <- readRDS("./data/cellranger-data-full/de.w-compare.o3.rds")
dim(compare.i)
dim(compare.m)
dim(compare.o) 
dim(compare.i1)
dim(compare.i2)
dim(compare.i3)
dim(compare.o1)
dim(compare.o2)
dim(compare.o3)
table(compare.i$p_val_adj<0.05)
table(compare.m$p_val_adj<0.05)
table(compare.o$p_val_adj<0.05)
table(compare.i1$p_val_adj<0.05)
table(compare.i2$p_val_adj<0.05)
table(compare.i3$p_val_adj<0.05)
table(compare.o1$p_val_adj<0.05)
table(compare.o2$p_val_adj<0.05)
table(compare.o3$p_val_adj<0.05)
head(compare.i, n=5)
head(compare.m, n=5)
head(compare.o, n=5)
head(compare.i1, n=5)
head(compare.i2, n=5)
head(compare.i3, n=5)
head(compare.o1, n=5)
head(compare.o2, n=5)
head(compare.o3, n=5)
compare.i$gene <- rownames(compare.i)
compare.m$gene <- rownames(compare.m)
compare.o$gene <- rownames(compare.o)
compare.i1$gene <- rownames(compare.i1)
compare.i2$gene <- rownames(compare.i2)
compare.i3$gene <- rownames(compare.i3)
compare.o1$gene <- rownames(compare.o1)
compare.o2$gene <- rownames(compare.o2)
compare.o3$gene <- rownames(compare.o3)
fracRibo(compare.i)
fracRibo(compare.m)
fracRibo(compare.o)
fracRibo(compare.i1)
fracRibo(compare.i2)
fracRibo(compare.i3)
fracRibo(compare.o1)
fracRibo(compare.o2)
fracRibo(compare.o3)
p.i <- goCCenrich(indv,compare.i)
p.m <- goCCenrich(indv,compare.m)
p.o <- goCCenrich(indv,compare.o)
p.i1 <- goCCenrich(indv,compare.i1)
p.i2 <- goCCenrich(indv,compare.i2)
p.i3 <- goCCenrich(indv,compare.i3)
p.o1 <- goCCenrich(indv,compare.o1)
p.o2 <- goCCenrich(indv,compare.o2)
p.o3 <- goCCenrich(indv,compare.o3)
p.i[[3]]
p.m[[3]]
p.o[[3]]
p.i1[[3]]
p.i2[[3]]
p.i3[[3]]
p.o1[[3]]
p.o2[[3]]
p.o3[[3]]
#go.i <- funcEnrich(indv, compare.i)
#go.m <- funcEnrich(indv, compare.m)
#go.o <- funcEnrich(indv, compare.o)
#go.i1 <- funcEnrich(indv, compare.i1)
#go.i2 <- funcEnrich(indv, compare.i2)
#go.i3 <- funcEnrich(indv, compare.i3)
#go.o1 <- funcEnrich(indv, compare.o1)
#go.o2 <- funcEnrich(indv, compare.o2)
#go.o3 <- funcEnrich(indv, compare.o3)
#dim(go.i)
#dim(go.m)
#dim(go.o)
#dim(go.i1)
#dim(go.i2)
#dim(go.i3)
#dim(go.o1)
#dim(go.o2)
#dim(go.o3)
#head(go.i, n=5)
#head(go.m, n=5)
#head(go.o, n=5)
#head(go.i1, n=5)
#head(go.i2, n=5)
#head(go.i3, n=5)
#head(go.o1, n=5)
#head(go.o2, n=5)
#head(go.o3, n=5)
compare.tot2 <- intersect(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                          intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                    rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #shared DE genes
compare.i.tot <- setdiff(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],c(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to i
compare.m.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to m
compare.o.tot <- setdiff(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.m)[which(compare.m$p_val_adj<0.05)])) #DE unique to o
compare.tot3 <- intersect(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],
                          intersect(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                    intersect(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                              intersect(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                        intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                  intersect(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                            intersect(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                      rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)]))))))) #shared DE
compare.i1.tot <- setdiff(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],c(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i1
compare.i2.tot <- setdiff(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i2
compare.i3.tot <- setdiff(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i3
compare.m1.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                 rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                 rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                 rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                 rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                 rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to m1
compare.o1.tot <- setdiff(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o1
compare.o2.tot <- setdiff(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o2
compare.o3.tot <- setdiff(rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)])) #DE unique to o3
length(compare.tot2)
length(compare.i.tot)
length(compare.m.tot)
length(compare.o.tot)
length(compare.tot3)
length(compare.i1.tot)
length(compare.i2.tot)
length(compare.i3.tot)
length(compare.m1.tot)
length(compare.o1.tot)
length(compare.o2.tot)
length(compare.o3.tot)
head(compare.i[which(rownames(compare.i)%in%compare.tot2),], n=5)
head(compare.i[which(rownames(compare.i)%in%compare.i.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m.tot),], n=5)
head(compare.o[which(rownames(compare.o)%in%compare.o.tot),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.tot3),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),], n=5)
head(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),], n=5)
head(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m1.tot),], n=5)
head(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),], n=5)
head(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),], n=5)
head(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),], n=5)
fracRibo(compare.i[which(rownames(compare.i)%in%compare.tot2),])
fracRibo(compare.i[which(rownames(compare.i)%in%compare.i.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m.tot),])
fracRibo(compare.o[which(rownames(compare.o)%in%compare.o.tot),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
fracRibo(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
fracRibo(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
fracRibo(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
fracRibo(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
fracRibo(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2 <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goCCenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goCCenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goCCenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goCCenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goCCenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goCCenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
p.tot2 <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goBPenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goBPenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goBPenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goBPenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goBPenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goBPenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.i$gene[which(compare.i$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o$gene[which(compare.o$p_val_adj<0.05)])),
                    Other=c((length(compare.i$gene[which(compare.i$p_val_adj<0.05)])-length(compare.i.tot)-length(compare.tot2)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m.tot)-length(compare.tot2)),
                            (length(compare.o$gene[which(compare.o$p_val_adj<0.05)])-length(compare.o.tot)-length(compare.tot2))),
                    Shared=c(rep(length(compare.tot2),3)),
                    Unique=c(length(compare.i.tot),
                             length(compare.m.tot),
                             length(compare.o.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))
DE.df <- data.frame(cell=(c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3")),
                    totalDEgenes=c(length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)]),
                                   length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)]),
                                   length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)]),
                                   length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)]),
                                   length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])),
                    Other=c((length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)])-length(compare.i1.tot)-length(compare.tot3)),
                            (length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)])-length(compare.i2.tot)-length(compare.tot3)),
                            (length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)])-length(compare.i3.tot)-length(compare.tot3)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m1.tot)-length(compare.tot3)),
                            (length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)])-length(compare.o1.tot)-length(compare.tot3)),
                            (length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)])-length(compare.o2.tot)-length(compare.tot3)),
                            (length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])-length(compare.o3.tot)-length(compare.tot3))),
                    Shared=c(rep(length(compare.tot3),7)),
                    Unique=c(length(compare.i1.tot),
                             length(compare.i2.tot),
                             length(compare.i3.tot),
                             length(compare.m1.tot),
                             length(compare.o1.tot),
                             length(compare.o2.tot),
                             length(compare.o3.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[8:28,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))



#Find differentially expressed genes using ad hoc definitions
compare.hoci  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="iPSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoci, file="./data/cellranger-data-full/de-compare.hoci.rds")
compare.hocm  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="MSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hocm, file="./data/cellranger-data-full/de-compare.hocm.rds")
compare.hoco  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="Osteoblast"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoco, file="./data/cellranger-data-full/de-compare.hoco.rds")

compare.hoci <- readRDS("./data/cellranger-data-full/de.w-compare.hoci.rds")
compare.hocm <- readRDS("./data/cellranger-data-full/de.w-compare.hocm.rds")
compare.hoco <- readRDS("./data/cellranger-data-full/de.w-compare.hoco.rds")
dim(compare.hoci)
dim(compare.hocm)
dim(compare.hoco)
table(compare.hoci$p_val_adj<0.05)
table(compare.hocm$p_val_adj<0.05)
table(compare.hoco$p_val_adj<0.05)
head(compare.hoci, n=5)
head(compare.hocm, n=5)
head(compare.hoco, n=5)
compare.hoci$gene <- rownames(compare.hoci)
compare.hocm$gene <- rownames(compare.hocm)
compare.hoco$gene <- rownames(compare.hoco)
fracRibo(compare.hoci)
fracRibo(compare.hocm)
fracRibo(compare.hoco)
p.hoci <- goCCenrich(indv,compare.hoci)
p.hocm <- goCCenrich(indv,compare.hocm)
p.hoco <- goCCenrich(indv,compare.hoco)
p.hoci[[3]]
p.hocm[[3]]
p.hoco[[3]]
#go.hoci <- funcEnrich(indv, compare.hoci)
#go.hocm <- funcEnrich(indv, compare.hocm)
#go.hoco <- funcEnrich(indv, compare.hoco)
#dim(go.hoci)
#dim(go.hocm)
#dim(go.hoco)
#head(go.hoci, n=5)
#head(go.hocm, n=5)
#head(go.hoco, n=5)
compare.tot5 <- intersect(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],intersect(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                                rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)]))
compare.hoci.tot <- setdiff(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],c(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hoci
compare.hocm.tot <- setdiff(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hocm
compare.hoco.tot <- setdiff(rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)])) #DE hoco
length(compare.tot5)
length(compare.hoci.tot)
length(compare.hocm.tot)
length(compare.hoco.tot)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),], n=5)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),], n=5)
head(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),], n=5)
head(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),], n=5)
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
fracRibo(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
fracRibo(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5 <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goCCenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goCCenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
p.tot5 <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goBPenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goBPenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)]),
                                   length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)]),
                                   length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])),
                    Other=c((length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)])-length(compare.hoci.tot)-length(compare.tot5)),
                            (length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)])-length(compare.hocm.tot)-length(compare.tot5)),
                            (length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])-length(compare.hoco.tot)-length(compare.tot5))),
                    Shared=c(rep(length(compare.tot5),3)),
                    Unique=c(length(compare.hoci.tot),
                             length(compare.hocm.tot),
                             length(compare.hoco.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

#Examine differentially expressed genes using osteogenic ad hoc definitions
compare.pre <- readRDS("./data/cellranger-data-full/de.w-compare.pre.rds")
compare.obl <- readRDS("./data/cellranger-data-full/de.w-compare.obl.rds")
compare.emb <- readRDS("./data/cellranger-data-full/de.w-compare.emb.rds")
dim(compare.pre)
dim(compare.obl)
dim(compare.emb)
table(compare.pre$p_val_adj<0.05)
table(compare.obl$p_val_adj<0.05)
table(compare.emb$p_val_adj<0.05)
head(compare.pre, n=5)
head(compare.obl, n=5)
head(compare.emb, n=5)
compare.pre$gene <- rownames(compare.pre)
compare.obl$gene <- rownames(compare.obl)
compare.emb$gene <- rownames(compare.emb)
fracRibo(compare.pre)
fracRibo(compare.obl)
fracRibo(compare.emb)
p.pre <- goCCenrich(indv,compare.pre)
p.obl <- goCCenrich(indv,compare.obl)
p.emb <- goCCenrich(indv,compare.emb)
p.pre[[3]]
p.obl[[3]]
p.emb[[3]]
#go.pre <- funcEnrich(indv, compare.pre)
#go.obl <- funcEnrich(indv, compare.obl)
#go.emb <- funcEnrich(indv, compare.emb)
#dim(go.pre)
#dim(go.obl)
#dim(go.emb)
#head(go.pre, n=5)
#head(go.obl, n=5)
#head(go.emb, n=5)
compare.tot6 <- intersect(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                          intersect(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                    rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #shared DE genes
compare.pre.tot <- setdiff(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)], c(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                                                                       rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE pre
compare.obl.tot <- setdiff(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE obl
compare.emb.tot <- setdiff(rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)])) #DE emb
length(compare.tot6)
length(compare.pre.tot)
length(compare.obl.tot)
length(compare.emb.tot)
head(compare.pre[which(rownames(compare.pre)%in%compare.tot6),], n=5)
head(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),], n=5)
head(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),], n=5)
head(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),], n=5)
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
fracRibo(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
fracRibo(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6 <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goCCenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goCCenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
p.tot6 <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goBPenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goBPenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
DE.df <- data.frame(cell=(c("preosteoblast","osteoblast","embedding osteoblast")),
                    totalDEgenes=c(length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)]),
                                   length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)]),
                                   length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])),
                    Other=c((length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)])-length(compare.pre.tot)-length(compare.tot6)),
                            (length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)])-length(compare.obl.tot)-length(compare.tot6)),
                            (length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])-length(compare.emb.tot)-length(compare.tot6))),
                    Shared=c(rep(length(compare.tot6),3)),
                    Unique=c(length(compare.pre.tot),
                             length(compare.obl.tot),
                             length(compare.emb.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("preosteoblast","osteoblast","embedding osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

#Examine differentially expressed genes using grade of membership definitions
compare.gomi <- readRDS("./data/cellranger-data-full/de.w-compare.gomi.rds")
compare.gomm <- readRDS("./data/cellranger-data-full/de.w-compare.gomm.rds")
compare.gomo <- readRDS("./data/cellranger-data-full/de.w-compare.gomo.rds")
dim(compare.gomi)
dim(compare.gomm)
dim(compare.gomo)
table(compare.gomi$p_val_adj<0.05)
table(compare.gomm$p_val_adj<0.05)
table(compare.gomo$p_val_adj<0.05)
head(compare.gomi, n=5)
head(compare.gomm, n=5)
head(compare.gomo, n=5)
compare.gomi$gene <- rownames(compare.gomi)
compare.gomm$gene <- rownames(compare.gomm)
compare.gomo$gene <- rownames(compare.gomo)
fracRibo(compare.gomi)
fracRibo(compare.gomm)
fracRibo(compare.gomo)
p.gomi <- goCCenrich(indv,compare.gomi)
p.gomm <- goCCenrich(indv,compare.gomm)
p.gomo <- goCCenrich(indv,compare.gomo)
p.gomi[[3]]
p.gomm[[3]]
p.gomo[[3]]
#go.gomi <- funcEnrich(indv, compare.gomi)
#go.gomm <- funcEnrich(indv, compare.gomm)
#go.gomo <- funcEnrich(indv, compare.gomo)
#dim(go.gomi)
#dim(go.gomm)
#dim(go.gomo)
#head(go.gomi, n=5)
#head(go.gomm, n=5)
#head(go.gomo, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.o$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))

```



