---
title: "10x-plot-replicate-correlation-tot"
author: "Genevieve Housman"
date: "September 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Examine Data Reproducibility Across Technical Replicates

Replicate Correlation Findings:

* across stages of differentiation: Time 0 > Time 1 > Time 2
* across unsupervized clusters: iPSC.c1 ~ iPCS.c2 > iPSC.c3 > MSC.c1 > Osteogenic.c1 ~ Osteogenic.c2
* across general ad hoc assignments: iPSC > MSC > Osteogenic
* across osteogenic ad hoc assignments: preosteoblast ~ osteoblast ~ embedding osteoblast ~ mineralizing osteoblast ~ maturing osteocyte
* for osteogenic cells, osteogenic ad hoc assignments ~ general ad hoc assignments ~ unsupervised clusters > stages

```{r}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)
library(ggplot2)
library(cowplot)
theme_set(theme_cowplot())

```

Check correlations using tot dataset.

```{r}

#Load data
scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds" #integrate across individuals - total (19k genes)
data <- readRDS(scrna)
data

```

Log normalized single-cell data are compared across technical replicates. Log normalized data were processed using NormalizedData(). Feature counts within a cell were divided by the total counts for that cell, multiplied by 10000, and natural-log transformed (log1p).

### Time 0 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9919808

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9878611

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 0" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9934725

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Time 1 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9713058

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9826974

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9255991

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="mediumorchid", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Time 2 Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9494355

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9799395

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Stage=="Time 2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8697377

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="dodgerblue", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### iPSC (ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9890889

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9856296

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="iPSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9923132

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### MSC (ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9523304

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9792015

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="MSC" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9288491

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Osteogenic (ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.976839

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9860837

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & AdHoc.Assign.median=="Osteogenic" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9218739

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### iPSC.c1 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9918167

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9880422

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9935021

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### iPSC.c2 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9878809

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9738256

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9898509

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### iPSC.c3 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9668283

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9333088

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="iPSC.c3" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9552462

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### MSC.c1 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9720904

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9826386

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="MSC.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9279012

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Osteogenic.c1 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9396369

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9785212

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="Osteogenic.c1" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8669608

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Osteogenic.c2 (cluster) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9647557

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9822405

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & Cluster=="Osteogenic.c2" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8832882

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Preosteoblast (ost ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9817919

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.980863

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="preosteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9420394

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Osteoblast (ost ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9690247

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9721283

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.876973

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Embedding Osteoblast (ost ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9818969

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.985008

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="embedding osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.8984163

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Mineralizing Osteoblast (ost ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9484144

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9848053

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="mineralizing osteoblast" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9219273

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```

### Maturing Osteocyte (ost ad hoc) Replicates

Keep human and chimp samples together.

```{r}

rep1 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9609351

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1",y="Average Gene Expression in Replicate 2") +
  theme_classic()

```

Look at humans separately.

```{r}

rep1 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Human" & OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9427392

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Human)",y="Average Gene Expression in Replicate 2 (Human)") +
  theme_classic()

```

Look at chimps separately.

```{r}

rep1 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="1"))
Idents(rep1) <- "rep1"
avg.rep1 <- log1p(AverageExpression(rep1, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep1$gene <- rownames(avg.rep1)

rep2 <- subset(data, subset=(Species=="Chimp" & OstAdHoc.Assign.5gene.median=="maturing osteocyte" & Pair=="H1+C1" & Replicate=="2"))
Idents(rep2) <- "rep2"
avg.rep2 <- log1p(AverageExpression(rep2, assays="RNA", slot="data", verbose=FALSE)$RNA)
avg.rep2$gene <- rownames(avg.rep2)

rep.avg <- merge(avg.rep1, avg.rep2, by="gene")
cor(rep.avg$rep1, rep.avg$rep2, method="pearson") #pearson correlation = 0.9344628

ggplot(rep.avg, aes(rep1, rep2)) +
  geom_point(size=1) +
  geom_smooth(method=lm, se=FALSE, col="pink", size=2) +
  labs(x="Average Gene Expression in Replicate 1 (Chimp)",y="Average Gene Expression in Replicate 2 (Chimp)") +
  theme_classic()

```
