---
title: "10x-data-cormotif-tot-cnts-v2"
author: "Genevieve Housman"
date: "September 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Correlation Motifs

Test for differential expression correlation motifs between different cell classifications in humans and chimpanzees.

Standard Cormotif (see 10x-data-6-cormotif-tot):

* calculates t statistics from running limma on log-cpm data

Instead of doing this, here we used a modified version of Cormotif that Kenneth prepared (VERSION 2):

* adds a call to voom before running limma so that limma can use vooms precision weights
* then uses these t statistics for the rest of the analysis
* voom.normalize.method = "none" (same as for pairwise DE analyses)
* https://github.com/kennethabarr/CormotifCounts/blob/main/R/Cormotif.R

```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(Cormotif)
library(gplots)
library(ComplexHeatmap)

source("./code/CormotifCounts_v2.R")

```

## Prepare data

```{r, eval=FALSE}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

#Define individual-replicate sets
data@meta.data$Individual.Replicate <- paste0(data@meta.data$Individual,".",data@meta.data$Replicate)

```

Define genes of interest: all except mitochondrial genes and ribosomal genes

```{r, eval=FALSE}

genes <- rownames(data@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)

```

## Run Cormotif to Identify DE Genes

Cormotif is a Bayesian clustering approach that identifies common temporal expression patterns (referred to as “correlation motifs”) that best fit the given data.

Cormotif helps to overcome issues of incomplete power affecting naïve pairwise DE comparisons and helps to account for dependency in data coming from different time points (stages of differentiation, etc). This joint modeling technique leverages expression information shared across time points to identify correlation motifs.

NOTE: In this modified version of Cormotif, we use raw counts instead of normalized log2 transformed cpm.

Plots of Interest:

* probability that motif k is differentially expressed in study d (bestmotif$motif.q)
* posterior probabilities for each gene to be differentially expressed in each study for the best fitted model (bestmotif$p.post)

Assemble normalized gene counts and run cormotif.
https://www.biostars.org/p/317701/

## Cormotif using Pseudobulk Data

Pseudobulk: Species vs. Stages

* each time point separately
	+ genes examined: 11579
	+ best fit model: 2 motifs
	+ unique DE genes (p.prob>0.5): 2906 in Time 0, 101 in Time 1, 0 in Time 2
* 3 time points and 4 comparisons
	+ genes examined: 11579
	+ best fit model: 4 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 89.10% (10327 genes)
	+ unique chimp DE genes (p.prob>0.5): 13 in t0-t1,   0 in t1-t2
	+ unique human DE genes (p.prob>0.5):  0 in t0-t1, 237 in t1-t2
* 3 time points and 4 comparisons
	+ genes examined: 11579
	+ best fit model: 8 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 78.65% (9107 genes)
	+ unique chimp DE genes (p.prob>0.5): 526 in Time 0,  0 in Time 1,  9 in Time 2
	+ unique human DE genes (p.prob>0.5):  48 in Time 0, 22 in Time 1, 91 in Time 2

Pseudobulk: Species vs. General Clusters (res=0.05)

* each cluster separately
	+ genes examined: 10742
	+ best fit model: 5 motifs
	+ unique DE genes (p.prob>0.5): 966 in iPSC.c1, 0 in iPSC.c2, 0 in iPSC.c3, 255 in MSC.c1, 0 in Osteogenic.c1, 0 in Osteogenic.c2
* 3 clusters and 4 comparisons (A)
	+ genes examined: 10742
	+ best fit model: 4 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 84.92% (9122 genes)
	+ unique chimp DE genes (p.prob>0.5): 11 in iPSC.c1-MSC.c1,   0 in MSC.c1-Osteogenic.c1
	+ unique human DE genes (p.prob>0.5):  0 in iPSC.c1-MSC.c1, 216 in MSC.c1-Osteogenic.c1
* 3 clusters and 4 comparisons (B)
	+ genes examined: 10742
	+ best fit model: 4 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 56.39% (6057 genes)
	+ unique chimp DE genes (p.prob>0.5): 12 in iPSC.c1-MSC.c1,   0 in MSC.c1-Osteogenic.c2
	+ unique human DE genes (p.prob>0.5):  0 in iPSC.c1-MSC.c1, 454 in MSC.c1-Osteogenic.c1
* 4 clusters and 8 comparisons
	+ genes examined: 10742
	+ best fit model: 12 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 32.96% (3541 genes)
	+ unique chimp DE genes (p.prob>0.5): 28 in iPSC.c1-MSC.c1,  0 in MSC.c1-Osteogenic.c1,   0 in MSC.c1-Osteogenic.c2
	+ unique human DE genes (p.prob>0.5):  0 in iPSC.c1-MSC.c1, 46 in MSC.c1-Osteogenic.c1, 144 in MSC.c1-Osteogenic.c2
* 3 clusters and 6 comparisons (A)
	+ genes examined: 10742
	+ best fit model: 9 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 73.91% (7939 genes)
	+ unique chimp DE genes (p.prob>0.5): 466 in iPSC.c1,  0 in MSC.c1,  6 in Osteogenic.c1
	+ unique human DE genes (p.prob>0.5): 196 in iPSC.c1, 11 in MSC.c1, 27 in Osteogenic.c1
* 3 clusters and 6 comparisons (B)
	+ genes examined: 10742
	+ best fit model: 7 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 49.26% (5291 genes)
	+ unique chimp DE genes (p.prob>0.5): 351 in iPSC.c1,  0 in MSC.c1,   0 in Osteogenic.c1
	+ unique human DE genes (p.prob>0.5):  19 in iPSC.c1, 18 in MSC.c1, 399 in Osteogenic.c1

Pseudobulk: Species vs. General Ad Hoc Classifications (thresh0)

* each ad hoc classification separately
	+ genes examined: 11824
	+ best fit model: 2 motifs
	+ unique DE genes (p.prob>0.5): 3065 in iPSCs, 8 in MSCs, 0 in Osteogenic
* 3 ad hoc classifications and 4 comparisons
	+ genes examined: 11824
	+ best fit model: 2 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 29.18% (3450 genes)
	+ unique chimp DE genes (p.prob>0.5): 315 in iPSC-MSC,  11 in MSC-Osteogenic
	+ unique human DE genes (p.prob>0.5):   0 in iPSC-MSC, 292 in MSC-Osteogenic
* 3 ad hoc classifications and 6 comparisons
	+ genes examined: 11824
	+ best fit model: 5 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 22.45% (2654 genes)
	+ unique chimp DE genes (p.prob>0.5): 191 in iPSC, 414 in MSC, 31 in Osteogenic 
	+ unique human DE genes (p.prob>0.5):  31 in iPSC,   0 in MSC, 50 in Osteogenic

Pseudobulk: Species vs. Osteogenic Clusters (res=0.50)

* each cluster separately
	+ genes examined: 10671
	+ best fit model: 2 motifs
	+ unique DE genes (p.prob>0.5): 2 in O.c1, 0 in O.c2, 0 in O.c3, 2 in O.c4
* 4 osteogenic clusters and 6 comparisons
	+ genes examined: 10671
	+ best fit model: 4 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 74.30% (7929 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in O.c1-O.c2, 70 in O.c2-O.c3,   0 in O.c3-O.c4
	+ unique human DE genes (p.prob>0.5): 0 in O.c1-O.c2,  0 in O.c2-O.c3, 470 in O.c3-O.c4

Pseudobulk: Species vs. Osteogenic Clusters (res=0.75)

* each cluster separately
	+ genes examined: 10397
	+ best fit model: 3 motifs
	+ unique DE genes (p.prob>0.5): 0 in O.c1, 0 in O.c2, 2 in O.c3, 66 in O.c4, 0 in O.c5, 0 in O.c6, 0 in O.c7, 0 in O.c8
* 8 osteogenic clusters and 14 comparisons
	+ genes examined: 10397
	+ best fit model: 14 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 33.84% (3518 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in O.c1-O.c2,  0 in O.c2-O.c3, 2 in O.c3-O.c4, 0 in O.c4-O.c5, 13 in O.c5-O.c6, 7 in O.c6-O.c7, 113 in O.c7-O.c8
	+ unique human DE genes (p.prob>0.5): 0 in O.c1-O.c2, 17 in O.c2-O.c3, 3 in O.c3-O.c4, 0 in O.c4-O.c5,  0 in O.c5-O.c6, 0 in O.c6-O.c7,   5 in O.c7-O.c8

Pseudobulk: Species vs. Osteogenic Clusters (res=0.50) (Time 2 only)

* each cluster separately
	+ genes examined: 10554
	+ best fit model: 3 motifs
	+ unique DE genes (p.prob>0.5): 4 in O.c1, 0 in O.c2, 0 in O.c3, 0 in O.c4
* 4 osteogenic clusters and 6 comparisons
	+ genes examined: 10554
	+ best fit model: 4 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 71.28% (7523 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in O.c1-O.c2, 72 in O.c2-O.c3,   0 in O.c3-O.c4
	+ unique human DE genes (p.prob>0.5): 0 in O.c1-O.c2,  0 in O.c2-O.c3, 807 in O.c3-O.c4

Pseudobulk: Species vs. Osteogenic Clusters (res=0.75) (Time 2 only)

* each cluster separately
	+ genes examined: 10212
	+ best fit model: 3 motifs
	+ unique DE genes (p.prob>0.5): 0 in O.c1, 2 in O.c2, 0 in O.c3, 0 in O.c4, 0 in O.c5, 0 in O.c6, 0 in O.c7, 0 in O.c8
* 8 osteogenic clusters and 14 comparisons
	+ genes examined: 10212
	+ best fit model: 15 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 31.99% (3267 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in O.c1-O.c2, 0 in O.c2-O.c3,  0 in O.c3-O.c4, 0 in O.c4-O.c5, 215 in O.c5-O.c6,  0 in O.c6-O.c7, 211 in O.c7-O.c8
	+ unique human DE genes (p.prob>0.5): 0 in O.c1-O.c2, 4 in O.c2-O.c3, 13 in O.c3-O.c4, 0 in O.c4-O.c5,   0 in O.c5-O.c6, 46 in O.c6-O.c7,   4 in O.c7-O.c8

Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0)

* each osteogenic ad hoc classification separately
	+ genes examined: 10669
	+ best fit model: 2 motifs
	+ unique DE genes (p.prob>0.5): 0 in Pre, 0 in Obl, 0 in Emb, 6 in Min, 0 in Mat
* 5 osteogenic ad hoc classifications and 8 comparisons
	+ genes examined: 10669
	+ best fit model: 1 motif
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 100.00% (10669 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat
	+ unique human DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat

Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0)

* each osteogenic ad hoc classification separately
	+ genes examined: 9999
	+ best fit model: 3 motifs
	+ unique DE genes (p.prob>0.5): 0 in Pre, 5 in Obl, 0 in Emb, 0 in Min, 0 in Mat
* 5 osteogenic ad hoc classifications and 8 comparisons
	+ genes examined: 9999
	+ best fit model: 1 motif
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 100.00% (999\89 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat
	+ unique human DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat

Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0) (Time 2 only)

* each osteogenic ad hoc classification separately
	+ genes examined: 10822
	+ best fit model: 2 motifs
	+ unique DE genes (p.prob>0.5): 0 in Pre, 0 in Obl, 0 in Emb, 52 in Min, 0 in Mat
* 5 osteogenic ad hoc classifications and 8 comparisons
	+ genes examined: 10822
	+ best fit model: 1 motif
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 100.00% (10822 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat
	+ unique human DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat

Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0) (Time 2 only)

* each osteogenic ad hoc classification separately
	+ genes examined: 10535
	+ best fit model: 3 motifs
	+ unique DE genes (p.prob>0.5): 2 in Pre, 4 in Obl, 0 in Emb, 0 in Min, 0 in Mat
* 5 osteogenic ad hoc classifications and 8 comparisons
	+ genes examined: 10535
	+ best fit model: 1 motifs
	+ conservation across stages (species difference in gene posterior probabilities <= 0.3): 100.00% (10535 genes)
	+ unique chimp DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat
	+ unique human DE genes (p.prob>0.5): 0 in Pre-Obl, 0 in Obl-Emb, 0 in Emb-Min, 0 in Min-Mat

### Pseudobulk: Species vs. Stages

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Time 0","Time 1","Time 2")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg <- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Stage")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.stg)

```

#### each time point separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(" ","",x)),sep = "."),
                      levels=c("Human.Time0","Human.Time1","Human.Time2","Chimp.Time0","Chimp.Time1","Chimp.Time2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID <- data.frame(c1 = c(1, 2, 3),
                        c2 = c(4, 5, 6))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Stage=c("Time 0","Time 1","Time 2"),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("Time.0","Time.1","Time.2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("Time.0","Time.1","Time.2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each time point
prob.motifT0  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5),])
length(prob.motifT0)
head(prob.motifT0)
#boxplot(tmm[prob.motifT0[1],]~groupFactor)

prob.motifT1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5),])
length(prob.motifT1)
head(prob.motifT1)
#boxplot(tmm[prob.motifT1[1],]~groupFactor)

prob.motifT2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5),])
length(prob.motifT2)
head(prob.motifT2)
#boxplot(tmm[prob.motifT2[1],]~groupFactor)

# Examine human/chimp differences shared across time points
prob.motifT012  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5),])
length(prob.motifT012)
head(prob.motifT012)
#boxplot(tmm[prob.motifT012[1],]~groupFactor)

prob.motifT01  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5),])
length(prob.motifT01)
head(prob.motifT01)
#boxplot(tmm[prob.motifT01[1],]~groupFactor)

prob.motifT02  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5),])
length(prob.motifT02)
head(prob.motifT02)
#boxplot(tmm[prob.motifT02[1],]~groupFactor)

prob.motifT12  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5),])
length(prob.motifT12)
head(prob.motifT12)
#boxplot(tmm[prob.motifT12[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-stage.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifT0,prob.motifT1,prob.motifT2)
rm(prob.motifT012,prob.motifT01,prob.motifT02,prob.motifT12)

```

#### 3 time points and 4 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(" ","",x)),sep = "."),
                      levels=c("Human.Time0","Human.Time1","Human.Time2","Chimp.Time0","Chimp.Time1","Chimp.Time2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID <- data.frame(c1 = c(1, 2, 4, 5),
                        c2 = c(2, 3, 5, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("Time 0","Time 1","Time 0","Time 1"),
                                       c("Time 1","Time 2","Time 1","Time 2")),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.t0-t1","H.t1-t2","C.t0-t1","C.t1-t2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.t0-t1","H.t1-t2","C.t0-t1","C.t1-t2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10327 genes) -> 89.19%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (8724 genes) -> 75.34%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (3429 genes) -> 29.61%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-stage4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)

```

#### 3 time points and 6 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Stage,function(x)sub(" ","",x)),sep = "."),
                      levels=c("Human.Time0","Human.Time1","Human.Time2","Chimp.Time0","Chimp.Time1","Chimp.Time2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID <- data.frame(c1 = c(1, 2, 1, 4, 5, 4),
                        c2 = c(2, 3, 3, 5, 6, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("Time 0","Time 1","Time 0","Time 0","Time 1","Time 0"),
                                       c("Time 1","Time 2","Time 2","Time 1","Time 2","Time 2")),
                           col=list(Stage=c("Time 0"="pink","Time 1"="mediumorchid","Time 2"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.t0-t1","H.t1-t2","H.t0-t2","C.t0-t1","C.t1-t2","C.t0-t2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.t0-t1","H.t1-t2","H.t0-t2","C.t0-t1","C.t1-t2","C.t0-t2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9107 genes) -> 78.65%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (7535 genes) -> 65.07%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (3577 genes) -> 30.89%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]>0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-stage6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. General Clusters (res=0.05)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=Cluster0.05==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt <- dataSub@meta.data$Cluster0.05[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Cluster")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)

```

#### each cluster separately

Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, iPSC.c2, iPSC.c3, MSC.c1, Osteogenic.c1, and Osteogenic.c2 for each species)
compareID <- data.frame(c1 = c(1, 2, 3,  4,  5,  6),
                        c2 = c(7, 8, 9, 10, 11, 12))

# Assess differences within each cluster
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Cluster=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "iPSC.c2"="pink4",
                                              "iPSC.c3"="grey",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        column_split=c(1,1,1,2,2,2),
        #column_split=c(1,1,2,3,3,4),
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each cluster
prob.motifI1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifI1)
head(prob.motifI1)
#boxplot(tmm[prob.motifI1[1],]~groupFactor)

prob.motifI2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifI2)
head(prob.motifI2)
#boxplot(tmm[prob.motifI2[1],]~groupFactor)

prob.motifI3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifI3)
head(prob.motifI3)
#boxplot(tmm[prob.motifI3[1],]~groupFactor)

prob.motifM1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifM1)
head(prob.motifM1)
#boxplot(tmm[prob.motifM1[1],]~groupFactor)

prob.motifO1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]<0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]>0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifI123M1O12  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifI123M1O12)
head(prob.motifI123M1O12)
#boxplot(tmm[prob.motifI123M1O12[1],]~groupFactor)

prob.motifI123M1     <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifI123M1)
head(prob.motifI123M1)
#boxplot(tmm[prob.motifI123M1[1],]~groupFactor)

prob.motifI123O12    <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifI123O12)
head(prob.motifI123O12)
#boxplot(tmm[prob.motifI123O12[1],]~groupFactor)

prob.motifM1O12     <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifM1O12)
head(prob.motifM1O12)
#boxplot(tmm[prob.motifM1O12[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifI1,prob.motifI2,prob.motifI3,prob.motifM1,prob.motifO1,prob.motifO2)
rm(prob.motifI123M1O12,prob.motifI123M1,prob.motifI123O12,prob.motifM1O12)

```

#### 3 clusters and 4 comparisons (A)

iPSC.c2, iPSC.c3, and Osteogenic.c2 are excluded here.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c1 for each species)
compareID <- data.frame(c1 = c(1, 4,  7, 10),
                        c2 = c(4, 5, 10, 11))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","MSC.c1"),
                                         c("MSC.c1","Osteogenic.c1","MSC.c1","Osteogenic.c1")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Clusters",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c1: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9122 genes) -> 84.92%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (7461 genes) -> 69.46%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2558 genes) -> 23.81%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4a.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)

```

#### 3 clusters and 4 comparisons (B)

iPSC.c2, iPSC.c3, and Osteogenic.c1 are excluded here.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c2 for each species)
compareID <- data.frame(c1 = c(1, 4,  7, 10),
                        c2 = c(4, 6, 10, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","MSC.c1"),
                                         c("MSC.c1","Osteogenic.c2","MSC.c1","Osteogenic.c2")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Clusters",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c2","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c2","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (6057 genes) -> 56.39%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (4243 genes) -> 39.50%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 ( 771 genes) ->  7.18%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster4b.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)

```

#### 4 clusters and 8 comparisons

Only iPSC.c2 and iPSC.c3 are excluded here.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, Osteogenic.c1, and Osteogenic.c2 for each species)
compareID <- data.frame(c1 = c(1, 4, 4, 5,  7, 10, 10, 11),
                        c2 = c(4, 5, 6, 6, 10, 11, 12, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:20,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 12 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human","Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","MSC.c1","Osteogenic.c1","iPSC.c1","MSC.c1","MSC.c1","Osteogenic.c1"),
                                         c("MSC.c1","Osteogenic.c1","Osteogenic.c2","Osteogenic.c2","MSC.c1","Osteogenic.c1","Osteogenic.c2","Osteogenic.c2")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Clusters",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","H.MSC.c1-Osteogenic.c2","H.Osteogenic.c1-Osteogenic.c2",
                         "C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1","C.MSC.c1-Osteogenic.c2","C.Osteogenic.c1-Osteogenic.c2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","H.MSC.c1-Osteogenic.c2","H.Osteogenic.c1-Osteogenic.c2",
                                             "C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1","C.MSC.c1-Osteogenic.c2","C.Osteogenic.c1-Osteogenic.c2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, Osteogenic.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (3541 genes) -> 32.96%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (3381 genes) -> 31.47%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.1 (2391 genes) -> 22.26%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) & (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster8.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)

```

#### 3 clusters and 6 comparisons (A)

iPSC.c2, iPSC.c3, and Osteogenic.c2 are excluded here.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c1 for each species)
compareID <- data.frame(c1 = c(1, 4, 1,  7, 10,  7),
                        c2 = c(4, 5, 5, 10, 11, 11))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 9 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","iPSC.c1","MSC.c1","iPSC.c1"),
                                         c("MSC.c1","Osteogenic.c1","Osteogenic.c1","MSC.c1","Osteogenic.c1","Osteogenic.c1")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c1"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","H.iPSC.c1-Osteogenic.c1","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1","C.iPSC.c1-Osteogenic.c1")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c1","H.iPSC.c1-Osteogenic.c1","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c1","C.iPSC.c1-Osteogenic.c1")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7939 genes) -> 73.91%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (6162 genes) -> 57.36%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2554 genes) -> 23.78%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]>0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-cluster6a.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)

```

#### 3 clusters and 6 comparisons (B)

iPSC.c2, iPSC.c3, and Osteogenic.c1 are excluded here.

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_iPSC.c1","Human_iPSC.c2","Human_iPSC.c3","Human_MSC.c1","Human_Osteogenic.c1","Human_Osteogenic.c2",
                               "Chimp_iPSC.c1","Chimp_iPSC.c2","Chimp_iPSC.c3","Chimp_MSC.c1","Chimp_Osteogenic.c1","Chimp_Osteogenic.c2"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = iPSC.c1, MSC.c1, and Osteogenic.c2 for each species)
compareID <- data.frame(c1 = c(1, 4, 1,  7, 10,  7),
                        c2 = c(4, 6, 6, 10, 12, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 7 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("iPSC.c1","MSC.c1","iPSC.c1","iPSC.c1","MSC.c1","iPSC.c1"),
                                         c("MSC.c1","Osteogenic.c2","Osteogenic.c2","MSC.c1","Osteogenic.c2","Osteogenic.c2")),
                           col=list(Cluster=c("iPSC.c1"="pink",
                                              "MSC.c1"="mediumorchid",
                                              "Osteogenic.c2"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c2","H.iPSC.c1-Osteogenic.c2","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c2","C.iPSC.c1-Osteogenic.c2")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC.c1-MSC.c1","H.MSC.c1-Osteogenic.c2","H.iPSC.c1-Osteogenic.c2","C.iPSC.c1-MSC.c1","C.MSC.c1-Osteogenic.c2","C.iPSC.c1-Osteogenic.c2")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for iPSC.c1, MSC.c1, and Osteogenic.c2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (5291 genes) -> 49.26%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (3615 genes) -> 33.65%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (1204 genes) -> 11.21%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]>0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.sc-cluster6b.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. General Ad Hoc Classifications (thresh0)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("iPSC","MSC","Osteogenic","other")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=AdHoc.Assign.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- as.character(dataSub@meta.data$AdHoc.Assign.thresh0[w][1])
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

#### each ad hoc classification separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = "_"),
                      levels=c("Human_iPSC","Human_MSC","Human_Osteogenic","Human_other","Chimp_iPSC","Chimp_MSC","Chimp_Osteogenic","Chimp_other"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each ad hoc assignment)
compareID <- data.frame(c1 = c(1, 2, 3),
                        c2 = c(5, 6, 7))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(AdHoc=c("iPSC","MSC","Osteogenic"),
                           col=list(AdHoc=c("iPSC"="pink","MSC"="mediumorchid","Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("iPSC","MSC","Osteogenic")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("iPSC","MSC","Osteogenic")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each time point
prob.motifI  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5),])
length(prob.motifI)
head(prob.motifI)
#boxplot(tmm[prob.motifI[1],]~groupFactor)

prob.motifM  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5),])
length(prob.motifM)
head(prob.motifM)
#boxplot(tmm[prob.motifM[1],]~groupFactor)

prob.motifO  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5),])
length(prob.motifO)
head(prob.motifO)
#boxplot(tmm[prob.motifO[1],]~groupFactor)

# Examine human/chimp differences shared across time points
prob.motifIMO  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5),])
length(prob.motifIMO)
head(prob.motifIMO)
#boxplot(tmm[prob.motifIMO[1],]~groupFactor)

prob.motifIM  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5),])
length(prob.motifIM)
head(prob.motifIM)
#boxplot(tmm[prob.motifIM[1],]~groupFactor)

prob.motifIO  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5),])
length(prob.motifIO)
head(prob.motifIO)
#boxplot(tmm[prob.motifIO[1],]~groupFactor)

prob.motifMO  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5),])
length(prob.motifMO)
head(prob.motifMO)
#boxplot(tmm[prob.motifMO[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifI,prob.motifM,prob.motifO)
rm(prob.motifIMO,prob.motifIM,prob.motifIO,prob.motifMO)

```

#### 3 ad hoc classifications and 4 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = "_"),
                      levels=c("Human_iPSC","Human_MSC","Human_Osteogenic","Human_other","Chimp_iPSC","Chimp_MSC","Chimp_Osteogenic","Chimp_other"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID <- data.frame(c1 = c(1, 2, 4, 5),
                        c2 = c(2, 3, 5, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("iPSC","MSC","iPSC","MSC"),
                                       c("MSC","Osteogenic","MSC","Osteogenic")),
                           col=list(Stage=c("iPSC"="pink",
                                            "MSC"="mediumorchid",
                                            "Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC-MSC","H.MSC-Osteogenic","C.iPSC-MSC","C.MSC-Osteogenic")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC-MSC","H.MSC-Osteogenic","C.iPSC-MSC","C.MSC-Osteogenic")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (3450 genes) -> 29.18%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (1933 genes) -> 16.35%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 ( 763 genes) ->  6.45%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,3]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,4]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc4.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC1,prob.motifC2)

```

#### 3 ad hoc classifications and 6 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,dge$samples$AdHoc,sep = "_"),
                      levels=c("Human_iPSC","Human_MSC","Human_Osteogenic","Human_other","Chimp_iPSC","Chimp_MSC","Chimp_Osteogenic","Chimp_other"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = time0, time1, and time2 for each species)
compareID <- data.frame(c1 = c(1, 2, 1, 4, 5, 4),
                        c2 = c(2, 3, 3, 5, 6, 6))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 5 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Stage=cbind(c("iPSC","MSC","iPSC","iPSC","MSC","iPSC"),
                                       c("MSC","Osteogenic","Osteogenic","MSC","Osteogenic","Osteogenic")),
                           col=list(Stage=c("iPSC"="pink",
                                            "MSC"="mediumorchid",
                                            "Osteogenic"="dodgerblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.iPSC-MSC","H.MSC-Osteogenic","H.iPSC-Osteogenic","C.iPSC-MSC","C.MSC-Osteogenic","C.iPSC-Osteogenic")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.iPSC-MSC","H.MSC-Osteogenic","H.iPSC-Osteogenic","C.iPSC-MSC","C.MSC-Osteogenic","C.iPSC-Osteogenic")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for time 0-2: posterior probability difference between the humans and the chimps within a motif 
#0.3 (2654 genes) -> 22.45%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (1500 genes) -> 12.69%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 ( 505 genes) ->  4.27%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) & (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) & (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC0  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5 & gene.prob[,6]>0.5),])
length(prob.motifC0)
head(prob.motifC0)
#boxplot(tmm[prob.motifC0[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5 & gene.prob[,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5 & gene.prob[,6]>0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH0  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH0)
head(prob.motifH0)
#boxplot(tmm[prob.motifH0[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5 & gene.prob[,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-adhoc6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH0,prob.motifH1,prob.motifH2)
rm(prob.motifC,prob.motifC0,prob.motifC1,prob.motifC2)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Clusters (res=0.50)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstCluster0.50==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt <- dataSub@meta.data$OstCluster0.50[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Cluster")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)

```

#### each cluster separately

Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.

```{r, eval=FALSE}

# Compile group labels of each sample (clusters are off by 1)
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID <- data.frame(c1 = c(1, 2, 3, 4),
                        c2 = c(5, 6, 7, 8))

# Assess differences within each cluster
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:20,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each cluster
prob.motifO1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO1234  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO1234)
head(prob.motifO1234)
#boxplot(tmm[prob.motifO1234[1],]~groupFactor)

prob.motifO134  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO134)
head(prob.motifO134)
#boxplot(tmm[prob.motifO134[1],]~groupFactor)

prob.motifO13  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifO13)
head(prob.motifO13)
#boxplot(tmm[prob.motifO13[1],]~groupFactor)

prob.motifO34  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO34)
head(prob.motifO34)
#boxplot(tmm[prob.motifO34[1],]~groupFactor)

prob.motifO14  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifO14)
head(prob.motifO14)
#boxplot(tmm[prob.motifO14[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4)
rm(prob.motifO1234,prob.motifO134,prob.motifO13,prob.motifO34,prob.motifO14)

```

#### 4 osteogenic clusters and 6 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 5, 6, 7),
                        c2 = c(2, 3, 4, 6, 7, 8))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3",
                                           "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                                         c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                           "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.O1-O2","H.O2-O3","H.O3-O4",
                         "C.O1-O2","C.O2-O3","C.O3-O4")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.O1-O2","H.O2-O3","H.O3-O4",
                                             "C.O1-O2","C.O2-O3","C.O3-O4")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7929 genes) -> 74.30%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (6427 genes) -> 60.23%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2219 genes) -> 20.79%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]>0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]>0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Clusters (res=0.75)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                 "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstCluster0.75==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt <- dataSub@meta.data$OstCluster0.75[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Cluster")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)

```

#### each cluster separately

Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.

```{r, eval=FALSE}

# Compile group labels of each sample (clusters are off by 1)
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Human_Osteogenic.c5","Human_Osteogenic.c6","Human_Osteogenic.c7","Human_Osteogenic.c8",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4",
                               "Chimp_Osteogenic.c5","Chimp_Osteogenic.c6","Chimp_Osteogenic.c7","Chimp_Osteogenic.c8"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID <- data.frame(c1 = c(1,  2,  3,  4,  5,  6,  7,  8),
                        c2 = c(9, 10, 11, 12, 13, 14, 15, 16))

# Assess differences within each cluster
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                     "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="dodgerblue4",
                                              "Osteogenic.c3"="deepskyblue2",
                                              "Osteogenic.c4"="navyblue",
                                              "Osteogenic.c5"="lightblue1",
                                              "Osteogenic.c6"="royalblue3",
                                              "Osteogenic.c7"="grey",
                                              "Osteogenic.c8"="black")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each cluster
prob.motifO1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

prob.motifO5  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO5)
head(prob.motifO5)
#boxplot(tmm[prob.motifO5[1],]~groupFactor)

prob.motifO6  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO6)
head(prob.motifO6)
#boxplot(tmm[prob.motifO6[1],]~groupFactor)

prob.motifO7  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifO7)
head(prob.motifO7)
#boxplot(tmm[prob.motifO7[1],]~groupFactor)

prob.motifO8  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifO8)
head(prob.motifO8)
#boxplot(tmm[prob.motifO8[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO12345678  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                            gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifO12345678)
head(prob.motifO12345678)
#boxplot(tmm[prob.motifO12345678[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3)
rm(prob.motifO12345678)

```

#### 8 osteogenic clusters and 14 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Human_Osteogenic.c5","Human_Osteogenic.c6","Human_Osteogenic.c7","Human_Osteogenic.c8",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4",
                               "Chimp_Osteogenic.c5","Chimp_Osteogenic.c6","Chimp_Osteogenic.c7","Chimp_Osteogenic.c8"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 5, 6, 7,  9, 10, 11, 12, 13, 14, 15),
                        c2 = c(2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:20,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 14 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7",
                                         "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7"),
                                       c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8",
                                         "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")),
                           col=list(AdHoc=c("Osteogenic.c1"="dodgerblue",
                                            "Osteogenic.c2"="dodgerblue4",
                                            "Osteogenic.c3"="deepskyblue2",
                                            "Osteogenic.c4"="navyblue",
                                            "Osteogenic.c5"="lightblue1",
                                            "Osteogenic.c6"="royalblue3",
                                            "Osteogenic.c7"="grey",
                                            "Osteogenic.c8"="black")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.O1-O2","H.O2-O3","H.O3-O4","H.O4-O5","H.O5-O6","H.O6-O7","H.O7-O8",
                         "C.O1-O2","C.O2-O3","C.O3-O4","C.O4-O5","C.O5-O6","C.O6-O7","C.O7-O8")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.O1-O2","H.O2-O3","H.O3-O4","H.O4-O5","H.O5-O6","H.O6-O7","H.O7-O8",
                                             "C.O1-O2","C.O2-O3","C.O3-O4","C.O4-O5","C.O5-O6","C.O6-O7","C.O7-O8")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (3518 genes) -> 33.84%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.3) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.3) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.3) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.3) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.3) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.3) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 ( 450 genes) ->  4.33%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.2) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.2) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.2) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.2) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.2) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.2) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.2), ]))/dim(dge$counts)[1]

#0.1 (  12 genes) ->  0.12%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[ ,9]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.1) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.1) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.1) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[ ,9]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.1) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.1) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.1) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]>0.5 & gene.prob[ ,9]>0.5 & gene.prob[,10]>0.5 & gene.prob[,11]>0.5 &
                                    gene.prob[,12]>0.5 & gene.prob[,13]>0.5 & gene.prob[,14]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]>0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]>0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]>0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]>0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifC5 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]>0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC5)
head(prob.motifC5)
#boxplot(tmm[prob.motifC5[1],]~groupFactor)

prob.motifC6 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]>0.5 & gene.prob[,14]<0.5),])
length(prob.motifC6)
head(prob.motifC6)
#boxplot(tmm[prob.motifC6[1],]~groupFactor)

prob.motifC7 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]>0.5),])
length(prob.motifC7)
head(prob.motifC7)
#boxplot(tmm[prob.motifC7[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]>0.5 &
                                    gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5 & gene.prob[ ,7]>0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

prob.motifH5 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH5)
head(prob.motifH5)
#boxplot(tmm[prob.motifH5[1],]~groupFactor)

prob.motifH6 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH6)
head(prob.motifH6)
#boxplot(tmm[prob.motifH6[1],]~groupFactor)

prob.motifH7 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]>0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH7)
head(prob.motifH7)
#boxplot(tmm[prob.motifH7[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4,prob.motifH5,prob.motifH6,prob.motifH7)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4,prob.motifC5,prob.motifC6,prob.motifC7)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Clusters (res=0.50) (Time 2 only)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstCluster0.50==j & Stage=="Time 2")
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt <- dataSub@meta.data$OstCluster0.50[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Cluster")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)

```

#### each cluster separately

Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.

```{r, eval=FALSE}

# Compile group labels of each sample (clusters are off by 1)
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID <- data.frame(c1 = c(1, 2, 3, 4),
                        c2 = c(5, 6, 7, 8))

# Assess differences within each cluster
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each cluster
prob.motifO1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO1234  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO1234)
head(prob.motifO1234)
#boxplot(tmm[prob.motifO1234[1],]~groupFactor)

prob.motifO134  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO134)
head(prob.motifO134)
#boxplot(tmm[prob.motifO134[1],]~groupFactor)

prob.motifO13  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5),])
length(prob.motifO13)
head(prob.motifO13)
#boxplot(tmm[prob.motifO13[1],]~groupFactor)

prob.motifO34  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5),])
length(prob.motifO34)
head(prob.motifO34)
#boxplot(tmm[prob.motifO34[1],]~groupFactor)

prob.motifO14  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5),])
length(prob.motifO14)
head(prob.motifO14)
#boxplot(tmm[prob.motifO14[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4)
rm(prob.motifO1234,prob.motifO134,prob.motifO13,prob.motifO34,prob.motifO14)

```

#### 4 osteogenic clusters and 6 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 5, 6, 7),
                        c2 = c(2, 3, 4, 6, 7, 8))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 4 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           Cluster=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3",
                                           "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                                         c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                           "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4")),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="deepskyblue2",
                                              "Osteogenic.c3"="dodgerblue4",
                                              "Osteogenic.c4"="navyblue")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.O1-O2","H.O2-O3","H.O3-O4",
                         "C.O1-O2","C.O2-O3","C.O3-O4")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.O1-O2","H.O2-O3","H.O3-O4",
                                             "C.O1-O2","C.O2-O3","C.O3-O4")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (7523 genes) -> 71.28%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (5777 genes) -> 54.74%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (2348 genes) -> 22.25%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,4]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,6]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]>0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]>0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 &
                                    gene.prob[ ,4]<0.5 & gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.50.T2.6.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Clusters (res=0.75) (Time 2 only)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                 "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstCluster0.75==j & Stage=="Time 2")
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.clt <- dataSub@meta.data$OstCluster0.75[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.clt))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Cluster")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.clt)

```

#### each cluster separately

Include all clusters here, as reducing the number of clusters included in cormotif does not reveal more interesting patterns.

```{r, eval=FALSE}

# Compile group labels of each sample (clusters are off by 1)
groupFactor <- factor(paste(dge$samples$Species,dge$samples$Cluster,sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Human_Osteogenic.c5","Human_Osteogenic.c6","Human_Osteogenic.c7","Human_Osteogenic.c8",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4",
                               "Chimp_Osteogenic.c5","Chimp_Osteogenic.c6","Chimp_Osteogenic.c7","Chimp_Osteogenic.c8"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species for each cluster)
compareID <- data.frame(c1 = c(1,  2,  3,  4,  5,  6,  7,  8),
                        c2 = c(9, 10, 11, 12, 13, 14, 15, 16))

# Assess differences within each cluster
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Cluster=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                     "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8"),
                           col=list(Cluster=c("Osteogenic.c1"="dodgerblue",
                                              "Osteogenic.c2"="dodgerblue4",
                                              "Osteogenic.c3"="deepskyblue2",
                                              "Osteogenic.c4"="navyblue",
                                              "Osteogenic.c5"="lightblue1",
                                              "Osteogenic.c6"="royalblue3",
                                              "Osteogenic.c7"="grey",
                                              "Osteogenic.c8"="black")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Species",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                             "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each cluster
prob.motifO1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO1)
head(prob.motifO1)
#boxplot(tmm[prob.motifO1[1],]~groupFactor)

prob.motifO2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO2)
head(prob.motifO2)
#boxplot(tmm[prob.motifO2[1],]~groupFactor)

prob.motifO3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO3)
head(prob.motifO3)
#boxplot(tmm[prob.motifO3[1],]~groupFactor)

prob.motifO4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO4)
head(prob.motifO4)
#boxplot(tmm[prob.motifO4[1],]~groupFactor)

prob.motifO5  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO5)
head(prob.motifO5)
#boxplot(tmm[prob.motifO5[1],]~groupFactor)

prob.motifO6  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifO6)
head(prob.motifO6)
#boxplot(tmm[prob.motifO6[1],]~groupFactor)

prob.motifO7  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifO7)
head(prob.motifO7)
#boxplot(tmm[prob.motifO7[1],]~groupFactor)

prob.motifO8  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifO8)
head(prob.motifO8)
#boxplot(tmm[prob.motifO8[1],]~groupFactor)

# Examine human/chimp differences shared across clusters
prob.motifO12345678  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                            gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifO12345678)
head(prob.motifO12345678)
#boxplot(tmm[prob.motifO12345678[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifO1,prob.motifO2,prob.motifO3,prob.motifO4,prob.motifO5,prob.motifO6,prob.motifO7,prob.motifO8)
rm(prob.motifO12345678)

```

#### 8 osteogenic clusters and 14 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$Cluster,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_Osteogenic.c1","Human_Osteogenic.c2","Human_Osteogenic.c3","Human_Osteogenic.c4",
                               "Human_Osteogenic.c5","Human_Osteogenic.c6","Human_Osteogenic.c7","Human_Osteogenic.c8",
                               "Chimp_Osteogenic.c1","Chimp_Osteogenic.c2","Chimp_Osteogenic.c3","Chimp_Osteogenic.c4",
                               "Chimp_Osteogenic.c5","Chimp_Osteogenic.c6","Chimp_Osteogenic.c7","Chimp_Osteogenic.c8"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 5, 6, 7,  9, 10, 11, 12, 13, 14, 15),
                        c2 = c(2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:20,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 15 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7",
                                         "Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7"),
                                       c("Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8",
                                         "Osteogenic.c2","Osteogenic.c3","Osteogenic.c4",
                                         "Osteogenic.c5","Osteogenic.c6","Osteogenic.c7","Osteogenic.c8")),
                           col=list(AdHoc=c("Osteogenic.c1"="dodgerblue",
                                            "Osteogenic.c2"="dodgerblue4",
                                            "Osteogenic.c3"="deepskyblue2",
                                            "Osteogenic.c4"="navyblue",
                                            "Osteogenic.c5"="lightblue1",
                                            "Osteogenic.c6"="royalblue3",
                                            "Osteogenic.c7"="grey",
                                            "Osteogenic.c8"="black")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.O1-O2","H.O2-O3","H.O3-O4","H.O4-O5","H.O5-O6","H.O6-O7","H.O7-O8",
                         "C.O1-O2","C.O2-O3","C.O3-O4","C.O4-O5","C.O5-O6","C.O6-O7","C.O7-O8")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.O1-O2","H.O2-O3","H.O3-O4","H.O4-O5","H.O5-O6","H.O6-O7","H.O7-O8",
                                             "C.O1-O2","C.O2-O3","C.O3-O4","C.O4-O5","C.O5-O6","C.O6-O7","C.O7-O8")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (3267 genes) -> 31.99%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.3) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.3) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.3) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.3) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.3) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.3) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (  66 genes) ->  0.65%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.2) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.2) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.2) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[, 8]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[, 9]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.2) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.2) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.2) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (   0 genes) ->   0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[ ,9]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.1) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.1) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.1) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[ ,8]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[ ,9]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,10]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,11]) <= 0.1) &
                          (abs(gene.prob[,5] - gene.prob[,12]) <= 0.1) &
                          (abs(gene.prob[,6] - gene.prob[,13]) <= 0.1) &
                          (abs(gene.prob[,7] - gene.prob[,14]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine genes from each motif (SKIPPED)

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]>0.5 & gene.prob[ ,9]>0.5 & gene.prob[,10]>0.5 & gene.prob[,11]>0.5 &
                                    gene.prob[,12]>0.5 & gene.prob[,13]>0.5 & gene.prob[,14]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]>0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]>0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]>0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]>0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifC5 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]>0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifC5)
head(prob.motifC5)
#boxplot(tmm[prob.motifC5[1],]~groupFactor)

prob.motifC6 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]>0.5 & gene.prob[,14]<0.5),])
length(prob.motifC6)
head(prob.motifC6)
#boxplot(tmm[prob.motifC6[1],]~groupFactor)

prob.motifC7 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]>0.5),])
length(prob.motifC7)
head(prob.motifC7)
#boxplot(tmm[prob.motifC7[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]>0.5 &
                                    gene.prob[ ,5]>0.5 & gene.prob[ ,6]>0.5 & gene.prob[ ,7]>0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[ ,1]>0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]>0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]>0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]>0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

prob.motifH5 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]>0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH5)
head(prob.motifH5)
#boxplot(tmm[prob.motifH5[1],]~groupFactor)

prob.motifH6 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]>0.5 & gene.prob[ ,7]<0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH6)
head(prob.motifH6)
#boxplot(tmm[prob.motifH6[1],]~groupFactor)

prob.motifH7 <- rownames(gene.prob[(gene.prob[ ,1]<0.5 & gene.prob[ ,2]<0.5 & gene.prob[ ,3]<0.5 & gene.prob[ ,4]<0.5 &
                                    gene.prob[ ,5]<0.5 & gene.prob[ ,6]<0.5 & gene.prob[ ,7]>0.5 &
                                    gene.prob[ ,8]<0.5 & gene.prob[ ,9]<0.5 & gene.prob[,10]<0.5 & gene.prob[,11]<0.5 &
                                    gene.prob[,12]<0.5 & gene.prob[,13]<0.5 & gene.prob[,14]<0.5),])
length(prob.motifH7)
head(prob.motifH7)
#boxplot(tmm[prob.motifH7[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostcluster0.75.T2.14.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4,prob.motifH5,prob.motifH6,prob.motifH7)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4,prob.motifC5,prob.motifC6,prob.motifC7)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstAdHoc.Assign.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

#### each osteogenic ad hoc classification separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID <- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)

```

#### 5 osteogenic ad hoc classifications and 8 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast",
                                         "preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                                       c("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte",
                                         "osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10669 genes) ->  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10669 genes) ->  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (10669 genes) ->  100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene8.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=OstAdHoc.Assign.x.4gene.thresh0==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.x.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

#### each osteogenic ad hoc classification separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID <- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)

```

#### 5 osteogenic ad hoc classifications and 8 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast",
                                         "preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                                       c("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte",
                                         "osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (9999 genes) -> 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (   0 genes) ->   0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (   0 genes) ->   0.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX8.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes flexible, thresh0) (Time 2 only)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=(OstAdHoc.Assign.4gene.thresh0==j & Stage=="Time 2"))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

#### each osteogenic ad hoc classification separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID <- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 2 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)

```

#### 5 osteogenic ad hoc classifications and 8 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 1 motif (WHAT?!)
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast",
                                         "preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                                       c("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte",
                                         "osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10822 genes) -> 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10822 genes) -> 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (10822 genes) -> 100.00%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4gene8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)

```

```{r, eval=FALSE}

rm(dge)

```

### Pseudobulk: Species vs. Osteogenic Ad Hoc Classifications (4 genes strict, thresh0) (Time 2 only)

```{r, eval=FALSE}

# Compile log2 normalized gene expression data
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data,subset=(OstAdHoc.Assign.x.4gene.thresh0==j & Stage=="Time 2"))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.x.4gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
dge$counts <- dge$counts[keep,]
##normalize data
#dge <- calcNormFactors(dge, method="TMM")
##log2 transformed cpm
#tmm <- edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25)
###boxplot(tmm,xlab="",ylab="Log2 counts per million",las=2)

rm(cell.subset,labels,i,j,w,keep,dataSub)
rm(x.cnt,x.lab,x.spp,x.col,x.ind,x.rep,x.idr,x.adh)

```

#### each osteogenic ad hoc classification separately

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = species in each osteogenic ad hoc classification)
compareID <- data.frame(c1 = c(1, 2,  3,  4,  5),
                        c2 = c(8, 9, 10, 11, 12))

# Assess differences at each stage of differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 3 motifs
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(AdHoc=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Osteogenic Ad Hoc",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Examine human/chimp differences unique to each osteogenic ad hoc classification
prob.motifPre  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPre)
head(prob.motifPre)
#boxplot(tmm[prob.motifPre[1],]~groupFactor)

prob.motifObl  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifObl)
head(prob.motifObl)
#boxplot(tmm[prob.motifObl[1],]~groupFactor)

prob.motifEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmb)
head(prob.motifEmb)
#boxplot(tmm[prob.motifEmb[1],]~groupFactor)

prob.motifMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifMin)
head(prob.motifMin)
#boxplot(tmm[prob.motifMin[1],]~groupFactor)

prob.motifMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]>0.5),])
length(prob.motifMat)
head(prob.motifMat)
#boxplot(tmm[prob.motifMat[1],]~groupFactor)

# Examine human/chimp differences shared across osteogenic ad hoc classification
prob.motifPreOblEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifPreOblEmbMinMat)
head(prob.motifPreOblEmbMinMat)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifPreOblEmb  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreOblEmb)
head(prob.motifPreOblEmb)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmbMin)
head(prob.motifOblEmbMin)
#boxplot(tmm[prob.motifOblEmbMin[1],]~groupFactor)

prob.motifEmbMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifEmbMinMat)
head(prob.motifEmbMinMat)
#boxplot(tmm[prob.motifEmbMinMat[1],]~groupFactor)

prob.motifPreObl  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifPreObl)
head(prob.motifPreObl)
#boxplot(tmm[prob.motifPreOblEmbMinMat[1],]~groupFactor)

prob.motifOblEmb  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 & gene.prob[,5]<0.5),])
length(prob.motifOblEmb)
head(prob.motifOblEmb)
#boxplot(tmm[prob.motifOblEmb[1],]~groupFactor)

prob.motifEmbMin  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 & gene.prob[,5]<0.5),])
length(prob.motifEmbMin)
head(prob.motifEmbMin)
#boxplot(tmm[prob.motifEmbMin[1],]~groupFactor)

prob.motifMinMat  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 & gene.prob[,5]>0.5),])
length(prob.motifMinMat)
head(prob.motifMinMat)
#boxplot(tmm[prob.motifMinMat[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifPre,prob.motifObl,prob.motifEmb,prob.motifMin,prob.motifMat)
rm(prob.motifPreOblEmbMinMat,prob.motifPreOblEmb,prob.motifOblEmbMin,prob.motifEmbMinMat,prob.motifPreObl,prob.motifOblEmb,prob.motifEmbMin,prob.motifMinMat)

```

#### 5 osteogenic ad hoc classifications and 8 comparisons

```{r, eval=FALSE}

# Compile group labels of each sample
groupFactor <- factor(paste(dge$samples$Species,sapply(dge$samples$AdHoc,function(x)sub(" ",".",x)),sep="_"),
                      levels=c("Human_preosteoblast","Human_osteoblast","Human_embedding.osteoblast","Human_mineralizing.osteoblast","Human_maturing.osteocyte",
                               "Human_unknown.osteogenic","Human_not.osteogenic",
                               "Chimp_preosteoblast","Chimp_osteoblast","Chimp_embedding.osteoblast","Chimp_mineralizing.osteoblast","Chimp_maturing.osteocyte",
                               "Chimp_unknown.osteogenic","Chimp_not.osteogenic"))
groupID <- as.numeric(groupFactor)

# Compile comparison pattern
#(condiditions to compare = pre, obl, emb, min, and mat for each species)
compareID <- data.frame(c1 = c(1, 2, 3, 4, 8,  9, 10, 11),
                        c2 = c(2, 3, 4, 5, 9, 10, 11, 12))

# Assess differences during differentiation
set.seed(1)
cormotifData <- cormotiffit(exprs = dge$counts,
                            groupid = groupID,
                            compid = compareID,
                            K = 1:10,
                            max.iter = 1000,
                            BIC = FALSE,
                            voom.normalize.method = "none",
                            runtype = "counts")

#genes examined
dim(dge$counts)[1]

#which BIC and AIC are lowest? - 1 motif
cormotifData$bic
plotIC(cormotifData)
cormotifData$bestmotif$loglike

#what motif patterns were learned by the algorithm
#and what is the probability that each motif is differentially expressed in each comparison
plotMotif(cormotifData)

#make a nicer plot
HAtop <- HeatmapAnnotation(Species=anno_text(c("Human","Human","Human","Human",
                                               "Chimp","Chimp","Chimp","Chimp"),
                                             rot=360,
                                             just="center",
                                             location=0.5),
                           AdHoc=cbind(c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast",
                                         "preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                                       c("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte",
                                         "osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
                           col=list(AdHoc=c("preosteoblast"="lightblue1",
                                            "osteoblast"="deepskyblue",
                                            "embedding.osteoblast"="dodgerblue2",
                                            "mineralizing.osteoblast"="royalblue3",
                                            "maturing.osteocyte"="royalblue4")),
                           gp=gpar(col="black"),
                           gap=unit(5,"points"),
                           annotation_name_side="left")
HAside = rowAnnotation(No.Genes=anno_barplot(floor(cormotifData$bestmotif$motif.p * nrow(cormotifData$bestmotif$p.post))),
                       width=unit(3,"cm"),
                       border=FALSE)
Heatmap(cormotifData$bestmotif$motif.q,
        col=gray(seq(from=1,to=0,by=-0.1)),
        name="DE Prob.",
        rect_gp=gpar(col="black",lwd=.3),
        top_annotation=HAtop,
        right_annotation=HAside,
        column_title="Comparisons across Stages",
        row_title="Motif",
        column_dend_reorder=FALSE,
        row_dend_reorder=TRUE,
        #column_split=c(1,2,1,2),
        heatmap_legend_param=list(at=c(0,0.2,0.4,0.6,0.8,1)))

#posterior probability of DE for each gene
head(cormotifData$bestmotif$p.post)

#obtain DE pattern at 0.5 posterior distribution cutoff
gene.prob <- cormotifData$bestmotif$p.post
rownames(gene.prob) <- rownames(dge$counts)
colnames(gene.prob) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(gene.prob)
#add details to cormotif object
rownames(cormotifData$bestmotif$p.post) <- rownames(dge$counts)
colnames(cormotifData$bestmotif$p.post) <- c("H.pre-obl","H.obl-emb","H.emb-min","H.min-mat","C.pre-obl","C.obl-emb","C.emb-min","C.min-mat")
head(cormotifData$bestmotif$p.post)

heatmap.2(gene.prob,
          dendrogram="none",
          Colv=FALSE,
          breaks=c(0,0.2,0.4,0.6,0.8,1),
          col=c("#fef0d9","#fdcc8a","#fc8d59","#e34a33","#b30000"),
          margins=c(3,0),
          trace="none",
          key.title="",
          key.xlab="posterior prob.",
          key.ylab="",
          #ylab=paste0(length(rownames(gene.prob))," genes"),
          keysize=1,
          key.par=list(mar=c(3.5,0,3,0)),
          lmat=rbind(c(5,4,2),c(6,1,3)),
          lhei=c(2.5,5),
          lwid=c(1,10,1),
          labRow=FALSE,
          cexCol=1,
          srtCol=0,
          adjCol=c(0.5,0))

# Test degree of conservation for pre, obl, emb, min, and mat: posterior probability difference between the humans and the chimps within a motif 
#0.3 (10535 genes) -> 100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.3) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.3) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.3) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.3), ]))/dim(dge$counts)[1]
#0.2 (10535 genes) -> 100%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.2) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.2) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.2) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.2), ]))/dim(dge$counts)[1]
#0.1 (    0 genes) ->   0%
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))
length(rownames(gene.prob[(abs(gene.prob[,1] - gene.prob[,5]) <= 0.1) &
                          (abs(gene.prob[,2] - gene.prob[,6]) <= 0.1) &
                          (abs(gene.prob[,3] - gene.prob[,7]) <= 0.1) &
                          (abs(gene.prob[,4] - gene.prob[,8]) <= 0.1), ]))/dim(dge$counts)[1]

# Examine human/chimp differences
prob.motifC  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]>0.5 & gene.prob[,6]>0.5 & gene.prob[,7]>0.5 & gene.prob[,8]>0.5),])
length(prob.motifC)
head(prob.motifC)
#boxplot(tmm[prob.motifC[1],]~groupFactor)

prob.motifC1  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]>0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC1)
head(prob.motifC1)
#boxplot(tmm[prob.motifC1[1],]~groupFactor)

prob.motifC2  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]>0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifC2)
head(prob.motifC2)
#boxplot(tmm[prob.motifC2[1],]~groupFactor)

prob.motifC3  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]>0.5 & gene.prob[,8]<0.5),])
length(prob.motifC3)
head(prob.motifC3)
#boxplot(tmm[prob.motifC3[1],]~groupFactor)

prob.motifC4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]>0.5),])
length(prob.motifC4)
head(prob.motifC4)
#boxplot(tmm[prob.motifC4[1],]~groupFactor)

prob.motifH  <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]>0.5 & gene.prob[,3]>0.5 & gene.prob[,4]>0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH)
head(prob.motifH)
#boxplot(tmm[prob.motifH[1],]~groupFactor)

prob.motifH1 <- rownames(gene.prob[(gene.prob[,1]>0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH1)
head(prob.motifH1)
#boxplot(tmm[prob.motifH1[1],]~groupFactor)

prob.motifH2 <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]>0.5 & gene.prob[,3]<0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH2)
head(prob.motifH2)
#boxplot(tmm[prob.motifH2[1],]~groupFactor)

prob.motifH3 <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]>0.5 & gene.prob[,4]<0.5 &
                                    gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH3)
head(prob.motifH3)
#boxplot(tmm[prob.motifH3[1],]~groupFactor)

prob.motifH4  <- rownames(gene.prob[(gene.prob[,1]<0.5 & gene.prob[,2]<0.5 & gene.prob[,3]<0.5 & gene.prob[,4]>0.5 &
                                     gene.prob[,5]<0.5 & gene.prob[,6]<0.5 & gene.prob[,7]<0.5 & gene.prob[,8]<0.5),])
length(prob.motifH4)
head(prob.motifH4)
#boxplot(tmm[prob.motifH4[1],]~groupFactor)

```

```{r, eval=FALSE}

saveRDS(cormotifData,"./data/cormotif-data/cormotif.cnts.v2.pseudo-ostadhoc4geneX8.T2.data.filterC.log.indv-cell.intNo0.reg-tot.rds")

```

```{r, eval=FALSE}

rm(HAside,HAtop)
rm(groupFactor,groupID,compareID,cormotifData,gene.prob)
rm(prob.motifH,prob.motifH1,prob.motifH2,prob.motifH3,prob.motifH4)
rm(prob.motifC,prob.motifC1,prob.motifC2,prob.motifC3,prob.motifC4)

```

```{r, eval=FALSE}

rm(dge)

```
