---
title: "10x-plot-classification-adhoc"
author: "Genevieve Housman"
date: "May 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 10X Data Cellranger Analysis - Visual Cell Classifications (Ad Hoc)

Visualize ad hoc cell classification results.

```{r}

#Load libraries
library(ggplot2)
library(grid)
library(gridExtra)
library(Seurat)
library(UpSetR)

```

```{r}

#Load data
#data integrated across individuals - total (conservative cell filter + non-zero genes + regress out UMI/mito)
scrna <- "./data/cellranger-data-full/data.filterC.log.indv-cell.intNo0.reg-tot.assign.rds"
data <- readRDS(scrna)

```

## Examine several different candidate genes.

```{r}

#Look at candidate genes

#Time 0
#diagnostic: POU5F1 (OCT4), TDGF1, SOX2, EPCAM
#additional: NANOG, ALPL, STAT3, PROM1 (CD133), PODXL (TRA-1-60R), LIN28A, ZFP42, UTF1, MYC, BMP4, FGF2, TGFB1, DNMT3B, SALL4
#if needed: FOXD3, FUT4 (SSEA-1), KLF2, KLF4, KLF5, INHBA (Activin A), LIF, PRDM14
genes <- c("POU5F1","TDGF1","SOX2","EPCAM","NANOG","ALPL","STAT3","PROM1","PODXL","LIN28A","ZFP42","UTF1","MYC","BMP4","FGF2","TGFB1","DNMT3B",
           "SALL4","FOXD3","FUT4","KLF2","KLF4","KLF5","INHBA","LIF","PRDM14")
data@assays$integrated[genes,1:3]
FeaturePlot(data, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=3)

#Time 1
#diagnostic: THY1 (CD90), NT5E (CD73), ENG (CD105), CD44
#additional: ANPEP (CD13), NGFR (CD271), ITGB1 (CD29), NCAM1 (CD56)
#if needed: CD9, CD14, MME (CD10), ITGAL (CD11A), ITGAX (CD11C)*, CD36, ITGA3 (CD49C), ITGA4 (CD49D), ITGAV (CD51), ICAM1 (CD54), CD55, CD59, TFRC (CD71), VCAM1 (CD106), KIT (CD117), TNFRSF1A (CD120a), PDGFRB (CD140B), CDH5 (CD144), MCAM (CD146), ALCAM (CD166), ERBB2 (CD340)
#negative controls: ITGAM (CD11B), CD14, PECAM1 (CD31), CD34, PTPRC (CD45), CD79A, HLA-DRA (HLA-DR)
genes <- c("THY1","NT5E","ENG","CD44","ANPEP","NGFR","ITGB1","NCAM1","CD9","CD14","MME","ITGAL","ITGAX","CD36","ITGA3","ITGA4","ITGAV","ICAM1","CD55",
           "CD59","TFRC","VCAM1","KIT","TNFRSF1A","PDGFRB","CDH5","MCAM","ALCAM","ERBB2","ITGAM","CD14","PECAM1","CD34","PTPRC","CD79A","HLA-DRA")
data@assays$integrated[genes,1:3]
FeaturePlot(data, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=3)

#Time 2
#diagnostic: RUNX2, BGLAP, CAPG, [DMP1], [PHEX], MEPE, [SOST], [HYOU1]
#additional: COL1A1, COL1A2, ALPL, TNFRSF11B (OPG), TNFSF11 (RANKL), CSF1 (M-CSF), DKK1, SFRP1, PTGES2, PTGS2, SPP1 (OPN), SPARC (ONT)
#if needed: E11*
# * not in ortho-exon (osteo)
genes <- c("RUNX2","BGLAP","CAPG","DMP1","PHEX","MEPE","SOST","HYOU1",
           "COL1A1","COL1A2","ALPL","TNFRSF11B","TNFSF11",
           "CSF1","DKK1","SFRP1","PTGES2","PTGS2","SPP1","SPARC")
data@assays$integrated[genes,1:3]
FeaturePlot(data, features=genes[1:8], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=3)
FeaturePlot(data, features=genes[9:20], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
#markers that differentiated BM osteoblasts in Anastasia Tikhonova's research (ASBMR2020)
FeaturePlot(data, features=c("COL11A2","SPARC","SOX9","COMP","CHAD","COL10A1","EDIL3","MMP14","OSTN","COL12A1","ANGPTL2","COL16A1"), reduction="umap", pt.size=0.5, order=TRUE)
#markers used to differentiate different stages of osteogenic differentiation (BGLAP=preosteoblast,TNC=intermediate,PDPN+FBLN7=osteocytes) from Jialiang Wang's research (ASBMR2020)
FeaturePlot(data, features=c("BGLAP","TNC","PDPN","FBLN7"), reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

#chondrogenic diagnostic: ACAN, COL2A1, SOX9, COL10A1, COL11A1, SOX5, SOX6
#adipogenic diagnostic: FABP4, PPARG, EBF2, PDGFRA, FGF10, FGF16, FGF19, SLC7A10, SLC36A2, MYF5, P2RX5
genes <- c("ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2",
           "COL10A1","COL11A1","SOX5","SOX6",
           "PDGFRA","FGF10","FGF16","FGF19","SLC7A10","SLC36A2","P2RX5","MYF5")
data@assays$integrated[genes,1:5]
FeaturePlot(data, features=genes[c(1,4,2,5,3,6)], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

```

Isolate gene expression levels of interest.

```{r}

#Isolate genes of interest
genes <- c("POU5F1","TDGF1","SOX2","EPCAM",
           "THY1","NT5E","ENG","CD44",
           "RUNX2","BGLAP","CAPG","DMP1","PHEX","MEPE","SOST","HYOU1",
           "COL1A1","COL1A2","ALPL",
           "ACAN","COL2A1","SOX9",
           "FABP4","PPARG","EBF2",
           "TNC","PDPN","FBLN7")

```

## Assign Gene Expression Thresholds

Define functions for setting potential ad hoc thresholds.

```{r}

#Define functions for setting ad hoc thresholds

chk.thres.manual <- function(data, genes, thres) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres[i], 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.median <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- median(data[,genes[i]]) }
    else { thres <- median(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.q1 <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- quantile(data[,genes[i]],prob=0.25) }
    else { thres <- quantile(data[,genes[i]][data[,genes[i]]<cutoff],prob=0.25) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.q3 <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- quantile(data[,genes[i]],prob=0.75) }
    else { thres <- quantile(data[,genes[i]][data[,genes[i]]<cutoff],prob=0.75) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.mean <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- mean(data[,genes[i]]) }
    else { thres <- mean(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.meansd <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- sum(mean(data[,genes[i]]),sd(data[,genes[i]])) }
    else { thres <- sum(mean(data[,genes[i]][data[,genes[i]]<cutoff]),sd(data[,genes[i]][data[,genes[i]]<cutoff])) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

```

Calculate if genes reach expression thresholds. Start with several threshold limits.

* scaled integrated data (with confounding variables regressed out)
* log normalized data
* rescaled integrated gene expression data to range from 0 to 1
* rescaled log normalized gene expression data to range from 0 to 1
* non-scaled integrated data (with confounding variables regressed out)

```{r}

#Scaled integrated data (with confounding variables regressed out)
tot.genes <- cbind(as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes,]))),data@meta.data[,c("Species","Stage","Sample")])

assign.threshN0.1 <- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.1,length(genes))))
assign.threshN0.05 <- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.05,length(genes))))
assign.threshN0.01 <- chk.thres.manual(tot.genes, genes, thres=c(rep(-0.01,length(genes))))
assign.thresh0 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0,length(genes))))
assign.thresh0.01 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.01,length(genes))))
assign.thresh0.05 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.05,length(genes))))
assign.thresh0.1 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,length(genes))))
assign.thresh0.3 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.3,length(genes))))
assign.thresh0.5 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.5,length(genes))))
assign.thresh0.7 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.7,length(genes))))
assign.thresh0.9 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.9,length(genes))))
assign.thresh1 <- chk.thres.manual(tot.genes, genes, thres=c(rep(1,length(genes))))
assign.threshMulti01 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
assign.threshMulti0 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

assign.median <- chk.thres.median(tot.genes, genes, cutoff=NA)
assign.median1 <- chk.thres.median(tot.genes, genes, cutoff=1)
assign.q1 <- chk.thres.q1(tot.genes, genes, cutoff=NA)
assign.q3 <- chk.thres.q3(tot.genes, genes, cutoff=NA)
assign.mean <- chk.thres.mean(tot.genes, genes, cutoff=NA)
assign.mean1 <- chk.thres.mean(tot.genes, genes, cutoff=1)
assign.meansd <- chk.thres.meansd(tot.genes, genes, cutoff=NA)
assign.meansd1 <- chk.thres.meansd(tot.genes, genes, cutoff=1)
assign.multi.q1median <- cbind(assign.q1[,1:8],assign.median[,9:16],assign.q1[,17:31])
assign.multi.medianq1 <- cbind(assign.median[,1:8],assign.q1[,9:16],assign.median[,17:31])

```

```{r}

#Log normalized data
l.tot.genes <- cbind(as.data.frame(t(as.matrix(data@assays$RNA@data[genes,]))),data@meta.data[,c("Species","Stage","Sample")])

l.assign.threshN0.1 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(-0.1,length(genes))))
l.assign.threshN0.05 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(-0.05,length(genes))))
l.assign.threshN0.01 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(-0.01,length(genes))))
l.assign.thresh0 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0,length(genes))))
l.assign.thresh0.01 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.01,length(genes))))
l.assign.thresh0.05 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.05,length(genes))))
l.assign.thresh0.1 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,length(genes))))
l.assign.thresh0.3 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.3,length(genes))))
l.assign.thresh0.5 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.5,length(genes))))
l.assign.thresh0.7 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.7,length(genes))))
l.assign.thresh0.9 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.9,length(genes))))
l.assign.thresh1 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(1,length(genes))))
#l.assign.threshMulti01 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
#l.assign.threshMulti0 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

l.assign.median <- chk.thres.median(l.tot.genes, genes, cutoff=NA)
l.assign.median1 <- chk.thres.median(l.tot.genes, genes, cutoff=1)
l.assign.q1 <- chk.thres.q1(l.tot.genes, genes, cutoff=NA)
l.assign.q3 <- chk.thres.q3(l.tot.genes, genes, cutoff=NA)
l.assign.mean <- chk.thres.mean(l.tot.genes, genes, cutoff=NA)
l.assign.mean1 <- chk.thres.mean(l.tot.genes, genes, cutoff=1)
l.assign.meansd <- chk.thres.meansd(l.tot.genes, genes, cutoff=NA)
l.assign.meansd1 <- chk.thres.meansd(l.tot.genes, genes, cutoff=1)
#l.assign.multi.q1median <- cbind(assign.q1[,1:8],assign.median[,9:16],assign.q1[,17:31])
#l.assign.multi.medianq1 <- cbind(assign.median[,1:8],assign.q1[,9:16],assign.median[,17:31])

```

```{r}

#Rescaled integrated gene expression data to range from 0 to 1
z.tot.genes <- as.data.frame(apply(tot.genes[,1:28], MARGIN=2, FUN=function(X)(X-min(X))/diff(range(X))))
z.tot.genes <- cbind(z.tot.genes,data@meta.data[,c("Species","Stage","Sample")])

z.assign.threshN0.1 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(-0.1,length(genes))))
z.assign.threshN0.05 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(-0.05,length(genes))))
z.assign.threshN0.01 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(-0.01,length(genes))))
z.assign.thresh0 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0,length(genes))))
z.assign.thresh0.01 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.01,length(genes))))
z.assign.thresh0.05 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.05,length(genes))))
z.assign.thresh0.1 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.1,length(genes))))
z.assign.thresh0.3 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.3,length(genes))))
z.assign.thresh0.5 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.5,length(genes))))
z.assign.thresh0.7 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.7,length(genes))))
z.assign.thresh0.9 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.9,length(genes))))
z.assign.thresh1 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(1,length(genes))))
z.assign.median <- chk.thres.median(z.tot.genes, genes, cutoff=NA)
z.assign.median1 <- chk.thres.median(z.tot.genes, genes, cutoff=1)
z.assign.mean <- chk.thres.mean(z.tot.genes, genes, cutoff=NA)
z.assign.mean1 <- chk.thres.mean(z.tot.genes, genes, cutoff=1)
z.assign.meansd <- chk.thres.meansd(z.tot.genes, genes, cutoff=NA)
z.assign.meansd1 <- chk.thres.meansd(z.tot.genes, genes, cutoff=1)
z.assign.threshMulti01 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
z.assign.threshMulti0 <- chk.thres.manual(z.tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

#z.assign.median, z.assign.mean, z.assign.meansd -> produce the same results as assign.median, assign.mean, assign.meansd
#z.assign.median1, z.assign.mean1, z.assign.meansd1 -> produce very similar results to z.assign.median, z.assign.mean, z.assign.meansd

```

```{r}

#Rescaled log normalized gene expression data to range from 0 to 1
z.l.tot.genes <- as.data.frame(apply(l.tot.genes[,1:28], MARGIN=2, FUN=function(X)(X-min(X))/diff(range(X))))
z.l.tot.genes <- cbind(z.l.tot.genes,data@meta.data[,c("Species","Stage","Sample")])

z.l.assign.threshN0.1 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(-0.1,length(genes))))
z.l.assign.threshN0.05 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(-0.05,length(genes))))
z.l.assign.threshN0.01 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(-0.01,length(genes))))
z.l.assign.thresh0 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0,length(genes))))
z.l.assign.thresh0.01 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.01,length(genes))))
z.l.assign.thresh0.05 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.05,length(genes))))
z.l.assign.thresh0.1 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.1,length(genes))))
z.l.assign.thresh0.3 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.3,length(genes))))
z.l.assign.thresh0.5 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.5,length(genes))))
z.l.assign.thresh0.7 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.7,length(genes))))
z.l.assign.thresh0.9 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.9,length(genes))))
z.l.assign.thresh1 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(1,length(genes))))
z.l.assign.median <- chk.thres.median(z.l.tot.genes, genes, cutoff=NA)
z.l.assign.median1 <- chk.thres.median(z.l.tot.genes, genes, cutoff=1)
z.l.assign.mean <- chk.thres.mean(z.l.tot.genes, genes, cutoff=NA)
z.l.assign.mean1 <- chk.thres.mean(z.l.tot.genes, genes, cutoff=1)
z.l.assign.meansd <- chk.thres.meansd(z.l.tot.genes, genes, cutoff=NA)
z.l.assign.meansd1 <- chk.thres.meansd(z.l.tot.genes, genes, cutoff=1)
z.l.assign.threshMulti01 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
z.l.assign.threshMulti0 <- chk.thres.manual(z.l.tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

#z.l.assign.median, z.l.assign.mean, z.l.assign.meansd -> produce the same results as assign.median, assign.mean, assign.meansd
#z.l.assign.median1, z.l.assign.mean1, z.l.assign.meansd1 -> produce very similar results to z.assign.median, z.assign.mean, z.assign.meansd

```

```{r}

#Non-scaled integrated data (with confounding variables regressed out)
d.tot.genes <- cbind(as.data.frame(t(as.matrix(data@assays$integrated@data[genes,]))),data@meta.data[,c("Species","Stage","Sample")])

d.assign.threshN0.1 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(-0.1,length(genes))))
d.assign.threshN0.05 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(-0.05,length(genes))))
d.assign.threshN0.01 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(-0.01,length(genes))))
d.assign.thresh0 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0,length(genes))))
d.assign.thresh0.01 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.01,length(genes))))
d.assign.thresh0.05 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.05,length(genes))))
d.assign.thresh0.1 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.1,length(genes))))
d.assign.thresh0.3 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.3,length(genes))))
d.assign.thresh0.5 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.5,length(genes))))
d.assign.thresh0.7 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.7,length(genes))))
d.assign.thresh0.9 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.9,length(genes))))
d.assign.thresh1 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(1,length(genes))))
d.assign.threshMulti01 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
d.assign.threshMulti0 <- chk.thres.manual(d.tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

d.assign.median <- chk.thres.median(d.tot.genes, genes, cutoff=NA)
d.assign.median1 <- chk.thres.median(d.tot.genes, genes, cutoff=1)
d.assign.q1 <- chk.thres.q1(d.tot.genes, genes, cutoff=NA)
d.assign.q3 <- chk.thres.q3(d.tot.genes, genes, cutoff=NA)
d.assign.mean <- chk.thres.mean(d.tot.genes, genes, cutoff=NA)
d.assign.mean1 <- chk.thres.mean(d.tot.genes, genes, cutoff=1)
d.assign.meansd <- chk.thres.meansd(d.tot.genes, genes, cutoff=NA)
d.assign.meansd1 <- chk.thres.meansd(d.tot.genes, genes, cutoff=1)
d.assign.multi.q1median <- cbind(d.assign.q1[,1:8],d.assign.median[,9:16],d.assign.q1[,17:31])
d.assign.multi.medianq1 <- cbind(d.assign.median[,1:8],d.assign.q1[,9:16],d.assign.median[,17:31])

```

## Examine Gene Expression Distributions

Takeaways regarding marker genes:

* for scaled integrated data (with confounding variables regressed out):
    + iPSC markers (POU5F1, TDGF1, SOX2, EPCAM): clear bimodal distributions
    + MSC markers (THY1, NT5E, ENG, CD44): bimodal distributions squished but still distinct (least clear for CD44)
    + Osteogenic markers (COL1A1, COL1A2, ALPL): clear bimodal distributions (relatively squished for ALPL)
    + Some specific osteogenic markers (RUNX2, BGLAP, CAPG, PHEX, HYOU1): okay distributions
    + Some specific osteogenic markers (DMP1, MEPE, SOST): extremely narrow distributions
    + Chondrogenic markers (ACAN, COL2A1, SOX9): narrow distributions (wider for SOX9)
    + Adipogenic markers (FABP4, PPARG, EBF2): narrow distributions (wider for PPARG)
    + Other osteogenic markers (TNC, PDPN, FBLN7): okay distributions (very narrow for FBLN7)
  
* for log normalized data: 
    + iPSC markers (POU5F1, TDGF1, SOX2, EPCAM): zero-inflated
    + MSC markers (THY1, NT5E, ENG, CD44): zero-inflated
    + Osteogenic markers (COL1A1, COL1A2, ALPL): zero-inflated (bimodal distributions for COL1A1 and COL1A2)
    + Some specific osteogenic markers (RUNX2, CAPG, HYOU1): okay distributions
    + Some specific osteogenic markers (BGLAP, DMP1, PHEX, MEPE, SOST): only zeros
    + Chondrogenic markers (ACAN, COL2A1, SOX9): zero-inflated (only zeros for ACAN)
    + Adipogenic markers (FABP4, PPARG, EBF2): only zeros (zero-inflated for PPARG)
    + Other osteogenic markers (TNC, PDPN, FBLN7): zero-inflated (only zeros for FBLN7)

* for rescaled integrated gene expression data to range from 0 to 1:
    + iPSC markers (POU5F1, TDGF1, SOX2, EPCAM): clear bimodal distributions
    + MSC markers (THY1, NT5E, ENG, CD44): skewed distributions
    + Osteogenic markers (COL1A1, COL1A2, ALPL): clear bimodal distributions (skewed distribution for ALPL)
    + Some specific osteogenic markers (RUNX2, BGLAP, CAPG, PHEX, MEPE, SOST, HYOU1): okay distributions
    + Some specific osteogenic markers (DMP1): extremely narrow distributions
    + Chondrogenic markers (ACAN, COL2A1, SOX9): narrow distributions (wider for SOX9)
    + Adipogenic markers (FABP4, PPARG, EBF2): narrow distributions (wider for PPARG)
    + Other osteogenic markers (TNC, PDPN, FBLN7): okay distributions (very narrow for FBLN7)

* for rescaled log normalized gene expression data to range from 0 to 1
    + iPSC markers (POU5F1, TDGF1, SOX2, EPCAM): zero-inflated
    + MSC markers (THY1, NT5E, ENG, CD44): zero-inflated
    + Osteogenic markers (COL1A1, COL1A2, ALPL): zero-inflated (bimodal distributions for COL1A1 and COL1A2)
    + Some specific osteogenic markers (RUNX2, CAPG, PHEX, HYOU1): okay distributions
    + Some specific osteogenic markers (BGLAP, DMP1, MEPE, SOST): only zeros
    + Chondrogenic markers (ACAN, COL2A1, SOX9): zero-inflated (only zeros for ACAN)
    + Adipogenic markers (FABP4, PPARG, EBF2): zero-inflated (only zeros for EBF2)
    + Other osteogenic markers (TNC, PDPN, FBLN7): zero-inflated (only zeros for FBLN7)

* for non-scaled integrated data (with confounding variables regressed out)
    + iPSC markers (POU5F1, TDGF1, SOX2, EPCAM): clear bimodal distributions
    + MSC markers (THY1, NT5E, ENG, CD44): bimodal distributions squished but still distinct (least clear for CD44)
    + Osteogenic markers (COL1A1, COL1A2, ALPL): clear bimodal distributions (relatively squished for ALPL)
    + Some specific osteogenic markers (RUNX2, CAPG, HYOU1): okay distributions
    + Some specific osteogenic markers (BGLAP, DMP1, PHEX, MEPE, SOST): extremely narrow distributions
    + Chondrogenic markers (ACAN, COL2A1, SOX9): narrow distributions (wider for SOX9)
    + Adipogenic markers (FABP4, PPARG, EBF2): narrow distributions (wider for PPARG)
    + Other osteogenic markers (TNC, PDPN, FBLN7): okay distributions (very narrow for FBLN7)

NEED TO FIX THIS:

* Variable numbers of cells express iPSC markers depending on the threshold (POU5F1 + TDGF1 + SOX2 + EPCAM) (bad markers)
  + for TDGF1 + SOX2, the distribution is so narrow that minor changes in the cutoff drastically change the number of positive cells
* Large proportion of cells express MSC markers (THY1 + NT5E + ENG + CD44) (okay markers)
  + most cells express at least 1 MSC marker
* Large proportion of cells express general osteogenic markers (COL1A1 + COL1A2 + ALPL) (okay markers)
  + most cells express ALPL (most important marker to be called osteogenic)
  + some cells only express COL1A1+COL1A2
* Proportion of cells expressing more specific osteogenic markers (RUNX2 + BGLAP + CAPG + DMP1 + PHEX + MEPE + SOST + HYOU1) hard to discern (okay markers)
  + should exclude DMP1 because it is not expressed in any cell
  + should exclude SOST because it is only expressed in 1 cell
  + should exclude HYOU1 because it is a heat shock protein and may get turned on for any number of reasons aside from osteogenic differentiation
* Variable numbers of cells express chondrogenic markers (ACAN + COL2A1 + SOX9)
  + for ACAN + COL2A1, the distribution is so narrow that minor changes in the cutoff drastically change the number of positive cells
* Variable numbers of cells express adipogenic markers depending on the threshold (FABP4 + PPARG + EBF2)
  + for EBF2, the distribution is so narrow that minor changes in the cutoff drastically change the number of positive cells

While several thresholds were assigned, some cutoffs make more sense than others given the gene expression distributions.

* for scaled integrated data (with confounding variables regressed out):
    + the mean is always 0
    + aside from COL1A1 and COL1A2, the median is always less than the mean

* for log normalized data: 
    + using a cutoff of > 0 may be sufficient
    + the lowest value is 0, so do not use negative thresholds
    + however, confounding variables are not regressed out of these data

* for rescaled integrated gene expression data to range from 0 to 1:
    + the lowest value is 0, so do not use negative thresholds
    + rescaling already scaled data seems unnecessary
    + also the rescaling squishes the data making it harder to pick an optimal cutoff

* for rescaled log normalized gene expression data to range from 0 to 1
    + seems okay
    + using a cutoff of > 0 may be sufficient
    + the lowest value is 0, so do not use negative thresholds
    + any cutoff that exceed 0.25 may be too conservative
    + however, confounding variables are not regressed out of these data

* for non-scaled integrated data (with confounding variables regressed out)
    + using the mean may make more sense than the median based on how the distributions look

* do not consider using different cutoffs for different genes
* using a cutoff based on the mean calculated from all values >1 seems too conservative
* using a cutoff that adds the standard deviation to the mean seems too conservative

### Scaled integrated data (with confounding variables regressed out)

```{r}

#Scaled integrated data (with confounding variables regressed out)

ggplot(tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

### Log normalized data

```{r}

#Log normalized data

ggplot(l.tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

### Rescaled integrated gene expression data to range from 0 to 1

```{r}

#Rescaled integrated gene expression data to range from 0 to 1

ggplot(z.tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

### Rescaled log normalized gene expression data to range from 0 to 1

```{r}

#Rescaled log normalized gene expression data to range from 0 to 1

ggplot(z.l.tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(z.l.tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

### Non-scaled integrated data (with confounding variables regressed out)

```{r}

#Non-scaled integrated data (with confounding variables regressed out)

ggplot(d.tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(d.tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

## Visualize Ad Hoc Assignments

Define functions for visualizing ad hoc assignments.

```{r}

#Define functions for making upsetr plots

plot.upset.HCcellcompare <- function(data, genes, ymax) {
  
  x=1
  T0.H.geneList=c()
  T1.H.geneList=c()
  T2.H.geneList=c()
  T0.C.geneList=c()
  T1.C.geneList=c()
  T2.C.geneList=c()
  while(x <= length(genes)) {
    if(sum(data[which(data$Stage=="Time 0" & data$Species=="Human"),genes[x]])>0){ T0.H.geneList=c(T0.H.geneList,genes[x]) }
    if(sum(data[which(data$Stage=="Time 1" & data$Species=="Human"),genes[x]])>0){ T1.H.geneList=c(T1.H.geneList,genes[x]) }
    if(sum(data[which(data$Stage=="Time 2" & data$Species=="Human"),genes[x]])>0){ T2.H.geneList=c(T2.H.geneList,genes[x]) }
    if(sum(data[which(data$Stage=="Time 0" & data$Species=="Chimp"),genes[x]])>0){ T0.C.geneList=c(T0.C.geneList,genes[x]) }
    if(sum(data[which(data$Stage=="Time 1" & data$Species=="Chimp"),genes[x]])>0){ T1.C.geneList=c(T1.C.geneList,genes[x]) }
    if(sum(data[which(data$Stage=="Time 2" & data$Species=="Chimp"),genes[x]])>0){ T2.C.geneList=c(T2.C.geneList,genes[x]) }
    x=x+1
  }
    
  plot.ls <- list(
    T0.Human = upset(data[which(data$Stage=="Time 0" & data$Species=="Human"),T0.H.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T0.H.geneList, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage=="Time 1" & data$Species=="Human"),T1.H.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T1.H.geneList, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage=="Time 2" & data$Species=="Human"),T2.H.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T2.H.geneList, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage=="Time 0" & data$Species=="Chimp"),T0.C.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T0.C.geneList, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage=="Time 1" & data$Species=="Chimp"),T1.C.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T1.C.geneList, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage=="Time 2" & data$Species=="Chimp"),T2.C.geneList],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE,
                     order.by="degree", decreasing=TRUE, sets=T2.C.geneList, mainbar.y.max=ymax)
  )
  
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
  }
    
}

plot.upset.sub.HCcellcompare <- function(data, genes, intersections, ymax) {
  
  plot.ls <- list(
    T0.Human = upset(data[which(data$Stage=="Time 0" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage=="Time 1" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage=="Time 2" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage=="Time 0" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage=="Time 1" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage=="Time 2" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=FALSE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax)
  )
  
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
  }

}

```

## Visualize General Ad Hoc Assignments

Look at which cells meet threshold cutoffs.
(note: if all cells in a subset were negative for a marker gene, that gene is excludeded from the upsetr plot)

Takeaways regarding assignment cutoffs (number of cells reaching cutoffs):

* for scaled integrated data (with confounding variables regressed out):
    + mean = thresh0
    + threshN0.1 > threshN0.05 > threshN0.01 > thresh0 > thresh0.01 > thresh0.05 > thresh0.1 > thresh0.3 > thresh0.5 > thresh0.7 > thresh0.9 > thresh1
    + q1 > median > q3

* for log normalized data: 
    + thresh0 = thresh0.01 = thresh0.05 = thresh0.1 > thresh0.3 > thresh0.5 > thresh0.7 > thresh0.9 > thresh1
    + q1 > median ~ mean > q3

* for rescaled integrated gene expression data to range from 0 to 1:
    + mean and median cutoffs produce the same results as those for scaled integrated data (with confounding variables regressed out)

* for rescaled log normalized gene expression data to range from 0 to 1
    + thresh0, mean, and median cutoffs produce the same results as those for log normalized data

* for non-scaled integrated data (with confounding variables regressed out)
    + most of these cutoffs look too lenient

The most sensible cutoff for general cell types (iPSC, MSC, Osteogenic):

* thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)
* for good measure run a couple other assignments at different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)
* ips = +POU5F1,+TDGF,+SOX2,+EPCAM
* msc = +THY1,+NT5E,+ENG,+CD44
* ost = -POU5F1,+COL1A1,+COL1A2,+ALPL

### Scaled integrated data (with confounding variables regressed out)

```{r}

#Scaled integrated data (with confounding variables regressed out)

assignments <- list("thresh0"=assign.thresh0,
                    "thresh0.01"=assign.thresh0.01,
                    "thresh0.05"=assign.thresh0.05,
                    "thresh0.1"=assign.thresh0.1,
                    "thresh0.3"=assign.thresh0.3,
                    "thresh0.5"=assign.thresh0.5,
                    "thresh0.7"=assign.thresh0.7,
                    "thresh0.9"=assign.thresh0.9,
                    "thresh1"=assign.thresh1,
                    "threshN0.01"=assign.threshN0.01,
                    "threshN0.05"=assign.threshN0.05,
                    "threshN0.1"=assign.threshN0.1,
                    "median"=assign.median,
                    "q1"=assign.q1,
                    "q3"=assign.q3,
                    "mean"=assign.mean)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #ipsc candidate genes
  print(paste0(names(assignments[x]),": iPSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[1:4],20000)

  #msc candidate genes
  print(paste0(names(assignments[x]),": MSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[5:8],20000)

  #osteogenic candidate genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

  #chondrogenic candidate genes
  print(paste0(names(assignments[x]),": Chondrogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[20:22],20000)

  #adipogenic candidate genes
  print(paste0(names(assignments[x]),": Adipogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[23:25],20000)

  x=x+1
  
}

```

### Log normalized data

```{r}

#Log normalized data

assignments <- list("l.thresh0"=l.assign.thresh0,
                    "l.thresh0.01"=l.assign.thresh0.01,
                    "l.thresh0.05"=l.assign.thresh0.05,
                    "l.thresh0.1"=l.assign.thresh0.1,
                    "l.thresh0.3"=l.assign.thresh0.3,
                    "l.thresh0.5"=l.assign.thresh0.5,
                    "l.thresh0.7"=l.assign.thresh0.7,
                    "l.thresh0.9"=l.assign.thresh0.9,
                    "l.thresh1"=l.assign.thresh1,
                    "l.median"=l.assign.median,
                    "l.q1"=l.assign.q1,
                    "l.q3"=l.assign.q3,
                    "l.mean"=l.assign.mean)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #ipsc candidate genes
  print(paste0(names(assignments[x]),": iPSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[1:4],20000)

  #msc candidate genes
  print(paste0(names(assignments[x]),": MSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[5:8],20000)

  #osteogenic candidate genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

  #chondrogenic candidate genes
  print(paste0(names(assignments[x]),": Chondrogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[20:22],20000)

  #adipogenic candidate genes
  print(paste0(names(assignments[x]),": Adipogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[23:25],20000)
  
  x=x+1

}

```

### Rescaled integrated gene expression data to range from 0 to 1

```{r}

#Rescaled integrated gene expression data to range from 0 to 1

assignments <- list("z.median"=z.assign.median,
                    "z.mean"=z.assign.mean)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #ipsc candidate genes
  print(paste0(names(assignments[x]),": iPSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[1:4],20000)

  #msc candidate genes
  print(paste0(names(assignments[x]),": MSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[5:8],20000)

  #osteogenic candidate genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

  #chondrogenic candidate genes
  print(paste0(names(assignments[x]),": Chondrogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[20:22],20000)

  #adipogenic candidate genes
  print(paste0(names(assignments[x]),": Adipogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[23:25],20000)

  x=x+1

}

```

### Rescaled log normalized gene expression data to range from 0 to 1

```{r}

#Rescaled log normalized gene expression data to range from 0 to 1

assignments <- list("z.l.thresh0"=z.l.assign.thresh0,
                    "z.l.thresh0.01"=z.l.assign.thresh0.01,
                    "z.l.thresh0.05"=z.l.assign.thresh0.05,
                    "z.l.thresh0.1"=z.l.assign.thresh0.1,
                    "z.l.median"=z.l.assign.median,
                    "z.l.mean"=z.l.assign.mean)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #ipsc candidate genes
  print(paste0(names(assignments[x]),": iPSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[1:4],20000)

  #msc candidate genes
  print(paste0(names(assignments[x]),": MSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[5:8],20000)

  #osteogenic candidate genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

  #chondrogenic candidate genes
  print(paste0(names(assignments[x]),": Chondrogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[20:22],20000)

  #adipogenic candidate genes
  print(paste0(names(assignments[x]),": Adipogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[23:25],20000)

  x=x+1

}

```

### Non-scaled integrated data (with confounding variables regressed out)

```{r}

#Non-scaled integrated data (with confounding variables regressed out)

assignments <- list("d.thresh0"=d.assign.thresh0,
                    "d.thresh0.01"=d.assign.thresh0.01,
                    "d.thresh0.05"=d.assign.thresh0.05,
                    "d.thresh0.1"=d.assign.thresh0.1,
                    "d.thresh0.3"=d.assign.thresh0.3,
                    "d.thresh0.5"=d.assign.thresh0.5,
                    "d.thresh0.7"=d.assign.thresh0.7,
                    "d.thresh0.9"=d.assign.thresh0.9,
                    "d.thresh1"=d.assign.thresh1,
                    "d.threshN0.01"=d.assign.threshN0.01,
                    "d.threshN0.05"=d.assign.threshN0.05,
                    "d.threshN0.1"=d.assign.threshN0.1,
                    "d.median"=d.assign.median,
                    "d.q1"=d.assign.q1,
                    "d.q3"=d.assign.q3,
                    "d.mean"=d.assign.mean)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #ipsc candidate genes
  print(paste0(names(assignments[x]),": iPSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[1:4],20000)

  #msc candidate genes
  print(paste0(names(assignments[x]),": MSC Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[5:8],20000)

  #osteogenic candidate genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

  #chondrogenic candidate genes
  print(paste0(names(assignments[x]),": Chondrogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[20:22],20000)

  #adipogenic candidate genes
  print(paste0(names(assignments[x]),": Adipogenic Candidate Genes"))
  plot.upset.HCcellcompare(assignment,genes[23:25],20000)

  x=x+1

}

```

## Visualize Osteogenic Ad Hoc Assignments

Since the most sensible cutoff for general cell types (iPSC, MSC, Osteogenic) are

* thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)
* for good measure run a couple other assignments at different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)

we only consider these for more specific osteogenic ad hoc assignments.

The remaining question for these is which genes are the most appropriate to use. Decided to use the candidate genes specified in Dallas et al. 2013. Endocr Rev. (see Figure 3). However, not all of these genes may be appropriate:

* RUNX2 (good marker)
    + directs early osteoblast differentiation
* BGLAP (good marker)
    + expressed by mature osteoblasts and early osteocytes
    + also known as osteocalcin (OCN)
* PHEX (okay marker)
    + involved in phosphate homeostasis
    + expressed in later stages of osteogenesis
    + [in my data, this gene is lowly expressed across cell types]
* DMP1 (okay marker)
    + expressed in mineralizing osteoblasts and mature osteocytes
    + [in my data, this gene is lowly expressed across cell types]
* CAPG (okay marker)
    + expressed in mineralizing osteoblasts and mature osteocytes
    + [in my data, this gene appears widely expressed across cell types]
* MEPE (okay marker)
    + expressed in mineralizing osteoblasts and mature osteocytes
    + [in my data, this gene is lowly expressed across cell types]
* E11 (bad marker)
    + earliest osteocyte marker expressed during differentiation but not found in mature osteocytes
    + [in my data, this gene not part of the ortho-exon file]
* SOST (okay marker)
    + expression is confined to mature osteocytes
    + [in my data, this gene is lowly expressed across cell types]
* HYOU1 (bad marker)
    + only found in mature osteocytes within the hypoxic environment of the mineralized bone matrix
    + also known as ORP150
    + [in my data, this gene appears widely expressed across cell types]
    + [heat shock proteins might not be a good candidate markers for cell culture systems]

Notes on cutoffs and gene combinations:

* using higher cutoffs - increases the number of preosteoblasts and osteoblasts
* using fewer genes - increases the number of preosteoblasts and osteoblasts
* using lower cutoffs - increases the number of embedding osteoblasts
* using the -0.05 cutoff - seems excessive
* using 5B genes (RUNX2, BGLAP, CAPG, PHEX, and MEPE) - increases the number of mineralizing osteoblasts

The most sensible gene combination for osteogenic cell types (preosteoblasts, osteoblasts, embedding/mineralizing osteoblasts, maturing osteocytes):

* 8 genes - most consistent with literature even though it is the most conservative criteria
* 6A genes - intermediate number of genes that removes a highly expressed gene in all cell types (CAPG) and a heat-shock protein (HYOU1)
* 4 genes - simplest gene combination

### Osteogenic Genes Only

```{r}

#Scaled integrated data (with confounding variables regressed out)

assignments <- list("thresh0"=assign.thresh0,
                    "thresh0.01"=assign.thresh0.01,
                    "thresh0.05"=assign.thresh0.05,
                    "threshN0.01"=assign.threshN0.01,
                    "threshN0.05"=assign.threshN0.05,
                    "threshN0.1"=assign.threshN0.1)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #more specific osteogenic candidate genes - all 8 potential genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (8 genes)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:16)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","CAPG","DMP1","PHEX"),
                                                  list("BGLAP","CAPG","DMP1","PHEX","MEPE"),
                                                  list("CAPG","DMP1","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude HYOU1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:15)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","CAPG","DMP1","PHEX"),
                                                  list("BGLAP","CAPG","DMP1","PHEX","MEPE"),
                                                  list("CAPG","DMP1","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude CAPG)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:10,12:16)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","DMP1","PHEX"),
                                                  list("BGLAP","DMP1","PHEX","MEPE"),
                                                  list("DMP1","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option C)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:11,13:16)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","CAPG","PHEX"),
                                                  list("BGLAP","CAPG","PHEX","MEPE"),
                                                  list("CAPG","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  #more specific osteogenic candidate genes - 6 potential genes (exclude HYOU1+CAPG)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:10,12:15)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","DMP1","PHEX"),
                                                  list("BGLAP","DMP1","PHEX","MEPE"),
                                                  list("DMP1","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 6 potential genes (exclude HYOU1+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:11,13:15)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","CAPG","PHEX"),
                                                  list("BGLAP","CAPG","PHEX","MEPE"),
                                                  list("CAPG","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 6 potential genes (exclude CAPG+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option C)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:10,13:16)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","PHEX"),
                                                  list("BGLAP","PHEX","MEPE"),
                                                  list("PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 5 potential genes (exclude HYOU1+CAPG+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (5 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:10,13:15)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","PHEX"),
                                                  list("BGLAP","PHEX","MEPE"),
                                                  list("PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 5 potential genes (exclude HYOU1+DMP1+SOST)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (5 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:11,13:14)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","CAPG","PHEX"),
                                                  list("BGLAP","CAPG","PHEX","MEPE"),
                                                  list("CAPG","PHEX","MEPE")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 4 potential genes (exclude HYOU1+CAPG+DMP1+SOST)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (4 genes)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(9:10,13:14)],
                               intersections=list(list("RUNX2"),
                                                  list("RUNX2","BGLAP"),
                                                  list("BGLAP","PHEX"),
                                                  list("BGLAP","PHEX","MEPE"),
                                                  list("PHEX","MEPE")),
                               ymax=15000)
  
  x=x+1
  
}

```

### Osteogenic and General Genes Only

Include general iPSC markers to filter out non-osteogenic cells (-POU5F1, -TDGF, -SOX2, -EPCAM) and general osteogenic markers to ensure retention of osteogenic cells (+ALPL).

Using these general ad hoc markers improves the osteogenic ad hoc results. Thus, these will be used along with the following cutoff criteria to define osteogenic cell types (preosteoblasts, osteoblasts, embedding/mineralizing osteoblasts, maturing osteocytes):

* thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)
* for good measure run a couple other assignments at different cutoffs but not as many as for the general ad hoc assignments (thresh-0.01,thresh0.01)
* 8 genes - most consistent with literature even though it is the most conservative criteria
    + pre = +RUNX2,-BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,+SOST,+HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
* 6A genes - intermediate number of genes that removes a highly expressed gene in all cell types (CAPG) and a heat-shock protein (HYOU1)
    + pre = +RUNX2,-BGLAP,-DMP1,-PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-DMP1,-PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+DMP1,+PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+DMP1,+PHEX,+MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+DMP1,+PHEX,+MEPE,+SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
* 4 genes - simplest gene combination
    + pre = +RUNX2,-BGLAP,-PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+PHEX,+MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+PHEX,+MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL

```{r}

#Scaled integrated data (with confounding variables regressed out)

assignments <- list("thresh0"=assign.thresh0,
                    "thresh0.01"=assign.thresh0.01,
                    "thresh0.05"=assign.thresh0.05,
                    "threshN0.01"=assign.threshN0.01,
                    "threshN0.05"=assign.threshN0.05,
                    "threshN0.1"=assign.threshN0.1)

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])

  #more specific osteogenic candidate genes - all 8 potential genes
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (8 genes)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:16)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","CAPG","DMP1","PHEX"),
                                                  list("ALPL","BGLAP","CAPG","DMP1","PHEX","MEPE"),
                                                  list("ALPL","CAPG","DMP1","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude HYOU1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:15)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","CAPG","DMP1","PHEX"),
                                                  list("ALPL","BGLAP","CAPG","DMP1","PHEX","MEPE"),
                                                  list("ALPL","CAPG","DMP1","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude CAPG)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:10,12:16)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","DMP1","PHEX"),
                                                  list("ALPL","BGLAP","DMP1","PHEX","MEPE"),
                                                  list("ALPL","DMP1","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 7 potential genes (exclude DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (7 genes, option C)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:11,13:16)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","CAPG","PHEX"),
                                                  list("ALPL","BGLAP","CAPG","PHEX","MEPE"),
                                                  list("ALPL","CAPG","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  #more specific osteogenic candidate genes - 6 potential genes (exclude HYOU1+CAPG)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:10,12:15)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","DMP1","PHEX"),
                                                  list("ALPL","BGLAP","DMP1","PHEX","MEPE"),
                                                  list("ALPL","DMP1","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 6 potential genes (exclude HYOU1+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:11,13:15)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","CAPG","PHEX"),
                                                  list("ALPL","BGLAP","CAPG","PHEX","MEPE"),
                                                  list("ALPL","CAPG","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 6 potential genes (exclude CAPG+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (6 genes, option C)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:10,13:16)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","PHEX"),
                                                  list("ALPL","BGLAP","PHEX","MEPE"),
                                                  list("ALPL","PHEX","MEPE","SOST","HYOU1")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 5 potential genes (exclude HYOU1+CAPG+DMP1)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (5 genes, option A)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:10,13:15)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","PHEX"),
                                                  list("ALPL","BGLAP","PHEX","MEPE"),
                                                  list("ALPL","PHEX","MEPE","SOST")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 5 potential genes (exclude HYOU1+DMP1+SOST)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (5 genes, option B)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:11,13:14)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","CAPG","PHEX"),
                                                  list("ALPL","BGLAP","CAPG","PHEX","MEPE"),
                                                  list("ALPL","CAPG","PHEX","MEPE")),
                               ymax=15000)
  
  #more specific osteogenic candidate genes - 4 potential genes (exclude HYOU1+CAPG+DMP1+SOST)
  print(paste0(names(assignments[x]),": Osteogenic Candidate Genes (4 genes)"))
  plot.upset.sub.HCcellcompare(data=assignment,
                               genes=genes[c(1:4,19,9:10,13:14)],
                               intersections=list(list("ALPL","RUNX2"),
                                                  list("ALPL","RUNX2","BGLAP"),
                                                  list("ALPL","BGLAP","PHEX"),
                                                  list("ALPL","BGLAP","PHEX","MEPE"),
                                                  list("ALPL","PHEX","MEPE")),
                               ymax=15000)
  
  x=x+1
  
}

```

```{r, eval=FALSE}

#Additional plots to possibly consider

#all genes
plot.upset.HCcellcompare(assignments,genes[c(1:8,17:19)],20000)

#msc candidate genes
plot.upset.HCcellcompare(assignments,genes[c(19,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2
plot.upset.HCcellcompare(assignments,genes[c(17,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2
plot.upset.HCcellcompare(assignments,genes[c(17:18,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2
  
#osteogenic candidate genes
plot.upset.HCcellcompare(assignments,genes[17:19],20000)

#more specific osteogenic candidate genes
plot.upset.HCcellcompare(assignment,genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignment,genes[c(9:10,13:14)],20000)
plot.upset.HCcellcompare(assignments,genes[c(9:16)],20000)
plot.upset.HCcellcompare(assignments,genes[c(9:11,17:18)],20000)
plot.upset.HCcellcompare(assignments,genes[c(11,13:14,17:18)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$ALPL==1),],genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1),],genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1),],genes[c(9:11,13:14)],20000)

```

## Examine General Ad Hoc Plots

Cutoff scheme:

* thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)
* for good measure run a couple other assignments at different cutoffs (thresh-0.01,thresh-0.05,thresh-0.1,thresh0.01,thresh0.05,thresh0.1)
* ips = +POU5F1,+TDGF,+SOX2,+EPCAM
* msc = +THY1,+NT5E,+ENG,+CD44
* ost = -POU5F1,+COL1A1,+COL1A2,+ALPL

To make this cutoff scheme work, the coded cutoff actually needs to be:

* ips = +POU5F1 [or] +TDGF [or] +SOX2 [or] +EPCAM [&] -NT5E,-ENG,-CD44
* msc = +THY1   [or] +NT5E [or] +ENG  [or] +CD44  [&] -POU5F1,-TDGF,-SOX2,-EPCAM,-ALPL
* ost = +ALPL [&] -POU5F1,-TDGF,-SOX2,-EPCAM

```{r}

hoc.assign <- function(assignments) {
  assignments$assign <- "other"
  i=1
  while(i <= length(rownames(assignments))) {
    if((assignments$POU5F1[i] == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0) |
       (assignments$TDGF1[i]  == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0) |
       (assignments$SOX2[i]   == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0) |
       (assignments$EPCAM[i]  == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0))
      {assignments$assign[i] <- "iPSC"}
    if((assignments$THY1[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$NT5E[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$ENG[i]  == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$CD44[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0))
      {assignments$assign[i] <- "MSC"}
    if((assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0))
      {assignments$assign[i] <- "Osteogenic"}
    i=i+1
  }
 return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
```

```{r}

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.thresh0 <- factor(data@meta.data$AdHoc.Assign.thresh0,
#                                              levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.01)
ad.hoc.assign <- hoc.assign(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.threshN0.01 <- factor(data@meta.data$AdHoc.Assign.threshN0.01,
#                                                  levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.05)
ad.hoc.assign <- hoc.assign(assign.threshN0.05)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.threshN0.05 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.threshN0.05 <- factor(data@meta.data$AdHoc.Assign.threshN0.05,
#                                                  levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, threshN0.1)
ad.hoc.assign <- hoc.assign(assign.threshN0.1)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.threshN0.1 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.threshN0.1 <- factor(data@meta.data$AdHoc.Assign.threshN0.1,
#                                                 levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.thresh0.01 <- factor(data@meta.data$AdHoc.Assign.thresh0.01,
#                                                 levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.05)
ad.hoc.assign <- hoc.assign(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.thresh0.05 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.thresh0.05 <- factor(data@meta.data$AdHoc.Assign.thresh0.05,
#                                                 levels=c("iPSC","MSC","Osteogenic","other"))

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.1)
ad.hoc.assign <- hoc.assign(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$AdHoc.Assign.thresh0.1 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$AdHoc.Assign.thresh0.1 <- factor(data@meta.data$AdHoc.Assign.thresh0.1,
#                                                levels=c("iPSC","MSC","Osteogenic","other"))

```

```{r}

assignments <- list("AdHoc.Assign.thresh0"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.thresh0")],
                    "AdHoc.Assign.threshN0.01"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.threshN0.01")],
                    "AdHoc.Assign.threshN0.05"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.threshN0.05")],
                    "AdHoc.Assign.threshN0.1"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.threshN0.1")],
                    "AdHoc.Assign.thresh0.01"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.thresh0.01")],
                    "AdHoc.Assign.thresh0.05"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.thresh0.05")],
                    "AdHoc.Assign.thresh0.1"=data@meta.data[,c("Species","Stage","Individual","AdHoc.Assign.thresh0.1")])

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])
  
  cutoff <-colnames(assignment)[4]
  print(cutoff)
  
  colnames(assignment) <- c("Species","Stage","Individual","AdHoc.Assign")
  
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 2")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 2")])
  
  print(ggplot(assignment, aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  print(ggplot(assignment, aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Species), stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  
  print(ggplot(assignment, aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(stat="count", position="dodge"))
  print(ggplot(assignment, aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Individual), stat="count", position="dodge"))
  
  print(DimPlot(data,
                group.by="Species",
                split.by=cutoff,
                reduction="umap",
                cols=c("grey","black")) +
          labs(x="UMAP1",y="UMAP2"))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("pink","mediumorchid","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Species=="Chimp"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("pink","mediumorchid","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("pink")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 1"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("mediumorchid")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 2"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))

  x=x+1

}

```

## Examine Osteogenic Ad Hoc Plots

Cutoff scheme:

* thresh0 (or mean) for scaled integrated data (with confounding variables regressed out)
* for good measure run a couple other assignments at different cutoffs but not as many as for the general ad hoc assignments (thresh-0.01,thresh0.01)
* 8 genes - most consistent with literature even though it is the most conservative criteria
    + pre = +RUNX2,-BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,-MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,-SOST,-HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+CAPG,+DMP1,+PHEX,+MEPE,+SOST,+HYOU1 & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
* 6A genes - intermediate number of genes that removes a highly expressed gene in all cell types (CAPG) and a heat-shock protein (HYOU1)
    + pre = +RUNX2,-BGLAP,-DMP1,-PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-DMP1,-PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+DMP1,+PHEX,-MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+DMP1,+PHEX,+MEPE,-SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+DMP1,+PHEX,+MEPE,+SOST & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
* 4 genes - simplest gene combination
    + pre = +RUNX2,-BGLAP,-PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + obl = +RUNX2,+BGLAP,-PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + emb = -RUNX2,+BGLAP,+PHEX,-MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + min = -RUNX2,+BGLAP,+PHEX,+MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL
    + mat = -RUNX2,-BGLAP,+PHEX,+MEPE & -POU5F1,-TDGF,-SOX2,-EPCAM,+ALPL

This cutoff scheme should work as is, but closely related gene expression patterns are also defined as x.osteoblast, x.embedding osteoblast, x.mineralizing osteoblast, and x.maturing osteocyte:

* 8 genes
    + x.obl = +BGLAP [and] -CAPG,-DMP1,-PHEX,-MEPE,-SOST,-HYOU1
    + x.emb = +CAPG [or] +DMP1 [or] +PHEX [and] -MEPE,-SOST,-HYOU1
    + x.min = +MEPE [and] -SOST,-HYOU1
    + x.mat = +SOST [or] +HYOU1
* 6A genes
    + x.obl = +BGLAP [and] -DMP1,-PHEX,-MEPE,-SOST
    + x.emb = +DMP1 [or] +PHEX [and] -MEPE,-SOST
    + x.min = +MEPE [and] -SOST
    + x.mat = +SOST
* 4 genes
    + x.obl = +BGLAP [and] -PHEX,-MEPE
    + x.emb = +PHEX [and] -MEPE
    + x.min = +MEPE
    + x.mat = +MEPE [and] -RUNX2,-BGLAP
    
Assignments using 8 genes

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes)

hoc.assign.x.8 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #lenient criteria
      if((assignments$SOST[i] ==1) |
         (assignments$HYOU1[i]==1)) {
        assignments$assign[i] <- "x.maturing osteocyte"
      }
      if(assignments$MEPE[i]==1 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) {
        assignments$assign[i] <- "x.mineralizing osteoblast"
      }
      if((assignments$CAPG[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) |
         (assignments$DMP[i] ==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) |
         (assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0)) {
        assignments$assign[i] <- "x.embedding osteoblast"
      }
      if(assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) {
        assignments$assign[i] <- "x.osteoblast"
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==1 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==1 & assignments$SOST[i] ==1 & assignments$HYOU1[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.8 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #lenient criteria
      if((assignments$SOST[i] ==1) |
         (assignments$HYOU1[i]==1)) {
        assignments$assign[i] <- "maturing osteocyte"
      }
      if(assignments$MEPE[i]==1 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) {
        assignments$assign[i] <- "mineralizing osteoblast"
      }
      if((assignments$CAPG[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) |
         (assignments$DMP[i] ==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) |
         (assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0)) {
        assignments$assign[i] <- "embedding osteoblast"
      }
      if(assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0 & assignments$HYOU1[i]==0) {
        assignments$assign[i] <- "osteoblast"
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==0 & assignments$DMP[i]  ==0 & assignments$PHEX[i] ==0 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==0 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==1 & assignments$SOST[i] ==0 & assignments$HYOU1[i]==0) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 &
         assignments$CAPG[i] ==1 & assignments$DMP[i]  ==1 & assignments$PHEX[i] ==1 &
         assignments$MEPE[i] ==1 & assignments$SOST[i] ==1 & assignments$HYOU1[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

```

```{r}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.8(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.8gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.8gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.8gene.thresh0,
#                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.8(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.8gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.8gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.8gene.thresh0,
#                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, threshN0.01)
ad.hoc.assign <- hoc.assign.x.8(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.8gene.threshN0.01,
#                                                             levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                      "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                      "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.8(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.8gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.8gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.8gene.threshN0.01,
#                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                    "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (8 genes) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.8(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.8gene.thresh0.01,
#                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.8(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.8gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.8gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.8gene.thresh0.01,
#                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                   "unknown osteogenic","not osteogenic","other"))

```

Assignments using 6A genes

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes)

hoc.assign.x.6 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #lenient criteria
      if(assignments$SOST[i]==1) {
        assignments$assign[i] <- "x.maturing osteocyte"
      }
      if(assignments$MEPE[i]==1 & assignments$SOST[i]==0) {
        assignments$assign[i] <- "x.mineralizing osteoblast"
      }
      if((assignments$DMP[i] ==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) |
         (assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0)) {
        assignments$assign[i] <- "x.embedding osteoblast"
      }
      if(assignments$BGLAP[i]==1 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {
        assignments$assign[i] <- "x.osteoblast"
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1 & assignments$SOST[i]==0) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1 & assignments$SOST[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.6 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #lenient criteria
      if(assignments$SOST[i]==1) {
        assignments$assign[i] <- "maturing osteocyte"
      }
      if(assignments$MEPE[i]==1 & assignments$SOST[i]==0) {
        assignments$assign[i] <- "mineralizing osteoblast"
      }
      if((assignments$DMP[i] ==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) |
         (assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0)) {
        assignments$assign[i] <- "embedding osteoblast"
      }
      if(assignments$BGLAP[i]==1 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {
        assignments$assign[i] <- "osteoblast"
      }
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$DMP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0 & assignments$SOST[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1 & assignments$SOST[i]==0) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$DMP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1 & assignments$SOST[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

```

```{r}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.6(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.6gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.6gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.6gene.thresh0,
#                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.6(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.6gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.6gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.6gene.thresh0,
#                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, threshN0.01)
ad.hoc.assign <- hoc.assign.x.6(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.6gene.threshN0.01,
#                                                             levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                      "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                      "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.6(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.6gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.6gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.6gene.threshN0.01,
#                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                    "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (6A genes) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.6(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.6gene.thresh0.01,
#                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.6(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.6gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.6gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.6gene.thresh0.01,
#                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                   "unknown osteogenic","not osteogenic","other"))

```

Assignments using 4 genes

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes)

hoc.assign.x.4 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

hoc.assign.4 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

```

```{r}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.4(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.4gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.4gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.4gene.thresh0,
#                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.4gene.thresh0 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.4gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.4gene.thresh0,
#                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, threshN0.01)
ad.hoc.assign <- hoc.assign.x.4(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.4gene.threshN0.01,
#                                                             levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                      "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                      "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4(assign.threshN0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.4gene.threshN0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.4gene.threshN0.01 <- factor(data@meta.data$OstAdHoc.Assign.4gene.threshN0.01,
#                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                    "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (4 genes) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.4(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.4gene.thresh0.01,
#                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
#                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
#data@meta.data$OstAdHoc.Assign.4gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
#data@meta.data$OstAdHoc.Assign.4gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.4gene.thresh0.01,
#                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
#                                                                   "unknown osteogenic","not osteogenic","other"))

```

```{r}

assignments <- list("OstAdHoc.Assign.x.8gene.thresh0"    =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.8gene.thresh0")],
                    "OstAdHoc.Assign.x.8gene.threshN0.01"=data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.8gene.threshN0.01")],
                    "OstAdHoc.Assign.x.8gene.thresh0.01" =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.8gene.thresh0.01")],
                    
                    "OstAdHoc.Assign.x.6gene.thresh0"    =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.6gene.thresh0")],
                    "OstAdHoc.Assign.x.6gene.threshN0.01"=data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.6gene.threshN0.01")],
                    "OstAdHoc.Assign.x.6gene.thresh0.01" =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.6gene.thresh0.01")],
                    
                    "OstAdHoc.Assign.x.4gene.thresh0"    =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.4gene.thresh0")],
                    "OstAdHoc.Assign.x.4gene.threshN0.01"=data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.4gene.threshN0.01")],
                    "OstAdHoc.Assign.x.4gene.thresh0.01" =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.x.4gene.thresh0.01")])

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])
  
  cutoff <-colnames(assignment)[4]
  print(cutoff)
  
  colnames(assignment) <- c("Species","Stage","Individual","AdHoc.Assign")
  
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 2")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 2")])

  assignment$AdHoc.Assign <- factor(assignment$AdHoc.Assign,
                                    levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                             "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte","unknown osteogenic"),
                                    labels=c("pre","obl","emb","min","mat","x.obl","x.emb","x.min","x.mat","unk"))
    
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Species), stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(stat="count", position="dodge"))
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Individual), stat="count", position="dodge"))
  
  print(DimPlot(data,
                group.by="Species",
                split.by=cutoff,
                reduction="umap",
                cols=c("grey","black")) +
          labs(x="UMAP1",y="UMAP2"))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey","lightgrey","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Species=="Chimp"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey","lightgrey","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 1"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 2"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))

  print(DimPlot(data,
                group.by="Stage",
                split.by=cutoff,
                reduction="umap",
                cols=c("lightgrey","lightgrey","dodgerblue")) +
          labs(x="UMAP1",y="UMAP2"))

  x=x+1

}

```

```{r}

assignments <- list("OstAdHoc.Assign.8gene.thresh0"      =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.8gene.thresh0")],
                    "OstAdHoc.Assign.8gene.threshN0.01"  =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.8gene.threshN0.01")],
                    "OstAdHoc.Assign.8gene.thresh0.01"   =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.8gene.thresh0.01")],
                    
                    "OstAdHoc.Assign.6gene.thresh0"      =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.6gene.thresh0")],
                    "OstAdHoc.Assign.6gene.threshN0.01"  =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.6gene.threshN0.01")],
                    "OstAdHoc.Assign.6gene.thresh0.01"   =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.6gene.thresh0.01")],
                    
                    "OstAdHoc.Assign.4gene.thresh0"      =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.4gene.thresh0")],
                    "OstAdHoc.Assign.4gene.threshN0.01"  =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.4gene.threshN0.01")],
                    "OstAdHoc.Assign.4gene.thresh0.01"   =data@meta.data[,c("Species","Stage","Individual","OstAdHoc.Assign.4gene.thresh0.01")])

x=1
while (x <= length(assignments)) {

  assignment <- as.data.frame(assignments[[x]])
  
  cutoff <-colnames(assignment)[4]
  print(cutoff)
  
  colnames(assignment) <- c("Species","Stage","Individual","AdHoc.Assign")
  
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Human" & assignment$Stage=="Time 2")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 0")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 1")])
  table(assignment$AdHoc.Assign[which(assignment$Species=="Chimp" & assignment$Stage=="Time 2")])

  assignment$AdHoc.Assign <- factor(assignment$AdHoc.Assign,
                                    levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic"),
                                    labels=c("pre","obl","emb","min","mat","unk"))
    
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Species)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Species), stat="count", position="dodge") +
        scale_fill_manual(values=c("grey","black")))
  
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(stat="count", position="dodge"))
  print(ggplot(subset(assignment, !is.na(AdHoc.Assign)), aes(x=AdHoc.Assign, fill=Individual)) +
        geom_bar(mapping = aes(x=AdHoc.Assign, y=..prop.., group=Individual), stat="count", position="dodge"))
  
  print(DimPlot(data,
                group.by="Species",
                split.by=cutoff,
                reduction="umap",
                cols=c("grey","black")) +
          labs(x="UMAP1",y="UMAP2"))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey","lightgrey","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Species=="Chimp"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey","lightgrey","dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))
  
  print(CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 1"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("lightgrey")) +
                                   labs(x="UMAP1",y="UMAP2")),
                                (DimPlot(subset(data,subset=Stage=="Time 2"),
                                         group.by="Stage",
                                         split.by=cutoff,
                                         reduction="umap",
                                         cols=c("dodgerblue")) +
                                   labs(x="UMAP1",y="UMAP2"))), ncol=1))

  print(DimPlot(data,
                group.by="Stage",
                split.by=cutoff,
                reduction="umap",
                cols=c("lightgrey","lightgrey","dodgerblue")) +
          labs(x="UMAP1",y="UMAP2"))

  x=x+1

}

```
