---
title: "10x-data-plot-clusters-cellidentity"
author: "Genevieve Housman"
date: "4/26/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Determining Cell Identity

Try different clustering methods on 10X data to assess cell identities.

Data being examined comes from 6 humans and 6 chimpanzees.
- GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
- GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
- GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
- HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
- HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)
- HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
- HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
- HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)
- HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
- HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
- HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)
- HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
- HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
- HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)
- HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
- HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
- HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)
- HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
- HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
- HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)

Each dataset being evaluated comprises gene count data that were seperately calculated using both the hg38 human genome and the panTro6 chimpanzee genome, as well as the corresponding ortho-exon annotation in cellranger.Species-specific assignments were determined using human-chimpanzee genome combination with ortho-exon annotation.

```{r load libraries}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)

library(tidyr)
library(UpSetR)
library(reshape2)

library(CountClust)
library(classtpx)

library(MAST)
library(DESeq2)
library(topGO)
library("org.Hs.eg.db")
library(clusterProfiler)

library(reshape2)

#library(pcaMethods)
#library(SC3)
#library(scater)
#library(SingleCellExperiment)
#library(pheatmap)
#library(mclust)
#set.seed(1234567)

#library(SingleR, lib.loc="") #only available on R/3.6.0??
#see cell atlas assignment below for more details

```

```{r load data}

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'
dir2 <- paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full")

#Load data
scrna <- "/data.filterMT.regMT.indv.log.10k.int.rds"            #integrate across individuals - total
#scrna <- "/data.filter.reg.indv-cell.log.10k.int.ipsc.rds"  #integrate across individuals - ipsc
#scrna <- "/data.filterMT.regMT.indv-cell.log.10k.int.msc.rds"   #integrate across individuals - msc
#scrna <- "/data.filterMT.regMT.indv-cell.log.10k.int.osteo.rds" #integrate across individuals - osteo
indv <- readRDS(paste0(dir2,scrna))

indv
head(indv@meta.data)

```

```{r seurat clustering}

pva <- indv@reductions$pca@stdev^2/indv@reductions$pca@misc$total.variance
length(which(pva>=0.001))

#Identify Clusters (increased resolution identifies greater number of clusters)
indv <- FindNeighbors(indv, k.param=20, dims=1:17)
indv <- FindClusters(indv, resolution=0.05)

#                                 ipsc /   msc / osteo / total
#indv1k, k=20, res=0.4 clusters:    13 /     8 /     8 /    15
#indv1k, k=20, res=0.3 clusters:       /       /     5 / 
#indv1k, k=20, res=0.2 clusters:       /       /     6 / 
#indv1k, k=20, res=0.1 clusters:       /       /     4 / 
#indv6k, k=20, res=0.1 clusters:       /       /     8 / 
#indv6k, k=40, res=0.1 clusters:       /       /     7 / 
#indv10k, k=20, res=0.1 clusters:      /       /    10 /    14
#indv10k, k=40, res=0.1 clusters:      /       /       /    12
#indv10k, k=40, res=0.05 clusters:     /       /       /     7

#CombinePlots(plots=list((DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(1,2)) + labs(x="PC1",y="PC2")),
#                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(3,4)) + labs(x="PC3",y="PC4")),
#                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(5,6)) + labs(x="PC5",y="PC6")),
#                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(7,8)) + labs(x="PC7",y="PC8")),
#                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(9,10)) + labs(x="PC9",y="PC10")),
#                        (DimPlot(subset(indv,subset=species=="Human"), group.by="seurat_clusters", reduction="pca", dims=c(1,2)) + labs(x="PC1",y="PC2")),
#                        (DimPlot(subset(indv,subset=species=="Human"), group.by="seurat_clusters", reduction="pca", dims=c(3,4)) + labs(x="PC3",y="PC4")),
#                        (DimPlot(subset(indv,subset=species=="Human"), group.by="seurat_clusters", reduction="pca", dims=c(5,6)) + labs(x="PC5",y="PC6")),
#                        (DimPlot(subset(indv,subset=species=="Human"), group.by="seurat_clusters", reduction="pca", dims=c(7,8)) + labs(x="PC7",y="PC8")),
#                        (DimPlot(subset(indv,subset=species=="Human"), group.by="seurat_clusters", reduction="pca", dims=c(9,10)) + labs(x="PC9",y="PC10")),
#                        (DimPlot(subset(indv,subset=species=="Chimp"), group.by="seurat_clusters", reduction="pca", dims=c(1,2)) + labs(x="PC1",y="PC2")),
#                        (DimPlot(subset(indv,subset=species=="Chimp"), group.by="seurat_clusters", reduction="pca", dims=c(3,4)) + labs(x="PC3",y="PC4")),
#                        (DimPlot(subset(indv,subset=species=="Chimp"), group.by="seurat_clusters", reduction="pca", dims=c(5,6)) + labs(x="PC5",y="PC6")),
#                        (DimPlot(subset(indv,subset=species=="Chimp"), group.by="seurat_clusters", reduction="pca", dims=c(7,8)) + labs(x="PC7",y="PC8")),
#                        (DimPlot(subset(indv,subset=species=="Chimp"), group.by="seurat_clusters", reduction="pca", dims=c(9,10)) + labs(x="PC9",y="PC10"))),
#             ncol=5)
CombinePlots(plots=list((DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (DimPlot(indv, group.by="seurat_clusters", reduction="pca", dims=c(9,10)) + labs(x="PC9",y="PC10"))), ncol=5)
DimPlot(indv, group.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"), split.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"), split.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Cell.Type=="iPSC"), split.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="MSC"), split.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="Osteoblast"), split.by="seurat_clusters", reduction="umap") + labs(x="UMAP1",y="UMAP2"))),
             ncol=1)

#Find markers for every cluster compared to all remaining cells, report only the positive ones
markers <- FindAllMarkers(indv, only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25)
markers %>% group_by(cluster) %>% top_n(n=5, wt=avg_logFC)
top5 <- markers %>% group_by(cluster) %>% top_n(n=5, wt=avg_logFC)
DoHeatmap(indv, features=top5$gene) + NoLegend()

#Find markers for groups of clusters to all remaining cells
markers.i <- FindMarkers(indv, ident.1=c(1,4,6), ident.2=c(0,2,3,5), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #ipsc
markers.m <- FindMarkers(indv, ident.1=c(0), ident.2=c(1,2,3,4,5,6), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #msc
markers.o <- FindMarkers(indv, ident.1=c(2,3,5), ident.2=c(0,1,4,6), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #osteo
head(markers, n=5)

VlnPlot(indv, features=c("MMP1","CD24","CTSD","MTRNR2L12","POU5F1","COL11A1","CNTNAP2"), slot="data", pt.size=0, ncol=7)
VlnPlot(indv, features=c("CD24","POU5F1","APOE","TUBB2B","L1TD1"), group.by="Cell.Type", slot="data", pt.size=0, ncol=5) #ipsc
VlnPlot(indv, features=c("MMP1","LGALS1","TAGLN2","RGS4","TPM1"), group.by="Cell.Type", slot="data", pt.size=0, ncol=5) #msc
VlnPlot(indv, features=c("NUPR1","TIMP1","IGFBP4","CTSD","GAS6"), group.by="Cell.Type", slot="data", pt.size=0, ncol=5) #osteo
VlnPlot(indv, features=c('nCount_RNA','nFeature_RNA','percent.mt'), pt.size=0)


indv <- AddMetaData(indv, rep(NA,length(rownames(indv@meta.data))), col.name="Cluster")
indv$Cluster[which(indv$seurat_clusters=="1")] <- "iPSC.c1"
indv$Cluster[which(indv$seurat_clusters=="4")] <- "iPSC.c2"
indv$Cluster[which(indv$seurat_clusters=="6")] <- "iPSC.c3"
indv$Cluster[which(indv$seurat_clusters=="0")] <- "MSC.c1"
indv$Cluster[which(indv$seurat_clusters=="2")] <- "Osteoblast.c1"
indv$Cluster[which(indv$seurat_clusters=="3")] <- "Osteoblast.c2"
indv$Cluster[which(indv$seurat_clusters=="5")] <- "Osteoblast.c3"

DimPlot(indv, group.by="Cluster", reduction="umap", cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
  labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(indv,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(1,2),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(indv,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(3,4),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(indv,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(5,6),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(indv,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(7,8),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(indv,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(9,10),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="PC9",y="PC10"))), ncol=5)
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=2)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Cell.Type=="iPSC"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="MSC"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="Osteoblast"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=3)
table(subset(indv,subset=species=="Human")@meta.data$Cluster)
table(subset(indv,subset=species=="Chimp")@meta.data$Cluster)
table(subset(indv,subset=Cell.Type=="iPSC")@meta.data$Cluster)
table(subset(indv,subset=Cell.Type=="MSC")@meta.data$Cluster)
table(subset(indv,subset=Cell.Type=="Osteoblast")@meta.data$Cluster)
table(subset(indv,subset=(species=="Human" & Cell.Type=="iPSC"))@meta.data$Cluster)
table(subset(indv,subset=(species=="Human" & Cell.Type=="MSC"))@meta.data$Cluster)
table(subset(indv,subset=(species=="Human" & Cell.Type=="Osteoblast"))@meta.data$Cluster)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="iPSC"))@meta.data$Cluster)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="MSC"))@meta.data$Cluster)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="Osteoblast"))@meta.data$Cluster)

```

```{r SC3 clustering: NOT ENOUGH MEMORY - NEED TO FIX}

#(do this poriton on cluster: sc3.sh + sc3.R)
#Create a SingleCellExperiment object
sce <- SingleCellExperiment(assays = list(logcounts = as.matrix(indv@assays$integrated@data[,which(indv@meta.data$Individual=="H6")])), 
                            colData = indv@meta.data[which(indv@meta.data$Individual=="H6"),])
#Define feature names in feature_symbol column
rowData(sce)$feature_symbol <- rownames(sce)
#Remove features with duplicated names
sce <- sce[!duplicated(rowData(sce)$feature_symbol), ]
#plotPCA(sce, colour_by="Cell.Type")
#Run SC3
sce <- sc3(sce, ks=3, gene_filter=FALSE, biology=TRUE, n_cores=1)
#Save to a file
saveRDS(sce, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/sc3.indv.log.10k.int.rds"))

#Read SC3 file
sce <- readRDS(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/sc3.indv.log.10k.int.rds"))

```

```{r cell atlas assignment}

#sinteractive --mem=24g --time=12:00:00 --partition=broadwl
#module load R/3.5.1
#R

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

###SCRIPTS GOTTEN FROM BINGQING XIE - AND EDITED BY GAH###

###compute correlation matrix between sample and cell lines in cellAtlas
###object: Seurat object
###cell.line: cell Atlas expression matrix
###CorMethod: correlation method to use
ComputeCorMat <- function(object, refs, CorMethod="pearson"){
	data <- GetAssayData(object, slot="data", assay="RNA") #additions - added slot and assay info because SCT assay can cause issues
  #data <- apply(data, 2, function(x) x/sum(x)) #normalize
  #data <- apply(data, 2, function(x) x * 1e+06) #library size
  gene.id <- rownames(data)
  gene.ref <- rownames(refs)
  common <- intersect(gene.id, gene.ref)
  #data <- apply(data[common, ], 2, function(x) { log(x + 0.1) }) #log-scale
  #refs <- apply(refs[common, ], 2, function(x) { x - mean(x) }) #adjust
  data <- data[common, ]
  refs <- refs[common, ]
  n <- ncol(data)
  m <- ncol(refs)
  cor.mat <- matrix(0, nrow=n, ncol=m)
  for (j in 1:m) {
    for (i in 1:n) {
      cor.mat[i,j] <- cor(data[,i], refs[,j], method=CorMethod)
      }
    }
  rownames(cor.mat) <- colnames(data)
  colnames(cor.mat) <- colnames(refs)
	return(cor.mat)
}

###Assign clustered cell atlas as cell type to all cells in the sample
###object: seurat object
###cor.mat: correlation matrix
###K: number of clusters generated from cell atlas
###topCorrelation: number of highest correlated cell lines in cell atlas to be assigned, set to 1 to extract only the top correlated cell line
AssignClusteredLabel <- function (object, cor.mat, K, topCorrelation, dist.method="euclidean", hclust.method="ward.D"){
  dist_mat <- dist(t(cor.mat), method=dist.method)
	hclust_avg_cellline <- hclust(dist_mat, method=hclust.method)
	#plot(hclust_avg_cellline)
	cut_cellline <- cutree(hclust_avg_cellline, k = K)
	cellline_abstract=sapply(strsplit(as.character(colnames(cor.mat)), "\\_"), "[[", 2)
	clusteredCellline=rbind(cut_cellline,cellline_abstract)
	nametable=rep('n',K)
	for(i in 1:K){
		indexy=as.character(i)
		nametable[i]=paste0(names(sort(table(clusteredCellline[2,clusteredCellline[1,]==indexy]),decreasing=TRUE)),collapse='_' )
	}
	assignm=apply(cor.mat,1,function(x){
		indexy=names(which.max(table(clusteredCellline[1,names(sort(x,decreasing=T)[1:topCorrelation])])))
		y=paste0(names( sort(table(clusteredCellline[2,clusteredCellline[1,]==indexy]),decreasing=TRUE)),collapse='_' )
		return(y)
	})
	object$clusteredCellAtlas=assignm
	return(object)
}

#Load cell atlas data matrix (curated in Mengjie's lab)
#references: Mabbott et al. BMC Genomics 2013; Wu C et al. Nucleic acids research 2016
load("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/cell_atlas_ref_panel")

#Load cell atlas data matrix (curated by GAH)
#references: Human Primary Cell Atlas (Mabbott et al., 2013)
#description: 713 microarray samples processed and normalized as described in Aran et al. (2019); each sample assigned to one of 37 main cell types and 157 subtypes
#library(SingleR, lib.loc="") #only available on R/3.6.0??? ...BUT able to install on local computer
#if(!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
#BiocManager::install("SingleR")
#library(SingleR)
#hpca.se <- HumanPrimaryCellAtlasData()
#write.table(hpca.se@assays@data$logcounts, file="G:/My Drive/Documents/University of Chicago/Gilad Lab/project notes/study-human-chimp-bone-cartilage-rna/study-collections/sc-rna/HumanPrimaryCellAtlasData-SingleR")
#write.csv(as.data.frame(hpca.se@colData),file="G:/My Drive/Documents/University of Chicago/Gilad Lab/project notes/study-human-chimp-bone-cartilage-rna/study-collections/sc-rna/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=TRUE)
#manually curated HumanPrimaryCellAtlasLabels-SingleR.csv
#transferred files to cluster
hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasData-SingleR")
hpca.names <- read.csv("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=1)
hpca.names$label <- paste0(hpca.names$geo.curated,"_",hpca.names$label.curated)
i=1
while(i <= length(colnames(hpca))) {
  colnames(hpca)[i] <- hpca.names$label[rownames(hpca.names)==colnames(hpca)[i]]
  i=i+1
}

#Subset cell atlas data matrix to only include certain cells
hpca.sub <- hpca[,c(239:261,416:419,470:474,496,499:515,548:551,592:599,608:616,651:665)] #iPSCs, MSCs, Osteoblasts
#hpca.sub <- hpca[,unique(c(grep("iPSC", colnames(hpca)),grep("MSC", colnames(hpca)),grep("Osteoblast", colnames(hpca)),
#                           grep("Chondrocyte", colnames(hpca)),grep("Adipocyte", colnames(hpca))))]
#hpca.sub <- hpca[,c(239:261,416:419,470:474,496,499:515,548:551,592:599,608:616,651:665,
#                    grep("Endothelial.cells", colnames(hpca)),grep("Smooth.muscle.cells", colnames(hpca)),
#                    grep("Epithelial.cells", colnames(hpca)),grep("Neurons", colnames(hpca)),
#                    grep("Fibroblasts", colnames(hpca)),grep("B.cell", colnames(hpca)),grep("T.cells", colnames(hpca)),
#                    grep("Monocyte", colnames(hpca)),grep("Macrophage", colnames(hpca)),grep("Neutrophils", colnames(hpca)))]



#Compute the correlation matrix use a seurat object and cell line matrix as input
cor.mat=ComputeCorMat(subset(indv,subset=Sample=="H6-O"), refs=cell.line)
cor.mat.hpca=ComputeCorMat(indv, refs=hpca.sub)
#cor.mat.hpca=ComputeCorMat(indv, refs=hpca.sub) #run these on cluster
#cor.mat.hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/corrmatrix.tot")
#cor.mat.hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/corrmatrix.ips")
#cor.mat.hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/corrmatrix.msc")
#cor.mat.hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/corrmatrix.ost")

#Save data
save(cor.mat.hpca, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/corrmatrix3.rds"))

#Read GoM data
cor.mat.hpca <- readRDS(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/corrmatrix3.rds"))

#Add a metadata column "clusteredCellAtlas" to the object
indv <- AssignClusteredLabel(indv, cor.mat, K=50, topCorrelation=10)
indv <- AssignClusteredLabel(indv, cor.mat.hpca, K=50, topCorrelation=5)

#Visualize assigned clusters
DimPlot(subset(indv,subset=Sample=="H6-O"), reduction="umap", group.by="clusteredCellAtlas", pt.size=0.5)
as.data.frame(table(indv$clusteredCellAtlas))

DimPlot(indv, reduction="umap", group.by="species", split.by="clusteredCellAtlas", pt.size=0.5) +
  theme(plot.title = element_text(size=1)) +
  labs(x="UMAP1",y="UMAP2")




#Check out average correlations with each cell line
#cor.mat.orth[grep("iPSC", rownames(cor.mat.orth)),]
#cor.mat.orth[grep("MSC", rownames(cor.mat.orth)),]
#cor.mat.orth[grep("osteo", rownames(cor.mat.orth)),]
#cor.mat.orth[grep("chondro", rownames(cor.mat.orth)),]
z <- c("hotpink","hotpink","hotpink","hotpink","hotpink","hotpink","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","pink","pink","pink","pink","mediumorchid","mediumorchid","mediumorchid","mediumorchid","mediumorchid","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","gold","gold","dodgerblue","dodgerblue","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","mediumorchid","mediumorchid","mediumorchid","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey","lightgrey")


x <- as.data.frame(colSums(cor.mat.orth[grep("iPSC", rownames(cor.mat.orth)),])/length(grep("iPSC", rownames(cor.mat.orth))))
colnames(x) <- "avg"
x$cell <- rownames(x)
x$label <- z
x <- x[order(x$avg, decreasing=TRUE),]
x$cell <- factor(x$cell, levels=x$cell)
ggplot(x, aes(x=cell, y=avg)) + geom_bar(stat="identity", fill=x$label) + labs(title="iPSC-orth-avg-cor") + ylim(0,0.5) +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5))

x <- as.data.frame(colSums(cor.mat.orth[grep("MSC", rownames(cor.mat.orth)),])/length(grep("MSC", rownames(cor.mat.orth))))
colnames(x) <- "avg"
x$cell <- rownames(x)
x$label <- z
x <- x[order(x$avg, decreasing=TRUE),]
x$cell <- factor(x$cell, levels=x$cell)
ggplot(x, aes(x=cell, y=avg)) + geom_bar(stat="identity", fill=x$label) + labs(title="MSC-orth-avg-cor") + ylim(0,0.5) +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5))

x <- as.data.frame(colSums(cor.mat.orth[grep("osteo", rownames(cor.mat.orth)),])/length(grep("osteo", rownames(cor.mat.orth))))
colnames(x) <- "avg"
x$cell <- rownames(x)
x$label <- z
x <- x[order(x$avg, decreasing=TRUE),]
x$cell <- factor(x$cell, levels=x$cell)
ggplot(x, aes(x=cell, y=avg)) + geom_bar(stat="identity", fill=x$label) + labs(title="osteo-orth-avg-cor") + ylim(0,0.5) +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5))

x <- as.data.frame(colSums(cor.mat.orth[grep("chondro", rownames(cor.mat.orth)),])/length(grep("chondro", rownames(cor.mat.orth))))
colnames(x) <- "avg"
x$cell <- rownames(x)
x$label <- z
x <- x[order(x$avg, decreasing=TRUE),]
x$cell <- factor(x$cell, levels=x$cell)
ggplot(x, aes(x=cell, y=avg)) + geom_bar(stat="identity", fill=x$label[1:94]) + labs(title="chondro-orth-avg-cor") + ylim(0,0.5) +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5))

```

```{r grades of membership - no reference: STILL NEED TO RUN ON CLUSTER}

#(do this poriton on cluster: gom.sh + gom.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load data
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.10k.int.rds")) #integrate across individuals - total

#Fit GoM model (maybe change tol=0.1 at some point)
Combined_GoM <- CountClust::FitGoM(t(as.matrix(indv@assays$integrated@data)), K=c(3:7), tol=1, control=list(tmax=100))

#Save data
save(Combined_GoM, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom3to7.indv.log.10k.int.rds"))

#Read GoM data
Combined_GoM <- readRDS(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom3to7.indv.log.10k.int.rds"))

#GoM Plotting
plot_structure_plot <- function(omega, labels){
     annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=labels)
     rownames(omega) <- annotation$sample_id
     StructureGGplot(omega=omega, annotation=annotation, order_sample=TRUE,
                     axis_tick=list(axis_ticks_length = .1,
                                    axis_ticks_lwd_y = .1,
                                    axis_ticks_lwd_x = .1,
                                    axis_label_size = 7,
                                    axis_label_face = "bold"))
     }
make_topic_plots <- function(omega, labels) {
  omega <- as.data.frame(omega)
  omega <- cbind(omega,labels)
  for (k in 1:(ncol(omega)-1)) {
    base_plot <- ggplot(data=omega, mapping=aes(x=labels, y=omega[,k])) +
      labs(y=paste0("Topic ", k, " value"), x=paste0("Cell Type")) +
      theme(axis.text.x=element_text(angle=-20, hjust=0, vjust=0))
    plot(base_plot + geom_violin(width=1.5, fill="skyblue",color="skyblue") + geom_boxplot(width=0.1,outlier.shape=NA, alpha=0.2))
   }
}
annotate_clusters <- function(theta_mat, countmatrix){
  top_features <- ExtractTopFeatures(theta_mat, top_features=100, method="poisson", options="min")
  gene_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(countmatrix)[top_features[[1]][x,]]))
  scores_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}

#k=3
plot_structure_plot(omega=Combined_GoM$clust_4$omega, labels=labels)
make_topic_plots(omega=Combined_GoM$clust_4$omega, labels=labels)
#genes <- annotate_clusters(theta_mat=Combined_GoM$clust_4$theta, countmatrix=merged_raw_data_counts)
#matrix(gene_anno$external_gene_name[match(genes[[1]], gene_anno$ensembl_gene_id)], ncol=100)
#genes[[2]]


```

```{r grades of membership - reference: STILL NEED TO RUN ON CLUSTER}

#(do this poriton on cluster: gom-ref.sh + gom-ref.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load data
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.10k.int.rds")) #integrate across individuals - total

#Load cell atlas data matrix (currated by GAH)
hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasData-SingleR")
hpca.names <- read.csv("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=1)
hpca.names$label <- paste0(hpca.names$label.curated,"_",hpca.names$geo.curated)
i=1
while(i <= length(colnames(hpca))) {
  colnames(hpca)[i] <- hpca.names$label[rownames(hpca.names)==colnames(hpca)[i]]
  i=i+1
}

#Subset cell atlas data matrix to only iPSCs, MSCs, and Osteoblasts
hpca.sub <- hpca[,c(239:261,416:419,470:474,496,499:515,548:551,592:599,608:616,651:665)]

#Merge with cell atlas data (integrated data and renormalizing data like ComputeCorMat() does not work)
gene.id <- rownames(indv@assays$RNA@counts)
gene.ref <- rownames(hpca.sub)
common <- intersect(gene.id, gene.ref)
merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,], hpca.sub[common,], by="row.names")
#merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,which(indv@meta.data$Sample=="H1-I")], hpca.sub[common,], by="row.names")
rownames(merged.data) <- merged.data$Row.names
merged.data <- merged.data[,c(2:length(colnames(merged.data)))]
labels <- sapply(strsplit(colnames(merged.data),"_"), `[`, 1)

#Fit GoM model (maybe change tol=0.1 at some point)
Combined_GoM <- CountClust::FitGoM(t(as.matrix(merged.data)), K=c(3:7), tol=1, control=list(tmax=100))

#Save data
save(Combined_GoM, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.ref.3to7.indv.log.10k.int.rds"))

#Read GoM data
Combined_GoM <- readRDS(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.ref.3to7.indv.log.10k.int.rds"))

#GoM Plotting
plot_structure_plot <- function(omega, labels){
     annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=labels)
     rownames(omega) <- annotation$sample_id
     StructureGGplot(omega=omega, annotation=annotation, order_sample=TRUE,
                     axis_tick=list(axis_ticks_length = .1,
                                    axis_ticks_lwd_y = .1,
                                    axis_ticks_lwd_x = .1,
                                    axis_label_size = 7,
                                    axis_label_face = "bold"))
     }
make_topic_plots <- function(omega, labels) {
  omega <- as.data.frame(omega)
  omega <- cbind(omega,labels)
  for (k in 1:(ncol(omega)-1)) {
    base_plot <- ggplot(data=omega, mapping=aes(x=labels, y=omega[,k])) +
      labs(y=paste0("Topic ", k, " value"), x=paste0("Cell Type")) +
      theme(axis.text.x=element_text(angle=-20, hjust=0, vjust=0))
    plot(base_plot + geom_violin(width=1.5, fill="skyblue",color="skyblue") + geom_boxplot(width=0.1,outlier.shape=NA, alpha=0.2))
   }
}
annotate_clusters <- function(theta_mat, countmatrix){
  top_features <- ExtractTopFeatures(theta_mat, top_features=100, method="poisson", options="min")
  gene_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(countmatrix)[top_features[[1]][x,]]))
  scores_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}

#k=3
plot_structure_plot(omega=Combined_GoM$clust_4$omega, labels=labels)
make_topic_plots(omega=Combined_GoM$clust_4$omega, labels=labels)
#genes <- annotate_clusters(theta_mat=Combined_GoM$clust_4$theta, countmatrix=merged_raw_data_counts)
#matrix(gene_anno$external_gene_name[match(genes[[1]], gene_anno$ensembl_gene_id)], ncol=100)
#genes[[2]]

```

```{r grades of membership - semisupervised: 3 known samples}

#(do this poriton on cluster: gom-sup.sh + gom-sup.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load data
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.10k.int.rds")) #integrate across individuals - total

#Load cell atlas data matrix (currated by GAH)
hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasData-SingleR")
hpca.names <- read.csv("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=1)
hpca.names$label <- paste0(hpca.names$label.curated,"_",hpca.names$geo.curated)
i=1
while(i <= length(colnames(hpca))) {
  colnames(hpca)[i] <- hpca.names$label[rownames(hpca.names)==colnames(hpca)[i]]
  i=i+1
}

#Subset cell atlas data matrix to only iPSCs, MSCs, and Osteoblasts
hpca.sub <- hpca[,c(239:261,416:419,470:474,496,499:515,548:551,592:599,608:616,651:665)]

#Merge with cell atlas data (integrated data and renormalizing data like ComputeCorMat() does not work)
gene.id <- rownames(indv@assays$RNA@counts)
gene.ref <- rownames(hpca.sub)
common <- intersect(gene.id, gene.ref)
merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,], hpca.sub[common,], by="row.names")
#merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,which(indv@meta.data$Sample=="H1-I")], hpca.sub[common,], by="row.names")
rownames(merged.data) <- merged.data$Row.names
merged.data <- merged.data[,c(2:length(colnames(merged.data)))]
labels <- sapply(strsplit(colnames(merged.data),"_"), `[`, 1)

#Semi-supervised topic modeling (define some known clusters prior to fitting topic model)

#labels for "known" clusters
idx.ips <- which(labels=="iPSC")
idx.msc <- which(labels=="MSC")
idx.ost <- which(labels=="Osteoblasts" | labels=="MSC-Osteoblast.D1" | labels=="MSC-Osteoblast.D3" | labels=="MSC-Osteoblast.D7" | labels=="MSC-Osteoblast")
kwn.smp <- c(idx.ips,idx.msc,idx.ost)

class.labs <- c(rep("iPSC", length(idx.ips)),
                rep("MSC", length(idx.msc)),
                rep("Osteoblast", length(idx.ost)))

#Perform topic modeling k=3:4 (maybe change tol=0.1 at some point)
#Save GoM data
for(k in c(3:4)) {
  print(k)
  tpx.clust <- classtpx::class_topics(
    t(merged.data),
    K=k,
    known_samples=kwn.smp,
    class_labs=class.labs,
    method="omega.fix",
    tol=1)
  save(tpx.clust, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.",k,".indv.log.10k.int.rds"))
  print(paste0("Finished k = ",k))
}

#Read GoM file
#tpx.clust <- get(load(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.3.indv.log.10k.int.rds")))
tpx.clust <- get(load(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.3.indv.filterMT.regMT.log.10k.int.rds")))

make_topic_plots <- function(omega, labels) {
  omega <- as.data.frame(omega)
  omega <- cbind(omega,labels)
  col <- c("pink","mediumorchid","dodgerblue")
  for (k in 1:(ncol(omega)-1)) {
    p <- ggplot(data=omega, mapping=aes(x=labels, y=omega[,k])) +
            labs(y=paste0("Topic ",k," Value"), x=paste0("Cell Type")) +
            theme(axis.text.x=element_text(angle=-90, hjust=0, vjust=0))
    plot(p + geom_boxplot(width=0.5, outlier.shape=NA, outlier.colour=NA, fill=col[k]))
  }
}
annotate_clusters <- function(theta_mat, top_features, countmatrix){
  top_features <- ExtractTopFeatures(theta_mat, top_features=top_features, method="poisson", options="min", shared=FALSE)
  gene_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(countmatrix)[top_features[[1]][x,]]))
  scores_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}

omega <- tpx.clust$omega
labels <- sapply(strsplit(rownames(omega),"_"), `[`, 1)
#labels[labels %in% c("C1.I","C1r2.I","C2.I","C3.I","C4.I","C5.I","C6.I")] <- "Chimp.iPSC"
#labels[labels %in% c("C1.M","C1r2.M","C2.M","C3.M","C4.M","C5.M","C6.M")] <- "Chimp.MSC"
#labels[labels %in% c("C1.O","C1r2.O","C2.O","C3.O","C4.O","C5.O","C6.O")] <- "Chimp.Osteoblast"
#labels[labels %in% c("H1.I","H1r2.I","H2.I","H3.I","H4.I","H5.I","H6.I")] <- "Human.iPSC"
#labels[labels %in% c("H1.M","H1r2.M","H2.M","H3.M","H4.M","H5.M","H6.M")] <- "Human.MSC"
#labels[labels %in% c("H1.O","H1r2.O","H2.O","H3.O","H4.O","H5.O","H6.O")] <- "Human.Osteoblast"
annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(labels))
rownames(omega) <- annotation$sample_id

x <- length(labels)-85
y <- length(labels)

CountClust::StructureGGplot(omega=omega[x:y,],
                annotation=annotation[x:y,],
                palette=c("pink","mediumorchid","dodgerblue"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))
CountClust::StructureGGplot(omega=omega[1:(x-1),],
                annotation=annotation[1:(x-1),],
                palette=c("pink","mediumorchid","dodgerblue"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))
CountClust::StructureGGplot(omega=omega,
                annotation=annotation,
                palette=c("pink","mediumorchid","dodgerblue"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

make_topic_plots(omega=omega, labels=labels)

genes <- annotate_clusters(theta_mat=tpx.clust$theta, top_features=10, countmatrix=merged.data)
genes[[1]]



#assign cell type labels based on grade of membership cutoffs
gom.assign <- as.data.frame(omega[1:(x-1),])
gom.assign[,1] <- ifelse(gom.assign[,1] > 0.75, 1, 0)
gom.assign[,2] <- ifelse(gom.assign[,2] > 0.75, 1, 0)
gom.assign[,3] <- ifelse(gom.assign[,3] > 0.75, 1, 0)
upset(gom.assign, nintersects=NA, empty.intersections="on")

gom.assign$assign <- "other"
i=1
while(i <= length(rownames(gom.assign))) {
  if(gom.assign[i,1] == 1) {gom.assign$assign[i] <- "iPSC"}
  if(gom.assign[i,2] == 1) {gom.assign$assign[i] <- "MSC"}
  if(gom.assign[i,3] == 1) {gom.assign$assign[i] <- "Osteoblast"}
  i=i+1
}

indv@meta.data$GoM.Assign <- gom.assign$assign


```

```{r grades of membership - semisupervised: 5 known samples: PENDING ON CLUSTER}

#(do this poriton on cluster: gom-sup5.sh + gom-sup5.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load data
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.10k.int.rds")) #integrate across individuals - total

#Load cell atlas data matrix (currated by GAH)
hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasData-SingleR")
hpca.names <- read.csv("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=1)
hpca.names$label <- paste0(hpca.names$label.curated,"_",hpca.names$geo.curated)
i=1
while(i <= length(colnames(hpca))) {
  colnames(hpca)[i] <- hpca.names$label[rownames(hpca.names)==colnames(hpca)[i]]
  i=i+1
}

#Subset cell atlas data matrix only iPSCs, MSCs, Osteoblasts, Chondrocytes, and Adipocytes
hpca.sub <- hpca[,unique(c(grep("iPSC", colnames(hpca)),grep("MSC", colnames(hpca)),grep("Osteoblast", colnames(hpca)),
                           grep("Chondrocyte", colnames(hpca)),grep("Adipocyte", colnames(hpca))))]

#Merge with cell atlas data (integrated data and renormalizing data like ComputeCorMat() does not work)
gene.id <- rownames(indv@assays$RNA@counts)
gene.ref <- rownames(hpca.sub)
common <- intersect(gene.id, gene.ref)
merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,], hpca.sub[common,], by="row.names")
#merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,which(indv@meta.data$Sample=="H1-I")], hpca.sub[common,], by="row.names")
rownames(merged.data) <- merged.data$Row.names
merged.data <- merged.data[,c(2:length(colnames(merged.data)))]
labels <- sapply(strsplit(colnames(merged.data),"_"), `[`, 1)

#Semi-supervised topic modeling (define some known clusters prior to fitting topic model)

#labels for "known" clusters
idx.ips <- which(labels=="iPSC")
idx.msc <- which(labels=="MSC")
idx.ost <- which(labels=="Osteoblast" | labels=="MSC-Osteoblast.D1" | labels=="MSC-Osteoblast.D3" | labels=="MSC-Osteoblast.D7" | labels=="MSC-Osteoblast")
idx.cho <- which(labels=="Chondrocytes" | labels=="MSC-Chondrocyte")
idx.adi <- which(labels=="MSC-Adipocyte")
kwn.smp <- c(idx.ips,idx.msc,idx.ost,idx.cho,idx.adi)

class.labs <- c(rep("iPSC", length(idx.ips)),
                rep("MSC", length(idx.msc)),
                rep("Osteoblast", length(idx.ost)),
                rep("Chondrocyte", length(idx.cho)),
                rep("Adipocyte", length(idx.adi)))

#Perform topic modeling k=5:6 (maybe change tol=0.1 at some point)
#Save GoM data
for(k in c(5:6)) {
  print(k)
  tpx.clust <- classtpx::class_topics(
    t(merged.data),
    K=k,
    known_samples=kwn.smp,
    class_labs=class.labs,
    method="omega.fix",
    tol=1)
  save(tpx.clust, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.",k,".indv.log.10k.int.rds"))
  print(paste0("Finished k = ",k))
}

#Read GoM file
tpx.clust <- get(load(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.5.indv.log.10k.int.rds")))

make_topic_plots <- function(omega, labels) {
  omega <- as.data.frame(omega)
  omega <- cbind(omega,labels)
  col <- c("pink","mediumorchid","dodgerblue")
  for (k in 1:(ncol(omega)-1)) {
    p <- ggplot(data=omega, mapping=aes(x=labels, y=omega[,k])) +
            labs(y=paste0("Topic ",k," Value"), x=paste0("Cell Type")) +
            theme(axis.text.x=element_text(angle=-90, hjust=0, vjust=0))
    plot(p + geom_boxplot(width=0.5, outlier.shape=NA, outlier.colour=NA, fill=col[k]))
  }
}
annotate_clusters <- function(theta_mat, top_features, countmatrix){
  top_features <- ExtractTopFeatures(theta_mat, top_features=top_features, method="poisson", options="min", shared=FALSE)
  gene_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(countmatrix)[top_features[[1]][x,]]))
  scores_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}

omega <- tpx.clust$omega
labels <- sapply(strsplit(rownames(omega),"_"), `[`, 1)
annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(labels))
rownames(omega) <- annotation$sample_id

x <- length(labels)-119
y <- length(labels)

CountClust::StructureGGplot(omega=omega[x:y,],
                annotation=annotation[x:y,],
                palette=RColorBrewer::brewer.pal(8, "Accent"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

CountClust::StructureGGplot(omega=omega[c(1:(x-1)),],
                annotation=annotation[c(1:(x-1)),],
                palette=RColorBrewer::brewer.pal(8, "Accent"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

```

```{r grades of membership - semisupervised: 13 known samples: PENDING ON CLUSTER}

#(do this poriton on cluster: gom-sup13.sh + gom-sup13.R)

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'

#Load data
indv <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.indv.log.10k.int.rds")) #integrate across individuals - total

#Load cell atlas data matrix (currated by GAH)
hpca <- read.table("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasData-SingleR")
hpca.names <- read.csv("/project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/data/HumanPrimaryCellAtlasLabels-SingleR.csv", row.names=1)
hpca.names$label <- paste0(hpca.names$label.curated,"_",hpca.names$geo.curated)
i=1
while(i <= length(colnames(hpca))) {
  colnames(hpca)[i] <- hpca.names$label[rownames(hpca.names)==colnames(hpca)[i]]
  i=i+1
}

#Subset cell atlas data matrix to several cell types
hpca.sub <- hpca[,c(239:261,416:419,470:474,496,499:515,548:551,592:599,608:616,651:665,
                    grep("Endothelial.cells", colnames(hpca)),grep("Smooth.muscle.cells", colnames(hpca)),
                    grep("Epithelial.cells", colnames(hpca)),grep("Neurons", colnames(hpca)),
                    grep("Fibroblasts", colnames(hpca)),grep("B.cell", colnames(hpca)),grep("T.cells", colnames(hpca)),
                    grep("Monocyte", colnames(hpca)),grep("Macrophage", colnames(hpca)),grep("Neutrophils", colnames(hpca)))]

#Merge with cell atlas data (integrated data and renormalizing data like ComputeCorMat() does not work)
gene.id <- rownames(indv@assays$RNA@counts)
gene.ref <- rownames(hpca.sub)
common <- intersect(gene.id, gene.ref)
merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,], hpca.sub[common,], by="row.names")
#merged.data <- merge(GetAssayData(indv, slot="data", assay="RNA")[common,which(indv@meta.data$Sample=="H1-I")], hpca.sub[common,], by="row.names")
rownames(merged.data) <- merged.data$Row.names
merged.data <- merged.data[,c(2:length(colnames(merged.data)))]
labels <- sapply(strsplit(colnames(merged.data),"_"), `[`, 1)

#Semi-supervised topic modeling (define some known clusters prior to fitting topic model)

#labels for "known" clusters
idx.ips <- which(labels=="iPSC")
idx.msc <- which(labels=="MSC")
idx.ost <- which(labels=="Osteoblast" | labels=="MSC-Osteoblast.D1" | labels=="MSC-Osteoblast.D3" | labels=="MSC-Osteoblast.D7" | labels=="MSC-Osteoblast")
idx.end <- which(labels=="Endothelial.cells")
idx.smo <- which(labels=="Smooth.muscle.cells")
idx.epi <- which(labels=="Epithelial.cells")
idx.nrn <- which(labels=="Neurons")
idx.fib <- which(labels=="Fibroblasts")
idx.bce <- which(labels=="B.cell")
idx.tce <- which(labels=="T.cells")
idx.mon <- which(labels=="Monocyte")
idx.mac <- which(labels=="Macrophage")
idx.neu <- which(labels=="Neutrophils")
kwn.smp <- c(idx.ips,idx.msc,idx.ost,idx.end,idx.smo,idx.epi,idx.nrn,idx.fib,idx.bce,idx.tce,idx.mon,idx.mac,idx.neu)

class.labs <- c(rep("iPSC", length(idx.ips)),
                rep("MSC", length(idx.msc)),
                rep("Osteoblast", length(idx.ost)),
                rep("Endothelial.cells", length(idx.end)),
                rep("Smooth.muscle.cells", length(idx.smo)),
                rep("Epithelial.cells", length(idx.epi)),
                rep("Neurons", length(idx.nrn)),
                rep("Fibroblasts", length(idx.fib)),
                rep("B.cell", length(idx.bce)),
                rep("T.cells", length(idx.tce)),
                rep("Monocyte", length(idx.mon)),
                rep("Macrophage", length(idx.mac)),
                rep("Neutrophils", length(idx.neu)))

#Perform topic modeling k=13:14 (maybe change tol=0.1 at some point)
#Save GoM data
for(k in c(13:14)) {
  print(k)
  tpx.clust <- classtpx::class_topics(
    t(merged.data),
    K=k,
    known_samples=kwn.smp,
    class_labs=class.labs,
    method="omega.fix",
    tol=1)
  save(tpx.clust, file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.",k,".indv.log.10k.int.rds"))
  print(paste0("Finished k = ",k))
}

#Read GoM file
tpx.clust <- get(load(file=paste0(dir,"human-chimp-skeletal-scRNA/data/cellranger-data-full/gom.sup.3.indv.log.10k.int.rds")))

omega <- tpx.clust$omega
labels <- sapply(strsplit(rownames(omega),"_"), `[`, 1)
annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(labels))
rownames(omega) <- annotation$sample_id
CountClust::StructureGGplot(omega=omega,
                annotation=annotation,
                palette=RColorBrewer::brewer.pal(8, "Accent"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

CountClust::StructureGGplot(omega=omega[c(1:50,2969:3054),],
                annotation=annotation[c(1:50,2969:3054),],
                palette=RColorBrewer::brewer.pal(8, "Accent"),
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

```

```{r ad hoc cell classification}

#Look at candidate genes

#ipsc
#diagnostic: POU5F1 (OCT4), TDGF1, SOX2, EPCAM
#additional: NANOG, ALPL, STAT3, PROM1 (CD133), PODXL (TRA-1-60R), LIN28A, ZFP42, UTF1, MYC, BMP4, FGF2, TGFB1, DNMT3B, SALL4
#if needed: FOXD3, FUT4 (SSEA-1)*, KLF2, KLF4, KLF5, INHBA (Activin A), LIF, PRDM14
#[not integrated in msc data: NANOG, POU5F1, UTF1, FUT4, PRDM14]
#[not integrated in osteo data: NANOG, TDGF1, LIN28A, UTF1, TGFB1, FUT4, PRDM14]
#[not integrated in I/M/O data: NANOG, STAT3, FUT4]
genes <- c("POU5F1","TDGF1","SOX2","EPCAM","NANOG","ALPL","STAT3","PROM1","PODXL","LIN28A","ZFP42","UTF1","MYC",
          "BMP4","FGF2","TGFB1","DNMT3B","SALL4","FOXD3","FUT4","KLF2","KLF4","KLF5","INHBA","LIF","PRDM14")

FeaturePlot(indv, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=4)

#msc
#diagnostic: THY1 (CD90), NT5E (CD73), ENG (CD105), CD44
#additional: ANPEP (CD13), NGFR (CD271), ITGB1 (CD29), NCAM1 (CD56)
#if needed: CD9, CD14, MME (CD10), ITGAL (CD11A)*, ITGAX (CD11C)*, CD36, ITGA3 (CD49C), ITGA4 (CD49D), ITGAV (CD51), ICAM1 (CD54), CD55, CD59, TFRC (CD71), VCAM1 (CD106), KIT (CD117), TNFRSF1A (CD120a), PDGFRB (CD140B), CDH5 (CD144)*, MCAM (CD146), ALCAM (CD166), ERBB2 (CD340)*
#negative controls: ITGAM (CD11B)*, CD14, PECAM1 (CD31), CD34*, PTPRC (CD45), CD79A, HLA-DRA (HLA-DR)*
# * not integrated (msc)
#[not integrated in ipsc data: ITGAL, ITGAX, CD34, ITGA4, CD55, VCAM1, CDH5, PTPRC, CD79A]
#[not integrated in osteo data: ITGAL, CDH5, ITGAM, CD34, CD79A, HLA-DRA]
#[not integrated in I/M/O data: ITGAL, CDH5, ITGAM, CD34, CD79A]
genes <- c("THY1","NT5E","ENG","CD44","ANPEP","NGFR","ITGB1","NCAM1","CD9","CD14","MME","ITGAL","ITGAX","CD36","ITGA3","ITGA4","ITGAV","ICAM1","CD55",
          "CD59","TFRC","VCAM1","KIT","TNFRSF1A","PDGFRB","CDH5","MCAM","ALCAM","ERBB2","ITGAM","CD14","PECAM1","CD34","PTPRC","CD79A","HLA-DRA")

FeaturePlot(indv, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=4)

#osteoblast
#diagnostic: RUNX2, BGLAP, CAPG, MEPE
#additional: COL1A1, COL1A2, ALPL, TNFRSF11B (OPG), TNFSF11 (RANKL), CSF1 (M-CSF), DKK1, SFRP1, PTGES2, PTGS2, SPP1 (OPN), SPARC (ONT)
#if needed: DMP1*, PHEX*, E11**, SOST*
# * not integrated (osteo) / ** not in ortho-exon (osteo)
#[not integrated in ipsc data: RUNX2, BGLAP, MEPE, DMP1]
#[not integrated in msc data: BGLAP, TNFSF11, DMP1, PHEX]
#[not integrated in I/M/O data: TNFSF11, PTGES2, DMP1, PHEX]
genes <- c("RUNX2","BGLAP","CAPG","MEPE","COL1A1","COL1A2","ALPL","TNFRSF11B","TNFSF11",
           "CSF1","DKK1","SFRP1","PTGES2","PTGS2","SPP1","SPARC","DMP1","PHEX","SOST")

FeaturePlot(indv, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(indv, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=5)

FeaturePlot(indv, features=genes[5:19], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((FeaturePlot(subset(indv,subset=species=="Human"), features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="Sample") +
                           labs(x="UMAP1",y="UMAP2")),
                        (FeaturePlot(subset(indv,subset=species=="Chimp"), features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="Sample") +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)

FeaturePlot(indv, features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="species")
FeaturePlot(indv, features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="Individual", ncol=6)

#chondrocyte diagnostic: ACAN, COL2A1, SOX9, COL10A1, COL11A1, SOX5, SOX6
#adipocyte diagnostic: FABP4, PPARG, EBF2, PDGFRA, FGF10, FGF16, FGF19, SLC7A10*, SLC36A2*, MYF5*, P2RX5
# * not integrated (ipsc, msc, osteo)
#[not integrated in ipsc data: COL10A1, PPARG, SLC7A10, SLC36A2, MYF5]
#[not integrated in msc data: SLC7A10, SLC36A2, MYF5, FGF16]
#[not integrated in osteo data: COL2A1, SLC7A10, SLC36A2, MYF5, FGF16]
#[not integrated in I/M/O data: SLC7A10, SLC36A2, P2RX5, MYF5]
genes <- c("ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2",
           "COL10A1","COL11A1","SOX5","SOX6",
           "PDGFRA","FGF10","FGF16","FGF19","SLC7A10","SLC36A2","P2RX5","MYF5")

FeaturePlot(indv, features=genes[c(1,4,2,5,3,6)], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")


#Isolate genes of interest

genes <- c("POU5F1","TDGF1","SOX2","EPCAM","THY1","NT5E","ENG","CD44","RUNX2","BGLAP","CAPG","MEPE","COL1A1","COL1A2","ALPL",
           "ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2")
#ips.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[1:4],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
#msc.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[5:8],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
#ost.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[9:12],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
#cho.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[13:15],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
#adi.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[16:18],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
tot.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes,]))),indv@meta.data[,c("species","Cell.Type","Sample")])

                   
#Deterimine Gene Expression Cutoffs for Ad Hoc Cell Type Assignments

gene=genes[15]

hist(x=tot.genes[,gene], breaks=100, main=gene, xlab=paste0(gene," Expression"))
abline(v=c(median(tot.genes[,gene]),
           mean(tot.genes[,gene]),
           mean(tot.genes[,gene]) + sd(tot.genes[,gene]),
           median(tot.genes[,gene][tot.genes[,gene]<1]),
           mean(tot.genes[,gene][tot.genes[,gene]<1]),
           mean(tot.genes[,gene][tot.genes[,gene]<1]) + sd(tot.genes[,gene][tot.genes[,gene]<1])),
       col=c("blue","black","red","blue","black","red"),
       lty=c(1,1,1,2,2,2))
text(x=3, y=10000, cex=0.75,
     labels=paste0("median(tot): ",round(median(tot.genes[,gene]),digits=4),
                   "\nmean(tot): ",round(mean(tot.genes[,gene]),digits=4),
                   "\nmean+sd(tot): ",round(mean(tot.genes[,gene]) + sd(tot.genes[,gene]),digits=4),
                   "\nmedian(<1): ",round(median(tot.genes[,gene][tot.genes[,gene]<1]),digits=4),
                   "\nmean(<1): ",round(mean(tot.genes[,gene][tot.genes[,gene]<1]),digits=4),
                   "\nmean+sd(<1): ",round(mean(tot.genes[,gene][tot.genes[,gene]<1]) + sd(tot.genes[,gene][tot.genes[,gene]<1]),digits=4)))

round(median(tot.genes[,gene]),digits=4) #median(tot)
round(mean(tot.genes[,gene]),digits=4)   #mean(tot)
round(mean(tot.genes[,gene]) + sd(tot.genes[,gene]),digits=4) #mean+sd
round(median(tot.genes[,gene][tot.genes[,gene]<1]),digits=4) #median(<1)
round(mean(tot.genes[,gene][tot.genes[,gene]<1]),digits=4)   #mean(<1)
round(mean(tot.genes[,gene][tot.genes[,gene]<1]) + sd(tot.genes[,gene][tot.genes[,gene]<1]),digits=4) #mean+sd(<1)


gene=genes[15]
hist(x=tot.genes[,gene], breaks=100, main=gene, xlab=paste0(gene," Expression"))
abline(v=0.15, col="green", lty=1, lwd=2)
text(x=2, y=10000, cex=0.75, labels="manual: 0.15")

#             POU5F1-tot  TDGF1-tot SOX2-tot  EPCAM-tot THY1-tot  NT5E-tot  ENG-tot CD44-tot  RUNX2-osteo BGLAP-osteo CAPG-osteo  MEPE-osteo
#>0           -           -         -         -         -         -         -       -         -           -           -           -
#>0.1         -           -         -         -         -         -         -       -         -           -           -           -
#>0.3         -           -         -         -         -         -         -       -         -           -           -           -
#>0.5         -           -         -         -         -         -         -       -         -           -           -           -
#>0.7         -           -         -         -         -         -         -       -         -           -           -           -
#>0.9         -           -         -         -         -         -         -       -         -           -           -           -
#>1           -           -         -         -         -         -         -       -         -           -           -           -
#median(tot)  0.0066      0.0010    0.0160    0.0162    0.6896    0.0717    0.1447  0.1329    0.0040      0.0218      0.4322      0.0000
#mean(tot)    0.9519      0.4972    0.4496    0.4766    0.6800    0.2614    0.2643  0.3722    0.0727      0.0584      0.4863     -0.0013
#mean+sd(tot) 2.2700      1.2676    1.1567    1.2339    1.3942    0.6630    0.6108  0.9096    0.2927      0.2265      0.9814      0.0164
#median(<1)   0.0000      0.0000    0.0059    0.0076    0.2752    0.0279    0.1155  0.0360    0.0035      0.0215      0.3337      0.0000
#mean(<1)     0.0040      0.0669    0.0737    0.0714    0.2648    0.1898    0.2193  0.2015    0.0564      0.0514      0.3383     -0.0014
#mean+sd(<1)  0.1012      0.2783    0.2883    0.2793    0.7407    0.4776    0.4843  0.5487    0.2113      0.1823      0.6667      0.0115
#manual       0.3         0.2       0.2       0.4       0.3       0.4       0.15    0.1       0.2         0.2         0.2         0.05

#             ACAN-tot  COL2A1-tot  SOX9-tot  FABP4-tot PPARG-tot EBF2-tot  COL1A1-tot  COL1A2-tot  ALPL-tot
#>0           -         -           -         -         -         -         -           -           -
#>0.1         -         -           -         -         -         -         -           -           -
#>0.3         -         -           -         -         -         -         -           -           -
#>0.5         -         -           -         -         -         -         -           -           -
#>0.7         -         -           -         -         -         -         -           -           -
#>0.9         -         -           -         -         -         -         -           -           -
#>1           -         -           -         -         -         -         -           -           -
#median(tot)  0.0000    0.0000      0.0001    0.0000    0.0000    0.0000    3.5837      2.4306      0.0331
#mean(tot)    0.0028    0.0130      0.0286    0.0021    0.0209    0.0009    2.7469      1.9487      0.1899
#mean+sd(tot) 0.0734    0.1331      0.1885    0.1230    0.1740    0.0291    4.7741      3.6127      0.5524
#median(<1)   0.0000    0.0000      0.0001    0.0000    0.0000    0.0000    0.1335      0.0005      0.0270
#mean(<1)     0.0019    0.0084      0.0226   -0.0027    0.0145    0.0007    0.2379      0.1029      0.1422
#mean+sd(<1)  0.0626    0.0927      0.1520    0.0722    0.1312    0.0244    0.5136      0.3875      0.4099
#manual       0.05      0.1         0.1       0.1       0.15      0.05      0.3         0.2         0.15



#Make upsetr plotS

chk.thres.manual <- function(data, genes, thres) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres[i], 1, 0)
    i=i+1
  }
  return(assignments)
}
chk.thres.median <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- median(data[,genes[i]]) }
    else { thres <- median(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}
chk.thres.mean <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- mean(data[,genes[i]]) }
    else { thres <- mean(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}
chk.thres.meansd <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- sum(mean(data[,genes[i]]),sd(data[,genes[i]])) }
    else { thres <- sum(mean(data[,genes[i]][data[,genes[i]]<cutoff]),sd(data[,genes[i]][data[,genes[i]]<cutoff])) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}
plot.upset.HCcellcompare <- function(data, genes, ymax) {
  plot.ls <- list(
    iPSC.Human       = upset(data[which(data$Cell.Type=="iPSC" & data$species=="Human"),genes],
                            nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    MSC.Human        = upset(data[which(data$Cell.Type=="MSC" & data$species=="Human"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    Osteoblast.Human = upset(data[which(data$Cell.Type=="Osteoblast" & data$species=="Human"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    iPSC.Chimp       = upset(data[which(data$Cell.Type=="iPSC" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    MSC.Chimp        = upset(data[which(data$Cell.Type=="MSC" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    Osteoblast.Chimp = upset(data[which(data$Cell.Type=="Osteoblast" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit('arrange', name=v)
    vp <- grid.grab()
    plot.ls[[v]] <- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}
plot.upset.sub.HCcellcompare <- function(data, genes, intersections, ymax) {
  plot.ls <- list(
    iPSC.Human       = upset(data[which(data$Cell.Type=="iPSC" & data$species=="Human"),genes],
                            nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                            sets=genes, intersections=intersections, mainbar.y.max=ymax),
    MSC.Human        = upset(data[which(data$Cell.Type=="MSC" & data$species=="Human"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                             sets=genes, intersections=intersections, mainbar.y.max=ymax),
    Osteoblast.Human = upset(data[which(data$Cell.Type=="Osteoblast" & data$species=="Human"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                             sets=genes, intersections=intersections, mainbar.y.max=ymax),
    iPSC.Chimp       = upset(data[which(data$Cell.Type=="iPSC" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                             sets=genes, intersections=intersections, mainbar.y.max=ymax),
    MSC.Chimp        = upset(data[which(data$Cell.Type=="MSC" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                             sets=genes, intersections=intersections, mainbar.y.max=ymax),
    Osteoblast.Chimp = upset(data[which(data$Cell.Type=="Osteoblast" & data$species=="Chimp"),genes],
                             nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                             sets=genes, intersections=intersections, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit('arrange', name=v)
    vp <- grid.grab()
    plot.ls[[v]] <- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}

genes <- c("POU5F1","TDGF1","SOX2","EPCAM","THY1","NT5E","ENG","CD44","RUNX2","BGLAP","CAPG","MEPE","COL1A1","COL1A2","ALPL",
           "ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2")
tot.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes,]))),indv@meta.data[,c("species","Cell.Type","Sample")])

#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0,length(genes))))
assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,length(genes))))
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.3,length(genes))))
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.5,length(genes))))
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.7,length(genes))))
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.9,length(genes))))
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(rep(1,length(genes))))
#assignments <- chk.thres.median(tot.genes, genes, cutoff=NA)
#assignments <- chk.thres.median(tot.genes, genes, cutoff=1)
#ssignments <- chk.thres.mean(tot.genes, genes, cutoff=NA)
#assignments <- chk.thres.mean(tot.genes, genes, cutoff=1)
#assignments <- chk.thres.meansd(tot.genes, genes, cutoff=NA)
#assignments <- chk.thres.meansd(tot.genes, genes, cutoff=1)
#assignments <- chk.thres.manual(tot.genes, genes, thres=c(0.3,0.2,0.2,0.4,0.3,0.4,0.15,0.1,0.2,0.2,0.2,0.05,0.3,0.2,0.15,0.05,0.1,0.1,0.1,0.15,0.05))

plot.upset.HCcellcompare(assignments,genes[1:4],20000)
plot.upset.HCcellcompare(assignments,genes[5:8],20000)
plot.upset.HCcellcompare(assignments,genes[9:12],20000)
plot.upset.HCcellcompare(assignments,genes[13:15],20000)
plot.upset.HCcellcompare(assignments,genes[16:18],20000)
plot.upset.HCcellcompare(assignments,genes[19:21],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1),],genes[9:12],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0),],genes[9:12],20000)
upset(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0 &
                          assignments$Cell.Type=="Osteoblast" & assignments$species=="Chimp"),genes[9:12]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[9:12], mainbar.y.max=2000)


CombinePlots(list((DimPlot(indv, group.by="Cell.Type", reduction="umap") +
                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
                  (DimPlot(indv, group.by="species", reduction="umap",
                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1),])) +
                    labs(x="UMAP1",y="UMAP2",title="iPSCs")),
                  (DimPlot(indv, group.by="species", reduction="umap",
                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1),])) +
                    labs(x="UMAP1",y="UMAP2",title="MSCs")),
                  (DimPlot(indv, group.by="species", reduction="umap",
                          cells=rownames(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1),])) +
                    labs(x="UMAP1",y="UMAP2",title="Osteoblasts"))), ncol=4)

CombinePlots(list((DimPlot(indv, group.by="Cell.Type", reduction="umap") +
                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
                  (DimPlot(indv, group.by="species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1),])) +
                    labs(x="UMAP1",y="UMAP2",title="iPSCs")),
                  (DimPlot(indv, group.by="species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1),])) +
                    labs(x="UMAP1",y="UMAP2",title="MSCs")),
                  (DimPlot(indv, group.by="species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0),])) +
                    labs(x="UMAP1",y="UMAP2",title="Osteoblasts"))), ncol=4)



upset(assignments[which(assignments$Cell.Type=="iPSC"),genes[1:4]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[1:4], mainbar.y.max=30000)
upset(assignments[which(assignments$Cell.Type=="Osteoblast"),genes[1:4]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[1:4], mainbar.y.max=20000)
upset(assignments[which(assignments$Cell.Type=="Osteoblast"),genes[1:4]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[1:4], mainbar.y.max=20000)

upset(assignments[,genes[9:12]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[9:12], mainbar.y.max=15000)
upset(assignments[which(assignments$species=="Human"),genes[9:12]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[9:12], mainbar.y.max=15000)
upset(assignments[which(assignments$species=="Chimp"),genes[9:12]],
      nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes[9:12], mainbar.y.max=15000)

#plot.upset.HCcellcompare(assignments,genes[c(9:12,19:29)],20000)

#genes <- c("POU5F1","TDGF1","SOX2","EPCAM","THY1","NT5E","ENG","CD44","RUNX2","BGLAP","CAPG","MEPE","ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2",
#           "COL1A1","COL1A2","ALPL","TNFRSF11B","CSF1","DKK1","SFRP1","PTGS2","SPP1","SPARC","SOST")
#ost.genes <- cbind(as.data.frame(t(as.matrix(indv@assays$integrated@data[genes[c(9:12,19:29)],]))),indv@meta.data[,c("species","Cell.Type","Sample")])
#assignments <- chk.thres.manual(ost.genes, genes[c(9:12,19:29)], thres=c(rep(0.1,length(genes))))
#plot.upset.HCcellcompare(assignments,genes[c(9:12,29)],20000)
#plot.upset.HCcellcompare(assignments,genes[c(19:21,12)],20000)
#plot.upset.sub.HCcellcompare(assignments,genes[c(9:10,12,19:21)],ymax=20000,
#                             intersections=list(list("COL1A1","COL1A2"),
#                                                list("COL1A1","COL1A2","ALPL"),
#                                                list("COL1A1","COL1A2","ALPL","RUNX2"),
#                                                list("COL1A1","COL1A2","ALPL","RUNX2","BGLAP"),
#                                                list("COL1A1","COL1A2","ALPL","BGLAP"),
#                                                list("COL1A1","COL1A2","ALPL","BGLAP","MEPE"),
#                                                list("COL1A1","COL1A2","ALPL","MEPE"),
#                                                list("RUNX2"),
#                                                list("RUNX2","BGLAP"),
#                                                list("BGLAP"),
#                                                list("BGLAP","MEPE"),
#                                                list("MEPE")))
#plot.upset.sub.HCcellcompare(assignments,genes[c(9:12,19:21)],ymax=20000,
#                             intersections=list(list("COL1A1","COL1A2"),
#                                                list("COL1A1","COL1A2","CAPG"),
#                                                list("COL1A1","COL1A2","ALPL"),
#                                                list("COL1A1","COL1A2","ALPL","CAPG"),
#                                                list("COL1A1","COL1A2","ALPL","RUNX2"),
#                                                list("COL1A1","COL1A2","ALPL","RUNX2","BGLAP"),
#                                                list("COL1A1","COL1A2","ALPL","BGLAP","CAPG"),
#                                                list("COL1A1","COL1A2","ALPL","BGLAP","CAPG","MEPE"),
#                                                list("COL1A1","COL1A2","ALPL","CAPG","MEPE"),
#                                                list("RUNX2"),
#                                                list("RUNX2","BGLAP"),
#                                                list("BGLAP","CAPG"),
#                                                list("BGLAP","CAPG","MEPE"),
#                                                list("CAPG","MEPE")))

upset(assignments[,1:12], nintersects=NA, keep.order=TRUE, order.by="freq",
      sets=genes[1:12],
      intersections=list(list("POU5F1","TDGF1","SOX2","EPCAM"),
                         list("THY1","NT5E","ENG","CD44"),
                         list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG"),
                         list("BGLAP","CAPG","MEPE"),
                         list("CAPG","MEPE")))

upset(ost.genes, keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","MEPE"),
      intersections=list(list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG"),
                         list("BGLAP","CAPG","MEPE"),
                         list("CAPG","MEPE")),
      order.by="freq")

upset(ost.genes, keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","MEPE"),
      intersections=list(list("BGLAP"),
                         list("RUNX2","BGLAP","CAPG"),
                         list("RUNX2","CAPG"),
                         list("CAPG"),
                         list("RUNX2","BGLAP","CAPG","MEPE"),
                         list("RUNX2","BGLAP","MEPE"),
                         list("RUNX2","CAPG","MEPE"),
                         list("RUNX2","MEPE"),
                         list("BGLAP","MEPE"),
                        list("MEPE")),
      order.by=c("freq"))

upset(ost.genes, keep.order=TRUE, empty.intersections="on", group.by="degree",
      sets=c("RUNX2","BGLAP","CAPG","MEPE"))



#assign cell type labels based on ad hoc cutoffs
hoc.assign <- assignments

hoc.assign$assign <- "other"
i=1
while(i <= length(rownames(hoc.assign))) {
  if(hoc.assign$POU5F1[i] == 1 & hoc.assign$TDGF1[i] == 1 & hoc.assign$SOX2[i] == 1 & hoc.assign$EPCAM[i] == 1 &
     hoc.assign$NT5E[i] == 0 & hoc.assign$ENG[i] == 0 & hoc.assign$CD44[i] == 0) {hoc.assign$assign[i] <- "iPSC"}
  if(hoc.assign$THY1[i] == 1 & hoc.assign$NT5E[i] == 1 & hoc.assign$ENG[i] == 1 & hoc.assign$CD44[i] == 1 &
     hoc.assign$ALPL[i] == 0) {hoc.assign$assign[i] <- "MSC"}
  if(hoc.assign$POU5F1[i] == 0 & hoc.assign$COL1A1[i] == 1 & hoc.assign$COL1A2[i] == 1 & hoc.assign$ALPL[i] == 1) {hoc.assign$assign[i] <- "Osteoblast"}
  i=i+1
}
table(hoc.assign$assign)

indv@meta.data$AdHoc.Assign <- hoc.assign$assign

CombinePlots(plots=list((DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="pca",
                                 dims=c(1,2),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="pca",
                                 dims=c(3,4),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="pca",
                                 dims=c(5,6),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="pca",
                                 dims=c(7,8),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="pca",
                                 dims=c(9,10),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC9",y="PC10"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"),
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"),
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Cell.Type=="iPSC"),
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="umap",
                                 cols=c("pink")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="MSC"),
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="umap",
                                 cols=c("mediumorchid")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="Osteoblast"),
                                 group.by="Cell.Type",
                                 split.by="AdHoc.Assign",
                                 reduction="umap",
                                 cols=c("dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
table(subset(indv,subset=species=="Human")@meta.data$AdHoc.Assign)
table(subset(indv,subset=species=="Chimp")@meta.data$AdHoc.Assign)
table(subset(indv,subset=Cell.Type=="iPSC")@meta.data$AdHoc.Assign)
table(subset(indv,subset=Cell.Type=="MSC")@meta.data$AdHoc.Assign)
table(subset(indv,subset=Cell.Type=="Osteoblast")@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="iPSC"))@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="MSC"))@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="Osteoblast"))@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="iPSC"))@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="MSC"))@meta.data$AdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="Osteoblast"))@meta.data$AdHoc.Assign)



#assign osteogenic cell sub-type labels based on ad hoc cutoffs
ost.assign <- assignments

ost.assign$assign <- "other"
i=1
while(i <= length(rownames(ost.assign))) {

  #strict criteria
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "preosteoblast"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "osteoblast"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "embedding osteoblast"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mineralizing osteocyte"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mature osteocyte"}
  
  #lenient criteria
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "osteoblast"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "embedding osteoblast"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "embedding osteoblast"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0) {ost.assign$assign[i] <- "embedding osteoblast"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mineralizing osteocyte"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mineralizing osteocyte"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mature osteocyte"}
  #if(ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mature osteocyte"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mature osteocyte"}
  #if(ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1) {ost.assign$assign[i] <- "mature osteocyte"}
  
  #strict criteria with COL1A1, COL1A2, ALPL, POUF51
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "preosteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "embedding osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mineralizing osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mature osteocyte"}
  
  #lenient criteria with COL1A1, COL1A2, ALPL, POUF51
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "embedding osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "embedding osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==0)
    {ost.assign$assign[i] <- "embedding osteoblast"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mineralizing osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mineralizing osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==1 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mature osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==1 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mature osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==1 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mature osteocyte"}
  if(ost.assign$COL1A1[i]==1 & ost.assign$COL1A2[i]==1 & ost.assign$ALPL[i]==1 & ost.assign$POU5F1[i]==0 &
     ost.assign$RUNX2[i]==0 & ost.assign$BGLAP[i]==0 & ost.assign$CAPG[i]==0 & ost.assign$MEPE[i]==1)
    {ost.assign$assign[i] <- "mature osteocyte"}
  
  i=i+1
}
table(ost.assign$assign)

indv@meta.data$OstAdHoc.Assign <- ost.assign$assign
indv@meta.data$OstAdHoc.Assign <- factor(indv@meta.data$OstAdHoc.Assign,
                                         levels=c("preosteoblast","osteoblast","embedding osteoblast",
                                                  "mineralizing osteocyte","mature osteocyte","other"))

DimPlot(indv, group.by="Cell.Type", split.by="OstAdHoc.Assign", reduction="umap", cols=c("lightgrey","lightgrey","dodgerblue")) + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="pca",
                                 dims=c(1,2),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="pca",
                                 dims=c(3,4),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="pca",
                                 dims=c(5,6),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="pca",
                                 dims=c(7,8),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(indv,
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="pca",
                                 dims=c(9,10),
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="PC9",y="PC10"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"),
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="umap",
                                 cols=c("lightgrey","lightgrey","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"),
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="umap",
                                 cols=c("lightgrey","lightgrey","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Cell.Type=="iPSC"),
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="umap",
                                 cols=c("lightgrey")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="MSC"),
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="umap",
                                 cols=c("lightgrey")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Cell.Type=="Osteoblast"),
                                 group.by="Cell.Type",
                                 split.by="OstAdHoc.Assign",
                                 reduction="umap",
                                 cols=c("dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
table(subset(indv,subset=species=="Human")@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=species=="Chimp")@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=Cell.Type=="iPSC")@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=Cell.Type=="MSC")@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=Cell.Type=="Osteoblast")@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="iPSC"))@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="MSC"))@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Human" & Cell.Type=="Osteoblast"))@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="iPSC"))@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="MSC"))@meta.data$OstAdHoc.Assign)
table(subset(indv,subset=(species=="Chimp" & Cell.Type=="Osteoblast"))@meta.data$OstAdHoc.Assign)

```

```{r look at candidate genes - OLD}

#Look at candidate genes
#ipsc
FeaturePlot(indv, features="NANOG", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #NANOG  --- RNA
FeaturePlot(indv, features="POU5F1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")   #POU5F1 --- RNA
FeaturePlot(indv, features="TDGF1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #TDGF1  --- RNA
RidgePlot(indv, features=c("NANOG","POU5F1","TDGF1"), ncol=3)
#msc
FeaturePlot(indv, features="MIXL1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #MIXL1  --- RNA
FeaturePlot(indv, features="TBXT", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #TBXT   --- RNA
FeaturePlot(indv, features="TDGF1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #TDGF1  --- RNA
FeaturePlot(indv, features="CD44", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD44
FeaturePlot(indv, features="CD200", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD200  --- RNA (1K)
FeaturePlot(indv, features="NGFR", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD271  --- RNA (1K)
RidgePlot(indv, features=c("MIXL1","TBXT","TDGF1","CD44","CD200","NGFR"), ncol=3)
#osteoblast
FeaturePlot(indv, features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")   #COL1A1
FeaturePlot(indv, features="COL1A2", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")   #COL1A2
FeaturePlot(indv, features="ALPL", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #ALPL   --- RNA (1K)
FeaturePlot(indv, features="RUNX2", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #RUNX2  --- RNA (1K)
FeaturePlot(indv, features="BGLAP", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #OSC    --- RNA (1K)
FeaturePlot(indv, features="CAPG", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CAPG   --- RNA (1K)
FeaturePlot(indv, features="DMP1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #DMP1   --- RNA
FeaturePlot(indv, features="PHEX", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #PHEX   --- RNA
FeaturePlot(indv, features="MEPE", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #MEPE   --- RNA (1K, 6K)
FeaturePlot(indv, features="SPDYE11", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")  #E11    --- not in data
FeaturePlot(indv, features="SOST", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOST   --- RNA
RidgePlot(indv, features=c("COL1A1","COL1A2","ALPL","RUNX2","BGLAP","CAPG","DMP1","PHEX","MEPE","SOST"), ncol=5)
RidgePlot(indv, features=c("ALPL","RUNX2","BGLAP","DMP1","PHEX","MEPE","SOST"), ncol=4)
#all-ipsc/msc/osteoblast
FeaturePlot(indv, features="THY1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD90
#some-msc/osteoblast
FeaturePlot(indv, features="CD44", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD44
FeaturePlot(indv, features="NT5E", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD73
FeaturePlot(indv, features="ENG", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")      #CD105
FeaturePlot(indv, features="ITGB1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD29   --- RNA (1K)
FeaturePlot(indv, features="ALCAM", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD166
FeaturePlot(indv, features="MCAM", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD146
#negative-msc
FeaturePlot(indv, features="CD34", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD34   --- RNA
FeaturePlot(indv, features="PTPRC", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD45   --- RNA
FeaturePlot(indv, features="ITGAM", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD11b  --- RNA
FeaturePlot(indv, features="CD19", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD19   --- RNA
FeaturePlot(indv, features="HLA-DRB1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2") #HLA-DR --- RNA
FeaturePlot(indv, features="CD14", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #CD14
FeaturePlot(indv, features="CD79A", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #CD79A  --- RNA
#chondrocyte
FeaturePlot(indv, features="ACAN", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #ACAN   --- RNA (1K)
FeaturePlot(indv, features="COL2A1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")   #COL2A1 --- RNA
FeaturePlot(indv, features="SOX9", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOX9   --- RNA (1K)
FeaturePlot(indv, features="COL10A1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")  #COL10A1--- RNA (1K)
FeaturePlot(indv, features="COL11A1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")  #COL11A1
FeaturePlot(indv, features="SOX5", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOX5   --- RNA (1K)
FeaturePlot(indv, features="SOX6", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOX6   --- RNA (1K)
#adipocyte
FeaturePlot(indv, features="FABP4", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #FABP4  --- RNA (1K)
FeaturePlot(indv, features="PPARG", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #PPARG  --- RNA (1K)
#endomesoderm
FeaturePlot(indv, features="EOMES", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #EOMES  --- RNA
#endoderm
FeaturePlot(indv, features="SOX17", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #SOX17  --- RNA
#neuroectoderm
FeaturePlot(indv, features="SOX1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOX1   --- RNA
#neuroepithelial
FeaturePlot(indv, features="SOX2", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #SOX2   --- RNA
#epithelial
FeaturePlot(indv, features="EPCAM", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #EPCAM  --- RNA (1K)
#hepatocyte
FeaturePlot(indv, features="ALB", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")      #ALB    --- RNA
#heart/neural crest
FeaturePlot(indv, features="HAND1", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #HAND1  --- RNA (1K)
#cardiomyocyte
FeaturePlot(indv, features="TNNT2", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #TNNT2  --- RNA (1K)
#cardiac muscle
FeaturePlot(indv, features="MYL7", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #MYL7   --- RNA (1K)
FeaturePlot(indv, features="MYH6", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #MYH6   --- RNA
#neutral
FeaturePlot(indv, features="GAPDH", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #GAPDH  --- RNA
FeaturePlot(indv, features="GUSB", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")     #GUSB   --- RNA
FeaturePlot(indv, features="YWHAZ", reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")    #YWHAZ  --- RNA (1K,6K)
RidgePlot(indv, features=c("GAPDH","GUSB","YWHAZ"), ncol=3)
 

```

```{r ad hoc cell classification - OLD}

CountPlot <- function(gene,ymax) {
  par(mfrow=c(2,8))
  barplot(table(subset(indv,subset=species=="Human")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("All Human Cells")
  barplot(table(subset(indv,subset=Sample=="H1-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H1-O")
  barplot(table(subset(indv,subset=Sample=="H1-O-r2")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H1-O-r2")
  barplot(table(subset(indv,subset=Sample=="H2-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H2-O")
  barplot(table(subset(indv,subset=Sample=="H3-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H3-O")
  barplot(table(subset(indv,subset=Sample=="H4-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H4-O")
  barplot(table(subset(indv,subset=Sample=="H5-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H5-O")
  barplot(table(subset(indv,subset=Sample=="H6-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("H6-O")
  barplot(table(subset(indv,subset=species=="Chimp")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("All Chimp Cells")
  barplot(table(subset(indv,subset=Sample=="C1-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C1-O")
  barplot(table(subset(indv,subset=Sample=="C1-O-r2")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C1-O-r2")
  barplot(table(subset(indv,subset=Sample=="C2-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C2-O")
  barplot(table(subset(indv,subset=Sample=="C3-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C3-O")
  barplot(table(subset(indv,subset=Sample=="C4-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C4-O")
  barplot(table(subset(indv,subset=Sample=="C5-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C5-O")
  barplot(table(subset(indv,subset=Sample=="C6-O")@assays$integrated@data[gene,]>0), ylim=c(0,ymax))
  title("C6-O")
}
CountPlotRNA <- function(gene,ymax) {
  par(mfrow=c(2,8))
  barplot(table(subset(indv,subset=species=="Human")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("All Human Cells")
  barplot(table(subset(indv,subset=Sample=="H1-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H1-O")
  barplot(table(subset(indv,subset=Sample=="H1-O-r2")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H1-O-r2")
  barplot(table(subset(indv,subset=Sample=="H2-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H2-O")
  barplot(table(subset(indv,subset=Sample=="H3-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H3-O")
  barplot(table(subset(indv,subset=Sample=="H4-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H4-O")
  barplot(table(subset(indv,subset=Sample=="H5-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H5-O")
  barplot(table(subset(indv,subset=Sample=="H6-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("H6-O")
  barplot(table(subset(indv,subset=species=="Chimp")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("All Chimp Cells")
  barplot(table(subset(indv,subset=Sample=="C1-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C1-O")
  barplot(table(subset(indv,subset=Sample=="C1-O-r2")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C1-O-r2")
  barplot(table(subset(indv,subset=Sample=="C2-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C2-O")
  barplot(table(subset(indv,subset=Sample=="C3-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C3-O")
  barplot(table(subset(indv,subset=Sample=="C4-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C4-O")
  barplot(table(subset(indv,subset=Sample=="C5-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C5-O")
  barplot(table(subset(indv,subset=Sample=="C6-O")@assays$RNA@data[gene,]>0), ylim=c(0,ymax))
  title("C6-O")
}

CountPlot("POU5F1",14000)
CountPlot("TDGF1",14000)
CountPlot("SOX2",14000)
CountPlot("EPCAM",14000)

CountPlot("THY1",14000)
CountPlot("NT5E",14000)
CountPlot("ENG",14000)
CountPlot("CD44",14000)

CountPlot("RUNX2",14000)
CountPlot("BGLAP",14000)
CountPlot("CAPG",14000)
CountPlot("MEPE",14000)

CountPlot("ACAN",14000)
CountPlot("COL2A1",14000)
CountPlot("SOX9",14000)
CountPlot("FABP4",14000)
CountPlot("PPARG",14000)
CountPlot("EBF2",14000)


#Deterimine Gene Expression Cutoffs for Ad Hoc Cell Type Assignments

hist(x=tot.genes[,genes[1]], breaks=1000, main=genes[1], xlab=paste0(genes[1]," Expression"), xlim=c(0,0.1))
abline(v=0.025,col="blue") #lenient
abline(v=0.1,col="black") #mild
abline(v=0.3,col="red") #strict

hist(x=tot.genes[,genes[1]], breaks=1000, main=genes[1], xlab=paste0(genes[1]," Expression"))
abline(v=0.025,col="blue")
abline(v=0.1,col="black")
abline(v=0.3,col="red")
text(x=1.5, y=2500, labels="lenient cutoff = 0.025", col="blue")
text(x=1.5, y=2000, labels="mild cutoff = 0.1", col="black")
text(x=1.5, y=1500, labels="strict cutoff = 0.3", col="red")

median(tot.genes[,genes[1]]) #median(tot)
mean(tot.genes[,genes[1]])   #mean(tot)
mean(tot.genes[,genes[1]]) + sd(tot.genes[,genes[1]]) #mean+sd
median(tot.genes[,genes[1]][tot.genes[,genes[1]]<1]) #median(<1)
mean(tot.genes[,genes[1]][tot.genes[,genes[1]]<1])   #mean(<1)
mean(tot.genes[,genes[1]][tot.genes[,genes[1]]<1]) + sd(tot.genes[,genes[1]][tot.genes[,genes[1]]<1]) #mean+sd(<1)

#             RUNX2-osteo BGLAP-osteo CAPG-osteo  MEPE-osteo
#>0           -           -           -           -
#>0.1         -           -           -           -
#>0.3         -           -           -           -
#>0.5         -           -           -           -
#lenient      0.0250      0.0200      0.0450      0.0000
#mild         0.1000      0.0450      0.1050      0.0180
#strict       0.3000      0.2100      0.3250      0.0400
#median(tot)  0.0454      0.0406      0.5314     -0.0010
#mean(tot)    0.1679      0.0770      0.6470     -0.0036
#mean+sd(tot) 0.5337      0.3044      1.2465      0.0269
#median(<1)   0.0391      0.0396      0.3672     -0.0010
#mean(<1)     0.1103      0.0577      0.3805      0.0038


#Classify cell types (ad hoc) using <=0 and >0 expression cutoffs

#not osteogenic (705 cells)
table(indv@assays$integrated@data["RUNX2",]<=0
      & indv@assays$integrated@data["BGLAP",]<=0
      & indv@assays$integrated@data["CAPG",]<=0
      & indv@assays$integrated@data["MEPE",]<=0
      & indv@assays$RNA@data["SOST",]<=0)
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="not osteogenic",
     col=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="not osteogenic",
     col=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="not osteogenic",
     col=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), "blue", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), 0.5, 0.1))
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="not osteogenic",
     col=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), "black", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]<=0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), 0.5, 0.1))
par(mfrow=c(1,3))
barplot(table(indv@assays$integrated@data["RUNX2",]<=0
              & indv@assays$integrated@data["BGLAP",]<=0
              & indv@assays$integrated@data["CAPG",]<=0
              & indv@assays$integrated@data["MEPE",]<=0
              & indv@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Cells")
barplot(table(subset(indv,subset=species=="Human")@assays$integrated@data["RUNX2",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["BGLAP",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Human")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Human Cells")
barplot(table(subset(indv,subset=species=="Chimp")@assays$integrated@data["RUNX2",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["BGLAP",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Chimp")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Chimp Cells")

#preosteoblast (112 cells)
table(indv@assays$integrated@data["RUNX2",]>0
      & indv@assays$integrated@data["BGLAP",]<=0
      & indv@assays$integrated@data["CAPG",]<=0
      & indv@assays$integrated@data["MEPE",]<=0
      & indv@assays$RNA@data["SOST",]<=0)
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="preosteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="preosteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="preosteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), "blue", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), 0.5, 0.1))
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="preosteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), "black", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]<=0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), 0.5, 0.1))
par(mfrow=c(1,3))
barplot(table(indv@assays$integrated@data["RUNX2",]>0
              & indv@assays$integrated@data["BGLAP",]<=0
              & indv@assays$integrated@data["CAPG",]<=0
              & indv@assays$integrated@data["MEPE",]<=0
              & indv@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Cells")
barplot(table(subset(indv,subset=species=="Human")@assays$integrated@data["RUNX2",]>0
              & subset(indv,subset=species=="Human")@assays$integrated@data["BGLAP",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Human")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Human Cells")
barplot(table(subset(indv,subset=species=="Chimp")@assays$integrated@data["RUNX2",]>0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["BGLAP",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Chimp")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Chimp Cells")

#osteoblast (1322 cells)
table(indv@assays$integrated@data["RUNX2",]>0
      & indv@assays$integrated@data["BGLAP",]>0
      & indv@assays$integrated@data["CAPG",]<=0
      & indv@assays$integrated@data["MEPE",]<=0
      & indv@assays$RNA@data["SOST",]<=0)
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="osteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="osteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["CAPG",], y=indv@assays$integrated@data["MEPE",], xlab="CAPG", ylab="MEPE", pch=20, main="osteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), "blue", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0 &
                   (indv@assays$integrated@data["CAPG",]>0 | indv@assays$integrated@data["MEPE",]>0)), 0.5, 0.1))
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["BGLAP",], xlab="RUNX2", ylab="BGLAP", pch=20, main="osteoblast",
     col=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), "black", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["RUNX2",]>0 & indv@assays$integrated@data["BGLAP",]>0 &
                   indv@assays$integrated@data["CAPG",]<=0 & indv@assays$integrated@data["MEPE",]<=0), 0.5, 0.1))
par(mfrow=c(1,3))
barplot(table(indv@assays$integrated@data["RUNX2",]>0
              & indv@assays$integrated@data["BGLAP",]>0
              & indv@assays$integrated@data["CAPG",]<=0
              & indv@assays$integrated@data["MEPE",]<=0
              & indv@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Cells")
barplot(table(subset(indv,subset=species=="Human")@assays$integrated@data["RUNX2",]>0
              & subset(indv,subset=species=="Human")@assays$integrated@data["BGLAP",]>0
              & subset(indv,subset=species=="Human")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Human")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Human Cells")
barplot(table(subset(indv,subset=species=="Chimp")@assays$integrated@data["RUNX2",]>0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["BGLAP",]>0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["CAPG",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["MEPE",]<=0
              & subset(indv,subset=species=="Chimp")@assays$RNA@data["SOST",]<=0),
        ylim=c(0,25000))
title("All Chimp Cells")

#osteoblast-depositing osteoid / osteocyte-mineralizing osteoid (392 cells)
table(indv@assays$integrated@data["BGLAP",]>0
      & indv@assays$integrated@data["CAPG",]>0
      & indv@assays$integrated@data["MEPE",]>0
      & indv@assays$RNA@data["SOST",]<=0
      & indv@assays$integrated@data["RUNX2",]<=0)
plot(x=indv@assays$integrated@data["BGLAP",], y=indv@assays$integrated@data["CAPG",], xlab="BGLAP", ylab="CAPG", pch=20, main="OB-deposit/OC-mineral osteoid",
     col=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["MEPE",], xlab="RUNX2", ylab="MEPE", pch=20, main="OB-deposit/OC-mineral osteoid",
     col=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0), "red", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["RUNX2",], y=indv@assays$integrated@data["MEPE",], xlab="RUNX2", ylab="MEPE", pch=20, main="OB-deposit/OC-mineral osteoid",
     col=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0 &
                   indv@assays$integrated@data["RUNX2",]>0), "blue", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0 &
                   indv@assays$integrated@data["RUNX2",]>0), 0.5, 0.1))
plot(x=indv@assays$integrated@data["BGLAP",], y=indv@assays$integrated@data["CAPG",], xlab="BGLAP", ylab="CAPG", pch=20, main="OB-deposit/OC-mineral osteoid",
     col=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0 &
                   indv@assays$integrated@data["RUNX2",]<=0), "black", "darkgrey"),
     cex=ifelse((indv@assays$integrated@data["BGLAP",]>0 & indv@assays$integrated@data["CAPG",]>0 & indv@assays$integrated@data["MEPE",]>0 &
                   indv@assays$integrated@data["RUNX2",]<=0), 0.5, 0.1))
par(mfrow=c(1,3))
barplot(table(indv@assays$integrated@data["BGLAP",]>0
              & indv@assays$integrated@data["CAPG",]>0
              & indv@assays$integrated@data["MEPE",]>0
              & indv@assays$RNA@data["SOST",]<=0
              & indv@assays$integrated@data["RUNX2",]<=0),
        ylim=c(0,25000))
title("All Cells")
barplot(table(subset(indv,subset=species=="Human")@assays$integrated@data["BGLAP",]>0
              & subset(indv,subset=species=="Human")@assays$integrated@data["CAPG",]>0
              & subset(indv,subset=species=="Human")@assays$integrated@data["MEPE",]>0
              & subset(indv,subset=species=="Human")@assays$RNA@data["SOST",]<=0
              & subset(indv,subset=species=="Human")@assays$integrated@data["RUNX2",]<=0),
        ylim=c(0,25000))
title("All Human Cells")
barplot(table(subset(indv,subset=species=="Chimp")@assays$integrated@data["BGLAP",]>0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["CAPG",]>0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["MEPE",]>0
              & subset(indv,subset=species=="Chimp")@assays$RNA@data["SOST",]<=0
              & subset(indv,subset=species=="Chimp")@assays$integrated@data["RUNX2",]<=0),
        ylim=c(0,25000))
title("All Chimp Cells")

#mature osteocyte (1 cell)
table(indv@assays$RNA@data["SOST",]>0)



ost.genes <- as.data.frame(t(as.matrix(indv.o@assays$integrated@data[genes[9:12],])))
#Check cell assignment numbers for different thresholds
chk.thres <- function(data, runx2, bglap, capg, mepe) {
  assign <- ""
  if(data[1] > runx2) { assign <- paste(assign,"RUNX2") }
  if(data[2] > bglap) { assign <- paste(assign,"BGLAP") }
  if(data[3] > capg) { assign <- paste(assign,"CAPG") }
  if(data[4] > mepe) { assign <- paste(assign,"MEPE") }
  return(assign)
}
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0, bglap=0, capg=0, mepe=0) #threshold = 0
ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.1, bglap=0.1, capg=0.1, mepe=0.1) #threshold = 0.1
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.3, bglap=0.3, capg=0.3, mepe=0.3) #threshold = 0.3
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.5, bglap=0.5, capg=0.5, mepe=0.5) #threshold = 0.5
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.0250, bglap=0.0200, capg=0.0450, mepe=0.0000) #threshold = lenient
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.1000, bglap=0.0450, capg=0.1050, mepe=0.0180) #threshold = mild
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.3000, bglap=0.2100, capg=0.3250, mepe=0.0400) #threshold = strict
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.0454, bglap=0.0406, capg=0.5314, mepe=-0.0010) #threshold = median (total)
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.1679, bglap=0.0770, capg=0.6470, mepe=-0.0036) #threshold = mean (total)
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.5337, bglap=0.3044, capg=1.2465, mepe=0.0269) #threshold = mean+sd (total)
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.0391, bglap=0.0396, capg=0.3672, mepe=-0.0010) #threshold = median (total)
#ost.genes$assign <- apply(ost.genes, 1, chk.thres, runx2=0.1103, bglap=0.0577, capg=0.3805, mepe=-0.0038) #threshold = mean (total)

ost.genes %>% group_by(assign) %>% summarise(total=n()) %>% dplyr::mutate(freq=total/sum(total))
#ost.genes %>% group_by(assign) %>% tally()
#ost.genes %>% group_by(assign)



indv <- AddMetaData(indv, ost.genes$assign, col.name="Assignment")
DimPlot(indv, group.by="species", split.by="Assignment", reduction="umap", ncol=8, cols=c("grey","black")) + labs(x="UMAP1",y="UMAP2")
FeaturePlot(indv, split.by="Assignment", reduction="umap", ncol=8, features="Alizarin") + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((DimPlot(indv, group.by="Assignment", reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (DimPlot(indv, group.by="Assignment", reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (DimPlot(indv, group.by="Assignment", reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (DimPlot(indv, group.by="Assignment", reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8"))), ncol=4)
DimPlot(subset(indv,subset=Assignment==" RUNX2"), group.by="Sample", reduction="umap") + labs(x="UMAP1",y="UMAP2",title="preosteoblast")
DimPlot(subset(indv,subset=Assignment==" RUNX2 BGLAP"), group.by="Sample", reduction="umap") + labs(x="UMAP1",y="UMAP2",title="osteoblast")
DimPlot(subset(indv,subset=Assignment==" BGLAP CAPG"), group.by="Sample", reduction="umap") + labs(x="UMAP1",y="UMAP2", title="embedding osteoblast")
DimPlot(subset(indv,subset=Assignment==" BGLAP CAPG MEPE"), group.by="Sample", reduction="umap") + labs(x="UMAP1",y="UMAP2", title="mineralizing/osteoid osteocyte")
DimPlot(subset(indv,subset=Assignment==" CAPG MEPE"), group.by="Sample", reduction="umap") + labs(x="UMAP1",y="UMAP2", title="mature osteocyte")

CombinePlots(plots=list((FeaturePlot(subset(indv,subset=Assignment==""), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" BGLAP"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" BGLAP CAPG"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" BGLAP CAPG MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" BGLAP MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" CAPG"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" CAPG MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 BGLAP"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 BGLAP CAPG"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 BGLAP CAPG MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 BGLAP MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 CAPG"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 CAPG MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP")),
                        (FeaturePlot(subset(indv,subset=Assignment==" RUNX2 MEPE"), features="Alizarin", split.by="Assignment",
                                     reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP"))),
                        ncol=8)

#Make upsetr plotS
ost.genes[ost.genes$RUNX2 > 0, "RUNX2"] <- TRUE
ost.genes[ost.genes$RUNX2 <= 0, "RUNX2"] <- FALSE
ost.genes[ost.genes$BGLAP > 0, "BGLAP"] <- TRUE
ost.genes[ost.genes$BGLAP <= 0, "BGLAP"] <- FALSE
ost.genes[ost.genes$CAPG > 0, "CAPG"] <- TRUE
ost.genes[ost.genes$CAPG <= 0, "CAPG"] <- FALSE
ost.genes[ost.genes$MEPE > 0, "MEPE"] <- TRUE
ost.genes[ost.genes$MEPE <= 0, "MEPE"] <- FALSE

#test <- ost.genes
#test$cell <- rownames(test)
#test <- melt(test, id.vars="cell", measure.vars=c("RUNX2","BGLAP","CAPG","MEPE"))
#class(test$variable) <- "list"
#class(test$value) <- "list"
#library(ggupset)
#ggplot(test, aes(x=value)) +
#  geom_bar() +
#  scale_x_upset()

```

```{r DE temporary code}

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indv <- CellCycleScoring(indv, s.features=s.genes, g2m.features=g2m.genes, set.ident=TRUE)

#TEMPORARY - Add new cell assignments to merged data
data <- readRDS(file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.merge.l.rds"))
cells <- sapply(colnames(data), function(x) str_split(x,"_")[[1]], USE.NAMES=FALSE)
cell.name <- cells[1,]
cell.id <- cells[2,]
new.name <- c()
i=1
while(i <= length(cell.name)) {
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1.I") {x="C1.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1-r2.I") {x="C1r2.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H2C2.I") {x="C2.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H3C3.I") {x="C3.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H4C4.I") {x="C4.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H5C5.I") {x="C5.I"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H6C6.I") {x="C6.I"}
  
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1.M") {x="C1.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1-r2.M") {x="C1r2.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H2C2.M") {x="C2.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H3C3.M") {x="C3.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H4C4.M") {x="C4.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H5C5.M") {x="C5.M"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H6C6.M") {x="C6.M"}
  
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1.O") {x="C1.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H1C1-r2.O") {x="C1r2.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H2C2.O") {x="C2.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H3C3.O") {x="C3.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H4C4.O") {x="C4.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H5C5.O") {x="C5.O"}
  if(data@meta.data$species[i]=="Chimp" & cell.name[i]=="H6C6.O") {x="C6.O"}
  
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1.I") {x="H1.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1-r2.I") {x="H1r2.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H2C2.I") {x="H2.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H3C3.I") {x="H3.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H4C4.I") {x="H4.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H5C5.I") {x="H5.I"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H6C6.I") {x="H6.I"}
  
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1.M") {x="H1.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1-r2.M") {x="H1r2.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H2C2.M") {x="H2.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H3C3.M") {x="H3.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H4C4.M") {x="H4.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H5C5.M") {x="H5.M"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H6C6.M") {x="H6.M"}
  
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1.O") {x="H1.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H1C1-r2.O") {x="H1r2.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H2C2.O") {x="H2.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H3C3.O") {x="H3.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H4C4.O") {x="H4.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H5C5.O") {x="H5.O"}
  if(data@meta.data$species[i]=="Human" & cell.name[i]=="H6C6.O") {x="H6.O"}
  new.name <- c(new.name, x)
  i=i+1
}
cells.new <- paste(new.name,cell.id,sep="_")
data.rename <- RenameCells(data, new.names=cells.new)
data.rename <- data.rename[,colnames(data.rename)[which(colnames(data.rename) %in% colnames(indv))]]
new.meta <- 
  data.rename@meta.data %>% 
      tibble::rownames_to_column("cell.barcode") %>% 
  inner_join(
    indv@meta.data %>% 
      tibble::rownames_to_column("cell.barcode") %>% 
      dplyr::select(cell.barcode,seurat_clusters,AdHoc.Assign,OstAdHoc.Assign,GoM.Assign,Sample,SequencingBatch,Phase),
    by="cell.barcode"
  )
i=1
while(i <= length(colnames(data.rename))) {
  if(colnames(data.rename)[i] != new.meta$cell.barcode[i]) {print(i)}
  i=i+1
}
data.rename@meta.data$seurat_clusters <- new.meta$seurat_clusters
data.rename@meta.data$AdHoc.Assign <- new.meta$AdHoc.Assign
data.rename@meta.data$OstAdHoc.Assign <- new.meta$OstAdHoc.Assign
data.rename@meta.data$GoM.Assign <- new.meta$GoM.Assign
data.rename@meta.data$Sample <- new.meta$Sample
data.rename@meta.data$SequencingBatch <- new.meta$SequencingBatch
data.rename@meta.data$Phase <- new.meta$Phase
saveRDS(data.rename, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/data.filterMT.merge.l-TEMP.rds"))

#TEMPORARY - Test for DE Genes
data <- readRDS(paste0(dir2,"/data.filterMT.merge.l-TEMP.rds"))
Idents(data) <- factor(data@meta.data$species, levels=c("Chimp","Human"))
trial1  <- FindMarkers(subset(data,subset=Cell.Type=="Osteoblast"),
                              assay="RNA", slot="data",
                              ident.1="Human", ident.2="Chimp",
                              logfc.threshold=0.25, min.pct=0.1,
                              test.use="MAST", latent.vars=c("Sample","nCount_RNA"))
trial1$gene <- rownames(trial1)
fracRibo(trial1)
p <- goCCenrich(data,trial1)
p[[1]]
p[[2]]

#TEMPORARY - Test of latent.vars regression
#diff-exp-TEST.sh and diff-exp-TEST.R on cluster
#latent.vars=NULL
ips0 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ips0.rds"))
msc0 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-msc0.rds"))
ost0 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ost0.rds"))
#latent.vars="Sample"
ips1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ips1.rds"))
msc1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-msc1.rds"))
ost1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ost1.rds"))
#latent.vars="nCount_RNA"
ips2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ips2.rds"))
msc2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-msc2.rds"))
ost2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ost2.rds"))
#latent.vars=c("Sample,"nCount_RNA")
ips3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ips3.rds"))
msc3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-msc3.rds"))
ost3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-ost3.rds"))
#                  no.0 no.1 no.2 no.3
dim(ips3)  #genes:  968  968  968  968
dim(msc3)  #genes: 1795 1795 1795 2795
dim(ost3)  #genes: 2557 2557 2557 2557
#same number of genes were tested in all cases
#                                      no.0 no.1 no.2 no.3
table(ips3$p_val_adj<0.05)  #DE genes:  968  929  949  862
table(msc3$p_val_adj<0.05)  #DE genes: 1795 1557 1791 1534
table(ost3$p_val_adj<0.05)  #DE genes: 2557 1677 2105 1618
head(ips1, n=5)
head(msc1, n=5)
head(ost1, n=5)
go.ips <- funcEnrich(indv, ips3)
go.msc <- funcEnrich(indv, msc3)
go.ost <- funcEnrich(indv, ost3)
#                          no.0 no.1 no.2 no.3
dim(go.ips) #GO functions:   99  106   90   93
dim(go.msc) #GO functions:  259  235  259  246
dim(go.ost) #GO functions:  283  234  256  264
head(go.ips, n=5)
head(go.msc, n=5)
head(go.ost, n=5)

#RANDOM
#https://stats.stackexchange.com/questions/70558/diagnostic-plots-for-count-regression
#test and graph original count data - which distribution best fits count data
library(MASS)
library(vcd)
fit <- goodfit(dge$counts[1,], type="nbinomial", method="ML")
summary(fit) 
rootogram(fit)
Ord_plot(dge$counts[1,])
distplot(dge$counts[1,], type="nbinom")
#inspect usual goodness-of-fit measures
mod1 <- glm(dge$counts[1,]~dge$samples$species, family="poisson")
summary(mod1)
#check for over/underdispersion
library(AER)
deviance(mod1)/mod1$df.residual
dispersiontest(mod1)
#check for influential and leverage points
library(car)
influencePlot(mod1)
#check for zero inflation
library(pscl)
mod2 <- zeroinfl(dge$counts[1,]~dge$samples$species, dist="poisson")
AIC(mod1, mod2)
#plot the residuals (y-axis) vs. the log predicted values (x-axis)
res <- residuals(mod1, type="deviance")
plot(log(predict(mod1)), res)
abline(h=0, lty=2)
qqnorm(res)
qqline(res)
#plot a half normal probability plot (ordered absolute residuals vs. expected normal values)
library(faraway)
halfnorm(residuals(mod1))
#diagnostic plots
plot(dge$counts[1,]~dge$samples$species) 
prs  <- predict(mod1, type="response", se.fit=TRUE)
pris <- data.frame("pest"=prs[[1]], "lwr"=prs[[1]]-prs[[2]], "upr"=prs[[1]]+prs[[2]])
points(pris$pest ~ dge$samples$species, col="red")
points(pris$lwr  ~ dge$samples$species, col="pink", pch=19)
points(pris$upr  ~ dge$samples$species, col="pink", pch=19)

```

```{r differential expression}

#Find differentially expressed genes between humans and chimpanzees
Idents(indv) <- factor(indv@meta.data$species, levels=c("Chimp","Human"))

data <- readRDS(paste0(dir2,"/data.filterMT.merge.l-TEMP.rds"))
Idents(data) <- factor(data@meta.data$species, levels=c("Chimp","Human"))

#Determine fraction of ribosomal genes
fracRibo <- function(markers) {
  genes <- unique(markers[markers$p_val_adj<0.05,'gene'])
  ribo.genes <- grep('^RP',genes,value=T)
  frac <- length(ribo.genes)/length(genes)
  return(frac)
}
removeRibo <- function(markers) {
  genes <- unique(markers[markers$p_val_adj<0.05,'gene'])
  ribo.genes <- grep('^RP',genes,value=T)
  return(markers <- markers[which(!(markers1$gene %in% ribo.genes)),])
}

#And assess functional enrichment of differentially expressed genes
goCCenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "CC",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
goBPenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "BP",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
funcEnrich <- function(data, markers) {
  genes <- structure(rep(1, length(rownames(data@assays$RNA@data))), names=rownames(data@assays$RNA@data))
  x=1
  while (x <= length(rownames(markers))) {
    genes[which(names(genes)==rownames(markers)[x])] <- markers$p_val_adj[x]
    x=x+1
  }
  go_data <- new("topGOdata", ontology="BP", allGenes=genes, geneSel=function(allScore){return(allScore < 0.01)},
                 nodeSize=5, annotationFun=annFUN.org, mapping="org.Hs.eg.db", ID="symbol")
  go_test <- runTest(go_data, algorithm="weight01", statistic="fisher")
  go_table <- GenTable(go_data, weightFisher=go_test, orderBy="weightFisher", ranksOf="weightFisher", topNodes=sum(score(go_test) < .01))
  return(go_table)
}

#notes: Seurat identifes positive and negative markers of a single cluster compared to all other cells by default
#notes: you can designate which groups of clusters vs. each other, or against all cells
#notes: FindAllMarkers automates this process (identifying positive and negative markers) for all clusters

#notes: min.pct argument requires a feature to be detected at a minimum percentage in either of the two groups of cells
#notes: thresh.test argument requires a feature to be differentially expressed (on average) by some amount between the two groups
#notes: max.cells.per.ident can be set to downsample each identity class to have no more cells than whatever this is set to (option to speed up computations)
#notes: test.use parameter can designate which of several tests for differential expression to use

#using negative binomial with collection batch, individual, and nUMI as covariates (return.thresh=1 so that all results will be shown)



#RUN FINDMARKERS ON CLUSTER: diff-exp.sh and diff-exp.R

#Find differentially expressed genes using differentiation stage at collection
compare.ips  <- FindMarkers(subset(indv,subset=Cell.Type=="iPSC"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.ips, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.ips.rds"))
compare.msc  <- FindMarkers(subset(indv,subset=Cell.Type=="MSC"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.msc, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.msc.rds"))
compare.ost  <- FindMarkers(subset(data,subset=Cell.Type=="Osteoblast"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="MAST", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.ost, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.ost.rds"))
rm(compare.ips,compare.msc,compare.ost)

compare.ips <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.ips.rds"))
compare.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.msc.rds"))
compare.ost <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.ost.rds"))
dim(compare.ips)
dim(compare.msc)
dim(compare.ost)
table(compare.ips$p_val_adj<0.05)
table(compare.msc$p_val_adj<0.05)
table(compare.ost$p_val_adj<0.05)
head(compare.ips, n=5)
head(compare.msc, n=5)
head(compare.ost, n=5)
compare.ips$gene <- rownames(compare.ips)
compare.msc$gene <- rownames(compare.msc)
compare.ost$gene <- rownames(compare.ost)
fracRibo(compare.ips)
fracRibo(compare.msc)
fracRibo(compare.ost)
p.ips <- goCCenrich(indv,compare.ips)
p.msc <- goCCenrich(indv,compare.msc)
p.ost <- goCCenrich(indv,compare.ost)
p.ips[[3]]
p.msc[[3]]
p.ost[[3]]
#go.ips <- funcEnrich(indv, compare.ips)
#go.msc <- funcEnrich(indv, compare.msc)
#go.ost <- funcEnrich(indv, compare.ost)
#dim(go.ips)
#dim(go.msc)
#dim(go.ost)
#head(go.ips, n=5)
#head(go.msc, n=5)
#head(go.ost, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


#Find differentially expressed genes using cluster definitions
compare.i  <- FindMarkers(subset(indv,subset=(seurat_clusters==1|seurat_clusters==4|seurat_clusters==6)),
                          assay="RNA", slot="data",
                          ident.1="Human", ident.2="Chimp",
                          logfc.threshold=0.25, min.pct=0.25,
                          test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.i, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.i.rds"))
compare.m  <- FindMarkers(subset(indv,subset=seurat_clusters==0),
                          assay="RNA", slot="data",
                          ident.1="Human", ident.2="Chimp",
                          logfc.threshold=0.25, min.pct=0.25,
                          test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.m, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.m.rds"))
compare.o  <- FindMarkers(subset(indv,subset=(seurat_clusters==2|seurat_clusters==3|seurat_clusters==5)),
                          assay="RNA", slot="data",
                          ident.1="Human", ident.2="Chimp",
                          logfc.threshold=0.25, min.pct=0.25,
                          test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.o, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.o.rds"))
rm(compare.i,compare.m,compare.o)
compare.i1  <- FindMarkers(subset(indv,subset=seurat_clusters==1),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.i1, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.i1.rds"))
compare.i2  <- FindMarkers(subset(indv,subset=seurat_clusters==4),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.i2, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.i2.rds"))
compare.i3  <- FindMarkers(subset(indv,subset=seurat_clusters==6),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.i3, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.i3.rds"))
rm(compare.i1,compare.i2,compare.i3)
compare.o1  <- FindMarkers(subset(indv,subset=seurat_clusters==2),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.o1, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.o1.rds"))
compare.o2  <- FindMarkers(subset(indv,subset=seurat_clusters==3),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.o2, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.o2.rds"))
compare.o3  <- FindMarkers(subset(indv,subset=seurat_clusters==5),
                           assay="RNA", slot="data",
                           ident.1="Human", ident.2="Chimp",
                           logfc.threshold=0.25, min.pct=0.25,
                           test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.o3, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.o3.rds"))
rm(compare.o1,compare.o2,compare.o3)

compare.i <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i.rds"))
compare.m <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.m.rds"))
compare.o <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o.rds"))
compare.i1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i1.rds"))
compare.i2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i2.rds"))
compare.i3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i3.rds"))
compare.o1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o1.rds"))
compare.o2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o2.rds"))
compare.o3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o3.rds"))
dim(compare.i)
dim(compare.m)
dim(compare.o) 
dim(compare.i1)
dim(compare.i2)
dim(compare.i3)
dim(compare.o1)
dim(compare.o2)
dim(compare.o3)
table(compare.i$p_val_adj<0.05)
table(compare.m$p_val_adj<0.05)
table(compare.o$p_val_adj<0.05)
table(compare.i1$p_val_adj<0.05)
table(compare.i2$p_val_adj<0.05)
table(compare.i3$p_val_adj<0.05)
table(compare.o1$p_val_adj<0.05)
table(compare.o2$p_val_adj<0.05)
table(compare.o3$p_val_adj<0.05)
head(compare.i, n=5)
head(compare.m, n=5)
head(compare.o, n=5)
head(compare.i1, n=5)
head(compare.i2, n=5)
head(compare.i3, n=5)
head(compare.o1, n=5)
head(compare.o2, n=5)
head(compare.o3, n=5)
compare.i$gene <- rownames(compare.i)
compare.m$gene <- rownames(compare.m)
compare.o$gene <- rownames(compare.o)
compare.i1$gene <- rownames(compare.i1)
compare.i2$gene <- rownames(compare.i2)
compare.i3$gene <- rownames(compare.i3)
compare.o1$gene <- rownames(compare.o1)
compare.o2$gene <- rownames(compare.o2)
compare.o3$gene <- rownames(compare.o3)
fracRibo(compare.i)
fracRibo(compare.m)
fracRibo(compare.o)
fracRibo(compare.i1)
fracRibo(compare.i2)
fracRibo(compare.i3)
fracRibo(compare.o1)
fracRibo(compare.o2)
fracRibo(compare.o3)
p.i <- goCCenrich(indv,compare.i)
p.m <- goCCenrich(indv,compare.m)
p.o <- goCCenrich(indv,compare.o)
p.i1 <- goCCenrich(indv,compare.i1)
p.i2 <- goCCenrich(indv,compare.i2)
p.i3 <- goCCenrich(indv,compare.i3)
p.o1 <- goCCenrich(indv,compare.o1)
p.o2 <- goCCenrich(indv,compare.o2)
p.o3 <- goCCenrich(indv,compare.o3)
p.i[[3]]
p.m[[3]]
p.o[[3]]
p.i1[[3]]
p.i2[[3]]
p.i3[[3]]
p.o1[[3]]
p.o2[[3]]
p.o3[[3]]
#go.i <- funcEnrich(indv, compare.i)
#go.m <- funcEnrich(indv, compare.m)
#go.o <- funcEnrich(indv, compare.o)
#go.i1 <- funcEnrich(indv, compare.i1)
#go.i2 <- funcEnrich(indv, compare.i2)
#go.i3 <- funcEnrich(indv, compare.i3)
#go.o1 <- funcEnrich(indv, compare.o1)
#go.o2 <- funcEnrich(indv, compare.o2)
#go.o3 <- funcEnrich(indv, compare.o3)
#dim(go.i)
#dim(go.m)
#dim(go.o)
#dim(go.i1)
#dim(go.i2)
#dim(go.i3)
#dim(go.o1)
#dim(go.o2)
#dim(go.o3)
#head(go.i, n=5)
#head(go.m, n=5)
#head(go.o, n=5)
#head(go.i1, n=5)
#head(go.i2, n=5)
#head(go.i3, n=5)
#head(go.o1, n=5)
#head(go.o2, n=5)
#head(go.o3, n=5)
compare.tot2 <- intersect(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                          intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                    rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #shared DE genes
compare.i.tot <- setdiff(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],c(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to i
compare.m.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to m
compare.o.tot <- setdiff(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.m)[which(compare.m$p_val_adj<0.05)])) #DE unique to o
compare.tot3 <- intersect(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],
                          intersect(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                    intersect(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                              intersect(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                        intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                  intersect(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                            intersect(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                      rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)]))))))) #shared DE
compare.i1.tot <- setdiff(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],c(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i1
compare.i2.tot <- setdiff(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i2
compare.i3.tot <- setdiff(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i3
compare.m1.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                 rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                 rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                 rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                 rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                 rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to m1
compare.o1.tot <- setdiff(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o1
compare.o2.tot <- setdiff(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o2
compare.o3.tot <- setdiff(rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)])) #DE unique to o3
length(compare.tot2)
length(compare.i.tot)
length(compare.m.tot)
length(compare.o.tot)
length(compare.tot3)
length(compare.i1.tot)
length(compare.i2.tot)
length(compare.i3.tot)
length(compare.m1.tot)
length(compare.o1.tot)
length(compare.o2.tot)
length(compare.o3.tot)
head(compare.i[which(rownames(compare.i)%in%compare.tot2),], n=5)
head(compare.i[which(rownames(compare.i)%in%compare.i.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m.tot),], n=5)
head(compare.o[which(rownames(compare.o)%in%compare.o.tot),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.tot3),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),], n=5)
head(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),], n=5)
head(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m1.tot),], n=5)
head(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),], n=5)
head(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),], n=5)
head(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),], n=5)
fracRibo(compare.i[which(rownames(compare.i)%in%compare.tot2),])
fracRibo(compare.i[which(rownames(compare.i)%in%compare.i.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m.tot),])
fracRibo(compare.o[which(rownames(compare.o)%in%compare.o.tot),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
fracRibo(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
fracRibo(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
fracRibo(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
fracRibo(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
fracRibo(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2 <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goCCenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goCCenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goCCenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goCCenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goCCenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goCCenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
p.tot2 <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goBPenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goBPenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goBPenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goBPenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goBPenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goBPenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.i$gene[which(compare.i$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o$gene[which(compare.o$p_val_adj<0.05)])),
                    Other=c((length(compare.i$gene[which(compare.i$p_val_adj<0.05)])-length(compare.i.tot)-length(compare.tot2)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m.tot)-length(compare.tot2)),
                            (length(compare.o$gene[which(compare.o$p_val_adj<0.05)])-length(compare.o.tot)-length(compare.tot2))),
                    Shared=c(rep(length(compare.tot2),3)),
                    Unique=c(length(compare.i.tot),
                             length(compare.m.tot),
                             length(compare.o.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))
DE.df <- data.frame(cell=(c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3")),
                    totalDEgenes=c(length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)]),
                                   length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)]),
                                   length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)]),
                                   length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)]),
                                   length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])),
                    Other=c((length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)])-length(compare.i1.tot)-length(compare.tot3)),
                            (length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)])-length(compare.i2.tot)-length(compare.tot3)),
                            (length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)])-length(compare.i3.tot)-length(compare.tot3)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m1.tot)-length(compare.tot3)),
                            (length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)])-length(compare.o1.tot)-length(compare.tot3)),
                            (length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)])-length(compare.o2.tot)-length(compare.tot3)),
                            (length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])-length(compare.o3.tot)-length(compare.tot3))),
                    Shared=c(rep(length(compare.tot3),7)),
                    Unique=c(length(compare.i1.tot),
                             length(compare.i2.tot),
                             length(compare.i3.tot),
                             length(compare.m1.tot),
                             length(compare.o1.tot),
                             length(compare.o2.tot),
                             length(compare.o3.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[8:28,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))



#Find differentially expressed genes using ad hoc definitions
compare.hoci  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="iPSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoci, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.hoci.rds"))
compare.hocm  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="MSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hocm, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.hocm.rds"))
compare.hoco  <- FindMarkers(subset(indv,subset=AdHoc.Assign=="Osteoblast"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.hoco, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.hoco.rds"))

compare.hoci <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hoci.rds"))
compare.hocm <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hocm.rds"))
compare.hoco <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hoco.rds"))
dim(compare.hoci)
dim(compare.hocm)
dim(compare.hoco)
table(compare.hoci$p_val_adj<0.05)
table(compare.hocm$p_val_adj<0.05)
table(compare.hoco$p_val_adj<0.05)
head(compare.hoci, n=5)
head(compare.hocm, n=5)
head(compare.hoco, n=5)
compare.hoci$gene <- rownames(compare.hoci)
compare.hocm$gene <- rownames(compare.hocm)
compare.hoco$gene <- rownames(compare.hoco)
fracRibo(compare.hoci)
fracRibo(compare.hocm)
fracRibo(compare.hoco)
p.hoci <- goCCenrich(indv,compare.hoci)
p.hocm <- goCCenrich(indv,compare.hocm)
p.hoco <- goCCenrich(indv,compare.hoco)
p.hoci[[3]]
p.hocm[[3]]
p.hoco[[3]]
#go.hoci <- funcEnrich(indv, compare.hoci)
#go.hocm <- funcEnrich(indv, compare.hocm)
#go.hoco <- funcEnrich(indv, compare.hoco)
#dim(go.hoci)
#dim(go.hocm)
#dim(go.hoco)
#head(go.hoci, n=5)
#head(go.hocm, n=5)
#head(go.hoco, n=5)
compare.tot5 <- intersect(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],intersect(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                                rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)]))
compare.hoci.tot <- setdiff(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],c(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hoci
compare.hocm.tot <- setdiff(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hocm
compare.hoco.tot <- setdiff(rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)])) #DE hoco
length(compare.tot5)
length(compare.hoci.tot)
length(compare.hocm.tot)
length(compare.hoco.tot)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),], n=5)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),], n=5)
head(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),], n=5)
head(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),], n=5)
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
fracRibo(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
fracRibo(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5 <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goCCenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goCCenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
p.tot5 <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goBPenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goBPenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)]),
                                   length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)]),
                                   length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])),
                    Other=c((length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)])-length(compare.hoci.tot)-length(compare.tot5)),
                            (length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)])-length(compare.hocm.tot)-length(compare.tot5)),
                            (length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])-length(compare.hoco.tot)-length(compare.tot5))),
                    Shared=c(rep(length(compare.tot5),3)),
                    Unique=c(length(compare.hoci.tot),
                             length(compare.hocm.tot),
                             length(compare.hoco.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))



#Find differentially expressed genes using osteogenic ad hoc definitions
compare.pre  <- FindMarkers(subset(indv,subset=OstAdHoc.Assign=="preosteoblast"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.pre, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.pre.rds"))
compare.obl  <- FindMarkers(subset(indv,subset=OstAdHoc.Assign=="osteoblast"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.obl, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.obl.rds"))
compare.emb  <- FindMarkers(subset(indv,subset=OstAdHoc.Assign=="embedding osteoblast"),
                            assay="RNA", slot="data",
                            ident.1="Human", ident.2="Chimp",
                            logfc.threshold=0.25, min.pct=0.25,
                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.emb, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.emb.rds"))
#compare.min  <- FindMarkers(subset(indv,subset=OstAdHoc.Assign=="mineralizing osteocyte"),
#                            assay="RNA", slot="data",
#                            ident.1="Human", ident.2="Chimp",
#                            logfc.threshold=0.25, min.pct=0.25,
#                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
#saveRDS(compare.min, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.min.rds"))
#compare.mat  <- FindMarkers(subset(indv,subset=OstAdHoc.Assign=="mature osteocyte"),
#                            assay="RNA", slot="data",
#                            ident.1="Human", ident.2="Chimp",
#                            logfc.threshold=0.25, min.pct=0.25,
#                            test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
#saveRDS(compare.mat, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.mat.rds"))
rm(compare.pre,compare.ost,compare.emb)
#rm(compare.pre,compare.ost,compare.emb,compare.min,compare.mat)

compare.pre <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.pre.rds"))
compare.obl <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.obl.rds"))
compare.emb <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.emb.rds"))
dim(compare.pre)
dim(compare.obl)
dim(compare.emb)
table(compare.pre$p_val_adj<0.05)
table(compare.obl$p_val_adj<0.05)
table(compare.emb$p_val_adj<0.05)
head(compare.pre, n=5)
head(compare.obl, n=5)
head(compare.emb, n=5)
compare.pre$gene <- rownames(compare.pre)
compare.obl$gene <- rownames(compare.obl)
compare.emb$gene <- rownames(compare.emb)
fracRibo(compare.pre)
fracRibo(compare.obl)
fracRibo(compare.emb)
p.pre <- goCCenrich(indv,compare.pre)
p.obl <- goCCenrich(indv,compare.obl)
p.emb <- goCCenrich(indv,compare.emb)
p.pre[[3]]
p.obl[[3]]
p.emb[[3]]
#go.pre <- funcEnrich(indv, compare.pre)
#go.obl <- funcEnrich(indv, compare.obl)
#go.emb <- funcEnrich(indv, compare.emb)
#dim(go.pre)
#dim(go.obl)
#dim(go.emb)
#head(go.pre, n=5)
#head(go.obl, n=5)
#head(go.emb, n=5)
compare.tot6 <- intersect(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                          intersect(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                    rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #shared DE genes
compare.pre.tot <- setdiff(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)], c(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],
                                                                                       rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE pre
compare.obl.tot <- setdiff(rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)])) #DE obl
compare.emb.tot <- setdiff(rownames(compare.emb)[which(compare.emb$p_val_adj<0.05)],c(rownames(compare.pre)[which(compare.pre$p_val_adj<0.05)],
                                                                                      rownames(compare.obl)[which(compare.obl$p_val_adj<0.05)])) #DE emb
length(compare.tot6)
length(compare.pre.tot)
length(compare.obl.tot)
length(compare.emb.tot)
head(compare.pre[which(rownames(compare.pre)%in%compare.tot6),], n=5)
head(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),], n=5)
head(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),], n=5)
head(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),], n=5)
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
fracRibo(compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
fracRibo(compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
fracRibo(compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6 <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goCCenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goCCenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goCCenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
p.tot6 <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.tot6),])
p.pre.tot <- goBPenrich(indv,compare.pre[which(rownames(compare.pre)%in%compare.pre.tot),])
p.obl.tot <- goBPenrich(indv,compare.obl[which(rownames(compare.obl)%in%compare.obl.tot),])
p.emb.tot <- goBPenrich(indv,compare.emb[which(rownames(compare.emb)%in%compare.emb.tot),])
p.tot6[[3]]
p.pre.tot[[3]]
p.obl.tot[[3]]
p.emb.tot[[3]]
DE.df <- data.frame(cell=(c("preosteoblast","osteoblast","embedding osteoblast")),
                    totalDEgenes=c(length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)]),
                                   length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)]),
                                   length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])),
                    Other=c((length(compare.pre$gene[which(compare.pre$p_val_adj<0.05)])-length(compare.pre.tot)-length(compare.tot6)),
                            (length(compare.obl$gene[which(compare.obl$p_val_adj<0.05)])-length(compare.obl.tot)-length(compare.tot6)),
                            (length(compare.emb$gene[which(compare.emb$p_val_adj<0.05)])-length(compare.emb.tot)-length(compare.tot6))),
                    Shared=c(rep(length(compare.tot6),3)),
                    Unique=c(length(compare.pre.tot),
                             length(compare.obl.tot),
                             length(compare.emb.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("preosteoblast","osteoblast","embedding osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


#Find differentially expressed genes using grade of membership definitions
compare.gomi  <- FindMarkers(subset(indv,subset=GoM.Assign=="iPSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.gomi, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.gomi.rds"))
compare.gomm  <- FindMarkers(subset(indv,subset=GoM.Assign=="MSC"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.gomm, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.gomm.rds"))
compare.gomo  <- FindMarkers(subset(indv,subset=GoM.Assign=="Osteoblast"),
                             assay="RNA", slot="data",
                             ident.1="Human", ident.2="Chimp",
                             logfc.threshold=0.25, min.pct=0.25,
                             test.use="negbinom", latent.vars=c("Sample","nCount_RNA"))
saveRDS(compare.gomo, file=paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de-compare.gomo.rds"))
rm(compare.gomi,compare.gomm,compare.gomo)

compare.gomi <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.gomi.rds"))
compare.gomm <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.gomm.rds"))
compare.gomo <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.gomo.rds"))
dim(compare.gomi)
dim(compare.gomm)
dim(compare.gomo)
table(compare.gomi$p_val_adj<0.05)
table(compare.gomm$p_val_adj<0.05)
table(compare.gomo$p_val_adj<0.05)
head(compare.gomi, n=5)
head(compare.gomm, n=5)
head(compare.gomo, n=5)
compare.gomi$gene <- rownames(compare.gomi)
compare.gomm$gene <- rownames(compare.gomm)
compare.gomo$gene <- rownames(compare.gomo)
fracRibo(compare.gomi)
fracRibo(compare.gomm)
fracRibo(compare.gomo)
p.gomi <- goCCenrich(indv,compare.gomi)
p.gomm <- goCCenrich(indv,compare.gomm)
p.gomo <- goCCenrich(indv,compare.gomo)
p.gomi[[3]]
p.gomm[[3]]
p.gomo[[3]]
#go.gomi <- funcEnrich(indv, compare.gomi)
#go.gomm <- funcEnrich(indv, compare.gomm)
#go.gomo <- funcEnrich(indv, compare.gomo)
#dim(go.gomi)
#dim(go.gomm)
#dim(go.gomo)
#head(go.gomi, n=5)
#head(go.gomm, n=5)
#head(go.gomo, n=5)
compare.tot1 <- intersect(rownames(compare.ips),intersect(rownames(compare.msc),rownames(compare.ost))) #shared DE genes
compare.ips.tot <- setdiff(rownames(compare.ips),c(rownames(compare.msc),rownames(compare.ost))) #DE genes unique to ips
compare.msc.tot <- setdiff(rownames(compare.msc),c(rownames(compare.ips),rownames(compare.ost))) #DE genes unique to msc
compare.ost.tot <- setdiff(rownames(compare.ost),c(rownames(compare.ips),rownames(compare.msc))) #DE genes unique to ost
length(compare.tot1)
length(compare.ips.tot)
length(compare.msc.tot)
length(compare.ost.tot)
head(compare.ips[which(rownames(compare.ips)%in%compare.tot1),], n=5)
head(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),], n=5)
head(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),], n=5)
head(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),], n=5)
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
fracRibo(compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
fracRibo(compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
fracRibo(compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1 <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goCCenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goCCenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goCCenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
p.tot1 <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.tot1),])
p.ips.tot <- goBPenrich(indv,compare.ips[which(rownames(compare.ips)%in%compare.ips.tot),])
p.msc.tot <- goBPenrich(indv,compare.msc[which(rownames(compare.msc)%in%compare.msc.tot),])
p.ost.tot <- goBPenrich(indv,compare.ost[which(rownames(compare.ost)%in%compare.ost.tot),])
p.tot1[[3]]
p.ips.tot[[3]]
p.msc.tot[[3]]
p.ost.tot[[3]]
DE.df <- data.frame(cell=(c("Time 0","Time 1","Time 2")),
                    totalDEgenes=c(length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)]),
                                   length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)]),
                                   length(compare.ost$gene[which(compare.ost$p_val_adj<0.05)])),
                    Other=c((length(compare.ips$gene[which(compare.ips$p_val_adj<0.05)])-length(compare.ips.tot)-length(compare.tot1)),
                            (length(compare.msc$gene[which(compare.msc$p_val_adj<0.05)])-length(compare.msc.tot)-length(compare.tot1)),
                            (length(compare.ost$gene[which(compare.o$p_val_adj<0.05)])-length(compare.ost.tot)-length(compare.tot1))),
                    Shared=c(rep(length(compare.tot1),3)),
                    Unique=c(length(compare.ips.tot),
                             length(compare.msc.tot),
                             length(compare.ost.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("Time 0","Time 1","Time 2"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))






```

```{r differential expression TESTS}

#Load libraries
library(Seurat)
library(clusterProfiler)

#Define main directory
dir <- '/project2/gilad/ghousman/skeletal-human-chimp/'
dir2 <- paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full")

#Load data
indv <- readRDS(paste0(dir2,"/data.filterMT.regMT.indv.log.10k.int.rds" ))
indv@meta.data$Collection <- as.factor(indv@meta.data$Collection)
data <- readRDS(paste0(dir2,"/data.filterMT.merge.l-TEMP.rds"))
data@meta.data$Collection <- as.factor(data@meta.data$Collection)

#Find differentially expressed genes between humans and chimpanzees
Idents(indv) <- factor(indv@meta.data$species, levels=c("Chimp","Human"))
Idents(data) <- factor(data@meta.data$species, levels=c("Chimp","Human"))

#Isolate ribosomal genes
genes <- rownames(data@assays$RNA@counts)
genes.ribo <- grep('^RP',genes,value=T)
genes.no.ribo <- genes[which(!(genes %in% genes.ribo))]

#notes:    decided on method comparisons after consulting Sonesson & Robinson (2018) Nat Methods / Wang et al (2019) BMC Bioinf
#notes:    EXCLUDE SEQUENCING BATCH FOR NOW - MIGHT BE CONFOUNDED WITH COLLECTION
#          > dge$samples$SequencingBatch <- as.factor(as.numeric(dge$samples$SequencingBatch))
#notes:    USING ~0 INTERCEPT WITH CONTRASTS PRODUCES SAME RESULTS AS ~1 INTERCEPT WITH TEST ON COEFF
#          > design <- model.matrix(~0+species+Collection+nCount_RNA+Phase, data=dge$samples)
#          > contr <- makeContrasts(speciesChimp-speciesHuman, levels=design)
#          > qlf <- glmQLFTest(fit, contrast=contr)
#          > tr <- glmTreat(fit, contrast=contr, lfc=log2(1.2))
#QUESTION: SHOULD I INCLUDE nCount_RNA BECAUSE LIBRARY SIZE IS ALREADY ACCOUNTED FOR IN edgeR ANALYSES?
#          KEEPING FOR NOW BECAUSE IT REDUCES COMMON DISPERSION ESTIMATE

##Wilcox - DO NOT USE (NONPARAMETRIC - NOT ENOUGH POWER)
#compare.obl  <- FindMarkers(subset(data,subset=OstAdHoc.Assign=="osteoblast"),
#                            assay="RNA",
#                            slot="data",
#                            ident.1="Human",
#                            ident.2="Chimp",
#                            features=genes.no.ribo,
#                            logfc.threshold=0.25,
#                            min.pct=0.25,
#                            min.cells.feature=3,
#                            return.thresh=1,
#                            test.use="wilcox",
#                            latent.vars=NULL)
#d <- density(compare.obl$avg_logFC)
#plot(d)
#summary(compare.obl$p_val_adj)
#hist(compare.obl$p_val_adj)
#dim(compare.obl)
#table(compare.obl$p_val_adj<0.05)
#head(compare.obl, n=5)
#(counts): 678/2040
#(data): 482/1465

#Negative Binomial GLM (ignores return.thresh argument)
compare.obl  <- FindMarkers(subset(data,subset=OstAdHoc.Assign=="osteoblast"),
                            assay="RNA",
                            slot="counts",
                            ident.1="Human",
                            ident.2="Chimp",
                            features=genes.no.ribo,
                            logfc.threshold=0.25,
                            min.pct=0.25,
                            min.cells.feature=3,
                            test.use="negbinom",
                            latent.vars=c("Collection","nCount_RNA","Phase"))
d <- density(compare.obl$avg_logFC)
plot(d)
summary(compare.obl$p_val_adj)
hist(compare.obl$p_val_adj)
dim(compare.obl)
table(compare.obl$p_val_adj<0.05)
head(compare.obl, n=5)
#latent.vars=NULL (counts): 1131/2040
#latent.vars="Sample"(counts): 2/2000 -------------- (DO NOT INCLUDE - CONFOUNDED WITH SPECIES)
#latent.vars="DateCollected"(counts): 993/2037 ----- (DO NOT INCLUDE - BATCHES AND COLLECTION DATES ARE NOT 1:1)
#latent.vars="Collection"(counts): 832/2036
#latent.vars="SequencingBatch"(counts): 888/2039 --- (DO NOT INCLUDE - CONFOUNDED WITH COLLECTION)
#latent.vars="nCount_RNA"(counts & data): 493/2040
#latent.vars="Phase"(counts): 1173/2040
#latent.vars=c("Sample","nCount_RNA")(counts): 2/1988
#latent.vars=c("DateCollection","nCount_RNA")(counts): 449/2037
#latent.vars=c("Collection","nCount_RNA")(counts): 434/2036
#latent.vars=c("DateCollection","SequencingBatch","nCount_RNA")(counts): 449/2037
#latent.vars=c("Collection","SequencingBatch","nCount_RNA")(counts): 434/2036
#latent.vars=c("DateCollection","SequencingBatch","nCount_RNA","Phase")(counts): 448/2037
#latent.vars=c("Collection","SequencingBatch","nCount_RNA","Phase")(counts): 436/2036
#latent.vars=c("Collection","nCount_RNA","Phase")(counts): 435/2036 --------------------- (USE THIS MODEL FOR NOW)

#Poisson GLM
compare.obl  <- FindMarkers(subset(data,subset=OstAdHoc.Assign=="osteoblast"),
                            assay="RNA",
                            slot="counts",
                            ident.1="Human",
                            ident.2="Chimp",
                            features=genes.no.ribo,
                            logfc.threshold=0.25,
                            min.pct=0.25,
                            min.cells.feature=3,
                            test.use="poisson",
                            latent.vars=c("Collection","nCount_RNA","Phase"))
d <- density(compare.obl$avg_logFC)
plot(d)
summary(compare.obl$p_val_adj)
hist(compare.obl$p_val_adj)
dim(compare.obl)
table(compare.obl$p_val_adj<0.05)
head(compare.obl, n=5)
#latent.vars=c("Collection","nCount_RNA","Phase")(counts): 598/2040

#MAST (generalized linear hurdel model - does not support categorical SO cannot currently include "Collection" or "SequencingBatch" in model)
compare.obl  <- FindMarkers(subset(data,subset=OstAdHoc.Assign=="osteoblast"),
                            assay="RNA",
                            slot="counts",
                            ident.1="Human",
                            ident.2="Chimp",
                            features=genes.no.ribo,
                            logfc.threshold=0.25,
                            min.pct=0.25,
                            min.cells.feature=3,
                            test.use="MAST",
                            latent.vars="nCount_RNA")
d <- density(compare.obl$avg_logFC)
plot(d)
summary(compare.obl$p_val_adj)
hist(compare.obl$p_val_adj)
dim(compare.obl)
table(compare.obl$p_val_adj<0.05)
head(compare.obl, n=5)
#latent.vars="nCount_RNA"(counts): 596/2040

#edgeRQLF (Han et al., 2015)
#uses negative binomial model-based method to determine DE genes
#model: weighted trimmed mean of the log expression ratios used to normalize the sequencing depth and gene length between the samples
#       expression data fit to a negative binomial model (mean  and variance  have relationship of   =  + 2, and  is the dispersion factor)
#       dispersion factor estimated by combining:
#          (a) common dispersion across all the genes (estimated by a likelihood function), and
#          (b) gene-specific dispersion (estimated by the empirical Bayes method)
#de: exact test with FDR control is used to determine DE genes
#https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_edgeRQLFDetRate.R
#https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf
#edgeRQLFDetRate
library(edgeR)
library(scran)
library(SingleCellExperiment)
runEDGER <- function(dataSub, genes.no.ribo) {
  
  library(edgeR)

  #count matrix
  counts <- as.matrix(GetAssayData(dataSub, assay="RNA", slot="counts"))
  counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
  
  #metadata
  metadata <- dataSub@meta.data[,c("species","Collection","nCount_RNA","Phase")]
  metadata <- metadata[colnames(counts),]
  
  #make edgeR object
  dge <- DGEList(counts)
  meta_dge <- dge$samples[,c("lib.size","norm.factors")]
  meta_dge <- cbind(meta_dge, metadata)
  dge$samples <- meta_dge
  
  #filter genes
  keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
  table(keep)
  dge <- dge[keep, , keep.lib.sizes=FALSE]
  
  #normalize data
  dge <- calcNormFactors(dge, method="TMM")
  summary(dge$samples$norm.factors)
  
  #calculate cellular detection rate
  cdr <- scale(colMeans(dge$count > 0))
  
  #design matrix
  dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
  dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
  dge$samples$species <- as.factor(as.character(dge$samples$species))
  design <- model.matrix(~Collection+cdr+nCount_RNA+Phase+species, data=dge$samples)
  dge <- estimateDisp(dge, design=design)
  print(paste0("common dispersion: ",dge$common.dispersion))

  #QLF model fit (quasi-likelihood ratio test) - reflects uncertainty in estimating dispersion for each gene
  fit <- glmQLFit(dge, design=design, robust=TRUE)
  head(fit$coefficients)

  #differential expression
  qlf <- glmQLFTest(fit, coef=dim(design)[2]) #detect DE genes between species
  tr <- glmTreat(fit, coef=dim(design)[2], lfc=log2(1.2)) #narrow list of DE genes to those with at least a fold-change of 1.2
  
  #assess output
  tt <- topTags(qlf, n=Inf, adjust.method="BH", p.value=1)
  print("All DE Genes")
  print(dim(tt$table))
  print(table(tt$table$FDR<0.05))
  print(summary(decideTests(qlf, adjust.method="BH", p.value=0.05)))

  tt <- topTags(tr, n=Inf, adjust.method="BH", sort.by="PValue", p.value=1)
  print("DE Genes with log2(1.2) Fold Change")
  print(dim(tt$table))
  print(table(tt$table$FDR<0.05))
  print(summary(decideTests(tr, adjust.method="BH", p.value=0.05)))

  return(tt)

}
edgr <- runEDGER(subset(data,subset=OstAdHoc.Assign=="osteoblast"), genes.no.ribo)
#count matrix
obl <- subset(data,subset=OstAdHoc.Assign=="osteoblast")
counts <- as.matrix(GetAssayData(obl, assay="RNA", slot="counts"))
counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
#metadata
metadata <- obl@meta.data[,c("species","Collection","nCount_RNA","Phase")]
metadata <- metadata[colnames(counts),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
#filter genes
keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
table(keep)
dge <- dge[keep, , keep.lib.sizes=FALSE]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
summary(dge$samples$norm.factors)
#calculate cellular detection rate
cdr <- scale(colMeans(dge$count > 0))
#design matrix
dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
dge$samples$species <- as.factor(as.character(dge$samples$species))
design <- model.matrix(~Collection+cdr+nCount_RNA+Phase+species, data=dge$samples)
dge <- estimateDisp(dge, design=design)
dge$common.dispersion
plotBCV(dge)
#QLF model fit (quasi-likelihood ratio test) - reflects uncertainty in estimating dispersion for each gene
fit <- glmQLFit(dge, design=design, robust=TRUE)
head(fit$coefficients)
plotQLDisp(fit)
g <- gof(fit, pcutoff=0.05, adjust="holm", plot=TRUE) #deviance residuals don't have to be normally distributed for the model to be valid
hist(g$gof.pvalue)
hist(g$gof.statistics)
#differential expression
qlf <- glmQLFTest(fit, coef=dim(design)[2]) #detect DE genes between species
tr <- glmTreat(fit, coef=dim(design)[2], lfc=log2(1.2)) #narrow list of DE genes to those with at least a fold-change of 1.2
#assess output
tt <- topTags(qlf, n=Inf, adjust.method="BH", p.value=1)
d <- density(tt$table$logFC)
plot(d)
summary(tt$table$PValue)
hist(tt$table$PValue, 50)
hist(tt$table$FDR, 50)
dim(tt$table)
table(tt$table$PValue<0.05)
table(tt$table$FDR<0.05)
head(tt$table, n=5)
summary(decideTests(qlf, adjust.method="BH", p.value=0.05))
plotMD(qlf)
#counts (qlf): 1429/4937
tt <- topTags(tr, n=Inf, adjust.method="BH", sort.by="PValue", p.value=1)
d <- density(tt$table$logFC)
plot(d)
summary(tt$table$PValue)
hist(tt$table$PValue, 50)
hist(tt$table$FDR, 50)
dim(tt$table)
table(tt$table$PValue<0.05)
table(tt$table$FDR<0.05)
head(tt$table, n=5)
summary(decideTests(tr, adjust.method="BH", p.value=0.05))
plotMD(tr)
#counts (tr): 1222/4937
tt = tt
df = data.frame(pval = tt$table$PValue,
                         padj = tt$table$FDR,
                         row.names = rownames(tt$table))

##limma - SEE LIMMA TREND
##https://ucdavis-bioinformatics-training.github.io/2019-single-cell-RNA-sequencing-Workshop-UCD_UCSF/scrnaseq_analysis/scRNA_Workshop-PART6.html
#https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf
#library(limma)
#obl <- subset(data,subset=OstAdHoc.Assign=="osteoblast")
#expr <- as.matrix(GetAssayData(obl, assay="RNA", slot="data"))
#expr <- expr[which(rownames(expr) %in% genes.no.ribo),] #remove ribosomal genes
#bad <- which(rowSums(expr)==0) #filter out genes with no expression
#expr <- expr[-bad,]
#mm <- model.matrix(~0+species+Collection+SequencingBatch+nCount_RNA+Phase, data=obl@meta.data)
#fit <- lmFit(expr, mm)
#head(coef(fit)) #means in each sample for each gene
#contr <- makeContrasts(speciesChimp-speciesHuman, levels=colnames(coef(fit)))
#tmp <- contrasts.fit(fit, contrasts=contr)
#tmp <- eBayes(tmp)
#d <- density(topTable(tmp, n=dim(expr)[1])$logFC)
#plot(d)
#summary(topTable(tmp, n=dim(expr)[1])$adj.P.Val)
#hist(topTable(tmp, n=dim(expr)[1])$adj.P.Val)
#dim(topTable(tmp, n=dim(expr)[1]))
#table(topTable(tmp, n=dim(expr)[1])$adj.P.Val<0.05)
#topTable(tmp, sort.by="P", n=5)
##counts: 2841/13912
##data: 2018/13912

#limmatrend
#https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_limmatrend.R
library(limma)
runLIMMA <- function(dataSub, genes.no.ribo) {
  
  #count matrix
  counts <- as.matrix(GetAssayData(dataSub, assay="RNA", slot="counts"))
  counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
  
  #metadata
  metadata <- dataSub@meta.data[,c("species","Collection","nCount_RNA","Phase")]
  metadata <- metadata[colnames(counts),]
  
  #make edgeR object
  dge <- DGEList(counts)
  meta_dge <- dge$samples[,c("lib.size","norm.factors")]
  meta_dge <- cbind(meta_dge, metadata)
  dge$samples <- meta_dge
  
  #filter genes
  keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
  table(keep)
  dge <- dge[keep, , keep.lib.sizes=FALSE]
  
  #normalize data
  dge <- calcNormFactors(dge, method="TMM")
  summary(dge$samples$norm.factors)
  
  #lcpm
  y <- new("EList")
  y$E <- edgeR::cpm(dge, log=TRUE, prior.count=3)
  
  #design matrix
  dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
  dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
  dge$samples$species <- as.factor(as.character(dge$samples$species))
  design <- model.matrix(~Collection+nCount_RNA+Phase+species, data=dge$samples)
  
  #linear model
  fit <- lmFit(y, design=design)

  #differential expression
  contr <- contrasts.fit(fit, coefficients=dim(design)[2])
  lmf <- eBayes(contr, trend=TRUE, robust=TRUE)
  tr <- treat(contr, lfc=log2(1.2), trend=TRUE, robust=TRUE)
  
  #assess output
  tt <- topTable(lmf, n=Inf, adjust.method="BH", p.value=1)
  print("All DE Genes")
  print(dim(tt))
  print(table(tt$adj.P.Val<0.05))
  print(summary(decideTests(lmf, adjust.method="BH", p.value=0.05)))

  tt <- topTable(tr, n=Inf, adjust.method="BH", sort.by="p", p.value=1)
  print("DE Genes with log2(1.2) Fold Change")
  print(dim(tt))
  print(table(tt$adj.P.Val<0.05))
  print(summary(decideTests(tr, adjust.method="BH", p.value=0.05)))

  return(tt)

}
lmma <- runLIMMA(subset(data,subset=OstAdHoc.Assign=="osteoblast"), genes.no.ribo)
#count matrix
obl <- subset(data,subset=OstAdHoc.Assign=="osteoblast")
counts <- as.matrix(GetAssayData(obl, assay="RNA", slot="counts"))
counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
#metadata
metadata <- obl@meta.data[,c("species","Collection","nCount_RNA","Phase")]
metadata <- metadata[colnames(counts),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
#filter genes
keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
table(keep)
dge <- dge[keep, , keep.lib.sizes=FALSE]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
summary(dge$samples$norm.factors)
#lcpm
y <- new("EList")
y$E <- edgeR::cpm(dge, log=TRUE, prior.count=3)
#design matrix
dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
dge$samples$species <- as.factor(as.character(dge$samples$species))
design <- model.matrix(~Collection+nCount_RNA+Phase+species, data=dge$samples)
#linear model
fit <- lmFit(y, design=design)
head(fit$coefficients)
hist(fit$coefficients, 50)
plot(fit$coefficients, fit$df.residual)
#differential expression
contr <- contrasts.fit(fit, coefficients=dim(design)[2])
lmf <- eBayes(contr, trend=TRUE, robust=TRUE)
tr <- treat(contr, lfc=log2(1.2), trend=TRUE, robust=TRUE)
#assess output
tt <- topTable(lmf, n=Inf, adjust.method="BH", p.value=1)
d <- density(tt$logFC)
plot(d)
summary(tt$adj.P.Val)
hist(tt$P.Value, 50)
hist(tt$adj.P.Val, 50)
dim(tt)
table(tt$P.Value<0.05)
table(tt$adj.P.Val<0.05)
head(tt, n=5)
summary(decideTests(lmf, adjust.method="BH", p.value=0.05))
plotMD(lmf)
#lcpm (lmf): 1743/4937
tt <- topTable(tr, n=Inf, adjust.method="BH", sort.by="p", p.value=1)
d <- density(tt$logFC)
plot(d)
summary(tt$adj.P.Val)
hist(tt$P.Value, 50)
hist(tt$adj.P.Val, 50)
dim(tt)
table(tt$P.Value<0.05)
table(tt$adj.P.Val<0.05)
head(tt, n=5)
summary(decideTests(tr, adjust.method="BH", p.value=0.05))
plotMD(tr)
#lcpm (tr): 147/4937
tt = tt
df = data.frame(pval = tt$P.Value,
                       padj = tt$adj.P.Val,
                       row.names = rownames(tt))

#MAST (Finak et al., 2015)
#uses a twopart generalized linear model (hurdle model) that analyzes continuous expression levels rather than discrete count values
#model: continuous expression levels are modeled in two-parts:
#       (a) rate of gene expression level by logistic regression, and 
#       (b) transformed positive expression mean using a Gaussian linear model
#de: tested using a likelihood ratio test with asymptotic chi-square null distribution and FDR adjustment control
#https://github.com/csoneson/conquer_comparison/blob/master/scripts/apply_MASTcpmDetRate.R
#https://www.bioconductor.org/packages/release/bioc/vignettes/MAST/inst/doc/MAST-Intro.html#4_use_with_single_cell_rna-sequencing_data
#https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/MAST/inst/doc/MAITAnalysis.html
#MASTcpmDetRate
library(MAST)
runMAST <- function(dataSub, genes.no.ribo) {
  
  #count matrix
  counts <- as.matrix(GetAssayData(dataSub, assay="RNA", slot="counts"))
  counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
  
  #metadata
  metadata <- dataSub@meta.data[,c("species","Collection","nCount_RNA","Phase")]
  metadata <- metadata[colnames(counts),]
  
  #make edgeR object
  dge <- DGEList(counts)
  meta_dge <- dge$samples[,c("lib.size","norm.factors")]
  meta_dge <- cbind(meta_dge, metadata)
  dge$samples <- meta_dge
  
  #filter genes
  keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
  table(keep)
  dge <- dge[keep, , keep.lib.sizes=FALSE]
  
  #normalize data
  dge <- edgeR::calcNormFactors(dge, method="TMM")
  summary(dge$samples$norm.factors)
  cpms <- edgeR::cpm(dge)
  
  #calculate cellular detection rate
  cdr <- scale(colMeans(dge$count > 0))
  
  #design matrix
  dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
  dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
  dge$samples$species <- as.factor(as.character(dge$samples$species))
  sca <- FromMatrix(exprsArray=log2(cpms+1),
                    cData=data.frame(wellKey=rownames(dge$samples),
                                     species=dge$samples$species,
                                     Collection=dge$samples$Collection,
                                     nCount_RNA=dge$samples$nCount_RNA,
                                     Phase=dge$samples$Phase,
                                     cdr=cdr))
  
  #hurdle model fit
  zlmdata <- zlm(~Collection+cdr+nCount_RNA+Phase+species, sca)
  show(zlmdata)

  #differential expression
  mast <- lrTest(zlmdata, "species")
  
  #assess output
  print(length(mast[, "hurdle", "Pr(>Chisq)"]))
  print(table(mast[, "hurdle", "Pr(>Chisq)"]<0.05))

  return(mast)

}
mast <- runMAST(subset(data,subset=OstAdHoc.Assign=="osteoblast"), genes.no.ribo)
#count matrix
obl <- subset(data,subset=OstAdHoc.Assign=="osteoblast")
counts <- as.matrix(GetAssayData(obl, assay="RNA", slot="counts"))
counts <- counts[which(rownames(counts) %in% genes.no.ribo),] #remove ribosomal genes
#metadata
metadata <- obl@meta.data[,c("species","Collection","nCount_RNA","Phase")]
metadata <- metadata[colnames(counts),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
#filter genes
keep <- filterByExpr(dge, group=dge$samples$species, min.count=1, min.total.count=15)
table(keep)
dge <- dge[keep, , keep.lib.sizes=FALSE]
#normalize data
dge <- edgeR::calcNormFactors(dge, method="TMM")
summary(dge$samples$norm.factors)
cpms <- edgeR::cpm(dge)
#calculate cellular detection rate
cdr <- scale(colMeans(dge$count > 0))
#design matrix
dge$samples$Collection <- as.factor(as.numeric(dge$samples$Collection))
dge$samples$Phase <- as.factor(as.numeric(dge$samples$Phase))
dge$samples$species <- as.factor(as.character(dge$samples$species))
sca <- FromMatrix(exprsArray=log2(cpms+1),
                  cData=data.frame(wellKey=rownames(dge$samples),
                                   species=dge$samples$species,
                                   Collection=dge$samples$Collection,
                                   nCount_RNA=dge$samples$nCount_RNA,
                                   Phase=dge$samples$Phase,
                                   cdr=cdr))
#hurdle model fit
zlmdata <- zlm(~Collection+cdr+nCount_RNA+Phase+species, sca)
show(zlmdata)
coefAndCI <- summary(zlmdata, logFC=FALSE)$datatable
head(coefAndCI)
#differential expression
mast <- lrTest(zlmdata, "species")
#assess output
hist(mast[, "hurdle", "Pr(>Chisq)"], 50)
length(mast[, "hurdle", "Pr(>Chisq)"])
table(mast[, "hurdle", "Pr(>Chisq)"]<0.05)
head(mast[, "hurdle", ])
#cpm: 1666/4937
res = mast
df = data.frame(pval=mast[, "hurdle", "Pr(>Chisq)"],
                row.names=names(mast[, "hurdle", "Pr(>Chisq)"]))

#SCDE (Kharchenko et al., 2014)
#uses a mixture probabilistic model for gene expression values
#model: read counts of genes are modeled as a mixture of:
#        (a) drop-out events by a low-level Poisson distribution (to accommodate genes that are undetected across some of the cells), and
#        (b) amplification components by a negative binomial distribution (to account for overdispersed expression counts from detected transcripts)
#de: posterior probabilities of a given fold expression difference are calculated to test genes for differential expression between two subgroups of cells

#DESeq2 (Love et al., 2014)
#uses negative binomial distribution
#model: allows the use of a gene-specific shrinkage estimation for dispersions
#       uses all genes with a similar average expression to estimate dispersion
#de: fold-change estimation is employed to avoid identifying genes with small average expression values



#EVALUATE DIFFERENTIAL EXPRESSION TEST RESULTS

#Assess functional enrichment of differentially expressed genes
goCCenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "CC",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "CC",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
goBPenrich <- function(data, markers) {
  
  ref.df <- bitr(rownames(data@assays$RNA), fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC>0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_human <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC<0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego_chimp <- enrichGO(gene          = gene.df$ENSEMBL,
                        OrgDb         = org.Hs.eg.db,
                        universe      = ref.df$ENSEMBL,
                        keyType       = 'ENSEMBL',
                        ont           = "BP",
                        pAdjustMethod = "BH",
                        pvalueCutoff  = 0.01,
                        qvalueCutoff  = 0.05,
                        readable      = TRUE)
  
  gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  #gene <- markers$gene[markers$p_val_adj<0.05 & markers$avg_logFC!=0]
  gene.df <- bitr(gene, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = "BP",
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  
  return(list(barplot(ego_human, drop=TRUE, showCategory=10, title = 'Upregulated in Human GO Analysis'),
              barplot(ego_chimp, drop=TRUE, showCategory=10, title = 'Upregulated in Chimp GO Analysis'),
              barplot(ego, drop=TRUE, showCategory=10, title = 'DE Genes GO Analysis')))

}
funcEnrich <- function(data, markers) {
  genes <- structure(rep(1, length(rownames(data@assays$RNA@data))), names=rownames(data@assays$RNA@data))
  x=1
  while (x <= length(rownames(markers))) {
    genes[which(names(genes)==rownames(markers)[x])] <- markers$p_val_adj[x]
    x=x+1
  }
  go_data <- new("topGOdata", ontology="BP", allGenes=genes, geneSel=function(allScore){return(allScore < 0.01)},
                 nodeSize=5, annotationFun=annFUN.org, mapping="org.Hs.eg.db", ID="symbol")
  go_test <- runTest(go_data, algorithm="weight01", statistic="fisher")
  go_table <- GenTable(go_data, weightFisher=go_test, orderBy="weightFisher", ranksOf="weightFisher", topNodes=sum(score(go_test) < .01))
  return(go_table)
}

#Find differentially expressed genes using differentiation stage at collection
compare.ips <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.ips.rds"))
compare.msc <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.msc.rds"))
compare.ost <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.ost.rds"))
#Find differentially expressed genes using cluster definitions
compare.i <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i.rds"))
compare.m <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.m.rds"))
compare.o <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o.rds"))
compare.i1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i1.rds"))
compare.i2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i2.rds"))
compare.i3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.i3.rds"))
compare.o1 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o1.rds"))
compare.o2 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o2.rds"))
compare.o3 <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.o3.rds"))
#Find differentially expressed genes using ad hoc definitions
compare.hoci <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hoci.rds"))
compare.hocm <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hocm.rds"))
compare.hoco <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.hoco.rds"))
#Find differentially expressed genes using osteogenic ad hoc definitions
compare.pre <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.pre.rds"))
compare.obl <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.obl.rds"))
compare.emb <- readRDS(paste0(dir,"/human-chimp-skeletal-scRNA/data/cellranger-data-full/de.w-compare.emb.rds"))


dim(compare.hoci)
dim(compare.hocm)
dim(compare.hoco)
table(compare.hoci$p_val_adj<0.05)
table(compare.hocm$p_val_adj<0.05)
table(compare.hoco$p_val_adj<0.05)
head(compare.hoci, n=5)
head(compare.hocm, n=5)
head(compare.hoco, n=5)
compare.hoci$gene <- rownames(compare.hoci)
compare.hocm$gene <- rownames(compare.hocm)
compare.hoco$gene <- rownames(compare.hoco)
fracRibo(compare.hoci)
fracRibo(compare.hocm)
fracRibo(compare.hoco)
p.hoci <- goCCenrich(indv,compare.hoci)
p.hocm <- goCCenrich(indv,compare.hocm)
p.hoco <- goCCenrich(indv,compare.hoco)
p.hoci[[3]]
p.hocm[[3]]
p.hoco[[3]]
#go.hoci <- funcEnrich(indv, compare.hoci)
#go.hocm <- funcEnrich(indv, compare.hocm)
#go.hoco <- funcEnrich(indv, compare.hoco)
#dim(go.hoci)
#dim(go.hocm)
#dim(go.hoco)
#head(go.hoci, n=5)
#head(go.hocm, n=5)
#head(go.hoco, n=5)
compare.tot5 <- intersect(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],intersect(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                                rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)]))
compare.hoci.tot <- setdiff(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],c(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hoci
compare.hocm.tot <- setdiff(rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)])) #DE hocm
compare.hoco.tot <- setdiff(rownames(compare.hoco)[which(compare.hoco$p_val_adj<0.05)],c(rownames(compare.hoci)[which(compare.hoci$p_val_adj<0.05)],
                                                                                          rownames(compare.hocm)[which(compare.hocm$p_val_adj<0.05)])) #DE hoco
length(compare.tot5)
length(compare.hoci.tot)
length(compare.hocm.tot)
length(compare.hoco.tot)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),], n=5)
head(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),], n=5)
head(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),], n=5)
head(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),], n=5)
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
fracRibo(compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
fracRibo(compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
fracRibo(compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5 <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goCCenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goCCenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goCCenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
p.tot5 <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.tot5),])
p.hoci.tot <- goBPenrich(indv,compare.hoci[which(rownames(compare.hoci)%in%compare.hoci.tot),])
p.hocm.tot <- goBPenrich(indv,compare.hocm[which(rownames(compare.hocm)%in%compare.hocm.tot),])
p.hoco.tot <- goBPenrich(indv,compare.hoco[which(rownames(compare.hoco)%in%compare.hoco.tot),])
p.tot5[[3]]
p.hoci.tot[[3]]
p.hocm.tot[[3]]
p.hoco.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)]),
                                   length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)]),
                                   length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])),
                    Other=c((length(compare.hoci$gene[which(compare.hoci$p_val_adj<0.05)])-length(compare.hoci.tot)-length(compare.tot5)),
                            (length(compare.hocm$gene[which(compare.hocm$p_val_adj<0.05)])-length(compare.hocm.tot)-length(compare.tot5)),
                            (length(compare.hoco$gene[which(compare.hoco$p_val_adj<0.05)])-length(compare.hoco.tot)-length(compare.tot5))),
                    Shared=c(rep(length(compare.tot5),3)),
                    Unique=c(length(compare.hoci.tot),
                             length(compare.hocm.tot),
                             length(compare.hoco.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


dim(compare.i)
dim(compare.m)
dim(compare.o) 
dim(compare.i1)
dim(compare.i2)
dim(compare.i3)
dim(compare.o1)
dim(compare.o2)
dim(compare.o3)
table(compare.i$p_val_adj<0.05)
table(compare.m$p_val_adj<0.05)
table(compare.o$p_val_adj<0.05)
table(compare.i1$p_val_adj<0.05)
table(compare.i2$p_val_adj<0.05)
table(compare.i3$p_val_adj<0.05)
table(compare.o1$p_val_adj<0.05)
table(compare.o2$p_val_adj<0.05)
table(compare.o3$p_val_adj<0.05)
head(compare.i, n=5)
head(compare.m, n=5)
head(compare.o, n=5)
head(compare.i1, n=5)
head(compare.i2, n=5)
head(compare.i3, n=5)
head(compare.o1, n=5)
head(compare.o2, n=5)
head(compare.o3, n=5)
compare.i$gene <- rownames(compare.i)
compare.m$gene <- rownames(compare.m)
compare.o$gene <- rownames(compare.o)
compare.i1$gene <- rownames(compare.i1)
compare.i2$gene <- rownames(compare.i2)
compare.i3$gene <- rownames(compare.i3)
compare.o1$gene <- rownames(compare.o1)
compare.o2$gene <- rownames(compare.o2)
compare.o3$gene <- rownames(compare.o3)
fracRibo(compare.i)
fracRibo(compare.m)
fracRibo(compare.o)
fracRibo(compare.i1)
fracRibo(compare.i2)
fracRibo(compare.i3)
fracRibo(compare.o1)
fracRibo(compare.o2)
fracRibo(compare.o3)
p.i <- goCCenrich(indv,compare.i)
p.m <- goCCenrich(indv,compare.m)
p.o <- goCCenrich(indv,compare.o)
p.i1 <- goCCenrich(indv,compare.i1)
p.i2 <- goCCenrich(indv,compare.i2)
p.i3 <- goCCenrich(indv,compare.i3)
p.o1 <- goCCenrich(indv,compare.o1)
p.o2 <- goCCenrich(indv,compare.o2)
p.o3 <- goCCenrich(indv,compare.o3)
p.i[[3]]
p.m[[3]]
p.o[[3]]
p.i1[[3]]
p.i2[[3]]
p.i3[[3]]
p.o1[[3]]
p.o2[[3]]
p.o3[[3]]
#go.i <- funcEnrich(indv, compare.i)
#go.m <- funcEnrich(indv, compare.m)
#go.o <- funcEnrich(indv, compare.o)
#go.i1 <- funcEnrich(indv, compare.i1)
#go.i2 <- funcEnrich(indv, compare.i2)
#go.i3 <- funcEnrich(indv, compare.i3)
#go.o1 <- funcEnrich(indv, compare.o1)
#go.o2 <- funcEnrich(indv, compare.o2)
#go.o3 <- funcEnrich(indv, compare.o3)
#dim(go.i)
#dim(go.m)
#dim(go.o)
#dim(go.i1)
#dim(go.i2)
#dim(go.i3)
#dim(go.o1)
#dim(go.o2)
#dim(go.o3)
#head(go.i, n=5)
#head(go.m, n=5)
#head(go.o, n=5)
#head(go.i1, n=5)
#head(go.i2, n=5)
#head(go.i3, n=5)
#head(go.o1, n=5)
#head(go.o2, n=5)
#head(go.o3, n=5)
compare.tot2 <- intersect(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                          intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                    rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #shared DE genes
compare.i.tot <- setdiff(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],c(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to i
compare.m.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.o)[which(compare.o$p_val_adj<0.05)])) #DE unique to m
compare.o.tot <- setdiff(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],c(rownames(compare.i)[which(compare.i$p_val_adj<0.05)],
                                                                                rownames(compare.m)[which(compare.m$p_val_adj<0.05)])) #DE unique to o
compare.tot3 <- intersect(rownames(compare.o)[which(compare.o$p_val_adj<0.05)],
                          intersect(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                    intersect(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                              intersect(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                        intersect(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                  intersect(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                            intersect(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                      rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)]))))))) #shared DE
compare.i1.tot <- setdiff(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],c(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i1
compare.i2.tot <- setdiff(rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i2
compare.i3.tot <- setdiff(rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to i3
compare.m1.tot <- setdiff(rownames(compare.m)[which(compare.m$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                 rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                 rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                 rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                 rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                 rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to m1
compare.o1.tot <- setdiff(rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o1
compare.o2.tot <- setdiff(rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)])) #DE unique to o2
compare.o3.tot <- setdiff(rownames(compare.o3)[which(compare.o3$p_val_adj<0.05)],c(rownames(compare.i1)[which(compare.i1$p_val_adj<0.05)],
                                                                                   rownames(compare.i2)[which(compare.i2$p_val_adj<0.05)],
                                                                                   rownames(compare.i3)[which(compare.i3$p_val_adj<0.05)],
                                                                                   rownames(compare.m)[which(compare.m$p_val_adj<0.05)],
                                                                                   rownames(compare.o1)[which(compare.o1$p_val_adj<0.05)],
                                                                                   rownames(compare.o2)[which(compare.o2$p_val_adj<0.05)])) #DE unique to o3
length(compare.tot2)
length(compare.i.tot)
length(compare.m.tot)
length(compare.o.tot)
length(compare.tot3)
length(compare.i1.tot)
length(compare.i2.tot)
length(compare.i3.tot)
length(compare.m1.tot)
length(compare.o1.tot)
length(compare.o2.tot)
length(compare.o3.tot)
head(compare.i[which(rownames(compare.i)%in%compare.tot2),], n=5)
head(compare.i[which(rownames(compare.i)%in%compare.i.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m.tot),], n=5)
head(compare.o[which(rownames(compare.o)%in%compare.o.tot),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.tot3),], n=5)
head(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),], n=5)
head(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),], n=5)
head(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),], n=5)
head(compare.m[which(rownames(compare.m)%in%compare.m1.tot),], n=5)
head(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),], n=5)
head(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),], n=5)
head(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),], n=5)
fracRibo(compare.i[which(rownames(compare.i)%in%compare.tot2),])
fracRibo(compare.i[which(rownames(compare.i)%in%compare.i.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m.tot),])
fracRibo(compare.o[which(rownames(compare.o)%in%compare.o.tot),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
fracRibo(compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
fracRibo(compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
fracRibo(compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
fracRibo(compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
fracRibo(compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
fracRibo(compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
fracRibo(compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2 <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goCCenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goCCenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goCCenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goCCenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goCCenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goCCenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goCCenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goCCenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goCCenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
p.tot2 <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.tot2),])
p.i.tot <- goBPenrich(indv,compare.i[which(rownames(compare.i)%in%compare.i.tot),])
p.m.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m.tot),])
p.o.tot <- goBPenrich(indv,compare.o[which(rownames(compare.o)%in%compare.o.tot),])
p.tot3 <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.tot3),])
p.i1.tot <- goBPenrich(indv,compare.i1[which(rownames(compare.i1)%in%compare.i1.tot),])
p.i2.tot <- goBPenrich(indv,compare.i2[which(rownames(compare.i2)%in%compare.i2.tot),])
p.i3.tot <- goBPenrich(indv,compare.i3[which(rownames(compare.i3)%in%compare.i3.tot),])
p.m1.tot <- goBPenrich(indv,compare.m[which(rownames(compare.m)%in%compare.m1.tot),])
p.o1.tot <- goBPenrich(indv,compare.o1[which(rownames(compare.o1)%in%compare.o1.tot),])
p.o2.tot <- goBPenrich(indv,compare.o2[which(rownames(compare.o2)%in%compare.o2.tot),])
p.o3.tot <- goBPenrich(indv,compare.o3[which(rownames(compare.o3)%in%compare.o3.tot),])
p.tot2[[3]]
p.i.tot[[3]]
p.m.tot[[3]]
p.o.tot[[3]]
p.tot3[[3]]
p.i1.tot[[3]]
p.i2.tot[[3]]
p.i3.tot[[3]]
p.m1.tot[[3]]
p.o1.tot[[3]]
p.o2.tot[[3]]
p.o3.tot[[3]]
DE.df <- data.frame(cell=(c("iPSC","MSC","Osteoblast")),
                    totalDEgenes=c(length(compare.i$gene[which(compare.i$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o$gene[which(compare.o$p_val_adj<0.05)])),
                    Other=c((length(compare.i$gene[which(compare.i$p_val_adj<0.05)])-length(compare.i.tot)-length(compare.tot2)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m.tot)-length(compare.tot2)),
                            (length(compare.o$gene[which(compare.o$p_val_adj<0.05)])-length(compare.o.tot)-length(compare.tot2))),
                    Shared=c(rep(length(compare.tot2),3)),
                    Unique=c(length(compare.i.tot),
                             length(compare.m.tot),
                             length(compare.o.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC","MSC","Osteoblast"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[4:12,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))
DE.df <- data.frame(cell=(c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3")),
                    totalDEgenes=c(length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)]),
                                   length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)]),
                                   length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)]),
                                   length(compare.m$gene[which(compare.m$p_val_adj<0.05)]),
                                   length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)]),
                                   length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)]),
                                   length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])),
                    Other=c((length(compare.i1$gene[which(compare.i1$p_val_adj<0.05)])-length(compare.i1.tot)-length(compare.tot3)),
                            (length(compare.i2$gene[which(compare.i2$p_val_adj<0.05)])-length(compare.i2.tot)-length(compare.tot3)),
                            (length(compare.i3$gene[which(compare.i3$p_val_adj<0.05)])-length(compare.i3.tot)-length(compare.tot3)),
                            (length(compare.m$gene[which(compare.m$p_val_adj<0.05)])-length(compare.m1.tot)-length(compare.tot3)),
                            (length(compare.o1$gene[which(compare.o1$p_val_adj<0.05)])-length(compare.o1.tot)-length(compare.tot3)),
                            (length(compare.o2$gene[which(compare.o2$p_val_adj<0.05)])-length(compare.o2.tot)-length(compare.tot3)),
                            (length(compare.o3$gene[which(compare.o3$p_val_adj<0.05)])-length(compare.o3.tot)-length(compare.tot3))),
                    Shared=c(rep(length(compare.tot3),7)),
                    Unique=c(length(compare.i1.tot),
                             length(compare.i2.tot),
                             length(compare.i3.tot),
                             length(compare.m1.tot),
                             length(compare.o1.tot),
                             length(compare.o2.tot),
                             length(compare.o3.tot)))
DE.df$cell <- factor(DE.df$cell, levels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteo.c1","Osteo.c2","Osteo.c3"))
ggplot(DE.df, aes(x=cell, y=totalDEgenes)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
DE.df <- melt(DE.df, id.var="cell")
ggplot(DE.df[8:28,], aes(x=cell, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  labs(x="",y="Number of DE Genes") +
  theme(legend.title = element_blank()) +
  scale_fill_manual(values=c("grey80","grey40","black")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12), legend.title=element_text(size=0))


```

```{r plots for BoG presentation}

indv <- AddMetaData(indv, rep(NA,length(rownames(indv@meta.data))), col.name="Stage")
indv$Stage[which(indv$Cell.Type=="iPSC")] <- "Time 0" 
indv$Stage[which(indv$Cell.Type=="MSC")] <- "Time 1"
indv$Stage[which(indv$Cell.Type=="Osteoblast")] <- "Time 2"

Idents(indv) <- factor(indv@meta.data$species, levels=c("Chimp","Human"))

#cell counts
ggplot(indv@meta.data, aes(x=Stage, fill=Stage)) +
  geom_bar() +
  labs(y="Number of Cells", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(indv@meta.data[which(indv@meta.data$species=="Human"),], aes(x=Stage, fill=Stage)) +
  geom_bar() +
  labs(y="Number of Cells", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
ggplot(indv@meta.data[which(indv@meta.data$species=="Chimp"),], aes(x=Stage, fill=Stage)) +
  geom_bar() +
  labs(y="Number of Cells", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
#ggplot(indv@meta.data, aes(x=Stage, fill=species)) +
#  geom_bar(position="dodge") +
#  labs(y="Number of Cells", x="") +
#  scale_fill_manual(name="", values=c("grey70","grey30")) +
#  theme_classic() +
#  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))

#umi counts
ggplot(indv@meta.data, aes(x=Stage, y=nCount_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="UMIs per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(indv@meta.data[which(indv@meta.data$species=="Human"),], aes(x=Stage, y=nCount_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="UMIs per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
ggplot(indv@meta.data[which(indv@meta.data$species=="Chimp"),], aes(x=Stage, y=nCount_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="UMIs per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
#ggplot(indv@meta.data, aes(x=Stage, y=nCount_RNA, fill=species)) +
#  geom_violin() +
#  labs(y="UMIs per Cell", x="") +
#  scale_fill_manual(name="", values=c("grey70","grey30")) +
#  theme_classic() +
#  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))

#gene counts
ggplot(indv@meta.data, aes(x=Stage, y=nFeature_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="Genes per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))
ggplot(indv@meta.data[which(indv@meta.data$species=="Human"),], aes(x=Stage, y=nFeature_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="Genes per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
ggplot(indv@meta.data[which(indv@meta.data$species=="Chimp"),], aes(x=Stage, y=nFeature_RNA, fill=Stage)) +
  geom_violin() +
  labs(y="Genes per Cell", x="") +
  scale_fill_manual(name="", values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=8), axis.title=element_text(size=8), legend.text=element_text(size=8))
#ggplot(indv@meta.data, aes(x=Stage, y=nFeature_RNA, fill=species)) +
#  geom_violin() +
#  labs(y="Genes per Cell", x="") +
#  scale_fill_manual(name="", values=c("grey70","grey30")) +
#  theme_classic() +
#  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))

#stage of differentiation
DimPlot(indv, group.by="Stage", reduction="umap", cols=c("pink","mediumorchid","dodgerblue")) + labs(x="UMAP1",y="UMAP2")
DimPlot(subset(indv,subset=species=="Human"), group.by="Stage", reduction="umap", cols=c("pink","mediumorchid","dodgerblue")) + labs(x="UMAP1",y="UMAP2")
DimPlot(subset(indv,subset=species=="Chimp"), group.by="Stage", reduction="umap", cols=c("pink","mediumorchid","dodgerblue")) + labs(x="UMAP1",y="UMAP2")
DimPlot(subset(indv,subset=species=="Human"), group.by="Stage", split.by="Individual", reduction="umap", cols=c("pink","mediumorchid","dodgerblue")) +
  labs(x="UMAP1",y="UMAP2")
DimPlot(subset(indv,subset=species=="Chimp"), group.by="Stage", split.by="Individual", reduction="umap", cols=c("pink","mediumorchid","dodgerblue")) +
  labs(x="UMAP1",y="UMAP2")
#VlnPlot(indv, features=c("POU5F1","TDGF1","SOX2"), group.by="Stage", cols=c("pink","mediumorchid","dodgerblue"), pt.size=0)
#VlnPlot(indv, features=c("NT5E","ENG","CD44"), group.by="Stage", cols=c("pink","mediumorchid","dodgerblue"), pt.size=0)
#VlnPlot(indv, features=c("COL1A1","COL1A2","RUNX2"), group.by="Stage", cols=c("pink","mediumorchid","dodgerblue"), pt.size=0)
VlnPlot(subset(indv,subset=species=="Human"), features=c("POU5F1","CD44","COL1A1"), group.by="Stage", cols=c("pink","mediumorchid","dodgerblue"), pt.size=0)
VlnPlot(subset(indv,subset=species=="Chimp"), features=c("POU5F1","CD44","COL1A1"), group.by="Stage", cols=c("pink","mediumorchid","dodgerblue"), pt.size=0)

#unsupervised clustering
DimPlot(indv, group.by="Cluster", reduction="umap", cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4")) +
  labs(x="UMAP1",y="UMAP2")
#VlnPlot(indv, features=c("POU5F1","TDGF1","SOX2"), group.by="seurat_clusters", pt.size=0)
#VlnPlot(indv, features=c("NT5E","ENG","CD44"), group.by="seurat_clusters", pt.size=0)
#VlnPlot(indv, features=c("COL1A1","COL1A2","RUNX2"), group.by="seurat_clusters", pt.size=0)
VlnPlot(subset(indv,subset=species=="Human"), features=c("POU5F1","CD44","COL1A1"), group.by="Cluster",
        cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4"), pt.size=0)
VlnPlot(subset(indv,subset=species=="Chimp"), features=c("POU5F1","CD44","COL1A1"), group.by="Cluster",
        cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue3","dodgerblue4"), pt.size=0)

#grade of membership cell assignments
#DimPlot(indv, group.by="GoM.Assign", reduction="umap", cols=c("pink","mediumorchid","dodgerblue","lightgrey")) + labs(x="UMAP1",y="UMAP2")
#VlnPlot(indv, features=c("POU5F1","TDGF1","SOX2"), group.by="GoM.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(indv, features=c("NT5E","ENG","CD44"), group.by="GoM.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(indv, features=c("COL1A1","COL1A2","RUNX2"), group.by="GoM.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(subset(indv,subset=species=="Human"), features=c("POU5F1","CD44","COL1A1"), group.by="GoM.Assign",
#        cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(subset(indv,subset=species=="Chimp"), features=c("POU5F1","CD44","COL1A1"), group.by="GoM.Assign",
#        cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)

#ad hoc cell assignments
DimPlot(indv, group.by="AdHoc.Assign", reduction="umap", cols=c("pink","mediumorchid","dodgerblue","lightgrey")) + labs(x="UMAP1",y="UMAP2")
DimPlot(indv, group.by="AdHoc.Assign", reduction="umap", cols=c("lightgrey","pink","dodgerblue","mediumorchid"), order=c("MSC","Osteoblast","iPSC","other")) +
  labs(x="UMAP1",y="UMAP2")
DimPlot(indv, group.by="AdHoc.Assign", reduction="umap", cols=c("lightgrey","pink","dodgerblue","mediumorchid"), order=c("MSC","Osteoblast","iPSC","other")) +
  labs(x="UMAP1",y="UMAP2")
#VlnPlot(indv, features=c("POU5F1","TDGF1","SOX2"), group.by="AdHoc.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(indv, features=c("NT5E","ENG","CD44"), group.by="AdHoc.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
#VlnPlot(indv, features=c("COL1A1","COL1A2","RUNX2"), group.by="AdHoc.Assign", cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
VlnPlot(subset(indv,subset=species=="Human"), features=c("POU5F1","CD44","COL1A1"), group.by="AdHoc.Assign",
        cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)
VlnPlot(subset(indv,subset=species=="Chimp"), features=c("POU5F1","CD44","COL1A1"), group.by="AdHoc.Assign",
        cols=c("pink","mediumorchid","dodgerblue","lightgrey"), pt.size=0)

#ad hoc osteogenic cell assignments
DimPlot(indv, group.by="Stage", reduction="umap", cols=c("lightgrey","lightgrey","dodgerblue")) + labs(x="UMAP1",y="UMAP2")
DimPlot(indv, group.by="Stage", split.by="OstAdHoc.Assign", reduction="umap", cols=c("lightgrey","lightgrey","dodgerblue")) + labs(x="UMAP1",y="UMAP2")

VlnPlot(subset(indv,subset=species=="Human"), features=c("POU5F1","CD44","COL1A1"), group.by="OstAdHoc.Assign", pt.size=0)
VlnPlot(subset(indv,subset=species=="Chimp"), features=c("POU5F1","CD44","COL1A1"), group.by="OstAdHoc.Assign", pt.size=0)


```

```{r cell cycle}

#cell cycle markers (Tirosh et al, 2015) - loaded with seurat
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

#score cell cycle phases
indv <- CellCycleScoring(indv, s.features=s.genes, g2m.features=g2m.genes, set.ident=TRUE)

#view cell cycle scores and phase assignments
head(indv[[]])

#plot cell cycle scores
CombinePlots(plots=list((DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(1,2)) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(3,4)) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(5,6)) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(7,8)) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(indv,
                                 group.by="Phase",
                                 reduction="pca",
                                 dims=c(9,10)) +
                           labs(x="PC9",y="PC10"))), ncol=5)
CombinePlots(plots=list((DimPlot(subset(indv,subset=species=="Human"),
                                 group.by="Phase",
                                 reduction="umap") +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=species=="Chimp"),
                                 group.by="Phase",
                                 reduction="umap") +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
CombinePlots(plots=list((DimPlot(subset(indv,subset=Phase=="G1"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Phase=="G2M"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(indv,subset=Phase=="S"),
                                 group.by="Cell.Type",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=3)


```


