---
title: "10x-view-data-metrics"
author: "Genevieve Housman"
date: "May 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# View 10X Data Cellranger Metrics

Data being examined come from 6 humans and 6 chimpanzees.

* GHO-1:  H1-I + C1-I (iPSCs from H23555 and C8861)
* GHO-2:  H1-M + C1-M (iPSC-derived MSCs from H23555 and C8861)
* GHO-4:  H1-O + C1-O (iPSC-MSC-derived osteoblasts from H23555 and C8861)
* HOU-10: H1-I-r2 + C1-I-r2 (iPSCs from H23555 and C8861)
* HOU-12: H1-M-r2 + C1-M-r2 (iPSC-derived MSCs from H23555 and C8861)
* HOU-16: H1-O-r2 + C1-O-r2 (iPSC-MSC-derived osteoblasts from H23555 and C8861)
* HOU-1:  H2-I + C2-I (iPSCs from H20157 and C3647)
* HOU-2:  H2-M + C2-M (iPSC-derived MSCs from H20157 and C3647)
* HOU-4:  H2-O + C2-O (iPSC-MSC-derived osteoblasts from H20157 and C3647)
* HOU-9:  H3-I + C3-I (iPSCs from H28126 and C3649)
* HOU-11: H3-M + C3-M (iPSC-derived MSCs from H28126 and C3649)
* HOU-15: H3-O + C3-O (iPSC-MSC-derived osteoblasts from H28126 and C3649)
* HOU-5:  H4-I + C4-I (iPSCs from H28834 and C40210)
* HOU-6:  H4-M + C4-M (iPSC-derived MSCs from H28834 and C40210)
* HOU-8:  H4-O + C4-O (iPSC-MSC-derived osteoblasts from H28834 and C40210)
* HOU-17:	H5-I + C5-I (iPSCs from H21792 and C40280)
* HOU-19:	H5-M + C5-M (iPSC-derived MSCs from H21792 and C40280)
* HOU-21:	H5-O + C5-O (iPSC-MSC-derived osteoblasts from H21792 and C40280)
* HOU-18:	H6-I + C6-I (iPSCs from H20961 and C3624)
* HOU-20:	H6-M + C6-M (iPSC-derived MSCs from H20961 and C3624)
* HOU-22:	H6-O + C6-O (iPSC-MSC-derived osteoblasts from H20961 and C3624)

Collections are classified by the stage of differentiation at which cells were collected:

* Time 0 = iPSCs
* Time 1 = iPSC-derived MSCs
* Time 2 = iPSC-MSC-derived osteoblasts

```{r load libraries, message=FALSE}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)

```

```{r compile summary metrics, results="hide"}

#Load batch info
batch <- read.csv(file='./data/scrna-batch.csv', header=TRUE, sep=",")

#Define methods of alignment
align <- c("HumanOrthoV2","ChimpOrthoV2")

#Load metrics data from full data
metrics <- data.frame()
for (j in align) {
  i <- 1
  while (i <= length(batch$Sample_Name_at_Core)) {
    print(i)
    #Define directory containing data from one collection
    dir_data <- paste0('./../scRNA/cellranger-data-full/cellranger-data-',j,'/',batch$Sample_Name_at_Core[i],'/outs')
    #Load summary metrics
    temp <- read.csv(file=paste0(dir_data,'/metrics_summary.csv'))
    #Process metrics
    temp[,c(1:4,19:20)] <- lapply(temp[,c(1:4,19:20)], function(x) as.numeric(gsub(",","",as.character(x))))
    temp[,c(5:18)] <- lapply(temp[,c(5:18)], function(x) as.numeric(gsub("%","",as.character(x)))/100)
    temp$Collection.Name <- batch$Sample_Name_at_Core[i]
    temp$Alignment.Type <- j
    #Add data to dataframe
    metrics <- rbind(metrics,temp)
    i <- i+1
  }
}
rm(dir_data,temp,i,j)

#Fix metrics details so that plotting is easier
metrics$Collection.Name <- factor(metrics$Collection.Name,
                                  levels=c("GHO-1","GHO-2","GHO-4",
                                           "HOU-10","HOU-12","HOU-16",
                                           "HOU-1","HOU-2","HOU-4",
                                           "HOU-9","HOU-11","HOU-15",
                                           "HOU-5","HOU-6","HOU-8",
                                           "HOU-17","HOU-19","HOU-21",
                                           "HOU-18","HOU-20","HOU-22"))
metrics$reads_prop <- metrics$Number.of.Reads / metrics$Number.of.Reads
metrics$reads_mapped_to_genome <- metrics$Number.of.Reads * metrics$Reads.Mapped.to.Genome
metrics$reads_mapped_confidently_to_genome <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Genome
metrics$reads_mapped_confidently_to_exonic_regions <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Exonic.Regions
metrics$reads_mapped_confidently_to_transcriptome <- metrics$Number.of.Reads * metrics$Reads.Mapped.Confidently.to.Transcriptome

```

## Visualize read alignment metrics:

```{r}

#total number of reads
ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Number.of.Reads)) +
  geom_col(position="dodge") +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Number of Reads Sequenced",y="Number of Reads",x="Collection") +
  ylim(0,300000000)

```

```{r}

#percent sequence saturation
ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Sequencing.Saturation)) +
  geom_col(position="dodge") +
  theme_classic() +
  theme(axis.text.x=element_text(size=6)) +
  labs(title="Sequencing Saturation",y="Sequencing Saturation",x="Collection") +
  ylim(0,1)

```

```{r}

#mean reads per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads per Cell (hg38)",y="Mean Reads per Cell",x="Collection") +
                           ylim(0,110000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Mean.Reads.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads per Cell (panTro6)",y="Mean Reads per Cell",x="Collection") +
                           ylim(0,110000))), ncol=1)

```

```{r}

#median genes per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Genes per Cell (ortho-exon hg38)",y="Median Genes per Cell",x="Collection") +
                           ylim(0,6500)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Median.Genes.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Genes per Cell (ortho-exon panTro6)",y="Median Genes per Cell",x="Collection") +
                           ylim(0,6500))), ncol=1)

```

```{r}

#median UMI per cell
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="UMIs per Cell (ortho-exon hg38)",y="Median UMI Counts per Cell",x="Collection") +
                           ylim(0,31000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Median.UMI.Counts.per.Cell)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="UMIs per Cell (ortho-exon panTro6)",y="Median UMI Counts per Cell",x="Collection") +
                           ylim(0,31000))), ncol=1)

```

```{r}

#proportion of Q30 bases in barcode, rna read, and umi
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: Barcode (ortho-exon hg38)",y="Q30 Bases in Barcode",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: UMI (ortho-exon hg38)",y="Q30 Bases in UMI",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: RNA (ortho-exon hg38)",y="Q30 Bases in RNA Read",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.Barcode)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: Barcode (ortho-exon panTro6)",y="Q30 Bases in Barcode",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.UMI)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: UMI (ortho-exon panTro6)",y="Q30 Bases in UMI",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Q30.Bases.in.RNA.Read)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion Q30: RNA (ortho-exon panTro6)",y="Q30 Bases in RNA Read",x="Collection") +
                           ylim(0,1))), ncol=3)

```

```{r}

#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=reads_prop), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.to.Genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Exonic.Regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Proportion of Mapped Reads (ortho-exon panTro6)",y="Mapped Reads",x="Collection") +
                           ylim(0,1))), ncol=1)

```

```{r}

#reads mapped to genome, confidently mapped to genome, confidently mapped to exons, and confidently mapped to transcriptome
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Numbers of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection") +
                           ylim(0,300000000)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2')) +
                           geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                           geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Numbers of Mapped Reads (ortho-exon panTro6)",y="Mapped Reads",x="Collection") +
                           ylim(0,300000000))), ncol=1)

```

```{r}

#reads mapped to intergenic regions, intronic regions, and antisense to gene
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Reads Mapped to Intergenic Regions (ortho-exon hg38)",
                                y="Reads Mapped to Intergenic Regions",
                                x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intronic Regions (ortho-exon hg38)",
                                 y="Reads Mapped to Intronic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon hg38)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intergenic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Confidently.to.Intronic.Regions)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intronic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intronic Regions",
                                 x="Collection") +
                            ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'),
                                aes(x=Collection.Name, y=Reads.Mapped.Antisense.to.Gene)) +
                            geom_col(position="dodge") +
                            theme_classic() +
                            theme(axis.text.x=element_text(size=6)) +
                            labs(title="Reads Mapped to Intergenic Regions (ortho-exon panTro6)",
                                 y="Reads Mapped to Intergenic Regions",
                                 x="Collection") +
                            ylim(0,1))), ncol=3)

```

```{r}

#fraction of reads in cells
CombinePlots(plots=list((ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2'), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Fraction of Reads in Cells (ortho-exon hg38)",y="Fraction of Reads in Cells",x="Collection") +
                           ylim(0,1)),
                        (ggplot(subset(metrics,Alignment.Type=='ChimpOrthoV2'), aes(x=Collection.Name, y=Fraction.Reads.in.Cells)) +
                           geom_col(position="dodge") +
                           theme_classic() +
                           theme(axis.text.x=element_text(size=6)) +
                           labs(title="Fraction of Reads in Cells (ortho-exon panTro6)",y="Fraction of Reads in Cells",x="Collection") +
                           ylim(0,1))), ncol=1)

```

```{r}

#Compare alignment proportions
mean(subset(metrics,Alignment.Type=='HumanOrthoV2')$Reads.Mapped.Confidently.to.Transcriptome) #65.17%
mean(subset(metrics,Alignment.Type=='ChimpOrthoV2')$Reads.Mapped.Confidently.to.Transcriptome) #63.41%

#Summarize metrics in plots for figures
metrics.h <- subset(metrics,Alignment.Type=='HumanOrthoV2')
metrics.h$Collection <- c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                          "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                          "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                          "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                          "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                          "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                          "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O")
metrics.h$Collection <- factor(metrics.h$Collection,
                               levels=c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                                        "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                                        "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                                        "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                                        "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                                        "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                                        "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O"))
metrics.c <- subset(metrics,Alignment.Type=='ChimpOrthoV2')
metrics.c$Collection <- c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                          "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                          "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                          "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                          "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                          "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                          "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O")
metrics.c$Collection <- factor(metrics.c$Collection,
                               levels=c("H1-I/C1-I","H1-M/C1-M","H1-O/C1-O",
                                        "H1-r2-I/C1-r2-I","H1-r2-M/C1-r2-M","H1-r2-O/C1-r2-O",
                                        "H2-I/C2-I","H2-M/C2-M","H2-O/C2-O",
                                        "H3-I/C3-I","H3-M/C3-M","H3-O/C3-O",
                                        "H4-I/C4-I","H4-M/C4-M","H4-O/C4-O",
                                        "H5-I/C5-I","H5-M/C5-M","H5-O/C5-O",
                                        "H6-I/C6-I","H6-M/C6-M","H6-O/C6-O"))

```

```{r}

ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

```{r}

ggplot(metrics.c) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_genome), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.5) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: panTro6, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

```{r}

ggplot(metrics.h) +
  geom_col(aes(x=Collection.Name, y=Number.of.Reads), position="dodge", alpha=0.2) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.4) +
  geom_col(aes(x=Collection.Name, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
  theme_classic() +
  theme(axis.text.x=element_text(size=12), axis.text.y=element_text(size=12)) +
  labs(title="Numbers of Mapped Reads (Genome: hg38, Annotation: ortho-exon)",y="Mapped Reads",x="Collection") +
  ylim(0,300000000) +
  scale_x_discrete(labels=c("GHO-1"="H1+C1\nTime 0",
                            "GHO-2"="H1+C1\nTime 1",
                            "GHO-4"="H1+C1\nTime 2",
                            "HOU-10"="H1r2+C1r2\nTime 0",
                            "HOU-12"="H1r2+C1r2\nTime 1",
                            "HOU-16"="H1r2+C1r2\nTime 2",
                            "HOU-1"="H2+C2\nTime 0",
                            "HOU-2"="H2+C2\nTime 1",
                            "HOU-4"="H2+C2\nTime 2",
                            "HOU-9"="H3+C3\nTime 0",
                            "HOU-11"="H3+C3\nTime 1",
                            "HOU-15"="H3+C3\nTime 2",
                            "HOU-5"="H4+C4\nTime 0",
                            "HOU-6"="H4+C4\nTime 1",
                            "HOU-8"="H4+C4\nTime 2",
                            "HOU-17"="H5+C5\nTime 0",
                            "HOU-19"="H5+C5\nTime 1",
                            "HOU-21"="H5+C5\nTime 2",
                            "HOU-18"="H6+C6\nTime 0",
                            "HOU-20"="H6+C6\nTime 1",
                            "HOU-22"="H6+C6\nTime 2"))

```

```{r}

metrics$Stage <- factor(rep(c("Time 0","Time 1","Time 2"),7),
                        levels=c("Time 0","Time 1","Time 2"))

ggplot(subset(metrics,Alignment.Type=='HumanOrthoV2')) +
                                geom_col(aes(x=Stage, y=Number.of.Reads), position="dodge", alpha=0.2) +
                                geom_col(aes(x=Stage, y=reads_mapped_to_genome), position="dodge", alpha=0.3) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_exonic_regions), position="dodge", alpha=0.6) +
                                geom_col(aes(x=Stage, y=reads_mapped_confidently_to_transcriptome), position="dodge", alpha=1) +
                                theme(axis.text.x=element_text(size=6)) +
                                labs(title="Numbers of Mapped Reads (ortho-exon hg38)",y="Mapped Reads",x="Collection")

```

## Conclusions

* Number of total reads sequenced varied slightly across collections
* Number of reads that were retained (confidently mapped to the transcriptome) were similar across collections
* Number of reads that were retained (confidently mapped to the transcriptome) were similar whether using hg38 or panTro6 reference
* 10X collections from Time 2 generally had higher sequencing saturation than from Time 0 or Time 1
* Collection HOU-15 (H3+C3 Time 2) has higher reads/cell, UMIs/cell, and genes/per cell
