---
title: "10x-plot-diffexp"
author: "Genevieve Housman"
date: "October 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Differential Expression

View results of pairwise differential expression tests between humans and chimpanzees within different cell classifications.


```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(edgeR)
library(scran)
library(SingleCellExperiment)
library(purrr)
library(UpSetR)
library(clusterProfiler)
library(org.Hs.eg.db)
library(Rgraphviz)
library(enrichplot)
library(viridis)
library(VennDiagram)

```

### Compare DE gene results across different tests

```{r}

#tot results
compare1 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster50.rds")
compare2 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0.rds")
compare3 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocXscale0.rds")
DEdata.tot <- rbind(compare1,compare2,compare3)
colnames(DEdata.tot)[5] <- "FDR"
DEdata.tot <- as_tibble(DEdata.tot)
rm(compare1,compare2,compare3)

#t2 results using tot assignments
compare1 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.cluster50tot.rds")
compare2 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0tot.rds")
compare3 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocXscale0tot.rds")
DEdata.t2Wtot <- rbind(compare1,compare2,compare3)
colnames(DEdata.t2Wtot)[5] <- "FDR"
DEdata.t2Wtot <- as_tibble(DEdata.t2Wtot)
rm(compare1,compare2,compare3)

#t2 results
compare1 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.cluster.rds")
compare2 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0.rds")
compare3 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocXscale0.rds")
DEdata.t2 <- rbind(compare1,compare2,compare3)
colnames(DEdata.t2)[5] <- "FDR"
DEdata.t2 <- as_tibble(DEdata.t2)
rm(compare1,compare2,compare3)

#tot results using t2 assignments - EXACTLY THE SAME AS T2 RESULTS
compare1 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.clustert2.rds")
compare2 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0t2.rds")
compare3 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocXscale0t2.rds")
DEdata.totWt2 <- rbind(compare1,compare2,compare3)
colnames(DEdata.totWt2)[5] <- "FDR"
DEdata.totWt2 <- as_tibble(DEdata.totWt2)
rm(compare1,compare2,compare3)

```

Osteogenic Clusters (tot res=0.5, t2 res=0.1)

```{r}

#Osteogenic.c1

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c1") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50tot") %>%
    dplyr::filter(cell.subset=="Osteogenic.c1") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c1") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset=="Osteogenic.c1") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#Osteogenic.c2

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c2") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50tot") %>%
    dplyr::filter(cell.subset=="Osteogenic.c2") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c2") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset=="Osteogenic.c2") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#Osteogenic.c3

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c3") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50tot") %>%
    dplyr::filter(cell.subset=="Osteogenic.c3") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c3") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset=="Osteogenic.c5") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#Osteogenic.c5

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c5") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50tot") %>%
    dplyr::filter(cell.subset=="Osteogenic.c5") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset=="Osteogenic.c5") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset=="Osteogenic.c3") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

```

OstAdHoc (scaled integrated data, thresh=0)

```{r}

#preosteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#embedding osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#mineralizing osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#maturing osteocytes

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

```

OstAdHoc.x (scaled integrated data, thresh=0)

```{r}

#preosteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0tot") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="preosteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0tot") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#embedding osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0tot") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="embedding osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#mineralizing osteoblasts

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0tot") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="mineralizing osteoblast") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

#maturing osteocytes

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2Wtot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0tot") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

compare <- full_join(
  DEdata.tot %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  DEdata.t2 %>%
    dplyr::filter(FDR<=0.05) %>%
    dplyr::filter(cell.assign=="ostadhocXscale0") %>%
    dplyr::filter(cell.subset=="maturing osteocyte") %>% 
    dplyr::select(cell.assign,cell.subset,gene),
  by="gene"
)
#compare
table(is.na(compare$cell.assign.x))
table(is.na(compare$cell.assign.y))

```

```{r}

rm(DEdata.tot,DEdata.t2Wtot,DEdata.t2,DEdata.totWt2)

```

### Examine output of DE analyses in more detail

```{r}

#Load DE results

#compare1 <- readRDS("./data/de-data/DEdream.tot.sc.pch20.fml6.nofc.bh.stage.rds")
#compare2 <- readRDS("./data/de-data/DEdream.tot.sc.pch20.fml6.nofc.bh.cluster.rds")
#compare3 <- readRDS("./data/de-data/DEdream.tot.sc.pch20.fml6.nofc.bh.adhocscale0.rds")
#compare4 <- readRDS("./data/de-data/DEdream.tot.sc.pch20.fml6.nofc.bh.cluster50.rds")
#compare5 <- readRDS("./data/de-data/DEdream.t2.sc.pch20.fml6.nofc.bh.cluster50tot.rds")
#compare6 <- readRDS("./data/de-data/DEdream.t2.sc.pch20.fml6.nofc.bh.cluster.rds")
#compare7 <- readRDS("./data/de-data/DEdream.tot.sc.pch20.fml6.nofc.bh.ostadhocscale0.rds")
#compare8 <- readRDS("./data/de-data/DEdream.t2.sc.pch20.fml6.nofc.bh.ostadhocscale0tot.rds")
#compare9 <- readRDS("./data/de-data/DEdream.t2.sc.pch20.fml6.nofc.bh.ostadhocscale0.rds")

compare1 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.stage.rds")
compare2 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster.rds")
compare3 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.adhocscale0.rds")
compare4 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.cluster50.rds")
compare5 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.cluster50tot.rds")
compare6 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.cluster.rds")
compare7 <- readRDS("./data/de-data/DEdream.tot.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0.rds")
compare8 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0tot.rds")
compare9 <- readRDS("./data/de-data/DEdream.t2.pseudo.logcpm0.fml6.nofc.bh.ostadhocscale0.rds")

DEdata <- rbind(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9)
colnames(DEdata)[5] <- "FDR"
DEdata <- as_tibble(DEdata)
rm(compare1,compare2,compare3,compare4,compare5,compare6,compare7,compare8,compare9)

```

```{r}

#Isolate significant DE genes

#DEdataSig <- DEdata %>% filter(FDR<=0.05)
DEdataSig <- DEdata %>% filter(FDR<=0.01)
#DEdataSig <- DEdata %>% filter(abs(logFC)>=1.2 & FDR<=0.01)

```

### Examine Numbers, Overlap, and logFC of DE Genes

Stages of Differentiation at Collection

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Stage",
                    labels=c("Time 0","Time 1","Time 2"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("Time.0"),
                         list("Time.1"),
                         list("Time.2"),
                         list("Time.0","Time.1"),
                         list("Time.1","Time.2"),
                         list("Time.0","Time.2"),
                         list("Time.0","Time.1","Time.2")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))),
                  category.names = c("Time 0","Time 1","Time 2"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("pink","mediumorchid","dodgerblue"),
                  fill=c("pink","mediumorchid","dodgerblue"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Stage",
                    labels=c("Time 0","Time 1","Time 2"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 0")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 1")])) #FDR<0.05: p<2.2e-16 / FDR<0.01: p=3.3e-11
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Time 2")])) #FDR<0.05: p<2.2e-16 / FDR<0.01: p=4.1e-08

```

General Clusters (res=0.05)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds") %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset %in% c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                    values=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","iPSC.c2","iPSC.c3","MSC.c1","Osteogenic.c1","Osteogenic.c2"),
                    values=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

###

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds") %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset %in% c("iPSC.c1","MSC.c1","Osteogenic.c1")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC.c1") %>% dplyr::select(gene), "iPSC.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="MSC.c1") %>% dplyr::select(gene), "MSC.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC.c1"),
                         list("MSC.c1"),
                         list("Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1"),
                         list("MSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","Osteogenic.c1"),
                         list("iPSC.c1","MSC.c1","Osteogenic.c1")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="iPSC.c1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="MSC.c1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene))),
                  category.names = c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("pink","mediumorchid","dodgerblue"),
                  fill=c("pink","mediumorchid","dodgerblue"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("iPSC.c1","MSC.c1","Osteogenic.c1"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="iPSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]))       #FDR<0.05: p<2.2e-16 / FDR<0.01: p=1.1e-09
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")])) #FDR<0.05: p<2.2e-16 / FDR<0.01: p=6.0e-09

```

General Ad Hoc Classification of Cells (scaled integrated data, threshold=0)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="adhocscale0") %>%
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Ad Hoc",
                    labels=c("iPSC","MSC","Osteogenic"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="iPSC") %>% dplyr::select(gene), "iPSC"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="MSC") %>% dplyr::select(gene), "MSC"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene), "Osteogenic"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      intersections=list(list("iPSC"),
                         list("MSC"),
                         list("Osteogenic"),
                         list("iPSC","MSC"),
                         list("MSC","Osteogenic"),
                         list("iPSC","Osteogenic"),
                         list("iPSC","MSC","Osteogenic")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="iPSC") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="MSC") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic") %>% dplyr::select(gene))),
                  category.names = c("iPSC","MSC","Osteogenic"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("pink","mediumorchid","dodgerblue"),
                  fill=c("pink","mediumorchid","dodgerblue"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Ad Hoc",
                    labels=c("iPSC","MSC","Osteogenic"),
                    values=c("pink","mediumorchid","dodgerblue")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="iPSC")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC")]))       #FDR<0.05: p<2.2e-16 / FDR<0.01: p=1.5e-08
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="MSC")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic")])) #FDR<0.05: p<2.2e-16 / FDR<0.01: p=6.2e-08

```

Osteogenic Clusters (tot-cluster50, res=0.5 in tot data)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds") %>%
    dplyr::filter(cell.assign=="cluster50") %>%
    dplyr::filter(cell.subset %in% c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene), "Osteogenic.c5"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
      intersections=list(list("Osteogenic.c1"),
                         list("Osteogenic.c2"),
                         list("Osteogenic.c3"),
                         list("Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2"),
                         list("Osteogenic.c2","Osteogenic.c3"),
                         list("Osteogenic.c3","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                         list("Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene))),
                  category.names = c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")])) #FDR<0.05: p=0.14670 / FDR<0.01: p=0.13240
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")])) #FDR<0.05: p=0.01449 / FDR<0.01: p=0.00360
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c5")])) #FDR<0.05: p=0.00475 / FDR<0.01: NA

```

Osteogenic Clusters (t2-cluster50.tot, res=0.5 in tot data)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="cluster50tot") %>%
    dplyr::filter(cell.subset %in% c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
      intersections=list(list("Osteogenic.c1"),
                         list("Osteogenic.c2"),
                         list("Osteogenic.c3"),
                         list("Osteogenic.c1","Osteogenic.c2"),
                         list("Osteogenic.c2","Osteogenic.c3"),
                         list("Osteogenic.c1","Osteogenic.c3"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene))),
                  category.names = c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")])) #FDR<0.05: p=0.44620 / FDR<0.01: p=0.30010
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")])) #FDR<0.05: p=8.1e-11 / FDR<0.01: p=1.9e-05

```

Osteogenic Clusters (t2-clusters, res=0.1 in t2 data)

```{r}

#note: Osteogenic.c5 is the same as Osteogenic.c3 above (Osteogenic.c3 is the same as Osteogenic.c5 above)

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="cluster") %>%
    dplyr::filter(cell.subset %in% c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5")) %>% 
    dplyr::select(cell.subset,gene,logFC)

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue4","dodgerblue2")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene), "Osteogenic.c1"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene), "Osteogenic.c2"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene), "Osteogenic.c3"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene), "Osteogenic.c5"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
      intersections=list(list("Osteogenic.c1"),
                         list("Osteogenic.c2"),
                         list("Osteogenic.c3"),
                         list("Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2"),
                         list("Osteogenic.c2","Osteogenic.c3"),
                         list("Osteogenic.c3","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c3"),
                         list("Osteogenic.c1","Osteogenic.c5"),
                         list("Osteogenic.c2","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3"),
                         list("Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c3","Osteogenic.c5"),
                         list("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="Osteogenic.c1") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c2") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c3") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="Osteogenic.c5") %>% dplyr::select(gene))),
                  category.names = c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2","dodgerblue4"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Cluster",
                    labels=c("Osteogenic.c1","Osteogenic.c2","Osteogenic.c3","Osteogenic.c5"),
                    values=c("lightblue1","deepskyblue","dodgerblue4","dodgerblue2")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c1")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")])) #FDR<0.05: p=0.95920 / FDR<0.01: p=0.26690
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c2")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c5")])) #FDR<0.05: p=6.7e-07 / FDR<0.01: p=0.00029
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c5")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="Osteogenic.c3")])) #FDR<0.05: p=0.35390 / FDR<0.01: NA

```

Osteogenic Ad Hoc Classification of Cells (tot-ostadhocscale0, scaled integrated data, threshold=0)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))                       #FDR<0.05: p=0.00769 / FDR<0.01: p=0.46880
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))               #FDR<0.05: p<2.2e-16 / FDR<0.01: p=1.0e-06
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")])) #FDR<0.05: p=0.00538 / FDR<0.01: p=0.08167
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))   #FDR<0.05: p=2.2e-05 / FDR<0.01: p=0.00030

```

Osteogenic Ad Hoc Classification of Cells (t2-ostadhocscale0.tot, scaled integrated data, threshold=0)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))                       #FDR<0.05: p=0.00183 / FDR<0.01: p=0.74110
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))               #FDR<0.05: p<2.2e-16 / FDR<0.01: p=4.3e-06
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")])) #FDR<0.05: p=0.08559 / FDR<0.01: p=0.62810
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))   #FDR<0.05: p=1.6e-11 / FDR<0.01: p=9.7e-05

```

Osteogenic Ad Hoc Classification of Cells (t2-ostadhocscale0, scaled integrated data, threshold=0)

```{r}

DEgenes <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0") %>%
    dplyr::select(cell.subset,gene,logFC)

DEgenes$cell.subset <- factor(DEgenes$cell.subset,
                              levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                              labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))

# Number of DE genes
ggplot(DEgenes, aes(x=cell.subset, fill=cell.subset)) +
  geom_bar() +
  ylab('Number of DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

# Overlap of DE genes
combined <- reduce(list(data.frame(gene=DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "preosteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "embedding.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "mineralizing.osteoblast"=1),
                        data.frame(gene=DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "maturing.osteocyte"=1)
                        ), full_join)
combined[is.na(combined)] <- 0
upset(combined,
      keep.order=TRUE,
      empty.intersections="on")
upset(combined,
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
      intersections=list(list("preosteoblast"),
                         list("osteoblast"),
                         list("embedding.osteoblast"),
                         list("mineralizing.osteoblast"),
                         list("maturing.osteocyte"),
                         list("preosteoblast","osteoblast"),
                         list("osteoblast","embedding.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast"),
                         list("mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast"),
                         list("osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                         list("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte")),
      text.scale=2)
dev.off()
v <- venn.diagram(list(pull(DEgenes %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene)),
                       pull(DEgenes %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene))),
                  category.names = c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte"),
                  filename=NULL,
                  scaled=TRUE,
                  euler.d=TRUE,
                  col=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fill=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4"),
                  fontfamily="sans",
                  cat.fontfamily="sans")
grid.draw(v)

# Examine absoute logFC of DE genes
ggplot(DEgenes, aes(x=cell.subset, y=abs(logFC), fill=cell.subset)) +
  geom_boxplot() +
  ylab('Absolute Log2 Fold Change in DE Genes') +
  scale_fill_manual(name="Osteogenic Stage",
                    labels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                    values=c("lightblue1","deepskyblue","dodgerblue2","royalblue3","royalblue4")) +
  theme_classic() +
  theme(axis.text=element_text(size=10), axis.title=element_text(size=10), legend.text=element_text(size=10))

t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="preosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]))                       #FDR<0.05: p=0.28870 / FDR<0.01: p=0.01397
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="osteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]))               #FDR<0.05: p=1.4e-08 / FDR<0.01: p=0.03585
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="embedding\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")])) #FDR<0.05: p=0.83730 / FDR<0.01: p=0.76080
t.test(abs(DEgenes$logFC[which(DEgenes$cell.subset=="mineralizing\nosteoblast")]),abs(DEgenes$logFC[which(DEgenes$cell.subset=="maturing\nosteocyte")]))   #FDR<0.05: p=7.5e-09 / FDR<0.01: p=0.01472

```

### Make plots of interesting candidate DE genes

Identify Genes of Interest for Stages of Differentiation at Collection

```{r}

# DE genes and cell subset overlap
DEgenes.stg <-
  DEdataSig %>%
    dplyr::filter(cell.assign=="stage") %>%
    dplyr::select(cell.subset,gene,logFC)
combined.stg <- reduce(list(data.frame(gene=DEgenes.stg %>% filter(cell.subset=="Time 0") %>% dplyr::select(gene), "Time.0"=1),
                            data.frame(gene=DEgenes.stg %>% filter(cell.subset=="Time 1") %>% dplyr::select(gene), "Time.1"=1),
                            data.frame(gene=DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene), "Time.2"=1)
                            ), full_join)
combined.stg[is.na(combined.stg)] <- 0

# DE Genes Unique to Time 2
ostUniq <- DEgenes.stg %>% 
             filter(cell.subset=="Time 2",
                    gene %in% pull(combined.stg %>% filter(Time.0==0,Time.1==0,Time.2==1) %>% dplyr::select(gene))) %>% 
             arrange(gene)
             #arrange(desc(abs(logFC)))

```

Identify Genes of Interest for Osteogenic Ad Hoc Classification of Cells (t2-ostadhocscale0.tot, scaled integrated data, threshold=0)

```{r, eval=FALSE}

# DE genes and cell subset overlap
DEgenes.ost <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::select(cell.subset,gene,logFC)
DEgenes.ost$cell.subset <- factor(DEgenes.ost$cell.subset,
                                  levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                                  labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))
combined.ost <- reduce(list(data.frame(gene=DEgenes.ost %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "pre"=1),
                            data.frame(gene=DEgenes.ost %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "obl"=1),
                            data.frame(gene=DEgenes.ost %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "emb"=1),
                            data.frame(gene=DEgenes.ost %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "min"=1),
                            data.frame(gene=DEgenes.ost %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "mat"=1)
                            ), full_join)
combined.ost[is.na(combined.ost)] <- 0

# DE Genes Unique to Each Stage of Osteogenesis
preUniq <- DEgenes.ost %>% 
             filter(cell.subset=="preosteoblast",
                    gene %in% pull(combined.ost %>% filter(pre==1,obl==0,emb==0,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
oblUniq <- DEgenes.ost %>% 
             filter(cell.subset=="osteoblast",
                    gene %in% pull(combined.ost %>% filter(pre==0,obl==1,emb==0,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
embUniq <- DEgenes.ost %>% 
             filter(cell.subset=="embedding\nosteoblast",
                    gene %in% pull(combined.ost %>% filter(pre==0,obl==0,emb==1,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
minUniq <- DEgenes.ost %>% 
             filter(cell.subset=="mineralizing\nosteoblast",
                    gene %in% pull(combined.ost %>% filter(pre==0,obl==0,emb==0,min==1,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
matUniq <- DEgenes.ost %>% 
             filter(cell.subset=="maturing\nosteocyte",
                    gene %in% pull(combined.ost %>% filter(pre==0,obl==0,emb==0,min==0,mat==1) %>% dplyr::select(gene))) %>% 
             arrange(gene)

# DE Genes Unique to Each Stage of Osteogenesis that Do Not Overlap with Time 2 Specific DE Genes
preUniqX <- preUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
oblUniqX <- oblUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
embUniqX <- embUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
minUniqX <- minUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
matUniqX <- matUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))

# DE Genes Unique to Time 2 that Do Not Overlap with Stages of Osteogenesis
ostUniqX <- ostUniq %>% 
              filter(!(gene %in% pull(DEgenes.ost %>% dplyr::select(gene))))

```

```{r}

# Isolate genes that were tested across osteogenic cell types
DEdata.ost <-
  DEdata %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::select(cell.subset,gene,logFC,FDR)
DEdata.ost$cell.subset <- factor(DEdata.ost$cell.subset,
                                  levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                                  labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))
combined.ost0 <- reduce(list(data.frame(gene=DEdata.ost %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "pre"=1),
                             data.frame(gene=DEdata.ost %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "obl"=1),
                             data.frame(gene=DEdata.ost %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "emb"=1),
                             data.frame(gene=DEdata.ost %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "min"=1),
                             data.frame(gene=DEdata.ost %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "mat"=1)
                             ), full_join)
combined.ost0[is.na(combined.ost0)] <- 0
tested.ost <- pull(combined.ost0 %>% filter(pre==1,obl==1,emb==1,min==1,mat==1) %>% dplyr::select(gene))

# DE genes and cell subset overlap (only genes that were also tested in all other comparisons)
DEgenes.ost2 <-
  DEdataSig %>%
    dplyr::filter(data=="./../data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds") %>%
    dplyr::filter(cell.assign=="ostadhocscale0tot") %>%
    dplyr::filter(gene %in% tested.ost) %>% 
    dplyr::select(cell.subset,gene,logFC)
DEgenes.ost2$cell.subset <- factor(DEgenes.ost2$cell.subset,
                                   levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte"),
                                   labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte"))
combined.ost2 <- reduce(list(data.frame(gene=DEgenes.ost2 %>% filter(cell.subset=="preosteoblast")            %>% dplyr::select(gene), "pre"=1),
                             data.frame(gene=DEgenes.ost2 %>% filter(cell.subset=="osteoblast")               %>% dplyr::select(gene), "obl"=1),
                             data.frame(gene=DEgenes.ost2 %>% filter(cell.subset=="embedding\nosteoblast")    %>% dplyr::select(gene), "emb"=1),
                             data.frame(gene=DEgenes.ost2 %>% filter(cell.subset=="mineralizing\nosteoblast") %>% dplyr::select(gene), "min"=1),
                             data.frame(gene=DEgenes.ost2 %>% filter(cell.subset=="maturing\nosteocyte")      %>% dplyr::select(gene), "mat"=1)
                             ), full_join)
combined.ost2[is.na(combined.ost2)] <- 0


# DE Genes Unique to Each Stage of Osteogenesis (and that were also tested in all other comparisons)
preUniq <- DEgenes.ost %>% 
             filter(cell.subset=="preosteoblast",
                    gene %in% pull(combined.os2t %>% filter(pre==1,obl==0,emb==0,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
oblUniq <- DEgenes.ost %>% 
             filter(cell.subset=="osteoblast",
                    gene %in% pull(combined.ost2 %>% filter(pre==0,obl==1,emb==0,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
embUniq <- DEgenes.ost %>% 
             filter(cell.subset=="embedding\nosteoblast",
                    gene %in% pull(combined.ost2 %>% filter(pre==0,obl==0,emb==1,min==0,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
minUniq <- DEgenes.ost %>% 
             filter(cell.subset=="mineralizing\nosteoblast",
                    gene %in% pull(combined.ost2 %>% filter(pre==0,obl==0,emb==0,min==1,mat==0) %>% dplyr::select(gene))) %>% 
             arrange(gene)
matUniq <- DEgenes.ost %>% 
             filter(cell.subset=="maturing\nosteocyte",
                    gene %in% pull(combined.ost2 %>% filter(pre==0,obl==0,emb==0,min==0,mat==1) %>% dplyr::select(gene))) %>% 
             arrange(gene)

# DE Genes Unique to Each Stage of Osteogenesis that Do Not Overlap with Time 2 Specific DE Genes
preUniqX <- preUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
oblUniqX <- oblUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
embUniqX <- embUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
minUniqX <- minUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))
matUniqX <- matUniq %>% 
             filter(!(gene %in% pull(DEgenes.stg %>% filter(cell.subset=="Time 2") %>% dplyr::select(gene))))

# DE Genes Unique to Time 2 that Do Not Overlap with Stages of Osteogenesis
ostUniqX <- ostUniq %>% 
              filter(!(gene %in% pull(DEgenes.ost2 %>% dplyr::select(gene))))

```

Identify Genes of Interest from Cormotif Results

```{r}

#Species vs. Stages - 4 comparisons
gene.prob.stg <- readRDS("./output/cormotif.pseudo-stage4.data.filter.log.indv-cell.int19k-tot.rds")

# Examine genes from each motif
prob.motif1  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]<0.5),]))
prob.motif2  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]>0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]>0.5 & gene.prob.stg[,4]<0.5),]))
prob.motif3  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]>0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]>0.5),]))
prob.motif4  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]>0.5 & gene.prob.stg[,2]>0.5 & gene.prob.stg[,3]>0.5 & gene.prob.stg[,4]>0.5),]))

length(prob.motif1)
length(prob.motif2)
length(prob.motif3)
length(prob.motif4)

# Examine human/chimp differences
prob.motifC   <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]>0.5 & gene.prob.stg[,4]>0.5),]))
prob.motifC1  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]>0.5 & gene.prob.stg[,4]<0.5),]))
prob.motifC2  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]>0.5),]))
prob.motifH   <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]>0.5 & gene.prob.stg[,2]>0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]<0.5),]))
prob.motifH1  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]>0.5 & gene.prob.stg[,2]<0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]<0.5),]))
prob.motifH2  <- sort(rownames(gene.prob.stg[(gene.prob.stg[,1]<0.5 & gene.prob.stg[,2]>0.5 & gene.prob.stg[,3]<0.5 & gene.prob.stg[,4]<0.5),]))

length(prob.motifC)
length(prob.motifC1)
length(prob.motifC2)
length(prob.motifH)
length(prob.motifH1)
length(prob.motifH2)

```

Make plots with pseudobulk across stages of differentiation

```{r}

scrna.tot <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds" #integrate across individuals - total (19k genes)
data.tot <- readRDS(scrna.tot)
data.tot@meta.data$Individual.Replicate <- paste0(data.tot@meta.data$Individual,".",data.tot@meta.data$Replicate)

```

```{r}

# Compile log2 normalized gene expression data
genes <- rownames(data.tot@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("Time 0","Time 1","Time 2")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data.tot,subset=Stage==j)
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",j)
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.stg <- dataSub@meta.data$Stage[w][1]
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.stg))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","Stage")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
#keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
#dge$counts <- dge$counts[keep,]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
#log2 transformed cpm
tmm <- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
tmm$gene <- rownames(tmm)
tmm.melt <- melt(tmm,id.vars="gene",measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][1])
tmm.melt$stage <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][2])
tmm.melt$species <- ifelse(grepl("H",tmm.melt$ind.rep),"Human","Chimp")

```

```{r}

# Look at Gene Expression Levels
pdf(file="./output/de-ostuniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in ostUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-ostuniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in ostUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-2-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motif2) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-3-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motif3) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-4-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motif4) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-C1-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motifC1) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-C2-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motifC2) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-H1-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motifH1) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-cormotif-stage-H2-pseudo.pdf", onefile=TRUE, width=3, height=3)
for(i in prob.motifH2) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=stage, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

```

Make plots with pseudobulk across osteogenic ad hoc classifications (t2)

```{r}

scrna.t2 <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-t2.assign.rds" #integrate across individuals - time 2 (19k genes)
data.t2 <- readRDS(scrna.t2)
data.t2@meta.data$Individual.Replicate <- paste0(data.t2@meta.data$Individual,".",data.t2@meta.data$Replicate)

```

```{r}

# Compile log2 normalized gene expression data
genes <- rownames(data.t2@assays$RNA@counts)
genes.mito <- c("MT-ATP6","MT-ATP8","MT-CO1","MT-CO2","MT-CO3","MT-CYB","MT-ND1","MT-ND2","MT-ND3","MT-ND4","MT-ND4L","MT-ND5")
genes.ribo <- grep('^RP',genes,value=T)
genes.no.mito.ribo <- genes[which(!(genes %in% c(genes.mito,genes.ribo)))]
rm(genes,genes.mito,genes.ribo)
#make count matrix and metadata for edgeR object - pseudobulk (pseudocounts, filter genes mean(log2cpm)>0)
cell.subset <- c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte","unknown osteogenic","not osteogenic")
counts <- c()
metadata <- c()
labels <- c()
for(j in cell.subset) {
  dataSub <- subset(data.t2,subset=OstAdHoc.Assign.5gene.thresh0==j)
  #dataSub <- subset(data,subset=(OstAdHoc.Assign.5gene.median==j & Stage=="Time 2"))
  for (i in unique(dataSub@meta.data$Individual.Replicate)) {
    x.lab <- paste0(i,"_",sub(" ",".",j))
    w <- which(dataSub@meta.data$Individual.Replicate==i)
    if (length(w)>0) {
      x.spp <- dataSub@meta.data$Species[w][1]
      x.col <- dataSub@meta.data$Collection[w][1]
      x.ind <- dataSub@meta.data$Individual[w][1]
      x.rep <- dataSub@meta.data$Replicate[w][1]
      x.idr <- dataSub@meta.data$Individual.Replicate[w][1]
      x.adh <- sub(" ",".",as.character(dataSub@meta.data$OstAdHoc.Assign.5gene.thresh0[w][1]))
      if (length(w)==1) {
        x.cnt <- dataSub@assays$RNA@counts[,w]
      } else {
        x.cnt <- Matrix::rowSums(dataSub@assays$RNA@counts[,w])
      }
      counts <- cbind(counts, x.cnt)
      metadata <- rbind(metadata, c(x.lab, x.spp, x.col, x.ind, x.rep, x.idr, x.adh))
      labels <- c(labels, x.lab)
    }
  }
}
colnames(counts) <- labels
rownames(metadata) <- labels
colnames(metadata) <- c("Sample","Species","Collection","Individual","Replicate","Individual.Replicate","AdHoc")
metadata <- as.data.frame(metadata)
#remove mitochodrial and ribosomal genes
counts <- counts[which(rownames(counts) %in% genes.no.mito.ribo),]
#make edgeR object
dge <- DGEList(counts)
meta_dge <- dge$samples[,c("lib.size","norm.factors")]
meta_dge <- cbind(meta_dge, metadata)
dge$samples <- meta_dge
rm(counts,metadata,meta_dge)
#filter genes
#keep <- rowMeans(edgeR::cpm(dge,log=TRUE,prior.count=0.25))>0
#dge$counts <- dge$counts[keep,]
#normalize data
dge <- calcNormFactors(dge, method="TMM")
#log2 transformed cpm
tmm <- as.data.frame(edgeR::cpm(dge,normalized.lib.sizes=TRUE,log=TRUE,prior.count=0.25))
tmm$gene <- rownames(tmm)
tmm.melt <- melt(tmm,id.vars="gene",measure.vars=colnames(tmm)[1:(length(colnames(tmm))-1)])
tmm.melt$ind.rep <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][1])
tmm.melt$ostadhoc <- sapply(tmm.melt$variable,function(x)str_split(x,"_")[[1]][2])
tmm.melt$ostadhoc <- factor(tmm.melt$ostadhoc,
                            levels=c("preosteoblast","osteoblast","embedding.osteoblast","mineralizing.osteoblast","maturing.osteocyte","unknown.osteogenic","not.osteogenic"),
                            labels=c("preosteoblast","osteoblast","embedding\nosteoblast","mineralizing\nosteoblast","maturing\nosteocyte","unknown\nosteogenic","not\nosteogenic"))
tmm.melt$species <- ifelse(grepl("H",tmm.melt$ind.rep),"Human","Chimp")

```

```{r}

# Look at Gene Expression Levels
pdf(file="./output/de-preuniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in preUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-obluniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in oblUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-embuniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in embUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-minuniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in minUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-matuniq-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in matUniq$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

# Look at Gene Expression Levels
pdf(file="./output/de-preuniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in preUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-obluniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in oblUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-embuniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in embUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-minuniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in minUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

pdf(file="./output/de-matuniqX-fdr0.01-pseudo.pdf", onefile=TRUE, width=9, height=3)
for(i in matUniqX$gene) {
  p <- ggplot(tmm.melt[which(tmm.melt$gene==i),],
              aes(x=ostadhoc, y=value, fill=species)) +
         geom_boxplot() +
         labs(title=i, y="Genes Expression", x="") +
         scale_fill_manual(name="", values=c("lightgrey","black")) +
         theme_classic() +
         theme(axis.text=element_text(size=12), axis.title=element_text(size=12), legend.text=element_text(size=12))
  print(p)
}
dev.off()

```

