---
title: "10x-data-classification"
author: "Genevieve Housman"
date: "September 2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 10X Data Cellranger Analysis - Cell Classification

Define and assign cell type classifications.


```{r, message=FALSE}

#Load libraries
library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(pheatmap)
library(grid)
library(gbm)
library(colorspace)
library(RColorBrewer)
library(tidyr)
library(UpSetR)
library(reshape2)
library(CountClust)
library(classtpx)
library(fastTopics)

```

```{r}

#Load data
scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.rds" #integrate across individuals - total (19k genes + regress out UMI/mito)
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds" #integrate across individuals - total (19k genes + regress out UMI/mito)
#scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k-tot.rds" #integrate across individuals - total (19k genes)
data <- readRDS(scrna)

data
head(data@meta.data)

DimPlot(data,
        group.by="Stage",
        reduction="umap",
        cols=c("pink","mediumorchid","dodgerblue")) +
  labs(x="UMAP1",y="UMAP2")

```

### Seurat Clustering

                               tot19k.reg | tot19k
data, k=40, res=1.00 clusters:         24 |     22
data, k=40, res=0.75 clusters:         19 |     18
data, k=40, res=0.50 clusters:         17 |     17
data, k=40, res=0.25 clusters:         12 |     12
data, k=40, res=0.10 clusters:          8 |      8
data, k=40, res=0.05 clusters:          6 |      6


cell counts per cluster:      0 |     1 |     2 |     3 |    4 |    5 |    6 |    7 |    8 |    9 |   10 |   11 |   12 |   13 |   14 |   15 |   16 |   17 |   18 |   19 |  20 |  21 |  22 |  23 |

tot19k.reg.res.1           8988 |  8274 |  8263 |  8042 | 7730 | 7610 | 7342 | 7129 | 5639 | 4203 | 4130 | 4045 | 3821 | 3578 | 3144 | 1945 | 1915 | 1169 | 1041 | 1013 | 963 | 711 | 508 | 331 | 
tot19k.reg.res.0.75       12590 | 12470 | 11640 |  8051 | 7573 | 7513 | 6950 | 5683 | 5548 | 5342 | 4132 | 3800 | 3403 | 1648 | 1499 | 1038 | 1009 |  935 |  710 |
tot19k.reg.res.0.5        12257 | 11687 | 11242 |  9017 | 8764 | 8052 | 7619 | 7500 | 7149 | 6051 | 4131 | 2194 | 1642 | 1484 | 1035 | 1003 |  707 | 
tot19k.reg.res.0.25       27142 | 25838 | 12394 |  9143 | 7578 | 6818 | 4130 | 3290 | 2173 | 1988 |  707 |  333 | 
tot19k.reg.res.0.1        29475 | 29440 | 19067 |  9168 | 7573 | 4128 | 1980 |  703 | 
tot19k.reg.res.0.05       36958 | 29523 | 19205 |  9084 | 6060 |  704 | 

tot19k.res.1              11380 | 10542 |  8445 |  8121 | 7340 | 7337 | 6970 | 5305 | 4818 | 4398 | 4068 | 3874 | 3431 | 3331 | 2514 | 2505 | 1955 | 1643 | 1191 | 1050 | 704 | 612 | 
tot19k.res.0.75           12983 | 12575 | 10964 |  8129 | 8093 | 7667 | 6866 | 6851 | 5993 | 5279 | 4288 | 4075 | 3087 | 2115 | 1047 |  704 |  489 |  329 | 
tot19k.res.0.5            13555 | 11793 | 11540 | 10964 | 8164 | 7545 | 6774 | 6701 | 5895 | 4174 | 3768 | 3432 | 3072 | 2078 | 1046 |  704 |  329 | 
tot19k.res.0.25           25620 | 18457 | 13671 |  9113 | 7252 | 6732 | 5756 | 4260 | 4164 | 3399 | 2406 |  704 | 
tot19k.res.0.1            29031 | 27875 | 19390 |  9136 | 8840 | 4160 | 2398 |  704 | 
tot19k.res.0.05           36719 | 31528 | 19392 |  9130 | 4061 |  704 | 

```{r}

#use dims that explaim more than 0.1% of variance
pva <- data@reductions$pca@stdev^2/data@reductions$pca@misc$total.variance
ndim <- length(which(pva>=0.001))
print(ndim)

#Identify Clusters (increased resolution identifies greater number of clusters)
data <- FindNeighbors(data, k.param=40, dims=1:ndim)
data <- FindClusters(data, resolution=1.00)
data <- FindClusters(data, resolution=0.75)
data <- FindClusters(data, resolution=0.50)
data <- FindClusters(data, resolution=0.25)
data <- FindClusters(data, resolution=0.10)
data <- FindClusters(data, resolution=0.05)

rm(pva,ndim)

```

```{r, eval=FALSE}

table(data@meta.data$integrated_snn_res.1)
table(data@meta.data$integrated_snn_res.0.75)
table(data@meta.data$integrated_snn_res.0.5)
table(data@meta.data$integrated_snn_res.0.25)
table(data@meta.data$integrated_snn_res.0.1)
table(data@meta.data$integrated_snn_res.0.05)
table(data@meta.data$seurat_clusters)

```

Quickly check clusters.

```{r}

x <- "integrated_snn_res.0.05"

#CombinePlots(plots=list((DimPlot(data, group.by=x, reduction="pca", dims=c(1,2)) + labs(x="PC1",y="PC2")),
#                        (DimPlot(data, group.by=x, reduction="pca", dims=c(3,4)) + labs(x="PC3",y="PC4")),
#                        (DimPlot(data, group.by=x, reduction="pca", dims=c(5,6)) + labs(x="PC5",y="PC6")),
#                        (DimPlot(data, group.by=x, reduction="pca", dims=c(7,8)) + labs(x="PC7",y="PC8")),
#                        (DimPlot(data, group.by=x, reduction="pca", dims=c(9,10)) + labs(x="PC9",y="PC10"))), ncol=5)

DimPlot(data, group.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"), split.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Species=="Chimp"), split.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2"))),
             ncol=1)

CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"), split.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 1"), split.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 2"), split.by=x, reduction="umap") + labs(x="UMAP1",y="UMAP2"))),
             ncol=1)

```

##### DO I WANT TO DO THIS: Check expression of gene markers in each cluster. Using these levels of expression, clusters are assigned to stages of osteogenic differentiation.
(SEE LINES 136-223)

Define new metadata label for clusters (resolution=0.05).

```{r}

data <- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name="Cluster")
data$Cluster[which(data$integrated_snn_res.0.05=="1")] <- "iPSC.c1"
data$Cluster[which(data$integrated_snn_res.0.05=="4")] <- "iPSC.c2"
data$Cluster[which(data$integrated_snn_res.0.05=="5")] <- "iPSC.c3"
data$Cluster[which(data$integrated_snn_res.0.05=="0")] <- "MSC.c1"
data$Cluster[which(data$integrated_snn_res.0.05=="2")] <- "Osteogenic.c1"
data$Cluster[which(data$integrated_snn_res.0.05=="3")] <- "Osteogenic.c2"

```


```{r}

data <- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name="Cluster0.25")
data$Cluster0.25[which(data$integrated_snn_res.0.25=="1")]  <- "iPSC.c1"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="7")]  <- "iPSC.c2"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="9")]  <- "iPSC.c3"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="6")]  <- "iPSC.c3"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="11")] <- "iPSC.c4"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="10")] <- "iPSC.c5"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="0")]  <- "MSC.c1"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="4")]  <- "MSC.c2"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="8")]  <- "MSC.c3"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="5")]  <- "Osteogenic.c1"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="2")]  <- "Osteogenic.c2"
data$Cluster0.25[which(data$integrated_snn_res.0.25=="3")]  <- "Osteogenic.c3"

data <- AddMetaData(data, rep(NA,length(rownames(data@meta.data))), col.name="Cluster0.5")
data$Cluster0.5[which(data$integrated_snn_res.0.5=="1")]  <- "iPSC.c1"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="6")]  <- "iPSC.c2"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="3")]  <- "iPSC.c3"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="12")] <- "iPSC.c3"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="13")] <- "iPSC.c4"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="10")] <- "iPSC.c5"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="16")] <- "iPSC.c6"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="2")]  <- "MSC.c1"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="4")]  <- "MSC.c2"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="8")]  <- "MSC.c3"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="7")]  <- "MSC.c4"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="11")] <- "MSC.c5"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="9")]  <- "Osteogenic.c1"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="0")]  <- "Osteogenic.c2"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="5")]  <- "Osteogenic.c3"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="14")] <- "Osteogenic.c4"
data$Cluster0.5[which(data$integrated_snn_res.0.5=="15")] <- "Osteogenic.c5"

```

Calculate numbers of cells in each cluster given particular parameters.

                    iPSC.c1 | iPSC.c2 | iPSC.c3 | MSC.c1 | Osteogenic.c1 | Osteogenic.c2 |

tot19k.reg.res.0.05   29523 |    6060 |     704 |  36958 |         19205 |          9084 | 
Human:                13265 |    2987 |     439 |  17496 |          8227 |          5558 | 
Chimp:                16258 |    3073 |     265 |  19462 |         10978 |          3526 |
Time 0:               29523 |    6057 |     704 |      5 |             0 |             5 |  
Time 1:                   0 |       3 |       0 |  36044 |           525 |          1517 |  
Time 2:                   0 |       0 |       0 |    909 |         18680 |          7562 | 
Human Time 0:         13265 |    2984 |     439 |      5 |             0 |             1 |
Human Time 1:             0 |       3 |       0 |  17294 |           417 |           999 | 
Human Time 2:             0 |       0 |       0 |    197 |          7810 |          4558 | 
Chimp Time 0:         16258 |    3073 |     265 |      0 |             0 |             4 | 
Chimp Time 1:             0 |       0 |       0 |  18750 |           108 |           518 | 
Chimp Time 2:             0 |       0 |       0 |    712 |         10870 |          3004 | 

tot19k.res.0.05       31528 |    4061 |     704 |  36719 |         19392 |          9130 | 
Human:                14346 |    1908 |     440 |  17340 |          8341 |          5597 | 
Chimp:                17182 |    2153 |     264 |  19379 |         11051 |          3533 |
Time 0:               31528 |    4058 |     704 |      2 |             1 |             1 |  
Time 1:                   3 |       0 |       0 |  35867 |           663 |          1556 |  
Time 2:                   0 |       0 |       0 |    850 |         18728 |          7573 | 
Human Time 0:         14346 |    1905 |     440 |      2 |             1 |             0 |
Human Time 1:             0 |       3 |       0 |  17173 |           512 |          1025 | 
Human Time 2:             0 |       0 |       0 |    165 |          7828 |          4572 | 
Chimp Time 0:         17182 |    2153 |     264 |      0 |             0 |             1 | 
Chimp Time 1:             0 |       0 |       0 |  18694 |           151 |           531 | 
Chimp Time 2:             0 |       0 |       0 |    685 |         10900 |          3001 | 

```{r, eval=FALSE}

table(data@meta.data$Cluster)
table(subset(data,subset=Species=="Human")@meta.data$Cluster)
table(subset(data,subset=Species=="Chimp")@meta.data$Cluster)
table(subset(data,subset=Stage=="Time 0")@meta.data$Cluster)
table(subset(data,subset=Stage=="Time 1")@meta.data$Cluster)
table(subset(data,subset=Stage=="Time 2")@meta.data$Cluster)
table(subset(data,subset=(Species=="Human" & Stage=="Time 0"))@meta.data$Cluster)
table(subset(data,subset=(Species=="Human" & Stage=="Time 1"))@meta.data$Cluster)
table(subset(data,subset=(Species=="Human" & Stage=="Time 2"))@meta.data$Cluster)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 0"))@meta.data$Cluster)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 1"))@meta.data$Cluster)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 2"))@meta.data$Cluster)

```

Make nicer plots of clusters.

```{r}

DimPlot(data,
        group.by="Cluster",
        reduction="umap",
        cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
  labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(data,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(1,2),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="PC1",y="PC2")),
                        (DimPlot(data,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(3,4),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="PC3",y="PC4")),
                        (DimPlot(data,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(5,6),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="PC5",y="PC6")),
                        (DimPlot(data,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(7,8),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="PC7",y="PC8")),
                        (DimPlot(data,
                                 group.by="Cluster",
                                 reduction="pca",
                                 dims=c(9,10),
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="PC9",y="PC10"))), ncol=5)

CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Species=="Chimp"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=2)

CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("pink","pink3","pink4","mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 1"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 2"),
                                 group.by="Cluster",
                                 reduction="umap",
                                 cols=c("mediumorchid","dodgerblue","dodgerblue4")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=3)

```

Identify markers defining each cluster.

```{r, eval=FALSE}

#Find markers for every cluster compared to all remaining cells, report only the positive ones
markers <- FindAllMarkers(data, only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25)
markers %>% group_by(cluster) %>% top_n(n=5, wt=avg_logFC)
top5 <- markers %>% group_by(cluster) %>% top_n(n=5, wt=avg_logFC)
DoHeatmap(data, features=top5$gene) + NoLegend()

saveRDS(markers, file="./output/markers.data.filter.log.indv-cell.int19k.reg-tot.rds")
#saveRDS(markers, file="./output/markers.data.filter.log.indv-cell.int19k-tot.rds")

VlnPlot(data, features=top5$gene[1:5], group.by="Cluster", slot="data", pt.size=0, ncol=5)   #cluster 0 (MSC.c1): MMP1, LGALS1, TAGLN2, RGS4, VIM
VlnPlot(data, features=top5$gene[6:10], group.by="Cluster", slot="data", pt.size=0, ncol=5)  #cluster 1 (iPSC.c1): CD24, APOE, POU5F1, L1TD1, GAL
VlnPlot(data, features=top5$gene[11:15], group.by="Cluster", slot="data", pt.size=0, ncol=5) #cluster 2 (Osteogenic.c1): COL1A1, CTSD, COL1A2, POSTN, NUPR1
VlnPlot(data, features=top5$gene[16:20], group.by="Cluster", slot="data", pt.size=0, ncol=5) #cluster 3 (Osteogenic.c2): CCND1, MTRNR2L8, NUPR1, TIMP1, CCND2
VlnPlot(data, features=top5$gene[21:25], group.by="Cluster", slot="data", pt.size=0, ncol=5) #cluster 4 (iPSC.c2): POU5F1, HMGA1, TUBB2B, CD24, SNRPN
VlnPlot(data, features=top5$gene[26:30], group.by="Cluster", slot="data", pt.size=0, ncol=5) #cluster 5 (iPSC.c3): TTYH1, CNTNAP2, SOX2, DLK1, FABP7


#Find markers for groups of clusters to all remaining cells
markers.i <- FindMarkers(data, ident.1=c(1,4,5), ident.2=c(0,2,3), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #ipsc
markers.m <- FindMarkers(data, ident.1=c(0), ident.2=c(1,2,3,4,5), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #msc
markers.o <- FindMarkers(data, ident.1=c(2,3), ident.2=c(0,1,4,5), only.pos=TRUE, min.pct=0.25, logfc.threshold=0.25) #osteo

VlnPlot(data, features=rownames(markers.i)[1:5], group.by="Cluster", slot="data", pt.size=0, ncol=5) #CD24, POU5F1, APOE, L1TD1, TUBB2B
VlnPlot(data, features=rownames(markers.m)[1:5], group.by="Cluster", slot="data", pt.size=0, ncol=5) #MMP1, LGALS1, TAGLN2, RGS4, VIM
VlnPlot(data, features=rownames(markers.o)[1:5], group.by="Cluster", slot="data", pt.size=0, ncol=5) #NUPR1, TIMP1, IGFBP4, IFI6, CTSD

```

### Grades of Membership (Topic Modeling - unsupervised)

Alternative way to look at data structure. A topic model explains each sample (an expression profile) as a mixture of “gene programs” that are estimated from the data. In a topic modeling analysis, the gene expression programs are the topics. Overall, the topic modeling analysis retains the benefits of the clustering analysis while also revealing additional substructure that cannot be identified by standard clustering methods. We are interested in identifying which topics show major differences across groups of cells (what genes/pathways are enriched in these topics).

* fastTopics
  + This analytical method implements fast optimization algorithms that fit topic models and Poisson non-negative matrix factorizations to scRNA-seq count data. The methods exploit the close relationship between topic modeling and Poisson non-negative matrix factorization (NMF).
  + fit_poisson_nmf computes maximum-likelihood estimates of the Poisson NMF parameters and stores these parameters into a loadings matrix (fit$L) and a factors matrix (fit$F).
  + poisson2multinom recovers the multinomial topic model as a reparameterization of the Poisson NMF model
* CountClust
  + FitGoM fits a grade of membership model to count data using the topics() function implemented in Matt Taddy's maptpx pacakge

Reference: https://github.com/stephenslab/fastTopics

```{r, eval=FALSE, echo=TRUE}

#Load libraries
library(Seurat)
library(CountClust)
library(classtpx)
library(fastTopics)

#Load data
scrna <- "./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.rds"
data <- readRDS(scrna)

#Prepare data (unintegrated count data)
counts <- as.matrix(data@assays$RNA@counts) #do not use integrated data
counts <- counts[rowSums(counts>0)>0,]      #remove genes with 0 counts across cells
counts <- t(as.matrix(counts))              #fix matrix orientation (barcodes x features)
rm(data)

#Batch correct counts - don't correct for species
#labelBatch1 <- as.factor(paste0(data@meta.data$Pair,data@meta.data$Replicate)[which(rownames(data@meta.data) %in% rownames(counts))])
#countsBC1 <- BatchCorrectedCounts(counts, labelBatch1, use_parallel=FALSE)
#rm(data)

#Fit fastTopics model and save data
fit <- fit_poisson_nmf(counts, k=7, numiter=100)
saveRDS(fit, file="./data/gom-data/topics7.log.indv-cell.int19k.reg-tot.rds")
#fit <- fit_poisson_nmf(countsBC1, k=3, numiter=100)
#saveRDS(fit, file="./data/gom-data/topics3.log.indv-cell.int19k.reg-tot.bc1.rds")

#Fit GoM model (OLD METHOD) and save data
fit <- CountClust::FitGoM(counts, K=7, tol=1, control=list(tmax=100))
saveRDS(fit, file="./data/gom-data/gom7.log.indv-cell.int19k.reg-tot.rds")

```

```{bash}
#easties to run these scripts on the cluster
cd /project2/gilad/ghousman/skeletal-human-chimp/human-chimp-skeletal-scRNA/code/

sbatch ./GoM.sh 3 tot
sbatch ./GoM.sh 7 tot #error

sbatch ./fastTopics.sh 3 tot
sbatch ./fastTopics.sh 4 tot
sbatch ./fastTopics.sh 5 tot
sbatch ./fastTopics.sh 6 tot
sbatch ./fastTopics.sh 7 tot
sbatch ./fastTopics.sh 10 tot
sbatch ./fastTopics.sh 15 tot
sbatch ./fastTopics.sh 20 tot
sbatch ./fastTopics.sh 30 tot
sbatch ./fastTopics.sh 50 tot

sbatch ./fastTopicsSpp.sh 3 tot
sbatch ./fastTopicsSpp.sh 4 tot
sbatch ./fastTopicsSpp.sh 5 tot
sbatch ./fastTopicsSpp.sh 6 tot
sbatch ./fastTopicsSpp.sh 7 tot
sbatch ./fastTopicsSpp.sh 10 tot
sbatch ./fastTopicsSpp.sh 15 tot
sbatch ./fastTopicsSpp.sh 20 tot
sbatch ./fastTopicsSpp.sh 30 tot
sbatch ./fastTopicsSpp.sh 50 tot

sbatch ./fastTopicsBC.sh 3 tot
sbatch ./fastTopicsBC.sh 4 tot
sbatch ./fastTopicsBC.sh 5 tot
sbatch ./fastTopicsBC.sh 6 tot
sbatch ./fastTopicsBC.sh 7 tot
sbatch ./fastTopicsBC.sh 10 tot
sbatch ./fastTopicsBC.sh 15 tot
sbatch ./fastTopicsBC.sh 20 tot
sbatch ./fastTopicsBC.sh 30 tot
sbatch ./fastTopicsBC.sh 50 tot

sbatch ./fastTopicsSppBC.sh 3 tot
sbatch ./fastTopicsSppBC.sh 4 tot
sbatch ./fastTopicsSppBC.sh 5 tot
sbatch ./fastTopicsSppBC.sh 6 tot
sbatch ./fastTopicsSppBC.sh 7 tot
sbatch ./fastTopicsSppBC.sh 10 tot
sbatch ./fastTopicsSppBC.sh 15 tot
sbatch ./fastTopicsSppBC.sh 20 tot
sbatch ./fastTopicsSppBC.sh 30 tot
sbatch ./fastTopicsSppBC.sh 50 tot

```

```{r}

#GoM Functions

plot_structure_plot <- function(omega, labels){
     annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=labels)
     rownames(omega) <- annotation$sample_id
     StructureGGplot(omega=omega, annotation=annotation, order_sample=TRUE,
                     axis_tick=list(axis_ticks_length = .1,
                                    axis_ticks_lwd_y = .1,
                                    axis_ticks_lwd_x = .1,
                                    axis_label_size = 7,
                                    axis_label_face = "bold"))
     }

make_topic_plots <- function(omega, labels, angle) {
  i=1
  omega <- as.data.frame(omega)
  colNames <- names(omega)
  omega <- cbind(omega,labels)
  names(omega)
  p <- list()
  for (k in colNames) {
    p[[i]] <- ggplot(data=omega, mapping=aes(x=labels)) +
      geom_violin(mapping=aes_string(y=k),width=1,fill="lightgrey",color="lightgrey") +
      geom_boxplot(mapping=aes_string(y=k),width=0.1,outlier.shape=NA, alpha=0.2) +
      labs(y=paste0("Topic ", k, " value"), x=paste0("")) +
      theme_classic() +
      theme(axis.text.x=element_text(angle=angle, hjust=0, vjust=0))
    i=i+1
  }
  do.call(grid.arrange,c(p,ncol=1))
}

annotate_clusters <- function(theta_mat){
  top_features <- ExtractTopFeatures(theta_mat, top_features=100, method="poisson", options="min")
  gene_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) rownames(theta_mat)[top_features[[1]][x,]]))
  scores_vector <- do.call(rbind, lapply(1:dim(top_features[[1]])[1], function(x) top_features[[2]][x,]))
  return(list(gene_vector, scores_vector))
}

```

Assess fastTopics results

```{r, eval=FALSE}

#Check BIC

numclust <- 6
numclust <- "6.human"
numclust <- "6.chimp"

#Read fastTopics data
fit <- readRDS(paste0("./data/gom-data/topics",numclust,".log.indv-cell.int19k.reg-tot.rds"))
fit <- readRDS(paste0("./data/gom-data/topics",numclust,".log.indv-cell.int19k.reg-tot.bc1.rds"))

#Compute omega (weights, cells x k) and theta (topics, genes x k) (need to add up to 1)
l <- fit$L
f <- fit$F
omega <- sweep(l, MARGIN=2, colSums(f), `*`)
scale <- rowSums(omega)
omega <- omega / scale
theta <- f/colSums(f)

#BIC
loglik <- fit$progress$loglik[length(fit$progress$loglik)]
BIC <- -2*loglik + nrow(theta)*(ncol(theta)-1)*log(nrow(omega))
BIC

```

```{r}

#Plot BIC

par(mfrow=c(1,1))

#fastTopics BIC
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(2283025929,2247075247,2213763940,2185892949,2165452542,2116117855,2068051067,2035942645,1992928855,1955894217)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

#fastTopics.Human BIC
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(1088036999,1070043976,1055669438,1043500728,1029463798,1008507969,984980174,970892139,956580403,943899385)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

#fastTopics.Chimp BIC
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(1125119471,1101887355,1088917452,1079148382,1069878432,1048494671,1027287717,1015562562,1001838696,990882542)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

#fastTopics BIC - batch corrected
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(2245510167,2210485273,2176520656,2151057930,2134684345,2091974751,2049689737,2024091433,1988674503,1953253816)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

#fastTopics.Human BIC - batch corrected
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(1057255003,1043136501,1026386422,1019880111,1012852663,997914823,981717709,971139105,959363245,948201355)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

#fastTopics.Chimp BIC - batch corrected
x=c(3,4,5,6,7,10,15,20,30,50)
y=c(1092537173,1077076072,1066632327,1058052552,1051357116,1036910100,1022111755,1012964037,1000575470,991214123)
plot(x,y,xlab="Topics (k)",ylab="BIC")
lines(x[order(x)],y[order(x)],xlim=range(x),ylim=range(y),pch=16)

```

```{r}

#Examine structure plots and gene contributions to topics

numclust <- 6
numclust <- "6.human"
numclust <- "6.chimp"

#colclust <- c("pink","mediumorchid","dodgerblue") #k=3
#colclust <- c("pink","mediumorchid","orchid4","dodgerblue") #k=4
#colclust <- c("pink","mediumorchid","dodgerblue1","dodgerblue4") #k=4
#colclust <- c("pink","pink4","mediumorchid","orchid4") #k=4
#colclust <- c("pink","mediumorchid","dodgerblue","grey") #k=4
#colclust <- c("pink","pink4","mediumorchid","orchid4","dodgerblue") #k=5
#colclust <- c("pink","pink4","mediumorchid","dodgerblue1","dodgerblue4") #k=5
#colclust <- c("pink","mediumorchid","orchid4","dodgerblue1","dodgerblue4") #k=5
#colclust <- c("pink","pink4","mediumorchid","dodgerblue","grey") #k=5
#colclust <- c("pink","mediumorchid","orchid4","dodgerblue","grey") #k=5
colclust <- c("pink","pink4","mediumorchid","orchid4","dodgerblue1","dodgerblue4") #k=6
colclust <- c("pink","pink4","mediumorchid","dodgerblue1","dodgerblue1","grey") #k=6
colclust <- c("pink","pink4","mediumorchid","dodgerblue1","dodgerblue4","grey") #k=6
colclust <- c("pink","pink2","pink4","mediumorchid","dodgerblue1","dodgerblue4") #k=6
colclust <- c("pink","pink4","mediumorchid","orchid4","dodgerblue","grey") #k=6
#colclust <- c("pink","pink4","mediumorchid","orchid4","dodgerblue1","dodgerblue4","grey") #k=7

#Read fastTopics data
fit <- readRDS(paste0("./data/gom-data/topics",numclust,".log.indv-cell.int19k.reg-tot.rds"))
#fit <- readRDS(paste0("./data/gom-data/topics",numclust,".log.indv-cell.int19k.reg-tot.bc1.rds"))

#Check fitting procedure
quickplot(x=1:length(fit$progress$loglik),
          y=fit$progress$loglik,
          geom=c("point","line"),
          xlab="iteration",
          ylab="loglik")

#Compute omega (weights, cells x k) and theta (topics, genes x k) (need to add up to 1)
l <- fit$L
f <- fit$F
omega <- sweep(l, MARGIN=2, colSums(f), `*`)
scale <- rowSums(omega)
omega <- omega / scale
theta <- f/colSums(f)

#BIC
loglik <- fit$progress$loglik[length(fit$progress$loglik)]
BIC <- -2*loglik + nrow(theta)*(ncol(theta)-1)*log(nrow(omega))
BIC

#Recover the multinomial topic model
fit.multinom <- poisson2multinom(fit)
range(rowSums(fit.multinom$L))
range(colSums(fit.multinom$F))

#Topic annotations
genes <- annotate_clusters(theta_mat=theta)
genes[[1]][,1:10]
saveRDS(genes, file=paste0("./data/gom-data/topics",numclust,".100genes.log.indv-cell.int19k.reg-tot.rds"))
#saveRDS(genes, file=paste0("./data/gom-data/topics",numclust,".100genes.log.indv-cell.int19k.reg-tot.bc1.rds"))

#Create labels
labels <- sapply(strsplit(rownames(omega),"_"), `[`, 1)
annotation1 <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(labels))
celltype <- labels
celltype[celltype %in% c("C1.I","C1r2.I","C2.I","C3.I","C4.I","C5.I","C6.I")] <- "Chimp.iPSC"
celltype[celltype %in% c("C1.M","C1r2.M","C2.M","C3.M","C4.M","C5.M","C6.M")] <- "Chimp.MSC"
celltype[celltype %in% c("C1.O","C1r2.O","C2.O","C3.O","C4.O","C5.O","C6.O")] <- "Chimp.Osteoblast"
celltype[celltype %in% c("H1.I","H1r2.I","H2.I","H3.I","H4.I","H5.I","H6.I")] <- "Human.iPSC"
celltype[celltype %in% c("H1.M","H1r2.M","H2.M","H3.M","H4.M","H5.M","H6.M")] <- "Human.MSC"
celltype[celltype %in% c("H1.O","H1r2.O","H2.O","H3.O","H4.O","H5.O","H6.O")] <- "Human.Osteoblast"
annotation2 <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(celltype))

#Reorder k
omega <- omega[,c(3,2,1)] #k=3, data=tot,tot.BC,tot-h.BC
omega <- omega[,c(3,1,2)] #k=3, data=tot-h,tot-c,tot-c.BC
omega <- omega[,c(4,3,2,1)] #k=4, data=tot,tot-h
omega <- omega[,c(4,1,3,2)] #k=4, data=tot-c
omega <- omega[,c(1,2,4,3)] #k=4, data=tot.BC,tot-c.BC
omega <- omega[,c(1,3,4,2)] #k=4, data=tot-h.BC
omega <- omega[,c(3,2,5,4,1)] #k=5, data=tot
omega <- omega[,c(2,1,5,4,3)] #k=5, data=tot-h
omega <- omega[,c(5,3,1,4,2)] #k=5, data=tot-c
omega <- omega[,c(4,5,2,3,1)] #k=5, data=tot.BC
omega <- omega[,c(4,3,5,1,2)] #k=5, data=tot-h.BC
omega <- omega[,c(4,5,1,3,2)] #k=5, data=tot-c.BC
omega <- omega[,c(4,5,6,2,1,3)] #k=6, data=tot
omega <- omega[,c(3,2,4,5,6,1)] #k=6, data=tot-h
omega <- omega[,c(6,5,1,4,3,2)] #k=6, data=tot-c
omega <- omega[,c(4,2,6,1,5,3)] #k=6, data=tot.BC
omega <- omega[,c(1,6,5,4,2,3)] #k=6, data=tot-h.BC
omega <- omega[,c(5,3,6,2,4,1)] #k=6, data=tot-c.BC
omega <- omega[,c(7,1,4,6,5,3,2)] #k=7, data=tot
omega <- omega[,c(1,4,3,6,7,5,2)] #k=7, data=tot-h
omega <- omega[,c(7,6,5,1,4,2,3)] #k=7, data=tot-c
omega <- omega[,c(6,7,5,3,4,1,2)] #k=7, data=tot.BC
omega <- omega[,c(7,3,2,5,1,4,6)] #k=7, data=tot-h.BC
omega <- omega[,c(7,4,6,2,1,3,5)] #k=7, data=tot-c.BC

#Plot of topic membership
make_topic_plots(omega=omega, labels=labels, angle=-45)
make_topic_plots(omega=omega, labels=celltype, angle=-20)

#Structure plots
#plot_structure_plot(omega=omega, labels=labels)
#plot_structure_plot(omega=omega, labels=celltype)
rownames(omega) <- annotation1$sample_id
CountClust::StructureGGplot(omega=omega,
                annotation=annotation1,
                palette=colclust,
                yaxis_label="Samples",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))
rownames(omega) <- annotation2$sample_id
CountClust::StructureGGplot(omega=omega,
                annotation=annotation2,
                palette=colclust,
                yaxis_label="Species",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

#Heatmap of topic weights (omega)
annotation <- data.frame(row.names=paste0("X", c(1:NROW(omega))), labels=as.factor(labels))
rownames(omega) <- rownames(annotation)
pheatmap(omega,
         cluster_cols=FALSE,
         cluster_rows=FALSE,
         annotation_row=annotation,
         show_rownames=FALSE,
         main="Heatmap of topic weights")
annotation <- data.frame(row.names=paste0("X", c(1:NROW(omega))), labels=as.factor(celltype))
rownames(omega) <- rownames(annotation)
pheatmap(omega,
         cluster_cols=FALSE,
         cluster_rows=FALSE,
         annotation_row=annotation,
         show_rownames=FALSE,
         main="Heatmap of topic weights")
#pheatmap(omega,
#         cluster_cols=FALSE,
#         cluster_rows=TRUE,
#         annotation_row=annotation,
#         show_rownames=FALSE,
#         main="Heatmap of topic weights")

```

Assess CountClust results - Do not use these moving forward.

```{r, eval=FALSE}

numclust <- 3
colclust <- c("pink","mediumorchid","dodgerblue")

#Read GoM data
fit <- readRDS(paste0("./data/gom-data/gom",numclust,".log.indv-cell.int19k.reg-tot.rds"))

#BIC
fit$BIC

#Create labels
omega <- as.matrix(fit$fit$omega)
colnames(omega) <- paste0("k",1:ncol(omega))
labels <- sapply(strsplit(rownames(omega),"_"), `[`, 1)
celltype <- labels
celltype[celltype %in% c("C1.I","C1r2.I","C2.I","C3.I","C4.I","C5.I","C6.I")] <- "Chimp.iPSC"
celltype[celltype %in% c("C1.M","C1r2.M","C2.M","C3.M","C4.M","C5.M","C6.M")] <- "Chimp.MSC"
celltype[celltype %in% c("C1.O","C1r2.O","C2.O","C3.O","C4.O","C5.O","C6.O")] <- "Chimp.Osteoblast"
celltype[celltype %in% c("H1.I","H1r2.I","H2.I","H3.I","H4.I","H5.I","H6.I")] <- "Human.iPSC"
celltype[celltype %in% c("H1.M","H1r2.M","H2.M","H3.M","H4.M","H5.M","H6.M")] <- "Human.MSC"
celltype[celltype %in% c("H1.O","H1r2.O","H2.O","H3.O","H4.O","H5.O","H6.O")] <- "Human.Osteoblast"
annotation <- data.frame(sample_id=paste0("X", c(1:NROW(omega))), tissue_label=as.factor(celltype))

#Plot of topic membership
make_topic_plots(omega=omega, labels=labels)
make_topic_plots(omega=omega, labels=celltype)

#Structure plots
plot_structure_plot(omega=omega, labels=labels)
rownames(omega) <- annotation$sample_id
CountClust::StructureGGplot(omega=omega,
                annotation=annotation,
                palette=colclust,
                yaxis_label="Tissue/Cell Type",
                order_sample=TRUE,
                axis_tick=list(axis_ticks_length=0.1,
                               axis_ticks_lwd_y=0.1,
                               axis_ticks_lwd_x=0.1,
                               axis_label_size=7,
                               axis_label_face="bold"))

#Topic annotations
genes <- annotate_clusters(theta_mat=fit$fit$theta)
genes[[1]][,1:10]
saveRDS(genes, file=paste0("./data/gom-data/gom",numclust,".100genes.log.indv-cell.int19k.reg-tot.rds"))

```

Assess GO Functional Enrichment of Genes Defining Each Topic

```{r}

#Load libraries for GO enrichment analysis
library(clusterProfiler)
library(org.Hs.eg.db)
library(viridis)

#Define functions to assess GO functional enrichment
goEnrich <- function(totGenes, deGenes, ontology) {
  ref.df <- bitr(totGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  gene.df <- bitr(deGenes, fromType="SYMBOL", toType=c("ENSEMBL","ENTREZID"), OrgDb=org.Hs.eg.db)
  ego <- enrichGO(gene          = gene.df$ENSEMBL,
                  OrgDb         = org.Hs.eg.db,
                  universe      = ref.df$ENSEMBL,
                  keyType       = 'ENSEMBL',
                  ont           = ontology,
                  pAdjustMethod = "BH",
                  pvalueCutoff  = 0.01,
                  qvalueCutoff  = 0.05,
                  readable      = TRUE)
  return(ego)
}

```

```{r}

#Get total genes included in topics modeling
counts <- as.matrix(data@assays$RNA@counts) #do not use integrated data
counts <- counts[rowSums(counts>0)>0,]      #remove genes with 0 counts across cells
geneTot <- rownames(counts)

dataTmp <- subset(data, subset=Species=="Human")
counts <- as.matrix(dataTmp@assays$RNA@counts) #do not use integrated data
counts <- counts[rowSums(counts>0)>0,]         #remove genes with 0 counts across cells
geneTotH <- rownames(counts)

dataTmp <- subset(data, subset=Species=="Chimp")
counts <- as.matrix(dataTmp@assays$RNA@counts) #do not use integrated data
counts <- counts[rowSums(counts>0)>0,]         #remove genes with 0 counts across cells
geneTotC <- rownames(counts)

rm(dataTmp,counts)


#Get top 100 genes defining topics
numclust <- 7
numclust <- "7.human"
numclust <- "7.chimp"

#genes <- readRDS(paste0("./data/gom-data/topics",numclust,".100genes.log.indv-cell.int19k.reg-tot.rds"))
genes <- readRDS(paste0("./data/gom-data/topics",numclust,".100genes.log.indv-cell.int19k.reg-tot.bc1.rds"))


#Test for functional enrichment
goTot <- goEnrich(totGenes=geneTot, deGenes=genes[[1]][1,], ontology="ALL")

goTot@result$Description <- str_wrap(goTot@result$Description, width=30)
ggplot(goTot, aes(x=Count, y=Description, color=p.adjust, size=Count), split='ONTOLOGY') +
  geom_point() +
  theme(axis.text.x=element_text(angle=-40, hjust=0)) +
  facet_grid(ONTOLOGY~., scale="free") +
  scale_size_area(max_size=6) +
  scale_color_viridis(option="C") +
  theme_bw() +
  labs(title="GoM Genes")

```



### Ad Hoc Assignments

```{r, eval=FALSE}

#Look at candidate genes

#Time 0
#diagnostic: POU5F1 (OCT4), TDGF1, SOX2, EPCAM
#additional: NANOG, ALPL, STAT3, PROM1 (CD133), PODXL (TRA-1-60R), LIN28A, ZFP42, UTF1, MYC, BMP4, FGF2, TGFB1, DNMT3B, SALL4
#if needed: FOXD3, FUT4 (SSEA-1), KLF2, KLF4, KLF5, INHBA (Activin A), LIF, PRDM14
genes <- c("POU5F1","TDGF1","SOX2","EPCAM","NANOG","ALPL","STAT3","PROM1","PODXL","LIN28A","ZFP42","UTF1","MYC","BMP4","FGF2","TGFB1","DNMT3B",
           "SALL4","FOXD3","FUT4","KLF2","KLF4","KLF5","INHBA","LIF","PRDM14")
data@assays$integrated[genes,1:5]
FeaturePlot(data, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=4)

#Time 1
#diagnostic: THY1 (CD90), NT5E (CD73), ENG (CD105), CD44
#additional: ANPEP (CD13), NGFR (CD271), ITGB1 (CD29), NCAM1 (CD56)
#if needed: CD9, CD14, MME (CD10), ITGAL (CD11A), ITGAX (CD11C)*, CD36, ITGA3 (CD49C), ITGA4 (CD49D), ITGAV (CD51), ICAM1 (CD54), CD55, CD59, TFRC (CD71), VCAM1 (CD106), KIT (CD117), TNFRSF1A (CD120a), PDGFRB (CD140B), CDH5 (CD144), MCAM (CD146), ALCAM (CD166), ERBB2 (CD340)
#negative controls: ITGAM (CD11B), CD14, PECAM1 (CD31), CD34, PTPRC (CD45), CD79A, HLA-DRA (HLA-DR)
genes <- c("THY1","NT5E","ENG","CD44","ANPEP","NGFR","ITGB1","NCAM1","CD9","CD14","MME","ITGAL","ITGAX","CD36","ITGA3","ITGA4","ITGAV","ICAM1","CD55",
          "CD59","TFRC","VCAM1","KIT","TNFRSF1A","PDGFRB","CDH5","MCAM","ALCAM","ERBB2","ITGAM","CD14","PECAM1","CD34","PTPRC","CD79A","HLA-DRA")
data@assays$integrated[genes,1:5]
FeaturePlot(data, features=genes[1:4], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=4)

#Time 2
#diagnostic: RUNX2, BGLAP, CAPG, [DMP1], [PHEX], MEPE, [SOST], [HYOU1]
#additional: COL1A1, COL1A2, ALPL, TNFRSF11B (OPG), TNFSF11 (RANKL), CSF1 (M-CSF), DKK1, SFRP1, PTGES2, PTGS2, SPP1 (OPN), SPARC (ONT)
#if needed: E11*
# * not in ortho-exon (osteo)
genes <- c("RUNX2","BGLAP","CAPG","DMP1","PHEX","MEPE","SOST","HYOU1",
           "COL1A1","COL1A2","ALPL","TNFRSF11B","TNFSF11",
           "CSF1","DKK1","SFRP1","PTGES2","PTGS2","SPP1","SPARC")
data@assays$integrated[genes,1:5]
FeaturePlot(data, features=genes[1:8], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(1,2)) + labs(x="PC1",y="PC2")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(3,4)) + labs(x="PC3",y="PC4")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(5,6)) + labs(x="PC5",y="PC6")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(7,8)) + labs(x="PC7",y="PC8")),
                        (FeaturePlot(data, features=genes[1], reduction="pca", pt.size=0.5, order=TRUE, dims=c(9,10)) + labs(x="PC1",y="PC10"))), ncol=5)
FeaturePlot(data, features=genes[9:20], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")
CombinePlots(plots=list((FeaturePlot(subset(data,subset=Species=="Human"), features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="Sample") +
                           labs(x="UMAP1",y="UMAP2")),
                        (FeaturePlot(subset(data,subset=Species=="Chimp"), features="COL1A1", reduction="umap", pt.size=0.5, order=TRUE, split.by="Sample") +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)
#markers that differentiated BM osteoblasts in Anastasia Tikhonova's research (ASBMR2020)
FeaturePlot(data, features=c("COL11A2","SPARC","SOX9","COMP","CHAD","COL10A1""EDIL3","MMP14","OSTN","COL12A1","ANGPTL2","COL16A1"), reduction="umap", pt.size=0.5, order=TRUE)
#markers used to differentiate different stages of osteogenic differentiation (BGLAP=preosteoblast,TNC=intermediate,PDPN+FBLN7=osteocytes) from Jialiang Wang's research (ASBMR2020)
FeaturePlot(data, features=c("BGLAP","TNC","PDPN","FBLN7"), reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

#chondrocyte diagnostic: ACAN, COL2A1, SOX9, COL10A1, COL11A1, SOX5, SOX6
#adipocyte diagnostic: FABP4, PPARG, EBF2, PDGFRA, FGF10, FGF16, FGF19, SLC7A10, SLC36A2, MYF5, P2RX5
genes <- c("ACAN","COL2A1","SOX9","FABP4","PPARG","EBF2",
           "COL10A1","COL11A1","SOX5","SOX6",
           "PDGFRA","FGF10","FGF16","FGF19","SLC7A10","SLC36A2","P2RX5","MYF5")
data@assays$integrated[genes,1:5]
FeaturePlot(data, features=genes[c(1,4,2,5,3,6)], reduction="umap", pt.size=0.5, order=TRUE) + labs(x="UMAP1",y="UMAP2")

```

Isolate gene expression levels of interest.

```{r}

#Isolate genes of interest
genes <- c("POU5F1","TDGF1","SOX2","EPCAM",
           "THY1","NT5E","ENG","CD44",
           "RUNX2","BGLAP","CAPG","DMP1","PHEX","MEPE","SOST","HYOU1",
           "COL1A1","COL1A2","ALPL",
           "ACAN","COL2A1","SOX9",
           "FABP4","PPARG","EBF2",
           "TNC","PDPN","FBLN7")

```

Define functions for setting potential ad hoc thresholds.

```{r}

#Define functions for setting ad hoc thresholds

chk.thres.manual <- function(data, genes, thres) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres[i], 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.median <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- median(data[,genes[i]]) }
    else { thres <- median(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.q1 <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- quantile(data[,genes[i]],prob=0.25) }
    else { thres <- quantile(data[,genes[i]][data[,genes[i]]<cutoff],prob=0.25) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.q3 <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- quantile(data[,genes[i]],prob=0.75) }
    else { thres <- quantile(data[,genes[i]][data[,genes[i]]<cutoff],prob=0.75) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.mean <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- mean(data[,genes[i]]) }
    else { thres <- mean(data[,genes[i]][data[,genes[i]]<cutoff]) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

chk.thres.meansd <- function(data, genes, cutoff) {
  assignments <- data
  i=1
  while(i <= length(genes)) {
    if(is.na(cutoff)) { thres <- sum(mean(data[,genes[i]]),sd(data[,genes[i]])) }
    else { thres <- sum(mean(data[,genes[i]][data[,genes[i]]<cutoff]),sd(data[,genes[i]][data[,genes[i]]<cutoff])) }
    assignments[,genes[i]] <- ifelse(data[,genes[i]] > thres, 1, 0)
    i=i+1
  }
  return(assignments)
}

```

Define functions for visualizing ad hoc assignments.

```{r}

#Define functions for making upsetr plots

plot.upset.HCcellcompare <- function(data, genes, ymax) {
  plot.ls <- list(
    T0.Human = upset(data[which(data$Stage=="Time 0" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage=="Time 1" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage=="Time 2" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage=="Time 0" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage=="Time 1" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage=="Time 2" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE, sets=genes, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit('arrange', name=v)
    vp <- grid.grab()
    plot.ls[[v]] <- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}

plot.upset.sub.HCcellcompare <- function(data, genes, intersections, ymax) {
  plot.ls <- list(
    T0.Human = upset(data[which(data$Stage=="Time 0" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Human = upset(data[which(data$Stage=="Time 1" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Human = upset(data[which(data$Stage=="Time 2" & data$Species=="Human"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T0.Chimp = upset(data[which(data$Stage=="Time 0" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T1.Chimp = upset(data[which(data$Stage=="Time 1" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax),
    T2.Chimp = upset(data[which(data$Stage=="Time 2" & data$Species=="Chimp"),genes],
                     nintersects=NA, empty.intersections="on", keep.order=TRUE, order.by="degree", decreasing=TRUE,
                     sets=genes, intersections=intersections, mainbar.y.max=ymax))
  for (v in names(plot.ls)) {
    print(plot.ls[[v]])
    grid.text(v, x=0.65, y=0.97, gp=gpar(fontsize=10))
    grid.edit('arrange', name=v)
    vp <- grid.grab()
    plot.ls[[v]] <- vp
  }
  return(grid.arrange(grobs=plot.ls, ncol=3))
}

```

Calculate if genes reach expression thresholds. Start with several threshold limits.

```{r}

#Integrated data scaled and confounding variables regressed out
tot.genes <- cbind(as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes,]))),data@meta.data[,c("Species","Stage","Sample")])

assign.thresh0 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0,length(genes))))
assign.thresh0.01 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.01,length(genes))))
assign.thresh0.05 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.05,length(genes))))
assign.thresh0.1 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,length(genes))))
assign.thresh0.3 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.3,length(genes))))
assign.thresh0.5 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.5,length(genes))))
assign.thresh0.7 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.7,length(genes))))
assign.thresh0.9 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.9,length(genes))))
assign.thresh1 <- chk.thres.manual(tot.genes, genes, thres=c(rep(1,length(genes))))
assign.threshMulti01 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
assign.threshMulti0 <- chk.thres.manual(tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

assign.median <- chk.thres.median(tot.genes, genes, cutoff=NA)
assign.median1 <- chk.thres.median(tot.genes, genes, cutoff=1)
assign.q1 <- chk.thres.q1(tot.genes, genes, cutoff=NA)
assign.q3 <- chk.thres.q3(tot.genes, genes, cutoff=NA)
assign.mean <- chk.thres.mean(tot.genes, genes, cutoff=NA)
assign.mean1 <- chk.thres.mean(tot.genes, genes, cutoff=1)
assign.meansd <- chk.thres.meansd(tot.genes, genes, cutoff=NA)
assign.meansd1 <- chk.thres.meansd(tot.genes, genes, cutoff=1)
assign.multi.q1median <- cbind(assign.q1[,1:8],assign.median[,9:16],assign.q1[,17:31])
assign.multi.medianq1 <- cbind(assign.median[,1:8],assign.q1[,9:16],assign.median[,17:31])

```

```{r}

#Log normalized data
l.tot.genes <- cbind(as.data.frame(t(as.matrix(data@assays$RNA@data[genes,]))),data@meta.data[,c("Species","Stage","Sample")])

l.assign.thresh0 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0,length(genes))))
l.assign.thresh0.01 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.01,length(genes))))
l.assign.thresh0.05 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.05,length(genes))))
l.assign.thresh0.1 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,length(genes))))
l.assign.thresh0.3 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.3,length(genes))))
l.assign.thresh0.5 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.5,length(genes))))
l.assign.thresh0.7 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.7,length(genes))))
l.assign.thresh0.9 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.9,length(genes))))
l.assign.thresh1 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(1,length(genes))))
#l.assign.threshMulti01 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,8),rep(0.01,8),rep(0.1,9),rep(0.01,3)))
#l.assign.threshMulti0 <- chk.thres.manual(l.tot.genes, genes, thres=c(rep(0.1,8),rep(0,8),rep(0.1,9),rep(0,3)))

l.assign.median <- chk.thres.median(l.tot.genes, genes, cutoff=NA)
l.assign.median1 <- chk.thres.median(l.tot.genes, genes, cutoff=1)
l.assign.q1 <- chk.thres.q1(l.tot.genes, genes, cutoff=NA)
l.assign.q3 <- chk.thres.q3(l.tot.genes, genes, cutoff=NA)
l.assign.mean <- chk.thres.mean(l.tot.genes, genes, cutoff=NA)
l.assign.mean1 <- chk.thres.mean(l.tot.genes, genes, cutoff=1)
l.assign.meansd <- chk.thres.meansd(l.tot.genes, genes, cutoff=NA)
l.assign.meansd1 <- chk.thres.meansd(l.tot.genes, genes, cutoff=1)
#l.assign.multi.q1median <- cbind(assign.q1[,1:8],assign.median[,9:16],assign.q1[,17:31])
#l.assign.multi.medianq1 <- cbind(assign.median[,1:8],assign.q1[,9:16],assign.median[,17:31])

```

While several thresholds were tested, using the first quartile or median as an expression level cutoff makes the most sense given gene distributions.

```{r}

ggplot(tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

```{r}

ggplot(l.tot.genes, aes(x=POU5F1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(POU5F1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(POU5F1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(POU5F1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=TDGF1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TDGF1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TDGF1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TDGF1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=EPCAM)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EPCAM)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EPCAM)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EPCAM,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=THY1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(THY1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(THY1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(THY1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=NT5E)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(NT5E)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(NT5E)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(NT5E,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ENG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ENG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ENG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ENG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=CD44)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CD44)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CD44)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CD44,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=RUNX2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(RUNX2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(RUNX2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(RUNX2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=BGLAP)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(BGLAP)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(BGLAP)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(BGLAP,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=CAPG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(CAPG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(CAPG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(CAPG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=DMP1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(DMP1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(DMP1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(DMP1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PHEX)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PHEX)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PHEX)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PHEX,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=MEPE)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(MEPE)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(MEPE)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(MEPE,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOST)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOST)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOST)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOST,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=HYOU1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(HYOU1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(HYOU1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(HYOU1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL1A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL1A2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL1A2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL1A2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL1A2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ALPL)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ALPL)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ALPL)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ALPL,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=ACAN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(ACAN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(ACAN)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(ACAN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=COL2A1)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(COL2A1)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(COL2A1)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(COL2A1,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=SOX9)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(SOX9)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(SOX9)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(SOX9,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=FABP4)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FABP4)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FABP4)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FABP4,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PPARG)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PPARG)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PPARG)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PPARG,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=EBF2)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(EBF2)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(EBF2)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(EBF2,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=TNC)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(TNC)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(TNC)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(TNC,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=PDPN)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(PDPN)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(PDPN)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(PDPN,prob=0.75)), color="red", linetype="solid", size=0.5)

ggplot(l.tot.genes, aes(x=FBLN7)) +
  geom_histogram(binwidth=0.05, color="black", fill="white") +
  geom_vline(aes(xintercept=mean(FBLN7)), color="blue", linetype="dashed", size=0.5) +
  geom_vline(aes(xintercept=median(FBLN7)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.25)), color="red", linetype="solid", size=0.5) +
  geom_vline(aes(xintercept=quantile(FBLN7,prob=0.75)), color="red", linetype="solid", size=0.5)

```

Look at which cells meet threshold cutoffs.

Takeaways regarding assignment cutoffs:

* best cutoff for iPSC markers is mean or manual threshold = 0
  + median threshold is okay but shows some positive cells in Time 1 and 2
* best cutoff for MSC markers is median, mean, or manual threshold = 0
* best cutoff for osteogenic markers median, mean, or manual threshold = 0

Takeaways regarding marker genes:

* Highest proportion of cells expressing iPSC markers (POU5F1 + TDGF1 + SOX2 + EPCAM) come from Time 0 (good markers)
* Highest proportion of cells expressing MSC markers (THY1 + NT5E + ENG + CD44) come from Time 1 (okay markers)
    + relatively high proportion in Time 2 cells as well
    + better able to distinguish between Time 1 and Time 2 by including ALP
* Highest proportion of cells expressing general osteogenic markers (COL1A1 + COL1A2 + ALPL) come from Time 2 (okay markers)
    + relatively high proportion in Time 0 cells as well
    + better able to distinguish between Time 0 and Time 2 by including POU5F1
* Proportion of cells expressing more specific osteogenic markers (RUNX2 + BGLAP + CAPG + DMP1 + PHEX + MEPE + SOST + HYOU1) look similar in Time 1 and Time 2 (bad markers)
    + proportion slightly decreases in Time 1 when cells are first filtered using general osteogenic marker expression
* Variable numbers of cells express chondrogenic markers (ACAN + COL2A1 + SOX9)
* Variable numbers of cells express adipogenic markers depending on the threshold (FABP4 + PPARG + EBF2)


```{r}

chk.thres <- function(data, pou5f1, tdgf1, sox2, epcam, thy1, nt5e, eng, cd44, col1a1, col1a2, alpl) {
  assign <- ""
  if(data[1]  > pou5f1) { assign <- paste(assign,"POU5F1") }
  if(data[2]  > tdgf1)  { assign <- paste(assign,"TDGF1") }
  if(data[3]  > sox2)   { assign <- paste(assign,"SOX2") }
  if(data[4]  > epcam)  { assign <- paste(assign,"EPCAM") }
  if(data[5]  > thy1)   { assign <- paste(assign,"THY1") }
  if(data[6]  > nt5e)   { assign <- paste(assign,"NT5E") }
  if(data[7]  > eng)    { assign <- paste(assign,"ENG") }
  if(data[8]  > cd44)   { assign <- paste(assign,"CD44") }
  if(data[9]  > col1a1) { assign <- paste(assign,"COL1A1") }
  if(data[10] > col1a2) { assign <- paste(assign,"COL1A2") }
  if(data[11] > alpl)   { assign <- paste(assign,"ALPL") }
  return(assign)
}
test.genes <- as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes[c(1:8,17:19)],])))
test.genes$assign <- apply(test.genes, 1, chk.thres, pou5f1=0, tdgf1=0, sox2=0, epcam=0, thy1=0, nt5e=0, eng=0, cd44=0, col1a1=0, col1a2=0, alpl=0) #threshold = 0
test <- test.genes %>% group_by(assign) %>% summarise(total=n()) %>% dplyr::mutate(freq=total/sum(total))

chk.thres <- function(data, pou5f1, tdgf1, sox2, epcam, thy1, nt5e, eng, cd44, col1a1, col1a2, alpl) {
  assign <- ""
  if(data[1]  > pou5f1) { assign <- paste(assign,"POU5F1") }
  if(data[2]  > tdgf1)  { assign <- paste(assign,"TDGF1") }
  if(data[3]  > sox2)   { assign <- paste(assign,"SOX2") }
  if(data[4]  > epcam)  { assign <- paste(assign,"EPCAM") }
  if(data[5]  > thy1)   { assign <- paste(assign,"THY1") }
  if(data[6]  > nt5e)   { assign <- paste(assign,"NT5E") }
  if(data[7]  > eng)    { assign <- paste(assign,"ENG") }
  if(data[8]  > cd44)   { assign <- paste(assign,"CD44") }
  if(data[11] > alpl)   { assign <- paste(assign,"ALPL") }
  return(assign)
}
test.genes <- as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes[c(1:8,17:19)],])))
test.genes$assign <- apply(test.genes, 1, chk.thres, pou5f1=0, tdgf1=0, sox2=0, epcam=0, thy1=0, nt5e=0, eng=0, cd44=0, alpl=0) #threshold = 0
test <- test.genes %>% group_by(assign) %>% summarise(total=n()) %>% dplyr::mutate(freq=total/sum(total))

chk.thres <- function(data, col1a1, col1a2, alpl, runx2, bglap, capg, phex, mepe) {
  assign <- ""
  if(data[1] > col1a1) { assign <- paste(assign,"COL1A1") }
  if(data[2] > col1a2) { assign <- paste(assign,"COL1A2") }
  if(data[3] > alpl)   { assign <- paste(assign,"ALPL") }
  if(data[4] > runx2)  { assign <- paste(assign,"RUNX2") }
  if(data[5] > bglap)  { assign <- paste(assign,"BGLAP") }
  if(data[6] > capg)   { assign <- paste(assign,"CAPG") }
  if(data[7] > phex)   { assign <- paste(assign,"PHEX") }
  if(data[8] > mepe)   { assign <- paste(assign,"MEPE") }
  return(assign)
}
test.genes <- as.data.frame(t(as.matrix(data@assays$integrated@scale.data[genes[c(17:19,9:11,13:14)],])))
test.genes$assign <- apply(test.genes, 1, chk.thres, col1a1=0, col1a2=0, alpl=0, runx2=0, bglap=0, capg=0, phex=0, mepe=0) #threshold = 0
test <- test.genes %>% group_by(assign) %>% summarise(total=n()) %>% dplyr::mutate(freq=total/sum(total))


```

```{r}

assignments <- assign.thresh0
assignments <- assign.thresh0.01
assignments <- assign.thresh0.05
assignments <- assign.thresh0.1
assignments <- assign.median
#assignments <- assign.q1
#assignments <- assign.mean
#assignments <- assign.multi.q1median

assignments <- l.assign.thresh0
#assignments <- l.assign.thresh0.01
#assignments <- l.assign.thresh0.05
#assignments <- l.assign.thresh0.1
#assignments <- l.assign.thresh0.3
#assignments <- l.assign.thresh0.5
#assignments <- l.assign.thresh0.7
#assignments <- l.assign.thresh0.9
#assignments <- l.assign.thresh1
#assignments <- l.assign.median
assignments <- l.assign.q1
#assignments <- l.assign.mean

#all genes
plot.upset.HCcellcompare(assignments,genes[c(1:8,17:19)],20000)

#ipsc candidate genes
plot.upset.HCcellcompare(assignments,genes[1:4],20000)
plot.upset.sub.HCcellcompare(assignments,
                             genes[c(1:8,17:19)],
                             intersections=,
                             20000)

#msc candidate genes
plot.upset.HCcellcompare(assignments,genes[5:8],20000)
plot.upset.HCcellcompare(assignments,genes[c(19,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2
plot.upset.HCcellcompare(assignments,genes[c(17,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2
plot.upset.HCcellcompare(assignments,genes[c(17:18,5:8)],20000) #include osteo marker to better distinguish time 1 and time 2

#osteogenic candidate genes
plot.upset.HCcellcompare(assignments,genes[17:19],20000)
plot.upset.HCcellcompare(assignments,genes[c(1,17:19)],20000) #include ipsc marker to better distinguish time 2

#more specific osteogenic candidate genes
plot.upset.HCcellcompare(assignments,genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignments,genes[c(9:10,13:14)],20000)
plot.upset.HCcellcompare(assignments,genes[c(9:11,17:18)],20000)
plot.upset.HCcellcompare(assignments,genes[c(11,13:14,17:18)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$ALPL==1),],genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1),],genes[c(9:11,13:14)],20000)
plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1),],genes[c(9:11,13:14)],20000)
#plot.upset.HCcellcompare(assignments,genes[9:16],20000)
#plot.upset.HCcellcompare(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1),],genes[9:16],20000)

#chondrogenic candidate genes
plot.upset.HCcellcompare(assignments,genes[20:22],20000)

#adipogenic candidate genes
plot.upset.HCcellcompare(assignments,genes[23:25],20000)

```

Quickly look at possible ad hoc classification systems.

This one looks good for general ad hoc assignment:

```{r}

#assignments <- assign.thresh0
assignments <- assign.median
#assignments <- assign.q1
#assignments <- assign.mean
#assignments <- assign.multi.q1median

#general ad hoc classification system
CombinePlots(list((DimPlot(data, group.by="Stage", reduction="umap") +
                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1 &
                                                           assignments$NT5E==0 & assignments$ENG==0 & assignments$CD44==0 &
                                                           assignments$COL1A1==0 & assignments$COL1A2==0 & assignments$ALPL==0),])) +
                    labs(x="UMAP1",y="UMAP2",title="iPSCs (ad hoc)")),
                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1 & assignments$ALPL==0),])) +
                    labs(x="UMAP1",y="UMAP2",title="MSCs (ad hoc)")),
                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
                          cells=rownames(assignments[which(assignments$ALPL==1 & assignments$POU5F1==0 & assignments$TDGF1==0 & assignments$SOX2==0 & assignments$EPCAM==0),])) +
                    labs(x="UMAP1",y="UMAP2",title="Osteogenic Cells (ad hoc)"))), ncol=4)

#CombinePlots(list((DimPlot(data, group.by="Stage", reduction="umap") +
#                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1 &
#                                                           assignments$NT5E==0 & assignments$ENG==0 & assignments$CD44==0 &
#                                                           assignments$COL1A1==0 & assignments$COL1A2==0 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="iPSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="MSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$ALPL==1 & assignments$POU5F1==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="Osteogenic Cells (ad hoc)"))), ncol=4)

#CombinePlots(list((DimPlot(data, group.by="Stage", reduction="umap") +
#                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1 &
#                                                           assignments$NT5E==0 & assignments$ENG==0 & assignments$CD44==0 &
#                                                           assignments$COL1A1==0 & assignments$COL1A2==0 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="iPSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="MSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$ALPL==1),])) +
#                    labs(x="UMAP1",y="UMAP2",title="Osteogenic Cells (ad hoc)"))), ncol=4)
#

#CombinePlots(list((DimPlot(data, group.by="Stage", reduction="umap") +
#                    labs(x="UMAP1",y="UMAP2",title="All Cells")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                           cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1 &
#                                                            assignments$NT5E==0 & assignments$ENG==0 & assignments$CD44==0 &
#                                                            assignments$COL1A1==0 & assignments$COL1A2==0 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="iPSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="MSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1  & assignments$POU5F1==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="Osteogenic Cells (ad hoc)"))), ncol=4)

#CombinePlots(list((DimPlot(data, group.by="Stage", reduction="umap") +
#                      labs(x="UMAP1",y="UMAP2",title="All Cells")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$POU5F1==1 & assignments$TDGF1==1 & assignments$SOX2==1 & assignments$EPCAM==1),])) +
#                    labs(x="UMAP1",y="UMAP2",title="iPSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$THY1==1 & assignments$NT5E==1 & assignments$ENG==1 & assignments$CD44==1 & assignments$ALPL==0),])) +
#                    labs(x="UMAP1",y="UMAP2",title="MSCs (ad hoc)")),
#                  (DimPlot(data, group.by="Species", reduction="umap", cols=c("grey","black"),
#                          cells=rownames(assignments[which(assignments$ALPL==1),])) +
#                    labs(x="UMAP1",y="UMAP2",title="Osteogenic Cells (ad hoc)"))), ncol=4)

```

This one looks good for osteogenic ad hoc assignment:

```{r}

#assignments <- assign.thresh0
assignments <- assign.median
#assignments <- assign.q1
#assignments <- assign.mean
#assignments <- assign.multi.q1median

#osteogenic specific assignments
#cell assignments according to Dallas et al. 2013 (excluding DMP1 + SOST + HYOU1) - few preosteoblasts and osteoblasts present
#(notes: DMP1 excluded because it is not expressed in any cell)
#(notes: SOST excluded because it is only expressed in 1 cell)
#(notes: HYOU1 excluded because it is a heat shock protein and may get turned on for any number of reasons aside from osteogenic differentiation)

upset(assignments[,c(9:11,13:14)], keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      order.by="degree", nintersects=100)
upset(assignments[which(assignments$ALPL==1),c(9:11,13:14)], keep.order=TRUE,
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      order.by="degree", nintersects=100)
upset(assignments[which(assignments$ALPL==1 & assignments$POU5F1==0),c(9:11,13:14)], keep.order=TRUE,
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      order.by="degree", nintersects=100)
upset(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0),c(9:11,13:14)], keep.order=TRUE,
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      order.by="degree", nintersects=100)

upset(assignments[,c(9:11,13:14)], keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      intersections=list(list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG","PHEX"),
                         list("BGLAP","CAPG","PHEX","MEPE"),
                         list("CAPG","PHEX","MEPE")))
upset(assignments[which(assignments$ALPL==1),c(9:11,13:14)], keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      intersections=list(list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG","PHEX"),
                         list("BGLAP","CAPG","PHEX","MEPE"),
                         list("CAPG","PHEX","MEPE")))
upset(assignments[which(assignments$ALPL==1 & assignments$POU5F1==0),c(9:11,13:14)], keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      intersections=list(list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG","PHEX"),
                         list("BGLAP","CAPG","PHEX","MEPE"),
                         list("CAPG","PHEX","MEPE")))
upset(assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0),c(9:11,13:14)], keep.order=TRUE, empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      intersections=list(list("RUNX2"),
                         list("RUNX2","BGLAP"),
                         list("BGLAP","CAPG","PHEX"),
                         list("BGLAP","CAPG","PHEX","MEPE"),
                         list("CAPG","PHEX","MEPE")))

plot.upset.sub.HCcellcompare(data=assignments,
                             genes=genes[c(9:11,13:14)],
                             intersections=list(list("RUNX2"),
                                                list("RUNX2","BGLAP"),
                                                list("BGLAP","CAPG","PHEX"),
                                                list("BGLAP","CAPG","PHEX","MEPE"),
                                                list("CAPG","PHEX","MEPE")),
                             ymax=2000)
plot.upset.sub.HCcellcompare(data=assignments[which(assignments$ALPL==1),],
                             genes=genes[c(9:11,13:14)],
                             intersections=list(list("RUNX2"),
                                                list("RUNX2","BGLAP"),
                                                list("BGLAP","CAPG","PHEX"),
                                                list("BGLAP","CAPG","PHEX","MEPE"),
                                                list("CAPG","PHEX","MEPE")),
                             ymax=2000)
plot.upset.sub.HCcellcompare(data=assignments[which(assignments$ALPL==1 & assignments$POU5F1==0),],
                             genes=genes[c(9:11,13:14)],
                             intersections=list(list("RUNX2"),
                                                list("RUNX2","BGLAP"),
                                                list("BGLAP","CAPG","PHEX"),
                                                list("BGLAP","CAPG","PHEX","MEPE"),
                                                list("CAPG","PHEX","MEPE")),
                             ymax=2000)
plot.upset.sub.HCcellcompare(data=assignments[which(assignments$COL1A1==1 & assignments$COL1A2==1 & assignments$ALPL==1 & assignments$POU5F1==0),],
                             genes=genes[c(9:11,13:14)],
                             intersections=list(list("RUNX2"),
                                                list("RUNX2","BGLAP"),
                                                list("BGLAP","CAPG","PHEX"),
                                                list("BGLAP","CAPG","PHEX","MEPE"),
                                                list("CAPG","PHEX","MEPE")),
                             ymax=2000)

upset(assignments[which(assignments$ALPL==1 & assignments$POU5F1==0),c(9:11,13:14)],
      keep.order=TRUE,
      empty.intersections="on",
      sets=c("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
      intersections=list(#preosteoblasts
                         list("RUNX2"),
                         #osteoblasts
                         list("RUNX2","BGLAP"),
                         list("BGLAP"),
                         #embedding osteblasts
                         list("BGLAP","CAPG","PHEX"),
                         list("CAPG"),
                         list("PHEX"),
                         list("RUNX2","CAPG"),
                         list("RUNX2","PHEX"),
                         list("BGLAP","CAPG"),
                         list("BGLAP","PHEX"),
                         list("CAPG","PHEX"),
                         list("RUNX2","BGLAP","CAPG"),
                         list("RUNX2","BGLAP","PHEX"),
                         list("RUNX2","CAPG","PHEX"),
                         list("RUNX2","BGLAP","CAPG","PHEX"),
                         #mineralizing osteoblasts
                         list("BGLAP","CAPG","PHEX","MEPE"),
                         list("RUNX2","MEPE"),
                         list("BGLAP","MEPE"),
                         list("RUNX2","BGLAP","MEPE"),
                         list("RUNX2","CAPG","MEPE"),
                         list("RUNX2","PHEX","MEPE"),
                         list("BGLAP","CAPG","MEPE"),
                         list("BGLAP","PHEX","MEPE"),
                         list("RUNX2","BGLAP","CAPG","MEPE"),
                         list("RUNX2","BGLAP","PHEX","MEPE"),
                         list("RUNX2","CAPG","PHEX","MEPE"),
                         list("RUNX2","BGLAP","CAPG","PHEX","MEPE"),
                         #mature osteocytes
                         list("CAPG","PHEX","MEPE"),
                         list("MEPE"),
                         list("CAPG","MEPE"),
                         list("PHEX","MEPE")))

```

#### Ad Hoc Assignment (general)

Assign cell type labels based on ad hoc cutoffs.

How to classify cells:

* iPSCs = express POU5F1 + TDGF1 + SOX2 + EPCAM (but do not express NT5E + ENG + CD44 + COL1A1 + COL1A2 + ALPL)
* MSCs = express THY1 + NT5E + ENG + CD44 (but do not express ALPL)
* Osteogenic Cells = express ALPL (but do not express POU5F1 + TDGF1 + SOX2 + EPCAM) or COL1A1 + COL1A2 (but do not express POU5F1 + TDGF1 + SOX2 + EPCAM) 

[use median of integrated and scaled expression as the cutoff for genes]

```{r}

#define function to assign cell type labels based on ad hoc cutoffs
hoc.assign <- function(assignments) {
  assignments$assign <- "other"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$POU5F1[i] == 1 & assignments$TDGF1[i] == 1 & assignments$SOX2[i] == 1 & assignments$EPCAM[i] == 1 &
       assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0 &
       assignments$COL1A1[i] == 0 & assignments$COL1A2[i] == 0 & assignments$ALPL[i] == 0) {assignments$assign[i] <- "iPSC"}
    if(assignments$THY1[i] == 1 & assignments$NT5E[i] == 1 & assignments$ENG[i] == 1 & assignments$CD44[i] == 1 & assignments$ALPL[i] == 0) {assignments$assign[i] <- "MSC"}
    if((assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
       (assignments$COL1A1[i] == 1 & assignments$COL1A2[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0))
      {assignments$assign[i] <- "Osteogenic"}
    i=i+1
  }
 return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, median)
ad.hoc.assign <- hoc.assign(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.median <- ad.hoc.assign$`assignments$assign`

```

```{r}

#define function to assign cell type labels based on ad hoc cutoffs
#hoc.assign <- function(assignments) {
#  assignments$assign <- "other"
#  i=1
#  while(i <= length(rownames(assignments))) {
#    if(assignments$POU5F1[i] == 1 & assignments$TDGF1[i] == 1 & assignments$SOX2[i] == 1 & assignments$EPCAM[i] == 1 &
#       assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0 &
#       assignments$COL1A1[i] == 0 & assignments$COL1A2[i] == 0 & assignments$ALPL[i] == 0) {assignments$assign[i] <- "iPSC"}
#    if(assignments$THY1[i] == 1 & assignments$NT5E[i] == 1 & assignments$ENG[i] == 1 & assignments$CD44[i] == 1 & assignments$ALPL[i] == 0) {assignments$assign[i] <- "MSC"}
#    if(assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) {assignments$assign[i] <- "Osteogenic"}
#    i=i+1
#  }
# return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
#}
hoc.assign <- function(assignments) {
  assignments$assign <- "other"
  i=1
  while(i <= length(rownames(assignments))) {
    if((assignments$POU5F1[i] == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0) |
       (assignments$TDGF1[i]  == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0 ) |
       (assignments$SOX2[i]   == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0 ) |
       (assignments$EPCAM[i]  == 1 & assignments$NT5E[i] == 0 & assignments$ENG[i] == 0 & assignments$CD44[i] == 0))
      {assignments$assign[i] <- "iPSC"}
    if((assignments$THY1[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$NT5E[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$ENG[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0) |
       (assignments$CD44[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0 & assignments$ALPL[i] == 0))
      {assignments$assign[i] <- "MSC"}
    if((assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0))
      {assignments$assign[i] <- "Osteogenic"}
    #if((assignments$ALPL[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
    #   (assignments$RUNX2[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
    #   (assignments$RUNX2[i] == 1 & assignments$BGLAP[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
    #   (assignments$BGLAP[i] == 1 & assignments$CAPG[i] == 1 & assignments$PHEX[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
    #   (assignments$BGLAP[i] == 1 & assignments$PHEX[i] == 1 & assignments$MEPE[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
    #   (assignments$CAPG[i] == 1 & assignments$PHEX[i] == 1 & assignments$MEPE[i] == 1 &
    #    assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0)) {assignments$assign[i] <- "Osteogenic"}
    i=i+1
  }
 return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0 <- ad.hoc.assign$`assignments$assign`

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.01 <- ad.hoc.assign$`assignments$assign`

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.01 <- ad.hoc.assign$`assignments$assign`

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.05)
ad.hoc.assign <- hoc.assign(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.05 <- ad.hoc.assign$`assignments$assign`

#assign cell type labels based on ad hoc cutoffs (scaled integrated data, thresh0.1)
ad.hoc.assign <- hoc.assign(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.thresh0.1 <- ad.hoc.assign$`assignments$assign`

#assign cell type labels based on ad hoc cutoffs (log normalized data, thresh0)
ad.hoc.assign <- hoc.assign(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$AdHoc.Assign.l.thresh0 <- ad.hoc.assign$`assignments$assign`

```

```{r}

table(subset(data,subset=Species=="Human")@meta.data$AdHoc.Assign)
table(subset(data,subset=Species=="Chimp")@meta.data$AdHoc.Assign)
table(subset(data,subset=Stage=="Time 0")@meta.data$AdHoc.Assign)
table(subset(data,subset=Stage=="Time 1")@meta.data$AdHoc.Assign)
table(subset(data,subset=Stage=="Time 2")@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Human" & Stage=="Time 0"))@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Human" & Stage=="Time 1"))@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Human" & Stage=="Time 2"))@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 0"))@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 1"))@meta.data$AdHoc.Assign)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 2"))@meta.data$AdHoc.Assign)

```

```{r}

ggplot(data@meta.data, aes(x=AdHoc.Assign.median, fill=Species)) + geom_bar(stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black"))
ggplot(data@meta.data, aes(x=AdHoc.Assign.median, fill=Species)) +
  geom_bar(mapping = aes(x=AdHoc.Assign.median, y=..prop.., group=Species), stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black"))

ggplot(data@meta.data, aes(x=AdHoc.Assign.median, fill=Individual)) +
  geom_bar(stat="count", position="dodge")
ggplot(data@meta.data, aes(x=AdHoc.Assign.median, fill=Individual)) +
  geom_bar(mapping = aes(x=AdHoc.Assign.median, y=..prop.., group=Individual), stat="count", position="dodge")

DimPlot(data,
        group.by="Species",
        split.by="AdHoc.Assign.median",
        reduction="umap",
        cols=c("grey","black")) +
  labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                 group.by="Stage",
                                 split.by="AdHoc.Assign.median",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Species=="Chimp"),
                                 group.by="Stage",
                                 split.by="AdHoc.Assign.median",
                                 reduction="umap",
                                 cols=c("pink","mediumorchid","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)

CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                 group.by="Stage",
                                 split.by="AdHoc.Assign.median",
                                 reduction="umap",
                                 cols=c("pink")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 1"),
                                 group.by="Stage",
                                 split.by="AdHoc.Assign.median",
                                 reduction="umap",
                                 cols=c("mediumorchid")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 2"),
                                 group.by="Stage",
                                 split.by="AdHoc.Assign.median",
                                 reduction="umap",
                                 cols=c("dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)

```

### Ad Hoc Assignment (osteogenic)

Assign cell type labels based on ad hoc cutoffs.

How to classify cells (using 5 genes):

* not osteogenic = do not express ALPL or do not express COL1A1 + COL1A2
* osteogenic = express ALPL (but do not express POU5F1 + TDGF1 + SOX2 + EPCAM) or COL1A1 + COL1A2 (but do not express POU5F1 + TDGF1 + SOX2 + EPCAM) 
  + unknown osteogenic = do not express RUNX2 + BGLAP + CAPG + PHEX + MEPE
  + preosteoblasts = express RUNX2 (but do not express BGLAP + CAPG + PHEX + MEPE)
  + osteoblasts = express RUNX2 + BGLAP (but do not express CAPG + PHEX + MEPE)
  + embedding osteoblasts = express BGLAP + CAPG + PHEX (but do not express RUNX2 + MEPE)
  + mineralizing osteoblasts = express BGLAP + CAPG + PHEX + MEPE (but do not express RUNX2)
  + maturing osteoblasts = express CAPG + PHEX + MEPE (but do not express RUNX2 + BGLAP)

[use median of integrated and scaled expression as the cutoff for all genes]


```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes)
hoc.assign.x.5 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if((assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
       (assignments$COL1A1[i] == 1 & assignments$COL1A2[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0)) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.5 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if((assignments$ALPL[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0) |
       (assignments$COL1A1[i] == 1 & assignments$COL1A2[i] == 1 & assignments$POU5F1[i] == 0 & assignments$TDGF1[i] == 0 & assignments$SOX2[i] == 0 & assignments$EPCAM[i] == 0)) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (scaled integrated data, median)
ad.hoc.assign <- hoc.assign.x.5(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.median <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.median,
                                                        levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                 "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                 "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.median <- factor(data@meta.data$OstAdHoc.Assign.5gene.median,
                                                      levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                               "unknown osteogenic","not osteogenic","other"))

```

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes)
hoc.assign.x.5 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.5 <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.5(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.5gene.thresh0,
                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.5(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.01,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.5gene.thresh0.01,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (scaled integrated data, thresh0.05)
ad.hoc.assign <- hoc.assign.x.5(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.05,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.5gene.thresh0.05,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (scaled integrated data, thresh0.1)
ad.hoc.assign <- hoc.assign.x.5(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.thresh0.1,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.5gene.thresh0.1,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 5 genes) (log normalized data, thresh0)
ad.hoc.assign <- hoc.assign.x.5(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.5gene.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.5gene.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.5gene.l.thresh0,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.5(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.5gene.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.5gene.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.5gene.l.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

```

```{r}

table(subset(data,subset=Species=="Human")@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=Species=="Chimp")@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=Stage=="Time 0")@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=Stage=="Time 1")@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=Stage=="Time 2")@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 0"))@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 1"))@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 2"))@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 0"))@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 1"))@meta.data$OstAdHoc.Assign.x.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 2"))@meta.data$OstAdHoc.Assign.x.5gene.median)

table(subset(data,subset=Species=="Human")@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=Species=="Chimp")@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=Stage=="Time 0")@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=Stage=="Time 1")@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=Stage=="Time 2")@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 0"))@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 1"))@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Human" & Stage=="Time 2"))@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 0"))@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 1"))@meta.data$OstAdHoc.Assign.5gene.median)
table(subset(data,subset=(Species=="Chimp" & Stage=="Time 2"))@meta.data$OstAdHoc.Assign.5gene.median)

```

```{r}

ggplot(data@meta.data, aes(x=OstAdHoc.Assign.x.5gene.median, fill=Species)) + geom_bar(stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black"))
ggplot(data@meta.data, aes(x=OstAdHoc.Assign.x.5gene.median, fill=Species)) +
  geom_bar(mapping = aes(x=OstAdHoc.Assign.x.5gene.median, y=..prop.., group=Species), stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black"))

ggplot(data@meta.data, aes(x=OstAdHoc.Assign.x.5gene.median, fill=Individual)) +
  geom_bar(stat="count", position="dodge")
ggplot(data@meta.data, aes(x=OstAdHoc.Assign.x.5gene.median, fill=Individual)) +
  geom_bar(mapping = aes(x=OstAdHoc.Assign.x.5gene.median, y=..prop.., group=Individual), stat="count", position="dodge")

DimPlot(data,
        group.by="Species",
        split.by="OstAdHoc.Assign.x.5gene.median",
        reduction="umap",
        cols=c("grey","black")) +
  labs(x="UMAP1",y="UMAP2")

ggplot(data@meta.data, aes(x=OstAdHoc.Assign.5gene.median, fill=Species)) + geom_bar(stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black")) +
  ylim(c(0,10000))
ggplot(data@meta.data, aes(x=OstAdHoc.Assign.5gene.median, fill=Species)) +
  geom_bar(mapping = aes(x=OstAdHoc.Assign.5gene.median, y=..prop.., group=Species), stat="count", position="dodge") +
  scale_fill_manual(values=c("grey","black"))

ggplot(data@meta.data, aes(x=OstAdHoc.Assign.5gene.median, fill=Individual)) +
  geom_bar(stat="count", position="dodge")
ggplot(data@meta.data, aes(x=OstAdHoc.Assign.5gene.median, fill=Individual)) +
  geom_bar(mapping = aes(x=OstAdHoc.Assign.5gene.median, y=..prop.., group=Individual), stat="count", position="dodge")

DimPlot(data,
        group.by="Species",
        split.by="OstAdHoc.Assign.5gene.median",
        reduction="umap",
        cols=c("grey","black")) +
  labs(x="UMAP1",y="UMAP2")

CombinePlots(plots=list((DimPlot(subset(data,subset=Species=="Human"),
                                 group.by="Stage",
                                 split.by="OstAdHoc.Assign.5gene.median",
                                 reduction="umap",
                                 cols=c("lightgrey","lightgrey","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Species=="Chimp"),
                                 group.by="Stage",
                                 split.by="OstAdHoc.Assign.5gene.median",
                                 reduction="umap",
                                 cols=c("lightgrey","lightgrey","dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)

CombinePlots(plots=list((DimPlot(subset(data,subset=Stage=="Time 0"),
                                 group.by="Stage",
                                 split.by="OstAdHoc.Assign.5gene.median",
                                 reduction="umap",
                                 cols=c("lightgrey")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 1"),
                                 group.by="Stage",
                                 split.by="OstAdHoc.Assign.5gene.median",
                                 reduction="umap",
                                 cols=c("lightgrey")) +
                           labs(x="UMAP1",y="UMAP2")),
                        (DimPlot(subset(data,subset=Stage=="Time 2"),
                                 group.by="Stage",
                                 split.by="OstAdHoc.Assign.5gene.median",
                                 reduction="umap",
                                 cols=c("dodgerblue")) +
                           labs(x="UMAP1",y="UMAP2"))), ncol=1)

DimPlot(data, group.by="Stage", split.by="OstAdHoc.Assign.5gene.median", reduction="umap", cols=c("lightgrey","lightgrey","dodgerblue")) + labs(x="UMAP1",y="UMAP2")

```

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A)
hoc.assign.x.4a <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.4a <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, q1)
ad.hoc.assign <- hoc.assign.x.4a(assign.q1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.q1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.q1 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.q1,
                                                     levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                              "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                              "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.q1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.q1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.q1 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.q1,
                                                   levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                            "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, median)
ad.hoc.assign <- hoc.assign.x.4a(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.median <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.median,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.median <- factor(data@meta.data$OstAdHoc.Assign.4geneA.median,
                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, q1 for general genes / median for osteo specific genes)
ad.hoc.assign <- hoc.assign.x.4a(assign.multi.q1median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.q1median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.q1median <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.q1median,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.multi.q1median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.q1median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.q1median <- factor(data@meta.data$OstAdHoc.Assign.4geneA.q1median,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

```

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A)
hoc.assign.x.4a <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.4a <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$PHEX[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$PHEX[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.4a(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.thresh0,
                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.4a(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.01,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.thresh0.01,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.05)
ad.hoc.assign <- hoc.assign.x.4a(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.05,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.thresh0.05,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.1)
ad.hoc.assign <- hoc.assign.x.4a(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.thresh0.1,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.thresh0.1,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (log normalized data, thresh0)
ad.hoc.assign <- hoc.assign.x.4a(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneA.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneA.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneA.l.thresh0,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4a(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneA.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneA.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.4geneA.l.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

```


```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option B)
hoc.assign.x.4b <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.4b <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option B) (scaled integrated data, q1)
ad.hoc.assign <- hoc.assign.x.4b(assign.q1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.q1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.q1 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.q1,
                                                     levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                              "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                              "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.q1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.q1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.q1 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.q1,
                                                   levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                            "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option B) (scaled integrated data, median)
ad.hoc.assign <- hoc.assign.x.4b(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.median <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.median,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.median <- factor(data@meta.data$OstAdHoc.Assign.4geneB.median,
                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option B) (scaled integrated data, q1 for general genes / median for osteo specific genes)
ad.hoc.assign <- hoc.assign.x.4b(assign.multi.q1median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.q1median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.q1median <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.q1median,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.multi.q1median)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.q1median <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.q1median <- factor(data@meta.data$OstAdHoc.Assign.4geneB.q1median,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

```

```{r}

#define function to assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option B)
hoc.assign.x.4b <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "x.embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "x.maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}
hoc.assign.4b <- function(assignments) {
  assignments$assign <- "not osteogenic"
  i=1
  while(i <= length(rownames(assignments))) {
    if(assignments$ALPL[i]==1 & assignments$POU5F1[i]==0 & assignments$TDGF1[i]==0 & assignments$SOX2[i]==0 & assignments$EPCAM[i]==0) {
      #strict criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "unknown osteogenic"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "preosteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
      #lenient criteria
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==0) {assignments$assign[i] <- "osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==0) {assignments$assign[i] <- "embedding osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==1 & assignments$BGLAP[i]==1 & assignments$CAPG[i]==1 & assignments$MEPE[i]==1) {assignments$assign[i] <- "mineralizing osteoblast"}
      if(assignments$RUNX2[i]==0 & assignments$BGLAP[i]==0 & assignments$CAPG[i]==0 & assignments$MEPE[i]==1) {assignments$assign[i] <- "maturing osteocyte"}
    }
    i=i+1
  }
  return(as.data.frame(assignments$assign, row.names=rownames(assignments)))
}

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0)
ad.hoc.assign <- hoc.assign.x.4b(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.thresh0,
                                                       levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.01)
ad.hoc.assign <- hoc.assign.x.4b(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.01,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.thresh0.01)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.01 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.01 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.thresh0.01,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.05)
ad.hoc.assign <- hoc.assign.x.4b(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.05,
                                                            levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                     "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                     "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.thresh0.05)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.05 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.05 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.thresh0.05,
                                                          levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                   "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (scaled integrated data, thresh0.1)
ad.hoc.assign <- hoc.assign.x.4b(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.thresh0.1,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(assign.thresh0.1)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.1 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.thresh0.1 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.thresh0.1,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))

#assign osteogenic cell sub-type labels based on ad hoc cutoffs (using 4 genes - option A) (log normalized data, thresh0)
ad.hoc.assign <- hoc.assign.x.4b(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.x.4geneB.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.x.4geneB.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.x.4geneB.l.thresh0,
                                                           levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                    "x.osteoblast","x.embedding osteoblast","x.mineralizing osteoblast","x.maturing osteocyte",
                                                                    "unknown osteogenic","not osteogenic","other"))
ad.hoc.assign <- hoc.assign.4b(l.assign.thresh0)
table(ad.hoc.assign$`assignments$assign`)
data@meta.data$OstAdHoc.Assign.4geneB.l.thresh0 <- ad.hoc.assign$`assignments$assign`
data@meta.data$OstAdHoc.Assign.4geneB.l.thresh0 <- factor(data@meta.data$OstAdHoc.Assign.4geneB.l.thresh0,
                                                         levels=c("preosteoblast","osteoblast","embedding osteoblast","mineralizing osteoblast","maturing osteocyte",
                                                                  "unknown osteogenic","not osteogenic","other"))


```

```{r}

saveRDS(data, file="./data/cellranger-data-full/data.filter.log.indv-cell.int19k.reg-tot.assign.rds")
#saveRDS(data, file="./data/cellranger-data-full/data.filter.log.indv-cell.int19k-tot.assign.rds")

```
